<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ee1857a059c0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - TC-Hewitt/OatCrownRust: Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux environment or R.
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:141EB83:1D192A7:6605D559" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDoxNDFFQjgzOjFEMTkyQTc6NjYwNUQ1NTkiLCJ2aXNpdG9yX2lkIjoiNjQ1MjI0MTg1NjIzODgxNzgwNyIsInJlZ2lvbl9lZGdlIjoiaWFkIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-pjax-transient="true" name="visitor-payload">
           <meta content="bd84dcd92c4b1725032a79c6399ae8aac77fb721ad51664e383d77836ae0d452" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:560200753" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux environment or R. - TC-Hewitt/OatCrownRust" name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/TC-Hewitt/OatCrownRust" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/e160fffd4bbb2d44c92fc20d6988a62d3a4be0fefeefd6b1508c0bbb1ab75e88/TC-Hewitt/OatCrownRust" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - TC-Hewitt/OatCrownRust: Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux environment or R." name="twitter:title">
                    <meta content="Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux envi..." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/e160fffd4bbb2d44c92fc20d6988a62d3a4be0fefeefd6b1508c0bbb1ab75e88/TC-Hewitt/OatCrownRust" property="og:image">
                      <meta content="Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux envi..." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - TC-Hewitt/OatCrownRust: Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux environment or R." property="og:title"/>
                        <meta content="https://github.com/TC-Hewitt/OatCrownRust" property="og:url"/>
                        <meta content="Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux envi..." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="def536b5d69b8afc2117ff3aa904c09699292b9d809266f1d19b952c3b02461c" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="5aaf0d46cfb23ac0376126aa517fe17c5c6293814f8cfb7fb33598c257a2a6b9" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/TC-Hewitt/OatCrownRust git https://github.com/TC-Hewitt/OatCrownRust.git" name="go-import"/>
                        <meta content="33470968" name="octolytics-dimension-user_id">
                         <meta content="TC-Hewitt" name="octolytics-dimension-user_login">
                          <meta content="560200753" name="octolytics-dimension-repository_id">
                           <meta content="TC-Hewitt/OatCrownRust" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="560200753" name="octolytics-dimension-repository_network_root_id">
                               <meta content="TC-Hewitt/OatCrownRust" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/TC-Hewitt/OatCrownRust" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-99519581d0f8.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="c77c5dd907a6b456ef4ed3a6b0bdc75b3a559ea20b291a73983be2f7010c99fb" href="/login?return_to=https%3A%2F%2Fgithub.com%2FTC-Hewitt%2FOatCrownRust">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=6wmyQdTO8UqgFXT3WAMowjhPrk7gOjpPgWVZjOcoJmXOq5ecqVG1YMUnnP4a324NwJoVE4k879Yxc7rlrpGv8PeQjYFSmRQLOAI2JGuWc4jM%2FBVl9t7zYTCDOvzRlmfZYmMv7vztFkr0iszz0sbieed6IPA5w2fryQpe%2FqkfLbh0K%2FnkFVfrVMpubAtOWoIkTwvavh1jrLy%2Fhqpf88jF3U%2BhuC6sDWCgYRtCKmyId%2BNJG4oenxFy10iN4OGRtU7SKMDQ2%2BBgjLd4R5zJNM8IUGb2bSGTlDoQhH0xU8S7XQeN%2FMNCnoFCPgDjz80PtH%2B9ecoihMm3R4vtzvKqg4pxShrtTQ%2FMi4qZm5VcH6JBTLZQHBNQQjiwqas7eqjNuI%2By907Q8iZ%2FKj%2Fnl%2FvhzbWe5OWaGwaTqiv1mXgsP8mdaAR0IrBw2JggQj0bzGBwMmjxrauO9hFE00F2myOqktnASYhb4msft6aesU5Q%2FhEGazazFB1XaandGFIPFlYFInysXjkqoMDX2ccDIYi0VIW1aBR%2B8m0XHQ%3D%3D--t%2FqCQQjzYOrGz%2FDE--9TysMfwjkAhYFFb031yVZw%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="TC-Hewitt" data-current-repository="TC-Hewitt/OatCrownRust" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="lfIW-QpiIu0SXd9Uwsq9vkZxx9WXgIWHKfrXzVwUHlLET0CUUX4bAPxh7uXM0vUVECgTxTtyh48yewbHSS8UFg" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:TC-Hewitt/OatCrownRust">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-011a5439-e668-4bd4-a374-e3c4c52148d4" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-011a5439-e668-4bd4-a374-e3c4c52148d4">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="RikDM6wYcuO24MmA29N5YMbzVmaEn4+JYAuHMSPJjhL47RbF0xdUmPzhqPAKN6B4d0+NXAOEz1kpQmO08LfnIg=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="5QZlCOpvqUaXnZpGHwYl8joyQqvEMtkD+Kdk8JqUxZlvgctU6kUUkdslta0m03sq/1qhpcaKi//fARGr9AdudA==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="mV0t+Sb5aL8srEBBNREAxeq3dZ158xZfzK/WShL20jH0k4WlBAub3aZBMLuTQ9bLPuyy5vTNJNUcg7Ttb7OQUg==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="au/6dbZX+TTs4IGRt+SoOBaWyzzieujRSyj2coRnxAuc8j8TSsTJSilNLwgFCNwQoNZDFUrnZ5pVdE1ITxtoGw==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="c77c5dd907a6b456ef4ed3a6b0bdc75b3a559ea20b291a73983be2f7010c99fb" href="/login?return_to=https%3A%2F%2Fgithub.com%2FTC-Hewitt%2FOatCrownRust">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="c77c5dd907a6b456ef4ed3a6b0bdc75b3a559ea20b291a73983be2f7010c99fb" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=TC-Hewitt%2FOatCrownRust">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-9c6e34ad-f842-4682-9b59-157407db5a88" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-9430fbce-ed35-419a-bd18-df25a3cef86c" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-9430fbce-ed35-419a-bd18-df25a3cef86c" id="tooltip-9c6e34ad-f842-4682-9b59-157407db5a88" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/TC-Hewitt/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/TC-Hewitt" rel="author">
            TC-Hewitt
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/TC-Hewitt/OatCrownRust">
            OatCrownRust
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="853462875dd1d6c31ea28617c042478c4efde625a0d8f2938df970df41c27615" data-view-component="true" href="/login?return_to=%2FTC-Hewitt%2FOatCrownRust" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":560200753,"auth_type":"LOG_IN","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="8c6bf1abcd0819bb9df028812903bfef82feb5ee6556e449f20eb5d664ee850a" data-view-component="true" href="/login?return_to=%2FTC-Hewitt%2FOatCrownRust" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":560200753,"auth_type":"LOG_IN","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="9b4278d55a23e773e9572aa9299458d6dd59a8633d5b795634db273a8b8e9be3" data-view-component="true" href="/login?return_to=%2FTC-Hewitt%2FOatCrownRust" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <p class="f4 mb-3">
          Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux environment or R.
         </p>
         <div class="mb-2 d-flex flex-items-center Link--secondary">
          <svg aria-hidden="true" class="octicon octicon-link flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
           </path>
          </svg>
          <span class="flex-auto min-width-0 css-truncate css-truncate-target width-fit">
           <a class="text-bold" href="https://doi.org/10.1101/2023.09.18.557855" rel="noopener noreferrer" role="link" target="_blank" title="https://doi.org/10.1101/2023.09.18.557855">
            doi.org/10.1101/2023.09.18.557855
           </a>
          </span>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/TC-Hewitt/OatCrownRust/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/TC-Hewitt/OatCrownRust/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/TC-Hewitt/OatCrownRust/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/TC-Hewitt/OatCrownRust/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/TC-Hewitt/OatCrownRust/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":560200753,"auth_type":"LOG_IN","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="9b4278d55a23e773e9572aa9299458d6dd59a8633d5b795634db273a8b8e9be3" data-view-component="true" href="/login?return_to=%2FTC-Hewitt%2FOatCrownRust" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/TC-Hewitt/OatCrownRust","user_id":null}}' data-hydro-click-hmac="853462875dd1d6c31ea28617c042478c4efde625a0d8f2938df970df41c27615" data-view-component="true" href="/login?return_to=%2FTC-Hewitt%2FOatCrownRust" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /TC-Hewitt/OatCrownRust" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TC-Hewitt/OatCrownRust" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /TC-Hewitt/OatCrownRust/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TC-Hewitt/OatCrownRust/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /TC-Hewitt/OatCrownRust/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TC-Hewitt/OatCrownRust/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /TC-Hewitt/OatCrownRust/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TC-Hewitt/OatCrownRust/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /TC-Hewitt/OatCrownRust/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TC-Hewitt/OatCrownRust/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /TC-Hewitt/OatCrownRust/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TC-Hewitt/OatCrownRust/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/TC-Hewitt/OatCrownRust/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /TC-Hewitt/OatCrownRust/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TC-Hewitt/OatCrownRust/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-list" aria-haspopup="true" aria-labelledby="tooltip-b7d67007-5072-4d20-8c80-7f8b0454b546" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-button" popovertarget="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-button" id="tooltip-b7d67007-5072-4d20-8c80-7f8b0454b546" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-button" anchor-offset="normal" data-view-component="true" id="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-7e8e38a9-7644-44cf-a6b7-2ddbbe96e956-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TC-Hewitt/OatCrownRust" id="item-be952b42-b08c-403e-9188-c85be4390218" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TC-Hewitt/OatCrownRust/issues" id="item-9d813f7d-5679-4578-bf58-3ffad8ad91a1" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TC-Hewitt/OatCrownRust/pulls" id="item-a74dfd79-7ed1-4044-84c8-844b35b5ad9e" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TC-Hewitt/OatCrownRust/actions" id="item-d2e21274-5ccd-4225-a129-fac89b0374a9" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TC-Hewitt/OatCrownRust/projects" id="item-1e08036f-14d5-4773-bc96-3d1251d7a40d" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TC-Hewitt/OatCrownRust/security" id="item-473a2e4c-a435-4fcd-bc1f-94921a9d11d6" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TC-Hewitt/OatCrownRust/pulse" id="item-98a393be-969f-446f-a142-f7e60a609448" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         TC-Hewitt/OatCrownRust
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/TC-Hewitt/OatCrownRust/spoofed_commit_check/1534f6da53c00e6be92ee5ebdfd8f02d0017e557">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":560200753,"defaultBranch":"main","name":"OatCrownRust","ownerLogin":"TC-Hewitt","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2022-11-01T00:26:56.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/33470968?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1667263055.070705","canEdit":false,"refType":"branch","currentOid":"1534f6da53c00e6be92ee5ebdfd8f02d0017e557"},"tree":{"items":[{"name":"Orthogroups_txtConvert.py","path":"Orthogroups_txtConvert.py","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":2,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/TC-Hewitt/OatCrownRust/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/TC-Hewitt/OatCrownRust.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone TC-Hewitt/OatCrownRust","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/TC-Hewitt/OatCrownRust/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=560200753"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"33","overviewFiles":[{"displayName":"README.md","repoName":"OatCrownRust","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eOatCrownRust\u003c/h1\u003e\u003ca id=\"user-content-oatcrownrust\" class=\"anchor\" aria-label=\"Permalink: OatCrownRust\" href=\"#oatcrownrust\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRead mapping statistics and coverage plots\u003c/h2\u003e\u003ca id=\"user-content-read-mapping-statistics-and-coverage-plots\" class=\"anchor\" aria-label=\"Permalink: Read mapping statistics and coverage plots\" href=\"#read-mapping-statistics-and-coverage-plots\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAfter FASTQ files are mapped to a reference FASTA with an aligner such as \u003ca href=\"https://github.com/lh3/bwa\"\u003eBWA\u003c/a\u003e or \u003ca href=\"https://github.com/DaehwanKimLab/hisat2\"\u003eHISAT2\u003c/a\u003e, simple count and coverage statistics are obtained from the subsequent SAM/BAM files with the following scripts using \u003ca href=\"https://github.com/samtools/samtools\"\u003eSAMtools\u003c/a\u003e output piped into AWK operations\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1. for a single BAM, count total number of mapped reads and store in var \u003cem\u003emap_count\u003c/em\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load samtools/1.12 #your system might have different environment tool command\nmap_count=$(samtools view -c -F 260 sample01.bam)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emodule load samtools/1.12 #your system might have different environment tool command\nmap_count=$(samtools view -c -F 260 sample01.bam)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-c\u003c/em\u003e, counts total n alignments instead of printing each alignment\u003cbr\u003e\n\u003cem\u003e-F 260\u003c/em\u003e, exclude unmapped reads or non-primary alignments (\"260\" set in \u003cem\u003eFLAG\u003c/em\u003e field)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2. get read depth at each ref position in BAM and pipe to AWK to generate coverage stats\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"samtools depth -a sample01.bam | awk -v OFS='\\t' -v sname=\u0026quot;sample01\u0026quot; -v nmap=\u0026quot;$map_count\u0026quot; \\\n'{sum+=$3} ; $3\u0026gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' \u0026gt; sample01.covstats.out\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esamtools depth -a sample01.bam | awk -v OFS='\\t' -v sname=\"sample01\" -v nmap=\"$map_count\" \\\n'{sum+=$3} ; $3\u0026gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' \u0026gt; sample01.covstats.out\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-a\u003c/em\u003e, output all positions (including those with zero depth)\u003cbr\u003e\n\u003cem\u003e-v OFS='\\t'\u003c/em\u003e, sets awk output to be tab-delimited\u003cbr\u003e\n\u003cem\u003e-v sname=\"sample01\"\u003c/em\u003e, sets awk var for sample name\u003cbr\u003e\n\u003cem\u003e-v nmap=\"$map_count\"\u003c/em\u003e, sets awk var for \u003cem\u003emap_count\u003c/em\u003e determined in previous step\u003cbr\u003e\n\u003cem\u003e{sum+=$3}\u003c/em\u003e, cosecutively add depth value in 3rd field to \u003cem\u003esum\u003c/em\u003e var\u003cbr\u003e\n\u003cem\u003e$3\u0026gt;10{c++}\u003c/em\u003e, adds 1 count to var \u003cem\u003ec\u003c/em\u003e if depth value in 3rd field \u0026gt; 10 (i.e. 10X coverage)\u003cbr\u003e\n\u003cem\u003eEND{print ...}\u003c/em\u003e, after final line of \u003cem\u003esamtools depth\u003c/em\u003e output reached, writes the following to file \u003cem\u003esample01.covstats.out\u003c/em\u003e:\u003cbr\u003e\nsample name (\u003cem\u003esname\u003c/em\u003e)\u003cbr\u003e\ntotal mapped reads (\u003cem\u003enmap\u003c/em\u003e)\u003cbr\u003e\nmean coverage (\u003cem\u003esum/NR\u003c/em\u003e, where \u003cem\u003eNR\u003c/em\u003e=total n of rows processed =n ref positions)\u003cbr\u003e\ntotal n positions with coverage over 10X (\u003cem\u003ec+0\u003c/em\u003e)\u003cbr\u003e\npercent of all positions with coverage over 10X (\u003cem\u003ec*100/NR\u003c/em\u003e)\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003emultiple BAM files can be processed in parrallel to save time. For example, below is an array job script for 100 samples using the \u003ca href=\"https://github.com/SchedMD/slurm\"\u003eSlurm Workload Manager\u003c/a\u003e on a shared cluster\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/env bash\n\n#SBATCH --job-name=covstats\n#SBATCH --time=2:00:00\n#SBATCH --nodes=1\n#SBATCH --mem=2GB\n#SBATCH --array=0-99\n\nSAMPLES=( $(cat samplenames.txt) ) #file with 1 sample name per line (sans file exts)\n\nmodule load samtools/1.12\n\nif [ ! -z \u0026quot;$SLURM_ARRAY_TASK_ID\u0026quot; ]\nthen\n    i=$SLURM_ARRAY_TASK_ID\n    map_count=$(samtools view -c -F 260 ${SAMPLES[$i]}.bam)\n    samtools depth -a ${SAMPLES[$i]}.bam | awk -v OFS='\\t' -v sname=${SAMPLES[$i]} -v nmap=\u0026quot;$map_count\u0026quot; \\\n    '{sum+=$3} ; $3\u0026gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' \u0026gt; temp${i}\nelse\n    echo \u0026quot;Error: Missing array index as SLURM_ARRAY_TASK_ID\u0026quot;\nfi\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/usr/bin/env bash\n\n#SBATCH --job-name=covstats\n#SBATCH --time=2:00:00\n#SBATCH --nodes=1\n#SBATCH --mem=2GB\n#SBATCH --array=0-99\n\nSAMPLES=( $(cat samplenames.txt) ) #file with 1 sample name per line (sans file exts)\n\nmodule load samtools/1.12\n\nif [ ! -z \"$SLURM_ARRAY_TASK_ID\" ]\nthen\n    i=$SLURM_ARRAY_TASK_ID\n    map_count=$(samtools view -c -F 260 ${SAMPLES[$i]}.bam)\n    samtools depth -a ${SAMPLES[$i]}.bam | awk -v OFS='\\t' -v sname=${SAMPLES[$i]} -v nmap=\"$map_count\" \\\n    '{sum+=$3} ; $3\u0026gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' \u0026gt; temp${i}\nelse\n    echo \"Error: Missing array index as SLURM_ARRAY_TASK_ID\"\nfi\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ethe above can be wrapped in file \u003cem\u003ecovstats.sh\u003c/em\u003e and run as \u003cem\u003e\u0026gt;sbatch covstats.sh\u003c/em\u003e \u003cbr\u003e\nthe output line of each sample is written to seperate temp files assigned \u003cem\u003etemp0\u003c/em\u003e to \u003cem\u003etemp99\u003c/em\u003e \u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3. the output stats from each sample can now be collated into a single tab-delim table by creating a header and concatenating with all \u003cem\u003etemp\u003c/em\u003e files\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"echo -e \u0026quot;sample\\tmapped_reads\\tmean_cov\\tbp_over_10X\\tpcnt_ref_over_10X\u0026quot; \u0026gt; header\ncat header temp{0..99} \u0026gt; combined.covstats.txt\nrm header temp{0..99}\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eecho -e \"sample\\tmapped_reads\\tmean_cov\\tbp_over_10X\\tpcnt_ref_over_10X\" \u0026gt; header\ncat header temp{0..99} \u0026gt; combined.covstats.txt\nrm header temp{0..99}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eheader names describe the output fields\u003cbr\u003e\nfinal matrix is written to \u003cem\u003ecombined.covstats.txt\u003c/em\u003e \u003cbr\u003e\ntemporary \u003cem\u003eheader\u003c/em\u003e and \u003cem\u003etemp\u003c/em\u003e files are deleted\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e4. various stats can then be plotted from the table using a simple R script like below\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/env Rscript\ninfile = commandArgs(trailingOnly=TRUE)\nstats \u0026lt;- read.table(infile, header = T, sep = \u0026quot;\\t\u0026quot;)\noutname \u0026lt;- strsplit(infile, '.txt') \npng(file=paste0(outname, \u0026quot;_plots.png\u0026quot;), width=2000, height=600)\npar(mfrow=c(2,1), mar=c(6,4,4,4))\nbarplot(stats$mean_cov, names=stats$sample, main=\u0026quot;mean cov\u0026quot;, ylim=c(0,100), las=2)\nbarplot(stats$pcnt_ref_over_10X, names=stats$sample, main=\u0026quot;perc bp over 10X\u0026quot;, ylim=c(0,100), las=2)\ndev.off()\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/usr/bin/env Rscript\ninfile = commandArgs(trailingOnly=TRUE)\nstats \u0026lt;- read.table(infile, header = T, sep = \"\\t\")\noutname \u0026lt;- strsplit(infile, '.txt') \npng(file=paste0(outname, \"_plots.png\"), width=2000, height=600)\npar(mfrow=c(2,1), mar=c(6,4,4,4))\nbarplot(stats$mean_cov, names=stats$sample, main=\"mean cov\", ylim=c(0,100), las=2)\nbarplot(stats$pcnt_ref_over_10X, names=stats$sample, main=\"perc bp over 10X\", ylim=c(0,100), las=2)\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ethe above can be wrapped in file \u003cem\u003eplot_covstats.R\u003c/em\u003e and run as \u003cem\u003e\u0026gt;Rscript plot_covstats.R combined.covstats.txt\u003c/em\u003e \u003cbr\u003e\nthis creates two basic barplots, one for mean coverage and another for percent of ref bases over 10X coverage, in a single PNG image\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAssessing variation with population SNP counts\u003c/h2\u003e\u003ca id=\"user-content-assessing-variation-with-population-snp-counts\" class=\"anchor\" aria-label=\"Permalink: Assessing variation with population SNP counts\" href=\"#assessing-variation-with-population-snp-counts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHere we want to obtain the number of variable SNP sites (VSS) in a populations of crown rust fungi. These are genomic positions at which SNP genotypes are heterogeneous within a population (i.e. not all individuals possess the same genotype) and can therefore be used as a relative measure of overall genotypic diversity when comparing populations (provided each population is called against the same reference genome). Variant calling can be performed independently for each population (represented by collections of fungal isolates) using the same reference with appropriate filters applied.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn this example, we are comparing a population of North American (US) and South African (SA) isolates. However, variant calling was performed simultaneously on all isolates using \u003ca href=\"https://github.com/freebayes/freebayes\"\u003eFreeBayes\u003c/a\u003e to produce a single, multisample VCF containing variant data from 20 US isolates and 29 SA isolates. Extra steps are required to subset the US and SA populations and apply additional filtering before both the total SNP sites (TSS) and VSS of each population are counted.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1. initial filtering\u003c/strong\u003e - the raw VCF file \u003cem\u003ecr_collection.vcf\u003c/em\u003e is 1st filtered on quality using \u003ca href=\"https://github.com/vcflib/vcflib\"\u003evcffilter\u003c/a\u003e with various standard params\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load vcflib/1.0.1\nvcffilter -f \u0026quot;QUAL \u0026gt; 20 \u0026amp; QUAL / AO \u0026gt; 10 \u0026amp; SAF \u0026gt; 0 \u0026amp; SAR \u0026gt; 0 \u0026amp; RPR \u0026gt; 1 \u0026amp; RPL \u0026gt; 1 \u0026amp; AC \u0026gt; 0 \u0026amp; DP \u0026gt; 500\u0026quot; \\\ncr_collection.vcf \u0026gt; cr_collection.QA.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emodule load vcflib/1.0.1\nvcffilter -f \"QUAL \u0026gt; 20 \u0026amp; QUAL / AO \u0026gt; 10 \u0026amp; SAF \u0026gt; 0 \u0026amp; SAR \u0026gt; 0 \u0026amp; RPR \u0026gt; 1 \u0026amp; RPL \u0026gt; 1 \u0026amp; AC \u0026gt; 0 \u0026amp; DP \u0026gt; 500\" \\\ncr_collection.vcf \u0026gt; cr_collection.QA.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-f\u003c/em\u003e, specifies a filter to apply to the info fields of records, removes sites which do not pass the filter\u003cbr\u003e\n\u003cem\u003eQUAL \u0026gt; 20\u003c/em\u003e, requires quality score no less than 20\u003cbr\u003e\n\u003cem\u003eQUAL / AO \u0026gt; 10\u003c/em\u003e, requires additional contribution of each observation at least 10 log units (~Q10 per read)\u003cbr\u003e\n\u003cem\u003eSAF \u0026gt; 0 \u0026amp; SAR \u0026gt; 0\u003c/em\u003e, requires reads present on both strands\u003cbr\u003e\n\u003cem\u003eRPR \u0026gt; 1 \u0026amp; RPL \u0026gt; 1\u003c/em\u003e, requires at least two reads “balanced” to each side of the site\u003cbr\u003e\n\u003cem\u003eAC \u0026gt; 0\u003c/em\u003e, requires total number of alt alleles in called genotypes more than 0\u003cbr\u003e\n\u003cem\u003eDP \u0026gt; 500\u003c/em\u003e, requires greater than 500 total read depth (with 49 samples this means a min ~10X per sample on average)\u003cbr\u003e\n\u003cem\u003e\u0026gt; cr_collection.QA.vcf\u003c/em\u003e, write filtered records to new \"Quality Assured\" file\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2. feature filtering\u003c/strong\u003e - we are only interested in biallelic SNPs likely to be informative, so we will apply additional filtering to the VCF using \u003ca href=\"https://github.com/samtools/bcftools\"\u003eBCFtools\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load bcftools/1.15.1\nbcftools view -m2 -M2 -v snps -i 'F_MISSING \u0026lt; 0.1' -e 'MAF \u0026lt; 0.05' \\\n--output-type z -o cr_collection.QA.biaSNPs.vcf.gz cr_collection.QA.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emodule load bcftools/1.15.1\nbcftools view -m2 -M2 -v snps -i 'F_MISSING \u0026lt; 0.1' -e 'MAF \u0026lt; 0.05' \\\n--output-type z -o cr_collection.QA.biaSNPs.vcf.gz cr_collection.QA.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-m2 -M2\u003c/em\u003e, keep sites with min 2 alleles and max 2 alleles\u003cbr\u003e\n\u003cem\u003e-v snps\u003c/em\u003e, keep SNPs only (omits indels and MNPs)\u003cbr\u003e\n\u003cem\u003e-i 'F_MISSING \u0026lt; 0.1'\u003c/em\u003e, include only sites wih less than 10% missing data\u003cbr\u003e\n\u003cem\u003e-e 'MAF \u0026lt; 0.05'\u003c/em\u003e, exclude sites with a minor allele frequency (MAF) of less than 5% (note: skip MAF filtering if you dont want to discard potential rare/novel genotypes)\u003cbr\u003e\n\u003cem\u003e--output-type z\u003c/em\u003e, output gz compressed vcf\u003cbr\u003e\n\u003cem\u003e-o cr_collection.QA.biaSNPs.vcf.gz\u003c/em\u003e, output file name\u003cbr\u003e\n\u003cem\u003ecr_collection.QA.vcf\u003c/em\u003e, input file name\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3. subset populations\u003c/strong\u003e - we will now subset the US and SA populations from this VCF based on sample names\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bcftools view -S samplenames_US.txt cr_collection.QA.biaSNPs.vcf.gz \u0026gt; cr_pop_US.vcf\nbcftools view -S samplenames_SA.txt cr_collection.QA.biaSNPs.vcf.gz \u0026gt; cr_pop_SA.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebcftools view -S samplenames_US.txt cr_collection.QA.biaSNPs.vcf.gz \u0026gt; cr_pop_US.vcf\nbcftools view -S samplenames_SA.txt cr_collection.QA.biaSNPs.vcf.gz \u0026gt; cr_pop_SA.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-S samplenames_US.txt\u003c/em\u003e, subset US pop using sample names listed in samplenames_US.txt (1 per line)\u003cbr\u003e\n\u003cem\u003e-S samplenames_SA.txt\u003c/em\u003e, subset SA pop using sample names listed in samplenames_SA.txt\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eNames must match exactly as present in the VCF file. In case they are not known, it is possible to retrieve them from the VCF with a bash script\u003cbr\u003e\n\u003ccode\u003ezcat cr_collection.QA.biaSNPs.vcf.gz | egrep \"^#\" | tail -1\u003c/code\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003ezcat\u003c/em\u003e will print gz compressed files, otherwise use \u003cem\u003ecat\u003c/em\u003e if VCF uncompressed\u003cbr\u003e\n\u003cem\u003eegrep\u003c/em\u003e will find only header lines (starts with \"#\")\u003cbr\u003e\n\u003cem\u003etail -1\u003c/em\u003e will print out the final header line which should contain the names of the first 8 or 9 default fields followed by the sample fields\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e4. get TSS counts\u003c/strong\u003e - we now have seperate US and SA files to get their total SNP count. However, since they were subsetted from a single VCF file, each may contain genotypes that are homozygous reference (ref/ref, RR) in every sample. These can be excluded from the final count\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bcftools view -i 'GT[*]=\u0026quot;alt\u0026quot;' cr_pop_US.vcf \u0026gt; cr_pop_US.noRR.vcf\nbcftools view -i 'GT[*]=\u0026quot;alt\u0026quot;' cr_pop_SA.vcf \u0026gt; cr_pop_SA.noRR.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebcftools view -i 'GT[*]=\"alt\"' cr_pop_US.vcf \u0026gt; cr_pop_US.noRR.vcf\nbcftools view -i 'GT[*]=\"alt\"' cr_pop_SA.vcf \u0026gt; cr_pop_SA.noRR.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-i ...\u003c/em\u003e, this includes only sites where at least one sample has an alt allele, written to a new VCF\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eTSS and other stats can now be obtained from these VCFs, contained in \u003cem\u003ebcftools stats\u003c/em\u003e output\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bcftools stats cr_pop_US.noRR.vcf \u0026gt; cr_pop_US.noRR.stats.out\nbcftools stats cr_pop_US.noRR.vcf \u0026gt; cr_pop_US.noRR.stats.out\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebcftools stats cr_pop_US.noRR.vcf \u0026gt; cr_pop_US.noRR.stats.out\nbcftools stats cr_pop_US.noRR.vcf \u0026gt; cr_pop_US.noRR.stats.out\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ealternatively, TSS can be counted directly without writing new files\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bcftools view -i 'GT[*]=\u0026quot;alt\u0026quot;' cr_pop_US.vcf | egrep -v \u0026quot;^#\u0026quot; | wc -l\n# TSS for US pop =756,980\n\nbcftools view -i 'GT[*]=\u0026quot;alt\u0026quot;' cr_pop_US.vcf | egrep -v \u0026quot;^#\u0026quot; | wc -l\n# TSS for SA pop =548,091\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebcftools view -i 'GT[*]=\"alt\"' cr_pop_US.vcf | egrep -v \"^#\" | wc -l\n# TSS for US pop =756,980\n\nbcftools view -i 'GT[*]=\"alt\"' cr_pop_US.vcf | egrep -v \"^#\" | wc -l\n# TSS for SA pop =548,091\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eRR sites are filtered out as before\u003cbr\u003e\n\u003cem\u003eegrep -v\u003c/em\u003e, exclude lines that start with \"#\" (header lines)\u003cbr\u003e\n\u003cem\u003ewc -l\u003c/em\u003e, count and print number of output lines (= total n SNP records)\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e5. get VSS counts\u003c/strong\u003e - different filter params can be used to obtain the number of variable SNP sites. As before, we can write to new files and then run \u003cem\u003ebcftools stats\u003c/em\u003e, but here we will print the counts directly like above\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"nUS=20\nnSA=29\n\nbcftools view -e \u0026quot;COUNT(GT='RR')=$nUS | COUNT(GT='RA')=$nUS | COUNT(GT='AA')=$nUS | COUNT(GT='Aa')=$nUS\u0026quot; \\\ncr_pop_US.vcf | egrep -v \u0026quot;^#\u0026quot; | wc -l\n# VSS for US pop =652,504\n\nbcftools view -e \u0026quot;COUNT(GT='RR')=$nSA | COUNT(GT='RA')=$nSA | COUNT(GT='AA')=$nSA | COUNT(GT='Aa')=$nSA\u0026quot; \\\ncr_pop_SA.vcf | egrep -v \u0026quot;^#\u0026quot; | wc -l\n# VSS for SA pop =47,369\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003enUS=20\nnSA=29\n\nbcftools view -e \"COUNT(GT='RR')=$nUS | COUNT(GT='RA')=$nUS | COUNT(GT='AA')=$nUS | COUNT(GT='Aa')=$nUS\" \\\ncr_pop_US.vcf | egrep -v \"^#\" | wc -l\n# VSS for US pop =652,504\n\nbcftools view -e \"COUNT(GT='RR')=$nSA | COUNT(GT='RA')=$nSA | COUNT(GT='AA')=$nSA | COUNT(GT='Aa')=$nSA\" \\\ncr_pop_SA.vcf | egrep -v \"^#\" | wc -l\n# VSS for SA pop =47,369\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eexact sample sizes of the US and SA populations (as present in VCF subsets) are first assigned to to vars \u003cem\u003enUS\u003c/em\u003e and \u003cem\u003enSA\u003c/em\u003e \u003cbr\u003e\n\u003cem\u003e-e ...\u003c/em\u003e, this excludes sites where a count condition is equal to the sample size (\u003cem\u003e$nUS\u003c/em\u003e or \u003cem\u003e$nSA\u003c/em\u003e), i.e. identical genotype (RR=ref/ref hom, RA=ref/alt het, AA=alt/alt hom, Aa=alt/alt het) in every sample. Thus, what remains are only the variable sites\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003eTo save space, uncompresed leftover VCF files can be bgzipped with \u003cem\u003esamtools\u003c/em\u003e then indexed with \u003cem\u003ebcftools\u003c/em\u003e \u003cbr\u003e\n\u003ccode\u003ebgzip -@ 4 file.vcf \u0026amp;\u0026amp; bcftools index file.vcf.gz\u003c/code\u003e\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-@ 4\u003c/em\u003e, num threads to use to perform compression\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003eFor context, the SA population is entirely asexual, lacking any recombination, whereas the US population reproduces both sexually and asexually. Therefore, the US population is expected to be more diverse. This is highly evident in the VSS count, which is ~14 times higher than that of the SA population. TSS counts only indicate SNPs against the reference and are heavily influenced by reference choice. Subtracting the VSS from the TSS in the SA population (548,091-47,369) gives ~500K genotypes that were likely already present at the founding of this population, and diversity within this population is represented by only ~47K SNP sites.\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenerating chromosome maps with \u003ca href=\"https://github.com/bernatgel/karyoploteR\"\u003ekaryoploteR\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-generating-chromosome-maps-with-karyoploter\" class=\"anchor\" aria-label=\"Permalink: Generating chromosome maps with karyoploteR\" href=\"#generating-chromosome-maps-with-karyoploter\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThese are instructions to generate simple chromosome maps annotating regions of interest, similar to what appears in the paper figure. More detailed maps incorporating features such as density plots and coverage graphs are covered in the \u003ca href=\"https://bernatgel.github.io/karyoploter_tutorial/\" rel=\"nofollow\"\u003ekaryoploteR tutorial\u003c/a\u003e. Here we will make basic plots based on a diploid chromosome (chr) set where A and B haplotypes are separated.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1. Input for chr lengths\u003c/strong\u003e - this is simply a tab delim table of chr names, start and end coords which can be obtaind from the \u003cem\u003e.fasta.fai\u003c/em\u003e index of the genome FASTA file\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003esamtools faidx mygenome.fasta\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ethe following bash scripts can be used to create a regions table from \u003cem\u003emygenome.fasta.fai\u003c/em\u003e compatible with \u003ca href=\"https://bioconductor.org/packages/3.18/bioc/html/GenomicRanges.html\" rel=\"nofollow\"\u003eGRanges\u003c/a\u003e formatting\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"printf \u0026quot;chr\\tstart\\tend\\n\u0026quot; \u0026gt; mygenome.granges.txt\nTAB='\\t'\ncut -f1,2 mygenome.fasta.fai | sed \u0026quot;s/${TAB}/${TAB}0${TAB}/g\u0026quot; \u0026gt;\u0026gt; mygenome.granges.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eprintf \"chr\\tstart\\tend\\n\" \u0026gt; mygenome.granges.txt\nTAB='\\t'\ncut -f1,2 mygenome.fasta.fai | sed \"s/${TAB}/${TAB}0${TAB}/g\" \u0026gt;\u0026gt; mygenome.granges.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethis will look like the following in this example, where haplotypes are inidcated by _\u003cem\u003eA\u003c/em\u003e and _\u003cem\u003eB\u003c/em\u003e suffixes.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"chr start   end\nchr1_A  0   8018478\nchr2_A  0   6810466\nchr3_A  0   6979198\nchr4_A  0   7574082\nchr5_A  0   6841886\nchr6_A  0   5186056\nchr1_B  0   7419164\nchr2_B  0   6665088\nchr3_B  0   6167249\nchr4_B  0   8040728\nchr5_B  0   7117848\nchr6_B  0   5207541\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003echr start   end\nchr1_A  0   8018478\nchr2_A  0   6810466\nchr3_A  0   6979198\nchr4_A  0   7574082\nchr5_A  0   6841886\nchr6_A  0   5186056\nchr1_B  0   7419164\nchr2_B  0   6665088\nchr3_B  0   6167249\nchr4_B  0   8040728\nchr5_B  0   7117848\nchr6_B  0   5207541\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2. Input for regions of interest\u003c/strong\u003e - similarly, we will have a table \u003cem\u003eloci.granges.txt\u003c/em\u003e with start and end coords for specific loci, with extra fields for locus name and chosen colours as HEX values\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"chr start   end name    colour\nchr1_A  2140000 2200000 loc1A '#61d04f'\nchr1_B  1980000 2020000 loc1B '#61d04f'\nchr1_A  4010000 4030000 loc2A '#2297e6'\nchr1_B  3760000 3770000 loc2B '#2297e6'\nchr2_A  4590000 4910000 loc3A '#542a18'\nchr2_B  4680000 4990000 loc3B '#542a18'\nchr6_A  480000  700000  loc4A '#cd0bbc'\nchr6_B  570000  770000  loc4B '#cd0bbc'\nchr6_A  2740000 2840000 loc5A '#ffa500'\nchr6_B  2860000 2930000 loc5B '#ffa500'\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003echr start   end name    colour\nchr1_A  2140000 2200000 loc1A '#61d04f'\nchr1_B  1980000 2020000 loc1B '#61d04f'\nchr1_A  4010000 4030000 loc2A '#2297e6'\nchr1_B  3760000 3770000 loc2B '#2297e6'\nchr2_A  4590000 4910000 loc3A '#542a18'\nchr2_B  4680000 4990000 loc3B '#542a18'\nchr6_A  480000  700000  loc4A '#cd0bbc'\nchr6_B  570000  770000  loc4B '#cd0bbc'\nchr6_A  2740000 2840000 loc5A '#ffa500'\nchr6_B  2860000 2930000 loc5B '#ffa500'\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3. Input for text annotation\u003c/strong\u003e - we also need a table \u003cem\u003eloci.labels.txt\u003c/em\u003e indicating the position and label for our loci. The positions can be anywhere, but here we are just using the start coordinate of each locus with the colour values matching those of the regions\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"chr pos label   colour\nchr1_A  2140000 loc1A    '#61d04f'\nchr1_B  1980000 loc1B    '#61d04f'\nchr1_A  4010000 loc2A    '#2297e6'\nchr1_B  3760000 loc2B    '#2297e6'\nchr2_A  4590000 loc3A    '#542a18'\nchr2_B  4680000 loc3B    '#542a18'\nchr6_A  480000  loc4A    '#cd0bbc'\nchr6_B  570000  loc4B    '#cd0bbc'\nchr6_A  2740000 loc5A    '#ffa500'\nchr6_B  2860000 loc5B    '#ffa500'\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003echr pos label   colour\nchr1_A  2140000 loc1A    '#61d04f'\nchr1_B  1980000 loc1B    '#61d04f'\nchr1_A  4010000 loc2A    '#2297e6'\nchr1_B  3760000 loc2B    '#2297e6'\nchr2_A  4590000 loc3A    '#542a18'\nchr2_B  4680000 loc3B    '#542a18'\nchr6_A  480000  loc4A    '#cd0bbc'\nchr6_B  570000  loc4B    '#cd0bbc'\nchr6_A  2740000 loc5A    '#ffa500'\nchr6_B  2860000 loc5B    '#ffa500'\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e4. Input for other local features\u003c/strong\u003e - a previous analysis was performed with \u003ca href=\"https://github.com/alachins/sweed\"\u003eSweeD\u003c/a\u003e to identify potential seletive sweep regions. It generated the output \u003cem\u003eselective_sweep.granges.bed\u003c/em\u003e which we will also plot\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#chr start   end\nchr1_A  112428  172428\nchr1_A  890413  950413\nchr1_B  3223380 3292289\nchr1_B  4567533 4628523\nchr2_A  422444  484441\nchr2_B  1038753 1098753\nchr2_A  1899788 1961785\nchr2_B  3122417 3182417\nchr2_A  3752711 3812711\nchr2_B  4863466 4923466\nchr2_A  5693536 5755533\nchr2_B  6070114 6169070\nchr5_B  217674  277674\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#chr start   end\nchr1_A  112428  172428\nchr1_A  890413  950413\nchr1_B  3223380 3292289\nchr1_B  4567533 4628523\nchr2_A  422444  484441\nchr2_B  1038753 1098753\nchr2_A  1899788 1961785\nchr2_B  3122417 3182417\nchr2_A  3752711 3812711\nchr2_B  4863466 4923466\nchr2_A  5693536 5755533\nchr2_B  6070114 6169070\nchr5_B  217674  277674\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e5. Prepare inputs in R\u003c/strong\u003e - now in R, load the requisite libraries and import the data. We will also specify the different haplotypes by chr name.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"library(karyoplotR)\nlibrary(svglite)\n\nmygenome \u0026lt;- toGRanges(\u0026quot;mygenome.granges.txt\u0026quot;)\nloci \u0026lt;- toGRanges(\u0026quot;loci.granges.txt\u0026quot;)\nsweeps \u0026lt;- toGRanges(\u0026quot;selective_sweep.granges.bed\u0026quot;)\nlabels \u0026lt;- read.table(\u0026quot;loci.labels.txt\u0026quot;, header=TRUE)\nhapA \u0026lt;- c(\u0026quot;chr1_A\u0026quot;, \u0026quot;chr2_A\u0026quot;, \u0026quot;chr3_A\u0026quot;, \u0026quot;chr4_A\u0026quot;, \u0026quot;chr5_A\u0026quot;, \u0026quot;chr6_A\u0026quot;)\nhapB \u0026lt;- c(\u0026quot;chr1_B\u0026quot;, \u0026quot;chr2_B\u0026quot;, \u0026quot;chr3_B\u0026quot;, \u0026quot;chr4_B\u0026quot;, \u0026quot;chr5_B\u0026quot;, \u0026quot;chr6_B\u0026quot;)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003elibrary(karyoplotR)\nlibrary(svglite)\n\nmygenome \u0026lt;- toGRanges(\"mygenome.granges.txt\")\nloci \u0026lt;- toGRanges(\"loci.granges.txt\")\nsweeps \u0026lt;- toGRanges(\"selective_sweep.granges.bed\")\nlabels \u0026lt;- read.table(\"loci.labels.txt\", header=TRUE)\nhapA \u0026lt;- c(\"chr1_A\", \"chr2_A\", \"chr3_A\", \"chr4_A\", \"chr5_A\", \"chr6_A\")\nhapB \u0026lt;- c(\"chr1_B\", \"chr2_B\", \"chr3_B\", \"chr4_B\", \"chr5_B\", \"chr6_B\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6. Plot A chrs\u003c/strong\u003e - we will create separate plots for the A and B chrs so they can later be arranged side-by-side (without subsetting, all chrs will be plotted as a single stack).\nFirst initialise an output image \u003cem\u003echrsA.svg\u003c/em\u003e using \u003ca href=\"https://github.com/r-lib/svglite\"\u003esvglite\u003c/a\u003e to save the plot as a scalable vector graphic (SVG) that can be easily resized and modified in a free SVG editor such as \u003ca href=\"https://inkscape.org/\" rel=\"nofollow\"\u003eInkscape\u003c/a\u003e, then run the plot script\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"svglite(\u0026quot;chrsA.svg\u0026quot;, width=8, height=6)\n\nkp_hapA \u0026lt;- plotKaryotype(genome=mygenome, chromosomes=hapA)\nkpPlotRegions(\n    kp_hapA, loci, \n    data.panel=\u0026quot;ideogram\u0026quot;, \n    r0=NULL, r1=NULL, \n    col=loci$colour, \n    border='#000000', \n    avoid.overlapping = FALSE\n    )\nkpPlotRegions(\n    kp_hapA, sweeps, \n    data.panel=1, \n    col=\u0026quot;#c00000\u0026quot;, \n    avoid.overlapping = FALSE, \n    r0=-0.6, r1=-0.4)\nkpAddBaseNumbers(\n    kp_hapA, \n    tick.dist = 1000001, \n    tick.len = 20, \n    tick.col=\u0026quot;#000000\u0026quot;, \n    cex = 0.01\n    )\nkpPlotMarkers(\n    kp_hapA, \n    chr=labels$chr, \n    x=labels$pos, \n    labels=labels$label, \n    text.orientation = \u0026quot;horizontal\u0026quot;, \n    line.color = \u0026quot;#000000\u0026quot;, \n    label.color = labels$colour, \n    marker.parts = c(0.1, 0.8, 0.1), \n    r1=0.4, cex=0.8, \n    ignore.chromosome.ends=FALSE)\n\ndev.off()\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esvglite(\"chrsA.svg\", width=8, height=6)\n\nkp_hapA \u0026lt;- plotKaryotype(genome=mygenome, chromosomes=hapA)\nkpPlotRegions(\n    kp_hapA, loci, \n    data.panel=\"ideogram\", \n    r0=NULL, r1=NULL, \n    col=loci$colour, \n    border='#000000', \n    avoid.overlapping = FALSE\n    )\nkpPlotRegions(\n    kp_hapA, sweeps, \n    data.panel=1, \n    col=\"#c00000\", \n    avoid.overlapping = FALSE, \n    r0=-0.6, r1=-0.4)\nkpAddBaseNumbers(\n    kp_hapA, \n    tick.dist = 1000001, \n    tick.len = 20, \n    tick.col=\"#000000\", \n    cex = 0.01\n    )\nkpPlotMarkers(\n    kp_hapA, \n    chr=labels$chr, \n    x=labels$pos, \n    labels=labels$label, \n    text.orientation = \"horizontal\", \n    line.color = \"#000000\", \n    label.color = labels$colour, \n    marker.parts = c(0.1, 0.8, 0.1), \n    r1=0.4, cex=0.8, \n    ignore.chromosome.ends=FALSE)\n\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003ekp_hapA \u0026lt;- plotKaryotype(...)\u003c/em\u003e, this creates the plot object \u003cem\u003ekp_hapA\u003c/em\u003e based on the chr lengths in \u003cem\u003emygenome\u003c/em\u003e and the subset of chrs listed in \u003cem\u003ehapA\u003c/em\u003e \u003cbr\u003e\n\u003cem\u003ekpPlotRegions(kp_hapA, loci, ...)\u003c/em\u003e, this draws the loci regions overlayed on the chrs\u003cbr\u003e\n\u003cem\u003ekpPlotRegions(kp_hapA, sweeps, ...)\u003c/em\u003e, this draws the selective sweep regions below the chrs (coloured red)\u003cbr\u003e\n\u003cem\u003ekpAddBaseNumbers(kp_hapA, ...)\u003c/em\u003e, this adds distance ticks along each chr set 1Mb apart\u003cbr\u003e\n\u003cem\u003ekpPlotMarkers(kp_hapA, chr=labels$chr, x=labels$pos, labels=labels$label, ...)\u003c/em\u003e, this adds text labels to each locus specified by names and coordinates in \u003cem\u003elabels\u003c/em\u003e \u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/267542418-b4bc15d7-0b03-45d6-b128-94a61dfd4e42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0MTgtYjRiYzE1ZDctMGIwMy00NWQ2LWIxMjgtOTRhNjFkZmQ0ZTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwOTllYmU5MDhmYjc0NzY4ZjY3NWRhN2UzNzU3YjVmODg4YjQ2YTZkZGUzMWJhMDBkYmRkNGYxNjI0NDEwOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WVYZrZ6hiw4oOjb4Bt1OSwv8n7qe0zaILjT0kGpudSA\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/267542418-b4bc15d7-0b03-45d6-b128-94a61dfd4e42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0MTgtYjRiYzE1ZDctMGIwMy00NWQ2LWIxMjgtOTRhNjFkZmQ0ZTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwOTllYmU5MDhmYjc0NzY4ZjY3NWRhN2UzNzU3YjVmODg4YjQ2YTZkZGUzMWJhMDBkYmRkNGYxNjI0NDEwOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WVYZrZ6hiw4oOjb4Bt1OSwv8n7qe0zaILjT0kGpudSA\" alt=\"chrsA_small\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e7. Plot B chrs\u003c/strong\u003e - we will now do the same for the B chrs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"svglite(\u0026quot;chrsB.svg\u0026quot;, width=8, height=6)\n\nkp_hapB \u0026lt;- plotKaryotype(genome=mygenome, chromosomes=hapB)\nkpPlotRegions(\n    kp_hapB, loci, \n    data.panel=\u0026quot;ideogram\u0026quot;, \n    r0=NULL, r1=NULL, \n    col=loci$colour, \n    border='#000000', \n    avoid.overlapping = FALSE)\nkpPlotRegions(\n    kp_hapB, \n    sweeps, \n    data.panel=1, \n    col=\u0026quot;#c00000\u0026quot;, \n    avoid.overlapping = FALSE, \n    r0=-0.6, r1=-0.4\n    )\nkpAddBaseNumbers(\n    kp_hapB, \n    tick.dist = 1000001, \n    tick.len = 20, \n    tick.col=\u0026quot;#000000\u0026quot;, \n    cex = 0.01\n    )\nkpPlotMarkers(kp_hapB, \n    chr=labels$chr, \n    x=labels$pos, \n    labels=labels$label, \n    text.orientation = \u0026quot;horizontal\u0026quot;, \n    line.color = \u0026quot;#000000\u0026quot;, \n    label.color = labels$colour, \n    marker.parts = c(0.1, 0.8, 0.1), \n    r1=0.4, cex=0.8, \n    ignore.chromosome.ends=FALSE)\n\ndev.off()\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esvglite(\"chrsB.svg\", width=8, height=6)\n\nkp_hapB \u0026lt;- plotKaryotype(genome=mygenome, chromosomes=hapB)\nkpPlotRegions(\n    kp_hapB, loci, \n    data.panel=\"ideogram\", \n    r0=NULL, r1=NULL, \n    col=loci$colour, \n    border='#000000', \n    avoid.overlapping = FALSE)\nkpPlotRegions(\n    kp_hapB, \n    sweeps, \n    data.panel=1, \n    col=\"#c00000\", \n    avoid.overlapping = FALSE, \n    r0=-0.6, r1=-0.4\n    )\nkpAddBaseNumbers(\n    kp_hapB, \n    tick.dist = 1000001, \n    tick.len = 20, \n    tick.col=\"#000000\", \n    cex = 0.01\n    )\nkpPlotMarkers(kp_hapB, \n    chr=labels$chr, \n    x=labels$pos, \n    labels=labels$label, \n    text.orientation = \"horizontal\", \n    line.color = \"#000000\", \n    label.color = labels$colour, \n    marker.parts = c(0.1, 0.8, 0.1), \n    r1=0.4, cex=0.8, \n    ignore.chromosome.ends=FALSE)\n\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/267542455-19912205-27df-4b14-9097-49b358176bd2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0NTUtMTk5MTIyMDUtMjdkZi00YjE0LTkwOTctNDliMzU4MTc2YmQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5MDM1MzliOGY1MmIxMTc0ZDNlYzJlZDEyMDI0ZTM0ZTRjN2NkZDRmMGNmMjY2N2I4ODEzMWMwZjQ0NDM2ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KoBWjMk3VxeUPFJRie-2_uogunSzzGJc3QFhwG2CH3c\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/267542455-19912205-27df-4b14-9097-49b358176bd2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0NTUtMTk5MTIyMDUtMjdkZi00YjE0LTkwOTctNDliMzU4MTc2YmQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5MDM1MzliOGY1MmIxMTc0ZDNlYzJlZDEyMDI0ZTM0ZTRjN2NkZDRmMGNmMjY2N2I4ODEzMWMwZjQ0NDM2ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KoBWjMk3VxeUPFJRie-2_uogunSzzGJc3QFhwG2CH3c\" alt=\"chrsB_small\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ethis illustrates that one of the loci (\u003cem\u003eloc3B\u003c/em\u003e) overlaps a region that has a positive signal for selection\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenerating synteny plots with \u003ca href=\"https://github.com/thackl/gggenomes\"\u003egggenomes\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-generating-synteny-plots-with-gggenomes\" class=\"anchor\" aria-label=\"Permalink: Generating synteny plots with gggenomes\" href=\"#generating-synteny-plots-with-gggenomes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHere we are creating a synteny plot of genomic regions from three different strains of the same species. One of the assemblies is fully phased into full-length chromosomes, while the other two are partially phased into contigs only. Through GWAS, we have identified significant marker-trait association regions (MTAR) on chromosomes 6 (A and B). We want to visualise annotated gene content and synteny between orthologous regions from all three assemblies. Here, an MTAR is defined as a genomic interval given by the span of SNP markers above a \u003cem\u003ep\u003c/em\u003e-value threshold in an association peak, which can be retrieved from the output matrix of GWAS programs such as \u003ca href=\"https://www.maizegenetics.net/tassel\" rel=\"nofollow\"\u003eTASSEL\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/269470527-857a35bf-c785-4a87-af56-3ca3bff481e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1MjctODU3YTM1YmYtYzc4NS00YTg3LWFmNTYtM2NhM2JmZjQ4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4Y2Q2ZDA2NmFlNzU0ZjhkNDk2MDM3NDJiZDVkMDFhMTllNzI5ODllNTczNmYxZDZkNDk0MThlMDNmMWRkODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8BwQJDCMnn_TsEAmxLg37OlN6cgprAKcCzXXajZ2ZcI\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/269470527-857a35bf-c785-4a87-af56-3ca3bff481e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1MjctODU3YTM1YmYtYzc4NS00YTg3LWFmNTYtM2NhM2JmZjQ4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4Y2Q2ZDA2NmFlNzU0ZjhkNDk2MDM3NDJiZDVkMDFhMTllNzI5ODllNTczNmYxZDZkNDk0MThlMDNmMWRkODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8BwQJDCMnn_TsEAmxLg37OlN6cgprAKcCzXXajZ2ZcI\" alt=\"ROI_from_GWAS\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn our example, the MTARs on chrs 6 in \u003cem\u003emygenome.fasta\u003c/em\u003e are defined by the following coordinates:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"chr6_A:480000-700000\nchr6_B:570000-770000\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003echr6_A:480000-700000\nchr6_B:570000-770000\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eContigs from the other two assemblies \u003cem\u003e12SD80_refassem.fa\u003c/em\u003e and \u003cem\u003e12NC29_refassem.fa\u003c/em\u003e that are orthologous to these MTARs were established based on whole assembly alignments between \u003cem\u003emygenome.fasta\u003c/em\u003e, \u003cem\u003e12SD80_refassem.fa\u003c/em\u003e and \u003cem\u003e12NC29_refassem.fa\u003c/em\u003e using a tool such as \u003ca href=\"https://github.com/lh3/minimap2\"\u003eMinimap2\u003c/a\u003e or \u003ca href=\"https://dgenies.toulouse.inra.fr/\" rel=\"nofollow\"\u003eD-Genies\u003c/a\u003e to visualise dotplots. In the below illustration, a contig-level assembly was aligned to a chr-level assembly with the dotplot zoomed-in to region of interest on \u003cem\u003echr9\u003c/em\u003e. Here we see that \u003cem\u003econtig_1\u003c/em\u003e and \u003cem\u003econtig_8\u003c/em\u003e map to this region but wih some gaps remaining.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/269470576-165c0ee1-cd72-40e5-88a8-6ff39d3c8adc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1NzYtMTY1YzBlZTEtY2Q3Mi00MGU1LTg4YTgtNmZmMzlkM2M4YWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3MWMwNGM2MDcwZmIzYWEyYTZkMDY1OTkwZjM3ZDkzN2E1ZWIxNGI4NmU4MjRjYTYyOGEzOWVmNmQzY2E4MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PmmPG-4noOUf0QdDeALCPN4idwihLzf_atXYPZTX60w\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/269470576-165c0ee1-cd72-40e5-88a8-6ff39d3c8adc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1NzYtMTY1YzBlZTEtY2Q3Mi00MGU1LTg4YTgtNmZmMzlkM2M4YWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3MWMwNGM2MDcwZmIzYWEyYTZkMDY1OTkwZjM3ZDkzN2E1ZWIxNGI4NmU4MjRjYTYyOGEzOWVmNmQzY2E4MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PmmPG-4noOUf0QdDeALCPN4idwihLzf_atXYPZTX60w\" alt=\"dotplot_synteny_juxta\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn our example, the orthologous contigs (orthotigs) in \u003cem\u003e12SD80_refassem.fa\u003c/em\u003e and \u003cem\u003e12NC29_refassem.fa\u003c/em\u003e have the following sequence IDs:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# from 12SD80\n000065F_004\n000065F\n000065F_003\n\n# from 12NC29\n000020F\n000020F_001\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# from 12SD80\n000065F_004\n000065F\n000065F_003\n\n# from 12NC29\n000020F\n000020F_001\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe also have previously generated annotations for each assembly in the form of GFF files containing predicted gene models, and for \u003cem\u003emygenome.fasta\u003c/em\u003e, an additional GFF file for sequence repeats. We will first prepare all the inputs in UNIX/bash before moving to RStudio to make the synteny plot.\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e1. retrieve MTAR and orthotig sequences\u003c/strong\u003e\u003cbr\u003e\nextract MTAR seqs from \u003cem\u003emygenome.fasta\u003c/em\u003e for each haplotype (A and B)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load samtools/1.12\nsamtools faidx mygenome.fasta chr6_A:480000-700000 | sed -r 's/:[0-9]+-[0-9]+//g' \u0026gt; MTAR_6A.fa\nsamtools faidx mygenome.fasta chr6_B:570000-770000 | sed -r 's/:[0-9]+-[0-9]+//g' \u0026gt; MTAR_6B.fa\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emodule load samtools/1.12\nsamtools faidx mygenome.fasta chr6_A:480000-700000 | sed -r 's/:[0-9]+-[0-9]+//g' \u0026gt; MTAR_6A.fa\nsamtools faidx mygenome.fasta chr6_B:570000-770000 | sed -r 's/:[0-9]+-[0-9]+//g' \u0026gt; MTAR_6B.fa\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003esed -r ...\u003c/em\u003e, this removes the coordinates which samtools automatically adds to the output fasta headers (otherwise seq IDs won't match GFFs)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003enow pull out the orthotigs from \u003cem\u003e12SD80_refassem.fa\u003c/em\u003e and \u003cem\u003e12NC29_refassem.fa\u003c/em\u003e. This can be done with a tool such as \u003ca href=\"https://github.com/TC-Hewitt/Misc_NGS/blob/master/get_contigs.py\"\u003e\u003cem\u003eget_contigs.py\u003c/em\u003e\u003c/a\u003e from my other repo \u003ca href=\"https://github.com/TC-Hewitt/Misc_NGS\"\u003eMisc_NGS\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"~/Misc_NGS/get_contigs.py -i 12SD80_refassem.fa -l 000065F_004 000065F 000065F_003 -o MTAR_6_12SD80_orthotigs.fa\n~/Misc_NGS/get_contigs.py -i 12NC29_refassem.fa -l 000020F 000020F_001 -o MTAR_6_12NC29_orthotigs.fa\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e~/Misc_NGS/get_contigs.py -i 12SD80_refassem.fa -l 000065F_004 000065F 000065F_003 -o MTAR_6_12SD80_orthotigs.fa\n~/Misc_NGS/get_contigs.py -i 12NC29_refassem.fa -l 000020F 000020F_001 -o MTAR_6_12NC29_orthotigs.fa\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-i\u003c/em\u003e, input fasta file\u003cbr\u003e\n\u003cem\u003e-l\u003c/em\u003e, space-separated list of seq IDs\u003cbr\u003e\n\u003cem\u003e-o\u003c/em\u003e, output fasta file\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e2. retrieve corresponding annotations for MTAR seqs and orthotigs\u003c/strong\u003e\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eusing an AWK script, we will use the same coordinates as previously to pull out annotations for each MTAR from \u003cem\u003emygenome.models.gff\u003c/em\u003e then use \u003cem\u003egrep\u003c/em\u003e to keep only \u003cem\u003emRNA\u003c/em\u003e features\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"awk '$1 == \u0026quot;chr6_A\u0026quot; \u0026amp;\u0026amp; $4 \u0026gt;= 480000 \u0026amp;\u0026amp; $5 \u0026lt;= 700000' mygenome.models.gff | grep mRNA \u0026gt; MTAR_6A.mrna.gff\nawk '$1 == \u0026quot;chr6_B\u0026quot; \u0026amp;\u0026amp; $4 \u0026gt;= 570000 \u0026amp;\u0026amp; $5 \u0026lt;= 770000' mygenome.models.gff | grep mRNA \u0026gt; MTAR_6B.mrna.gff\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eawk '$1 == \"chr6_A\" \u0026amp;\u0026amp; $4 \u0026gt;= 480000 \u0026amp;\u0026amp; $5 \u0026lt;= 700000' mygenome.models.gff | grep mRNA \u0026gt; MTAR_6A.mrna.gff\nawk '$1 == \"chr6_B\" \u0026amp;\u0026amp; $4 \u0026gt;= 570000 \u0026amp;\u0026amp; $5 \u0026lt;= 770000' mygenome.models.gff | grep mRNA \u0026gt; MTAR_6B.mrna.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe will do the same for the repeat annotations\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"awk '$1 == \u0026quot;chr6_A\u0026quot; \u0026amp;\u0026amp; $4 \u0026gt;= 480000 \u0026amp;\u0026amp; $5 \u0026lt;= 700000' mygenome.repeats.gff \u0026gt; MTAR_6A.repeats.gff\nawk '$1 == \u0026quot;chr6_B\u0026quot; \u0026amp;\u0026amp; $4 \u0026gt;= 570000 \u0026amp;\u0026amp; $5 \u0026lt;= 770000' mygenome.repeats.gff \u0026gt; MTAR_6B.repeats.gff\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eawk '$1 == \"chr6_A\" \u0026amp;\u0026amp; $4 \u0026gt;= 480000 \u0026amp;\u0026amp; $5 \u0026lt;= 700000' mygenome.repeats.gff \u0026gt; MTAR_6A.repeats.gff\nawk '$1 == \"chr6_B\" \u0026amp;\u0026amp; $4 \u0026gt;= 570000 \u0026amp;\u0026amp; $5 \u0026lt;= 770000' mygenome.repeats.gff \u0026gt; MTAR_6B.repeats.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003enow we have annotations for just the MTARs; however, the \u003cem\u003estart\u003c/em\u003e and \u003cem\u003eend\u003c/em\u003e coordinates of each feature are still relative to the whole chrs 6A and 6B; to modify their positions to be relative to \u003cem\u003eMTAR_6A.fa\u003c/em\u003e and \u003cem\u003eMTAR_6B.fa\u003c/em\u003e we can use AWK to subtract just the start coordinate of the respective MTAR from both the \u003cem\u003estart\u003c/em\u003e and \u003cem\u003eend\u003c/em\u003e values of each feature\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"awk -F $'\\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\\t' MTAR_6A.mrna.gff \u0026gt; MTAR_6.mrna.modpos.gff\nawk -F $'\\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\\t' MTAR_6B.mrna.gff \u0026gt;\u0026gt; MTAR_6.mrna.modpos.gff\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eawk -F $'\\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\\t' MTAR_6A.mrna.gff \u0026gt; MTAR_6.mrna.modpos.gff\nawk -F $'\\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\\t' MTAR_6B.mrna.gff \u0026gt;\u0026gt; MTAR_6.mrna.modpos.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ehere we write the outputs from both haplotypes to the same file \u003cem\u003eMTAR_6.mrna.modpos.gff\u003c/em\u003e \u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"awk -F $'\\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\\t' MTAR_6A.repeats.gff \u0026gt; MTAR_6.repeats.modpos.gff\nawk -F $'\\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\\t' MTAR_6B.repeats.gff \u0026gt;\u0026gt; MTAR_6.repeats.modpos.gff\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eawk -F $'\\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\\t' MTAR_6A.repeats.gff \u0026gt; MTAR_6.repeats.modpos.gff\nawk -F $'\\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\\t' MTAR_6B.repeats.gff \u0026gt;\u0026gt; MTAR_6.repeats.modpos.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ehere we write the outputs from both haplotypes to the same file \u003cem\u003eMTAR_6.repeats.modpos.gff\u003c/em\u003e \u003cbr\u003e\nthis isn't always necessary since the R package \u003cem\u003egggenomes\u003c/em\u003e allows intervals to be custom defined in generating synteny plots; however, to save overhead of importing the entire chr or assembly and GFFs files into RStudio, we have opted to prep the inputs beforehand\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003elikewise, we will retrieve the \u003cem\u003emRNA\u003c/em\u003e annotations for the orthotigs from \u003cem\u003e12SD80\u003c/em\u003e and \u003cem\u003e12NC29\u003c/em\u003e using a \u003cem\u003ebash\u003c/em\u003e script; we do not need to modify the feature positions since they are already relative to the individual contigs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# 12SD80\nfor i in 000065F_004 000065F 000065F_003\ndo\n    grep \u0026quot;^$i\\s.*mRNA\u0026quot; 12SD80_refassem.models.gff3 \u0026gt;\u0026gt; MTAR_6_12SD80_orthotigs.mrna.gff\ndone\n\n#12NC29\nfor i in 000020F 000020F_001\ndo\n    grep \u0026quot;^$i\\s.*mRNA\u0026quot; 12NC29_refassem.models.gff3 \u0026gt;\u0026gt; MTAR_6_12NC29_orthotigs.mrna.gff\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# 12SD80\nfor i in 000065F_004 000065F 000065F_003\ndo\n    grep \"^$i\\s.*mRNA\" 12SD80_refassem.models.gff3 \u0026gt;\u0026gt; MTAR_6_12SD80_orthotigs.mrna.gff\ndone\n\n#12NC29\nfor i in 000020F 000020F_001\ndo\n    grep \"^$i\\s.*mRNA\" 12NC29_refassem.models.gff3 \u0026gt;\u0026gt; MTAR_6_12NC29_orthotigs.mrna.gff\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e3. compile fasta files and annotation files of MTARs and orthotigs for later export to RStudio\u003c/strong\u003e\u003cbr\u003e\ncombine MTAR and orthotig fastas to \u003cem\u003eMTAR_6_allseqs.fa\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat \\\nMTAR_6A.fa \\\nMTAR_6B.fa \\\nMTAR_6_12SD80_orthotigs.fa \\\nMTAR_6_12NC29_orthotigs.fa \\\n\u0026gt; MTAR_6_allseqs.fa\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecat \\\nMTAR_6A.fa \\\nMTAR_6B.fa \\\nMTAR_6_12SD80_orthotigs.fa \\\nMTAR_6_12NC29_orthotigs.fa \\\n\u0026gt; MTAR_6_allseqs.fa\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ecombine MTAR and orthotig gffs to \u003cem\u003eMTAR_6_allseqs.mrna.gff\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat \\\nMTAR_6.mrna.modpos.gff \\\nMTAR_6_12SD80_orthotigs.mrna.gff \\\nMTAR_6_12NC29_orthotigs.mrna.gff \\\n\u0026gt; MTAR_6_allseqs.mrna.gff\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecat \\\nMTAR_6.mrna.modpos.gff \\\nMTAR_6_12SD80_orthotigs.mrna.gff \\\nMTAR_6_12NC29_orthotigs.mrna.gff \\\n\u0026gt; MTAR_6_allseqs.mrna.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003esince we only have repeat annotations for the MTARs, we do not need to combine them with those of the orthotigs (unavailable)\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e4. perform an all-vs-all alignment of \u003cem\u003eMTAR_6_allseqs.fa\u003c/em\u003e with \u003ca href=\"https://github.com/lh3/minimap2\"\u003eMinimap2\u003c/a\u003e\u003c/strong\u003e \u003cbr\u003e\nthe PAF output will be later used for drawing links between regions in the synteny plot\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load minimap2/2.24\nminimap2 -X MTAR_6_allseqs.fa MTAR_6_allseqs.fa \u0026gt; MTAR_6_allseqs.paf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emodule load minimap2/2.24\nminimap2 -X MTAR_6_allseqs.fa MTAR_6_allseqs.fa \u0026gt; MTAR_6_allseqs.paf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-X\u003c/em\u003e, skip self and dual mappings (for the all-vs-all mode)\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e5. generate ortholog clusters of gene models using \u003ca href=\"https://github.com/davidemms/OrthoFinder\"\u003eOrthoFinder\u003c/a\u003e\u003c/strong\u003e \u003cbr\u003e\nwe will use protein fasta files previously created along with the GFF annotations of the three assemblies; a package such as \u003ca href=\"https://github.com/NBISweden/AGAT\"\u003eAGAT\u003c/a\u003e can be used to generate translated protein sequences from a gff and nucleotide fasta\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ethe orthogroups will be later used for colouring genes and drawing links between genes in the synteny plot\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efirst we will retrieve the matching sequences from each protein fasta with \u003ca href=\"https://github.com/TC-Hewitt/Misc_NGS/blob/master/get_contigs.py\"\u003e\u003cem\u003eget_contigs.py\u003c/em\u003e\u003c/a\u003e using \u003cem\u003eMTAR_6_allseqs.mrna.gff\u003c/em\u003e as a reference\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"~/Misc_NGS/get_contigs.py -i mygenome.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6.proteins.faa\n~/Misc_NGS/get_contigs.py -i 12SD80_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12SD80_orthotigs.proteins.faa\n~/Misc_NGS/get_contigs.py -i 12NC29_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12NC29_orthotigs.proteins.faa\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e~/Misc_NGS/get_contigs.py -i mygenome.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6.proteins.faa\n~/Misc_NGS/get_contigs.py -i 12SD80_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12SD80_orthotigs.proteins.faa\n~/Misc_NGS/get_contigs.py -i 12NC29_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12NC29_orthotigs.proteins.faa\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-i\u003c/em\u003e, input fasta file\u003cbr\u003e\n\u003cem\u003e-t\u003c/em\u003e, reference file containing IDs of seqs to retrieve\u003cbr\u003e\n\u003cem\u003e-s gene\u003c/em\u003e_, prefix of seq IDs to look for in reference file (i.e. starts with \"gene_\")\u003cbr\u003e\n\u003cem\u003e-o\u003c/em\u003e, output fasta file\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003enow we will create a separate directory \u003cem\u003etranslated\u003c/em\u003e to move our retrieved protein seqs and run \u003cem\u003eorthofinder\u003c/em\u003e on\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir translated\nmv *.faa translated/\n\nmodule load orthofinder/2.5.4\northofinder -a 4 -f translated/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emkdir translated\nmv *.faa translated/\n\nmodule load orthofinder/2.5.4\northofinder -a 4 -f translated/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e-a\u003c/em\u003e, number of parallel analysis threads\u003cbr\u003e\n\u003cem\u003e-f\u003c/em\u003e, dir containing fasta proteomes\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003ethe output we want for our purposes is found in \u003cem\u003etranslated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt\u003c/em\u003e where the wildcard \"?????\" substitutes for the date the orthofinder output was created (e.g. \"Feb28\"); \u003cem\u003eOrthogroups.txt\u003c/em\u003e is not in a tractable format, so we can convert it to long format using \u003ca href=\"https://github.com/TC-Hewitt/OatCrownRust/blob/main/Orthogroups_txtConvert.py\"\u003e\u003cem\u003eOrthogroups_txtConvert.py\u003c/em\u003e\u003c/a\u003e for later use in R\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"./Orthogroups_txtConvert.py -i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt \\\n\u0026gt; MTAR_6_allseqs.orthogroups.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e./Orthogroups_txtConvert.py -i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt \\\n\u0026gt; MTAR_6_allseqs.orthogroups.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e6. pull out only secreted subset from orthogroups\u003c/strong\u003e\u003cbr\u003e\nin this example, we are particularly interested in genes that encode secreted proteins and want to highlight these in our synteny plot later; here they are denoted by the string \"secreted\" in the attribute field (column 9) of the gff files; for example, column 9 of a single gene record in \u003cem\u003eMTAR_6_allseqs.mrna.gff\u003c/em\u003e looks like the following:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"id=gene_1-T1;parent=gene_1;product=secreted protein;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eid=gene_1-T1;parent=gene_1;product=secreted protein;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eof course, this depends on the specific annotation workflow used in assigning attributes to gene models, but for our example, our genes of interest can be easily retrieved with \u003cem\u003egrep\u003c/em\u003e; as we only need the transcript name (\u003cem\u003eid=...\u003c/em\u003e), we can strip other columns and attributes to output a simple list of IDs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"grep \u0026quot;secreted\u0026quot; MTAR_6_allseqs.mrna.gff \\\n| cut -f 9 | sed -r \u0026quot;s/id=([^;]+).*$/\\1/g\u0026quot; \\\n\u0026gt; MTAR_6_allseqs.secreted.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egrep \"secreted\" MTAR_6_allseqs.mrna.gff \\\n| cut -f 9 | sed -r \"s/id=([^;]+).*$/\\1/g\" \\\n\u0026gt; MTAR_6_allseqs.secreted.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003ecut -f 9\u003c/em\u003e, limit to col 9\u003cbr\u003e\n\u003cem\u003esed -r ...\u003c/em\u003e, regex operation discards all chars except those matching group 1 (any consecutive chars after \"id=\" except \";\" in the parentheses)\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003ewe will then use the IDs to extract the orthogroups from \u003cem\u003eOrthogroups.txt\u003c/em\u003e and write to a temp file; this will pull out all the genes of an orthogroup, as long as the group contains at least one secreted member in \u003cem\u003eMTAR_6_allseqs.secreted.txt\u003c/em\u003e (this is desirable since we may want to know if the secretion state varies between orthologs of different assemblies/strains)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for i in $(cat MTAR_allseqs.secreted.txt) \ndo\n    grep $i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt \u0026gt;\u0026gt; temp1.txt \ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efor i in $(cat MTAR_allseqs.secreted.txt) \ndo\n    grep $i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt \u0026gt;\u0026gt; temp1.txt \ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003egiven the above, \u003cem\u003etemp1.txt\u003c/em\u003e is likely to contain duplicate lines which we can remove with \u003cem\u003esort\u003c/em\u003e and \u003cem\u003euniq\u003c/em\u003e and write to a new temp file\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sort temp1.txt | uniq \u0026gt; temp2.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esort temp1.txt | uniq \u0026gt; temp2.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe can now convert this to a desirable format using \u003cem\u003eOrthogroups_txtConvert.py\u003c/em\u003e then delete the temp files\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"./Orthogroups_txtConvert.py -i temp2.txt \u0026gt; MTAR_6_allseqs.orthogroups.secreted.txt\nrm temp1.txt temp2.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e./Orthogroups_txtConvert.py -i temp2.txt \u0026gt; MTAR_6_allseqs.orthogroups.secreted.txt\nrm temp1.txt temp2.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eMTAR_6_allseqs.orthogroups.secreted.txt\u003c/em\u003e now contains only proteins that are either secreted or belong to the same orthogroup as a secreted protein\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e7. create zip folder of all input files for export to Rstudio\u003c/strong\u003e\u003cbr\u003e\nnow all input files for our synteny plot are ready, we will zip them in \u003cem\u003eexportR\u003c/em\u003e for easy transfer to RStudio\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"zip exportR \\\nMTAR_6_allseqs.fa \\\nMTAR_6_allseqs.gff \\\nMTAR_6.repeats.modpos.gff \\\nMTAR_6_allseqs.paf \\\nMTAR_6_allseqs.orthogroups.txt \\\nMTAR_6_allseqs.orthogroups.secreted.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ezip exportR \\\nMTAR_6_allseqs.fa \\\nMTAR_6_allseqs.gff \\\nMTAR_6.repeats.modpos.gff \\\nMTAR_6_allseqs.paf \\\nMTAR_6_allseqs.orthogroups.txt \\\nMTAR_6_allseqs.orthogroups.secreted.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003enow in RStudio, we want to draw a synteny plot using \u003cem\u003egggenomes\u003c/em\u003e with the following features:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003esequences arranged in bins based on haplotype\u003c/li\u003e\n\u003cli\u003elinks bewteen adjacent sequences showing blocks of homology\u003c/li\u003e\n\u003cli\u003edirectional gene annotations\u003c/li\u003e\n\u003cli\u003estranded repeat blocks\u003c/li\u003e\n\u003cli\u003egenes coloured by orthogroup (secreted only)\u003c/li\u003e\n\u003cli\u003elines between adjacent sequences linking genes of the same orthogroup\u003c/li\u003e\n\u003cli\u003egene labels (secreted only)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003ethe following was conducted in RStudio using R 4.0.2 and gggenomes 0.9.5.9000\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e8. load libraries and read in input files\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"library(gggenomes)\nlibrary(svglite)\n\n#assign inputs\nmy_seqs \u0026lt;- read_seqs(\u0026quot;MTAR_6_allseqs.fa\u0026quot;)\nmy_genes \u0026lt;- read_gff3(\u0026quot;MTAR_6_allseqs.gff\u0026quot;)\nmy_repeats \u0026lt;- read_gff(\u0026quot;MTAR_6.repeats.modpos.gff\u0026quot;)\nmy_links \u0026lt;- read_paf(\u0026quot;MTAR_6_allseqs.paf\u0026quot;)\nmy_clusts \u0026lt;- read.table(\u0026quot;MTAR_6_allseqs.orthogroups.txt\u0026quot;, header=T, sep=\u0026quot;\\t\u0026quot;)\nmy_clusts_sec \u0026lt;- read.table(\u0026quot;MTAR_6_allseqs.orthogroups.secreted.txt\u0026quot;, header=T, sep=\u0026quot;\\t\u0026quot;)\n\n#remove 3 leading zeroes from 12SD80 and 12NC29 derived seq_ids for cleaner labels later on\nmy_seqs$seq_id \u0026lt;- sub(\u0026quot;^000\u0026quot;, \u0026quot;\u0026quot;, my_seqs$seq_id)\nmy_genes$seq_id \u0026lt;- sub(\u0026quot;^000\u0026quot;, \u0026quot;\u0026quot;, my_genes$seq_id)\nmy_links$seq_id \u0026lt;- sub(\u0026quot;^000\u0026quot;, \u0026quot;\u0026quot;, my_links$seq_id)\nmy_links$seq_id2 \u0026lt;- sub(\u0026quot;^000\u0026quot;, \u0026quot;\u0026quot;, my_links$seq_id2)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003elibrary(gggenomes)\nlibrary(svglite)\n\n#assign inputs\nmy_seqs \u0026lt;- read_seqs(\"MTAR_6_allseqs.fa\")\nmy_genes \u0026lt;- read_gff3(\"MTAR_6_allseqs.gff\")\nmy_repeats \u0026lt;- read_gff(\"MTAR_6.repeats.modpos.gff\")\nmy_links \u0026lt;- read_paf(\"MTAR_6_allseqs.paf\")\nmy_clusts \u0026lt;- read.table(\"MTAR_6_allseqs.orthogroups.txt\", header=T, sep=\"\\t\")\nmy_clusts_sec \u0026lt;- read.table(\"MTAR_6_allseqs.orthogroups.secreted.txt\", header=T, sep=\"\\t\")\n\n#remove 3 leading zeroes from 12SD80 and 12NC29 derived seq_ids for cleaner labels later on\nmy_seqs$seq_id \u0026lt;- sub(\"^000\", \"\", my_seqs$seq_id)\nmy_genes$seq_id \u0026lt;- sub(\"^000\", \"\", my_genes$seq_id)\nmy_links$seq_id \u0026lt;- sub(\"^000\", \"\", my_links$seq_id)\nmy_links$seq_id2 \u0026lt;- sub(\"^000\", \"\", my_links$seq_id2)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ewe wont end up using \u003cem\u003emy_clusts\u003c/em\u003e in our final plot and will only use \u003cem\u003emy_clusts_sec\u003c/em\u003e as we want to highlight only secreted genes\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e9. create preliminary plot for assessing sequence arrangements\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"synplot1 \u0026lt;- gggenomes(\n     genes = my_genes,\n     seqs = my_seqs,\n     feats = NULL,\n     links = my_links,\n     .id = \u0026quot;file_id\u0026quot;,\n     spacing = 0.05,\n     wrap = NULL,\n     adjacent_only = FALSE, #will also plot links b/w non-adjacent seqs\n     infer_bin_id = seq_id,\n     infer_start = min(start, end),\n     infer_end = max(start, end),\n     infer_length = max(start, end),\n     theme = c(\u0026quot;clean\u0026quot;, NULL),\n     .layout = NULL) + geom_link(alpha=0.1) + geom_seq() + geom_bin_label()\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esynplot1 \u0026lt;- gggenomes(\n     genes = my_genes,\n     seqs = my_seqs,\n     feats = NULL,\n     links = my_links,\n     .id = \"file_id\",\n     spacing = 0.05,\n     wrap = NULL,\n     adjacent_only = FALSE, #will also plot links b/w non-adjacent seqs\n     infer_bin_id = seq_id,\n     infer_start = min(start, end),\n     infer_end = max(start, end),\n     infer_length = max(start, end),\n     theme = c(\"clean\", NULL),\n     .layout = NULL) + geom_link(alpha=0.1) + geom_seq() + geom_bin_label()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe can select a few sequences to plot at a time to help figure out how they should be ordered and binned\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"synplot1 %\u0026gt;% pick(\u0026quot;chr6_A\u0026quot;, \u0026quot;chr6_B\u0026quot;, \u0026quot;065F\u0026quot;, \u0026quot;065F_004\u0026quot;, \u0026quot;065F_003\u0026quot;)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esynplot1 %\u0026gt;% pick(\"chr6_A\", \"chr6_B\", \"065F\", \"065F_004\", \"065F_003\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003efor example, this plots the A and B MTARs along with orthotigs from \u003cem\u003e12SD80\u003c/em\u003e. Homology links are shown between all seqs.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/269470823-f7c40d1e-dd1d-4da0-bd57-bab18ca2080b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4MjMtZjdjNDBkMWUtZGQxZC00ZGEwLWJkNTctYmFiMThjYTIwODBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkY2Q1Y2YwNzQ1ZTJlNDM0NjBiMjZhYTJjMmY0ZTUwNjc5ZTJhMDZjNzQ5Y2NmYzQ1Y2VhYTdkOGMwZjFkM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rolAIhw09cOci1uJadwIUrBqUtqSbLxz1GlNr9DjS90\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/269470823-f7c40d1e-dd1d-4da0-bd57-bab18ca2080b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4MjMtZjdjNDBkMWUtZGQxZC00ZGEwLWJkNTctYmFiMThjYTIwODBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkY2Q1Y2YwNzQ1ZTJlNDM0NjBiMjZhYTJjMmY0ZTUwNjc5ZTJhMDZjNzQ5Y2NmYzQ1Y2VhYTdkOGMwZjFkM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rolAIhw09cOci1uJadwIUrBqUtqSbLxz1GlNr9DjS90\" alt=\"synplot_exp1\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eon first look we can see the plot may benefit from alternative ordering and orientation of seqs. For instance, contigs \u003cem\u003e065F_003\u003c/em\u003e and \u003cem\u003e065F_004\u003c/em\u003e appear to align side-by-side against the larger \u003cem\u003e065F\u003c/em\u003e contig, meaning they should be placed in the same bin (each bin representing an inferred haplotype). By default, all seqs are plotted as a single stack unless they are assigned to bins allowing seqs to be positioned in series\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e10. assign seqs to bins and create a new plot\u003c/strong\u003e\u003cbr\u003e\nfirst we need to know the lengths of our seqs to enter into the table. These can be found in the \u003cem\u003emy_seqs\u003c/em\u003e object created earlier\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"print(my_seqs)\n#   seq_id  seq_desc    length\n    \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;int\u0026gt;\n    chr6_A  NA  220001\n    chr6_B  NA  200001\n    000065F_004 NA  103286\n    000065F NA  373524\n    000065F_003 NA  237372\n    000020F NA  481180\n    000020F_001 NA  241894\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eprint(my_seqs)\n#   seq_id  seq_desc    length\n    \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;int\u0026gt;\n    chr6_A  NA  220001\n    chr6_B  NA  200001\n    000065F_004 NA  103286\n    000065F NA  373524\n    000065F_003 NA  237372\n    000020F NA  481180\n    000020F_001 NA  241894\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe can then define our bin information in a new table object\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"my_bins \u0026lt;- tribble(\n  ~bin_id, ~seq_id, ~length,\n  \u0026quot;mygenome_1\u0026quot;, \u0026quot;chr6_A\u0026quot;, 220001,\n  \u0026quot;mygenome_2\u0026quot;, \u0026quot;chr6_B\u0026quot;, 200001,\n  \u0026quot;12SD80_1\u0026quot;, \u0026quot;065F\u0026quot;, 373524,\n  \u0026quot;12SD80_2\u0026quot;, \u0026quot;065F_004\u0026quot;, 103286,\n  \u0026quot;12SD80_2\u0026quot;, \u0026quot;065F_003\u0026quot;, 237372,  \n  \u0026quot;12NC29_1\u0026quot;, \u0026quot;020F\u0026quot;, 481180,\n  \u0026quot;12NC29_2\u0026quot;, \u0026quot;020F_001\u0026quot;, 241894,\n)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emy_bins \u0026lt;- tribble(\n  ~bin_id, ~seq_id, ~length,\n  \"mygenome_1\", \"chr6_A\", 220001,\n  \"mygenome_2\", \"chr6_B\", 200001,\n  \"12SD80_1\", \"065F\", 373524,\n  \"12SD80_2\", \"065F_004\", 103286,\n  \"12SD80_2\", \"065F_003\", 237372,  \n  \"12NC29_1\", \"020F\", 481180,\n  \"12NC29_2\", \"020F_001\", 241894,\n)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe may also want to filter out smaller links from our \u003cem\u003emy_links\u003c/em\u003e object so they do not clutter the plot. Here we will only keep links of thickness 5kb or more and assign to \u003cem\u003emy_links_5kb\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"my_links_5kb \u0026lt;- filter(my_links, map_length \u0026gt;= 5000)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emy_links_5kb \u0026lt;- filter(my_links, map_length \u0026gt;= 5000)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe can now create our initial plot substituting in \u003cem\u003emy_bins\u003c/em\u003e and \u003cem\u003emy_links_5kb\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"synplot2 \u0026lt;- gggenomes(\n     genes = my_genes,\n     seqs = my_bins, # new seq table\n     feats = my_repeats,\n     links = my_links_5kb, # new links obj\n     .id = \u0026quot;file_id\u0026quot;,\n     spacing = 0.05,\n     wrap = NULL,\n     adjacent_only = TRUE, # links will only be shown between adjacent seqs\n     infer_bin_id = bin_id, # make sure changed to bin_id\n     infer_start = min(start, end),\n     infer_end = max(start, end),\n     infer_length = max(start, end),\n     theme = c(\u0026quot;clean\u0026quot;, NULL),\n     .layout = NULL) +\n     geom_bin_label() + \n     geom_seq() + \n     geom_link(fill=\u0026quot;lightgrey\u0026quot;, color=\u0026quot;lightgrey\u0026quot;) + \n     geom_seq_label()\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esynplot2 \u0026lt;- gggenomes(\n     genes = my_genes,\n     seqs = my_bins, # new seq table\n     feats = my_repeats,\n     links = my_links_5kb, # new links obj\n     .id = \"file_id\",\n     spacing = 0.05,\n     wrap = NULL,\n     adjacent_only = TRUE, # links will only be shown between adjacent seqs\n     infer_bin_id = bin_id, # make sure changed to bin_id\n     infer_start = min(start, end),\n     infer_end = max(start, end),\n     infer_length = max(start, end),\n     theme = c(\"clean\", NULL),\n     .layout = NULL) +\n     geom_bin_label() + \n     geom_seq() + \n     geom_link(fill=\"lightgrey\", color=\"lightgrey\") + \n     geom_seq_label()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/269470889-7bb20abd-d5b1-472f-bafc-8d03fb238211.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4ODktN2JiMjBhYmQtZDViMS00NzJmLWJhZmMtOGQwM2ZiMjM4MjExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiNjllN2JiZDZkNDFjZjM0ZTRhOWYxZDFmYzhiMjI3NmQ1ZDYyOWNkMzJkZWE3NzBiNTA0ZjE2YTBhYWRlNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FRldwEMOFGc4xQ0N2Vm-xuA299dKaqQJKGSpTddzNp8\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/269470889-7bb20abd-d5b1-472f-bafc-8d03fb238211.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4ODktN2JiMjBhYmQtZDViMS00NzJmLWJhZmMtOGQwM2ZiMjM4MjExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiNjllN2JiZDZkNDFjZjM0ZTRhOWYxZDFmYzhiMjI3NmQ1ZDYyOWNkMzJkZWE3NzBiNTA0ZjE2YTBhYWRlNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FRldwEMOFGc4xQ0N2Vm-xuA299dKaqQJKGSpTddzNp8\" alt=\"synplot_exp2\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eour seqs are now ordered the way we want, except their orientations are still mismatched\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e11. flip seqs to match orientations\u003c/strong\u003e\u003cbr\u003e\nwe can see the alignment can be fixed by flipping seqs \u003cem\u003e065F\u003c/em\u003e, \u003cem\u003e065F_004\u003c/em\u003e and \u003cem\u003e065F_003\u003c/em\u003e, which are seq numbers 3, 4 and 5, corresponding to the order they are listed in \u003cem\u003emy_bins\u003c/em\u003e. We assign the untwisted alignment to a new plot object\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"synplot2_flip \u0026lt;- synplot2 %\u0026gt;% flip_seqs(3, 4, 5)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esynplot2_flip \u0026lt;- synplot2 %\u0026gt;% flip_seqs(3, 4, 5)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/269470954-2611676c-0c5f-43e8-82b3-9a712155256b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA5NTQtMjYxMTY3NmMtMGM1Zi00M2U4LTgyYjMtOWE3MTIxNTUyNTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYTMxYTc4MTQwODg1NmM5ZjQyMzYzNjQ3N2VkNjAyMGQzMDgwM2U1NDUxZTgzNmM0YjgwMzRmMTQyY2Q0ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._zFxlRWlgHP_CeitZLwmeU0TmKEHsRwFWgnD1BmLRaY\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/269470954-2611676c-0c5f-43e8-82b3-9a712155256b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA5NTQtMjYxMTY3NmMtMGM1Zi00M2U4LTgyYjMtOWE3MTIxNTUyNTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYTMxYTc4MTQwODg1NmM5ZjQyMzYzNjQ3N2VkNjAyMGQzMDgwM2U1NDUxZTgzNmM0YjgwMzRmMTQyY2Q0ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._zFxlRWlgHP_CeitZLwmeU0TmKEHsRwFWgnD1BmLRaY\" alt=\"synplot_exp3\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eit is now looking much better, but seq \u003cem\u003e020F\u003c/em\u003e is showing some unaligned flanking regions which we might want to trim\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e12. assign seq boundaries to plot only desired regions\u003c/strong\u003e\u003cbr\u003e\nwe can create a table defining the start and end coordinates of each seq region we want to show. Note that coordinates are always based on the original seqs (unflipped), so the previous \u003cem\u003esynplot2\u003c/em\u003e can be used a reference when defining region coordinates. The default length bar at the bottom can be used as a guide.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"my_loci \u0026lt;- tribble(\n  ~seq_id, ~start, ~end,\n  \u0026quot;chr6_A\u0026quot;, 1, 220001,\n  \u0026quot;chr6_B\u0026quot;, 1, 200001,\n  \u0026quot;065F\u0026quot;, 1, 373524,\n  \u0026quot;065F_004\u0026quot;, 1, 103286,\n  \u0026quot;065F_003\u0026quot;, 1, 237372,  \n  \u0026quot;020F\u0026quot;, 70000, 450000,\n  \u0026quot;020F_001\u0026quot;, 1, 241894,\n)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emy_loci \u0026lt;- tribble(\n  ~seq_id, ~start, ~end,\n  \"chr6_A\", 1, 220001,\n  \"chr6_B\", 1, 200001,\n  \"065F\", 1, 373524,\n  \"065F_004\", 1, 103286,\n  \"065F_003\", 1, 237372,  \n  \"020F\", 70000, 450000,\n  \"020F_001\", 1, 241894,\n)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe can then use the focus function with \u003cem\u003emy_loci\u003c/em\u003e to create a new plot object showing only desired regions\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"syplot2_flip_focus \u0026lt;- synplot2_flip %\u0026gt;% focus(.track_id=seqs, .loci=my_loci, .locus_id=seq_id)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esyplot2_flip_focus \u0026lt;- synplot2_flip %\u0026gt;% focus(.track_id=seqs, .loci=my_loci, .locus_id=seq_id)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/269471107-4304ba3e-0f8c-472d-b8da-7f7e1ce0dff7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzExMDctNDMwNGJhM2UtMGY4Yy00NzJkLWI4ZGEtN2Y3ZTFjZTBkZmY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiY2JiNDA3M2QzNjgwYTFiZjVmMjFiMjc1YTIyNDM4MzEzZjJkYjhhNmFlMzBhYmJmZjQxOTE2ZDcyYTVlY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z2Zp3NrcNwgycAivYFMeGsagweeWsaw4a09Yg1dWwS4\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/269471107-4304ba3e-0f8c-472d-b8da-7f7e1ce0dff7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzExMDctNDMwNGJhM2UtMGY4Yy00NzJkLWI4ZGEtN2Y3ZTFjZTBkZmY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiY2JiNDA3M2QzNjgwYTFiZjVmMjFiMjc1YTIyNDM4MzEzZjJkYjhhNmFlMzBhYmJmZjQxOTE2ZDcyYTVlY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z2Zp3NrcNwgycAivYFMeGsagweeWsaw4a09Yg1dWwS4\" alt=\"synplot_exp4\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enow that we have a bare plot combining our desired ordering, orientation and boundaries, we can start adding features and aesthetics\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e13. modify package functions for custom fill and links\u003c/strong\u003e\u003cbr\u003e\nby default, \u003cem\u003egggenomes\u003c/em\u003e function \u003cem\u003egeom_gene()\u003c/em\u003e draws features of type \"mRNA\" from a gff with a lightened color scheme. However, we want our genes to have full solid colour. To temporarily disable automatic lighten,  open and edit the \u003cem\u003egeom_gene()\u003c/em\u003e code in the console using \u003ccode\u003etrace(gggenomes::geom_gene, edit=TRUE)\u003c/code\u003e. Now in the text editor window, change the line \u003ccode\u003erna_def \u0026lt;- aes(fill = colorspace::lighten(fill, .5), color = colorspace::lighten(colour, .5))\u003c/code\u003e to simply \u003ccode\u003erna_def \u0026lt;- aes()\u003c/code\u003e\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eour plot already shows links between regions of homology as connected polygons. We also want to draw links between genes in the the same cluster but as single lines. However, the default behaviour of \u003cem\u003egeom_link()\u003c/em\u003e is to draw links as polygons with the same thickness of the gene, which can look cluttered. To instead draw links as single lines, we can define a \u003cem\u003egeom_link_line()\u003c/em\u003e function using base \u003cem\u003eggplot::geom_segment\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"geom_link_line \u0026lt;- function(mapping = NULL, data = links(), stat = \u0026quot;identity\u0026quot;,\n          position = \u0026quot;identity\u0026quot;, na.rm = FALSE, show.legend = NA, inherit.aes = TRUE,\n          ...){\n  default_aes \u0026lt;- aes(y=y, yend=yend, x=(x+xend)/2, xend=(xmin+xmax)/2)\n  mapping \u0026lt;- gggenomes:::aes_intersect(mapping, default_aes)\n\n  layer(geom = GeomSegment, mapping = mapping, data = data, stat = stat, \n        position = position, show.legend = show.legend, inherit.aes = inherit.aes, \n        params = list(na.rm = na.rm, ...))\n}\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egeom_link_line \u0026lt;- function(mapping = NULL, data = links(), stat = \"identity\",\n          position = \"identity\", na.rm = FALSE, show.legend = NA, inherit.aes = TRUE,\n          ...){\n  default_aes \u0026lt;- aes(y=y, yend=yend, x=(x+xend)/2, xend=(xmin+xmax)/2)\n  mapping \u0026lt;- gggenomes:::aes_intersect(mapping, default_aes)\n\n  layer(geom = GeomSegment, mapping = mapping, data = data, stat = stat, \n        position = position, show.legend = show.legend, inherit.aes = inherit.aes, \n        params = list(na.rm = na.rm, ...))\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe can now include this function when adding features to the plot\u003cbr\u003e\n\u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e14. add features and custom aesthetics\u003c/strong\u003e\u003cbr\u003e\nwe want to colour only secreted genes according to orthgroup (OG) cluster so first setup a colour palette to use. From \u003ca href=\"https://r-graph-gallery.com/38-rcolorbrewers-palettes.html\" rel=\"nofollow\"\u003e\u003cem\u003eRColorBrewer\u003c/em\u003e\u003c/a\u003e we'll use \u003cem\u003eSet1\u003c/em\u003e, which has 9 discrete colours but we can expand it with a colour ramp as \u003cem\u003emy_clusts_sec\u003c/em\u003e has 17 OGs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"library(RColorBrewer)\ngetPalette = colorRampPalette(brewer.pal(8, \u0026quot;Set1\u0026quot;))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003elibrary(RColorBrewer)\ngetPalette = colorRampPalette(brewer.pal(8, \"Set1\"))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eonly using the first 8 colours of \u003cem\u003eSet1\u003c/em\u003e since the 9th colour is grey, which we want for \u003cem\u003ena.values\u003c/em\u003e, i.e. all other genes not in \u003cem\u003emy_clusts_sec\u003c/em\u003e\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003enow we will put everything together with previous \u003cem\u003esynplot2_flip_focus\u003c/em\u003e in new plot \u003cem\u003esynplot3\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"synplot3 \u0026lt;- synplot2_flip_focus %\u0026gt;% add_clusters(my_clusts_sec) +\ngeom_feat(position=position_strand(), colour=\u0026quot;black\u0026quot;) + \ngeom_gene(aes(fill=cluster_id, colour=cluster_id), size=7) + \ngeom_gene_tag(aes(label=parent_ids), vjust = -1, hjust = -0.15, size = 2) + \ngeom_link_line(data=links(.track_id=2), color=\u0026quot;black\u0026quot;, linetype=2) +\nscale_fill_manual(\u0026quot;cluster\u0026quot;, values=getPalette(17), na.value=\u0026quot;darkgrey\u0026quot;) + \nscale_color_manual(\u0026quot;cluster\u0026quot;, values=getPalette(17), na.value=\u0026quot;darkgrey\u0026quot;)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esynplot3 \u0026lt;- synplot2_flip_focus %\u0026gt;% add_clusters(my_clusts_sec) +\ngeom_feat(position=position_strand(), colour=\"black\") + \ngeom_gene(aes(fill=cluster_id, colour=cluster_id), size=7) + \ngeom_gene_tag(aes(label=parent_ids), vjust = -1, hjust = -0.15, size = 2) + \ngeom_link_line(data=links(.track_id=2), color=\"black\", linetype=2) +\nscale_fill_manual(\"cluster\", values=getPalette(17), na.value=\"darkgrey\") + \nscale_color_manual(\"cluster\", values=getPalette(17), na.value=\"darkgrey\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003egeom_feat\u003c/em\u003e, these are the repeat regions positioned by +/- strand and coloured black\u003cbr\u003e\n\u003cem\u003egeom_gene\u003c/em\u003e, these are the genes with border and fill according to \u003cem\u003ecluster_id\u003c/em\u003e from \u003cem\u003emy_clusts_sec\u003c/em\u003e \u003cbr\u003e\n\u003cem\u003egeom_gene_tag\u003c/em\u003e, gene labels from \u003cem\u003emy_seqs\u003c/em\u003e using \u003cem\u003eparent_id\u003c/em\u003e attribute so trancript number isn't shown (see step 6)\u003cbr\u003e\n\u003cem\u003egeom_link_line\u003c/em\u003e, our custom function (step 13) to link genes with lines instead of polygons\u003cbr\u003e\n\u003cem\u003escale_fill_manual\u003c/em\u003e, assign fill colour to clusters\u003cbr\u003e\n\u003cem\u003escale_color_manual\u003c/em\u003e, assign border colour to clusters (matching fill)\u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e15. shift bins for optimal display and save to final plot \u003cem\u003esynplot4\u003c/em\u003e\u003c/strong\u003e \u003cbr\u003e\nby default, the seq bins are left-justified but this can make the links look stretched; we can automatically centre the bins if desired\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"synplot4 \u0026lt;- synplot3 %\u0026gt;% shift(center=TRUE)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esynplot4 \u0026lt;- synplot3 %\u0026gt;% shift(center=TRUE)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ealternatively, it is easy to manually shift bins horizontally (this may take trial-and-error but plot scale bar can be used as an aid)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"synplot4 \u0026lt;- synplot3 %\u0026gt;% shift(bins=c(1,2,4,6), by=c(50000, 50000, 25000, 50000))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esynplot4 \u0026lt;- synplot3 %\u0026gt;% shift(bins=c(1,2,4,6), by=c(50000, 50000, 25000, 50000))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://private-user-images.githubusercontent.com/33470968/269471307-a8bb91a2-ff44-4775-8a22-a59f2867fcef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzEzMDctYThiYjkxYTItZmY0NC00Nzc1LThhMjItYTU5ZjI4NjdmY2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViYmM4NzhmYjE3MTgxN2Y1NWY1NDhhZDE0YWQ0OWM0OWQ0MTA0Y2EzMDJmYzg1YzI4OTY5ODNhNTE4YjhjNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DF5jF98Sf6zdbd4uAMEHRqboMvusyc7BAwVmQPFBfrA\"\u003e\u003cimg src=\"https://private-user-images.githubusercontent.com/33470968/269471307-a8bb91a2-ff44-4775-8a22-a59f2867fcef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzEzMDctYThiYjkxYTItZmY0NC00Nzc1LThhMjItYTU5ZjI4NjdmY2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViYmM4NzhmYjE3MTgxN2Y1NWY1NDhhZDE0YWQ0OWM0OWQ0MTA0Y2EzMDJmYzg1YzI4OTY5ODNhNTE4YjhjNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DF5jF98Sf6zdbd4uAMEHRqboMvusyc7BAwVmQPFBfrA\" alt=\"synplot_exp5_small\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003enow we are happy with our synteny plot, we can save it as an SVG by invoking \u003cem\u003esvglite\u003c/em\u003e within \u003cem\u003eggsave\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"ggsave(file=\u0026quot;MTAR_6_synteny_plot.svg\u0026quot;, device=svg, plot=synplot4, width=12, height=6)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eggsave(file=\"MTAR_6_synteny_plot.svg\", device=svg, plot=synplot4, width=12, height=6)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ein the above plot, only gene labels for the first bin are shown for simplicity, coordinates were manually added to the labels for \u003cem\u003echr6_A\u003c/em\u003e, \u003cem\u003echr6_B\u003c/em\u003e and \u003cem\u003e020F\u003c/em\u003e, some superfluous minor homolgy links were removed for cleaner visuals, and scale bar was reduced to show just one increment. The benefit of saving as SVG is that small adjustments and fixes like these are easy to do in a free SVG editor like \u003ca href=\"https://inkscape.org/\" rel=\"nofollow\"\u003eInkscape\u003c/a\u003e \u003cbr\u003e\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cbr\u003e\n\u003cp dir=\"auto\"\u003emore plotting capabilities can be found in the \u003cem\u003egggenomes\u003c/em\u003e \u003ca href=\"https://thackl.github.io/gggenomes/articles/emales.html\" rel=\"nofollow\"\u003etutorial\u003c/a\u003e pages by \u003ca href=\"https://github.com/thackl\"\u003eThomas Hackl\u003c/a\u003e.\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"OatCrownRust","anchor":"oatcrownrust","htmlText":"OatCrownRust"},{"level":2,"text":"Read mapping statistics and coverage plots","anchor":"read-mapping-statistics-and-coverage-plots","htmlText":"Read mapping statistics and coverage plots"},{"level":2,"text":"Assessing variation with population SNP counts","anchor":"assessing-variation-with-population-snp-counts","htmlText":"Assessing variation with population SNP counts"},{"level":2,"text":"Generating chromosome maps with karyoploteR","anchor":"generating-chromosome-maps-with-karyoploter","htmlText":"Generating chromosome maps with karyoploteR"},{"level":2,"text":"Generating synteny plots with gggenomes","anchor":"generating-synteny-plots-with-gggenomes","htmlText":"Generating synteny plots with gggenomes"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FTC-Hewitt%2FOatCrownRust"}}],"overviewFilesProcessingTime":44.632646}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/TC-Hewitt/OatCrownRust/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/TC-Hewitt/OatCrownRust/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/TC-Hewitt/OatCrownRust/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/TC-Hewitt/OatCrownRust/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/TC-Hewitt/OatCrownRust/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           33 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/TC-Hewitt/OatCrownRust/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Orthogroups_txtConvert.py, (File)" class="Link--primary" href="/TC-Hewitt/OatCrownRust/blob/main/Orthogroups_txtConvert.py" title="Orthogroups_txtConvert.py">
                          Orthogroups_txtConvert.py
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Orthogroups_txtConvert.py, (File)" class="Link--primary" href="/TC-Hewitt/OatCrownRust/blob/main/Orthogroups_txtConvert.py" title="Orthogroups_txtConvert.py">
                          Orthogroups_txtConvert.py
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/TC-Hewitt/OatCrownRust/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/TC-Hewitt/OatCrownRust/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      OatCrownRust
                     </h1>
                     <a aria-label="Permalink: OatCrownRust" class="anchor" href="#oatcrownrust" id="user-content-oatcrownrust">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Read mapping statistics and coverage plots
                     </h2>
                     <a aria-label="Permalink: Read mapping statistics and coverage plots" class="anchor" href="#read-mapping-statistics-and-coverage-plots" id="user-content-read-mapping-statistics-and-coverage-plots">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     After FASTQ files are mapped to a reference FASTA with an aligner such as
                     <a href="https://github.com/lh3/bwa">
                      BWA
                     </a>
                     or
                     <a href="https://github.com/DaehwanKimLab/hisat2">
                      HISAT2
                     </a>
                     , simple count and coverage statistics are obtained from the subsequent SAM/BAM files with the following scripts using
                     <a href="https://github.com/samtools/samtools">
                      SAMtools
                     </a>
                     output piped into AWK operations
                    </p>
                    <p dir="auto">
                     <strong>
                      1. for a single BAM, count total number of mapped reads and store in var
                      <em>
                       map_count
                      </em>
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="module load samtools/1.12 #your system might have different environment tool command
map_count=$(samtools view -c -F 260 sample01.bam)">
                     <pre class="notranslate"><code>module load samtools/1.12 #your system might have different environment tool command
map_count=$(samtools view -c -F 260 sample01.bam)
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -c
                      </em>
                      , counts total n alignments instead of printing each alignment
                      <br/>
                      <em>
                       -F 260
                      </em>
                      , exclude unmapped reads or non-primary alignments ("260" set in
                      <em>
                       FLAG
                      </em>
                      field)
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      2. get read depth at each ref position in BAM and pipe to AWK to generate coverage stats
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="samtools depth -a sample01.bam | awk -v OFS='\t' -v sname=&quot;sample01&quot; -v nmap=&quot;$map_count&quot; \
'{sum+=$3} ; $3&gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' &gt; sample01.covstats.out">
                     <pre class="notranslate"><code>samtools depth -a sample01.bam | awk -v OFS='\t' -v sname="sample01" -v nmap="$map_count" \
'{sum+=$3} ; $3&gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' &gt; sample01.covstats.out
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -a
                      </em>
                      , output all positions (including those with zero depth)
                      <br/>
                      <em>
                       -v OFS='\t'
                      </em>
                      , sets awk output to be tab-delimited
                      <br/>
                      <em>
                       -v sname="sample01"
                      </em>
                      , sets awk var for sample name
                      <br/>
                      <em>
                       -v nmap="$map_count"
                      </em>
                      , sets awk var for
                      <em>
                       map_count
                      </em>
                      determined in previous step
                      <br/>
                      <em>
                       {sum+=$3}
                      </em>
                      , cosecutively add depth value in 3rd field to
                      <em>
                       sum
                      </em>
                      var
                      <br/>
                      <em>
                       $3&gt;10{c++}
                      </em>
                      , adds 1 count to var
                      <em>
                       c
                      </em>
                      if depth value in 3rd field &gt; 10 (i.e. 10X coverage)
                      <br/>
                      <em>
                       END{print ...}
                      </em>
                      , after final line of
                      <em>
                       samtools depth
                      </em>
                      output reached, writes the following to file
                      <em>
                       sample01.covstats.out
                      </em>
                      :
                      <br/>
                      sample name (
                      <em>
                       sname
                      </em>
                      )
                      <br/>
                      total mapped reads (
                      <em>
                       nmap
                      </em>
                      )
                      <br/>
                      mean coverage (
                      <em>
                       sum/NR
                      </em>
                      , where
                      <em>
                       NR
                      </em>
                      =total n of rows processed =n ref positions)
                      <br/>
                      total n positions with coverage over 10X (
                      <em>
                       c+0
                      </em>
                      )
                      <br/>
                      percent of all positions with coverage over 10X (
                      <em>
                       c*100/NR
                      </em>
                      )
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     multiple BAM files can be processed in parrallel to save time. For example, below is an array job script for 100 samples using the
                     <a href="https://github.com/SchedMD/slurm">
                      Slurm Workload Manager
                     </a>
                     on a shared cluster
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/env bash

#SBATCH --job-name=covstats
#SBATCH --time=2:00:00
#SBATCH --nodes=1
#SBATCH --mem=2GB
#SBATCH --array=0-99

SAMPLES=( $(cat samplenames.txt) ) #file with 1 sample name per line (sans file exts)

module load samtools/1.12

if [ ! -z &quot;$SLURM_ARRAY_TASK_ID&quot; ]
then
    i=$SLURM_ARRAY_TASK_ID
    map_count=$(samtools view -c -F 260 ${SAMPLES[$i]}.bam)
    samtools depth -a ${SAMPLES[$i]}.bam | awk -v OFS='\t' -v sname=${SAMPLES[$i]} -v nmap=&quot;$map_count&quot; \
    '{sum+=$3} ; $3&gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' &gt; temp${i}
else
    echo &quot;Error: Missing array index as SLURM_ARRAY_TASK_ID&quot;
fi">
                     <pre class="notranslate"><code>#!/usr/bin/env bash

#SBATCH --job-name=covstats
#SBATCH --time=2:00:00
#SBATCH --nodes=1
#SBATCH --mem=2GB
#SBATCH --array=0-99

SAMPLES=( $(cat samplenames.txt) ) #file with 1 sample name per line (sans file exts)

module load samtools/1.12

if [ ! -z "$SLURM_ARRAY_TASK_ID" ]
then
    i=$SLURM_ARRAY_TASK_ID
    map_count=$(samtools view -c -F 260 ${SAMPLES[$i]}.bam)
    samtools depth -a ${SAMPLES[$i]}.bam | awk -v OFS='\t' -v sname=${SAMPLES[$i]} -v nmap="$map_count" \
    '{sum+=$3} ; $3&gt;10{c++} END{print sname, nmap, sum/NR, c+0, c*100/NR}' &gt; temp${i}
else
    echo "Error: Missing array index as SLURM_ARRAY_TASK_ID"
fi
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      the above can be wrapped in file
                      <em>
                       covstats.sh
                      </em>
                      and run as
                      <em>
                       &gt;sbatch covstats.sh
                      </em>
                      <br/>
                      the output line of each sample is written to seperate temp files assigned
                      <em>
                       temp0
                      </em>
                      to
                      <em>
                       temp99
                      </em>
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      3. the output stats from each sample can now be collated into a single tab-delim table by creating a header and concatenating with all
                      <em>
                       temp
                      </em>
                      files
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='echo -e "sample\tmapped_reads\tmean_cov\tbp_over_10X\tpcnt_ref_over_10X" &gt; header
cat header temp{0..99} &gt; combined.covstats.txt
rm header temp{0..99}'>
                     <pre class="notranslate"><code>echo -e "sample\tmapped_reads\tmean_cov\tbp_over_10X\tpcnt_ref_over_10X" &gt; header
cat header temp{0..99} &gt; combined.covstats.txt
rm header temp{0..99}
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      header names describe the output fields
                      <br/>
                      final matrix is written to
                      <em>
                       combined.covstats.txt
                      </em>
                      <br/>
                      temporary
                      <em>
                       header
                      </em>
                      and
                      <em>
                       temp
                      </em>
                      files are deleted
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      4. various stats can then be plotted from the table using a simple R script like below
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/env Rscript
infile = commandArgs(trailingOnly=TRUE)
stats &lt;- read.table(infile, header = T, sep = &quot;\t&quot;)
outname &lt;- strsplit(infile, '.txt') 
png(file=paste0(outname, &quot;_plots.png&quot;), width=2000, height=600)
par(mfrow=c(2,1), mar=c(6,4,4,4))
barplot(stats$mean_cov, names=stats$sample, main=&quot;mean cov&quot;, ylim=c(0,100), las=2)
barplot(stats$pcnt_ref_over_10X, names=stats$sample, main=&quot;perc bp over 10X&quot;, ylim=c(0,100), las=2)
dev.off()">
                     <pre class="notranslate"><code>#!/usr/bin/env Rscript
infile = commandArgs(trailingOnly=TRUE)
stats &lt;- read.table(infile, header = T, sep = "\t")
outname &lt;- strsplit(infile, '.txt') 
png(file=paste0(outname, "_plots.png"), width=2000, height=600)
par(mfrow=c(2,1), mar=c(6,4,4,4))
barplot(stats$mean_cov, names=stats$sample, main="mean cov", ylim=c(0,100), las=2)
barplot(stats$pcnt_ref_over_10X, names=stats$sample, main="perc bp over 10X", ylim=c(0,100), las=2)
dev.off()
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      the above can be wrapped in file
                      <em>
                       plot_covstats.R
                      </em>
                      and run as
                      <em>
                       &gt;Rscript plot_covstats.R combined.covstats.txt
                      </em>
                      <br/>
                      this creates two basic barplots, one for mean coverage and another for percent of ref bases over 10X coverage, in a single PNG image
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Assessing variation with population SNP counts
                     </h2>
                     <a aria-label="Permalink: Assessing variation with population SNP counts" class="anchor" href="#assessing-variation-with-population-snp-counts" id="user-content-assessing-variation-with-population-snp-counts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Here we want to obtain the number of variable SNP sites (VSS) in a populations of crown rust fungi. These are genomic positions at which SNP genotypes are heterogeneous within a population (i.e. not all individuals possess the same genotype) and can therefore be used as a relative measure of overall genotypic diversity when comparing populations (provided each population is called against the same reference genome). Variant calling can be performed independently for each population (represented by collections of fungal isolates) using the same reference with appropriate filters applied.
                    </p>
                    <p dir="auto">
                     In this example, we are comparing a population of North American (US) and South African (SA) isolates. However, variant calling was performed simultaneously on all isolates using
                     <a href="https://github.com/freebayes/freebayes">
                      FreeBayes
                     </a>
                     to produce a single, multisample VCF containing variant data from 20 US isolates and 29 SA isolates. Extra steps are required to subset the US and SA populations and apply additional filtering before both the total SNP sites (TSS) and VSS of each population are counted.
                    </p>
                    <p dir="auto">
                     <strong>
                      1. initial filtering
                     </strong>
                     - the raw VCF file
                     <em>
                      cr_collection.vcf
                     </em>
                     is 1st filtered on quality using
                     <a href="https://github.com/vcflib/vcflib">
                      vcffilter
                     </a>
                     with various standard params
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='module load vcflib/1.0.1
vcffilter -f "QUAL &gt; 20 &amp; QUAL / AO &gt; 10 &amp; SAF &gt; 0 &amp; SAR &gt; 0 &amp; RPR &gt; 1 &amp; RPL &gt; 1 &amp; AC &gt; 0 &amp; DP &gt; 500" \
cr_collection.vcf &gt; cr_collection.QA.vcf'>
                     <pre class="notranslate"><code>module load vcflib/1.0.1
vcffilter -f "QUAL &gt; 20 &amp; QUAL / AO &gt; 10 &amp; SAF &gt; 0 &amp; SAR &gt; 0 &amp; RPR &gt; 1 &amp; RPL &gt; 1 &amp; AC &gt; 0 &amp; DP &gt; 500" \
cr_collection.vcf &gt; cr_collection.QA.vcf
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -f
                      </em>
                      , specifies a filter to apply to the info fields of records, removes sites which do not pass the filter
                      <br/>
                      <em>
                       QUAL &gt; 20
                      </em>
                      , requires quality score no less than 20
                      <br/>
                      <em>
                       QUAL / AO &gt; 10
                      </em>
                      , requires additional contribution of each observation at least 10 log units (~Q10 per read)
                      <br/>
                      <em>
                       SAF &gt; 0 &amp; SAR &gt; 0
                      </em>
                      , requires reads present on both strands
                      <br/>
                      <em>
                       RPR &gt; 1 &amp; RPL &gt; 1
                      </em>
                      , requires at least two reads “balanced” to each side of the site
                      <br/>
                      <em>
                       AC &gt; 0
                      </em>
                      , requires total number of alt alleles in called genotypes more than 0
                      <br/>
                      <em>
                       DP &gt; 500
                      </em>
                      , requires greater than 500 total read depth (with 49 samples this means a min ~10X per sample on average)
                      <br/>
                      <em>
                       &gt; cr_collection.QA.vcf
                      </em>
                      , write filtered records to new "Quality Assured" file
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      2. feature filtering
                     </strong>
                     - we are only interested in biallelic SNPs likely to be informative, so we will apply additional filtering to the VCF using
                     <a href="https://github.com/samtools/bcftools">
                      BCFtools
                     </a>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="module load bcftools/1.15.1
bcftools view -m2 -M2 -v snps -i 'F_MISSING &lt; 0.1' -e 'MAF &lt; 0.05' \
--output-type z -o cr_collection.QA.biaSNPs.vcf.gz cr_collection.QA.vcf">
                     <pre class="notranslate"><code>module load bcftools/1.15.1
bcftools view -m2 -M2 -v snps -i 'F_MISSING &lt; 0.1' -e 'MAF &lt; 0.05' \
--output-type z -o cr_collection.QA.biaSNPs.vcf.gz cr_collection.QA.vcf
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -m2 -M2
                      </em>
                      , keep sites with min 2 alleles and max 2 alleles
                      <br/>
                      <em>
                       -v snps
                      </em>
                      , keep SNPs only (omits indels and MNPs)
                      <br/>
                      <em>
                       -i 'F_MISSING &lt; 0.1'
                      </em>
                      , include only sites wih less than 10% missing data
                      <br/>
                      <em>
                       -e 'MAF &lt; 0.05'
                      </em>
                      , exclude sites with a minor allele frequency (MAF) of less than 5% (note: skip MAF filtering if you dont want to discard potential rare/novel genotypes)
                      <br/>
                      <em>
                       --output-type z
                      </em>
                      , output gz compressed vcf
                      <br/>
                      <em>
                       -o cr_collection.QA.biaSNPs.vcf.gz
                      </em>
                      , output file name
                      <br/>
                      <em>
                       cr_collection.QA.vcf
                      </em>
                      , input file name
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      3. subset populations
                     </strong>
                     - we will now subset the US and SA populations from this VCF based on sample names
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bcftools view -S samplenames_US.txt cr_collection.QA.biaSNPs.vcf.gz &gt; cr_pop_US.vcf
bcftools view -S samplenames_SA.txt cr_collection.QA.biaSNPs.vcf.gz &gt; cr_pop_SA.vcf">
                     <pre class="notranslate"><code>bcftools view -S samplenames_US.txt cr_collection.QA.biaSNPs.vcf.gz &gt; cr_pop_US.vcf
bcftools view -S samplenames_SA.txt cr_collection.QA.biaSNPs.vcf.gz &gt; cr_pop_SA.vcf
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -S samplenames_US.txt
                      </em>
                      , subset US pop using sample names listed in samplenames_US.txt (1 per line)
                      <br/>
                      <em>
                       -S samplenames_SA.txt
                      </em>
                      , subset SA pop using sample names listed in samplenames_SA.txt
                      <br/>
                     </p>
                    </blockquote>
                    <p dir="auto">
                     Names must match exactly as present in the VCF file. In case they are not known, it is possible to retrieve them from the VCF with a bash script
                     <br/>
                     <code>
                      zcat cr_collection.QA.biaSNPs.vcf.gz | egrep "^#" | tail -1
                     </code>
                    </p>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       zcat
                      </em>
                      will print gz compressed files, otherwise use
                      <em>
                       cat
                      </em>
                      if VCF uncompressed
                      <br/>
                      <em>
                       egrep
                      </em>
                      will find only header lines (starts with "#")
                      <br/>
                      <em>
                       tail -1
                      </em>
                      will print out the final header line which should contain the names of the first 8 or 9 default fields followed by the sample fields
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      4. get TSS counts
                     </strong>
                     - we now have seperate US and SA files to get their total SNP count. However, since they were subsetted from a single VCF file, each may contain genotypes that are homozygous reference (ref/ref, RR) in every sample. These can be excluded from the final count
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bcftools view -i 'GT[*]=&quot;alt&quot;' cr_pop_US.vcf &gt; cr_pop_US.noRR.vcf
bcftools view -i 'GT[*]=&quot;alt&quot;' cr_pop_SA.vcf &gt; cr_pop_SA.noRR.vcf">
                     <pre class="notranslate"><code>bcftools view -i 'GT[*]="alt"' cr_pop_US.vcf &gt; cr_pop_US.noRR.vcf
bcftools view -i 'GT[*]="alt"' cr_pop_SA.vcf &gt; cr_pop_SA.noRR.vcf
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -i ...
                      </em>
                      , this includes only sites where at least one sample has an alt allele, written to a new VCF
                     </p>
                    </blockquote>
                    <p dir="auto">
                     TSS and other stats can now be obtained from these VCFs, contained in
                     <em>
                      bcftools stats
                     </em>
                     output
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bcftools stats cr_pop_US.noRR.vcf &gt; cr_pop_US.noRR.stats.out
bcftools stats cr_pop_US.noRR.vcf &gt; cr_pop_US.noRR.stats.out">
                     <pre class="notranslate"><code>bcftools stats cr_pop_US.noRR.vcf &gt; cr_pop_US.noRR.stats.out
bcftools stats cr_pop_US.noRR.vcf &gt; cr_pop_US.noRR.stats.out
</code></pre>
                    </div>
                    <p dir="auto">
                     alternatively, TSS can be counted directly without writing new files
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bcftools view -i 'GT[*]=&quot;alt&quot;' cr_pop_US.vcf | egrep -v &quot;^#&quot; | wc -l
# TSS for US pop =756,980

bcftools view -i 'GT[*]=&quot;alt&quot;' cr_pop_US.vcf | egrep -v &quot;^#&quot; | wc -l
# TSS for SA pop =548,091">
                     <pre class="notranslate"><code>bcftools view -i 'GT[*]="alt"' cr_pop_US.vcf | egrep -v "^#" | wc -l
# TSS for US pop =756,980

bcftools view -i 'GT[*]="alt"' cr_pop_US.vcf | egrep -v "^#" | wc -l
# TSS for SA pop =548,091
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      RR sites are filtered out as before
                      <br/>
                      <em>
                       egrep -v
                      </em>
                      , exclude lines that start with "#" (header lines)
                      <br/>
                      <em>
                       wc -l
                      </em>
                      , count and print number of output lines (= total n SNP records)
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      5. get VSS counts
                     </strong>
                     - different filter params can be used to obtain the number of variable SNP sites. As before, we can write to new files and then run
                     <em>
                      bcftools stats
                     </em>
                     , but here we will print the counts directly like above
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="nUS=20
nSA=29

bcftools view -e &quot;COUNT(GT='RR')=$nUS | COUNT(GT='RA')=$nUS | COUNT(GT='AA')=$nUS | COUNT(GT='Aa')=$nUS&quot; \
cr_pop_US.vcf | egrep -v &quot;^#&quot; | wc -l
# VSS for US pop =652,504

bcftools view -e &quot;COUNT(GT='RR')=$nSA | COUNT(GT='RA')=$nSA | COUNT(GT='AA')=$nSA | COUNT(GT='Aa')=$nSA&quot; \
cr_pop_SA.vcf | egrep -v &quot;^#&quot; | wc -l
# VSS for SA pop =47,369">
                     <pre class="notranslate"><code>nUS=20
nSA=29

bcftools view -e "COUNT(GT='RR')=$nUS | COUNT(GT='RA')=$nUS | COUNT(GT='AA')=$nUS | COUNT(GT='Aa')=$nUS" \
cr_pop_US.vcf | egrep -v "^#" | wc -l
# VSS for US pop =652,504

bcftools view -e "COUNT(GT='RR')=$nSA | COUNT(GT='RA')=$nSA | COUNT(GT='AA')=$nSA | COUNT(GT='Aa')=$nSA" \
cr_pop_SA.vcf | egrep -v "^#" | wc -l
# VSS for SA pop =47,369
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      exact sample sizes of the US and SA populations (as present in VCF subsets) are first assigned to to vars
                      <em>
                       nUS
                      </em>
                      and
                      <em>
                       nSA
                      </em>
                      <br/>
                      <em>
                       -e ...
                      </em>
                      , this excludes sites where a count condition is equal to the sample size (
                      <em>
                       $nUS
                      </em>
                      or
                      <em>
                       $nSA
                      </em>
                      ), i.e. identical genotype (RR=ref/ref hom, RA=ref/alt het, AA=alt/alt hom, Aa=alt/alt het) in every sample. Thus, what remains are only the variable sites
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     To save space, uncompresed leftover VCF files can be bgzipped with
                     <em>
                      samtools
                     </em>
                     then indexed with
                     <em>
                      bcftools
                     </em>
                     <br/>
                     <code>
                      bgzip -@ 4 file.vcf &amp;&amp; bcftools index file.vcf.gz
                     </code>
                    </p>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -@ 4
                      </em>
                      , num threads to use to perform compression
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     For context, the SA population is entirely asexual, lacking any recombination, whereas the US population reproduces both sexually and asexually. Therefore, the US population is expected to be more diverse. This is highly evident in the VSS count, which is ~14 times higher than that of the SA population. TSS counts only indicate SNPs against the reference and are heavily influenced by reference choice. Subtracting the VSS from the TSS in the SA population (548,091-47,369) gives ~500K genotypes that were likely already present at the founding of this population, and diversity within this population is represented by only ~47K SNP sites.
                     <br/>
                     <br/>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Generating chromosome maps with
                      <a href="https://github.com/bernatgel/karyoploteR">
                       karyoploteR
                      </a>
                     </h2>
                     <a aria-label="Permalink: Generating chromosome maps with karyoploteR" class="anchor" href="#generating-chromosome-maps-with-karyoploter" id="user-content-generating-chromosome-maps-with-karyoploter">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     These are instructions to generate simple chromosome maps annotating regions of interest, similar to what appears in the paper figure. More detailed maps incorporating features such as density plots and coverage graphs are covered in the
                     <a href="https://bernatgel.github.io/karyoploter_tutorial/" rel="nofollow">
                      karyoploteR tutorial
                     </a>
                     . Here we will make basic plots based on a diploid chromosome (chr) set where A and B haplotypes are separated.
                    </p>
                    <p dir="auto">
                     <strong>
                      1. Input for chr lengths
                     </strong>
                     - this is simply a tab delim table of chr names, start and end coords which can be obtaind from the
                     <em>
                      .fasta.fai
                     </em>
                     index of the genome FASTA file
                     <br/>
                    </p>
                    <p dir="auto">
                     <code>
                      samtools faidx mygenome.fasta
                     </code>
                    </p>
                    <p dir="auto">
                     the following bash scripts can be used to create a regions table from
                     <em>
                      mygenome.fasta.fai
                     </em>
                     compatible with
                     <a href="https://bioconductor.org/packages/3.18/bioc/html/GenomicRanges.html" rel="nofollow">
                      GRanges
                     </a>
                     formatting
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="printf &quot;chr\tstart\tend\n&quot; &gt; mygenome.granges.txt
TAB='\t'
cut -f1,2 mygenome.fasta.fai | sed &quot;s/${TAB}/${TAB}0${TAB}/g&quot; &gt;&gt; mygenome.granges.txt">
                     <pre class="notranslate"><code>printf "chr\tstart\tend\n" &gt; mygenome.granges.txt
TAB='\t'
cut -f1,2 mygenome.fasta.fai | sed "s/${TAB}/${TAB}0${TAB}/g" &gt;&gt; mygenome.granges.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     this will look like the following in this example, where haplotypes are inidcated by _
                     <em>
                      A
                     </em>
                     and _
                     <em>
                      B
                     </em>
                     suffixes.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="chr start   end
chr1_A  0   8018478
chr2_A  0   6810466
chr3_A  0   6979198
chr4_A  0   7574082
chr5_A  0   6841886
chr6_A  0   5186056
chr1_B  0   7419164
chr2_B  0   6665088
chr3_B  0   6167249
chr4_B  0   8040728
chr5_B  0   7117848
chr6_B  0   5207541">
                     <pre class="notranslate"><code>chr start   end
chr1_A  0   8018478
chr2_A  0   6810466
chr3_A  0   6979198
chr4_A  0   7574082
chr5_A  0   6841886
chr6_A  0   5186056
chr1_B  0   7419164
chr2_B  0   6665088
chr3_B  0   6167249
chr4_B  0   8040728
chr5_B  0   7117848
chr6_B  0   5207541
</code></pre>
                    </div>
                    <br/>
                    <p dir="auto">
                     <strong>
                      2. Input for regions of interest
                     </strong>
                     - similarly, we will have a table
                     <em>
                      loci.granges.txt
                     </em>
                     with start and end coords for specific loci, with extra fields for locus name and chosen colours as HEX values
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="chr start   end name    colour
chr1_A  2140000 2200000 loc1A '#61d04f'
chr1_B  1980000 2020000 loc1B '#61d04f'
chr1_A  4010000 4030000 loc2A '#2297e6'
chr1_B  3760000 3770000 loc2B '#2297e6'
chr2_A  4590000 4910000 loc3A '#542a18'
chr2_B  4680000 4990000 loc3B '#542a18'
chr6_A  480000  700000  loc4A '#cd0bbc'
chr6_B  570000  770000  loc4B '#cd0bbc'
chr6_A  2740000 2840000 loc5A '#ffa500'
chr6_B  2860000 2930000 loc5B '#ffa500'">
                     <pre class="notranslate"><code>chr start   end name    colour
chr1_A  2140000 2200000 loc1A '#61d04f'
chr1_B  1980000 2020000 loc1B '#61d04f'
chr1_A  4010000 4030000 loc2A '#2297e6'
chr1_B  3760000 3770000 loc2B '#2297e6'
chr2_A  4590000 4910000 loc3A '#542a18'
chr2_B  4680000 4990000 loc3B '#542a18'
chr6_A  480000  700000  loc4A '#cd0bbc'
chr6_B  570000  770000  loc4B '#cd0bbc'
chr6_A  2740000 2840000 loc5A '#ffa500'
chr6_B  2860000 2930000 loc5B '#ffa500'
</code></pre>
                    </div>
                    <br/>
                    <p dir="auto">
                     <strong>
                      3. Input for text annotation
                     </strong>
                     - we also need a table
                     <em>
                      loci.labels.txt
                     </em>
                     indicating the position and label for our loci. The positions can be anywhere, but here we are just using the start coordinate of each locus with the colour values matching those of the regions
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="chr pos label   colour
chr1_A  2140000 loc1A    '#61d04f'
chr1_B  1980000 loc1B    '#61d04f'
chr1_A  4010000 loc2A    '#2297e6'
chr1_B  3760000 loc2B    '#2297e6'
chr2_A  4590000 loc3A    '#542a18'
chr2_B  4680000 loc3B    '#542a18'
chr6_A  480000  loc4A    '#cd0bbc'
chr6_B  570000  loc4B    '#cd0bbc'
chr6_A  2740000 loc5A    '#ffa500'
chr6_B  2860000 loc5B    '#ffa500'">
                     <pre class="notranslate"><code>chr pos label   colour
chr1_A  2140000 loc1A    '#61d04f'
chr1_B  1980000 loc1B    '#61d04f'
chr1_A  4010000 loc2A    '#2297e6'
chr1_B  3760000 loc2B    '#2297e6'
chr2_A  4590000 loc3A    '#542a18'
chr2_B  4680000 loc3B    '#542a18'
chr6_A  480000  loc4A    '#cd0bbc'
chr6_B  570000  loc4B    '#cd0bbc'
chr6_A  2740000 loc5A    '#ffa500'
chr6_B  2860000 loc5B    '#ffa500'
</code></pre>
                    </div>
                    <br/>
                    <p dir="auto">
                     <strong>
                      4. Input for other local features
                     </strong>
                     - a previous analysis was performed with
                     <a href="https://github.com/alachins/sweed">
                      SweeD
                     </a>
                     to identify potential seletive sweep regions. It generated the output
                     <em>
                      selective_sweep.granges.bed
                     </em>
                     which we will also plot
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#chr start   end
chr1_A  112428  172428
chr1_A  890413  950413
chr1_B  3223380 3292289
chr1_B  4567533 4628523
chr2_A  422444  484441
chr2_B  1038753 1098753
chr2_A  1899788 1961785
chr2_B  3122417 3182417
chr2_A  3752711 3812711
chr2_B  4863466 4923466
chr2_A  5693536 5755533
chr2_B  6070114 6169070
chr5_B  217674  277674">
                     <pre class="notranslate"><code>#chr start   end
chr1_A  112428  172428
chr1_A  890413  950413
chr1_B  3223380 3292289
chr1_B  4567533 4628523
chr2_A  422444  484441
chr2_B  1038753 1098753
chr2_A  1899788 1961785
chr2_B  3122417 3182417
chr2_A  3752711 3812711
chr2_B  4863466 4923466
chr2_A  5693536 5755533
chr2_B  6070114 6169070
chr5_B  217674  277674
</code></pre>
                    </div>
                    <br/>
                    <p dir="auto">
                     <strong>
                      5. Prepare inputs in R
                     </strong>
                     - now in R, load the requisite libraries and import the data. We will also specify the different haplotypes by chr name.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(karyoplotR)
library(svglite)

mygenome &lt;- toGRanges("mygenome.granges.txt")
loci &lt;- toGRanges("loci.granges.txt")
sweeps &lt;- toGRanges("selective_sweep.granges.bed")
labels &lt;- read.table("loci.labels.txt", header=TRUE)
hapA &lt;- c("chr1_A", "chr2_A", "chr3_A", "chr4_A", "chr5_A", "chr6_A")
hapB &lt;- c("chr1_B", "chr2_B", "chr3_B", "chr4_B", "chr5_B", "chr6_B")'>
                     <pre class="notranslate"><code>library(karyoplotR)
library(svglite)

mygenome &lt;- toGRanges("mygenome.granges.txt")
loci &lt;- toGRanges("loci.granges.txt")
sweeps &lt;- toGRanges("selective_sweep.granges.bed")
labels &lt;- read.table("loci.labels.txt", header=TRUE)
hapA &lt;- c("chr1_A", "chr2_A", "chr3_A", "chr4_A", "chr5_A", "chr6_A")
hapB &lt;- c("chr1_B", "chr2_B", "chr3_B", "chr4_B", "chr5_B", "chr6_B")
</code></pre>
                    </div>
                    <br/>
                    <p dir="auto">
                     <strong>
                      6. Plot A chrs
                     </strong>
                     - we will create separate plots for the A and B chrs so they can later be arranged side-by-side (without subsetting, all chrs will be plotted as a single stack).
First initialise an output image
                     <em>
                      chrsA.svg
                     </em>
                     using
                     <a href="https://github.com/r-lib/svglite">
                      svglite
                     </a>
                     to save the plot as a scalable vector graphic (SVG) that can be easily resized and modified in a free SVG editor such as
                     <a href="https://inkscape.org/" rel="nofollow">
                      Inkscape
                     </a>
                     , then run the plot script
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="svglite(&quot;chrsA.svg&quot;, width=8, height=6)

kp_hapA &lt;- plotKaryotype(genome=mygenome, chromosomes=hapA)
kpPlotRegions(
    kp_hapA, loci, 
    data.panel=&quot;ideogram&quot;, 
    r0=NULL, r1=NULL, 
    col=loci$colour, 
    border='#000000', 
    avoid.overlapping = FALSE
    )
kpPlotRegions(
    kp_hapA, sweeps, 
    data.panel=1, 
    col=&quot;#c00000&quot;, 
    avoid.overlapping = FALSE, 
    r0=-0.6, r1=-0.4)
kpAddBaseNumbers(
    kp_hapA, 
    tick.dist = 1000001, 
    tick.len = 20, 
    tick.col=&quot;#000000&quot;, 
    cex = 0.01
    )
kpPlotMarkers(
    kp_hapA, 
    chr=labels$chr, 
    x=labels$pos, 
    labels=labels$label, 
    text.orientation = &quot;horizontal&quot;, 
    line.color = &quot;#000000&quot;, 
    label.color = labels$colour, 
    marker.parts = c(0.1, 0.8, 0.1), 
    r1=0.4, cex=0.8, 
    ignore.chromosome.ends=FALSE)

dev.off()">
                     <pre class="notranslate"><code>svglite("chrsA.svg", width=8, height=6)

kp_hapA &lt;- plotKaryotype(genome=mygenome, chromosomes=hapA)
kpPlotRegions(
    kp_hapA, loci, 
    data.panel="ideogram", 
    r0=NULL, r1=NULL, 
    col=loci$colour, 
    border='#000000', 
    avoid.overlapping = FALSE
    )
kpPlotRegions(
    kp_hapA, sweeps, 
    data.panel=1, 
    col="#c00000", 
    avoid.overlapping = FALSE, 
    r0=-0.6, r1=-0.4)
kpAddBaseNumbers(
    kp_hapA, 
    tick.dist = 1000001, 
    tick.len = 20, 
    tick.col="#000000", 
    cex = 0.01
    )
kpPlotMarkers(
    kp_hapA, 
    chr=labels$chr, 
    x=labels$pos, 
    labels=labels$label, 
    text.orientation = "horizontal", 
    line.color = "#000000", 
    label.color = labels$colour, 
    marker.parts = c(0.1, 0.8, 0.1), 
    r1=0.4, cex=0.8, 
    ignore.chromosome.ends=FALSE)

dev.off()
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       kp_hapA &lt;- plotKaryotype(...)
                      </em>
                      , this creates the plot object
                      <em>
                       kp_hapA
                      </em>
                      based on the chr lengths in
                      <em>
                       mygenome
                      </em>
                      and the subset of chrs listed in
                      <em>
                       hapA
                      </em>
                      <br/>
                      <em>
                       kpPlotRegions(kp_hapA, loci, ...)
                      </em>
                      , this draws the loci regions overlayed on the chrs
                      <br/>
                      <em>
                       kpPlotRegions(kp_hapA, sweeps, ...)
                      </em>
                      , this draws the selective sweep regions below the chrs (coloured red)
                      <br/>
                      <em>
                       kpAddBaseNumbers(kp_hapA, ...)
                      </em>
                      , this adds distance ticks along each chr set 1Mb apart
                      <br/>
                      <em>
                       kpPlotMarkers(kp_hapA, chr=labels$chr, x=labels$pos, labels=labels$label, ...)
                      </em>
                      , this adds text labels to each locus specified by names and coordinates in
                      <em>
                       labels
                      </em>
                      <br/>
                     </p>
                    </blockquote>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/267542418-b4bc15d7-0b03-45d6-b128-94a61dfd4e42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0MTgtYjRiYzE1ZDctMGIwMy00NWQ2LWIxMjgtOTRhNjFkZmQ0ZTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwOTllYmU5MDhmYjc0NzY4ZjY3NWRhN2UzNzU3YjVmODg4YjQ2YTZkZGUzMWJhMDBkYmRkNGYxNjI0NDEwOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WVYZrZ6hiw4oOjb4Bt1OSwv8n7qe0zaILjT0kGpudSA" rel="noopener noreferrer" target="_blank">
                      <img alt="chrsA_small" src="https://private-user-images.githubusercontent.com/33470968/267542418-b4bc15d7-0b03-45d6-b128-94a61dfd4e42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0MTgtYjRiYzE1ZDctMGIwMy00NWQ2LWIxMjgtOTRhNjFkZmQ0ZTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwOTllYmU5MDhmYjc0NzY4ZjY3NWRhN2UzNzU3YjVmODg4YjQ2YTZkZGUzMWJhMDBkYmRkNGYxNjI0NDEwOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WVYZrZ6hiw4oOjb4Bt1OSwv8n7qe0zaILjT0kGpudSA" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      7. Plot B chrs
                     </strong>
                     - we will now do the same for the B chrs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="svglite(&quot;chrsB.svg&quot;, width=8, height=6)

kp_hapB &lt;- plotKaryotype(genome=mygenome, chromosomes=hapB)
kpPlotRegions(
    kp_hapB, loci, 
    data.panel=&quot;ideogram&quot;, 
    r0=NULL, r1=NULL, 
    col=loci$colour, 
    border='#000000', 
    avoid.overlapping = FALSE)
kpPlotRegions(
    kp_hapB, 
    sweeps, 
    data.panel=1, 
    col=&quot;#c00000&quot;, 
    avoid.overlapping = FALSE, 
    r0=-0.6, r1=-0.4
    )
kpAddBaseNumbers(
    kp_hapB, 
    tick.dist = 1000001, 
    tick.len = 20, 
    tick.col=&quot;#000000&quot;, 
    cex = 0.01
    )
kpPlotMarkers(kp_hapB, 
    chr=labels$chr, 
    x=labels$pos, 
    labels=labels$label, 
    text.orientation = &quot;horizontal&quot;, 
    line.color = &quot;#000000&quot;, 
    label.color = labels$colour, 
    marker.parts = c(0.1, 0.8, 0.1), 
    r1=0.4, cex=0.8, 
    ignore.chromosome.ends=FALSE)

dev.off()">
                     <pre class="notranslate"><code>svglite("chrsB.svg", width=8, height=6)

kp_hapB &lt;- plotKaryotype(genome=mygenome, chromosomes=hapB)
kpPlotRegions(
    kp_hapB, loci, 
    data.panel="ideogram", 
    r0=NULL, r1=NULL, 
    col=loci$colour, 
    border='#000000', 
    avoid.overlapping = FALSE)
kpPlotRegions(
    kp_hapB, 
    sweeps, 
    data.panel=1, 
    col="#c00000", 
    avoid.overlapping = FALSE, 
    r0=-0.6, r1=-0.4
    )
kpAddBaseNumbers(
    kp_hapB, 
    tick.dist = 1000001, 
    tick.len = 20, 
    tick.col="#000000", 
    cex = 0.01
    )
kpPlotMarkers(kp_hapB, 
    chr=labels$chr, 
    x=labels$pos, 
    labels=labels$label, 
    text.orientation = "horizontal", 
    line.color = "#000000", 
    label.color = labels$colour, 
    marker.parts = c(0.1, 0.8, 0.1), 
    r1=0.4, cex=0.8, 
    ignore.chromosome.ends=FALSE)

dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/267542455-19912205-27df-4b14-9097-49b358176bd2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0NTUtMTk5MTIyMDUtMjdkZi00YjE0LTkwOTctNDliMzU4MTc2YmQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5MDM1MzliOGY1MmIxMTc0ZDNlYzJlZDEyMDI0ZTM0ZTRjN2NkZDRmMGNmMjY2N2I4ODEzMWMwZjQ0NDM2ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KoBWjMk3VxeUPFJRie-2_uogunSzzGJc3QFhwG2CH3c" rel="noopener noreferrer" target="_blank">
                      <img alt="chrsB_small" src="https://private-user-images.githubusercontent.com/33470968/267542455-19912205-27df-4b14-9097-49b358176bd2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjc1NDI0NTUtMTk5MTIyMDUtMjdkZi00YjE0LTkwOTctNDliMzU4MTc2YmQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5MDM1MzliOGY1MmIxMTc0ZDNlYzJlZDEyMDI0ZTM0ZTRjN2NkZDRmMGNmMjY2N2I4ODEzMWMwZjQ0NDM2ZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KoBWjMk3VxeUPFJRie-2_uogunSzzGJc3QFhwG2CH3c" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     this illustrates that one of the loci (
                     <em>
                      loc3B
                     </em>
                     ) overlaps a region that has a positive signal for selection
                     <br/>
                     <br/>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Generating synteny plots with
                      <a href="https://github.com/thackl/gggenomes">
                       gggenomes
                      </a>
                     </h2>
                     <a aria-label="Permalink: Generating synteny plots with gggenomes" class="anchor" href="#generating-synteny-plots-with-gggenomes" id="user-content-generating-synteny-plots-with-gggenomes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Here we are creating a synteny plot of genomic regions from three different strains of the same species. One of the assemblies is fully phased into full-length chromosomes, while the other two are partially phased into contigs only. Through GWAS, we have identified significant marker-trait association regions (MTAR) on chromosomes 6 (A and B). We want to visualise annotated gene content and synteny between orthologous regions from all three assemblies. Here, an MTAR is defined as a genomic interval given by the span of SNP markers above a
                     <em>
                      p
                     </em>
                     -value threshold in an association peak, which can be retrieved from the output matrix of GWAS programs such as
                     <a href="https://www.maizegenetics.net/tassel" rel="nofollow">
                      TASSEL
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/269470527-857a35bf-c785-4a87-af56-3ca3bff481e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1MjctODU3YTM1YmYtYzc4NS00YTg3LWFmNTYtM2NhM2JmZjQ4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4Y2Q2ZDA2NmFlNzU0ZjhkNDk2MDM3NDJiZDVkMDFhMTllNzI5ODllNTczNmYxZDZkNDk0MThlMDNmMWRkODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8BwQJDCMnn_TsEAmxLg37OlN6cgprAKcCzXXajZ2ZcI" rel="noopener noreferrer" target="_blank">
                      <img alt="ROI_from_GWAS" src="https://private-user-images.githubusercontent.com/33470968/269470527-857a35bf-c785-4a87-af56-3ca3bff481e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1MjctODU3YTM1YmYtYzc4NS00YTg3LWFmNTYtM2NhM2JmZjQ4MWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4Y2Q2ZDA2NmFlNzU0ZjhkNDk2MDM3NDJiZDVkMDFhMTllNzI5ODllNTczNmYxZDZkNDk0MThlMDNmMWRkODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8BwQJDCMnn_TsEAmxLg37OlN6cgprAKcCzXXajZ2ZcI" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     In our example, the MTARs on chrs 6 in
                     <em>
                      mygenome.fasta
                     </em>
                     are defined by the following coordinates:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="chr6_A:480000-700000
chr6_B:570000-770000">
                     <pre class="notranslate"><code>chr6_A:480000-700000
chr6_B:570000-770000
</code></pre>
                    </div>
                    <p dir="auto">
                     Contigs from the other two assemblies
                     <em>
                      12SD80_refassem.fa
                     </em>
                     and
                     <em>
                      12NC29_refassem.fa
                     </em>
                     that are orthologous to these MTARs were established based on whole assembly alignments between
                     <em>
                      mygenome.fasta
                     </em>
                     ,
                     <em>
                      12SD80_refassem.fa
                     </em>
                     and
                     <em>
                      12NC29_refassem.fa
                     </em>
                     using a tool such as
                     <a href="https://github.com/lh3/minimap2">
                      Minimap2
                     </a>
                     or
                     <a href="https://dgenies.toulouse.inra.fr/" rel="nofollow">
                      D-Genies
                     </a>
                     to visualise dotplots. In the below illustration, a contig-level assembly was aligned to a chr-level assembly with the dotplot zoomed-in to region of interest on
                     <em>
                      chr9
                     </em>
                     . Here we see that
                     <em>
                      contig_1
                     </em>
                     and
                     <em>
                      contig_8
                     </em>
                     map to this region but wih some gaps remaining.
                    </p>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/269470576-165c0ee1-cd72-40e5-88a8-6ff39d3c8adc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1NzYtMTY1YzBlZTEtY2Q3Mi00MGU1LTg4YTgtNmZmMzlkM2M4YWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3MWMwNGM2MDcwZmIzYWEyYTZkMDY1OTkwZjM3ZDkzN2E1ZWIxNGI4NmU4MjRjYTYyOGEzOWVmNmQzY2E4MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PmmPG-4noOUf0QdDeALCPN4idwihLzf_atXYPZTX60w" rel="noopener noreferrer" target="_blank">
                      <img alt="dotplot_synteny_juxta" src="https://private-user-images.githubusercontent.com/33470968/269470576-165c0ee1-cd72-40e5-88a8-6ff39d3c8adc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA1NzYtMTY1YzBlZTEtY2Q3Mi00MGU1LTg4YTgtNmZmMzlkM2M4YWRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3MWMwNGM2MDcwZmIzYWEyYTZkMDY1OTkwZjM3ZDkzN2E1ZWIxNGI4NmU4MjRjYTYyOGEzOWVmNmQzY2E4MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PmmPG-4noOUf0QdDeALCPN4idwihLzf_atXYPZTX60w" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     In our example, the orthologous contigs (orthotigs) in
                     <em>
                      12SD80_refassem.fa
                     </em>
                     and
                     <em>
                      12NC29_refassem.fa
                     </em>
                     have the following sequence IDs:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# from 12SD80
000065F_004
000065F
000065F_003

# from 12NC29
000020F
000020F_001">
                     <pre class="notranslate"><code># from 12SD80
000065F_004
000065F
000065F_003

# from 12NC29
000020F
000020F_001
</code></pre>
                    </div>
                    <p dir="auto">
                     We also have previously generated annotations for each assembly in the form of GFF files containing predicted gene models, and for
                     <em>
                      mygenome.fasta
                     </em>
                     , an additional GFF file for sequence repeats. We will first prepare all the inputs in UNIX/bash before moving to RStudio to make the synteny plot.
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      1. retrieve MTAR and orthotig sequences
                     </strong>
                     <br/>
                     extract MTAR seqs from
                     <em>
                      mygenome.fasta
                     </em>
                     for each haplotype (A and B)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="module load samtools/1.12
samtools faidx mygenome.fasta chr6_A:480000-700000 | sed -r 's/:[0-9]+-[0-9]+//g' &gt; MTAR_6A.fa
samtools faidx mygenome.fasta chr6_B:570000-770000 | sed -r 's/:[0-9]+-[0-9]+//g' &gt; MTAR_6B.fa">
                     <pre class="notranslate"><code>module load samtools/1.12
samtools faidx mygenome.fasta chr6_A:480000-700000 | sed -r 's/:[0-9]+-[0-9]+//g' &gt; MTAR_6A.fa
samtools faidx mygenome.fasta chr6_B:570000-770000 | sed -r 's/:[0-9]+-[0-9]+//g' &gt; MTAR_6B.fa
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       sed -r ...
                      </em>
                      , this removes the coordinates which samtools automatically adds to the output fasta headers (otherwise seq IDs won't match GFFs)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     now pull out the orthotigs from
                     <em>
                      12SD80_refassem.fa
                     </em>
                     and
                     <em>
                      12NC29_refassem.fa
                     </em>
                     . This can be done with a tool such as
                     <a href="https://github.com/TC-Hewitt/Misc_NGS/blob/master/get_contigs.py">
                      <em>
                       get_contigs.py
                      </em>
                     </a>
                     from my other repo
                     <a href="https://github.com/TC-Hewitt/Misc_NGS">
                      Misc_NGS
                     </a>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="~/Misc_NGS/get_contigs.py -i 12SD80_refassem.fa -l 000065F_004 000065F 000065F_003 -o MTAR_6_12SD80_orthotigs.fa
~/Misc_NGS/get_contigs.py -i 12NC29_refassem.fa -l 000020F 000020F_001 -o MTAR_6_12NC29_orthotigs.fa">
                     <pre class="notranslate"><code>~/Misc_NGS/get_contigs.py -i 12SD80_refassem.fa -l 000065F_004 000065F 000065F_003 -o MTAR_6_12SD80_orthotigs.fa
~/Misc_NGS/get_contigs.py -i 12NC29_refassem.fa -l 000020F 000020F_001 -o MTAR_6_12NC29_orthotigs.fa
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -i
                      </em>
                      , input fasta file
                      <br/>
                      <em>
                       -l
                      </em>
                      , space-separated list of seq IDs
                      <br/>
                      <em>
                       -o
                      </em>
                      , output fasta file
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      2. retrieve corresponding annotations for MTAR seqs and orthotigs
                     </strong>
                     <br/>
                    </p>
                    <p dir="auto">
                     using an AWK script, we will use the same coordinates as previously to pull out annotations for each MTAR from
                     <em>
                      mygenome.models.gff
                     </em>
                     then use
                     <em>
                      grep
                     </em>
                     to keep only
                     <em>
                      mRNA
                     </em>
                     features
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="awk '$1 == &quot;chr6_A&quot; &amp;&amp; $4 &gt;= 480000 &amp;&amp; $5 &lt;= 700000' mygenome.models.gff | grep mRNA &gt; MTAR_6A.mrna.gff
awk '$1 == &quot;chr6_B&quot; &amp;&amp; $4 &gt;= 570000 &amp;&amp; $5 &lt;= 770000' mygenome.models.gff | grep mRNA &gt; MTAR_6B.mrna.gff">
                     <pre class="notranslate"><code>awk '$1 == "chr6_A" &amp;&amp; $4 &gt;= 480000 &amp;&amp; $5 &lt;= 700000' mygenome.models.gff | grep mRNA &gt; MTAR_6A.mrna.gff
awk '$1 == "chr6_B" &amp;&amp; $4 &gt;= 570000 &amp;&amp; $5 &lt;= 770000' mygenome.models.gff | grep mRNA &gt; MTAR_6B.mrna.gff
</code></pre>
                    </div>
                    <p dir="auto">
                     we will do the same for the repeat annotations
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="awk '$1 == &quot;chr6_A&quot; &amp;&amp; $4 &gt;= 480000 &amp;&amp; $5 &lt;= 700000' mygenome.repeats.gff &gt; MTAR_6A.repeats.gff
awk '$1 == &quot;chr6_B&quot; &amp;&amp; $4 &gt;= 570000 &amp;&amp; $5 &lt;= 770000' mygenome.repeats.gff &gt; MTAR_6B.repeats.gff">
                     <pre class="notranslate"><code>awk '$1 == "chr6_A" &amp;&amp; $4 &gt;= 480000 &amp;&amp; $5 &lt;= 700000' mygenome.repeats.gff &gt; MTAR_6A.repeats.gff
awk '$1 == "chr6_B" &amp;&amp; $4 &gt;= 570000 &amp;&amp; $5 &lt;= 770000' mygenome.repeats.gff &gt; MTAR_6B.repeats.gff
</code></pre>
                    </div>
                    <p dir="auto">
                     now we have annotations for just the MTARs; however, the
                     <em>
                      start
                     </em>
                     and
                     <em>
                      end
                     </em>
                     coordinates of each feature are still relative to the whole chrs 6A and 6B; to modify their positions to be relative to
                     <em>
                      MTAR_6A.fa
                     </em>
                     and
                     <em>
                      MTAR_6B.fa
                     </em>
                     we can use AWK to subtract just the start coordinate of the respective MTAR from both the
                     <em>
                      start
                     </em>
                     and
                     <em>
                      end
                     </em>
                     values of each feature
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="awk -F $'\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\t' MTAR_6A.mrna.gff &gt; MTAR_6.mrna.modpos.gff
awk -F $'\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\t' MTAR_6B.mrna.gff &gt;&gt; MTAR_6.mrna.modpos.gff">
                     <pre class="notranslate"><code>awk -F $'\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\t' MTAR_6A.mrna.gff &gt; MTAR_6.mrna.modpos.gff
awk -F $'\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\t' MTAR_6B.mrna.gff &gt;&gt; MTAR_6.mrna.modpos.gff
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      here we write the outputs from both haplotypes to the same file
                      <em>
                       MTAR_6.mrna.modpos.gff
                      </em>
                      <br/>
                     </p>
                    </blockquote>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="awk -F $'\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\t' MTAR_6A.repeats.gff &gt; MTAR_6.repeats.modpos.gff
awk -F $'\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\t' MTAR_6B.repeats.gff &gt;&gt; MTAR_6.repeats.modpos.gff">
                     <pre class="notranslate"><code>awk -F $'\t' ' { $4 = $4 - 480000; $5 = $5 - 480000; print; } ' OFS=$'\t' MTAR_6A.repeats.gff &gt; MTAR_6.repeats.modpos.gff
awk -F $'\t' ' { $4 = $4 - 570000; $5 = $5 - 570000; print; } ' OFS=$'\t' MTAR_6B.repeats.gff &gt;&gt; MTAR_6.repeats.modpos.gff
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      here we write the outputs from both haplotypes to the same file
                      <em>
                       MTAR_6.repeats.modpos.gff
                      </em>
                      <br/>
                      this isn't always necessary since the R package
                      <em>
                       gggenomes
                      </em>
                      allows intervals to be custom defined in generating synteny plots; however, to save overhead of importing the entire chr or assembly and GFFs files into RStudio, we have opted to prep the inputs beforehand
                      <br/>
                     </p>
                    </blockquote>
                    <p dir="auto">
                     likewise, we will retrieve the
                     <em>
                      mRNA
                     </em>
                     annotations for the orthotigs from
                     <em>
                      12SD80
                     </em>
                     and
                     <em>
                      12NC29
                     </em>
                     using a
                     <em>
                      bash
                     </em>
                     script; we do not need to modify the feature positions since they are already relative to the individual contigs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# 12SD80
for i in 000065F_004 000065F 000065F_003
do
    grep "^$i\s.*mRNA" 12SD80_refassem.models.gff3 &gt;&gt; MTAR_6_12SD80_orthotigs.mrna.gff
done

#12NC29
for i in 000020F 000020F_001
do
    grep "^$i\s.*mRNA" 12NC29_refassem.models.gff3 &gt;&gt; MTAR_6_12NC29_orthotigs.mrna.gff
done'>
                     <pre class="notranslate"><code># 12SD80
for i in 000065F_004 000065F 000065F_003
do
    grep "^$i\s.*mRNA" 12SD80_refassem.models.gff3 &gt;&gt; MTAR_6_12SD80_orthotigs.mrna.gff
done

#12NC29
for i in 000020F 000020F_001
do
    grep "^$i\s.*mRNA" 12NC29_refassem.models.gff3 &gt;&gt; MTAR_6_12NC29_orthotigs.mrna.gff
done
</code></pre>
                    </div>
                    <br/>
                    <p dir="auto">
                     <strong>
                      3. compile fasta files and annotation files of MTARs and orthotigs for later export to RStudio
                     </strong>
                     <br/>
                     combine MTAR and orthotig fastas to
                     <em>
                      MTAR_6_allseqs.fa
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat \
MTAR_6A.fa \
MTAR_6B.fa \
MTAR_6_12SD80_orthotigs.fa \
MTAR_6_12NC29_orthotigs.fa \
&gt; MTAR_6_allseqs.fa">
                     <pre class="notranslate"><code>cat \
MTAR_6A.fa \
MTAR_6B.fa \
MTAR_6_12SD80_orthotigs.fa \
MTAR_6_12NC29_orthotigs.fa \
&gt; MTAR_6_allseqs.fa
</code></pre>
                    </div>
                    <p dir="auto">
                     combine MTAR and orthotig gffs to
                     <em>
                      MTAR_6_allseqs.mrna.gff
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat \
MTAR_6.mrna.modpos.gff \
MTAR_6_12SD80_orthotigs.mrna.gff \
MTAR_6_12NC29_orthotigs.mrna.gff \
&gt; MTAR_6_allseqs.mrna.gff">
                     <pre class="notranslate"><code>cat \
MTAR_6.mrna.modpos.gff \
MTAR_6_12SD80_orthotigs.mrna.gff \
MTAR_6_12NC29_orthotigs.mrna.gff \
&gt; MTAR_6_allseqs.mrna.gff
</code></pre>
                    </div>
                    <p dir="auto">
                     since we only have repeat annotations for the MTARs, we do not need to combine them with those of the orthotigs (unavailable)
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      4. perform an all-vs-all alignment of
                      <em>
                       MTAR_6_allseqs.fa
                      </em>
                      with
                      <a href="https://github.com/lh3/minimap2">
                       Minimap2
                      </a>
                     </strong>
                     <br/>
                     the PAF output will be later used for drawing links between regions in the synteny plot
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="module load minimap2/2.24
minimap2 -X MTAR_6_allseqs.fa MTAR_6_allseqs.fa &gt; MTAR_6_allseqs.paf">
                     <pre class="notranslate"><code>module load minimap2/2.24
minimap2 -X MTAR_6_allseqs.fa MTAR_6_allseqs.fa &gt; MTAR_6_allseqs.paf
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -X
                      </em>
                      , skip self and dual mappings (for the all-vs-all mode)
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      5. generate ortholog clusters of gene models using
                      <a href="https://github.com/davidemms/OrthoFinder">
                       OrthoFinder
                      </a>
                     </strong>
                     <br/>
                     we will use protein fasta files previously created along with the GFF annotations of the three assemblies; a package such as
                     <a href="https://github.com/NBISweden/AGAT">
                      AGAT
                     </a>
                     can be used to generate translated protein sequences from a gff and nucleotide fasta
                     <br/>
                    </p>
                    <p dir="auto">
                     the orthogroups will be later used for colouring genes and drawing links between genes in the synteny plot
                     <br/>
                    </p>
                    <p dir="auto">
                     first we will retrieve the matching sequences from each protein fasta with
                     <a href="https://github.com/TC-Hewitt/Misc_NGS/blob/master/get_contigs.py">
                      <em>
                       get_contigs.py
                      </em>
                     </a>
                     using
                     <em>
                      MTAR_6_allseqs.mrna.gff
                     </em>
                     as a reference
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="~/Misc_NGS/get_contigs.py -i mygenome.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6.proteins.faa
~/Misc_NGS/get_contigs.py -i 12SD80_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12SD80_orthotigs.proteins.faa
~/Misc_NGS/get_contigs.py -i 12NC29_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12NC29_orthotigs.proteins.faa">
                     <pre class="notranslate"><code>~/Misc_NGS/get_contigs.py -i mygenome.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6.proteins.faa
~/Misc_NGS/get_contigs.py -i 12SD80_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12SD80_orthotigs.proteins.faa
~/Misc_NGS/get_contigs.py -i 12NC29_refassem.proteins.faa -t MTAR_6_allseqs.mrna.gff -s gene_ -o MTAR_6_12NC29_orthotigs.proteins.faa
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -i
                      </em>
                      , input fasta file
                      <br/>
                      <em>
                       -t
                      </em>
                      , reference file containing IDs of seqs to retrieve
                      <br/>
                      <em>
                       -s gene
                      </em>
                      _, prefix of seq IDs to look for in reference file (i.e. starts with "gene_")
                      <br/>
                      <em>
                       -o
                      </em>
                      , output fasta file
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     now we will create a separate directory
                     <em>
                      translated
                     </em>
                     to move our retrieved protein seqs and run
                     <em>
                      orthofinder
                     </em>
                     on
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mkdir translated
mv *.faa translated/

module load orthofinder/2.5.4
orthofinder -a 4 -f translated/">
                     <pre class="notranslate"><code>mkdir translated
mv *.faa translated/

module load orthofinder/2.5.4
orthofinder -a 4 -f translated/
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       -a
                      </em>
                      , number of parallel analysis threads
                      <br/>
                      <em>
                       -f
                      </em>
                      , dir containing fasta proteomes
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     the output we want for our purposes is found in
                     <em>
                      translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt
                     </em>
                     where the wildcard "?????" substitutes for the date the orthofinder output was created (e.g. "Feb28");
                     <em>
                      Orthogroups.txt
                     </em>
                     is not in a tractable format, so we can convert it to long format using
                     <a href="https://github.com/TC-Hewitt/OatCrownRust/blob/main/Orthogroups_txtConvert.py">
                      <em>
                       Orthogroups_txtConvert.py
                      </em>
                     </a>
                     for later use in R
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="./Orthogroups_txtConvert.py -i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt \
&gt; MTAR_6_allseqs.orthogroups.txt">
                     <pre class="notranslate"><code>./Orthogroups_txtConvert.py -i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt \
&gt; MTAR_6_allseqs.orthogroups.txt
</code></pre>
                    </div>
                    <br/>
                    <p dir="auto">
                     <strong>
                      6. pull out only secreted subset from orthogroups
                     </strong>
                     <br/>
                     in this example, we are particularly interested in genes that encode secreted proteins and want to highlight these in our synteny plot later; here they are denoted by the string "secreted" in the attribute field (column 9) of the gff files; for example, column 9 of a single gene record in
                     <em>
                      MTAR_6_allseqs.mrna.gff
                     </em>
                     looks like the following:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="id=gene_1-T1;parent=gene_1;product=secreted protein;">
                     <pre class="notranslate"><code>id=gene_1-T1;parent=gene_1;product=secreted protein;
</code></pre>
                    </div>
                    <p dir="auto">
                     of course, this depends on the specific annotation workflow used in assigning attributes to gene models, but for our example, our genes of interest can be easily retrieved with
                     <em>
                      grep
                     </em>
                     ; as we only need the transcript name (
                     <em>
                      id=...
                     </em>
                     ), we can strip other columns and attributes to output a simple list of IDs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='grep "secreted" MTAR_6_allseqs.mrna.gff \
| cut -f 9 | sed -r "s/id=([^;]+).*$/\1/g" \
&gt; MTAR_6_allseqs.secreted.txt'>
                     <pre class="notranslate"><code>grep "secreted" MTAR_6_allseqs.mrna.gff \
| cut -f 9 | sed -r "s/id=([^;]+).*$/\1/g" \
&gt; MTAR_6_allseqs.secreted.txt
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       cut -f 9
                      </em>
                      , limit to col 9
                      <br/>
                      <em>
                       sed -r ...
                      </em>
                      , regex operation discards all chars except those matching group 1 (any consecutive chars after "id=" except ";" in the parentheses)
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     we will then use the IDs to extract the orthogroups from
                     <em>
                      Orthogroups.txt
                     </em>
                     and write to a temp file; this will pull out all the genes of an orthogroup, as long as the group contains at least one secreted member in
                     <em>
                      MTAR_6_allseqs.secreted.txt
                     </em>
                     (this is desirable since we may want to know if the secretion state varies between orthologs of different assemblies/strains)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for i in $(cat MTAR_allseqs.secreted.txt) 
do
    grep $i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt &gt;&gt; temp1.txt 
done">
                     <pre class="notranslate"><code>for i in $(cat MTAR_allseqs.secreted.txt) 
do
    grep $i translated/OrthoFinder/Results_?????/Orthogroups/Orthogroups.txt &gt;&gt; temp1.txt 
done
</code></pre>
                    </div>
                    <p dir="auto">
                     given the above,
                     <em>
                      temp1.txt
                     </em>
                     is likely to contain duplicate lines which we can remove with
                     <em>
                      sort
                     </em>
                     and
                     <em>
                      uniq
                     </em>
                     and write to a new temp file
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sort temp1.txt | uniq &gt; temp2.txt">
                     <pre class="notranslate"><code>sort temp1.txt | uniq &gt; temp2.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     we can now convert this to a desirable format using
                     <em>
                      Orthogroups_txtConvert.py
                     </em>
                     then delete the temp files
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="./Orthogroups_txtConvert.py -i temp2.txt &gt; MTAR_6_allseqs.orthogroups.secreted.txt
rm temp1.txt temp2.txt">
                     <pre class="notranslate"><code>./Orthogroups_txtConvert.py -i temp2.txt &gt; MTAR_6_allseqs.orthogroups.secreted.txt
rm temp1.txt temp2.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      MTAR_6_allseqs.orthogroups.secreted.txt
                     </em>
                     now contains only proteins that are either secreted or belong to the same orthogroup as a secreted protein
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      7. create zip folder of all input files for export to Rstudio
                     </strong>
                     <br/>
                     now all input files for our synteny plot are ready, we will zip them in
                     <em>
                      exportR
                     </em>
                     for easy transfer to RStudio
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="zip exportR \
MTAR_6_allseqs.fa \
MTAR_6_allseqs.gff \
MTAR_6.repeats.modpos.gff \
MTAR_6_allseqs.paf \
MTAR_6_allseqs.orthogroups.txt \
MTAR_6_allseqs.orthogroups.secreted.txt">
                     <pre class="notranslate"><code>zip exportR \
MTAR_6_allseqs.fa \
MTAR_6_allseqs.gff \
MTAR_6.repeats.modpos.gff \
MTAR_6_allseqs.paf \
MTAR_6_allseqs.orthogroups.txt \
MTAR_6_allseqs.orthogroups.secreted.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     now in RStudio, we want to draw a synteny plot using
                     <em>
                      gggenomes
                     </em>
                     with the following features:
                    </p>
                    <ul dir="auto">
                     <li>
                      sequences arranged in bins based on haplotype
                     </li>
                     <li>
                      links bewteen adjacent sequences showing blocks of homology
                     </li>
                     <li>
                      directional gene annotations
                     </li>
                     <li>
                      stranded repeat blocks
                     </li>
                     <li>
                      genes coloured by orthogroup (secreted only)
                     </li>
                     <li>
                      lines between adjacent sequences linking genes of the same orthogroup
                     </li>
                     <li>
                      gene labels (secreted only)
                     </li>
                    </ul>
                    <p dir="auto">
                     the following was conducted in RStudio using R 4.0.2 and gggenomes 0.9.5.9000
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      8. load libraries and read in input files
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(gggenomes)
library(svglite)

#assign inputs
my_seqs &lt;- read_seqs("MTAR_6_allseqs.fa")
my_genes &lt;- read_gff3("MTAR_6_allseqs.gff")
my_repeats &lt;- read_gff("MTAR_6.repeats.modpos.gff")
my_links &lt;- read_paf("MTAR_6_allseqs.paf")
my_clusts &lt;- read.table("MTAR_6_allseqs.orthogroups.txt", header=T, sep="\t")
my_clusts_sec &lt;- read.table("MTAR_6_allseqs.orthogroups.secreted.txt", header=T, sep="\t")

#remove 3 leading zeroes from 12SD80 and 12NC29 derived seq_ids for cleaner labels later on
my_seqs$seq_id &lt;- sub("^000", "", my_seqs$seq_id)
my_genes$seq_id &lt;- sub("^000", "", my_genes$seq_id)
my_links$seq_id &lt;- sub("^000", "", my_links$seq_id)
my_links$seq_id2 &lt;- sub("^000", "", my_links$seq_id2)'>
                     <pre class="notranslate"><code>library(gggenomes)
library(svglite)

#assign inputs
my_seqs &lt;- read_seqs("MTAR_6_allseqs.fa")
my_genes &lt;- read_gff3("MTAR_6_allseqs.gff")
my_repeats &lt;- read_gff("MTAR_6.repeats.modpos.gff")
my_links &lt;- read_paf("MTAR_6_allseqs.paf")
my_clusts &lt;- read.table("MTAR_6_allseqs.orthogroups.txt", header=T, sep="\t")
my_clusts_sec &lt;- read.table("MTAR_6_allseqs.orthogroups.secreted.txt", header=T, sep="\t")

#remove 3 leading zeroes from 12SD80 and 12NC29 derived seq_ids for cleaner labels later on
my_seqs$seq_id &lt;- sub("^000", "", my_seqs$seq_id)
my_genes$seq_id &lt;- sub("^000", "", my_genes$seq_id)
my_links$seq_id &lt;- sub("^000", "", my_links$seq_id)
my_links$seq_id2 &lt;- sub("^000", "", my_links$seq_id2)
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      we wont end up using
                      <em>
                       my_clusts
                      </em>
                      in our final plot and will only use
                      <em>
                       my_clusts_sec
                      </em>
                      as we want to highlight only secreted genes
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      9. create preliminary plot for assessing sequence arrangements
                     </strong>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='synplot1 &lt;- gggenomes(
     genes = my_genes,
     seqs = my_seqs,
     feats = NULL,
     links = my_links,
     .id = "file_id",
     spacing = 0.05,
     wrap = NULL,
     adjacent_only = FALSE, #will also plot links b/w non-adjacent seqs
     infer_bin_id = seq_id,
     infer_start = min(start, end),
     infer_end = max(start, end),
     infer_length = max(start, end),
     theme = c("clean", NULL),
     .layout = NULL) + geom_link(alpha=0.1) + geom_seq() + geom_bin_label()'>
                     <pre class="notranslate"><code>synplot1 &lt;- gggenomes(
     genes = my_genes,
     seqs = my_seqs,
     feats = NULL,
     links = my_links,
     .id = "file_id",
     spacing = 0.05,
     wrap = NULL,
     adjacent_only = FALSE, #will also plot links b/w non-adjacent seqs
     infer_bin_id = seq_id,
     infer_start = min(start, end),
     infer_end = max(start, end),
     infer_length = max(start, end),
     theme = c("clean", NULL),
     .layout = NULL) + geom_link(alpha=0.1) + geom_seq() + geom_bin_label()
</code></pre>
                    </div>
                    <p dir="auto">
                     we can select a few sequences to plot at a time to help figure out how they should be ordered and binned
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='synplot1 %&gt;% pick("chr6_A", "chr6_B", "065F", "065F_004", "065F_003")'>
                     <pre class="notranslate"><code>synplot1 %&gt;% pick("chr6_A", "chr6_B", "065F", "065F_004", "065F_003")
</code></pre>
                    </div>
                    <p dir="auto">
                     for example, this plots the A and B MTARs along with orthotigs from
                     <em>
                      12SD80
                     </em>
                     . Homology links are shown between all seqs.
                    </p>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/269470823-f7c40d1e-dd1d-4da0-bd57-bab18ca2080b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4MjMtZjdjNDBkMWUtZGQxZC00ZGEwLWJkNTctYmFiMThjYTIwODBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkY2Q1Y2YwNzQ1ZTJlNDM0NjBiMjZhYTJjMmY0ZTUwNjc5ZTJhMDZjNzQ5Y2NmYzQ1Y2VhYTdkOGMwZjFkM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rolAIhw09cOci1uJadwIUrBqUtqSbLxz1GlNr9DjS90" rel="noopener noreferrer" target="_blank">
                      <img alt="synplot_exp1" src="https://private-user-images.githubusercontent.com/33470968/269470823-f7c40d1e-dd1d-4da0-bd57-bab18ca2080b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4MjMtZjdjNDBkMWUtZGQxZC00ZGEwLWJkNTctYmFiMThjYTIwODBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkY2Q1Y2YwNzQ1ZTJlNDM0NjBiMjZhYTJjMmY0ZTUwNjc5ZTJhMDZjNzQ5Y2NmYzQ1Y2VhYTdkOGMwZjFkM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rolAIhw09cOci1uJadwIUrBqUtqSbLxz1GlNr9DjS90" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     on first look we can see the plot may benefit from alternative ordering and orientation of seqs. For instance, contigs
                     <em>
                      065F_003
                     </em>
                     and
                     <em>
                      065F_004
                     </em>
                     appear to align side-by-side against the larger
                     <em>
                      065F
                     </em>
                     contig, meaning they should be placed in the same bin (each bin representing an inferred haplotype). By default, all seqs are plotted as a single stack unless they are assigned to bins allowing seqs to be positioned in series
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      10. assign seqs to bins and create a new plot
                     </strong>
                     <br/>
                     first we need to know the lengths of our seqs to enter into the table. These can be found in the
                     <em>
                      my_seqs
                     </em>
                     object created earlier
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="print(my_seqs)
#   seq_id  seq_desc    length
    &lt;chr&gt;   &lt;chr&gt;   &lt;int&gt;
    chr6_A  NA  220001
    chr6_B  NA  200001
    000065F_004 NA  103286
    000065F NA  373524
    000065F_003 NA  237372
    000020F NA  481180
    000020F_001 NA  241894">
                     <pre class="notranslate"><code>print(my_seqs)
#   seq_id  seq_desc    length
    &lt;chr&gt;   &lt;chr&gt;   &lt;int&gt;
    chr6_A  NA  220001
    chr6_B  NA  200001
    000065F_004 NA  103286
    000065F NA  373524
    000065F_003 NA  237372
    000020F NA  481180
    000020F_001 NA  241894
</code></pre>
                    </div>
                    <p dir="auto">
                     we can then define our bin information in a new table object
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='my_bins &lt;- tribble(
  ~bin_id, ~seq_id, ~length,
  "mygenome_1", "chr6_A", 220001,
  "mygenome_2", "chr6_B", 200001,
  "12SD80_1", "065F", 373524,
  "12SD80_2", "065F_004", 103286,
  "12SD80_2", "065F_003", 237372,  
  "12NC29_1", "020F", 481180,
  "12NC29_2", "020F_001", 241894,
)'>
                     <pre class="notranslate"><code>my_bins &lt;- tribble(
  ~bin_id, ~seq_id, ~length,
  "mygenome_1", "chr6_A", 220001,
  "mygenome_2", "chr6_B", 200001,
  "12SD80_1", "065F", 373524,
  "12SD80_2", "065F_004", 103286,
  "12SD80_2", "065F_003", 237372,  
  "12NC29_1", "020F", 481180,
  "12NC29_2", "020F_001", 241894,
)
</code></pre>
                    </div>
                    <p dir="auto">
                     we may also want to filter out smaller links from our
                     <em>
                      my_links
                     </em>
                     object so they do not clutter the plot. Here we will only keep links of thickness 5kb or more and assign to
                     <em>
                      my_links_5kb
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="my_links_5kb &lt;- filter(my_links, map_length &gt;= 5000)">
                     <pre class="notranslate"><code>my_links_5kb &lt;- filter(my_links, map_length &gt;= 5000)
</code></pre>
                    </div>
                    <p dir="auto">
                     we can now create our initial plot substituting in
                     <em>
                      my_bins
                     </em>
                     and
                     <em>
                      my_links_5kb
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='synplot2 &lt;- gggenomes(
     genes = my_genes,
     seqs = my_bins, # new seq table
     feats = my_repeats,
     links = my_links_5kb, # new links obj
     .id = "file_id",
     spacing = 0.05,
     wrap = NULL,
     adjacent_only = TRUE, # links will only be shown between adjacent seqs
     infer_bin_id = bin_id, # make sure changed to bin_id
     infer_start = min(start, end),
     infer_end = max(start, end),
     infer_length = max(start, end),
     theme = c("clean", NULL),
     .layout = NULL) +
     geom_bin_label() + 
     geom_seq() + 
     geom_link(fill="lightgrey", color="lightgrey") + 
     geom_seq_label()'>
                     <pre class="notranslate"><code>synplot2 &lt;- gggenomes(
     genes = my_genes,
     seqs = my_bins, # new seq table
     feats = my_repeats,
     links = my_links_5kb, # new links obj
     .id = "file_id",
     spacing = 0.05,
     wrap = NULL,
     adjacent_only = TRUE, # links will only be shown between adjacent seqs
     infer_bin_id = bin_id, # make sure changed to bin_id
     infer_start = min(start, end),
     infer_end = max(start, end),
     infer_length = max(start, end),
     theme = c("clean", NULL),
     .layout = NULL) +
     geom_bin_label() + 
     geom_seq() + 
     geom_link(fill="lightgrey", color="lightgrey") + 
     geom_seq_label()
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/269470889-7bb20abd-d5b1-472f-bafc-8d03fb238211.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4ODktN2JiMjBhYmQtZDViMS00NzJmLWJhZmMtOGQwM2ZiMjM4MjExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiNjllN2JiZDZkNDFjZjM0ZTRhOWYxZDFmYzhiMjI3NmQ1ZDYyOWNkMzJkZWE3NzBiNTA0ZjE2YTBhYWRlNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FRldwEMOFGc4xQ0N2Vm-xuA299dKaqQJKGSpTddzNp8" rel="noopener noreferrer" target="_blank">
                      <img alt="synplot_exp2" src="https://private-user-images.githubusercontent.com/33470968/269470889-7bb20abd-d5b1-472f-bafc-8d03fb238211.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA4ODktN2JiMjBhYmQtZDViMS00NzJmLWJhZmMtOGQwM2ZiMjM4MjExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiNjllN2JiZDZkNDFjZjM0ZTRhOWYxZDFmYzhiMjI3NmQ1ZDYyOWNkMzJkZWE3NzBiNTA0ZjE2YTBhYWRlNmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FRldwEMOFGc4xQ0N2Vm-xuA299dKaqQJKGSpTddzNp8" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     our seqs are now ordered the way we want, except their orientations are still mismatched
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      11. flip seqs to match orientations
                     </strong>
                     <br/>
                     we can see the alignment can be fixed by flipping seqs
                     <em>
                      065F
                     </em>
                     ,
                     <em>
                      065F_004
                     </em>
                     and
                     <em>
                      065F_003
                     </em>
                     , which are seq numbers 3, 4 and 5, corresponding to the order they are listed in
                     <em>
                      my_bins
                     </em>
                     . We assign the untwisted alignment to a new plot object
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="synplot2_flip &lt;- synplot2 %&gt;% flip_seqs(3, 4, 5)">
                     <pre class="notranslate"><code>synplot2_flip &lt;- synplot2 %&gt;% flip_seqs(3, 4, 5)
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/269470954-2611676c-0c5f-43e8-82b3-9a712155256b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA5NTQtMjYxMTY3NmMtMGM1Zi00M2U4LTgyYjMtOWE3MTIxNTUyNTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYTMxYTc4MTQwODg1NmM5ZjQyMzYzNjQ3N2VkNjAyMGQzMDgwM2U1NDUxZTgzNmM0YjgwMzRmMTQyY2Q0ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._zFxlRWlgHP_CeitZLwmeU0TmKEHsRwFWgnD1BmLRaY" rel="noopener noreferrer" target="_blank">
                      <img alt="synplot_exp3" src="https://private-user-images.githubusercontent.com/33470968/269470954-2611676c-0c5f-43e8-82b3-9a712155256b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzA5NTQtMjYxMTY3NmMtMGM1Zi00M2U4LTgyYjMtOWE3MTIxNTUyNTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYTMxYTc4MTQwODg1NmM5ZjQyMzYzNjQ3N2VkNjAyMGQzMDgwM2U1NDUxZTgzNmM0YjgwMzRmMTQyY2Q0ZTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._zFxlRWlgHP_CeitZLwmeU0TmKEHsRwFWgnD1BmLRaY" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     it is now looking much better, but seq
                     <em>
                      020F
                     </em>
                     is showing some unaligned flanking regions which we might want to trim
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      12. assign seq boundaries to plot only desired regions
                     </strong>
                     <br/>
                     we can create a table defining the start and end coordinates of each seq region we want to show. Note that coordinates are always based on the original seqs (unflipped), so the previous
                     <em>
                      synplot2
                     </em>
                     can be used a reference when defining region coordinates. The default length bar at the bottom can be used as a guide.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='my_loci &lt;- tribble(
  ~seq_id, ~start, ~end,
  "chr6_A", 1, 220001,
  "chr6_B", 1, 200001,
  "065F", 1, 373524,
  "065F_004", 1, 103286,
  "065F_003", 1, 237372,  
  "020F", 70000, 450000,
  "020F_001", 1, 241894,
)'>
                     <pre class="notranslate"><code>my_loci &lt;- tribble(
  ~seq_id, ~start, ~end,
  "chr6_A", 1, 220001,
  "chr6_B", 1, 200001,
  "065F", 1, 373524,
  "065F_004", 1, 103286,
  "065F_003", 1, 237372,  
  "020F", 70000, 450000,
  "020F_001", 1, 241894,
)
</code></pre>
                    </div>
                    <p dir="auto">
                     we can then use the focus function with
                     <em>
                      my_loci
                     </em>
                     to create a new plot object showing only desired regions
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="syplot2_flip_focus &lt;- synplot2_flip %&gt;% focus(.track_id=seqs, .loci=my_loci, .locus_id=seq_id)">
                     <pre class="notranslate"><code>syplot2_flip_focus &lt;- synplot2_flip %&gt;% focus(.track_id=seqs, .loci=my_loci, .locus_id=seq_id)
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/269471107-4304ba3e-0f8c-472d-b8da-7f7e1ce0dff7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzExMDctNDMwNGJhM2UtMGY4Yy00NzJkLWI4ZGEtN2Y3ZTFjZTBkZmY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiY2JiNDA3M2QzNjgwYTFiZjVmMjFiMjc1YTIyNDM4MzEzZjJkYjhhNmFlMzBhYmJmZjQxOTE2ZDcyYTVlY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z2Zp3NrcNwgycAivYFMeGsagweeWsaw4a09Yg1dWwS4" rel="noopener noreferrer" target="_blank">
                      <img alt="synplot_exp4" src="https://private-user-images.githubusercontent.com/33470968/269471107-4304ba3e-0f8c-472d-b8da-7f7e1ce0dff7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzExMDctNDMwNGJhM2UtMGY4Yy00NzJkLWI4ZGEtN2Y3ZTFjZTBkZmY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiY2JiNDA3M2QzNjgwYTFiZjVmMjFiMjc1YTIyNDM4MzEzZjJkYjhhNmFlMzBhYmJmZjQxOTE2ZDcyYTVlY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z2Zp3NrcNwgycAivYFMeGsagweeWsaw4a09Yg1dWwS4" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     now that we have a bare plot combining our desired ordering, orientation and boundaries, we can start adding features and aesthetics
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      13. modify package functions for custom fill and links
                     </strong>
                     <br/>
                     by default,
                     <em>
                      gggenomes
                     </em>
                     function
                     <em>
                      geom_gene()
                     </em>
                     draws features of type "mRNA" from a gff with a lightened color scheme. However, we want our genes to have full solid colour. To temporarily disable automatic lighten,  open and edit the
                     <em>
                      geom_gene()
                     </em>
                     code in the console using
                     <code>
                      trace(gggenomes::geom_gene, edit=TRUE)
                     </code>
                     . Now in the text editor window, change the line
                     <code>
                      rna_def &lt;- aes(fill = colorspace::lighten(fill, .5), color = colorspace::lighten(colour, .5))
                     </code>
                     to simply
                     <code>
                      rna_def &lt;- aes()
                     </code>
                     <br/>
                    </p>
                    <p dir="auto">
                     our plot already shows links between regions of homology as connected polygons. We also want to draw links between genes in the the same cluster but as single lines. However, the default behaviour of
                     <em>
                      geom_link()
                     </em>
                     is to draw links as polygons with the same thickness of the gene, which can look cluttered. To instead draw links as single lines, we can define a
                     <em>
                      geom_link_line()
                     </em>
                     function using base
                     <em>
                      ggplot::geom_segment
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='geom_link_line &lt;- function(mapping = NULL, data = links(), stat = "identity",
          position = "identity", na.rm = FALSE, show.legend = NA, inherit.aes = TRUE,
          ...){
  default_aes &lt;- aes(y=y, yend=yend, x=(x+xend)/2, xend=(xmin+xmax)/2)
  mapping &lt;- gggenomes:::aes_intersect(mapping, default_aes)

  layer(geom = GeomSegment, mapping = mapping, data = data, stat = stat, 
        position = position, show.legend = show.legend, inherit.aes = inherit.aes, 
        params = list(na.rm = na.rm, ...))
}'>
                     <pre class="notranslate"><code>geom_link_line &lt;- function(mapping = NULL, data = links(), stat = "identity",
          position = "identity", na.rm = FALSE, show.legend = NA, inherit.aes = TRUE,
          ...){
  default_aes &lt;- aes(y=y, yend=yend, x=(x+xend)/2, xend=(xmin+xmax)/2)
  mapping &lt;- gggenomes:::aes_intersect(mapping, default_aes)

  layer(geom = GeomSegment, mapping = mapping, data = data, stat = stat, 
        position = position, show.legend = show.legend, inherit.aes = inherit.aes, 
        params = list(na.rm = na.rm, ...))
}
</code></pre>
                    </div>
                    <p dir="auto">
                     we can now include this function when adding features to the plot
                     <br/>
                     <br/>
                    </p>
                    <p dir="auto">
                     <strong>
                      14. add features and custom aesthetics
                     </strong>
                     <br/>
                     we want to colour only secreted genes according to orthgroup (OG) cluster so first setup a colour palette to use. From
                     <a href="https://r-graph-gallery.com/38-rcolorbrewers-palettes.html" rel="nofollow">
                      <em>
                       RColorBrewer
                      </em>
                     </a>
                     we'll use
                     <em>
                      Set1
                     </em>
                     , which has 9 discrete colours but we can expand it with a colour ramp as
                     <em>
                      my_clusts_sec
                     </em>
                     has 17 OGs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(RColorBrewer)
getPalette = colorRampPalette(brewer.pal(8, "Set1"))'>
                     <pre class="notranslate"><code>library(RColorBrewer)
getPalette = colorRampPalette(brewer.pal(8, "Set1"))
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      only using the first 8 colours of
                      <em>
                       Set1
                      </em>
                      since the 9th colour is grey, which we want for
                      <em>
                       na.values
                      </em>
                      , i.e. all other genes not in
                      <em>
                       my_clusts_sec
                      </em>
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     now we will put everything together with previous
                     <em>
                      synplot2_flip_focus
                     </em>
                     in new plot
                     <em>
                      synplot3
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='synplot3 &lt;- synplot2_flip_focus %&gt;% add_clusters(my_clusts_sec) +
geom_feat(position=position_strand(), colour="black") + 
geom_gene(aes(fill=cluster_id, colour=cluster_id), size=7) + 
geom_gene_tag(aes(label=parent_ids), vjust = -1, hjust = -0.15, size = 2) + 
geom_link_line(data=links(.track_id=2), color="black", linetype=2) +
scale_fill_manual("cluster", values=getPalette(17), na.value="darkgrey") + 
scale_color_manual("cluster", values=getPalette(17), na.value="darkgrey")'>
                     <pre class="notranslate"><code>synplot3 &lt;- synplot2_flip_focus %&gt;% add_clusters(my_clusts_sec) +
geom_feat(position=position_strand(), colour="black") + 
geom_gene(aes(fill=cluster_id, colour=cluster_id), size=7) + 
geom_gene_tag(aes(label=parent_ids), vjust = -1, hjust = -0.15, size = 2) + 
geom_link_line(data=links(.track_id=2), color="black", linetype=2) +
scale_fill_manual("cluster", values=getPalette(17), na.value="darkgrey") + 
scale_color_manual("cluster", values=getPalette(17), na.value="darkgrey")
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      <em>
                       geom_feat
                      </em>
                      , these are the repeat regions positioned by +/- strand and coloured black
                      <br/>
                      <em>
                       geom_gene
                      </em>
                      , these are the genes with border and fill according to
                      <em>
                       cluster_id
                      </em>
                      from
                      <em>
                       my_clusts_sec
                      </em>
                      <br/>
                      <em>
                       geom_gene_tag
                      </em>
                      , gene labels from
                      <em>
                       my_seqs
                      </em>
                      using
                      <em>
                       parent_id
                      </em>
                      attribute so trancript number isn't shown (see step 6)
                      <br/>
                      <em>
                       geom_link_line
                      </em>
                      , our custom function (step 13) to link genes with lines instead of polygons
                      <br/>
                      <em>
                       scale_fill_manual
                      </em>
                      , assign fill colour to clusters
                      <br/>
                      <em>
                       scale_color_manual
                      </em>
                      , assign border colour to clusters (matching fill)
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     <strong>
                      15. shift bins for optimal display and save to final plot
                      <em>
                       synplot4
                      </em>
                     </strong>
                     <br/>
                     by default, the seq bins are left-justified but this can make the links look stretched; we can automatically centre the bins if desired
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="synplot4 &lt;- synplot3 %&gt;% shift(center=TRUE)">
                     <pre class="notranslate"><code>synplot4 &lt;- synplot3 %&gt;% shift(center=TRUE)
</code></pre>
                    </div>
                    <p dir="auto">
                     alternatively, it is easy to manually shift bins horizontally (this may take trial-and-error but plot scale bar can be used as an aid)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="synplot4 &lt;- synplot3 %&gt;% shift(bins=c(1,2,4,6), by=c(50000, 50000, 25000, 50000))">
                     <pre class="notranslate"><code>synplot4 &lt;- synplot3 %&gt;% shift(bins=c(1,2,4,6), by=c(50000, 50000, 25000, 50000))
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://private-user-images.githubusercontent.com/33470968/269471307-a8bb91a2-ff44-4775-8a22-a59f2867fcef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzEzMDctYThiYjkxYTItZmY0NC00Nzc1LThhMjItYTU5ZjI4NjdmY2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViYmM4NzhmYjE3MTgxN2Y1NWY1NDhhZDE0YWQ0OWM0OWQ0MTA0Y2EzMDJmYzg1YzI4OTY5ODNhNTE4YjhjNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DF5jF98Sf6zdbd4uAMEHRqboMvusyc7BAwVmQPFBfrA" rel="noopener noreferrer" target="_blank">
                      <img alt="synplot_exp5_small" src="https://private-user-images.githubusercontent.com/33470968/269471307-a8bb91a2-ff44-4775-8a22-a59f2867fcef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTg2MzIsIm5iZiI6MTcxMTY1ODMzMiwicGF0aCI6Ii8zMzQ3MDk2OC8yNjk0NzEzMDctYThiYjkxYTItZmY0NC00Nzc1LThhMjItYTU5ZjI4NjdmY2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDIwMzg1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViYmM4NzhmYjE3MTgxN2Y1NWY1NDhhZDE0YWQ0OWM0OWQ0MTA0Y2EzMDJmYzg1YzI4OTY5ODNhNTE4YjhjNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DF5jF98Sf6zdbd4uAMEHRqboMvusyc7BAwVmQPFBfrA" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     now we are happy with our synteny plot, we can save it as an SVG by invoking
                     <em>
                      svglite
                     </em>
                     within
                     <em>
                      ggsave
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='ggsave(file="MTAR_6_synteny_plot.svg", device=svg, plot=synplot4, width=12, height=6)'>
                     <pre class="notranslate"><code>ggsave(file="MTAR_6_synteny_plot.svg", device=svg, plot=synplot4, width=12, height=6)
</code></pre>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      in the above plot, only gene labels for the first bin are shown for simplicity, coordinates were manually added to the labels for
                      <em>
                       chr6_A
                      </em>
                      ,
                      <em>
                       chr6_B
                      </em>
                      and
                      <em>
                       020F
                      </em>
                      , some superfluous minor homolgy links were removed for cleaner visuals, and scale bar was reduced to show just one increment. The benefit of saving as SVG is that small adjustments and fixes like these are easy to do in a free SVG editor like
                      <a href="https://inkscape.org/" rel="nofollow">
                       Inkscape
                      </a>
                      <br/>
                     </p>
                    </blockquote>
                    <br/>
                    <p dir="auto">
                     more plotting capabilities can be found in the
                     <em>
                      gggenomes
                     </em>
                     <a href="https://thackl.github.io/gggenomes/articles/emales.html" rel="nofollow">
                      tutorial
                     </a>
                     pages by
                     <a href="https://github.com/thackl">
                      Thomas Hackl
                     </a>
                     .
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="EcGrHPzIfR/tj1tZdTqfLAs/eGChrJe+nWb2/n606Psn5LH/rsBN3IaWo5eaPmBHMF1IJsbzaE1XVU0rOYjdrg==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Custom scripts and workflows used in the genomics and population analysis of oat crown rust (Puccinia coronata f. sp. avenae) preprint (bioRxiv 2023.09.18.557855). All steps performed in Linux environment or R.
                </p>
                <div class="my-3 d-flex flex-items-center">
                 <svg aria-hidden="true" class="octicon octicon-link flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                  </path>
                 </svg>
                 <span class="flex-auto min-width-0 css-truncate css-truncate-target width-fit">
                  <a class="text-bold" href="https://doi.org/10.1101/2023.09.18.557855" rel="noopener noreferrer nofollow" role="link" target="_blank" title="https://doi.org/10.1101/2023.09.18.557855">
                   doi.org/10.1101/2023.09.18.557855
                  </a>
                 </span>
                </div>
                <h3 class="sr-only">
                 Topics
                </h3>
                <div class="my-3">
                 <div class="f6">
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:gwas" data-view-component="true" href="/topics/gwas" title="Topic: gwas">
                   gwas
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:synteny" data-view-component="true" href="/topics/synteny" title="Topic: synteny">
                   synteny
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:genomic-data-analysis" data-view-component="true" href="/topics/genomic-data-analysis" title="Topic: genomic-data-analysis">
                   genomic-data-analysis
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:snp-genotyping" data-view-component="true" href="/topics/snp-genotyping" title="Topic: snp-genotyping">
                   snp-genotyping
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:ngs-analysis" data-view-component="true" href="/topics/ngs-analysis" title="Topic: ngs-analysis">
                   ngs-analysis
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:read-alignment" data-view-component="true" href="/topics/read-alignment" title="Topic: read-alignment">
                   read-alignment
                  </a>
                 </div>
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/TC-Hewitt/OatCrownRust/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TC-Hewitt/OatCrownRust/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TC-Hewitt/OatCrownRust/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TC-Hewitt/OatCrownRust/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TC-Hewitt/OatCrownRust/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FTC-Hewitt%2FOatCrownRust&amp;report=TC-Hewitt+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/TC-Hewitt/OatCrownRust/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/TC-Hewitt/packages?repo_name=OatCrownRust">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/TC-Hewitt/OatCrownRust/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Python 100.0" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#3572A5 !important;;width: 100.0%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/TC-Hewitt/OatCrownRust/search?l=python">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#3572A5;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Python
                  </span>
                  <span>
                   100.0%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
