<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light-0eace2597ca3.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark-a167e256da9c.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed-d11f2cf8009b.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast-ea7373db06c8.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind-afa99dcf40f7.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind-af6c685139ba.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast-578cdbc8a5a9.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia-5cb699a7e247.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia-9b32204967c6.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ca8ae01e33b4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-fb4b8d40f206.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         GitHub - TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="0727:1B9964:C402D2:11FE996:66059FCC" data-pjax-transient="true" name="request-id">
         <meta content="4b888b49b71a013bb4fd18dfe7b6c11b4d25fb45698c123a6e3fd2c966ff90bc" data-pjax-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6IiIsInJlcXVlc3RfaWQiOiIwNzI3OjFCOTk2NDpDNDAyRDI6MTFGRTk5Njo2NjA1OUZDQyIsInZpc2l0b3JfaWQiOiI2NDUyMjQxODU2MjM4ODE3ODA3IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-pjax-transient="true" name="visitor-payload">
           <meta content="da693a14b6ba49634ce7d765d1844d4116984003dd8f145195d41252a6720a24" data-pjax-transient="true" name="visitor-hmac">
            <meta content="repository:414932614" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
               <meta content="" name="user-login"/>
               <meta content="width=device-width" name="viewport"/>
               <meta content="Contribute to TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii development by creating an account on GitHub." name="description"/>
               <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
               <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
               <meta content="1401488693436528" property="fb:app_id"/>
               <meta content="app-id=1477376905, app-argument=https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" name="apple-itunes-app">
                <meta content="https://opengraph.githubassets.com/f1c1f3fc666c0742b23fe9532f8172e43b3616a77d711797917162185d9ea64d/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" name="twitter:image:src">
                 <meta content="@github" name="twitter:site">
                  <meta content="summary_large_image" name="twitter:card">
                   <meta content="GitHub - TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" name="twitter:title">
                    <meta content="Contribute to TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii development by creating an account on GitHub." name="twitter:description">
                     <meta content="https://opengraph.githubassets.com/f1c1f3fc666c0742b23fe9532f8172e43b3616a77d711797917162185d9ea64d/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" property="og:image">
                      <meta content="Contribute to TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii development by creating an account on GitHub." property="og:image:alt">
                       <meta content="1200" property="og:image:width">
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="GitHub - TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" property="og:title"/>
                        <meta content="https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" property="og:url"/>
                        <meta content="Contribute to TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii development by creating an account on GitHub." property="og:description"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="584826db69b6459b3c4c29172349cd9ed6729235b962a93107214b2469aa5b40" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="a2a6f0b866a9de2e69b1f990e0730e22d502d619ef6d55aae603cacc6e2f5285" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii git https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii.git" name="go-import"/>
                        <meta content="74647948" name="octolytics-dimension-user_id">
                         <meta content="TommasoBarberis" name="octolytics-dimension-user_login">
                          <meta content="414932614" name="octolytics-dimension-repository_id">
                           <meta content="TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="414932614" name="octolytics-dimension-repository_network_root_id">
                               <meta content="TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" rel="canonical"/>
                                <meta content="logged-out env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-out env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-out env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="px-2 py-4 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-94fd67-11f6759e1cef.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/sessions-694c8423e347.js" type="application/javascript">
    </script>
    <header class="Header-old header-logged-out js-details-container Details position-relative f4 py-3" data-color-mode="light" data-dark-theme="dark" data-light-theme="light" role="banner">
     <button aria-label="Toggle navigation" class="Header-backdrop d-lg-none border-0 position-fixed top-0 left-0 width-full height-full js-details-target" type="button">
      <span class="d-none">
       Toggle navigation
      </span>
     </button>
     <div class="d-flex flex-column flex-lg-row flex-items-center p-responsive height-full position-relative z-1">
      <div class="d-flex flex-justify-between flex-items-center width-full width-lg-auto">
       <a aria-label="Homepage" class="mr-lg-3 color-fg-inherit flex-order-2" data-ga-click="(Logged out) Header, go to homepage, icon:logo-wordmark" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="flex-1">
        <a class="d-inline-block d-lg-none flex-order-1 f5 no-underline border color-border-default rounded-2 px-2 py-1 color-fg-inherit" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="8c7892079bc2c642b14a5647f71c867a5690ac091f489a2fa7311cefc0566cc3" href="/login?return_to=https%3A%2F%2Fgithub.com%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii">
         Sign in
        </a>
       </div>
       <div class="flex-1 flex-order-2 text-right">
        <button aria-expanded="false" aria-label="Toggle navigation" class="js-details-target Button--link Button--medium Button d-lg-none color-fg-inherit p-1" data-view-component="true" type="button">
         <span class="Button-content">
          <span class="Button-label">
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
           <div class="HeaderMenu-toggle-bar rounded my-1">
           </div>
          </span>
         </span>
        </button>
       </div>
      </div>
      <div class="HeaderMenu--logged-out p-responsive height-fit position-lg-relative d-lg-flex flex-column flex-auto pt-7 pb-4 top-0">
       <div class="header-menu-wrapper d-flex flex-column flex-self-end flex-lg-row flex-justify-between flex-auto p-3 p-lg-0 rounded rounded-lg-0 mt-3 mt-lg-0">
        <nav aria-label="Global" class="mt-0 px-3 px-lg-0 mb-3 mb-lg-0">
         <ul class="d-lg-flex list-style-none">
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Product
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 d-lg-flex dropdown-menu-wide">
            <div class="px-lg-4 border-lg-right mb-4 mb-lg-0 pr-lg-7">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Actions","label":"ref_cta:Actions;"}' href="/features/actions">
                <svg aria-hidden="true" class="octicon octicon-workflow color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1 3a2 2 0 0 1 2-2h6.5a2 2 0 0 1 2 2v6.5a2 2 0 0 1-2 2H7v4.063C7 16.355 7.644 17 8.438 17H12.5v-2.5a2 2 0 0 1 2-2H21a2 2 0 0 1 2 2V21a2 2 0 0 1-2 2h-6.5a2 2 0 0 1-2-2v-2.5H8.437A2.939 2.939 0 0 1 5.5 15.562V11.5H3a2 2 0 0 1-2-2Zm2-.5a.5.5 0 0 0-.5.5v6.5a.5.5 0 0 0 .5.5h6.5a.5.5 0 0 0 .5-.5V3a.5.5 0 0 0-.5-.5ZM14.5 14a.5.5 0 0 0-.5.5V21a.5.5 0 0 0 .5.5H21a.5.5 0 0 0 .5-.5v-6.5a.5.5 0 0 0-.5-.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Actions
                 </div>
                 Automate any workflow
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Packages","label":"ref_cta:Packages;"}' href="/features/packages">
                <svg aria-hidden="true" class="octicon octicon-package color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12.876.64V.639l8.25 4.763c.541.313.875.89.875 1.515v9.525a1.75 1.75 0 0 1-.875 1.516l-8.25 4.762a1.748 1.748 0 0 1-1.75 0l-8.25-4.763a1.75 1.75 0 0 1-.875-1.515V6.917c0-.625.334-1.202.875-1.515L11.126.64a1.748 1.748 0 0 1 1.75 0Zm-1 1.298L4.251 6.34l7.75 4.474 7.75-4.474-7.625-4.402a.248.248 0 0 0-.25 0Zm.875 19.123 7.625-4.402a.25.25 0 0 0 .125-.216V7.639l-7.75 4.474ZM3.501 7.64v8.803c0 .09.048.172.125.216l7.625 4.402v-8.947Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Packages
                 </div>
                 Host and manage packages
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Security","label":"ref_cta:Security;"}' href="/features/security">
                <svg aria-hidden="true" class="octicon octicon-shield-check color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M16.53 9.78a.75.75 0 0 0-1.06-1.06L11 13.19l-1.97-1.97a.75.75 0 0 0-1.06 1.06l2.5 2.5a.75.75 0 0 0 1.06 0l5-5Z">
                 </path>
                 <path d="m12.54.637 8.25 2.675A1.75 1.75 0 0 1 22 4.976V10c0 6.19-3.771 10.704-9.401 12.83a1.704 1.704 0 0 1-1.198 0C5.77 20.705 2 16.19 2 10V4.976c0-.758.489-1.43 1.21-1.664L11.46.637a1.748 1.748 0 0 1 1.08 0Zm-.617 1.426-8.25 2.676a.249.249 0 0 0-.173.237V10c0 5.46 3.28 9.483 8.43 11.426a.199.199 0 0 0 .14 0C17.22 19.483 20.5 15.461 20.5 10V4.976a.25.25 0 0 0-.173-.237l-8.25-2.676a.253.253 0 0 0-.154 0Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Security
                 </div>
                 Find and fix vulnerabilities
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Codespaces","label":"ref_cta:Codespaces;"}' href="/features/codespaces">
                <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M3.5 3.75C3.5 2.784 4.284 2 5.25 2h13.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 18.75 13H5.25a1.75 1.75 0 0 1-1.75-1.75Zm-2 12c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v4a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75ZM5.25 3.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h13.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Zm-2 12a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25Z">
                 </path>
                 <path d="M10 17.75a.75.75 0 0 1 .75-.75h6.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Codespaces
                 </div>
                 Instant dev environments
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Copilot","label":"ref_cta:Copilot;"}' href="/features/copilot">
                <svg aria-hidden="true" class="octicon octicon-copilot color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M23.922 16.992c-.861 1.495-5.859 5.023-11.922 5.023-6.063 0-11.061-3.528-11.922-5.023A.641.641 0 0 1 0 16.736v-2.869a.841.841 0 0 1 .053-.22c.372-.935 1.347-2.292 2.605-2.656.167-.429.414-1.055.644-1.517a10.195 10.195 0 0 1-.052-1.086c0-1.331.282-2.499 1.132-3.368.397-.406.89-.717 1.474-.952 1.399-1.136 3.392-2.093 6.122-2.093 2.731 0 4.767.957 6.166 2.093.584.235 1.077.546 1.474.952.85.869 1.132 2.037 1.132 3.368 0 .368-.014.733-.052 1.086.23.462.477 1.088.644 1.517 1.258.364 2.233 1.721 2.605 2.656a.832.832 0 0 1 .053.22v2.869a.641.641 0 0 1-.078.256ZM12.172 11h-.344a4.323 4.323 0 0 1-.355.508C10.703 12.455 9.555 13 7.965 13c-1.725 0-2.989-.359-3.782-1.259a2.005 2.005 0 0 1-.085-.104L4 11.741v6.585c1.435.779 4.514 2.179 8 2.179 3.486 0 6.565-1.4 8-2.179v-6.585l-.098-.104s-.033.045-.085.104c-.793.9-2.057 1.259-3.782 1.259-1.59 0-2.738-.545-3.508-1.492a4.323 4.323 0 0 1-.355-.508h-.016.016Zm.641-2.935c.136 1.057.403 1.913.878 2.497.442.544 1.134.938 2.344.938 1.573 0 2.292-.337 2.657-.751.384-.435.558-1.15.558-2.361 0-1.14-.243-1.847-.705-2.319-.477-.488-1.319-.862-2.824-1.025-1.487-.161-2.192.138-2.533.529-.269.307-.437.808-.438 1.578v.021c0 .265.021.562.063.893Zm-1.626 0c.042-.331.063-.628.063-.894v-.02c-.001-.77-.169-1.271-.438-1.578-.341-.391-1.046-.69-2.533-.529-1.505.163-2.347.537-2.824 1.025-.462.472-.705 1.179-.705 2.319 0 1.211.175 1.926.558 2.361.365.414 1.084.751 2.657.751 1.21 0 1.902-.394 2.344-.938.475-.584.742-1.44.878-2.497Z">
                 </path>
                 <path d="M14.5 14.25a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Zm-5 0a1 1 0 0 1 1 1v2a1 1 0 0 1-2 0v-2a1 1 0 0 1 1-1Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Copilot
                 </div>
                 Write better code with AI
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Code review","label":"ref_cta:Code review;"}' href="/features/code-review">
                <svg aria-hidden="true" class="octicon octicon-code-review color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M10.3 6.74a.75.75 0 0 1-.04 1.06l-2.908 2.7 2.908 2.7a.75.75 0 1 1-1.02 1.1l-3.5-3.25a.75.75 0 0 1 0-1.1l3.5-3.25a.75.75 0 0 1 1.06.04Zm3.44 1.06a.75.75 0 1 1 1.02-1.1l3.5 3.25a.75.75 0 0 1 0 1.1l-3.5 3.25a.75.75 0 1 1-1.02-1.1l2.908-2.7-2.908-2.7Z">
                 </path>
                 <path d="M1.5 4.25c0-.966.784-1.75 1.75-1.75h17.5c.966 0 1.75.784 1.75 1.75v12.5a1.75 1.75 0 0 1-1.75 1.75h-9.69l-3.573 3.573A1.458 1.458 0 0 1 5 21.043V18.5H3.25a1.75 1.75 0 0 1-1.75-1.75ZM3.25 4a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 .75.75v3.19l3.72-3.72a.749.749 0 0 1 .53-.22h10a.25.25 0 0 0 .25-.25V4.25a.25.25 0 0 0-.25-.25Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Code review
                 </div>
                 Manage code changes
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center pb-lg-3" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Issues","label":"ref_cta:Issues;"}' href="/features/issues">
                <svg aria-hidden="true" class="octicon octicon-issue-opened color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M12 1c6.075 0 11 4.925 11 11s-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1ZM2.5 12a9.5 9.5 0 0 0 9.5 9.5 9.5 9.5 0 0 0 9.5-9.5A9.5 9.5 0 0 0 12 2.5 9.5 9.5 0 0 0 2.5 12Zm9.5 2a2 2 0 1 1-.001-3.999A2 2 0 0 1 12 14Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Issues
                 </div>
                 Plan and track work
                </div>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Discussions","label":"ref_cta:Discussions;"}' href="/features/discussions">
                <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-subtle mr-3" data-view-component="true" height="24" version="1.1" viewbox="0 0 24 24" width="24">
                 <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 14.25 14H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 15.543V14H1.75A1.75 1.75 0 0 1 0 12.25v-9.5C0 1.784.784 1 1.75 1ZM1.5 2.75v9.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-9.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Z">
                 </path>
                 <path d="M22.5 8.75a.25.25 0 0 0-.25-.25h-3.5a.75.75 0 0 1 0-1.5h3.5c.966 0 1.75.784 1.75 1.75v9.5A1.75 1.75 0 0 1 22.25 20H21v1.543a1.457 1.457 0 0 1-2.487 1.03L15.939 20H10.75A1.75 1.75 0 0 1 9 18.25v-1.465a.75.75 0 0 1 1.5 0v1.465c0 .138.112.25.25.25h5.5a.75.75 0 0 1 .53.22l2.72 2.72v-2.19a.75.75 0 0 1 .75-.75h2a.25.25 0 0 0 .25-.25v-9.5Z">
                 </path>
                </svg>
                <div>
                 <div class="color-fg-default h4">
                  Discussions
                 </div>
                 Collaborate outside of code
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="px-lg-4">
             <span class="d-block h4 color-fg-default my-1" id="product-explore-heading">
              Explore
             </span>
             <ul aria-labelledby="product-explore-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to All features","label":"ref_cta:All features;"}' href="/features">
                All features
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Documentation","label":"ref_cta:Documentation;"}' href="https://docs.github.com" target="_blank">
                Documentation
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to GitHub Skills","label":"ref_cta:GitHub Skills;"}' href="https://skills.github.com/" target="_blank">
                GitHub Skills
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Product","action":"click to go to Blog","label":"ref_cta:Blog;"}' href="https://github.blog" target="_blank">
                Blog
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Solutions
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-for-heading">
              For
             </span>
             <ul aria-labelledby="solutions-for-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Enterprise","label":"ref_cta:Enterprise;"}' href="/enterprise">
                Enterprise
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Teams","label":"ref_cta:Teams;"}' href="/team">
                Teams
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Startups","label":"ref_cta:Startups;"}' href="/enterprise/startups">
                Startups
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Education","label":"ref_cta:Education;"}' href="https://education.github.com" target="_blank">
                Education
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <span class="d-block h4 color-fg-default my-1" id="solutions-by-solution-heading">
              By Solution
             </span>
             <ul aria-labelledby="solutions-by-solution-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to CI/CD &amp;amp; Automation","label":"ref_cta:CI/CD &amp;amp; Automation;"}' href="/solutions/ci-cd/">
                CI/CD &amp; Automation
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevOps","label":"ref_cta:DevOps;"}' href="/solutions/devops/">
                DevOps
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to DevSecOps","label":"ref_cta:DevSecOps;"}' href="https://resources.github.com/devops/fundamentals/devsecops/" target="_blank">
                DevSecOps
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="solutions-resources-heading">
              Resources
             </span>
             <ul aria-labelledby="solutions-resources-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Learning Pathways","label":"ref_cta:Learning Pathways;"}' href="https://resources.github.com/learn/pathways/" target="_blank">
                Learning Pathways
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to White papers, Ebooks, Webinars","label":"ref_cta:White papers, Ebooks, Webinars;"}' href="https://resources.github.com/" target="_blank">
                White papers, Ebooks, Webinars
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Customer Stories","label":"ref_cta:Customer Stories;"}' href="/customer-stories">
                Customer Stories
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Solutions","action":"click to go to Partners","label":"ref_cta:Partners;"}' href="https://partner.github.com/" target="_blank">
                Partners
                <svg aria-hidden="true" class="octicon octicon-link-external HeaderMenu-external-icon color-fg-subtle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.75 2h3.5a.75.75 0 0 1 0 1.5h-3.5a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-3.5a.75.75 0 0 1 1.5 0v3.5A1.75 1.75 0 0 1 12.25 14h-8.5A1.75 1.75 0 0 1 2 12.25v-8.5C2 2.784 2.784 2 3.75 2Zm6.854-1h4.146a.25.25 0 0 1 .25.25v4.146a.25.25 0 0 1-.427.177L13.03 4.03 9.28 7.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.75-3.75-1.543-1.543A.25.25 0 0 1 10.604 1Z">
                 </path>
                </svg>
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <button aria-expanded="false" class="HeaderMenu-link border-0 width-full width-lg-auto px-0 px-lg-2 py-3 py-lg-2 no-wrap d-flex flex-items-center flex-justify-between js-details-target" type="button">
            Open Source
            <svg aria-hidden="true" class="octicon octicon-chevron-down HeaderMenu-icon ml-1" data-view-component="true" height="16" opacity="0.5" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M12.78 5.22a.749.749 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.06 0L3.22 6.28a.749.749 0 1 1 1.06-1.06L8 8.939l3.72-3.719a.749.749 0 0 1 1.06 0Z">
             </path>
            </svg>
           </button>
           <div class="HeaderMenu-dropdown dropdown-menu rounded m-0 p-0 py-2 py-lg-4 position-relative position-lg-absolute left-0 left-lg-n3 px-lg-4">
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to GitHub Sponsors","label":"ref_cta:GitHub Sponsors;"}' href="/sponsors">
                <div>
                 <div class="color-fg-default h4">
                  GitHub Sponsors
                 </div>
                 Fund open source developers
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="border-bottom pb-3 mb-3">
             <ul class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary d-flex flex-items-center" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to The ReadME Project","label":"ref_cta:The ReadME Project;"}' href="/readme">
                <div>
                 <div class="color-fg-default h4">
                  The ReadME Project
                 </div>
                 GitHub community articles
                </div>
               </a>
              </li>
             </ul>
            </div>
            <div class="">
             <span class="d-block h4 color-fg-default my-1" id="open-source-repositories-heading">
              Repositories
             </span>
             <ul aria-labelledby="open-source-repositories-heading" class="list-style-none f5">
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Topics","label":"ref_cta:Topics;"}' href="/topics">
                Topics
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Trending","label":"ref_cta:Trending;"}' href="/trending">
                Trending
               </a>
              </li>
              <li>
               <a class="HeaderMenu-dropdown-link lh-condensed d-block no-underline position-relative py-2 Link--secondary" data-analytics-event='{"category":"Header dropdown (logged out), Open Source","action":"click to go to Collections","label":"ref_cta:Collections;"}' href="/collections">
                Collections
               </a>
              </li>
             </ul>
            </div>
           </div>
          </li>
          <li class="HeaderMenu-item position-relative flex-wrap flex-justify-between flex-items-center d-block d-lg-flex flex-lg-nowrap flex-lg-items-center js-details-container js-header-menu-item">
           <a class="HeaderMenu-link no-underline px-0 px-lg-2 py-3 py-lg-2 d-block d-lg-inline-block" data-analytics-event='{"category":"Header menu top item (logged out)","action":"click to go to Pricing","label":"ref_cta:Pricing;"}' href="/pricing">
            Pricing
           </a>
          </li>
         </ul>
        </nav>
        <div class="d-lg-flex flex-items-center mb-3 mb-lg-0 text-center text-lg-left ml-3" style="">
         <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;esi:include src="/_esi/rails_csrf_token_form_hidden?r=qrvD4bRT%2F%2B4xAX48sXgNVNje9CIZ%2F4H8r%2B9gYZJwcuoXEQc29GkQBQJCTfq2fjsnv4KJr0BcJjvLniHfBd0k48xqKsMU82FKBSuahbGxlY%2BexyVqvGxcUpdaDWaARHMZYya0YOMcH7Po6Kr3wqZ8bRMuMtJiwkN81nDX%2BLhet4EygKkC%2FUHAavVUSjYfKk4Yl1fYVtvubFLxapNSpib9piVvmUUkr7irAx7m9uOPXMVrsDmL2Nk6W5VZPAwb6c7VVF%2BSUiYMgUR4Gyunh4DhceA2mZcpH9fKdydwnMaYmwaTawTgXpbnmPQOUSvZx6DK8IpJPiPiNb4ENKZ50j55nCnTAqbdGY1lqRRn4%2FGPezFAYbdYG3ZYlpoCCWnjvK%2Ffak30%2FTlNLgLlduFVhhQfKFUNfsf%2FTrtwa2UOCvsxQNU%2Bu51Lj8RmQXr4wNd9U6xb9bAK6aasancJKDBuXXY4b0Qdu24hWamKvBaoCwQ%2BpbFwvJh%2FDgQLQ3J0SodYwYoJbAdMLJrt7Ym1UDQfCQ6%2BxYcAGhjSq7VIgdBZINYzZfPLcKnlpkIHbaYKnuCSGMmFaYHpEQY66Ngh2aDTpjiL5O4rk%2BRCbnKEAFX4eQDu--Nj%2FdS5OZoitfAlXw--MLr7IN%2Fgv7%2FtAIz0YIBCHA%3D%3D" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="TommasoBarberis" data-current-repository="TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="rgYvQ4f5yqIaC8ND99e1lQnIghRFqpPW9sRE9-T6Lis5lqZnjHSN9V9yLUUZjjnUs3et3mdQXGoHQMRtBmFpoQ" data-header-redesign-enabled="false" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="false" data-max-custom-scopes="10" data-scope="repo:TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii">
          <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center mr-4 rounded" data-action="click:qbsearch-input#searchInputContainerClicked">
           <button autocapitalize="off" class="header-search-button placeholder input-button form-control d-flex flex-1 flex-self-stretch flex-items-center no-wrap width-full py-0 pl-2 pr-0 text-left border-0 box-shadow-none" data-action="click:qbsearch-input#handleExpand" data-hotkey="s,/" data-target="qbsearch-input.inputButton" placeholder="Search or jump to..." type="button">
            <div class="mr-2 color-fg-muted">
             <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
              </path>
             </svg>
            </div>
            <span class="flex-1" data-target="qbsearch-input.inputButtonText">
             Search or jump to...
            </span>
            <div class="d-flex" data-target="qbsearch-input.hotkeyIndicator">
             <svg aria-hidden="true" class="mr-1" height="20" width="22" xmlns="http://www.w3.org/2000/svg">
              <path d="M3.5.5h12c1.7 0 3 1.3 3 3v13c0 1.7-1.3 3-3 3h-12c-1.7 0-3-1.3-3-3v-13c0-1.7 1.3-3 3-3z" fill="none" opacity=".4" stroke="#979A9C">
              </path>
              <path d="M11.8 6L8 15.1h-.9L10.8 6h1z" fill="#979A9C">
              </path>
             </svg>
            </div>
           </button>
           <input class="js-site-search-type-field" name="type" type="hidden"/>
           <div class="Overlay--hidden" data-modal-dialog-overlay="">
            <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-large Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
             <h1 class="sr-only" id="search-suggestions-dialog-header">
              Search code, repositories, users, issues, pull requests...
             </h1>
             <div class="Overlay-body Overlay-body--paddingNone">
              <div data-view-component="true">
               <div class="search-suggestions position-fixed width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                 <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                  <div class="FormControl FormControl--fullWidth">
                   <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                    Search
                   </label>
                   <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                    <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                     <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                      </path>
                     </svg>
                    </span>
                    <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                     <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                     </div>
                     <div class="QueryBuilder-InputWrapper">
                      <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                      </div>
                      <input aria-describedby="validation-e5fb8f6e-ceb4-4a13-82bc-b533df87ff28" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                      </input>
                     </div>
                    </div>
                    <span class="sr-only" id="query-builder-test-clear">
                     Clear
                    </span>
                    <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                     <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                      </path>
                     </svg>
                    </button>
                   </div>
                   <template id="search-icon">
                    <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-icon">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="file-code-icon">
                    <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                     </path>
                    </svg>
                   </template>
                   <template id="history-icon">
                    <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                     </path>
                    </svg>
                   </template>
                   <template id="repo-icon">
                    <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="bookmark-icon">
                    <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="plus-circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="circle-icon">
                    <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                     </path>
                    </svg>
                   </template>
                   <template id="trash-icon">
                    <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="team-icon">
                    <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="project-icon">
                    <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                     </path>
                    </svg>
                   </template>
                   <template id="pencil-icon">
                    <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </template>
                   <template id="copilot-icon">
                    <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                     </path>
                     <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="workflow-icon">
                    <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="book-icon">
                    <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-review-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </template>
                   <template id="codespaces-icon">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="comment-discussion-icon">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </template>
                   <template id="organization-icon">
                    <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="rocket-icon">
                    <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="shield-check-icon">
                    <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="heart-icon">
                    <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="server-icon">
                    <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                     </path>
                    </svg>
                   </template>
                   <template id="globe-icon">
                    <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                     </path>
                    </svg>
                   </template>
                   <template id="issue-opened-icon">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-mobile-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </template>
                   <template id="package-icon">
                    <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                     </path>
                    </svg>
                   </template>
                   <template id="credit-card-icon">
                    <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                     </path>
                     <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                     </path>
                    </svg>
                   </template>
                   <template id="play-icon">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </template>
                   <template id="gift-icon">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </template>
                   <template id="code-square-icon">
                    <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </template>
                   <template id="device-desktop-icon">
                    <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                     </path>
                    </svg>
                   </template>
                   <div class="position-relative">
                    <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                    </ul>
                   </div>
                   <div class="FormControl-inlineValidation" hidden="hidden" id="validation-e5fb8f6e-ceb4-4a13-82bc-b533df87ff28">
                    <span class="FormControl-inlineValidation--visual">
                     <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                      <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                      </path>
                     </svg>
                    </span>
                    <span>
                    </span>
                   </div>
                  </div>
                  <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                  </div>
                 </query-builder>
                </form>
                <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                 <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                  Search syntax tips
                 </a>
                 <div class="d-flex flex-1">
                 </div>
                </div>
               </div>
              </div>
             </div>
            </modal-dialog>
           </div>
          </div>
          <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
          </div>
          <div class="color-fg-default">
           <dialog-helper>
            <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
             <div class="Overlay-header" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="feedback-dialog-title">
                 Provide feedback
                </h1>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="feedback-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
                <input data-csrf="true" name="authenticity_token" type="hidden" value="nWPMLizzCiCdo711hRbp4SjbXsdHVwozjU1uDZaF24gjp9nYU/wsW9ei3AVU8jD5mWeF/cBMSuPEBIqIRfuyuA=="/>
                <p>
                 We read every piece of feedback, and take your input very seriously.
                </p>
                <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
                <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
                <label for="include_email" style="font-weight: normal">
                 Include my email address so I can be contacted
                </label>
               </form>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
              <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
               Submit feedback
              </button>
             </div>
            </dialog>
           </dialog-helper>
           <custom-scopes data-target="qbsearch-input.customScopesManager">
            <dialog-helper>
             <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
              <div class="Overlay-header Overlay-header--divided" data-view-component="true">
               <div class="Overlay-headerContentWrap">
                <div class="Overlay-titleWrap">
                 <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                  Saved searches
                 </h1>
                 <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                  Use saved searches to filter your results more quickly
                 </h2>
                </div>
                <div class="Overlay-actionWrap">
                 <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </div>
               </div>
              </div>
              <scrollable-region data-labelled-by="custom-scopes-dialog-title">
               <div class="Overlay-body" data-view-component="true">
                <div data-target="custom-scopes.customScopesModalDialogFlash">
                </div>
                <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                  <input data-csrf="true" name="authenticity_token" type="hidden" value="M+0y4qCry5MemWXVd7AZ4U1Xz51p6uZGGn3ZFMn0IDO5apy+oIF2RFIhSj5OZUc5iD8sk2tStLo926xPp2eL3g==">
                   <div data-target="custom-scopes.customScopesModalDialogFlash">
                   </div>
                   <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                   <div class="form-group">
                    <label for="custom_scope_name">
                     Name
                    </label>
                    <auto-check required="" src="/search/custom_scopes/check_name">
                     <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                     <input data-csrf="true" type="hidden" value="EiaVv1wo3X0tKvOEzD5DVMwkx96GK2+SPw3QoRoaL1V/6D3jftouH6fHg35qbJVaGH8ApQsVXRjvIbIGZ19tNg==">
                     </input>
                    </auto-check>
                   </div>
                   <div class="form-group">
                    <label for="custom_scope_query">
                     Query
                    </label>
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                   </div>
                   <p class="text-small color-fg-muted">
                    To see all available qualifiers, see our
                    <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                     documentation
                    </a>
                    .
                   </p>
                  </input>
                 </form>
                </div>
                <div data-target="custom-scopes.manageCustomScopesForm">
                 <div data-target="custom-scopes.list">
                 </div>
                </div>
               </div>
              </scrollable-region>
              <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
               <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
                Cancel
               </button>
               <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
                Create saved search
               </button>
              </div>
             </dialog>
            </dialog-helper>
           </custom-scopes>
          </div>
         </qbsearch-input>
         <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="4On3XHpVv0uDwHxL31ZPyapO/EYTY9QjCpiY4tOkeAcW9DI6hsaPNUZt0tJtujvhHA50b7v+W2gUxCPYGNjUFw==">
          <div class="position-relative mr-lg-3 d-lg-inline-block">
           <a class="HeaderMenu-link HeaderMenu-link--sign-in flex-shrink-0 no-underline d-block d-lg-inline-block border border-lg-0 rounded rounded-lg-0 p-2 p-lg-0" data-ga-click="(Logged out) Header, clicked Sign in, text:sign-in" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="8c7892079bc2c642b14a5647f71c867a5690ac091f489a2fa7311cefc0566cc3" href="/login?return_to=https%3A%2F%2Fgithub.com%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii">
            Sign in
           </a>
          </div>
          <a class="HeaderMenu-link HeaderMenu-link--sign-up flex-shrink-0 d-none d-lg-inline-block no-underline border color-border-default rounded px-2 py-1" data-analytics-event='{"category":"Sign up","action":"click to sign up for account","label":"ref_page:/&lt;user-name&gt;/&lt;repo-name&gt;;ref_cta:Sign up;ref_loc:header logged out"}' data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"site header menu","repository_id":null,"auth_type":"SIGN_UP","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="8c7892079bc2c642b14a5647f71c867a5690ac091f489a2fa7311cefc0566cc3" href="/signup?ref_cta=Sign+up&amp;ref_loc=header+logged+out&amp;ref_page=%2F%3Cuser-name%3E%2F%3Crepo-name%3E&amp;source=header-repo&amp;source_repo=TommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii">
           Sign up
          </a>
         </input>
        </div>
       </div>
      </div>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-33bf9fab-f963-4b3c-9c2e-786ea40b1bcd" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-44caf349-c64f-44d5-a89d-aa2ed3880225" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-44caf349-c64f-44d5-a89d-aa2ed3880225" id="tooltip-33bf9fab-f963-4b3c-9c2e-786ea40b1bcd" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <include-fragment class="js-notification-shelf-include-fragment" data-base-src="https://github.com/notifications/beta/shelf">
   </include-fragment>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header" style="background-color: var(--page-header-bgColor, var(--color-page-header-bg));">
       <div class="d-flex flex-wrap flex-justify-end mb-3 px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal">
          <svg aria-hidden="true" class="octicon octicon-repo color-fg-muted mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
           </path>
          </svg>
          <span class="author flex-self-stretch" itemprop="author">
           <a class="url fn" data-hovercard-type="user" data-hovercard-url="/users/TommasoBarberis/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="/TommasoBarberis" rel="author">
            TommasoBarberis
           </a>
          </span>
          <span class="mx-1 flex-self-stretch color-fg-muted">
           /
          </span>
          <strong class="mr-2 flex-self-stretch" itemprop="name">
           <a data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii">
            Genomic-and-transcriptomic-evolution-of-D.-Suzukii
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1">
           Public
          </span>
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="e0f620aebbeea47a88e2f3e48b70b0eee946f4370d46e64d03648282e70ca54c" data-view-component="true" href="/login?return_to=%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </li>
          <li>
           <a class="btn-sm btn" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"repo details fork button","repository_id":414932614,"auth_type":"LOG_IN","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="e91b5652401adf0f01d11287416bc655d034460127ee917739e0ba31b3b41bc2" data-view-component="true" href="/login?return_to=%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii" icon="repo-forked" id="fork-button" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
             </path>
            </svg>
            Fork
            <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
             0
            </span>
           </a>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":414932614,"auth_type":"LOG_IN","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="f4a38956217bbfe541f9f5de3d82c3f868d4a2b5eff08778f5d453b4472cda96" data-view-component="true" href="/login?return_to=%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
             <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
              0
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div data-turbo-replace="" id="responsive-meta-container">
        <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
         <h3 class="sr-only">
          License
         </h3>
         <div class="mb-2">
          <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/LICENSE">
           <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
            </path>
           </svg>
           MIT license
          </a>
         </div>
         <div class="mb-3">
          <a class="Link--secondary no-underline mr-3" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/stargazers">
           <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           stars
          </a>
          <a class="Link--secondary no-underline mr-3" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/forks">
           <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
            </path>
           </svg>
           <span class="text-bold">
            0
           </span>
           forks
          </a>
          <a class="Link--secondary no-underline mr-3 d-inline-block" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/branches">
           <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
            </path>
           </svg>
           <span>
            Branches
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tags">
           <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
            </path>
           </svg>
           <span>
            Tags
           </span>
          </a>
          <a class="Link--secondary no-underline d-inline-block" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/activity">
           <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
            </path>
           </svg>
           <span>
            Activity
           </span>
          </a>
         </div>
         <div class="d-flex flex-wrap gap-2">
          <div class="flex-1">
           <div class="BtnGroup d-flex" data-view-component="true">
            <a aria-label="You must be signed in to star a repository" class="tooltipped tooltipped-s btn-sm btn btn-block BtnGroup-item" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"star button","repository_id":414932614,"auth_type":"LOG_IN","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="f4a38956217bbfe541f9f5de3d82c3f868d4a2b5eff08778f5d453b4472cda96" data-view-component="true" href="/login?return_to=%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii" rel="nofollow">
             <svg aria-hidden="true" class="octicon octicon-star v-align-text-bottom d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="d-inline" data-view-component="true">
              Star
             </span>
            </a>
            <button aria-label="You must be signed in to add this repository to a list" class="btn-sm btn BtnGroup-item px-2" data-view-component="true" disabled="disabled" type="button">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </button>
           </div>
          </div>
          <div class="flex-1">
           <a aria-label="You must be signed in to change notification settings" class="tooltipped tooltipped-s btn-sm btn btn-block" data-hydro-click='{"event_type":"authentication.click","payload":{"location_in_page":"notification subscription menu watch","repository_id":null,"auth_type":"LOG_IN","originating_url":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","user_id":null}}' data-hydro-click-hmac="e0f620aebbeea47a88e2f3e48b70b0eee946f4370d46e64d03648282e70ca54c" data-view-component="true" href="/login?return_to=%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii" rel="nofollow">
            <svg aria-hidden="true" class="octicon octicon-bell mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
             </path>
            </svg>
            Notifications
           </a>
          </div>
          <span>
          </span>
         </div>
        </div>
       </div>
       <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav px-3 px-md-4 px-lg-5" data-pjax="#js-repo-pjax-container" data-view-component="true">
        <ul class="UnderlineNav-body list-style-none" data-view-component="true">
         <li class="d-inline-flex" data-view-component="true">
          <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" id="code-tab">
           <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
            </path>
           </svg>
           <span data-content="Code">
            Code
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/issues" id="issues-tab">
           <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
            </path>
           </svg>
           <span data-content="Issues">
            Issues
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/pulls" id="pull-requests-tab">
           <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
            </path>
           </svg>
           <span data-content="Pull requests">
            Pull requests
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/actions" id="actions-tab">
           <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
            </path>
           </svg>
           <span data-content="Actions">
            Actions
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/projects" id="projects-tab">
           <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
            </path>
           </svg>
           <span data-content="Projects">
            Projects
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
            0
           </span>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/security" id="security-tab">
           <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
            </path>
           </svg>
           <span data-content="Security">
            Security
           </span>
           <include-fragment accept="text/fragment+html" src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/security/overall-count">
           </include-fragment>
          </a>
         </li>
         <li class="d-inline-flex" data-view-component="true">
          <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/pulse" id="insights-tab">
           <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
            </path>
           </svg>
           <span data-content="Insights">
            Insights
           </span>
           <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
           </span>
          </a>
         </li>
        </ul>
        <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
         <action-menu data-select-variant="none" data-view-component="true">
          <focus-group direction="vertical" mnemonics="" retain="">
           <button aria-controls="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-list" aria-haspopup="true" aria-labelledby="tooltip-30c09cc4-d547-4a42-8249-ccd65f27d086" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-button" popovertarget="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-overlay" type="button">
            <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
             </path>
            </svg>
           </button>
           <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-button" id="tooltip-30c09cc4-d547-4a42-8249-ccd65f27d086" popover="manual">
            Additional navigation options
           </tool-tip>
           <anchored-position align="start" anchor="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-button" anchor-offset="normal" data-view-component="true" id="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-overlay" popover="auto" side="outside-bottom">
            <div class="Overlay Overlay--size-auto" data-view-component="true">
             <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
              <action-list>
               <div data-view-component="true">
                <ul aria-labelledby="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-8e6ce352-54b1-48d0-a683-353c968b24bc-list" role="menu">
                 <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii" id="item-2b30eaa4-7587-434e-a250-1c681e779173" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Code
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/issues" id="item-5f074e4b-3908-47d5-8176-4ad1d82302c8" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/pulls" id="item-313de23a-5e9e-47ac-b36e-11b307283cd8" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/actions" id="item-69198d85-8134-4547-ac8a-1ccafa1451cb" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Actions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/projects" id="item-6a52eb47-8e11-4496-b87e-1528e7448696" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/security" id="item-f6aebc9a-5a74-4d40-a730-918c7489253f" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Security
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                  <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/pulse" id="item-2704b95c-9226-48f5-b3f0-a4b2470bd0df" role="menuitem" tabindex="-1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Insights
                   </span>
                  </a>
                 </li>
                </ul>
               </div>
              </action-list>
             </div>
            </div>
           </anchored-position>
          </focus-group>
         </action-menu>
        </div>
       </nav>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <h1 class="sr-only">
         TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/spoofed_commit_check/f98ba06d69d1e01a39c0946a919b1e7c01c7e2fd">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":false,"path":"/","repo":{"id":414932614,"defaultBranch":"main","name":"Genomic-and-transcriptomic-evolution-of-D.-Suzukii","ownerLogin":"TommasoBarberis","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-10-08T09:53:54.000Z","ownerAvatar":"https://avatars.githubusercontent.com/u/74647948?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":null,"refInfo":{"name":"main","listCacheKey":"v0:1639038639.3231108","canEdit":false,"refType":"branch","currentOid":"f98ba06d69d1e01a39c0946a919b1e7c01c7e2fd"},"tree":{"items":[{"name":"TE_analysis","path":"TE_analysis","contentType":"directory"},{"name":"img","path":"img","contentType":"directory"},{"name":"quality_control","path":"quality_control","contentType":"directory"},{"name":"scripts","path":"scripts","contentType":"directory"},{"name":".gitattributes","path":".gitattributes","contentType":"file"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"Book of specifications.pdf","path":"Book of specifications.pdf","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":9,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":null,"httpUrl":"https://github.com/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii.git","showCloneWarning":null,"sshUrl":null,"sshCertificatesRequired":null,"sshCertificatesAvailable":null,"ghCliUrl":"gh repo clone TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"https://desktop.github.com","showVisualStudioCloneButton":false,"visualStudioCloneUrl":"https://windows.github.com","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/archive/refs/heads/main.zip"}},"newCodespacePath":"/codespaces/new?hide_repo_select=true\u0026repo=414932614"},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"62","overviewFiles":[{"displayName":"README.md","repoName":"Genomic-and-transcriptomic-evolution-of-D.-Suzukii","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenomic and transcriptomic evolution of \u003cem\u003eD. suzukii\u003c/em\u003e\u003c/h1\u003e\u003ca id=\"user-content-genomic-and-transcriptomic-evolution-of-d-suzukii\" class=\"anchor\" aria-label=\"Permalink: Genomic and transcriptomic evolution of D. suzukii\" href=\"#genomic-and-transcriptomic-evolution-of-d-suzukii\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTable of contents\u003c/h2\u003e\u003ca id=\"user-content-table-of-contents\" class=\"anchor\" aria-label=\"Permalink: Table of contents\" href=\"#table-of-contents\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#1-context\"\u003eContext\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#2-introduction-to-the-project\"\u003eIntroduction to the project\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#3-data\"\u003eData\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#4-snp-calling-workflow\"\u003eSNP Calling Workflow\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#5-diversity-analysis\"\u003eDiversity Analysis\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#6-te-analysis\"\u003eTE Analysis\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#7-authors\"\u003eAuthors\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#8-references\"\u003eReferences\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1. Context\u003c/h2\u003e\u003ca id=\"user-content-1-context\" class=\"anchor\" aria-label=\"Permalink: 1. Context\" href=\"#1-context\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSince its recent invasion in the European and American continents, \u003cem\u003eDrosophila suzukii\u003c/em\u003e, has become a major pest of berry crops.\nContrary to other \u003cem\u003eDrosophila\u003c/em\u003e species that develop themselves on damaged or rotten fruit, \u003cem\u003eD. suzukii\u003c/em\u003e is able to lay its eggs in healthy fruit before harvest, using its \u003cstrong\u003esclerotinized ovipositor\u003c/strong\u003e. In order to control the \u003cem\u003eD. suzukii\u003c/em\u003e population it is therefore required to improve our knowledge about its ability to adapt to a local environment outside its origin areas.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eA previous study focused on phenotipic changes during a selective experiment showed a temporal adaptation of \u003cem\u003eD. suzukii\u003c/em\u003e populations in the three different fruit environments. This process produces a \u003cstrong\u003efitness\u003c/strong\u003e improvement which corresponds to a greater ability to reproduce and could explain the adaptation abilities  of \u003cem\u003eD. suzukii\u003c/em\u003e. Based on the same experiment, a second study has been started by the LBBE to see if the \u003cem\u003eD. suzukii\u003c/em\u003e genome and transcriptome are affected by the environment.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe project is focused on both genes and Transposable Elements (TE) expression in order to find the variants explaining the local environment adaptation of \u003cem\u003eD. suzukii\u003c/em\u003e.\nA significant number of differentially expressed genes across generations has been found with \u003cstrong\u003eRNAseq analysis\u003c/strong\u003e between the G0 and the G7 generations. These variations seems to be independant of the fruit media. Moreover, in differentially expressed (DE) genes an increase of the number of Single Nucleotide Polymorphism (SNP) was found in G7 compared to G0, but this result was not expected. Natural selection generally keeps alleles which allow the best adaptation, and it is synonymous with a decrease in SNP number.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2. Introduction to the project\u003c/h2\u003e\u003ca id=\"user-content-2-introduction-to-the-project\" class=\"anchor\" aria-label=\"Permalink: 2. Introduction to the project\" href=\"#2-introduction-to-the-project\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.1 Polymorphism analysis\u003c/h3\u003e\u003ca id=\"user-content-21-polymorphism-analysis\" class=\"anchor\" aria-label=\"Permalink: 2.1 Polymorphism analysis\" href=\"#21-polymorphism-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe goal here is to analyse \u003cem\u003eD. suzukii\u003c/em\u003e polymorphism performing a \u003cstrong\u003evariant calling\u003c/strong\u003e on the whole genome (using poolseq data) in order to count the number of different genomic positions between G0 and G12, and between fruits media. \u003cbr\u003e\nThe position of the genetic variations will be compared with previous SNP analysis conducted on RNA-seq data to see if we also find an increase of the SNP number. To have a view more global on the genome's diversity, several diversity measures will be calculated: \u003cstrong\u003epi's\u003c/strong\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://render.githubusercontent.com/render/math?math=\\theta\"\u003e\u003cimg src=\"https://render.githubusercontent.com/render/math?math=\\theta\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e, \u003cstrong\u003eWatterson's\u003c/strong\u003e \u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://render.githubusercontent.com/render/math?math=\\theta\"\u003e\u003cimg src=\"https://render.githubusercontent.com/render/math?math=\\theta\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e, \u003cstrong\u003eTajima's D\u003c/strong\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2 TE analysis\u003c/h3\u003e\u003ca id=\"user-content-22-te-analysis\" class=\"anchor\" aria-label=\"Permalink: 2.2 TE analysis\" href=\"#22-te-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTransposable elements\u003c/strong\u003e (\u003cstrong\u003eTE\u003c/strong\u003e) represent \u003cstrong\u003e47%\u003c/strong\u003e of the \u003cem\u003eD. suzukii\u003c/em\u003e genome and can be important in the adaptation process, interfering in gene regulation during evolution. The aim is to determine TEs age on our data and compute their abundance. Comparing abundance graphs between G0 and G12 will inform us if some recent TE insertions happened.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAll the project objectives are presented more precisely in the book of specifications file.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3. Data\u003c/h2\u003e\u003ca id=\"user-content-3-data\" class=\"anchor\" aria-label=\"Permalink: 3. Data\" href=\"#3-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eDNA poolseq fastq from \u003cem\u003eD.suzukii\u003c/em\u003e obtained with paired-end Illumina sequencing (2x100pb), with 40 individuals for each pool:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eOne poolseq for G0;\u003c/li\u003e\n\u003cli\u003eOne poolseq for each media in G12 (strawberry, cherry and cranberry).\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv align=\"center\" dir=\"auto\"\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"center\"\u003eFile name\u003c/th\u003e\n\u003cth align=\"center\"\u003eNumber of reads\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG0-MTP\u003c/td\u003e\n\u003ctd align=\"center\"\u003e242 990 252\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG12-cerise\u003c/td\u003e\n\u003ctd align=\"center\"\u003e214 319 262\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG12-cranb\u003c/td\u003e\n\u003ctd align=\"center\"\u003e261 528 888\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG12-fraise\u003c/td\u003e\n\u003ctd align=\"center\"\u003e216 028 586\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLocation: \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/data\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/data\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePartial assembly of \u003cem\u003eD. suzukii\u003c/em\u003e genome as reference (4 chromosomes, 542 contigs);\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eLocation: \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-contig.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-contig.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eAnnotation file at the format \u003ccode\u003egff3\u003c/code\u003e.\nLocation: \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-annotation-3-ws.gff3\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-annotation-3-ws.gff3\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4. SNP Calling Workflow\u003c/h2\u003e\u003ca id=\"user-content-4-snp-calling-workflow\" class=\"anchor\" aria-label=\"Permalink: 4. SNP Calling Workflow\" href=\"#4-snp-calling-workflow\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/schema snp calling (1).jpg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/schema snp calling (1).jpg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1 Quality control\u003c/h3\u003e\u003ca id=\"user-content-41-quality-control\" class=\"anchor\" aria-label=\"Permalink: 4.1 Quality control\" href=\"#41-quality-control\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFirst of all we checked the quality of the data, using \u003ccode\u003efastqc\u003c/code\u003e \u003cem\u003e(version v0.11.9)\u003c/em\u003e:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"fastqc sample.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efastqc sample.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLocation: \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/quality_control\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/quality_control\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen, we merged results from \u003ccode\u003efastqc\u003c/code\u003e using \u003ccode\u003emultiqc\u003c/code\u003e in the folder with \u003ccode\u003efastqc\u003c/code\u003e reports. The \u003ccode\u003ehtml\u003c/code\u003e report is available in the \u003ccode\u003equality_control\u003c/code\u003e folder. \u003cbr\u003e\nWe noticed a high \u003cstrong\u003eGC\u003c/strong\u003e rate in the reads, but it seems to be in accordance with the results obtained by other researchs on \u003cem\u003eD. suzukii\u003c/em\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2 Mapping\u003c/h3\u003e\u003ca id=\"user-content-42-mapping\" class=\"anchor\" aria-label=\"Permalink: 4.2 Mapping\" href=\"#42-mapping\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eGenome indexing:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bwa index \\\n    -p Drosophila-suzukii-contig.fasta.fai \\\n    -a bwtsw Drosophila-suzukii-contig.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebwa index \\\n    -p Drosophila-suzukii-contig.fasta.fai \\\n    -a bwtsw Drosophila-suzukii-contig.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-p\u003c/code\u003e: index name;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-a\u003c/code\u003e: index algorithm (\u003cem\u003ebwtsw\u003c/em\u003e for long genome).\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eMapping samples on the reference:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_1.fastq.gz \u0026gt; sample_1.sai\nbwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_2.fastq.gz \u0026gt; sample_2.sai\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_1.fastq.gz \u0026gt; sample_1.sai\nbwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_2.fastq.gz \u0026gt; sample_2.sai\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-t\u003c/code\u003e: number of threads.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eMerge the \u003ccode\u003e.sai\u003c/code\u003e files of forward and reverse reads:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bwa sampe Drosophila-suzukii-contig.fasta.fai \\\n    sample_1.sai sample_2.sai \\\n    sample_1.fastq.gz sample_2.fastq.gz \u0026gt; sample_pe.sam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebwa sampe Drosophila-suzukii-contig.fasta.fai \\\n    sample_1.sai sample_2.sai \\\n    sample_1.fastq.gz sample_2.fastq.gz \u0026gt; sample_pe.sam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003esampe\u003c/code\u003e: take in account to have paired ends reads.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMapping stats\u003c/h4\u003e\u003ca id=\"user-content-mapping-stats\" class=\"anchor\" aria-label=\"Permalink: Mapping stats\" href=\"#mapping-stats\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eComputation of mapped files statistics using:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"samtools stats sample.sam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esamtools stats sample.sam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"center\"\u003esample\u003c/th\u003e\n\u003cth align=\"center\"\u003eraw sequences\u003c/th\u003e\n\u003cth align=\"center\"\u003emapped and paired squences\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG0-MTP_pe\u003c/td\u003e\n\u003ctd align=\"center\"\u003e242495252\u003c/td\u003e\n\u003ctd align=\"center\"\u003e194326801\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG12-fraise_pe\u003c/td\u003e\n\u003ctd align=\"center\"\u003e216028586\u003c/td\u003e\n\u003ctd align=\"center\"\u003e172898366\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG12-cerise_pe\u003c/td\u003e\n\u003ctd align=\"center\"\u003e214319262\u003c/td\u003e\n\u003ctd align=\"center\"\u003e172440052\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eG12-cranb_pe\u003c/td\u003e\n\u003ctd align=\"center\"\u003e261528888\u003c/td\u003e\n\u003ctd align=\"center\"\u003e212135984\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConversion to \u003ccode\u003e.bam\u003c/code\u003e\u003c/h4\u003e\u003ca id=\"user-content-conversion-to-bam\" class=\"anchor\" aria-label=\"Permalink: Conversion to .bam\" href=\"#conversion-to-bam\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"samtools view -bS -@ 3 sample.sam \u0026gt; sample.bam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esamtools view -bS -@ 3 sample.sam \u0026gt; sample.bam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-bS\u003c/code\u003e: output in the \u003cstrong\u003eBAM\u003c/strong\u003e format ignoring for compatibility with previous samtools version;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-@\u003c/code\u003e: number of threads.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eLocation: \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/mapping_bwa\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/mapping_bwa\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNB:\u003c/strong\u003e Before using \u003ccode\u003ebwa\u003c/code\u003e, we tried to use \u003ccode\u003ehisat2\u003c/code\u003e but, for an unkown reason it could not map the strawberry sample.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3 Variant Calling\u003c/h3\u003e\u003ca id=\"user-content-43-variant-calling\" class=\"anchor\" aria-label=\"Permalink: 4.3 Variant Calling\" href=\"#43-variant-calling\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe SNP calling on pooled data is still not largely used because it is very innovative. For this reason it has required several tests (with different tools) and a long time to be completed. \u003cbr\u003e\nFinally, we used \u003ccode\u003eGATK\u003c/code\u003e to perform our SNP calling and for that we were inspired by M. Tabourin's workflow (\u003ca href=\"https://github.com/mtabourin/Analyse-experience-d-evolution-experimentale-D.suzukii\"\u003eAnalyse experience d'évolution expérimentale \u003cem\u003eD.suzukii\u003c/em\u003e\u003c/a\u003e) and from \u003ca href=\"https://yeamanlab.weebly.com/uploads/5/7/9/5/57959825/snp_calling_pipeline.pdf\" rel=\"nofollow\"\u003ethis\u003c/a\u003e document.\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eSorting \u003ccode\u003e.bam\u003c/code\u003e files:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# GATK version: gatk-4.2.2.0\ngatk SortSam -I sample.bam -O sample_sorted.bam -SO queryname -VALIDATION_STRINGENCY SILENT\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# GATK version: gatk-4.2.2.0\ngatk SortSam -I sample.bam -O sample_sorted.bam -SO queryname -VALIDATION_STRINGENCY SILENT\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-I\u003c/code\u003e: input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-SO\u003c/code\u003e: sort order of output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-VALIDATION_STRINGENCY\u003c/code\u003e: Validation stringency for all SAM files read by this program. Setting stringency to \u003ccode\u003eSILENT\u003c/code\u003e can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eAdd read group info:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gatk AddOrReplaceReadGroups \\\n    -I sample_sorted.bam \\\n    -O sample_ReadGroups.bam \\\n    -RGID name \\\n    -RGLB name \\\n    -RGPL ILLUMINA \\\n    -RGPU name \\\n    -RGSM sample \\\n    -VALIDATION_STRINGENCY SILENT\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egatk AddOrReplaceReadGroups \\\n    -I sample_sorted.bam \\\n    -O sample_ReadGroups.bam \\\n    -RGID name \\\n    -RGLB name \\\n    -RGPL ILLUMINA \\\n    -RGPU name \\\n    -RGSM sample \\\n    -VALIDATION_STRINGENCY SILENT\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-I\u003c/code\u003e: input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-RGID\u003c/code\u003e: Read-Group ID;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-RGLB\u003c/code\u003e: Read-Group library;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-RGPL\u003c/code\u003e: Read-Group platform;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-RGPU\u003c/code\u003e: Read-Group platform unit(barcode);\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-RGSM\u003c/code\u003e: Read-Group sample name;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-VALIDATION_STRINGENCY\u003c/code\u003e: Validation stringency for all SAM files read by this program.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNB:\u003c/strong\u003e Read group information flags (\u003ccode\u003e-RG*\u003c/code\u003e) are mandatory, even if you don't have them.\u003c/p\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eRemove duplicates:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gatk MarkDuplicatesSpark \\\n    -I sample_ReadGroups.bam \\\n    -O sample_MarkDuplicated.bam \\\n    --spark-master local[16]\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egatk MarkDuplicatesSpark \\\n    -I sample_ReadGroups.bam \\\n    -O sample_MarkDuplicated.bam \\\n    --spark-master local[16]\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-I\u003c/code\u003e: input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e--spark-master\u003c/code\u003e: to parallelize the work.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eDetect variant sites in the dataset:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gatk HaplotypeCaller \\\n    -R reference.fasta \\\n    -I sample_MarkDuplicated.bam \\\n    --sample-ploidy 80 -O sample.vcf \\\n    --max-genotype-count 91881\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egatk HaplotypeCaller \\\n    -R reference.fasta \\\n    -I sample_MarkDuplicated.bam \\\n    --sample-ploidy 80 -O sample.vcf \\\n    --max-genotype-count 91881\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-R\u003c/code\u003e: genome reference;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-I\u003c/code\u003e: input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file (\u003ccode\u003evcf\u003c/code\u003e format);\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e--sample-ploidy\u003c/code\u003e: it allows to take in account of using pooled data:\n\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://render.githubusercontent.com/render/math?math=sample\\_ploidy = organism\\_ploidy \\times pool\\_size\"\u003e\u003cimg src=\"https://render.githubusercontent.com/render/math?math=sample\\_ploidy = organism\\_ploidy \\times pool\\_size\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \u003cbr\u003e\nIn our case we have a ploidy of 2 (\u003cem\u003eD. suzukii\u003c/em\u003e is a diploïd organism) and a pool size of 40.\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e--max-genotype-count\u003c/code\u003e: maximum number of genotypes to consider at any site:\u003c/li\u003e\n\u003cli\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://render.githubusercontent.com/render/math?math=max\\_genotype\\_count = \\binom{P +  A - 1}{A - 1} = \\frac{(P + A - 1)!}{P!(A - 1)!}\"\u003e\u003cimg src=\"https://render.githubusercontent.com/render/math?math=max\\_genotype\\_count = \\binom{P +  A - 1}{A - 1} = \\frac{(P + A - 1)!}{P!(A - 1)!}\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e \u003cbr\u003e\nWith \u003ccode\u003eP\u003c/code\u003e la ploidy of the sample (previous formula) and \u003ccode\u003eA\u003c/code\u003e (\u003ca target=\"_blank\" rel=\"noopener noreferrer nofollow\" href=\"https://render.githubusercontent.com/render/math?math=A = 3\"\u003e\u003cimg src=\"https://render.githubusercontent.com/render/math?math=A = 3\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e) the allele count.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3.1 SNPs filtering\u003c/h4\u003e\u003ca id=\"user-content-431-snps-filtering\" class=\"anchor\" aria-label=\"Permalink: 4.3.1 SNPs filtering\" href=\"#431-snps-filtering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eCrete a subset keeping only \u003cstrong\u003eSNPs\u003c/strong\u003e:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gatk SelectVariants \\\n    -R reference_genome.fasta \\\n    -V input.vcf \\\n    -O output.vcf \\\n    --select-type-to-include SNP\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egatk SelectVariants \\\n    -R reference_genome.fasta \\\n    -V input.vcf \\\n    -O output.vcf \\\n    --select-type-to-include SNP\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-R\u003c/code\u003e: genome reference;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-V\u003c/code\u003e: \u003ccode\u003e.vcf\u003c/code\u003e input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e--select-type-to-include\u003c/code\u003e select a type of variant.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eGenerate a table that can be used to find filter thresholds:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\tgatk VariantsToTable \\\n\t\t-R reference_genome.fasta \\\n\t\t-V input.vcf \\\n\t\t-O output.table \\\n\t\t# scores to save to the table\n\t\t-F CHROM \\\n\t\t-F POS \\\n\t\t-F QUAL \\\n\t\t-F AC \\\n\t\t-F AF \\\n\t\t-F DP \\\n\t\t-F QD \\\n\t\t-F MQ \\\n\t\t-F FS \\\n\t\t-F SOR \\\n\t\t-F MQRankSum \\\n\t\t-F ReadPosRankSum\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\tgatk VariantsToTable \\\n\t\t-R reference_genome.fasta \\\n\t\t-V input.vcf \\\n\t\t-O output.table \\\n\t\t# scores to save to the table\n\t\t-F CHROM \\\n\t\t-F POS \\\n\t\t-F QUAL \\\n\t\t-F AC \\\n\t\t-F AF \\\n\t\t-F DP \\\n\t\t-F QD \\\n\t\t-F MQ \\\n\t\t-F FS \\\n\t\t-F SOR \\\n\t\t-F MQRankSum \\\n\t\t-F ReadPosRankSum\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-R\u003c/code\u003e: reference;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-V\u003c/code\u003e: \u003ccode\u003e.vcf\u003c/code\u003e input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-F\u003c/code\u003e: selected field.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThe best way to choose the right filter, is to plot the several scores across the VCF file, for example using \u003ccode\u003eggplot2\u003c/code\u003e library in \u003ccode\u003eR\u003c/code\u003e.\nYou can find here the plot used to define some scores: \u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/DP.svg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/DP.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/FS.svg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/FS.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/MQ.svg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/MQ.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/QD.svg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/QD.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/QUAL.svg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/QUAL.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/SOR.svg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/SOR.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cbr\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eUse filter on \u003ccode\u003evcf\u003c/code\u003e files:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\tgatk VariantFiltration \\\n\t\t-R reference_genome.fasta \\\n\t\t-V input.vcf \\\n\t\t-O output.vcf \\\n\t\t# following results observed in plots\n\t\t-filter \u0026quot;QUAL \u0026lt; 70\u0026quot; --filter-name Low_Qual\\\n\t\t-filter \u0026quot;DP \u0026lt; 90\u0026quot; --filter-name Low_Cov \\\n\t\t-filter \u0026quot;QD \u0026lt; 20.0 || MQ \u0026lt; 28.0 || FS \u0026gt; 1.0 || \\\n\t\tSOR \u0026gt; 1.4 --filter-name Secondary_filter\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\tgatk VariantFiltration \\\n\t\t-R reference_genome.fasta \\\n\t\t-V input.vcf \\\n\t\t-O output.vcf \\\n\t\t# following results observed in plots\n\t\t-filter \"QUAL \u0026lt; 70\" --filter-name Low_Qual\\\n\t\t-filter \"DP \u0026lt; 90\" --filter-name Low_Cov \\\n\t\t-filter \"QD \u0026lt; 20.0 || MQ \u0026lt; 28.0 || FS \u0026gt; 1.0 || \\\n\t\tSOR \u0026gt; 1.4 --filter-name Secondary_filter\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-R\u003c/code\u003e: reference;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-V\u003c/code\u003e: \u003ccode\u003evcf\u003c/code\u003e input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-filter\u003c/code\u003e: define a filter;\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003eQUAL\u003c/code\u003e: Phred-scaled quality score for the assertion made in alternative;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eDP\u003c/code\u003e: read depth at this position for this sample;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eQD\u003c/code\u003e: this is the variant confidence (from the QUAL field) divided by the unfiltered depth of non-hom-ref samples;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eMQ\u003c/code\u003e: mapping quality;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eFS\u003c/code\u003e: this is the Phred-scaled probability that there is strand bias at the site;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eSOR\u003c/code\u003e: this is another way to estimate strand bias using a test similar to the symmetric odds ratio test;\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e--filter-name\u003c/code\u003e: used to assign a name to the filter.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eExtract \u003cstrong\u003eSNPs\u003c/strong\u003e who passed the filters:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gatk SelectVariants \\\n\t\t-R reference_genome.fasta \\\n\t\t-V input.vcf \\\n\t\t-O output.vcf \\\n\t\t--exclude-filtered\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egatk SelectVariants \\\n\t\t-R reference_genome.fasta \\\n\t\t-V input.vcf \\\n\t\t-O output.vcf \\\n\t\t--exclude-filtered\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-R\u003c/code\u003e: reference;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-V\u003c/code\u003e: \u003ccode\u003evcf\u003c/code\u003e input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-O\u003c/code\u003e: output file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e--exclude-filtered\u003c/code\u003e: exclude \u003cstrong\u003eSNPs\u003c/strong\u003e marked to be filtered.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3.2 Get \u003cstrong\u003eSNPs\u003c/strong\u003e that are in transcripts (exons)\u003c/h4\u003e\u003ca id=\"user-content-432-get-snps-that-are-in-transcripts-exons\" class=\"anchor\" aria-label=\"Permalink: 4.3.2 Get SNPs that are in transcripts (exons)\" href=\"#432-get-snps-that-are-in-transcripts-exons\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eFrom the \u003ccode\u003e.gff3\u003c/code\u003e file of the reference, we can extract a subset having only exons:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tail -n +2 Drosophila-suzukii-annotation-3-ws.gff3 | \\\nawk -F \u0026quot;\\t\u0026quot; '{if($3==\u0026quot;exon\u0026quot;) print $0}' \u0026gt; ~/exon.gff3\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etail -n +2 Drosophila-suzukii-annotation-3-ws.gff3 | \\\nawk -F \"\\t\" '{if($3==\"exon\") print $0}' \u0026gt; ~/exon.gff3\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003etail -n +2\u003c/code\u003e: to skip \u003ccode\u003egff3\u003c/code\u003e header;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eawk\u003c/code\u003e: if the third field has 'exon' tag, it print the line in the \u003ccode\u003eexon.gff3\u003c/code\u003e file.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eConvert the \u003ccode\u003e.gff3\u003c/code\u003e into \u003ccode\u003e.bed\u003c/code\u003e format:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gff2bed \u0026lt; exon.gff3 \u0026gt; exon.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egff2bed \u0026lt; exon.gff3 \u0026gt; exon.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eIntersect the \u003ccode\u003e.vcf\u003c/code\u003e file with the \u003ccode\u003e.bed\u003c/code\u003e file:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"intersectBed -a input.vcf -b exon.bed -header \u0026gt; output.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eintersectBed -a input.vcf -b exon.bed -header \u0026gt; output.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-a\u003c/code\u003e: \u003ccode\u003e.vcf\u003c/code\u003e input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-b\u003c/code\u003e: \u003ccode\u003e.bed\u003c/code\u003e input file;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-header\u003c/code\u003e: if the \u003ccode\u003e.bed\u003c/code\u003e file has header.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eCount the number of gene for a given sample:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003efor DNA data we have used \u003ccode\u003egene_count.py\u003c/code\u003e that can be founded in the \u003ccode\u003escripts/\u003c/code\u003e folder;\u003c/li\u003e\n\u003cli\u003efor the RNA data we have used the following command (car genes are used as chromosome/contig in these files):\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"grep -v \u0026quot;#\u0026quot; file.vcf | cut -f1 | sort | uniq | wc -l\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egrep -v \"#\" file.vcf | cut -f1 | sort | uniq | wc -l\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3.3  Get \u003cstrong\u003eSNPs\u003c/strong\u003e that are in DE genes\u003c/h4\u003e\u003ca id=\"user-content-433--get-snps-that-are-in-de-genes\" class=\"anchor\" aria-label=\"Permalink: 4.3.3  Get SNPs that are in DE genes\" href=\"#433--get-snps-that-are-in-de-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eList of the DE genes can be found here: \u003cbr\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLocation: \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/DE_gene\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/DE_gene\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis list is obtained by the M. Tabourin's work.\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eObtain a \u003ccode\u003e.gff3\u003c/code\u003e file containing only DE genes:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python3 get_de.py\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython3 get_de.py\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003eget_de.py\u003c/code\u003e: it is a script written by us. It allows, from the \u003ccode\u003egff3\u003c/code\u003e and the \u003ccode\u003elist.txt\u003c/code\u003e having DE genes, to obtain a \u003ccode\u003e.gff3\u003c/code\u003e file having only DE genes.\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eThen we have transformed the obtained \u003ccode\u003e.gff3\u003c/code\u003e file into \u003ccode\u003e.bed\u003c/code\u003e file using the command \u003ccode\u003e3\u003c/code\u003e at the previous paragraph and we have used it to obtain \u003cstrong\u003eSNPs\u003c/strong\u003e in DE genes using command \u003ccode\u003e4\u003c/code\u003e of the previous paragraph.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFinally we have compute the intersect of the \u003ccode\u003evcf\u003c/code\u003e files from DNA and RNA. Unfortunately, these \u003ccode\u003evcf\u003c/code\u003e files (DNA and RNA) were generated with a different reference, so we have wrote a script to convert genomic positions of the RNA \u003ccode\u003evcf\u003c/code\u003e files to DNA positions. It can be found in the \u003ccode\u003escripts/\u003c/code\u003e folder as \u003ccode\u003econvert_vcf_pos.py\u003c/code\u003e. Then we have run the following commands:\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Compress VCF files\nbgzip input_DNA_file.vcf\nbgzip input_RNA_file.vcf\n\n# Build indexes for compressed files\ntabix -p vcf input_DNA_file.vcf.gz\ntabix -p vcf input_RNA_file.vcf.gz\n\n# run the intersection between the two files\nbcftools isec -c snps -p output/ input_DNA_file.vcf.gz input_RNA_file.vcf.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Compress VCF files\nbgzip input_DNA_file.vcf\nbgzip input_RNA_file.vcf\n\n# Build indexes for compressed files\ntabix -p vcf input_DNA_file.vcf.gz\ntabix -p vcf input_RNA_file.vcf.gz\n\n# run the intersection between the two files\nbcftools isec -c snps -p output/ input_DNA_file.vcf.gz input_RNA_file.vcf.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3.4 Results\u003c/h4\u003e\u003ca id=\"user-content-434-results\" class=\"anchor\" aria-label=\"Permalink: 4.3.4 Results\" href=\"#434-results\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLocation of \u003ccode\u003e.vcf\u003c/code\u003e result files: \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# DNA\n/localdata/pandata/students/M2_projet_15/GATK_pipeline/gatk_ploidy \n# RNA\n/localdata/pandata/students/M2_projet_15/RNAseq/vcf \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# DNA\n/localdata/pandata/students/M2_projet_15/GATK_pipeline/gatk_ploidy \n# RNA\n/localdata/pandata/students/M2_projet_15/RNAseq/vcf \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3.4.1 SNPs founded on exons\u003c/h5\u003e\u003ca id=\"user-content-4341-snps-founded-on-exons\" class=\"anchor\" aria-label=\"Permalink: 4.3.4.1 SNPs founded on exons\" href=\"#4341-snps-founded-on-exons\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"center\"\u003e\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0\u003c/th\u003e\n\u003cth align=\"center\"\u003ecerise\u003c/th\u003e\n\u003cth align=\"center\"\u003efraise\u003c/th\u003e\n\u003cth align=\"center\"\u003egene count\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eDNA\u003c/td\u003e\n\u003ctd align=\"center\"\u003e59,055\u003c/td\u003e\n\u003ctd align=\"center\"\u003e50,077\u003c/td\u003e\n\u003ctd align=\"center\"\u003e52,243\u003c/td\u003e\n\u003ctd align=\"center\"\u003e9,837\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eRNA\u003c/td\u003e\n\u003ctd align=\"center\"\u003e28,150\u003c/td\u003e\n\u003ctd align=\"center\"\u003e29,823\u003c/td\u003e\n\u003ctd align=\"center\"\u003e31,052\u003c/td\u003e\n\u003ctd align=\"center\"\u003e5,248\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eIn the DNA data we find almost double compared to the RNA, but this can be easily explained by the number of genes that is also double. Nevertheless we observe a decrease from G0 to G12 for DNA, which is rather consistent with the basic hypothesis of having a selection effect.\nOn the other hand, in RNA, the number does not decrease.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.3.4.1 SNPs founded on exons of DE genes\u003c/h5\u003e\u003ca id=\"user-content-4341-snps-founded-on-exons-of-de-genes\" class=\"anchor\" aria-label=\"Permalink: 4.3.4.1 SNPs founded on exons of DE genes\" href=\"#4341-snps-founded-on-exons-of-de-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"center\"\u003e\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0\u003c/th\u003e\n\u003cth align=\"center\"\u003ecerise\u003c/th\u003e\n\u003cth align=\"center\"\u003efraise\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eDNA\u003c/td\u003e\n\u003ctd align=\"center\"\u003e536\u003c/td\u003e\n\u003ctd align=\"center\"\u003e415\u003c/td\u003e\n\u003ctd align=\"center\"\u003e492\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eRNA\u003c/td\u003e\n\u003ctd align=\"center\"\u003e17\u003c/td\u003e\n\u003ctd align=\"center\"\u003e36\u003c/td\u003e\n\u003ctd align=\"center\"\u003e51\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eintersect\u003c/td\u003e\n\u003ctd align=\"center\"\u003e1\u003c/td\u003e\n\u003ctd align=\"center\"\u003e8\u003c/td\u003e\n\u003ctd align=\"center\"\u003e7\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eOn differentially expressed genes, there are many more SNPs on the DNA than on the RNA. It is suspected that this may be due to having poolseq data for DNA.\nWe find again a decrease between G0 and G12 for DNA (is it significant?) but not between G0 and G7 for RNA.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe SNP founde both in RNA and DNA is on Contig 9. The 8 SNPs for the cherry are founded on Contig 9 (2), Contig 44 (5) and Contig 49 (1). The 7 SNPs for the strawberry are founded on Andromeda (1), Cepheus (1), Contig (5), Contig 9 (1), Contig 10 (1) and Contig 44 (2).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5. Diversity \u0026amp; Differentiation Analysis\u003c/h2\u003e\u003ca id=\"user-content-5-diversity--differentiation-analysis\" class=\"anchor\" aria-label=\"Permalink: 5. Diversity \u0026amp; Differentiation Analysis\" href=\"#5-diversity--differentiation-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInstallating PoPoolation: Diversity Analysis\u003c/h1\u003e\u003ca id=\"user-content-installating-popoolation-diversity-analysis\" class=\"anchor\" aria-label=\"Permalink: Installating PoPoolation: Diversity Analysis\" href=\"#installating-popoolation-diversity-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll the steps come from PoPoolation website (\u003ca href=\"https://sourceforge.net/p/popoolation/wiki/browse_pages/\" rel=\"nofollow\"\u003ehttps://sourceforge.net/p/popoolation/wiki/browse_pages/\u003c/a\u003e) and Dr. Kofler presentation (\u003ca href=\"https://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part1.pdf\" rel=\"nofollow\"\u003ehttps://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part1.pdf\u003c/a\u003e)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"svn checkout https://svn.code.sf.net/p/popoolation/code/trunk popoolation-code\ncd popoolation-code\nsvn update\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esvn checkout https://svn.code.sf.net/p/popoolation/code/trunk popoolation-code\ncd popoolation-code\nsvn update\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRemove low quality alignments\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time samtools view -q 20 -f 0x0002 -F 0x0004 -F 0x0008 -b G0_MarkDuplicated.bam \u0026gt; G0_q20_MarkDuplicated.bam #26m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime samtools view -q 20 -f 0x0002 -F 0x0004 -F 0x0008 -b G0_MarkDuplicated.bam \u0026gt; G0_q20_MarkDuplicated.bam #26m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-q 20\u003c/code\u003e only keep reads with a mapping quality higher than 20 (remove ambiguously aligned reads)\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-f 0x0002\u003c/code\u003e only keep proper pairs (remember flags from sam file)\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-F 0x0004\u003c/code\u003e remove reads that are not mapped\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-F 0x0008\u003c/code\u003e remove reads with an un-mapped mate\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eNote \u003ccode\u003e-f\u003c/code\u003e means only keep reads having the given flag and \u003ccode\u003e-F\u003c/code\u003e discard all reads having the given flag.\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eCreating a mpileup file\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam \u0026gt; G0.mpileup #20m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam \u0026gt; G0.mpileup #20m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-B\u003c/code\u003e disable BAQ computation (base alignment quality)\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-Q\u003c/code\u003e skip bases with base quality smaller than the given value\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-f\u003c/code\u003e path to reference sequence\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eMPILEUP FILE:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ecol1 reference chromosome\u003c/li\u003e\n\u003cli\u003ecol2 position\u003c/li\u003e\n\u003cli\u003ecol3 reference character\u003c/li\u003e\n\u003cli\u003ecol4 coverage\u003c/li\u003e\n\u003cli\u003ecol5 bases for the given position (’.’ identical to reference character -forward strand; ’,’ identical to reference - reverse strand)\u003c/li\u003e\n\u003cli\u003ecol6 quality for the bases\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eFiltering indels\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/identify-genomic-indel-regions.pl --indel-window 5 --min-count 2 --input G0.mpileup --output indels_G0.gtf #25m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/identify-genomic-indel-regions.pl --indel-window 5 --min-count 2 --input G0.mpileup --output indels_G0.gtf #25m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e–indel-window how many bases surrounding indels should be ignored\u003c/li\u003e\n\u003cli\u003e–min-count minimum count for calling an indel. Note that indels may be sequencing errors as well\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/filter-pileup-by-gtf.pl --input G0.mpileup --gtf indels_G0.gtf --output G0.idf.mpileup #20m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/filter-pileup-by-gtf.pl --input G0.mpileup --gtf indels_G0.gtf --output G0.idf.mpileup #20m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eSubsampling to uniform coverage\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eSeveral population genetic estimators are sensitive to sequencing errors. For example a very low Tajima’s D, usually indicative of a selective sweep, may be, as an artifact, frequently be found in highly covered regions because these regions have just more sequencing errors. To avoid these kinds of biases we recommend to subsample to an uniform coverage.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/subsample-pileup.pl --min-qual 20 --method withoutreplace --max-coverage 50 --fastq-type sanger --target-coverage 10 --input G0.idf.mpileup --output G0.ss10.idf.mpileup #343m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/subsample-pileup.pl --min-qual 20 --method withoutreplace --max-coverage 50 --fastq-type sanger --target-coverage 10 --input G0.idf.mpileup --output G0.ss10.idf.mpileup #343m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e–min-qual minimum base quality\u003c/li\u003e\n\u003cli\u003e–method method for subsampling, we recommend without replacement\u003c/li\u003e\n\u003cli\u003e–target-coverage which coverage should the resulting mpileup file have\u003c/li\u003e\n\u003cli\u003e–max-coverage the maximum allowed coverage, regions having higher coverages will be ignored (they may be copy number variations and lead to wrong SNPs)\u003c/li\u003e\n\u003cli\u003e–fastq-type (sanger means offset 33)\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eCalculating Tajima’s Pi\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/Variance-sliding.pl --fastq-type sanger --measure pi --input G0.ss10.idf.mpileup --min-count 2 --min-coverage 4 --max-coverage 10 --min-covered-fraction 0.5 --pool-size 80 --window-size 1000 --step-size 1000 --output G0.pi --snp-output G0.snps\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime perl ~/Documents/Projet_S3/Popoolation/popoolation-code/Variance-sliding.pl --fastq-type sanger --measure pi --input G0.ss10.idf.mpileup --min-count 2 --min-coverage 4 --max-coverage 10 --min-covered-fraction 0.5 --pool-size 80 --window-size 1000 --step-size 1000 --output G0.pi --snp-output G0.snps\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e–min-coverage –max-coverage: for subsampled files not important; should contain target\ncoverage, i.e.: 10\u003c/li\u003e\n\u003cli\u003e–min-covered-fraction minimum percentage of sites having sufficient coverage in the given window\u003c/li\u003e\n\u003cli\u003e–min-count minimum occurrence of allele for calling a SNP\u003c/li\u003e\n\u003cli\u003e–measure which population genetics measure should be computed (pi/theta/D)\u003c/li\u003e\n\u003cli\u003e–pool-size number of chromosomes (thus number of diploids times two)\u003c/li\u003e\n\u003cli\u003e–region compute the measure only for a small region; default is the whole genome\u003c/li\u003e\n\u003cli\u003e–output a file containing the measure (π) for the windows\u003c/li\u003e\n\u003cli\u003e–snp-output a file containing for every window the SNPs that have been used for computing the measure (e.g. π)\u003c/li\u003e\n\u003cli\u003e–window-size –step-size control behavior of sliding window; if step size is smaller than window size than the windows will be overlapping.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eOutput G0.pi:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ecol 1: reference chromosome\u003c/li\u003e\n\u003cli\u003ecol 2: position of window (mean value)\u003c/li\u003e\n\u003cli\u003ecol 3: number of SNPs in the given window\u003c/li\u003e\n\u003cli\u003ecol 4: fraction of sites in the window having sufficient coverage (min ≤ x ≤ max)\u003c/li\u003e\n\u003cli\u003ecol 5: measure for the window (π)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eOutput G0.snps:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ecol 1: reference chromosome\u003c/li\u003e\n\u003cli\u003ecol 2: position of SNP\u003c/li\u003e\n\u003cli\u003ecol 3: reference character\u003c/li\u003e\n\u003cli\u003ecol 4: coverage\u003c/li\u003e\n\u003cli\u003ecol 5-9: counts of A, T, C, G, N respectively\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eRESULTS:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv align=\"center\" dir=\"auto\"\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"center\"\u003eTheta Pi\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0\u003c/th\u003e\n\u003cth align=\"center\"\u003eG12 cranberry\u003c/th\u003e\n\u003cth align=\"center\"\u003eG12 cerise\u003c/th\u003e\n\u003cth align=\"center\"\u003eG12 fraise\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003emoyenne\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.00830\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.00771\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.00939\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.00864\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eécart type\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.01250\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.01289\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.01468\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.01418\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInstallating PoPoolation2: Differentiation Analysis\u003c/h1\u003e\u003ca id=\"user-content-installating-popoolation2-differentiation-analysis\" class=\"anchor\" aria-label=\"Permalink: Installating PoPoolation2: Differentiation Analysis\" href=\"#installating-popoolation2-differentiation-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll the steps comes from PoPoolation website (\u003ca href=\"https://sourceforge.net/p/popoolation2/wiki/Home/\" rel=\"nofollow\"\u003ehttps://sourceforge.net/p/popoolation2/wiki/Home/\u003c/a\u003e) and Dr. Kofler presentation (\u003ca href=\"https://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part2.pdf\" rel=\"nofollow\"\u003ehttps://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part2.pdf\u003c/a\u003e)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"svn checkout https://svn.code.sf.net/p/popoolation2/code/trunk popoolation2-code\ncd popoolation2-code/\nsvn update\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esvn checkout https://svn.code.sf.net/p/popoolation2/code/trunk popoolation2-code\ncd popoolation2-code/\nsvn update\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eCreate a mpileup\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam G12_cerise_q20_MarkDuplicated.bam G12_cran_q20_MarkDuplicated.bam G12_fraise_q20_MarkDuplicated.bam \u0026gt; p1-4.mpileup # 77m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam G12_cerise_q20_MarkDuplicated.bam G12_cran_q20_MarkDuplicated.bam G12_fraise_q20_MarkDuplicated.bam \u0026gt; p1-4.mpileup # 77m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMPILEUP:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVery similar to the mpileup with one sample. In this multi-pileup (mpileup) three additional columns are created for each additional sample. Thus for each sample the following information is provided:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ethe coverage, columns 4 + n ∗ 3\u003c/li\u003e\n\u003cli\u003ethe bases, columns 5 + n ∗ 3\u003c/li\u003e\n\u003cli\u003ethe corresponding base quality, columns 6 + n ∗ 3\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eConversion to sync-file\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eIn order to use PoPoolation2 we have to convert the mpileup to a sync file.\nThis may seem unnecessary to you, but it serves to speed up the analysis,\nbecause the time consuming part of the analysis - parsing of the mpileup file -\nneeds just to be performed once.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time java -jar ~/Documents/Projet_S3/Popoolation/popoolation2-code/mpileup2sync.jar --input p1-4.mpileup --output p1-4.sync --fastq-type sanger --min-qual 20 --threads 15 #78m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime java -jar ~/Documents/Projet_S3/Popoolation/popoolation2-code/mpileup2sync.jar --input p1-4.mpileup --output p1-4.sync --fastq-type sanger --min-qual 20 --threads 15 #78m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTHE SYNC-FILE:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecol 1: reference chromosome\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecol 2: position\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecol 3: reference character\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecol 4: allele counts for first population\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecol 5: allele counts for second population\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecol n: allele counts for n-3 population\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAllele counts are in the form ”A:T:C:G:N:del”\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ethe sync-file provides a convenient summary of the allele counts of several populations (there\nis no upper threshold of the population number).\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003esubsampling to an uniform coverage with the PoPoolation2 pipeline should be done using such a sync-file\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eCalculating the Fst\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"time perl ~/Documents/Projet_S3/Popoolation/popoolation2-code/fst-sliding.pl --window-size 1 --step-size 1 --suppress-noninformative --input p1-4.sync --min-covered-fraction 1.0 --min-coverage 4 --max-coverage 120 --min-count 3 --output fst_tot.txt --pool-size 80 #386m\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etime perl ~/Documents/Projet_S3/Popoolation/popoolation2-code/fst-sliding.pl --window-size 1 --step-size 1 --suppress-noninformative --input p1-4.sync --min-covered-fraction 1.0 --min-coverage 4 --max-coverage 120 --min-count 3 --output fst_tot.txt --pool-size 80 #386m\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e--suppress-noninformative suppresses output for windows not containing a SNP. When applied to windows of size one, this option suppresses output for bases that are no SNP.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eOUTPUT OF THE Fst SCRIPT:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ecol 1: reference chromosome\u003c/li\u003e\n\u003cli\u003ecol 2: position\u003c/li\u003e\n\u003cli\u003ecol 3: window size (1 for single SNPs)\u003c/li\u003e\n\u003cli\u003ecol 4: covered fraction (relevant for minimum covered fraction)\u003c/li\u003e\n\u003cli\u003ecol 5: average minimum coverage for the window across all populations (the higher the more reliable the estimate)\u003c/li\u003e\n\u003cli\u003ecol 6: pairwise Fst comparing population 1 with population 2\u003c/li\u003e\n\u003cli\u003ecol 7: etc for ALL pairwise comparisons of the populations present in the sync file\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eRESULTS:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe separate the fst_tot.txt file in 3 files each containing one comparison with G0.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat fst_tot.txt | sed 's/1:2=//' | awk  -F\u0026quot; \u0026quot; '{print $1\u0026quot; \u0026quot; $2\u0026quot; \u0026quot; $6}'\u0026gt; Fst_1-2.txt\ncat fst_tot.txt | sed 's/1:3=//' | awk  -F\u0026quot; \u0026quot; '{print $1\u0026quot; \u0026quot; $2\u0026quot; \u0026quot; $7}'\u0026gt; Fst_1-3.txt\ncat fst_tot.txt | sed 's/1:4=//' | awk  -F\u0026quot; \u0026quot; '{print $1\u0026quot; \u0026quot; $2\u0026quot; \u0026quot; $8}'\u0026gt; Fst_1-4.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecat fst_tot.txt | sed 's/1:2=//' | awk  -F\" \" '{print $1\" \" $2\" \" $6}'\u0026gt; Fst_1-2.txt\ncat fst_tot.txt | sed 's/1:3=//' | awk  -F\" \" '{print $1\" \" $2\" \" $7}'\u0026gt; Fst_1-3.txt\ncat fst_tot.txt | sed 's/1:4=//' | awk  -F\" \" '{print $1\" \" $2\" \" $8}'\u0026gt; Fst_1-4.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFst_1-2.txt: G0-G12_cerise\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFst_1-3.txt: G0-G12_cran\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFst_1-4.txt: G0-G12_fraise\u003c/p\u003e\n\u003cdiv align=\"center\" dir=\"auto\"\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"center\"\u003eFst totalité des SNP\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0-G12 cranberry\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0-G12 cerise\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0-G12 fraise\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003emoyenne\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.02347\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.03759\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.03145\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eécart type\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.03630\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.05538\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.04696\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\n\u003cdiv align=\"center\" dir=\"auto\"\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth align=\"center\"\u003eFst gènes DE\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0-G12 cranberry\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0-G12 cerise\u003c/th\u003e\n\u003cth align=\"center\"\u003eG0-G12 fraise\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003emoyenne\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.02188\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.03429\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.02229\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd align=\"center\"\u003eécart type\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.03374\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.04879\u003c/td\u003e\n\u003ctd align=\"center\"\u003e0.03373\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eCalculating Fst for DE genes:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eWe write a script using Fst results and a list of DE gene in a bed format to calculate Fst for DE gene.\nAs we use dichotomous search, the Fst file must be sorted like the pipeline does.\nYou find it on the github repo:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\" scripts/Fst_DE_Genes.py\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e scripts/Fst_DE_Genes.py\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eVisualisation of Fst on chromosomes:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eWe write a function in R to visualize the Fst on a chosen chromosome.\nYou find it on the github repo:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\" scripts/VisualisationFst.R\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e scripts/VisualisationFst.R\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/Fst.png\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/Fst.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e6. TE Analysis\u003c/h2\u003e\u003ca id=\"user-content-6-te-analysis\" class=\"anchor\" aria-label=\"Permalink: 6. TE Analysis\" href=\"#6-te-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll the steps are included in a tool named \u003ca href=\"https://github.com/clemgoub/dnaPipeTE\"\u003ednaPipeTE\u003c/a\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eUniform samplings of the reads to produce low coverage data sets (read sampling to have \u0026lt;1X coverage of the genome to keep only repeated regions)\u003c/li\u003e\n\u003cli\u003eTrinity, repeated assembly of contigs\u003c/li\u003e\n\u003cli\u003eRepeatMasker, contigs annotation\u003c/li\u003e\n\u003cli\u003eBlastn, repeat quantification\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/TE_analysis/dnapipete.jpg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/TE_analysis/dnapipete.jpg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFigure from the \u003ca href=\"https://academic.oup.com/gbe/article/7/4/1192/533768\" rel=\"nofollow\"\u003etool article\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e6.1 Running DnaPipeTE\u003c/h3\u003e\u003ca id=\"user-content-61-running-dnapipete\" class=\"anchor\" aria-label=\"Permalink: 6.1 Running DnaPipeTE\" href=\"#61-running-dnapipete\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#dnaPipeTE: version 1.3 (uses Perl 5, R v3.0.2, Python v3.8.5, Trinity v2.5.1, \n\t\t\t      RepeatMasker v4.0.5 including RMblastn, ncbi-blast v2.2.28+)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#dnaPipeTE: version 1.3 (uses Perl 5, R v3.0.2, Python v3.8.5, Trinity v2.5.1, \n\t\t\t      RepeatMasker v4.0.5 including RMblastn, ncbi-blast v2.2.28+)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe had a lot of dependancies problems running dnaPipeTE so we used the \u003ca href=\"https://hub.docker.com/r/clemgoub/dnapipete\" rel=\"nofollow\"\u003edocker version\u003c/a\u003e.\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eCopy input data\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eTwo different types of input are necessary to run the pipeline :\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eThe fastq files for each samples\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eDrosophila_Transposable_Element_all.fasta\u003c/code\u003e which is the Drosophila TE library from Repbase\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eLocation :  \u003cem\u003epedago-ngs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/localdata/pandata/students/M2_projet_15/TE/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/localdata/pandata/students/M2_projet_15/TE/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThey all had to be in a \u003ccode\u003e~/Project\u003c/code\u003e folder which is the communication folder between the computer and the container.\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eStarting the container\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sudo docker pull clemgoub/dnapipete:latest\nsudo docker run -it -v ~/Project:/mnt clemgoub/dnapipete:latest\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esudo docker pull clemgoub/dnapipete:latest\nsudo docker run -it -v ~/Project:/mnt clemgoub/dnapipete:latest\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eRun dnaPipeTE\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /opt/dnaPipeTE\npython3 dnaPipeTE.py \\ \n\t\t-input /mnt/data/sample_1.fastq.gz \n\t\t-output /mnt/output_dnapipete/sample \n\t\t-cpu 6 \n\t\t-genome_size 270000000 \n\t\t-genome_coverage 0.2 \n\t\t-sample_number 2 \n\t\t-RM_lib /mnt/Drosophila_Transposable_Element_all.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /opt/dnaPipeTE\npython3 dnaPipeTE.py \\ \n\t\t-input /mnt/data/sample_1.fastq.gz \n\t\t-output /mnt/output_dnapipete/sample \n\t\t-cpu 6 \n\t\t-genome_size 270000000 \n\t\t-genome_coverage 0.2 \n\t\t-sample_number 2 \n\t\t-RM_lib /mnt/Drosophila_Transposable_Element_all.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003e-input\u003c/code\u003e: input fastq or fastq.gz files (single end only);\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-output\u003c/code\u003e: complete path with name for the outputs;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-cpu\u003c/code\u003e: maximum number of cpu to use;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-genome_size\u003c/code\u003e: here we know that \u003cem\u003eD.suzukii\u003c/em\u003e is 270Mb long;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-genome_coverage\u003c/code\u003e: for this genome size 0.2X is recommended in the Supplementary Material of the \u003ca href=\"https://academic.oup.com/gbe/article/7/4/1192/533768\" rel=\"nofollow\"\u003etool article\u003c/a\u003e;\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-sample_number\u003c/code\u003e: number of Trinity iterations (2 is the minimum and recommended number of iterations);\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003e-RM_lib\u003c/code\u003e: path to custom repeat library for RepeatMasker.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e6.2 Results\u003c/h3\u003e\u003ca id=\"user-content-62-results\" class=\"anchor\" aria-label=\"Permalink: 6.2 Results\" href=\"#62-results\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eExample of sampling for G0 :\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"total number of reads: 121247626\nmaximum number of reads to sample:  710526\nfastq :  /mnt/G0-MTP_1.fastq\nsampling 2 samples of max 355263 reads to reach coverage...\n53999976 bases sampled in 355263 reads\ns_G0-MTP_1.fastq done.\n53999976 bases sampled in 710526 reads\ns_G0-MTP_1.fastq done.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003etotal number of reads: 121247626\nmaximum number of reads to sample:  710526\nfastq :  /mnt/G0-MTP_1.fastq\nsampling 2 samples of max 355263 reads to reach coverage...\n53999976 bases sampled in 355263 reads\ns_G0-MTP_1.fastq done.\n53999976 bases sampled in 710526 reads\ns_G0-MTP_1.fastq done.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe pipeline gives several useful outputs such as the \u003cstrong\u003eLandscapes\u003c/strong\u003e (TE divergence to estimate the insertions) and \u003cstrong\u003ePieChart\u003c/strong\u003e (TE families proportions in the genome).\nOther graphs have been created with the script \u003ccode\u003egraphs.R\u003c/code\u003e to compare TE throught different sample generations.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/TE_analysis/prop_ech_par_family.jpg\"\u003e\u003cimg src=\"/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/TE_analysis/prop_ech_par_family.jpg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOther graphs can be found in the \u003ccode\u003egraphs.pdf\u003c/code\u003e document.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e7. Authors\u003c/h2\u003e\u003ca id=\"user-content-7-authors\" class=\"anchor\" aria-label=\"Permalink: 7. Authors\" href=\"#7-authors\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe project was developped by:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eChloé AUJOULAT\u003c/li\u003e\n\u003cli\u003eTommaso BARBERIS\u003c/li\u003e\n\u003cli\u003eBertrand HUGUENIN-BIZOT\u003c/li\u003e\n\u003cli\u003eMarie VERNERET\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e8. References\u003c/h2\u003e\u003ca id=\"user-content-8-references\" class=\"anchor\" aria-label=\"Permalink: 8. References\" href=\"#8-references\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eAnand, Santosh, Eleonora Mangano, Nadia Barizzone, Roberta Bordoni, Melissa Sorosina, Ferdinando Clarelli, Lucia Corrado, Filippo Martinelli Boneschi, Sandra D’Alfonso, et Gianluca De Bellis. « Next Generation Sequencing of Pooled Samples: Guideline for Variants’ Filtering ». Scientific Reports 6, no 1 (27 septembre 2016): 33735. \u003ca href=\"https://doi.org/10.1038/srep33735\" rel=\"nofollow\"\u003ehttps://doi.org/10.1038/srep33735\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eChiu, Joanna C, Xuanting Jiang, Li Zhao, Christopher A Hamm, Julie M Cridland, Perot Saelao, Kelly A Hamby, et al. « Genome of Drosophila suzukii, the Spotted Wing Drosophila ». G3 Genes|Genomes|Genetics 3, no 12 (1 décembre 2013): 2257‑71. \u003ca href=\"https://doi.org/10.1534/g3.113.008185\" rel=\"nofollow\"\u003ehttps://doi.org/10.1534/g3.113.008185\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eKofler, Robert, Daniel Gómez-Sánchez, et Christian Schlötterer. « PoPoolationTE2: Comparative Population Genomics of Transposable Elements Using Pool-Seq ». Molecular Biology and Evolution 33, no 10 (octobre 2016): 2759‑64. \u003ca href=\"https://doi.org/10.1093/molbev/msw137\" rel=\"nofollow\"\u003ehttps://doi.org/10.1093/molbev/msw137\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eOlazcuaga, Laure, Julien Foucaud, Mathieu Gautier, Candice Deschamps, Anne Loiseau, Nicolas Leménager, Benoit Facon, et al. « Adaptation and Correlated Fitness Responses over Two Time Scales in Drosophila Suzukii Populations Evolving in Different Environments ». Journal of Evolutionary Biology 34, no 8 (2021): 1225‑40. \u003ca href=\"https://doi.org/10.1111/jeb.13878\" rel=\"nofollow\"\u003ehttps://doi.org/10.1111/jeb.13878\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eClément Goubert, Laurent Modolo, Cristina Vieira, Claire ValienteMoro, Patrick Mavingui, Matthieu Boulesteix, De Novo Assembly and Annotation of the Asian Tiger Mosquito (Aedes albopictus) Repeatome with dnaPipeTE from Raw Genomic Reads and Comparative Analysis with the Yellow Fever Mosquito (Aedes aegypti), Genome Biology and Evolution, Volume 7, Issue 4, April 2015, Pages 1192–1205, \u003ca href=\"https://doi.org/10.1093/gbe/evv050\" rel=\"nofollow\"\u003ehttps://doi.org/10.1093/gbe/evv050\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eMarie  Tabourin  et  al.  “Analyse  de  l’expression  des  gènes  et  des  éléments  transposableschez Drosophila suzukii suite à une expérience d’évolution expérimentale.\u003c/li\u003e\n\u003cli\u003eSimon Andrews et al.FastQC. Babraham Institute. Babraham, UK, Jan. 2012, \u003ca href=\"https://www.bioinformatics.babraham.ac.uk/projects/fastqc/\" rel=\"nofollow\"\u003ehttps://www.bioinformatics.babraham.ac.uk/projects/fastqc/\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eMultiQC: Summarize analysis results for multiple tools and samples in a single report, Philip Ewels, Måns Magnusson, Sverker, Lundin and Max Käller, Bioinformatics (2016), doi: 10.1093/bioinformatics/btw354, PMID: 27312411.\u003c/li\u003e\n\u003cli\u003eLi H. and Durbin R. (2009) Fast and accurate short read alignment with Burrows-Wheeler Transform. Bioinformatics, 25:1754-60. [PMID: 19451168].\u003c/li\u003e\n\u003cli\u003eHeng Li, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, Richard Durbin, 1000 Genome Project Data Processing Subgroup, The Sequence Alignment/Map format and SAMtools, Bioinformatics, Volume 25, Issue 16, 15 August 2009, Pages 2078–2079, \u003ca href=\"https://doi.org/10.1093/bioinformatics/btp352\" rel=\"nofollow\"\u003ehttps://doi.org/10.1093/bioinformatics/btp352\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eVan der Auwera GA \u0026amp; O'Connor BD. (2020). Genomics in the Cloud: Using Docker, GATK, and WDL in Terra (1st Edition). O'Reilly Media.\u003c/li\u003e\n\u003cli\u003eWickham H (2016). ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York. ISBN 978-3-319-24277-4, \u003ca href=\"https://ggplot2.tidyverse.org\" rel=\"nofollow\"\u003ehttps://ggplot2.tidyverse.org\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eR Core Team (2021). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL \u003ca href=\"https://www.R-project.org/\" rel=\"nofollow\"\u003ehttps://www.R-project.org/\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003eBedops.readthedocs.io. 2021. 6.3.3.5. gff2bed — BEDOPS v2.4.40. (online) Available at: \u003ca href=\"https://bedops.readthedocs.io/en/latest/content/reference/file-management/conversion/gff2bed.html\" rel=\"nofollow\"\u003ehttps://bedops.readthedocs.io/en/latest/content/reference/file-management/conversion/gff2bed.html\u003c/a\u003e (Accessed 2 December 2021).\u003c/li\u003e\n\u003cli\u003eBedtools.readthedocs.io. 2021. intersect — bedtools 2.30.0 documentation. (online) Available at: \u003ca href=\"https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html\" rel=\"nofollow\"\u003ehttps://bedtools.readthedocs.io/en/latest/content/tools/intersect.html\u003c/a\u003e (Accessed 2 December 2021).\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Genomic and transcriptomic evolution of D. suzukii","anchor":"genomic-and-transcriptomic-evolution-of-d-suzukii","htmlText":"Genomic and transcriptomic evolution of D. suzukii"},{"level":2,"text":"Table of contents","anchor":"table-of-contents","htmlText":"Table of contents"},{"level":2,"text":"1. Context","anchor":"1-context","htmlText":"1. Context"},{"level":2,"text":"2. Introduction to the project","anchor":"2-introduction-to-the-project","htmlText":"2. Introduction to the project"},{"level":3,"text":"2.1 Polymorphism analysis","anchor":"21-polymorphism-analysis","htmlText":"2.1 Polymorphism analysis"},{"level":3,"text":"2.2 TE analysis","anchor":"22-te-analysis","htmlText":"2.2 TE analysis"},{"level":2,"text":"3. Data","anchor":"3-data","htmlText":"3. Data"},{"level":2,"text":"4. SNP Calling Workflow","anchor":"4-snp-calling-workflow","htmlText":"4. SNP Calling Workflow"},{"level":3,"text":"4.1 Quality control","anchor":"41-quality-control","htmlText":"4.1 Quality control"},{"level":3,"text":"4.2 Mapping","anchor":"42-mapping","htmlText":"4.2 Mapping"},{"level":4,"text":"Mapping stats","anchor":"mapping-stats","htmlText":"Mapping stats"},{"level":4,"text":"Conversion to .bam","anchor":"conversion-to-bam","htmlText":"Conversion to .bam"},{"level":3,"text":"4.3 Variant Calling","anchor":"43-variant-calling","htmlText":"4.3 Variant Calling"},{"level":4,"text":"4.3.1 SNPs filtering","anchor":"431-snps-filtering","htmlText":"4.3.1 SNPs filtering"},{"level":4,"text":"4.3.2 Get SNPs that are in transcripts (exons)","anchor":"432-get-snps-that-are-in-transcripts-exons","htmlText":"4.3.2 Get SNPs that are in transcripts (exons)"},{"level":4,"text":"4.3.3  Get SNPs that are in DE genes","anchor":"433--get-snps-that-are-in-de-genes","htmlText":"4.3.3  Get SNPs that are in DE genes"},{"level":4,"text":"4.3.4 Results","anchor":"434-results","htmlText":"4.3.4 Results"},{"level":5,"text":"4.3.4.1 SNPs founded on exons","anchor":"4341-snps-founded-on-exons","htmlText":"4.3.4.1 SNPs founded on exons"},{"level":5,"text":"4.3.4.1 SNPs founded on exons of DE genes","anchor":"4341-snps-founded-on-exons-of-de-genes","htmlText":"4.3.4.1 SNPs founded on exons of DE genes"},{"level":2,"text":"5. Diversity \u0026 Differentiation Analysis","anchor":"5-diversity--differentiation-analysis","htmlText":"5. Diversity \u0026amp; Differentiation Analysis"},{"level":1,"text":"Installating PoPoolation: Diversity Analysis","anchor":"installating-popoolation-diversity-analysis","htmlText":"Installating PoPoolation: Diversity Analysis"},{"level":1,"text":"Installating PoPoolation2: Differentiation Analysis","anchor":"installating-popoolation2-differentiation-analysis","htmlText":"Installating PoPoolation2: Differentiation Analysis"},{"level":2,"text":"6. TE Analysis","anchor":"6-te-analysis","htmlText":"6. TE Analysis"},{"level":3,"text":"6.1 Running DnaPipeTE","anchor":"61-running-dnapipete","htmlText":"6.1 Running DnaPipeTE"},{"level":3,"text":"6.2 Results","anchor":"62-results","htmlText":"6.2 Results"},{"level":2,"text":"7. Authors","anchor":"7-authors","htmlText":"7. Authors"},{"level":2,"text":"8. References","anchor":"8-references","htmlText":"8. References"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii"}},{"displayName":"LICENSE","repoName":"Genomic-and-transcriptomic-evolution-of-D.-Suzukii","refName":"main","path":"LICENSE","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii"}}],"overviewFilesProcessingTime":78.43409}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":null,"enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":false,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":false}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                  </div>
                  <div class="react-directory-remove-file-icon">
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           62 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="TE_analysis, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/TE_analysis" title="TE_analysis">
                          TE_analysis
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="TE_analysis, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/TE_analysis" title="TE_analysis">
                          TE_analysis
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="img, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/img" title="img">
                          img
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="img, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/img" title="img">
                          img
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="quality_control, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/quality_control" title="quality_control">
                          quality_control
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="quality_control, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/quality_control" title="quality_control">
                          quality_control
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="scripts, (Directory)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/tree/main/scripts" title="scripts">
                          scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitattributes, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/.gitattributes" title=".gitattributes">
                          .gitattributes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitattributes, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/.gitattributes" title=".gitattributes">
                          .gitattributes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Book of specifications.pdf, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/Book%20of%20specifications.pdf" title="Book of specifications.pdf">
                          Book of specifications.pdf
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Book of specifications.pdf, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/Book%20of%20specifications.pdf" title="Book of specifications.pdf">
                          Book of specifications.pdf
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Genomic and transcriptomic evolution of
                      <em>
                       D. suzukii
                      </em>
                     </h1>
                     <a aria-label="Permalink: Genomic and transcriptomic evolution of D. suzukii" class="anchor" href="#genomic-and-transcriptomic-evolution-of-d-suzukii" id="user-content-genomic-and-transcriptomic-evolution-of-d-suzukii">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Table of contents
                     </h2>
                     <a aria-label="Permalink: Table of contents" class="anchor" href="#table-of-contents" id="user-content-table-of-contents">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <a href="#1-context">
                       Context
                      </a>
                     </li>
                     <li>
                      <a href="#2-introduction-to-the-project">
                       Introduction to the project
                      </a>
                     </li>
                     <li>
                      <a href="#3-data">
                       Data
                      </a>
                     </li>
                     <li>
                      <a href="#4-snp-calling-workflow">
                       SNP Calling Workflow
                      </a>
                     </li>
                     <li>
                      <a href="#5-diversity-analysis">
                       Diversity Analysis
                      </a>
                     </li>
                     <li>
                      <a href="#6-te-analysis">
                       TE Analysis
                      </a>
                     </li>
                     <li>
                      <a href="#7-authors">
                       Authors
                      </a>
                     </li>
                     <li>
                      <a href="#8-references">
                       References
                      </a>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      1. Context
                     </h2>
                     <a aria-label="Permalink: 1. Context" class="anchor" href="#1-context" id="user-content-1-context">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Since its recent invasion in the European and American continents,
                     <em>
                      Drosophila suzukii
                     </em>
                     , has become a major pest of berry crops.
Contrary to other
                     <em>
                      Drosophila
                     </em>
                     species that develop themselves on damaged or rotten fruit,
                     <em>
                      D. suzukii
                     </em>
                     is able to lay its eggs in healthy fruit before harvest, using its
                     <strong>
                      sclerotinized ovipositor
                     </strong>
                     . In order to control the
                     <em>
                      D. suzukii
                     </em>
                     population it is therefore required to improve our knowledge about its ability to adapt to a local environment outside its origin areas.
                    </p>
                    <p dir="auto">
                     A previous study focused on phenotipic changes during a selective experiment showed a temporal adaptation of
                     <em>
                      D. suzukii
                     </em>
                     populations in the three different fruit environments. This process produces a
                     <strong>
                      fitness
                     </strong>
                     improvement which corresponds to a greater ability to reproduce and could explain the adaptation abilities  of
                     <em>
                      D. suzukii
                     </em>
                     . Based on the same experiment, a second study has been started by the LBBE to see if the
                     <em>
                      D. suzukii
                     </em>
                     genome and transcriptome are affected by the environment.
                    </p>
                    <p dir="auto">
                     The project is focused on both genes and Transposable Elements (TE) expression in order to find the variants explaining the local environment adaptation of
                     <em>
                      D. suzukii
                     </em>
                     .
A significant number of differentially expressed genes across generations has been found with
                     <strong>
                      RNAseq analysis
                     </strong>
                     between the G0 and the G7 generations. These variations seems to be independant of the fruit media. Moreover, in differentially expressed (DE) genes an increase of the number of Single Nucleotide Polymorphism (SNP) was found in G7 compared to G0, but this result was not expected. Natural selection generally keeps alleles which allow the best adaptation, and it is synonymous with a decrease in SNP number.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2. Introduction to the project
                     </h2>
                     <a aria-label="Permalink: 2. Introduction to the project" class="anchor" href="#2-introduction-to-the-project" id="user-content-2-introduction-to-the-project">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.1 Polymorphism analysis
                     </h3>
                     <a aria-label="Permalink: 2.1 Polymorphism analysis" class="anchor" href="#21-polymorphism-analysis" id="user-content-21-polymorphism-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The goal here is to analyse
                     <em>
                      D. suzukii
                     </em>
                     polymorphism performing a
                     <strong>
                      variant calling
                     </strong>
                     on the whole genome (using poolseq data) in order to count the number of different genomic positions between G0 and G12, and between fruits media.
                     <br/>
                     The position of the genetic variations will be compared with previous SNP analysis conducted on RNA-seq data to see if we also find an increase of the SNP number. To have a view more global on the genome's diversity, several diversity measures will be calculated:
                     <strong>
                      pi's
                     </strong>
                     <a href="https://render.githubusercontent.com/render/math?math=\theta" rel="noopener noreferrer nofollow" target="_blank">
                      <img src="https://render.githubusercontent.com/render/math?math=\theta" style="max-width: 100%;"/>
                     </a>
                     ,
                     <strong>
                      Watterson's
                     </strong>
                     <a href="https://render.githubusercontent.com/render/math?math=\theta" rel="noopener noreferrer nofollow" target="_blank">
                      <img src="https://render.githubusercontent.com/render/math?math=\theta" style="max-width: 100%;"/>
                     </a>
                     ,
                     <strong>
                      Tajima's D
                     </strong>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.2 TE analysis
                     </h3>
                     <a aria-label="Permalink: 2.2 TE analysis" class="anchor" href="#22-te-analysis" id="user-content-22-te-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Transposable elements
                     </strong>
                     (
                     <strong>
                      TE
                     </strong>
                     ) represent
                     <strong>
                      47%
                     </strong>
                     of the
                     <em>
                      D. suzukii
                     </em>
                     genome and can be important in the adaptation process, interfering in gene regulation during evolution. The aim is to determine TEs age on our data and compute their abundance. Comparing abundance graphs between G0 and G12 will inform us if some recent TE insertions happened.
                    </p>
                    <p dir="auto">
                     All the project objectives are presented more precisely in the book of specifications file.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      3. Data
                     </h2>
                     <a aria-label="Permalink: 3. Data" class="anchor" href="#3-data" id="user-content-3-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      DNA poolseq fastq from
                      <em>
                       D.suzukii
                      </em>
                      obtained with paired-end Illumina sequencing (2x100pb), with 40 individuals for each pool:
                      <ul dir="auto">
                       <li>
                        One poolseq for G0;
                       </li>
                       <li>
                        One poolseq for each media in G12 (strawberry, cherry and cranberry).
                       </li>
                      </ul>
                     </li>
                    </ul>
                    <div align="center" dir="auto">
                     <table>
                      <thead>
                       <tr>
                        <th align="center">
                         File name
                        </th>
                        <th align="center">
                         Number of reads
                        </th>
                       </tr>
                      </thead>
                      <tbody>
                       <tr>
                        <td align="center">
                         G0-MTP
                        </td>
                        <td align="center">
                         242 990 252
                        </td>
                       </tr>
                       <tr>
                        <td align="center">
                         G12-cerise
                        </td>
                        <td align="center">
                         214 319 262
                        </td>
                       </tr>
                       <tr>
                        <td align="center">
                         G12-cranb
                        </td>
                        <td align="center">
                         261 528 888
                        </td>
                       </tr>
                       <tr>
                        <td align="center">
                         G12-fraise
                        </td>
                        <td align="center">
                         216 028 586
                        </td>
                       </tr>
                      </tbody>
                     </table>
                    </div>
                    <p dir="auto">
                     Location:
                     <em>
                      pedago-ngs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/data">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/data
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      Partial assembly of
                      <em>
                       D. suzukii
                      </em>
                      genome as reference (4 chromosomes, 542 contigs);
                     </li>
                    </ul>
                    <p dir="auto">
                     Location:
                     <em>
                      pedago-ngs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-contig.fasta">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-contig.fasta
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      Annotation file at the format
                      <code>
                       gff3
                      </code>
                      .
Location:
                      <em>
                       pedago-ngs
                      </em>
                     </li>
                    </ul>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-annotation-3-ws.gff3">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/ref/Drosophila-suzukii-annotation-3-ws.gff3
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4. SNP Calling Workflow
                     </h2>
                     <a aria-label="Permalink: 4. SNP Calling Workflow" class="anchor" href="#4-snp-calling-workflow" id="user-content-4-snp-calling-workflow">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/schema snp calling (1).jpg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/schema snp calling (1).jpg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.1 Quality control
                     </h3>
                     <a aria-label="Permalink: 4.1 Quality control" class="anchor" href="#41-quality-control" id="user-content-41-quality-control">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     First of all we checked the quality of the data, using
                     <code>
                      fastqc
                     </code>
                     <em>
                      (version v0.11.9)
                     </em>
                     :
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="fastqc sample.fastq.gz">
                     <pre class="notranslate"><code>fastqc sample.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     Location:
                     <em>
                      pedago-ngs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/quality_control">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/quality_control
</code></pre>
                    </div>
                    <p dir="auto">
                     Then, we merged results from
                     <code>
                      fastqc
                     </code>
                     using
                     <code>
                      multiqc
                     </code>
                     in the folder with
                     <code>
                      fastqc
                     </code>
                     reports. The
                     <code>
                      html
                     </code>
                     report is available in the
                     <code>
                      quality_control
                     </code>
                     folder.
                     <br/>
                     We noticed a high
                     <strong>
                      GC
                     </strong>
                     rate in the reads, but it seems to be in accordance with the results obtained by other researchs on
                     <em>
                      D. suzukii
                     </em>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.2 Mapping
                     </h3>
                     <a aria-label="Permalink: 4.2 Mapping" class="anchor" href="#42-mapping" id="user-content-42-mapping">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Genome indexing:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bwa index \
    -p Drosophila-suzukii-contig.fasta.fai \
    -a bwtsw Drosophila-suzukii-contig.fasta">
                     <pre class="notranslate"><code>bwa index \
    -p Drosophila-suzukii-contig.fasta.fai \
    -a bwtsw Drosophila-suzukii-contig.fasta
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -p
                      </code>
                      : index name;
                     </li>
                     <li>
                      <code>
                       -a
                      </code>
                      : index algorithm (
                      <em>
                       bwtsw
                      </em>
                      for long genome).
                     </li>
                    </ul>
                    <ol dir="auto" start="2">
                     <li>
                      Mapping samples on the reference:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_1.fastq.gz &gt; sample_1.sai
bwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_2.fastq.gz &gt; sample_2.sai">
                     <pre class="notranslate"><code>bwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_1.fastq.gz &gt; sample_1.sai
bwa aln -t 14 Drosophila-suzukii-contig.fasta.fai sample_2.fastq.gz &gt; sample_2.sai
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -t
                      </code>
                      : number of threads.
                     </li>
                    </ul>
                    <ol dir="auto" start="3">
                     <li>
                      Merge the
                      <code>
                       .sai
                      </code>
                      files of forward and reverse reads:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bwa sampe Drosophila-suzukii-contig.fasta.fai \
    sample_1.sai sample_2.sai \
    sample_1.fastq.gz sample_2.fastq.gz &gt; sample_pe.sam">
                     <pre class="notranslate"><code>bwa sampe Drosophila-suzukii-contig.fasta.fai \
    sample_1.sai sample_2.sai \
    sample_1.fastq.gz sample_2.fastq.gz &gt; sample_pe.sam
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       sampe
                      </code>
                      : take in account to have paired ends reads.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Mapping stats
                     </h4>
                     <a aria-label="Permalink: Mapping stats" class="anchor" href="#mapping-stats" id="user-content-mapping-stats">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Computation of mapped files statistics using:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="samtools stats sample.sam">
                     <pre class="notranslate"><code>samtools stats sample.sam
</code></pre>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th align="center">
                        sample
                       </th>
                       <th align="center">
                        raw sequences
                       </th>
                       <th align="center">
                        mapped and paired squences
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td align="center">
                        G0-MTP_pe
                       </td>
                       <td align="center">
                        242495252
                       </td>
                       <td align="center">
                        194326801
                       </td>
                      </tr>
                      <tr>
                       <td align="center">
                        G12-fraise_pe
                       </td>
                       <td align="center">
                        216028586
                       </td>
                       <td align="center">
                        172898366
                       </td>
                      </tr>
                      <tr>
                       <td align="center">
                        G12-cerise_pe
                       </td>
                       <td align="center">
                        214319262
                       </td>
                       <td align="center">
                        172440052
                       </td>
                      </tr>
                      <tr>
                       <td align="center">
                        G12-cranb_pe
                       </td>
                       <td align="center">
                        261528888
                       </td>
                       <td align="center">
                        212135984
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Conversion to
                      <code>
                       .bam
                      </code>
                     </h4>
                     <a aria-label="Permalink: Conversion to .bam" class="anchor" href="#conversion-to-bam" id="user-content-conversion-to-bam">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="samtools view -bS -@ 3 sample.sam &gt; sample.bam">
                     <pre class="notranslate"><code>samtools view -bS -@ 3 sample.sam &gt; sample.bam
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -bS
                      </code>
                      : output in the
                      <strong>
                       BAM
                      </strong>
                      format ignoring for compatibility with previous samtools version;
                     </li>
                     <li>
                      <code>
                       -@
                      </code>
                      : number of threads.
                     </li>
                    </ul>
                    <p dir="auto">
                     Location:
                     <em>
                      pedago-ngs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/mapping_bwa">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/mapping_bwa
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      NB:
                     </strong>
                     Before using
                     <code>
                      bwa
                     </code>
                     , we tried to use
                     <code>
                      hisat2
                     </code>
                     but, for an unkown reason it could not map the strawberry sample.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.3 Variant Calling
                     </h3>
                     <a aria-label="Permalink: 4.3 Variant Calling" class="anchor" href="#43-variant-calling" id="user-content-43-variant-calling">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The SNP calling on pooled data is still not largely used because it is very innovative. For this reason it has required several tests (with different tools) and a long time to be completed.
                     <br/>
                     Finally, we used
                     <code>
                      GATK
                     </code>
                     to perform our SNP calling and for that we were inspired by M. Tabourin's workflow (
                     <a href="https://github.com/mtabourin/Analyse-experience-d-evolution-experimentale-D.suzukii">
                      Analyse experience d'évolution expérimentale
                      <em>
                       D.suzukii
                      </em>
                     </a>
                     ) and from
                     <a href="https://yeamanlab.weebly.com/uploads/5/7/9/5/57959825/snp_calling_pipeline.pdf" rel="nofollow">
                      this
                     </a>
                     document.
                    </p>
                    <ol dir="auto">
                     <li>
                      Sorting
                      <code>
                       .bam
                      </code>
                      files:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# GATK version: gatk-4.2.2.0
gatk SortSam -I sample.bam -O sample_sorted.bam -SO queryname -VALIDATION_STRINGENCY SILENT">
                     <pre class="notranslate"><code># GATK version: gatk-4.2.2.0
gatk SortSam -I sample.bam -O sample_sorted.bam -SO queryname -VALIDATION_STRINGENCY SILENT
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -I
                      </code>
                      : input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file;
                     </li>
                     <li>
                      <code>
                       -SO
                      </code>
                      : sort order of output file;
                     </li>
                     <li>
                      <code>
                       -VALIDATION_STRINGENCY
                      </code>
                      : Validation stringency for all SAM files read by this program. Setting stringency to
                      <code>
                       SILENT
                      </code>
                      can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.
                     </li>
                    </ul>
                    <ol dir="auto" start="2">
                     <li>
                      Add read group info:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gatk AddOrReplaceReadGroups \
    -I sample_sorted.bam \
    -O sample_ReadGroups.bam \
    -RGID name \
    -RGLB name \
    -RGPL ILLUMINA \
    -RGPU name \
    -RGSM sample \
    -VALIDATION_STRINGENCY SILENT">
                     <pre class="notranslate"><code>gatk AddOrReplaceReadGroups \
    -I sample_sorted.bam \
    -O sample_ReadGroups.bam \
    -RGID name \
    -RGLB name \
    -RGPL ILLUMINA \
    -RGPU name \
    -RGSM sample \
    -VALIDATION_STRINGENCY SILENT
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -I
                      </code>
                      : input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file;
                     </li>
                     <li>
                      <code>
                       -RGID
                      </code>
                      : Read-Group ID;
                     </li>
                     <li>
                      <code>
                       -RGLB
                      </code>
                      : Read-Group library;
                     </li>
                     <li>
                      <code>
                       -RGPL
                      </code>
                      : Read-Group platform;
                     </li>
                     <li>
                      <code>
                       -RGPU
                      </code>
                      : Read-Group platform unit(barcode);
                     </li>
                     <li>
                      <code>
                       -RGSM
                      </code>
                      : Read-Group sample name;
                     </li>
                     <li>
                      <code>
                       -VALIDATION_STRINGENCY
                      </code>
                      : Validation stringency for all SAM files read by this program.
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      NB:
                     </strong>
                     Read group information flags (
                     <code>
                      -RG*
                     </code>
                     ) are mandatory, even if you don't have them.
                    </p>
                    <ol dir="auto" start="3">
                     <li>
                      Remove duplicates:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gatk MarkDuplicatesSpark \
    -I sample_ReadGroups.bam \
    -O sample_MarkDuplicated.bam \
    --spark-master local[16]">
                     <pre class="notranslate"><code>gatk MarkDuplicatesSpark \
    -I sample_ReadGroups.bam \
    -O sample_MarkDuplicated.bam \
    --spark-master local[16]
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -I
                      </code>
                      : input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file;
                     </li>
                     <li>
                      <code>
                       --spark-master
                      </code>
                      : to parallelize the work.
                     </li>
                    </ul>
                    <ol dir="auto" start="4">
                     <li>
                      Detect variant sites in the dataset:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gatk HaplotypeCaller \
    -R reference.fasta \
    -I sample_MarkDuplicated.bam \
    --sample-ploidy 80 -O sample.vcf \
    --max-genotype-count 91881">
                     <pre class="notranslate"><code>gatk HaplotypeCaller \
    -R reference.fasta \
    -I sample_MarkDuplicated.bam \
    --sample-ploidy 80 -O sample.vcf \
    --max-genotype-count 91881
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -R
                      </code>
                      : genome reference;
                     </li>
                     <li>
                      <code>
                       -I
                      </code>
                      : input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file (
                      <code>
                       vcf
                      </code>
                      format);
                     </li>
                     <li>
                      <code>
                       --sample-ploidy
                      </code>
                      : it allows to take in account of using pooled data:
                      <a href="https://render.githubusercontent.com/render/math?math=sample\_ploidy = organism\_ploidy \times pool\_size" rel="noopener noreferrer nofollow" target="_blank">
                       <img src="https://render.githubusercontent.com/render/math?math=sample\_ploidy = organism\_ploidy \times pool\_size" style="max-width: 100%;"/>
                      </a>
                      <br/>
                      In our case we have a ploidy of 2 (
                      <em>
                       D. suzukii
                      </em>
                      is a diploïd organism) and a pool size of 40.
                     </li>
                     <li>
                      <code>
                       --max-genotype-count
                      </code>
                      : maximum number of genotypes to consider at any site:
                     </li>
                     <li>
                      <a href="https://render.githubusercontent.com/render/math?math=max\_genotype\_count = \binom{P +  A - 1}{A - 1} = \frac{(P + A - 1)!}{P!(A - 1)!}" rel="noopener noreferrer nofollow" target="_blank">
                       <img src="https://render.githubusercontent.com/render/math?math=max\_genotype\_count = \binom{P +  A - 1}{A - 1} = \frac{(P + A - 1)!}{P!(A - 1)!}" style="max-width: 100%;"/>
                      </a>
                      <br/>
                      With
                      <code>
                       P
                      </code>
                      la ploidy of the sample (previous formula) and
                      <code>
                       A
                      </code>
                      (
                      <a href="https://render.githubusercontent.com/render/math?math=A = 3" rel="noopener noreferrer nofollow" target="_blank">
                       <img src="https://render.githubusercontent.com/render/math?math=A = 3" style="max-width: 100%;"/>
                      </a>
                      ) the allele count.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.3.1 SNPs filtering
                     </h4>
                     <a aria-label="Permalink: 4.3.1 SNPs filtering" class="anchor" href="#431-snps-filtering" id="user-content-431-snps-filtering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Crete a subset keeping only
                      <strong>
                       SNPs
                      </strong>
                      :
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gatk SelectVariants \
    -R reference_genome.fasta \
    -V input.vcf \
    -O output.vcf \
    --select-type-to-include SNP">
                     <pre class="notranslate"><code>gatk SelectVariants \
    -R reference_genome.fasta \
    -V input.vcf \
    -O output.vcf \
    --select-type-to-include SNP
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -R
                      </code>
                      : genome reference;
                     </li>
                     <li>
                      <code>
                       -V
                      </code>
                      :
                      <code>
                       .vcf
                      </code>
                      input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file;
                     </li>
                     <li>
                      <code>
                       --select-type-to-include
                      </code>
                      select a type of variant.
                     </li>
                    </ul>
                    <ol dir="auto" start="2">
                     <li>
                      Generate a table that can be used to find filter thresholds:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	gatk VariantsToTable \
		-R reference_genome.fasta \
		-V input.vcf \
		-O output.table \
		# scores to save to the table
		-F CHROM \
		-F POS \
		-F QUAL \
		-F AC \
		-F AF \
		-F DP \
		-F QD \
		-F MQ \
		-F FS \
		-F SOR \
		-F MQRankSum \
		-F ReadPosRankSum">
                     <pre class="notranslate"><code>	gatk VariantsToTable \
		-R reference_genome.fasta \
		-V input.vcf \
		-O output.table \
		# scores to save to the table
		-F CHROM \
		-F POS \
		-F QUAL \
		-F AC \
		-F AF \
		-F DP \
		-F QD \
		-F MQ \
		-F FS \
		-F SOR \
		-F MQRankSum \
		-F ReadPosRankSum
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -R
                      </code>
                      : reference;
                     </li>
                     <li>
                      <code>
                       -V
                      </code>
                      :
                      <code>
                       .vcf
                      </code>
                      input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file;
                     </li>
                     <li>
                      <code>
                       -F
                      </code>
                      : selected field.
                     </li>
                    </ul>
                    <p dir="auto">
                     The best way to choose the right filter, is to plot the several scores across the VCF file, for example using
                     <code>
                      ggplot2
                     </code>
                     library in
                     <code>
                      R
                     </code>
                     .
You can find here the plot used to define some scores:
                     <br/>
                    </p>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/DP.svg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/DP.svg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/FS.svg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/FS.svg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/MQ.svg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/MQ.svg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/QD.svg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/QD.svg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/QUAL.svg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/QUAL.svg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/SOR.svg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/SOR.svg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <br/>
                    <ol dir="auto" start="3">
                     <li>
                      Use filter on
                      <code>
                       vcf
                      </code>
                      files:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='	gatk VariantFiltration \
		-R reference_genome.fasta \
		-V input.vcf \
		-O output.vcf \
		# following results observed in plots
		-filter "QUAL &lt; 70" --filter-name Low_Qual\
		-filter "DP &lt; 90" --filter-name Low_Cov \
		-filter "QD &lt; 20.0 || MQ &lt; 28.0 || FS &gt; 1.0 || \
		SOR &gt; 1.4 --filter-name Secondary_filter'>
                     <pre class="notranslate"><code>	gatk VariantFiltration \
		-R reference_genome.fasta \
		-V input.vcf \
		-O output.vcf \
		# following results observed in plots
		-filter "QUAL &lt; 70" --filter-name Low_Qual\
		-filter "DP &lt; 90" --filter-name Low_Cov \
		-filter "QD &lt; 20.0 || MQ &lt; 28.0 || FS &gt; 1.0 || \
		SOR &gt; 1.4 --filter-name Secondary_filter
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -R
                      </code>
                      : reference;
                     </li>
                     <li>
                      <code>
                       -V
                      </code>
                      :
                      <code>
                       vcf
                      </code>
                      input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file;
                     </li>
                     <li>
                      <code>
                       -filter
                      </code>
                      : define a filter;
                      <ul dir="auto">
                       <li>
                        <code>
                         QUAL
                        </code>
                        : Phred-scaled quality score for the assertion made in alternative;
                       </li>
                       <li>
                        <code>
                         DP
                        </code>
                        : read depth at this position for this sample;
                       </li>
                       <li>
                        <code>
                         QD
                        </code>
                        : this is the variant confidence (from the QUAL field) divided by the unfiltered depth of non-hom-ref samples;
                       </li>
                       <li>
                        <code>
                         MQ
                        </code>
                        : mapping quality;
                       </li>
                       <li>
                        <code>
                         FS
                        </code>
                        : this is the Phred-scaled probability that there is strand bias at the site;
                       </li>
                       <li>
                        <code>
                         SOR
                        </code>
                        : this is another way to estimate strand bias using a test similar to the symmetric odds ratio test;
                       </li>
                      </ul>
                     </li>
                     <li>
                      <code>
                       --filter-name
                      </code>
                      : used to assign a name to the filter.
                     </li>
                    </ul>
                    <ol dir="auto" start="4">
                     <li>
                      Extract
                      <strong>
                       SNPs
                      </strong>
                      who passed the filters:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gatk SelectVariants \
		-R reference_genome.fasta \
		-V input.vcf \
		-O output.vcf \
		--exclude-filtered">
                     <pre class="notranslate"><code>gatk SelectVariants \
		-R reference_genome.fasta \
		-V input.vcf \
		-O output.vcf \
		--exclude-filtered
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -R
                      </code>
                      : reference;
                     </li>
                     <li>
                      <code>
                       -V
                      </code>
                      :
                      <code>
                       vcf
                      </code>
                      input file;
                     </li>
                     <li>
                      <code>
                       -O
                      </code>
                      : output file;
                     </li>
                     <li>
                      <code>
                       --exclude-filtered
                      </code>
                      : exclude
                      <strong>
                       SNPs
                      </strong>
                      marked to be filtered.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.3.2 Get
                      <strong>
                       SNPs
                      </strong>
                      that are in transcripts (exons)
                     </h4>
                     <a aria-label="Permalink: 4.3.2 Get SNPs that are in transcripts (exons)" class="anchor" href="#432-get-snps-that-are-in-transcripts-exons" id="user-content-432-get-snps-that-are-in-transcripts-exons">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      From the
                      <code>
                       .gff3
                      </code>
                      file of the reference, we can extract a subset having only exons:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="tail -n +2 Drosophila-suzukii-annotation-3-ws.gff3 | \
awk -F &quot;\t&quot; '{if($3==&quot;exon&quot;) print $0}' &gt; ~/exon.gff3">
                     <pre class="notranslate"><code>tail -n +2 Drosophila-suzukii-annotation-3-ws.gff3 | \
awk -F "\t" '{if($3=="exon") print $0}' &gt; ~/exon.gff3
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       tail -n +2
                      </code>
                      : to skip
                      <code>
                       gff3
                      </code>
                      header;
                     </li>
                     <li>
                      <code>
                       awk
                      </code>
                      : if the third field has 'exon' tag, it print the line in the
                      <code>
                       exon.gff3
                      </code>
                      file.
                     </li>
                    </ul>
                    <ol dir="auto" start="2">
                     <li>
                      Convert the
                      <code>
                       .gff3
                      </code>
                      into
                      <code>
                       .bed
                      </code>
                      format:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gff2bed &lt; exon.gff3 &gt; exon.bed">
                     <pre class="notranslate"><code>gff2bed &lt; exon.gff3 &gt; exon.bed
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Intersect the
                      <code>
                       .vcf
                      </code>
                      file with the
                      <code>
                       .bed
                      </code>
                      file:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="intersectBed -a input.vcf -b exon.bed -header &gt; output.vcf">
                     <pre class="notranslate"><code>intersectBed -a input.vcf -b exon.bed -header &gt; output.vcf
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -a
                      </code>
                      :
                      <code>
                       .vcf
                      </code>
                      input file;
                     </li>
                     <li>
                      <code>
                       -b
                      </code>
                      :
                      <code>
                       .bed
                      </code>
                      input file;
                     </li>
                     <li>
                      <code>
                       -header
                      </code>
                      : if the
                      <code>
                       .bed
                      </code>
                      file has header.
                     </li>
                    </ul>
                    <ol dir="auto" start="4">
                     <li>
                      Count the number of gene for a given sample:
                     </li>
                    </ol>
                    <ul dir="auto">
                     <li>
                      for DNA data we have used
                      <code>
                       gene_count.py
                      </code>
                      that can be founded in the
                      <code>
                       scripts/
                      </code>
                      folder;
                     </li>
                     <li>
                      for the RNA data we have used the following command (car genes are used as chromosome/contig in these files):
                     </li>
                    </ul>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='grep -v "#" file.vcf | cut -f1 | sort | uniq | wc -l'>
                     <pre class="notranslate"><code>grep -v "#" file.vcf | cut -f1 | sort | uniq | wc -l
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.3.3  Get
                      <strong>
                       SNPs
                      </strong>
                      that are in DE genes
                     </h4>
                     <a aria-label="Permalink: 4.3.3  Get SNPs that are in DE genes" class="anchor" href="#433--get-snps-that-are-in-de-genes" id="user-content-433--get-snps-that-are-in-de-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     List of the DE genes can be found here:
                     <br/>
                    </p>
                    <p dir="auto">
                     Location:
                     <em>
                      pedago-ngs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/DE_gene">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/DE_gene
</code></pre>
                    </div>
                    <p dir="auto">
                     This list is obtained by the M. Tabourin's work.
                    </p>
                    <ol dir="auto">
                     <li>
                      Obtain a
                      <code>
                       .gff3
                      </code>
                      file containing only DE genes:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python3 get_de.py">
                     <pre class="notranslate"><code>python3 get_de.py
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       get_de.py
                      </code>
                      : it is a script written by us. It allows, from the
                      <code>
                       gff3
                      </code>
                      and the
                      <code>
                       list.txt
                      </code>
                      having DE genes, to obtain a
                      <code>
                       .gff3
                      </code>
                      file having only DE genes.
                     </li>
                    </ul>
                    <ol dir="auto" start="2">
                     <li>
                      <p dir="auto">
                       Then we have transformed the obtained
                       <code>
                        .gff3
                       </code>
                       file into
                       <code>
                        .bed
                       </code>
                       file using the command
                       <code>
                        3
                       </code>
                       at the previous paragraph and we have used it to obtain
                       <strong>
                        SNPs
                       </strong>
                       in DE genes using command
                       <code>
                        4
                       </code>
                       of the previous paragraph.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Finally we have compute the intersect of the
                       <code>
                        vcf
                       </code>
                       files from DNA and RNA. Unfortunately, these
                       <code>
                        vcf
                       </code>
                       files (DNA and RNA) were generated with a different reference, so we have wrote a script to convert genomic positions of the RNA
                       <code>
                        vcf
                       </code>
                       files to DNA positions. It can be found in the
                       <code>
                        scripts/
                       </code>
                       folder as
                       <code>
                        convert_vcf_pos.py
                       </code>
                       . Then we have run the following commands:
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Compress VCF files
bgzip input_DNA_file.vcf
bgzip input_RNA_file.vcf

# Build indexes for compressed files
tabix -p vcf input_DNA_file.vcf.gz
tabix -p vcf input_RNA_file.vcf.gz

# run the intersection between the two files
bcftools isec -c snps -p output/ input_DNA_file.vcf.gz input_RNA_file.vcf.gz">
                     <pre class="notranslate"><code># Compress VCF files
bgzip input_DNA_file.vcf
bgzip input_RNA_file.vcf

# Build indexes for compressed files
tabix -p vcf input_DNA_file.vcf.gz
tabix -p vcf input_RNA_file.vcf.gz

# run the intersection between the two files
bcftools isec -c snps -p output/ input_DNA_file.vcf.gz input_RNA_file.vcf.gz
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      4.3.4 Results
                     </h4>
                     <a aria-label="Permalink: 4.3.4 Results" class="anchor" href="#434-results" id="user-content-434-results">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Location of
                     <code>
                      .vcf
                     </code>
                     result files:
                     <em>
                      pedago-ngs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# DNA
/localdata/pandata/students/M2_projet_15/GATK_pipeline/gatk_ploidy 
# RNA
/localdata/pandata/students/M2_projet_15/RNAseq/vcf ">
                     <pre class="notranslate"><code># DNA
/localdata/pandata/students/M2_projet_15/GATK_pipeline/gatk_ploidy 
# RNA
/localdata/pandata/students/M2_projet_15/RNAseq/vcf 
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      4.3.4.1 SNPs founded on exons
                     </h5>
                     <a aria-label="Permalink: 4.3.4.1 SNPs founded on exons" class="anchor" href="#4341-snps-founded-on-exons" id="user-content-4341-snps-founded-on-exons">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th align="center">
                       </th>
                       <th align="center">
                        G0
                       </th>
                       <th align="center">
                        cerise
                       </th>
                       <th align="center">
                        fraise
                       </th>
                       <th align="center">
                        gene count
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td align="center">
                        DNA
                       </td>
                       <td align="center">
                        59,055
                       </td>
                       <td align="center">
                        50,077
                       </td>
                       <td align="center">
                        52,243
                       </td>
                       <td align="center">
                        9,837
                       </td>
                      </tr>
                      <tr>
                       <td align="center">
                        RNA
                       </td>
                       <td align="center">
                        28,150
                       </td>
                       <td align="center">
                        29,823
                       </td>
                       <td align="center">
                        31,052
                       </td>
                       <td align="center">
                        5,248
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     In the DNA data we find almost double compared to the RNA, but this can be easily explained by the number of genes that is also double. Nevertheless we observe a decrease from G0 to G12 for DNA, which is rather consistent with the basic hypothesis of having a selection effect.
On the other hand, in RNA, the number does not decrease.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      4.3.4.1 SNPs founded on exons of DE genes
                     </h5>
                     <a aria-label="Permalink: 4.3.4.1 SNPs founded on exons of DE genes" class="anchor" href="#4341-snps-founded-on-exons-of-de-genes" id="user-content-4341-snps-founded-on-exons-of-de-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th align="center">
                       </th>
                       <th align="center">
                        G0
                       </th>
                       <th align="center">
                        cerise
                       </th>
                       <th align="center">
                        fraise
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td align="center">
                        DNA
                       </td>
                       <td align="center">
                        536
                       </td>
                       <td align="center">
                        415
                       </td>
                       <td align="center">
                        492
                       </td>
                      </tr>
                      <tr>
                       <td align="center">
                        RNA
                       </td>
                       <td align="center">
                        17
                       </td>
                       <td align="center">
                        36
                       </td>
                       <td align="center">
                        51
                       </td>
                      </tr>
                      <tr>
                       <td align="center">
                        intersect
                       </td>
                       <td align="center">
                        1
                       </td>
                       <td align="center">
                        8
                       </td>
                       <td align="center">
                        7
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     On differentially expressed genes, there are many more SNPs on the DNA than on the RNA. It is suspected that this may be due to having poolseq data for DNA.
We find again a decrease between G0 and G12 for DNA (is it significant?) but not between G0 and G7 for RNA.
                    </p>
                    <p dir="auto">
                     The SNP founde both in RNA and DNA is on Contig 9. The 8 SNPs for the cherry are founded on Contig 9 (2), Contig 44 (5) and Contig 49 (1). The 7 SNPs for the strawberry are founded on Andromeda (1), Cepheus (1), Contig (5), Contig 9 (1), Contig 10 (1) and Contig 44 (2).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5. Diversity &amp; Differentiation Analysis
                     </h2>
                     <a aria-label="Permalink: 5. Diversity &amp; Differentiation Analysis" class="anchor" href="#5-diversity--differentiation-analysis" id="user-content-5-diversity--differentiation-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Installating PoPoolation: Diversity Analysis
                     </h1>
                     <a aria-label="Permalink: Installating PoPoolation: Diversity Analysis" class="anchor" href="#installating-popoolation-diversity-analysis" id="user-content-installating-popoolation-diversity-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     All the steps come from PoPoolation website (
                     <a href="https://sourceforge.net/p/popoolation/wiki/browse_pages/" rel="nofollow">
                      https://sourceforge.net/p/popoolation/wiki/browse_pages/
                     </a>
                     ) and Dr. Kofler presentation (
                     <a href="https://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part1.pdf" rel="nofollow">
                      https://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part1.pdf
                     </a>
                     )
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="svn checkout https://svn.code.sf.net/p/popoolation/code/trunk popoolation-code
cd popoolation-code
svn update">
                     <pre class="notranslate"><code>svn checkout https://svn.code.sf.net/p/popoolation/code/trunk popoolation-code
cd popoolation-code
svn update
</code></pre>
                    </div>
                    <ol dir="auto">
                     <li>
                      Remove low quality alignments
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time samtools view -q 20 -f 0x0002 -F 0x0004 -F 0x0008 -b G0_MarkDuplicated.bam &gt; G0_q20_MarkDuplicated.bam #26m">
                     <pre class="notranslate"><code>time samtools view -q 20 -f 0x0002 -F 0x0004 -F 0x0008 -b G0_MarkDuplicated.bam &gt; G0_q20_MarkDuplicated.bam #26m
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -q 20
                      </code>
                      only keep reads with a mapping quality higher than 20 (remove ambiguously aligned reads)
                     </li>
                     <li>
                      <code>
                       -f 0x0002
                      </code>
                      only keep proper pairs (remember flags from sam file)
                     </li>
                     <li>
                      <code>
                       -F 0x0004
                      </code>
                      remove reads that are not mapped
                     </li>
                     <li>
                      <code>
                       -F 0x0008
                      </code>
                      remove reads with an un-mapped mate
                     </li>
                    </ul>
                    <p dir="auto">
                     Note
                     <code>
                      -f
                     </code>
                     means only keep reads having the given flag and
                     <code>
                      -F
                     </code>
                     discard all reads having the given flag.
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      Creating a mpileup file
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam &gt; G0.mpileup #20m">
                     <pre class="notranslate"><code>time samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam &gt; G0.mpileup #20m
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -B
                      </code>
                      disable BAQ computation (base alignment quality)
                     </li>
                     <li>
                      <code>
                       -Q
                      </code>
                      skip bases with base quality smaller than the given value
                     </li>
                     <li>
                      <code>
                       -f
                      </code>
                      path to reference sequence
                     </li>
                    </ul>
                    <p dir="auto">
                     MPILEUP FILE:
                    </p>
                    <ul dir="auto">
                     <li>
                      col1 reference chromosome
                     </li>
                     <li>
                      col2 position
                     </li>
                     <li>
                      col3 reference character
                     </li>
                     <li>
                      col4 coverage
                     </li>
                     <li>
                      col5 bases for the given position (’.’ identical to reference character -forward strand; ’,’ identical to reference - reverse strand)
                     </li>
                     <li>
                      col6 quality for the bases
                     </li>
                    </ul>
                    <ol dir="auto" start="3">
                     <li>
                      Filtering indels
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/identify-genomic-indel-regions.pl --indel-window 5 --min-count 2 --input G0.mpileup --output indels_G0.gtf #25m">
                     <pre class="notranslate"><code>time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/identify-genomic-indel-regions.pl --indel-window 5 --min-count 2 --input G0.mpileup --output indels_G0.gtf #25m
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      –indel-window how many bases surrounding indels should be ignored
                     </li>
                     <li>
                      –min-count minimum count for calling an indel. Note that indels may be sequencing errors as well
                     </li>
                    </ul>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/filter-pileup-by-gtf.pl --input G0.mpileup --gtf indels_G0.gtf --output G0.idf.mpileup #20m">
                     <pre class="notranslate"><code>time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/filter-pileup-by-gtf.pl --input G0.mpileup --gtf indels_G0.gtf --output G0.idf.mpileup #20m
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      Subsampling to uniform coverage
                     </li>
                    </ol>
                    <p dir="auto">
                     Several population genetic estimators are sensitive to sequencing errors. For example a very low Tajima’s D, usually indicative of a selective sweep, may be, as an artifact, frequently be found in highly covered regions because these regions have just more sequencing errors. To avoid these kinds of biases we recommend to subsample to an uniform coverage.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/subsample-pileup.pl --min-qual 20 --method withoutreplace --max-coverage 50 --fastq-type sanger --target-coverage 10 --input G0.idf.mpileup --output G0.ss10.idf.mpileup #343m">
                     <pre class="notranslate"><code>time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/basic-pipeline/subsample-pileup.pl --min-qual 20 --method withoutreplace --max-coverage 50 --fastq-type sanger --target-coverage 10 --input G0.idf.mpileup --output G0.ss10.idf.mpileup #343m
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      –min-qual minimum base quality
                     </li>
                     <li>
                      –method method for subsampling, we recommend without replacement
                     </li>
                     <li>
                      –target-coverage which coverage should the resulting mpileup file have
                     </li>
                     <li>
                      –max-coverage the maximum allowed coverage, regions having higher coverages will be ignored (they may be copy number variations and lead to wrong SNPs)
                     </li>
                     <li>
                      –fastq-type (sanger means offset 33)
                     </li>
                    </ul>
                    <ol dir="auto" start="5">
                     <li>
                      Calculating Tajima’s Pi
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/Variance-sliding.pl --fastq-type sanger --measure pi --input G0.ss10.idf.mpileup --min-count 2 --min-coverage 4 --max-coverage 10 --min-covered-fraction 0.5 --pool-size 80 --window-size 1000 --step-size 1000 --output G0.pi --snp-output G0.snps">
                     <pre class="notranslate"><code>time perl ~/Documents/Projet_S3/Popoolation/popoolation-code/Variance-sliding.pl --fastq-type sanger --measure pi --input G0.ss10.idf.mpileup --min-count 2 --min-coverage 4 --max-coverage 10 --min-covered-fraction 0.5 --pool-size 80 --window-size 1000 --step-size 1000 --output G0.pi --snp-output G0.snps
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      –min-coverage –max-coverage: for subsampled files not important; should contain target
coverage, i.e.: 10
                     </li>
                     <li>
                      –min-covered-fraction minimum percentage of sites having sufficient coverage in the given window
                     </li>
                     <li>
                      –min-count minimum occurrence of allele for calling a SNP
                     </li>
                     <li>
                      –measure which population genetics measure should be computed (pi/theta/D)
                     </li>
                     <li>
                      –pool-size number of chromosomes (thus number of diploids times two)
                     </li>
                     <li>
                      –region compute the measure only for a small region; default is the whole genome
                     </li>
                     <li>
                      –output a file containing the measure (π) for the windows
                     </li>
                     <li>
                      –snp-output a file containing for every window the SNPs that have been used for computing the measure (e.g. π)
                     </li>
                     <li>
                      –window-size –step-size control behavior of sliding window; if step size is smaller than window size than the windows will be overlapping.
                     </li>
                    </ul>
                    <p dir="auto">
                     Output G0.pi:
                    </p>
                    <ul dir="auto">
                     <li>
                      col 1: reference chromosome
                     </li>
                     <li>
                      col 2: position of window (mean value)
                     </li>
                     <li>
                      col 3: number of SNPs in the given window
                     </li>
                     <li>
                      col 4: fraction of sites in the window having sufficient coverage (min ≤ x ≤ max)
                     </li>
                     <li>
                      col 5: measure for the window (π)
                     </li>
                    </ul>
                    <p dir="auto">
                     Output G0.snps:
                    </p>
                    <ul dir="auto">
                     <li>
                      col 1: reference chromosome
                     </li>
                     <li>
                      col 2: position of SNP
                     </li>
                     <li>
                      col 3: reference character
                     </li>
                     <li>
                      col 4: coverage
                     </li>
                     <li>
                      col 5-9: counts of A, T, C, G, N respectively
                     </li>
                    </ul>
                    <p dir="auto">
                     RESULTS:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation
</code></pre>
                    </div>
                    <div align="center" dir="auto">
                     <table>
                      <thead>
                       <tr>
                        <th align="center">
                         Theta Pi
                        </th>
                        <th align="center">
                         G0
                        </th>
                        <th align="center">
                         G12 cranberry
                        </th>
                        <th align="center">
                         G12 cerise
                        </th>
                        <th align="center">
                         G12 fraise
                        </th>
                       </tr>
                      </thead>
                      <tbody>
                       <tr>
                        <td align="center">
                         moyenne
                        </td>
                        <td align="center">
                         0.00830
                        </td>
                        <td align="center">
                         0.00771
                        </td>
                        <td align="center">
                         0.00939
                        </td>
                        <td align="center">
                         0.00864
                        </td>
                       </tr>
                       <tr>
                        <td align="center">
                         écart type
                        </td>
                        <td align="center">
                         0.01250
                        </td>
                        <td align="center">
                         0.01289
                        </td>
                        <td align="center">
                         0.01468
                        </td>
                        <td align="center">
                         0.01418
                        </td>
                       </tr>
                      </tbody>
                     </table>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Installating PoPoolation2: Differentiation Analysis
                     </h1>
                     <a aria-label="Permalink: Installating PoPoolation2: Differentiation Analysis" class="anchor" href="#installating-popoolation2-differentiation-analysis" id="user-content-installating-popoolation2-differentiation-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     All the steps comes from PoPoolation website (
                     <a href="https://sourceforge.net/p/popoolation2/wiki/Home/" rel="nofollow">
                      https://sourceforge.net/p/popoolation2/wiki/Home/
                     </a>
                     ) and Dr. Kofler presentation (
                     <a href="https://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part2.pdf" rel="nofollow">
                      https://www.kofler.or.at/bioinformatic/wp-content/uploads/2018/07/pooledAnalysis_part2.pdf
                     </a>
                     )
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="svn checkout https://svn.code.sf.net/p/popoolation2/code/trunk popoolation2-code
cd popoolation2-code/
svn update">
                     <pre class="notranslate"><code>svn checkout https://svn.code.sf.net/p/popoolation2/code/trunk popoolation2-code
cd popoolation2-code/
svn update
</code></pre>
                    </div>
                    <ol dir="auto">
                     <li>
                      Create a mpileup
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam G12_cerise_q20_MarkDuplicated.bam G12_cran_q20_MarkDuplicated.bam G12_fraise_q20_MarkDuplicated.bam &gt; p1-4.mpileup # 77m">
                     <pre class="notranslate"><code>time samtools mpileup -B -Q 0 -f Drosophila-suzukii-contig.fasta G0_q20_MarkDuplicated.bam G12_cerise_q20_MarkDuplicated.bam G12_cran_q20_MarkDuplicated.bam G12_fraise_q20_MarkDuplicated.bam &gt; p1-4.mpileup # 77m
</code></pre>
                    </div>
                    <p dir="auto">
                     MPILEUP:
                    </p>
                    <p dir="auto">
                     Very similar to the mpileup with one sample. In this multi-pileup (mpileup) three additional columns are created for each additional sample. Thus for each sample the following information is provided:
                    </p>
                    <ul dir="auto">
                     <li>
                      the coverage, columns 4 + n ∗ 3
                     </li>
                     <li>
                      the bases, columns 5 + n ∗ 3
                     </li>
                     <li>
                      the corresponding base quality, columns 6 + n ∗ 3
                     </li>
                    </ul>
                    <ol dir="auto" start="2">
                     <li>
                      Conversion to sync-file
                     </li>
                    </ol>
                    <p dir="auto">
                     In order to use PoPoolation2 we have to convert the mpileup to a sync file.
This may seem unnecessary to you, but it serves to speed up the analysis,
because the time consuming part of the analysis - parsing of the mpileup file -
needs just to be performed once.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time java -jar ~/Documents/Projet_S3/Popoolation/popoolation2-code/mpileup2sync.jar --input p1-4.mpileup --output p1-4.sync --fastq-type sanger --min-qual 20 --threads 15 #78m">
                     <pre class="notranslate"><code>time java -jar ~/Documents/Projet_S3/Popoolation/popoolation2-code/mpileup2sync.jar --input p1-4.mpileup --output p1-4.sync --fastq-type sanger --min-qual 20 --threads 15 #78m
</code></pre>
                    </div>
                    <p dir="auto">
                     THE SYNC-FILE:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       col 1: reference chromosome
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       col 2: position
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       col 3: reference character
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       col 4: allele counts for first population
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       col 5: allele counts for second population
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       col n: allele counts for n-3 population
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Allele counts are in the form ”A:T:C:G:N:del”
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       the sync-file provides a convenient summary of the allele counts of several populations (there
is no upper threshold of the population number).
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       subsampling to an uniform coverage with the PoPoolation2 pipeline should be done using such a sync-file
                      </p>
                     </li>
                    </ul>
                    <ol dir="auto" start="3">
                     <li>
                      Calculating the Fst
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="time perl ~/Documents/Projet_S3/Popoolation/popoolation2-code/fst-sliding.pl --window-size 1 --step-size 1 --suppress-noninformative --input p1-4.sync --min-covered-fraction 1.0 --min-coverage 4 --max-coverage 120 --min-count 3 --output fst_tot.txt --pool-size 80 #386m">
                     <pre class="notranslate"><code>time perl ~/Documents/Projet_S3/Popoolation/popoolation2-code/fst-sliding.pl --window-size 1 --step-size 1 --suppress-noninformative --input p1-4.sync --min-covered-fraction 1.0 --min-coverage 4 --max-coverage 120 --min-count 3 --output fst_tot.txt --pool-size 80 #386m
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      --suppress-noninformative suppresses output for windows not containing a SNP. When applied to windows of size one, this option suppresses output for bases that are no SNP.
                     </li>
                    </ul>
                    <p dir="auto">
                     OUTPUT OF THE Fst SCRIPT:
                    </p>
                    <ul dir="auto">
                     <li>
                      col 1: reference chromosome
                     </li>
                     <li>
                      col 2: position
                     </li>
                     <li>
                      col 3: window size (1 for single SNPs)
                     </li>
                     <li>
                      col 4: covered fraction (relevant for minimum covered fraction)
                     </li>
                     <li>
                      col 5: average minimum coverage for the window across all populations (the higher the more reliable the estimate)
                     </li>
                     <li>
                      col 6: pairwise Fst comparing population 1 with population 2
                     </li>
                     <li>
                      col 7: etc for ALL pairwise comparisons of the populations present in the sync file
                     </li>
                    </ul>
                    <p dir="auto">
                     RESULTS:
                    </p>
                    <p dir="auto">
                     We separate the fst_tot.txt file in 3 files each containing one comparison with G0.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat fst_tot.txt | sed 's/1:2=//' | awk  -F&quot; &quot; '{print $1&quot; &quot; $2&quot; &quot; $6}'&gt; Fst_1-2.txt
cat fst_tot.txt | sed 's/1:3=//' | awk  -F&quot; &quot; '{print $1&quot; &quot; $2&quot; &quot; $7}'&gt; Fst_1-3.txt
cat fst_tot.txt | sed 's/1:4=//' | awk  -F&quot; &quot; '{print $1&quot; &quot; $2&quot; &quot; $8}'&gt; Fst_1-4.txt">
                     <pre class="notranslate"><code>cat fst_tot.txt | sed 's/1:2=//' | awk  -F" " '{print $1" " $2" " $6}'&gt; Fst_1-2.txt
cat fst_tot.txt | sed 's/1:3=//' | awk  -F" " '{print $1" " $2" " $7}'&gt; Fst_1-3.txt
cat fst_tot.txt | sed 's/1:4=//' | awk  -F" " '{print $1" " $2" " $8}'&gt; Fst_1-4.txt
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation2">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/GATK_pipeline/PoPoolation2
</code></pre>
                    </div>
                    <p dir="auto">
                     Fst_1-2.txt: G0-G12_cerise
                    </p>
                    <p dir="auto">
                     Fst_1-3.txt: G0-G12_cran
                    </p>
                    <p dir="auto">
                     Fst_1-4.txt: G0-G12_fraise
                    </p>
                    <div align="center" dir="auto">
                     <table>
                      <thead>
                       <tr>
                        <th align="center">
                         Fst totalité des SNP
                        </th>
                        <th align="center">
                         G0-G12 cranberry
                        </th>
                        <th align="center">
                         G0-G12 cerise
                        </th>
                        <th align="center">
                         G0-G12 fraise
                        </th>
                       </tr>
                      </thead>
                      <tbody>
                       <tr>
                        <td align="center">
                         moyenne
                        </td>
                        <td align="center">
                         0.02347
                        </td>
                        <td align="center">
                         0.03759
                        </td>
                        <td align="center">
                         0.03145
                        </td>
                       </tr>
                       <tr>
                        <td align="center">
                         écart type
                        </td>
                        <td align="center">
                         0.03630
                        </td>
                        <td align="center">
                         0.05538
                        </td>
                        <td align="center">
                         0.04696
                        </td>
                       </tr>
                      </tbody>
                     </table>
                    </div>
                    <div align="center" dir="auto">
                     <table>
                      <thead>
                       <tr>
                        <th align="center">
                         Fst gènes DE
                        </th>
                        <th align="center">
                         G0-G12 cranberry
                        </th>
                        <th align="center">
                         G0-G12 cerise
                        </th>
                        <th align="center">
                         G0-G12 fraise
                        </th>
                       </tr>
                      </thead>
                      <tbody>
                       <tr>
                        <td align="center">
                         moyenne
                        </td>
                        <td align="center">
                         0.02188
                        </td>
                        <td align="center">
                         0.03429
                        </td>
                        <td align="center">
                         0.02229
                        </td>
                       </tr>
                       <tr>
                        <td align="center">
                         écart type
                        </td>
                        <td align="center">
                         0.03374
                        </td>
                        <td align="center">
                         0.04879
                        </td>
                        <td align="center">
                         0.03373
                        </td>
                       </tr>
                      </tbody>
                     </table>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      Calculating Fst for DE genes:
                     </li>
                    </ol>
                    <p dir="auto">
                     We write a script using Fst results and a list of DE gene in a bed format to calculate Fst for DE gene.
As we use dichotomous search, the Fst file must be sorted like the pipeline does.
You find it on the github repo:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" scripts/Fst_DE_Genes.py">
                     <pre class="notranslate"><code> scripts/Fst_DE_Genes.py
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      Visualisation of Fst on chromosomes:
                     </li>
                    </ol>
                    <p dir="auto">
                     We write a function in R to visualize the Fst on a chosen chromosome.
You find it on the github repo:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" scripts/VisualisationFst.R">
                     <pre class="notranslate"><code> scripts/VisualisationFst.R
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/img/Fst.png" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/img/Fst.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      6. TE Analysis
                     </h2>
                     <a aria-label="Permalink: 6. TE Analysis" class="anchor" href="#6-te-analysis" id="user-content-6-te-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     All the steps are included in a tool named
                     <a href="https://github.com/clemgoub/dnaPipeTE">
                      dnaPipeTE
                     </a>
                    </p>
                    <ul dir="auto">
                     <li>
                      Uniform samplings of the reads to produce low coverage data sets (read sampling to have &lt;1X coverage of the genome to keep only repeated regions)
                     </li>
                     <li>
                      Trinity, repeated assembly of contigs
                     </li>
                     <li>
                      RepeatMasker, contigs annotation
                     </li>
                     <li>
                      Blastn, repeat quantification
                     </li>
                    </ul>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/TE_analysis/dnapipete.jpg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/TE_analysis/dnapipete.jpg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Figure from the
                     <a href="https://academic.oup.com/gbe/article/7/4/1192/533768" rel="nofollow">
                      tool article
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      6.1 Running DnaPipeTE
                     </h3>
                     <a aria-label="Permalink: 6.1 Running DnaPipeTE" class="anchor" href="#61-running-dnapipete" id="user-content-61-running-dnapipete">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#dnaPipeTE: version 1.3 (uses Perl 5, R v3.0.2, Python v3.8.5, Trinity v2.5.1, 
			      RepeatMasker v4.0.5 including RMblastn, ncbi-blast v2.2.28+)">
                     <pre class="notranslate"><code>#dnaPipeTE: version 1.3 (uses Perl 5, R v3.0.2, Python v3.8.5, Trinity v2.5.1, 
			      RepeatMasker v4.0.5 including RMblastn, ncbi-blast v2.2.28+)
</code></pre>
                    </div>
                    <p dir="auto">
                     We had a lot of dependancies problems running dnaPipeTE so we used the
                     <a href="https://hub.docker.com/r/clemgoub/dnapipete" rel="nofollow">
                      docker version
                     </a>
                     .
                    </p>
                    <ol dir="auto">
                     <li>
                      Copy input data
                     </li>
                    </ol>
                    <p dir="auto">
                     Two different types of input are necessary to run the pipeline :
                    </p>
                    <ul dir="auto">
                     <li>
                      The fastq files for each samples
                     </li>
                     <li>
                      <code>
                       Drosophila_Transposable_Element_all.fasta
                      </code>
                      which is the Drosophila TE library from Repbase
                     </li>
                    </ul>
                    <p dir="auto">
                     Location :
                     <em>
                      pedago-ngs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/localdata/pandata/students/M2_projet_15/TE/">
                     <pre class="notranslate"><code>/localdata/pandata/students/M2_projet_15/TE/
</code></pre>
                    </div>
                    <p dir="auto">
                     They all had to be in a
                     <code>
                      ~/Project
                     </code>
                     folder which is the communication folder between the computer and the container.
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      Starting the container
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sudo docker pull clemgoub/dnapipete:latest
sudo docker run -it -v ~/Project:/mnt clemgoub/dnapipete:latest">
                     <pre class="notranslate"><code>sudo docker pull clemgoub/dnapipete:latest
sudo docker run -it -v ~/Project:/mnt clemgoub/dnapipete:latest
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Run dnaPipeTE
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /opt/dnaPipeTE
python3 dnaPipeTE.py \ 
		-input /mnt/data/sample_1.fastq.gz 
		-output /mnt/output_dnapipete/sample 
		-cpu 6 
		-genome_size 270000000 
		-genome_coverage 0.2 
		-sample_number 2 
		-RM_lib /mnt/Drosophila_Transposable_Element_all.fasta">
                     <pre class="notranslate"><code>cd /opt/dnaPipeTE
python3 dnaPipeTE.py \ 
		-input /mnt/data/sample_1.fastq.gz 
		-output /mnt/output_dnapipete/sample 
		-cpu 6 
		-genome_size 270000000 
		-genome_coverage 0.2 
		-sample_number 2 
		-RM_lib /mnt/Drosophila_Transposable_Element_all.fasta
</code></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <code>
                       -input
                      </code>
                      : input fastq or fastq.gz files (single end only);
                     </li>
                     <li>
                      <code>
                       -output
                      </code>
                      : complete path with name for the outputs;
                     </li>
                     <li>
                      <code>
                       -cpu
                      </code>
                      : maximum number of cpu to use;
                     </li>
                     <li>
                      <code>
                       -genome_size
                      </code>
                      : here we know that
                      <em>
                       D.suzukii
                      </em>
                      is 270Mb long;
                     </li>
                     <li>
                      <code>
                       -genome_coverage
                      </code>
                      : for this genome size 0.2X is recommended in the Supplementary Material of the
                      <a href="https://academic.oup.com/gbe/article/7/4/1192/533768" rel="nofollow">
                       tool article
                      </a>
                      ;
                     </li>
                     <li>
                      <code>
                       -sample_number
                      </code>
                      : number of Trinity iterations (2 is the minimum and recommended number of iterations);
                     </li>
                     <li>
                      <code>
                       -RM_lib
                      </code>
                      : path to custom repeat library for RepeatMasker.
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      6.2 Results
                     </h3>
                     <a aria-label="Permalink: 6.2 Results" class="anchor" href="#62-results" id="user-content-62-results">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Example of sampling for G0 :
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="total number of reads: 121247626
maximum number of reads to sample:  710526
fastq :  /mnt/G0-MTP_1.fastq
sampling 2 samples of max 355263 reads to reach coverage...
53999976 bases sampled in 355263 reads
s_G0-MTP_1.fastq done.
53999976 bases sampled in 710526 reads
s_G0-MTP_1.fastq done.">
                     <pre class="notranslate"><code>total number of reads: 121247626
maximum number of reads to sample:  710526
fastq :  /mnt/G0-MTP_1.fastq
sampling 2 samples of max 355263 reads to reach coverage...
53999976 bases sampled in 355263 reads
s_G0-MTP_1.fastq done.
53999976 bases sampled in 710526 reads
s_G0-MTP_1.fastq done.
</code></pre>
                    </div>
                    <p dir="auto">
                     The pipeline gives several useful outputs such as the
                     <strong>
                      Landscapes
                     </strong>
                     (TE divergence to estimate the insertions) and
                     <strong>
                      PieChart
                     </strong>
                     (TE families proportions in the genome).
Other graphs have been created with the script
                     <code>
                      graphs.R
                     </code>
                     to compare TE throught different sample generations.
                    </p>
                    <p dir="auto">
                     <a href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/blob/main/TE_analysis/prop_ech_par_family.jpg" rel="noopener noreferrer" target="_blank">
                      <img src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/raw/main/TE_analysis/prop_ech_par_family.jpg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Other graphs can be found in the
                     <code>
                      graphs.pdf
                     </code>
                     document.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      7. Authors
                     </h2>
                     <a aria-label="Permalink: 7. Authors" class="anchor" href="#7-authors" id="user-content-7-authors">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The project was developped by:
                    </p>
                    <ul dir="auto">
                     <li>
                      Chloé AUJOULAT
                     </li>
                     <li>
                      Tommaso BARBERIS
                     </li>
                     <li>
                      Bertrand HUGUENIN-BIZOT
                     </li>
                     <li>
                      Marie VERNERET
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      8. References
                     </h2>
                     <a aria-label="Permalink: 8. References" class="anchor" href="#8-references" id="user-content-8-references">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Anand, Santosh, Eleonora Mangano, Nadia Barizzone, Roberta Bordoni, Melissa Sorosina, Ferdinando Clarelli, Lucia Corrado, Filippo Martinelli Boneschi, Sandra D’Alfonso, et Gianluca De Bellis. « Next Generation Sequencing of Pooled Samples: Guideline for Variants’ Filtering ». Scientific Reports 6, no 1 (27 septembre 2016): 33735.
                      <a href="https://doi.org/10.1038/srep33735" rel="nofollow">
                       https://doi.org/10.1038/srep33735
                      </a>
                      .
                     </li>
                     <li>
                      Chiu, Joanna C, Xuanting Jiang, Li Zhao, Christopher A Hamm, Julie M Cridland, Perot Saelao, Kelly A Hamby, et al. « Genome of Drosophila suzukii, the Spotted Wing Drosophila ». G3 Genes|Genomes|Genetics 3, no 12 (1 décembre 2013): 2257‑71.
                      <a href="https://doi.org/10.1534/g3.113.008185" rel="nofollow">
                       https://doi.org/10.1534/g3.113.008185
                      </a>
                      .
                     </li>
                     <li>
                      Kofler, Robert, Daniel Gómez-Sánchez, et Christian Schlötterer. « PoPoolationTE2: Comparative Population Genomics of Transposable Elements Using Pool-Seq ». Molecular Biology and Evolution 33, no 10 (octobre 2016): 2759‑64.
                      <a href="https://doi.org/10.1093/molbev/msw137" rel="nofollow">
                       https://doi.org/10.1093/molbev/msw137
                      </a>
                      .
                     </li>
                     <li>
                      Olazcuaga, Laure, Julien Foucaud, Mathieu Gautier, Candice Deschamps, Anne Loiseau, Nicolas Leménager, Benoit Facon, et al. « Adaptation and Correlated Fitness Responses over Two Time Scales in Drosophila Suzukii Populations Evolving in Different Environments ». Journal of Evolutionary Biology 34, no 8 (2021): 1225‑40.
                      <a href="https://doi.org/10.1111/jeb.13878" rel="nofollow">
                       https://doi.org/10.1111/jeb.13878
                      </a>
                      .
                     </li>
                     <li>
                      Clément Goubert, Laurent Modolo, Cristina Vieira, Claire ValienteMoro, Patrick Mavingui, Matthieu Boulesteix, De Novo Assembly and Annotation of the Asian Tiger Mosquito (Aedes albopictus) Repeatome with dnaPipeTE from Raw Genomic Reads and Comparative Analysis with the Yellow Fever Mosquito (Aedes aegypti), Genome Biology and Evolution, Volume 7, Issue 4, April 2015, Pages 1192–1205,
                      <a href="https://doi.org/10.1093/gbe/evv050" rel="nofollow">
                       https://doi.org/10.1093/gbe/evv050
                      </a>
                     </li>
                     <li>
                      Marie  Tabourin  et  al.  “Analyse  de  l’expression  des  gènes  et  des  éléments  transposableschez Drosophila suzukii suite à une expérience d’évolution expérimentale.
                     </li>
                     <li>
                      Simon Andrews et al.FastQC. Babraham Institute. Babraham, UK, Jan. 2012,
                      <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" rel="nofollow">
                       https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
                      </a>
                      .
                     </li>
                     <li>
                      MultiQC: Summarize analysis results for multiple tools and samples in a single report, Philip Ewels, Måns Magnusson, Sverker, Lundin and Max Käller, Bioinformatics (2016), doi: 10.1093/bioinformatics/btw354, PMID: 27312411.
                     </li>
                     <li>
                      Li H. and Durbin R. (2009) Fast and accurate short read alignment with Burrows-Wheeler Transform. Bioinformatics, 25:1754-60. [PMID: 19451168].
                     </li>
                     <li>
                      Heng Li, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, Richard Durbin, 1000 Genome Project Data Processing Subgroup, The Sequence Alignment/Map format and SAMtools, Bioinformatics, Volume 25, Issue 16, 15 August 2009, Pages 2078–2079,
                      <a href="https://doi.org/10.1093/bioinformatics/btp352" rel="nofollow">
                       https://doi.org/10.1093/bioinformatics/btp352
                      </a>
                      .
                     </li>
                     <li>
                      Van der Auwera GA &amp; O'Connor BD. (2020). Genomics in the Cloud: Using Docker, GATK, and WDL in Terra (1st Edition). O'Reilly Media.
                     </li>
                     <li>
                      Wickham H (2016). ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York. ISBN 978-3-319-24277-4,
                      <a href="https://ggplot2.tidyverse.org" rel="nofollow">
                       https://ggplot2.tidyverse.org
                      </a>
                      .
                     </li>
                     <li>
                      R Core Team (2021). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL
                      <a href="https://www.R-project.org/" rel="nofollow">
                       https://www.R-project.org/
                      </a>
                      .
                     </li>
                     <li>
                      Bedops.readthedocs.io. 2021. 6.3.3.5. gff2bed — BEDOPS v2.4.40. (online) Available at:
                      <a href="https://bedops.readthedocs.io/en/latest/content/reference/file-management/conversion/gff2bed.html" rel="nofollow">
                       https://bedops.readthedocs.io/en/latest/content/reference/file-management/conversion/gff2bed.html
                      </a>
                      (Accessed 2 December 2021).
                     </li>
                     <li>
                      Bedtools.readthedocs.io. 2021. intersect — bedtools 2.30.0 documentation. (online) Available at:
                      <a href="https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html" rel="nofollow">
                       https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html
                      </a>
                      (Accessed 2 December 2021).
                     </li>
                    </ul>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" type="hidden" value="CS8sGHNBmllT/rcqHPK53/bDLLUO677xlI0EAG93f24/Cjb7IUmqmjjnT+Tz9ka0zaEc82m0QQJevr/VKEtKOw==">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description or website provided.
                </div>
                <h3 class="sr-only">
                 Topics
                </h3>
                <div class="my-3">
                 <div class="f6">
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:bioinformatics" data-view-component="true" href="/topics/bioinformatics" title="Topic: bioinformatics">
                   bioinformatics
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:genomics" data-view-component="true" href="/topics/genomics" title="Topic: genomics">
                   genomics
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:ngs" data-view-component="true" href="/topics/ngs" title="Topic: ngs">
                   ngs
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:drosophila" data-view-component="true" href="/topics/drosophila" title="Topic: drosophila">
                   drosophila
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:dnaseq" data-view-component="true" href="/topics/dnaseq" title="Topic: dnaseq">
                   dnaseq
                  </a>
                 </div>
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#MIT-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  MIT license
                 </a>
                </div>
                <include-fragment src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FTommasoBarberis%2FGenomic-and-transcriptomic-evolution-of-D.-Suzukii&amp;report=TommasoBarberis+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/graphs/contributors">
                 Contributors
                 <span class="Counter ml-1" data-view-component="true" title="4">
                  4
                 </span>
                </a>
               </h2>
               <include-fragment aria-busy="true" aria-label="Loading contributors" src="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/contributors_list?count=4&amp;current_repository=Genomic-and-transcriptomic-evolution-of-D.-Suzukii&amp;items_to_show=4">
                <ul class="list-style-none">
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                </ul>
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="R 66.7" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 66.7%;">
                 </span>
                 <span aria-label="Python 31.3" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#3572A5 !important;;width: 31.3%;">
                 </span>
                 <span aria-label="Shell 2.0" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#89e051 !important;;width: 2.0%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   66.7%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/search?l=python">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#3572A5;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Python
                  </span>
                  <span>
                   31.3%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/TommasoBarberis/Genomic-and-transcriptomic-evolution-of-D.-Suzukii/search?l=shell">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#89e051;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Shell
                  </span>
                  <span>
                   2.0%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
