<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         acpearson888/advanced_genomics_apearson
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="90A8:5F4D:35D4A0:4C175A:6605CBB2" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjkwQTg6NUY0RDozNUQ0QTA6NEMxNzVBOjY2MDVDQkIyIiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="909c37fe1eb0facaee2ad523323835dd593ae684aabf6f67c2c87146678c2dce" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:331965963" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Contribute to acpearson888/advanced_genomics_apearson development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/acpearson888/advanced_genomics_apearson" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/89f3cf72cd7fba2e1943a38ce08bbf2a11b25811a28bd4ce846e9659bdd9b736/acpearson888/advanced_genomics_apearson" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="acpearson888/advanced_genomics_apearson" name="twitter:title">
                       <meta content="Contribute to acpearson888/advanced_genomics_apearson development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/89f3cf72cd7fba2e1943a38ce08bbf2a11b25811a28bd4ce846e9659bdd9b736/acpearson888/advanced_genomics_apearson" property="og:image"/>
                        <meta content="Contribute to acpearson888/advanced_genomics_apearson development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="acpearson888/advanced_genomics_apearson" property="og:title"/>
                        <meta content="https://github.com/acpearson888/advanced_genomics_apearson" property="og:url"/>
                        <meta content="Contribute to acpearson888/advanced_genomics_apearson development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjI5NzEzNzc2MDAsInQiOjE3MTE2NTU4NjJ9--30b5d987e18ad5c51282728d15365d7ad6542d701ea67f8c5446b1a413afb40a" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/acpearson888/advanced_genomics_apearson git https://github.com/acpearson888/advanced_genomics_apearson.git" name="go-import"/>
                        <meta content="48361260" name="octolytics-dimension-user_id">
                         <meta content="acpearson888" name="octolytics-dimension-user_login">
                          <meta content="331965963" name="octolytics-dimension-repository_id">
                           <meta content="acpearson888/advanced_genomics_apearson" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="331965963" name="octolytics-dimension-repository_network_root_id">
                               <meta content="acpearson888/advanced_genomics_apearson" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/acpearson888/advanced_genomics_apearson" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8" data-view-component="true" id="dialog-show-dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8-description" aria-labelledby="dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-fa1b1af7-dc55-44cb-a6c8-9b1cde3eafb8-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-5e4d696d-dbf6-46a5-9e6f-eda0958da0f2">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-eab68cb9-d027-453a-9735-ac81effb6128">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-34680fb5-ffed-4141-af5f-3778c5571bb1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-0c9a9d29-da30-474e-9bc7-0df09f7b99b0">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-618a78e7-b338-4579-9770-495d08d4f24e">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-2c304a32-fbad-40f9-9655-29a50271cd99">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-ad59c590-f2e0-410d-b0e5-57ab7e8a8e3c">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-a972a620-0b40-46a8-bc64-12118f546835">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               Â© 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: acpearson888 / advanced_genomics_apearson" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              acpearson888
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              advanced_genomics_apearson
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"acpearson888","screen_size":"compact"}' data-view-component="true" href="/acpearson888">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 acpearson888
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"advanced_genomics_apearson","screen_size":"compact"}' data-view-component="true" href="/acpearson888/advanced_genomics_apearson">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 advanced_genomics_apearson
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"acpearson888","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/acpearson888/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/acpearson888">
             <span class="AppHeader-context-item-label">
              acpearson888
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"advanced_genomics_apearson","screen_size":"full"}' data-view-component="true" href="/acpearson888/advanced_genomics_apearson">
             <span class="AppHeader-context-item-label">
              advanced_genomics_apearson
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="cSIKvnXv2i4MAhDAP-3JazihDATA9z0kX9l0Mv2f_vchAQY252IF4buYSO6aeiswDC3M7_TciLWBogqMUFtDHw" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="acpearson888" data-current-repository="acpearson888/advanced_genomics_apearson" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="3UA56mNFtLSvBNsoIafvdqeRf_j37vpiQ88RcfJOFmWVii7ARQP9wfioqfHHoh9n9TU9UdGJCg0vLVDaFXANKQ" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:acpearson888/advanced_genomics_apearson">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump toâ¦" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump toâ¦" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-044138eb-e843-44ff-a572-ac7caa8efb32" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-6de2459e-dfd8-497c-ad1e-a9d34c4cdb29" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-6de2459e-dfd8-497c-ad1e-a9d34c4cdb29">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="cu7Sih4NRqK_O2SzWGDT8HfG-AQ9PddF3TZLUaeOC6HgXMuy1tt1OA8ZardmedIBqs01x2cdMRDTtl7rUd8OGg"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="8NYN6x3NZIj8o-TNVJJx_1HAAZUVfl0ppIGcX70-MQm8JG9RRUw6UbfKWWAeR7zCmJBlgkN7H_IH2U3nonloXA">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="7Qp1iPqyqJX-Kua7NPBQvSEZ7yUD3fD9blJlvqz3DN31VzgurTABln0-rzh0og7Xl97qoi0EEGUD8sdslJjfFQ">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="s004I5I1il7tDG1hHUxoHFLcq0vmPH122uAP-K_DZDJ0GFcBizJMPArX95zU9xMPgmRzgWOau0bVg1FdN1cFOA">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-4edad9e3-9939-4c44-a372-137993ee17be" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-093a63cd-7909-4db4-afeb-5b206ee08b0f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-149d59d0-ce08-4d86-bb51-baa4f584ebc1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-acb55d7f-c213-437b-b531-edeb8b9541f5" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-631d63a6-c6e0-4941-ac87-cd1fdce9f474" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-b244fbb0-8f49-4603-9f5d-807d112262b0" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-ac256817-c216-4949-aed5-20a895bae260" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-9dad00bb-6200-42f6-936e-40f4960bdedc">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-9dad00bb-6200-42f6-936e-40f4960bdedc" id="tooltip-ac256817-c216-4949-aed5-20a895bae260" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-e2f71a03-0e22-4db1-b9f4-2687d2e9c970" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-5c045249-081d-4fe2-a067-76d45fc88a48">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-5c045249-081d-4fe2-a067-76d45fc88a48" id="tooltip-e2f71a03-0e22-4db1-b9f4-2687d2e9c970" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTg2Mn0=--8d35f53fbbc8f17375cd671d23eaa7e8c7a007e12a80174a6a65466f3d1f58d4" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-8b23b5ff-3449-4f11-bf74-e62e80ca7587" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=331965963">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-ea0d81d7-9578-4228-9f60-3907191507f9" data-view-component="true" id="dialog-show-dialog-ea0d81d7-9578-4228-9f60-3907191507f9" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-ea0d81d7-9578-4228-9f60-3907191507f9-description" aria-labelledby="dialog-ea0d81d7-9578-4228-9f60-3907191507f9-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-ea0d81d7-9578-4228-9f60-3907191507f9">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-ea0d81d7-9578-4228-9f60-3907191507f9-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-303877d2-4462-438c-ae12-baa95d575498" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-f19c4f22-91ea-4385-a5d8-8a039b731ef6" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-408c97e7-344c-43eb-a95a-33bb7cd967b0" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-7f637760-5e7d-408b-ac69-3a351f0f300e" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-566a5312-a4f3-43c8-9aa9-b5a22e070e31" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-d83ac1c2-0c73-4a96-8d91-0eedcddd8aa5" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-ea0d81d7-9578-4228-9f60-3907191507f9" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-ea0d81d7-9578-4228-9f60-3907191507f9-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-a341aab7-a385-4c67-a762-637b63a8175d" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-f3741931-e2ed-47a6-81a2-ee535e7df46b">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-e446164d-296d-4e40-8029-27e789cd5555" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-f24feee2-2da1-4c15-a161-18444fcb14a0">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-47a24bfc-1dde-444c-88aa-41560fab771e">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-3579aa3c-e1b6-4171-8407-c463c5c8391a" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-aef151ac-f379-4c3b-a175-c17538b6067d" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-6d3e102c-78bf-43aa-a3b3-d9f8821342c6">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-b851bbfa-23ea-43eb-a6e3-aceb64be71d0">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-a872570d-2f97-480d-a380-a45fa4d5aaae">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-e274da5e-e658-4db3-8002-7303ce97f9cc" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-93ad0a2e-16c6-4229-8218-76127841c8a3" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-122afe81-a3b1-430a-8f2e-7a75bac1694f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-d136fada-36dd-4e6d-85dc-432ef4f6ae17">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-41a65a8c-361b-4006-9653-3fcae7ae8f68">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-c77b7167-8b85-4a00-8b46-1ac404834849">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /acpearson888/advanced_genomics_apearson" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/acpearson888/advanced_genomics_apearson" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /acpearson888/advanced_genomics_apearson/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /acpearson888/advanced_genomics_apearson/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /acpearson888/advanced_genomics_apearson/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /acpearson888/advanced_genomics_apearson/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /acpearson888/advanced_genomics_apearson/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/acpearson888/advanced_genomics_apearson/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /acpearson888/advanced_genomics_apearson/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-list" aria-haspopup="true" aria-labelledby="tooltip-49ffc47b-822b-467e-9eee-2cbd9895f751" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-button" popovertarget="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-button" id="tooltip-49ffc47b-822b-467e-9eee-2cbd9895f751" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-button" anchor-offset="normal" data-view-component="true" id="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-7e8aa32d-25b7-412f-b54b-f45289e73965-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/acpearson888/advanced_genomics_apearson" id="item-92fc8a47-512f-4c3d-b103-56aefc036a2b" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/issues" id="item-03e74887-8883-4831-80c8-9205f36c93b1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/pulls" id="item-0ae982d6-f390-4a0a-94b3-24eb5b4a7f40" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/actions" id="item-f70aa892-5c4f-4145-800b-4cda9327a60b" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/projects" id="item-6e058975-3713-4378-9182-90c3fcfc5142" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/security" id="item-9807a343-c839-4e72-9bb7-407d59393201" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/pulse" id="item-31ca9517-4b05-4642-944d-52bb4b287cd6" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-7ddba38e-def7-4e56-9678-1f9847b09528" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-09605037-d591-4e8f-9824-db21fe3bea5f" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-09605037-d591-4e8f-9824-db21fe3bea5f" id="tooltip-7ddba38e-def7-4e56-9678-1f9847b09528" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTg2Mn0=--8d35f53fbbc8f17375cd671d23eaa7e8c7a007e12a80174a6a65466f3d1f58d4" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/acpearson888/advanced_genomics_apearson" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjQ4MzYxMjYw" data-targets="command-palette-scope.tokens" data-text="acpearson888" data-type="owner" data-value="acpearson888" style="white-space:nowrap;line-height:20px;">
         acpearson888
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzMzE5NjU5NjM=" data-targets="command-palette-scope.tokens" data-text="advanced_genomics_apearson" data-type="repository" data-value="advanced_genomics_apearson" style="white-space:nowrap;line-height:20px;">
         advanced_genomics...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-9ebfd919-09f4-4b47-bf89-239ad45715e7" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5NjU5NjM=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         Weâve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="acpearson888" data-scope-id="MDQ6VXNlcjQ4MzYxMjYw" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="advanced_genomics_apearson" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5NjU5NjM=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="acpearson888" data-scope-id="MDQ6VXNlcjQ4MzYxMjYw" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="advanced_genomics_apearson" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5NjU5NjM=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="a0vzlsUYOBZ0aWwXa-P3r2oUKbPisKauPaBJn7aW_Vq7J1myUMLusZdUxnD4O24JzxfixmaXj5KjCmwqAkfr7Q">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="â­ï¸ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="JJoDruCOB6jAreiNfHPXWCKTWC1FFw0H6xWmktKufg1Esl9fL6BVzBbmd8mfcRNuI3kGXOnlsz1HP9xbMMbKow">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/48361260?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/acpearson888/advanced_genomics_apearson">
            advanced_genomics_apearson
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/331965963/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":331965963,"originating_url":"https://github.com/acpearson888/advanced_genomics_apearson","user_id":126681664}}' data-hydro-click-hmac="ea6b6e12ae572766a33e798094e92162209441f428f2ebe8c099a3e45e944a22" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-faf5eda8-8f1f-41dd-a216-e1bc8a8bdc08" popover="manual">
              Fork your own copy of acpearson888/advanced_genomics_apearson
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-331965963">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/acpearson888/advanced_genomics_apearson/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-331965963" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-20f29e">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/acpearson888/advanced_genomics_apearson/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="G3Yqu1g9Nmi2vr8rC7KQtoUb-On8pEcP3ECp9nnfTj8_3Sx6Aup14FptxFS4JBLOuSoIrGPXwe7Uao42nWUC-w">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="Tyt9oH7D_ckLwB9tzJ2GqIS-g3UQxTsaI7G8m7tS86prgHthJBS-QecTZBJ_CwTQuI9zMI-2vfsrm5tbX-i_bg">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (0)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":331965963,"originating_url":"https://github.com/acpearson888/advanced_genomics_apearson","user_id":126681664}}' data-hydro-click-hmac="29074188ab3e5a796fa519c37f9a3e89c3d0563c2e5948bc7da0b1b7693e2c76" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="0">
                  0
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-331965963-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/acpearson888/advanced_genomics_apearson/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-331965963-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-194a01">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/acpearson888/advanced_genomics_apearson/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="fnkjLNMDfrBIFFz9TZMSqGhl3JZyBiYB0eQul8L9YuBqc2AUO3Sof_sfpTK24L2piCu_9PLuoZbl_AXy5nKHaQ">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (0)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":331965963,"originating_url":"https://github.com/acpearson888/advanced_genomics_apearson","user_id":126681664}}' data-hydro-click-hmac="c4d521afb69b811537c93bb8d51314bcc8f6a18fcbac0014352a011c34127ec9" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
                 0
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-331965963-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/acpearson888/advanced_genomics_apearson/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-331965963-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-cec9d6">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":331965963,"originating_url":"https://github.com/acpearson888/advanced_genomics_apearson","user_id":126681664}}' data-hydro-click-hmac="a3fac46900aa8b5f6158b3b3725bfd80b6f79f2bbad34d8a698b8c9b0911337d" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="4BFIy7QvCUymxFwCbhQEMXF823lkb_gwCO6eZc-U19D40USxqpB8TUrXawiy9IJGLllIbLE1U-QqFWyRmCVpSQ">
                   <input name="repository_id" type="hidden" value="331965963"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="vTCWxaLNb0vLcv9YM2Zx0nGkUFzopikfrSMTHyJRTaWl8Jq_vHIaSidhyFLvhvelLoHDST38gsuP2OHrdeDzPA">
                   <input name="repository_id" type="hidden" value="331965963"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-c0620f54-4ed2-42fc-9d78-3705e9a9b7a7" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-fac97dfe-abd7-48cd-baba-27af9950936f" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":331965963,"originating_url":"https://github.com/acpearson888/advanced_genomics_apearson","user_id":126681664}}' data-hydro-click-hmac="ea6b6e12ae572766a33e798094e92162209441f428f2ebe8c099a3e45e944a22" data-view-component="true" group_item="true" href="/acpearson888/advanced_genomics_apearson/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-fac97dfe-abd7-48cd-baba-27af9950936f" popover="manual">
              Fork your own copy of acpearson888/advanced_genomics_apearson
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/acpearson888/advanced_genomics_apearson/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="C4aRx5ZzyjIRAJ0JV2NT5ts-JyAqcU6DVE7r7XbpbgMvLZcGzKSJuv3T5nbk9dGe5w_XZbUCyGJcZMwtklMixw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-81827267-b2c1-4025-a9fa-cf642336671c" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":331965963,"originating_url":"https://github.com/acpearson888/advanced_genomics_apearson","user_id":126681664}}' data-hydro-click-hmac="29074188ab3e5a796fa519c37f9a3e89c3d0563c2e5948bc7da0b1b7693e2c76" data-view-component="true" id="icon-button-9dcade9d-5d69-4f5b-9625-65b7d02cb6ad" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-9dcade9d-5d69-4f5b-9625-65b7d02cb6ad" id="tooltip-81827267-b2c1-4025-a9fa-cf642336671c" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/acpearson888/advanced_genomics_apearson/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="L_zWtCs3u3qOYGiKScKDUczIk7mH7c88SpD8-DOrJVc79pWMw0BttT1rkUWysSxQLIbw2wcFSKt-iNedFyTA3g">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-af19fed4-286a-401b-9f14-0191dd78b49f" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":331965963,"originating_url":"https://github.com/acpearson888/advanced_genomics_apearson","user_id":126681664}}' data-hydro-click-hmac="c4d521afb69b811537c93bb8d51314bcc8f6a18fcbac0014352a011c34127ec9" data-view-component="true" id="icon-button-b96e9b33-fa0c-411b-be94-8353babc5f7f" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-b96e9b33-fa0c-411b-be94-8353babc5f7f" id="tooltip-af19fed4-286a-401b-9f14-0191dd78b49f" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/acpearson888/advanced_genomics_apearson/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/acpearson888/advanced_genomics_apearson/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/acpearson888/advanced_genomics_apearson/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/acpearson888/advanced_genomics_apearson/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/acpearson888/advanced_genomics_apearson/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/acpearson888/advanced_genomics_apearson/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/acpearson888/advanced_genomics_apearson/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/acpearson888/advanced_genomics_apearson?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         acpearson888/advanced_genomics_apearson
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/acpearson888/advanced_genomics_apearson/spoofed_commit_check/27bc6239c0de8e00d7c47e8006d5a3acfc6c2771">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":331965963,"defaultBranch":"main","name":"advanced_genomics_apearson","ownerLogin":"acpearson888","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-01-22T22:14:43.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/48361260?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1611325304.0","canEdit":true,"refType":"branch","currentOid":"27bc6239c0de8e00d7c47e8006d5a3acfc6c2771"},"tree":{"items":[{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":1,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/acpearson888/advanced_genomics_apearson/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/acpearson888/advanced_genomics_apearson.git","showCloneWarning":false,"sshUrl":"git@github.com:acpearson888/advanced_genomics_apearson.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone acpearson888/advanced_genomics_apearson","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/acpearson888/advanced_genomics_apearson","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Facpearson888%2Fadvanced_genomics_apearson","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/acpearson888/advanced_genomics_apearson/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"38","overviewFiles":[{"displayName":"README.md","repoName":"advanced_genomics_apearson","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAndrew's 2021 Advanced Bioinformatics Course Notebook\u003c/h1\u003e\u003ca id=\"user-content-andrews-2021-advanced-bioinformatics-course-notebook\" class=\"anchor\" aria-label=\"Permalink: Andrew's 2021 Advanced Bioinformatics Course Notebook\" href=\"#andrews-2021-advanced-bioinformatics-course-notebook\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 02 22-Jan-2021\u003c/h2\u003e\u003ca id=\"user-content-day-02-22-jan-2021\" class=\"anchor\" aria-label=\"Permalink: Day 02 22-Jan-2021\" href=\"#day-02-22-jan-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eexecute a pwd command and start a log of your commands with a header for today's date in your README.md github logfile in your workspace\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 exercises]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 exercises]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003ecp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise* .\n[apear012@turing1 exercises]$ ls\nExercise2.fasta.gz  Exercise2.fastq.tar.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise* .\n[apear012@turing1 exercises]$ ls\nExercise2.fasta.gz  Exercise2.fastq.tar.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eunzip and untar the files\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 exercises]$ gunzip Exercise2.fasta.gz\n[apear012@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq.tar.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 exercises]$ gunzip Exercise2.fasta.gz\n[apear012@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq.tar.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIf you wanted to gunzip without losing the original file, you would use:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"$ gunzip -c Exercise2.fasta.gz \u0026gt; Exercise2.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e$ gunzip -c Exercise2.fasta.gz \u0026gt; Exercise2.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 exercises]$ tar -xzvf Exercise2.fastq.tar.gz \nExercise2.fastq\n[apear012@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 exercises]$ tar -xzvf Exercise2.fastq.tar.gz \nExercise2.fastq\n[apear012@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003ecalculate how many sequences are in each file and add these results to your notebook file\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 exercises]$ grep -c '\u0026gt;' Exercise2.fasta\n138\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 exercises]$ grep -c '\u0026gt;' Exercise2.fasta\n138\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 exercises]$ wc -l Exercise2.fastq\n245216 Exercise2.fastq\n[apear012@turing1 exercises]$ echo 245216/4 | bc \n61304\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 exercises]$ wc -l Exercise2.fastq\n245216 Exercise2.fastq\n[apear012@turing1 exercises]$ echo 245216/4 | bc \n61304\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003ecp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py .\n[apear012@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py .\n[apear012@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"10\" dir=\"auto\"\u003e\n\u003cli\u003estart an interactive compute session and re-navigate to your exercises directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"wahab-01:~\u0026gt; salloc\nsalloc: Pending job allocation 133736\nsalloc: job 133736 queued and waiting for resources\nsalloc: job 133736 has been allocated resources\nsalloc: Granted job allocation 133736\nThis session will be terminated in 7 days. If your application requires\na longer excution time, please use command \u0026quot;salloc -t N-0\u0026quot; where N is the\nnumber of days that you need.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ewahab-01:~\u0026gt; salloc\nsalloc: Pending job allocation 133736\nsalloc: job 133736 queued and waiting for resources\nsalloc: job 133736 has been allocated resources\nsalloc: Granted job allocation 133736\nThis session will be terminated in 7 days. If your application requires\na longer excution time, please use command \"salloc -t N-0\" where N is the\nnumber of days that you need.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"11\" dir=\"auto\"\u003e\n\u003cli\u003erun the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\" e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises\u0026gt; ../../scripts//avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 138\ncontigs \u0026gt;= 1000bp = 138\ncontigs \u0026gt;= 2000bp = 135\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises\u0026gt; ../../scripts//avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 138\ncontigs \u0026gt;= 1000bp = 138\ncontigs \u0026gt;= 2000bp = 135\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 03 27-Jan-2021\u003c/h2\u003e\u003ca id=\"user-content-day-03-27-jan-2021\" class=\"anchor\" aria-label=\"Permalink: Day 03 27-Jan-2021\" href=\"#day-03-27-jan-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBelow is technically the day02 homework.\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eWrite an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory, Andrew-HADB01\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-25-072 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@coreV2-25-072 data]$ nano APCopyLane01.sh\n[apear012@coreV2-25-072 data]$ ls\nAPCopyLane01.sh  exercises\n[apear012@coreV2-25-072 data]$ cat APCopyLane01.sh \n#!/bin/bash -l\n\n#SBATCH -o APCopyLane01.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APCopyLane01\n\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-25-072 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@coreV2-25-072 data]$ nano APCopyLane01.sh\n[apear012@coreV2-25-072 data]$ ls\nAPCopyLane01.sh  exercises\n[apear012@coreV2-25-072 data]$ cat APCopyLane01.sh \n#!/bin/bash -l\n\n#SBATCH -o APCopyLane01.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APCopyLane01\n\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAdd the content of your sbatch script to your logfile\nDid this above.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003esubmit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-25-072 data]$ sbatch APCopyLane01.sh\nSubmitted batch job 9270452\n[apear012@coreV2-25-072 data]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270428      main       sh apear012  R      39:36      1 coreV2-25-072 \n           9270452      main APCopyLa apear012  R       0:29      1 coreV2-25-072 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-25-072 data]$ sbatch APCopyLane01.sh\nSubmitted batch job 9270452\n[apear012@coreV2-25-072 data]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270428      main       sh apear012  R      39:36      1 coreV2-25-072 \n           9270452      main APCopyLa apear012  R       0:29      1 coreV2-25-072 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eMake sure this is all documented on your github notebook\nSee above.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eWrite a sbatch script to gunzip all the fastq.gz files in your data directory\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@turing1 data]$ cat APGunzipLane01Files.sh \n#!/bin/bash -l\n\n#SBATCH -o APGunzipLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APGunzipLane01Files\n\ngunzip *.fastq.gz\n[apear012@turing1 data]$ sbatch APGunzipLane01Files.sh \nSubmitted batch job 9270467\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@turing1 data]$ cat APGunzipLane01Files.sh \n#!/bin/bash -l\n\n#SBATCH -o APGunzipLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APGunzipLane01Files\n\ngunzip *.fastq.gz\n[apear012@turing1 data]$ sbatch APGunzipLane01Files.sh \nSubmitted batch job 9270467\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eThis is the day03 homework:\u003c/strong\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ecp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 apearson]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson\n[apear012@turing1 apearson]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 .\n[apear012@turing1 apearson]$ ls\nAP_exercise1.txt  data  day03  groundrules.txt  scripts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 apearson]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson\n[apear012@turing1 apearson]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 .\n[apear012@turing1 apearson]$ ls\nAP_exercise1.txt  data  day03  groundrules.txt  scripts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003emkdir a fastq directory in your data directory and mv all the .fastq files into this directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 data]$ mkdir fastq\n[apear012@turing1 data]$ mv ./*.fastq fastq\n[apear012@turing1 data]$ cd fastq\n[apear012@turing1 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq\nHADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq\nHADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq\nHADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 data]$ mkdir fastq\n[apear012@turing1 data]$ mv ./*.fastq fastq\n[apear012@turing1 data]$ cd fastq\n[apear012@turing1 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq\nHADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq\nHADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq\nHADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003ecp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder and less the new script and check out the usage statement\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/renamingtable_complete.txt .\n[apear012@turing1 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq\nHADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq\nHADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt\nHADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/renamingtable_complete.txt .\n[apear012@turing1 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq\nHADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq\nHADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt\nHADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py .\n[apear012@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py\n[apear012@turing1 scripts]$ less renamer_advbioinf.py \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py .\n[apear012@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py\n[apear012@turing1 scripts]$ less renamer_advbioinf.py \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003erenamer_advbioinf.py :\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\tif linecount\u0026gt;=2:\n\t\tcols=line.rstrip().split('\\t')\n\t\tprint 'mv %s %s' %(cols[0], cols[1])\n#\t\tos.popen('mv %s %s' %(cols[0], cols[1]))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\tif linecount\u0026gt;=2:\n\t\tcols=line.rstrip().split('\\t')\n\t\tprint 'mv %s %s' %(cols[0], cols[1])\n#\t\tos.popen('mv %s %s' %(cols[0], cols[1]))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003erun the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ python ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\nmv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\nmv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\nmv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\nmv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\nmv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\nmv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\nmv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\nmv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\nmv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\nmv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\nmv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\nmv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\nmv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\nmv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\nmv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\nmv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\nmv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\nmv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\nmv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\nmv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\nmv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\nmv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\nmv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\nmv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\nmv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\nmv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\nmv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\nmv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\nmv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\nmv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\nmv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\nmv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\nmv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\nmv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\nmv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\nmv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\nmv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\nmv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\nmv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\nmv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\nmv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\nmv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\nmv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\nmv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\nmv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\nmv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\nmv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\nmv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\nmv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\nmv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\nmv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\nmv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\nmv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\nmv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\nmv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\nmv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\nmv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\nmv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\nmv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\nmv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\nmv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\nmv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\nmv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\nmv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\nmv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\nmv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\nmv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\nmv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\nmv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\nmv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\nmv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\nmv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\nmv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\nmv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\nmv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\nmv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\nmv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\nmv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\nmv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\nmv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\nmv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\nmv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\nmv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\nmv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\nmv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\nmv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\nmv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\nmv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\nmv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\nmv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\nmv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\nmv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\nmv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\nmv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\nmv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\nmv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ python ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\nmv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\nmv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\nmv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\nmv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\nmv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\nmv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\nmv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\nmv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\nmv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\nmv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\nmv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\nmv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\nmv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\nmv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\nmv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\nmv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\nmv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\nmv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\nmv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\nmv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\nmv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\nmv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\nmv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\nmv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\nmv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\nmv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\nmv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\nmv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\nmv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\nmv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\nmv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\nmv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\nmv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\nmv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\nmv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\nmv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\nmv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\nmv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\nmv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\nmv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\nmv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\nmv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\nmv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\nmv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\nmv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\nmv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\nmv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\nmv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\nmv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\nmv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\nmv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\nmv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\nmv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\nmv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\nmv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\nmv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\nmv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\nmv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\nmv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\nmv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\nmv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\nmv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\nmv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\nmv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\nmv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\nmv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\nmv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\nmv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\nmv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\nmv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\nmv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\nmv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\nmv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\nmv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\nmv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\nmv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\nmv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\nmv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\nmv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\nmv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\nmv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\nmv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\nmv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\nmv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\nmv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\nmv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\nmv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\nmv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\nmv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\nmv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\nmv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\nmv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\nmv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\nmv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\nmv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\nmv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eUncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@turing1 scripts]$ cat renamer_advbioinf.py \n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\tif linecount\u0026gt;=2:\n\t\tcols=line.rstrip().split('\\t')\n\t\tprint 'mv %s %s' %(cols[0], cols[1])\n#\t\tos.popen('mv %s %s' %(cols[0], cols[1]))\n[apear012@turing1 scripts]$ nano renamer_advbioinf.py \n[apear012@turing1 scripts]$ cat renamer_advbioinf.py\n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\tif linecount\u0026gt;=2:\n\t\tcols=line.rstrip().split('\\t')\n#\t\tprint 'mv %s %s' %(cols[0], cols[1])\n\t\tos.popen('mv %s %s' %(cols[0], cols[1]))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@turing1 scripts]$ cat renamer_advbioinf.py \n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\tif linecount\u0026gt;=2:\n\t\tcols=line.rstrip().split('\\t')\n\t\tprint 'mv %s %s' %(cols[0], cols[1])\n#\t\tos.popen('mv %s %s' %(cols[0], cols[1]))\n[apear012@turing1 scripts]$ nano renamer_advbioinf.py \n[apear012@turing1 scripts]$ cat renamer_advbioinf.py\n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\tif linecount\u0026gt;=2:\n\t\tcols=line.rstrip().split('\\t')\n#\t\tprint 'mv %s %s' %(cols[0], cols[1])\n\t\tos.popen('mv %s %s' %(cols[0], cols[1]))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003ewrite a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ nano APRenameLane01Files.sh\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ nano APRenameLane01Files.sh\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAPRenameLane01Files.sh:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o APRenameLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APRenameLane01Files\n\n../../scripts/renamer_advbioinf.py renamingtable_complete.txt ./*.fastq*\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o APRenameLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APRenameLane01Files\n\n../../scripts/renamer_advbioinf.py renamingtable_complete.txt ./*.fastq*\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ sbatch APRenameLane01Files.sh \nSubmitted batch job 9270537\n[apear012@turing1 fastq]$ ls\nAPRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq\nAPRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq\nrenamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq\nRI_B_01_14.fastq            RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ sbatch APRenameLane01Files.sh \nSubmitted batch job 9270537\n[apear012@turing1 fastq]$ ls\nAPRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq\nAPRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq\nrenamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq\nRI_B_01_14.fastq            RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eMake sure this is all documented on your github page\nsee above\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eThe naming convention for the files is as follows:\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq\u003c/li\u003e\n\u003cli\u003eThere are 2 sources: Virginia and Rhode Island\u003c/li\u003e\n\u003cli\u003eThere are 2 symbiotic states: Brown and White\u003c/li\u003e\n\u003c/ul\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eNext, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n\n[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py .\n[apear012@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n\n[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py .\n[apear012@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"11\" dir=\"auto\"\u003e\n\u003cli\u003eLess/head the new script and check out the usage statement\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py \n#!/usr/bin/env python\n# Written by Dan Barshis\n\nimport sys, os\n\n########### Usage #############\n# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\n# This should be run from within the folder with all of your original .fastqstrim\n# Will quality trim multiple SINGLE-END fastq files\n# Things to customize for your particular platform:\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py \n#!/usr/bin/env python\n# Written by Dan Barshis\n\nimport sys, os\n\n########### Usage #############\n# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\n# This should be run from within the folder with all of your original .fastqstrim\n# Will quality trim multiple SINGLE-END fastq files\n# Things to customize for your particular platform:\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"12\" dir=\"auto\"\u003e\n\u003cli\u003ecp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt .\n[apear012@turing1 fastq]$ ls\nadapterlist_advbioinf.txt   RI_B_01_14.fastq   RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq\nAPRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq\nAPRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq\nrenamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt .\n[apear012@turing1 fastq]$ ls\nadapterlist_advbioinf.txt   RI_B_01_14.fastq   RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq\nAPRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq\nAPRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq\nrenamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"13\" dir=\"auto\"\u003e\n\u003cli\u003eMake a sbatch script for the Trimclipfilter... script and run it on your fastq files\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ nano APTrimLane01Files.sh\n[apear012@turing1 fastq]$ cat APTrimLane01Files.sh \n#!/bin/bash -l\n\n#SBATCH -o APTrimLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APTrimLane01Files\n\n../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt ./*.fastq*\n[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh \nSubmitted batch job 9270540\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ nano APTrimLane01Files.sh\n[apear012@turing1 fastq]$ cat APTrimLane01Files.sh \n#!/bin/bash -l\n\n#SBATCH -o APTrimLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APTrimLane01Files\n\n../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt ./*.fastq*\n[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh \nSubmitted batch job 9270540\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEmail received immediately saying my job had failed.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ nano APTrimLane01Files.sh \n[apear012@turing1 fastq]$ cat APTrimLane01Files.sh\n#!/bin/bash -l\n\n#SBATCH -o APTrimLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APTrimLane01Files\n\n../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ nano APTrimLane01Files.sh \n[apear012@turing1 fastq]$ cat APTrimLane01Files.sh\n#!/bin/bash -l\n\n#SBATCH -o APTrimLane01Files.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APTrimLane01Files\n\n../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh\nSubmitted batch job 9270839\n[apear012@turing1 fastq]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270839      main APTrimLa apear012  R       0:00      1 coreV1-22-016 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh\nSubmitted batch job 9270839\n[apear012@turing1 fastq]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270839      main APTrimLa apear012  R       0:00      1 coreV1-22-016 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"14\" dir=\"auto\"\u003e\n\u003cli\u003eThis will take a while (like days)\u003c/li\u003e\n\u003cli\u003eNow might be a good time to update everything on your github\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 04 29-Jan-2021\u003c/h2\u003e\u003ca id=\"user-content-day-04-29-jan-2021\" class=\"anchor\" aria-label=\"Permalink: Day 04 29-Jan-2021\" href=\"#day-04-29-jan-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eAdd your trimclipstats.txt output to the full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e1a. Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-25-072 filteringstats]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/filteringstats\n[apear012@turing1 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\n\nThis script takes a stats output file from fastx_clipper and converts it into a table.\n\nUsage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt\n\nOptions (-c and -v must be listed separately to run together):\n-h\tDisplay this help message\n-c\tUse comma delimiter instead of tabs\n-v\tVerbose mode (print steps to stdout)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-25-072 filteringstats]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/filteringstats\n[apear012@turing1 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\n\nThis script takes a stats output file from fastx_clipper and converts it into a table.\n\nUsage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt\n\nOptions (-c and -v must be listed separately to run together):\n-h\tDisplay this help message\n-c\tUse comma delimiter instead of tabs\n-v\tVerbose mode (print steps to stdout)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e1b. Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-25-072 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt ANDREW_trimclipstatsout.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-25-072 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt ANDREW_trimclipstatsout.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e1c. Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 filteringstats]$ tail -n +2 ANDREW_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 filteringstats]$ tail -n +2 ANDREW_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eNow we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline \u003ca href=\"https://github.com/bethsheets/SNPcalling_tutorial\"\u003ehttps://github.com/bethsheets/SNPcalling_tutorial\u003c/a\u003e)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ewrite a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load bowtie2/2.2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \\\n--rg SM:${i%_clippedtrimmed.fastq} \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \\\n\u0026gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emodule load bowtie2/2.2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \\\n--rg SM:${i%_clippedtrimmed.fastq} \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \\\n\u0026gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ nano APbowtie.sh\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eapear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ nano APbowtie.sh\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAPbowtie.sh:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o APbowtie.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APbowtie\n\nmodule load bowtie2/2.2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \\\n--rg SM:${i%_clippedtrimmed.fastq} \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \\\n\u0026gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o APbowtie.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APbowtie\n\nmodule load bowtie2/2.2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \\\n--rg SM:${i%_clippedtrimmed.fastq} \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \\\n\u0026gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ sbatch APbowtie.sh \nSubmitted batch job 9271109\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9271094      main       sh apear012  R      54:07      1 coreV2-25-072 \n           9271109      main APbowtie apear012  R       0:07      1 coreV2-25-007 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eapear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ sbatch APbowtie.sh \nSubmitted batch job 9271109\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9271094      main       sh apear012  R      54:07      1 coreV2-25-072 \n           9271109      main APbowtie apear012  R       0:07      1 coreV2-25-007 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eSubmit and add everything to your logfile.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eSee above.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 05 03-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-05-03-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 05 03-Feb-2021\" href=\"#day-05-03-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eTeam up with a partner for this one and work only on a combined set of data for your two lanes of sequences\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003ePartner: Areej\nLanes: 01, 06, and 07\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRun the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eModify the below sbatch script (note the differences in the header compared to the previous one)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eoriginal script:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o OUTFILE.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=YOUREMAIL@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=YOURJOBNAME\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single ALLSINGLEENDREADSSEPARATEDBYCOMMAS --CPU 32\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o OUTFILE.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=YOUREMAIL@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=YOURJOBNAME\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single ALLSINGLEENDREADSSEPARATEDBYCOMMAS --CPU 32\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEdited script (renamed APAMAssemble.sh in /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o APAMAssemble.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APAMAssemble\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o APAMAssemble.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APAMAssemble\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCheck \u003ca href=\"https://trinityrnaseq.github.io/\" rel=\"nofollow\"\u003ehttps://trinityrnaseq.github.io/\u003c/a\u003e for usage info\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSubmit your trinity script\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 apam]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam\n[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh \nSubmitted batch job 9272358\n[apear012@coreV2-22-007 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources) \n           9272334      main       sh apear012  R    1:19:02      1 coreV2-25-002 \n           9272357      main       sh apear012  R       6:01      1 coreV2-22-007 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 apam]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam\n[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh \nSubmitted batch job 9272358\n[apear012@coreV2-22-007 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources) \n           9272334      main       sh apear012  R    1:19:02      1 coreV2-25-002 \n           9272357      main       sh apear012  R       6:01      1 coreV2-22-007 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eJob failed after ~15min.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eedited APAMAssemble.sh (removed --normalize_reads):\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o APAMAssemble.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APAMAssemble\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o APAMAssemble.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APAMAssemble\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 apam]$ sbatch APAMAssemble.sh\nSubmitted batch job 9272385\n[apear012@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 \n           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 apam]$ sbatch APAMAssemble.sh\nSubmitted batch job 9272385\n[apear012@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 \n           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 06 05-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-06-05-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 06 05-Feb-2021\" href=\"#day-06-05-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003estart an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 apam]$ salloc\nsalloc: Pending job allocation 9272856\nsalloc: job 9272856 queued and waiting for resources\nsalloc: job 9272856 has been allocated resources\nsalloc: Granted job allocation 9272856\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 apam]$ salloc\nsalloc: Pending job allocation 9272856\nsalloc: job 9272856 queued and waiting for resources\nsalloc: job 9272856 has been allocated resources\nsalloc: Granted job allocation 9272856\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-22-028 djbtestassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly\n[apear012@coreV2-22-028 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta \n\n\n################################\n## Counts of transcripts, etc.\n################################\nTotal trinity 'genes':\t20980\nTotal trinity transcripts:\t21992\nPercent GC: 46.21\n\n########################################\nStats based on ALL transcript contigs:\n########################################\n\n\tContig N10: 1178\n\tContig N20: 694\n\tContig N30: 514\n\tContig N40: 414\n\tContig N50: 347\n\n\tMedian contig length: 273\n\tAverage contig: 356.95\n\tTotal assembled bases: 7850006\n\n\n#####################################################\n## Stats based on ONLY LONGEST ISOFORM per 'GENE':\n#####################################################\n\n\tContig N10: 1027\n\tContig N20: 643\n\tContig N30: 486\n\tContig N40: 397\n\tContig N50: 336\n\n\tMedian contig length: 271\n\tAverage contig: 347.72\n\tTotal assembled bases: 7295061\n\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-22-028 djbtestassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly\n[apear012@coreV2-22-028 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta \n\n\n################################\n## Counts of transcripts, etc.\n################################\nTotal trinity 'genes':\t20980\nTotal trinity transcripts:\t21992\nPercent GC: 46.21\n\n########################################\nStats based on ALL transcript contigs:\n########################################\n\n\tContig N10: 1178\n\tContig N20: 694\n\tContig N30: 514\n\tContig N40: 414\n\tContig N50: 347\n\n\tMedian contig length: 273\n\tAverage contig: 356.95\n\tTotal assembled bases: 7850006\n\n\n#####################################################\n## Stats based on ONLY LONGEST ISOFORM per 'GENE':\n#####################################################\n\n\tContig N10: 1027\n\tContig N20: 643\n\tContig N30: 486\n\tContig N40: 397\n\tContig N50: 336\n\n\tMedian contig length: 271\n\tAverage contig: 347.72\n\tTotal assembled bases: 7295061\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003ecompare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-22-028 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\nThe total number of sequences is 15079\nThe average sequence length is 876\nThe total number of bases is 13210470\nThe minimum sequence length is 500\nThe maximum sequence length is 10795\nThe N50 is 881\nMedian Length = 578\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 15079\ncontigs \u0026gt;= 1000bp = 3660\ncontigs \u0026gt;= 2000bp = 536\n[apear012@coreV2-22-028 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-22-028 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\nThe total number of sequences is 15079\nThe average sequence length is 876\nThe total number of bases is 13210470\nThe minimum sequence length is 500\nThe maximum sequence length is 10795\nThe N50 is 881\nMedian Length = 578\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 15079\ncontigs \u0026gt;= 1000bp = 3660\ncontigs \u0026gt;= 2000bp = 536\n[apear012@coreV2-22-028 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly/Trinity.fasta \nThe total number of sequences is 21992\nThe average sequence length is 356\nThe total number of bases is 7850006\nThe minimum sequence length is 192\nThe maximum sequence length is 10795\nThe N50 is 347\nMedian Length = 355\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 2862\ncontigs \u0026gt;= 1000bp = 607\ncontigs \u0026gt;= 2000bp = 91\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly/Trinity.fasta \nThe total number of sequences is 21992\nThe average sequence length is 356\nThe total number of bases is 7850006\nThe minimum sequence length is 192\nThe maximum sequence length is 10795\nThe N50 is 347\nMedian Length = 355\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 2862\ncontigs \u0026gt;= 1000bp = 607\ncontigs \u0026gt;= 2000bp = 91\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eless or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:\na-the mean percent \"overall alignment rate\"\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ grep 'overall alignment rate' APbowtie.txt | cut -d '%' -f 1\n3.22\n1.74\n1.41\n1.65\n2.01\n1.86\n1.34\n1.49\n2.68\n7.09\n1.84\n1.83\n1.75\n1.70\n52.97\n1.47\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ grep 'overall alignment rate' APbowtie.txt | cut -d '%' -f 1\n3.22\n1.74\n1.41\n1.65\n2.01\n1.86\n1.34\n1.49\n2.68\n7.09\n1.84\n1.83\n1.75\n1.70\n52.97\n1.47\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"b-the mean percent reads \u0026quot;aligned exactly 1 time\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eb-the mean percent reads \"aligned exactly 1 time\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2\n1.63\n1.18\n1.00\n1.18\n0.67\n1.21\n0.88\n0.94\n1.71\n5.19\n1.29\n1.23\n1.09\n1.18\n0.55\n1.03\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2\n1.63\n1.18\n1.00\n1.18\n0.67\n1.21\n0.88\n0.94\n1.71\n5.19\n1.29\n1.23\n1.09\n1.18\n0.55\n1.03\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"c-the mean number of reads \u0026quot;aligned exactly 1 time\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ec-the mean number of reads \"aligned exactly 1 time\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d ' ' -f 5\n117712\n281939\n240500\n256601\n40180\n298308\n182262\n381640\n354945\n1169727\n374006\n332280\n168092\n330623\n397\n276577\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d ' ' -f 5\n117712\n281939\n240500\n256601\n40180\n298308\n182262\n381640\n354945\n1169727\n374006\n332280\n168092\n330623\n397\n276577\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"d-the mean percent reads \u0026quot;aligned \u0026gt;1 times\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ed-the mean percent reads \"aligned \u0026gt;1 times\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ grep 'aligned \u0026gt;1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2\n1.59\n0.56\n0.41\n0.47\n1.34\n0.65\n0.45\n0.55\n0.97\n1.91\n0.55\n0.60\n0.66\n0.51\n52.42\n0.44\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ grep 'aligned \u0026gt;1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2\n1.59\n0.56\n0.41\n0.47\n1.34\n0.65\n0.45\n0.55\n0.97\n1.91\n0.55\n0.60\n0.66\n0.51\n52.42\n0.44\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"hint use grep and paste into excel\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ehint use grep and paste into excel\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCopied each of these outputs into excel and averaged. This was used for the next step to make the alignstatsAP.txt\u003c/p\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eadd your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:\nLaneX_yourinitials\tb-the mean percent \"overall alignment rate\"\tc-the mean percent reads \"aligned exactly 1 time\"\td-the mean number of reads \"aligned exactly 1 time\"\te-the mean percent reads \"aligned \u0026gt;1 times\"\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ nano alignstatsAP.txt\n[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt \nLane1AP\t5.378125\t1.3725\t300361.8125\t4.005\n[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\n[apear012@turing1 QCFastqs]$ tail /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\nLaneX\toverallalignmentrate\tpercsinglyaligned\tnumsinglyaligned\tpercmultiplyaligned\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLane1AP\t5.378125\t1.3725\t300361.8125\t4.005\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ nano alignstatsAP.txt\n[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt \nLane1AP\t5.378125\t1.3725\t300361.8125\t4.005\n[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\n[apear012@turing1 QCFastqs]$ tail /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\nLaneX\toverallalignmentrate\tpercsinglyaligned\tnumsinglyaligned\tpercmultiplyaligned\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLane1AP\t5.378125\t1.3725\t300361.8125\t4.005\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eData cleanup and archiving:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003ea. mv your Trinity.fasta file from your trinity_out_dir to a folder called testassembly in your data directory\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 trinity_out_dir]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/trinity_out_dir\n[apear012@turing1 trinity_out_dir]$ mkdir ../../../../testassembly\n[apear012@turing1 trinity_out_dir]$ mv Trinity.fasta ../../../../testassembly/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 trinity_out_dir]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/trinity_out_dir\n[apear012@turing1 trinity_out_dir]$ mkdir ../../../../testassembly\n[apear012@turing1 trinity_out_dir]$ mv Trinity.fasta ../../../../testassembly/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eb. set up a sbatch script to:\nrm -r YOURtrinity_out_dir\nrm -r YOURoriginalfastqs\nrm -r YOURfilteringstats # as long as you've already appended your filteringstats output to the class table as part of homework_day04.txt\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ nano AP_Cleanup.sh\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq\n[apear012@turing1 fastq]$ nano AP_Cleanup.sh\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAP_Cleanup.sh:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o AP_Cleanup.txt\n#SBATCH -n 1\n#SBATCH --mail-user=acpearson@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Cleanup\n\nrm -r ./QCFastqs/apam/trinity_out_dir\nrm -r filteringstats\nrm -r originalfastqs\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o AP_Cleanup.txt\n#SBATCH -n 1\n#SBATCH --mail-user=acpearson@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Cleanup\n\nrm -r ./QCFastqs/apam/trinity_out_dir\nrm -r filteringstats\nrm -r originalfastqs\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 fastq]$ sbatch AP_Cleanup.sh \nSubmitted batch job 9273047\n[apear012@turing1 fastq]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9273047      main AP_Clean apear012  R       0:01      1 coreV1-22-003 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 fastq]$ sbatch AP_Cleanup.sh \nSubmitted batch job 9273047\n[apear012@turing1 fastq]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9273047      main AP_Clean apear012  R       0:01      1 coreV1-22-003 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003esubmit a blast against sprot from your testassembly folder using the following command\n#!/bin/bash -l\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o OUTFILENAME.txt\n#SBATCH -n 6\u003cbr\u003e\n#SBATCH --mail-user=EMAIL\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \u003cbr\u003e\n-evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n###Exploring our SAM files, check out \u003ca href=\"http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output\" rel=\"nofollow\"\u003ehttp://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output\u003c/a\u003e for bowtie2 specific output and \u003ca href=\"http://bio-bwa.sourceforge.net/bwa.shtml#4\" rel=\"nofollow\"\u003ehttp://bio-bwa.sourceforge.net/bwa.shtml#4\u003c/a\u003e for general SAM output\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly\n[apear012@turing1 testassembly]$ nano AP_Blast.sh\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly\n[apear012@turing1 testassembly]$ nano AP_Blast.sh\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAP_Blast.sh:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o AP_Blast.txt\n#SBATCH -n 1\n#SBATCH --mail-user=acpearson@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Blast\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o AP_Blast.txt\n#SBATCH -n 1\n#SBATCH --mail-user=acpearson@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Blast\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 testassembly]$ ls\nAP_Blast.sh  Trinity.fasta\n[apear012@turing1 testassembly]$ sbatch AP_Blast.sh \nSubmitted batch job 9273048\n[apear012@turing1 testassembly]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9273048      main AP_Blast apear012  R       0:02      1 coreV1-22-016 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 testassembly]$ ls\nAP_Blast.sh  Trinity.fasta\n[apear012@turing1 testassembly]$ sbatch AP_Blast.sh \nSubmitted batch job 9273048\n[apear012@turing1 testassembly]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9273048      main AP_Blast apear012  R       0:02      1 coreV1-22-016 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003ehead one of your .sam files to look at the header\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ head RI_B_01_14_clippedtrimmed.fastq.sam \n@HD\tVN:1.0\tSO:unsorted\n@SQ\tSN:TR78|c0_g1_i1_coral\tLN:1109\n@SQ\tSN:TR78|c0_g2_i1_coral\tLN:1109\n@SQ\tSN:TR79|c0_g1_i1_coral\tLN:610\n@SQ\tSN:TR79|c0_g2_i1_coral\tLN:1549\n@SQ\tSN:TR87|c0_g1_i1_coral\tLN:732\n@SQ\tSN:TR93|c0_g1_i1_coral\tLN:550\n@SQ\tSN:TR101|c0_g1_i1_coral\tLN:673\n@SQ\tSN:TR104|c0_g1_i1_coral\tLN:607\n@SQ\tSN:TR105|c0_g1_i1_coral\tLN:587\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ head RI_B_01_14_clippedtrimmed.fastq.sam \n@HD\tVN:1.0\tSO:unsorted\n@SQ\tSN:TR78|c0_g1_i1_coral\tLN:1109\n@SQ\tSN:TR78|c0_g2_i1_coral\tLN:1109\n@SQ\tSN:TR79|c0_g1_i1_coral\tLN:610\n@SQ\tSN:TR79|c0_g2_i1_coral\tLN:1549\n@SQ\tSN:TR87|c0_g1_i1_coral\tLN:732\n@SQ\tSN:TR93|c0_g1_i1_coral\tLN:550\n@SQ\tSN:TR101|c0_g1_i1_coral\tLN:673\n@SQ\tSN:TR104|c0_g1_i1_coral\tLN:607\n@SQ\tSN:TR105|c0_g1_i1_coral\tLN:587\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003egrep -v '@' your.sam | head to look at the sequence read lines, why does this work to exclude the header lines?\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ grep -v '@' RI_B_01_14_clippedtrimmed.fastq.sam | head\nK00188:59:HMTFHBBXX:2:1101:25814:1701\t16\tTR14006|c0_g2_i1_coral\t317\t255\t51M\t*\t0\tCAGCTGGAAATTCACGTGGTGGCAGTGGAAGTGGCTTTTGACAGAAATCTG\tAJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFFAA\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:19827:1912\t0\tTR6438|c0_g1_i3_coral\t30\t1\t51M\t*\t0\tTGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA\tAAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:19827:1912\t256\tTR6438|c0_g1_i1_coral\t30\t255\t51M\t*\t0\tTGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA\tAAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:19827:1912\t256\tTR6438|c0_g1_i4_coral\t30\t255\t51M\t*\t0\tTGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA\tAAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23622:1912\t16\tTR43756|c0_g1_i1_coral\t7449\t1\t51M\t*\t0\tGGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC\tJJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23622:1912\t272\tTR43756|c0_g1_i1_coral\t2063\t255\t51M\t*\t0\tGGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC\tJJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23622:1912\t272\tTR43756|c0_g2_i1_coral\t2077\t255\t51M\t*\t0\tGGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC\tJJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:20212:1947\t0\tTR11796|c0_g1_i1_sym\t24\t1\t51M\t*\t0\tTCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG\t-AAFAFFFJA--FJJJ\u0026lt;FJ77AFJ7-\u0026lt;JAJFF7-F\u0026lt;A7\u0026lt;F-FJA\u0026lt;FFJJFF\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:20212:1947\t256\tTR11796|c0_g2_i1_sym\t552\t255\t51M\t*\t0\tTCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG\t-AAFAFFFJA--FJJJ\u0026lt;FJ77AFJ7-\u0026lt;JAJFF7-F\u0026lt;A7\u0026lt;F-FJA\u0026lt;FFJJFF\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23723:1947\t0\tTR43756|c0_g2_i1_coral\t1579\t1\t51M\t*\t0\tAACCATAACGAGCATCATCTTGATTAAGCTCATTAGGGTTAGCCTCGGTAC\tAAFFFJJJJJJJJJJJJJJJJJJJJJJJJ7\u0026lt;FFJJJJJJJJFJJJJJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ grep -v '@' RI_B_01_14_clippedtrimmed.fastq.sam | head\nK00188:59:HMTFHBBXX:2:1101:25814:1701\t16\tTR14006|c0_g2_i1_coral\t317\t255\t51M\t*\t0\tCAGCTGGAAATTCACGTGGTGGCAGTGGAAGTGGCTTTTGACAGAAATCTG\tAJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFFAA\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:19827:1912\t0\tTR6438|c0_g1_i3_coral\t30\t1\t51M\t*\t0\tTGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA\tAAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:19827:1912\t256\tTR6438|c0_g1_i1_coral\t30\t255\t51M\t*\t0\tTGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA\tAAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:19827:1912\t256\tTR6438|c0_g1_i4_coral\t30\t255\t51M\t*\t0\tTGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA\tAAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23622:1912\t16\tTR43756|c0_g1_i1_coral\t7449\t1\t51M\t*\t0\tGGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC\tJJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23622:1912\t272\tTR43756|c0_g1_i1_coral\t2063\t255\t51M\t*\t0\tGGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC\tJJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23622:1912\t272\tTR43756|c0_g2_i1_coral\t2077\t255\t51M\t*\t0\tGGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC\tJJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:20212:1947\t0\tTR11796|c0_g1_i1_sym\t24\t1\t51M\t*\t0\tTCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG\t-AAFAFFFJA--FJJJ\u0026lt;FJ77AFJ7-\u0026lt;JAJFF7-F\u0026lt;A7\u0026lt;F-FJA\u0026lt;FFJJFF\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:20212:1947\t256\tTR11796|c0_g2_i1_sym\t552\t255\t51M\t*\t0\tTCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG\t-AAFAFFFJA--FJJJ\u0026lt;FJ77AFJ7-\u0026lt;JAJFF7-F\u0026lt;A7\u0026lt;F-FJA\u0026lt;FFJJFF\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\nK00188:59:HMTFHBBXX:2:1101:23723:1947\t0\tTR43756|c0_g2_i1_coral\t1579\t1\t51M\t*\t0\tAACCATAACGAGCATCATCTTGATTAAGCTCATTAGGGTTAGCCTCGGTAC\tAAFFFJJJJJJJJJJJJJJJJJJJJJJJJ7\u0026lt;FFJJJJJJJJFJJJJJJJJJ\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_01_14\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis works because the -v flag for grep means find all lines that DO NOT include '@'. Since all the header lines include @, they are excluded.\u003c/p\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003ein an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9273049\nsalloc: job 9273049 queued and waiting for resources\nsalloc: job 9273049 has been allocated resources\nsalloc: Granted job allocation 9273049\n[apear012@coreV1-22-016 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_08_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_09_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_08_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9273049\nsalloc: job 9273049 queued and waiting for resources\nsalloc: job 9273049 has been allocated resources\nsalloc: Granted job allocation 9273049\n[apear012@coreV1-22-016 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_01_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_01_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_01_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_W_08_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_B_09_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nVA_W_08_SNP_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"10\" dir=\"auto\"\u003e\n\u003cli\u003ewe need to run the read sorting step required for SNP calling, so if you have time, set up and run the following script on your .sam files to finish before Wednesday:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o OUTFILENAME.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=EMAIL\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o OUTFILENAME.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=EMAIL\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV1-22-016 QCFastqs]$ nano AP_Sort.sh\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV1-22-016 QCFastqs]$ nano AP_Sort.sh\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAP_Sort.sh:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o AP_Sort.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Sort\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o AP_Sort.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Sort\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV1-22-016 QCFastqs]$ sbatch AP_Sort.sh \nSubmitted batch job 9273050\n[apear012@coreV1-22-016 QCFastqs]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9273050      main  AP_Sort apear012  R       0:11      1 coreV1-22-016 \n           9273049      main       sh apear012  R      14:11      1 coreV1-22-016 \n           9273048      main AP_Blast apear012  R      21:11      1 coreV1-22-016 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV1-22-016 QCFastqs]$ sbatch AP_Sort.sh \nSubmitted batch job 9273050\n[apear012@coreV1-22-016 QCFastqs]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9273050      main  AP_Sort apear012  R       0:11      1 coreV1-22-016 \n           9273049      main       sh apear012  R      14:11      1 coreV1-22-016 \n           9273048      main AP_Blast apear012  R      21:11      1 coreV1-22-016 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 07 10-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-07-10-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 07 10-Feb-2021\" href=\"#day-07-10-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun the following command on your sprot output file to process into the contig length/match format that trinity examines\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o OUTFILE.TXT\n#SBATCH -n 1\n#SBATCH --mail-user=YOUREMAIL.odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o OUTFILE.TXT\n#SBATCH -n 1\n#SBATCH --mail-user=YOUREMAIL.odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly\n[apear012@turing1 testassembly]$ nano AP_BlastxToSprot.sh\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly\n[apear012@turing1 testassembly]$ nano AP_BlastxToSprot.sh\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAP_BlastxToSprot.sh:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o AP_BlastxToSprot.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_BlastxToSprot\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o AP_BlastxToSprot.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_BlastxToSprot\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 testassembly]$ sbatch AP_BlastxToSprot.sh \nSubmitted batch job 9276483\n[apear012@turing1 testassembly]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9276483      main AP_Blast apear012 PD       0:00      1 (Priority) \n[apear012@turing1 testassembly]$ ls\nAP_Blast.sh   AP_BlastxToSprot.sh   blastx.outfmt6       blastx.outfmt6.hist.list         Trinity.fasta\nAP_Blast.txt  AP_BlastxToSprot.txt  blastx.outfmt6.hist  blastx.outfmt6.w_pct_hit_length\n[apear012@turing1 testassembly]$ cat *.hist\n#hit_pct_cov_bin\tcount_in_bin\t\u0026gt;bin_below\n100\t198\t198\n90\t87\t285\n80\t88\t373\n70\t110\t483\n60\t146\t629\n50\t210\t839\n40\t339\t1178\n30\t650\t1828\n20\t1178\t3006\n10\t909\t3915\n[apear012@turing1 testassembly]$ grep -c '\u0026gt;' Trinity.fasta \n30194\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 testassembly]$ sbatch AP_BlastxToSprot.sh \nSubmitted batch job 9276483\n[apear012@turing1 testassembly]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9276483      main AP_Blast apear012 PD       0:00      1 (Priority) \n[apear012@turing1 testassembly]$ ls\nAP_Blast.sh   AP_BlastxToSprot.sh   blastx.outfmt6       blastx.outfmt6.hist.list         Trinity.fasta\nAP_Blast.txt  AP_BlastxToSprot.txt  blastx.outfmt6.hist  blastx.outfmt6.w_pct_hit_length\n[apear012@turing1 testassembly]$ cat *.hist\n#hit_pct_cov_bin\tcount_in_bin\t\u0026gt;bin_below\n100\t198\t198\n90\t87\t285\n80\t88\t373\n70\t110\t483\n60\t146\t629\n50\t210\t839\n40\t339\t1178\n30\t650\t1828\n20\t1178\t3006\n10\t909\t3915\n[apear012@turing1 testassembly]$ grep -c '\u0026gt;' Trinity.fasta \n30194\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eRm UNSORTED.bam\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-046 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ ls\nalignstatsAP.txt                               RI_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam\napam                                           *stats.txt.bam\nAPbowtie.sh                                    *stats.txt.sam\nAPbowtie.txt                                   *stats.txt_UNSORTED.bam\nAP_Sort.sh                                     VA_B_01_14_clippedtrimmed.fastq\nAP_Sort.txt                                    VA_B_01_14_clippedtrimmed.fastq.bam\nRI_B_01_14_clippedtrimmed.fastq                VA_B_01_14_clippedtrimmed.fastq.bam.bai\nRI_B_01_14_clippedtrimmed.fastq.bam            VA_B_01_14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.bam.bai        VA_B_01_14_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_01_14_clippedtrimmed.fastq.sam            VA_B_01_18_clippedtrimmed.fastq\nRI_B_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_18_clippedtrimmed.fastq.bam\nRI_B_01_18_clippedtrimmed.fastq                VA_B_01_18_clippedtrimmed.fastq.bam.bai\nRI_B_01_18_clippedtrimmed.fastq.bam            VA_B_01_18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.bam.bai        VA_B_01_18_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_01_18_clippedtrimmed.fastq.sam            VA_B_01_22_clippedtrimmed.fastq\nRI_B_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_22_clippedtrimmed.fastq.bam\nRI_B_01_22_clippedtrimmed.fastq                VA_B_01_22_clippedtrimmed.fastq.bam.bai\nRI_B_01_22_clippedtrimmed.fastq.bam            VA_B_01_22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.bam.bai        VA_B_01_22_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_01_22_clippedtrimmed.fastq.sam            VA_B_09_SNP_clippedtrimmed.fastq\nRI_B_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_B_09_SNP_clippedtrimmed.fastq.bam\nRI_B_08_SNP_clippedtrimmed.fastq               VA_B_09_SNP_clippedtrimmed.fastq.bam.bai\nRI_B_08_SNP_clippedtrimmed.fastq.bam           VA_B_09_SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.bam.bai       VA_B_09_SNP_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_08_SNP_clippedtrimmed.fastq.sam           VA_W_01_14_clippedtrimmed.fastq\nRI_B_08_SNP_clippedtrimmed.fastq_UNSORTED.bam  VA_W_01_14_clippedtrimmed.fastq.bam\nRI_W_01_14_clippedtrimmed.fastq                VA_W_01_14_clippedtrimmed.fastq.bam.bai\nRI_W_01_14_clippedtrimmed.fastq.bam            VA_W_01_14_clippedtrimmed.fastq.sam\nRI_W_01_14_clippedtrimmed.fastq.bam.bai        VA_W_01_14_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_01_14_clippedtrimmed.fastq.sam            VA_W_01_18_clippedtrimmed.fastq\nRI_W_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_18_clippedtrimmed.fastq.bam\nRI_W_01_18_clippedtrimmed.fastq                VA_W_01_18_clippedtrimmed.fastq.bam.bai\nRI_W_01_18_clippedtrimmed.fastq.bam            VA_W_01_18_clippedtrimmed.fastq.sam\nRI_W_01_18_clippedtrimmed.fastq.bam.bai        VA_W_01_18_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_01_18_clippedtrimmed.fastq.sam            VA_W_01_22_clippedtrimmed.fastq\nRI_W_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_22_clippedtrimmed.fastq.bam\nRI_W_01_22_clippedtrimmed.fastq                VA_W_01_22_clippedtrimmed.fastq.bam.bai\nRI_W_01_22_clippedtrimmed.fastq.bam            VA_W_01_22_clippedtrimmed.fastq.sam\nRI_W_01_22_clippedtrimmed.fastq.bam.bai        VA_W_01_22_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_01_22_clippedtrimmed.fastq.sam            VA_W_08_SNP_clippedtrimmed.fastq\nRI_W_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_W_08_SNP_clippedtrimmed.fastq.bam\nRI_W_08_SNP_clippedtrimmed.fastq               VA_W_08_SNP_clippedtrimmed.fastq.bam.bai\nRI_W_08_SNP_clippedtrimmed.fastq.bam           VA_W_08_SNP_clippedtrimmed.fastq.sam\nRI_W_08_SNP_clippedtrimmed.fastq.bam.bai       VA_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_08_SNP_clippedtrimmed.fastq.sam\n[apear012@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9276491\nsalloc: job 9276491 queued and waiting for resources\nsalloc: job 9276491 has been allocated resources\nsalloc: Granted job allocation 9276491\n[apear012@coreV3-23-046 QCFastqs]$ rm *UNSORTED.bam\n[apear012@coreV3-23-046 QCFastqs]$ ls\nalignstatsAP.txt                          RI_W_08_SNP_clippedtrimmed.fastq.bam.bai\napam                                      RI_W_08_SNP_clippedtrimmed.fastq.sam\nAPbowtie.sh                               *stats.txt.bam\nAPbowtie.txt                              *stats.txt.sam\nAP_Sort.sh                                VA_B_01_14_clippedtrimmed.fastq\nAP_Sort.txt                               VA_B_01_14_clippedtrimmed.fastq.bam\nRI_B_01_14_clippedtrimmed.fastq           VA_B_01_14_clippedtrimmed.fastq.bam.bai\nRI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq\nRI_B_01_14_clippedtrimmed.fastq.sam       VA_B_01_18_clippedtrimmed.fastq.bam\nRI_B_01_18_clippedtrimmed.fastq           VA_B_01_18_clippedtrimmed.fastq.bam.bai\nRI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq\nRI_B_01_18_clippedtrimmed.fastq.sam       VA_B_01_22_clippedtrimmed.fastq.bam\nRI_B_01_22_clippedtrimmed.fastq           VA_B_01_22_clippedtrimmed.fastq.bam.bai\nRI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_09_SNP_clippedtrimmed.fastq\nRI_B_01_22_clippedtrimmed.fastq.sam       VA_B_09_SNP_clippedtrimmed.fastq.bam\nRI_B_08_SNP_clippedtrimmed.fastq          VA_B_09_SNP_clippedtrimmed.fastq.bam.bai\nRI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_01_14_clippedtrimmed.fastq\nRI_B_08_SNP_clippedtrimmed.fastq.sam      VA_W_01_14_clippedtrimmed.fastq.bam\nRI_W_01_14_clippedtrimmed.fastq           VA_W_01_14_clippedtrimmed.fastq.bam.bai\nRI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.sam\nRI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq\nRI_W_01_14_clippedtrimmed.fastq.sam       VA_W_01_18_clippedtrimmed.fastq.bam\nRI_W_01_18_clippedtrimmed.fastq           VA_W_01_18_clippedtrimmed.fastq.bam.bai\nRI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.sam\nRI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq\nRI_W_01_18_clippedtrimmed.fastq.sam       VA_W_01_22_clippedtrimmed.fastq.bam\nRI_W_01_22_clippedtrimmed.fastq           VA_W_01_22_clippedtrimmed.fastq.bam.bai\nRI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.sam\nRI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_08_SNP_clippedtrimmed.fastq\nRI_W_01_22_clippedtrimmed.fastq.sam       VA_W_08_SNP_clippedtrimmed.fastq.bam\nRI_W_08_SNP_clippedtrimmed.fastq          VA_W_08_SNP_clippedtrimmed.fastq.bam.bai\nRI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.sam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-046 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ ls\nalignstatsAP.txt                               RI_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam\napam                                           *stats.txt.bam\nAPbowtie.sh                                    *stats.txt.sam\nAPbowtie.txt                                   *stats.txt_UNSORTED.bam\nAP_Sort.sh                                     VA_B_01_14_clippedtrimmed.fastq\nAP_Sort.txt                                    VA_B_01_14_clippedtrimmed.fastq.bam\nRI_B_01_14_clippedtrimmed.fastq                VA_B_01_14_clippedtrimmed.fastq.bam.bai\nRI_B_01_14_clippedtrimmed.fastq.bam            VA_B_01_14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.bam.bai        VA_B_01_14_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_01_14_clippedtrimmed.fastq.sam            VA_B_01_18_clippedtrimmed.fastq\nRI_B_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_18_clippedtrimmed.fastq.bam\nRI_B_01_18_clippedtrimmed.fastq                VA_B_01_18_clippedtrimmed.fastq.bam.bai\nRI_B_01_18_clippedtrimmed.fastq.bam            VA_B_01_18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.bam.bai        VA_B_01_18_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_01_18_clippedtrimmed.fastq.sam            VA_B_01_22_clippedtrimmed.fastq\nRI_B_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_22_clippedtrimmed.fastq.bam\nRI_B_01_22_clippedtrimmed.fastq                VA_B_01_22_clippedtrimmed.fastq.bam.bai\nRI_B_01_22_clippedtrimmed.fastq.bam            VA_B_01_22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.bam.bai        VA_B_01_22_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_01_22_clippedtrimmed.fastq.sam            VA_B_09_SNP_clippedtrimmed.fastq\nRI_B_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_B_09_SNP_clippedtrimmed.fastq.bam\nRI_B_08_SNP_clippedtrimmed.fastq               VA_B_09_SNP_clippedtrimmed.fastq.bam.bai\nRI_B_08_SNP_clippedtrimmed.fastq.bam           VA_B_09_SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.bam.bai       VA_B_09_SNP_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_08_SNP_clippedtrimmed.fastq.sam           VA_W_01_14_clippedtrimmed.fastq\nRI_B_08_SNP_clippedtrimmed.fastq_UNSORTED.bam  VA_W_01_14_clippedtrimmed.fastq.bam\nRI_W_01_14_clippedtrimmed.fastq                VA_W_01_14_clippedtrimmed.fastq.bam.bai\nRI_W_01_14_clippedtrimmed.fastq.bam            VA_W_01_14_clippedtrimmed.fastq.sam\nRI_W_01_14_clippedtrimmed.fastq.bam.bai        VA_W_01_14_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_01_14_clippedtrimmed.fastq.sam            VA_W_01_18_clippedtrimmed.fastq\nRI_W_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_18_clippedtrimmed.fastq.bam\nRI_W_01_18_clippedtrimmed.fastq                VA_W_01_18_clippedtrimmed.fastq.bam.bai\nRI_W_01_18_clippedtrimmed.fastq.bam            VA_W_01_18_clippedtrimmed.fastq.sam\nRI_W_01_18_clippedtrimmed.fastq.bam.bai        VA_W_01_18_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_01_18_clippedtrimmed.fastq.sam            VA_W_01_22_clippedtrimmed.fastq\nRI_W_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_22_clippedtrimmed.fastq.bam\nRI_W_01_22_clippedtrimmed.fastq                VA_W_01_22_clippedtrimmed.fastq.bam.bai\nRI_W_01_22_clippedtrimmed.fastq.bam            VA_W_01_22_clippedtrimmed.fastq.sam\nRI_W_01_22_clippedtrimmed.fastq.bam.bai        VA_W_01_22_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_01_22_clippedtrimmed.fastq.sam            VA_W_08_SNP_clippedtrimmed.fastq\nRI_W_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_W_08_SNP_clippedtrimmed.fastq.bam\nRI_W_08_SNP_clippedtrimmed.fastq               VA_W_08_SNP_clippedtrimmed.fastq.bam.bai\nRI_W_08_SNP_clippedtrimmed.fastq.bam           VA_W_08_SNP_clippedtrimmed.fastq.sam\nRI_W_08_SNP_clippedtrimmed.fastq.bam.bai       VA_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_08_SNP_clippedtrimmed.fastq.sam\n[apear012@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9276491\nsalloc: job 9276491 queued and waiting for resources\nsalloc: job 9276491 has been allocated resources\nsalloc: Granted job allocation 9276491\n[apear012@coreV3-23-046 QCFastqs]$ rm *UNSORTED.bam\n[apear012@coreV3-23-046 QCFastqs]$ ls\nalignstatsAP.txt                          RI_W_08_SNP_clippedtrimmed.fastq.bam.bai\napam                                      RI_W_08_SNP_clippedtrimmed.fastq.sam\nAPbowtie.sh                               *stats.txt.bam\nAPbowtie.txt                              *stats.txt.sam\nAP_Sort.sh                                VA_B_01_14_clippedtrimmed.fastq\nAP_Sort.txt                               VA_B_01_14_clippedtrimmed.fastq.bam\nRI_B_01_14_clippedtrimmed.fastq           VA_B_01_14_clippedtrimmed.fastq.bam.bai\nRI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq\nRI_B_01_14_clippedtrimmed.fastq.sam       VA_B_01_18_clippedtrimmed.fastq.bam\nRI_B_01_18_clippedtrimmed.fastq           VA_B_01_18_clippedtrimmed.fastq.bam.bai\nRI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq\nRI_B_01_18_clippedtrimmed.fastq.sam       VA_B_01_22_clippedtrimmed.fastq.bam\nRI_B_01_22_clippedtrimmed.fastq           VA_B_01_22_clippedtrimmed.fastq.bam.bai\nRI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_09_SNP_clippedtrimmed.fastq\nRI_B_01_22_clippedtrimmed.fastq.sam       VA_B_09_SNP_clippedtrimmed.fastq.bam\nRI_B_08_SNP_clippedtrimmed.fastq          VA_B_09_SNP_clippedtrimmed.fastq.bam.bai\nRI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_01_14_clippedtrimmed.fastq\nRI_B_08_SNP_clippedtrimmed.fastq.sam      VA_W_01_14_clippedtrimmed.fastq.bam\nRI_W_01_14_clippedtrimmed.fastq           VA_W_01_14_clippedtrimmed.fastq.bam.bai\nRI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.sam\nRI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq\nRI_W_01_14_clippedtrimmed.fastq.sam       VA_W_01_18_clippedtrimmed.fastq.bam\nRI_W_01_18_clippedtrimmed.fastq           VA_W_01_18_clippedtrimmed.fastq.bam.bai\nRI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.sam\nRI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq\nRI_W_01_18_clippedtrimmed.fastq.sam       VA_W_01_22_clippedtrimmed.fastq.bam\nRI_W_01_22_clippedtrimmed.fastq           VA_W_01_22_clippedtrimmed.fastq.bam.bai\nRI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.sam\nRI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_08_SNP_clippedtrimmed.fastq\nRI_W_01_22_clippedtrimmed.fastq.sam       VA_W_08_SNP_clippedtrimmed.fastq.bam\nRI_W_08_SNP_clippedtrimmed.fastq          VA_W_08_SNP_clippedtrimmed.fastq.bam.bai\nRI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.sam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-046 QCFastqs]$ enable lmod\nenable: Command not found.\n[apear012@coreV3-23-046 QCFastqs]$ enable_lmod\n[apear012@coreV3-23-046 QCFastqs]$ module load samtools\n[apear012@coreV3-23-046 QCFastqs]$ samtools tview RI_B_01_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\n1         11        21        31        41        51        61        71        \nTCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT\n                                                        ........................\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-046 QCFastqs]$ enable lmod\nenable: Command not found.\n[apear012@coreV3-23-046 QCFastqs]$ enable_lmod\n[apear012@coreV3-23-046 QCFastqs]$ module load samtools\n[apear012@coreV3-23-046 QCFastqs]$ samtools tview RI_B_01_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\n1         11        21        31        41        51        61        71        \nTCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT\n                                                        ........................\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\n                                                        ,,,,,,,,,,,,,,,,,,,,,,,,\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eRun the following to start genotyping your SNPs for filtering next class\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o OUTFILENAME.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=EMAIL\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam \u0026gt; YOURNAMEmergedfastqs.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o OUTFILENAME.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=EMAIL\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam \u0026gt; YOURNAMEmergedfastqs.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ nano AP_Freebayes.sh\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ nano AP_Freebayes.sh\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAP_Freebayes.sh:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o AP_Freebayes.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Freebayes\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam \u0026gt; APmergedfastqs.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e#!/bin/bash -l\n\n#SBATCH -o AP_Freebayes.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_Freebayes\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam \u0026gt; APmergedfastqs.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ sbatch AP_Freebayes.sh \nSubmitted batch job 9276576\n[apear012@turing1 QCFastqs]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9276576      main AP_Freeb apear012  R       0:02      1 coreV3-23-026 \n           9276491      main       sh apear012  R    1:11:32      1 coreV3-23-046 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@turing1 QCFastqs]$ sbatch AP_Freebayes.sh \nSubmitted batch job 9276576\n[apear012@turing1 QCFastqs]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9276576      main AP_Freeb apear012  R       0:02      1 coreV3-23-026 \n           9276491      main       sh apear012  R    1:11:32      1 coreV3-23-046 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 08 12-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-08-12-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 08 12-Feb-2021\" href=\"#day-08-12-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eClean up your data directory by:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e-Making a SAMS folder and mv all your .sam files into that directory\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@turing1 data]$ mkdir SAMS\n[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.sam SAMS\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@turing1 data]$ mkdir SAMS\n[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.sam SAMS\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 data]$ mkdir BAMS\n[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.bam BAMS\n[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.bai BAMS\n[apear012@coreV3-23-040 data]$ ls BAMS\nRI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.bam\nRI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_14_clippedtrimmed.fastq.bam.bai\nRI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.bam\nRI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq.bam.bai\nRI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.bam\nRI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq.bam.bai\nRI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.bam\nRI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_B_09_SNP_clippedtrimmed.fastq.bam.bai\nRI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.bam\nRI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_14_clippedtrimmed.fastq.bam.bai\nRI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.bam\nRI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq.bam.bai\nRI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.bam\nRI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq.bam.bai\nRI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.bam\nRI_W_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_08_SNP_clippedtrimmed.fastq.bam.bai\n*stats.txt.bam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 data]$ mkdir BAMS\n[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.bam BAMS\n[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.bai BAMS\n[apear012@coreV3-23-040 data]$ ls BAMS\nRI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.bam\nRI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_14_clippedtrimmed.fastq.bam.bai\nRI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.bam\nRI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq.bam.bai\nRI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.bam\nRI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq.bam.bai\nRI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.bam\nRI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_B_09_SNP_clippedtrimmed.fastq.bam.bai\nRI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.bam\nRI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_14_clippedtrimmed.fastq.bam.bai\nRI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.bam\nRI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq.bam.bai\nRI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.bam\nRI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq.bam.bai\nRI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.bam\nRI_W_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_08_SNP_clippedtrimmed.fastq.bam.bai\n*stats.txt.bam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"-rm your _unfiltered.vcf files if you have any\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e-rm your _unfiltered.vcf files if you have any\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eI didn't have any.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"-rm your .fastq files\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e-rm your .fastq files\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-040 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@coreV3-23-040 QCFastqs]$ rm *.fastq\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-040 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs\n[apear012@coreV3-23-040 QCFastqs]$ rm *.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"-Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e-Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-040 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@coreV3-23-040 data]$ mkdir VCF\n[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.vcf VCF\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-040 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data\n[apear012@coreV3-23-040 data]$ mkdir VCF\n[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.vcf VCF\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eStart an interactive session via salloc\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eDid this during previous steps.\u003c/p\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003ecp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-040 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF\n[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf .\n[apear012@coreV3-23-040 VCF]$ ls\nAPmergedfastqs.vcf  mergedfastq_HEAAstrangiaAssembly.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-040 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF\n[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf .\n[apear012@coreV3-23-040 VCF]$ ls\nAPmergedfastqs.vcf  mergedfastq_HEAAstrangiaAssembly.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eDetermine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-040 VCF]$ vcftools --vcf APmergedfastqs.vcf \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf APmergedfastqs.vcf\n\nAfter filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 148757 out of a possible 148757 Sites\nRun Time = 1.00 seconds\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-040 VCF]$ vcftools --vcf APmergedfastqs.vcf \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf APmergedfastqs.vcf\n\nAfter filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 148757 out of a possible 148757 Sites\nRun Time = 1.00 seconds\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003ecp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt .\n[apear012@coreV3-23-040 VCF]$ ls\nAPmergedfastqs.vcf  GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf  out.log\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt .\n[apear012@coreV3-23-040 VCF]$ ls\nAPmergedfastqs.vcf  GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf  out.log\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eRun our host vcf extractor on your merged vcf file using the following syntax:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-026 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-026 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003eCompare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 VCF]$ ls\nAPmergedfastqs.vcf                    mergedfastq_HEAAstrangiaAssembly_subset.vcf  out.log\nGoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf\n[apear012@turing1 VCF]$ vcftools --vcf mer\nmergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*\n[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 432676 out of a possible 432676 Sites\nRun Time = 6.00 seconds\n[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 11.00 seconds\n[apear012@turing1 VCF]$ vcftools --vcf APmergedfastqs.vcf \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf APmergedfastqs.vcf\n\nAfter filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 148757 out of a possible 148757 Sites\nRun Time = 2.00 seconds\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 VCF]$ ls\nAPmergedfastqs.vcf                    mergedfastq_HEAAstrangiaAssembly_subset.vcf  out.log\nGoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf\n[apear012@turing1 VCF]$ vcftools --vcf mer\nmergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*\n[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 432676 out of a possible 432676 Sites\nRun Time = 6.00 seconds\n[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 11.00 seconds\n[apear012@turing1 VCF]$ vcftools --vcf APmergedfastqs.vcf \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf APmergedfastqs.vcf\n\nAfter filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 148757 out of a possible 148757 Sites\nRun Time = 2.00 seconds\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003eWork through the \u003ca href=\"http://www.ddocent.com/filtering/\" rel=\"nofollow\"\u003eVCF filtering tutorial\u003c/a\u003e until the following step:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eNow that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003evcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 VCF]$ vcftools --vcf mer --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\nmergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*\n[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\t--recode-INFO-all\n\t--mac 3\n\t--minQ 30\n\t--max-missing 0.5\n\t--out raw.g5mac3\n\t--recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 432676 Sites\nRun Time = 33.00 seconds\n[apear012@turing1 VCF]$ ls\nAPmergedfastqs.vcf                           out.log\nGoodCoralGenelistForVCFSubsetter.txt         raw.g5mac3.log\nmergedfastq_HEAAstrangiaAssembly_subset.vcf  raw.g5mac3.recode.vcf\nmergedfastq_HEAAstrangiaAssembly.vcf\n[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3 \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf raw.g5mac3.recode.vcf\n\t--recode-INFO-all\n\t--minDP 3\n\t--out raw.g5mac3dp3\n\t--recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 30.00 seconds\n[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf raw.g5mac3dp3.recode.vcf\n\t--missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 3.00 seconds\n[apear012@turing1 VCF]$ bash\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 \u0026gt; totalmissing\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ gnuplot \u0026lt;\u0026lt; \\EOF \n\u0026gt; set terminal dumb size 120, 30\n\u0026gt; set autoscale \n\u0026gt; unset label\n\u0026gt; set title \u0026quot;Histogram of % missing data per individual\u0026quot;\n\u0026gt; set ylabel \u0026quot;Number of Occurrences\u0026quot;\n\u0026gt; set xlabel \u0026quot;% of missing data\u0026quot;\n\u0026gt; #set yr [0:100000]\n\u0026gt; binwidth=0.01\n\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\n\u0026gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes\n\u0026gt; pause -1\n\u0026gt; EOF\n\n                                         Histogram of % missing data per individual\n  Number of Occurrences\n      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++\n        +           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n    2.5 ++                                                               * *                                        ++\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n      2 ++                                                       **      * * ***  ******    ** ****   ****          ++\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +\n      1 *********************************************************************************************************---++\n       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\n                                                      % of missing data\n\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ exit\nexit\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 VCF]$ vcftools --vcf mer --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\nmergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*\n[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\t--recode-INFO-all\n\t--mac 3\n\t--minQ 30\n\t--max-missing 0.5\n\t--out raw.g5mac3\n\t--recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 432676 Sites\nRun Time = 33.00 seconds\n[apear012@turing1 VCF]$ ls\nAPmergedfastqs.vcf                           out.log\nGoodCoralGenelistForVCFSubsetter.txt         raw.g5mac3.log\nmergedfastq_HEAAstrangiaAssembly_subset.vcf  raw.g5mac3.recode.vcf\nmergedfastq_HEAAstrangiaAssembly.vcf\n[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3 \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf raw.g5mac3.recode.vcf\n\t--recode-INFO-all\n\t--minDP 3\n\t--out raw.g5mac3dp3\n\t--recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 30.00 seconds\n[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf raw.g5mac3dp3.recode.vcf\n\t--missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 3.00 seconds\n[apear012@turing1 VCF]$ bash\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 \u0026gt; totalmissing\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ gnuplot \u0026lt;\u0026lt; \\EOF \n\u0026gt; set terminal dumb size 120, 30\n\u0026gt; set autoscale \n\u0026gt; unset label\n\u0026gt; set title \"Histogram of % missing data per individual\"\n\u0026gt; set ylabel \"Number of Occurrences\"\n\u0026gt; set xlabel \"% of missing data\"\n\u0026gt; #set yr [0:100000]\n\u0026gt; binwidth=0.01\n\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\n\u0026gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes\n\u0026gt; pause -1\n\u0026gt; EOF\n\n                                         Histogram of % missing data per individual\n  Number of Occurrences\n      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++\n        +           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n    2.5 ++                                                               * *                                        ++\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n      2 ++                                                       **      * * ***  ******    ** ****   ****          ++\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +\n      1 *********************************************************************************************************---++\n       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\n                                                      % of missing data\n\napear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ exit\nexit\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 09 17-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-09-17-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 09 17-Feb-2021\" href=\"#day-09-17-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters\n##Half the class run the following (I was instructed to do this one)\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n##the other half run the filters we used from the paper\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-015 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF\n[apear012@coreV3-23-015 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\t--recode-INFO-all\n\t--maf 0.015\n\t--max-alleles 2\n\t--minDP 3\n\t--minGQ 20\n\t--hwe 0.01\n\t--minQ 30\n\t--max-missing 0.5\n\t--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\t--recode\n\t--remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 18718 out of a possible 432676 Sites\nRun Time = 10.00 seconds\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-015 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF\n[apear012@coreV3-23-015 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\t--recode-INFO-all\n\t--maf 0.015\n\t--max-alleles 2\n\t--minDP 3\n\t--minGQ 20\n\t--hwe 0.01\n\t--minQ 30\n\t--max-missing 0.5\n\t--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\t--recode\n\t--remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 18718 out of a possible 432676 Sites\nRun Time = 10.00 seconds\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eMake a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV3-23-015 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf \u0026gt; samplenames.txt\n[apear012@coreV3-23-015 VCF]$ cat samplenames.txt \n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tRI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_merged\tVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_mergedRI_W_05_merged\tRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV3-23-015 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf \u0026gt; samplenames.txt\n[apear012@coreV3-23-015 VCF]$ cat samplenames.txt \n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tRI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_merged\tVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_mergedRI_W_05_merged\tRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCopied the contents of samplenames.txt into BBedit and used regular expressions in find and replace as such:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"FIND: (\\w\\w_\\w)(_.*)\nREPLACE: \\1\\2\\t\\1\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eFIND: (\\w\\w_\\w)(_.*)\nREPLACE: \\1\\2\\t\\1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-22-030 VCF]$ nano popfile.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-22-030 VCF]$ nano popfile.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003epopfile.txt:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"RI_W_06_merged\tRI_W\nRI_W_07_merged\tRI_W\nVA_B_03_merged\tVA_B\nRI_W_02_merged\tRI_W\nRI_W_04_merged\tRI_W\nVA_W_09_SNP_clipped\tVA_W\nRI_B_08_SNP_clipped\tRI_B\nVA_W_08_SNP_clipped\tVA_W\nVA_B_08_SNP_clipped\tVA_B\nVA_W_02_merged\tVA_W\nVA_B_07_merged\tVA_B\nRI_B_05_merged\tRI_B\nVA_W_06_merged\tVA_W\nVA_W_04_merged\tVA_W\nVA_W_01_merged\tVA_W\nVA_B_10_SNP_clipped\tVA_B\nVA_B_06_merged\tVA_B\nVA_W_05_merged\tVA_W\nRI_B_09_SNP_clipped\tRI_B\nVA_W_10_SNP_clipped\tVA_W\nRI_W_08_SNP_clipped\tRI_W\nRI_B_06_merged\tRI_B\nRI_W_10_SNP_clipped\tRI_W\nRI_B_04_merged\tRI_B\nVA_W_03_merged\tVA_W\nRI_B_07_merged\tRI_B\nRI_W_05_merged\tRI_W\nRI_W_09_SNP_clipped\tRI_W\nVA_B_01_merged\tVA_B\nVA_B_09_SNP_clipped\tVA_B\nRI_B_10_SNP_clipped\tRI_B\nRI_W_01_merged\tRI_W\nRI_B_01_merged\tRI_B\nVA_B_04_merged\tVA_B\nRI_B_02_merged\tRI_B\nRI_W_03_merged\tRI_W\nVA_B_02_merged\tVA_B\nVA_W_07_merged\tVA_W\nVA_B_05_merged\tVA_B\nRI_B_03_merged\tRI_B\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eRI_W_06_merged\tRI_W\nRI_W_07_merged\tRI_W\nVA_B_03_merged\tVA_B\nRI_W_02_merged\tRI_W\nRI_W_04_merged\tRI_W\nVA_W_09_SNP_clipped\tVA_W\nRI_B_08_SNP_clipped\tRI_B\nVA_W_08_SNP_clipped\tVA_W\nVA_B_08_SNP_clipped\tVA_B\nVA_W_02_merged\tVA_W\nVA_B_07_merged\tVA_B\nRI_B_05_merged\tRI_B\nVA_W_06_merged\tVA_W\nVA_W_04_merged\tVA_W\nVA_W_01_merged\tVA_W\nVA_B_10_SNP_clipped\tVA_B\nVA_B_06_merged\tVA_B\nVA_W_05_merged\tVA_W\nRI_B_09_SNP_clipped\tRI_B\nVA_W_10_SNP_clipped\tVA_W\nRI_W_08_SNP_clipped\tRI_W\nRI_B_06_merged\tRI_B\nRI_W_10_SNP_clipped\tRI_W\nRI_B_04_merged\tRI_B\nVA_W_03_merged\tVA_W\nRI_B_07_merged\tRI_B\nRI_W_05_merged\tRI_W\nRI_W_09_SNP_clipped\tRI_W\nVA_B_01_merged\tVA_B\nVA_B_09_SNP_clipped\tVA_B\nRI_B_10_SNP_clipped\tRI_B\nRI_W_01_merged\tRI_W\nRI_B_01_merged\tRI_B\nVA_B_04_merged\tVA_B\nRI_B_02_merged\tRI_B\nRI_W_03_merged\tRI_W\nVA_B_02_merged\tVA_B\nVA_W_07_merged\tVA_W\nVA_B_05_merged\tVA_B\nRI_B_03_merged\tRI_B\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eConvert your filtered .vcf file to genepop format using the following command:\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-22-030 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt \nIndivs with genotypes in vcf file: RI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_merged\tVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_merged\tRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n44 18718 18718 18718 18718 40\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-22-030 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt \nIndivs with genotypes in vcf file: RI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_merged\tVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_merged\tRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n44 18718 18718 18718 18718 40\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eSCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"(base) Andrews-MacBook-Air:day09 AndrewsComputer$ pwd\n/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/day09\n(base) Andrews-MacBook-Air:day09 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen .\napear012@turing.hpc.odu.edu's password: \n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.ge 100% 4062KB 180.0KB/s   00:22    \n(base) Andrews-MacBook-Air:day09 AndrewsComputer$ ls\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen\nadegenet_PCAs.R\nadegenet_PCAs_HEAOutliers.R\nhomework_day09.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e(base) Andrews-MacBook-Air:day09 AndrewsComputer$ pwd\n/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/day09\n(base) Andrews-MacBook-Air:day09 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen .\napear012@turing.hpc.odu.edu's password: \n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.ge 100% 4062KB 180.0KB/s   00:22    \n(base) Andrews-MacBook-Air:day09 AndrewsComputer$ ls\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen\nadegenet_PCAs.R\nadegenet_PCAs_HEAOutliers.R\nhomework_day09.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eSwitch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eI did this. I had to import a number of dependencies for the R package 'adegenet' manually to get this to work. Specifically, the R package 'sf', a dependancy of 'adegent', gave me trouble and I had to go to download it in Chrome and manually add the package from the file browser.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 10 19-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-10-19-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 10 19-Feb-2021\" href=\"#day-10-19-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eWork through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eI did this.\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003ecd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 SAMS]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS\n[apear012@coreV2-25-017 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 SAMS]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS\n[apear012@coreV2-25-017 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eOnce your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt YOURNAMEFullCounts_summed.txt NoMatch *_counts.txt\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e(I also had to rename the match_counts.txt file to matchcounts.txt so this command would not include it.)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@turing1 SAMS]$ mv match_counts.txt matchcounts.txt\n[apear012@turing1 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt APFullCounts_summed.txt NoMatch *_counts.txt\nHits not matchedRI_B_01_14_clippedtrimmed.fastq_counts.txt=1698\tRI_B_01_18_clippedtrimmed.fastq_counts.txt=169RI_B_01_22_clippedtrimmed.fastq_counts.txt=1698\tRI_B_08_SNP_clippedtrimmed.fastq_counts.txt=1698\tRI_W_01_14_clippedtrimmed.fastq_counts.txt=1698\tRI_W_01_18_clippedtrimmed.fastq_counts.txt=1698\tRI_W_01_22_clippedtrimmed.fastq_counts.txt=1698\tRI_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698\tVA_B_01_14_clippedtrimmed.fastq_counts.txt=1698\tVA_B_01_18_clippedtrimmed.fastq_counts.txt=1698\tVA_B_01_22_clippedtrimmed.fastq_counts.txt=1698\tVA_B_09_SNP_clippedtrimmed.fastq_counts.txt=1698\tVA_W_01_14_clippedtrimmed.fastq_counts.txt=1698\tVA_W_01_18_clippedtrimmed.fastq_counts.txt=1698\tVA_W_01_22_clippedtrimmed.fastq_counts.txt=1698\tVA_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698\n[apear012@turing1 SAMS]$ ls\nAPFullCounts_summed.txt                      stats.txt\nmatchcounts.txt                              VA_B_01_14_clippedtrimmed.fastq_counts.txt\nRI_B_01_14_clippedtrimmed.fastq_counts.txt   VA_B_01_14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.sam          VA_B_01_18_clippedtrimmed.fastq_counts.txt\nRI_B_01_18_clippedtrimmed.fastq_counts.txt   VA_B_01_18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.sam          VA_B_01_22_clippedtrimmed.fastq_counts.txt\nRI_B_01_22_clippedtrimmed.fastq_counts.txt   VA_B_01_22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.sam          VA_B_09_SNP_clippedtrimmed.fastq_counts.txt\nRI_B_08_SNP_clippedtrimmed.fastq_counts.txt  VA_B_09_SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.sam         VA_W_01_14_clippedtrimmed.fastq_counts.txt\nRI_W_01_14_clippedtrimmed.fastq_counts.txt   VA_W_01_14_clippedtrimmed.fastq.sam\nRI_W_01_14_clippedtrimmed.fastq.sam          VA_W_01_18_clippedtrimmed.fastq_counts.txt\nRI_W_01_18_clippedtrimmed.fastq_counts.txt   VA_W_01_18_clippedtrimmed.fastq.sam\nRI_W_01_18_clippedtrimmed.fastq.sam          VA_W_01_22_clippedtrimmed.fastq_counts.txt\nRI_W_01_22_clippedtrimmed.fastq_counts.txt   VA_W_01_22_clippedtrimmed.fastq.sam\nRI_W_01_22_clippedtrimmed.fastq.sam          VA_W_08_SNP_clippedtrimmed.fastq_counts.txt\nRI_W_08_SNP_clippedtrimmed.fastq_counts.txt  VA_W_08_SNP_clippedtrimmed.fastq.sam\nRI_W_08_SNP_clippedtrimmed.fastq.sam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@turing1 SAMS]$ mv match_counts.txt matchcounts.txt\n[apear012@turing1 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt APFullCounts_summed.txt NoMatch *_counts.txt\nHits not matchedRI_B_01_14_clippedtrimmed.fastq_counts.txt=1698\tRI_B_01_18_clippedtrimmed.fastq_counts.txt=169RI_B_01_22_clippedtrimmed.fastq_counts.txt=1698\tRI_B_08_SNP_clippedtrimmed.fastq_counts.txt=1698\tRI_W_01_14_clippedtrimmed.fastq_counts.txt=1698\tRI_W_01_18_clippedtrimmed.fastq_counts.txt=1698\tRI_W_01_22_clippedtrimmed.fastq_counts.txt=1698\tRI_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698\tVA_B_01_14_clippedtrimmed.fastq_counts.txt=1698\tVA_B_01_18_clippedtrimmed.fastq_counts.txt=1698\tVA_B_01_22_clippedtrimmed.fastq_counts.txt=1698\tVA_B_09_SNP_clippedtrimmed.fastq_counts.txt=1698\tVA_W_01_14_clippedtrimmed.fastq_counts.txt=1698\tVA_W_01_18_clippedtrimmed.fastq_counts.txt=1698\tVA_W_01_22_clippedtrimmed.fastq_counts.txt=1698\tVA_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698\n[apear012@turing1 SAMS]$ ls\nAPFullCounts_summed.txt                      stats.txt\nmatchcounts.txt                              VA_B_01_14_clippedtrimmed.fastq_counts.txt\nRI_B_01_14_clippedtrimmed.fastq_counts.txt   VA_B_01_14_clippedtrimmed.fastq.sam\nRI_B_01_14_clippedtrimmed.fastq.sam          VA_B_01_18_clippedtrimmed.fastq_counts.txt\nRI_B_01_18_clippedtrimmed.fastq_counts.txt   VA_B_01_18_clippedtrimmed.fastq.sam\nRI_B_01_18_clippedtrimmed.fastq.sam          VA_B_01_22_clippedtrimmed.fastq_counts.txt\nRI_B_01_22_clippedtrimmed.fastq_counts.txt   VA_B_01_22_clippedtrimmed.fastq.sam\nRI_B_01_22_clippedtrimmed.fastq.sam          VA_B_09_SNP_clippedtrimmed.fastq_counts.txt\nRI_B_08_SNP_clippedtrimmed.fastq_counts.txt  VA_B_09_SNP_clippedtrimmed.fastq.sam\nRI_B_08_SNP_clippedtrimmed.fastq.sam         VA_W_01_14_clippedtrimmed.fastq_counts.txt\nRI_W_01_14_clippedtrimmed.fastq_counts.txt   VA_W_01_14_clippedtrimmed.fastq.sam\nRI_W_01_14_clippedtrimmed.fastq.sam          VA_W_01_18_clippedtrimmed.fastq_counts.txt\nRI_W_01_18_clippedtrimmed.fastq_counts.txt   VA_W_01_18_clippedtrimmed.fastq.sam\nRI_W_01_18_clippedtrimmed.fastq.sam          VA_W_01_22_clippedtrimmed.fastq_counts.txt\nRI_W_01_22_clippedtrimmed.fastq_counts.txt   VA_W_01_22_clippedtrimmed.fastq.sam\nRI_W_01_22_clippedtrimmed.fastq.sam          VA_W_08_SNP_clippedtrimmed.fastq_counts.txt\nRI_W_08_SNP_clippedtrimmed.fastq_counts.txt  VA_W_08_SNP_clippedtrimmed.fastq.sam\nRI_W_08_SNP_clippedtrimmed.fastq.sam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003escp YOURNAMEFullCounts_summed.txt to your laptop\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ pwd\n/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/Day10\n(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS/APFullCounts_summed.txt .\napear012@turing.hpc.odu.edu's password: \nAPFullCounts_summed.txt                                                      100%  762KB 182.5KB/s   00:04    \n(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ ls\nAPFullCounts_summed.txt\ncoral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\ncoral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen\nhomework_day10.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ pwd\n/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/Day10\n(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS/APFullCounts_summed.txt .\napear012@turing.hpc.odu.edu's password: \nAPFullCounts_summed.txt                                                      100%  762KB 182.5KB/s   00:04    \n(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ ls\nAPFullCounts_summed.txt\ncoral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\ncoral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen\nhomework_day10.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eedit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eI used the find a replace feature of BBedit (without GREP) to replace each instance of '_clippedtrimmed.fastq_counts.txt_UniqueTotReads' from the end of the file names\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 11 24-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-11-24-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 11 24-Feb-2021\" href=\"#day-11-24-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eDownload the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eI did this.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 12 26-Feb-2021\u003c/h2\u003e\u003ca id=\"user-content-day-12-26-feb-2021\" class=\"anchor\" aria-label=\"Permalink: Day 12 26-Feb-2021\" href=\"#day-12-26-feb-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 13 3-Mar-2021\u003c/h2\u003e\u003ca id=\"user-content-day-13-3-mar-2021\" class=\"anchor\" aria-label=\"Permalink: Day 13 3-Mar-2021\" href=\"#day-13-3-mar-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecreate an sbatch script to rm -r your sandbox/YOURNAME directory\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003esubmit your sbatch script and add the entry to your logfile/readme\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[apear012@coreV2-25-072 sandboxes]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes\n[apear012@coreV2-25-072 sandboxes]$ cat AP_SandboxRemoval.sh\n#!/bin/bash -l\n\n#SBATCH -o AP_SandboxRemoval.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_SandboxRemoval\n\nrm -r ./apearson\n[apear012@coreV2-25-072 sandboxes]$ sbatch AP_SandboxRemoval.sh \nSubmitted batch job 9284108\n[apear012@coreV2-25-072 sandboxes]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9284108      main AP_Sandb apear012 PD       0:00      1 (Priority) \n           9284102      main       sh apear012  R       0:58      1 coreV2-25-072 \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[apear012@coreV2-25-072 sandboxes]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes\n[apear012@coreV2-25-072 sandboxes]$ cat AP_SandboxRemoval.sh\n#!/bin/bash -l\n\n#SBATCH -o AP_SandboxRemoval.txt\n#SBATCH -n 1\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AP_SandboxRemoval\n\nrm -r ./apearson\n[apear012@coreV2-25-072 sandboxes]$ sbatch AP_SandboxRemoval.sh \nSubmitted batch job 9284108\n[apear012@coreV2-25-072 sandboxes]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9284108      main AP_Sandb apear012 PD       0:00      1 (Priority) \n           9284102      main       sh apear012  R       0:58      1 coreV2-25-072 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Andrew's 2021 Advanced Bioinformatics Course Notebook","anchor":"andrews-2021-advanced-bioinformatics-course-notebook","htmlText":"Andrew's 2021 Advanced Bioinformatics Course Notebook"},{"level":2,"text":"Day 02 22-Jan-2021","anchor":"day-02-22-jan-2021","htmlText":"Day 02 22-Jan-2021"},{"level":2,"text":"Day 03 27-Jan-2021","anchor":"day-03-27-jan-2021","htmlText":"Day 03 27-Jan-2021"},{"level":2,"text":"Day 04 29-Jan-2021","anchor":"day-04-29-jan-2021","htmlText":"Day 04 29-Jan-2021"},{"level":2,"text":"Day 05 03-Feb-2021","anchor":"day-05-03-feb-2021","htmlText":"Day 05 03-Feb-2021"},{"level":2,"text":"Day 06 05-Feb-2021","anchor":"day-06-05-feb-2021","htmlText":"Day 06 05-Feb-2021"},{"level":2,"text":"Day 07 10-Feb-2021","anchor":"day-07-10-feb-2021","htmlText":"Day 07 10-Feb-2021"},{"level":2,"text":"Day 08 12-Feb-2021","anchor":"day-08-12-feb-2021","htmlText":"Day 08 12-Feb-2021"},{"level":2,"text":"Day 09 17-Feb-2021","anchor":"day-09-17-feb-2021","htmlText":"Day 09 17-Feb-2021"},{"level":2,"text":"Day 10 19-Feb-2021","anchor":"day-10-19-feb-2021","htmlText":"Day 10 19-Feb-2021"},{"level":2,"text":"Day 11 24-Feb-2021","anchor":"day-11-24-feb-2021","htmlText":"Day 11 24-Feb-2021"},{"level":2,"text":"Day 12 26-Feb-2021","anchor":"day-12-26-feb-2021","htmlText":"Day 12 26-Feb-2021"},{"level":2,"text":"Day 13 3-Mar-2021","anchor":"day-13-3-mar-2021","htmlText":"Day 13 3-Mar-2021"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Facpearson888%2Fadvanced_genomics_apearson"}}],"overviewFilesProcessingTime":73.08004,"createFromTemplatePath":"/new?template_name=advanced_genomics_apearson\u0026template_owner=acpearson888"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/acpearson888/advanced_genomics_apearson/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/acpearson888/advanced_genomics_apearson/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/acpearson888/advanced_genomics_apearson/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/acpearson888/advanced_genomics_apearson/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/acpearson888/advanced_genomics_apearson/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           38 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/acpearson888/advanced_genomics_apearson/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/acpearson888/advanced_genomics_apearson/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/acpearson888/advanced_genomics_apearson/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Andrew's 2021 Advanced Bioinformatics Course Notebook
                     </h1>
                     <a aria-label="Permalink: Andrew's 2021 Advanced Bioinformatics Course Notebook" class="anchor" href="#andrews-2021-advanced-bioinformatics-course-notebook" id="user-content-andrews-2021-advanced-bioinformatics-course-notebook">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 02 22-Jan-2021
                     </h2>
                     <a aria-label="Permalink: Day 02 22-Jan-2021" class="anchor" href="#day-02-22-jan-2021" id="user-content-day-02-22-jan-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      execute a pwd command and start a log of your commands with a header for today's date in your README.md github logfile in your workspace
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 exercises]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises">
                     <pre class="notranslate"><code>[apear012@turing1 exercises]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise* .
[apear012@turing1 exercises]$ ls
Exercise2.fasta.gz  Exercise2.fastq.tar.gz">
                     <pre class="notranslate"><code>[apear012@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise* .
[apear012@turing1 exercises]$ ls
Exercise2.fasta.gz  Exercise2.fastq.tar.gz
</code></pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      unzip and untar the files
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 exercises]$ gunzip Exercise2.fasta.gz
[apear012@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq.tar.gz">
                     <pre class="notranslate"><code>[apear012@turing1 exercises]$ gunzip Exercise2.fasta.gz
[apear012@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq.tar.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     If you wanted to gunzip without losing the original file, you would use:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="$ gunzip -c Exercise2.fasta.gz &gt; Exercise2.fasta">
                     <pre class="notranslate"><code>$ gunzip -c Exercise2.fasta.gz &gt; Exercise2.fasta
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 exercises]$ tar -xzvf Exercise2.fastq.tar.gz 
Exercise2.fastq
[apear012@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz">
                     <pre class="notranslate"><code>[apear012@turing1 exercises]$ tar -xzvf Exercise2.fastq.tar.gz 
Exercise2.fastq
[apear012@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz
</code></pre>
                    </div>
                    <ol dir="auto" start="8">
                     <li>
                      calculate how many sequences are in each file and add these results to your notebook file
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 exercises]$ grep -c '&gt;' Exercise2.fasta
138">
                     <pre class="notranslate"><code>[apear012@turing1 exercises]$ grep -c '&gt;' Exercise2.fasta
138
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 exercises]$ wc -l Exercise2.fastq
245216 Exercise2.fastq
[apear012@turing1 exercises]$ echo 245216/4 | bc 
61304">
                     <pre class="notranslate"><code>[apear012@turing1 exercises]$ wc -l Exercise2.fastq
245216 Exercise2.fastq
[apear012@turing1 exercises]$ echo 245216/4 | bc 
61304
</code></pre>
                    </div>
                    <ol dir="auto" start="9">
                     <li>
                      cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py .
[apear012@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py">
                     <pre class="notranslate"><code>[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py .
[apear012@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py
</code></pre>
                    </div>
                    <ol dir="auto" start="10">
                     <li>
                      start an interactive compute session and re-navigate to your exercises directory
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='wahab-01:~&gt; salloc
salloc: Pending job allocation 133736
salloc: job 133736 queued and waiting for resources
salloc: job 133736 has been allocated resources
salloc: Granted job allocation 133736
This session will be terminated in 7 days. If your application requires
a longer excution time, please use command "salloc -t N-0" where N is the
number of days that you need.'>
                     <pre class="notranslate"><code>wahab-01:~&gt; salloc
salloc: Pending job allocation 133736
salloc: job 133736 queued and waiting for resources
salloc: job 133736 has been allocated resources
salloc: Granted job allocation 133736
This session will be terminated in 7 days. If your application requires
a longer excution time, please use command "salloc -t N-0" where N is the
number of days that you need.
</code></pre>
                    </div>
                    <ol dir="auto" start="11">
                     <li>
                      run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises&gt; ../../scripts//avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 138
contigs &gt;= 1000bp = 138
contigs &gt;= 2000bp = 135">
                     <pre class="notranslate"><code> e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/exercises&gt; ../../scripts//avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 138
contigs &gt;= 1000bp = 138
contigs &gt;= 2000bp = 135
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 03 27-Jan-2021
                     </h2>
                     <a aria-label="Permalink: Day 03 27-Jan-2021" class="anchor" href="#day-03-27-jan-2021" id="user-content-day-03-27-jan-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Below is technically the day02 homework.
                    </p>
                    <ol dir="auto">
                     <li>
                      Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory, Andrew-HADB01
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-25-072 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@coreV2-25-072 data]$ nano APCopyLane01.sh
[apear012@coreV2-25-072 data]$ ls
APCopyLane01.sh  exercises
[apear012@coreV2-25-072 data]$ cat APCopyLane01.sh 
#!/bin/bash -l

#SBATCH -o APCopyLane01.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APCopyLane01

cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data">
                     <pre class="notranslate"><code>[apear012@coreV2-25-072 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@coreV2-25-072 data]$ nano APCopyLane01.sh
[apear012@coreV2-25-072 data]$ ls
APCopyLane01.sh  exercises
[apear012@coreV2-25-072 data]$ cat APCopyLane01.sh 
#!/bin/bash -l

#SBATCH -o APCopyLane01.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APCopyLane01

cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      <p dir="auto">
                       Add the content of your sbatch script to your logfile
Did this above.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       submit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-25-072 data]$ sbatch APCopyLane01.sh
Submitted batch job 9270452
[apear012@coreV2-25-072 data]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270428      main       sh apear012  R      39:36      1 coreV2-25-072 
           9270452      main APCopyLa apear012  R       0:29      1 coreV2-25-072 ">
                     <pre class="notranslate"><code>[apear012@coreV2-25-072 data]$ sbatch APCopyLane01.sh
Submitted batch job 9270452
[apear012@coreV2-25-072 data]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270428      main       sh apear012  R      39:36      1 coreV2-25-072 
           9270452      main APCopyLa apear012  R       0:29      1 coreV2-25-072 
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      <p dir="auto">
                       Make sure this is all documented on your github notebook
See above.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Write a sbatch script to gunzip all the fastq.gz files in your data directory
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@turing1 data]$ cat APGunzipLane01Files.sh 
#!/bin/bash -l

#SBATCH -o APGunzipLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APGunzipLane01Files

gunzip *.fastq.gz
[apear012@turing1 data]$ sbatch APGunzipLane01Files.sh 
Submitted batch job 9270467">
                     <pre class="notranslate"><code>[apear012@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@turing1 data]$ cat APGunzipLane01Files.sh 
#!/bin/bash -l

#SBATCH -o APGunzipLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APGunzipLane01Files

gunzip *.fastq.gz
[apear012@turing1 data]$ sbatch APGunzipLane01Files.sh 
Submitted batch job 9270467
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      This is the day03 homework:
                     </strong>
                    </p>
                    <ol dir="auto">
                     <li>
                      cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox.
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 apearson]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson
[apear012@turing1 apearson]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 .
[apear012@turing1 apearson]$ ls
AP_exercise1.txt  data  day03  groundrules.txt  scripts">
                     <pre class="notranslate"><code>[apear012@turing1 apearson]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson
[apear012@turing1 apearson]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 .
[apear012@turing1 apearson]$ ls
AP_exercise1.txt  data  day03  groundrules.txt  scripts
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      mkdir a fastq directory in your data directory and mv all the .fastq files into this directory
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 data]$ mkdir fastq
[apear012@turing1 data]$ mv ./*.fastq fastq
[apear012@turing1 data]$ cd fastq
[apear012@turing1 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq
HADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq
HADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq
HADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq">
                     <pre class="notranslate"><code>[apear012@turing1 data]$ mkdir fastq
[apear012@turing1 data]$ mv ./*.fastq fastq
[apear012@turing1 data]$ cd fastq
[apear012@turing1 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq
HADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq
HADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq
HADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder and less the new script and check out the usage statement
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/renamingtable_complete.txt .
[apear012@turing1 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq
HADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq
HADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt
HADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/renamingtable_complete.txt .
[apear012@turing1 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq
HADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq
HADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt
HADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py .
[apear012@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py
[apear012@turing1 scripts]$ less renamer_advbioinf.py ">
                     <pre class="notranslate"><code>[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py .
[apear012@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py
[apear012@turing1 scripts]$ less renamer_advbioinf.py 
</code></pre>
                    </div>
                    <p dir="auto">
                     renamer_advbioinf.py :
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1
	if linecount&gt;=2:
		cols=line.rstrip().split('\t')
		print 'mv %s %s' %(cols[0], cols[1])
#		os.popen('mv %s %s' %(cols[0], cols[1]))">
                     <pre class="notranslate"><code>#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1
	if linecount&gt;=2:
		cols=line.rstrip().split('\t')
		print 'mv %s %s' %(cols[0], cols[1])
#		os.popen('mv %s %s' %(cols[0], cols[1]))
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ python ../../scripts/renamer_advbioinf.py renamingtable_complete.txt
mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq
mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq
mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq
mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq
mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq
mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq
mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq
mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq
mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq
mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq
mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq
mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq
mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq
mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq
mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq
mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq
mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq
mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq
mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq
mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq
mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq
mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq
mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq
mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq
mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq
mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq
mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq
mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq
mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq
mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq
mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq
mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq
mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq
mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq
mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq
mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq
mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq
mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq
mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq
mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq
mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq
mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq
mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq
mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq
mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq
mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq
mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq
mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq
mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq
mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq
mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq
mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq
mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq
mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq
mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq
mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq
mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq
mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq
mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq
mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq
mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq
mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq
mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq
mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq
mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq
mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq
mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq
mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq
mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq
mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq
mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq
mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq
mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq
mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq
mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq
mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq
mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq
mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq
mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq
mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq
mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq
mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq
mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq
mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq
mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq
mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq
mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq
mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq
mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq
mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq
mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq
mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq
mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq
mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq
mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq
mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ python ../../scripts/renamer_advbioinf.py renamingtable_complete.txt
mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq
mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq
mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq
mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq
mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq
mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq
mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq
mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq
mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq
mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq
mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq
mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq
mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq
mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq
mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq
mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq
mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq
mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq
mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq
mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq
mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq
mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq
mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq
mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq
mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq
mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq
mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq
mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq
mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq
mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq
mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq
mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq
mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq
mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq
mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq
mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq
mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq
mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq
mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq
mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq
mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq
mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq
mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq
mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq
mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq
mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq
mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq
mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq
mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq
mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq
mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq
mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq
mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq
mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq
mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq
mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq
mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq
mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq
mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq
mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq
mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq
mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq
mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq
mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq
mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq
mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq
mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq
mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq
mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq
mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq
mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq
mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq
mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq
mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq
mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq
mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq
mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq
mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq
mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq
mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq
mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq
mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq
mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq
mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq
mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq
mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq
mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq
mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq
mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq
mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq
mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq
mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq
mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq
mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq
mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq
mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print.
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@turing1 scripts]$ cat renamer_advbioinf.py 
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1
	if linecount&gt;=2:
		cols=line.rstrip().split('\t')
		print 'mv %s %s' %(cols[0], cols[1])
#		os.popen('mv %s %s' %(cols[0], cols[1]))
[apear012@turing1 scripts]$ nano renamer_advbioinf.py 
[apear012@turing1 scripts]$ cat renamer_advbioinf.py
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1
	if linecount&gt;=2:
		cols=line.rstrip().split('\t')
#		print 'mv %s %s' %(cols[0], cols[1])
		os.popen('mv %s %s' %(cols[0], cols[1]))">
                     <pre class="notranslate"><code>[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@turing1 scripts]$ cat renamer_advbioinf.py 
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1
	if linecount&gt;=2:
		cols=line.rstrip().split('\t')
		print 'mv %s %s' %(cols[0], cols[1])
#		os.popen('mv %s %s' %(cols[0], cols[1]))
[apear012@turing1 scripts]$ nano renamer_advbioinf.py 
[apear012@turing1 scripts]$ cat renamer_advbioinf.py
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1
	if linecount&gt;=2:
		cols=line.rstrip().split('\t')
#		print 'mv %s %s' %(cols[0], cols[1])
		os.popen('mv %s %s' %(cols[0], cols[1]))
</code></pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ nano APRenameLane01Files.sh">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ nano APRenameLane01Files.sh
</code></pre>
                    </div>
                    <p dir="auto">
                     APRenameLane01Files.sh:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o APRenameLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APRenameLane01Files

../../scripts/renamer_advbioinf.py renamingtable_complete.txt ./*.fastq*">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o APRenameLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APRenameLane01Files

../../scripts/renamer_advbioinf.py renamingtable_complete.txt ./*.fastq*
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ sbatch APRenameLane01Files.sh 
Submitted batch job 9270537
[apear012@turing1 fastq]$ ls
APRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq
APRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq
renamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq
RI_B_01_14.fastq            RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ sbatch APRenameLane01Files.sh 
Submitted batch job 9270537
[apear012@turing1 fastq]$ ls
APRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq
APRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq
renamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq
RI_B_01_14.fastq            RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq
</code></pre>
                    </div>
                    <ol dir="auto" start="7">
                     <li>
                      <p dir="auto">
                       Make sure this is all documented on your github page
see above
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       The naming convention for the files is as follows:
                      </p>
                     </li>
                    </ol>
                    <ul dir="auto">
                     <li>
                      SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq
                     </li>
                     <li>
                      There are 2 sources: Virginia and Rhode Island
                     </li>
                     <li>
                      There are 2 symbiotic states: Brown and White
                     </li>
                    </ul>
                    <ol dir="auto" start="9">
                     <li>
                      <p dir="auto">
                       Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory.
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts

[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py .
[apear012@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py">
                     <pre class="notranslate"><code>[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts

[apear012@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py .
[apear012@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py
</code></pre>
                    </div>
                    <ol dir="auto" start="11">
                     <li>
                      Less/head the new script and check out the usage statement
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py 
#!/usr/bin/env python
# Written by Dan Barshis

import sys, os

########### Usage #############
# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles
# This should be run from within the folder with all of your original .fastqstrim
# Will quality trim multiple SINGLE-END fastq files
# Things to customize for your particular platform:">
                     <pre class="notranslate"><code>[apear012@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py 
#!/usr/bin/env python
# Written by Dan Barshis

import sys, os

########### Usage #############
# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles
# This should be run from within the folder with all of your original .fastqstrim
# Will quality trim multiple SINGLE-END fastq files
# Things to customize for your particular platform:
</code></pre>
                    </div>
                    <ol dir="auto" start="12">
                     <li>
                      cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files.
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt .
[apear012@turing1 fastq]$ ls
adapterlist_advbioinf.txt   RI_B_01_14.fastq   RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq
APRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq
APRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq
renamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt .
[apear012@turing1 fastq]$ ls
adapterlist_advbioinf.txt   RI_B_01_14.fastq   RI_W_01_14.fastq   VA_B_01_14.fastq   VA_W_01_14.fastq
APRenameLane01Files.sh      RI_B_01_18.fastq   RI_W_01_18.fastq   VA_B_01_18.fastq   VA_W_01_18.fastq
APRenameLane01Files.txt     RI_B_01_22.fastq   RI_W_01_22.fastq   VA_B_01_22.fastq   VA_W_01_22.fastq
renamingtable_complete.txt  RI_B_08_SNP.fastq  RI_W_08_SNP.fastq  VA_B_09_SNP.fastq  VA_W_08_SNP.fastq
</code></pre>
                    </div>
                    <ol dir="auto" start="13">
                     <li>
                      Make a sbatch script for the Trimclipfilter... script and run it on your fastq files
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ nano APTrimLane01Files.sh
[apear012@turing1 fastq]$ cat APTrimLane01Files.sh 
#!/bin/bash -l

#SBATCH -o APTrimLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APTrimLane01Files

../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt ./*.fastq*
[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh 
Submitted batch job 9270540">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ nano APTrimLane01Files.sh
[apear012@turing1 fastq]$ cat APTrimLane01Files.sh 
#!/bin/bash -l

#SBATCH -o APTrimLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APTrimLane01Files

../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt ./*.fastq*
[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh 
Submitted batch job 9270540
</code></pre>
                    </div>
                    <p dir="auto">
                     Email received immediately saying my job had failed.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ nano APTrimLane01Files.sh 
[apear012@turing1 fastq]$ cat APTrimLane01Files.sh
#!/bin/bash -l

#SBATCH -o APTrimLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APTrimLane01Files

../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ nano APTrimLane01Files.sh 
[apear012@turing1 fastq]$ cat APTrimLane01Files.sh
#!/bin/bash -l

#SBATCH -o APTrimLane01Files.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APTrimLane01Files

../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh
Submitted batch job 9270839
[apear012@turing1 fastq]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270839      main APTrimLa apear012  R       0:00      1 coreV1-22-016 ">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ sbatch APTrimLane01Files.sh
Submitted batch job 9270839
[apear012@turing1 fastq]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270839      main APTrimLa apear012  R       0:00      1 coreV1-22-016 
</code></pre>
                    </div>
                    <ol dir="auto" start="14">
                     <li>
                      This will take a while (like days)
                     </li>
                     <li>
                      Now might be a good time to update everything on your github
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 04 29-Jan-2021
                     </h2>
                     <a aria-label="Permalink: Day 04 29-Jan-2021" class="anchor" href="#day-04-29-jan-2021" id="user-content-day-04-29-jan-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Add your trimclipstats.txt output to the full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps
                     </li>
                    </ol>
                    <p dir="auto">
                     1a. Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-25-072 filteringstats]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/filteringstats
[apear012@turing1 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015

This script takes a stats output file from fastx_clipper and converts it into a table.

Usage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt

Options (-c and -v must be listed separately to run together):
-h	Display this help message
-c	Use comma delimiter instead of tabs
-v	Verbose mode (print steps to stdout)">
                     <pre class="notranslate"><code>[apear012@coreV2-25-072 filteringstats]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/filteringstats
[apear012@turing1 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015

This script takes a stats output file from fastx_clipper and converts it into a table.

Usage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt

Options (-c and -v must be listed separately to run together):
-h	Display this help message
-c	Use comma delimiter instead of tabs
-v	Verbose mode (print steps to stdout)
</code></pre>
                    </div>
                    <p dir="auto">
                     1b. Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-25-072 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt ANDREW_trimclipstatsout.txt">
                     <pre class="notranslate"><code>[apear012@coreV2-25-072 filteringstats]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt ANDREW_trimclipstatsout.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     1c. Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 filteringstats]$ tail -n +2 ANDREW_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt">
                     <pre class="notranslate"><code>[apear012@turing1 filteringstats]$ tail -n +2 ANDREW_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      <p dir="auto">
                       Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline
                       <a href="https://github.com/bethsheets/SNPcalling_tutorial">
                        https://github.com/bethsheets/SNPcalling_tutorial
                       </a>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="module load bowtie2/2.2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \
--rg SM:${i%_clippedtrimmed.fastq} \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \
&gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done">
                     <pre class="notranslate"><code>module load bowtie2/2.2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \
--rg SM:${i%_clippedtrimmed.fastq} \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \
&gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ nano APbowtie.sh">
                     <pre class="notranslate"><code>apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ nano APbowtie.sh
</code></pre>
                    </div>
                    <p dir="auto">
                     APbowtie.sh:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o APbowtie.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APbowtie

module load bowtie2/2.2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \
--rg SM:${i%_clippedtrimmed.fastq} \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \
&gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o APbowtie.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APbowtie

module load bowtie2/2.2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \
--rg SM:${i%_clippedtrimmed.fastq} \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \
&gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ sbatch APbowtie.sh 
Submitted batch job 9271109
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9271094      main       sh apear012  R      54:07      1 coreV2-25-072 
           9271109      main APbowtie apear012  R       0:07      1 coreV2-25-007 ">
                     <pre class="notranslate"><code>apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ sbatch APbowtie.sh 
Submitted batch job 9271109
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9271094      main       sh apear012  R      54:07      1 coreV2-25-072 
           9271109      main APbowtie apear012  R       0:07      1 coreV2-25-007 
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      Submit and add everything to your logfile.
                     </li>
                    </ol>
                    <p dir="auto">
                     See above.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 05 03-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 05 03-Feb-2021" class="anchor" href="#day-05-03-feb-2021" id="user-content-day-05-03-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Team up with a partner for this one and work only on a combined set of data for your two lanes of sequences
                     </li>
                    </ol>
                    <p dir="auto">
                     Partner: Areej
Lanes: 01, 06, and 07
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      <p dir="auto">
                       Run the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Modify the below sbatch script (note the differences in the header compared to the previous one)
                      </p>
                     </li>
                    </ol>
                    <p dir="auto">
                     original script:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o OUTFILE.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=YOUREMAIL@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=YOURJOBNAME

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single ALLSINGLEENDREADSSEPARATEDBYCOMMAS --CPU 32">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o OUTFILE.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=YOUREMAIL@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=YOURJOBNAME

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single ALLSINGLEENDREADSSEPARATEDBYCOMMAS --CPU 32
</code></pre>
                    </div>
                    <p dir="auto">
                     Edited script (renamed APAMAssemble.sh in /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o APAMAssemble.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APAMAssemble

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o APAMAssemble.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APAMAssemble

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      <p dir="auto">
                       Check
                       <a href="https://trinityrnaseq.github.io/" rel="nofollow">
                        https://trinityrnaseq.github.io/
                       </a>
                       for usage info
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Submit your trinity script
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 apam]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam
[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh 
Submitted batch job 9272358
[apear012@coreV2-22-007 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources) 
           9272334      main       sh apear012  R    1:19:02      1 coreV2-25-002 
           9272357      main       sh apear012  R       6:01      1 coreV2-22-007 ">
                     <pre class="notranslate"><code>[apear012@turing1 apam]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam
[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh 
Submitted batch job 9272358
[apear012@coreV2-22-007 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources) 
           9272334      main       sh apear012  R    1:19:02      1 coreV2-25-002 
           9272357      main       sh apear012  R       6:01      1 coreV2-22-007 
</code></pre>
                    </div>
                    <p dir="auto">
                     Job failed after ~15min.
                    </p>
                    <p dir="auto">
                     edited APAMAssemble.sh (removed --normalize_reads):
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o APAMAssemble.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APAMAssemble

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o APAMAssemble.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APAMAssemble

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 apam]$ sbatch APAMAssemble.sh
Submitted batch job 9272385
[apear012@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 
           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003 ">
                     <pre class="notranslate"><code>[apear012@turing1 apam]$ sbatch APAMAssemble.sh
Submitted batch job 9272385
[apear012@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 
           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003 
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 06 05-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 06 05-Feb-2021" class="anchor" href="#day-06-05-feb-2021" id="user-content-day-06-05-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 apam]$ salloc
salloc: Pending job allocation 9272856
salloc: job 9272856 queued and waiting for resources
salloc: job 9272856 has been allocated resources
salloc: Granted job allocation 9272856">
                     <pre class="notranslate"><code>[apear012@turing1 apam]$ salloc
salloc: Pending job allocation 9272856
salloc: job 9272856 queued and waiting for resources
salloc: job 9272856 has been allocated resources
salloc: Granted job allocation 9272856
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-22-028 djbtestassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly
[apear012@coreV2-22-028 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta 


################################
## Counts of transcripts, etc.
################################
Total trinity 'genes':	20980
Total trinity transcripts:	21992
Percent GC: 46.21

########################################
Stats based on ALL transcript contigs:
########################################

	Contig N10: 1178
	Contig N20: 694
	Contig N30: 514
	Contig N40: 414
	Contig N50: 347

	Median contig length: 273
	Average contig: 356.95
	Total assembled bases: 7850006


#####################################################
## Stats based on ONLY LONGEST ISOFORM per 'GENE':
#####################################################

	Contig N10: 1027
	Contig N20: 643
	Contig N30: 486
	Contig N40: 397
	Contig N50: 336

	Median contig length: 271
	Average contig: 347.72
	Total assembled bases: 7295061
">
                     <pre class="notranslate"><code>[apear012@coreV2-22-028 djbtestassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly
[apear012@coreV2-22-028 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta 


################################
## Counts of transcripts, etc.
################################
Total trinity 'genes':	20980
Total trinity transcripts:	21992
Percent GC: 46.21

########################################
Stats based on ALL transcript contigs:
########################################

	Contig N10: 1178
	Contig N20: 694
	Contig N30: 514
	Contig N40: 414
	Contig N50: 347

	Median contig length: 273
	Average contig: 356.95
	Total assembled bases: 7850006


#####################################################
## Stats based on ONLY LONGEST ISOFORM per 'GENE':
#####################################################

	Contig N10: 1027
	Contig N20: 643
	Contig N30: 486
	Contig N40: 397
	Contig N50: 336

	Median contig length: 271
	Average contig: 347.72
	Total assembled bases: 7295061

</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-22-028 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
The total number of sequences is 15079
The average sequence length is 876
The total number of bases is 13210470
The minimum sequence length is 500
The maximum sequence length is 10795
The N50 is 881
Median Length = 578
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 15079
contigs &gt;= 1000bp = 3660
contigs &gt;= 2000bp = 536
[apear012@coreV2-22-028 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts">
                     <pre class="notranslate"><code>[apear012@coreV2-22-028 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
The total number of sequences is 15079
The average sequence length is 876
The total number of bases is 13210470
The minimum sequence length is 500
The maximum sequence length is 10795
The N50 is 881
Median Length = 578
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 15079
contigs &gt;= 1000bp = 3660
contigs &gt;= 2000bp = 536
[apear012@coreV2-22-028 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/scripts
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly/Trinity.fasta 
The total number of sequences is 21992
The average sequence length is 356
The total number of bases is 7850006
The minimum sequence length is 192
The maximum sequence length is 10795
The N50 is 347
Median Length = 355
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 2862
contigs &gt;= 1000bp = 607
contigs &gt;= 2000bp = 91">
                     <pre class="notranslate"><code>[apear012@coreV2-22-028 scripts]$ python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/djbtestassembly/Trinity.fasta 
The total number of sequences is 21992
The average sequence length is 356
The total number of bases is 7850006
The minimum sequence length is 192
The maximum sequence length is 10795
The N50 is 347
Median Length = 355
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 2862
contigs &gt;= 1000bp = 607
contigs &gt;= 2000bp = 91
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:
a-the mean percent "overall alignment rate"
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ grep 'overall alignment rate' APbowtie.txt | cut -d '%' -f 1
3.22
1.74
1.41
1.65
2.01
1.86
1.34
1.49
2.68
7.09
1.84
1.83
1.75
1.70
52.97
1.47">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ grep 'overall alignment rate' APbowtie.txt | cut -d '%' -f 1
3.22
1.74
1.41
1.65
2.01
1.86
1.34
1.49
2.68
7.09
1.84
1.83
1.75
1.70
52.97
1.47
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='b-the mean percent reads "aligned exactly 1 time"'>
                     <pre class="notranslate"><code>b-the mean percent reads "aligned exactly 1 time"
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2
1.63
1.18
1.00
1.18
0.67
1.21
0.88
0.94
1.71
5.19
1.29
1.23
1.09
1.18
0.55
1.03">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2
1.63
1.18
1.00
1.18
0.67
1.21
0.88
0.94
1.71
5.19
1.29
1.23
1.09
1.18
0.55
1.03
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='c-the mean number of reads "aligned exactly 1 time"'>
                     <pre class="notranslate"><code>c-the mean number of reads "aligned exactly 1 time"
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d ' ' -f 5
117712
281939
240500
256601
40180
298308
182262
381640
354945
1169727
374006
332280
168092
330623
397
276577">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ grep 'aligned exactly 1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d ' ' -f 5
117712
281939
240500
256601
40180
298308
182262
381640
354945
1169727
374006
332280
168092
330623
397
276577
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='d-the mean percent reads "aligned &gt;1 times"'>
                     <pre class="notranslate"><code>d-the mean percent reads "aligned &gt;1 times"
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ grep 'aligned &gt;1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2
1.59
0.56
0.41
0.47
1.34
0.65
0.45
0.55
0.97
1.91
0.55
0.60
0.66
0.51
52.42
0.44">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ grep 'aligned &gt;1 time' APbowtie.txt | cut -d '%' -f 1 | cut -d '(' -f 2
1.59
0.56
0.41
0.47
1.34
0.65
0.45
0.55
0.97
1.91
0.55
0.60
0.66
0.51
52.42
0.44
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="hint use grep and paste into excel">
                     <pre class="notranslate"><code>hint use grep and paste into excel
</code></pre>
                    </div>
                    <p dir="auto">
                     Copied each of these outputs into excel and averaged. This was used for the next step to make the alignstatsAP.txt
                    </p>
                    <ol dir="auto" start="4">
                     <li>
                      add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:
LaneX_yourinitials	b-the mean percent "overall alignment rate"	c-the mean percent reads "aligned exactly 1 time"	d-the mean number of reads "aligned exactly 1 time"	e-the mean percent reads "aligned &gt;1 times"
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ nano alignstatsAP.txt
[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt 
Lane1AP	5.378125	1.3725	300361.8125	4.005
[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
[apear012@turing1 QCFastqs]$ tail /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
LaneX	overallalignmentrate	percsinglyaligned	numsinglyaligned	percmultiplyaligned
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
Lane1AP	5.378125	1.3725	300361.8125	4.005">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ nano alignstatsAP.txt
[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt 
Lane1AP	5.378125	1.3725	300361.8125	4.005
[apear012@turing1 QCFastqs]$ cat alignstatsAP.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
[apear012@turing1 QCFastqs]$ tail /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
LaneX	overallalignmentrate	percsinglyaligned	numsinglyaligned	percmultiplyaligned
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
Lane1AP	5.378125	1.3725	300361.8125	4.005
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      Data cleanup and archiving:
                     </li>
                    </ol>
                    <p dir="auto">
                     a. mv your Trinity.fasta file from your trinity_out_dir to a folder called testassembly in your data directory
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 trinity_out_dir]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/trinity_out_dir
[apear012@turing1 trinity_out_dir]$ mkdir ../../../../testassembly
[apear012@turing1 trinity_out_dir]$ mv Trinity.fasta ../../../../testassembly/">
                     <pre class="notranslate"><code>[apear012@turing1 trinity_out_dir]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/trinity_out_dir
[apear012@turing1 trinity_out_dir]$ mkdir ../../../../testassembly
[apear012@turing1 trinity_out_dir]$ mv Trinity.fasta ../../../../testassembly/
</code></pre>
                    </div>
                    <p dir="auto">
                     b. set up a sbatch script to:
rm -r YOURtrinity_out_dir
rm -r YOURoriginalfastqs
rm -r YOURfilteringstats # as long as you've already appended your filteringstats output to the class table as part of homework_day04.txt
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ nano AP_Cleanup.sh">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq
[apear012@turing1 fastq]$ nano AP_Cleanup.sh
</code></pre>
                    </div>
                    <p dir="auto">
                     AP_Cleanup.sh:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o AP_Cleanup.txt
#SBATCH -n 1
#SBATCH --mail-user=acpearson@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Cleanup

rm -r ./QCFastqs/apam/trinity_out_dir
rm -r filteringstats
rm -r originalfastqs">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o AP_Cleanup.txt
#SBATCH -n 1
#SBATCH --mail-user=acpearson@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Cleanup

rm -r ./QCFastqs/apam/trinity_out_dir
rm -r filteringstats
rm -r originalfastqs
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 fastq]$ sbatch AP_Cleanup.sh 
Submitted batch job 9273047
[apear012@turing1 fastq]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9273047      main AP_Clean apear012  R       0:01      1 coreV1-22-003 ">
                     <pre class="notranslate"><code>[apear012@turing1 fastq]$ sbatch AP_Cleanup.sh 
Submitted batch job 9273047
[apear012@turing1 fastq]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9273047      main AP_Clean apear012  R       0:01      1 coreV1-22-003 
</code></pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      submit a blast against sprot from your testassembly folder using the following command
#!/bin/bash -l
                     </li>
                    </ol>
                    <p dir="auto">
                     #SBATCH -o OUTFILENAME.txt
#SBATCH -n 6
                     <br/>
                     #SBATCH --mail-user=EMAIL
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME
                    </p>
                    <p dir="auto">
                     enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6
                     <br/>
                     -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
###Exploring our SAM files, check out
                     <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output" rel="nofollow">
                      http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output
                     </a>
                     for bowtie2 specific output and
                     <a href="http://bio-bwa.sourceforge.net/bwa.shtml#4" rel="nofollow">
                      http://bio-bwa.sourceforge.net/bwa.shtml#4
                     </a>
                     for general SAM output
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly
[apear012@turing1 testassembly]$ nano AP_Blast.sh">
                     <pre class="notranslate"><code>[apear012@turing1 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly
[apear012@turing1 testassembly]$ nano AP_Blast.sh
</code></pre>
                    </div>
                    <p dir="auto">
                     AP_Blast.sh:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o AP_Blast.txt
#SBATCH -n 1
#SBATCH --mail-user=acpearson@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Blast

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o AP_Blast.txt
#SBATCH -n 1
#SBATCH --mail-user=acpearson@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Blast

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 testassembly]$ ls
AP_Blast.sh  Trinity.fasta
[apear012@turing1 testassembly]$ sbatch AP_Blast.sh 
Submitted batch job 9273048
[apear012@turing1 testassembly]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9273048      main AP_Blast apear012  R       0:02      1 coreV1-22-016 ">
                     <pre class="notranslate"><code>[apear012@turing1 testassembly]$ ls
AP_Blast.sh  Trinity.fasta
[apear012@turing1 testassembly]$ sbatch AP_Blast.sh 
Submitted batch job 9273048
[apear012@turing1 testassembly]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9273048      main AP_Blast apear012  R       0:02      1 coreV1-22-016 
</code></pre>
                    </div>
                    <ol dir="auto" start="7">
                     <li>
                      head one of your .sam files to look at the header
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ head RI_B_01_14_clippedtrimmed.fastq.sam 
@HD	VN:1.0	SO:unsorted
@SQ	SN:TR78|c0_g1_i1_coral	LN:1109
@SQ	SN:TR78|c0_g2_i1_coral	LN:1109
@SQ	SN:TR79|c0_g1_i1_coral	LN:610
@SQ	SN:TR79|c0_g2_i1_coral	LN:1549
@SQ	SN:TR87|c0_g1_i1_coral	LN:732
@SQ	SN:TR93|c0_g1_i1_coral	LN:550
@SQ	SN:TR101|c0_g1_i1_coral	LN:673
@SQ	SN:TR104|c0_g1_i1_coral	LN:607
@SQ	SN:TR105|c0_g1_i1_coral	LN:587">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ head RI_B_01_14_clippedtrimmed.fastq.sam 
@HD	VN:1.0	SO:unsorted
@SQ	SN:TR78|c0_g1_i1_coral	LN:1109
@SQ	SN:TR78|c0_g2_i1_coral	LN:1109
@SQ	SN:TR79|c0_g1_i1_coral	LN:610
@SQ	SN:TR79|c0_g2_i1_coral	LN:1549
@SQ	SN:TR87|c0_g1_i1_coral	LN:732
@SQ	SN:TR93|c0_g1_i1_coral	LN:550
@SQ	SN:TR101|c0_g1_i1_coral	LN:673
@SQ	SN:TR104|c0_g1_i1_coral	LN:607
@SQ	SN:TR105|c0_g1_i1_coral	LN:587
</code></pre>
                    </div>
                    <ol dir="auto" start="8">
                     <li>
                      grep -v '@' your.sam | head to look at the sequence read lines, why does this work to exclude the header lines?
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ grep -v '@' RI_B_01_14_clippedtrimmed.fastq.sam | head
K00188:59:HMTFHBBXX:2:1101:25814:1701	16	TR14006|c0_g2_i1_coral	317	255	51M	*	0	CAGCTGGAAATTCACGTGGTGGCAGTGGAAGTGGCTTTTGACAGAAATCTG	AJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFFAA	AS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:19827:1912	0	TR6438|c0_g1_i3_coral	30	1	51M	*	0	TGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA	AAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:19827:1912	256	TR6438|c0_g1_i1_coral	30	255	51M	*	0	TGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA	AAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:19827:1912	256	TR6438|c0_g1_i4_coral	30	255	51M	*	0	TGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA	AAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23622:1912	16	TR43756|c0_g1_i1_coral	7449	1	51M	*	0	GGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC	JJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23622:1912	272	TR43756|c0_g1_i1_coral	2063	255	51M	*	0	GGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC	JJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23622:1912	272	TR43756|c0_g2_i1_coral	2077	255	51M	*	0	GGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC	JJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:20212:1947	0	TR11796|c0_g1_i1_sym	24	1	51M	*	0	TCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG	-AAFAFFFJA--FJJJ&lt;FJ77AFJ7-&lt;JAJFF7-F&lt;A7&lt;F-FJA&lt;FFJJFF	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:20212:1947	256	TR11796|c0_g2_i1_sym	552	255	51M	*	0	TCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG	-AAFAFFFJA--FJJJ&lt;FJ77AFJ7-&lt;JAJFF7-F&lt;A7&lt;F-FJA&lt;FFJJFF	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23723:1947	0	TR43756|c0_g2_i1_coral	1579	1	51M	*	0	AACCATAACGAGCATCATCTTGATTAAGCTCATTAGGGTTAGCCTCGGTAC	AAFFFJJJJJJJJJJJJJJJJJJJJJJJJ7&lt;FFJJJJJJJJFJJJJJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ grep -v '@' RI_B_01_14_clippedtrimmed.fastq.sam | head
K00188:59:HMTFHBBXX:2:1101:25814:1701	16	TR14006|c0_g2_i1_coral	317	255	51M	*	0	CAGCTGGAAATTCACGTGGTGGCAGTGGAAGTGGCTTTTGACAGAAATCTG	AJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFFAA	AS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:19827:1912	0	TR6438|c0_g1_i3_coral	30	1	51M	*	0	TGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA	AAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:19827:1912	256	TR6438|c0_g1_i1_coral	30	255	51M	*	0	TGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA	AAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:19827:1912	256	TR6438|c0_g1_i4_coral	30	255	51M	*	0	TGAACATAGGCCGCTAACATCAAGGCATAAAAGTTTTGTGGCCGTTTTCGA	AAFFFJJJJJJJJJJJJJJJFJJJFJJJJJJJJJJJJJJJJJJJJFJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23622:1912	16	TR43756|c0_g1_i1_coral	7449	1	51M	*	0	GGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC	JJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23622:1912	272	TR43756|c0_g1_i1_coral	2063	255	51M	*	0	GGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC	JJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23622:1912	272	TR43756|c0_g2_i1_coral	2077	255	51M	*	0	GGGGCCGAAGCCCCTGCAATTAAAATTGTTGACCACCTACATACCAAAGAC	JJJJJJJJJJJJFJJJJJJJJJJJFJFJJJJJJJJJJJJJJFJJJJFAAAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:20212:1947	0	TR11796|c0_g1_i1_sym	24	1	51M	*	0	TCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG	-AAFAFFFJA--FJJJ&lt;FJ77AFJ7-&lt;JAJFF7-F&lt;A7&lt;F-FJA&lt;FFJJFF	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:20212:1947	256	TR11796|c0_g2_i1_sym	552	255	51M	*	0	TCCACCTTCCTGCGCATACCATCCATCCTACGCTTGCTGCTGAAAACGTTG	-AAFAFFFJA--FJJJ&lt;FJ77AFJ7-&lt;JAJFF7-F&lt;A7&lt;F-FJA&lt;FFJJFF	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
K00188:59:HMTFHBBXX:2:1101:23723:1947	0	TR43756|c0_g2_i1_coral	1579	1	51M	*	0	AACCATAACGAGCATCATCTTGATTAAGCTCATTAGGGTTAGCCTCGGTAC	AAFFFJJJJJJJJJJJJJJJJJJJJJJJJ7&lt;FFJJJJJJJJFJJJJJJJJJ	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_01_14
</code></pre>
                    </div>
                    <p dir="auto">
                     This works because the -v flag for grep means find all lines that DO NOT include '@'. Since all the header lines include @, they are excluded.
                    </p>
                    <ol dir="auto" start="9">
                     <li>
                      in an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9273049
salloc: job 9273049 queued and waiting for resources
salloc: job 9273049 has been allocated resources
salloc: Granted job allocation 9273049
[apear012@coreV1-22-016 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_08_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_09_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_08_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9273049
salloc: job 9273049 queued and waiting for resources
salloc: job 9273049 has been allocated resources
salloc: Granted job allocation 9273049
[apear012@coreV1-22-016 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_01_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_01_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_01_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
[apear012@coreV1-22-016 QCFastqs]$ python /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py *SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_W_08_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_B_09_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
VA_W_08_SNP_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
</code></pre>
                    </div>
                    <ol dir="auto" start="10">
                     <li>
                      we need to run the read sorting step required for SNP calling, so if you have time, set up and run the following script on your .sam files to finish before Wednesday:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o OUTFILENAME.txt
#SBATCH -n 1         
#SBATCH --mail-user=EMAIL
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o OUTFILENAME.txt
#SBATCH -n 1         
#SBATCH --mail-user=EMAIL
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV1-22-016 QCFastqs]$ nano AP_Sort.sh">
                     <pre class="notranslate"><code>[apear012@coreV1-22-016 QCFastqs]$ nano AP_Sort.sh
</code></pre>
                    </div>
                    <p dir="auto">
                     AP_Sort.sh:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o AP_Sort.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Sort

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o AP_Sort.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Sort

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV1-22-016 QCFastqs]$ sbatch AP_Sort.sh 
Submitted batch job 9273050
[apear012@coreV1-22-016 QCFastqs]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9273050      main  AP_Sort apear012  R       0:11      1 coreV1-22-016 
           9273049      main       sh apear012  R      14:11      1 coreV1-22-016 
           9273048      main AP_Blast apear012  R      21:11      1 coreV1-22-016 ">
                     <pre class="notranslate"><code>[apear012@coreV1-22-016 QCFastqs]$ sbatch AP_Sort.sh 
Submitted batch job 9273050
[apear012@coreV1-22-016 QCFastqs]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9273050      main  AP_Sort apear012  R       0:11      1 coreV1-22-016 
           9273049      main       sh apear012  R      14:11      1 coreV1-22-016 
           9273048      main AP_Blast apear012  R      21:11      1 coreV1-22-016 
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 07 10-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 07 10-Feb-2021" class="anchor" href="#day-07-10-feb-2021" id="user-content-day-07-10-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Run the following command on your sprot output file to process into the contig length/match format that trinity examines
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o OUTFILE.TXT
#SBATCH -n 1
#SBATCH --mail-user=YOUREMAIL.odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o OUTFILE.TXT
#SBATCH -n 1
#SBATCH --mail-user=YOUREMAIL.odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly
[apear012@turing1 testassembly]$ nano AP_BlastxToSprot.sh">
                     <pre class="notranslate"><code>[apear012@turing1 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/testassembly
[apear012@turing1 testassembly]$ nano AP_BlastxToSprot.sh
</code></pre>
                    </div>
                    <p dir="auto">
                     AP_BlastxToSprot.sh:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o AP_BlastxToSprot.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_BlastxToSprot

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o AP_BlastxToSprot.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_BlastxToSprot

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 testassembly]$ sbatch AP_BlastxToSprot.sh 
Submitted batch job 9276483
[apear012@turing1 testassembly]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9276483      main AP_Blast apear012 PD       0:00      1 (Priority) 
[apear012@turing1 testassembly]$ ls
AP_Blast.sh   AP_BlastxToSprot.sh   blastx.outfmt6       blastx.outfmt6.hist.list         Trinity.fasta
AP_Blast.txt  AP_BlastxToSprot.txt  blastx.outfmt6.hist  blastx.outfmt6.w_pct_hit_length
[apear012@turing1 testassembly]$ cat *.hist
#hit_pct_cov_bin	count_in_bin	&gt;bin_below
100	198	198
90	87	285
80	88	373
70	110	483
60	146	629
50	210	839
40	339	1178
30	650	1828
20	1178	3006
10	909	3915
[apear012@turing1 testassembly]$ grep -c '&gt;' Trinity.fasta 
30194">
                     <pre class="notranslate"><code>[apear012@turing1 testassembly]$ sbatch AP_BlastxToSprot.sh 
Submitted batch job 9276483
[apear012@turing1 testassembly]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9276483      main AP_Blast apear012 PD       0:00      1 (Priority) 
[apear012@turing1 testassembly]$ ls
AP_Blast.sh   AP_BlastxToSprot.sh   blastx.outfmt6       blastx.outfmt6.hist.list         Trinity.fasta
AP_Blast.txt  AP_BlastxToSprot.txt  blastx.outfmt6.hist  blastx.outfmt6.w_pct_hit_length
[apear012@turing1 testassembly]$ cat *.hist
#hit_pct_cov_bin	count_in_bin	&gt;bin_below
100	198	198
90	87	285
80	88	373
70	110	483
60	146	629
50	210	839
40	339	1178
30	650	1828
20	1178	3006
10	909	3915
[apear012@turing1 testassembly]$ grep -c '&gt;' Trinity.fasta 
30194
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      Rm UNSORTED.bam
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-046 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ ls
alignstatsAP.txt                               RI_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam
apam                                           *stats.txt.bam
APbowtie.sh                                    *stats.txt.sam
APbowtie.txt                                   *stats.txt_UNSORTED.bam
AP_Sort.sh                                     VA_B_01_14_clippedtrimmed.fastq
AP_Sort.txt                                    VA_B_01_14_clippedtrimmed.fastq.bam
RI_B_01_14_clippedtrimmed.fastq                VA_B_01_14_clippedtrimmed.fastq.bam.bai
RI_B_01_14_clippedtrimmed.fastq.bam            VA_B_01_14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.bam.bai        VA_B_01_14_clippedtrimmed.fastq_UNSORTED.bam
RI_B_01_14_clippedtrimmed.fastq.sam            VA_B_01_18_clippedtrimmed.fastq
RI_B_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_18_clippedtrimmed.fastq.bam
RI_B_01_18_clippedtrimmed.fastq                VA_B_01_18_clippedtrimmed.fastq.bam.bai
RI_B_01_18_clippedtrimmed.fastq.bam            VA_B_01_18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.bam.bai        VA_B_01_18_clippedtrimmed.fastq_UNSORTED.bam
RI_B_01_18_clippedtrimmed.fastq.sam            VA_B_01_22_clippedtrimmed.fastq
RI_B_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_22_clippedtrimmed.fastq.bam
RI_B_01_22_clippedtrimmed.fastq                VA_B_01_22_clippedtrimmed.fastq.bam.bai
RI_B_01_22_clippedtrimmed.fastq.bam            VA_B_01_22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.bam.bai        VA_B_01_22_clippedtrimmed.fastq_UNSORTED.bam
RI_B_01_22_clippedtrimmed.fastq.sam            VA_B_09_SNP_clippedtrimmed.fastq
RI_B_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_B_09_SNP_clippedtrimmed.fastq.bam
RI_B_08_SNP_clippedtrimmed.fastq               VA_B_09_SNP_clippedtrimmed.fastq.bam.bai
RI_B_08_SNP_clippedtrimmed.fastq.bam           VA_B_09_SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.bam.bai       VA_B_09_SNP_clippedtrimmed.fastq_UNSORTED.bam
RI_B_08_SNP_clippedtrimmed.fastq.sam           VA_W_01_14_clippedtrimmed.fastq
RI_B_08_SNP_clippedtrimmed.fastq_UNSORTED.bam  VA_W_01_14_clippedtrimmed.fastq.bam
RI_W_01_14_clippedtrimmed.fastq                VA_W_01_14_clippedtrimmed.fastq.bam.bai
RI_W_01_14_clippedtrimmed.fastq.bam            VA_W_01_14_clippedtrimmed.fastq.sam
RI_W_01_14_clippedtrimmed.fastq.bam.bai        VA_W_01_14_clippedtrimmed.fastq_UNSORTED.bam
RI_W_01_14_clippedtrimmed.fastq.sam            VA_W_01_18_clippedtrimmed.fastq
RI_W_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_18_clippedtrimmed.fastq.bam
RI_W_01_18_clippedtrimmed.fastq                VA_W_01_18_clippedtrimmed.fastq.bam.bai
RI_W_01_18_clippedtrimmed.fastq.bam            VA_W_01_18_clippedtrimmed.fastq.sam
RI_W_01_18_clippedtrimmed.fastq.bam.bai        VA_W_01_18_clippedtrimmed.fastq_UNSORTED.bam
RI_W_01_18_clippedtrimmed.fastq.sam            VA_W_01_22_clippedtrimmed.fastq
RI_W_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_22_clippedtrimmed.fastq.bam
RI_W_01_22_clippedtrimmed.fastq                VA_W_01_22_clippedtrimmed.fastq.bam.bai
RI_W_01_22_clippedtrimmed.fastq.bam            VA_W_01_22_clippedtrimmed.fastq.sam
RI_W_01_22_clippedtrimmed.fastq.bam.bai        VA_W_01_22_clippedtrimmed.fastq_UNSORTED.bam
RI_W_01_22_clippedtrimmed.fastq.sam            VA_W_08_SNP_clippedtrimmed.fastq
RI_W_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_W_08_SNP_clippedtrimmed.fastq.bam
RI_W_08_SNP_clippedtrimmed.fastq               VA_W_08_SNP_clippedtrimmed.fastq.bam.bai
RI_W_08_SNP_clippedtrimmed.fastq.bam           VA_W_08_SNP_clippedtrimmed.fastq.sam
RI_W_08_SNP_clippedtrimmed.fastq.bam.bai       VA_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam
RI_W_08_SNP_clippedtrimmed.fastq.sam
[apear012@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9276491
salloc: job 9276491 queued and waiting for resources
salloc: job 9276491 has been allocated resources
salloc: Granted job allocation 9276491
[apear012@coreV3-23-046 QCFastqs]$ rm *UNSORTED.bam
[apear012@coreV3-23-046 QCFastqs]$ ls
alignstatsAP.txt                          RI_W_08_SNP_clippedtrimmed.fastq.bam.bai
apam                                      RI_W_08_SNP_clippedtrimmed.fastq.sam
APbowtie.sh                               *stats.txt.bam
APbowtie.txt                              *stats.txt.sam
AP_Sort.sh                                VA_B_01_14_clippedtrimmed.fastq
AP_Sort.txt                               VA_B_01_14_clippedtrimmed.fastq.bam
RI_B_01_14_clippedtrimmed.fastq           VA_B_01_14_clippedtrimmed.fastq.bam.bai
RI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq
RI_B_01_14_clippedtrimmed.fastq.sam       VA_B_01_18_clippedtrimmed.fastq.bam
RI_B_01_18_clippedtrimmed.fastq           VA_B_01_18_clippedtrimmed.fastq.bam.bai
RI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq
RI_B_01_18_clippedtrimmed.fastq.sam       VA_B_01_22_clippedtrimmed.fastq.bam
RI_B_01_22_clippedtrimmed.fastq           VA_B_01_22_clippedtrimmed.fastq.bam.bai
RI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_09_SNP_clippedtrimmed.fastq
RI_B_01_22_clippedtrimmed.fastq.sam       VA_B_09_SNP_clippedtrimmed.fastq.bam
RI_B_08_SNP_clippedtrimmed.fastq          VA_B_09_SNP_clippedtrimmed.fastq.bam.bai
RI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_01_14_clippedtrimmed.fastq
RI_B_08_SNP_clippedtrimmed.fastq.sam      VA_W_01_14_clippedtrimmed.fastq.bam
RI_W_01_14_clippedtrimmed.fastq           VA_W_01_14_clippedtrimmed.fastq.bam.bai
RI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.sam
RI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq
RI_W_01_14_clippedtrimmed.fastq.sam       VA_W_01_18_clippedtrimmed.fastq.bam
RI_W_01_18_clippedtrimmed.fastq           VA_W_01_18_clippedtrimmed.fastq.bam.bai
RI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.sam
RI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq
RI_W_01_18_clippedtrimmed.fastq.sam       VA_W_01_22_clippedtrimmed.fastq.bam
RI_W_01_22_clippedtrimmed.fastq           VA_W_01_22_clippedtrimmed.fastq.bam.bai
RI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.sam
RI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_08_SNP_clippedtrimmed.fastq
RI_W_01_22_clippedtrimmed.fastq.sam       VA_W_08_SNP_clippedtrimmed.fastq.bam
RI_W_08_SNP_clippedtrimmed.fastq          VA_W_08_SNP_clippedtrimmed.fastq.bam.bai
RI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.sam">
                     <pre class="notranslate"><code>[apear012@coreV3-23-046 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ ls
alignstatsAP.txt                               RI_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam
apam                                           *stats.txt.bam
APbowtie.sh                                    *stats.txt.sam
APbowtie.txt                                   *stats.txt_UNSORTED.bam
AP_Sort.sh                                     VA_B_01_14_clippedtrimmed.fastq
AP_Sort.txt                                    VA_B_01_14_clippedtrimmed.fastq.bam
RI_B_01_14_clippedtrimmed.fastq                VA_B_01_14_clippedtrimmed.fastq.bam.bai
RI_B_01_14_clippedtrimmed.fastq.bam            VA_B_01_14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.bam.bai        VA_B_01_14_clippedtrimmed.fastq_UNSORTED.bam
RI_B_01_14_clippedtrimmed.fastq.sam            VA_B_01_18_clippedtrimmed.fastq
RI_B_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_18_clippedtrimmed.fastq.bam
RI_B_01_18_clippedtrimmed.fastq                VA_B_01_18_clippedtrimmed.fastq.bam.bai
RI_B_01_18_clippedtrimmed.fastq.bam            VA_B_01_18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.bam.bai        VA_B_01_18_clippedtrimmed.fastq_UNSORTED.bam
RI_B_01_18_clippedtrimmed.fastq.sam            VA_B_01_22_clippedtrimmed.fastq
RI_B_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_B_01_22_clippedtrimmed.fastq.bam
RI_B_01_22_clippedtrimmed.fastq                VA_B_01_22_clippedtrimmed.fastq.bam.bai
RI_B_01_22_clippedtrimmed.fastq.bam            VA_B_01_22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.bam.bai        VA_B_01_22_clippedtrimmed.fastq_UNSORTED.bam
RI_B_01_22_clippedtrimmed.fastq.sam            VA_B_09_SNP_clippedtrimmed.fastq
RI_B_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_B_09_SNP_clippedtrimmed.fastq.bam
RI_B_08_SNP_clippedtrimmed.fastq               VA_B_09_SNP_clippedtrimmed.fastq.bam.bai
RI_B_08_SNP_clippedtrimmed.fastq.bam           VA_B_09_SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.bam.bai       VA_B_09_SNP_clippedtrimmed.fastq_UNSORTED.bam
RI_B_08_SNP_clippedtrimmed.fastq.sam           VA_W_01_14_clippedtrimmed.fastq
RI_B_08_SNP_clippedtrimmed.fastq_UNSORTED.bam  VA_W_01_14_clippedtrimmed.fastq.bam
RI_W_01_14_clippedtrimmed.fastq                VA_W_01_14_clippedtrimmed.fastq.bam.bai
RI_W_01_14_clippedtrimmed.fastq.bam            VA_W_01_14_clippedtrimmed.fastq.sam
RI_W_01_14_clippedtrimmed.fastq.bam.bai        VA_W_01_14_clippedtrimmed.fastq_UNSORTED.bam
RI_W_01_14_clippedtrimmed.fastq.sam            VA_W_01_18_clippedtrimmed.fastq
RI_W_01_14_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_18_clippedtrimmed.fastq.bam
RI_W_01_18_clippedtrimmed.fastq                VA_W_01_18_clippedtrimmed.fastq.bam.bai
RI_W_01_18_clippedtrimmed.fastq.bam            VA_W_01_18_clippedtrimmed.fastq.sam
RI_W_01_18_clippedtrimmed.fastq.bam.bai        VA_W_01_18_clippedtrimmed.fastq_UNSORTED.bam
RI_W_01_18_clippedtrimmed.fastq.sam            VA_W_01_22_clippedtrimmed.fastq
RI_W_01_18_clippedtrimmed.fastq_UNSORTED.bam   VA_W_01_22_clippedtrimmed.fastq.bam
RI_W_01_22_clippedtrimmed.fastq                VA_W_01_22_clippedtrimmed.fastq.bam.bai
RI_W_01_22_clippedtrimmed.fastq.bam            VA_W_01_22_clippedtrimmed.fastq.sam
RI_W_01_22_clippedtrimmed.fastq.bam.bai        VA_W_01_22_clippedtrimmed.fastq_UNSORTED.bam
RI_W_01_22_clippedtrimmed.fastq.sam            VA_W_08_SNP_clippedtrimmed.fastq
RI_W_01_22_clippedtrimmed.fastq_UNSORTED.bam   VA_W_08_SNP_clippedtrimmed.fastq.bam
RI_W_08_SNP_clippedtrimmed.fastq               VA_W_08_SNP_clippedtrimmed.fastq.bam.bai
RI_W_08_SNP_clippedtrimmed.fastq.bam           VA_W_08_SNP_clippedtrimmed.fastq.sam
RI_W_08_SNP_clippedtrimmed.fastq.bam.bai       VA_W_08_SNP_clippedtrimmed.fastq_UNSORTED.bam
RI_W_08_SNP_clippedtrimmed.fastq.sam
[apear012@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9276491
salloc: job 9276491 queued and waiting for resources
salloc: job 9276491 has been allocated resources
salloc: Granted job allocation 9276491
[apear012@coreV3-23-046 QCFastqs]$ rm *UNSORTED.bam
[apear012@coreV3-23-046 QCFastqs]$ ls
alignstatsAP.txt                          RI_W_08_SNP_clippedtrimmed.fastq.bam.bai
apam                                      RI_W_08_SNP_clippedtrimmed.fastq.sam
APbowtie.sh                               *stats.txt.bam
APbowtie.txt                              *stats.txt.sam
AP_Sort.sh                                VA_B_01_14_clippedtrimmed.fastq
AP_Sort.txt                               VA_B_01_14_clippedtrimmed.fastq.bam
RI_B_01_14_clippedtrimmed.fastq           VA_B_01_14_clippedtrimmed.fastq.bam.bai
RI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq
RI_B_01_14_clippedtrimmed.fastq.sam       VA_B_01_18_clippedtrimmed.fastq.bam
RI_B_01_18_clippedtrimmed.fastq           VA_B_01_18_clippedtrimmed.fastq.bam.bai
RI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq
RI_B_01_18_clippedtrimmed.fastq.sam       VA_B_01_22_clippedtrimmed.fastq.bam
RI_B_01_22_clippedtrimmed.fastq           VA_B_01_22_clippedtrimmed.fastq.bam.bai
RI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_09_SNP_clippedtrimmed.fastq
RI_B_01_22_clippedtrimmed.fastq.sam       VA_B_09_SNP_clippedtrimmed.fastq.bam
RI_B_08_SNP_clippedtrimmed.fastq          VA_B_09_SNP_clippedtrimmed.fastq.bam.bai
RI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_01_14_clippedtrimmed.fastq
RI_B_08_SNP_clippedtrimmed.fastq.sam      VA_W_01_14_clippedtrimmed.fastq.bam
RI_W_01_14_clippedtrimmed.fastq           VA_W_01_14_clippedtrimmed.fastq.bam.bai
RI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.sam
RI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq
RI_W_01_14_clippedtrimmed.fastq.sam       VA_W_01_18_clippedtrimmed.fastq.bam
RI_W_01_18_clippedtrimmed.fastq           VA_W_01_18_clippedtrimmed.fastq.bam.bai
RI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.sam
RI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq
RI_W_01_18_clippedtrimmed.fastq.sam       VA_W_01_22_clippedtrimmed.fastq.bam
RI_W_01_22_clippedtrimmed.fastq           VA_W_01_22_clippedtrimmed.fastq.bam.bai
RI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.sam
RI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_08_SNP_clippedtrimmed.fastq
RI_W_01_22_clippedtrimmed.fastq.sam       VA_W_08_SNP_clippedtrimmed.fastq.bam
RI_W_08_SNP_clippedtrimmed.fastq          VA_W_08_SNP_clippedtrimmed.fastq.bam.bai
RI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.sam
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-046 QCFastqs]$ enable lmod
enable: Command not found.
[apear012@coreV3-23-046 QCFastqs]$ enable_lmod
[apear012@coreV3-23-046 QCFastqs]$ module load samtools
[apear012@coreV3-23-046 QCFastqs]$ samtools tview RI_B_01_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
1         11        21        31        41        51        61        71        
TCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT
                                                        ........................
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,">
                     <pre class="notranslate"><code>[apear012@coreV3-23-046 QCFastqs]$ enable lmod
enable: Command not found.
[apear012@coreV3-23-046 QCFastqs]$ enable_lmod
[apear012@coreV3-23-046 QCFastqs]$ module load samtools
[apear012@coreV3-23-046 QCFastqs]$ samtools tview RI_B_01_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
1         11        21        31        41        51        61        71        
TCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT
                                                        ........................
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,
                                                        ,,,,,,,,,,,,,,,,,,,,,,,,
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Run the following to start genotyping your SNPs for filtering next class
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o OUTFILENAME.txt
#SBATCH -n 1         
#SBATCH --mail-user=EMAIL
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam &gt; YOURNAMEmergedfastqs.vcf">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o OUTFILENAME.txt
#SBATCH -n 1         
#SBATCH --mail-user=EMAIL
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam &gt; YOURNAMEmergedfastqs.vcf
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ nano AP_Freebayes.sh">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ nano AP_Freebayes.sh
</code></pre>
                    </div>
                    <p dir="auto">
                     AP_Freebayes.sh:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o AP_Freebayes.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Freebayes

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam &gt; APmergedfastqs.vcf">
                     <pre class="notranslate"><code>#!/bin/bash -l

#SBATCH -o AP_Freebayes.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_Freebayes

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam &gt; APmergedfastqs.vcf
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ sbatch AP_Freebayes.sh 
Submitted batch job 9276576
[apear012@turing1 QCFastqs]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9276576      main AP_Freeb apear012  R       0:02      1 coreV3-23-026 
           9276491      main       sh apear012  R    1:11:32      1 coreV3-23-046 ">
                     <pre class="notranslate"><code>[apear012@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@turing1 QCFastqs]$ sbatch AP_Freebayes.sh 
Submitted batch job 9276576
[apear012@turing1 QCFastqs]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9276576      main AP_Freeb apear012  R       0:02      1 coreV3-23-026 
           9276491      main       sh apear012  R    1:11:32      1 coreV3-23-046 
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 08 12-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 08 12-Feb-2021" class="anchor" href="#day-08-12-feb-2021" id="user-content-day-08-12-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       Clean up your data directory by:
                      </p>
                      <p dir="auto">
                       -Making a SAMS folder and mv all your .sam files into that directory
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@turing1 data]$ mkdir SAMS
[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.sam SAMS">
                     <pre class="notranslate"><code>[apear012@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@turing1 data]$ mkdir SAMS
[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.sam SAMS
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory">
                     <pre class="notranslate"><code>-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 data]$ mkdir BAMS
[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.bam BAMS
[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.bai BAMS
[apear012@coreV3-23-040 data]$ ls BAMS
RI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.bam
RI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_14_clippedtrimmed.fastq.bam.bai
RI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.bam
RI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq.bam.bai
RI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.bam
RI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq.bam.bai
RI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.bam
RI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_B_09_SNP_clippedtrimmed.fastq.bam.bai
RI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.bam
RI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_14_clippedtrimmed.fastq.bam.bai
RI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.bam
RI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq.bam.bai
RI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.bam
RI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq.bam.bai
RI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.bam
RI_W_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_08_SNP_clippedtrimmed.fastq.bam.bai
*stats.txt.bam">
                     <pre class="notranslate"><code>[apear012@turing1 data]$ mkdir BAMS
[apear012@turing1 data]$ mv ./fastq/QCFastqs/*.bam BAMS
[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.bai BAMS
[apear012@coreV3-23-040 data]$ ls BAMS
RI_B_01_14_clippedtrimmed.fastq.bam       VA_B_01_14_clippedtrimmed.fastq.bam
RI_B_01_14_clippedtrimmed.fastq.bam.bai   VA_B_01_14_clippedtrimmed.fastq.bam.bai
RI_B_01_18_clippedtrimmed.fastq.bam       VA_B_01_18_clippedtrimmed.fastq.bam
RI_B_01_18_clippedtrimmed.fastq.bam.bai   VA_B_01_18_clippedtrimmed.fastq.bam.bai
RI_B_01_22_clippedtrimmed.fastq.bam       VA_B_01_22_clippedtrimmed.fastq.bam
RI_B_01_22_clippedtrimmed.fastq.bam.bai   VA_B_01_22_clippedtrimmed.fastq.bam.bai
RI_B_08_SNP_clippedtrimmed.fastq.bam      VA_B_09_SNP_clippedtrimmed.fastq.bam
RI_B_08_SNP_clippedtrimmed.fastq.bam.bai  VA_B_09_SNP_clippedtrimmed.fastq.bam.bai
RI_W_01_14_clippedtrimmed.fastq.bam       VA_W_01_14_clippedtrimmed.fastq.bam
RI_W_01_14_clippedtrimmed.fastq.bam.bai   VA_W_01_14_clippedtrimmed.fastq.bam.bai
RI_W_01_18_clippedtrimmed.fastq.bam       VA_W_01_18_clippedtrimmed.fastq.bam
RI_W_01_18_clippedtrimmed.fastq.bam.bai   VA_W_01_18_clippedtrimmed.fastq.bam.bai
RI_W_01_22_clippedtrimmed.fastq.bam       VA_W_01_22_clippedtrimmed.fastq.bam
RI_W_01_22_clippedtrimmed.fastq.bam.bai   VA_W_01_22_clippedtrimmed.fastq.bam.bai
RI_W_08_SNP_clippedtrimmed.fastq.bam      VA_W_08_SNP_clippedtrimmed.fastq.bam
RI_W_08_SNP_clippedtrimmed.fastq.bam.bai  VA_W_08_SNP_clippedtrimmed.fastq.bam.bai
*stats.txt.bam
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="-rm your _unfiltered.vcf files if you have any">
                     <pre class="notranslate"><code>-rm your _unfiltered.vcf files if you have any
</code></pre>
                    </div>
                    <p dir="auto">
                     I didn't have any.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="-rm your .fastq files">
                     <pre class="notranslate"><code>-rm your .fastq files
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-040 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@coreV3-23-040 QCFastqs]$ rm *.fastq">
                     <pre class="notranslate"><code>[apear012@coreV3-23-040 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs
[apear012@coreV3-23-040 QCFastqs]$ rm *.fastq
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="-Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)">
                     <pre class="notranslate"><code>-Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-040 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@coreV3-23-040 data]$ mkdir VCF
[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.vcf VCF">
                     <pre class="notranslate"><code>[apear012@coreV3-23-040 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data
[apear012@coreV3-23-040 data]$ mkdir VCF
[apear012@coreV3-23-040 data]$ mv ./fastq/QCFastqs/*.vcf VCF
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      Start an interactive session via salloc
                     </li>
                    </ol>
                    <p dir="auto">
                     Did this during previous steps.
                    </p>
                    <ol dir="auto" start="3">
                     <li>
                      cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-040 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF
[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf .
[apear012@coreV3-23-040 VCF]$ ls
APmergedfastqs.vcf  mergedfastq_HEAAstrangiaAssembly.vcf">
                     <pre class="notranslate"><code>[apear012@coreV3-23-040 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF
[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf .
[apear012@coreV3-23-040 VCF]$ ls
APmergedfastqs.vcf  mergedfastq_HEAAstrangiaAssembly.vcf
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-040 VCF]$ vcftools --vcf APmergedfastqs.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf APmergedfastqs.vcf

After filtering, kept 16 out of 16 Individuals
After filtering, kept 148757 out of a possible 148757 Sites
Run Time = 1.00 seconds">
                     <pre class="notranslate"><code>[apear012@coreV3-23-040 VCF]$ vcftools --vcf APmergedfastqs.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf APmergedfastqs.vcf

After filtering, kept 16 out of 16 Individuals
After filtering, kept 148757 out of a possible 148757 Sites
Run Time = 1.00 seconds
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt .
[apear012@coreV3-23-040 VCF]$ ls
APmergedfastqs.vcf  GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf  out.log">
                     <pre class="notranslate"><code>[apear012@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt .
[apear012@coreV3-23-040 VCF]$ ls
APmergedfastqs.vcf  GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf  out.log
</code></pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      Run our host vcf extractor on your merged vcf file using the following syntax:
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf">
                     <pre class="notranslate"><code>/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-026 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf">
                     <pre class="notranslate"><code>[apear012@coreV3-23-026 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
</code></pre>
                    </div>
                    <ol dir="auto" start="7">
                     <li>
                      Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 VCF]$ ls
APmergedfastqs.vcf                    mergedfastq_HEAAstrangiaAssembly_subset.vcf  out.log
GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf
[apear012@turing1 VCF]$ vcftools --vcf mer
mergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*
[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 432676 out of a possible 432676 Sites
Run Time = 6.00 seconds
[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 11.00 seconds
[apear012@turing1 VCF]$ vcftools --vcf APmergedfastqs.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf APmergedfastqs.vcf

After filtering, kept 16 out of 16 Individuals
After filtering, kept 148757 out of a possible 148757 Sites
Run Time = 2.00 seconds">
                     <pre class="notranslate"><code>[apear012@turing1 VCF]$ ls
APmergedfastqs.vcf                    mergedfastq_HEAAstrangiaAssembly_subset.vcf  out.log
GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf
[apear012@turing1 VCF]$ vcftools --vcf mer
mergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*
[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 432676 out of a possible 432676 Sites
Run Time = 6.00 seconds
[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 11.00 seconds
[apear012@turing1 VCF]$ vcftools --vcf APmergedfastqs.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf APmergedfastqs.vcf

After filtering, kept 16 out of 16 Individuals
After filtering, kept 148757 out of a possible 148757 Sites
Run Time = 2.00 seconds
</code></pre>
                    </div>
                    <ol dir="auto" start="8">
                     <li>
                      Work through the
                      <a href="http://www.ddocent.com/filtering/" rel="nofollow">
                       VCF filtering tutorial
                      </a>
                      until the following step:
                     </li>
                    </ol>
                    <p dir="auto">
                     Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.
                    </p>
                    <p dir="auto">
                     vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 VCF]$ vcftools --vcf mer --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3
mergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*
[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
	--recode-INFO-all
	--mac 3
	--minQ 30
	--max-missing 0.5
	--out raw.g5mac3
	--recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 432676 Sites
Run Time = 33.00 seconds
[apear012@turing1 VCF]$ ls
APmergedfastqs.vcf                           out.log
GoodCoralGenelistForVCFSubsetter.txt         raw.g5mac3.log
mergedfastq_HEAAstrangiaAssembly_subset.vcf  raw.g5mac3.recode.vcf
mergedfastq_HEAAstrangiaAssembly.vcf
[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf raw.g5mac3.recode.vcf
	--recode-INFO-all
	--minDP 3
	--out raw.g5mac3dp3
	--recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 30.00 seconds
[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf raw.g5mac3dp3.recode.vcf
	--missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 3.00 seconds
[apear012@turing1 VCF]$ bash
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 &gt; totalmissing
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ gnuplot &lt;&lt; \EOF 
&gt; set terminal dumb size 120, 30
&gt; set autoscale 
&gt; unset label
&gt; set title &quot;Histogram of % missing data per individual&quot;
&gt; set ylabel &quot;Number of Occurrences&quot;
&gt; set xlabel &quot;% of missing data&quot;
&gt; #set yr [0:100000]
&gt; binwidth=0.01
&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0
&gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes
&gt; pause -1
&gt; EOF

                                         Histogram of % missing data per individual
  Number of Occurrences
      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++
        +           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
    2.5 ++                                                               * *                                        ++
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
      2 ++                                                       **      * * ***  ******    ** ****   ****          ++
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +
      1 *********************************************************************************************************---++
       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1
                                                      % of missing data

apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ exit
exit">
                     <pre class="notranslate"><code>[apear012@turing1 VCF]$ vcftools --vcf mer --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3
mergedfastq_HEAAstrangiaAssembly_subset.vcf* mergedfastq_HEAAstrangiaAssembly.vcf*
[apear012@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
	--recode-INFO-all
	--mac 3
	--minQ 30
	--max-missing 0.5
	--out raw.g5mac3
	--recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 432676 Sites
Run Time = 33.00 seconds
[apear012@turing1 VCF]$ ls
APmergedfastqs.vcf                           out.log
GoodCoralGenelistForVCFSubsetter.txt         raw.g5mac3.log
mergedfastq_HEAAstrangiaAssembly_subset.vcf  raw.g5mac3.recode.vcf
mergedfastq_HEAAstrangiaAssembly.vcf
[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf raw.g5mac3.recode.vcf
	--recode-INFO-all
	--minDP 3
	--out raw.g5mac3dp3
	--recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 30.00 seconds
[apear012@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf raw.g5mac3dp3.recode.vcf
	--missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 3.00 seconds
[apear012@turing1 VCF]$ bash
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 &gt; totalmissing
apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ gnuplot &lt;&lt; \EOF 
&gt; set terminal dumb size 120, 30
&gt; set autoscale 
&gt; unset label
&gt; set title "Histogram of % missing data per individual"
&gt; set ylabel "Number of Occurrences"
&gt; set xlabel "% of missing data"
&gt; #set yr [0:100000]
&gt; binwidth=0.01
&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0
&gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes
&gt; pause -1
&gt; EOF

                                         Histogram of % missing data per individual
  Number of Occurrences
      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++
        +           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
    2.5 ++                                                               * *                                        ++
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
      2 ++                                                       **      * * ***  ******    ** ****   ****          ++
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +
      1 *********************************************************************************************************---++
       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1
                                                      % of missing data

apear012@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF$ exit
exit
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 09 17-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 09 17-Feb-2021" class="anchor" href="#day-09-17-feb-2021" id="user-content-day-09-17-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters
##Half the class run the following (I was instructed to do this one)
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
##the other half run the filters we used from the paper
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-015 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF
[apear012@coreV3-23-015 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
	--recode-INFO-all
	--maf 0.015
	--max-alleles 2
	--minDP 3
	--minGQ 20
	--hwe 0.01
	--minQ 30
	--max-missing 0.5
	--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
	--recode
	--remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 18718 out of a possible 432676 Sites
Run Time = 10.00 seconds">
                     <pre class="notranslate"><code>[apear012@coreV3-23-015 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF
[apear012@coreV3-23-015 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
	--recode-INFO-all
	--maf 0.015
	--max-alleles 2
	--minDP 3
	--minGQ 20
	--hwe 0.01
	--minQ 30
	--max-missing 0.5
	--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
	--recode
	--remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 18718 out of a possible 432676 Sites
Run Time = 10.00 seconds
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV3-23-015 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf &gt; samplenames.txt
[apear012@coreV3-23-015 VCF]$ cat samplenames.txt 
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_merged	VA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_mergedRI_W_05_merged	RI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged">
                     <pre class="notranslate"><code>[apear012@coreV3-23-015 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf &gt; samplenames.txt
[apear012@coreV3-23-015 VCF]$ cat samplenames.txt 
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_merged	VA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_mergedRI_W_05_merged	RI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
</code></pre>
                    </div>
                    <p dir="auto">
                     Copied the contents of samplenames.txt into BBedit and used regular expressions in find and replace as such:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="FIND: (\w\w_\w)(_.*)
REPLACE: \1\2\t\1">
                     <pre class="notranslate"><code>FIND: (\w\w_\w)(_.*)
REPLACE: \1\2\t\1
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-22-030 VCF]$ nano popfile.txt">
                     <pre class="notranslate"><code>[apear012@coreV2-22-030 VCF]$ nano popfile.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     popfile.txt:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="RI_W_06_merged	RI_W
RI_W_07_merged	RI_W
VA_B_03_merged	VA_B
RI_W_02_merged	RI_W
RI_W_04_merged	RI_W
VA_W_09_SNP_clipped	VA_W
RI_B_08_SNP_clipped	RI_B
VA_W_08_SNP_clipped	VA_W
VA_B_08_SNP_clipped	VA_B
VA_W_02_merged	VA_W
VA_B_07_merged	VA_B
RI_B_05_merged	RI_B
VA_W_06_merged	VA_W
VA_W_04_merged	VA_W
VA_W_01_merged	VA_W
VA_B_10_SNP_clipped	VA_B
VA_B_06_merged	VA_B
VA_W_05_merged	VA_W
RI_B_09_SNP_clipped	RI_B
VA_W_10_SNP_clipped	VA_W
RI_W_08_SNP_clipped	RI_W
RI_B_06_merged	RI_B
RI_W_10_SNP_clipped	RI_W
RI_B_04_merged	RI_B
VA_W_03_merged	VA_W
RI_B_07_merged	RI_B
RI_W_05_merged	RI_W
RI_W_09_SNP_clipped	RI_W
VA_B_01_merged	VA_B
VA_B_09_SNP_clipped	VA_B
RI_B_10_SNP_clipped	RI_B
RI_W_01_merged	RI_W
RI_B_01_merged	RI_B
VA_B_04_merged	VA_B
RI_B_02_merged	RI_B
RI_W_03_merged	RI_W
VA_B_02_merged	VA_B
VA_W_07_merged	VA_W
VA_B_05_merged	VA_B
RI_B_03_merged	RI_B">
                     <pre class="notranslate"><code>RI_W_06_merged	RI_W
RI_W_07_merged	RI_W
VA_B_03_merged	VA_B
RI_W_02_merged	RI_W
RI_W_04_merged	RI_W
VA_W_09_SNP_clipped	VA_W
RI_B_08_SNP_clipped	RI_B
VA_W_08_SNP_clipped	VA_W
VA_B_08_SNP_clipped	VA_B
VA_W_02_merged	VA_W
VA_B_07_merged	VA_B
RI_B_05_merged	RI_B
VA_W_06_merged	VA_W
VA_W_04_merged	VA_W
VA_W_01_merged	VA_W
VA_B_10_SNP_clipped	VA_B
VA_B_06_merged	VA_B
VA_W_05_merged	VA_W
RI_B_09_SNP_clipped	RI_B
VA_W_10_SNP_clipped	VA_W
RI_W_08_SNP_clipped	RI_W
RI_B_06_merged	RI_B
RI_W_10_SNP_clipped	RI_W
RI_B_04_merged	RI_B
VA_W_03_merged	VA_W
RI_B_07_merged	RI_B
RI_W_05_merged	RI_W
RI_W_09_SNP_clipped	RI_W
VA_B_01_merged	VA_B
VA_B_09_SNP_clipped	VA_B
RI_B_10_SNP_clipped	RI_B
RI_W_01_merged	RI_W
RI_B_01_merged	RI_B
VA_B_04_merged	VA_B
RI_B_02_merged	RI_B
RI_W_03_merged	RI_W
VA_B_02_merged	VA_B
VA_W_07_merged	VA_W
VA_B_05_merged	VA_B
RI_B_03_merged	RI_B
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Convert your filtered .vcf file to genepop format using the following command:
                     </li>
                    </ol>
                    <p dir="auto">
                     /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-22-030 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt 
Indivs with genotypes in vcf file: RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_merged	VA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_merged	RI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
44 18718 18718 18718 18718 40">
                     <pre class="notranslate"><code>[apear012@coreV2-22-030 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt 
Indivs with genotypes in vcf file: RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_merged	VA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_merged	RI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
44 18718 18718 18718 18718 40
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="(base) Andrews-MacBook-Air:day09 AndrewsComputer$ pwd
/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/day09
(base) Andrews-MacBook-Air:day09 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen .
apear012@turing.hpc.odu.edu's password: 
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.ge 100% 4062KB 180.0KB/s   00:22    
(base) Andrews-MacBook-Air:day09 AndrewsComputer$ ls
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen
adegenet_PCAs.R
adegenet_PCAs_HEAOutliers.R
homework_day09.txt">
                     <pre class="notranslate"><code>(base) Andrews-MacBook-Air:day09 AndrewsComputer$ pwd
/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/day09
(base) Andrews-MacBook-Air:day09 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen .
apear012@turing.hpc.odu.edu's password: 
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.ge 100% 4062KB 180.0KB/s   00:22    
(base) Andrews-MacBook-Air:day09 AndrewsComputer$ ls
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen
adegenet_PCAs.R
adegenet_PCAs_HEAOutliers.R
homework_day09.txt
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.
                     </li>
                    </ol>
                    <p dir="auto">
                     I did this. I had to import a number of dependencies for the R package 'adegenet' manually to get this to work. Specifically, the R package 'sf', a dependancy of 'adegent', gave me trouble and I had to go to download it in Chrome and manually add the package from the file browser.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 10 19-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 10 19-Feb-2021" class="anchor" href="#day-10-19-feb-2021" id="user-content-day-10-19-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.
                     </li>
                    </ol>
                    <p dir="auto">
                     I did this.
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 SAMS]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS
[apear012@coreV2-25-017 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam">
                     <pre class="notranslate"><code>[apear012@turing1 SAMS]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS
[apear012@coreV2-25-017 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt YOURNAMEFullCounts_summed.txt NoMatch *_counts.txt
                     </li>
                    </ol>
                    <p dir="auto">
                     (I also had to rename the match_counts.txt file to matchcounts.txt so this command would not include it.)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@turing1 SAMS]$ mv match_counts.txt matchcounts.txt
[apear012@turing1 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt APFullCounts_summed.txt NoMatch *_counts.txt
Hits not matchedRI_B_01_14_clippedtrimmed.fastq_counts.txt=1698	RI_B_01_18_clippedtrimmed.fastq_counts.txt=169RI_B_01_22_clippedtrimmed.fastq_counts.txt=1698	RI_B_08_SNP_clippedtrimmed.fastq_counts.txt=1698	RI_W_01_14_clippedtrimmed.fastq_counts.txt=1698	RI_W_01_18_clippedtrimmed.fastq_counts.txt=1698	RI_W_01_22_clippedtrimmed.fastq_counts.txt=1698	RI_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698	VA_B_01_14_clippedtrimmed.fastq_counts.txt=1698	VA_B_01_18_clippedtrimmed.fastq_counts.txt=1698	VA_B_01_22_clippedtrimmed.fastq_counts.txt=1698	VA_B_09_SNP_clippedtrimmed.fastq_counts.txt=1698	VA_W_01_14_clippedtrimmed.fastq_counts.txt=1698	VA_W_01_18_clippedtrimmed.fastq_counts.txt=1698	VA_W_01_22_clippedtrimmed.fastq_counts.txt=1698	VA_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698
[apear012@turing1 SAMS]$ ls
APFullCounts_summed.txt                      stats.txt
matchcounts.txt                              VA_B_01_14_clippedtrimmed.fastq_counts.txt
RI_B_01_14_clippedtrimmed.fastq_counts.txt   VA_B_01_14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.sam          VA_B_01_18_clippedtrimmed.fastq_counts.txt
RI_B_01_18_clippedtrimmed.fastq_counts.txt   VA_B_01_18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.sam          VA_B_01_22_clippedtrimmed.fastq_counts.txt
RI_B_01_22_clippedtrimmed.fastq_counts.txt   VA_B_01_22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.sam          VA_B_09_SNP_clippedtrimmed.fastq_counts.txt
RI_B_08_SNP_clippedtrimmed.fastq_counts.txt  VA_B_09_SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.sam         VA_W_01_14_clippedtrimmed.fastq_counts.txt
RI_W_01_14_clippedtrimmed.fastq_counts.txt   VA_W_01_14_clippedtrimmed.fastq.sam
RI_W_01_14_clippedtrimmed.fastq.sam          VA_W_01_18_clippedtrimmed.fastq_counts.txt
RI_W_01_18_clippedtrimmed.fastq_counts.txt   VA_W_01_18_clippedtrimmed.fastq.sam
RI_W_01_18_clippedtrimmed.fastq.sam          VA_W_01_22_clippedtrimmed.fastq_counts.txt
RI_W_01_22_clippedtrimmed.fastq_counts.txt   VA_W_01_22_clippedtrimmed.fastq.sam
RI_W_01_22_clippedtrimmed.fastq.sam          VA_W_08_SNP_clippedtrimmed.fastq_counts.txt
RI_W_08_SNP_clippedtrimmed.fastq_counts.txt  VA_W_08_SNP_clippedtrimmed.fastq.sam
RI_W_08_SNP_clippedtrimmed.fastq.sam">
                     <pre class="notranslate"><code>[apear012@turing1 SAMS]$ mv match_counts.txt matchcounts.txt
[apear012@turing1 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt APFullCounts_summed.txt NoMatch *_counts.txt
Hits not matchedRI_B_01_14_clippedtrimmed.fastq_counts.txt=1698	RI_B_01_18_clippedtrimmed.fastq_counts.txt=169RI_B_01_22_clippedtrimmed.fastq_counts.txt=1698	RI_B_08_SNP_clippedtrimmed.fastq_counts.txt=1698	RI_W_01_14_clippedtrimmed.fastq_counts.txt=1698	RI_W_01_18_clippedtrimmed.fastq_counts.txt=1698	RI_W_01_22_clippedtrimmed.fastq_counts.txt=1698	RI_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698	VA_B_01_14_clippedtrimmed.fastq_counts.txt=1698	VA_B_01_18_clippedtrimmed.fastq_counts.txt=1698	VA_B_01_22_clippedtrimmed.fastq_counts.txt=1698	VA_B_09_SNP_clippedtrimmed.fastq_counts.txt=1698	VA_W_01_14_clippedtrimmed.fastq_counts.txt=1698	VA_W_01_18_clippedtrimmed.fastq_counts.txt=1698	VA_W_01_22_clippedtrimmed.fastq_counts.txt=1698	VA_W_08_SNP_clippedtrimmed.fastq_counts.txt=1698
[apear012@turing1 SAMS]$ ls
APFullCounts_summed.txt                      stats.txt
matchcounts.txt                              VA_B_01_14_clippedtrimmed.fastq_counts.txt
RI_B_01_14_clippedtrimmed.fastq_counts.txt   VA_B_01_14_clippedtrimmed.fastq.sam
RI_B_01_14_clippedtrimmed.fastq.sam          VA_B_01_18_clippedtrimmed.fastq_counts.txt
RI_B_01_18_clippedtrimmed.fastq_counts.txt   VA_B_01_18_clippedtrimmed.fastq.sam
RI_B_01_18_clippedtrimmed.fastq.sam          VA_B_01_22_clippedtrimmed.fastq_counts.txt
RI_B_01_22_clippedtrimmed.fastq_counts.txt   VA_B_01_22_clippedtrimmed.fastq.sam
RI_B_01_22_clippedtrimmed.fastq.sam          VA_B_09_SNP_clippedtrimmed.fastq_counts.txt
RI_B_08_SNP_clippedtrimmed.fastq_counts.txt  VA_B_09_SNP_clippedtrimmed.fastq.sam
RI_B_08_SNP_clippedtrimmed.fastq.sam         VA_W_01_14_clippedtrimmed.fastq_counts.txt
RI_W_01_14_clippedtrimmed.fastq_counts.txt   VA_W_01_14_clippedtrimmed.fastq.sam
RI_W_01_14_clippedtrimmed.fastq.sam          VA_W_01_18_clippedtrimmed.fastq_counts.txt
RI_W_01_18_clippedtrimmed.fastq_counts.txt   VA_W_01_18_clippedtrimmed.fastq.sam
RI_W_01_18_clippedtrimmed.fastq.sam          VA_W_01_22_clippedtrimmed.fastq_counts.txt
RI_W_01_22_clippedtrimmed.fastq_counts.txt   VA_W_01_22_clippedtrimmed.fastq.sam
RI_W_01_22_clippedtrimmed.fastq.sam          VA_W_08_SNP_clippedtrimmed.fastq_counts.txt
RI_W_08_SNP_clippedtrimmed.fastq_counts.txt  VA_W_08_SNP_clippedtrimmed.fastq.sam
RI_W_08_SNP_clippedtrimmed.fastq.sam
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      scp YOURNAMEFullCounts_summed.txt to your laptop
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ pwd
/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/Day10
(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS/APFullCounts_summed.txt .
apear012@turing.hpc.odu.edu's password: 
APFullCounts_summed.txt                                                      100%  762KB 182.5KB/s   00:04    
(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ ls
APFullCounts_summed.txt
coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen
coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen
homework_day10.txt">
                     <pre class="notranslate"><code>(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ pwd
/Users/AndrewsComputer/Documents/Bio_ODU_2021/21sp_advgenomics/assignments_exercises/Day10
(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ scp apear012@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/SAMS/APFullCounts_summed.txt .
apear012@turing.hpc.odu.edu's password: 
APFullCounts_summed.txt                                                      100%  762KB 182.5KB/s   00:04    
(base) Andrews-MacBook-Air:Day10 AndrewsComputer$ ls
APFullCounts_summed.txt
coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen
coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen
homework_day10.txt
</code></pre>
                    </div>
                    <ol dir="auto" start="5">
                     <li>
                      edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)
                     </li>
                    </ol>
                    <p dir="auto">
                     I used the find a replace feature of BBedit (without GREP) to replace each instance of '_clippedtrimmed.fastq_counts.txt_UniqueTotReads' from the end of the file names
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 11 24-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 11 24-Feb-2021" class="anchor" href="#day-11-24-feb-2021" id="user-content-day-11-24-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file
                     </li>
                    </ol>
                    <p dir="auto">
                     I did this.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 12 26-Feb-2021
                     </h2>
                     <a aria-label="Permalink: Day 12 26-Feb-2021" class="anchor" href="#day-12-26-feb-2021" id="user-content-day-12-26-feb-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 13 3-Mar-2021
                     </h2>
                     <a aria-label="Permalink: Day 13 3-Mar-2021" class="anchor" href="#day-13-3-mar-2021" id="user-content-day-13-3-mar-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       create an sbatch script to rm -r your sandbox/YOURNAME directory
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       submit your sbatch script and add the entry to your logfile/readme
                      </p>
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[apear012@coreV2-25-072 sandboxes]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes
[apear012@coreV2-25-072 sandboxes]$ cat AP_SandboxRemoval.sh
#!/bin/bash -l

#SBATCH -o AP_SandboxRemoval.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_SandboxRemoval

rm -r ./apearson
[apear012@coreV2-25-072 sandboxes]$ sbatch AP_SandboxRemoval.sh 
Submitted batch job 9284108
[apear012@coreV2-25-072 sandboxes]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9284108      main AP_Sandb apear012 PD       0:00      1 (Priority) 
           9284102      main       sh apear012  R       0:58      1 coreV2-25-072 ">
                     <pre class="notranslate"><code>[apear012@coreV2-25-072 sandboxes]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes
[apear012@coreV2-25-072 sandboxes]$ cat AP_SandboxRemoval.sh
#!/bin/bash -l

#SBATCH -o AP_SandboxRemoval.txt
#SBATCH -n 1
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AP_SandboxRemoval

rm -r ./apearson
[apear012@coreV2-25-072 sandboxes]$ sbatch AP_SandboxRemoval.sh 
Submitted batch job 9284108
[apear012@coreV2-25-072 sandboxes]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9284108      main AP_Sandb apear012 PD       0:00      1 (Priority) 
           9284102      main       sh apear012  R       0:58      1 coreV2-25-072 
</code></pre>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="pLpr6T59kyXyVphZVPS1_1Mlu-0MnoK0GfZcych12_ezYvRGonbrRfEaYiACoCgxJZ-WDuw3-S5oJXhizdZF1A">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/acpearson888/advanced_genomics_apearson/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Facpearson888%2Fadvanced_genomics_apearson&amp;report=acpearson888+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/acpearson888/advanced_genomics_apearson/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/acpearson888/packages?repo_name=advanced_genomics_apearson">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/acpearson888/advanced_genomics_apearson/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       Â© 2024 GitHub,Â Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You canât perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
