<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","report_hydro_web_vitals","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         areejmalik/AreejsAdvancedGenomicsLog
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="90A8:5F4D:363BC1:4CA5A0:6605CBDD" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjkwQTg6NUY0RDozNjNCQzE6NENBNUEwOjY2MDVDQkREIiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="59a34591f7ff87903866cb8d67417553c38652d301c48b252fc5ca9fd2a6b5d0" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:332036467" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Contribute to areejmalik/AreejsAdvancedGenomicsLog development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/areejmalik/AreejsAdvancedGenomicsLog" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/7d98093a58f18dffb69e83a9612d2063276dfe2b0cefcda1bf29a62e8575cfdf/areejmalik/AreejsAdvancedGenomicsLog" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="areejmalik/AreejsAdvancedGenomicsLog" name="twitter:title">
                       <meta content="Contribute to areejmalik/AreejsAdvancedGenomicsLog development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/7d98093a58f18dffb69e83a9612d2063276dfe2b0cefcda1bf29a62e8575cfdf/areejmalik/AreejsAdvancedGenomicsLog" property="og:image"/>
                        <meta content="Contribute to areejmalik/AreejsAdvancedGenomicsLog development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="areejmalik/AreejsAdvancedGenomicsLog" property="og:title"/>
                        <meta content="https://github.com/areejmalik/AreejsAdvancedGenomicsLog" property="og:url"/>
                        <meta content="Contribute to areejmalik/AreejsAdvancedGenomicsLog development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjI5ODY0OTEyNywidCI6MTcxMTY1NTkwNX0=--5d11b9e304bb6fd1aa363ca83ffc29c8297478d388dba6114ce13aab78d7dffe" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/areejmalik/AreejsAdvancedGenomicsLog git https://github.com/areejmalik/AreejsAdvancedGenomicsLog.git" name="go-import"/>
                        <meta content="77842593" name="octolytics-dimension-user_id">
                         <meta content="areejmalik" name="octolytics-dimension-user_login">
                          <meta content="332036467" name="octolytics-dimension-repository_id">
                           <meta content="areejmalik/AreejsAdvancedGenomicsLog" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="332036467" name="octolytics-dimension-repository_network_root_id">
                               <meta content="areejmalik/AreejsAdvancedGenomicsLog" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/areejmalik/AreejsAdvancedGenomicsLog" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8" data-view-component="true" id="dialog-show-dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8-description" aria-labelledby="dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-576f1bbd-3a4c-4c27-9d78-2e1f315bf6b8-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-607b9ea1-b096-4ef8-8541-b253e989ba94">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-58901a46-3fb6-4ac5-aa1d-3567433390bb">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-8c81f622-0488-4d00-abc7-4a32adb09b97">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-6332a901-1668-48c6-a091-947ed3c3dd7a">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-52a5835f-ac65-4333-904b-332f8464ed84">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-7a65f696-6625-46f7-8514-092bd245809a">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-153ba36c-df2c-4de9-9d89-d5d42bf9186c">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-0f30b82a-4767-4a0b-ac5b-19f14037338e">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               © 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: areejmalik / AreejsAdvancedGenomicsLog" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              areejmalik
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              AreejsAdvancedGenomicsLog
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"areejmalik","screen_size":"compact"}' data-view-component="true" href="/areejmalik">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 areejmalik
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"AreejsAdvancedGenomicsLog","screen_size":"compact"}' data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 AreejsAdvancedGenomicsLog
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"areejmalik","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/areejmalik/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/areejmalik">
             <span class="AppHeader-context-item-label">
              areejmalik
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"AreejsAdvancedGenomicsLog","screen_size":"full"}' data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog">
             <span class="AppHeader-context-item-label">
              AreejsAdvancedGenomicsLog
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="cHqMQEMRkyo4PvEZJEknAeQk4Nq7vbGFm1u0_nvFlm4gWYDI0ZxM5Y-kqTeB3sVa0KggMY-WBBRFIMpA1gErhg" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="areejmalik" data-current-repository="areejmalik/AreejsAdvancedGenomicsLog" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="vbkoxGrrCH97mci0I7xPVUhByZe0lRjRL60tSzIJjn_1cz_uTK1BCiw1um3Fub9EGuWLPpLy6L5DT2zg1TeVMw" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:areejmalik/AreejsAdvancedGenomicsLog">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump to…" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump to…" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-42157612-398a-4068-b8cb-2f5c74464fe5" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-71085d3b-9964-49f7-ac00-bee5f1d7ed5c" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-71085d3b-9964-49f7-ac00-bee5f1d7ed5c">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="Pn_5tGy0OEnuhVuBv6-ox1t6exhbQCe2LgH6ZGOPci6szeCMpGIL016nVYWBtqk2hnG22wFgweMgge_eld53lQ"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="DMG6szWny7jDIIAr3bhEMq2Q_TuCg0DUZZaThZWfnaxAM9gJbSaVYYhJPYaXbYkPZMCZLNSGAg_GzkI9itjE-Q">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="mQ4zxAoHt_bCaPPfc2l076tPl7Y1JMYtsgkKNtv6hOWBU35iXYUe9UF8ulwzOyqFHYiSMRv9JrXfqajk45VXLQ">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="uJPIKh9l-5X3SsOc9hzSOy8mbXrD47hJQb4K9_4lh6V_xqcIBmI99xCRWWE_p6ko_561sEZFfnlO3VRSZrHmrw">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-896e23b0-2ac3-4682-a694-8380069828db" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-f84e6d4c-886e-4fb5-9967-409d6f382547" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-fe8fd3ad-149e-4e51-89e2-fd36115bbe4f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-06e890b5-39cd-438f-9e2b-f531c732785b" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-7eedf0d2-7fec-4bcf-9ace-96c014b7bc06" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-e173ea4f-ca51-4213-9eb3-76f16de1bd33" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-23a7ac74-d3ba-40a3-a404-e4cd7dd59605" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-0ce81717-2e90-4cf2-9617-651d5e0fcc20">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-0ce81717-2e90-4cf2-9617-651d5e0fcc20" id="tooltip-23a7ac74-d3ba-40a3-a404-e4cd7dd59605" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-fa1f8aea-73ad-4443-9fee-50aa96df20dc" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-1966c141-f73f-403d-b14c-1103bc0348ac">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-1966c141-f73f-403d-b14c-1103bc0348ac" id="tooltip-fa1f8aea-73ad-4443-9fee-50aa96df20dc" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTkwNX0=--7514569c7cb6708c2223e2c575ff597ea1d9802ff9e90736fab3c81cd71dcf55" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-6ac9717a-64dd-438a-8e9c-cb0de8f2b8d3" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=332036467">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd" data-view-component="true" id="dialog-show-dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd-description" aria-labelledby="dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-14c79ec9-a0c3-4492-bda3-1269819181f9" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-921c6e2f-13d5-4737-bcbd-79a82d275a74" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-7d9f9284-01c9-451d-94ea-fbf3a52e9984" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-ab4b3280-0939-456f-b117-f0251d198522" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-054b9118-9a3e-4f67-b221-e084bdb0c5b7" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-aeacae30-e177-4884-b368-c80c89b61dcb" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-eae20b1c-61a7-494a-9f7e-ad2719ebaacd-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-350353dc-73e8-4eb5-9c61-2fe0a49d18e2" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-3cfabbd9-33bf-4076-9dcc-8cf00b2c3a48">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-47b8034c-0bcf-4ef4-b2bf-f894c6a58162" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-76e25ed2-a9b1-4f42-b977-40eca6b72cb0">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-54b8230b-2788-4015-aa14-8eae749d0d4d">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-ef1f5e4f-0ba9-44a0-bf90-c8c136bf1d4d" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-70dc960a-53dd-4e78-8d55-2333e87e28e1" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-1e7d6bf8-ce4c-439e-868f-f588342bb4e3">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-4c65dab3-48fe-4a28-9577-8b211359eb3d">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-14ebcbdb-26fa-44aa-b56e-ceb4331b14a5">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-d0e7ce67-6155-45f8-a799-21309db4a630" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-c8faf32f-90f6-4bfa-b45b-72b5fd31fd4e" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-f5086d67-7f17-47ec-8757-a6e1d8861d1d">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-4d904a88-38f1-45f8-913d-9f62c3308215">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-5469b7c2-c014-487f-b84a-519762230743">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-1523ccb0-53d8-419d-8445-e0ee17d06256">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /areejmalik/AreejsAdvancedGenomicsLog" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /areejmalik/AreejsAdvancedGenomicsLog/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /areejmalik/AreejsAdvancedGenomicsLog/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /areejmalik/AreejsAdvancedGenomicsLog/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /areejmalik/AreejsAdvancedGenomicsLog/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /areejmalik/AreejsAdvancedGenomicsLog/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/areejmalik/AreejsAdvancedGenomicsLog/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /areejmalik/AreejsAdvancedGenomicsLog/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-list" aria-haspopup="true" aria-labelledby="tooltip-150a98e6-0c25-4d8d-abe4-5a941d02396d" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-button" popovertarget="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-button" id="tooltip-150a98e6-0c25-4d8d-abe4-5a941d02396d" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-button" anchor-offset="normal" data-view-component="true" id="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-2e534c88-107d-4912-a14a-d6892bacbf24-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog" id="item-1d3bad98-e239-46d1-96b2-a6b618425aef" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/issues" id="item-b4534087-669b-46ff-8a6f-67ea22d9654b" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/pulls" id="item-2aaea21d-7d50-46af-a4e8-23a519af837e" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/actions" id="item-219f72f8-4c96-4992-bb04-5aff2929655a" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/projects" id="item-ce5091d1-39df-4f78-903d-febb3e1bc264" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/security" id="item-d626d79b-e7b3-46ae-9a86-0b3f3f3d0de9" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/pulse" id="item-30551904-346f-416c-acd0-f127714ce917" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-9f4b6231-4b31-4fa3-82e4-21dde890ce55" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-edc6d518-680a-4967-a43b-1a9bfc4cbb3c" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-edc6d518-680a-4967-a43b-1a9bfc4cbb3c" id="tooltip-9f4b6231-4b31-4fa3-82e4-21dde890ce55" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTkwNX0=--7514569c7cb6708c2223e2c575ff597ea1d9802ff9e90736fab3c81cd71dcf55" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/areejmalik/AreejsAdvancedGenomicsLog" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjc3ODQyNTkz" data-targets="command-palette-scope.tokens" data-text="areejmalik" data-type="owner" data-value="areejmalik" style="white-space:nowrap;line-height:20px;">
         areejmalik
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzMzIwMzY0Njc=" data-targets="command-palette-scope.tokens" data-text="AreejsAdvancedGenomicsLog" data-type="repository" data-value="AreejsAdvancedGenomicsLog" style="white-space:nowrap;line-height:20px;">
         AreejsAdvancedGen...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-d4c32f26-7c29-4e53-8ce3-315af7f1ed6d" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzMzIwMzY0Njc=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         We’ve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="areejmalik" data-scope-id="MDQ6VXNlcjc3ODQyNTkz" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="AreejsAdvancedGenomicsLog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzIwMzY0Njc=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="areejmalik" data-scope-id="MDQ6VXNlcjc3ODQyNTkz" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="AreejsAdvancedGenomicsLog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzIwMzY0Njc=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="ElONuoYtk-JMQeHgYMcJp4JJgJ3FEeWybRNVey7ji_fCPyeeE_dFRa98S4fzH5ABJ0pL6EE2zI7zuXDOmjKdQA">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="⭐️ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="urgLgZAQ2badwj8LYbBhuAzA-4KzGdqrMm5cPGvmrHrakFdwXz6L0kuJoE-CsqWODSql8x_rZJGeRCb1iY4Y1A">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/77842593?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/areejmalik/AreejsAdvancedGenomicsLog">
            AreejsAdvancedGenomicsLog
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/332036467/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":332036467,"originating_url":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="7dbdafc4b5379e6201b415e480db2643be7f99308220b6e63ff3a5f1894e762c" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-b1dd8cfe-ee61-44c9-a59e-b9d20f258a22" popover="manual">
              Fork your own copy of areejmalik/AreejsAdvancedGenomicsLog
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-332036467">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/areejmalik/AreejsAdvancedGenomicsLog/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-332036467" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-34b1ef">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/areejmalik/AreejsAdvancedGenomicsLog/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="5I7R90agxCrssSzoNMrUFKJ7pIk3sdW8KIyEKO6hj-KyTNxyu5QZB0ugcA1u8SEC9AVKmbRNowZ4wTbuW_thQA">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="grKRIbJWy3DWcms1p9m_D-N5bbY5KnchqLK7shUEd7LUcJykT2IWXXFjN9D94koZtQeDprrWAZv4_wl0oF6ZEA">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (0)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":332036467,"originating_url":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="3958930efb2be3f74e1c11664542d4df62990800b209c484ba4171caa7e5ad04" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="0">
                  0
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-332036467-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/areejmalik/AreejsAdvancedGenomicsLog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-332036467-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-3360f6">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/areejmalik/AreejsAdvancedGenomicsLog/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="lVyRk5StRVIZYs2by55xWypt2gxOO-lYTgVDSZZx08W_mU7mAdX99EG_-8YROVKw__Yx2HFGC81944WnmuJ4aA">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (0)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":332036467,"originating_url":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="883376a6cde0af1168c2207bf3f8163cd783613b59d7119203191139297168a0" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
                 0
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-332036467-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/areejmalik/AreejsAdvancedGenomicsLog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-332036467-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-549fea">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":332036467,"originating_url":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="0e6f22d558240d724cc939e85fa88cc361a9a549553e323ff0c8e4c032810650" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="6LrYvjmfCUBgU2cTWz2lSP6awDKs0UCNVdgLIpKMZ0jwetTEJyB8QYxAUBmH3SM_ob9TJ3mL61l3I_nWxT3Z0Q">
                   <input name="repository_id" type="hidden" value="332036467"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="Yiigxdfsg_XblaTmFNtF5C-a6iVE4MPWVXMkA_x5sG166Ky_yVP29DeGk-zIO8OTcL95MJG6aAJ3iNb3q8gO9A">
                   <input name="repository_id" type="hidden" value="332036467"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-a8ed36d8-88e9-433a-8f99-5d1ea9d7af1e" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-28b7a34c-cc5f-456b-8215-1be31ebb9828" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":332036467,"originating_url":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="7dbdafc4b5379e6201b415e480db2643be7f99308220b6e63ff3a5f1894e762c" data-view-component="true" group_item="true" href="/areejmalik/AreejsAdvancedGenomicsLog/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-28b7a34c-cc5f-456b-8215-1be31ebb9828" popover="manual">
              Fork your own copy of areejmalik/AreejsAdvancedGenomicsLog
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/areejmalik/AreejsAdvancedGenomicsLog/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="txwAPOu6P4b3J7mdXAbRqnN0VGrMESONJPBL_yJJ87Xh3g25Fo7iq1A25XgGPSS8JQq6ek_tVTd0vfk5lxMdFw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-5ad342ff-3a63-45f4-8b76-cf04170f7a9a" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":332036467,"originating_url":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="3958930efb2be3f74e1c11664542d4df62990800b209c484ba4171caa7e5ad04" data-view-component="true" id="icon-button-63f3b0f9-dc62-4abd-b479-a97fec9ee306" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-63f3b0f9-dc62-4abd-b479-a97fec9ee306" id="tooltip-5ad342ff-3a63-45f4-8b76-cf04170f7a9a" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/areejmalik/AreejsAdvancedGenomicsLog/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="NGMnpQ0gGhb6BWJC4naSky4Pg38kbmqeHBjZk3v0YYUepvjQmFiisKLYVB840bF4-5RoqxsTiAsv_h99d2fKKA">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-8a155c08-d4aa-495c-ba1d-f63304c9d8db" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":332036467,"originating_url":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="883376a6cde0af1168c2207bf3f8163cd783613b59d7119203191139297168a0" data-view-component="true" id="icon-button-4567a0c1-6a37-4e34-bcd0-350f89f977b6" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-4567a0c1-6a37-4e34-bcd0-350f89f977b6" id="tooltip-8a155c08-d4aa-495c-ba1d-f63304c9d8db" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/areejmalik/AreejsAdvancedGenomicsLog/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/areejmalik/AreejsAdvancedGenomicsLog/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/areejmalik/AreejsAdvancedGenomicsLog/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/areejmalik/AreejsAdvancedGenomicsLog/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/areejmalik/AreejsAdvancedGenomicsLog/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/areejmalik/AreejsAdvancedGenomicsLog/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/areejmalik/AreejsAdvancedGenomicsLog/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/areejmalik/AreejsAdvancedGenomicsLog?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         areejmalik/AreejsAdvancedGenomicsLog
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/areejmalik/AreejsAdvancedGenomicsLog/spoofed_commit_check/fc0ef4d355ee1c6a558e8c102119fdd29f95825e">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":332036467,"defaultBranch":"main","name":"AreejsAdvancedGenomicsLog","ownerLogin":"areejmalik","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-01-23T02:57:34.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/77842593?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1611342015.0","canEdit":true,"refType":"branch","currentOid":"fc0ef4d355ee1c6a558e8c102119fdd29f95825e"},"tree":{"items":[{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":1,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/areejmalik/AreejsAdvancedGenomicsLog/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/areejmalik/AreejsAdvancedGenomicsLog.git","showCloneWarning":false,"sshUrl":"git@github.com:areejmalik/AreejsAdvancedGenomicsLog.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone areejmalik/AreejsAdvancedGenomicsLog","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/areejmalik/AreejsAdvancedGenomicsLog","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fareejmalik%2FAreejsAdvancedGenomicsLog","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/areejmalik/AreejsAdvancedGenomicsLog/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"32","overviewFiles":[{"displayName":"README.md","repoName":"AreejsAdvancedGenomicsLog","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cp dir=\"auto\"\u003e2021_Spring_Areej's Advanced Genomics Notebook\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e##Day 02 22-Jan-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day02\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n#1 - Log on to the cluster @turing.hpc.odu.edu\n[amali010@turing1 ~]$ ls\ndata  exercises  hpc-intro\n\n#2 - Make a directory in your course workspace called data\n[amali010@turing1 ~]$ mkdir data\n\n#3 - Make a directory called exercises in your data directory\n[amali010@turing1 data]$ mkdir exercises\n\n#4 - Execute a pwd command and start a log of your commands with a \nheader for today's date in your README.md github logfile in your workspace\n[amali010@turing1 exercises]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/exercises \n\n#5 - cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory\n[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./\n[amali010@turing1 exercises]$ ls\nExercise2_djb.fasta.gz  Exercise2.fasta.gz  Exercise2.fastq.tar.gz\nExercise2.fasta         Exercise2.fastq\n[amali010@turing1 exercises]$ ls -alh\ntotal 37M\ndrwxrwxrwx 2 amali010 users  146 Jan 26 20:48 .\ndrwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz\n-rwxr-xr-x 1 amali010 users    0 Jan 29 09:53 Exercise2.fasta\n-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz\n\n#6 - Unzip and untar the files\n[amali010@turing1 exercises]$ gunzip -c Exercise2_djb.fasta.gz \u0026gt; Exercise2.fasta\n[amali010@turing1 exercises]$ ls\nExercise2_djb.fasta.gz  Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz\n[amali010@turing1 exercises]$ ls -alh\ntotal 61M\ndrwxrwxrwx 2 amali010 users  146 Jan 26 20:48 .\ndrwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz\n-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta\n-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz\n\n[amali010@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz\nExercise2.fastq\n[amali010@turing1 exercises]$ ls -alh\ntotal 69M\ndrwxrwxrwx 2 amali010 users  182 Jan 29 10:11 .\ndrwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz\n-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta\n-rwxr-xr-x 1 amali010 users 4.3M Jan 29 10:06 Exercise2.fasta.gz\n-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz\n\n#7 - Add these commands to your notebook\nDone\n\n#8 - Calculate how many sequences are in each file and add these results to your notebook file\n[amali010@turing1 exercises]$ head -1 Exercise2.fasta\n\u0026gt;scaffold10078|size20675\n[amali010@turing1 exercises]$ grep -c '\u0026gt;' Exercise2.fasta\n138\n[amali010@turing1 exercises]$ head -1 Exercise2.fastq\n@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC\n[amali010@turing1 exercises]$ grep -c '@HS3' Exercise2.fastq\n61304\n[amali010@turing1 exercises]$ wc -l Exercise2.fastq\n245216 Exercise2.fastq\n[amali010@turing1 exercises]$ echo 245216/4 | bc -l\n61304.00000000000000000000\n\n#9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory\n[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../scripts/\ncp: cannot create regular file `../scripts/': Is a directory\n[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/\n[amali010@turing1 exercises]$ cd ..\n[amali010@turing1 data]$ cd ..\n[amali010@turing1 areej]$ ls\nam_exercise1.txt  data  day03  groundrules.txt  scripts\n[amali010@turing1 areej]$ cd scripts\n[amali010@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\nrenamer_advbioinf.py\n\n#10 - Start an interactive compute session and re-navigate to your exercises directory\n[amali010@turing1 exercises]$ salloc\nsalloc: Pending job allocation 9271058\nsalloc: job 9271058 queued and waiting for resources\nsalloc: job 9271058 has been allocated resources\nsalloc: Granted job allocation 9271058\n\n#11 - Run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name\n[amali010@coreV2-22-007 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 138\ncontigs \u0026gt;= 1000bp = 138\ncontigs \u0026gt;= 2000bp = 135\n\n#12 - Did you add all these entries to your notebook file, including the results of the script?\nYes!\n\n#13 - Push your notebook file to your github page\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Log on to the cluster @turing.hpc.odu.edu\u003c/span\u003e\n[amali010@turing1 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e]$ ls\ndata  exercises  hpc-intro\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - Make a directory in your course workspace called data\u003c/span\u003e\n[amali010@turing1 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e]$ mkdir data\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Make a directory called exercises in your data directory\u003c/span\u003e\n[amali010@turing1 data]$ mkdir exercises\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Execute a pwd command and start a log of your commands with a \u003c/span\u003e\nheader \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etoday\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es date\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e your README.md github logfile \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e your workspace\n[amali010@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/exercises \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5 - cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory\u003c/span\u003e\n[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e ./\n[amali010@turing1 exercises]$ ls\nExercise2_djb.fasta.gz  Exercise2.fasta.gz  Exercise2.fastq.tar.gz\nExercise2.fasta         Exercise2.fastq\n[amali010@turing1 exercises]$ ls -alh\ntotal 37M\ndrwxrwxrwx 2 amali010 users  146 Jan 26 20:48 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz\n-rwxr-xr-x 1 amali010 users    0 Jan 29 09:53 Exercise2.fasta\n-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6 - Unzip and untar the files\u003c/span\u003e\n[amali010@turing1 exercises]$ gunzip -c Exercise2_djb.fasta.gz \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e Exercise2.fasta\n[amali010@turing1 exercises]$ ls\nExercise2_djb.fasta.gz  Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz\n[amali010@turing1 exercises]$ ls -alh\ntotal 61M\ndrwxrwxrwx 2 amali010 users  146 Jan 26 20:48 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz\n-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta\n-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz\n\n[amali010@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz\nExercise2.fastq\n[amali010@turing1 exercises]$ ls -alh\ntotal 69M\ndrwxrwxrwx 2 amali010 users  182 Jan 29 10:11 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz\n-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta\n-rwxr-xr-x 1 amali010 users 4.3M Jan 29 10:06 Exercise2.fasta.gz\n-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq\n-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e7 - Add these commands to your notebook\u003c/span\u003e\nDone\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e8 - Calculate how many sequences are in each file and add these results to your notebook file\u003c/span\u003e\n[amali010@turing1 exercises]$ head -1 Exercise2.fasta\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003escaffold10078\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003esize20675\n[amali010@turing1 exercises]$ grep -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e Exercise2.fasta\n138\n[amali010@turing1 exercises]$ head -1 Exercise2.fastq\n@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC\n[amali010@turing1 exercises]$ grep -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e@HS3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e Exercise2.fastq\n61304\n[amali010@turing1 exercises]$ wc -l Exercise2.fastq\n245216 Exercise2.fastq\n[amali010@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e 245216/4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e bc -l\n61304.00000000000000000000\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory\u003c/span\u003e\n[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../scripts/\ncp: cannot create regular file \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e../scripts/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e: Is a directory\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@turing1 exercises]$ cd ..\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@turing1 data]$ cd ..\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@turing1 areej]$ ls\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eam_exercise1.txt  data  day03  groundrules.txt  scripts\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@turing1 areej]$ cd scripts\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@turing1 scripts]$ ls\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eavg_cov_len_fasta_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003erenamer_advbioinf.py\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e#10 - Start an interactive compute session and re-navigate to your exercises directory\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@turing1 exercises]$ salloc\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003esalloc: Pending job allocation 9271058\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003esalloc: job 9271058 queued and waiting for resources\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003esalloc: job 9271058 has been allocated resources\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003esalloc: Granted job allocation 9271058\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e#11 - Run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-007 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eThe total number of sequences is 138\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eThe average sequence length is 126640\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eThe total number of bases is 17476447\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eThe minimum sequence length is 1122\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eThe maximum sequence length is 562680\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eThe N50 is 187037\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eMedian Length = 92612\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003econtigs \u0026lt; 150bp = 0\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003econtigs \u0026gt;= 500bp = 138\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003econtigs \u0026gt;= 1000bp = 138\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003econtigs \u0026gt;= 2000bp = 135\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e#12 - Did you add all these entries to your notebook file, including the results of the script?\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003eYes!\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003e#13 - Push your notebook file to your github page\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003egit add README.md\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003egit commit -m \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003eupdating readme\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-s\"\u003egit push -u origin main\u003c/span\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 03 27-Jan-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eHomework day02 (originally meant for 22-Jan-2021 but done on 27-Jan-2021)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 - Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory\n   Kristina-HADB01\n   Ivan-HADB02\n   KatieP-HADB03\n   KatieC-HADB04\n   Stephanie-HADB05\n   Areej-HADB06\n   Andrew-HADB01\n   John-HADB02\n   dan-All\n   \n[amali010@coreV2-25-072 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n\n#2 - Add the content of your sbatch script to your logfile\n[amali010@coreV2-25-072 data]$ cat AMCopyLane06.sh\n#!/bin/bash -l\n\n#SBATCH -o AMCopyLane06.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMCopyLane06\n\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB06*.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n\n#3 - Submit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)\n[amali010@coreV2-25-072 data]$ sbatch AMCopyLane06.sh\nSubmitted batch job 9270443\n\n#4 - Make sure this is all documented on your github notebook\nDone\n\n#5 - Write a sbatch script to gunzip all the fastq.gz files\n[amali010@coreV2-25-072 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n[amali010@coreV2-25-072 data]$ cat AMGunzipLane06.sh\n#!/bin/bash -l\n\n#SBATCH -o AMGunzipLane06.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMGunzipLane06\n\ngunzip *.fastq.gz\n\n[amali010@coreV2-25-072 data]$ sbatch AMGunzipLane06.sh\nSubmitted batch job 9270466\n\n#6 - Push your notebook file to your github page\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory\u003c/span\u003e\n   Kristina-HADB01\n   Ivan-HADB02\n   KatieP-HADB03\n   KatieC-HADB04\n   Stephanie-HADB05\n   Areej-HADB06\n   Andrew-HADB01\n   John-HADB02\n   dan-All\n   \n[amali010@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - Add the content of your sbatch script to your logfile\u003c/span\u003e\n[amali010@coreV2-25-072 data]$ cat AMCopyLane06.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMCopyLane06.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMCopyLane06\u003c/span\u003e\n\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB06\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Submit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)\u003c/span\u003e\n[amali010@coreV2-25-072 data]$ sbatch AMCopyLane06.sh\nSubmitted batch job 9270443\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Make sure this is all documented on your github notebook\u003c/span\u003e\nDone\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5 - Write a sbatch script to gunzip all the fastq.gz files\u003c/span\u003e\n[amali010@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n[amali010@coreV2-25-072 data]$ cat AMGunzipLane06.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMGunzipLane06.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMGunzipLane06\u003c/span\u003e\n\ngunzip \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.gz\n\n[amali010@coreV2-25-072 data]$ sbatch AMGunzipLane06.sh\nSubmitted batch job 9270466\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6 - Push your notebook file to your github page\u003c/span\u003e\ngit add README.md\ngit commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\ngit push -u origin main\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 03 27-Jan-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eHomework day03\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[amali010@turing1 areej]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej\n\n#1 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox\n[amali010@turing1 areej]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 ./\n\n#2 - mkdir a fastq directory in your data directory and mv all the .fastq files into this directory\n[amali010@turing1 areej]$ cd data\n[amali010@turing1 data]$ mkdir fastq\n[amali010@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n[amali010@turing1 data]$ mv *.fastq ./fastq/\n\n#3 - cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your  sandbox scripts folder and less the new script and check out the usage statement\n[amali010@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/\n[amali010@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ../scripts/\n[amali010@turing1 data]$ head ../scripts/renamer_advbioinf.py\n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\n\n#4 - Run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand\n[amali010@turing1 data]$ cd fastq/\n[amali010@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\nmv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\nmv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\nmv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\nmv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\nmv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\nmv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\nmv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\nmv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\nmv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\nmv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\nmv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\nmv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\nmv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\nmv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\nmv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\nmv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\nmv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\nmv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\nmv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\nmv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\nmv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\nmv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\nmv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\nmv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\nmv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\nmv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\nmv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\nmv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\nmv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\nmv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\nmv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\nmv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\nmv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\nmv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\nmv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\nmv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\nmv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\nmv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\nmv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\nmv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\nmv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\nmv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\nmv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\nmv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\nmv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\nmv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\nmv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\nmv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\nmv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\nmv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\nmv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\nmv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\nmv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\nmv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\nmv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\nmv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\nmv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\nmv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\nmv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\nmv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\nmv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\nmv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\nmv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\nmv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\nmv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\nmv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\nmv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\nmv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\nmv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\nmv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\nmv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\nmv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\nmv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\nmv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\nmv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\nmv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\nmv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\nmv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\nmv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\nmv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\nmv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\nmv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\nmv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\nmv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\nmv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\nmv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\nmv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\nmv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\nmv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\nmv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\nmv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\nmv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\nmv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\nmv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\nmv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\nmv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\n\n#5 - Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print\n[amali010@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py\n\n#6 - Write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script\n[amali010@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ nano areejsrenamer.sh\n[amali010@turing1 fastq]$ cat areejsrenamer.sh\n#!/bin/bash -l\n\n#SBATCH -o areejsrenamer.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=FastqRename\n\n../../scripts/renamer_advbioinf.py renamingtable_complete.txt\n\n[amali010@turing1 fastq]$ sbatch ./areejsrenamer.sh\nSubmitted batch job 9270936\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n\n#7 - Make sure this is all documented on your github page\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)\n$ git add README.md\n\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)\n$ git commit -m 'updating readme'\n[main 4326621] updating readme\n 1 file changed, 20 insertions(+), 1 deletion(-)\n\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)\n$ git push -u origin main\nEnumerating objects: 5, done.\nCounting objects: 100% (5/5), done.\nDelta compression using up to 8 threads\nCompressing objects: 100% (2/2), done.\nWriting objects: 100% (3/3), 491 bytes | 245.00 KiB/s, done.\nTotal 3 (delta 1), reused 0 (delta 0), pack-reused 0\nremote: Resolving deltas: 100% (1/1), completed with 1 local object.\nTo https://github.com/areejmalik/AreejsAdvancedGenomicsLog.git\n   aa0237a..4326621  main -\u0026gt; main\nBranch 'main' set up to track remote branch 'main' from 'origin'.\n\n#8 - The naming convention for the files is as follows:\nSOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq\nThere are 2 sources: Virginia and Rhode Island\nThere are 2 symbiotic states: Brown and White\n\t\n#9 - Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane\nOk!\n\n#10 - cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory\n[amali010@coreV1-22-016 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n[amali010@coreV1-22-016 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/\n\n#11 - Less/head the new script and check out the usage statement\n[amali010@coreV1-22-016 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py\n#!/usr/bin/env python\n# Written by Dan Barshis\n\nimport sys, os\n\n########### Usage #############\n# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\n# This should be run from within the folder with all of your original .fastqstrim\n# Will quality trim multiple SINGLE-END fastq files\n# Things to customize for your particular platform:\n# qualoffset = 33 Quality score offset\n# -t option in qualitytrim (the lower threshold quality score for trimming)\n# -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)\n\n#12 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt \ninto the working directory with your fastq files\n[amali010@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./\n\n#13 - Make a sbatch script for the Trimclipfilter... script and run it on your fastq files\n[amali010@turing1 ~]$ salloc\nsalloc: Pending job allocation 9270954\nsalloc: job 9270954 queued and waiting for resources\nsalloc: job 9270954 has been allocated resources\nsalloc: Granted job allocation 9270954\nNOTE: The job allocation number matches that of the one that is currently the only one running\n[amali010@coreV1-22-005 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@coreV1-22-005 fastq]$ nano FullTrimClip.sh\n[amali010@coreV1-22-005 fastq]$ cat FullTrimClip.sh\n#!/bin/bash -l\n\n#SBATCH -o AMFullTrimclip.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMTrimTest\n\n../../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\n\n[amali010@coreV1-22-005 fastq]$ sbatch FullTrimClip.sh\nSubmitted batch job 9270961\n[amali010@coreV1-22-005 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270961      main AMTrimTe amali010 PD       0:00      1 (Priority)\n           9270954      main       sh amali010  R      18:11      1 coreV1-22-005\n[amali010@coreV1-22-005 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270954      main       sh amali010  R      19:42      1 coreV1-22-005\nNOTE: I don't know why it keeps failing the job partition I even tried logging \nout the core and remained on turing but that was rejected as well. :(\n\nAfter multiple attempts!!\n[amali010@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ nano amFullTrimClip.sh\n[amali010@turing1 fastq]$ cat amFullTrimClip.sh\n#!/bin/bash -l\n\n#SBATCH -o amFullTrimclip.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=amTrimFull\n\n../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\n[amali010@turing1 fastq]$ sbatch amFullTrimClip.sh\nSubmitted batch job 9270966\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R       0:03      1 coreV1-22-005\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R       0:22      1 coreV1-22-005\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R       1:17      1 coreV1-22-005\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R       2:40      1 coreV1-22-005\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R       3:25      1 coreV1-22-005\n[amali010@turing1 fastq]$ ls\nadapterlist_advbioinf.txt   RI_B_06_18.fastq   VA_B_06_18.fastq\namFullTrimClip.sh           RI_B_07_14.fastq   VA_B_07_14.fastq\namFullTrimclip.txt          RI_B_07_18.fastq   VA_B_07_18.fastq\nAMFullTrimclip.txt          RI_B_07_22.fastq   VA_B_07_22.fastq\nareejsrenamer.sh            RI_W_06_18.fastq   VA_W_06_18.fastq\nAreejsRenamer.sh            RI_W_07_14.fastq   VA_W_07_14.fastq\nareejsrenamer.txt           RI_W_07_18.fastq   VA_W_07_18.fastq\nFullTrimClip.sh             RI_W_07_22.fastq   VA_W_07_22.fastq\nrenamingtable_complete.txt  Testing\nRI_B_06_18_clipped.fastq    trimclipstats.txt\n\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R       5:20      1 coreV1-22-005\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R       9:01      1 coreV1-22-005\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270966      main amTrimFu amali010  R      17:31      1 coreV1-22-005\n\n#14 - Push your notebook file to your github page\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e[amali010@turing1 areej]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox\u003c/span\u003e\n[amali010@turing1 areej]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 ./\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - mkdir a fastq directory in your data directory and mv all the .fastq files into this directory\u003c/span\u003e\n[amali010@turing1 areej]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data\n[amali010@turing1 data]$ mkdir fastq\n[amali010@turing1 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n[amali010@turing1 data]$ mv \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq ./fastq/\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your  sandbox scripts folder and less the new script and check out the usage statement\u003c/span\u003e\n[amali010@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/\n[amali010@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ../scripts/\n[amali010@turing1 data]$ head ../scripts/renamer_advbioinf.py\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###usage renamer.py renamingtable\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand\u003c/span\u003e\n[amali010@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e fastq/\n[amali010@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\nmv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\nmv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\nmv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\nmv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\nmv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\nmv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\nmv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\nmv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\nmv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\nmv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\nmv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\nmv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\nmv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\nmv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\nmv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\nmv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\nmv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\nmv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\nmv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\nmv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\nmv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\nmv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\nmv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\nmv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\nmv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\nmv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\nmv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\nmv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\nmv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\nmv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\nmv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\nmv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\nmv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\nmv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\nmv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\nmv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\nmv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\nmv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\nmv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\nmv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\nmv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\nmv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\nmv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\nmv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\nmv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\nmv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\nmv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\nmv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\nmv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\nmv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\nmv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\nmv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\nmv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\nmv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\nmv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\nmv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\nmv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\nmv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\nmv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\nmv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\nmv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\nmv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\nmv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\nmv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\nmv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\nmv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\nmv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\nmv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\nmv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\nmv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\nmv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\nmv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\nmv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\nmv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\nmv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\nmv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\nmv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\nmv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\nmv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\nmv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\nmv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\nmv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\nmv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\nmv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\nmv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\nmv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\nmv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\nmv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\nmv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\nmv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\nmv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\nmv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\nmv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\nmv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\nmv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\nmv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5 - Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print\u003c/span\u003e\n[amali010@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6 - Write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script\u003c/span\u003e\n[amali010@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ nano areejsrenamer.sh\n[amali010@turing1 fastq]$ cat areejsrenamer.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o areejsrenamer.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=FastqRename\u003c/span\u003e\n\n../../scripts/renamer_advbioinf.py renamingtable_complete.txt\n\n[amali010@turing1 fastq]$ sbatch ./areejsrenamer.sh\nSubmitted batch job 9270936\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e7 - Make sure this is all documented on your github page\u003c/span\u003e\nareej@DESKTOP-RF6VF4N MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)\n$ git add README.md\n\nareej@DESKTOP-RF6VF4N MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)\n$ git commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n[main 4326621] updating readme\n 1 file changed, 20 insertions(+), 1 deletion(-)\n\nareej@DESKTOP-RF6VF4N MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)\n$ git push -u origin main\nEnumerating objects: 5, done.\nCounting objects: 100% (5/5), done.\nDelta compression using up to 8 threads\nCompressing objects: 100% (2/2), done.\nWriting objects: 100% (3/3), 491 bytes \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e 245.00 KiB/s, done.\nTotal 3 (delta 1), reused 0 (delta 0), pack-reused 0\nremote: Resolving deltas: 100% (1/1), completed with 1 \u003cspan class=\"pl-k\"\u003elocal\u003c/span\u003e object.\nTo https://github.com/areejmalik/AreejsAdvancedGenomicsLog.git\n   aa0237a..4326621  main -\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e main\nBranch \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emain\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eset\u003c/span\u003e up to track remote branch \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emain\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e from \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eorigin\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e.\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e8 - The naming convention for the files is as follows:\u003c/span\u003e\nSOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq\nThere are 2 sources: Virginia and Rhode Island\nThere are 2 symbiotic states: Brown and White\n\t\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e9 - Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane\u003c/span\u003e\nOk\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e10 - cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory\u003c/span\u003e\n[amali010@coreV1-22-016 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data\n[amali010@coreV1-22-016 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e11 - Less/head the new script and check out the usage statement\u003c/span\u003e\n[amali010@coreV1-22-016 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Written by Dan Barshis\u003c/span\u003e\n\nimport sys, os\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########## Usage #############\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This should be run from within the folder with all of your original .fastqstrim\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Will quality trim multiple SINGLE-END fastq files\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Things to customize for your particular platform:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e qualoffset = 33 Quality score offset\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e -t option in qualitytrim (the lower threshold quality score for trimming)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e12 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt \u003c/span\u003e\ninto the working directory with your fastq files\n[amali010@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e13 - Make a sbatch script for the Trimclipfilter... script and run it on your fastq files\u003c/span\u003e\n[amali010@turing1 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e]$ salloc\nsalloc: Pending job allocation 9270954\nsalloc: job 9270954 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9270954 has been allocated resources\nsalloc: Granted job allocation 9270954\nNOTE: The job allocation number matches that of the one that is currently the only one running\n[amali010@coreV1-22-005 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@coreV1-22-005 fastq]$ nano FullTrimClip.sh\n[amali010@coreV1-22-005 fastq]$ cat FullTrimClip.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMFullTrimclip.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMTrimTest\u003c/span\u003e\n\n../../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq\n\n[amali010@coreV1-22-005 fastq]$ sbatch FullTrimClip.sh\nSubmitted batch job 9270961\n[amali010@coreV1-22-005 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270961      main AMTrimTe amali010 PD       0:00      1 (Priority)\n           9270954      main       sh amali010  R      18:11      1 coreV1-22-005\n[amali010@coreV1-22-005 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9270954      main       sh amali010  R      19:42      1 coreV1-22-005\nNOTE: I don\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003et know why it keeps failing the job partition I even tried logging \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eout the core and remained on turing but that was rejected as well. :(\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter multiple attempts!!\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ nano amFullTrimClip.sh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ cat amFullTrimClip.sh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#!/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH -o amFullTrimclip.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH --job-name=amTrimFull\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ sbatch amFullTrimClip.sh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSubmitted batch job 9270966\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R       0:03      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R       0:22      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R       1:17      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R       2:40      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R       3:25      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eadapterlist_advbioinf.txt   RI_B_06_18.fastq   VA_B_06_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eamFullTrimClip.sh           RI_B_07_14.fastq   VA_B_07_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eamFullTrimclip.txt          RI_B_07_18.fastq   VA_B_07_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAMFullTrimclip.txt          RI_B_07_22.fastq   VA_B_07_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eareejsrenamer.sh            RI_W_06_18.fastq   VA_W_06_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAreejsRenamer.sh            RI_W_07_14.fastq   VA_W_07_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eareejsrenamer.txt           RI_W_07_18.fastq   VA_W_07_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eFullTrimClip.sh             RI_W_07_22.fastq   VA_W_07_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003erenamingtable_complete.txt  Testing\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_06_18_clipped.fastq    trimclipstats.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R       5:20      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R       9:01      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 fastq]$ squeue -u amali010\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e           9270966      main amTrimFu amali010  R      17:31      1 coreV1-22-005\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#14 - Push your notebook file to your github page\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit add README.md\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit commit -m \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003eupdating readme\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit push -u origin main\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 04 29-Jan-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eHomework day04\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Checked the success of the job submission from the day before\n[amali010@turing1 areej]$ cd data\n[amali010@turing1 data]$ cd fastq\n[amali010@turing1 fastq]$ ls\n[amali010@turing1 fastq]$ ls filteringstats/\n[amali010@turing1 fastq]$ ls -alh filteringstats/\n\n#1 - Add your trimclipstats.txt output to the full class \n\n#1a - Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps\n[amali010@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\n\nThis script takes a stats output file from fastx_clipper and converts it into a table.\n\nUsage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt\n\nOptions (-c and -v must be listed separately to run together):\n-h      Display this help message\n-c      Use comma delimiter instead of tabs\n-v      Verbose mode (print steps to stdout)\n\n#1b - Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt\n[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt AREEJ_trimclipstatsout.txt\n\n#1c -  Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n[amali010@turing1 fastq]$ tail -n +2 AREEJ_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\n#2 - Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)\nOk!\n\n#3 - Write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data\n[amali010@turing1 fastq]$ cd QCFastqs\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ nano ajbowtie2.sh\n[amali010@turing1 QCFastqs]$ ls\najbowtie2.sh                     VA_B_06_18_clippedtrimmed.fastq\nRI_B_06_18_clippedtrimmed.fastq  VA_B_07_14_clippedtrimmed.fastq\nRI_B_07_14_clippedtrimmed.fastq  VA_B_07_18_clippedtrimmed.fastq\nRI_B_07_18_clippedtrimmed.fastq  VA_B_07_22_clippedtrimmed.fastq\nRI_B_07_22_clippedtrimmed.fastq  VA_W_06_18_clippedtrimmed.fastq\nRI_W_06_18_clippedtrimmed.fastq  VA_W_07_14_clippedtrimmed.fastq\nRI_W_07_14_clippedtrimmed.fastq  VA_W_07_18_clippedtrimmed.fastq\nRI_W_07_18_clippedtrimmed.fastq  VA_W_07_22_clippedtrimmed.fastq\nRI_W_07_22_clippedtrimmed.fastq\n[amali010@turing1 QCFastqs]$ cat ajbowtie2.sh\n#!/bin/bash -l\n\n#SBATCH -o ambowtie2.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=ambowtie\n\nmodule load bowtie2/2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \\\n--rg SM:${i%_clippedtrimmed.fastq} \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \\\n\u0026gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done\n\n[amali010@turing1 QCFastqs]$ sbatch ajbowtie2.sh\nSubmitted batch job 9271117\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271117      main ambowtie amali010 PD       0:00      1 (Priority)\n  [amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271117      main ambowtie amali010  R       1:49      1 coreV2-25-049\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271117      main ambowtie amali010  R       5:07      1 coreV2-25-049\n\n#4 - Submit and add everything to your logfile\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003eChecked the success of the job submission from the day before\n[amali010@turing1 areej]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data\n[amali010@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e fastq\n[amali010@turing1 fastq]$ ls\n[amali010@turing1 fastq]$ ls filteringstats/\n[amali010@turing1 fastq]$ ls -alh filteringstats/\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Add your trimclipstats.txt output to the full class \u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1a - Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps\u003c/span\u003e\n[amali010@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq\n[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\n\nThis script takes a stats output file from fastx_clipper and converts it into a table.\n\nUsage: Schafran_trimstatstable.py [-c, -v, \u003cspan class=\"pl-k\"\u003e-h\u003c/span\u003e] inputfile.txt outputfile.txt\n\nOptions (-c and -v must be listed separately to run together):\n-h      Display this \u003cspan class=\"pl-c1\"\u003ehelp\u003c/span\u003e message\n-c      Use comma delimiter instead of tabs\n-v      Verbose mode (print steps to stdout)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1b - Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt\u003c/span\u003e\n[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt AREEJ_trimclipstatsout.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1c -  Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\u003c/span\u003e\n[amali010@turing1 fastq]$ tail -n +2 AREEJ_trimclipstatsout.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)\u003c/span\u003e\nOk\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data\u003c/span\u003e\n[amali010@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e QCFastqs\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ nano ajbowtie2.sh\n[amali010@turing1 QCFastqs]$ ls\najbowtie2.sh                     VA_B_06_18_clippedtrimmed.fastq\nRI_B_06_18_clippedtrimmed.fastq  VA_B_07_14_clippedtrimmed.fastq\nRI_B_07_14_clippedtrimmed.fastq  VA_B_07_18_clippedtrimmed.fastq\nRI_B_07_18_clippedtrimmed.fastq  VA_B_07_22_clippedtrimmed.fastq\nRI_B_07_22_clippedtrimmed.fastq  VA_W_06_18_clippedtrimmed.fastq\nRI_W_06_18_clippedtrimmed.fastq  VA_W_07_14_clippedtrimmed.fastq\nRI_W_07_14_clippedtrimmed.fastq  VA_W_07_18_clippedtrimmed.fastq\nRI_W_07_18_clippedtrimmed.fastq  VA_W_07_22_clippedtrimmed.fastq\nRI_W_07_22_clippedtrimmed.fastq\n[amali010@turing1 QCFastqs]$ cat ajbowtie2.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o ambowtie2.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=ambowtie\u003c/span\u003e\n\nmodule load bowtie2/2.4\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_clippedtrimmed.fastq\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e bowtie2 --rg-id \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmed.fastq}\u003c/span\u003e \\\n--rg SM:\u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmed.fastq}\u003c/span\u003e \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmedfilterd.fastq}\u003c/span\u003e.sam --no-unal -k 5\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n[amali010@turing1 QCFastqs]$ sbatch ajbowtie2.sh\nSubmitted batch job 9271117\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271117      main ambowtie amali010 PD       0:00      1 (Priority)\n  [amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271117      main ambowtie amali010  R       1:49      1 coreV2-25-049\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271117      main ambowtie amali010  R       5:07      1 coreV2-25-049\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Submit and add everything to your logfile\u003c/span\u003e\ngit add README.md\ngit commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\ngit push -u origin main\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 05 03-Feb-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day05\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 - Team up with a partner for this one and work only on a combined set of data for your two lanes of sequences\nPartner: Andrew\nCombined set of data lanes: 1, 6, \u0026amp; 7\n\n#2 - Run the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together\nDone!\n\n#3 - Modify the below sbatch script (note the differences in the header compared to the previous one)\nOriginal script (used from homework_day05.txt file)\n\nEdited script\nNew name: APAMAssemble.sh\nLocation: /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/\n#!/bin/bash -l\n\n#SBATCH -o APAMAssemble.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APAMAssemble\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\n\n#4 - Check https://trinityrnaseq.github.io/ for usage info\nDone!\n\n#5 - Submit your trinity script\nNote: It was done by Andrew, so the following is from his notebook.\n[apear012@turing1 apam]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam\n[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh \nSubmitted batch job 9272358\n\n[amali010@turing1 apam]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n[amali010@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)\n           9272334      main       sh apear012  R    1:19:50      1 coreV2-25-002\n           9272357      main       sh apear012  R       6:49      1 coreV2-22-007\n[amali010@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)\n           9272334      main       sh apear012  R    1:20:18      1 coreV2-25-002\n           9272357      main       sh apear012  R       7:17      1 coreV2-22-007\nNote: Job failed after approximately 15 minutes :(\n\nEdited APAMAssemble.sh file (removed --normalize_reads). This was also done by Andrew.\n#!/bin/bash -l\n\n#SBATCH -o APAMAssemble.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=pearsoac@evms.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=APAMAssemble\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\n\n[apear012@turing1 apam]$ sbatch APAMAssemble.sh\nSubmitted batch job 9272385\n[apear012@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 \n           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003\n           \n#6 - Submit and add everything to your logfile\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Team up with a partner for this one and work only on a combined set of data for your two lanes of sequences\u003c/span\u003e\nPartner: Andrew\nCombined \u003cspan class=\"pl-c1\"\u003eset\u003c/span\u003e of data lanes: 1, 6, \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e 7\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - Run the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together\u003c/span\u003e\nDone\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Modify the below sbatch script (note the differences in the header compared to the previous one)\u003c/span\u003e\nOriginal script (used from homework_day05.txt file)\n\nEdited script\nNew name: APAMAssemble.sh\nLocation: /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o APAMAssemble.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 32\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -p himem\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=pearsoac@evms.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=APAMAssemble\u003c/span\u003e\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Check https://trinityrnaseq.github.io/ for usage info\u003c/span\u003e\nDone\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5 - Submit your trinity script\u003c/span\u003e\nNote: It was \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e by Andrew, so the following is from his notebook.\n[apear012@turing1 apam]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam\n[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh \nSubmitted batch job 9272358\n\n[amali010@turing1 apam]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n[amali010@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)\n           9272334      main       sh apear012  R    1:19:50      1 coreV2-25-002\n           9272357      main       sh apear012  R       6:49      1 coreV2-22-007\n[amali010@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)\n           9272334      main       sh apear012  R    1:20:18      1 coreV2-25-002\n           9272357      main       sh apear012  R       7:17      1 coreV2-22-007\nNote: Job failed after approximately 15 minutes :(\n\nEdited APAMAssemble.sh file (removed --normalize_reads). This was also \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e by Andrew.\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o APAMAssemble.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 32\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -p himem\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=pearsoac@evms.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=APAMAssemble\u003c/span\u003e\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32\n\n[apear012@turing1 apam]$ sbatch APAMAssemble.sh\nSubmitted batch job 9272385\n[apear012@turing1 apam]$ squeue -u apear012\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 \n           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003\n           \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6 - Submit and add everything to your logfile\u003c/span\u003e\ngit add README.md\ngit commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\ngit push -u origin main\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 05 03-Feb-2021\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"No homework\"\u003e\u003cpre\u003eNo homework\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 06 05-Feb-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day06\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 - Start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly\n[amali010@turing1 djbtestassembly]$ salloc\nsalloc: Pending job allocation 9272860\nsalloc: job 9272860 queued and waiting for resources\nsalloc: job 9272860 has been allocated resources\nsalloc: Granted job allocation 9272860\n\n[amali010@turing1 djbtestassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/djbtestassembly\n[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta\n\n\n################################\n## Counts of transcripts, etc.\n################################\nTotal trinity 'genes':  20980\nTotal trinity transcripts:      21992\nPercent GC: 46.21\n\n########################################\nStats based on ALL transcript contigs:\n########################################\n\n        Contig N10: 1178\n        Contig N20: 694\n        Contig N30: 514\n        Contig N40: 414\n        Contig N50: 347\n\n        Median contig length: 273\n        Average contig: 356.95\n        Total assembled bases: 7850006\n\n\n#####################################################\n## Stats based on ONLY LONGEST ISOFORM per 'GENE':\n#####################################################\n\n        Contig N10: 1027\n        Contig N20: 643\n        Contig N30: 486\n        Contig N40: 397\n        Contig N50: 336\n\n        Median contig length: 271\n        Average contig: 347.72\n        Total assembled bases: 7295061\n\n#2 - Compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile\n[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\nThe total number of sequences is 15079\nThe average sequence length is 876\nThe total number of bases is 13210470\nThe minimum sequence length is 500\nThe maximum sequence length is 10795\nThe N50 is 881\nMedian Length = 578\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 15079\ncontigs \u0026gt;= 1000bp = 3660\ncontigs \u0026gt;= 2000bp = 536\n\n#3 - Less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:\n\n#3a - The mean percent \u0026quot;overall alignment rate\u0026quot;\n[amali010@coreV2-22-007 QCFastqs]$ less ambowtie2.txt\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV2-22-007 QCFastqs]$ grep \u0026quot;overall alignment rate\u0026quot; ambowtie2.txt\n1.87% overall alignment rate\n1.53% overall alignment rate\n1.46% overall alignment rate\n1.92% overall alignment rate\n1.65% overall alignment rate\n1.48% overall alignment rate\n1.25% overall alignment rate\n1.60% overall alignment rate\n1.94% overall alignment rate\n2.00% overall alignment rate\n1.67% overall alignment rate\n2.09% overall alignment rate\n1.62% overall alignment rate\n1.54% overall alignment rate\n1.70% overall alignment rate\n1.41% overall alignment rate\n\n[amali010@coreV2-22-007 QCFastqs]$ grep \u0026quot;overall alignment rate\u0026quot; ambowtie2.txt | cut -d \u0026quot;%\u0026quot; -f 1\n1.87\n1.53\n1.46\n1.92\n1.65\n1.48\n1.25\n1.60\n1.94\n2.00\n1.67\n2.09\n1.62\n1.54\n1.70\n1.41\n\n#3b - The mean percent reads \u0026quot;aligned exactly 1 time\u0026quot;\n[amali010@coreV2-22-007 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; ambowtie2.txt\n    312173 (1.32%) aligned exactly 1 time\n    279609 (1.10%) aligned exactly 1 time\n    226384 (0.96%) aligned exactly 1 time\n    29903 (1.24%) aligned exactly 1 time\n    210196 (1.09%) aligned exactly 1 time\n    359056 (0.96%) aligned exactly 1 time\n    225886 (0.88%) aligned exactly 1 time\n    183780 (1.06%) aligned exactly 1 time\n    207948 (1.27%) aligned exactly 1 time\n    228135 (1.33%) aligned exactly 1 time\n    250518 (1.09%) aligned exactly 1 time\n    195159 (1.31%) aligned exactly 1 time\n    195302 (1.13%) aligned exactly 1 time\n    234155 (1.07%) aligned exactly 1 time\n    176134 (0.87%) aligned exactly 1 time\n    243143 (0.89%) aligned exactly 1 time\n\n[amali010@coreV2-22-007 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; ambowtie2.txt | cut -d \u0026quot;%\u0026quot; -f 1 | cut -d \u0026quot;(\u0026quot; -f 2\n1.32\n1.10\n0.96\n1.24\n1.09\n0.96\n0.88\n1.06\n1.27\n1.33\n1.09\n1.31\n1.13\n1.07\n0.87\n0.89\n\n#3c - The mean number of reads \u0026quot;aligned exactly 1 time\u0026quot;\n[amali010@turing1 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; ambowtie2.txt\n    312173 (1.32%) aligned exactly 1 time\n    279609 (1.10%) aligned exactly 1 time\n    226384 (0.96%) aligned exactly 1 time\n    29903 (1.24%) aligned exactly 1 time\n    210196 (1.09%) aligned exactly 1 time\n    359056 (0.96%) aligned exactly 1 time\n    225886 (0.88%) aligned exactly 1 time\n    183780 (1.06%) aligned exactly 1 time\n    207948 (1.27%) aligned exactly 1 time\n    228135 (1.33%) aligned exactly 1 time\n    250518 (1.09%) aligned exactly 1 time\n    195159 (1.31%) aligned exactly 1 time\n    195302 (1.13%) aligned exactly 1 time\n    234155 (1.07%) aligned exactly 1 time\n    176134 (0.87%) aligned exactly 1 time\n    243143 (0.89%) aligned exactly 1 time\n    \n    [amali010@coreV2-22-007 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; ambowtie2.txt | cut -d \u0026quot; \u0026quot; -f 5\n312173\n279609\n226384\n29903\n210196\n359056\n225886\n183780\n207948\n228135\n250518\n195159\n195302\n234155\n176134\n243143\n\n#3d - The mean percent reads \u0026quot;aligned \u0026gt;1 times\u0026quot;\n[amali010@coreV2-22-007 QCFastqs]$ grep \u0026quot;aligned \u0026gt;1 times\u0026quot; ambowtie2.txt\n    131960 (0.56%) aligned \u0026gt;1 times\n    109989 (0.43%) aligned \u0026gt;1 times\n    119408 (0.50%) aligned \u0026gt;1 times\n    16432 (0.68%) aligned \u0026gt;1 times\n    107800 (0.56%) aligned \u0026gt;1 times\n    194361 (0.52%) aligned \u0026gt;1 times\n    95504 (0.37%) aligned \u0026gt;1 times\n    93466 (0.54%) aligned \u0026gt;1 times\n    110827 (0.68%) aligned \u0026gt;1 times\n    115424 (0.67%) aligned \u0026gt;1 times\n    131960 (0.58%) aligned \u0026gt;1 times\n    116526 (0.78%) aligned \u0026gt;1 times\n    83427 (0.48%) aligned \u0026gt;1 times\n    103215 (0.47%) aligned \u0026gt;1 times\n    169138 (0.83%) aligned \u0026gt;1 times\n    140557 (0.52%) aligned \u0026gt;1 times\n\n[amali010@coreV2-22-007 QCFastqs]$ grep \u0026quot;aligned \u0026gt;1 times\u0026quot; ambowtie2.txt | cut -d \u0026quot;%\u0026quot; -f 1 | cut -d \u0026quot;(\u0026quot; -f 2\n0.56\n0.43\n0.50\n0.68\n0.56\n0.52\n0.37\n0.54\n0.68\n0.67\n0.58\n0.78\n0.48\n0.47\n0.83\n0.52\nhint use grep and paste into excel\n\n#4 - Add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:\nLaneX_yourinitials\tb-the mean percent \u0026quot;overall alignment rate\u0026quot;\tc-the mean percent reads \u0026quot;aligned exactly 1 time\u0026quot;\td-the mean number of reads \u0026quot;aligned exactly 1 time\u0026quot;\te-the mean percent reads \u0026quot;aligned \u0026gt;1 times\u0026quot;\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV2-22-007 QCFastqs]$ nano alignstatsam.txt\n[amali010@coreV2-22-007 QCFastqs]$ cat alignstatsam.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\n[amali010@coreV2-22-007 QCFastqs]$ tail -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\nLaneX   overallalignmentrate    percsinglyaligned       numsinglyaligned        percmultiplyaligned\nLaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857\nLaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857\nLane1AP 5.378125        1.3725  300361.8125     4.005\nlane5_sg        2.03%   1.11%   235455.75       0.92%\nLaneX_jcw       2.495714286     1.251428571     247656.7143     1.247142857\nLane4_KCrid     2.48    1.30    289072.88       1.18\nLaneX_am 1.670625       1.098125        222342.5625     0.573125\n\n#5 - Data cleanup and archiving:\n#5a - mv your Trinity.fast file from your trinity_out_dir to a folder called testassembly in your data directory\nNote: our job was still running at the time. So to avoid confusion I decided to work with the Trinity.fasta file that was copied into my data directory during class.\n[amali010@turing1 data]$ mkdir testassembly\n[amali010@turing1 data]$ cd djbtestassembly/\n[amali010@turing1 djbtestassembly]$ ls\nTrinity.fasta\n[amali010@turing1 djbtestassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly/\n[amali010@turing1 testassembly]$ ls\nTrinity.fasta\n\n#5b - Set up a sbatch script to:\n\trm -r YOURtrinity_out_dir (don't have one so this will be skipped in the sbatch script)\n\trm -r YOURoriginalfastqs\n\trm -r YOURfilterstats # as long as you've already appended your filteringstats output to the class as part of homework_day04.txt\noriginal script (used from homework_ day06.txt file)\n#!/bin/bash -l\n\n#SBATCH -o OUTFILENAME.txt\n#SBATCH -n NUMTHREADS(1 if not multi-threaded)\n#SBATCH --mail-user=YOUREMAIL@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/QCFastqs/21sampleassembly_trinity_out_dir\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/filteringstats\n\n[amali010@turing1 originalfastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs\n\n[amali010@turing1 filteringstats]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats\n\n[amali010@turing1 fastq]$ nano AMCleanup.sh\n[amali010@turing1 fastq]$ head AMCleanup.sh\n#!/bin/bash -l\n\n#SBATCH -o AMCleanup.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMCleanup\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats\n\n[amali010@turing1 fastq]$ sbatch AMCleanup.sh\nSubmitted batch job 9273090\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273090      main AMCleanu amali010  R       0:01      1 coreV2-22-007\n\n#6 - Submit a blast against sprot from your testassembly folder using the following command:\n#!/bin/bash -l\n\n#SBATCH -o OUTFILENAME.txt\n#SBATCH -n 6         \n#SBATCH --mail-user=EMAIL\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n\n[amali010@turing1 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly\n[amali010@turing1 testassembly]$ nano AMBlast.sh\n[amali010@turing1 testassembly]$ cat AMBlast.sh\n#!/bin/bash -l\n\n#SBATCH -o AMBlast.txt\n#SBATCH -n 6\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMBlast\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n[amali010@turing1 testassembly]$ sbatch AMBlast.sh\nSubmitted batch job 9273094\n[amali010@turing1 testassembly]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R       0:03      1 coreV2-22-028\n[amali010@turing1 areej]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R      58:38      1 coreV2-22-028\n\n#7 - Head one of your .sam files to look at the header\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ head RI_B_06_18_clippedtrimmed.fastq.sam\n@HD     VN:1.0  SO:unsorted\n@SQ     SN:TR78|c0_g1_i1_coral  LN:1109\n@SQ     SN:TR78|c0_g2_i1_coral  LN:1109\n@SQ     SN:TR79|c0_g1_i1_coral  LN:610\n@SQ     SN:TR79|c0_g2_i1_coral  LN:1549\n@SQ     SN:TR87|c0_g1_i1_coral  LN:732\n@SQ     SN:TR93|c0_g1_i1_coral  LN:550\n@SQ     SN:TR101|c0_g1_i1_coral LN:673\n@SQ     SN:TR104|c0_g1_i1_coral LN:607\n@SQ     SN:TR105|c0_g1_i1_coral LN:587\n\n#8 - grep -v '@' your.sam | head to look at the sequence read lines, \nwhy does this work to exclude the header lines?\n[amali010@turing1 QCFastqs]$ grep -v \u0026quot;@\u0026quot; RI_B_06_18_clippedtrimmed.fastq.sam | head\nK00188:59:HMTFHBBXX:7:1101:12845:1666   16      TR13647|c0_g1_i1_coral  40      255\n51M     *       0       0       TTTGGATTCTTGATCACGGTATGATCGATAAAACACAGACCTGTAAAGTCC\nJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XN:i:0  XM:i:0  XO:i:0      XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:17645:1683   16      TR43786|c0_g1_i16_coral 125     1  51M      *       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC\nJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:17645:1683   272     TR43786|c0_g1_i5_coral  151     255\n51M     *       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC\nJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:20750:1683   0       TR24696|c0_g1_i2_coral  329     1  51M      *       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC\nAAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:20750:1683   256     TR24696|c0_g1_i3_coral  329     255\n51M     *       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC\nAAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:22658:1683   16      TR43764|c4_g1_i1_coral  1687    255\n51M     *       0       0       TCCATTCCTAATCCGTCTGTTAGTGGTCTGCAGTTCCACGCCGTCTTTTCG\nJJJJJJJFJJFJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJFJFFAAA     AS:i:-17        XN:i:0  XM:i:3      XO:i:0  XG:i:0  NM:i:3  MD:Z:25T7C14G2  YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:27854:1683   16      TR29978|c0_g2_i1_coral  10      255\n51M     *       0       0       GGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAG\nJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-28        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:30G1C12G1C0A2      YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:2392:1701    0       TR35685|c1_g2_i1_coral  534     255\n21M     *       0       0       TGACAACCCCGTCTGTGGCGC   AAFFFFJJJJJFFJJJJJFFJ   AS:i:-6     XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:7A13       YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:19705:1701   16      TR29978|c0_g2_i1_coral  14      255\n51M     *       0       0       TCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAGCGGG\nJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-30        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:26G1C12G1C0A6      YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:23094:1701   16      TR29978|c0_g2_i1_coral  5       255\n51M     *       0       0       CTGGGGGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAA\nJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-29        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:2A0A31G1C12G0      YT:Z:UU RG:Z:RI_B_06_18\n\nHeader lines include '@'. The -v in grep -v excludes '@' and therefore the header lines as well.\n\n#9 - In an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.\n[amali010@turing1 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_06_18_clippedtrimmed.fastq.sam RI_B_07_18_clippedtrimmed.fastq.sam RI_W_07_18_clippedtrimmed.fastq.sam\nRI_B_06_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n        read reverse strand\n        not primary alignment\n256 :\n        not primary alignment\n16 :\n        read reverse strand\nRI_B_07_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n        read reverse strand\n        not primary alignment\n256 :\n        not primary alignment\n16 :\n        read reverse strand\nRI_W_07_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n        read reverse strand\n        not primary alignment\n256 :\n        not primary alignment\n16 :\n        read reverse strand\n\n#10 - we need to run the read sorting step required for SNP calling, \nso if you have time, set up and run the following script on your .sam files to finish before Wednesday\n#!/bin/bash -l\n\n#SBATCH -o OUTFILENAME.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=EMAIL\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\n\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ nano AMReadSort.sh\n[amali010@turing1 QCFastqs]$ cat AMReadSort.sh\n#!/bin/bash -l\n\n#SBATCH -o AMReadSort.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMReadSort\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\n\ndone\n\n[amali010@turing1 QCFastqs]$ sbatch AMReadSort.sh\nSubmitted batch job 9273103\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R    1:21:32      1 coreV2-22-028\n           9273103      main AMReadSo amali010  R       0:04      1 coreV3-23-020\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R    1:22:36      1 coreV2-22-028\n           9273103      main AMReadSo amali010  R       1:08      1 coreV3-23-020\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R    1:25:19      1 coreV2-22-028\n\n#11 - submit and add everything to your logfile\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly\u003c/span\u003e\n[amali010@turing1 djbtestassembly]$ salloc\nsalloc: Pending job allocation 9272860\nsalloc: job 9272860 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9272860 has been allocated resources\nsalloc: Granted job allocation 9272860\n\n[amali010@turing1 djbtestassembly]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/djbtestassembly\n[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###############################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Counts of transcripts, etc.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###############################\u003c/span\u003e\nTotal trinity \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egenes\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e:  20980\nTotal trinity transcripts:      21992\nPercent GC: 46.21\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#######################################\u003c/span\u003e\nStats based on ALL transcript contigs:\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#######################################\u003c/span\u003e\n\n        Contig N10: 1178\n        Contig N20: 694\n        Contig N30: 514\n        Contig N40: 414\n        Contig N50: 347\n\n        Median contig length: 273\n        Average contig: 356.95\n        Total assembled bases: 7850006\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e####################################################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Stats based on ONLY LONGEST ISOFORM per 'GENE':\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e####################################################\u003c/span\u003e\n\n        Contig N10: 1027\n        Contig N20: 643\n        Contig N30: 486\n        Contig N40: 397\n        Contig N50: 336\n\n        Median contig length: 271\n        Average contig: 347.72\n        Total assembled bases: 7295061\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - Compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile\u003c/span\u003e\n[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\nThe total number of sequences is 15079\nThe average sequence length is 876\nThe total number of bases is 13210470\nThe minimum sequence length is 500\nThe maximum sequence length is 10795\nThe N50 is 881\nMedian Length = 578\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 150bp = 0\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 500bp = 15079\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 1000bp = 3660\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 2000bp = 536\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3a - The mean percent \"overall alignment rate\"\u003c/span\u003e\n[amali010@coreV2-22-007 QCFastqs]$ less ambowtie2.txt\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV2-22-007 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt\n1.87% overall alignment rate\n1.53% overall alignment rate\n1.46% overall alignment rate\n1.92% overall alignment rate\n1.65% overall alignment rate\n1.48% overall alignment rate\n1.25% overall alignment rate\n1.60% overall alignment rate\n1.94% overall alignment rate\n2.00% overall alignment rate\n1.67% overall alignment rate\n2.09% overall alignment rate\n1.62% overall alignment rate\n1.54% overall alignment rate\n1.70% overall alignment rate\n1.41% overall alignment rate\n\n[amali010@coreV2-22-007 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1\n1.87\n1.53\n1.46\n1.92\n1.65\n1.48\n1.25\n1.60\n1.94\n2.00\n1.67\n2.09\n1.62\n1.54\n1.70\n1.41\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3b - The mean percent reads \"aligned exactly 1 time\"\u003c/span\u003e\n[amali010@coreV2-22-007 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt\n    312173 (1.32%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    279609 (1.10%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    226384 (0.96%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    29903 (1.24%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    210196 (1.09%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    359056 (0.96%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    225886 (0.88%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    183780 (1.06%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    207948 (1.27%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    228135 (1.33%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    250518 (1.09%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    195159 (1.31%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    195302 (1.13%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    234155 (1.07%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    176134 (0.87%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    243143 (0.89%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n\n[amali010@coreV2-22-007 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e(\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 2\n1.32\n1.10\n0.96\n1.24\n1.09\n0.96\n0.88\n1.06\n1.27\n1.33\n1.09\n1.31\n1.13\n1.07\n0.87\n0.89\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3c - The mean number of reads \"aligned exactly 1 time\"\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt\n    312173 (1.32%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    279609 (1.10%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    226384 (0.96%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    29903 (1.24%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    210196 (1.09%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    359056 (0.96%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    225886 (0.88%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    183780 (1.06%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    207948 (1.27%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    228135 (1.33%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    250518 (1.09%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    195159 (1.31%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    195302 (1.13%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    234155 (1.07%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    176134 (0.87%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    243143 (0.89%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    \n    [amali010@coreV2-22-007 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 5\n312173\n279609\n226384\n29903\n210196\n359056\n225886\n183780\n207948\n228135\n250518\n195159\n195302\n234155\n176134\n243143\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3d - The mean percent reads \"aligned \u0026gt;1 times\"\u003c/span\u003e\n[amali010@coreV2-22-007 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt\n    131960 (0.56%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    109989 (0.43%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    119408 (0.50%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    16432 (0.68%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    107800 (0.56%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    194361 (0.52%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    95504 (0.37%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    93466 (0.54%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    110827 (0.68%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    115424 (0.67%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    131960 (0.58%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    116526 (0.78%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    83427 (0.48%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    103215 (0.47%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    169138 (0.83%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    140557 (0.52%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n\n[amali010@coreV2-22-007 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ambowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e(\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 2\n0.56\n0.43\n0.50\n0.68\n0.56\n0.52\n0.37\n0.54\n0.68\n0.67\n0.58\n0.78\n0.48\n0.47\n0.83\n0.52\nhint use grep and paste into excel\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:\u003c/span\u003e\nLaneX_yourinitials\tb-the mean percent \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\tc-the mean percent reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\td-the mean number of reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\te-the mean percent reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV2-22-007 QCFastqs]$ nano alignstatsam.txt\n[amali010@coreV2-22-007 QCFastqs]$ cat alignstatsam.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\n[amali010@coreV2-22-007 QCFastqs]$ tail -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\nLaneX   overallalignmentrate    percsinglyaligned       numsinglyaligned        percmultiplyaligned\nLaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857\nLaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857\nLane1AP 5.378125        1.3725  300361.8125     4.005\nlane5_sg        2.03%   1.11%   235455.75       0.92%\nLaneX_jcw       2.495714286     1.251428571     247656.7143     1.247142857\nLane4_KCrid     2.48    1.30    289072.88       1.18\nLaneX_am 1.670625       1.098125        222342.5625     0.573125\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5 - Data cleanup and archiving:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5a - mv your Trinity.fast file from your trinity_out_dir to a folder called testassembly in your data directory\u003c/span\u003e\nNote: our job was still running at the time. So to avoid confusion I decided to work with the Trinity.fasta file that was copied into my data directory during class.\n[amali010@turing1 data]$ mkdir testassembly\n[amali010@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e djbtestassembly/\n[amali010@turing1 djbtestassembly]$ ls\nTrinity.fasta\n[amali010@turing1 djbtestassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly/\n[amali010@turing1 testassembly]$ ls\nTrinity.fasta\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5b - Set up a sbatch script to:\u003c/span\u003e\n\trm -r YOURtrinity_out_dir (don\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003et have one so this will be skipped in the sbatch script)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\trm -r YOURoriginalfastqs\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\trm -r YOURfilterstats # as long as you\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003eve already appended your filteringstats output to the class as part of homework_day04.txt\noriginal script (used from homework_ day06.txt file)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o OUTFILENAME.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n NUMTHREADS(1 if not multi-threaded)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=YOUREMAIL@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=JOBNAME\u003c/span\u003e\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/QCFastqs/21sampleassembly_trinity_out_dir\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/filteringstats\n\n[amali010@turing1 originalfastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs\n\n[amali010@turing1 filteringstats]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats\n\n[amali010@turing1 fastq]$ nano AMCleanup.sh\n[amali010@turing1 fastq]$ head AMCleanup.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMCleanup.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMCleanup\u003c/span\u003e\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats\n\n[amali010@turing1 fastq]$ sbatch AMCleanup.sh\nSubmitted batch job 9273090\n[amali010@turing1 fastq]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273090      main AMCleanu amali010  R       0:01      1 coreV2-22-007\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6 - Submit a blast against sprot from your testassembly folder using the following command:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o OUTFILENAME.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 6         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=EMAIL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=JOBNAME\u003c/span\u003e\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n\n[amali010@turing1 testassembly]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly\n[amali010@turing1 testassembly]$ nano AMBlast.sh\n[amali010@turing1 testassembly]$ cat AMBlast.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMBlast.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 6\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMBlast\u003c/span\u003e\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n[amali010@turing1 testassembly]$ sbatch AMBlast.sh\nSubmitted batch job 9273094\n[amali010@turing1 testassembly]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R       0:03      1 coreV2-22-028\n[amali010@turing1 areej]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R      58:38      1 coreV2-22-028\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e7 - Head one of your .sam files to look at the header\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ head RI_B_06_18_clippedtrimmed.fastq.sam\n@HD     VN:1.0  SO:unsorted\n@SQ     SN:TR78\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  LN:1109\n@SQ     SN:TR78\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  LN:1109\n@SQ     SN:TR79\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  LN:610\n@SQ     SN:TR79\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  LN:1549\n@SQ     SN:TR87\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  LN:732\n@SQ     SN:TR93\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  LN:550\n@SQ     SN:TR101\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral LN:673\n@SQ     SN:TR104\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral LN:607\n@SQ     SN:TR105\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral LN:587\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e8 - grep -v '@' your.sam | head to look at the sequence read lines, \u003c/span\u003e\nwhy does this work to exclude the header lines\u003cspan class=\"pl-k\"\u003e?\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e@\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e RI_B_06_18_clippedtrimmed.fastq.sam \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e head\nK00188:59:HMTFHBBXX:7:1101:12845:1666   16      TR13647\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  40      255\n51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       TTTGGATTCTTGATCACGGTATGATCGATAAAACACAGACCTGTAAAGTCC\nJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XN:i:0  XM:i:0  XO:i:0      XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:17645:1683   16      TR43786\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i16_coral 125     1  51M      \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC\nJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:17645:1683   272     TR43786\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i5_coral  151     255\n51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC\nJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:20750:1683   0       TR24696\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  329     1  51M      \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC\nAAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:20750:1683   256     TR24696\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i3_coral  329     255\n51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC\nAAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:22658:1683   16      TR43764\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec4_g1_i1_coral  1687    255\n51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       TCCATTCCTAATCCGTCTGTTAGTGGTCTGCAGTTCCACGCCGTCTTTTCG\nJJJJJJJFJJFJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJFJFFAAA     AS:i:-17        XN:i:0  XM:i:3      XO:i:0  XG:i:0  NM:i:3  MD:Z:25T7C14G2  YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:27854:1683   16      TR29978\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  10      255\n51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       GGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAG\nJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-28        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:30G1C12G1C0A2      YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:2392:1701    0       TR35685\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g2_i1_coral  534     255\n21M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       TGACAACCCCGTCTGTGGCGC   AAFFFFJJJJJFFJJJJJFFJ   AS:i:-6     XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:7A13       YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:19705:1701   16      TR29978\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  14      255\n51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       TCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAGCGGG\nJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-30        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:26G1C12G1C0A6      YT:Z:UU RG:Z:RI_B_06_18\nK00188:59:HMTFHBBXX:7:1101:23094:1701   16      TR29978\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  5       255\n51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       CTGGGGGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAA\nJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-29        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:2A0A31G1C12G0      YT:Z:UU RG:Z:RI_B_06_18\n\nHeader lines include \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e@\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e. The -v \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e grep -v excludes \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e@\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e and therefore the header lines as well.\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e9 - In an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_06_18_clippedtrimmed.fastq.sam RI_B_07_18_clippedtrimmed.fastq.sam RI_W_07_18_clippedtrimmed.fastq.sam\nRI_B_06_18_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e272\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e256\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n272 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n        not primary alignment\n256 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        not primary alignment\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\nRI_B_07_18_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e272\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e256\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n272 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n        not primary alignment\n256 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        not primary alignment\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\nRI_W_07_18_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e272\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e256\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n272 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n        not primary alignment\n256 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        not primary alignment\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e10 - we need to run the read sorting step required for SNP calling, \u003c/span\u003e\nso \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e you have time, \u003cspan class=\"pl-c1\"\u003eset\u003c/span\u003e up and run the following script on your .sam files to finish before Wednesday\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o OUTFILENAME.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=EMAIL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=JOBNAME\u003c/span\u003e\n\nenable_lmod\nmodule load samtools/1\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esamtools view -bS \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.sam}\u003c/span\u003e_UNSORTED.bam\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e samtools sort \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\nsamtools index \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ nano AMReadSort.sh\n[amali010@turing1 QCFastqs]$ cat AMReadSort.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMReadSort.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMReadSort\u003c/span\u003e\n\nenable_lmod\nmodule load samtools/1\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esamtools view -bS \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.sam}\u003c/span\u003e_UNSORTED.bam\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e samtools sort \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\nsamtools index \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n[amali010@turing1 QCFastqs]$ sbatch AMReadSort.sh\nSubmitted batch job 9273103\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R    1:21:32      1 coreV2-22-028\n           9273103      main AMReadSo amali010  R       0:04      1 coreV3-23-020\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R    1:22:36      1 coreV2-22-028\n           9273103      main AMReadSo amali010  R       1:08      1 coreV3-23-020\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9273094      main  AMBlast amali010  R    1:25:19      1 coreV2-22-028\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e11 - submit and add everything to your logfile\u003c/span\u003e\ngit add README.md\ngit commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\ngit push -u origin main\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 07 10-Feb-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day07\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 - Run the following command on your sprot output file to process into the contig length/match format that trinity examines\n#!/bin/bash -l\n\n#SBATCH -o OUTFILE.TXT\n#SBATCH -n 1\n#SBATCH --mail-user=YOUREMAIL.odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=JOBNAME\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n\n[amali010@turing1 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly\n[amali010@turing1 testassembly]$ cat AMBlastparse.sh\n#!/bin/bash -l\n\n#SBATCH -o AMBlastparse.TXT\n#SBATCH -n 1\n#SBATCH --mail-user=amali010.odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMBlastparse\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n[amali010@turing1 testassembly]$ sbatch AMBlastparse.sh\nSubmitted batch job 9276474\n[amali010@turing1 testassembly]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276474      main AMBlastp amali010 PD       0:00      1 (Priority)\n[amali010@turing1 testassembly]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276474      main AMBlastp amali010  R       0:14      1 coreV3-23-040\n\n[amali010@turing1 testassembly]$ cat blastx.outfmt6.hist\n#hit_pct_cov_bin        count_in_bin    \u0026gt;bin_below\n100     143     143\n90      52      195\n80      77      272\n70      87      359\n60      110     469\n50      133     602\n40      219     821\n30      395     1216\n20      726     1942\n10      674     2616\n\n#2 - rm UNSORTED.bam's from your QCFastqs directory (or wherever your .bams and .sams are)\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ cat AMrmunsortbam.sh\n#!/bin/bash -l\n\n#SBATCH -o AMrmunsortbam.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMrmunsortbam\n\nrm *UNSORTED.bam\n[amali010@turing1 QCFastqs]$ sbatch AMrmunsortbam.sh\nSubmitted batch job 9276485\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276485      main AMrmunso amali010  R       0:01      1 coreV3-23-046\n\n#2a - Look at one of your files using sam tools\n[amali010@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9276494\n[amali010@coreV3-23-031 QCFastqs]$ enable_lmod\n[amali010@coreV3-23-031 QCFastqs]$ module load samtools\n[amali010@coreV3-23-031 QCFastqs]$ samtools tview RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\n1         11        21        31        41        51        61        71\nTCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT\n                                                        ........................\n                                                        ........................\n\n[amali010@coreV3-23-031 QCFastqs]$ samtools tview -p 'TR78|c0_g2_i1_coral' RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\n1         11        21        31        41        51        61        71\nTCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT\n                                                        ........................\n                                                        ........................\n\n#3 - Run the following to start genotyping your SNPs for filtering next week\n[amali010@coreV3-23-031 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV3-23-031 QCFastqs]$ cat AMfreebayessubref.sh\n#!/bin/bash -l\n\n#SBATCH -o AMfreebayessubref.txt\n#SBATCH -n 1\n#SBATCH --mail-user=amali010.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=AMfreebayessubref\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.fastq.bam \u0026gt; AMmergedfastqs.vcf\n[amali010@coreV3-23-031 QCFastqs]$ ls *.fastq.bam\nRI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam\nRI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam\nRI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam\nRI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam\nRI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam\nRI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam\nRI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam\nRI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam\n[amali010@coreV3-23-031 QCFastqs]$ ls *.bam\nRI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam\nRI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam\nRI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam\nRI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam\nRI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam\nRI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam\nRI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam\nRI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam\n[amali010@coreV3-23-031 QCFastqs]$ sbatch AMfreebayessubref.sh\nSubmitted batch job 9276571\n[amali010@coreV3-23-031 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276571      main AMfreeba amali010  R       0:02      1 coreV3-23-002\n           9276494      main       sh amali010  R      58:02      1 coreV3-23-031\n\n\n#4 - Submit and add everything to your logfile\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Run the following command on your sprot output file to process into the contig length/match format that trinity examines\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o OUTFILE.TXT\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=YOUREMAIL.odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=JOBNAME\u003c/span\u003e\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n\n[amali010@turing1 testassembly]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly\n[amali010@turing1 testassembly]$ cat AMBlastparse.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMBlastparse.TXT\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010.odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMBlastparse\u003c/span\u003e\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n[amali010@turing1 testassembly]$ sbatch AMBlastparse.sh\nSubmitted batch job 9276474\n[amali010@turing1 testassembly]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276474      main AMBlastp amali010 PD       0:00      1 (Priority)\n[amali010@turing1 testassembly]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276474      main AMBlastp amali010  R       0:14      1 coreV3-23-040\n\n[amali010@turing1 testassembly]$ cat blastx.outfmt6.hist\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ehit_pct_cov_bin        count_in_bin    \u0026gt;bin_below\u003c/span\u003e\n100     143     143\n90      52      195\n80      77      272\n70      87      359\n60      110     469\n50      133     602\n40      219     821\n30      395     1216\n20      726     1942\n10      674     2616\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - rm UNSORTED.bam's from your QCFastqs directory (or wherever your .bams and .sams are)\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ cat AMrmunsortbam.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMrmunsortbam.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMrmunsortbam\u003c/span\u003e\n\nrm \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\n[amali010@turing1 QCFastqs]$ sbatch AMrmunsortbam.sh\nSubmitted batch job 9276485\n[amali010@turing1 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276485      main AMrmunso amali010  R       0:01      1 coreV3-23-046\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2a - Look at one of your files using sam tools\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9276494\n[amali010@coreV3-23-031 QCFastqs]$ enable_lmod\n[amali010@coreV3-23-031 QCFastqs]$ module load samtools\n[amali010@coreV3-23-031 QCFastqs]$ samtools tview RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\n1         11        21        31        41        51        61        71\nTCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT\n                                                        ........................\n                                                        ........................\n\n[amali010@coreV3-23-031 QCFastqs]$ samtools tview -p \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eTR78|c0_g2_i1_coral\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\n1         11        21        31        41        51        61        71\nTCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT\n                                                        ........................\n                                                        ........................\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Run the following to start genotyping your SNPs for filtering next week\u003c/span\u003e\n[amali010@coreV3-23-031 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV3-23-031 QCFastqs]$ cat AMfreebayessubref.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o AMfreebayessubref.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=amali010.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=AMfreebayessubref\u003c/span\u003e\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.bam \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e AMmergedfastqs.vcf\n[amali010@coreV3-23-031 QCFastqs]$ ls \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.bam\nRI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam\nRI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam\nRI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam\nRI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam\nRI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam\nRI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam\nRI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam\nRI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam\n[amali010@coreV3-23-031 QCFastqs]$ ls \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.bam\nRI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam\nRI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam\nRI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam\nRI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam\nRI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam\nRI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam\nRI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam\nRI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam\n[amali010@coreV3-23-031 QCFastqs]$ sbatch AMfreebayessubref.sh\nSubmitted batch job 9276571\n[amali010@coreV3-23-031 QCFastqs]$ squeue -u amali010\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9276571      main AMfreeba amali010  R       0:02      1 coreV3-23-002\n           9276494      main       sh amali010  R      58:02      1 coreV3-23-031\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Submit and add everything to your logfile\u003c/span\u003e\ngit add README.md\ngit commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\ngit push -u origin main\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 08 12-Feb-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day08\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 - Clean up your data directory by:\n#1a - Make a SAMS folder and mv all your .sam files into that directory\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ mkdir ../../SAMS\n[amali010@turing1 QCFastqs]$ ls *.sam\n[amali010@turing1 QCFastqs]$ mv *.sam ../../SAMS/\n\n#1b - Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../BAMS\n[amali010@coreV3-23-003 QCFastqs]$ ls *.bam *.bam.bai\n[amali010@coreV3-23-003 QCFastqs]$ mv *bam *bam.bai ../../BAMS/\n\n#1c - rm your _unfiltered.vcf files if you have any\nI did not do this because I didn't have any.\n\n#1d - rm your .fastq files\n[amali010@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ rm *.fastq\n\n#1e - Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)\nI did this even though I do not have a 'YOURNAMEmergedfastqs.vcf' file from the job that submitted in the previous class.\n[amali010@coreV2-22-013 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../VCF\n\n#2 - Start an interactive session via salloc\n[amali010@coreV2-22-013 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9278251\nsalloc: job 9278251 queued and waiting for resources\nsalloc: job 9278251 has been allocated resources\nsalloc: Granted job allocation 9278251\n\n#3 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder\n[amali010@coreV2-22-013 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./\n[amali010@coreV2-22-013 VCF]$ ls -alh\ntotal 2.1G\ndrwxrwxrwx 2 amali010 users   54 Feb 12 12:59 .\ndrwxrwxrwx 9 amali010 users  316 Feb 12 12:56 ..\n-rwxr-xr-x 1 amali010 users 1.5G Feb 12 12:59 mergedfastq_HEAAstrangiaAssembly.vcf\n\n\n#4 - Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nI did it with the class vcf file since I didn't have one of my own.\n\n[amali010@coreV2-22-013 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n[amali010@coreV2-22-013 VCF]$ enable_lmod\n[amali010@coreV2-22-013 VCF]$ module load dDocent\n[amali010@coreV2-22-013 VCF]$ vcftools\n\nVCFtools (0.1.14)\n© Adam Auton and Anthony Marcketta 2009\n\nProcess Variant Call Format files\n\nFor a list of options, please go to:\n        https://vcftools.github.io/man_latest.html\n\nAlternatively, a man page is available, type:\n        man vcftools\n\nQuestions, comments, and suggestions should be emailed to:\n        vcftools-help@lists.sourceforge.net\n\n[amali010@coreV2-22-013 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 14.00 seconds\n\n#5 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files\n[amali010@coreV2-22-013 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt  ./\n\n#6 - Run our host vcf extractor on your merged vcf file using the following syntax:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\n[amali010@coreV2-22-013 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\nRead in ContigList\n\n#7 - Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf\n\nI did not have a .vcf file of my own for comparison. So I compared the other two files in the parenthesis.\n\n[amali010@turing1 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\nvcftools: Command not found.\n[amali010@turing1 VCF]$ enable_lmod\n[amali010@turing1 VCF]$ module load dDocent\n[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 11.00 seconds\n\n[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 432676 out of a possible 432676 Sites\nRun Time = 6.00 seconds\n\n#8 - Work through the VCF filtering tutorial until the following step:\n\nNow that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.\n\nvcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm\n[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n        --recode-INFO-all\n        --mac 3\n        --minQ 30\n        --max-missing 0.5\n        --out raw.g5mac3\n        --recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 432676 Sites\nRun Time = 33.00 seconds\n\n[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf raw.g5mac3.recode.vcf\n        --recode-INFO-all\n        --minDP 3\n        --out raw.g5mac3dp3\n        --recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 31.00 seconds\n\n[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf raw.g5mac3dp3.recode.vcf\n        --missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 3.00 seconds\n\n[amali010@turing1 VCF]$ cat out.imiss\nINDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS\nRI_W_06_merged  99853   0       67075   0.671737\nRI_W_07_merged  99853   0       63917   0.640111\nVA_B_03_merged  99853   0       56586   0.566693\nRI_W_02_merged  99853   0       73105   0.732126\nRI_W_04_merged  99853   0       71159   0.712638\nVA_W_09_SNP_clipped     99853   0       19459   0.194876\nRI_B_08_SNP_clipped     99853   0       88110   0.882397\nVA_W_08_SNP_clipped     99853   0       83344   0.834667\nVA_B_08_SNP_clipped     99853   0       94221   0.943597\nVA_W_02_merged  99853   0       71050   0.711546\nVA_B_07_merged  99853   0       62687   0.627793\nRI_B_05_merged  99853   0       43945   0.440097\nVA_W_06_merged  99853   0       65164   0.652599\nVA_W_04_merged  99853   0       50911   0.509859\nVA_W_01_merged  99853   0       67150   0.672489\nVA_B_10_SNP_clipped     99853   0       82831   0.829529\nVA_B_06_merged  99853   0       57724   0.57809\nVA_W_05_merged  99853   0       66297   0.663946\nRI_B_09_SNP_clipped     99853   0       85107   0.852323\nVA_W_10_SNP_clipped     99853   0       83234   0.833565\nRI_W_08_SNP_clipped     99853   0       77802   0.779165\nRI_B_06_merged  99853   0       79766   0.798834\nRI_W_10_SNP_clipped     99853   0       91323   0.914574\nRI_B_04_merged  99853   0       46717   0.467858\nVA_W_03_merged  99853   0       64016   0.641102\nRI_B_07_merged  99853   0       70798   0.709022\nRI_W_05_merged  99853   0       57117   0.572011\nRI_W_09_SNP_clipped     99853   0       88492   0.886223\nVA_B_01_merged  99853   0       47217   0.472865\nVA_B_09_SNP_clipped     99853   0       79187   0.793036\nRI_B_10_SNP_clipped     99853   0       82779   0.829009\nRI_W_01_merged  99853   0       73169   0.732767\nRI_B_01_merged  99853   0       69619   0.697215\nVA_B_04_merged  99853   0       58190   0.582757\nRI_B_02_merged  99853   0       87439   0.875677\nRI_W_03_merged  99853   0       38760   0.388171\nVA_B_02_merged  99853   0       78381   0.784964\nVA_W_07_merged  99853   0       64592   0.646871\nVA_B_05_merged  99853   0       54401   0.544811\nRI_B_03_merged  99853   0       80369   0.804873\n\namali010@turing1 VCF]$ bash\namali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 \u0026gt; totalmissing\namali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ gnuplot \u0026lt;\u0026lt; \\EOF\n\u0026gt; set terminal dumb size 120, 30\n\u0026gt; set autoscale\n\u0026gt; unset label\n\u0026gt; set title \u0026quot;Histogram of % missing data per individual\u0026quot;\n\u0026gt; set ylabel \u0026quot;Number of Occurrences\u0026quot;\n\u0026gt; set xlabel \u0026quot;% of missing data\u0026quot;\n\u0026gt; #set yr [0:100000]\n\u0026gt; binwidth=0.01\n\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\n\u0026gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes\n\u0026gt; pause -1\n\u0026gt; EOF\n♀                                         Histogram of % missing data per individual\n  Number of Occurrences\n      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++\n        +           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n    2.5 ++                                                               * *                                        ++\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n        |                                                                * *                                         |\n      2 ++                                                       **      * * ***  ******    ** ****   ****          ++\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\n        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +\n      1 *********************************************************************************************************---++\n       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\n                                                      % of missing data\n#9 - Submit and add everything to your logfile\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Clean up your data directory by:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1a - Make a SAMS folder and mv all your .sam files into that directory\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ mkdir ../../SAMS\n[amali010@turing1 QCFastqs]$ ls \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\n[amali010@turing1 QCFastqs]$ mv \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam ../../SAMS/\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1b - Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\u003c/span\u003e\n[amali010@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../BAMS\n[amali010@coreV3-23-003 QCFastqs]$ ls \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.bam \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.bam.bai\n[amali010@coreV3-23-003 QCFastqs]$ mv \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003ebam \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003ebam.bai ../../BAMS/\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1c - rm your _unfiltered.vcf files if you have any\u003c/span\u003e\nI did not \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e this because I didn\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003et have any.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#1d - rm your .fastq files\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 QCFastqs]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 QCFastqs]$ rm *.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#1e - Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003et \u003cspan class=\"pl-c1\"\u003ecomplete\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e skip this step)\nI did this even though I \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e not have a \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eYOURNAMEmergedfastqs.vcf\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e file from the job that submitted \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e the previous class.\n[amali010@coreV2-22-013 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../VCF\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - Start an interactive session via salloc\u003c/span\u003e\n[amali010@coreV2-22-013 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs\n[amali010@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9278251\nsalloc: job 9278251 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9278251 has been allocated resources\nsalloc: Granted job allocation 9278251\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder\u003c/span\u003e\n[amali010@coreV2-22-013 VCF]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./\n[amali010@coreV2-22-013 VCF]$ ls -alh\ntotal 2.1G\ndrwxrwxrwx 2 amali010 users   54 Feb 12 12:59 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 9 amali010 users  316 Feb 12 12:56 ..\n-rwxr-xr-x 1 amali010 users 1.5G Feb 12 12:59 mergedfastq_HEAAstrangiaAssembly.vcf\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:\u003c/span\u003e\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nI did it with the class vcf file since I didn\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003et have one of my own.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ enable_lmod\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ module load dDocent\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ vcftools\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools (0.1.14)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e© Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eProcess Variant Call Format files\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eFor a list of options, please go to:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        https://vcftools.github.io/man_latest.html\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAlternatively, a man page is available, type:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        man vcftools\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eQuestions, comments, and suggestions should be emailed to:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        vcftools-help@lists.sourceforge.net\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 1214003 out of a possible 1214003 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 14.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#5 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt  ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#6 - Run our host vcf extractor on your merged vcf file using the following syntax:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@coreV2-22-013 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRead in ContigList\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#7 - Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eI did not have a .vcf file of my own for comparison. So I compared the other two files in the parenthesis.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003evcftools: Command not found.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ enable_lmod\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ module load dDocent\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 1214003 out of a possible 1214003 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 11.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 432676 out of a possible 432676 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 6.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#8 - Work through the VCF filtering tutorial until the following step:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eNow that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003evcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --recode-INFO-all\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --mac 3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --minQ 30\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --max-missing 0.5\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --out raw.g5mac3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --recode\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutputting VCF file...\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 99853 out of a possible 432676 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 33.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --vcf raw.g5mac3.recode.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --recode-INFO-all\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --minDP 3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --out raw.g5mac3dp3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --recode\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutputting VCF file...\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 99853 out of a possible 99853 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 31.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --vcf raw.g5mac3dp3.recode.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        --missing-indv\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutputting Individual Missingness\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 99853 out of a possible 99853 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 3.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[amali010@turing1 VCF]$ cat out.imiss\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eINDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_06_merged  99853   0       67075   0.671737\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_07_merged  99853   0       63917   0.640111\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_03_merged  99853   0       56586   0.566693\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_02_merged  99853   0       73105   0.732126\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_04_merged  99853   0       71159   0.712638\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_09_SNP_clipped     99853   0       19459   0.194876\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_08_SNP_clipped     99853   0       88110   0.882397\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_08_SNP_clipped     99853   0       83344   0.834667\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_08_SNP_clipped     99853   0       94221   0.943597\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_02_merged  99853   0       71050   0.711546\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_07_merged  99853   0       62687   0.627793\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_05_merged  99853   0       43945   0.440097\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_06_merged  99853   0       65164   0.652599\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_04_merged  99853   0       50911   0.509859\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_01_merged  99853   0       67150   0.672489\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_10_SNP_clipped     99853   0       82831   0.829529\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_06_merged  99853   0       57724   0.57809\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_05_merged  99853   0       66297   0.663946\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_09_SNP_clipped     99853   0       85107   0.852323\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_10_SNP_clipped     99853   0       83234   0.833565\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_08_SNP_clipped     99853   0       77802   0.779165\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_06_merged  99853   0       79766   0.798834\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_10_SNP_clipped     99853   0       91323   0.914574\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_04_merged  99853   0       46717   0.467858\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_03_merged  99853   0       64016   0.641102\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_07_merged  99853   0       70798   0.709022\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_05_merged  99853   0       57117   0.572011\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_09_SNP_clipped     99853   0       88492   0.886223\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_01_merged  99853   0       47217   0.472865\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_09_SNP_clipped     99853   0       79187   0.793036\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_10_SNP_clipped     99853   0       82779   0.829009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_01_merged  99853   0       73169   0.732767\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_01_merged  99853   0       69619   0.697215\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_04_merged  99853   0       58190   0.582757\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_02_merged  99853   0       87439   0.875677\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_03_merged  99853   0       38760   0.388171\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_02_merged  99853   0       78381   0.784964\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_07_merged  99853   0       64592   0.646871\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_05_merged  99853   0       54401   0.544811\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_03_merged  99853   0       80369   0.804873\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eamali010@turing1 VCF]$ bash\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eamali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ mawk \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e/IN/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e out.imiss | cut -f5 \u0026gt; totalmissing\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eamali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ gnuplot \u0026lt;\u0026lt; \\EOF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set terminal dumb size 120, 30\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set autoscale\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; unset label\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set title \"Histogram of % missing data per individual\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set ylabel \"Number of Occurrences\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set xlabel \"% of missing data\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; #set yr [0:100000]\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; binwidth=0.01\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; plot \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003etotalmissing\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e using (bin($1,binwidth)):(1.0) smooth freq with boxes\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; pause -1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; EOF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e♀                                         Histogram of % missing data per individual\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e  Number of Occurrences\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        +           +           +           +           +       \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003etotalmissing\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e using (bin($1,binwidth)):(1.0) ****** +\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    2.5 ++                                                               * *                                        ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                                * *                                         |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      2 ++                                                       **      * * ***  ******    ** ****   ****          ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                        **      * * * *  * *  *    ** ** *   *  *           |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      1 *********************************************************************************************************---++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e                                                      % of missing data\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#9 - Submit and add everything to your logfile\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit add README.md\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit commit -m \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003eupdating readme\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit push -u origin main\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 09 17-Feb-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day09\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 -  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters\n##Half the class run the following\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n##the other half run the filters we used from the paper\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n\n#1a - Started a salloc session.\n[amali010@turing1 areej]$ salloc\nsalloc: Pending job allocation 9280291\nsalloc: job 9280291 queued and waiting for resources\nsalloc: job 9280291 has been allocated resources\nsalloc: Granted job allocation 9280291\n\n#1b - Navigated to my VCF directory.\n[amali010@coreV2-25-017 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n\n#1c - Ran the following to make sure that the .vcf file is correct: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 15.00 seconds\n\n#1d - Chosen to ran the second filter. Ran the following filter: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n        --recode-INFO-all\n        --mac 3\n        --maf 0.015\n        --max-alleles 2\n        --minDP 10\n        --minQ 30\n        --max-missing 0.5\n        --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n        --recode\n        --remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 1578 out of a possible 432676 Sites\nRun Time = 8.00 seconds\n\n#2 - Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs\n#2a - Looked at missing individuals\n[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf --missing-indv\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf\n        --missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 1578 out of a possible 1578 Sites\nRun Time = 1.00 seconds\n\n[amali010@coreV2-25-017 VCF]$ less mergedfastq_HEAAstrangiaAssembly_subset.vcf\n[amali010@coreV2-25-017 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf\n#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged\n[amali010@coreV2-25-017 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf \u0026gt; samplenames.text\n[amali010@coreV2-25-017 VCF]$ cat samplenames.text\n#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged\n[amali010@coreV2-25-017 VCF]$ cat out.imiss\nINDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS\nRI_W_06_merged  1578    0       316     0.200253\nRI_W_07_merged  1578    0       183     0.11597\nVA_B_03_merged  1578    0       79      0.0500634\nRI_W_02_merged  1578    0       442     0.280101\nRI_W_04_merged  1578    0       581     0.368188\nVA_W_09_SNP_clipped     1578    0       17      0.0107731\nRI_B_08_SNP_clipped     1578    0       1180    0.747782\nVA_W_08_SNP_clipped     1578    0       892     0.565272\nVA_B_08_SNP_clipped     1578    0       1273    0.806717\nVA_W_02_merged  1578    0       293     0.185678\nVA_B_07_merged  1578    0       116     0.0735108\nRI_B_05_merged  1578    0       57      0.0361217\nVA_W_06_merged  1578    0       164     0.103929\nVA_W_04_merged  1578    0       91      0.0576679\nVA_W_01_merged  1578    0       337     0.213561\nVA_B_10_SNP_clipped     1578    0       683     0.432826\nVA_B_06_merged  1578    0       82      0.0519645\nVA_W_05_merged  1578    0       187     0.118504\nRI_B_09_SNP_clipped     1578    0       1081    0.685044\nVA_W_10_SNP_clipped     1578    0       619     0.392269\nRI_W_08_SNP_clipped     1578    0       558     0.353612\nRI_B_06_merged  1578    0       613     0.388466\nRI_W_10_SNP_clipped     1578    0       1219    0.772497\nRI_B_04_merged  1578    0       107     0.0678074\nVA_W_03_merged  1578    0       197     0.124842\nRI_B_07_merged  1578    0       319     0.202155\nRI_W_05_merged  1578    0       79      0.0500634\nRI_W_09_SNP_clipped     1578    0       1159    0.734474\nVA_B_01_merged  1578    0       198     0.125475\nVA_B_09_SNP_clipped     1578    0       510     0.323194\nRI_B_10_SNP_clipped     1578    0       800     0.506971\nRI_W_01_merged  1578    0       499     0.316223\nRI_B_01_merged  1578    0       231     0.146388\nVA_B_04_merged  1578    0       139     0.0880862\nRI_B_02_merged  1578    0       1031    0.653359\nRI_W_03_merged  1578    0       109     0.0690748\nVA_B_02_merged  1578    0       597     0.378327\nVA_W_07_merged  1578    0       129     0.081749\nVA_B_05_merged  1578    0       54      0.0342205\nRI_B_03_merged  1578    0       762     0.48289\n[amali010@coreV2-25-017 VCF]$ cut -f 1 out.imiss | tail -n +2\nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n\nUsed excel to make a second column with just the header\n\nRI_W_06_merged\tRI_W\nRI_W_07_merged\tRI_W\nVA_B_03_merged\tVA_B\nRI_W_02_merged\tRI_W\nRI_W_04_merged\tRI_W\nVA_W_09_SNP_clipped\tVA_W\nRI_B_08_SNP_clipped\tRI_B\nVA_W_08_SNP_clipped\tVA_W\nVA_B_08_SNP_clipped\tVA_B\nVA_W_02_merged\tVA_W\nVA_B_07_merged\tVA_B\nRI_B_05_merged\tRI_B\nVA_W_06_merged\tVA_W\nVA_W_04_merged\tVA_W\nVA_W_01_merged\tVA_W\nVA_B_10_SNP_clipped\tVA_B\nVA_B_06_merged\tVA_B\nVA_W_05_merged\tVA_W\nRI_B_09_SNP_clipped\tRI_B\nVA_W_10_SNP_clipped\tVA_W\nRI_W_08_SNP_clipped\tRI_W\nRI_B_06_merged\tRI_B\nRI_W_10_SNP_clipped\tRI_W\nRI_B_04_merged\tRI_B\nVA_W_03_merged\tVA_W\nRI_B_07_merged\tRI_B\nRI_W_05_merged\tRI_W\nRI_W_09_SNP_clipped\tRI_W\nVA_B_01_merged\tVA_B\nVA_B_09_SNP_clipped\tVA_B\nRI_B_10_SNP_clipped\tRI_B\nRI_W_01_merged\tRI_W\nRI_B_01_merged\tRI_B\nVA_B_04_merged\tVA_B\nRI_B_02_merged\tRI_B\nRI_W_03_merged\tRI_W\nVA_B_02_merged\tVA_B\nVA_W_07_merged\tVA_W\nVA_B_05_merged\tVA_B\nRI_B_03_merged\tRI_B\n\n[amali010@coreV2-25-017 VCF]$ nano popfile2.txt\nPasted the two columns in this .txt file\n\n#3 - Convert your filtered .vcf file to genepop format using the following command:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt\n\n[amali010@coreV2-25-017 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile2.txt\nIndivs with genotypes in vcf file: RI_W_06_merged       RI_W_07_merged  VA_B_03_merged      RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped     RI_B_08_SNP_clipped\nVA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged  VA_B_07_merged  RI_B_05_merged      VA_W_06_merged  VA_W_04_merged  VA_W_01_merged  VA_B_10_SNP_clipped\nVA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped     VA_W_10_SNP_clipped     RI_W_08_SNP_clipped RI_B_06_merged  RI_W_10_SNP_clipped     RI_B_04_merged  VA_W_03_merged      RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped     VA_B_01_merged  VA_B_09_SNP_clipped RI_B_10_SNP_clipped     RI_W_01_merged  RI_B_01_merged  VA_B_04_merged      RI_B_02_merged  RI_W_03_merged  VA_B_02_merged  VA_W_07_merged  VA_B_05_merged      RI_B_03_merged\n44 1578 1578 1578 1578 40\n\n[amali010@coreV2-25-017 VCF]$ ls\nNow I have a .gen file\n[amali010@coreV2-25-017 VCF]$ less 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\nNoticed that the contig name has now merged with SNPs\n\n#4 - SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop\n[amali010@turing1 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)\n$ pwd\n/c/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09\n\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)\n$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen .\namali010@turing.hpc.odu.edu's password:\n1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFi 100%  343KB   2.1MB/s   00:00\n\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)\n$ ls\n1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\nadegenet_PCAs.R\nadegenet_PCAs_HEAOutliers.R\nhomework_day09.txt\n\n#5 - Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\nInput:\nlibrary(\u0026quot;ape\u0026quot;)\nlibrary(\u0026quot;pegas\u0026quot;)\nlibrary(\u0026quot;seqinr\u0026quot;)\nlibrary(\u0026quot;ggplot2\u0026quot;)\nlibrary(\u0026quot;adegenet\u0026quot;)\nsetwd(\u0026quot;/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09\u0026quot;)\ndatafile\u0026lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)\nsum(is.na(datafile$tab))\ndatafile #shows info\n\nOutput:\n\u0026gt; library(\u0026quot;ape\u0026quot;)\n\u0026gt; library(\u0026quot;pegas\u0026quot;)\n\u0026gt; library(\u0026quot;seqinr\u0026quot;)\n\u0026gt; library(\u0026quot;ggplot2\u0026quot;)\n\u0026gt; library(\u0026quot;adegenet\u0026quot;)\n\u0026gt; setwd(\u0026quot;/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09\u0026quot;)\n\u0026gt; datafile\u0026lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)\n\n Converting data from a Genepop .gen file to a genind object... \n\n\nFile description:  AllSNPs \n\n...done.\n\n\u0026gt; sum(is.na(datafile$tab))\n[1] 35966\n\u0026gt; datafile #shows info\n/// GENIND OBJECT /////////\n\n // 40 individuals; 1,578 loci; 3,156 alleles; size: 1.4 Mb\n\n // Basic content\n   @tab:  40 x 3156 matrix of allele counts\n   @loc.n.all: number of alleles per locus (range: 2-2)\n   @loc.fac: locus factor for the 3156 columns of @tab\n   @all.names: list of allele names for each locus\n   @ploidy: ploidy of each individual  (range: 2-2)\n   @type:  codom\n   @call: read.genepop(file = \u0026quot;1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\u0026quot;, \n    ncode = 2)\n\n // Optional content\n   @pop: population of each individual (group size range: 10-10)\n\nInput:\nYOURdata\u0026lt;-scaleGen(datafile, NA.method='mean')\nX\u0026lt;-YOURdata\nsubstring(pop(datafile),1,4)\nY\u0026lt;-as.factor(substring(pop(datafile),1,4))\npca1 \u0026lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\n\nOutput:\n\u0026gt; YOURdata\u0026lt;-scaleGen(datafile, NA.method='mean')\nWarning message:\nIn .local(x, ...) : Some scaling values are null.\n Corresponding alleles are removed.\n\u0026gt; X\u0026lt;-YOURdata\n\u0026gt; substring(pop(datafile),1,4)\n [1] \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot; \u0026quot;RI_B\u0026quot;\n[10] \u0026quot;RI_B\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot;\n[19] \u0026quot;VA_W\u0026quot; \u0026quot;VA_W\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot;\n[28] \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;VA_B\u0026quot; \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot;\n[37] \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot; \u0026quot;RI_W\u0026quot;\n\u0026gt; Y\u0026lt;-as.factor(substring(pop(datafile),1,4))\n\u0026gt; pca1 \u0026lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\n\nInput:\n#### PCAs ####\n# individual labels\ns.label(pca1$li)\n\nOutput:\n#### PCAs ####\n\u0026gt; # individual labels\n\u0026gt; s.label(pca1$li)\nPlotted each sample name as a point\n\nInput:\n# population elipses\ns.class(pca1$li, pop(datafile))\n\nOutput:\n\u0026gt; # population elipses\n\u0026gt; s.class(pca1$li, pop(datafile))\nPlotted population\n\nInput:\n#color symbols, pop names\n#pdf(\u0026quot;YOURINITIALS_ColorPCA1v2.pdf\u0026quot;)\ncol \u0026lt;- c(\u0026quot;blue\u0026quot;,\u0026quot;red\u0026quot;, \u0026quot;green\u0026quot;, \u0026quot;black\u0026quot;)\ns.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\u0026quot;topright\u0026quot;)\ntitle(\u0026quot;PCA of DJB_data\\naxes 1-2\u0026quot;)\n\nOutput: \n\u0026gt; #color symbols, pop names\n\u0026gt; #pdf(\u0026quot;YOURINITIALS_ColorPCA1v2.pdf\u0026quot;)\n\u0026gt; col \u0026lt;- c(\u0026quot;blue\u0026quot;,\u0026quot;red\u0026quot;, \u0026quot;green\u0026quot;, \u0026quot;black\u0026quot;)\n\u0026gt; s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\n\u0026gt; add.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\u0026quot;topright\u0026quot;)\n\u0026gt; title(\u0026quot;PCA of DJB_data\\naxes 1-2\u0026quot;)\nPlotted population with color\nPoints are individuals with elipses around the populations\nNot as many outliers as the other .gen file\n\nInput:\n#### Snapclust ####\na.clust\u0026lt;-snapclust(datafile, k = 2)\nclass(a.clust)\nnames(a.clust)\na.tab \u0026lt;- table(pop(datafile), a.clust$group)\ntable.value(a.tab, col.labels = 1:2)\n\nOutput:\n\u0026gt; #### Snapclust ####\n\u0026gt; a.clust\u0026lt;-snapclust(datafile, k = 2)\n\u0026gt; class(a.clust)\n[1] \u0026quot;snapclust\u0026quot; \u0026quot;list\u0026quot;     \n\u0026gt; names(a.clust)\n[1] \u0026quot;group\u0026quot;     \u0026quot;ll\u0026quot;        \u0026quot;proba\u0026quot;     \u0026quot;converged\u0026quot; \u0026quot;n.iter\u0026quot;   \n[6] \u0026quot;n.param\u0026quot;  \n\u0026gt; a.tab \u0026lt;- table(pop(datafile), a.clust$group)\n\u0026gt; table.value(a.tab, col.labels = 1:2)\nGave us a table that shows out of your total populations which are the individuals from this population assigning too. \nMost individuals are being assigned to blue populations with a few being assigned to brown population.\n\nInput:\ncompoplot(a.clust)\n\nOutput:\n\u0026gt; compoplot(a.clust)\nGave us cluster diagram that showed the same thing where the majority are being assigned to the blue population with the exception of a few that are assigned to brown population.\n\n#6 - Submit and add everything to your logfile\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 -  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Half the class run the following\u003c/span\u003e\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#the other half run the filters we used from the paper\u003c/span\u003e\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1a - Started a salloc session.\u003c/span\u003e\n[amali010@turing1 areej]$ salloc\nsalloc: Pending job allocation 9280291\nsalloc: job 9280291 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9280291 has been allocated resources\nsalloc: Granted job allocation 9280291\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1b - Navigated to my VCF directory.\u003c/span\u003e\n[amali010@coreV2-25-017 VCF]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1c - Ran the following to make sure that the .vcf file is correct: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 15.00 seconds\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1d - Chosen to ran the second filter. Ran the following filter: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\u003c/span\u003e\n[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n        --recode-INFO-all\n        --mac 3\n        --maf 0.015\n        --max-alleles 2\n        --minDP 10\n        --minQ 30\n        --max-missing 0.5\n        --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n        --recode\n        --remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 1578 out of a possible 432676 Sites\nRun Time = 8.00 seconds\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2a - Looked at missing individuals\u003c/span\u003e\n[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf --missing-indv\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n        --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf\n        --missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 1578 out of a possible 1578 Sites\nRun Time = 1.00 seconds\n\n[amali010@coreV2-25-017 VCF]$ less mergedfastq_HEAAstrangiaAssembly_subset.vcf\n[amali010@coreV2-25-017 VCF]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e#CHROM\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged\u003c/span\u003e\n[amali010@coreV2-25-017 VCF]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e#CHROM\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e mergedfastq_HEAAstrangiaAssembly_subset.vcf \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e samplenames.text\n[amali010@coreV2-25-017 VCF]$ cat samplenames.text\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged\u003c/span\u003e\n[amali010@coreV2-25-017 VCF]$ cat out.imiss\nINDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS\nRI_W_06_merged  1578    0       316     0.200253\nRI_W_07_merged  1578    0       183     0.11597\nVA_B_03_merged  1578    0       79      0.0500634\nRI_W_02_merged  1578    0       442     0.280101\nRI_W_04_merged  1578    0       581     0.368188\nVA_W_09_SNP_clipped     1578    0       17      0.0107731\nRI_B_08_SNP_clipped     1578    0       1180    0.747782\nVA_W_08_SNP_clipped     1578    0       892     0.565272\nVA_B_08_SNP_clipped     1578    0       1273    0.806717\nVA_W_02_merged  1578    0       293     0.185678\nVA_B_07_merged  1578    0       116     0.0735108\nRI_B_05_merged  1578    0       57      0.0361217\nVA_W_06_merged  1578    0       164     0.103929\nVA_W_04_merged  1578    0       91      0.0576679\nVA_W_01_merged  1578    0       337     0.213561\nVA_B_10_SNP_clipped     1578    0       683     0.432826\nVA_B_06_merged  1578    0       82      0.0519645\nVA_W_05_merged  1578    0       187     0.118504\nRI_B_09_SNP_clipped     1578    0       1081    0.685044\nVA_W_10_SNP_clipped     1578    0       619     0.392269\nRI_W_08_SNP_clipped     1578    0       558     0.353612\nRI_B_06_merged  1578    0       613     0.388466\nRI_W_10_SNP_clipped     1578    0       1219    0.772497\nRI_B_04_merged  1578    0       107     0.0678074\nVA_W_03_merged  1578    0       197     0.124842\nRI_B_07_merged  1578    0       319     0.202155\nRI_W_05_merged  1578    0       79      0.0500634\nRI_W_09_SNP_clipped     1578    0       1159    0.734474\nVA_B_01_merged  1578    0       198     0.125475\nVA_B_09_SNP_clipped     1578    0       510     0.323194\nRI_B_10_SNP_clipped     1578    0       800     0.506971\nRI_W_01_merged  1578    0       499     0.316223\nRI_B_01_merged  1578    0       231     0.146388\nVA_B_04_merged  1578    0       139     0.0880862\nRI_B_02_merged  1578    0       1031    0.653359\nRI_W_03_merged  1578    0       109     0.0690748\nVA_B_02_merged  1578    0       597     0.378327\nVA_W_07_merged  1578    0       129     0.081749\nVA_B_05_merged  1578    0       54      0.0342205\nRI_B_03_merged  1578    0       762     0.48289\n[amali010@coreV2-25-017 VCF]$ cut -f 1 out.imiss \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n +2\nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n\nUsed excel to make a second column with just the header\n\nRI_W_06_merged\tRI_W\nRI_W_07_merged\tRI_W\nVA_B_03_merged\tVA_B\nRI_W_02_merged\tRI_W\nRI_W_04_merged\tRI_W\nVA_W_09_SNP_clipped\tVA_W\nRI_B_08_SNP_clipped\tRI_B\nVA_W_08_SNP_clipped\tVA_W\nVA_B_08_SNP_clipped\tVA_B\nVA_W_02_merged\tVA_W\nVA_B_07_merged\tVA_B\nRI_B_05_merged\tRI_B\nVA_W_06_merged\tVA_W\nVA_W_04_merged\tVA_W\nVA_W_01_merged\tVA_W\nVA_B_10_SNP_clipped\tVA_B\nVA_B_06_merged\tVA_B\nVA_W_05_merged\tVA_W\nRI_B_09_SNP_clipped\tRI_B\nVA_W_10_SNP_clipped\tVA_W\nRI_W_08_SNP_clipped\tRI_W\nRI_B_06_merged\tRI_B\nRI_W_10_SNP_clipped\tRI_W\nRI_B_04_merged\tRI_B\nVA_W_03_merged\tVA_W\nRI_B_07_merged\tRI_B\nRI_W_05_merged\tRI_W\nRI_W_09_SNP_clipped\tRI_W\nVA_B_01_merged\tVA_B\nVA_B_09_SNP_clipped\tVA_B\nRI_B_10_SNP_clipped\tRI_B\nRI_W_01_merged\tRI_W\nRI_B_01_merged\tRI_B\nVA_B_04_merged\tVA_B\nRI_B_02_merged\tRI_B\nRI_W_03_merged\tRI_W\nVA_B_02_merged\tVA_B\nVA_W_07_merged\tVA_W\nVA_B_05_merged\tVA_B\nRI_B_03_merged\tRI_B\n\n[amali010@coreV2-25-017 VCF]$ nano popfile2.txt\nPasted the two columns \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e this .txt file\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Convert your filtered .vcf file to genepop format using the following command:\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt\n\n[amali010@coreV2-25-017 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile2.txt\nIndivs with genotypes \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e vcf file: RI_W_06_merged       RI_W_07_merged  VA_B_03_merged      RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped     RI_B_08_SNP_clipped\nVA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged  VA_B_07_merged  RI_B_05_merged      VA_W_06_merged  VA_W_04_merged  VA_W_01_merged  VA_B_10_SNP_clipped\nVA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped     VA_W_10_SNP_clipped     RI_W_08_SNP_clipped RI_B_06_merged  RI_W_10_SNP_clipped     RI_B_04_merged  VA_W_03_merged      RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped     VA_B_01_merged  VA_B_09_SNP_clipped RI_B_10_SNP_clipped     RI_W_01_merged  RI_B_01_merged  VA_B_04_merged      RI_B_02_merged  RI_W_03_merged  VA_B_02_merged  VA_W_07_merged  VA_B_05_merged      RI_B_03_merged\n44 1578 1578 1578 1578 40\n\n[amali010@coreV2-25-017 VCF]$ ls\nNow I have a .gen file\n[amali010@coreV2-25-017 VCF]$ less 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\nNoticed that the contig name has now merged with SNPs\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop\u003c/span\u003e\n[amali010@turing1 VCF]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF\n\nareej@DESKTOP-RF6VF4N MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)\n$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/c/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09\n\nareej@DESKTOP-RF6VF4N MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)\n$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\namali010@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFi 100%  343KB   2.1MB/s   00:00\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eadegenet_PCAs.R\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eadegenet_PCAs_HEAOutliers.R\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ehomework_day09.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#5 - Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003elibrary(\"ape\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003elibrary(\"pegas\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003elibrary(\"seqinr\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003elibrary(\"ggplot2\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003elibrary(\"adegenet\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esetwd(\"/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003edatafile\u0026lt;-read.genepop(\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e, ncode=2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esum(is.na(datafile$tab))\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003edatafile #shows info\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; library(\"ape\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; library(\"pegas\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; library(\"seqinr\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; library(\"ggplot2\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; library(\"adegenet\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; setwd(\"/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; datafile\u0026lt;-read.genepop(\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e, ncode=2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e Converting data from a Genepop .gen file to a genind object... \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eFile description:  AllSNPs \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e...done.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; sum(is.na(datafile$tab))\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[1] 35966\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; datafile #shows info\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/// GENIND OBJECT /////////\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e // 40 individuals; 1,578 loci; 3,156 alleles; size: 1.4 Mb\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e // Basic content\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @tab:  40 x 3156 matrix of allele counts\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @loc.n.all: number of alleles per locus (range: 2-2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @loc.fac: locus factor for the 3156 columns of @tab\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @all.names: list of allele names for each locus\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @ploidy: ploidy of each individual  (range: 2-2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @type:  codom\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @call: read.genepop(file = \"1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\", \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    ncode = 2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e // Optional content\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e   @pop: population of each individual (group size range: 10-10)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eYOURdata\u0026lt;-scaleGen(datafile, NA.method=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003emean\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eX\u0026lt;-YOURdata\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esubstring(pop(datafile),1,4)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eY\u0026lt;-as.factor(substring(pop(datafile),1,4))\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003epca1 \u0026lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; YOURdata\u0026lt;-scaleGen(datafile, NA.method=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003emean\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eWarning message:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eIn .local(x, ...) : Some scaling values are null.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e Corresponding alleles are removed.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; X\u0026lt;-YOURdata\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; substring(pop(datafile),1,4)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e [1] \"RI_B\" \"RI_B\" \"RI_B\" \"RI_B\" \"RI_B\" \"RI_B\" \"RI_B\" \"RI_B\" \"RI_B\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[10] \"RI_B\" \"VA_W\" \"VA_W\" \"VA_W\" \"VA_W\" \"VA_W\" \"VA_W\" \"VA_W\" \"VA_W\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[19] \"VA_W\" \"VA_W\" \"VA_B\" \"VA_B\" \"VA_B\" \"VA_B\" \"VA_B\" \"VA_B\" \"VA_B\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[28] \"VA_B\" \"VA_B\" \"VA_B\" \"RI_W\" \"RI_W\" \"RI_W\" \"RI_W\" \"RI_W\" \"RI_W\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[37] \"RI_W\" \"RI_W\" \"RI_W\" \"RI_W\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; Y\u0026lt;-as.factor(substring(pop(datafile),1,4))\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; pca1 \u0026lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#### PCAs ####\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e# individual labels\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003es.label(pca1$li)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#### PCAs ####\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; # individual labels\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; s.label(pca1$li)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ePlotted each sample name as a point\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e# population elipses\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003es.class(pca1$li, pop(datafile))\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; # population elipses\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; s.class(pca1$li, pop(datafile))\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ePlotted population\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#color symbols, pop names\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#pdf(\"YOURINITIALS_ColorPCA1v2.pdf\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecol \u0026lt;- c(\"blue\",\"red\", \"green\", \"black\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003es.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eadd.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\"topright\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003etitle(\"PCA of DJB_data\\naxes 1-2\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutput: \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; #color symbols, pop names\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; #pdf(\"YOURINITIALS_ColorPCA1v2.pdf\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; col \u0026lt;- c(\"blue\",\"red\", \"green\", \"black\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; add.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\"topright\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; title(\"PCA of DJB_data\\naxes 1-2\")\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ePlotted population with color\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ePoints are individuals with elipses around the populations\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eNot as many outliers as the other .gen file\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#### Snapclust ####\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea.clust\u0026lt;-snapclust(datafile, k = 2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eclass(a.clust)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003enames(a.clust)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ea.tab \u0026lt;- table(pop(datafile), a.clust$group)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003etable.value(a.tab, col.labels = 1:2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; #### Snapclust ####\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; a.clust\u0026lt;-snapclust(datafile, k = 2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; class(a.clust)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[1] \"snapclust\" \"list\"     \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; names(a.clust)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[1] \"group\"     \"ll\"        \"proba\"     \"converged\" \"n.iter\"   \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[6] \"n.param\"  \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; a.tab \u0026lt;- table(pop(datafile), a.clust$group)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; table.value(a.tab, col.labels = 1:2)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eGave us a table that shows out of your total populations which are the individuals from this population assigning too. \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eMost individuals are being assigned to blue populations with a few being assigned to brown population.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecompoplot(a.clust)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutput:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; compoplot(a.clust)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eGave us cluster diagram that showed the same thing where the majority are being assigned to the blue population with the exception of a few that are assigned to brown population.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#6 - Submit and add everything to your logfile\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit add README.md\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit commit -m \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003eupdating readme\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit push -u origin main\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 10 19-Feb-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day10\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 - Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\nI did this.\n\n#2 - cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\n[amali010@turing1 SAMS]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS\n[amali010@turing1 SAMS]$ salloc\nsalloc: Pending job allocation 9280964\nsalloc: job 9280964 queued and waiting for resources\nsalloc: job 9280964 has been allocated resources\nsalloc: Granted job allocation 9280964\n[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\n\n\n#3 - Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt YOURNAMEFullCounts_summed.txt NoMatch *_counts.txt\n[amali010@coreV1-22-016 SAMS]$ mv match_counts.txt match_countsoutput.txt\n[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt AMFullCounts_summed.txt NoMatch *counts.txt\nHits not matchedRI_B_06_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_14_clippedtrimmed.fastq_counts.txt=1698     RI_B_07_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_22_clippedtrimmed.fastq_counts.txt=1698     RI_W_06_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_14_clippedtrimmed.fastq_counts.txt=1698 RI_W_07_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_22_clippedtrimmed.fastq_counts.txt=1698 VA_B_06_18_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_18_clippedtrimmed.fastq_counts.txt=1698 VA_B_07_22_clippedtrimmed.fastq_counts.txt=1698     VA_W_06_18_clippedtrimmed.fastq_counts.txt=1698 VA_W_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_18_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_22_clippedtrimmed.fastq_counts.txt=1698\n[amali010@coreV1-22-016 SAMS]$ ls\nAMFullCounts_summed.txt\nmatch_countsoutput.txt\nRI_B_06_18_clippedtrimmed.fastq_counts.txt\nRI_B_06_18_clippedtrimmed.fastq.sam\nRI_B_07_14_clippedtrimmed.fastq_counts.txt\nRI_B_07_14_clippedtrimmed.fastq.sam\nRI_B_07_18_clippedtrimmed.fastq_counts.txt\nRI_B_07_18_clippedtrimmed.fastq.sam\nRI_B_07_22_clippedtrimmed.fastq_counts.txt\nRI_B_07_22_clippedtrimmed.fastq.sam\nRI_W_06_18_clippedtrimmed.fastq_counts.txt\nRI_W_06_18_clippedtrimmed.fastq.sam\nRI_W_07_14_clippedtrimmed.fastq_counts.txt\nRI_W_07_14_clippedtrimmed.fastq.sam\nRI_W_07_18_clippedtrimmed.fastq_counts.txt\nRI_W_07_18_clippedtrimmed.fastq.sam\nRI_W_07_22_clippedtrimmed.fastq_counts.txt\nRI_W_07_22_clippedtrimmed.fastq.sam\nVA_B_06_18_clippedtrimmed.fastq_counts.txt\nVA_B_06_18_clippedtrimmed.fastq.sam\nVA_B_07_14_clippedtrimmed.fastq_counts.txt\nVA_B_07_14_clippedtrimmed.fastq.sam\nVA_B_07_18_clippedtrimmed.fastq_counts.txt\nVA_B_07_18_clippedtrimmed.fastq.sam\nVA_B_07_22_clippedtrimmed.fastq_counts.txt\nVA_B_07_22_clippedtrimmed.fastq.sam\nVA_W_06_18_clippedtrimmed.fastq_counts.txt\nVA_W_06_18_clippedtrimmed.fastq.sam\nVA_W_07_14_clippedtrimmed.fastq_counts.txt\nVA_W_07_14_clippedtrimmed.fastq.sam\nVA_W_07_18_clippedtrimmed.fastq_counts.txt\nVA_W_07_18_clippedtrimmed.fastq.sam\nVA_W_07_22_clippedtrimmed.fastq_counts.txt\nVA_W_07_22_clippedtrimmed.fastq.sam\n\n#4 - scp YOURNAMEFullCounts_summed.txt to your laptop\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)\n$ pwd\n/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10\n\n[amali010@coreV1-22-016 SAMS]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS\n[amali010@coreV1-22-016 SAMS]$ ls /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt\n\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)\n$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt .\namali010@turing.hpc.odu.edu's password:\nAMFullCounts_summed.txt                           100%  756KB   4.8MB/s   00:00\n\nareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)\n$ ls\nAMFullCounts_summed.txt\ncoral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\ncoral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen\nhomework_day10.txt\n\n#5 - Edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)\nI did this in Notepad++ where I had to replace _clippedtrimmed.fastq_counts.txt_UniqueTotReads instead of just the _counts.txt_UniqueTotReads to retain the informative part of the sample names.\n\n#6 - Submit and add everything to your logfile\ngit add README.md\ngit commit -m 'updating readme'\ngit push -u origin main\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/span\u003e\nI did this.\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 - cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\n[amali010@turing1 SAMS]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS\n[amali010@turing1 SAMS]$ salloc\nsalloc: Pending job allocation 9280964\nsalloc: job 9280964 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9280964 has been allocated resources\nsalloc: Granted job allocation 9280964\n[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 - Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt YOURNAMEFullCounts_summed.txt NoMatch \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_counts.txt\n[amali010@coreV1-22-016 SAMS]$ mv match_counts.txt match_countsoutput.txt\n[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt AMFullCounts_summed.txt NoMatch \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003ecounts.txt\nHits not matchedRI_B_06_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_14_clippedtrimmed.fastq_counts.txt=1698     RI_B_07_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_22_clippedtrimmed.fastq_counts.txt=1698     RI_W_06_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_14_clippedtrimmed.fastq_counts.txt=1698 RI_W_07_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_22_clippedtrimmed.fastq_counts.txt=1698 VA_B_06_18_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_18_clippedtrimmed.fastq_counts.txt=1698 VA_B_07_22_clippedtrimmed.fastq_counts.txt=1698     VA_W_06_18_clippedtrimmed.fastq_counts.txt=1698 VA_W_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_18_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_22_clippedtrimmed.fastq_counts.txt=1698\n[amali010@coreV1-22-016 SAMS]$ ls\nAMFullCounts_summed.txt\nmatch_countsoutput.txt\nRI_B_06_18_clippedtrimmed.fastq_counts.txt\nRI_B_06_18_clippedtrimmed.fastq.sam\nRI_B_07_14_clippedtrimmed.fastq_counts.txt\nRI_B_07_14_clippedtrimmed.fastq.sam\nRI_B_07_18_clippedtrimmed.fastq_counts.txt\nRI_B_07_18_clippedtrimmed.fastq.sam\nRI_B_07_22_clippedtrimmed.fastq_counts.txt\nRI_B_07_22_clippedtrimmed.fastq.sam\nRI_W_06_18_clippedtrimmed.fastq_counts.txt\nRI_W_06_18_clippedtrimmed.fastq.sam\nRI_W_07_14_clippedtrimmed.fastq_counts.txt\nRI_W_07_14_clippedtrimmed.fastq.sam\nRI_W_07_18_clippedtrimmed.fastq_counts.txt\nRI_W_07_18_clippedtrimmed.fastq.sam\nRI_W_07_22_clippedtrimmed.fastq_counts.txt\nRI_W_07_22_clippedtrimmed.fastq.sam\nVA_B_06_18_clippedtrimmed.fastq_counts.txt\nVA_B_06_18_clippedtrimmed.fastq.sam\nVA_B_07_14_clippedtrimmed.fastq_counts.txt\nVA_B_07_14_clippedtrimmed.fastq.sam\nVA_B_07_18_clippedtrimmed.fastq_counts.txt\nVA_B_07_18_clippedtrimmed.fastq.sam\nVA_B_07_22_clippedtrimmed.fastq_counts.txt\nVA_B_07_22_clippedtrimmed.fastq.sam\nVA_W_06_18_clippedtrimmed.fastq_counts.txt\nVA_W_06_18_clippedtrimmed.fastq.sam\nVA_W_07_14_clippedtrimmed.fastq_counts.txt\nVA_W_07_14_clippedtrimmed.fastq.sam\nVA_W_07_18_clippedtrimmed.fastq_counts.txt\nVA_W_07_18_clippedtrimmed.fastq.sam\nVA_W_07_22_clippedtrimmed.fastq_counts.txt\nVA_W_07_22_clippedtrimmed.fastq.sam\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 - scp YOURNAMEFullCounts_summed.txt to your laptop\u003c/span\u003e\nareej@DESKTOP-RF6VF4N MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)\n$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10\n\n[amali010@coreV1-22-016 SAMS]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS\n[amali010@coreV1-22-016 SAMS]$ ls /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt\n\nareej@DESKTOP-RF6VF4N MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)\n$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\namali010@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAMFullCounts_summed.txt                           100%  756KB   4.8MB/s   00:00\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eareej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAMFullCounts_summed.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecoral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecoral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ehomework_day10.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#5 - Edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eI did this in Notepad++ where I had to replace _clippedtrimmed.fastq_counts.txt_UniqueTotReads instead of just the _counts.txt_UniqueTotReads to retain the informative part of the sample names.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#6 - Submit and add everything to your logfile\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit add README.md\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit commit -m \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003eupdating readme\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003egit push -u origin main\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##Day 11 24-Feb-2021\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExercise day11\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1 - Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file\nDid this during class\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 - Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file\u003c/span\u003e\nDid this during class\u003c/pre\u003e\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fareejmalik%2FAreejsAdvancedGenomicsLog"}}],"overviewFilesProcessingTime":172.091939,"createFromTemplatePath":"/new?template_name=AreejsAdvancedGenomicsLog\u0026template_owner=areejmalik"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/areejmalik/AreejsAdvancedGenomicsLog/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/areejmalik/AreejsAdvancedGenomicsLog/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/areejmalik/AreejsAdvancedGenomicsLog/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/areejmalik/AreejsAdvancedGenomicsLog/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/areejmalik/AreejsAdvancedGenomicsLog/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           32 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/areejmalik/AreejsAdvancedGenomicsLog/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/areejmalik/AreejsAdvancedGenomicsLog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/areejmalik/AreejsAdvancedGenomicsLog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <p dir="auto">
                     2021_Spring_Areej's Advanced Genomics Notebook
                    </p>
                    <p dir="auto">
                     ##Day 02 22-Jan-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day02
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
#1 - Log on to the cluster @turing.hpc.odu.edu
[amali010@turing1 ~]$ ls
data  exercises  hpc-intro

#2 - Make a directory in your course workspace called data
[amali010@turing1 ~]$ mkdir data

#3 - Make a directory called exercises in your data directory
[amali010@turing1 data]$ mkdir exercises

#4 - Execute a pwd command and start a log of your commands with a 
header for today's date in your README.md github logfile in your workspace
[amali010@turing1 exercises]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/exercises 

#5 - cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory
[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./
[amali010@turing1 exercises]$ ls
Exercise2_djb.fasta.gz  Exercise2.fasta.gz  Exercise2.fastq.tar.gz
Exercise2.fasta         Exercise2.fastq
[amali010@turing1 exercises]$ ls -alh
total 37M
drwxrwxrwx 2 amali010 users  146 Jan 26 20:48 .
drwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..
-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz
-rwxr-xr-x 1 amali010 users    0 Jan 29 09:53 Exercise2.fasta
-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq
-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz

#6 - Unzip and untar the files
[amali010@turing1 exercises]$ gunzip -c Exercise2_djb.fasta.gz &gt; Exercise2.fasta
[amali010@turing1 exercises]$ ls
Exercise2_djb.fasta.gz  Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz
[amali010@turing1 exercises]$ ls -alh
total 61M
drwxrwxrwx 2 amali010 users  146 Jan 26 20:48 .
drwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..
-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz
-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta
-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq
-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz

[amali010@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz
Exercise2.fastq
[amali010@turing1 exercises]$ ls -alh
total 69M
drwxrwxrwx 2 amali010 users  182 Jan 29 10:11 .
drwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..
-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz
-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta
-rwxr-xr-x 1 amali010 users 4.3M Jan 29 10:06 Exercise2.fasta.gz
-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq
-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz

#7 - Add these commands to your notebook
Done

#8 - Calculate how many sequences are in each file and add these results to your notebook file
[amali010@turing1 exercises]$ head -1 Exercise2.fasta
&gt;scaffold10078|size20675
[amali010@turing1 exercises]$ grep -c '&gt;' Exercise2.fasta
138
[amali010@turing1 exercises]$ head -1 Exercise2.fastq
@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC
[amali010@turing1 exercises]$ grep -c '@HS3' Exercise2.fastq
61304
[amali010@turing1 exercises]$ wc -l Exercise2.fastq
245216 Exercise2.fastq
[amali010@turing1 exercises]$ echo 245216/4 | bc -l
61304.00000000000000000000

#9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory
[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../scripts/
cp: cannot create regular file `../scripts/': Is a directory
[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/
[amali010@turing1 exercises]$ cd ..
[amali010@turing1 data]$ cd ..
[amali010@turing1 areej]$ ls
am_exercise1.txt  data  day03  groundrules.txt  scripts
[amali010@turing1 areej]$ cd scripts
[amali010@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py
renamer_advbioinf.py

#10 - Start an interactive compute session and re-navigate to your exercises directory
[amali010@turing1 exercises]$ salloc
salloc: Pending job allocation 9271058
salloc: job 9271058 queued and waiting for resources
salloc: job 9271058 has been allocated resources
salloc: Granted job allocation 9271058

#11 - Run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name
[amali010@coreV2-22-007 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 138
contigs &gt;= 1000bp = 138
contigs &gt;= 2000bp = 135

#12 - Did you add all these entries to your notebook file, including the results of the script?
Yes!

#13 - Push your notebook file to your github page
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Log on to the cluster @turing.hpc.odu.edu</span>
[amali010@turing1 <span class="pl-k">~</span>]$ ls
data  exercises  hpc-intro

<span class="pl-c"><span class="pl-c">#</span>2 - Make a directory in your course workspace called data</span>
[amali010@turing1 <span class="pl-k">~</span>]$ mkdir data

<span class="pl-c"><span class="pl-c">#</span>3 - Make a directory called exercises in your data directory</span>
[amali010@turing1 data]$ mkdir exercises

<span class="pl-c"><span class="pl-c">#</span>4 - Execute a pwd command and start a log of your commands with a </span>
header <span class="pl-k">for</span> <span class="pl-smi">today<span class="pl-s"><span class="pl-pds">'</span>s date</span></span> <span class="pl-k">in</span> your README.md github logfile <span class="pl-k">in</span> your workspace
[amali010@turing1 exercises]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/exercises 

<span class="pl-c"><span class="pl-c">#</span>5 - cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory</span>
[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast<span class="pl-k">*</span> ./
[amali010@turing1 exercises]$ ls
Exercise2_djb.fasta.gz  Exercise2.fasta.gz  Exercise2.fastq.tar.gz
Exercise2.fasta         Exercise2.fastq
[amali010@turing1 exercises]$ ls -alh
total 37M
drwxrwxrwx 2 amali010 users  146 Jan 26 20:48 <span class="pl-c1">.</span>
drwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..
-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz
-rwxr-xr-x 1 amali010 users    0 Jan 29 09:53 Exercise2.fasta
-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq
-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz

<span class="pl-c"><span class="pl-c">#</span>6 - Unzip and untar the files</span>
[amali010@turing1 exercises]$ gunzip -c Exercise2_djb.fasta.gz <span class="pl-k">&gt;</span> Exercise2.fasta
[amali010@turing1 exercises]$ ls
Exercise2_djb.fasta.gz  Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz
[amali010@turing1 exercises]$ ls -alh
total 61M
drwxrwxrwx 2 amali010 users  146 Jan 26 20:48 <span class="pl-c1">.</span>
drwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..
-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz
-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta
-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq
-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz

[amali010@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz
Exercise2.fastq
[amali010@turing1 exercises]$ ls -alh
total 69M
drwxrwxrwx 2 amali010 users  182 Jan 29 10:11 <span class="pl-c1">.</span>
drwxrwxrwx 4 amali010 users  188 Jan 28 17:31 ..
-rwxr-xr-x 1 dbarshis users 4.3M Jan 26 20:48 Exercise2_djb.fasta.gz
-rwxr-xr-x 1 amali010 users  17M Jan 29 09:55 Exercise2.fasta
-rwxr-xr-x 1 amali010 users 4.3M Jan 29 10:06 Exercise2.fasta.gz
-rw-r--r-- 1 amali010 users  18M Sep  2  2015 Exercise2.fastq
-rwxr-xr-x 1 dbarshis users 4.3M Jan 22 13:51 Exercise2.fastq.tar.gz

<span class="pl-c"><span class="pl-c">#</span>7 - Add these commands to your notebook</span>
Done

<span class="pl-c"><span class="pl-c">#</span>8 - Calculate how many sequences are in each file and add these results to your notebook file</span>
[amali010@turing1 exercises]$ head -1 Exercise2.fasta
<span class="pl-k">&gt;</span>scaffold10078<span class="pl-k">|</span>size20675
[amali010@turing1 exercises]$ grep -c <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> Exercise2.fasta
138
[amali010@turing1 exercises]$ head -1 Exercise2.fastq
@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC
[amali010@turing1 exercises]$ grep -c <span class="pl-s"><span class="pl-pds">'</span>@HS3<span class="pl-pds">'</span></span> Exercise2.fastq
61304
[amali010@turing1 exercises]$ wc -l Exercise2.fastq
245216 Exercise2.fastq
[amali010@turing1 exercises]$ <span class="pl-c1">echo</span> 245216/4 <span class="pl-k">|</span> bc -l
61304.00000000000000000000

<span class="pl-c"><span class="pl-c">#</span>9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory</span>
[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../scripts/
cp: cannot create regular file <span class="pl-s"><span class="pl-pds">`</span>../scripts/<span class="pl-s"><span class="pl-pds">'</span>: Is a directory</span></span>
<span class="pl-s"><span class="pl-s">[amali010@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/</span></span>
<span class="pl-s"><span class="pl-s">[amali010@turing1 exercises]$ cd ..</span></span>
<span class="pl-s"><span class="pl-s">[amali010@turing1 data]$ cd ..</span></span>
<span class="pl-s"><span class="pl-s">[amali010@turing1 areej]$ ls</span></span>
<span class="pl-s"><span class="pl-s">am_exercise1.txt  data  day03  groundrules.txt  scripts</span></span>
<span class="pl-s"><span class="pl-s">[amali010@turing1 areej]$ cd scripts</span></span>
<span class="pl-s"><span class="pl-s">[amali010@turing1 scripts]$ ls</span></span>
<span class="pl-s"><span class="pl-s">avg_cov_len_fasta_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py</span></span>
<span class="pl-s"><span class="pl-s">renamer_advbioinf.py</span></span>
<span class="pl-s"><span class="pl-s"></span></span>
<span class="pl-s"><span class="pl-s">#10 - Start an interactive compute session and re-navigate to your exercises directory</span></span>
<span class="pl-s"><span class="pl-s">[amali010@turing1 exercises]$ salloc</span></span>
<span class="pl-s"><span class="pl-s">salloc: Pending job allocation 9271058</span></span>
<span class="pl-s"><span class="pl-s">salloc: job 9271058 queued and waiting for resources</span></span>
<span class="pl-s"><span class="pl-s">salloc: job 9271058 has been allocated resources</span></span>
<span class="pl-s"><span class="pl-s">salloc: Granted job allocation 9271058</span></span>
<span class="pl-s"><span class="pl-s"></span></span>
<span class="pl-s"><span class="pl-s">#11 - Run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name</span></span>
<span class="pl-s"><span class="pl-s">[amali010@coreV2-22-007 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta</span></span>
<span class="pl-s"><span class="pl-s">The total number of sequences is 138</span></span>
<span class="pl-s"><span class="pl-s">The average sequence length is 126640</span></span>
<span class="pl-s"><span class="pl-s">The total number of bases is 17476447</span></span>
<span class="pl-s"><span class="pl-s">The minimum sequence length is 1122</span></span>
<span class="pl-s"><span class="pl-s">The maximum sequence length is 562680</span></span>
<span class="pl-s"><span class="pl-s">The N50 is 187037</span></span>
<span class="pl-s"><span class="pl-s">Median Length = 92612</span></span>
<span class="pl-s"><span class="pl-s">contigs &lt; 150bp = 0</span></span>
<span class="pl-s"><span class="pl-s">contigs &gt;= 500bp = 138</span></span>
<span class="pl-s"><span class="pl-s">contigs &gt;= 1000bp = 138</span></span>
<span class="pl-s"><span class="pl-s">contigs &gt;= 2000bp = 135</span></span>
<span class="pl-s"><span class="pl-s"></span></span>
<span class="pl-s"><span class="pl-s">#12 - Did you add all these entries to your notebook file, including the results of the script?</span></span>
<span class="pl-s"><span class="pl-s">Yes!</span></span>
<span class="pl-s"><span class="pl-s"></span></span>
<span class="pl-s"><span class="pl-s">#13 - Push your notebook file to your github page</span></span>
<span class="pl-s"><span class="pl-s">git add README.md</span></span>
<span class="pl-s"><span class="pl-s">git commit -m <span class="pl-pds">'</span></span>updating readme<span class="pl-s"><span class="pl-pds">'</span></span></span>
<span class="pl-s"><span class="pl-s">git push -u origin main</span></span></pre>
                    </div>
                    <p dir="auto">
                     ##Day 03 27-Jan-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Homework day02 (originally meant for 22-Jan-2021 but done on 27-Jan-2021)
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 - Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory
   Kristina-HADB01
   Ivan-HADB02
   KatieP-HADB03
   KatieC-HADB04
   Stephanie-HADB05
   Areej-HADB06
   Andrew-HADB01
   John-HADB02
   dan-All
   
[amali010@coreV2-25-072 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data

#2 - Add the content of your sbatch script to your logfile
[amali010@coreV2-25-072 data]$ cat AMCopyLane06.sh
#!/bin/bash -l

#SBATCH -o AMCopyLane06.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMCopyLane06

cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB06*.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data

#3 - Submit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)
[amali010@coreV2-25-072 data]$ sbatch AMCopyLane06.sh
Submitted batch job 9270443

#4 - Make sure this is all documented on your github notebook
Done

#5 - Write a sbatch script to gunzip all the fastq.gz files
[amali010@coreV2-25-072 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data
[amali010@coreV2-25-072 data]$ cat AMGunzipLane06.sh
#!/bin/bash -l

#SBATCH -o AMGunzipLane06.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMGunzipLane06

gunzip *.fastq.gz

[amali010@coreV2-25-072 data]$ sbatch AMGunzipLane06.sh
Submitted batch job 9270466

#6 - Push your notebook file to your github page
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory</span>
   Kristina-HADB01
   Ivan-HADB02
   KatieP-HADB03
   KatieC-HADB04
   Stephanie-HADB05
   Areej-HADB06
   Andrew-HADB01
   John-HADB02
   dan-All
   
[amali010@coreV2-25-072 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data

<span class="pl-c"><span class="pl-c">#</span>2 - Add the content of your sbatch script to your logfile</span>
[amali010@coreV2-25-072 data]$ cat AMCopyLane06.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMCopyLane06.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMCopyLane06</span>

cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB06<span class="pl-k">*</span>.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data

<span class="pl-c"><span class="pl-c">#</span>3 - Submit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)</span>
[amali010@coreV2-25-072 data]$ sbatch AMCopyLane06.sh
Submitted batch job 9270443

<span class="pl-c"><span class="pl-c">#</span>4 - Make sure this is all documented on your github notebook</span>
Done

<span class="pl-c"><span class="pl-c">#</span>5 - Write a sbatch script to gunzip all the fastq.gz files</span>
[amali010@coreV2-25-072 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data
[amali010@coreV2-25-072 data]$ cat AMGunzipLane06.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMGunzipLane06.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMGunzipLane06</span>

gunzip <span class="pl-k">*</span>.fastq.gz

[amali010@coreV2-25-072 data]$ sbatch AMGunzipLane06.sh
Submitted batch job 9270466

<span class="pl-c"><span class="pl-c">#</span>6 - Push your notebook file to your github page</span>
git add README.md
git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
git push -u origin main</pre>
                    </div>
                    <p dir="auto">
                     ##Day 03 27-Jan-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Homework day03
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[amali010@turing1 areej]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej

#1 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox
[amali010@turing1 areej]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 ./

#2 - mkdir a fastq directory in your data directory and mv all the .fastq files into this directory
[amali010@turing1 areej]$ cd data
[amali010@turing1 data]$ mkdir fastq
[amali010@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data
[amali010@turing1 data]$ mv *.fastq ./fastq/

#3 - cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your  sandbox scripts folder and less the new script and check out the usage statement
[amali010@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/
[amali010@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ../scripts/
[amali010@turing1 data]$ head ../scripts/renamer_advbioinf.py
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column

#4 - Run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand
[amali010@turing1 data]$ cd fastq/
[amali010@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt
mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq
mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq
mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq
mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq
mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq
mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq
mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq
mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq
mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq
mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq
mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq
mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq
mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq
mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq
mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq
mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq
mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq
mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq
mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq
mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq
mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq
mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq
mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq
mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq
mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq
mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq
mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq
mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq
mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq
mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq
mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq
mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq
mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq
mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq
mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq
mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq
mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq
mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq
mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq
mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq
mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq
mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq
mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq
mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq
mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq
mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq
mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq
mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq
mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq
mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq
mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq
mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq
mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq
mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq
mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq
mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq
mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq
mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq
mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq
mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq
mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq
mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq
mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq
mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq
mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq
mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq
mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq
mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq
mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq
mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq
mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq
mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq
mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq
mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq
mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq
mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq
mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq
mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq
mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq
mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq
mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq
mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq
mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq
mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq
mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq
mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq
mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq
mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq
mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq
mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq
mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq
mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq
mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq
mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq
mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq
mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq

#5 - Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print
[amali010@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py

#6 - Write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script
[amali010@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ nano areejsrenamer.sh
[amali010@turing1 fastq]$ cat areejsrenamer.sh
#!/bin/bash -l

#SBATCH -o areejsrenamer.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=FastqRename

../../scripts/renamer_advbioinf.py renamingtable_complete.txt

[amali010@turing1 fastq]$ sbatch ./areejsrenamer.sh
Submitted batch job 9270936
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)

#7 - Make sure this is all documented on your github page
areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)
$ git add README.md

areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)
$ git commit -m 'updating readme'
[main 4326621] updating readme
 1 file changed, 20 insertions(+), 1 deletion(-)

areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)
$ git push -u origin main
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 491 bytes | 245.00 KiB/s, done.
Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/areejmalik/AreejsAdvancedGenomicsLog.git
   aa0237a..4326621  main -&gt; main
Branch 'main' set up to track remote branch 'main' from 'origin'.

#8 - The naming convention for the files is as follows:
SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq
There are 2 sources: Virginia and Rhode Island
There are 2 symbiotic states: Brown and White
	
#9 - Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane
Ok!

#10 - cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory
[amali010@coreV1-22-016 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data
[amali010@coreV1-22-016 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/

#11 - Less/head the new script and check out the usage statement
[amali010@coreV1-22-016 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py
#!/usr/bin/env python
# Written by Dan Barshis

import sys, os

########### Usage #############
# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles
# This should be run from within the folder with all of your original .fastqstrim
# Will quality trim multiple SINGLE-END fastq files
# Things to customize for your particular platform:
# qualoffset = 33 Quality score offset
# -t option in qualitytrim (the lower threshold quality score for trimming)
# -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)

#12 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt 
into the working directory with your fastq files
[amali010@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./

#13 - Make a sbatch script for the Trimclipfilter... script and run it on your fastq files
[amali010@turing1 ~]$ salloc
salloc: Pending job allocation 9270954
salloc: job 9270954 queued and waiting for resources
salloc: job 9270954 has been allocated resources
salloc: Granted job allocation 9270954
NOTE: The job allocation number matches that of the one that is currently the only one running
[amali010@coreV1-22-005 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@coreV1-22-005 fastq]$ nano FullTrimClip.sh
[amali010@coreV1-22-005 fastq]$ cat FullTrimClip.sh
#!/bin/bash -l

#SBATCH -o AMFullTrimclip.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMTrimTest

../../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq

[amali010@coreV1-22-005 fastq]$ sbatch FullTrimClip.sh
Submitted batch job 9270961
[amali010@coreV1-22-005 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270961      main AMTrimTe amali010 PD       0:00      1 (Priority)
           9270954      main       sh amali010  R      18:11      1 coreV1-22-005
[amali010@coreV1-22-005 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270954      main       sh amali010  R      19:42      1 coreV1-22-005
NOTE: I don't know why it keeps failing the job partition I even tried logging 
out the core and remained on turing but that was rejected as well. :(

After multiple attempts!!
[amali010@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ nano amFullTrimClip.sh
[amali010@turing1 fastq]$ cat amFullTrimClip.sh
#!/bin/bash -l

#SBATCH -o amFullTrimclip.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=amTrimFull

../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq
[amali010@turing1 fastq]$ sbatch amFullTrimClip.sh
Submitted batch job 9270966
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R       0:03      1 coreV1-22-005
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R       0:22      1 coreV1-22-005
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R       1:17      1 coreV1-22-005
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R       2:40      1 coreV1-22-005
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R       3:25      1 coreV1-22-005
[amali010@turing1 fastq]$ ls
adapterlist_advbioinf.txt   RI_B_06_18.fastq   VA_B_06_18.fastq
amFullTrimClip.sh           RI_B_07_14.fastq   VA_B_07_14.fastq
amFullTrimclip.txt          RI_B_07_18.fastq   VA_B_07_18.fastq
AMFullTrimclip.txt          RI_B_07_22.fastq   VA_B_07_22.fastq
areejsrenamer.sh            RI_W_06_18.fastq   VA_W_06_18.fastq
AreejsRenamer.sh            RI_W_07_14.fastq   VA_W_07_14.fastq
areejsrenamer.txt           RI_W_07_18.fastq   VA_W_07_18.fastq
FullTrimClip.sh             RI_W_07_22.fastq   VA_W_07_22.fastq
renamingtable_complete.txt  Testing
RI_B_06_18_clipped.fastq    trimclipstats.txt

[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R       5:20      1 coreV1-22-005
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R       9:01      1 coreV1-22-005
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270966      main amTrimFu amali010  R      17:31      1 coreV1-22-005

#14 - Push your notebook file to your github page
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre>[amali010@turing1 areej]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej

<span class="pl-c"><span class="pl-c">#</span>1 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox</span>
[amali010@turing1 areej]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 ./

<span class="pl-c"><span class="pl-c">#</span>2 - mkdir a fastq directory in your data directory and mv all the .fastq files into this directory</span>
[amali010@turing1 areej]$ <span class="pl-c1">cd</span> data
[amali010@turing1 data]$ mkdir fastq
[amali010@turing1 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data
[amali010@turing1 data]$ mv <span class="pl-k">*</span>.fastq ./fastq/

<span class="pl-c"><span class="pl-c">#</span>3 - cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your  sandbox scripts folder and less the new script and check out the usage statement</span>
[amali010@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/
[amali010@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ../scripts/
[amali010@turing1 data]$ head ../scripts/renamer_advbioinf.py
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python</span>
<span class="pl-c"><span class="pl-c">#</span>###usage renamer.py renamingtable</span>
<span class="pl-c"><span class="pl-c">#</span>### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column</span>

<span class="pl-c"><span class="pl-c">#</span>4 - Run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand</span>
[amali010@turing1 data]$ <span class="pl-c1">cd</span> fastq/
[amali010@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt
mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq
mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq
mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq
mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq
mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq
mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq
mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq
mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq
mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq
mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq
mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq
mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq
mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq
mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq
mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq
mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq
mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq
mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq
mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq
mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq
mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq
mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq
mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq
mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq
mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq
mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq
mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq
mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq
mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq
mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq
mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq
mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq
mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq
mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq
mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq
mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq
mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq
mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq
mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq
mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq
mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq
mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq
mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq
mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq
mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq
mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq
mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq
mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq
mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq
mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq
mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq
mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq
mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq
mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq
mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq
mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq
mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq
mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq
mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq
mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq
mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq
mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq
mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq
mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq
mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq
mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq
mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq
mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq
mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq
mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq
mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq
mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq
mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq
mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq
mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq
mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq
mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq
mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq
mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq
mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq
mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq
mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq
mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq
mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq
mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq
mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq
mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq
mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq
mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq
mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq
mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq
mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq
mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq
mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq
mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq
mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq

<span class="pl-c"><span class="pl-c">#</span>5 - Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print</span>
[amali010@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py

<span class="pl-c"><span class="pl-c">#</span>6 - Write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script</span>
[amali010@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ nano areejsrenamer.sh
[amali010@turing1 fastq]$ cat areejsrenamer.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o areejsrenamer.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=FastqRename</span>

../../scripts/renamer_advbioinf.py renamingtable_complete.txt

[amali010@turing1 fastq]$ sbatch ./areejsrenamer.sh
Submitted batch job 9270936
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)

<span class="pl-c"><span class="pl-c">#</span>7 - Make sure this is all documented on your github page</span>
areej@DESKTOP-RF6VF4N MINGW64 <span class="pl-k">~</span>/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)
$ git add README.md

areej@DESKTOP-RF6VF4N MINGW64 <span class="pl-k">~</span>/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)
$ git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
[main 4326621] updating readme
 1 file changed, 20 insertions(+), 1 deletion(-)

areej@DESKTOP-RF6VF4N MINGW64 <span class="pl-k">~</span>/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/AreejsAdvancedGenomicsLog (main)
$ git push -u origin main
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 491 bytes <span class="pl-k">|</span> 245.00 KiB/s, done.
Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 <span class="pl-k">local</span> object.
To https://github.com/areejmalik/AreejsAdvancedGenomicsLog.git
   aa0237a..4326621  main -<span class="pl-k">&gt;</span> main
Branch <span class="pl-s"><span class="pl-pds">'</span>main<span class="pl-pds">'</span></span> <span class="pl-c1">set</span> up to track remote branch <span class="pl-s"><span class="pl-pds">'</span>main<span class="pl-pds">'</span></span> from <span class="pl-s"><span class="pl-pds">'</span>origin<span class="pl-pds">'</span></span>.

<span class="pl-c"><span class="pl-c">#</span>8 - The naming convention for the files is as follows:</span>
SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq
There are 2 sources: Virginia and Rhode Island
There are 2 symbiotic states: Brown and White
	
<span class="pl-c"><span class="pl-c">#</span>9 - Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane</span>
Ok<span class="pl-k">!</span>

<span class="pl-c"><span class="pl-c">#</span>10 - cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory</span>
[amali010@coreV1-22-016 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data
[amali010@coreV1-22-016 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/

<span class="pl-c"><span class="pl-c">#</span>11 - Less/head the new script and check out the usage statement</span>
[amali010@coreV1-22-016 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python</span>
<span class="pl-c"><span class="pl-c">#</span> Written by Dan Barshis</span>

import sys, os

<span class="pl-c"><span class="pl-c">#</span>########## Usage #############</span>
<span class="pl-c"><span class="pl-c">#</span> Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles</span>
<span class="pl-c"><span class="pl-c">#</span> This should be run from within the folder with all of your original .fastqstrim</span>
<span class="pl-c"><span class="pl-c">#</span> Will quality trim multiple SINGLE-END fastq files</span>
<span class="pl-c"><span class="pl-c">#</span> Things to customize for your particular platform:</span>
<span class="pl-c"><span class="pl-c">#</span> qualoffset = 33 Quality score offset</span>
<span class="pl-c"><span class="pl-c">#</span> -t option in qualitytrim (the lower threshold quality score for trimming)</span>
<span class="pl-c"><span class="pl-c">#</span> -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)</span>

<span class="pl-c"><span class="pl-c">#</span>12 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt </span>
into the working directory with your fastq files
[amali010@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./

<span class="pl-c"><span class="pl-c">#</span>13 - Make a sbatch script for the Trimclipfilter... script and run it on your fastq files</span>
[amali010@turing1 <span class="pl-k">~</span>]$ salloc
salloc: Pending job allocation 9270954
salloc: job 9270954 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9270954 has been allocated resources
salloc: Granted job allocation 9270954
NOTE: The job allocation number matches that of the one that is currently the only one running
[amali010@coreV1-22-005 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@coreV1-22-005 fastq]$ nano FullTrimClip.sh
[amali010@coreV1-22-005 fastq]$ cat FullTrimClip.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMFullTrimclip.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMTrimTest</span>

../../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt <span class="pl-k">*</span>.fastq

[amali010@coreV1-22-005 fastq]$ sbatch FullTrimClip.sh
Submitted batch job 9270961
[amali010@coreV1-22-005 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270961      main AMTrimTe amali010 PD       0:00      1 (Priority)
           9270954      main       sh amali010  R      18:11      1 coreV1-22-005
[amali010@coreV1-22-005 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9270954      main       sh amali010  R      19:42      1 coreV1-22-005
NOTE: I don<span class="pl-s"><span class="pl-pds">'</span>t know why it keeps failing the job partition I even tried logging </span>
<span class="pl-s">out the core and remained on turing but that was rejected as well. :(</span>
<span class="pl-s"></span>
<span class="pl-s">After multiple attempts!!</span>
<span class="pl-s">[amali010@turing1 fastq]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq</span>
<span class="pl-s">[amali010@turing1 fastq]$ nano amFullTrimClip.sh</span>
<span class="pl-s">[amali010@turing1 fastq]$ cat amFullTrimClip.sh</span>
<span class="pl-s">#!/bin/bash -l</span>
<span class="pl-s"></span>
<span class="pl-s">#SBATCH -o amFullTrimclip.txt</span>
<span class="pl-s">#SBATCH -n 1</span>
<span class="pl-s">#SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-s">#SBATCH --mail-type=END</span>
<span class="pl-s">#SBATCH --job-name=amTrimFull</span>
<span class="pl-s"></span>
<span class="pl-s">../../scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq</span>
<span class="pl-s">[amali010@turing1 fastq]$ sbatch amFullTrimClip.sh</span>
<span class="pl-s">Submitted batch job 9270966</span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R       0:03      1 coreV1-22-005</span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R       0:22      1 coreV1-22-005</span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R       1:17      1 coreV1-22-005</span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R       2:40      1 coreV1-22-005</span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R       3:25      1 coreV1-22-005</span>
<span class="pl-s">[amali010@turing1 fastq]$ ls</span>
<span class="pl-s">adapterlist_advbioinf.txt   RI_B_06_18.fastq   VA_B_06_18.fastq</span>
<span class="pl-s">amFullTrimClip.sh           RI_B_07_14.fastq   VA_B_07_14.fastq</span>
<span class="pl-s">amFullTrimclip.txt          RI_B_07_18.fastq   VA_B_07_18.fastq</span>
<span class="pl-s">AMFullTrimclip.txt          RI_B_07_22.fastq   VA_B_07_22.fastq</span>
<span class="pl-s">areejsrenamer.sh            RI_W_06_18.fastq   VA_W_06_18.fastq</span>
<span class="pl-s">AreejsRenamer.sh            RI_W_07_14.fastq   VA_W_07_14.fastq</span>
<span class="pl-s">areejsrenamer.txt           RI_W_07_18.fastq   VA_W_07_18.fastq</span>
<span class="pl-s">FullTrimClip.sh             RI_W_07_22.fastq   VA_W_07_22.fastq</span>
<span class="pl-s">renamingtable_complete.txt  Testing</span>
<span class="pl-s">RI_B_06_18_clipped.fastq    trimclipstats.txt</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R       5:20      1 coreV1-22-005</span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R       9:01      1 coreV1-22-005</span>
<span class="pl-s">[amali010@turing1 fastq]$ squeue -u amali010</span>
<span class="pl-s">             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)</span>
<span class="pl-s">           9270966      main amTrimFu amali010  R      17:31      1 coreV1-22-005</span>
<span class="pl-s"></span>
<span class="pl-s">#14 - Push your notebook file to your github page</span>
<span class="pl-s">git add README.md</span>
<span class="pl-s">git commit -m <span class="pl-pds">'</span></span>updating readme<span class="pl-s"><span class="pl-pds">'</span></span>
<span class="pl-s">git push -u origin main</span></pre>
                    </div>
                    <p dir="auto">
                     ##Day 04 29-Jan-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Homework day04
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Checked the success of the job submission from the day before
[amali010@turing1 areej]$ cd data
[amali010@turing1 data]$ cd fastq
[amali010@turing1 fastq]$ ls
[amali010@turing1 fastq]$ ls filteringstats/
[amali010@turing1 fastq]$ ls -alh filteringstats/

#1 - Add your trimclipstats.txt output to the full class 

#1a - Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps
[amali010@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015

This script takes a stats output file from fastx_clipper and converts it into a table.

Usage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt

Options (-c and -v must be listed separately to run together):
-h      Display this help message
-c      Use comma delimiter instead of tabs
-v      Verbose mode (print steps to stdout)

#1b - Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt
[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt AREEJ_trimclipstatsout.txt

#1c -  Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt
[amali010@turing1 fastq]$ tail -n +2 AREEJ_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt

#2 - Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)
Ok!

#3 - Write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data
[amali010@turing1 fastq]$ cd QCFastqs
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ nano ajbowtie2.sh
[amali010@turing1 QCFastqs]$ ls
ajbowtie2.sh                     VA_B_06_18_clippedtrimmed.fastq
RI_B_06_18_clippedtrimmed.fastq  VA_B_07_14_clippedtrimmed.fastq
RI_B_07_14_clippedtrimmed.fastq  VA_B_07_18_clippedtrimmed.fastq
RI_B_07_18_clippedtrimmed.fastq  VA_B_07_22_clippedtrimmed.fastq
RI_B_07_22_clippedtrimmed.fastq  VA_W_06_18_clippedtrimmed.fastq
RI_W_06_18_clippedtrimmed.fastq  VA_W_07_14_clippedtrimmed.fastq
RI_W_07_14_clippedtrimmed.fastq  VA_W_07_18_clippedtrimmed.fastq
RI_W_07_18_clippedtrimmed.fastq  VA_W_07_22_clippedtrimmed.fastq
RI_W_07_22_clippedtrimmed.fastq
[amali010@turing1 QCFastqs]$ cat ajbowtie2.sh
#!/bin/bash -l

#SBATCH -o ambowtie2.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=ambowtie

module load bowtie2/2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \
--rg SM:${i%_clippedtrimmed.fastq} \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \
&gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done

[amali010@turing1 QCFastqs]$ sbatch ajbowtie2.sh
Submitted batch job 9271117
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271117      main ambowtie amali010 PD       0:00      1 (Priority)
  [amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271117      main ambowtie amali010  R       1:49      1 coreV2-25-049
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271117      main ambowtie amali010  R       5:07      1 coreV2-25-049

#4 - Submit and add everything to your logfile
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre>Checked the success of the job submission from the day before
[amali010@turing1 areej]$ <span class="pl-c1">cd</span> data
[amali010@turing1 data]$ <span class="pl-c1">cd</span> fastq
[amali010@turing1 fastq]$ ls
[amali010@turing1 fastq]$ ls filteringstats/
[amali010@turing1 fastq]$ ls -alh filteringstats/

<span class="pl-c"><span class="pl-c">#</span>1 - Add your trimclipstats.txt output to the full class </span>

<span class="pl-c"><span class="pl-c">#</span>1a - Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps</span>
[amali010@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq
[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015

This script takes a stats output file from fastx_clipper and converts it into a table.

Usage: Schafran_trimstatstable.py [-c, -v, <span class="pl-k">-h</span>] inputfile.txt outputfile.txt

Options (-c and -v must be listed separately to run together):
-h      Display this <span class="pl-c1">help</span> message
-c      Use comma delimiter instead of tabs
-v      Verbose mode (print steps to stdout)

<span class="pl-c"><span class="pl-c">#</span>1b - Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt</span>
[amali010@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt AREEJ_trimclipstatsout.txt

<span class="pl-c"><span class="pl-c">#</span>1c -  Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt</span>
[amali010@turing1 fastq]$ tail -n +2 AREEJ_trimclipstatsout.txt <span class="pl-k">&gt;&gt;</span> /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt

<span class="pl-c"><span class="pl-c">#</span>2 - Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)</span>
Ok<span class="pl-k">!</span>

<span class="pl-c"><span class="pl-c">#</span>3 - Write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data</span>
[amali010@turing1 fastq]$ <span class="pl-c1">cd</span> QCFastqs
[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ nano ajbowtie2.sh
[amali010@turing1 QCFastqs]$ ls
ajbowtie2.sh                     VA_B_06_18_clippedtrimmed.fastq
RI_B_06_18_clippedtrimmed.fastq  VA_B_07_14_clippedtrimmed.fastq
RI_B_07_14_clippedtrimmed.fastq  VA_B_07_18_clippedtrimmed.fastq
RI_B_07_18_clippedtrimmed.fastq  VA_B_07_22_clippedtrimmed.fastq
RI_B_07_22_clippedtrimmed.fastq  VA_W_06_18_clippedtrimmed.fastq
RI_W_06_18_clippedtrimmed.fastq  VA_W_07_14_clippedtrimmed.fastq
RI_W_07_14_clippedtrimmed.fastq  VA_W_07_18_clippedtrimmed.fastq
RI_W_07_18_clippedtrimmed.fastq  VA_W_07_22_clippedtrimmed.fastq
RI_W_07_22_clippedtrimmed.fastq
[amali010@turing1 QCFastqs]$ cat ajbowtie2.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o ambowtie2.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=ambowtie</span>

module load bowtie2/2.4
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>_clippedtrimmed.fastq<span class="pl-k">;</span> <span class="pl-k">do</span> bowtie2 --rg-id <span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmed.fastq}</span> \
--rg SM:<span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmed.fastq}</span> \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U <span class="pl-smi">$i</span> \
<span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmedfilterd.fastq}</span>.sam --no-unal -k 5<span class="pl-k">;</span> <span class="pl-k">done</span>

[amali010@turing1 QCFastqs]$ sbatch ajbowtie2.sh
Submitted batch job 9271117
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271117      main ambowtie amali010 PD       0:00      1 (Priority)
  [amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271117      main ambowtie amali010  R       1:49      1 coreV2-25-049
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271117      main ambowtie amali010  R       5:07      1 coreV2-25-049

<span class="pl-c"><span class="pl-c">#</span>4 - Submit and add everything to your logfile</span>
git add README.md
git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
git push -u origin main</pre>
                    </div>
                    <p dir="auto">
                     ##Day 05 03-Feb-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day05
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 - Team up with a partner for this one and work only on a combined set of data for your two lanes of sequences
Partner: Andrew
Combined set of data lanes: 1, 6, &amp; 7

#2 - Run the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together
Done!

#3 - Modify the below sbatch script (note the differences in the header compared to the previous one)
Original script (used from homework_day05.txt file)

Edited script
New name: APAMAssemble.sh
Location: /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/
#!/bin/bash -l

#SBATCH -o APAMAssemble.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APAMAssemble

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32

#4 - Check https://trinityrnaseq.github.io/ for usage info
Done!

#5 - Submit your trinity script
Note: It was done by Andrew, so the following is from his notebook.
[apear012@turing1 apam]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam
[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh 
Submitted batch job 9272358

[amali010@turing1 apam]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
[amali010@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)
           9272334      main       sh apear012  R    1:19:50      1 coreV2-25-002
           9272357      main       sh apear012  R       6:49      1 coreV2-22-007
[amali010@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)
           9272334      main       sh apear012  R    1:20:18      1 coreV2-25-002
           9272357      main       sh apear012  R       7:17      1 coreV2-22-007
Note: Job failed after approximately 15 minutes :(

Edited APAMAssemble.sh file (removed --normalize_reads). This was also done by Andrew.
#!/bin/bash -l

#SBATCH -o APAMAssemble.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=pearsoac@evms.edu
#SBATCH --mail-type=END
#SBATCH --job-name=APAMAssemble

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32

[apear012@turing1 apam]$ sbatch APAMAssemble.sh
Submitted batch job 9272385
[apear012@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 
           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003
           
#6 - Submit and add everything to your logfile
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Team up with a partner for this one and work only on a combined set of data for your two lanes of sequences</span>
Partner: Andrew
Combined <span class="pl-c1">set</span> of data lanes: 1, 6, <span class="pl-k">&amp;</span> 7

<span class="pl-c"><span class="pl-c">#</span>2 - Run the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together</span>
Done<span class="pl-k">!</span>

<span class="pl-c"><span class="pl-c">#</span>3 - Modify the below sbatch script (note the differences in the header compared to the previous one)</span>
Original script (used from homework_day05.txt file)

Edited script
New name: APAMAssemble.sh
Location: /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam/
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o APAMAssemble.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 32</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -p himem</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=pearsoac@evms.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=APAMAssemble</span>

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32

<span class="pl-c"><span class="pl-c">#</span>4 - Check https://trinityrnaseq.github.io/ for usage info</span>
Done<span class="pl-k">!</span>

<span class="pl-c"><span class="pl-c">#</span>5 - Submit your trinity script</span>
Note: It was <span class="pl-k">done</span> by Andrew, so the following is from his notebook.
[apear012@turing1 apam]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/apearson/data/fastq/QCFastqs/apam
[apear012@coreV2-22-007 apam]$ sbatch APAMAssemble.sh 
Submitted batch job 9272358

[amali010@turing1 apam]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
[amali010@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)
           9272334      main       sh apear012  R    1:19:50      1 coreV2-25-002
           9272357      main       sh apear012  R       6:49      1 coreV2-22-007
[amali010@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9272358     himem APAMAsse apear012 PD       0:00      1 (Resources)
           9272334      main       sh apear012  R    1:20:18      1 coreV2-25-002
           9272357      main       sh apear012  R       7:17      1 coreV2-22-007
Note: Job failed after approximately 15 minutes :(

Edited APAMAssemble.sh file (removed --normalize_reads). This was also <span class="pl-k">done</span> by Andrew.
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o APAMAssemble.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 32</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -p himem</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=pearsoac@evms.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=APAMAssemble</span>

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --single RI_B_01_14_clippedtrimmed.fastq,RI_W_06_18_clippedtrimmed.fastq,VA_B_07_22_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_W_07_14_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_W_07_18_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,RI_B_06_18_clippedtrimmed.fastq,RI_W_07_22_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,RI_B_07_14_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,RI_B_07_18_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_W_06_18_clippedtrimmed.fastq,RI_B_07_22_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_W_07_14_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_W_07_18_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,VA_B_06_18_clippedtrimmed.fastq,VA_W_07_22_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,VA_B_07_14_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,VA_B_07_18_clippedtrimmed.fastq --CPU 32

[apear012@turing1 apam]$ sbatch APAMAssemble.sh
Submitted batch job 9272385
[apear012@turing1 apam]$ squeue -u apear012
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9272357      main       sh apear012  R    1:15:39      1 coreV2-22-007 
           9272385     himem APAMAsse apear012  R       0:14      1 coreV4-21-himem-003
           
<span class="pl-c"><span class="pl-c">#</span>6 - Submit and add everything to your logfile</span>
git add README.md
git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
git push -u origin main</pre>
                    </div>
                    <p dir="auto">
                     ##Day 05 03-Feb-2021
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="No homework" dir="auto">
                     <pre>No homework</pre>
                    </div>
                    <p dir="auto">
                     ##Day 06 05-Feb-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day06
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 - Start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly
[amali010@turing1 djbtestassembly]$ salloc
salloc: Pending job allocation 9272860
salloc: job 9272860 queued and waiting for resources
salloc: job 9272860 has been allocated resources
salloc: Granted job allocation 9272860

[amali010@turing1 djbtestassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/djbtestassembly
[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta


################################
## Counts of transcripts, etc.
################################
Total trinity 'genes':  20980
Total trinity transcripts:      21992
Percent GC: 46.21

########################################
Stats based on ALL transcript contigs:
########################################

        Contig N10: 1178
        Contig N20: 694
        Contig N30: 514
        Contig N40: 414
        Contig N50: 347

        Median contig length: 273
        Average contig: 356.95
        Total assembled bases: 7850006


#####################################################
## Stats based on ONLY LONGEST ISOFORM per 'GENE':
#####################################################

        Contig N10: 1027
        Contig N20: 643
        Contig N30: 486
        Contig N40: 397
        Contig N50: 336

        Median contig length: 271
        Average contig: 347.72
        Total assembled bases: 7295061

#2 - Compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile
[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
The total number of sequences is 15079
The average sequence length is 876
The total number of bases is 13210470
The minimum sequence length is 500
The maximum sequence length is 10795
The N50 is 881
Median Length = 578
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 15079
contigs &gt;= 1000bp = 3660
contigs &gt;= 2000bp = 536

#3 - Less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:

#3a - The mean percent &quot;overall alignment rate&quot;
[amali010@coreV2-22-007 QCFastqs]$ less ambowtie2.txt
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV2-22-007 QCFastqs]$ grep &quot;overall alignment rate&quot; ambowtie2.txt
1.87% overall alignment rate
1.53% overall alignment rate
1.46% overall alignment rate
1.92% overall alignment rate
1.65% overall alignment rate
1.48% overall alignment rate
1.25% overall alignment rate
1.60% overall alignment rate
1.94% overall alignment rate
2.00% overall alignment rate
1.67% overall alignment rate
2.09% overall alignment rate
1.62% overall alignment rate
1.54% overall alignment rate
1.70% overall alignment rate
1.41% overall alignment rate

[amali010@coreV2-22-007 QCFastqs]$ grep &quot;overall alignment rate&quot; ambowtie2.txt | cut -d &quot;%&quot; -f 1
1.87
1.53
1.46
1.92
1.65
1.48
1.25
1.60
1.94
2.00
1.67
2.09
1.62
1.54
1.70
1.41

#3b - The mean percent reads &quot;aligned exactly 1 time&quot;
[amali010@coreV2-22-007 QCFastqs]$ grep &quot;aligned exactly 1 time&quot; ambowtie2.txt
    312173 (1.32%) aligned exactly 1 time
    279609 (1.10%) aligned exactly 1 time
    226384 (0.96%) aligned exactly 1 time
    29903 (1.24%) aligned exactly 1 time
    210196 (1.09%) aligned exactly 1 time
    359056 (0.96%) aligned exactly 1 time
    225886 (0.88%) aligned exactly 1 time
    183780 (1.06%) aligned exactly 1 time
    207948 (1.27%) aligned exactly 1 time
    228135 (1.33%) aligned exactly 1 time
    250518 (1.09%) aligned exactly 1 time
    195159 (1.31%) aligned exactly 1 time
    195302 (1.13%) aligned exactly 1 time
    234155 (1.07%) aligned exactly 1 time
    176134 (0.87%) aligned exactly 1 time
    243143 (0.89%) aligned exactly 1 time

[amali010@coreV2-22-007 QCFastqs]$ grep &quot;aligned exactly 1 time&quot; ambowtie2.txt | cut -d &quot;%&quot; -f 1 | cut -d &quot;(&quot; -f 2
1.32
1.10
0.96
1.24
1.09
0.96
0.88
1.06
1.27
1.33
1.09
1.31
1.13
1.07
0.87
0.89

#3c - The mean number of reads &quot;aligned exactly 1 time&quot;
[amali010@turing1 QCFastqs]$ grep &quot;aligned exactly 1 time&quot; ambowtie2.txt
    312173 (1.32%) aligned exactly 1 time
    279609 (1.10%) aligned exactly 1 time
    226384 (0.96%) aligned exactly 1 time
    29903 (1.24%) aligned exactly 1 time
    210196 (1.09%) aligned exactly 1 time
    359056 (0.96%) aligned exactly 1 time
    225886 (0.88%) aligned exactly 1 time
    183780 (1.06%) aligned exactly 1 time
    207948 (1.27%) aligned exactly 1 time
    228135 (1.33%) aligned exactly 1 time
    250518 (1.09%) aligned exactly 1 time
    195159 (1.31%) aligned exactly 1 time
    195302 (1.13%) aligned exactly 1 time
    234155 (1.07%) aligned exactly 1 time
    176134 (0.87%) aligned exactly 1 time
    243143 (0.89%) aligned exactly 1 time
    
    [amali010@coreV2-22-007 QCFastqs]$ grep &quot;aligned exactly 1 time&quot; ambowtie2.txt | cut -d &quot; &quot; -f 5
312173
279609
226384
29903
210196
359056
225886
183780
207948
228135
250518
195159
195302
234155
176134
243143

#3d - The mean percent reads &quot;aligned &gt;1 times&quot;
[amali010@coreV2-22-007 QCFastqs]$ grep &quot;aligned &gt;1 times&quot; ambowtie2.txt
    131960 (0.56%) aligned &gt;1 times
    109989 (0.43%) aligned &gt;1 times
    119408 (0.50%) aligned &gt;1 times
    16432 (0.68%) aligned &gt;1 times
    107800 (0.56%) aligned &gt;1 times
    194361 (0.52%) aligned &gt;1 times
    95504 (0.37%) aligned &gt;1 times
    93466 (0.54%) aligned &gt;1 times
    110827 (0.68%) aligned &gt;1 times
    115424 (0.67%) aligned &gt;1 times
    131960 (0.58%) aligned &gt;1 times
    116526 (0.78%) aligned &gt;1 times
    83427 (0.48%) aligned &gt;1 times
    103215 (0.47%) aligned &gt;1 times
    169138 (0.83%) aligned &gt;1 times
    140557 (0.52%) aligned &gt;1 times

[amali010@coreV2-22-007 QCFastqs]$ grep &quot;aligned &gt;1 times&quot; ambowtie2.txt | cut -d &quot;%&quot; -f 1 | cut -d &quot;(&quot; -f 2
0.56
0.43
0.50
0.68
0.56
0.52
0.37
0.54
0.68
0.67
0.58
0.78
0.48
0.47
0.83
0.52
hint use grep and paste into excel

#4 - Add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:
LaneX_yourinitials	b-the mean percent &quot;overall alignment rate&quot;	c-the mean percent reads &quot;aligned exactly 1 time&quot;	d-the mean number of reads &quot;aligned exactly 1 time&quot;	e-the mean percent reads &quot;aligned &gt;1 times&quot;
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV2-22-007 QCFastqs]$ nano alignstatsam.txt
[amali010@coreV2-22-007 QCFastqs]$ cat alignstatsam.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
[amali010@coreV2-22-007 QCFastqs]$ tail -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
LaneX   overallalignmentrate    percsinglyaligned       numsinglyaligned        percmultiplyaligned
LaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857
LaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857
Lane1AP 5.378125        1.3725  300361.8125     4.005
lane5_sg        2.03%   1.11%   235455.75       0.92%
LaneX_jcw       2.495714286     1.251428571     247656.7143     1.247142857
Lane4_KCrid     2.48    1.30    289072.88       1.18
LaneX_am 1.670625       1.098125        222342.5625     0.573125

#5 - Data cleanup and archiving:
#5a - mv your Trinity.fast file from your trinity_out_dir to a folder called testassembly in your data directory
Note: our job was still running at the time. So to avoid confusion I decided to work with the Trinity.fasta file that was copied into my data directory during class.
[amali010@turing1 data]$ mkdir testassembly
[amali010@turing1 data]$ cd djbtestassembly/
[amali010@turing1 djbtestassembly]$ ls
Trinity.fasta
[amali010@turing1 djbtestassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly/
[amali010@turing1 testassembly]$ ls
Trinity.fasta

#5b - Set up a sbatch script to:
	rm -r YOURtrinity_out_dir (don't have one so this will be skipped in the sbatch script)
	rm -r YOURoriginalfastqs
	rm -r YOURfilterstats # as long as you've already appended your filteringstats output to the class as part of homework_day04.txt
original script (used from homework_ day06.txt file)
#!/bin/bash -l

#SBATCH -o OUTFILENAME.txt
#SBATCH -n NUMTHREADS(1 if not multi-threaded)
#SBATCH --mail-user=YOUREMAIL@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/QCFastqs/21sampleassembly_trinity_out_dir
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/filteringstats

[amali010@turing1 originalfastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs

[amali010@turing1 filteringstats]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats

[amali010@turing1 fastq]$ nano AMCleanup.sh
[amali010@turing1 fastq]$ head AMCleanup.sh
#!/bin/bash -l

#SBATCH -o AMCleanup.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMCleanup

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats

[amali010@turing1 fastq]$ sbatch AMCleanup.sh
Submitted batch job 9273090
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273090      main AMCleanu amali010  R       0:01      1 coreV2-22-007

#6 - Submit a blast against sprot from your testassembly folder using the following command:
#!/bin/bash -l

#SBATCH -o OUTFILENAME.txt
#SBATCH -n 6         
#SBATCH --mail-user=EMAIL
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6

[amali010@turing1 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly
[amali010@turing1 testassembly]$ nano AMBlast.sh
[amali010@turing1 testassembly]$ cat AMBlast.sh
#!/bin/bash -l

#SBATCH -o AMBlast.txt
#SBATCH -n 6
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMBlast

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
[amali010@turing1 testassembly]$ sbatch AMBlast.sh
Submitted batch job 9273094
[amali010@turing1 testassembly]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R       0:03      1 coreV2-22-028
[amali010@turing1 areej]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R      58:38      1 coreV2-22-028

#7 - Head one of your .sam files to look at the header
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ head RI_B_06_18_clippedtrimmed.fastq.sam
@HD     VN:1.0  SO:unsorted
@SQ     SN:TR78|c0_g1_i1_coral  LN:1109
@SQ     SN:TR78|c0_g2_i1_coral  LN:1109
@SQ     SN:TR79|c0_g1_i1_coral  LN:610
@SQ     SN:TR79|c0_g2_i1_coral  LN:1549
@SQ     SN:TR87|c0_g1_i1_coral  LN:732
@SQ     SN:TR93|c0_g1_i1_coral  LN:550
@SQ     SN:TR101|c0_g1_i1_coral LN:673
@SQ     SN:TR104|c0_g1_i1_coral LN:607
@SQ     SN:TR105|c0_g1_i1_coral LN:587

#8 - grep -v '@' your.sam | head to look at the sequence read lines, 
why does this work to exclude the header lines?
[amali010@turing1 QCFastqs]$ grep -v &quot;@&quot; RI_B_06_18_clippedtrimmed.fastq.sam | head
K00188:59:HMTFHBBXX:7:1101:12845:1666   16      TR13647|c0_g1_i1_coral  40      255
51M     *       0       0       TTTGGATTCTTGATCACGGTATGATCGATAAAACACAGACCTGTAAAGTCC
JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XN:i:0  XM:i:0  XO:i:0      XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:17645:1683   16      TR43786|c0_g1_i16_coral 125     1  51M      *       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC
JJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:17645:1683   272     TR43786|c0_g1_i5_coral  151     255
51M     *       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC
JJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:20750:1683   0       TR24696|c0_g1_i2_coral  329     1  51M      *       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC
AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:20750:1683   256     TR24696|c0_g1_i3_coral  329     255
51M     *       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC
AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:22658:1683   16      TR43764|c4_g1_i1_coral  1687    255
51M     *       0       0       TCCATTCCTAATCCGTCTGTTAGTGGTCTGCAGTTCCACGCCGTCTTTTCG
JJJJJJJFJJFJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJFJFFAAA     AS:i:-17        XN:i:0  XM:i:3      XO:i:0  XG:i:0  NM:i:3  MD:Z:25T7C14G2  YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:27854:1683   16      TR29978|c0_g2_i1_coral  10      255
51M     *       0       0       GGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAG
JJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-28        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:30G1C12G1C0A2      YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:2392:1701    0       TR35685|c1_g2_i1_coral  534     255
21M     *       0       0       TGACAACCCCGTCTGTGGCGC   AAFFFFJJJJJFFJJJJJFFJ   AS:i:-6     XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:7A13       YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:19705:1701   16      TR29978|c0_g2_i1_coral  14      255
51M     *       0       0       TCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAGCGGG
JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-30        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:26G1C12G1C0A6      YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:23094:1701   16      TR29978|c0_g2_i1_coral  5       255
51M     *       0       0       CTGGGGGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAA
JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-29        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:2A0A31G1C12G0      YT:Z:UU RG:Z:RI_B_06_18

Header lines include '@'. The -v in grep -v excludes '@' and therefore the header lines as well.

#9 - In an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.
[amali010@turing1 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_06_18_clippedtrimmed.fastq.sam RI_B_07_18_clippedtrimmed.fastq.sam RI_W_07_18_clippedtrimmed.fastq.sam
RI_B_06_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
        read reverse strand
        not primary alignment
256 :
        not primary alignment
16 :
        read reverse strand
RI_B_07_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
        read reverse strand
        not primary alignment
256 :
        not primary alignment
16 :
        read reverse strand
RI_W_07_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
        read reverse strand
        not primary alignment
256 :
        not primary alignment
16 :
        read reverse strand

#10 - we need to run the read sorting step required for SNP calling, 
so if you have time, set up and run the following script on your .sam files to finish before Wednesday
#!/bin/bash -l

#SBATCH -o OUTFILENAME.txt
#SBATCH -n 1         
#SBATCH --mail-user=EMAIL
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done

[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ nano AMReadSort.sh
[amali010@turing1 QCFastqs]$ cat AMReadSort.sh
#!/bin/bash -l

#SBATCH -o AMReadSort.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMReadSort

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam

done

[amali010@turing1 QCFastqs]$ sbatch AMReadSort.sh
Submitted batch job 9273103
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R    1:21:32      1 coreV2-22-028
           9273103      main AMReadSo amali010  R       0:04      1 coreV3-23-020
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R    1:22:36      1 coreV2-22-028
           9273103      main AMReadSo amali010  R       1:08      1 coreV3-23-020
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R    1:25:19      1 coreV2-22-028

#11 - submit and add everything to your logfile
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly</span>
[amali010@turing1 djbtestassembly]$ salloc
salloc: Pending job allocation 9272860
salloc: job 9272860 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9272860 has been allocated resources
salloc: Granted job allocation 9272860

[amali010@turing1 djbtestassembly]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/djbtestassembly
[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta


<span class="pl-c"><span class="pl-c">#</span>###############################</span>
<span class="pl-c"><span class="pl-c">#</span># Counts of transcripts, etc.</span>
<span class="pl-c"><span class="pl-c">#</span>###############################</span>
Total trinity <span class="pl-s"><span class="pl-pds">'</span>genes<span class="pl-pds">'</span></span>:  20980
Total trinity transcripts:      21992
Percent GC: 46.21

<span class="pl-c"><span class="pl-c">#</span>#######################################</span>
Stats based on ALL transcript contigs:
<span class="pl-c"><span class="pl-c">#</span>#######################################</span>

        Contig N10: 1178
        Contig N20: 694
        Contig N30: 514
        Contig N40: 414
        Contig N50: 347

        Median contig length: 273
        Average contig: 356.95
        Total assembled bases: 7850006


<span class="pl-c"><span class="pl-c">#</span>####################################################</span>
<span class="pl-c"><span class="pl-c">#</span># Stats based on ONLY LONGEST ISOFORM per 'GENE':</span>
<span class="pl-c"><span class="pl-c">#</span>####################################################</span>

        Contig N10: 1027
        Contig N20: 643
        Contig N30: 486
        Contig N40: 397
        Contig N50: 336

        Median contig length: 271
        Average contig: 347.72
        Total assembled bases: 7295061

<span class="pl-c"><span class="pl-c">#</span>2 - Compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile</span>
[amali010@coreV2-22-007 djbtestassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
The total number of sequences is 15079
The average sequence length is 876
The total number of bases is 13210470
The minimum sequence length is 500
The maximum sequence length is 10795
The N50 is 881
Median Length = 578
contigs <span class="pl-k">&lt;</span> 150bp = 0
contigs <span class="pl-k">&gt;</span>= 500bp = 15079
contigs <span class="pl-k">&gt;</span>= 1000bp = 3660
contigs <span class="pl-k">&gt;</span>= 2000bp = 536

<span class="pl-c"><span class="pl-c">#</span>3 - Less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:</span>

<span class="pl-c"><span class="pl-c">#</span>3a - The mean percent "overall alignment rate"</span>
[amali010@coreV2-22-007 QCFastqs]$ less ambowtie2.txt
[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV2-22-007 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span> ambowtie2.txt
1.87% overall alignment rate
1.53% overall alignment rate
1.46% overall alignment rate
1.92% overall alignment rate
1.65% overall alignment rate
1.48% overall alignment rate
1.25% overall alignment rate
1.60% overall alignment rate
1.94% overall alignment rate
2.00% overall alignment rate
1.67% overall alignment rate
2.09% overall alignment rate
1.62% overall alignment rate
1.54% overall alignment rate
1.70% overall alignment rate
1.41% overall alignment rate

[amali010@coreV2-22-007 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span> ambowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1
1.87
1.53
1.46
1.92
1.65
1.48
1.25
1.60
1.94
2.00
1.67
2.09
1.62
1.54
1.70
1.41

<span class="pl-c"><span class="pl-c">#</span>3b - The mean percent reads "aligned exactly 1 time"</span>
[amali010@coreV2-22-007 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> ambowtie2.txt
    312173 (1.32%) aligned exactly 1 <span class="pl-k">time</span>
    279609 (1.10%) aligned exactly 1 <span class="pl-k">time</span>
    226384 (0.96%) aligned exactly 1 <span class="pl-k">time</span>
    29903 (1.24%) aligned exactly 1 <span class="pl-k">time</span>
    210196 (1.09%) aligned exactly 1 <span class="pl-k">time</span>
    359056 (0.96%) aligned exactly 1 <span class="pl-k">time</span>
    225886 (0.88%) aligned exactly 1 <span class="pl-k">time</span>
    183780 (1.06%) aligned exactly 1 <span class="pl-k">time</span>
    207948 (1.27%) aligned exactly 1 <span class="pl-k">time</span>
    228135 (1.33%) aligned exactly 1 <span class="pl-k">time</span>
    250518 (1.09%) aligned exactly 1 <span class="pl-k">time</span>
    195159 (1.31%) aligned exactly 1 <span class="pl-k">time</span>
    195302 (1.13%) aligned exactly 1 <span class="pl-k">time</span>
    234155 (1.07%) aligned exactly 1 <span class="pl-k">time</span>
    176134 (0.87%) aligned exactly 1 <span class="pl-k">time</span>
    243143 (0.89%) aligned exactly 1 <span class="pl-k">time</span>

[amali010@coreV2-22-007 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> ambowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span> -f 2
1.32
1.10
0.96
1.24
1.09
0.96
0.88
1.06
1.27
1.33
1.09
1.31
1.13
1.07
0.87
0.89

<span class="pl-c"><span class="pl-c">#</span>3c - The mean number of reads "aligned exactly 1 time"</span>
[amali010@turing1 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> ambowtie2.txt
    312173 (1.32%) aligned exactly 1 <span class="pl-k">time</span>
    279609 (1.10%) aligned exactly 1 <span class="pl-k">time</span>
    226384 (0.96%) aligned exactly 1 <span class="pl-k">time</span>
    29903 (1.24%) aligned exactly 1 <span class="pl-k">time</span>
    210196 (1.09%) aligned exactly 1 <span class="pl-k">time</span>
    359056 (0.96%) aligned exactly 1 <span class="pl-k">time</span>
    225886 (0.88%) aligned exactly 1 <span class="pl-k">time</span>
    183780 (1.06%) aligned exactly 1 <span class="pl-k">time</span>
    207948 (1.27%) aligned exactly 1 <span class="pl-k">time</span>
    228135 (1.33%) aligned exactly 1 <span class="pl-k">time</span>
    250518 (1.09%) aligned exactly 1 <span class="pl-k">time</span>
    195159 (1.31%) aligned exactly 1 <span class="pl-k">time</span>
    195302 (1.13%) aligned exactly 1 <span class="pl-k">time</span>
    234155 (1.07%) aligned exactly 1 <span class="pl-k">time</span>
    176134 (0.87%) aligned exactly 1 <span class="pl-k">time</span>
    243143 (0.89%) aligned exactly 1 <span class="pl-k">time</span>
    
    [amali010@coreV2-22-007 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> ambowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span> -f 5
312173
279609
226384
29903
210196
359056
225886
183780
207948
228135
250518
195159
195302
234155
176134
243143

<span class="pl-c"><span class="pl-c">#</span>3d - The mean percent reads "aligned &gt;1 times"</span>
[amali010@coreV2-22-007 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span> ambowtie2.txt
    131960 (0.56%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    109989 (0.43%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    119408 (0.50%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    16432 (0.68%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    107800 (0.56%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    194361 (0.52%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    95504 (0.37%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    93466 (0.54%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    110827 (0.68%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    115424 (0.67%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    131960 (0.58%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    116526 (0.78%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    83427 (0.48%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    103215 (0.47%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    169138 (0.83%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
    140557 (0.52%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>

[amali010@coreV2-22-007 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span> ambowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span> -f 2
0.56
0.43
0.50
0.68
0.56
0.52
0.37
0.54
0.68
0.67
0.58
0.78
0.48
0.47
0.83
0.52
hint use grep and paste into excel

<span class="pl-c"><span class="pl-c">#</span>4 - Add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:</span>
LaneX_yourinitials	b-the mean percent <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span>	c-the mean percent reads <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span>	d-the mean number of reads <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span>	e-the mean percent reads <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span>
[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV2-22-007 QCFastqs]$ nano alignstatsam.txt
[amali010@coreV2-22-007 QCFastqs]$ cat alignstatsam.txt <span class="pl-k">&gt;&gt;</span> /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
[amali010@coreV2-22-007 QCFastqs]$ tail -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
LaneX   overallalignmentrate    percsinglyaligned       numsinglyaligned        percmultiplyaligned
LaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857
LaneX_djb       2.495714286     1.251428571     247656.7143     1.247142857
Lane1AP 5.378125        1.3725  300361.8125     4.005
lane5_sg        2.03%   1.11%   235455.75       0.92%
LaneX_jcw       2.495714286     1.251428571     247656.7143     1.247142857
Lane4_KCrid     2.48    1.30    289072.88       1.18
LaneX_am 1.670625       1.098125        222342.5625     0.573125

<span class="pl-c"><span class="pl-c">#</span>5 - Data cleanup and archiving:</span>
<span class="pl-c"><span class="pl-c">#</span>5a - mv your Trinity.fast file from your trinity_out_dir to a folder called testassembly in your data directory</span>
Note: our job was still running at the time. So to avoid confusion I decided to work with the Trinity.fasta file that was copied into my data directory during class.
[amali010@turing1 data]$ mkdir testassembly
[amali010@turing1 data]$ <span class="pl-c1">cd</span> djbtestassembly/
[amali010@turing1 djbtestassembly]$ ls
Trinity.fasta
[amali010@turing1 djbtestassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly/
[amali010@turing1 testassembly]$ ls
Trinity.fasta

<span class="pl-c"><span class="pl-c">#</span>5b - Set up a sbatch script to:</span>
	rm -r YOURtrinity_out_dir (don<span class="pl-s"><span class="pl-pds">'</span>t have one so this will be skipped in the sbatch script)</span>
<span class="pl-s">	rm -r YOURoriginalfastqs</span>
<span class="pl-s">	rm -r YOURfilterstats # as long as you<span class="pl-pds">'</span></span>ve already appended your filteringstats output to the class as part of homework_day04.txt
original script (used from homework_ day06.txt file)
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o OUTFILENAME.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n NUMTHREADS(1 if not multi-threaded)</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=YOUREMAIL@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=JOBNAME</span>

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/QCFastqs/21sampleassembly_trinity_out_dir
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/fastq/filteringstats

[amali010@turing1 originalfastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs

[amali010@turing1 filteringstats]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats

[amali010@turing1 fastq]$ nano AMCleanup.sh
[amali010@turing1 fastq]$ head AMCleanup.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMCleanup.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMCleanup</span>

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/filteringstats

[amali010@turing1 fastq]$ sbatch AMCleanup.sh
Submitted batch job 9273090
[amali010@turing1 fastq]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273090      main AMCleanu amali010  R       0:01      1 coreV2-22-007

<span class="pl-c"><span class="pl-c">#</span>6 - Submit a blast against sprot from your testassembly folder using the following command:</span>
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o OUTFILENAME.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 6         </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=EMAIL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=JOBNAME</span>

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6

[amali010@turing1 testassembly]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly
[amali010@turing1 testassembly]$ nano AMBlast.sh
[amali010@turing1 testassembly]$ cat AMBlast.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMBlast.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 6</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMBlast</span>

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
[amali010@turing1 testassembly]$ sbatch AMBlast.sh
Submitted batch job 9273094
[amali010@turing1 testassembly]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R       0:03      1 coreV2-22-028
[amali010@turing1 areej]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R      58:38      1 coreV2-22-028

<span class="pl-c"><span class="pl-c">#</span>7 - Head one of your .sam files to look at the header</span>
[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ head RI_B_06_18_clippedtrimmed.fastq.sam
@HD     VN:1.0  SO:unsorted
@SQ     SN:TR78<span class="pl-k">|</span>c0_g1_i1_coral  LN:1109
@SQ     SN:TR78<span class="pl-k">|</span>c0_g2_i1_coral  LN:1109
@SQ     SN:TR79<span class="pl-k">|</span>c0_g1_i1_coral  LN:610
@SQ     SN:TR79<span class="pl-k">|</span>c0_g2_i1_coral  LN:1549
@SQ     SN:TR87<span class="pl-k">|</span>c0_g1_i1_coral  LN:732
@SQ     SN:TR93<span class="pl-k">|</span>c0_g1_i1_coral  LN:550
@SQ     SN:TR101<span class="pl-k">|</span>c0_g1_i1_coral LN:673
@SQ     SN:TR104<span class="pl-k">|</span>c0_g1_i1_coral LN:607
@SQ     SN:TR105<span class="pl-k">|</span>c0_g1_i1_coral LN:587

<span class="pl-c"><span class="pl-c">#</span>8 - grep -v '@' your.sam | head to look at the sequence read lines, </span>
why does this work to exclude the header lines<span class="pl-k">?</span>
[amali010@turing1 QCFastqs]$ grep -v <span class="pl-s"><span class="pl-pds">"</span>@<span class="pl-pds">"</span></span> RI_B_06_18_clippedtrimmed.fastq.sam <span class="pl-k">|</span> head
K00188:59:HMTFHBBXX:7:1101:12845:1666   16      TR13647<span class="pl-k">|</span>c0_g1_i1_coral  40      255
51M     <span class="pl-k">*</span>       0       0       TTTGGATTCTTGATCACGGTATGATCGATAAAACACAGACCTGTAAAGTCC
JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XN:i:0  XM:i:0  XO:i:0      XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:17645:1683   16      TR43786<span class="pl-k">|</span>c0_g1_i16_coral 125     1  51M      <span class="pl-k">*</span>       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC
JJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:17645:1683   272     TR43786<span class="pl-k">|</span>c0_g1_i5_coral  151     255
51M     <span class="pl-k">*</span>       0       0       AAGAGTACTAAATGGTAACCAGCGAGTCCGTGTAGAAGTATTTTGTAGGTC
JJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:20750:1683   0       TR24696<span class="pl-k">|</span>c0_g1_i2_coral  329     1  51M      <span class="pl-k">*</span>       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC
AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:20750:1683   256     TR24696<span class="pl-k">|</span>c0_g1_i3_coral  329     255
51M     <span class="pl-k">*</span>       0       0       CCCCACTACCAGAAATTATACGCTCGAGTTACCCACATTTGGGGTAATCGC
AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJ     AS:i:0  XS:i:0  XN:i:0  XM:i:0      XO:i:0  XG:i:0  NM:i:0  MD:Z:51 YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:22658:1683   16      TR43764<span class="pl-k">|</span>c4_g1_i1_coral  1687    255
51M     <span class="pl-k">*</span>       0       0       TCCATTCCTAATCCGTCTGTTAGTGGTCTGCAGTTCCACGCCGTCTTTTCG
JJJJJJJFJJFJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJFJFFAAA     AS:i:-17        XN:i:0  XM:i:3      XO:i:0  XG:i:0  NM:i:3  MD:Z:25T7C14G2  YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:27854:1683   16      TR29978<span class="pl-k">|</span>c0_g2_i1_coral  10      255
51M     <span class="pl-k">*</span>       0       0       GGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAG
JJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-28        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:30G1C12G1C0A2      YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:2392:1701    0       TR35685<span class="pl-k">|</span>c1_g2_i1_coral  534     255
21M     <span class="pl-k">*</span>       0       0       TGACAACCCCGTCTGTGGCGC   AAFFFFJJJJJFFJJJJJFFJ   AS:i:-6     XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:7A13       YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:19705:1701   16      TR29978<span class="pl-k">|</span>c0_g2_i1_coral  14      255
51M     <span class="pl-k">*</span>       0       0       TCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAATTCAGCGGG
JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-30        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:26G1C12G1C0A6      YT:Z:UU RG:Z:RI_B_06_18
K00188:59:HMTFHBBXX:7:1101:23094:1701   16      TR29978<span class="pl-k">|</span>c0_g2_i1_coral  5       255
51M     <span class="pl-k">*</span>       0       0       CTGGGGGAATCCTTGTTAGTTTCTTTTCCTCCGCTTATTAATATGCTTAAA
JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFFAA     AS:i:-29        XN:i:0  XM:i:5      XO:i:0  XG:i:0  NM:i:5  MD:Z:2A0A31G1C12G0      YT:Z:UU RG:Z:RI_B_06_18

Header lines include <span class="pl-s"><span class="pl-pds">'</span>@<span class="pl-pds">'</span></span>. The -v <span class="pl-k">in</span> grep -v excludes <span class="pl-s"><span class="pl-pds">'</span>@<span class="pl-pds">'</span></span> and therefore the header lines as well.

<span class="pl-c"><span class="pl-c">#</span>9 - In an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.</span>
[amali010@turing1 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_06_18_clippedtrimmed.fastq.sam RI_B_07_18_clippedtrimmed.fastq.sam RI_W_07_18_clippedtrimmed.fastq.sam
RI_B_06_18_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>272<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>256<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
272 <span class="pl-c1">:</span>
        <span class="pl-c1">read</span> reverse strand
        not primary alignment
256 <span class="pl-c1">:</span>
        not primary alignment
16 <span class="pl-c1">:</span>
        <span class="pl-c1">read</span> reverse strand
RI_B_07_18_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>272<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>256<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
272 <span class="pl-c1">:</span>
        <span class="pl-c1">read</span> reverse strand
        not primary alignment
256 <span class="pl-c1">:</span>
        not primary alignment
16 <span class="pl-c1">:</span>
        <span class="pl-c1">read</span> reverse strand
RI_W_07_18_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>272<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>256<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
272 <span class="pl-c1">:</span>
        <span class="pl-c1">read</span> reverse strand
        not primary alignment
256 <span class="pl-c1">:</span>
        not primary alignment
16 <span class="pl-c1">:</span>
        <span class="pl-c1">read</span> reverse strand

<span class="pl-c"><span class="pl-c">#</span>10 - we need to run the read sorting step required for SNP calling, </span>
so <span class="pl-k">if</span> you have time, <span class="pl-c1">set</span> up and run the following script on your .sam files to finish before Wednesday
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o OUTFILENAME.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1         </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=EMAIL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=JOBNAME</span>

enable_lmod
module load samtools/1
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>.sam<span class="pl-k">;</span> <span class="pl-k">do</span> <span class="pl-s"><span class="pl-pds">`</span>samtools view -bS <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>.sam}</span>_UNSORTED.bam<span class="pl-pds">`</span></span><span class="pl-k">;</span> <span class="pl-k">done</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>UNSORTED.bam<span class="pl-k">;</span> <span class="pl-k">do</span> samtools sort <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
samtools index <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
<span class="pl-k">done</span>

[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ nano AMReadSort.sh
[amali010@turing1 QCFastqs]$ cat AMReadSort.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMReadSort.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMReadSort</span>

enable_lmod
module load samtools/1
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>.sam<span class="pl-k">;</span> <span class="pl-k">do</span> <span class="pl-s"><span class="pl-pds">`</span>samtools view -bS <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>.sam}</span>_UNSORTED.bam<span class="pl-pds">`</span></span><span class="pl-k">;</span> <span class="pl-k">done</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>UNSORTED.bam<span class="pl-k">;</span> <span class="pl-k">do</span> samtools sort <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
samtools index <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam

<span class="pl-k">done</span>

[amali010@turing1 QCFastqs]$ sbatch AMReadSort.sh
Submitted batch job 9273103
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R    1:21:32      1 coreV2-22-028
           9273103      main AMReadSo amali010  R       0:04      1 coreV3-23-020
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R    1:22:36      1 coreV2-22-028
           9273103      main AMReadSo amali010  R       1:08      1 coreV3-23-020
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9273094      main  AMBlast amali010  R    1:25:19      1 coreV2-22-028

<span class="pl-c"><span class="pl-c">#</span>11 - submit and add everything to your logfile</span>
git add README.md
git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
git push -u origin main</pre>
                    </div>
                    <p dir="auto">
                     ##Day 07 10-Feb-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day07
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 - Run the following command on your sprot output file to process into the contig length/match format that trinity examines
#!/bin/bash -l

#SBATCH -o OUTFILE.TXT
#SBATCH -n 1
#SBATCH --mail-user=YOUREMAIL.odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=JOBNAME

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta

[amali010@turing1 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly
[amali010@turing1 testassembly]$ cat AMBlastparse.sh
#!/bin/bash -l

#SBATCH -o AMBlastparse.TXT
#SBATCH -n 1
#SBATCH --mail-user=amali010.odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMBlastparse

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
[amali010@turing1 testassembly]$ sbatch AMBlastparse.sh
Submitted batch job 9276474
[amali010@turing1 testassembly]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276474      main AMBlastp amali010 PD       0:00      1 (Priority)
[amali010@turing1 testassembly]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276474      main AMBlastp amali010  R       0:14      1 coreV3-23-040

[amali010@turing1 testassembly]$ cat blastx.outfmt6.hist
#hit_pct_cov_bin        count_in_bin    &gt;bin_below
100     143     143
90      52      195
80      77      272
70      87      359
60      110     469
50      133     602
40      219     821
30      395     1216
20      726     1942
10      674     2616

#2 - rm UNSORTED.bam's from your QCFastqs directory (or wherever your .bams and .sams are)
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ cat AMrmunsortbam.sh
#!/bin/bash -l

#SBATCH -o AMrmunsortbam.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMrmunsortbam

rm *UNSORTED.bam
[amali010@turing1 QCFastqs]$ sbatch AMrmunsortbam.sh
Submitted batch job 9276485
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276485      main AMrmunso amali010  R       0:01      1 coreV3-23-046

#2a - Look at one of your files using sam tools
[amali010@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9276494
[amali010@coreV3-23-031 QCFastqs]$ enable_lmod
[amali010@coreV3-23-031 QCFastqs]$ module load samtools
[amali010@coreV3-23-031 QCFastqs]$ samtools tview RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
1         11        21        31        41        51        61        71
TCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT
                                                        ........................
                                                        ........................

[amali010@coreV3-23-031 QCFastqs]$ samtools tview -p 'TR78|c0_g2_i1_coral' RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
1         11        21        31        41        51        61        71
TCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT
                                                        ........................
                                                        ........................

#3 - Run the following to start genotyping your SNPs for filtering next week
[amali010@coreV3-23-031 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV3-23-031 QCFastqs]$ cat AMfreebayessubref.sh
#!/bin/bash -l

#SBATCH -o AMfreebayessubref.txt
#SBATCH -n 1
#SBATCH --mail-user=amali010.edu
#SBATCH --mail-type=END
#SBATCH --job-name=AMfreebayessubref

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.fastq.bam &gt; AMmergedfastqs.vcf
[amali010@coreV3-23-031 QCFastqs]$ ls *.fastq.bam
RI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam
RI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam
RI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam
RI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam
RI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam
RI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam
RI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam
RI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam
[amali010@coreV3-23-031 QCFastqs]$ ls *.bam
RI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam
RI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam
RI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam
RI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam
RI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam
RI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam
RI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam
RI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam
[amali010@coreV3-23-031 QCFastqs]$ sbatch AMfreebayessubref.sh
Submitted batch job 9276571
[amali010@coreV3-23-031 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276571      main AMfreeba amali010  R       0:02      1 coreV3-23-002
           9276494      main       sh amali010  R      58:02      1 coreV3-23-031


#4 - Submit and add everything to your logfile
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Run the following command on your sprot output file to process into the contig length/match format that trinity examines</span>
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o OUTFILE.TXT</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=YOUREMAIL.odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=JOBNAME</span>

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta

[amali010@turing1 testassembly]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/testassembly
[amali010@turing1 testassembly]$ cat AMBlastparse.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMBlastparse.TXT</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010.odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMBlastparse</span>

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
[amali010@turing1 testassembly]$ sbatch AMBlastparse.sh
Submitted batch job 9276474
[amali010@turing1 testassembly]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276474      main AMBlastp amali010 PD       0:00      1 (Priority)
[amali010@turing1 testassembly]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276474      main AMBlastp amali010  R       0:14      1 coreV3-23-040

[amali010@turing1 testassembly]$ cat blastx.outfmt6.hist
<span class="pl-c"><span class="pl-c">#</span>hit_pct_cov_bin        count_in_bin    &gt;bin_below</span>
100     143     143
90      52      195
80      77      272
70      87      359
60      110     469
50      133     602
40      219     821
30      395     1216
20      726     1942
10      674     2616

<span class="pl-c"><span class="pl-c">#</span>2 - rm UNSORTED.bam's from your QCFastqs directory (or wherever your .bams and .sams are)</span>
[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ cat AMrmunsortbam.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMrmunsortbam.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMrmunsortbam</span>

rm <span class="pl-k">*</span>UNSORTED.bam
[amali010@turing1 QCFastqs]$ sbatch AMrmunsortbam.sh
Submitted batch job 9276485
[amali010@turing1 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276485      main AMrmunso amali010  R       0:01      1 coreV3-23-046

<span class="pl-c"><span class="pl-c">#</span>2a - Look at one of your files using sam tools</span>
[amali010@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9276494
[amali010@coreV3-23-031 QCFastqs]$ enable_lmod
[amali010@coreV3-23-031 QCFastqs]$ module load samtools
[amali010@coreV3-23-031 QCFastqs]$ samtools tview RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
1         11        21        31        41        51        61        71
TCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT
                                                        ........................
                                                        ........................

[amali010@coreV3-23-031 QCFastqs]$ samtools tview -p <span class="pl-s"><span class="pl-pds">'</span>TR78|c0_g2_i1_coral<span class="pl-pds">'</span></span> RI_W_07_14_clippedtrimmed.fastq.bam /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
1         11        21        31        41        51        61        71
TCAGGACCAAGTCCACTCATGATCGGAAGAGAAAACTTCTTTTTGGGATCGAATGGCCGGGCTCCAGACTTAGATATTAT
                                                        ........................
                                                        ........................

<span class="pl-c"><span class="pl-c">#</span>3 - Run the following to start genotyping your SNPs for filtering next week</span>
[amali010@coreV3-23-031 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV3-23-031 QCFastqs]$ cat AMfreebayessubref.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o AMfreebayessubref.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=amali010.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=AMfreebayessubref</span>

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta <span class="pl-k">*</span>.fastq.bam <span class="pl-k">&gt;</span> AMmergedfastqs.vcf
[amali010@coreV3-23-031 QCFastqs]$ ls <span class="pl-k">*</span>.fastq.bam
RI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam
RI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam
RI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam
RI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam
RI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam
RI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam
RI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam
RI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam
[amali010@coreV3-23-031 QCFastqs]$ ls <span class="pl-k">*</span>.bam
RI_B_06_18_clippedtrimmed.fastq.bam  VA_B_06_18_clippedtrimmed.fastq.bam
RI_B_07_14_clippedtrimmed.fastq.bam  VA_B_07_14_clippedtrimmed.fastq.bam
RI_B_07_18_clippedtrimmed.fastq.bam  VA_B_07_18_clippedtrimmed.fastq.bam
RI_B_07_22_clippedtrimmed.fastq.bam  VA_B_07_22_clippedtrimmed.fastq.bam
RI_W_06_18_clippedtrimmed.fastq.bam  VA_W_06_18_clippedtrimmed.fastq.bam
RI_W_07_14_clippedtrimmed.fastq.bam  VA_W_07_14_clippedtrimmed.fastq.bam
RI_W_07_18_clippedtrimmed.fastq.bam  VA_W_07_18_clippedtrimmed.fastq.bam
RI_W_07_22_clippedtrimmed.fastq.bam  VA_W_07_22_clippedtrimmed.fastq.bam
[amali010@coreV3-23-031 QCFastqs]$ sbatch AMfreebayessubref.sh
Submitted batch job 9276571
[amali010@coreV3-23-031 QCFastqs]$ squeue -u amali010
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9276571      main AMfreeba amali010  R       0:02      1 coreV3-23-002
           9276494      main       sh amali010  R      58:02      1 coreV3-23-031


<span class="pl-c"><span class="pl-c">#</span>4 - Submit and add everything to your logfile</span>
git add README.md
git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
git push -u origin main</pre>
                    </div>
                    <p dir="auto">
                     ##Day 08 12-Feb-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day08
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 - Clean up your data directory by:
#1a - Make a SAMS folder and mv all your .sam files into that directory
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ mkdir ../../SAMS
[amali010@turing1 QCFastqs]$ ls *.sam
[amali010@turing1 QCFastqs]$ mv *.sam ../../SAMS/

#1b - Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../BAMS
[amali010@coreV3-23-003 QCFastqs]$ ls *.bam *.bam.bai
[amali010@coreV3-23-003 QCFastqs]$ mv *bam *bam.bai ../../BAMS/

#1c - rm your _unfiltered.vcf files if you have any
I did not do this because I didn't have any.

#1d - rm your .fastq files
[amali010@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ rm *.fastq

#1e - Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)
I did this even though I do not have a 'YOURNAMEmergedfastqs.vcf' file from the job that submitted in the previous class.
[amali010@coreV2-22-013 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../VCF

#2 - Start an interactive session via salloc
[amali010@coreV2-22-013 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9278251
salloc: job 9278251 queued and waiting for resources
salloc: job 9278251 has been allocated resources
salloc: Granted job allocation 9278251

#3 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder
[amali010@coreV2-22-013 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF
[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./
[amali010@coreV2-22-013 VCF]$ ls -alh
total 2.1G
drwxrwxrwx 2 amali010 users   54 Feb 12 12:59 .
drwxrwxrwx 9 amali010 users  316 Feb 12 12:56 ..
-rwxr-xr-x 1 amali010 users 1.5G Feb 12 12:59 mergedfastq_HEAAstrangiaAssembly.vcf


#4 - Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

I did it with the class vcf file since I didn't have one of my own.

[amali010@coreV2-22-013 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF
[amali010@coreV2-22-013 VCF]$ enable_lmod
[amali010@coreV2-22-013 VCF]$ module load dDocent
[amali010@coreV2-22-013 VCF]$ vcftools

VCFtools (0.1.14)
© Adam Auton and Anthony Marcketta 2009

Process Variant Call Format files

For a list of options, please go to:
        https://vcftools.github.io/man_latest.html

Alternatively, a man page is available, type:
        man vcftools

Questions, comments, and suggestions should be emailed to:
        vcftools-help@lists.sourceforge.net

[amali010@coreV2-22-013 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 14.00 seconds

#5 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files
[amali010@coreV2-22-013 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF
[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt  ./

#6 - Run our host vcf extractor on your merged vcf file using the following syntax:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
[amali010@coreV2-22-013 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
Read in ContigList

#7 - Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf

I did not have a .vcf file of my own for comparison. So I compared the other two files in the parenthesis.

[amali010@turing1 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF
[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
vcftools: Command not found.
[amali010@turing1 VCF]$ enable_lmod
[amali010@turing1 VCF]$ module load dDocent
[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 11.00 seconds

[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 432676 out of a possible 432676 Sites
Run Time = 6.00 seconds

#8 - Work through the VCF filtering tutorial until the following step:

Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.

vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm
[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
        --recode-INFO-all
        --mac 3
        --minQ 30
        --max-missing 0.5
        --out raw.g5mac3
        --recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 432676 Sites
Run Time = 33.00 seconds

[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf raw.g5mac3.recode.vcf
        --recode-INFO-all
        --minDP 3
        --out raw.g5mac3dp3
        --recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 31.00 seconds

[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf raw.g5mac3dp3.recode.vcf
        --missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 3.00 seconds

[amali010@turing1 VCF]$ cat out.imiss
INDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS
RI_W_06_merged  99853   0       67075   0.671737
RI_W_07_merged  99853   0       63917   0.640111
VA_B_03_merged  99853   0       56586   0.566693
RI_W_02_merged  99853   0       73105   0.732126
RI_W_04_merged  99853   0       71159   0.712638
VA_W_09_SNP_clipped     99853   0       19459   0.194876
RI_B_08_SNP_clipped     99853   0       88110   0.882397
VA_W_08_SNP_clipped     99853   0       83344   0.834667
VA_B_08_SNP_clipped     99853   0       94221   0.943597
VA_W_02_merged  99853   0       71050   0.711546
VA_B_07_merged  99853   0       62687   0.627793
RI_B_05_merged  99853   0       43945   0.440097
VA_W_06_merged  99853   0       65164   0.652599
VA_W_04_merged  99853   0       50911   0.509859
VA_W_01_merged  99853   0       67150   0.672489
VA_B_10_SNP_clipped     99853   0       82831   0.829529
VA_B_06_merged  99853   0       57724   0.57809
VA_W_05_merged  99853   0       66297   0.663946
RI_B_09_SNP_clipped     99853   0       85107   0.852323
VA_W_10_SNP_clipped     99853   0       83234   0.833565
RI_W_08_SNP_clipped     99853   0       77802   0.779165
RI_B_06_merged  99853   0       79766   0.798834
RI_W_10_SNP_clipped     99853   0       91323   0.914574
RI_B_04_merged  99853   0       46717   0.467858
VA_W_03_merged  99853   0       64016   0.641102
RI_B_07_merged  99853   0       70798   0.709022
RI_W_05_merged  99853   0       57117   0.572011
RI_W_09_SNP_clipped     99853   0       88492   0.886223
VA_B_01_merged  99853   0       47217   0.472865
VA_B_09_SNP_clipped     99853   0       79187   0.793036
RI_B_10_SNP_clipped     99853   0       82779   0.829009
RI_W_01_merged  99853   0       73169   0.732767
RI_B_01_merged  99853   0       69619   0.697215
VA_B_04_merged  99853   0       58190   0.582757
RI_B_02_merged  99853   0       87439   0.875677
RI_W_03_merged  99853   0       38760   0.388171
VA_B_02_merged  99853   0       78381   0.784964
VA_W_07_merged  99853   0       64592   0.646871
VA_B_05_merged  99853   0       54401   0.544811
RI_B_03_merged  99853   0       80369   0.804873

amali010@turing1 VCF]$ bash
amali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 &gt; totalmissing
amali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ gnuplot &lt;&lt; \EOF
&gt; set terminal dumb size 120, 30
&gt; set autoscale
&gt; unset label
&gt; set title &quot;Histogram of % missing data per individual&quot;
&gt; set ylabel &quot;Number of Occurrences&quot;
&gt; set xlabel &quot;% of missing data&quot;
&gt; #set yr [0:100000]
&gt; binwidth=0.01
&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0
&gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes
&gt; pause -1
&gt; EOF
♀                                         Histogram of % missing data per individual
  Number of Occurrences
      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++
        +           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
    2.5 ++                                                               * *                                        ++
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
        |                                                                * *                                         |
      2 ++                                                       **      * * ***  ******    ** ****   ****          ++
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        |                                                        **      * * * *  * *  *    ** ** *   *  *           |
        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +
      1 *********************************************************************************************************---++
       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1
                                                      % of missing data
#9 - Submit and add everything to your logfile
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Clean up your data directory by:</span>
<span class="pl-c"><span class="pl-c">#</span>1a - Make a SAMS folder and mv all your .sam files into that directory</span>
[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ mkdir ../../SAMS
[amali010@turing1 QCFastqs]$ ls <span class="pl-k">*</span>.sam
[amali010@turing1 QCFastqs]$ mv <span class="pl-k">*</span>.sam ../../SAMS/

<span class="pl-c"><span class="pl-c">#</span>1b - Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory</span>
[amali010@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../BAMS
[amali010@coreV3-23-003 QCFastqs]$ ls <span class="pl-k">*</span>.bam <span class="pl-k">*</span>.bam.bai
[amali010@coreV3-23-003 QCFastqs]$ mv <span class="pl-k">*</span>bam <span class="pl-k">*</span>bam.bai ../../BAMS/

<span class="pl-c"><span class="pl-c">#</span>1c - rm your _unfiltered.vcf files if you have any</span>
I did not <span class="pl-k">do</span> this because I didn<span class="pl-s"><span class="pl-pds">'</span>t have any.</span>
<span class="pl-s"></span>
<span class="pl-s">#1d - rm your .fastq files</span>
<span class="pl-s">[amali010@turing1 QCFastqs]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs</span>
<span class="pl-s">[amali010@turing1 QCFastqs]$ rm *.fastq</span>
<span class="pl-s"></span>
<span class="pl-s">#1e - Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn<span class="pl-pds">'</span></span>t <span class="pl-c1">complete</span> <span class="pl-k">then</span> skip this step)
I did this even though I <span class="pl-k">do</span> not have a <span class="pl-s"><span class="pl-pds">'</span>YOURNAMEmergedfastqs.vcf<span class="pl-pds">'</span></span> file from the job that submitted <span class="pl-k">in</span> the previous class.
[amali010@coreV2-22-013 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@coreV3-23-003 QCFastqs]$ mkdir ../../VCF

<span class="pl-c"><span class="pl-c">#</span>2 - Start an interactive session via salloc</span>
[amali010@coreV2-22-013 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/fastq/QCFastqs
[amali010@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9278251
salloc: job 9278251 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9278251 has been allocated resources
salloc: Granted job allocation 9278251

<span class="pl-c"><span class="pl-c">#</span>3 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder</span>
[amali010@coreV2-22-013 VCF]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF
[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./
[amali010@coreV2-22-013 VCF]$ ls -alh
total 2.1G
drwxrwxrwx 2 amali010 users   54 Feb 12 12:59 <span class="pl-c1">.</span>
drwxrwxrwx 9 amali010 users  316 Feb 12 12:56 ..
-rwxr-xr-x 1 amali010 users 1.5G Feb 12 12:59 mergedfastq_HEAAstrangiaAssembly.vcf


<span class="pl-c"><span class="pl-c">#</span>4 - Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:</span>
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

I did it with the class vcf file since I didn<span class="pl-s"><span class="pl-pds">'</span>t have one of my own.</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF</span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ enable_lmod</span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ module load dDocent</span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ vcftools</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools (0.1.14)</span>
<span class="pl-s">© Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Process Variant Call Format files</span>
<span class="pl-s"></span>
<span class="pl-s">For a list of options, please go to:</span>
<span class="pl-s">        https://vcftools.github.io/man_latest.html</span>
<span class="pl-s"></span>
<span class="pl-s">Alternatively, a man page is available, type:</span>
<span class="pl-s">        man vcftools</span>
<span class="pl-s"></span>
<span class="pl-s">Questions, comments, and suggestions should be emailed to:</span>
<span class="pl-s">        vcftools-help@lists.sourceforge.net</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">        --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">After filtering, kept 1214003 out of a possible 1214003 Sites</span>
<span class="pl-s">Run Time = 14.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">#5 - cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files</span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF</span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt  ./</span>
<span class="pl-s"></span>
<span class="pl-s">#6 - Run our host vcf extractor on your merged vcf file using the following syntax:</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">[amali010@coreV2-22-013 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">Read in ContigList</span>
<span class="pl-s"></span>
<span class="pl-s">#7 - Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:</span>
<span class="pl-s">/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf</span>
<span class="pl-s"></span>
<span class="pl-s">I did not have a .vcf file of my own for comparison. So I compared the other two files in the parenthesis.</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@turing1 VCF]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF</span>
<span class="pl-s">[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">vcftools: Command not found.</span>
<span class="pl-s">[amali010@turing1 VCF]$ enable_lmod</span>
<span class="pl-s">[amali010@turing1 VCF]$ module load dDocent</span>
<span class="pl-s">[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">        --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">After filtering, kept 1214003 out of a possible 1214003 Sites</span>
<span class="pl-s">Run Time = 11.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">After filtering, kept 432676 out of a possible 432676 Sites</span>
<span class="pl-s">Run Time = 6.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">#8 - Work through the VCF filtering tutorial until the following step:</span>
<span class="pl-s"></span>
<span class="pl-s">Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.</span>
<span class="pl-s"></span>
<span class="pl-s">vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm</span>
<span class="pl-s">[amali010@turing1 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf</span>
<span class="pl-s">        --recode-INFO-all</span>
<span class="pl-s">        --mac 3</span>
<span class="pl-s">        --minQ 30</span>
<span class="pl-s">        --max-missing 0.5</span>
<span class="pl-s">        --out raw.g5mac3</span>
<span class="pl-s">        --recode</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">Outputting VCF file...</span>
<span class="pl-s">After filtering, kept 99853 out of a possible 432676 Sites</span>
<span class="pl-s">Run Time = 33.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">        --vcf raw.g5mac3.recode.vcf</span>
<span class="pl-s">        --recode-INFO-all</span>
<span class="pl-s">        --minDP 3</span>
<span class="pl-s">        --out raw.g5mac3dp3</span>
<span class="pl-s">        --recode</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">Outputting VCF file...</span>
<span class="pl-s">After filtering, kept 99853 out of a possible 99853 Sites</span>
<span class="pl-s">Run Time = 31.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@turing1 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">        --vcf raw.g5mac3dp3.recode.vcf</span>
<span class="pl-s">        --missing-indv</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">Outputting Individual Missingness</span>
<span class="pl-s">After filtering, kept 99853 out of a possible 99853 Sites</span>
<span class="pl-s">Run Time = 3.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">[amali010@turing1 VCF]$ cat out.imiss</span>
<span class="pl-s">INDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS</span>
<span class="pl-s">RI_W_06_merged  99853   0       67075   0.671737</span>
<span class="pl-s">RI_W_07_merged  99853   0       63917   0.640111</span>
<span class="pl-s">VA_B_03_merged  99853   0       56586   0.566693</span>
<span class="pl-s">RI_W_02_merged  99853   0       73105   0.732126</span>
<span class="pl-s">RI_W_04_merged  99853   0       71159   0.712638</span>
<span class="pl-s">VA_W_09_SNP_clipped     99853   0       19459   0.194876</span>
<span class="pl-s">RI_B_08_SNP_clipped     99853   0       88110   0.882397</span>
<span class="pl-s">VA_W_08_SNP_clipped     99853   0       83344   0.834667</span>
<span class="pl-s">VA_B_08_SNP_clipped     99853   0       94221   0.943597</span>
<span class="pl-s">VA_W_02_merged  99853   0       71050   0.711546</span>
<span class="pl-s">VA_B_07_merged  99853   0       62687   0.627793</span>
<span class="pl-s">RI_B_05_merged  99853   0       43945   0.440097</span>
<span class="pl-s">VA_W_06_merged  99853   0       65164   0.652599</span>
<span class="pl-s">VA_W_04_merged  99853   0       50911   0.509859</span>
<span class="pl-s">VA_W_01_merged  99853   0       67150   0.672489</span>
<span class="pl-s">VA_B_10_SNP_clipped     99853   0       82831   0.829529</span>
<span class="pl-s">VA_B_06_merged  99853   0       57724   0.57809</span>
<span class="pl-s">VA_W_05_merged  99853   0       66297   0.663946</span>
<span class="pl-s">RI_B_09_SNP_clipped     99853   0       85107   0.852323</span>
<span class="pl-s">VA_W_10_SNP_clipped     99853   0       83234   0.833565</span>
<span class="pl-s">RI_W_08_SNP_clipped     99853   0       77802   0.779165</span>
<span class="pl-s">RI_B_06_merged  99853   0       79766   0.798834</span>
<span class="pl-s">RI_W_10_SNP_clipped     99853   0       91323   0.914574</span>
<span class="pl-s">RI_B_04_merged  99853   0       46717   0.467858</span>
<span class="pl-s">VA_W_03_merged  99853   0       64016   0.641102</span>
<span class="pl-s">RI_B_07_merged  99853   0       70798   0.709022</span>
<span class="pl-s">RI_W_05_merged  99853   0       57117   0.572011</span>
<span class="pl-s">RI_W_09_SNP_clipped     99853   0       88492   0.886223</span>
<span class="pl-s">VA_B_01_merged  99853   0       47217   0.472865</span>
<span class="pl-s">VA_B_09_SNP_clipped     99853   0       79187   0.793036</span>
<span class="pl-s">RI_B_10_SNP_clipped     99853   0       82779   0.829009</span>
<span class="pl-s">RI_W_01_merged  99853   0       73169   0.732767</span>
<span class="pl-s">RI_B_01_merged  99853   0       69619   0.697215</span>
<span class="pl-s">VA_B_04_merged  99853   0       58190   0.582757</span>
<span class="pl-s">RI_B_02_merged  99853   0       87439   0.875677</span>
<span class="pl-s">RI_W_03_merged  99853   0       38760   0.388171</span>
<span class="pl-s">VA_B_02_merged  99853   0       78381   0.784964</span>
<span class="pl-s">VA_W_07_merged  99853   0       64592   0.646871</span>
<span class="pl-s">VA_B_05_merged  99853   0       54401   0.544811</span>
<span class="pl-s">RI_B_03_merged  99853   0       80369   0.804873</span>
<span class="pl-s"></span>
<span class="pl-s">amali010@turing1 VCF]$ bash</span>
<span class="pl-s">amali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ mawk <span class="pl-pds">'</span></span><span class="pl-k">!</span>/IN/<span class="pl-s"><span class="pl-pds">'</span> out.imiss | cut -f5 &gt; totalmissing</span>
<span class="pl-s">amali010@turing1:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF$ gnuplot &lt;&lt; \EOF</span>
<span class="pl-s">&gt; set terminal dumb size 120, 30</span>
<span class="pl-s">&gt; set autoscale</span>
<span class="pl-s">&gt; unset label</span>
<span class="pl-s">&gt; set title "Histogram of % missing data per individual"</span>
<span class="pl-s">&gt; set ylabel "Number of Occurrences"</span>
<span class="pl-s">&gt; set xlabel "% of missing data"</span>
<span class="pl-s">&gt; #set yr [0:100000]</span>
<span class="pl-s">&gt; binwidth=0.01</span>
<span class="pl-s">&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0</span>
<span class="pl-s">&gt; plot <span class="pl-pds">'</span></span>totalmissing<span class="pl-s"><span class="pl-pds">'</span> using (bin($1,binwidth)):(1.0) smooth freq with boxes</span>
<span class="pl-s">&gt; pause -1</span>
<span class="pl-s">&gt; EOF</span>
<span class="pl-s">♀                                         Histogram of % missing data per individual</span>
<span class="pl-s">  Number of Occurrences</span>
<span class="pl-s">      3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++</span>
<span class="pl-s">        +           +           +           +           +       <span class="pl-pds">'</span></span>totalmissing<span class="pl-s"><span class="pl-pds">'</span> using (bin($1,binwidth)):(1.0) ****** +</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">    2.5 ++                                                               * *                                        ++</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">        |                                                                * *                                         |</span>
<span class="pl-s">      2 ++                                                       **      * * ***  ******    ** ****   ****          ++</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">    1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        |                                                        **      * * * *  * *  *    ** ** *   *  *           |</span>
<span class="pl-s">        +           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *           +</span>
<span class="pl-s">      1 *********************************************************************************************************---++</span>
<span class="pl-s">       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1</span>
<span class="pl-s">                                                      % of missing data</span>
<span class="pl-s">#9 - Submit and add everything to your logfile</span>
<span class="pl-s">git add README.md</span>
<span class="pl-s">git commit -m <span class="pl-pds">'</span></span>updating readme<span class="pl-s"><span class="pl-pds">'</span></span>
<span class="pl-s">git push -u origin main</span></pre>
                    </div>
                    <p dir="auto">
                     ##Day 09 17-Feb-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day09
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 -  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters
##Half the class run the following
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
##the other half run the filters we used from the paper
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters

#1a - Started a salloc session.
[amali010@turing1 areej]$ salloc
salloc: Pending job allocation 9280291
salloc: job 9280291 queued and waiting for resources
salloc: job 9280291 has been allocated resources
salloc: Granted job allocation 9280291

#1b - Navigated to my VCF directory.
[amali010@coreV2-25-017 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF

#1c - Ran the following to make sure that the .vcf file is correct: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 15.00 seconds

#1d - Chosen to ran the second filter. Ran the following filter: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
        --recode-INFO-all
        --mac 3
        --maf 0.015
        --max-alleles 2
        --minDP 10
        --minQ 30
        --max-missing 0.5
        --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
        --recode
        --remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 1578 out of a possible 432676 Sites
Run Time = 8.00 seconds

#2 - Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs
#2a - Looked at missing individuals
[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf --missing-indv

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf
        --missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 1578 out of a possible 1578 Sites
Run Time = 1.00 seconds

[amali010@coreV2-25-017 VCF]$ less mergedfastq_HEAAstrangiaAssembly_subset.vcf
[amali010@coreV2-25-017 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged
[amali010@coreV2-25-017 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf &gt; samplenames.text
[amali010@coreV2-25-017 VCF]$ cat samplenames.text
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged
[amali010@coreV2-25-017 VCF]$ cat out.imiss
INDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS
RI_W_06_merged  1578    0       316     0.200253
RI_W_07_merged  1578    0       183     0.11597
VA_B_03_merged  1578    0       79      0.0500634
RI_W_02_merged  1578    0       442     0.280101
RI_W_04_merged  1578    0       581     0.368188
VA_W_09_SNP_clipped     1578    0       17      0.0107731
RI_B_08_SNP_clipped     1578    0       1180    0.747782
VA_W_08_SNP_clipped     1578    0       892     0.565272
VA_B_08_SNP_clipped     1578    0       1273    0.806717
VA_W_02_merged  1578    0       293     0.185678
VA_B_07_merged  1578    0       116     0.0735108
RI_B_05_merged  1578    0       57      0.0361217
VA_W_06_merged  1578    0       164     0.103929
VA_W_04_merged  1578    0       91      0.0576679
VA_W_01_merged  1578    0       337     0.213561
VA_B_10_SNP_clipped     1578    0       683     0.432826
VA_B_06_merged  1578    0       82      0.0519645
VA_W_05_merged  1578    0       187     0.118504
RI_B_09_SNP_clipped     1578    0       1081    0.685044
VA_W_10_SNP_clipped     1578    0       619     0.392269
RI_W_08_SNP_clipped     1578    0       558     0.353612
RI_B_06_merged  1578    0       613     0.388466
RI_W_10_SNP_clipped     1578    0       1219    0.772497
RI_B_04_merged  1578    0       107     0.0678074
VA_W_03_merged  1578    0       197     0.124842
RI_B_07_merged  1578    0       319     0.202155
RI_W_05_merged  1578    0       79      0.0500634
RI_W_09_SNP_clipped     1578    0       1159    0.734474
VA_B_01_merged  1578    0       198     0.125475
VA_B_09_SNP_clipped     1578    0       510     0.323194
RI_B_10_SNP_clipped     1578    0       800     0.506971
RI_W_01_merged  1578    0       499     0.316223
RI_B_01_merged  1578    0       231     0.146388
VA_B_04_merged  1578    0       139     0.0880862
RI_B_02_merged  1578    0       1031    0.653359
RI_W_03_merged  1578    0       109     0.0690748
VA_B_02_merged  1578    0       597     0.378327
VA_W_07_merged  1578    0       129     0.081749
VA_B_05_merged  1578    0       54      0.0342205
RI_B_03_merged  1578    0       762     0.48289
[amali010@coreV2-25-017 VCF]$ cut -f 1 out.imiss | tail -n +2
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged

Used excel to make a second column with just the header

RI_W_06_merged	RI_W
RI_W_07_merged	RI_W
VA_B_03_merged	VA_B
RI_W_02_merged	RI_W
RI_W_04_merged	RI_W
VA_W_09_SNP_clipped	VA_W
RI_B_08_SNP_clipped	RI_B
VA_W_08_SNP_clipped	VA_W
VA_B_08_SNP_clipped	VA_B
VA_W_02_merged	VA_W
VA_B_07_merged	VA_B
RI_B_05_merged	RI_B
VA_W_06_merged	VA_W
VA_W_04_merged	VA_W
VA_W_01_merged	VA_W
VA_B_10_SNP_clipped	VA_B
VA_B_06_merged	VA_B
VA_W_05_merged	VA_W
RI_B_09_SNP_clipped	RI_B
VA_W_10_SNP_clipped	VA_W
RI_W_08_SNP_clipped	RI_W
RI_B_06_merged	RI_B
RI_W_10_SNP_clipped	RI_W
RI_B_04_merged	RI_B
VA_W_03_merged	VA_W
RI_B_07_merged	RI_B
RI_W_05_merged	RI_W
RI_W_09_SNP_clipped	RI_W
VA_B_01_merged	VA_B
VA_B_09_SNP_clipped	VA_B
RI_B_10_SNP_clipped	RI_B
RI_W_01_merged	RI_W
RI_B_01_merged	RI_B
VA_B_04_merged	VA_B
RI_B_02_merged	RI_B
RI_W_03_merged	RI_W
VA_B_02_merged	VA_B
VA_W_07_merged	VA_W
VA_B_05_merged	VA_B
RI_B_03_merged	RI_B

[amali010@coreV2-25-017 VCF]$ nano popfile2.txt
Pasted the two columns in this .txt file

#3 - Convert your filtered .vcf file to genepop format using the following command:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt

[amali010@coreV2-25-017 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile2.txt
Indivs with genotypes in vcf file: RI_W_06_merged       RI_W_07_merged  VA_B_03_merged      RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped     RI_B_08_SNP_clipped
VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged  VA_B_07_merged  RI_B_05_merged      VA_W_06_merged  VA_W_04_merged  VA_W_01_merged  VA_B_10_SNP_clipped
VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped     VA_W_10_SNP_clipped     RI_W_08_SNP_clipped RI_B_06_merged  RI_W_10_SNP_clipped     RI_B_04_merged  VA_W_03_merged      RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped     VA_B_01_merged  VA_B_09_SNP_clipped RI_B_10_SNP_clipped     RI_W_01_merged  RI_B_01_merged  VA_B_04_merged      RI_B_02_merged  RI_W_03_merged  VA_B_02_merged  VA_W_07_merged  VA_B_05_merged      RI_B_03_merged
44 1578 1578 1578 1578 40

[amali010@coreV2-25-017 VCF]$ ls
Now I have a .gen file
[amali010@coreV2-25-017 VCF]$ less 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen
Noticed that the contig name has now merged with SNPs

#4 - SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop
[amali010@turing1 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF

areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)
$ pwd
/c/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09

areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)
$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen .
amali010@turing.hpc.odu.edu's password:
1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFi 100%  343KB   2.1MB/s   00:00

areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)
$ ls
1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen
adegenet_PCAs.R
adegenet_PCAs_HEAOutliers.R
homework_day09.txt

#5 - Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.
Input:
library(&quot;ape&quot;)
library(&quot;pegas&quot;)
library(&quot;seqinr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;adegenet&quot;)
setwd(&quot;/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09&quot;)
datafile&lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)
sum(is.na(datafile$tab))
datafile #shows info

Output:
&gt; library(&quot;ape&quot;)
&gt; library(&quot;pegas&quot;)
&gt; library(&quot;seqinr&quot;)
&gt; library(&quot;ggplot2&quot;)
&gt; library(&quot;adegenet&quot;)
&gt; setwd(&quot;/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09&quot;)
&gt; datafile&lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)

 Converting data from a Genepop .gen file to a genind object... 


File description:  AllSNPs 

...done.

&gt; sum(is.na(datafile$tab))
[1] 35966
&gt; datafile #shows info
/// GENIND OBJECT /////////

 // 40 individuals; 1,578 loci; 3,156 alleles; size: 1.4 Mb

 // Basic content
   @tab:  40 x 3156 matrix of allele counts
   @loc.n.all: number of alleles per locus (range: 2-2)
   @loc.fac: locus factor for the 3156 columns of @tab
   @all.names: list of allele names for each locus
   @ploidy: ploidy of each individual  (range: 2-2)
   @type:  codom
   @call: read.genepop(file = &quot;1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen&quot;, 
    ncode = 2)

 // Optional content
   @pop: population of each individual (group size range: 10-10)

Input:
YOURdata&lt;-scaleGen(datafile, NA.method='mean')
X&lt;-YOURdata
substring(pop(datafile),1,4)
Y&lt;-as.factor(substring(pop(datafile),1,4))
pca1 &lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)

Output:
&gt; YOURdata&lt;-scaleGen(datafile, NA.method='mean')
Warning message:
In .local(x, ...) : Some scaling values are null.
 Corresponding alleles are removed.
&gt; X&lt;-YOURdata
&gt; substring(pop(datafile),1,4)
 [1] &quot;RI_B&quot; &quot;RI_B&quot; &quot;RI_B&quot; &quot;RI_B&quot; &quot;RI_B&quot; &quot;RI_B&quot; &quot;RI_B&quot; &quot;RI_B&quot; &quot;RI_B&quot;
[10] &quot;RI_B&quot; &quot;VA_W&quot; &quot;VA_W&quot; &quot;VA_W&quot; &quot;VA_W&quot; &quot;VA_W&quot; &quot;VA_W&quot; &quot;VA_W&quot; &quot;VA_W&quot;
[19] &quot;VA_W&quot; &quot;VA_W&quot; &quot;VA_B&quot; &quot;VA_B&quot; &quot;VA_B&quot; &quot;VA_B&quot; &quot;VA_B&quot; &quot;VA_B&quot; &quot;VA_B&quot;
[28] &quot;VA_B&quot; &quot;VA_B&quot; &quot;VA_B&quot; &quot;RI_W&quot; &quot;RI_W&quot; &quot;RI_W&quot; &quot;RI_W&quot; &quot;RI_W&quot; &quot;RI_W&quot;
[37] &quot;RI_W&quot; &quot;RI_W&quot; &quot;RI_W&quot; &quot;RI_W&quot;
&gt; Y&lt;-as.factor(substring(pop(datafile),1,4))
&gt; pca1 &lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)

Input:
#### PCAs ####
# individual labels
s.label(pca1$li)

Output:
#### PCAs ####
&gt; # individual labels
&gt; s.label(pca1$li)
Plotted each sample name as a point

Input:
# population elipses
s.class(pca1$li, pop(datafile))

Output:
&gt; # population elipses
&gt; s.class(pca1$li, pop(datafile))
Plotted population

Input:
#color symbols, pop names
#pdf(&quot;YOURINITIALS_ColorPCA1v2.pdf&quot;)
col &lt;- c(&quot;blue&quot;,&quot;red&quot;, &quot;green&quot;, &quot;black&quot;)
s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca1$eig[1:3], 3,1,2, posi=&quot;topright&quot;)
title(&quot;PCA of DJB_data\naxes 1-2&quot;)

Output: 
&gt; #color symbols, pop names
&gt; #pdf(&quot;YOURINITIALS_ColorPCA1v2.pdf&quot;)
&gt; col &lt;- c(&quot;blue&quot;,&quot;red&quot;, &quot;green&quot;, &quot;black&quot;)
&gt; s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
&gt; add.scatter.eig(pca1$eig[1:3], 3,1,2, posi=&quot;topright&quot;)
&gt; title(&quot;PCA of DJB_data\naxes 1-2&quot;)
Plotted population with color
Points are individuals with elipses around the populations
Not as many outliers as the other .gen file

Input:
#### Snapclust ####
a.clust&lt;-snapclust(datafile, k = 2)
class(a.clust)
names(a.clust)
a.tab &lt;- table(pop(datafile), a.clust$group)
table.value(a.tab, col.labels = 1:2)

Output:
&gt; #### Snapclust ####
&gt; a.clust&lt;-snapclust(datafile, k = 2)
&gt; class(a.clust)
[1] &quot;snapclust&quot; &quot;list&quot;     
&gt; names(a.clust)
[1] &quot;group&quot;     &quot;ll&quot;        &quot;proba&quot;     &quot;converged&quot; &quot;n.iter&quot;   
[6] &quot;n.param&quot;  
&gt; a.tab &lt;- table(pop(datafile), a.clust$group)
&gt; table.value(a.tab, col.labels = 1:2)
Gave us a table that shows out of your total populations which are the individuals from this population assigning too. 
Most individuals are being assigned to blue populations with a few being assigned to brown population.

Input:
compoplot(a.clust)

Output:
&gt; compoplot(a.clust)
Gave us cluster diagram that showed the same thing where the majority are being assigned to the blue population with the exception of a few that are assigned to brown population.

#6 - Submit and add everything to your logfile
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 -  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters</span>
<span class="pl-c"><span class="pl-c">#</span>#Half the class run the following</span>
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
<span class="pl-c"><span class="pl-c">#</span>#the other half run the filters we used from the paper</span>
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters

<span class="pl-c"><span class="pl-c">#</span>1a - Started a salloc session.</span>
[amali010@turing1 areej]$ salloc
salloc: Pending job allocation 9280291
salloc: job 9280291 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9280291 has been allocated resources
salloc: Granted job allocation 9280291

<span class="pl-c"><span class="pl-c">#</span>1b - Navigated to my VCF directory.</span>
[amali010@coreV2-25-017 VCF]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF

<span class="pl-c"><span class="pl-c">#</span>1c - Ran the following to make sure that the .vcf file is correct: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 15.00 seconds

<span class="pl-c"><span class="pl-c">#</span>1d - Chosen to ran the second filter. Ran the following filter: /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters</span>
[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
        --recode-INFO-all
        --mac 3
        --maf 0.015
        --max-alleles 2
        --minDP 10
        --minQ 30
        --max-missing 0.5
        --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
        --recode
        --remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 1578 out of a possible 432676 Sites
Run Time = 8.00 seconds

<span class="pl-c"><span class="pl-c">#</span>2 - Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs</span>
<span class="pl-c"><span class="pl-c">#</span>2a - Looked at missing individuals</span>
[amali010@coreV2-25-017 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf --missing-indv

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
        --vcf 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf
        --missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 1578 out of a possible 1578 Sites
Run Time = 1.00 seconds

[amali010@coreV2-25-017 VCF]$ less mergedfastq_HEAAstrangiaAssembly_subset.vcf
[amali010@coreV2-25-017 VCF]$ grep <span class="pl-s"><span class="pl-pds">'</span>#CHROM<span class="pl-pds">'</span></span> mergedfastq_HEAAstrangiaAssembly_subset.vcf
<span class="pl-c"><span class="pl-c">#</span>CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged</span>
[amali010@coreV2-25-017 VCF]$ grep <span class="pl-s"><span class="pl-pds">'</span>#CHROM<span class="pl-pds">'</span></span> mergedfastq_HEAAstrangiaAssembly_subset.vcf <span class="pl-k">&gt;</span> samplenames.text
[amali010@coreV2-25-017 VCF]$ cat samplenames.text
<span class="pl-c"><span class="pl-c">#</span>CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  RI_W_06_merged      RI_W_07_merged  VA_B_03_merged  RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped RI_B_08_SNP_clipped     VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged      VA_B_07_merged  RI_B_05_merged  VA_W_06_merged  VA_W_04_merged  VA_W_01_merged      VA_B_10_SNP_clipped     VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped VA_W_10_SNP_clipped     RI_W_08_SNP_clipped     RI_B_06_merged  RI_W_10_SNP_clipped RI_B_04_merged  VA_W_03_merged  RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped VA_B_01_merged  VA_B_09_SNP_clipped     RI_B_10_SNP_clipped     RI_W_01_merged      RI_B_01_merged  VA_B_04_merged  RI_B_02_merged  RI_W_03_merged  VA_B_02_merged      VA_W_07_merged  VA_B_05_merged  RI_B_03_merged</span>
[amali010@coreV2-25-017 VCF]$ cat out.imiss
INDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS
RI_W_06_merged  1578    0       316     0.200253
RI_W_07_merged  1578    0       183     0.11597
VA_B_03_merged  1578    0       79      0.0500634
RI_W_02_merged  1578    0       442     0.280101
RI_W_04_merged  1578    0       581     0.368188
VA_W_09_SNP_clipped     1578    0       17      0.0107731
RI_B_08_SNP_clipped     1578    0       1180    0.747782
VA_W_08_SNP_clipped     1578    0       892     0.565272
VA_B_08_SNP_clipped     1578    0       1273    0.806717
VA_W_02_merged  1578    0       293     0.185678
VA_B_07_merged  1578    0       116     0.0735108
RI_B_05_merged  1578    0       57      0.0361217
VA_W_06_merged  1578    0       164     0.103929
VA_W_04_merged  1578    0       91      0.0576679
VA_W_01_merged  1578    0       337     0.213561
VA_B_10_SNP_clipped     1578    0       683     0.432826
VA_B_06_merged  1578    0       82      0.0519645
VA_W_05_merged  1578    0       187     0.118504
RI_B_09_SNP_clipped     1578    0       1081    0.685044
VA_W_10_SNP_clipped     1578    0       619     0.392269
RI_W_08_SNP_clipped     1578    0       558     0.353612
RI_B_06_merged  1578    0       613     0.388466
RI_W_10_SNP_clipped     1578    0       1219    0.772497
RI_B_04_merged  1578    0       107     0.0678074
VA_W_03_merged  1578    0       197     0.124842
RI_B_07_merged  1578    0       319     0.202155
RI_W_05_merged  1578    0       79      0.0500634
RI_W_09_SNP_clipped     1578    0       1159    0.734474
VA_B_01_merged  1578    0       198     0.125475
VA_B_09_SNP_clipped     1578    0       510     0.323194
RI_B_10_SNP_clipped     1578    0       800     0.506971
RI_W_01_merged  1578    0       499     0.316223
RI_B_01_merged  1578    0       231     0.146388
VA_B_04_merged  1578    0       139     0.0880862
RI_B_02_merged  1578    0       1031    0.653359
RI_W_03_merged  1578    0       109     0.0690748
VA_B_02_merged  1578    0       597     0.378327
VA_W_07_merged  1578    0       129     0.081749
VA_B_05_merged  1578    0       54      0.0342205
RI_B_03_merged  1578    0       762     0.48289
[amali010@coreV2-25-017 VCF]$ cut -f 1 out.imiss <span class="pl-k">|</span> tail -n +2
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged

Used excel to make a second column with just the header

RI_W_06_merged	RI_W
RI_W_07_merged	RI_W
VA_B_03_merged	VA_B
RI_W_02_merged	RI_W
RI_W_04_merged	RI_W
VA_W_09_SNP_clipped	VA_W
RI_B_08_SNP_clipped	RI_B
VA_W_08_SNP_clipped	VA_W
VA_B_08_SNP_clipped	VA_B
VA_W_02_merged	VA_W
VA_B_07_merged	VA_B
RI_B_05_merged	RI_B
VA_W_06_merged	VA_W
VA_W_04_merged	VA_W
VA_W_01_merged	VA_W
VA_B_10_SNP_clipped	VA_B
VA_B_06_merged	VA_B
VA_W_05_merged	VA_W
RI_B_09_SNP_clipped	RI_B
VA_W_10_SNP_clipped	VA_W
RI_W_08_SNP_clipped	RI_W
RI_B_06_merged	RI_B
RI_W_10_SNP_clipped	RI_W
RI_B_04_merged	RI_B
VA_W_03_merged	VA_W
RI_B_07_merged	RI_B
RI_W_05_merged	RI_W
RI_W_09_SNP_clipped	RI_W
VA_B_01_merged	VA_B
VA_B_09_SNP_clipped	VA_B
RI_B_10_SNP_clipped	RI_B
RI_W_01_merged	RI_W
RI_B_01_merged	RI_B
VA_B_04_merged	VA_B
RI_B_02_merged	RI_B
RI_W_03_merged	RI_W
VA_B_02_merged	VA_B
VA_W_07_merged	VA_W
VA_B_05_merged	VA_B
RI_B_03_merged	RI_B

[amali010@coreV2-25-017 VCF]$ nano popfile2.txt
Pasted the two columns <span class="pl-k">in</span> this .txt file

<span class="pl-c"><span class="pl-c">#</span>3 - Convert your filtered .vcf file to genepop format using the following command:</span>
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt

[amali010@coreV2-25-017 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile2.txt
Indivs with genotypes <span class="pl-k">in</span> vcf file: RI_W_06_merged       RI_W_07_merged  VA_B_03_merged      RI_W_02_merged  RI_W_04_merged  VA_W_09_SNP_clipped     RI_B_08_SNP_clipped
VA_W_08_SNP_clipped     VA_B_08_SNP_clipped     VA_W_02_merged  VA_B_07_merged  RI_B_05_merged      VA_W_06_merged  VA_W_04_merged  VA_W_01_merged  VA_B_10_SNP_clipped
VA_B_06_merged  VA_W_05_merged  RI_B_09_SNP_clipped     VA_W_10_SNP_clipped     RI_W_08_SNP_clipped RI_B_06_merged  RI_W_10_SNP_clipped     RI_B_04_merged  VA_W_03_merged      RI_B_07_merged  RI_W_05_merged  RI_W_09_SNP_clipped     VA_B_01_merged  VA_B_09_SNP_clipped RI_B_10_SNP_clipped     RI_W_01_merged  RI_B_01_merged  VA_B_04_merged      RI_B_02_merged  RI_W_03_merged  VA_B_02_merged  VA_W_07_merged  VA_B_05_merged      RI_B_03_merged
44 1578 1578 1578 1578 40

[amali010@coreV2-25-017 VCF]$ ls
Now I have a .gen file
[amali010@coreV2-25-017 VCF]$ less 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen
Noticed that the contig name has now merged with SNPs

<span class="pl-c"><span class="pl-c">#</span>4 - SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop</span>
[amali010@turing1 VCF]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF

areej@DESKTOP-RF6VF4N MINGW64 <span class="pl-k">~</span>/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)
$ <span class="pl-c1">pwd</span>
/c/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09

areej@DESKTOP-RF6VF4N MINGW64 <span class="pl-k">~</span>/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)
$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/VCF/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen <span class="pl-c1">.</span>
amali010@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFi 100%  343KB   2.1MB/s   00:00</span>
<span class="pl-s"></span>
<span class="pl-s">areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09 (master)</span>
<span class="pl-s">$ ls</span>
<span class="pl-s">1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen</span>
<span class="pl-s">adegenet_PCAs.R</span>
<span class="pl-s">adegenet_PCAs_HEAOutliers.R</span>
<span class="pl-s">homework_day09.txt</span>
<span class="pl-s"></span>
<span class="pl-s">#5 - Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.</span>
<span class="pl-s">Input:</span>
<span class="pl-s">library("ape")</span>
<span class="pl-s">library("pegas")</span>
<span class="pl-s">library("seqinr")</span>
<span class="pl-s">library("ggplot2")</span>
<span class="pl-s">library("adegenet")</span>
<span class="pl-s">setwd("/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09")</span>
<span class="pl-s">datafile&lt;-read.genepop(<span class="pl-pds">'</span></span>1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen<span class="pl-s"><span class="pl-pds">'</span>, ncode=2)</span>
<span class="pl-s">sum(is.na(datafile$tab))</span>
<span class="pl-s">datafile #shows info</span>
<span class="pl-s"></span>
<span class="pl-s">Output:</span>
<span class="pl-s">&gt; library("ape")</span>
<span class="pl-s">&gt; library("pegas")</span>
<span class="pl-s">&gt; library("seqinr")</span>
<span class="pl-s">&gt; library("ggplot2")</span>
<span class="pl-s">&gt; library("adegenet")</span>
<span class="pl-s">&gt; setwd("/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day09")</span>
<span class="pl-s">&gt; datafile&lt;-read.genepop(<span class="pl-pds">'</span></span>1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen<span class="pl-s"><span class="pl-pds">'</span>, ncode=2)</span>
<span class="pl-s"></span>
<span class="pl-s"> Converting data from a Genepop .gen file to a genind object... </span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">File description:  AllSNPs </span>
<span class="pl-s"></span>
<span class="pl-s">...done.</span>
<span class="pl-s"></span>
<span class="pl-s">&gt; sum(is.na(datafile$tab))</span>
<span class="pl-s">[1] 35966</span>
<span class="pl-s">&gt; datafile #shows info</span>
<span class="pl-s">/// GENIND OBJECT /////////</span>
<span class="pl-s"></span>
<span class="pl-s"> // 40 individuals; 1,578 loci; 3,156 alleles; size: 1.4 Mb</span>
<span class="pl-s"></span>
<span class="pl-s"> // Basic content</span>
<span class="pl-s">   @tab:  40 x 3156 matrix of allele counts</span>
<span class="pl-s">   @loc.n.all: number of alleles per locus (range: 2-2)</span>
<span class="pl-s">   @loc.fac: locus factor for the 3156 columns of @tab</span>
<span class="pl-s">   @all.names: list of allele names for each locus</span>
<span class="pl-s">   @ploidy: ploidy of each individual  (range: 2-2)</span>
<span class="pl-s">   @type:  codom</span>
<span class="pl-s">   @call: read.genepop(file = "1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen", </span>
<span class="pl-s">    ncode = 2)</span>
<span class="pl-s"></span>
<span class="pl-s"> // Optional content</span>
<span class="pl-s">   @pop: population of each individual (group size range: 10-10)</span>
<span class="pl-s"></span>
<span class="pl-s">Input:</span>
<span class="pl-s">YOURdata&lt;-scaleGen(datafile, NA.method=<span class="pl-pds">'</span></span>mean<span class="pl-s"><span class="pl-pds">'</span>)</span>
<span class="pl-s">X&lt;-YOURdata</span>
<span class="pl-s">substring(pop(datafile),1,4)</span>
<span class="pl-s">Y&lt;-as.factor(substring(pop(datafile),1,4))</span>
<span class="pl-s">pca1 &lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)</span>
<span class="pl-s"></span>
<span class="pl-s">Output:</span>
<span class="pl-s">&gt; YOURdata&lt;-scaleGen(datafile, NA.method=<span class="pl-pds">'</span></span>mean<span class="pl-s"><span class="pl-pds">'</span>)</span>
<span class="pl-s">Warning message:</span>
<span class="pl-s">In .local(x, ...) : Some scaling values are null.</span>
<span class="pl-s"> Corresponding alleles are removed.</span>
<span class="pl-s">&gt; X&lt;-YOURdata</span>
<span class="pl-s">&gt; substring(pop(datafile),1,4)</span>
<span class="pl-s"> [1] "RI_B" "RI_B" "RI_B" "RI_B" "RI_B" "RI_B" "RI_B" "RI_B" "RI_B"</span>
<span class="pl-s">[10] "RI_B" "VA_W" "VA_W" "VA_W" "VA_W" "VA_W" "VA_W" "VA_W" "VA_W"</span>
<span class="pl-s">[19] "VA_W" "VA_W" "VA_B" "VA_B" "VA_B" "VA_B" "VA_B" "VA_B" "VA_B"</span>
<span class="pl-s">[28] "VA_B" "VA_B" "VA_B" "RI_W" "RI_W" "RI_W" "RI_W" "RI_W" "RI_W"</span>
<span class="pl-s">[37] "RI_W" "RI_W" "RI_W" "RI_W"</span>
<span class="pl-s">&gt; Y&lt;-as.factor(substring(pop(datafile),1,4))</span>
<span class="pl-s">&gt; pca1 &lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)</span>
<span class="pl-s"></span>
<span class="pl-s">Input:</span>
<span class="pl-s">#### PCAs ####</span>
<span class="pl-s"># individual labels</span>
<span class="pl-s">s.label(pca1$li)</span>
<span class="pl-s"></span>
<span class="pl-s">Output:</span>
<span class="pl-s">#### PCAs ####</span>
<span class="pl-s">&gt; # individual labels</span>
<span class="pl-s">&gt; s.label(pca1$li)</span>
<span class="pl-s">Plotted each sample name as a point</span>
<span class="pl-s"></span>
<span class="pl-s">Input:</span>
<span class="pl-s"># population elipses</span>
<span class="pl-s">s.class(pca1$li, pop(datafile))</span>
<span class="pl-s"></span>
<span class="pl-s">Output:</span>
<span class="pl-s">&gt; # population elipses</span>
<span class="pl-s">&gt; s.class(pca1$li, pop(datafile))</span>
<span class="pl-s">Plotted population</span>
<span class="pl-s"></span>
<span class="pl-s">Input:</span>
<span class="pl-s">#color symbols, pop names</span>
<span class="pl-s">#pdf("YOURINITIALS_ColorPCA1v2.pdf")</span>
<span class="pl-s">col &lt;- c("blue","red", "green", "black")</span>
<span class="pl-s">s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)</span>
<span class="pl-s">add.scatter.eig(pca1$eig[1:3], 3,1,2, posi="topright")</span>
<span class="pl-s">title("PCA of DJB_data\naxes 1-2")</span>
<span class="pl-s"></span>
<span class="pl-s">Output: </span>
<span class="pl-s">&gt; #color symbols, pop names</span>
<span class="pl-s">&gt; #pdf("YOURINITIALS_ColorPCA1v2.pdf")</span>
<span class="pl-s">&gt; col &lt;- c("blue","red", "green", "black")</span>
<span class="pl-s">&gt; s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)</span>
<span class="pl-s">&gt; add.scatter.eig(pca1$eig[1:3], 3,1,2, posi="topright")</span>
<span class="pl-s">&gt; title("PCA of DJB_data\naxes 1-2")</span>
<span class="pl-s">Plotted population with color</span>
<span class="pl-s">Points are individuals with elipses around the populations</span>
<span class="pl-s">Not as many outliers as the other .gen file</span>
<span class="pl-s"></span>
<span class="pl-s">Input:</span>
<span class="pl-s">#### Snapclust ####</span>
<span class="pl-s">a.clust&lt;-snapclust(datafile, k = 2)</span>
<span class="pl-s">class(a.clust)</span>
<span class="pl-s">names(a.clust)</span>
<span class="pl-s">a.tab &lt;- table(pop(datafile), a.clust$group)</span>
<span class="pl-s">table.value(a.tab, col.labels = 1:2)</span>
<span class="pl-s"></span>
<span class="pl-s">Output:</span>
<span class="pl-s">&gt; #### Snapclust ####</span>
<span class="pl-s">&gt; a.clust&lt;-snapclust(datafile, k = 2)</span>
<span class="pl-s">&gt; class(a.clust)</span>
<span class="pl-s">[1] "snapclust" "list"     </span>
<span class="pl-s">&gt; names(a.clust)</span>
<span class="pl-s">[1] "group"     "ll"        "proba"     "converged" "n.iter"   </span>
<span class="pl-s">[6] "n.param"  </span>
<span class="pl-s">&gt; a.tab &lt;- table(pop(datafile), a.clust$group)</span>
<span class="pl-s">&gt; table.value(a.tab, col.labels = 1:2)</span>
<span class="pl-s">Gave us a table that shows out of your total populations which are the individuals from this population assigning too. </span>
<span class="pl-s">Most individuals are being assigned to blue populations with a few being assigned to brown population.</span>
<span class="pl-s"></span>
<span class="pl-s">Input:</span>
<span class="pl-s">compoplot(a.clust)</span>
<span class="pl-s"></span>
<span class="pl-s">Output:</span>
<span class="pl-s">&gt; compoplot(a.clust)</span>
<span class="pl-s">Gave us cluster diagram that showed the same thing where the majority are being assigned to the blue population with the exception of a few that are assigned to brown population.</span>
<span class="pl-s"></span>
<span class="pl-s">#6 - Submit and add everything to your logfile</span>
<span class="pl-s">git add README.md</span>
<span class="pl-s">git commit -m <span class="pl-pds">'</span></span>updating readme<span class="pl-s"><span class="pl-pds">'</span></span>
<span class="pl-s">git push -u origin main</span></pre>
                    </div>
                    <p dir="auto">
                     ##Day 10 19-Feb-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day10
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 - Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.
I did this.

#2 - cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam
[amali010@turing1 SAMS]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS
[amali010@turing1 SAMS]$ salloc
salloc: Pending job allocation 9280964
salloc: job 9280964 queued and waiting for resources
salloc: job 9280964 has been allocated resources
salloc: Granted job allocation 9280964
[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam


#3 - Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt YOURNAMEFullCounts_summed.txt NoMatch *_counts.txt
[amali010@coreV1-22-016 SAMS]$ mv match_counts.txt match_countsoutput.txt
[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt AMFullCounts_summed.txt NoMatch *counts.txt
Hits not matchedRI_B_06_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_14_clippedtrimmed.fastq_counts.txt=1698     RI_B_07_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_22_clippedtrimmed.fastq_counts.txt=1698     RI_W_06_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_14_clippedtrimmed.fastq_counts.txt=1698 RI_W_07_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_22_clippedtrimmed.fastq_counts.txt=1698 VA_B_06_18_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_18_clippedtrimmed.fastq_counts.txt=1698 VA_B_07_22_clippedtrimmed.fastq_counts.txt=1698     VA_W_06_18_clippedtrimmed.fastq_counts.txt=1698 VA_W_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_18_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_22_clippedtrimmed.fastq_counts.txt=1698
[amali010@coreV1-22-016 SAMS]$ ls
AMFullCounts_summed.txt
match_countsoutput.txt
RI_B_06_18_clippedtrimmed.fastq_counts.txt
RI_B_06_18_clippedtrimmed.fastq.sam
RI_B_07_14_clippedtrimmed.fastq_counts.txt
RI_B_07_14_clippedtrimmed.fastq.sam
RI_B_07_18_clippedtrimmed.fastq_counts.txt
RI_B_07_18_clippedtrimmed.fastq.sam
RI_B_07_22_clippedtrimmed.fastq_counts.txt
RI_B_07_22_clippedtrimmed.fastq.sam
RI_W_06_18_clippedtrimmed.fastq_counts.txt
RI_W_06_18_clippedtrimmed.fastq.sam
RI_W_07_14_clippedtrimmed.fastq_counts.txt
RI_W_07_14_clippedtrimmed.fastq.sam
RI_W_07_18_clippedtrimmed.fastq_counts.txt
RI_W_07_18_clippedtrimmed.fastq.sam
RI_W_07_22_clippedtrimmed.fastq_counts.txt
RI_W_07_22_clippedtrimmed.fastq.sam
VA_B_06_18_clippedtrimmed.fastq_counts.txt
VA_B_06_18_clippedtrimmed.fastq.sam
VA_B_07_14_clippedtrimmed.fastq_counts.txt
VA_B_07_14_clippedtrimmed.fastq.sam
VA_B_07_18_clippedtrimmed.fastq_counts.txt
VA_B_07_18_clippedtrimmed.fastq.sam
VA_B_07_22_clippedtrimmed.fastq_counts.txt
VA_B_07_22_clippedtrimmed.fastq.sam
VA_W_06_18_clippedtrimmed.fastq_counts.txt
VA_W_06_18_clippedtrimmed.fastq.sam
VA_W_07_14_clippedtrimmed.fastq_counts.txt
VA_W_07_14_clippedtrimmed.fastq.sam
VA_W_07_18_clippedtrimmed.fastq_counts.txt
VA_W_07_18_clippedtrimmed.fastq.sam
VA_W_07_22_clippedtrimmed.fastq_counts.txt
VA_W_07_22_clippedtrimmed.fastq.sam

#4 - scp YOURNAMEFullCounts_summed.txt to your laptop
areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)
$ pwd
/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10

[amali010@coreV1-22-016 SAMS]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS
[amali010@coreV1-22-016 SAMS]$ ls /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt

areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)
$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt .
amali010@turing.hpc.odu.edu's password:
AMFullCounts_summed.txt                           100%  756KB   4.8MB/s   00:00

areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)
$ ls
AMFullCounts_summed.txt
coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen
coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen
homework_day10.txt

#5 - Edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)
I did this in Notepad++ where I had to replace _clippedtrimmed.fastq_counts.txt_UniqueTotReads instead of just the _counts.txt_UniqueTotReads to retain the informative part of the sample names.

#6 - Submit and add everything to your logfile
git add README.md
git commit -m 'updating readme'
git push -u origin main" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.</span>
I did this.

<span class="pl-c"><span class="pl-c">#</span>2 - cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:</span>
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py <span class="pl-k">*</span>.sam
[amali010@turing1 SAMS]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS
[amali010@turing1 SAMS]$ salloc
salloc: Pending job allocation 9280964
salloc: job 9280964 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9280964 has been allocated resources
salloc: Granted job allocation 9280964
[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py <span class="pl-k">*</span>.sam


<span class="pl-c"><span class="pl-c">#</span>3 - Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:</span>
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt YOURNAMEFullCounts_summed.txt NoMatch <span class="pl-k">*</span>_counts.txt
[amali010@coreV1-22-016 SAMS]$ mv match_counts.txt match_countsoutput.txt
[amali010@coreV1-22-016 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt AMFullCounts_summed.txt NoMatch <span class="pl-k">*</span>counts.txt
Hits not matchedRI_B_06_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_14_clippedtrimmed.fastq_counts.txt=1698     RI_B_07_18_clippedtrimmed.fastq_counts.txt=1698 RI_B_07_22_clippedtrimmed.fastq_counts.txt=1698     RI_W_06_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_14_clippedtrimmed.fastq_counts.txt=1698 RI_W_07_18_clippedtrimmed.fastq_counts.txt=1698     RI_W_07_22_clippedtrimmed.fastq_counts.txt=1698 VA_B_06_18_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_B_07_18_clippedtrimmed.fastq_counts.txt=1698 VA_B_07_22_clippedtrimmed.fastq_counts.txt=1698     VA_W_06_18_clippedtrimmed.fastq_counts.txt=1698 VA_W_07_14_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_18_clippedtrimmed.fastq_counts.txt=1698     VA_W_07_22_clippedtrimmed.fastq_counts.txt=1698
[amali010@coreV1-22-016 SAMS]$ ls
AMFullCounts_summed.txt
match_countsoutput.txt
RI_B_06_18_clippedtrimmed.fastq_counts.txt
RI_B_06_18_clippedtrimmed.fastq.sam
RI_B_07_14_clippedtrimmed.fastq_counts.txt
RI_B_07_14_clippedtrimmed.fastq.sam
RI_B_07_18_clippedtrimmed.fastq_counts.txt
RI_B_07_18_clippedtrimmed.fastq.sam
RI_B_07_22_clippedtrimmed.fastq_counts.txt
RI_B_07_22_clippedtrimmed.fastq.sam
RI_W_06_18_clippedtrimmed.fastq_counts.txt
RI_W_06_18_clippedtrimmed.fastq.sam
RI_W_07_14_clippedtrimmed.fastq_counts.txt
RI_W_07_14_clippedtrimmed.fastq.sam
RI_W_07_18_clippedtrimmed.fastq_counts.txt
RI_W_07_18_clippedtrimmed.fastq.sam
RI_W_07_22_clippedtrimmed.fastq_counts.txt
RI_W_07_22_clippedtrimmed.fastq.sam
VA_B_06_18_clippedtrimmed.fastq_counts.txt
VA_B_06_18_clippedtrimmed.fastq.sam
VA_B_07_14_clippedtrimmed.fastq_counts.txt
VA_B_07_14_clippedtrimmed.fastq.sam
VA_B_07_18_clippedtrimmed.fastq_counts.txt
VA_B_07_18_clippedtrimmed.fastq.sam
VA_B_07_22_clippedtrimmed.fastq_counts.txt
VA_B_07_22_clippedtrimmed.fastq.sam
VA_W_06_18_clippedtrimmed.fastq_counts.txt
VA_W_06_18_clippedtrimmed.fastq.sam
VA_W_07_14_clippedtrimmed.fastq_counts.txt
VA_W_07_14_clippedtrimmed.fastq.sam
VA_W_07_18_clippedtrimmed.fastq_counts.txt
VA_W_07_18_clippedtrimmed.fastq.sam
VA_W_07_22_clippedtrimmed.fastq_counts.txt
VA_W_07_22_clippedtrimmed.fastq.sam

<span class="pl-c"><span class="pl-c">#</span>4 - scp YOURNAMEFullCounts_summed.txt to your laptop</span>
areej@DESKTOP-RF6VF4N MINGW64 <span class="pl-k">~</span>/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)
$ <span class="pl-c1">pwd</span>
/Users/areej/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10

[amali010@coreV1-22-016 SAMS]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS
[amali010@coreV1-22-016 SAMS]$ ls /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt

areej@DESKTOP-RF6VF4N MINGW64 <span class="pl-k">~</span>/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)
$ scp amali010@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/areej/data/SAMS/AMFullCounts_summed.txt <span class="pl-c1">.</span>
amali010@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">AMFullCounts_summed.txt                           100%  756KB   4.8MB/s   00:00</span>
<span class="pl-s"></span>
<span class="pl-s">areej@DESKTOP-RF6VF4N MINGW64 ~/Desktop/BIOL_803_Advanced_Genomics_Data_Analysis/In_class/21sp_advgenomics/assignments_exercises/day10 (master)</span>
<span class="pl-s">$ ls</span>
<span class="pl-s">AMFullCounts_summed.txt</span>
<span class="pl-s">coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen</span>
<span class="pl-s">coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen</span>
<span class="pl-s">homework_day10.txt</span>
<span class="pl-s"></span>
<span class="pl-s">#5 - Edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)</span>
<span class="pl-s">I did this in Notepad++ where I had to replace _clippedtrimmed.fastq_counts.txt_UniqueTotReads instead of just the _counts.txt_UniqueTotReads to retain the informative part of the sample names.</span>
<span class="pl-s"></span>
<span class="pl-s">#6 - Submit and add everything to your logfile</span>
<span class="pl-s">git add README.md</span>
<span class="pl-s">git commit -m <span class="pl-pds">'</span></span>updating readme<span class="pl-s"><span class="pl-pds">'</span></span>
<span class="pl-s">git push -u origin main</span></pre>
                    </div>
                    <p dir="auto">
                     ##Day 11 24-Feb-2021
                    </p>
                    <ul dir="auto">
                     <li>
                      Exercise day11
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1 - Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file
Did this during class" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1 - Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file</span>
Did this during class</pre>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="btTcbx4TJDxMHDP9MPnOkhQXi8ACFQ9Wrs_r4TENKtt5DEPAghhcXE9QyYRmrVNcYq2mI-K8dMzfHM9KNK60-A">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/areejmalik/AreejsAdvancedGenomicsLog/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fareejmalik%2FAreejsAdvancedGenomicsLog&amp;report=areejmalik+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/areejmalik/AreejsAdvancedGenomicsLog/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/areejmalik/packages?repo_name=AreejsAdvancedGenomicsLog">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/areejmalik/AreejsAdvancedGenomicsLog/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
