<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","report_hydro_web_vitals","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-ee1857a059c0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         gnrobinson/Eth_foc_project: Population genomics of Ethiopian Fusarium spps
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="5E8A:2F5DAB:1483D6D:1D9C8B4:6605D4F8" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjVFOEE6MkY1REFCOjE0ODNENkQ6MUQ5QzhCNDo2NjA1RDRGOCIsInZpc2l0b3JfaWQiOiI1NTYzNjg1OTkxMDg0ODI5Mjg2IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-turbo-transient="true" name="visitor-payload">
           <meta content="761aa5bcd2c4dff0c28b97b1798ac2ed71474acfbeea2f5115a10392ed9be10e" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:351660251" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Population genomics of Ethiopian Fusarium spps. Contribute to gnrobinson/Eth_foc_project development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/gnrobinson/Eth_foc_project" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/61b68404acd701c51fcf201ee20c2a8a69fe38ec18e0c78b73a92e3b226c3444/gnrobinson/Eth_foc_project" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="gnrobinson/Eth_foc_project: Population genomics of Ethiopian Fusarium spps" name="twitter:title">
                       <meta content="Population genomics of Ethiopian Fusarium spps. Contribute to gnrobinson/Eth_foc_project development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/61b68404acd701c51fcf201ee20c2a8a69fe38ec18e0c78b73a92e3b226c3444/gnrobinson/Eth_foc_project" property="og:image"/>
                        <meta content="Population genomics of Ethiopian Fusarium spps. Contribute to gnrobinson/Eth_foc_project development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="gnrobinson/Eth_foc_project: Population genomics of Ethiopian Fusarium spps" property="og:title"/>
                        <meta content="https://github.com/gnrobinson/Eth_foc_project" property="og:url"/>
                        <meta content="Population genomics of Ethiopian Fusarium spps. Contribute to gnrobinson/Eth_foc_project development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjExODY3NTUzNywidCI6MTcxMTY1ODIzN30=--dbe8f7c985ec8efcc4ffd88265700e0cb703bf4fa175a4611c9fafd9a52d860e" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="def536b5d69b8afc2117ff3aa904c09699292b9d809266f1d19b952c3b02461c" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="5aaf0d46cfb23ac0376126aa517fe17c5c6293814f8cfb7fb33598c257a2a6b9" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/gnrobinson/Eth_foc_project git https://github.com/gnrobinson/Eth_foc_project.git" name="go-import"/>
                        <meta content="31085908" name="octolytics-dimension-user_id">
                         <meta content="gnrobinson" name="octolytics-dimension-user_login">
                          <meta content="351660251" name="octolytics-dimension-repository_id">
                           <meta content="gnrobinson/Eth_foc_project" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="351660251" name="octolytics-dimension-repository_network_root_id">
                               <meta content="gnrobinson/Eth_foc_project" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/gnrobinson/Eth_foc_project" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b" data-view-component="true" id="dialog-show-dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b-description" aria-labelledby="dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-43f03cbe-8e32-4446-ad64-b0ab6b5d345b-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-e1323595-d927-4160-a625-ea1e36998602">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-f1c0c4fd-ade3-41de-9cae-d34efb387e20">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-62e3d180-3933-4cd7-82ad-526ecefe1de0">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-b01d6b6d-e813-45bf-b67e-266449e00c9c">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-39e2b9e9-d748-4f6a-8bd7-49e5f6d4c752">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-ddbfa59f-4da2-4c1a-8588-ed3052c30fa6">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-02702d25-b0d9-4309-93df-1d4002e694ad">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-652a69ae-d7c3-4f35-b662-b6c4027136d8">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               © 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: gnrobinson / Eth_foc_project" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              gnrobinson
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              Eth_foc_project
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"gnrobinson","screen_size":"compact"}' data-view-component="true" href="/gnrobinson">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 gnrobinson
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"Eth_foc_project","screen_size":"compact"}' data-view-component="true" href="/gnrobinson/Eth_foc_project">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 Eth_foc_project
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"gnrobinson","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/gnrobinson/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/gnrobinson">
             <span class="AppHeader-context-item-label">
              gnrobinson
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"Eth_foc_project","screen_size":"full"}' data-view-component="true" href="/gnrobinson/Eth_foc_project">
             <span class="AppHeader-context-item-label">
              Eth_foc_project
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="DeMV_6mNSuIibA4zGC0kjaR7U-vnSk-MlXKui3twB_hdwBl3OwCVLZX2Vh29usbWkPeTANNh-h1LCdA11rS6EA" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="gnrobinson" data-current-repository="gnrobinson/Eth_foc_project" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="1EoeeJigNXWCBurQfe5ugRKUhvDFyo6L122G5cbFk_-cgAlSvuZ8ANWqmAmb656QQDDEWeOtfuS7j8dOIfuIsw" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:gnrobinson/Eth_foc_project">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump to…" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump to…" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-2a3f0f29-f835-4a1a-9e37-2651fdb1c3be" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-eeea1ea7-cdb7-4ddb-b1e2-b2a1f75d38cb" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-eeea1ea7-cdb7-4ddb-b1e2-b2a1f75d38cb">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="Eca3q4wg7FxTh8bmFSdJtB8lngdzojq2bOYhZneGpnWDdK6TRPbfxuOlyOIrPkhFwi5TxCmC3ONiZjTcgdejzg"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="zmiJ3RK2221n06Q3Nj81KiffyEVNGslmXmIFygPfJj6CmutnSjeFtCy6GZp86vgX7o-sUhsfi739OtRyHJh_aw">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="HxxVfdLic5-uEA_weLWGr4NxfY9W3T0TniCG37xA-_sHQRjbhWDanC0ERnM459jFNbZ4CHgE3YvzgCQNhC8oMw">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="fsILjcxzXo1Y62O7GJRRrZAhuXSNrUK3RegNDK263YC5l2Sv1XSY778w-UbRLyq-QJlhvggLhIdKi1OpNS68ig">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-278a0cc7-4e13-438c-9865-d91fe0dae718" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-2ac58014-0719-490b-8273-8b4391afec8c" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-5bcc5315-f813-4681-955b-9656fb91ee58" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-b78e326a-a1c7-4e3f-b9ab-3db0db20757c" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-4519e1a0-e02b-4160-851c-25a841ee67ac" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-b24530ee-2532-46f5-91b2-dd61d16e6e28" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-888ce3f1-e30e-401f-b2e9-f18a0736615d" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-9104313c-b111-4121-9c33-88c05b18f442">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-9104313c-b111-4121-9c33-88c05b18f442" id="tooltip-888ce3f1-e30e-401f-b2e9-f18a0736615d" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-1e02e762-96a7-4849-8429-0cea0b8b919d" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-8c2aa5dd-f51e-4125-b46d-ffa9b2fe530b">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-8c2aa5dd-f51e-4125-b46d-ffa9b2fe530b" id="tooltip-1e02e762-96a7-4849-8429-0cea0b8b919d" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1ODIzN30=--a3d58ff09385c52ae3552274138d61887291be9e0a89eb7de4e1f44381cb90fd" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-935b0820-801d-4f3d-87cc-494ce0bf1aa4" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=351660251">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-f1767452-65c6-4024-9caa-53abb01decbf" data-view-component="true" id="dialog-show-dialog-f1767452-65c6-4024-9caa-53abb01decbf" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-f1767452-65c6-4024-9caa-53abb01decbf-description" aria-labelledby="dialog-f1767452-65c6-4024-9caa-53abb01decbf-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-f1767452-65c6-4024-9caa-53abb01decbf">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-f1767452-65c6-4024-9caa-53abb01decbf-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-96d74e6b-96fb-447d-97e9-b1f0f687a860" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-ff851888-5826-4e0a-a8ac-a7476fd15852" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-b0aa99c0-6e76-4967-950e-45eb8d31b998" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-557f6bc6-39f0-4c13-ab75-ac5237b96748" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-3c7f04fa-40ca-4abf-956c-6c634ce470e6" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-baaeb2d3-8bf9-49f6-993a-256dc5a92fcf" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-f1767452-65c6-4024-9caa-53abb01decbf" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-f1767452-65c6-4024-9caa-53abb01decbf-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-5547c2df-0c3d-4d63-b619-0e028823e823" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-ae141cdc-5d28-4922-8545-718c203a91ec">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-127f3b5f-6fa9-472d-ae57-6b4c3f45541b" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-7564fb2c-b91e-4aee-824a-46d50db0312f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-cbdc8b5c-7a47-4c2a-928d-18146f8e7cfd">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-70c6c168-a305-4f21-aedb-6afb79da83c6" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-7eef3239-69b1-4185-927e-339d5868da6f" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-815c4163-7057-4d23-8a8e-9bd299607e3c">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-3976102a-98d5-4950-be25-0d323e838e03">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-b8801465-09c0-48f3-901b-b2edc14802c1">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-c980320b-00e1-4394-9788-b63910f9027a" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-36f8bd6a-add1-4d7d-95dd-0204fbea16c0" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-bd074db6-a041-4145-a0d3-8aaff0d6791a">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-d4d7fee5-44ed-4cdc-9a64-167d26f52c85">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-f8597567-f573-4cbb-947f-53cee2a6a0a5">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-121d43da-5507-40b6-a4fa-01330920f455">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /gnrobinson/Eth_foc_project" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/gnrobinson/Eth_foc_project" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /gnrobinson/Eth_foc_project/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/gnrobinson/Eth_foc_project/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /gnrobinson/Eth_foc_project/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/gnrobinson/Eth_foc_project/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /gnrobinson/Eth_foc_project/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/gnrobinson/Eth_foc_project/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /gnrobinson/Eth_foc_project/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/gnrobinson/Eth_foc_project/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /gnrobinson/Eth_foc_project/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/gnrobinson/Eth_foc_project/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/gnrobinson/Eth_foc_project/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /gnrobinson/Eth_foc_project/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/gnrobinson/Eth_foc_project/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-list" aria-haspopup="true" aria-labelledby="tooltip-faec54e1-5d33-43a6-a4f9-3b9cab8b8ee2" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-button" popovertarget="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-button" id="tooltip-faec54e1-5d33-43a6-a4f9-3b9cab8b8ee2" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-button" anchor-offset="normal" data-view-component="true" id="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-26a60e05-63a8-4392-a056-89f1027cbbbc-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/gnrobinson/Eth_foc_project" id="item-a94e2742-cfaa-472c-a9d0-a8c3c52b5093" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/gnrobinson/Eth_foc_project/issues" id="item-f656fbd6-d31c-4c57-aa70-dc4224223c1d" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/gnrobinson/Eth_foc_project/pulls" id="item-f7f0c262-bb38-40bb-bab5-e34777a2c65d" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/gnrobinson/Eth_foc_project/actions" id="item-b73a36ee-ccc9-49ea-9f60-430901101b77" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/gnrobinson/Eth_foc_project/projects" id="item-4d3bf207-23e6-4274-b6d1-666633d66eba" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/gnrobinson/Eth_foc_project/security" id="item-ee47513e-2fef-4487-bcec-a598a922084e" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/gnrobinson/Eth_foc_project/pulse" id="item-31a48505-b1c8-4665-8a8a-5d30cc4a6dd5" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-492fa27b-93f8-4964-b29a-1a913fd24a8a" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-a02e67b3-6d3e-490e-8d75-4bca87cf8b8f" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-a02e67b3-6d3e-490e-8d75-4bca87cf8b8f" id="tooltip-492fa27b-93f8-4964-b29a-1a913fd24a8a" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1ODIzN30=--a3d58ff09385c52ae3552274138d61887291be9e0a89eb7de4e1f44381cb90fd" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/gnrobinson/Eth_foc_project" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjMxMDg1OTA4" data-targets="command-palette-scope.tokens" data-text="gnrobinson" data-type="owner" data-value="gnrobinson" style="white-space:nowrap;line-height:20px;">
         gnrobinson
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzNTE2NjAyNTE=" data-targets="command-palette-scope.tokens" data-text="Eth_foc_project" data-type="repository" data-value="Eth_foc_project" style="white-space:nowrap;line-height:20px;">
         Eth_foc_project
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-3feb9b43-b954-451f-8074-0b66162f59fd" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzNTE2NjAyNTE=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         We’ve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="gnrobinson" data-scope-id="MDQ6VXNlcjMxMDg1OTA4" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="Eth_foc_project" data-scope-id="MDEwOlJlcG9zaXRvcnkzNTE2NjAyNTE=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="gnrobinson" data-scope-id="MDQ6VXNlcjMxMDg1OTA4" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="Eth_foc_project" data-scope-id="MDEwOlJlcG9zaXRvcnkzNTE2NjAyNTE=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="L7zSRhKd9GWaSyBvu2zpgMGEvyBUQ7RAM2y4cqPaFOL_0Hhih0ciwnl2iggotHAmZId0VdBknXytxp3HFwsCVQ">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="⭐️ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="-Q_V7j5O2ENCkw3rsZjrccO5YkN8FQ6QU3HKdAOYue2ZJ4kf8WCKJ5TYkq9Smi9HwlM8MtDnsKr_W7C94fANQw">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/31085908?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/gnrobinson/Eth_foc_project">
            Eth_foc_project
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/351660251/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":351660251,"originating_url":"https://github.com/gnrobinson/Eth_foc_project","user_id":126681664}}' data-hydro-click-hmac="b9a857bdb2d6f69912a02f11c81c6ca33487362c13b81f290b7bc87e5b3f5695" data-view-component="true" href="/gnrobinson/Eth_foc_project/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-b9d2e045-a0ed-4c32-8dad-660afbc4bcd6" popover="manual">
              Fork your own copy of gnrobinson/Eth_foc_project
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-351660251">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/gnrobinson/Eth_foc_project/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-351660251" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-dbc223">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/gnrobinson/Eth_foc_project/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="JssQvPizjFOqLVEX6RX3QiY6ODmsmB9-cDJRuvdUn-u6ovmzOJ-B4_fupbSQuwbbnp6T14RMF4HARVYtrmBa3Q">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="TSm3mEuZofg9LpHFUOCthRojtbfPy661X-6oUjzn_JfRQF6Xi7WsSGDtZWYpTlwcooceWecfpkrvma_FZdM5oQ">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (1)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":351660251,"originating_url":"https://github.com/gnrobinson/Eth_foc_project","user_id":126681664}}' data-hydro-click-hmac="90500bbef8a51aed060bbba97d1eb8e69fbbfd6647b0326c362a2761511ff61f" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="1 user starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="1">
                  1
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-351660251-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/gnrobinson/Eth_foc_project/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-351660251-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-9b97d4">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/gnrobinson/Eth_foc_project/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="TsX-tGEL1YBvPfxq7EJvJMuCUx_ye21pQxrwQSPR5EeDAvjoeZdZH8QMUrRRUmspthVijxOUgRuZkUk-xfRehA">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (1)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":351660251,"originating_url":"https://github.com/gnrobinson/Eth_foc_project","user_id":126681664}}' data-hydro-click-hmac="df5ec9eff2b60260b4c7c8f77cacc538a0f080c291f48181bf3582aada9f04b4" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="1 user starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="1">
                 1
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-351660251-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/gnrobinson/Eth_foc_project/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-351660251-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-6cab39">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":351660251,"originating_url":"https://github.com/gnrobinson/Eth_foc_project","user_id":126681664}}' data-hydro-click-hmac="91f732722bba116e15e8081a1c93e706c3f704a8de3cc15b77df2f8c34d9bc7d" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="QswtQIHDRi4Ku8GnNnwsk9uwhENG4-aX5IHM7Ua0xExaDCE6n3wzL-ao9q3qnKrkhJUXVpO5TUPGej4ZEQV61Q">
                   <input name="repository_id" type="hidden" value="351660251"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="dI0Mk876o7ftRlrvqIWC79lYdUqawsnDazeDXFUbSANsTQDp0EXWtgFVbeV0ZQSYhn3mX0-YYhdJzHGoAqr2mg">
                   <input name="repository_id" type="hidden" value="351660251"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-250d2cbf-98f7-43b2-b063-13accd00f662" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-9264b5fd-90e5-406b-b849-5a78f89a9387" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":351660251,"originating_url":"https://github.com/gnrobinson/Eth_foc_project","user_id":126681664}}' data-hydro-click-hmac="b9a857bdb2d6f69912a02f11c81c6ca33487362c13b81f290b7bc87e5b3f5695" data-view-component="true" group_item="true" href="/gnrobinson/Eth_foc_project/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-9264b5fd-90e5-406b-b849-5a78f89a9387" popover="manual">
              Fork your own copy of gnrobinson/Eth_foc_project
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/gnrobinson/Eth_foc_project/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="qu6hkRTOESMMvYfDT06MTHOtwVHweMTSIBJraz_Eazo2h0ie1OIck1F-c2A24H3Vywlqv9iszC2QZWz8ZvCuDA">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-20354305-544b-4512-a25f-cf1df1426318" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":351660251,"originating_url":"https://github.com/gnrobinson/Eth_foc_project","user_id":126681664}}' data-hydro-click-hmac="90500bbef8a51aed060bbba97d1eb8e69fbbfd6647b0326c362a2761511ff61f" data-view-component="true" id="icon-button-f273c104-1734-4aa4-b324-6eef84f85719" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-f273c104-1734-4aa4-b324-6eef84f85719" id="tooltip-20354305-544b-4512-a25f-cf1df1426318" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/gnrobinson/Eth_foc_project/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="NMQai3t1dKyP0q6uGdtPBpchj8NRBrKrUWQL8PKZvBT5AxzXY-n4MyTjAHCky0sL6ra-U7DpXtmL77KPFLwG1w">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-75be2295-4a03-4879-b091-95b19749e731" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":351660251,"originating_url":"https://github.com/gnrobinson/Eth_foc_project","user_id":126681664}}' data-hydro-click-hmac="df5ec9eff2b60260b4c7c8f77cacc538a0f080c291f48181bf3582aada9f04b4" data-view-component="true" id="icon-button-c808fa87-5972-48be-bce8-a8d6564f1472" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-c808fa87-5972-48be-bce8-a8d6564f1472" id="tooltip-75be2295-4a03-4879-b091-95b19749e731" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           Population genomics of Ethiopian Fusarium spps
          </p>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/gnrobinson/Eth_foc_project/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             star
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/gnrobinson/Eth_foc_project/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/gnrobinson/Eth_foc_project/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/gnrobinson/Eth_foc_project/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/gnrobinson/Eth_foc_project/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/gnrobinson/Eth_foc_project/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/gnrobinson/Eth_foc_project/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/gnrobinson/Eth_foc_project?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         gnrobinson/Eth_foc_project
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/gnrobinson/Eth_foc_project/spoofed_commit_check/87769a72fd0205dfb639e0deafd5bfaecfd75c8d">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":351660251,"defaultBranch":"main","name":"Eth_foc_project","ownerLogin":"gnrobinson","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-03-26T12:19:40.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/31085908?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1681767735.0","canEdit":true,"refType":"branch","currentOid":"87769a72fd0205dfb639e0deafd5bfaecfd75c8d"},"tree":{"items":[{"name":"Supplemental_Methods","path":"Supplemental_Methods","contentType":"directory"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"calculating_coverage_diversity.R","path":"calculating_coverage_diversity.R","contentType":"file"},{"name":"calculating_haplotype_diversity.R","path":"calculating_haplotype_diversity.R","contentType":"file"},{"name":"geo_plot_ethiopia.R","path":"geo_plot_ethiopia.R","contentType":"file"},{"name":"ia_analysis.Rmd","path":"ia_analysis.Rmd","contentType":"file"},{"name":"ld_decay.R","path":"ld_decay.R","contentType":"file"},{"name":"making_ld_windows.Rmd","path":"making_ld_windows.Rmd","contentType":"file"},{"name":"making_pi_windows.R","path":"making_pi_windows.R","contentType":"file"},{"name":"making_tajimaD_windows.Rmd","path":"making_tajimaD_windows.Rmd","contentType":"file"},{"name":"pangenome_plotting.R","path":"pangenome_plotting.R","contentType":"file"},{"name":"plotting_structure_network.R","path":"plotting_structure_network.R","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":12,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/gnrobinson/Eth_foc_project/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/gnrobinson/Eth_foc_project.git","showCloneWarning":false,"sshUrl":"git@github.com:gnrobinson/Eth_foc_project.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone gnrobinson/Eth_foc_project","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/gnrobinson/Eth_foc_project","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fgnrobinson%2FEth_foc_project","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/gnrobinson/Eth_foc_project/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"21","overviewFiles":[{"displayName":"README.md","repoName":"Eth_foc_project","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBioinformatic Pipeline - \"Hiding in Plain Sight - Genome-wide recombination and a dynamic accessory genome drive diversity in Fusarium oxysporum fsp ciceris\"\u003c/h1\u003e\u003ca id=\"user-content-bioinformatic-pipeline---hiding-in-plain-sight---genome-wide-recombination-and-a-dynamic-accessory-genome-drive-diversity-in-fusarium-oxysporum-fsp-ciceris\" class=\"anchor\" aria-label=\"Permalink: Bioinformatic Pipeline - \u0026quot;Hiding in Plain Sight - Genome-wide recombination and a dynamic accessory genome drive diversity in Fusarium oxysporum fsp ciceris\u0026quot;\" href=\"#bioinformatic-pipeline---hiding-in-plain-sight---genome-wide-recombination-and-a-dynamic-accessory-genome-drive-diversity-in-fusarium-oxysporum-fsp-ciceris\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFayyaz A, Robinson G, Chang PL, Bekele D, Yimer SM, Carrasquilla-Garcia N, Negash K, Anand K. Surendrarao, Eric J.B. von Wettberg, Robbertse B, Seid Ahmedi, Tesfaye K, Fikre Ab, Farmer AD and Cook DR\u003c/h2\u003e\u003ca id=\"user-content-fayyaz-a-robinson-g-chang-pl-bekele-d-yimer-sm-carrasquilla-garcia-n-negash-k-anand-k-surendrarao-eric-jb-von-wettberg-robbertse-b-seid-ahmedi-tesfaye-k-fikre-ab-farmer-ad-and-cook-dr\" class=\"anchor\" aria-label=\"Permalink: Fayyaz A, Robinson G, Chang PL, Bekele D, Yimer SM, Carrasquilla-Garcia N, Negash K, Anand K. Surendrarao, Eric J.B. von Wettberg, Robbertse B, Seid Ahmedi, Tesfaye K, Fikre Ab, Farmer AD and Cook DR\" href=\"#fayyaz-a-robinson-g-chang-pl-bekele-d-yimer-sm-carrasquilla-garcia-n-negash-k-anand-k-surendrarao-eric-jb-von-wettberg-robbertse-b-seid-ahmedi-tesfaye-k-fikre-ab-farmer-ad-and-cook-dr\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eContents\u003c/h2\u003e\u003ca id=\"user-content-contents\" class=\"anchor\" aria-label=\"Permalink: Contents\" href=\"#contents\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1. \u003ca href=\"#Overview\"\u003eOverview\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-1-overview\" class=\"anchor\" aria-label=\"Permalink: 1. Overview\" href=\"#1-overview\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2. \u003ca href=\"#assembly\"\u003eGenome Assembly\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-2-genome-assembly\" class=\"anchor\" aria-label=\"Permalink: 2. Genome Assembly\" href=\"#2-genome-assembly\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3. \u003ca href=\"#gene_annotation\"\u003eGene Annotation and Analyses\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-3-gene-annotation-and-analyses\" class=\"anchor\" aria-label=\"Permalink: 3. Gene Annotation and Analyses\" href=\"#3-gene-annotation-and-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4. \u003ca href=\"#read_mapping\"\u003eIdentification of SNP and Genetic Groups\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-4-identification-of-snp-and-genetic-groups\" class=\"anchor\" aria-label=\"Permalink: 4. Identification of SNP and Genetic Groups\" href=\"#4-identification-of-snp-and-genetic-groups\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5. \u003ca href=\"#pop_gen\"\u003ePopulation Analyses\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-5-population-analyses\" class=\"anchor\" aria-label=\"Permalink: 5. Population Analyses\" href=\"#5-population-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e6. \u003ca href=\"#refs\"\u003eReferences\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-6-references\" class=\"anchor\" aria-label=\"Permalink: 6. References\" href=\"#6-references\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca id=\"user-content-Overview\"\u003e\u003c/a\u003e1. Overview\u003c/h1\u003e\u003ca id=\"user-content-1-overview-1\" class=\"anchor\" aria-label=\"Permalink: 1. Overview\" href=\"#1-overview-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_pnas_overview.png\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_pnas_overview.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca id=\"user-content-assembly\"\u003e\u003c/a\u003e2. Genome Assembly\u003c/h1\u003e\u003ca id=\"user-content-2-genome-assembly-1\" class=\"anchor\" aria-label=\"Permalink: 2. Genome Assembly\" href=\"#2-genome-assembly-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFastQC v0.11.9 (\u003ca href=\"https://github.com/s-andrews/FastQC\"\u003ehttps://github.com/s-andrews/FastQC\u003c/a\u003e)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eTrimmomatic v0.36 (Bolger et al, 2014)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eALLPATHS-LG v52488 (Gnerre et al, 2010)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eA5 (Tritt et al, 2012)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eHifiasm v16.0 (Cheng et al, 2021)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSingularity v3.5.2\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFCG-GX (\u003ca href=\"https://github.com/ncbi/fcs\"\u003ehttps://github.com/ncbi/fcs\u003c/a\u003e)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBUSCO v3 (Simao et al, 2015)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA. Quality check with FastQC\u003c/h3\u003e\u003ca id=\"user-content-a-quality-check-with-fastqc\" class=\"anchor\" aria-label=\"Permalink: A. Quality check with FastQC\" href=\"#a-quality-check-with-fastqc\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFastQC was used for the initial sequencing QC report.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load fastqc\nfastqc $IN.fastq.gz\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003emodule load fastqc\nfastqc $IN.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB. Low quality reads and adapters trimmed using Trimmomatic v36\u003c/h3\u003e\u003ca id=\"user-content-b-low-quality-reads-and-adapters-trimmed-using-trimmomatic-v36\" class=\"anchor\" aria-label=\"Permalink: B. Low quality reads and adapters trimmed using Trimmomatic v36\" href=\"#b-low-quality-reads-and-adapters-trimmed-using-trimmomatic-v36\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eReads that were lower than an average phred score \u0026lt; 33 were removed.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAll adapter sequences contained within Illumina's adaptor database were removed.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir results/trimmomatic/\u0026quot;$SAMPLE\u0026quot;\n\njava -jar bin/Trimmomatic-0.36/trimmomatic-0.36.jar PE -phred33 \\\ndata/atsede-redo-libs/\u0026quot;$SAMPLE\u0026quot;_L006_R1_001.fastq.gz \\\ndata/atsede-redo-libs/\u0026quot;$SAMPLE\u0026quot;_L006_R2_001.fastq.gz \\\nresults/trimmomatic/\u0026quot;$SAMPLE\u0026quot;/\u0026quot;$SAMPLE\u0026quot;.R1.paired.fastq.gz \\\nresults/trimmomatic/\u0026quot;$SAMPLE\u0026quot;/\u0026quot;$SAMPLE\u0026quot;.R1.unpaired.fastq.gz \\\nresults/trimmomatic/\u0026quot;$SAMPLE\u0026quot;/\u0026quot;$SAMPLE\u0026quot;.R2.paired.fastq.gz \\\nresults/trimmomatic/\u0026quot;$SAMPLE\u0026quot;/\u0026quot;$SAMPLE\u0026quot;.R2.unpaired.fastq.gz \\\nILLUMINACLIP:bin/Trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003emkdir results/trimmomatic/\"$SAMPLE\"\n\njava -jar bin/Trimmomatic-0.36/trimmomatic-0.36.jar PE -phred33 \\\ndata/atsede-redo-libs/\"$SAMPLE\"_L006_R1_001.fastq.gz \\\ndata/atsede-redo-libs/\"$SAMPLE\"_L006_R2_001.fastq.gz \\\nresults/trimmomatic/\"$SAMPLE\"/\"$SAMPLE\".R1.paired.fastq.gz \\\nresults/trimmomatic/\"$SAMPLE\"/\"$SAMPLE\".R1.unpaired.fastq.gz \\\nresults/trimmomatic/\"$SAMPLE\"/\"$SAMPLE\".R2.paired.fastq.gz \\\nresults/trimmomatic/\"$SAMPLE\"/\"$SAMPLE\".R2.unpaired.fastq.gz \\\nILLUMINACLIP:bin/Trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC. Error correction using ALLPATHS-LG\u003c/h3\u003e\u003ca id=\"user-content-c-error-correction-using-allpaths-lg\" class=\"anchor\" aria-label=\"Permalink: C. Error correction using ALLPATHS-LG\" href=\"#c-error-correction-using-allpaths-lg\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo remove erroneous mutations during sequencing, ALLPATHS-LG was used to error correct reads.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#ALLPATHS-LG was run with default parameters\n#e.g.\nErrorCorrectReads.pl read_p1.fastq.gz read_p2.fastq.gz PHRED_ENCODING=33 READS_OUT=$OUT\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#ALLPATHS-LG was run with default parameters\n#e.g.\nErrorCorrectReads.pl read_p1.fastq.gz read_p2.fastq.gz PHRED_ENCODING=33 READS_OUT=$OUT\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eD. Assembly using A5\u003c/h3\u003e\u003ca id=\"user-content-d-assembly-using-a5\" class=\"anchor\" aria-label=\"Permalink: D. Assembly using A5\" href=\"#d-assembly-using-a5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAssembly of all Illumina genomes was conducted with default parameters.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#A5 was run with default parameters\n#e.g.\na5_pipeline.pl read_p1.fastq.gz read_p2.fastq.gz mygenome\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#A5 was run with default parameters\n#e.g.\na5_pipeline.pl read_p1.fastq.gz read_p2.fastq.gz mygenome\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eE. Assembly using Hifiasm (PacBio)\u003c/h3\u003e\u003ca id=\"user-content-e-assembly-using-hifiasm-pacbio\" class=\"anchor\" aria-label=\"Permalink: E. Assembly using Hifiasm (PacBio)\" href=\"#e-assembly-using-hifiasm-pacbio\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAssembly of Fo-Et-0028 PacBio genome\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"module load hifiasm\nsource activate hifiasm-0.16.1\nhifiasm -o fo_et_0028_pacbio.asm --primary -t 32 demultiplex.bc1001--bc1001.hifi_reads.fastq.gz\nconda deactivate\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003emodule load hifiasm\nsource activate hifiasm-0.16.1\nhifiasm -o fo_et_0028_pacbio.asm --primary -t 32 demultiplex.bc1001--bc1001.hifi_reads.fastq.gz\nconda deactivate\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eF. Contaminant identification using FCS-GX\u003c/h3\u003e\u003ca id=\"user-content-f-contaminant-identification-using-fcs-gx\" class=\"anchor\" aria-label=\"Permalink: F. Contaminant identification using FCS-GX\" href=\"#f-contaminant-identification-using-fcs-gx\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo identify and remove contaminants from assemblies, we used NCBI's FCS-GX (available at \u003ca href=\"https://github.com/ncbi/fcs\"\u003ehttps://github.com/ncbi/fcs\u003c/a\u003e)\nThis program screens for contaminants by identifying contigs/scaffolds that do not originate from the biological source organism (as defined by Refseq)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#load singularity\nmodule load singularity\nmodule load python3/3.7.4\n\n#gathering fcs-gx singularity image\ncurl -LO https://github.com/ncbi/fcs/raw/main/dist/run_fcsgx.py\ncurl -LO https://github.com/ncbi/fcs/raw/main/examples/fcsgx_test.fa.gz\ncurl https://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/FCS/releases/0.3.0/fcs-gx.0.3.0.sif -Lo fcsgx.sif\n\n#create temporary shared memory\nSHM_LOC=/home/melland/cook/long_read/PB852_4plex_Fusarium_HiFiv3/r64069_20230119_233108/A1/DemuxBarcodes/outputs/assemblies_hifiasm/contaminant_removal\nmkdir -p \u0026quot;${SHM_LOC}/gxdb\u0026quot;\nmkdir -p ./gxdb\nmkdir -p ./gx_out\n\n#verify functionality using small test-only database\n#python3 ./run_fcsgx.py --fasta ./fcsgx_test.fa.gz --out-dir ./gx_out/ --gx-db \u0026quot;${SHM_LOC}/gxdb/test-only\u0026quot; --gx-db-disk ./gxdb --split-fasta --tax-id 6973 --container-engine=singularity --image=fcsgx.sif\n\n#do full contamination analysis\n#uses manifest file containing list of all fasta files paths\npython3 ./run_fcsgx.py --fasta ./manifest_fo.mft --out-dir ./gx_out/ --gx-db \u0026quot;${SHM_LOC}/gxdb/all\u0026quot; --split-fasta --tax-id 5507 --container-engine=singularity --image=fcsgx.sif\n\n# all contaminant contigs/scaffolds were removed unless they contained a bonafide Fusarium oxysporum gene (\u0026gt;95% identity)\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#load singularity\nmodule load singularity\nmodule load python3/3.7.4\n\n#gathering fcs-gx singularity image\ncurl -LO https://github.com/ncbi/fcs/raw/main/dist/run_fcsgx.py\ncurl -LO https://github.com/ncbi/fcs/raw/main/examples/fcsgx_test.fa.gz\ncurl https://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/FCS/releases/0.3.0/fcs-gx.0.3.0.sif -Lo fcsgx.sif\n\n#create temporary shared memory\nSHM_LOC=/home/melland/cook/long_read/PB852_4plex_Fusarium_HiFiv3/r64069_20230119_233108/A1/DemuxBarcodes/outputs/assemblies_hifiasm/contaminant_removal\nmkdir -p \"${SHM_LOC}/gxdb\"\nmkdir -p ./gxdb\nmkdir -p ./gx_out\n\n#verify functionality using small test-only database\n#python3 ./run_fcsgx.py --fasta ./fcsgx_test.fa.gz --out-dir ./gx_out/ --gx-db \"${SHM_LOC}/gxdb/test-only\" --gx-db-disk ./gxdb --split-fasta --tax-id 6973 --container-engine=singularity --image=fcsgx.sif\n\n#do full contamination analysis\n#uses manifest file containing list of all fasta files paths\npython3 ./run_fcsgx.py --fasta ./manifest_fo.mft --out-dir ./gx_out/ --gx-db \"${SHM_LOC}/gxdb/all\" --split-fasta --tax-id 5507 --container-engine=singularity --image=fcsgx.sif\n\n# all contaminant contigs/scaffolds were removed unless they contained a bonafide Fusarium oxysporum gene (\u0026gt;95% identity)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eG. Genome completeness assessed using BUSCO v3 (Sordariomyceta odb9 dataset)\u003c/h3\u003e\u003ca id=\"user-content-g-genome-completeness-assessed-using-busco-v3-sordariomyceta-odb9-dataset\" class=\"anchor\" aria-label=\"Permalink: G. Genome completeness assessed using BUSCO v3 (Sordariomyceta odb9 dataset)\" href=\"#g-genome-completeness-assessed-using-busco-v3-sordariomyceta-odb9-dataset\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe completeness of each genome was assessed using BUSCO. This was used as a filtering step for the pangenome construction, as only genomes with greater than 98% BUSCO complete. The Sordariomyceta ODB9 database was used to specify the lineage and the Fusarium graminearum was used for the gene finding parameters (Fusarium oxysporum is not available).\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## parameters to use for running BUSCO\ndocker run -it --rm -v $(pwd):/home/working -w /home/working chrishah/busco-docker run_BUSCO.py --in ./Fusarium_oxysporum.fasta --out Fusarium-BUSCO -l ./sordariomyceta_odb9 --mode genome --sp fusarium_graminearum --c 20\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## parameters to use for running BUSCO\ndocker run -it --rm -v $(pwd):/home/working -w /home/working chrishah/busco-docker run_BUSCO.py --in ./Fusarium_oxysporum.fasta --out Fusarium-BUSCO -l ./sordariomyceta_odb9 --mode genome --sp fusarium_graminearum --c 20\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca id=\"user-content-gene_annotation\"\u003e\u003c/a\u003e3. Gene Annotation and Analyses\u003c/h1\u003e\u003ca id=\"user-content-3-gene-annotation-and-analyses-1\" class=\"anchor\" aria-label=\"Permalink: 3. Gene Annotation and Analyses\" href=\"#3-gene-annotation-and-analyses-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePyani v0.2.9 (Pritchard et al, 2016)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eRepeatModeler v1.0.11 (\u003ca href=\"http://www.repeatmasker.org\" rel=\"nofollow\"\u003ehttp://www.repeatmasker.org\u003c/a\u003e)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eMAKER v2.31 (Cantarel et al, 2008)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eInterproscan v5.46 (Jones et al, 2014)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBLASTp v2.7.1\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAHRD v2.0 (\u003ca href=\"https://github.com/groupschoof/AHRD\"\u003ehttps://github.com/groupschoof/AHRD\u003c/a\u003e)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBLAST2GO GUI (\u003ca href=\"https://www.blast2go.com/\" rel=\"nofollow\"\u003ehttps://www.blast2go.com/\u003c/a\u003e)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eOrthofinder v2.5.4 (Emms and Kelly, 2019)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA. Average nucleotide identity calculated with Pyani\u003c/h3\u003e\u003ca id=\"user-content-a-average-nucleotide-identity-calculated-with-pyani\" class=\"anchor\" aria-label=\"Permalink: A. Average nucleotide identity calculated with Pyani\" href=\"#a-average-nucleotide-identity-calculated-with-pyani\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/afayyaz/bin/singularity run /home/afayyaz/singularityImages/average_nucleotide_identity-v0.2.9.simg -i all-genomes/ -g -m ANIb --workers 20 -o pyani-results\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e/home/afayyaz/bin/singularity run /home/afayyaz/singularityImages/average_nucleotide_identity-v0.2.9.simg -i all-genomes/ -g -m ANIb --workers 20 -o pyani-results\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB. Gene Annotation\u003c/h3\u003e\u003ca id=\"user-content-b-gene-annotation\" class=\"anchor\" aria-label=\"Permalink: B. Gene Annotation\" href=\"#b-gene-annotation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ei. The general gene annotation was done By using MAKER, RepeatModeler, Interproscan, BlastP and AHRD\u003c/h4\u003e\u003ca id=\"user-content-i-the-general-gene-annotation-was-done-by-using-maker-repeatmodeler-interproscan-blastp-and-ahrd\" class=\"anchor\" aria-label=\"Permalink: i. The general gene annotation was done By using MAKER, RepeatModeler, Interproscan, BlastP and AHRD\" href=\"#i-the-general-gene-annotation-was-done-by-using-maker-repeatmodeler-interproscan-blastp-and-ahrd\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRepeatModeler: We generated the library for custom repeats using RepeatModeler and used these custom library of repeats to feed into MAKER.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## to generate library for custom repeats using RepeatModeler\n\n $:/home/localhost/adf/sw/bin/singularity shell /erdos/adf/sw/singularityImages/repeatmodeler-1.0.11--pl5.22.0_0.img\n\n./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- BuildDatabase -name fusariumdatabase -engine ncbi Fo-Et-0090.fasta\n./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- RepeatModeler -database fusariumdatabase -LTRStruct -pa 20 \u0026gt;\u0026amp; repeats.out\n\n```{bash, eval=F}\n##structural annotation using MAKER\n\ni=`ls *.fasta`\nfor j in $i; do  maker -g $j 2\u0026gt;\u0026amp;1 | tee round1_$j.log ; done\nmkdir snap\ncd snap\nmaker2zff -n ../Fo-Et-0090.all.gff\nfathom genome.ann genome.dna -categorize 1000\nfathom uni.ann uni.dna -export 1000 -plus\nforge export.ann export.dna\nhmm-assembler.pl Fo-Et-0090 . \u0026gt; Fo-Et-0090.hmm\n##used this hmm file to train next round of MAKER for all of the genome\nfor j in $i; do  maker -g $j 2\u0026gt;\u0026amp;1 | tee round1_$j.log ; done\n\n##functional annotation using iprscan, blastp and AHRD for example using one genome \u0026quot;Fo-Et-0000\u0026quot;\n\nhandle_stop_codons.pl --stop_char '\\*' Fo-Et-0000.all.maker.proteins.fasta \u0026gt; Fo-Et-0000.all.maker.proteins-stop-remove.fasta\nmkdir chunks\nfasta2chunks.pl --chunksize=1000 --chunk_prefix=chunks/chunk Fo-Et-0000.all.maker.proteins-stop-remove.fasta\nIPRSCAN_HELPERS=/home/afayyaz/chado_preprocessing\nJOB_LIMIT=32\nexport ANALYSIS_DATA_ROOT=`pwd`\nmkdir iprscan; pushd iprscan\nif [[ `which qsub` != \u0026quot;\u0026quot; ]]; then\nfor f in ../chunks/*; do echo ${IPRSCAN_HELPERS}/run_iprscan.bash $f | qsub -cwd -pe smp 8; done\nelse\nfor f in ../chunks/*; do\nwhile (( `jobs | wc -l` \u0026gt;= $JOB_LIMIT )); do sleep 5; done;\n${IPRSCAN_HELPERS}/run_iprscan.bash $f \u0026amp;\ndone\nfi\n#then to get iprscan \u0026quot;raw\u0026quot; output for AHRD\n${IPRSCAN_HELPERS}/iprscan_convert.bash\nmkdir ../iprscan_raw\nmv *raw ../iprscan_raw\ncd ../\nmkdir ipr2go\nfor f in iprscan_raw/*; do ${IPRSCAN_HELPERS}/ipr2go.pl $f \u0026gt; `echo $f | sed 's/^iprscan_raw/ipr2go/'`; done\nforkjobs_blastp_ahrd.pl --forks $JOB_LIMIT chunks blastp\n\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/fol/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/yeast/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/gram/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/vene/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/poae/*\nmkdir AHRD\nperl -p -e 's/\\$\\{([^}]+)\\}/defined $ENV{$1} ? $ENV{$1} : $\u0026amp;/eg' /home/afayyaz/batcher_input_example_template.yml \u0026gt; AHRD/batcher_input_example.yml\nln -s ${INTERPRO_DB_ROOT}/interpro.dtd AHRD/interpro.dtd\npushd AHRD\nmkdir batch_yml\njava -cp ${AHRD_GIT_ROOT}/bin/ahrd.jar ahrd.controller.Batcher batcher_input_example.yml\nperl -pi -e 's/$/ \u0026amp;/;' batcher.bash\nsource batcher.bash\nfor f in *csv; do if [[ ! -s all.ahrd.tsv ]]; then grep '^Protein-Accession' $f \u0026gt; all.ahrd.tsv; fi; sed -n '4,$p' $f \u0026gt;\u0026gt; all.ahrd.tsv; done\n${IPRSCAN_HELPERS}/clean_AHRD.sh all.ahrd.tsv | sed 's/-mRNA-1//' \u0026gt; all.ahrd.tsv.cleaned\npopd\ncat iprscan_raw/* \u0026gt; all.iprscan_raw\nawk 'BEGIN {FS=\u0026quot;\\t\u0026quot;} NF==9 \u0026amp;\u0026amp; $2 == \u0026quot;maker\u0026quot; {print}' Fo-Et-0000.all.gff \u0026gt; Fo-Et-0000.all.maker.gff\n##We added the annotation into gff files\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img ipr_update_gff Fo-Et-0000.all.maker.gff all.iprscan_raw \u0026gt; Fo-Et-0000.all.gff.iprscan_updated\n${IPRSCAN_HELPERS}/add_note_attr_inGFF.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.gff.iprscan_updated \u0026gt; Fo-Et-0000.all.gff.iprscan_updated+AHRD\n##We generated the gene id's and replace it with existing gene identifiers in the fasta and gff files.\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img maker_map_ids --prefix FO0000- Fo-Et-0000.all.gff \u0026gt; map\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.proteins.fasta \nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.transcripts.fasta\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff.iprscan_updated+AHRD\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_data_ids map AHRD/all.ahrd.tsv.cleaned\ncut -d'|' -f1,1 Fo-Et-0000.all.maker.proteins.fasta | sed 's!-RA!!g' \u0026gt;\u0026gt; Fo-Et-0000.all.maker.proteins.final.fasta\ncut -d'|' -f1,1 Fo-Et-0000.all.maker.transcripts.fasta | sed 's!-RA!!g' \u0026gt;\u0026gt; Fo-Et-0000.all.maker.transcripts.final.fasta\n/home/localhost/adf/ahrdify_fasta.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.maker.proteins.fasta Fo-Et-0000.all.maker.transcripts.fasta\n##Repeated the same for rest of the genome\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## to generate library for custom repeats using RepeatModeler\n\n $:/home/localhost/adf/sw/bin/singularity shell /erdos/adf/sw/singularityImages/repeatmodeler-1.0.11--pl5.22.0_0.img\n\n./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- BuildDatabase -name fusariumdatabase -engine ncbi Fo-Et-0090.fasta\n./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- RepeatModeler -database fusariumdatabase -LTRStruct -pa 20 \u0026gt;\u0026amp; repeats.out\n\n```{bash, eval=F}\n##structural annotation using MAKER\n\ni=`ls *.fasta`\nfor j in $i; do  maker -g $j 2\u0026gt;\u0026amp;1 | tee round1_$j.log ; done\nmkdir snap\ncd snap\nmaker2zff -n ../Fo-Et-0090.all.gff\nfathom genome.ann genome.dna -categorize 1000\nfathom uni.ann uni.dna -export 1000 -plus\nforge export.ann export.dna\nhmm-assembler.pl Fo-Et-0090 . \u0026gt; Fo-Et-0090.hmm\n##used this hmm file to train next round of MAKER for all of the genome\nfor j in $i; do  maker -g $j 2\u0026gt;\u0026amp;1 | tee round1_$j.log ; done\n\n##functional annotation using iprscan, blastp and AHRD for example using one genome \"Fo-Et-0000\"\n\nhandle_stop_codons.pl --stop_char '\\*' Fo-Et-0000.all.maker.proteins.fasta \u0026gt; Fo-Et-0000.all.maker.proteins-stop-remove.fasta\nmkdir chunks\nfasta2chunks.pl --chunksize=1000 --chunk_prefix=chunks/chunk Fo-Et-0000.all.maker.proteins-stop-remove.fasta\nIPRSCAN_HELPERS=/home/afayyaz/chado_preprocessing\nJOB_LIMIT=32\nexport ANALYSIS_DATA_ROOT=`pwd`\nmkdir iprscan; pushd iprscan\nif [[ `which qsub` != \"\" ]]; then\nfor f in ../chunks/*; do echo ${IPRSCAN_HELPERS}/run_iprscan.bash $f | qsub -cwd -pe smp 8; done\nelse\nfor f in ../chunks/*; do\nwhile (( `jobs | wc -l` \u0026gt;= $JOB_LIMIT )); do sleep 5; done;\n${IPRSCAN_HELPERS}/run_iprscan.bash $f \u0026amp;\ndone\nfi\n#then to get iprscan \"raw\" output for AHRD\n${IPRSCAN_HELPERS}/iprscan_convert.bash\nmkdir ../iprscan_raw\nmv *raw ../iprscan_raw\ncd ../\nmkdir ipr2go\nfor f in iprscan_raw/*; do ${IPRSCAN_HELPERS}/ipr2go.pl $f \u0026gt; `echo $f | sed 's/^iprscan_raw/ipr2go/'`; done\nforkjobs_blastp_ahrd.pl --forks $JOB_LIMIT chunks blastp\n\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/fol/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/yeast/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/gram/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/vene/*\nperl -i ahrd_untruncate_blast_query_ids.pl blastp/poae/*\nmkdir AHRD\nperl -p -e 's/\\$\\{([^}]+)\\}/defined $ENV{$1} ? $ENV{$1} : $\u0026amp;/eg' /home/afayyaz/batcher_input_example_template.yml \u0026gt; AHRD/batcher_input_example.yml\nln -s ${INTERPRO_DB_ROOT}/interpro.dtd AHRD/interpro.dtd\npushd AHRD\nmkdir batch_yml\njava -cp ${AHRD_GIT_ROOT}/bin/ahrd.jar ahrd.controller.Batcher batcher_input_example.yml\nperl -pi -e 's/$/ \u0026amp;/;' batcher.bash\nsource batcher.bash\nfor f in *csv; do if [[ ! -s all.ahrd.tsv ]]; then grep '^Protein-Accession' $f \u0026gt; all.ahrd.tsv; fi; sed -n '4,$p' $f \u0026gt;\u0026gt; all.ahrd.tsv; done\n${IPRSCAN_HELPERS}/clean_AHRD.sh all.ahrd.tsv | sed 's/-mRNA-1//' \u0026gt; all.ahrd.tsv.cleaned\npopd\ncat iprscan_raw/* \u0026gt; all.iprscan_raw\nawk 'BEGIN {FS=\"\\t\"} NF==9 \u0026amp;\u0026amp; $2 == \"maker\" {print}' Fo-Et-0000.all.gff \u0026gt; Fo-Et-0000.all.maker.gff\n##We added the annotation into gff files\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img ipr_update_gff Fo-Et-0000.all.maker.gff all.iprscan_raw \u0026gt; Fo-Et-0000.all.gff.iprscan_updated\n${IPRSCAN_HELPERS}/add_note_attr_inGFF.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.gff.iprscan_updated \u0026gt; Fo-Et-0000.all.gff.iprscan_updated+AHRD\n##We generated the gene id's and replace it with existing gene identifiers in the fasta and gff files.\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img maker_map_ids --prefix FO0000- Fo-Et-0000.all.gff \u0026gt; map\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.proteins.fasta \nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.transcripts.fasta\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff.iprscan_updated+AHRD\nsingularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_data_ids map AHRD/all.ahrd.tsv.cleaned\ncut -d'|' -f1,1 Fo-Et-0000.all.maker.proteins.fasta | sed 's!-RA!!g' \u0026gt;\u0026gt; Fo-Et-0000.all.maker.proteins.final.fasta\ncut -d'|' -f1,1 Fo-Et-0000.all.maker.transcripts.fasta | sed 's!-RA!!g' \u0026gt;\u0026gt; Fo-Et-0000.all.maker.transcripts.final.fasta\n/home/localhost/adf/ahrdify_fasta.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.maker.proteins.fasta Fo-Et-0000.all.maker.transcripts.fasta\n##Repeated the same for rest of the genome\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eii. Gene Enrichment analysis was performed using BLAST2GO\u003c/h4\u003e\u003ca id=\"user-content-ii-gene-enrichment-analysis-was-performed-using-blast2go\" class=\"anchor\" aria-label=\"Permalink: ii. Gene Enrichment analysis was performed using BLAST2GO\" href=\"#ii-gene-enrichment-analysis-was-performed-using-blast2go\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eImplemented using GUI software (\u003ca href=\"https://www.blast2go.com/\" rel=\"nofollow\"\u003ehttps://www.blast2go.com/\u003c/a\u003e)\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eiii. SIX genes homologs were identified using BLASTp\u003c/h4\u003e\u003ca id=\"user-content-iii-six-genes-homologs-were-identified-using-blastp\" class=\"anchor\" aria-label=\"Permalink: iii. SIX genes homologs were identified using BLASTp\" href=\"#iii-six-genes-homologs-were-identified-using-blastp\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe used an E-value cutoff of 1E-6 to ensure that they were true homologs\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eiv. Orthologs were identified with Orthofinder\u003c/h4\u003e\u003ca id=\"user-content-iv-orthologs-were-identified-with-orthofinder\" class=\"anchor\" aria-label=\"Permalink: iv. Orthologs were identified with Orthofinder\" href=\"#iv-orthologs-were-identified-with-orthofinder\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\" singularity exec /home/afayyaz/singularity-images/orthofinder_2.5.4.sif orthofinder -t 50 -a 10 -f /home/afayyaz/99-isolate-orth-run/ \u0026gt; ortho_log_file 2\u0026gt;\u0026amp;1 \"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e singularity exec /home/afayyaz/singularity-images/orthofinder_2.5.4.sif orthofinder -t 50 -a 10 -f /home/afayyaz/99-isolate-orth-run/ \u0026gt; ortho_log_file 2\u0026gt;\u0026amp;1 \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC. Phylogenetic Analyses\u003c/h3\u003e\u003ca id=\"user-content-c-phylogenetic-analyses\" class=\"anchor\" aria-label=\"Permalink: C. Phylogenetic Analyses\" href=\"#c-phylogenetic-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools Required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBEAST v1.10.4 (Suchard et al, 2018)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eETE3 Toolkit (Huerta-Cepas et al, 2016)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eTo understand the evolutionary history of the Ethiopian collection, the species phylogenies were constructed using BEAST. The phylogenies were built from concatenated BUSCO gene alignments.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"##Firstly, We extracted the single copy gene names/list form BUSCO results\nperl busco_find_conserved_single_copy_across_all.pl run_*/full_table* \u0026gt; busco_find_conserved_single_copy_across_all.txt\n##then we extracted the fasta sequences from the genoem files\nfor i in `cat busco_find_conserved_single_copy_across_all.txt`; do grep -A1 $i all-busco-single-copy-genes.fna --no-group-separator \u0026gt;\u0026gt; $i.fna; done\n##We used clustalo for alignment of the 1556 genes\n/home/afayyaz/bin/clustalo -i genes.fna -o genes-aligned.fasta --outfmt=fasta --threads 40\n##then we used the aligned file, convert it into nexus file by using seqmagick(python package) and concatenate all of the nexus file into one file for beauti software \npython3.8 seqmagick.py convert --output-format nexus --alphabet dna /home/afayyaz/aligned-genes-fasta/EOG***.fna /home/afayyaz/edit-genes/EOG***.nex\npython3.8 cat-nex.py\n##generated the xml file in beauti and used in Beast \njava -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml\n\n##used standard parameters for all of the phylogenetic trees and visulaized in iTOL\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e##Firstly, We extracted the single copy gene names/list form BUSCO results\nperl busco_find_conserved_single_copy_across_all.pl run_*/full_table* \u0026gt; busco_find_conserved_single_copy_across_all.txt\n##then we extracted the fasta sequences from the genoem files\nfor i in `cat busco_find_conserved_single_copy_across_all.txt`; do grep -A1 $i all-busco-single-copy-genes.fna --no-group-separator \u0026gt;\u0026gt; $i.fna; done\n##We used clustalo for alignment of the 1556 genes\n/home/afayyaz/bin/clustalo -i genes.fna -o genes-aligned.fasta --outfmt=fasta --threads 40\n##then we used the aligned file, convert it into nexus file by using seqmagick(python package) and concatenate all of the nexus file into one file for beauti software \npython3.8 seqmagick.py convert --output-format nexus --alphabet dna /home/afayyaz/aligned-genes-fasta/EOG***.fna /home/afayyaz/edit-genes/EOG***.nex\npython3.8 cat-nex.py\n##generated the xml file in beauti and used in Beast \njava -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml\n\n##used standard parameters for all of the phylogenetic trees and visulaized in iTOL\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRobinson-Foulds (RF) is a metric that measures the distance between different phylogenetic trees. To check the congruence between different trees, we used the normalized Robinson-Foulds (nRF) with ETE3 toolkit. For phylogenetic trees that are highly congruent, the expectation under highly clonal populations, the nRF value would be close to 0, whereas highly incongruent phylogenetic trees would be closer to 1.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSNPs were called from the concatenated BUSCO alignments and filtered to remove any SNP sites with missing data. Because allele frequency could potentially play a role in the nRF calculations, each SNP site was binned based on allele frequency and normalized accordingly. The nRF was calculated by subsampling these SNPs randomly 10 times with each containing 356,463 SNPs (all SNPs) and 601,905 (ancestral SNPs only)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\n##We called SNP's on 1556 BUSCO single copy genes using snp-sites(https://github.com/sanger-pathogens/snp-sites)\nfor i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done\n\n##We removed all entries with missing data\ncat *fasta.vcf \u0026gt; all_genes.fasta.vcf \n$awk '$0 !~ \u0026quot;_\u0026quot; {print}' all_genes.fasta.vcf \u0026gt; all_gene_snp_nomissing.vcf \ncat all_gene_snp_nomissing.vcf | awk -v OFS=\u0026quot;\\t\u0026quot; '$0 !~ \u0026quot;^#\u0026quot; {hom_REF = 0; hom_ALT = 0; ; hom_var_2 = 0; hom_var_3 = 0; het = 0; for(i=10;i\u0026lt;=NF;i++) { if($i ~ /0/) hom_REF++; else if($i ~ /1/) hom_ALT++; else if($i ~ /2/) hom_var_2++;else if($i ~ /3/) hom_var_3++; else het++; } print $1, $2,$3, $4, $5, hom_REF, hom_ALT, hom_var_2, hom_var_3, het}' \u0026gt;\u0026gt; maf-column.txt\npaste all_gene_snp_nomissing.vcf maf-column.txt \u0026gt; all_gene_snp-maf-file.vcf\n\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e\n##We called SNP's on 1556 BUSCO single copy genes using snp-sites(https://github.com/sanger-pathogens/snp-sites)\nfor i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done\n\n##We removed all entries with missing data\ncat *fasta.vcf \u0026gt; all_genes.fasta.vcf \n$awk '$0 !~ \"_\" {print}' all_genes.fasta.vcf \u0026gt; all_gene_snp_nomissing.vcf \ncat all_gene_snp_nomissing.vcf | awk -v OFS=\"\\t\" '$0 !~ \"^#\" {hom_REF = 0; hom_ALT = 0; ; hom_var_2 = 0; hom_var_3 = 0; het = 0; for(i=10;i\u0026lt;=NF;i++) { if($i ~ /0/) hom_REF++; else if($i ~ /1/) hom_ALT++; else if($i ~ /2/) hom_var_2++;else if($i ~ /3/) hom_var_3++; else het++; } print $1, $2,$3, $4, $5, hom_REF, hom_ALT, hom_var_2, hom_var_3, het}' \u0026gt;\u0026gt; maf-column.txt\npaste all_gene_snp_nomissing.vcf maf-column.txt \u0026gt; all_gene_snp-maf-file.vcf\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"##R code for selecting SNP's\nlibrary(dplyr)\nfile \u0026lt;- read.delim(\u0026quot;all_gene_snp-maf-file.vcf.txt\u0026quot;)\nmydata \u0026lt;- data.frame(file)\ndata \u0026lt;- arrange(mydata, desc(maf))\nsubset1 \u0026lt;- subset(data, maf \u0026lt; 0.1 )\nsubset2 \u0026lt;- subset(data, maf \u0026gt;= 0.1 \u0026amp; maf \u0026lt; 0.2)\nsubset3 \u0026lt;- subset(data, maf \u0026gt;= 0.2 \u0026amp; maf \u0026lt; 0.3)\nsubset4 \u0026lt;- subset(data, maf \u0026gt;= 0.3 \u0026amp; maf \u0026lt; 0.4)\nsubset5 \u0026lt;- subset(data, maf \u0026gt; 0.4)\n###Total number of SNP's are 706149\n###No of SNP's in subset1 -\u0026gt; 299273\n###No of SNP's in subset2 -\u0026gt; 402709\n###No of SNP's in subset3 -\u0026gt; 2103\n###No of SNP's in subset4 -\u0026gt; 1052\n###No of SNP's in subset5 -\u0026gt; 1012\n##We randomly selected snps from subsets based on the percentage representation in the total amount\n\nx \u0026lt;- slice_sample(subset1, n=126831)\ny \u0026lt;- slice_sample(subset2, n=229624) \nz \u0026lt;- slice_sample(subset3, n=6)\na \u0026lt;- slice_sample(subset4, n=1)\nb \u0026lt;- slice_sample(subset5, n=1)\n\n#randomly selecting snps from subsets based on the percentage representation in the total amount\nwrite.table(x, file = \u0026quot;x-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(y, file = \u0026quot;y-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(z, file = \u0026quot;z-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(a, file = \u0026quot;a-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(b, file = \u0026quot;b-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\n#concatente in bash and add header of the original vcf file.\"\u003e\u003cpre lang=\"{r,\" class=\"notranslate\"\u003e\u003ccode\u003e##R code for selecting SNP's\nlibrary(dplyr)\nfile \u0026lt;- read.delim(\"all_gene_snp-maf-file.vcf.txt\")\nmydata \u0026lt;- data.frame(file)\ndata \u0026lt;- arrange(mydata, desc(maf))\nsubset1 \u0026lt;- subset(data, maf \u0026lt; 0.1 )\nsubset2 \u0026lt;- subset(data, maf \u0026gt;= 0.1 \u0026amp; maf \u0026lt; 0.2)\nsubset3 \u0026lt;- subset(data, maf \u0026gt;= 0.2 \u0026amp; maf \u0026lt; 0.3)\nsubset4 \u0026lt;- subset(data, maf \u0026gt;= 0.3 \u0026amp; maf \u0026lt; 0.4)\nsubset5 \u0026lt;- subset(data, maf \u0026gt; 0.4)\n###Total number of SNP's are 706149\n###No of SNP's in subset1 -\u0026gt; 299273\n###No of SNP's in subset2 -\u0026gt; 402709\n###No of SNP's in subset3 -\u0026gt; 2103\n###No of SNP's in subset4 -\u0026gt; 1052\n###No of SNP's in subset5 -\u0026gt; 1012\n##We randomly selected snps from subsets based on the percentage representation in the total amount\n\nx \u0026lt;- slice_sample(subset1, n=126831)\ny \u0026lt;- slice_sample(subset2, n=229624) \nz \u0026lt;- slice_sample(subset3, n=6)\na \u0026lt;- slice_sample(subset4, n=1)\nb \u0026lt;- slice_sample(subset5, n=1)\n\n#randomly selecting snps from subsets based on the percentage representation in the total amount\nwrite.table(x, file = \"x-export-file.txt\", sep = \"\\t\" )\nwrite.table(y, file = \"y-export-file.txt\", sep = \"\\t\" )\nwrite.table(z, file = \"z-export-file.txt\", sep = \"\\t\" )\nwrite.table(a, file = \"a-export-file.txt\", sep = \"\\t\" )\nwrite.table(b, file = \"b-export-file.txt\", sep = \"\\t\" )\n#concatente in bash and add header of the original vcf file.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#To filter for ancestral SNP's on a per clade basis\nfor F in *.vcf.gz; do bgzip -dc $F | //\nawk '{for(i=11;i\u0026lt;=NF;i++)if($i!=$(i-1)\u0026amp;\u0026amp;$1!~/#/)next}1' | //\nbgzip -c \u0026gt; ${F}_filtered.vcf.gz; done \n\n#The filtered VCF files for all clades were then combined to calculate the ancestral SNP Robinson-Foulds values\n\n###Total number of SNP's are 613071\n###No of SNP's in subset1 -\u0026gt; 607962\n###No of SNP's in subset2 -\u0026gt; 3486\n###No of SNP's in subset3 -\u0026gt; 795\n###No of SNP's in subset4 -\u0026gt; 455\n###No of SNP's in subset5 -\u0026gt; 329\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#To filter for ancestral SNP's on a per clade basis\nfor F in *.vcf.gz; do bgzip -dc $F | //\nawk '{for(i=11;i\u0026lt;=NF;i++)if($i!=$(i-1)\u0026amp;\u0026amp;$1!~/#/)next}1' | //\nbgzip -c \u0026gt; ${F}_filtered.vcf.gz; done \n\n#The filtered VCF files for all clades were then combined to calculate the ancestral SNP Robinson-Foulds values\n\n###Total number of SNP's are 613071\n###No of SNP's in subset1 -\u0026gt; 607962\n###No of SNP's in subset2 -\u0026gt; 3486\n###No of SNP's in subset3 -\u0026gt; 795\n###No of SNP's in subset4 -\u0026gt; 455\n###No of SNP's in subset5 -\u0026gt; 329\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"x \u0026lt;- slice_sample(subset1, n=601882)\ny \u0026lt;- slice_sample(subset2, n=20) \nz \u0026lt;- slice_sample(subset3, n=1)\na \u0026lt;- slice_sample(subset4, n=1)\nb \u0026lt;- slice_sample(subset5, n=1)\nwrite.table(x, file = \u0026quot;x-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(y, file = \u0026quot;y-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(z, file = \u0026quot;z-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(a, file = \u0026quot;a-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\nwrite.table(b, file = \u0026quot;b-export-file.txt\u0026quot;, sep = \u0026quot;\\t\u0026quot; )\"\u003e\u003cpre lang=\"{r,\" class=\"notranslate\"\u003e\u003ccode\u003ex \u0026lt;- slice_sample(subset1, n=601882)\ny \u0026lt;- slice_sample(subset2, n=20) \nz \u0026lt;- slice_sample(subset3, n=1)\na \u0026lt;- slice_sample(subset4, n=1)\nb \u0026lt;- slice_sample(subset5, n=1)\nwrite.table(x, file = \"x-export-file.txt\", sep = \"\\t\" )\nwrite.table(y, file = \"y-export-file.txt\", sep = \"\\t\" )\nwrite.table(z, file = \"z-export-file.txt\", sep = \"\\t\" )\nwrite.table(a, file = \"a-export-file.txt\", sep = \"\\t\" )\nwrite.table(b, file = \"b-export-file.txt\", sep = \"\\t\" )\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#concatente in bash and add header of the original vcf file.\n\n##converted vcf format to fasta format and used \ncat subset.vcf | perl vcf_to_fasta.pl \u0026gt; vcftofasta.fasta\n##generated the xml file in beauti and used in Beast \njava -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml\n##generated 10 different trees newick files labeld as \u0026quot;1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt\u0026quot;\nete3 compare -taboutput -t 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt -r 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt \u0026gt; result.txt\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#concatente in bash and add header of the original vcf file.\n\n##converted vcf format to fasta format and used \ncat subset.vcf | perl vcf_to_fasta.pl \u0026gt; vcftofasta.fasta\n##generated the xml file in beauti and used in Beast \njava -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml\n##generated 10 different trees newick files labeld as \"1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt\"\nete3 compare -taboutput -t 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt -r 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt \u0026gt; result.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eD. \u003ca id=\"user-content-pangenome\"\u003e\u003c/a\u003ePangenome\u003c/h3\u003e\u003ca id=\"user-content-d-pangenome\" class=\"anchor\" aria-label=\"Permalink: D. Pangenome\" href=\"#d-pangenome\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePanOCT v3.23 (Fouts et al, 2012)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eVCFtools v0.1.15 (Danecek et al, 2011)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003ePangenome was constructed using PanOCT software. Genes predicted by MAKER were clustered together if the percentage identity was \u0026gt;80% and the genes were genomically syntenic.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"singularity exec /home/afayyaz/singularity-images/panoct_3.23--pl526_1.sif panoct.pl -t blast-results.txt -g final-e-attribute.txt -P 99-all-protein.pep -f list-e.txt -b /home/afayyaz/group-e-panoct/ -i 95 -L 20 -M Y -F 1.33 -N Y -H Y -V Y -S Y -C Y -B Y -c 0,5,25,50,75,90,95,98,100 \u0026gt; panoct-test 2\u0026gt;\u0026amp;1\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003esingularity exec /home/afayyaz/singularity-images/panoct_3.23--pl526_1.sif panoct.pl -t blast-results.txt -g final-e-attribute.txt -P 99-all-protein.pep -f list-e.txt -b /home/afayyaz/group-e-panoct/ -i 95 -L 20 -M Y -F 1.33 -N Y -H Y -V Y -S Y -C Y -B Y -c 0,5,25,50,75,90,95,98,100 \u0026gt; panoct-test 2\u0026gt;\u0026amp;1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBefore we calculated the nucleotide diversity for the pangenome, we removed the paralogs from the gene presence absence file\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"###Removed paralogs from panoct result file\nPython3 script-paralogs.py\n###To calculate nucleotide diversity for pan-genome, we needed the files of each cluster in the pangenome containing number of genomes from 5 to 99\nawk -F\u0026quot;\\t\u0026quot; '{print \u0026gt;\u0026gt; ($101\u0026quot;.txt\u0026quot;)}' without-paralogs.txt  \n###Then we made a list of the cluster in the pangenome\nfor i in `ls cluster*`; do sed $'s/\\t/\\\\\\n/g' $i| sed '/^$/d' | sed '$d' | sed '$d' \u0026gt; $i.txt; done\n###We extracted the cluster sequence from each genome\nfor i in `ls cluster*.txt`; do python3 extract_seq.py final-99-isolate-transcript.fasta $i $i.fasta; done\n###We aligned the sequence using mafft\nfor i in `ls *.fasta`; do mafft --thread 20 $i \u0026gt; $i.aligned.fasta; done\n###We called SNP's using python package snp-sites(https://github.com/sanger-pathogens/snp-sites)\nfor i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done\nfor i in `ls *.vcf`; do sed 's/*/_/g' $i | awk '$0 !~ \u0026quot;_\u0026quot; {print}' | awk '{for(i=1;i\u0026lt;=NF;i++){if($i==0){$i=\u0026quot;0/0\u0026quot;}}}{$1=$1} 1' OFS=\u0026quot;\\t\u0026quot; | awk '{for(i=10;i\u0026lt;=NF;i++){if($i==1){$i=\u0026quot;1/1\u0026quot;}}}{$1=$1} 1' OFS=\u0026quot;\\t\u0026quot;| awk '{for(i=10;i\u0026lt;=NF;i++){if($i==2){$i=\u0026quot;2/2\u0026quot;}}}{$1=$1} 1' OFS=\u0026quot;\\t\u0026quot; \u0026gt; $i-p.vcf; done\n###We calculated nucleotide diversity using vcftools\nfor i in `ls *-p.vcf`; do vcftools --vcf $i --site-pi --out $i-nucl_diversity; done\nfor i in `ls *.pi`; do sed '1d' $i \u0026gt; $i.txt ; done\n##We took the average of each file\nls *.txt \u0026gt; file_list\nwhile read line; do awk '{sum+=$2} END {print sum/NR}' $line \u0026gt;\u0026gt; avg-pi-file; done\u0026lt;list-avg-pi.txt\nmv file_new avg-values\ncp *-aligned-files/avg* .\nfor i in `ls avg-*`; do awk '{print FILENAME (NF?\u0026quot;\\t\u0026quot;:\u0026quot;\u0026quot;) $0}' $i \u0026gt; $i.txt; done\ncat *.txt \u0026gt; all-cluster-pi.txt\n###We calculated tajimas-d using vcftools\nfor i in `ls *-p.vcf`; do vcftools --vcf $i --TajimaD 100000000 --out $i-tajimad.txt; done\nfor i in `ls *.Tajima.D.txt`; do sed '1d' $i\u0026gt; $i.txt; done\n##We took the average of each file\nwhile read line; do awk '{sum+=$5} END {print sum/NR}' $line \u0026gt;\u0026gt; avg-tajima-file; done\u0026lt;list-avg-tajima.txt\nmv file_new avg-values\ncp *-aligned-files/avg* .\nfor i in `ls avg-*`; do awk '{print FILENAME (NF?\u0026quot;\\t\u0026quot;:\u0026quot;\u0026quot;) $0}' $i \u0026gt; $i.txt; done\ncat *.D.txt \u0026gt; tajima.d.txt\n##repeat these steps for all clusters containing number of genomes from 5 to 99\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e###Removed paralogs from panoct result file\nPython3 script-paralogs.py\n###To calculate nucleotide diversity for pan-genome, we needed the files of each cluster in the pangenome containing number of genomes from 5 to 99\nawk -F\"\\t\" '{print \u0026gt;\u0026gt; ($101\".txt\")}' without-paralogs.txt  \n###Then we made a list of the cluster in the pangenome\nfor i in `ls cluster*`; do sed $'s/\\t/\\\\\\n/g' $i| sed '/^$/d' | sed '$d' | sed '$d' \u0026gt; $i.txt; done\n###We extracted the cluster sequence from each genome\nfor i in `ls cluster*.txt`; do python3 extract_seq.py final-99-isolate-transcript.fasta $i $i.fasta; done\n###We aligned the sequence using mafft\nfor i in `ls *.fasta`; do mafft --thread 20 $i \u0026gt; $i.aligned.fasta; done\n###We called SNP's using python package snp-sites(https://github.com/sanger-pathogens/snp-sites)\nfor i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done\nfor i in `ls *.vcf`; do sed 's/*/_/g' $i | awk '$0 !~ \"_\" {print}' | awk '{for(i=1;i\u0026lt;=NF;i++){if($i==0){$i=\"0/0\"}}}{$1=$1} 1' OFS=\"\\t\" | awk '{for(i=10;i\u0026lt;=NF;i++){if($i==1){$i=\"1/1\"}}}{$1=$1} 1' OFS=\"\\t\"| awk '{for(i=10;i\u0026lt;=NF;i++){if($i==2){$i=\"2/2\"}}}{$1=$1} 1' OFS=\"\\t\" \u0026gt; $i-p.vcf; done\n###We calculated nucleotide diversity using vcftools\nfor i in `ls *-p.vcf`; do vcftools --vcf $i --site-pi --out $i-nucl_diversity; done\nfor i in `ls *.pi`; do sed '1d' $i \u0026gt; $i.txt ; done\n##We took the average of each file\nls *.txt \u0026gt; file_list\nwhile read line; do awk '{sum+=$2} END {print sum/NR}' $line \u0026gt;\u0026gt; avg-pi-file; done\u0026lt;list-avg-pi.txt\nmv file_new avg-values\ncp *-aligned-files/avg* .\nfor i in `ls avg-*`; do awk '{print FILENAME (NF?\"\\t\":\"\") $0}' $i \u0026gt; $i.txt; done\ncat *.txt \u0026gt; all-cluster-pi.txt\n###We calculated tajimas-d using vcftools\nfor i in `ls *-p.vcf`; do vcftools --vcf $i --TajimaD 100000000 --out $i-tajimad.txt; done\nfor i in `ls *.Tajima.D.txt`; do sed '1d' $i\u0026gt; $i.txt; done\n##We took the average of each file\nwhile read line; do awk '{sum+=$5} END {print sum/NR}' $line \u0026gt;\u0026gt; avg-tajima-file; done\u0026lt;list-avg-tajima.txt\nmv file_new avg-values\ncp *-aligned-files/avg* .\nfor i in `ls avg-*`; do awk '{print FILENAME (NF?\"\\t\":\"\") $0}' $i \u0026gt; $i.txt; done\ncat *.D.txt \u0026gt; tajima.d.txt\n##repeat these steps for all clusters containing number of genomes from 5 to 99\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca id=\"user-content-read_mapping\"\u003e\u003c/a\u003e4. Identification of SNP and genetic groups\u003c/h1\u003e\u003ca id=\"user-content-4-identification-of-snp-and-genetic-groups-1\" class=\"anchor\" aria-label=\"Permalink: 4. Identification of SNP and genetic groups\" href=\"#4-identification-of-snp-and-genetic-groups-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA. Read mapping and Variant calling\u003c/h3\u003e\u003ca id=\"user-content-a-read-mapping-and-variant-calling\" class=\"anchor\" aria-label=\"Permalink: A. Read mapping and Variant calling\" href=\"#a-read-mapping-and-variant-calling\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBWA MEM v0.7.9a-r786 (Li, 2013)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSAMtools v1.3.1 (Li et al, 2009)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eVCFtools v0.1.15 (Danecek et al, 2011)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBCFtools v1.16 (Danecek et al, 2021)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePicard (\u003ca href=\"http://broadinstitute.github.io/picard/\" rel=\"nofollow\"\u003ehttp://broadinstitute.github.io/picard/\u003c/a\u003e)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eGATK v4.1 (McKenna et al, 2010)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#create index for F. oxysporum reference genome (Fol4287 - GCF_000149955.1)\nbwa index [-a bwtsw|is] reference.fasta index_prefix\nsamtools faidx reference.fasta\n##created sequence dictionary\njava -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar CreateSequenceDictionary REFERENCE=reference.fasta OUTPUT=reference.dict\n##lign reads and assign read group\nfor i in $(ls *.fastq.gz | cut -d '_' -f1,2,3| uniq); do bwa mem -t 60 -R \u0026quot;@RG\\tID:${i}\\tSM:${i}\\tPL:illumina\\tPU:Lane1\\tLB:${i}\u0026quot; reference.fasta ${i}_R1_pair_001.fastq.gz ${i}_R2_pair_001.fastq.gz \u0026gt; ${i}.sam; done\n##sorting sam files\nfor i in `ls *.sam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar SortSam I=$i O=$i.bam SORT_ORDER=coordinate; done\n##mark duplicates\nfor i in `ls *.bam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar MarkDuplicates I=$i O=$i-edit.bam REMOVE_DUPLICATES=true METRICS_FILE=$i-metrics.txt; done\n##sort bam file\nfor i in `ls *-edit.bam`; do java -jar /erdos/cook_lab/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar BuildBamIndex INPUT=$i ; done\n##called variants\nfor i in `ls *-edit.bam`; do gatk HaplotypeCaller -R reference.fasta -I $i -ploidy 1 --native-pair-hmm-threads 40 -ERC GVCF -O $i-raw_gVCF.vcf --spark-runner LOCAL; done\nfor i in `ls *-raw_gVCF.vcf`; do gatk GenotypeGVCFs -O $i-raw.vcf -R reference.fasta --spark-runner LOCAL -V $i; done\nfor i in `ls *-raw.vcf`; do bgzip -c $i \u0026gt; $i.gz; done\nfor i in `ls *-raw.vcf.gz`; do tabix -p vcf $i ; done\nbcftools merge *vcf.gz -o a-merged.vcf\n##extract SNP's and stats\ngatk SelectVariants -R reference.fasta -V a-merged.vcf -select-type SNP -O c-snps.vcf\nbcftools stats c-snps.vcf \u0026gt; stats-snps.txt\nvcftools --vcf c-snps.vcf --out c-filtered --minDP 10 --max-missing 0.95 --minQ 30 --recode --recode-INFO-all\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#create index for F. oxysporum reference genome (Fol4287 - GCF_000149955.1)\nbwa index [-a bwtsw|is] reference.fasta index_prefix\nsamtools faidx reference.fasta\n##created sequence dictionary\njava -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar CreateSequenceDictionary REFERENCE=reference.fasta OUTPUT=reference.dict\n##lign reads and assign read group\nfor i in $(ls *.fastq.gz | cut -d '_' -f1,2,3| uniq); do bwa mem -t 60 -R \"@RG\\tID:${i}\\tSM:${i}\\tPL:illumina\\tPU:Lane1\\tLB:${i}\" reference.fasta ${i}_R1_pair_001.fastq.gz ${i}_R2_pair_001.fastq.gz \u0026gt; ${i}.sam; done\n##sorting sam files\nfor i in `ls *.sam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar SortSam I=$i O=$i.bam SORT_ORDER=coordinate; done\n##mark duplicates\nfor i in `ls *.bam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar MarkDuplicates I=$i O=$i-edit.bam REMOVE_DUPLICATES=true METRICS_FILE=$i-metrics.txt; done\n##sort bam file\nfor i in `ls *-edit.bam`; do java -jar /erdos/cook_lab/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar BuildBamIndex INPUT=$i ; done\n##called variants\nfor i in `ls *-edit.bam`; do gatk HaplotypeCaller -R reference.fasta -I $i -ploidy 1 --native-pair-hmm-threads 40 -ERC GVCF -O $i-raw_gVCF.vcf --spark-runner LOCAL; done\nfor i in `ls *-raw_gVCF.vcf`; do gatk GenotypeGVCFs -O $i-raw.vcf -R reference.fasta --spark-runner LOCAL -V $i; done\nfor i in `ls *-raw.vcf`; do bgzip -c $i \u0026gt; $i.gz; done\nfor i in `ls *-raw.vcf.gz`; do tabix -p vcf $i ; done\nbcftools merge *vcf.gz -o a-merged.vcf\n##extract SNP's and stats\ngatk SelectVariants -R reference.fasta -V a-merged.vcf -select-type SNP -O c-snps.vcf\nbcftools stats c-snps.vcf \u0026gt; stats-snps.txt\nvcftools --vcf c-snps.vcf --out c-filtered --minDP 10 --max-missing 0.95 --minQ 30 --recode --recode-INFO-all\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB. Filtering VCF file\u003c/h3\u003e\u003ca id=\"user-content-b-filtering-vcf-file\" class=\"anchor\" aria-label=\"Permalink: B. Filtering VCF file\" href=\"#b-filtering-vcf-file\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBCFtools v1.16 (Danecek et al, 2021)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eVCFtools v0.1.15 (Danecek et al, 2011)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eMUMmer v3.23 (Marcais et al, 2018)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eFor most population analyses, we exclusively used high quality genomes (based on BUSCO completeness \u0026gt; 98%, N50 \u0026gt; 1,200 and genome size) - see \"Genome Assembly\". This provided a total list of 120 genomes, 99 of which were taxonomically within the FOSC (Fig. 2). This is reduced the total number of isolates (320) (Supplemental Fig. 1).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ei. We split the VCF into separate single isolate VCF files\u003c/h4\u003e\u003ca id=\"user-content-i-we-split-the-vcf-into-separate-single-isolate-vcf-files\" class=\"anchor\" aria-label=\"Permalink: i. We split the VCF into separate single isolate VCF files\" href=\"#i-we-split-the-vcf-into-separate-single-isolate-vcf-files\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"######## ./splitting_vcfs_retaining.sh ###########\n\n#!/usr/bin/env bash\nfor file in *.vcf.gz; do\n  for sample in `bcftools query -l $file`; do\n    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file\n  done\ndone\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e######## ./splitting_vcfs_retaining.sh ###########\n\n#!/usr/bin/env bash\nfor file in *.vcf.gz; do\n  for sample in `bcftools query -l $file`; do\n    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file\n  done\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eii. We indexed all the single isolate VCF files\u003c/h4\u003e\u003ca id=\"user-content-ii-we-indexed-all-the-single-isolate-vcf-files\" class=\"anchor\" aria-label=\"Permalink: ii. We indexed all the single isolate VCF files\" href=\"#ii-we-indexed-all-the-single-isolate-vcf-files\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for f in *vcf.gz; do tabix -f -p vcf $f; done\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003efor f in *vcf.gz; do tabix -f -p vcf $f; done\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eiii. We merged isolates from list of high quality genomes (information obtained from assembly data)\u003c/h4\u003e\u003ca id=\"user-content-iii-we-merged-isolates-from-list-of-high-quality-genomes-information-obtained-from-assembly-data\" class=\"anchor\" aria-label=\"Permalink: iii. We merged isolates from list of high quality genomes (information obtained from assembly data)\" href=\"#iii-we-merged-isolates-from-list-of-high-quality-genomes-information-obtained-from-assembly-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bcftools merge --file-list isolate_list \u0026gt; ethiopia_fo_structure.vcf\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ebcftools merge --file-list isolate_list \u0026gt; ethiopia_fo_structure.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eiv. We removed heterozygous/ambiguous SNPs using sed\u003c/h4\u003e\u003ca id=\"user-content-iv-we-removed-heterozygousambiguous-snps-using-sed\" class=\"anchor\" aria-label=\"Permalink: iv. We removed heterozygous/ambiguous SNPs using sed\" href=\"#iv-we-removed-heterozygousambiguous-snps-using-sed\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sed -i 's#0/1|0/2|0/3|0/4|1/2|1/3|1/4|2/3|2/4|3/4#[.]/[.]#g' ethiopia_fo_structure.vcf\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003esed -i 's#0/1|0/2|0/3|0/4|1/2|1/3|1/4|2/3|2/4|3/4#[.]/[.]#g' ethiopia_fo_structure.vcf\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ev. We removed SNPs with high missingness and evaluated MAF filtering\u003c/h4\u003e\u003ca id=\"user-content-v-we-removed-snps-with-high-missingness-and-evaluated-maf-filtering\" class=\"anchor\" aria-label=\"Permalink: v. We removed SNPs with high missingness and evaluated MAF filtering\" href=\"#v-we-removed-snps-with-high-missingness-and-evaluated-maf-filtering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo ensure that the subsequent VCF was high quality, especially for linkage disequilibrium, we adjusted filters to learn how different filters affect the subsequent linkage disequilibrium analyses.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe main two parameters were missingness and minor allele frequency (MAF). MAF was of particular importance to us, as often researchers adjust MAF cutoff as \u0026gt;0.05 without regard for the effects on the population.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#VCF file for population genetic analyses\nvcftools --vcf ethiopia_fo_structure.vcf --max-missing 0.9 --recode --stdout | bgzip -c \u0026gt; all_ethiopia_isolates.vcf.gz #total number of SNPs = 136772\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#VCF file for population genetic analyses\nvcftools --vcf ethiopia_fo_structure.vcf --max-missing 0.9 --recode --stdout | bgzip -c \u0026gt; all_ethiopia_isolates.vcf.gz #total number of SNPs = 136772\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis filtering step was evaluated to understand the effects of using MAF cutoffs (no MAF filter vs \"typical\" MAF \u0026gt;0.05). There were only minor differences in the outcomes. The clonal groupings remained the same. The LD decay curve only showed minor differences (see below)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/ld_total_maf.png\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/ld_total_maf.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003evi. To ensure SNPs were core components of the genome, mapping was informed by additional whole genome alignments (MUMmer)\u003c/h4\u003e\u003ca id=\"user-content-vi-to-ensure-snps-were-core-components-of-the-genome-mapping-was-informed-by-additional-whole-genome-alignments-mummer\" class=\"anchor\" aria-label=\"Permalink: vi. To ensure SNPs were core components of the genome, mapping was informed by additional whole genome alignments (MUMmer)\" href=\"#vi-to-ensure-snps-were-core-components-of-the-genome-mapping-was-informed-by-additional-whole-genome-alignments-mummer\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis was done on per chromosome basis (Fol4287 1-15)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#Here is the example of chromosome 7 (Suppl. Fig. 2)\nfor f in Fo-Et-0*; do nucmer -maxmatch -c 100 -p ${f}_chr7 chr7.fna ${f}; done #outputs delta file\n\n# Whole genome alignments were visualized and filtered in R\nRscript ./mummer_coverage.R\n\n# This led to the removal of any SNPs mapped to chromosomes 3, 6, 14 and 15 as well as supercontig 27 (end of chromosome 1)\n# Any reference contigs that mapped poorly to the Illumina genomes were removed running awk itieratively\n# e.g. awk '($0 !~ \u0026quot;Supercontig_2.27\u0026quot;){print}' all_ethiopia_isolates.vcf.gz\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#Here is the example of chromosome 7 (Suppl. Fig. 2)\nfor f in Fo-Et-0*; do nucmer -maxmatch -c 100 -p ${f}_chr7 chr7.fna ${f}; done #outputs delta file\n\n# Whole genome alignments were visualized and filtered in R\nRscript ./mummer_coverage.R\n\n# This led to the removal of any SNPs mapped to chromosomes 3, 6, 14 and 15 as well as supercontig 27 (end of chromosome 1)\n# Any reference contigs that mapped poorly to the Illumina genomes were removed running awk itieratively\n# e.g. awk '($0 !~ \"Supercontig_2.27\"){print}' all_ethiopia_isolates.vcf.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC. Identification of clonal groups using STRUCTURE\u003c/h3\u003e\u003ca id=\"user-content-c-identification-of-clonal-groups-using-structure\" class=\"anchor\" aria-label=\"Permalink: C. Identification of clonal groups using STRUCTURE\" href=\"#c-identification-of-clonal-groups-using-structure\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSTRUCTURE v2.3.4 (Pritchard et al, 2000)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eStructure Harvester (\u003ca href=\"https://taylor0.biology.ucla.edu/structureHarvester/\" rel=\"nofollow\"\u003ehttps://taylor0.biology.ucla.edu/structureHarvester/\u003c/a\u003e) (Earl et al, 2012)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCLUMPP v1.1.2 (Jakobsson and Rosenberg, 2007)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eSubdividing clonal lineages in fungal populations remains challenging. Traditionally, reasearchers have determined clonality based on identical MLGs. This approach fails on a whole genome scale due to the accumulation of erroneous mutations. We employed two different methods to subdivide the lineages: phylogenetics (see Gene Annotation and Analyses - BEAST) and K clustering (via STRUCTURE).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ei. \u003ca id=\"user-content-struct\"\u003e\u003c/a\u003eRandom subsampling SNPs\u003c/h4\u003e\u003ca id=\"user-content-i-random-subsampling-snps\" class=\"anchor\" aria-label=\"Permalink: i. Random subsampling SNPs\" href=\"#i-random-subsampling-snps\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo avoid SNP selection bias, the VCF file was randomly subsampled at a rate of ~2.5%.\nNote that this script does not provide exactly 2.5% of the total number of SNPs (Table 1 - Non-clone corrected = 136,772), it gives each SNP a 2.5% chance of being included within the final subsampling.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub1.vcf.gz\n\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub2.vcf.gz\n\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub3.vcf.gz\n\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub4.vcf.gz\n\n#etc.\n\n#Subsample SNP Totals:\n#Sub1 = 3658\n#Sub2 = 3816\n#Sub3 = 3642\n#Sub4 = 3754\n#Sub5 = 3769\n#Sub6 = 3671\n#Sub7 = 3737\n#Sub8 = 3774\n#Sub9 = 3745\n#Sub10 =3744\n\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub1.vcf.gz\n\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub2.vcf.gz\n\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub3.vcf.gz\n\nbgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025) print $0}' | bgzip -c \u0026gt; ethiopia_fo_structure.0_05.vcf.gz_sub4.vcf.gz\n\n#etc.\n\n#Subsample SNP Totals:\n#Sub1 = 3658\n#Sub2 = 3816\n#Sub3 = 3642\n#Sub4 = 3754\n#Sub5 = 3769\n#Sub6 = 3671\n#Sub7 = 3737\n#Sub8 = 3774\n#Sub9 = 3745\n#Sub10 =3744\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eVCF subsample files were converted into genotype files using a manual bash script\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"############ ./convertvcftogeno.sh ############\n#!/usr/bin/env bash\nfile=$1\nout=${file}.out\nbgzip -dc ${file} | awk '$1 ~ \u0026quot;Super\u0026quot; || $1 ~ \u0026quot;#CHROM\u0026quot; {print}' | awk '$1 !~ \u0026quot;##\u0026quot; {print}' | sed 's#0/0#0#g;s#1/1#1#g;s#2/2#2#g;s#3/3#3#g;s#4/4#4#g;s#0/1#-9#g;s#0/2#-9#g;s#0/3#-9#g;s#0/4#-9#g;s#1/2#-9#g;s#1/3#-9#g;s#1/4#-9#g;s#2/3#-9#g;s#2/4#-9#g;s#3/4#-9#g;s#./.#-9#g' | cut -f 2,10- | awk '\n{ \n    for (i=1; i\u0026lt;=NF; i++)  {\n        a[NR,i] = $i\n    }\n}\nNF\u0026gt;p { p = NF }\nEND {    \n    for(j=1; j\u0026lt;=p; j++) {\n        str=a[1,j]\n        for(i=2; i\u0026lt;=NR; i++){\n            str=str\u0026quot; \u0026quot;a[i,j];\n        }\n        print str\n    }\n}' \u0026gt; $out\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e############ ./convertvcftogeno.sh ############\n#!/usr/bin/env bash\nfile=$1\nout=${file}.out\nbgzip -dc ${file} | awk '$1 ~ \"Super\" || $1 ~ \"#CHROM\" {print}' | awk '$1 !~ \"##\" {print}' | sed 's#0/0#0#g;s#1/1#1#g;s#2/2#2#g;s#3/3#3#g;s#4/4#4#g;s#0/1#-9#g;s#0/2#-9#g;s#0/3#-9#g;s#0/4#-9#g;s#1/2#-9#g;s#1/3#-9#g;s#1/4#-9#g;s#2/3#-9#g;s#2/4#-9#g;s#3/4#-9#g;s#./.#-9#g' | cut -f 2,10- | awk '\n{ \n    for (i=1; i\u0026lt;=NF; i++)  {\n        a[NR,i] = $i\n    }\n}\nNF\u0026gt;p { p = NF }\nEND {    \n    for(j=1; j\u0026lt;=p; j++) {\n        str=a[1,j]\n        for(i=2; i\u0026lt;=NR; i++){\n            str=str\" \"a[i,j];\n        }\n        print str\n    }\n}' \u0026gt; $out\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eii. STRUCTURE\u003c/h4\u003e\u003ca id=\"user-content-ii-structure\" class=\"anchor\" aria-label=\"Permalink: ii. STRUCTURE\" href=\"#ii-structure\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRan STRUCTURE on each subsampling 10 times: burnin = 10,000 and MCMC reps = 50,000. K values = 2-20.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# e.g.\n#!/bin/bash\nmodule load structure-console/2.3.4\nstructure -K 10 -o structure_large10-1-2\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e# e.g.\n#!/bin/bash\nmodule load structure-console/2.3.4\nstructure -K 10 -o structure_large10-1-2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eiii. STRUCTUREharvester\u003c/h4\u003e\u003ca id=\"user-content-iii-structureharvester\" class=\"anchor\" aria-label=\"Permalink: iii. STRUCTUREharvester\" href=\"#iii-structureharvester\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsing outputs of STRUCTURE, combine them using STRUCTURE_HARVESTER (online web server: \u003ca href=\"http://taylor0.biology.ucla.edu/structureHarvester/\" rel=\"nofollow\"\u003ehttp://taylor0.biology.ucla.edu/structureHarvester/\u003c/a\u003e) to produce .indfiles for CLUMPP.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eiv. CLUMPP\u003c/h4\u003e\u003ca id=\"user-content-iv-clumpp\" class=\"anchor\" aria-label=\"Permalink: iv. CLUMPP\" href=\"#iv-clumpp\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRan CLUMPP on HPC cluster to collate 10 permutations of each subsample. Used LargeKGreedy algorithm.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#e.g.\n#!/bin/bash\nmodule load CLUMPP/1.1.2\nCLUMPP K2.paramfile -i K12.indfile -k 12 -c 133 -r 10 -m 3 -w 0 -s 2 -o K12.outfile -j K12.miscfile\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#e.g.\n#!/bin/bash\nmodule load CLUMPP/1.1.2\nCLUMPP K2.paramfile -i K12.indfile -k 12 -c 133 -r 10 -m 3 -w 0 -s 2 -o K12.outfile -j K12.miscfile\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ev. STRUCTURE plotted with R script\u003c/h4\u003e\u003ca id=\"user-content-v-structure-plotted-with-r-script\" class=\"anchor\" aria-label=\"Permalink: v. STRUCTURE plotted with R script\" href=\"#v-structure-plotted-with-r-script\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"######### ./structure_plotting_larger_collection #############\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e######### ./structure_plotting_larger_collection #############\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003evi. Clonal groups were assigned based on Euclidean distances\u003c/h4\u003e\u003ca id=\"user-content-vi-clonal-groups-were-assigned-based-on-euclidean-distances\" class=\"anchor\" aria-label=\"Permalink: vi. Clonal groups were assigned based on Euclidean distances\" href=\"#vi-clonal-groups-were-assigned-based-on-euclidean-distances\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"######### ./plotting_structure_network.R ##########\n./plotting_structure_network.R --list \u0026quot;list of all .outfile paths\u0026quot; --label \u0026quot;list of isolate names (in order)\u0026quot; --title \u0026quot;title of graph (usually K val included)\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e######### ./plotting_structure_network.R ##########\n./plotting_structure_network.R --list \"list of all .outfile paths\" --label \"list of isolate names (in order)\" --title \"title of graph (usually K val included)\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca id=\"user-content-pop_gen\"\u003e\u003c/a\u003e5. Population Analyses\u003c/h1\u003e\u003ca id=\"user-content-5-population-analyses-1\" class=\"anchor\" aria-label=\"Permalink: 5. Population Analyses\" href=\"#5-population-analyses-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA. Geographic Analyses\u003c/h3\u003e\u003ca id=\"user-content-a-geographic-analyses\" class=\"anchor\" aria-label=\"Permalink: A. Geographic Analyses\" href=\"#a-geographic-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ei. Plotting onto maps in R\u003c/h4\u003e\u003ca id=\"user-content-i-plotting-onto-maps-in-r\" class=\"anchor\" aria-label=\"Permalink: i. Plotting onto maps in R\" href=\"#i-plotting-onto-maps-in-r\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo create plots of geographic sampling (both chickpea and Fungal sampling)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"######### ./geo_plot_ethiopia.R ###########\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e######### ./geo_plot_ethiopia.R ###########\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_figure1.gif\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_figure1.gif\" alt=\"\" data-animated-image=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_geog_subgroups.gif\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_geog_subgroups.gif\" alt=\"\" data-animated-image=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_geog_mat.png\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_geog_mat.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eii. Mantel tests\u003c/h4\u003e\u003ca id=\"user-content-ii-mantel-tests\" class=\"anchor\" aria-label=\"Permalink: ii. Mantel tests\" href=\"#ii-mantel-tests\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe Mantel test is a measure of correlation between two distance matrices. Here we want to understand how genetic difference (both sequence and presence/absence variation) is correlated with geography.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"##We prepared the co-ordinates for all infected sample into data frame for mental test\nlibrary(ade4)\nlocation = read.delim(\u0026quot;location-for-distance-plot-99-isolates.txt\u0026quot;)\nrownames(location) \u0026lt;- location[, \u0026quot;Sample\u0026quot;]\nlocation \u0026lt;- location[, -1]\nx = as.matrix(location)\ny \u0026lt;- distm(x, fun = distGeo)\nphysical_distance \u0026lt;- melt(y)\n\nn \u0026lt;- max(table(physical_distance$Sample)) + 1  # +1,  so we have diagonal of \nres \u0026lt;- lapply(with(physical_distance, split(value, physical_distance$Sample)), function(x) c(rep(NA, n - length(x)), x))\nres \u0026lt;- do.call(\u0026quot;rbind\u0026quot;, res)\nres \u0026lt;- rbind(res, rep(NA, n))\nres \u0026lt;- as.dist(t(res))\nres\n##mantel test for accessary genome usimg Jaccarad distance\nm \u0026lt;- max(table(pan_distance$X)) + 1  # +1,  so we have diagonal of \nres2 \u0026lt;- lapply(with(pan_distance, split(value, pan_distance$X)), function(x) c(rep(NA, m - length(x)), x))\nres2 \u0026lt;- do.call(\u0026quot;rbind\u0026quot;, res2)\nres2 \u0026lt;- rbind(res2, rep(NA, m))\nres2 \u0026lt;- as.dist(t(res2))\nres2\nmantel.rtest(res, res2, nrepet = 9999)\n###manetl test for core using Conserved Ortholog genes were also performed the same.\"\u003e\u003cpre lang=\"{r,\" class=\"notranslate\"\u003e\u003ccode\u003e##We prepared the co-ordinates for all infected sample into data frame for mental test\nlibrary(ade4)\nlocation = read.delim(\"location-for-distance-plot-99-isolates.txt\")\nrownames(location) \u0026lt;- location[, \"Sample\"]\nlocation \u0026lt;- location[, -1]\nx = as.matrix(location)\ny \u0026lt;- distm(x, fun = distGeo)\nphysical_distance \u0026lt;- melt(y)\n\nn \u0026lt;- max(table(physical_distance$Sample)) + 1  # +1,  so we have diagonal of \nres \u0026lt;- lapply(with(physical_distance, split(value, physical_distance$Sample)), function(x) c(rep(NA, n - length(x)), x))\nres \u0026lt;- do.call(\"rbind\", res)\nres \u0026lt;- rbind(res, rep(NA, n))\nres \u0026lt;- as.dist(t(res))\nres\n##mantel test for accessary genome usimg Jaccarad distance\nm \u0026lt;- max(table(pan_distance$X)) + 1  # +1,  so we have diagonal of \nres2 \u0026lt;- lapply(with(pan_distance, split(value, pan_distance$X)), function(x) c(rep(NA, m - length(x)), x))\nres2 \u0026lt;- do.call(\"rbind\", res2)\nres2 \u0026lt;- rbind(res2, rep(NA, m))\nres2 \u0026lt;- as.dist(t(res2))\nres2\nmantel.rtest(res, res2, nrepet = 9999)\n###manetl test for core using Conserved Ortholog genes were also performed the same.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB. Linkage disequilibrium\u003c/h3\u003e\u003ca id=\"user-content-b-linkage-disequilibrium\" class=\"anchor\" aria-label=\"Permalink: B. Linkage disequilibrium\" href=\"#b-linkage-disequilibrium\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe calculated linkage disequilibrium using two different methods:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ePearson's correlation coefficient (r2) [implemented in PLINK]\u003c/li\u003e\n\u003cli\u003eIndex of association (rbarD) [implemented in PoppR]\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ei. Linkage Disequilibrium via r2\u003c/h4\u003e\u003ca id=\"user-content-i-linkage-disequilibrium-via-r2\" class=\"anchor\" aria-label=\"Permalink: i. Linkage Disequilibrium via r2\" href=\"#i-linkage-disequilibrium-via-r2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ePlink v1.90 (Purcell et al, 2007)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eBCFtools v1.16 (Danecek et al, 2021)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePreparing datasets for \"clonally-derived\" vs \"ancestrally-derived\" SNPs (Figure 4 and Table 1)\u003c/h4\u003e\u003ca id=\"user-content-preparing-datasets-for-clonally-derived-vs-ancestrally-derived-snps-figure-4-and-table-1\" class=\"anchor\" aria-label=\"Permalink: Preparing datasets for \u0026quot;clonally-derived\u0026quot; vs \u0026quot;ancestrally-derived\u0026quot; SNPs (Figure 4 and Table 1)\" href=\"#preparing-datasets-for-clonally-derived-vs-ancestrally-derived-snps-figure-4-and-table-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\"Clonally-derived\" SNPs are variants that are variable within clonal subgroups.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Ancestrally-derived\" SNPs are variants that are fixed within clonal subgroups\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo do this, we separated the final VCF file (See Filtering VCF) into the subgroups defined by BEAST (phylogenetics) and STRUCTURE (K clustering).\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eSplit the VCf file into groups\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"######## ./splitting_vcfs_retaining.sh ###########\n\n#!/usr/bin/env bash\nfor file in *.vcf.gz; do\n  for sample in `bcftools query -l $file`; do\n    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file\n  done\ndone\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e######## ./splitting_vcfs_retaining.sh ###########\n\n#!/usr/bin/env bash\nfor file in *.vcf.gz; do\n  for sample in `bcftools query -l $file`; do\n    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file\n  done\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eCombine into VCF files on a clade basis using BCFtools\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#e.g. This is an example for clonal subgroup E. Note that this was done for all groups E-U\n#note that \u0026quot;clade_e.txt\u0026quot; is a text file containing the list of isolates contained in subgroup E\nbcftools merge -l clade_e.txt \u0026gt; clade_e.vcf.gz\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#e.g. This is an example for clonal subgroup E. Note that this was done for all groups E-U\n#note that \"clade_e.txt\" is a text file containing the list of isolates contained in subgroup E\nbcftools merge -l clade_e.txt \u0026gt; clade_e.vcf.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eFiltering step for clonally-derived SNPs\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bgzip -dc  $group.vcf.gz | // #input file lineages\nawk '($0 ~ \u0026quot;1/1\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;0/0\u0026quot;) || ($0 ~ \u0026quot;1/1\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;2/2\u0026quot;) || //\n($0 ~ \u0026quot;2/2\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;0/0\u0026quot;) || ($0 ~ \u0026quot;3/3\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;0/0\u0026quot;) || //\n($0 ~ \u0026quot;4/4\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;0/0\u0026quot;) || ($0 ~ \u0026quot;3/3\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;1/1\u0026quot;) || //\n($0 ~ \u0026quot;4/4\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;1/1\u0026quot;) || ($0 ~ \u0026quot;3/3\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;2/2\u0026quot;) || //\n($0 ~ \u0026quot;4/4\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;2/2\u0026quot;) || ($0 ~ \u0026quot;3/3\u0026quot; \u0026amp;\u0026amp; $0 ~ \u0026quot;4/4\u0026quot;) || //\n$1 ~ /#/ {print $0}' | bgzip -c \u0026gt; $output.vcf.gz #output SNPs variable across each clonal group\n#note that these files were used to generate Figure 4C (decay plot for each clonal lineage)\n\nbcftools merge -l $(ls clade*) \u0026gt; var_clones.vcf.gz\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ebgzip -dc  $group.vcf.gz | // #input file lineages\nawk '($0 ~ \"1/1\" \u0026amp;\u0026amp; $0 ~ \"0/0\") || ($0 ~ \"1/1\" \u0026amp;\u0026amp; $0 ~ \"2/2\") || //\n($0 ~ \"2/2\" \u0026amp;\u0026amp; $0 ~ \"0/0\") || ($0 ~ \"3/3\" \u0026amp;\u0026amp; $0 ~ \"0/0\") || //\n($0 ~ \"4/4\" \u0026amp;\u0026amp; $0 ~ \"0/0\") || ($0 ~ \"3/3\" \u0026amp;\u0026amp; $0 ~ \"1/1\") || //\n($0 ~ \"4/4\" \u0026amp;\u0026amp; $0 ~ \"1/1\") || ($0 ~ \"3/3\" \u0026amp;\u0026amp; $0 ~ \"2/2\") || //\n($0 ~ \"4/4\" \u0026amp;\u0026amp; $0 ~ \"2/2\") || ($0 ~ \"3/3\" \u0026amp;\u0026amp; $0 ~ \"4/4\") || //\n$1 ~ /#/ {print $0}' | bgzip -c \u0026gt; $output.vcf.gz #output SNPs variable across each clonal group\n#note that these files were used to generate Figure 4C (decay plot for each clonal lineage)\n\nbcftools merge -l $(ls clade*) \u0026gt; var_clones.vcf.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eFiltering step for ancestrally-derived SNPs\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#filter across clonal subgroup\nfor F in *.vcf.gz; do bgzip -dc $F | //\nawk '{for(i=11;i\u0026lt;=NF;i++)if($i!=$(i-1)\u0026amp;\u0026amp;$1!~/#/)next}1' | //\nbgzip -c \u0026gt; ${F}_filtered.vcf.gz; done #output SNPs fixed across entire clonal group\n\n#choose one sequence from the subgroup VCF files\nfor F in *_filtered.vcf.gz; do bgzip -dc $F | //\ncut -f 1-11 | //\nbgzip -c \u0026gt; ${F}_represent.vcf.gz; done\n\n#merge representative isolates\nbcftools merge -l $(ls *_represent.vcf.gz) \u0026gt; eth_clonecorrected.vcf.gz\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#filter across clonal subgroup\nfor F in *.vcf.gz; do bgzip -dc $F | //\nawk '{for(i=11;i\u0026lt;=NF;i++)if($i!=$(i-1)\u0026amp;\u0026amp;$1!~/#/)next}1' | //\nbgzip -c \u0026gt; ${F}_filtered.vcf.gz; done #output SNPs fixed across entire clonal group\n\n#choose one sequence from the subgroup VCF files\nfor F in *_filtered.vcf.gz; do bgzip -dc $F | //\ncut -f 1-11 | //\nbgzip -c \u0026gt; ${F}_represent.vcf.gz; done\n\n#merge representative isolates\nbcftools merge -l $(ls *_represent.vcf.gz) \u0026gt; eth_clonecorrected.vcf.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLinkage disequilibrium - r2\u003c/h4\u003e\u003ca id=\"user-content-linkage-disequilibrium---r2\" class=\"anchor\" aria-label=\"Permalink: Linkage disequilibrium - r2\" href=\"#linkage-disequilibrium---r2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun intra-chromosomal PLINK LD\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#code for clonally-derived SNPs\nplink --vcf var_clones.vcf.gz --allow-extra-chr --r2 //\n--ld-window-r2 0 --ld-window 99999 --out var_clone\n\n#code for ancestrally-derived SNPs\nplink --vcf total_clone.vcf.gz --allow-extra-chr --r2 --ld-window 99999 //\n--ld-window-r2 0 --out total_clone\n\n#note that PLINK automatically retains only biallelic SNPs\n\n#calculating bp difference from PLINK LD output\nfor f in *.ld; do awk -F' ' '{print $1, $2, $5, $7, ($2-$5) }' $f | sed 's/-//g;s/ /\\t/g' \u0026gt; diff_calculated/$f.diff; done\n\n#filtering for bp difference \u0026lt;75000 (saves memory to permit R-based analysis)\nfor f in *diff; do awk '($5 \u0026lt; 75000) {print}' $f \u0026gt; rm_large_diffs/$f.75000.ld; done\n\n#rest of plotting done in R\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#code for clonally-derived SNPs\nplink --vcf var_clones.vcf.gz --allow-extra-chr --r2 //\n--ld-window-r2 0 --ld-window 99999 --out var_clone\n\n#code for ancestrally-derived SNPs\nplink --vcf total_clone.vcf.gz --allow-extra-chr --r2 --ld-window 99999 //\n--ld-window-r2 0 --out total_clone\n\n#note that PLINK automatically retains only biallelic SNPs\n\n#calculating bp difference from PLINK LD output\nfor f in *.ld; do awk -F' ' '{print $1, $2, $5, $7, ($2-$5) }' $f | sed 's/-//g;s/ /\\t/g' \u0026gt; diff_calculated/$f.diff; done\n\n#filtering for bp difference \u0026lt;75000 (saves memory to permit R-based analysis)\nfor f in *diff; do awk '($5 \u0026lt; 75000) {print}' $f \u0026gt; rm_large_diffs/$f.75000.ld; done\n\n#rest of plotting done in R\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eRun inter-chromosomal PLINK LD\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eSince inter-chromosomal SNPs can create unreasonably large file sizes (\u0026gt;100GB), we subset the SNP files before running\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#selects ~2.5% of the total SNPs (~2000-3000 SNPs)\n#run on both var_clone.vcf.gz and eth_clonecorrected.vcf.gz\nawk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025 || FNR==1) print $0}' $IN.vcf.gz \u0026gt; $OUT-filtered.vcf.gz\n\n#calculating interchromosomal LD\nplink --vcf var_clone.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //\n--ld-window-r2 0 --out var_clone_inter\n\n#calculating interchromosomal LD\nplink --vcf eth_clonecorrected.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //\n--ld-window-r2 0 --out total_clone_inter\n\n#mean of each LD was calculated (background LD)\n# file.ld = any inter-chromosomal PLINK LD output\ncount=0; total=0; for i in $( awk '{ print $7; }' file.ld );\\\ndo total=$(echo $total+$i | bc ); \\\n((count++)); done; echo \u0026quot;scale=2; $total / $count\u0026quot; | bc\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#selects ~2.5% of the total SNPs (~2000-3000 SNPs)\n#run on both var_clone.vcf.gz and eth_clonecorrected.vcf.gz\nawk 'BEGIN {srand()} !/^$/ { if (rand() \u0026lt;= .025 || FNR==1) print $0}' $IN.vcf.gz \u0026gt; $OUT-filtered.vcf.gz\n\n#calculating interchromosomal LD\nplink --vcf var_clone.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //\n--ld-window-r2 0 --out var_clone_inter\n\n#calculating interchromosomal LD\nplink --vcf eth_clonecorrected.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //\n--ld-window-r2 0 --out total_clone_inter\n\n#mean of each LD was calculated (background LD)\n# file.ld = any inter-chromosomal PLINK LD output\ncount=0; total=0; for i in $( awk '{ print $7; }' file.ld );\\\ndo total=$(echo $total+$i | bc ); \\\n((count++)); done; echo \"scale=2; $total / $count\" | bc\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003ePlotting r2 decay curves and genome-wide r2\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003er2 decay plots\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"############# ld_decay.R ##################\"\u003e\u003cpre lang=\"{r,\" class=\"notranslate\"\u003e\u003ccode\u003e############# ld_decay.R ##################\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_ld_decay_figures.gif\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_ld_decay_figures.gif\" alt=\"\" data-animated-image=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003er2 genome-wide plots\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"############## making_ld_windows.Rmd ###############\"\u003e\u003cpre lang=\"{r,\" class=\"notranslate\"\u003e\u003ccode\u003e############## making_ld_windows.Rmd ###############\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_ld_across_window.gif\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_ld_across_window.gif\" alt=\"\" data-animated-image=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eii. Linkage disequilibrium - rbarD\u003c/h4\u003e\u003ca id=\"user-content-ii-linkage-disequilibrium---rbard\" class=\"anchor\" aria-label=\"Permalink: ii. Linkage disequilibrium - rbarD\" href=\"#ii-linkage-disequilibrium---rbard\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo complement the Pearson's correlation coefficient (r2) LD data, we included the index of association (rbarD)(Agapow and Burt, 2001)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOne issue that we experienced with this metric though is that it is very computationally intensive (at least in PoppR). As a result, most SNP sets had to be downsampled to a size of ~3000 SNPs\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe used the same subsamples as for \u003ca href=\"#struct\"\u003eSTRUCTURE\u003c/a\u003e. As a result, the rbarD calculations were averaged over the 10 subsamples.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe used PoppR (Kamvar et al, 2014) to run rbarD. The VCF files were converted into geno files using vcfR (Knaus et al, 2017)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe generalized framework for calculating the index of association (rbarD) in R:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"##########################################################################\n## For full code, see ./ia_analysis.Rmd. Here is an example of how the ###\n## index of association (rbarD) was calculated using Poppr. This example #\n## contains only Subsample 1. ############################################\n##########################################################################\n\n#load required libraries\nlibrary(tidyverse)\nlibrary(RColorBrewer)\nlibrary(ggplot2)\nlibrary(vcfR)\nlibrary(adegenet)\nlibrary(poppr)\nlibrary(magrittr)\n\n##############################################################################\n#function to prep VCF data\nfile_prep \u0026lt;- function(file) {\n  isolate_vcf \u0026lt;- read.vcfR(file = file, verbose = FALSE) #import vcf file (can also be vcf.gz)\n  isolate_genind \u0026lt;- vcfR2genind(isolate_vcf) #converts vcf into genind format\n  isolate_genind$other$lineage \u0026lt;- c(\u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;Q\u0026quot;, \u0026quot;Q\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;J\u0026quot;, \u0026quot;N\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;N\u0026quot;, \u0026quot;M\u0026quot;, \u0026quot;M\u0026quot;, \u0026quot;N\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;K\u0026quot;, \u0026quot;N\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;Q\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;N\u0026quot;, \u0026quot;N\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;J\u0026quot;, \u0026quot;J\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;S\u0026quot;, \u0026quot;S\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;U\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;E\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;Q\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;Q\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;Q\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;I\u0026quot;, \u0026quot;M\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;E\u0026quot;, \u0026quot;H\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;G\u0026quot;, \u0026quot;H\u0026quot;, \u0026quot;F\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;H\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;L\u0026quot;, \u0026quot;L\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;H\u0026quot;, \u0026quot;H\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;K\u0026quot;, \u0026quot;E\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;F\u0026quot;, \u0026quot;G\u0026quot;, \u0026quot;G\u0026quot;, \u0026quot;M\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;Q\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;E\u0026quot;, \u0026quot;T\u0026quot;, \u0026quot;K\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;I\u0026quot;, \u0026quot;U\u0026quot;, \u0026quot;E\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;P\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;SING\u0026quot;, \u0026quot;N\u0026quot;, \u0026quot;R\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;O\u0026quot;, \u0026quot;SING\u0026quot;) #predetermined lineages from STRUCTURE and phylogenetic analysis\n  pop(isolate_genind) \u0026lt;- isolate_genind$other$lineage #assigns lineages as populations\n  isolate_genclone \u0026lt;- as.genclone(isolate_genind) #creates genclone file for poppr analysis\n}\n\n###################################################################\nss1 \u0026lt;- file_prep(\u0026quot;subsample_1.vcf\u0026quot;)\n#etc.\n# All 10 subsamples were loaded into R like this.\n\n# Note\n# (Subsample #) : (# of SNPs)\n# Subsample 1 : 3658\n# Subsample 2 : 3816\n# Subsample 3 : 3642\n# Subsample 4 : 3754\n# Subsample 5 : 3769\n# Subsample 6 : 3671\n# Subsample 7 : 3737\n# Subsample 8 : 3774\n# Subsample 9 : 3745\n# Subsample 10 : 3744\n\n##Same SNP subsamples as used to delimit clonal groups in STRUCUTURE\n\n###################################################################\n# non-clone corrected Ia and Rbar calculations\nset.seed(100)\nresmpl_ia_ss1 \u0026lt;- resample.ia(ss1, reps = 999, n=132)\n#etc.\n\n#combine all subsampled data together\nresmpl_ia_all \u0026lt;- rbind(resmpl_ia_ss1,\n    resmpl_ia_ss2,\n    resmpl_ia_ss3,\n    resmpl_ia_ss4,\n    resmpl_ia_ss5,\n    resmpl_ia_ss6,\n    resmpl_ia_ss7,\n    resmpl_ia_ss8,\n    resmpl_ia_ss9,\n    resmpl_ia_ss10)\n\n  write_delim(resmpl_ia_all, file = \u0026quot;results/all_isolates_ia_vals.txt\u0026quot;, delim = \u0026quot;\\t\u0026quot;)\nresmpl_ia_all \u0026lt;- read_delim(\u0026quot;results/all_isolates_ia_vals.txt\u0026quot;, delim = \u0026quot;\\t\u0026quot;)\nstd(resmpl_ia_all$rbarD)\n\n\nt \u0026lt;- ggplot(resmpl_ia_all, aes(x = rbarD)) +\n  geom_histogram(bins=100) +\n  geom_vline(xintercept = ia(ss1)[2]) +\n#  geom_vline(xintercept = 0.0702, linetype=2) +\n  geom_vline(xintercept = ia(clonecorrect(ss1))[2], linetype = 2) +\n  xlab(expression(bar(r)[d]))\nt\n\n######################################################################\n#separating populations before index of association calculation\nobj1 \u0026lt;- seppop(ss1) #separates the genclone objects into a list of different populations specified earlier\n#etc.\n\n####SS1\nsing_genclone1 \u0026lt;- obj1$SING\ne_genclone1 \u0026lt;- obj1$E\nf_genclone1 \u0026lt;- obj1$F\ng_genclone1 \u0026lt;- obj1$G\nh_genclone1 \u0026lt;- obj1$H\ni_genclone1 \u0026lt;- obj1$I\nk_genclone1 \u0026lt;- obj1$K\nl_genclone1 \u0026lt;- obj1$L\nm_genclone1 \u0026lt;- obj1$M\nn_genclone1 \u0026lt;- obj1$N\no_genclone1 \u0026lt;- obj1$O\np_genclone1 \u0026lt;- obj1$P\nq_genclone1 \u0026lt;- obj1$Q\nr_genclone1 \u0026lt;- obj1$R\ns_genclone1 \u0026lt;- obj1$S\nt_genclone1 \u0026lt;- obj1$T\n\n# calculate Ia and rbarD for subgroup E, subsample 1\nresmpl_ia_ss1_e \u0026lt;- resample.ia(e_genclone1, reps = 999, n=(nrow(as.data.frame(e_genclone1))-1))\n#etc. \n# calculated for all other subsamples (E) too\n#combined all subsamples for subgroup E together\n\nresmpl_ia_e \u0026lt;- rbind(resmpl_ia_ss1_e,\n    resmpl_ia_ss2_e,\n    resmpl_ia_ss3_e,\n    resmpl_ia_ss4_e,\n    resmpl_ia_ss5_e,\n    resmpl_ia_ss6_e,\n    resmpl_ia_ss7_e,\n    resmpl_ia_ss8_e,\n    resmpl_ia_ss9_e,\n    resmpl_ia_ss10_e)\n\n# save data for subgroup E\nwrite_delim(resmpl_ia_e, file = \u0026quot;results/E_isolates_ia_vals.txt\u0026quot;, delim = \u0026quot;\\t\u0026quot;)\n\n\n###### calculated for all other subgroups too #########\n\n# Averages (Mean) and Standard Deviations were calculated in R #\"\u003e\u003cpre lang=\"{r,\" class=\"notranslate\"\u003e\u003ccode\u003e##########################################################################\n## For full code, see ./ia_analysis.Rmd. Here is an example of how the ###\n## index of association (rbarD) was calculated using Poppr. This example #\n## contains only Subsample 1. ############################################\n##########################################################################\n\n#load required libraries\nlibrary(tidyverse)\nlibrary(RColorBrewer)\nlibrary(ggplot2)\nlibrary(vcfR)\nlibrary(adegenet)\nlibrary(poppr)\nlibrary(magrittr)\n\n##############################################################################\n#function to prep VCF data\nfile_prep \u0026lt;- function(file) {\n  isolate_vcf \u0026lt;- read.vcfR(file = file, verbose = FALSE) #import vcf file (can also be vcf.gz)\n  isolate_genind \u0026lt;- vcfR2genind(isolate_vcf) #converts vcf into genind format\n  isolate_genind$other$lineage \u0026lt;- c(\"SING\", \"SING\", \"SING\", \"SING\", \"SING\", \"T\", \"SING\", \"T\", \"O\", \"SING\", \"Q\", \"Q\", \"O\", \"J\", \"N\", \"SING\", \"N\", \"M\", \"M\", \"N\", \"SING\", \"P\", \"SING\", \"K\", \"N\", \"T\", \"Q\", \"SING\", \"P\", \"T\", \"N\", \"N\", \"P\", \"J\", \"J\", \"SING\", \"SING\", \"S\", \"S\", \"SING\", \"O\", \"SING\", \"SING\", \"T\", \"P\", \"T\", \"P\", \"R\", \"SING\", \"SING\", \"SING\", \"O\", \"R\", \"SING\", \"SING\", \"U\", \"SING\", \"SING\", \"E\", \"SING\", \"SING\", \"Q\", \"SING\", \"R\", \"O\", \"Q\", \"R\", \"SING\", \"SING\", \"R\", \"R\", \"R\", \"SING\", \"Q\", \"SING\", \"SING\", \"R\", \"O\", \"I\", \"M\", \"R\", \"E\", \"H\", \"SING\", \"G\", \"H\", \"F\", \"SING\", \"H\", \"SING\", \"L\", \"L\", \"O\", \"H\", \"H\", \"P\", \"R\", \"SING\", \"O\", \"T\", \"K\", \"E\", \"O\", \"SING\", \"O\", \"SING\", \"F\", \"G\", \"G\", \"M\", \"O\", \"Q\", \"P\", \"E\", \"T\", \"K\", \"R\", \"R\", \"SING\", \"O\", \"I\", \"U\", \"E\", \"SING\", \"P\", \"P\", \"R\", \"SING\", \"N\", \"R\", \"O\", \"O\", \"SING\") #predetermined lineages from STRUCTURE and phylogenetic analysis\n  pop(isolate_genind) \u0026lt;- isolate_genind$other$lineage #assigns lineages as populations\n  isolate_genclone \u0026lt;- as.genclone(isolate_genind) #creates genclone file for poppr analysis\n}\n\n###################################################################\nss1 \u0026lt;- file_prep(\"subsample_1.vcf\")\n#etc.\n# All 10 subsamples were loaded into R like this.\n\n# Note\n# (Subsample #) : (# of SNPs)\n# Subsample 1 : 3658\n# Subsample 2 : 3816\n# Subsample 3 : 3642\n# Subsample 4 : 3754\n# Subsample 5 : 3769\n# Subsample 6 : 3671\n# Subsample 7 : 3737\n# Subsample 8 : 3774\n# Subsample 9 : 3745\n# Subsample 10 : 3744\n\n##Same SNP subsamples as used to delimit clonal groups in STRUCUTURE\n\n###################################################################\n# non-clone corrected Ia and Rbar calculations\nset.seed(100)\nresmpl_ia_ss1 \u0026lt;- resample.ia(ss1, reps = 999, n=132)\n#etc.\n\n#combine all subsampled data together\nresmpl_ia_all \u0026lt;- rbind(resmpl_ia_ss1,\n    resmpl_ia_ss2,\n    resmpl_ia_ss3,\n    resmpl_ia_ss4,\n    resmpl_ia_ss5,\n    resmpl_ia_ss6,\n    resmpl_ia_ss7,\n    resmpl_ia_ss8,\n    resmpl_ia_ss9,\n    resmpl_ia_ss10)\n\n  write_delim(resmpl_ia_all, file = \"results/all_isolates_ia_vals.txt\", delim = \"\\t\")\nresmpl_ia_all \u0026lt;- read_delim(\"results/all_isolates_ia_vals.txt\", delim = \"\\t\")\nstd(resmpl_ia_all$rbarD)\n\n\nt \u0026lt;- ggplot(resmpl_ia_all, aes(x = rbarD)) +\n  geom_histogram(bins=100) +\n  geom_vline(xintercept = ia(ss1)[2]) +\n#  geom_vline(xintercept = 0.0702, linetype=2) +\n  geom_vline(xintercept = ia(clonecorrect(ss1))[2], linetype = 2) +\n  xlab(expression(bar(r)[d]))\nt\n\n######################################################################\n#separating populations before index of association calculation\nobj1 \u0026lt;- seppop(ss1) #separates the genclone objects into a list of different populations specified earlier\n#etc.\n\n####SS1\nsing_genclone1 \u0026lt;- obj1$SING\ne_genclone1 \u0026lt;- obj1$E\nf_genclone1 \u0026lt;- obj1$F\ng_genclone1 \u0026lt;- obj1$G\nh_genclone1 \u0026lt;- obj1$H\ni_genclone1 \u0026lt;- obj1$I\nk_genclone1 \u0026lt;- obj1$K\nl_genclone1 \u0026lt;- obj1$L\nm_genclone1 \u0026lt;- obj1$M\nn_genclone1 \u0026lt;- obj1$N\no_genclone1 \u0026lt;- obj1$O\np_genclone1 \u0026lt;- obj1$P\nq_genclone1 \u0026lt;- obj1$Q\nr_genclone1 \u0026lt;- obj1$R\ns_genclone1 \u0026lt;- obj1$S\nt_genclone1 \u0026lt;- obj1$T\n\n# calculate Ia and rbarD for subgroup E, subsample 1\nresmpl_ia_ss1_e \u0026lt;- resample.ia(e_genclone1, reps = 999, n=(nrow(as.data.frame(e_genclone1))-1))\n#etc. \n# calculated for all other subsamples (E) too\n#combined all subsamples for subgroup E together\n\nresmpl_ia_e \u0026lt;- rbind(resmpl_ia_ss1_e,\n    resmpl_ia_ss2_e,\n    resmpl_ia_ss3_e,\n    resmpl_ia_ss4_e,\n    resmpl_ia_ss5_e,\n    resmpl_ia_ss6_e,\n    resmpl_ia_ss7_e,\n    resmpl_ia_ss8_e,\n    resmpl_ia_ss9_e,\n    resmpl_ia_ss10_e)\n\n# save data for subgroup E\nwrite_delim(resmpl_ia_e, file = \"results/E_isolates_ia_vals.txt\", delim = \"\\t\")\n\n\n###### calculated for all other subgroups too #########\n\n# Averages (Mean) and Standard Deviations were calculated in R #\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC. Nucleotide Diversity (pi)\u003c/h2\u003e\u003ca id=\"user-content-c-nucleotide-diversity-pi\" class=\"anchor\" aria-label=\"Permalink: C. Nucleotide Diversity (pi)\" href=\"#c-nucleotide-diversity-pi\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eVCFtools v0.1.15 (Danecek et al, 2011)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eNucleotide diversity (pi) was calculated from VCF files (both \u003ca href=\"#read_mapping\"\u003eread-mapping\u003c/a\u003e using VCFtools (Danecek et al, 2011)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"vcftools --gzvcf $IN.vcf.gz --window-pi  10000 --out $OUT \"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003evcftools --gzvcf $IN.vcf.gz --window-pi  10000 --out $OUT \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eD. Tajima's D\u003c/h2\u003e\u003ca id=\"user-content-d-tajimas-d\" class=\"anchor\" aria-label=\"Permalink: D. Tajima's D\" href=\"#d-tajimas-d\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eVCFtools v0.1.15 (Danecek et al, 2011)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eTajima's D was calculated from VCF files for \u003ca href=\"#read_mapping\"\u003eread-mapping\u003c/a\u003e directly using VCFtools (Danecek et al, 2011). The Tajima's D analysis calculated for the pangenome required additional steps and so is described above under \"\u003ca href=\"#pangenome\"\u003ePangenome\u003c/a\u003e\".\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor the readmapping VCF files, Tajima's D was calculated per SNP. These were then averaged across the genome in R.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"vcftools --gzvcf $IN.vcf.gz --TajimaD  1 --out $OUT \"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003evcftools --gzvcf $IN.vcf.gz --TajimaD  1 --out $OUT \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eE. Genetic Differentiation (Fst) (Weighted)\u003c/h2\u003e\u003ca id=\"user-content-e-genetic-differentiation-fst-weighted\" class=\"anchor\" aria-label=\"Permalink: E. Genetic Differentiation (Fst) (Weighted)\" href=\"#e-genetic-differentiation-fst-weighted\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGenetic differentiation (Fst) is a commonly used proxy to determine gene flow between multiple populations, because more gene flow = lower genetic differentiation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe wanted to understand the levels of genetic differentiation between different clonal subgroups (and therefore estimate the level of gene flow between purely clonal lineages). The idea behind this is to understand whether gene flow is still ongoing even after a transition to clonality. As a result, the Fst of sequence variation (nongenic core SNPs - Figure 6) was compared to presence/absence variation. Hypothesis: If gene flow is ongoing, we expect to see lower genetic differentiation between clonal subgroups that core SNP variation. Furthermore, they should display different demographic histories (i.e. the two Fst sets should be correlated).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo account for the role that genic selection, we included a supplemental figure (Supplemental Figure 3) displaying sequence variation of conserved orthologs. This should serve as a sandwich with the core gene Fst (under highest purifying selection) and core read-mapped Fst (under no selection) serving as the two extremes. If selection is the only reason for lower genetic differentiation (Figure 6), we expect that the accessory gene presence-absence genetic differentiation to be sandwiched between the Fst values in Figure 6A and Supplemental Figure 3A. Since the accessory genome is less differentiated than both core SNP sets, we can deduce that neither selection or genome-wide recombination is the primary mechanism diversifying the accessory genome. Either gene flow is ongoing or ancestral gene loss is driving diversity in the accessory genome in our Ethiopian F. oxysporum collection.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe used custom R scripts to calculate weighted Fst values. The input files for each R script are VCF files (haplotype) or presence/absence matrix (coverage) containing only specified subgroups.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"#sequence variation (core genes and core SNPs)\n# input is any VCF file\n############# calculating_haplotype_diversity.R ##################\nRscript ./calculating_haplotype_diversity.R --populationa $clonal_lineageA.vcf --populationb $clonal_lineageB.vcf \u0026gt; $OUT\n\n\n#presence/absence variation (accessory genes)\n############# calculating_coverage_diversity.R ##################\nRscript ./calculating_coverage_diversity.R --populationa $clonal_lineageA.txt --populationb $clonal_lineageB.txt \u0026gt; $OUT\n\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e#sequence variation (core genes and core SNPs)\n# input is any VCF file\n############# calculating_haplotype_diversity.R ##################\nRscript ./calculating_haplotype_diversity.R --populationa $clonal_lineageA.vcf --populationb $clonal_lineageB.vcf \u0026gt; $OUT\n\n\n#presence/absence variation (accessory genes)\n############# calculating_coverage_diversity.R ##################\nRscript ./calculating_coverage_diversity.R --populationa $clonal_lineageA.txt --populationb $clonal_lineageB.txt \u0026gt; $OUT\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOne can see that this phenomenon is not due to selection exclusively in the boxplot below\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/selection_snps_tajD.png\"\u003e\u003cimg src=\"/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/selection_snps_tajD.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eF. Effector Annotation\u003c/h2\u003e\u003ca id=\"user-content-f-effector-annotation\" class=\"anchor\" aria-label=\"Permalink: F. Effector Annotation\" href=\"#f-effector-annotation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTools required:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eTMHMM v2.0 (Krogh et al, 2001)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSignalP v4.1 (Petersen et al, 2011)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eEffectorP (Sperschneider et al, 2022)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eWe relied on EffectorP and previous GO term annotations (see \u003ca id=\"user-content-gene_annotation\"\u003e\u003c/a\u003eB. Gene Annotation, Part II) to understand the functional annotation and enrichment within the putative effector dataset.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs recommended, SignalP and TMHMM were run on representative pangenome gene cluster sequences, prior to EffectorP. This was ensure that effectors are predicted from the secretome.\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# running SignalP\nfor f in *.fa; do signalp -c 0 -f short -t euk $f \u0026gt; ${f}.signalp; done\n\n# running TMHMM\nfor f in *.fa; do cat $f | cat decodeanhmm.Linux_x86_64 -f /share/apps/tmhmm-2.0c/lib/TMHMM2.0.options -modelfile /share/apps/tmhmm-2.0c/lib/TMHMM2.0.model \u0026gt; ${f}.tmhmm; done\n\n#filtering correct sequences\ncat *.signalp | awk '($0 !~ \u0026quot;SignalP-TM\u0026quot;){print}' \u0026gt; putative_secrete.list\nfor i in $(readline);do cat $i \u0026gt;\u0026gt; putative_secrete.fa; done\nwhile read -r line; do $1=fasta | cat $fasta \u0026gt;\u0026gt; putative_secrete.fa; done \u0026lt; putative_secrete.list\n\n# running effectorP\npython EffectorP.py -i putative_secrete.fasta\n\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e# running SignalP\nfor f in *.fa; do signalp -c 0 -f short -t euk $f \u0026gt; ${f}.signalp; done\n\n# running TMHMM\nfor f in *.fa; do cat $f | cat decodeanhmm.Linux_x86_64 -f /share/apps/tmhmm-2.0c/lib/TMHMM2.0.options -modelfile /share/apps/tmhmm-2.0c/lib/TMHMM2.0.model \u0026gt; ${f}.tmhmm; done\n\n#filtering correct sequences\ncat *.signalp | awk '($0 !~ \"SignalP-TM\"){print}' \u0026gt; putative_secrete.list\nfor i in $(readline);do cat $i \u0026gt;\u0026gt; putative_secrete.fa; done\nwhile read -r line; do $1=fasta | cat $fasta \u0026gt;\u0026gt; putative_secrete.fa; done \u0026lt; putative_secrete.list\n\n# running effectorP\npython EffectorP.py -i putative_secrete.fasta\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003ca id=\"user-content-refs\"\u003e\u003c/a\u003e6. References\u003c/h1\u003e\u003ca id=\"user-content-6-references-1\" class=\"anchor\" aria-label=\"Permalink: 6. References\" href=\"#6-references-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAgapow, P.M. and Burt, A., 2001. Indices of multilocus linkage disequilibrium. Molecular Ecology Notes, 1(1‐2), pp.101-102.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAlmagro Armenteros, J.J., Tsirigos, K.D., Sønderby, C.K., Petersen, T.N., Winther, O., Brunak, S., von Heijne, G. and Nielsen, H., 2019. SignalP 5.0 improves signal peptide predictions using deep neural networks. Nature biotechnology, 37(4), pp.420-423.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBolger, A.M., Lohse, M. and Usadel, B., 2014. Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), pp.2114-2120.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCantarel, B.L., Korf, I., Robb, S.M., Parra, G., Ross, E., Moore, B., Holt, C., Alvarado, A.S. and Yandell, M., 2008. MAKER: an easy-to-use annotation pipeline designed for emerging model organism genomes. Genome research, 18(1), pp.188-196.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCheng, H., Concepcion, G.T., Feng, X., Zhang, H. and Li, H. 2021. Haplotype-resolved de novo assembly using phased assembly graphs with Hifiasm. Nat methods, 18: 170-175.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDanecek, P., Auton, A., Abecasis, G., Albers, C.A., Banks, E., DePristo, M.A., Handsaker, R.E., Lunter, G., Marth, G.T., Sherry, S.T. and McVean, G., 2011. The variant call format and VCFtools. Bioinformatics, 27(15), pp.2156-2158.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDanecek, P., Bonfield, J.K., Liddle, J., Marshall, J., Ohan, V., Pollard, M.O., Whitwham, A., Keane, T., McCarthy, S.A., Davies, R.M. and Li, H., 2021. Twelve years of SAMtools and BCFtools. Gigascience, 10(2), p.giab008.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEarl, D.A. and VonHoldt, B.M., 2012. STRUCTURE HARVESTER: a website and program for visualizing STRUCTURE output and implementing the Evanno method. Conservation genetics resources, 4, pp.359-361.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eEmms, D.M., and Kelly, S. 2019. OrthoFinder: phylogenetic orthology inference for comparative genomics. Genome Biol 20, 238.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFouts, D.E., Brinkac, L., Beck, E., Inman, J. and Sutton, G., 2012. PanOCT: automated clustering of orthologs using conserved gene neighborhood for pan-genomic analysis of bacterial strains and closely related species. Nucleic acids research, 40(22), pp.e172-e172.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGnerre, S., MacCallum, I., Przybylski, D., Ribeiro, F.J., Burton, J.N., Walker, B.J., Sharpe, T., Hall, G., Shea, T.P., Sykes, S. and Berlin, A.M., 2011. High-quality draft assemblies of mammalian genomes from massively parallel sequence data. Proceedings of the National Academy of Sciences, 108(4), pp.1513-1518.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHuerta-Cepas, J., Serra, F. and Bork, P., 2016. ETE 3: reconstruction, analysis, and visualization of phylogenomic data. Molecular biology and evolution, 33(6), pp.1635-1638.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eJakobsson, M. and Rosenberg, N.A., 2007. CLUMPP: a cluster matching and permutation program for dealing with label switching and multimodality in analysis of population structure. Bioinformatics, 23(14), pp.1801-1806.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eJones, P., Binns, D., Chang, H.Y., Fraser, M., Li, W., McAnulla, C., McWilliam, H., Maslen, J., Mitchell, A., Nuka, G. and Pesseat, S., 2014. InterProScan 5: genome-scale protein function classification. Bioinformatics, 30(9), pp.1236-1240.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKamvar, Z.N., Tabima, J.F. and Grünwald, N.J., 2014. Poppr: an R package for genetic analysis of populations with clonal, partially clonal, and/or sexual reproduction. PeerJ, 2, p.e281.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKnaus, B.J. and Grünwald, N.J., 2017. vcfr: a package to manipulate and visualize variant call format data in R. Molecular ecology resources, 17(1), pp.44-53.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eKrogh, A., Larsson, B., Von Heijne, G. and Sonnhammer, E.L., 2001. Predicting transmembrane protein topology with a hidden Markov model: application to complete genomes. Journal of molecular biology, 305(3), pp.567-580.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLi, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., Marth, G., Abecasis, G., Durbin, R. and 1000 Genome Project Data Processing Subgroup, 2009. The sequence alignment/map format and SAMtools. bioinformatics, 25(16), pp.2078-2079.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLi, H., 2013. Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM. arXiv preprint arXiv:1303.3997.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMarçais, G., Delcher, A.L., Phillippy, A.M., Coston, R., Salzberg, S.L. and Zimin, A., 2018. MUMmer4: A fast and versatile genome alignment system. PLoS computational biology, 14(1), p.e1005944.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMcKenna, A., Hanna, M., Banks, E., Sivachenko, A., Cibulskis, K., Kernytsky, A., Garimella, K., Altshuler, D., Gabriel, S., Daly, M. and DePristo, M.A., 2010. The Genome Analysis Toolkit: a MapReduce framework for analyzing next-generation DNA sequencing data. Genome research, 20(9), pp.1297-1303.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePetersen, T.N., Brunak, S., Von Heijne, G. and Nielsen, H., 2011. SignalP 4.0: discriminating signal peptides from transmembrane regions. Nature methods, 8(10), pp.785-786.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePritchard, J.K., Stephens, M. and Donnelly, P., 2000. Inference of population structure using multilocus genotype data. Genetics, 155(2), pp.945-959.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePritchard, L., Glover, R.H., Humphris, S., Elphinstone, J.G. and Toth, I.K., 2016. Genomics and taxonomy in diagnostics for food security: soft-rotting enterobacterial plant pathogens. Analytical Methods, 8(1), pp.12-24.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePurcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M.A., Bender, D., Maller, J., Sklar, P., De Bakker, P.I., Daly, M.J. and Sham, P.C., 2007. PLINK: a tool set for whole-genome association and population-based linkage analyses. The American journal of human genetics, 81(3), pp.559-575.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSimão, F.A., Waterhouse, R.M., Ioannidis, P., Kriventseva, E.V. and Zdobnov, E.M., 2015. BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs. Bioinformatics, 31(19), pp.3210-3212.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSperschneider, J. and Dodds, P.N., 2022. EffectorP 3.0: prediction of apoplastic and cytoplasmic effectors in fungi and oomycetes. Molecular Plant-Microbe Interactions, 35(2), pp.146-156.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSuchard, M.A., Lemey, P., Baele, G., Ayres, D.L., Drummond, A.J. and Rambaut, A., 2018. Bayesian phylogenetic and phylodynamic data integration using BEAST 1.10. Virus evolution, 4(1), p.vey016.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTritt, A., Eisen, J.A., Facciotti, M.T. and Darling, A.E., 2012. An integrated pipeline for de novo assembly of microbial genomes.\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Bioinformatic Pipeline - \"Hiding in Plain Sight - Genome-wide recombination and a dynamic accessory genome drive diversity in Fusarium oxysporum fsp ciceris\"","anchor":"bioinformatic-pipeline---hiding-in-plain-sight---genome-wide-recombination-and-a-dynamic-accessory-genome-drive-diversity-in-fusarium-oxysporum-fsp-ciceris","htmlText":"Bioinformatic Pipeline - \"Hiding in Plain Sight - Genome-wide recombination and a dynamic accessory genome drive diversity in Fusarium oxysporum fsp ciceris\""},{"level":2,"text":"Fayyaz A, Robinson G, Chang PL, Bekele D, Yimer SM, Carrasquilla-Garcia N, Negash K, Anand K. Surendrarao, Eric J.B. von Wettberg, Robbertse B, Seid Ahmedi, Tesfaye K, Fikre Ab, Farmer AD and Cook DR","anchor":"fayyaz-a-robinson-g-chang-pl-bekele-d-yimer-sm-carrasquilla-garcia-n-negash-k-anand-k-surendrarao-eric-jb-von-wettberg-robbertse-b-seid-ahmedi-tesfaye-k-fikre-ab-farmer-ad-and-cook-dr","htmlText":"Fayyaz A, Robinson G, Chang PL, Bekele D, Yimer SM, Carrasquilla-Garcia N, Negash K, Anand K. Surendrarao, Eric J.B. von Wettberg, Robbertse B, Seid Ahmedi, Tesfaye K, Fikre Ab, Farmer AD and Cook DR"},{"level":2,"text":"Contents","anchor":"contents","htmlText":"Contents"},{"level":3,"text":"1. Overview","anchor":"1-overview","htmlText":"1. Overview"},{"level":3,"text":"2. Genome Assembly","anchor":"2-genome-assembly","htmlText":"2. Genome Assembly"},{"level":3,"text":"3. Gene Annotation and Analyses","anchor":"3-gene-annotation-and-analyses","htmlText":"3. Gene Annotation and Analyses"},{"level":3,"text":"4. Identification of SNP and Genetic Groups","anchor":"4-identification-of-snp-and-genetic-groups","htmlText":"4. Identification of SNP and Genetic Groups"},{"level":3,"text":"5. Population Analyses","anchor":"5-population-analyses","htmlText":"5. Population Analyses"},{"level":3,"text":"6. References","anchor":"6-references","htmlText":"6. References"},{"level":1,"text":"1. Overview","anchor":"1-overview-1","htmlText":"1. Overview"},{"level":1,"text":"2. Genome Assembly","anchor":"2-genome-assembly-1","htmlText":"2. Genome Assembly"},{"level":3,"text":"A. Quality check with FastQC","anchor":"a-quality-check-with-fastqc","htmlText":"A. Quality check with FastQC"},{"level":3,"text":"B. Low quality reads and adapters trimmed using Trimmomatic v36","anchor":"b-low-quality-reads-and-adapters-trimmed-using-trimmomatic-v36","htmlText":"B. Low quality reads and adapters trimmed using Trimmomatic v36"},{"level":3,"text":"C. Error correction using ALLPATHS-LG","anchor":"c-error-correction-using-allpaths-lg","htmlText":"C. Error correction using ALLPATHS-LG"},{"level":3,"text":"D. Assembly using A5","anchor":"d-assembly-using-a5","htmlText":"D. Assembly using A5"},{"level":3,"text":"E. Assembly using Hifiasm (PacBio)","anchor":"e-assembly-using-hifiasm-pacbio","htmlText":"E. Assembly using Hifiasm (PacBio)"},{"level":3,"text":"F. Contaminant identification using FCS-GX","anchor":"f-contaminant-identification-using-fcs-gx","htmlText":"F. Contaminant identification using FCS-GX"},{"level":3,"text":"G. Genome completeness assessed using BUSCO v3 (Sordariomyceta odb9 dataset)","anchor":"g-genome-completeness-assessed-using-busco-v3-sordariomyceta-odb9-dataset","htmlText":"G. Genome completeness assessed using BUSCO v3 (Sordariomyceta odb9 dataset)"},{"level":1,"text":"3. Gene Annotation and Analyses","anchor":"3-gene-annotation-and-analyses-1","htmlText":"3. Gene Annotation and Analyses"},{"level":3,"text":"A. Average nucleotide identity calculated with Pyani","anchor":"a-average-nucleotide-identity-calculated-with-pyani","htmlText":"A. Average nucleotide identity calculated with Pyani"},{"level":3,"text":"B. Gene Annotation","anchor":"b-gene-annotation","htmlText":"B. Gene Annotation"},{"level":4,"text":"i. The general gene annotation was done By using MAKER, RepeatModeler, Interproscan, BlastP and AHRD","anchor":"i-the-general-gene-annotation-was-done-by-using-maker-repeatmodeler-interproscan-blastp-and-ahrd","htmlText":"i. The general gene annotation was done By using MAKER, RepeatModeler, Interproscan, BlastP and AHRD"},{"level":4,"text":"ii. Gene Enrichment analysis was performed using BLAST2GO","anchor":"ii-gene-enrichment-analysis-was-performed-using-blast2go","htmlText":"ii. Gene Enrichment analysis was performed using BLAST2GO"},{"level":4,"text":"iii. SIX genes homologs were identified using BLASTp","anchor":"iii-six-genes-homologs-were-identified-using-blastp","htmlText":"iii. SIX genes homologs were identified using BLASTp"},{"level":4,"text":"iv. Orthologs were identified with Orthofinder","anchor":"iv-orthologs-were-identified-with-orthofinder","htmlText":"iv. Orthologs were identified with Orthofinder"},{"level":3,"text":"C. Phylogenetic Analyses","anchor":"c-phylogenetic-analyses","htmlText":"C. Phylogenetic Analyses"},{"level":3,"text":"D. Pangenome","anchor":"d-pangenome","htmlText":"D. Pangenome"},{"level":1,"text":"4. Identification of SNP and genetic groups","anchor":"4-identification-of-snp-and-genetic-groups-1","htmlText":"4. Identification of SNP and genetic groups"},{"level":3,"text":"A. Read mapping and Variant calling","anchor":"a-read-mapping-and-variant-calling","htmlText":"A. Read mapping and Variant calling"},{"level":3,"text":"B. Filtering VCF file","anchor":"b-filtering-vcf-file","htmlText":"B. Filtering VCF file"},{"level":4,"text":"i. We split the VCF into separate single isolate VCF files","anchor":"i-we-split-the-vcf-into-separate-single-isolate-vcf-files","htmlText":"i. We split the VCF into separate single isolate VCF files"},{"level":4,"text":"ii. We indexed all the single isolate VCF files","anchor":"ii-we-indexed-all-the-single-isolate-vcf-files","htmlText":"ii. We indexed all the single isolate VCF files"},{"level":4,"text":"iii. We merged isolates from list of high quality genomes (information obtained from assembly data)","anchor":"iii-we-merged-isolates-from-list-of-high-quality-genomes-information-obtained-from-assembly-data","htmlText":"iii. We merged isolates from list of high quality genomes (information obtained from assembly data)"},{"level":4,"text":"iv. We removed heterozygous/ambiguous SNPs using sed","anchor":"iv-we-removed-heterozygousambiguous-snps-using-sed","htmlText":"iv. We removed heterozygous/ambiguous SNPs using sed"},{"level":4,"text":"v. We removed SNPs with high missingness and evaluated MAF filtering","anchor":"v-we-removed-snps-with-high-missingness-and-evaluated-maf-filtering","htmlText":"v. We removed SNPs with high missingness and evaluated MAF filtering"},{"level":4,"text":"vi. To ensure SNPs were core components of the genome, mapping was informed by additional whole genome alignments (MUMmer)","anchor":"vi-to-ensure-snps-were-core-components-of-the-genome-mapping-was-informed-by-additional-whole-genome-alignments-mummer","htmlText":"vi. To ensure SNPs were core components of the genome, mapping was informed by additional whole genome alignments (MUMmer)"},{"level":3,"text":"C. Identification of clonal groups using STRUCTURE","anchor":"c-identification-of-clonal-groups-using-structure","htmlText":"C. Identification of clonal groups using STRUCTURE"},{"level":4,"text":"i. Random subsampling SNPs","anchor":"i-random-subsampling-snps","htmlText":"i. Random subsampling SNPs"},{"level":4,"text":"ii. STRUCTURE","anchor":"ii-structure","htmlText":"ii. STRUCTURE"},{"level":4,"text":"iii. STRUCTUREharvester","anchor":"iii-structureharvester","htmlText":"iii. STRUCTUREharvester"},{"level":4,"text":"iv. CLUMPP","anchor":"iv-clumpp","htmlText":"iv. CLUMPP"},{"level":4,"text":"v. STRUCTURE plotted with R script","anchor":"v-structure-plotted-with-r-script","htmlText":"v. STRUCTURE plotted with R script"},{"level":4,"text":"vi. Clonal groups were assigned based on Euclidean distances","anchor":"vi-clonal-groups-were-assigned-based-on-euclidean-distances","htmlText":"vi. Clonal groups were assigned based on Euclidean distances"},{"level":1,"text":"5. Population Analyses","anchor":"5-population-analyses-1","htmlText":"5. Population Analyses"},{"level":3,"text":"A. Geographic Analyses","anchor":"a-geographic-analyses","htmlText":"A. Geographic Analyses"},{"level":4,"text":"i. Plotting onto maps in R","anchor":"i-plotting-onto-maps-in-r","htmlText":"i. Plotting onto maps in R"},{"level":4,"text":"ii. Mantel tests","anchor":"ii-mantel-tests","htmlText":"ii. Mantel tests"},{"level":3,"text":"B. Linkage disequilibrium","anchor":"b-linkage-disequilibrium","htmlText":"B. Linkage disequilibrium"},{"level":4,"text":"i. Linkage Disequilibrium via r2","anchor":"i-linkage-disequilibrium-via-r2","htmlText":"i. Linkage Disequilibrium via r2"},{"level":4,"text":"Preparing datasets for \"clonally-derived\" vs \"ancestrally-derived\" SNPs (Figure 4 and Table 1)","anchor":"preparing-datasets-for-clonally-derived-vs-ancestrally-derived-snps-figure-4-and-table-1","htmlText":"Preparing datasets for \"clonally-derived\" vs \"ancestrally-derived\" SNPs (Figure 4 and Table 1)"},{"level":4,"text":"Linkage disequilibrium - r2","anchor":"linkage-disequilibrium---r2","htmlText":"Linkage disequilibrium - r2"},{"level":4,"text":"ii. Linkage disequilibrium - rbarD","anchor":"ii-linkage-disequilibrium---rbard","htmlText":"ii. Linkage disequilibrium - rbarD"},{"level":2,"text":"C. Nucleotide Diversity (pi)","anchor":"c-nucleotide-diversity-pi","htmlText":"C. Nucleotide Diversity (pi)"},{"level":2,"text":"D. Tajima's D","anchor":"d-tajimas-d","htmlText":"D. Tajima's D"},{"level":2,"text":"E. Genetic Differentiation (Fst) (Weighted)","anchor":"e-genetic-differentiation-fst-weighted","htmlText":"E. Genetic Differentiation (Fst) (Weighted)"},{"level":2,"text":"F. Effector Annotation","anchor":"f-effector-annotation","htmlText":"F. Effector Annotation"},{"level":1,"text":"6. References","anchor":"6-references-1","htmlText":"6. References"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fgnrobinson%2FEth_foc_project"}}],"overviewFilesProcessingTime":39.089873999999995,"createFromTemplatePath":"/new?template_name=Eth_foc_project\u0026template_owner=gnrobinson"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/gnrobinson/Eth_foc_project/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/gnrobinson/Eth_foc_project/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/gnrobinson/Eth_foc_project/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/gnrobinson/Eth_foc_project/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/gnrobinson/Eth_foc_project/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           21 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/gnrobinson/Eth_foc_project/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Supplemental_Methods, (Directory)" class="Link--primary" href="/gnrobinson/Eth_foc_project/tree/main/Supplemental_Methods" title="Supplemental_Methods">
                          Supplemental_Methods
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Supplemental_Methods, (Directory)" class="Link--primary" href="/gnrobinson/Eth_foc_project/tree/main/Supplemental_Methods" title="Supplemental_Methods">
                          Supplemental_Methods
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="calculating_coverage_diversity.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/calculating_coverage_diversity.R" title="calculating_coverage_diversity.R">
                          calculating_coverage_diversity.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="calculating_coverage_diversity.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/calculating_coverage_diversity.R" title="calculating_coverage_diversity.R">
                          calculating_coverage_diversity.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="calculating_haplotype_diversity.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/calculating_haplotype_diversity.R" title="calculating_haplotype_diversity.R">
                          calculating_haplotype_diversity.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="calculating_haplotype_diversity.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/calculating_haplotype_diversity.R" title="calculating_haplotype_diversity.R">
                          calculating_haplotype_diversity.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="geo_plot_ethiopia.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/geo_plot_ethiopia.R" title="geo_plot_ethiopia.R">
                          geo_plot_ethiopia.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="geo_plot_ethiopia.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/geo_plot_ethiopia.R" title="geo_plot_ethiopia.R">
                          geo_plot_ethiopia.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="ia_analysis.Rmd, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/ia_analysis.Rmd" title="ia_analysis.Rmd">
                          ia_analysis.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="ia_analysis.Rmd, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/ia_analysis.Rmd" title="ia_analysis.Rmd">
                          ia_analysis.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="ld_decay.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/ld_decay.R" title="ld_decay.R">
                          ld_decay.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="ld_decay.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/ld_decay.R" title="ld_decay.R">
                          ld_decay.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="making_ld_windows.Rmd, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/making_ld_windows.Rmd" title="making_ld_windows.Rmd">
                          making_ld_windows.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="making_ld_windows.Rmd, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/making_ld_windows.Rmd" title="making_ld_windows.Rmd">
                          making_ld_windows.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="making_pi_windows.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/making_pi_windows.R" title="making_pi_windows.R">
                          making_pi_windows.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="making_pi_windows.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/making_pi_windows.R" title="making_pi_windows.R">
                          making_pi_windows.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-9">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="making_tajimaD_windows.Rmd, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/making_tajimaD_windows.Rmd" title="making_tajimaD_windows.Rmd">
                          making_tajimaD_windows.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="making_tajimaD_windows.Rmd, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/making_tajimaD_windows.Rmd" title="making_tajimaD_windows.Rmd">
                          making_tajimaD_windows.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-10">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pangenome_plotting.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/pangenome_plotting.R" title="pangenome_plotting.R">
                          pangenome_plotting.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pangenome_plotting.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/pangenome_plotting.R" title="pangenome_plotting.R">
                          pangenome_plotting.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-11">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="plotting_structure_network.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/plotting_structure_network.R" title="plotting_structure_network.R">
                          plotting_structure_network.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="plotting_structure_network.R, (File)" class="Link--primary" href="/gnrobinson/Eth_foc_project/blob/main/plotting_structure_network.R" title="plotting_structure_network.R">
                          plotting_structure_network.R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd show-for-mobile" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Bioinformatic Pipeline - "Hiding in Plain Sight - Genome-wide recombination and a dynamic accessory genome drive diversity in Fusarium oxysporum fsp ciceris"
                     </h1>
                     <a aria-label='Permalink: Bioinformatic Pipeline - "Hiding in Plain Sight - Genome-wide recombination and a dynamic accessory genome drive diversity in Fusarium oxysporum fsp ciceris"' class="anchor" href="#bioinformatic-pipeline---hiding-in-plain-sight---genome-wide-recombination-and-a-dynamic-accessory-genome-drive-diversity-in-fusarium-oxysporum-fsp-ciceris" id="user-content-bioinformatic-pipeline---hiding-in-plain-sight---genome-wide-recombination-and-a-dynamic-accessory-genome-drive-diversity-in-fusarium-oxysporum-fsp-ciceris">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Fayyaz A, Robinson G, Chang PL, Bekele D, Yimer SM, Carrasquilla-Garcia N, Negash K, Anand K. Surendrarao, Eric J.B. von Wettberg, Robbertse B, Seid Ahmedi, Tesfaye K, Fikre Ab, Farmer AD and Cook DR
                     </h2>
                     <a aria-label="Permalink: Fayyaz A, Robinson G, Chang PL, Bekele D, Yimer SM, Carrasquilla-Garcia N, Negash K, Anand K. Surendrarao, Eric J.B. von Wettberg, Robbertse B, Seid Ahmedi, Tesfaye K, Fikre Ab, Farmer AD and Cook DR" class="anchor" href="#fayyaz-a-robinson-g-chang-pl-bekele-d-yimer-sm-carrasquilla-garcia-n-negash-k-anand-k-surendrarao-eric-jb-von-wettberg-robbertse-b-seid-ahmedi-tesfaye-k-fikre-ab-farmer-ad-and-cook-dr" id="user-content-fayyaz-a-robinson-g-chang-pl-bekele-d-yimer-sm-carrasquilla-garcia-n-negash-k-anand-k-surendrarao-eric-jb-von-wettberg-robbertse-b-seid-ahmedi-tesfaye-k-fikre-ab-farmer-ad-and-cook-dr">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Contents
                     </h2>
                     <a aria-label="Permalink: Contents" class="anchor" href="#contents" id="user-content-contents">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      1.
                      <a href="#Overview">
                       Overview
                      </a>
                     </h3>
                     <a aria-label="Permalink: 1. Overview" class="anchor" href="#1-overview" id="user-content-1-overview">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.
                      <a href="#assembly">
                       Genome Assembly
                      </a>
                     </h3>
                     <a aria-label="Permalink: 2. Genome Assembly" class="anchor" href="#2-genome-assembly" id="user-content-2-genome-assembly">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      3.
                      <a href="#gene_annotation">
                       Gene Annotation and Analyses
                      </a>
                     </h3>
                     <a aria-label="Permalink: 3. Gene Annotation and Analyses" class="anchor" href="#3-gene-annotation-and-analyses" id="user-content-3-gene-annotation-and-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.
                      <a href="#read_mapping">
                       Identification of SNP and Genetic Groups
                      </a>
                     </h3>
                     <a aria-label="Permalink: 4. Identification of SNP and Genetic Groups" class="anchor" href="#4-identification-of-snp-and-genetic-groups" id="user-content-4-identification-of-snp-and-genetic-groups">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      5.
                      <a href="#pop_gen">
                       Population Analyses
                      </a>
                     </h3>
                     <a aria-label="Permalink: 5. Population Analyses" class="anchor" href="#5-population-analyses" id="user-content-5-population-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      6.
                      <a href="#refs">
                       References
                      </a>
                     </h3>
                     <a aria-label="Permalink: 6. References" class="anchor" href="#6-references" id="user-content-6-references">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      <a id="user-content-Overview">
                      </a>
                      1. Overview
                     </h1>
                     <a aria-label="Permalink: 1. Overview" class="anchor" href="#1-overview-1" id="user-content-1-overview-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_pnas_overview.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_pnas_overview.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      <a id="user-content-assembly">
                      </a>
                      2. Genome Assembly
                     </h1>
                     <a aria-label="Permalink: 2. Genome Assembly" class="anchor" href="#2-genome-assembly-1" id="user-content-2-genome-assembly-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       FastQC v0.11.9 (
                       <a href="https://github.com/s-andrews/FastQC">
                        https://github.com/s-andrews/FastQC
                       </a>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Trimmomatic v0.36 (Bolger et al, 2014)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       ALLPATHS-LG v52488 (Gnerre et al, 2010)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       A5 (Tritt et al, 2012)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Hifiasm v16.0 (Cheng et al, 2021)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Singularity v3.5.2
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       FCG-GX (
                       <a href="https://github.com/ncbi/fcs">
                        https://github.com/ncbi/fcs
                       </a>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       BUSCO v3 (Simao et al, 2015)
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      A. Quality check with FastQC
                     </h3>
                     <a aria-label="Permalink: A. Quality check with FastQC" class="anchor" href="#a-quality-check-with-fastqc" id="user-content-a-quality-check-with-fastqc">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     FastQC was used for the initial sequencing QC report.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="module load fastqc
fastqc $IN.fastq.gz">
                     <pre class="notranslate" lang="{bash,"><code>module load fastqc
fastqc $IN.fastq.gz
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      B. Low quality reads and adapters trimmed using Trimmomatic v36
                     </h3>
                     <a aria-label="Permalink: B. Low quality reads and adapters trimmed using Trimmomatic v36" class="anchor" href="#b-low-quality-reads-and-adapters-trimmed-using-trimmomatic-v36" id="user-content-b-low-quality-reads-and-adapters-trimmed-using-trimmomatic-v36">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Reads that were lower than an average phred score &lt; 33 were removed.
                    </p>
                    <p dir="auto">
                     All adapter sequences contained within Illumina's adaptor database were removed.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='mkdir results/trimmomatic/"$SAMPLE"

java -jar bin/Trimmomatic-0.36/trimmomatic-0.36.jar PE -phred33 \
data/atsede-redo-libs/"$SAMPLE"_L006_R1_001.fastq.gz \
data/atsede-redo-libs/"$SAMPLE"_L006_R2_001.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R1.paired.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R1.unpaired.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R2.paired.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R2.unpaired.fastq.gz \
ILLUMINACLIP:bin/Trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
'>
                     <pre class="notranslate" lang="{bash,"><code>mkdir results/trimmomatic/"$SAMPLE"

java -jar bin/Trimmomatic-0.36/trimmomatic-0.36.jar PE -phred33 \
data/atsede-redo-libs/"$SAMPLE"_L006_R1_001.fastq.gz \
data/atsede-redo-libs/"$SAMPLE"_L006_R2_001.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R1.paired.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R1.unpaired.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R2.paired.fastq.gz \
results/trimmomatic/"$SAMPLE"/"$SAMPLE".R2.unpaired.fastq.gz \
ILLUMINACLIP:bin/Trimmomatic-0.36/adapters/NexteraPE-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36

</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      C. Error correction using ALLPATHS-LG
                     </h3>
                     <a aria-label="Permalink: C. Error correction using ALLPATHS-LG" class="anchor" href="#c-error-correction-using-allpaths-lg" id="user-content-c-error-correction-using-allpaths-lg">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To remove erroneous mutations during sequencing, ALLPATHS-LG was used to error correct reads.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#ALLPATHS-LG was run with default parameters
#e.g.
ErrorCorrectReads.pl read_p1.fastq.gz read_p2.fastq.gz PHRED_ENCODING=33 READS_OUT=$OUT">
                     <pre class="notranslate" lang="{bash,"><code>#ALLPATHS-LG was run with default parameters
#e.g.
ErrorCorrectReads.pl read_p1.fastq.gz read_p2.fastq.gz PHRED_ENCODING=33 READS_OUT=$OUT
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      D. Assembly using A5
                     </h3>
                     <a aria-label="Permalink: D. Assembly using A5" class="anchor" href="#d-assembly-using-a5" id="user-content-d-assembly-using-a5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Assembly of all Illumina genomes was conducted with default parameters.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#A5 was run with default parameters
#e.g.
a5_pipeline.pl read_p1.fastq.gz read_p2.fastq.gz mygenome">
                     <pre class="notranslate" lang="{bash,"><code>#A5 was run with default parameters
#e.g.
a5_pipeline.pl read_p1.fastq.gz read_p2.fastq.gz mygenome
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      E. Assembly using Hifiasm (PacBio)
                     </h3>
                     <a aria-label="Permalink: E. Assembly using Hifiasm (PacBio)" class="anchor" href="#e-assembly-using-hifiasm-pacbio" id="user-content-e-assembly-using-hifiasm-pacbio">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Assembly of Fo-Et-0028 PacBio genome
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="module load hifiasm
source activate hifiasm-0.16.1
hifiasm -o fo_et_0028_pacbio.asm --primary -t 32 demultiplex.bc1001--bc1001.hifi_reads.fastq.gz
conda deactivate">
                     <pre class="notranslate" lang="{bash,"><code>module load hifiasm
source activate hifiasm-0.16.1
hifiasm -o fo_et_0028_pacbio.asm --primary -t 32 demultiplex.bc1001--bc1001.hifi_reads.fastq.gz
conda deactivate
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      F. Contaminant identification using FCS-GX
                     </h3>
                     <a aria-label="Permalink: F. Contaminant identification using FCS-GX" class="anchor" href="#f-contaminant-identification-using-fcs-gx" id="user-content-f-contaminant-identification-using-fcs-gx">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To identify and remove contaminants from assemblies, we used NCBI's FCS-GX (available at
                     <a href="https://github.com/ncbi/fcs">
                      https://github.com/ncbi/fcs
                     </a>
                     )
This program screens for contaminants by identifying contigs/scaffolds that do not originate from the biological source organism (as defined by Refseq)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#load singularity
module load singularity
module load python3/3.7.4

#gathering fcs-gx singularity image
curl -LO https://github.com/ncbi/fcs/raw/main/dist/run_fcsgx.py
curl -LO https://github.com/ncbi/fcs/raw/main/examples/fcsgx_test.fa.gz
curl https://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/FCS/releases/0.3.0/fcs-gx.0.3.0.sif -Lo fcsgx.sif

#create temporary shared memory
SHM_LOC=/home/melland/cook/long_read/PB852_4plex_Fusarium_HiFiv3/r64069_20230119_233108/A1/DemuxBarcodes/outputs/assemblies_hifiasm/contaminant_removal
mkdir -p "${SHM_LOC}/gxdb"
mkdir -p ./gxdb
mkdir -p ./gx_out

#verify functionality using small test-only database
#python3 ./run_fcsgx.py --fasta ./fcsgx_test.fa.gz --out-dir ./gx_out/ --gx-db "${SHM_LOC}/gxdb/test-only" --gx-db-disk ./gxdb --split-fasta --tax-id 6973 --container-engine=singularity --image=fcsgx.sif

#do full contamination analysis
#uses manifest file containing list of all fasta files paths
python3 ./run_fcsgx.py --fasta ./manifest_fo.mft --out-dir ./gx_out/ --gx-db "${SHM_LOC}/gxdb/all" --split-fasta --tax-id 5507 --container-engine=singularity --image=fcsgx.sif

# all contaminant contigs/scaffolds were removed unless they contained a bonafide Fusarium oxysporum gene (&gt;95% identity)'>
                     <pre class="notranslate" lang="{bash,"><code>#load singularity
module load singularity
module load python3/3.7.4

#gathering fcs-gx singularity image
curl -LO https://github.com/ncbi/fcs/raw/main/dist/run_fcsgx.py
curl -LO https://github.com/ncbi/fcs/raw/main/examples/fcsgx_test.fa.gz
curl https://ftp.ncbi.nlm.nih.gov/genomes/TOOLS/FCS/releases/0.3.0/fcs-gx.0.3.0.sif -Lo fcsgx.sif

#create temporary shared memory
SHM_LOC=/home/melland/cook/long_read/PB852_4plex_Fusarium_HiFiv3/r64069_20230119_233108/A1/DemuxBarcodes/outputs/assemblies_hifiasm/contaminant_removal
mkdir -p "${SHM_LOC}/gxdb"
mkdir -p ./gxdb
mkdir -p ./gx_out

#verify functionality using small test-only database
#python3 ./run_fcsgx.py --fasta ./fcsgx_test.fa.gz --out-dir ./gx_out/ --gx-db "${SHM_LOC}/gxdb/test-only" --gx-db-disk ./gxdb --split-fasta --tax-id 6973 --container-engine=singularity --image=fcsgx.sif

#do full contamination analysis
#uses manifest file containing list of all fasta files paths
python3 ./run_fcsgx.py --fasta ./manifest_fo.mft --out-dir ./gx_out/ --gx-db "${SHM_LOC}/gxdb/all" --split-fasta --tax-id 5507 --container-engine=singularity --image=fcsgx.sif

# all contaminant contigs/scaffolds were removed unless they contained a bonafide Fusarium oxysporum gene (&gt;95% identity)
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      G. Genome completeness assessed using BUSCO v3 (Sordariomyceta odb9 dataset)
                     </h3>
                     <a aria-label="Permalink: G. Genome completeness assessed using BUSCO v3 (Sordariomyceta odb9 dataset)" class="anchor" href="#g-genome-completeness-assessed-using-busco-v3-sordariomyceta-odb9-dataset" id="user-content-g-genome-completeness-assessed-using-busco-v3-sordariomyceta-odb9-dataset">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The completeness of each genome was assessed using BUSCO. This was used as a filtering step for the pangenome construction, as only genomes with greater than 98% BUSCO complete. The Sordariomyceta ODB9 database was used to specify the lineage and the Fusarium graminearum was used for the gene finding parameters (Fusarium oxysporum is not available).
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## parameters to use for running BUSCO
docker run -it --rm -v $(pwd):/home/working -w /home/working chrishah/busco-docker run_BUSCO.py --in ./Fusarium_oxysporum.fasta --out Fusarium-BUSCO -l ./sordariomyceta_odb9 --mode genome --sp fusarium_graminearum --c 20">
                     <pre class="notranslate" lang="{bash,"><code>## parameters to use for running BUSCO
docker run -it --rm -v $(pwd):/home/working -w /home/working chrishah/busco-docker run_BUSCO.py --in ./Fusarium_oxysporum.fasta --out Fusarium-BUSCO -l ./sordariomyceta_odb9 --mode genome --sp fusarium_graminearum --c 20
</code></pre>
                    </div>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      <a id="user-content-gene_annotation">
                      </a>
                      3. Gene Annotation and Analyses
                     </h1>
                     <a aria-label="Permalink: 3. Gene Annotation and Analyses" class="anchor" href="#3-gene-annotation-and-analyses-1" id="user-content-3-gene-annotation-and-analyses-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       Pyani v0.2.9 (Pritchard et al, 2016)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       RepeatModeler v1.0.11 (
                       <a href="http://www.repeatmasker.org" rel="nofollow">
                        http://www.repeatmasker.org
                       </a>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       MAKER v2.31 (Cantarel et al, 2008)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Interproscan v5.46 (Jones et al, 2014)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       BLASTp v2.7.1
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       AHRD v2.0 (
                       <a href="https://github.com/groupschoof/AHRD">
                        https://github.com/groupschoof/AHRD
                       </a>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       BLAST2GO GUI (
                       <a href="https://www.blast2go.com/" rel="nofollow">
                        https://www.blast2go.com/
                       </a>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Orthofinder v2.5.4 (Emms and Kelly, 2019)
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      A. Average nucleotide identity calculated with Pyani
                     </h3>
                     <a aria-label="Permalink: A. Average nucleotide identity calculated with Pyani" class="anchor" href="#a-average-nucleotide-identity-calculated-with-pyani" id="user-content-a-average-nucleotide-identity-calculated-with-pyani">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/afayyaz/bin/singularity run /home/afayyaz/singularityImages/average_nucleotide_identity-v0.2.9.simg -i all-genomes/ -g -m ANIb --workers 20 -o pyani-results">
                     <pre class="notranslate" lang="{bash,"><code>/home/afayyaz/bin/singularity run /home/afayyaz/singularityImages/average_nucleotide_identity-v0.2.9.simg -i all-genomes/ -g -m ANIb --workers 20 -o pyani-results
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      B. Gene Annotation
                     </h3>
                     <a aria-label="Permalink: B. Gene Annotation" class="anchor" href="#b-gene-annotation" id="user-content-b-gene-annotation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      i. The general gene annotation was done By using MAKER, RepeatModeler, Interproscan, BlastP and AHRD
                     </h4>
                     <a aria-label="Permalink: i. The general gene annotation was done By using MAKER, RepeatModeler, Interproscan, BlastP and AHRD" class="anchor" href="#i-the-general-gene-annotation-was-done-by-using-maker-repeatmodeler-interproscan-blastp-and-ahrd" id="user-content-i-the-general-gene-annotation-was-done-by-using-maker-repeatmodeler-interproscan-blastp-and-ahrd">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     RepeatModeler: We generated the library for custom repeats using RepeatModeler and used these custom library of repeats to feed into MAKER.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## to generate library for custom repeats using RepeatModeler

 $:/home/localhost/adf/sw/bin/singularity shell /erdos/adf/sw/singularityImages/repeatmodeler-1.0.11--pl5.22.0_0.img

./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- BuildDatabase -name fusariumdatabase -engine ncbi Fo-Et-0090.fasta
./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- RepeatModeler -database fusariumdatabase -LTRStruct -pa 20 &gt;&amp; repeats.out

```{bash, eval=F}
##structural annotation using MAKER

i=`ls *.fasta`
for j in $i; do  maker -g $j 2&gt;&amp;1 | tee round1_$j.log ; done
mkdir snap
cd snap
maker2zff -n ../Fo-Et-0090.all.gff
fathom genome.ann genome.dna -categorize 1000
fathom uni.ann uni.dna -export 1000 -plus
forge export.ann export.dna
hmm-assembler.pl Fo-Et-0090 . &gt; Fo-Et-0090.hmm
##used this hmm file to train next round of MAKER for all of the genome
for j in $i; do  maker -g $j 2&gt;&amp;1 | tee round1_$j.log ; done

##functional annotation using iprscan, blastp and AHRD for example using one genome &quot;Fo-Et-0000&quot;

handle_stop_codons.pl --stop_char '\*' Fo-Et-0000.all.maker.proteins.fasta &gt; Fo-Et-0000.all.maker.proteins-stop-remove.fasta
mkdir chunks
fasta2chunks.pl --chunksize=1000 --chunk_prefix=chunks/chunk Fo-Et-0000.all.maker.proteins-stop-remove.fasta
IPRSCAN_HELPERS=/home/afayyaz/chado_preprocessing
JOB_LIMIT=32
export ANALYSIS_DATA_ROOT=`pwd`
mkdir iprscan; pushd iprscan
if [[ `which qsub` != &quot;&quot; ]]; then
for f in ../chunks/*; do echo ${IPRSCAN_HELPERS}/run_iprscan.bash $f | qsub -cwd -pe smp 8; done
else
for f in ../chunks/*; do
while (( `jobs | wc -l` &gt;= $JOB_LIMIT )); do sleep 5; done;
${IPRSCAN_HELPERS}/run_iprscan.bash $f &amp;
done
fi
#then to get iprscan &quot;raw&quot; output for AHRD
${IPRSCAN_HELPERS}/iprscan_convert.bash
mkdir ../iprscan_raw
mv *raw ../iprscan_raw
cd ../
mkdir ipr2go
for f in iprscan_raw/*; do ${IPRSCAN_HELPERS}/ipr2go.pl $f &gt; `echo $f | sed 's/^iprscan_raw/ipr2go/'`; done
forkjobs_blastp_ahrd.pl --forks $JOB_LIMIT chunks blastp

perl -i ahrd_untruncate_blast_query_ids.pl blastp/fol/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/yeast/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/gram/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/vene/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/poae/*
mkdir AHRD
perl -p -e 's/\$\{([^}]+)\}/defined $ENV{$1} ? $ENV{$1} : $&amp;/eg' /home/afayyaz/batcher_input_example_template.yml &gt; AHRD/batcher_input_example.yml
ln -s ${INTERPRO_DB_ROOT}/interpro.dtd AHRD/interpro.dtd
pushd AHRD
mkdir batch_yml
java -cp ${AHRD_GIT_ROOT}/bin/ahrd.jar ahrd.controller.Batcher batcher_input_example.yml
perl -pi -e 's/$/ &amp;/;' batcher.bash
source batcher.bash
for f in *csv; do if [[ ! -s all.ahrd.tsv ]]; then grep '^Protein-Accession' $f &gt; all.ahrd.tsv; fi; sed -n '4,$p' $f &gt;&gt; all.ahrd.tsv; done
${IPRSCAN_HELPERS}/clean_AHRD.sh all.ahrd.tsv | sed 's/-mRNA-1//' &gt; all.ahrd.tsv.cleaned
popd
cat iprscan_raw/* &gt; all.iprscan_raw
awk 'BEGIN {FS=&quot;\t&quot;} NF==9 &amp;&amp; $2 == &quot;maker&quot; {print}' Fo-Et-0000.all.gff &gt; Fo-Et-0000.all.maker.gff
##We added the annotation into gff files
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img ipr_update_gff Fo-Et-0000.all.maker.gff all.iprscan_raw &gt; Fo-Et-0000.all.gff.iprscan_updated
${IPRSCAN_HELPERS}/add_note_attr_inGFF.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.gff.iprscan_updated &gt; Fo-Et-0000.all.gff.iprscan_updated+AHRD
##We generated the gene id's and replace it with existing gene identifiers in the fasta and gff files.
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img maker_map_ids --prefix FO0000- Fo-Et-0000.all.gff &gt; map
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.proteins.fasta 
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.transcripts.fasta
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff.iprscan_updated+AHRD
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_data_ids map AHRD/all.ahrd.tsv.cleaned
cut -d'|' -f1,1 Fo-Et-0000.all.maker.proteins.fasta | sed 's!-RA!!g' &gt;&gt; Fo-Et-0000.all.maker.proteins.final.fasta
cut -d'|' -f1,1 Fo-Et-0000.all.maker.transcripts.fasta | sed 's!-RA!!g' &gt;&gt; Fo-Et-0000.all.maker.transcripts.final.fasta
/home/localhost/adf/ahrdify_fasta.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.maker.proteins.fasta Fo-Et-0000.all.maker.transcripts.fasta
##Repeated the same for rest of the genome">
                     <pre class="notranslate" lang="{bash,"><code>## to generate library for custom repeats using RepeatModeler

 $:/home/localhost/adf/sw/bin/singularity shell /erdos/adf/sw/singularityImages/repeatmodeler-1.0.11--pl5.22.0_0.img

./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- BuildDatabase -name fusariumdatabase -engine ncbi Fo-Et-0090.fasta
./dfam-tetools.sh --singularity --container=dfam/tetools:latest --trf_prgm=/erdos/adf/sw/trf409.linux64 -- RepeatModeler -database fusariumdatabase -LTRStruct -pa 20 &gt;&amp; repeats.out

```{bash, eval=F}
##structural annotation using MAKER

i=`ls *.fasta`
for j in $i; do  maker -g $j 2&gt;&amp;1 | tee round1_$j.log ; done
mkdir snap
cd snap
maker2zff -n ../Fo-Et-0090.all.gff
fathom genome.ann genome.dna -categorize 1000
fathom uni.ann uni.dna -export 1000 -plus
forge export.ann export.dna
hmm-assembler.pl Fo-Et-0090 . &gt; Fo-Et-0090.hmm
##used this hmm file to train next round of MAKER for all of the genome
for j in $i; do  maker -g $j 2&gt;&amp;1 | tee round1_$j.log ; done

##functional annotation using iprscan, blastp and AHRD for example using one genome "Fo-Et-0000"

handle_stop_codons.pl --stop_char '\*' Fo-Et-0000.all.maker.proteins.fasta &gt; Fo-Et-0000.all.maker.proteins-stop-remove.fasta
mkdir chunks
fasta2chunks.pl --chunksize=1000 --chunk_prefix=chunks/chunk Fo-Et-0000.all.maker.proteins-stop-remove.fasta
IPRSCAN_HELPERS=/home/afayyaz/chado_preprocessing
JOB_LIMIT=32
export ANALYSIS_DATA_ROOT=`pwd`
mkdir iprscan; pushd iprscan
if [[ `which qsub` != "" ]]; then
for f in ../chunks/*; do echo ${IPRSCAN_HELPERS}/run_iprscan.bash $f | qsub -cwd -pe smp 8; done
else
for f in ../chunks/*; do
while (( `jobs | wc -l` &gt;= $JOB_LIMIT )); do sleep 5; done;
${IPRSCAN_HELPERS}/run_iprscan.bash $f &amp;
done
fi
#then to get iprscan "raw" output for AHRD
${IPRSCAN_HELPERS}/iprscan_convert.bash
mkdir ../iprscan_raw
mv *raw ../iprscan_raw
cd ../
mkdir ipr2go
for f in iprscan_raw/*; do ${IPRSCAN_HELPERS}/ipr2go.pl $f &gt; `echo $f | sed 's/^iprscan_raw/ipr2go/'`; done
forkjobs_blastp_ahrd.pl --forks $JOB_LIMIT chunks blastp

perl -i ahrd_untruncate_blast_query_ids.pl blastp/fol/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/yeast/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/gram/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/vene/*
perl -i ahrd_untruncate_blast_query_ids.pl blastp/poae/*
mkdir AHRD
perl -p -e 's/\$\{([^}]+)\}/defined $ENV{$1} ? $ENV{$1} : $&amp;/eg' /home/afayyaz/batcher_input_example_template.yml &gt; AHRD/batcher_input_example.yml
ln -s ${INTERPRO_DB_ROOT}/interpro.dtd AHRD/interpro.dtd
pushd AHRD
mkdir batch_yml
java -cp ${AHRD_GIT_ROOT}/bin/ahrd.jar ahrd.controller.Batcher batcher_input_example.yml
perl -pi -e 's/$/ &amp;/;' batcher.bash
source batcher.bash
for f in *csv; do if [[ ! -s all.ahrd.tsv ]]; then grep '^Protein-Accession' $f &gt; all.ahrd.tsv; fi; sed -n '4,$p' $f &gt;&gt; all.ahrd.tsv; done
${IPRSCAN_HELPERS}/clean_AHRD.sh all.ahrd.tsv | sed 's/-mRNA-1//' &gt; all.ahrd.tsv.cleaned
popd
cat iprscan_raw/* &gt; all.iprscan_raw
awk 'BEGIN {FS="\t"} NF==9 &amp;&amp; $2 == "maker" {print}' Fo-Et-0000.all.gff &gt; Fo-Et-0000.all.maker.gff
##We added the annotation into gff files
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img ipr_update_gff Fo-Et-0000.all.maker.gff all.iprscan_raw &gt; Fo-Et-0000.all.gff.iprscan_updated
${IPRSCAN_HELPERS}/add_note_attr_inGFF.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.gff.iprscan_updated &gt; Fo-Et-0000.all.gff.iprscan_updated+AHRD
##We generated the gene id's and replace it with existing gene identifiers in the fasta and gff files.
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img maker_map_ids --prefix FO0000- Fo-Et-0000.all.gff &gt; map
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.proteins.fasta 
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_fasta_ids map Fo-Et-0000.all.maker.transcripts.fasta
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_gff_ids map Fo-Et-0000.all.gff.iprscan_updated+AHRD
singularity exec --bind `pwd`:/pwd --pwd /pwd /erdos/adf/sw/singularityImages/maker-2.31.9--3.img map_data_ids map AHRD/all.ahrd.tsv.cleaned
cut -d'|' -f1,1 Fo-Et-0000.all.maker.proteins.fasta | sed 's!-RA!!g' &gt;&gt; Fo-Et-0000.all.maker.proteins.final.fasta
cut -d'|' -f1,1 Fo-Et-0000.all.maker.transcripts.fasta | sed 's!-RA!!g' &gt;&gt; Fo-Et-0000.all.maker.transcripts.final.fasta
/home/localhost/adf/ahrdify_fasta.pl AHRD/all.ahrd.tsv.cleaned Fo-Et-0000.all.maker.proteins.fasta Fo-Et-0000.all.maker.transcripts.fasta
##Repeated the same for rest of the genome
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      ii. Gene Enrichment analysis was performed using BLAST2GO
                     </h4>
                     <a aria-label="Permalink: ii. Gene Enrichment analysis was performed using BLAST2GO" class="anchor" href="#ii-gene-enrichment-analysis-was-performed-using-blast2go" id="user-content-ii-gene-enrichment-analysis-was-performed-using-blast2go">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Implemented using GUI software (
                     <a href="https://www.blast2go.com/" rel="nofollow">
                      https://www.blast2go.com/
                     </a>
                     )
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      iii. SIX genes homologs were identified using BLASTp
                     </h4>
                     <a aria-label="Permalink: iii. SIX genes homologs were identified using BLASTp" class="anchor" href="#iii-six-genes-homologs-were-identified-using-blastp" id="user-content-iii-six-genes-homologs-were-identified-using-blastp">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We used an E-value cutoff of 1E-6 to ensure that they were true homologs
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      iv. Orthologs were identified with Orthofinder
                     </h4>
                     <a aria-label="Permalink: iv. Orthologs were identified with Orthofinder" class="anchor" href="#iv-orthologs-were-identified-with-orthofinder" id="user-content-iv-orthologs-were-identified-with-orthofinder">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" singularity exec /home/afayyaz/singularity-images/orthofinder_2.5.4.sif orthofinder -t 50 -a 10 -f /home/afayyaz/99-isolate-orth-run/ &gt; ortho_log_file 2&gt;&amp;1 ">
                     <pre class="notranslate" lang="{bash,"><code> singularity exec /home/afayyaz/singularity-images/orthofinder_2.5.4.sif orthofinder -t 50 -a 10 -f /home/afayyaz/99-isolate-orth-run/ &gt; ortho_log_file 2&gt;&amp;1 
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      C. Phylogenetic Analyses
                     </h3>
                     <a aria-label="Permalink: C. Phylogenetic Analyses" class="anchor" href="#c-phylogenetic-analyses" id="user-content-c-phylogenetic-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools Required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       BEAST v1.10.4 (Suchard et al, 2018)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       ETE3 Toolkit (Huerta-Cepas et al, 2016)
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     To understand the evolutionary history of the Ethiopian collection, the species phylogenies were constructed using BEAST. The phylogenies were built from concatenated BUSCO gene alignments.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="##Firstly, We extracted the single copy gene names/list form BUSCO results
perl busco_find_conserved_single_copy_across_all.pl run_*/full_table* &gt; busco_find_conserved_single_copy_across_all.txt
##then we extracted the fasta sequences from the genoem files
for i in `cat busco_find_conserved_single_copy_across_all.txt`; do grep -A1 $i all-busco-single-copy-genes.fna --no-group-separator &gt;&gt; $i.fna; done
##We used clustalo for alignment of the 1556 genes
/home/afayyaz/bin/clustalo -i genes.fna -o genes-aligned.fasta --outfmt=fasta --threads 40
##then we used the aligned file, convert it into nexus file by using seqmagick(python package) and concatenate all of the nexus file into one file for beauti software 
python3.8 seqmagick.py convert --output-format nexus --alphabet dna /home/afayyaz/aligned-genes-fasta/EOG***.fna /home/afayyaz/edit-genes/EOG***.nex
python3.8 cat-nex.py
##generated the xml file in beauti and used in Beast 
java -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml

##used standard parameters for all of the phylogenetic trees and visulaized in iTOL">
                     <pre class="notranslate" lang="{bash,"><code>##Firstly, We extracted the single copy gene names/list form BUSCO results
perl busco_find_conserved_single_copy_across_all.pl run_*/full_table* &gt; busco_find_conserved_single_copy_across_all.txt
##then we extracted the fasta sequences from the genoem files
for i in `cat busco_find_conserved_single_copy_across_all.txt`; do grep -A1 $i all-busco-single-copy-genes.fna --no-group-separator &gt;&gt; $i.fna; done
##We used clustalo for alignment of the 1556 genes
/home/afayyaz/bin/clustalo -i genes.fna -o genes-aligned.fasta --outfmt=fasta --threads 40
##then we used the aligned file, convert it into nexus file by using seqmagick(python package) and concatenate all of the nexus file into one file for beauti software 
python3.8 seqmagick.py convert --output-format nexus --alphabet dna /home/afayyaz/aligned-genes-fasta/EOG***.fna /home/afayyaz/edit-genes/EOG***.nex
python3.8 cat-nex.py
##generated the xml file in beauti and used in Beast 
java -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml

##used standard parameters for all of the phylogenetic trees and visulaized in iTOL
</code></pre>
                    </div>
                    <p dir="auto">
                     Robinson-Foulds (RF) is a metric that measures the distance between different phylogenetic trees. To check the congruence between different trees, we used the normalized Robinson-Foulds (nRF) with ETE3 toolkit. For phylogenetic trees that are highly congruent, the expectation under highly clonal populations, the nRF value would be close to 0, whereas highly incongruent phylogenetic trees would be closer to 1.
                    </p>
                    <p dir="auto">
                     SNPs were called from the concatenated BUSCO alignments and filtered to remove any SNP sites with missing data. Because allele frequency could potentially play a role in the nRF calculations, each SNP site was binned based on allele frequency and normalized accordingly. The nRF was calculated by subsampling these SNPs randomly 10 times with each containing 356,463 SNPs (all SNPs) and 601,905 (ancestral SNPs only)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
##We called SNP's on 1556 BUSCO single copy genes using snp-sites(https://github.com/sanger-pathogens/snp-sites)
for i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done

##We removed all entries with missing data
cat *fasta.vcf &gt; all_genes.fasta.vcf 
$awk '$0 !~ &quot;_&quot; {print}' all_genes.fasta.vcf &gt; all_gene_snp_nomissing.vcf 
cat all_gene_snp_nomissing.vcf | awk -v OFS=&quot;\t&quot; '$0 !~ &quot;^#&quot; {hom_REF = 0; hom_ALT = 0; ; hom_var_2 = 0; hom_var_3 = 0; het = 0; for(i=10;i&lt;=NF;i++) { if($i ~ /0/) hom_REF++; else if($i ~ /1/) hom_ALT++; else if($i ~ /2/) hom_var_2++;else if($i ~ /3/) hom_var_3++; else het++; } print $1, $2,$3, $4, $5, hom_REF, hom_ALT, hom_var_2, hom_var_3, het}' &gt;&gt; maf-column.txt
paste all_gene_snp_nomissing.vcf maf-column.txt &gt; all_gene_snp-maf-file.vcf
">
                     <pre class="notranslate" lang="{bash,"><code>
##We called SNP's on 1556 BUSCO single copy genes using snp-sites(https://github.com/sanger-pathogens/snp-sites)
for i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done

##We removed all entries with missing data
cat *fasta.vcf &gt; all_genes.fasta.vcf 
$awk '$0 !~ "_" {print}' all_genes.fasta.vcf &gt; all_gene_snp_nomissing.vcf 
cat all_gene_snp_nomissing.vcf | awk -v OFS="\t" '$0 !~ "^#" {hom_REF = 0; hom_ALT = 0; ; hom_var_2 = 0; hom_var_3 = 0; het = 0; for(i=10;i&lt;=NF;i++) { if($i ~ /0/) hom_REF++; else if($i ~ /1/) hom_ALT++; else if($i ~ /2/) hom_var_2++;else if($i ~ /3/) hom_var_3++; else het++; } print $1, $2,$3, $4, $5, hom_REF, hom_ALT, hom_var_2, hom_var_3, het}' &gt;&gt; maf-column.txt
paste all_gene_snp_nomissing.vcf maf-column.txt &gt; all_gene_snp-maf-file.vcf

</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="##R code for selecting SNP's
library(dplyr)
file &lt;- read.delim(&quot;all_gene_snp-maf-file.vcf.txt&quot;)
mydata &lt;- data.frame(file)
data &lt;- arrange(mydata, desc(maf))
subset1 &lt;- subset(data, maf &lt; 0.1 )
subset2 &lt;- subset(data, maf &gt;= 0.1 &amp; maf &lt; 0.2)
subset3 &lt;- subset(data, maf &gt;= 0.2 &amp; maf &lt; 0.3)
subset4 &lt;- subset(data, maf &gt;= 0.3 &amp; maf &lt; 0.4)
subset5 &lt;- subset(data, maf &gt; 0.4)
###Total number of SNP's are 706149
###No of SNP's in subset1 -&gt; 299273
###No of SNP's in subset2 -&gt; 402709
###No of SNP's in subset3 -&gt; 2103
###No of SNP's in subset4 -&gt; 1052
###No of SNP's in subset5 -&gt; 1012
##We randomly selected snps from subsets based on the percentage representation in the total amount

x &lt;- slice_sample(subset1, n=126831)
y &lt;- slice_sample(subset2, n=229624) 
z &lt;- slice_sample(subset3, n=6)
a &lt;- slice_sample(subset4, n=1)
b &lt;- slice_sample(subset5, n=1)

#randomly selecting snps from subsets based on the percentage representation in the total amount
write.table(x, file = &quot;x-export-file.txt&quot;, sep = &quot;\t&quot; )
write.table(y, file = &quot;y-export-file.txt&quot;, sep = &quot;\t&quot; )
write.table(z, file = &quot;z-export-file.txt&quot;, sep = &quot;\t&quot; )
write.table(a, file = &quot;a-export-file.txt&quot;, sep = &quot;\t&quot; )
write.table(b, file = &quot;b-export-file.txt&quot;, sep = &quot;\t&quot; )
#concatente in bash and add header of the original vcf file.">
                     <pre class="notranslate" lang="{r,"><code>##R code for selecting SNP's
library(dplyr)
file &lt;- read.delim("all_gene_snp-maf-file.vcf.txt")
mydata &lt;- data.frame(file)
data &lt;- arrange(mydata, desc(maf))
subset1 &lt;- subset(data, maf &lt; 0.1 )
subset2 &lt;- subset(data, maf &gt;= 0.1 &amp; maf &lt; 0.2)
subset3 &lt;- subset(data, maf &gt;= 0.2 &amp; maf &lt; 0.3)
subset4 &lt;- subset(data, maf &gt;= 0.3 &amp; maf &lt; 0.4)
subset5 &lt;- subset(data, maf &gt; 0.4)
###Total number of SNP's are 706149
###No of SNP's in subset1 -&gt; 299273
###No of SNP's in subset2 -&gt; 402709
###No of SNP's in subset3 -&gt; 2103
###No of SNP's in subset4 -&gt; 1052
###No of SNP's in subset5 -&gt; 1012
##We randomly selected snps from subsets based on the percentage representation in the total amount

x &lt;- slice_sample(subset1, n=126831)
y &lt;- slice_sample(subset2, n=229624) 
z &lt;- slice_sample(subset3, n=6)
a &lt;- slice_sample(subset4, n=1)
b &lt;- slice_sample(subset5, n=1)

#randomly selecting snps from subsets based on the percentage representation in the total amount
write.table(x, file = "x-export-file.txt", sep = "\t" )
write.table(y, file = "y-export-file.txt", sep = "\t" )
write.table(z, file = "z-export-file.txt", sep = "\t" )
write.table(a, file = "a-export-file.txt", sep = "\t" )
write.table(b, file = "b-export-file.txt", sep = "\t" )
#concatente in bash and add header of the original vcf file.
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#To filter for ancestral SNP's on a per clade basis
for F in *.vcf.gz; do bgzip -dc $F | //
awk '{for(i=11;i&lt;=NF;i++)if($i!=$(i-1)&amp;&amp;$1!~/#/)next}1' | //
bgzip -c &gt; ${F}_filtered.vcf.gz; done 

#The filtered VCF files for all clades were then combined to calculate the ancestral SNP Robinson-Foulds values

###Total number of SNP's are 613071
###No of SNP's in subset1 -&gt; 607962
###No of SNP's in subset2 -&gt; 3486
###No of SNP's in subset3 -&gt; 795
###No of SNP's in subset4 -&gt; 455
###No of SNP's in subset5 -&gt; 329">
                     <pre class="notranslate" lang="{bash,"><code>#To filter for ancestral SNP's on a per clade basis
for F in *.vcf.gz; do bgzip -dc $F | //
awk '{for(i=11;i&lt;=NF;i++)if($i!=$(i-1)&amp;&amp;$1!~/#/)next}1' | //
bgzip -c &gt; ${F}_filtered.vcf.gz; done 

#The filtered VCF files for all clades were then combined to calculate the ancestral SNP Robinson-Foulds values

###Total number of SNP's are 613071
###No of SNP's in subset1 -&gt; 607962
###No of SNP's in subset2 -&gt; 3486
###No of SNP's in subset3 -&gt; 795
###No of SNP's in subset4 -&gt; 455
###No of SNP's in subset5 -&gt; 329
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='x &lt;- slice_sample(subset1, n=601882)
y &lt;- slice_sample(subset2, n=20) 
z &lt;- slice_sample(subset3, n=1)
a &lt;- slice_sample(subset4, n=1)
b &lt;- slice_sample(subset5, n=1)
write.table(x, file = "x-export-file.txt", sep = "\t" )
write.table(y, file = "y-export-file.txt", sep = "\t" )
write.table(z, file = "z-export-file.txt", sep = "\t" )
write.table(a, file = "a-export-file.txt", sep = "\t" )
write.table(b, file = "b-export-file.txt", sep = "\t" )'>
                     <pre class="notranslate" lang="{r,"><code>x &lt;- slice_sample(subset1, n=601882)
y &lt;- slice_sample(subset2, n=20) 
z &lt;- slice_sample(subset3, n=1)
a &lt;- slice_sample(subset4, n=1)
b &lt;- slice_sample(subset5, n=1)
write.table(x, file = "x-export-file.txt", sep = "\t" )
write.table(y, file = "y-export-file.txt", sep = "\t" )
write.table(z, file = "z-export-file.txt", sep = "\t" )
write.table(a, file = "a-export-file.txt", sep = "\t" )
write.table(b, file = "b-export-file.txt", sep = "\t" )
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#concatente in bash and add header of the original vcf file.

##converted vcf format to fasta format and used 
cat subset.vcf | perl vcf_to_fasta.pl &gt; vcftofasta.fasta
##generated the xml file in beauti and used in Beast 
java -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml
##generated 10 different trees newick files labeld as "1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt"
ete3 compare -taboutput -t 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt -r 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt &gt; result.txt'>
                     <pre class="notranslate" lang="{bash,"><code>#concatente in bash and add header of the original vcf file.

##converted vcf format to fasta format and used 
cat subset.vcf | perl vcf_to_fasta.pl &gt; vcftofasta.fasta
##generated the xml file in beauti and used in Beast 
java -jar /home/afayyaz/bin/BEASTv1.10.4/lib/beast.jar -threads 20 gene_alignment.xml
##generated 10 different trees newick files labeld as "1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt"
ete3 compare -taboutput -t 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt -r 1.txt 2.txt 3.txt 4.txt 5.txt 6.txt 7.txt 8.txt 9.txt 10.txt &gt; result.txt
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      D.
                      <a id="user-content-pangenome">
                      </a>
                      Pangenome
                     </h3>
                     <a aria-label="Permalink: D. Pangenome" class="anchor" href="#d-pangenome" id="user-content-d-pangenome">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       PanOCT v3.23 (Fouts et al, 2012)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       VCFtools v0.1.15 (Danecek et al, 2011)
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     Pangenome was constructed using PanOCT software. Genes predicted by MAKER were clustered together if the percentage identity was &gt;80% and the genes were genomically syntenic.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="singularity exec /home/afayyaz/singularity-images/panoct_3.23--pl526_1.sif panoct.pl -t blast-results.txt -g final-e-attribute.txt -P 99-all-protein.pep -f list-e.txt -b /home/afayyaz/group-e-panoct/ -i 95 -L 20 -M Y -F 1.33 -N Y -H Y -V Y -S Y -C Y -B Y -c 0,5,25,50,75,90,95,98,100 &gt; panoct-test 2&gt;&amp;1">
                     <pre class="notranslate" lang="{bash,"><code>singularity exec /home/afayyaz/singularity-images/panoct_3.23--pl526_1.sif panoct.pl -t blast-results.txt -g final-e-attribute.txt -P 99-all-protein.pep -f list-e.txt -b /home/afayyaz/group-e-panoct/ -i 95 -L 20 -M Y -F 1.33 -N Y -H Y -V Y -S Y -C Y -B Y -c 0,5,25,50,75,90,95,98,100 &gt; panoct-test 2&gt;&amp;1
</code></pre>
                    </div>
                    <p dir="auto">
                     Before we calculated the nucleotide diversity for the pangenome, we removed the paralogs from the gene presence absence file
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="###Removed paralogs from panoct result file
Python3 script-paralogs.py
###To calculate nucleotide diversity for pan-genome, we needed the files of each cluster in the pangenome containing number of genomes from 5 to 99
awk -F&quot;\t&quot; '{print &gt;&gt; ($101&quot;.txt&quot;)}' without-paralogs.txt  
###Then we made a list of the cluster in the pangenome
for i in `ls cluster*`; do sed $'s/\t/\\\n/g' $i| sed '/^$/d' | sed '$d' | sed '$d' &gt; $i.txt; done
###We extracted the cluster sequence from each genome
for i in `ls cluster*.txt`; do python3 extract_seq.py final-99-isolate-transcript.fasta $i $i.fasta; done
###We aligned the sequence using mafft
for i in `ls *.fasta`; do mafft --thread 20 $i &gt; $i.aligned.fasta; done
###We called SNP's using python package snp-sites(https://github.com/sanger-pathogens/snp-sites)
for i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done
for i in `ls *.vcf`; do sed 's/*/_/g' $i | awk '$0 !~ &quot;_&quot; {print}' | awk '{for(i=1;i&lt;=NF;i++){if($i==0){$i=&quot;0/0&quot;}}}{$1=$1} 1' OFS=&quot;\t&quot; | awk '{for(i=10;i&lt;=NF;i++){if($i==1){$i=&quot;1/1&quot;}}}{$1=$1} 1' OFS=&quot;\t&quot;| awk '{for(i=10;i&lt;=NF;i++){if($i==2){$i=&quot;2/2&quot;}}}{$1=$1} 1' OFS=&quot;\t&quot; &gt; $i-p.vcf; done
###We calculated nucleotide diversity using vcftools
for i in `ls *-p.vcf`; do vcftools --vcf $i --site-pi --out $i-nucl_diversity; done
for i in `ls *.pi`; do sed '1d' $i &gt; $i.txt ; done
##We took the average of each file
ls *.txt &gt; file_list
while read line; do awk '{sum+=$2} END {print sum/NR}' $line &gt;&gt; avg-pi-file; done&lt;list-avg-pi.txt
mv file_new avg-values
cp *-aligned-files/avg* .
for i in `ls avg-*`; do awk '{print FILENAME (NF?&quot;\t&quot;:&quot;&quot;) $0}' $i &gt; $i.txt; done
cat *.txt &gt; all-cluster-pi.txt
###We calculated tajimas-d using vcftools
for i in `ls *-p.vcf`; do vcftools --vcf $i --TajimaD 100000000 --out $i-tajimad.txt; done
for i in `ls *.Tajima.D.txt`; do sed '1d' $i&gt; $i.txt; done
##We took the average of each file
while read line; do awk '{sum+=$5} END {print sum/NR}' $line &gt;&gt; avg-tajima-file; done&lt;list-avg-tajima.txt
mv file_new avg-values
cp *-aligned-files/avg* .
for i in `ls avg-*`; do awk '{print FILENAME (NF?&quot;\t&quot;:&quot;&quot;) $0}' $i &gt; $i.txt; done
cat *.D.txt &gt; tajima.d.txt
##repeat these steps for all clusters containing number of genomes from 5 to 99">
                     <pre class="notranslate" lang="{bash,"><code>###Removed paralogs from panoct result file
Python3 script-paralogs.py
###To calculate nucleotide diversity for pan-genome, we needed the files of each cluster in the pangenome containing number of genomes from 5 to 99
awk -F"\t" '{print &gt;&gt; ($101".txt")}' without-paralogs.txt  
###Then we made a list of the cluster in the pangenome
for i in `ls cluster*`; do sed $'s/\t/\\\n/g' $i| sed '/^$/d' | sed '$d' | sed '$d' &gt; $i.txt; done
###We extracted the cluster sequence from each genome
for i in `ls cluster*.txt`; do python3 extract_seq.py final-99-isolate-transcript.fasta $i $i.fasta; done
###We aligned the sequence using mafft
for i in `ls *.fasta`; do mafft --thread 20 $i &gt; $i.aligned.fasta; done
###We called SNP's using python package snp-sites(https://github.com/sanger-pathogens/snp-sites)
for i in `ls *.fasta`; do snp-sites -rmv -o $i $i; done
for i in `ls *.vcf`; do sed 's/*/_/g' $i | awk '$0 !~ "_" {print}' | awk '{for(i=1;i&lt;=NF;i++){if($i==0){$i="0/0"}}}{$1=$1} 1' OFS="\t" | awk '{for(i=10;i&lt;=NF;i++){if($i==1){$i="1/1"}}}{$1=$1} 1' OFS="\t"| awk '{for(i=10;i&lt;=NF;i++){if($i==2){$i="2/2"}}}{$1=$1} 1' OFS="\t" &gt; $i-p.vcf; done
###We calculated nucleotide diversity using vcftools
for i in `ls *-p.vcf`; do vcftools --vcf $i --site-pi --out $i-nucl_diversity; done
for i in `ls *.pi`; do sed '1d' $i &gt; $i.txt ; done
##We took the average of each file
ls *.txt &gt; file_list
while read line; do awk '{sum+=$2} END {print sum/NR}' $line &gt;&gt; avg-pi-file; done&lt;list-avg-pi.txt
mv file_new avg-values
cp *-aligned-files/avg* .
for i in `ls avg-*`; do awk '{print FILENAME (NF?"\t":"") $0}' $i &gt; $i.txt; done
cat *.txt &gt; all-cluster-pi.txt
###We calculated tajimas-d using vcftools
for i in `ls *-p.vcf`; do vcftools --vcf $i --TajimaD 100000000 --out $i-tajimad.txt; done
for i in `ls *.Tajima.D.txt`; do sed '1d' $i&gt; $i.txt; done
##We took the average of each file
while read line; do awk '{sum+=$5} END {print sum/NR}' $line &gt;&gt; avg-tajima-file; done&lt;list-avg-tajima.txt
mv file_new avg-values
cp *-aligned-files/avg* .
for i in `ls avg-*`; do awk '{print FILENAME (NF?"\t":"") $0}' $i &gt; $i.txt; done
cat *.D.txt &gt; tajima.d.txt
##repeat these steps for all clusters containing number of genomes from 5 to 99
</code></pre>
                    </div>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      <a id="user-content-read_mapping">
                      </a>
                      4. Identification of SNP and genetic groups
                     </h1>
                     <a aria-label="Permalink: 4. Identification of SNP and genetic groups" class="anchor" href="#4-identification-of-snp-and-genetic-groups-1" id="user-content-4-identification-of-snp-and-genetic-groups-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      A. Read mapping and Variant calling
                     </h3>
                     <a aria-label="Permalink: A. Read mapping and Variant calling" class="anchor" href="#a-read-mapping-and-variant-calling" id="user-content-a-read-mapping-and-variant-calling">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       BWA MEM v0.7.9a-r786 (Li, 2013)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       SAMtools v1.3.1 (Li et al, 2009)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       VCFtools v0.1.15 (Danecek et al, 2011)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       BCFtools v1.16 (Danecek et al, 2021)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Picard (
                       <a href="http://broadinstitute.github.io/picard/" rel="nofollow">
                        http://broadinstitute.github.io/picard/
                       </a>
                       )
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       GATK v4.1 (McKenna et al, 2010)
                      </p>
                     </li>
                    </ul>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#create index for F. oxysporum reference genome (Fol4287 - GCF_000149955.1)
bwa index [-a bwtsw|is] reference.fasta index_prefix
samtools faidx reference.fasta
##created sequence dictionary
java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar CreateSequenceDictionary REFERENCE=reference.fasta OUTPUT=reference.dict
##lign reads and assign read group
for i in $(ls *.fastq.gz | cut -d '_' -f1,2,3| uniq); do bwa mem -t 60 -R &quot;@RG\tID:${i}\tSM:${i}\tPL:illumina\tPU:Lane1\tLB:${i}&quot; reference.fasta ${i}_R1_pair_001.fastq.gz ${i}_R2_pair_001.fastq.gz &gt; ${i}.sam; done
##sorting sam files
for i in `ls *.sam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar SortSam I=$i O=$i.bam SORT_ORDER=coordinate; done
##mark duplicates
for i in `ls *.bam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar MarkDuplicates I=$i O=$i-edit.bam REMOVE_DUPLICATES=true METRICS_FILE=$i-metrics.txt; done
##sort bam file
for i in `ls *-edit.bam`; do java -jar /erdos/cook_lab/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar BuildBamIndex INPUT=$i ; done
##called variants
for i in `ls *-edit.bam`; do gatk HaplotypeCaller -R reference.fasta -I $i -ploidy 1 --native-pair-hmm-threads 40 -ERC GVCF -O $i-raw_gVCF.vcf --spark-runner LOCAL; done
for i in `ls *-raw_gVCF.vcf`; do gatk GenotypeGVCFs -O $i-raw.vcf -R reference.fasta --spark-runner LOCAL -V $i; done
for i in `ls *-raw.vcf`; do bgzip -c $i &gt; $i.gz; done
for i in `ls *-raw.vcf.gz`; do tabix -p vcf $i ; done
bcftools merge *vcf.gz -o a-merged.vcf
##extract SNP's and stats
gatk SelectVariants -R reference.fasta -V a-merged.vcf -select-type SNP -O c-snps.vcf
bcftools stats c-snps.vcf &gt; stats-snps.txt
vcftools --vcf c-snps.vcf --out c-filtered --minDP 10 --max-missing 0.95 --minQ 30 --recode --recode-INFO-all">
                     <pre class="notranslate" lang="{bash,"><code>#create index for F. oxysporum reference genome (Fol4287 - GCF_000149955.1)
bwa index [-a bwtsw|is] reference.fasta index_prefix
samtools faidx reference.fasta
##created sequence dictionary
java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar CreateSequenceDictionary REFERENCE=reference.fasta OUTPUT=reference.dict
##lign reads and assign read group
for i in $(ls *.fastq.gz | cut -d '_' -f1,2,3| uniq); do bwa mem -t 60 -R "@RG\tID:${i}\tSM:${i}\tPL:illumina\tPU:Lane1\tLB:${i}" reference.fasta ${i}_R1_pair_001.fastq.gz ${i}_R2_pair_001.fastq.gz &gt; ${i}.sam; done
##sorting sam files
for i in `ls *.sam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar SortSam I=$i O=$i.bam SORT_ORDER=coordinate; done
##mark duplicates
for i in `ls *.bam`; do java -jar /home/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar MarkDuplicates I=$i O=$i-edit.bam REMOVE_DUPLICATES=true METRICS_FILE=$i-metrics.txt; done
##sort bam file
for i in `ls *-edit.bam`; do java -jar /erdos/cook_lab/afayyaz/software/picard/build/libs/picard-2.25.7-SNAPSHOT-all.jar BuildBamIndex INPUT=$i ; done
##called variants
for i in `ls *-edit.bam`; do gatk HaplotypeCaller -R reference.fasta -I $i -ploidy 1 --native-pair-hmm-threads 40 -ERC GVCF -O $i-raw_gVCF.vcf --spark-runner LOCAL; done
for i in `ls *-raw_gVCF.vcf`; do gatk GenotypeGVCFs -O $i-raw.vcf -R reference.fasta --spark-runner LOCAL -V $i; done
for i in `ls *-raw.vcf`; do bgzip -c $i &gt; $i.gz; done
for i in `ls *-raw.vcf.gz`; do tabix -p vcf $i ; done
bcftools merge *vcf.gz -o a-merged.vcf
##extract SNP's and stats
gatk SelectVariants -R reference.fasta -V a-merged.vcf -select-type SNP -O c-snps.vcf
bcftools stats c-snps.vcf &gt; stats-snps.txt
vcftools --vcf c-snps.vcf --out c-filtered --minDP 10 --max-missing 0.95 --minQ 30 --recode --recode-INFO-all
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      B. Filtering VCF file
                     </h3>
                     <a aria-label="Permalink: B. Filtering VCF file" class="anchor" href="#b-filtering-vcf-file" id="user-content-b-filtering-vcf-file">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       BCFtools v1.16 (Danecek et al, 2021)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       VCFtools v0.1.15 (Danecek et al, 2011)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       MUMmer v3.23 (Marcais et al, 2018)
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     For most population analyses, we exclusively used high quality genomes (based on BUSCO completeness &gt; 98%, N50 &gt; 1,200 and genome size) - see "Genome Assembly". This provided a total list of 120 genomes, 99 of which were taxonomically within the FOSC (Fig. 2). This is reduced the total number of isolates (320) (Supplemental Fig. 1).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      i. We split the VCF into separate single isolate VCF files
                     </h4>
                     <a aria-label="Permalink: i. We split the VCF into separate single isolate VCF files" class="anchor" href="#i-we-split-the-vcf-into-separate-single-isolate-vcf-files" id="user-content-i-we-split-the-vcf-into-separate-single-isolate-vcf-files">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="######## ./splitting_vcfs_retaining.sh ###########

#!/usr/bin/env bash
for file in *.vcf.gz; do
  for sample in `bcftools query -l $file`; do
    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file
  done
done">
                     <pre class="notranslate" lang="{bash,"><code>######## ./splitting_vcfs_retaining.sh ###########

#!/usr/bin/env bash
for file in *.vcf.gz; do
  for sample in `bcftools query -l $file`; do
    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file
  done
done
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      ii. We indexed all the single isolate VCF files
                     </h4>
                     <a aria-label="Permalink: ii. We indexed all the single isolate VCF files" class="anchor" href="#ii-we-indexed-all-the-single-isolate-vcf-files" id="user-content-ii-we-indexed-all-the-single-isolate-vcf-files">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for f in *vcf.gz; do tabix -f -p vcf $f; done">
                     <pre class="notranslate" lang="{bash,"><code>for f in *vcf.gz; do tabix -f -p vcf $f; done
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      iii. We merged isolates from list of high quality genomes (information obtained from assembly data)
                     </h4>
                     <a aria-label="Permalink: iii. We merged isolates from list of high quality genomes (information obtained from assembly data)" class="anchor" href="#iii-we-merged-isolates-from-list-of-high-quality-genomes-information-obtained-from-assembly-data" id="user-content-iii-we-merged-isolates-from-list-of-high-quality-genomes-information-obtained-from-assembly-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bcftools merge --file-list isolate_list &gt; ethiopia_fo_structure.vcf">
                     <pre class="notranslate" lang="{bash,"><code>bcftools merge --file-list isolate_list &gt; ethiopia_fo_structure.vcf
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      iv. We removed heterozygous/ambiguous SNPs using sed
                     </h4>
                     <a aria-label="Permalink: iv. We removed heterozygous/ambiguous SNPs using sed" class="anchor" href="#iv-we-removed-heterozygousambiguous-snps-using-sed" id="user-content-iv-we-removed-heterozygousambiguous-snps-using-sed">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sed -i 's#0/1|0/2|0/3|0/4|1/2|1/3|1/4|2/3|2/4|3/4#[.]/[.]#g' ethiopia_fo_structure.vcf">
                     <pre class="notranslate" lang="{bash,"><code>sed -i 's#0/1|0/2|0/3|0/4|1/2|1/3|1/4|2/3|2/4|3/4#[.]/[.]#g' ethiopia_fo_structure.vcf
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      v. We removed SNPs with high missingness and evaluated MAF filtering
                     </h4>
                     <a aria-label="Permalink: v. We removed SNPs with high missingness and evaluated MAF filtering" class="anchor" href="#v-we-removed-snps-with-high-missingness-and-evaluated-maf-filtering" id="user-content-v-we-removed-snps-with-high-missingness-and-evaluated-maf-filtering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To ensure that the subsequent VCF was high quality, especially for linkage disequilibrium, we adjusted filters to learn how different filters affect the subsequent linkage disequilibrium analyses.
                    </p>
                    <p dir="auto">
                     The main two parameters were missingness and minor allele frequency (MAF). MAF was of particular importance to us, as often researchers adjust MAF cutoff as &gt;0.05 without regard for the effects on the population.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#VCF file for population genetic analyses
vcftools --vcf ethiopia_fo_structure.vcf --max-missing 0.9 --recode --stdout | bgzip -c &gt; all_ethiopia_isolates.vcf.gz #total number of SNPs = 136772">
                     <pre class="notranslate" lang="{bash,"><code>#VCF file for population genetic analyses
vcftools --vcf ethiopia_fo_structure.vcf --max-missing 0.9 --recode --stdout | bgzip -c &gt; all_ethiopia_isolates.vcf.gz #total number of SNPs = 136772
</code></pre>
                    </div>
                    <p dir="auto">
                     This filtering step was evaluated to understand the effects of using MAF cutoffs (no MAF filter vs "typical" MAF &gt;0.05). There were only minor differences in the outcomes. The clonal groupings remained the same. The LD decay curve only showed minor differences (see below)
                    </p>
                    <p dir="auto">
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/ld_total_maf.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/ld_total_maf.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      vi. To ensure SNPs were core components of the genome, mapping was informed by additional whole genome alignments (MUMmer)
                     </h4>
                     <a aria-label="Permalink: vi. To ensure SNPs were core components of the genome, mapping was informed by additional whole genome alignments (MUMmer)" class="anchor" href="#vi-to-ensure-snps-were-core-components-of-the-genome-mapping-was-informed-by-additional-whole-genome-alignments-mummer" id="user-content-vi-to-ensure-snps-were-core-components-of-the-genome-mapping-was-informed-by-additional-whole-genome-alignments-mummer">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     This was done on per chromosome basis (Fol4287 1-15)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#Here is the example of chromosome 7 (Suppl. Fig. 2)
for f in Fo-Et-0*; do nucmer -maxmatch -c 100 -p ${f}_chr7 chr7.fna ${f}; done #outputs delta file

# Whole genome alignments were visualized and filtered in R
Rscript ./mummer_coverage.R

# This led to the removal of any SNPs mapped to chromosomes 3, 6, 14 and 15 as well as supercontig 27 (end of chromosome 1)
# Any reference contigs that mapped poorly to the Illumina genomes were removed running awk itieratively
# e.g. awk '($0 !~ &quot;Supercontig_2.27&quot;){print}' all_ethiopia_isolates.vcf.gz">
                     <pre class="notranslate" lang="{bash,"><code>#Here is the example of chromosome 7 (Suppl. Fig. 2)
for f in Fo-Et-0*; do nucmer -maxmatch -c 100 -p ${f}_chr7 chr7.fna ${f}; done #outputs delta file

# Whole genome alignments were visualized and filtered in R
Rscript ./mummer_coverage.R

# This led to the removal of any SNPs mapped to chromosomes 3, 6, 14 and 15 as well as supercontig 27 (end of chromosome 1)
# Any reference contigs that mapped poorly to the Illumina genomes were removed running awk itieratively
# e.g. awk '($0 !~ "Supercontig_2.27"){print}' all_ethiopia_isolates.vcf.gz
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      C. Identification of clonal groups using STRUCTURE
                     </h3>
                     <a aria-label="Permalink: C. Identification of clonal groups using STRUCTURE" class="anchor" href="#c-identification-of-clonal-groups-using-structure" id="user-content-c-identification-of-clonal-groups-using-structure">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       STRUCTURE v2.3.4 (Pritchard et al, 2000)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Structure Harvester (
                       <a href="https://taylor0.biology.ucla.edu/structureHarvester/" rel="nofollow">
                        https://taylor0.biology.ucla.edu/structureHarvester/
                       </a>
                       ) (Earl et al, 2012)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       CLUMPP v1.1.2 (Jakobsson and Rosenberg, 2007)
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     Subdividing clonal lineages in fungal populations remains challenging. Traditionally, reasearchers have determined clonality based on identical MLGs. This approach fails on a whole genome scale due to the accumulation of erroneous mutations. We employed two different methods to subdivide the lineages: phylogenetics (see Gene Annotation and Analyses - BEAST) and K clustering (via STRUCTURE).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      i.
                      <a id="user-content-struct">
                      </a>
                      Random subsampling SNPs
                     </h4>
                     <a aria-label="Permalink: i. Random subsampling SNPs" class="anchor" href="#i-random-subsampling-snps" id="user-content-i-random-subsampling-snps">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To avoid SNP selection bias, the VCF file was randomly subsampled at a rate of ~2.5%.
Note that this script does not provide exactly 2.5% of the total number of SNPs (Table 1 - Non-clone corrected = 136,772), it gives each SNP a 2.5% chance of being included within the final subsampling.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub1.vcf.gz

bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub2.vcf.gz

bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub3.vcf.gz

bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub4.vcf.gz

#etc.

#Subsample SNP Totals:
#Sub1 = 3658
#Sub2 = 3816
#Sub3 = 3642
#Sub4 = 3754
#Sub5 = 3769
#Sub6 = 3671
#Sub7 = 3737
#Sub8 = 3774
#Sub9 = 3745
#Sub10 =3744
">
                     <pre class="notranslate" lang="{bash,"><code>
bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub1.vcf.gz

bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub2.vcf.gz

bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub3.vcf.gz

bgzip -dc all_ethiopia_isolates.vcf.gz | awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025) print $0}' | bgzip -c &gt; ethiopia_fo_structure.0_05.vcf.gz_sub4.vcf.gz

#etc.

#Subsample SNP Totals:
#Sub1 = 3658
#Sub2 = 3816
#Sub3 = 3642
#Sub4 = 3754
#Sub5 = 3769
#Sub6 = 3671
#Sub7 = 3737
#Sub8 = 3774
#Sub9 = 3745
#Sub10 =3744

</code></pre>
                    </div>
                    <p dir="auto">
                     VCF subsample files were converted into genotype files using a manual bash script
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="############ ./convertvcftogeno.sh ############
#!/usr/bin/env bash
file=$1
out=${file}.out
bgzip -dc ${file} | awk '$1 ~ &quot;Super&quot; || $1 ~ &quot;#CHROM&quot; {print}' | awk '$1 !~ &quot;##&quot; {print}' | sed 's#0/0#0#g;s#1/1#1#g;s#2/2#2#g;s#3/3#3#g;s#4/4#4#g;s#0/1#-9#g;s#0/2#-9#g;s#0/3#-9#g;s#0/4#-9#g;s#1/2#-9#g;s#1/3#-9#g;s#1/4#-9#g;s#2/3#-9#g;s#2/4#-9#g;s#3/4#-9#g;s#./.#-9#g' | cut -f 2,10- | awk '
{ 
    for (i=1; i&lt;=NF; i++)  {
        a[NR,i] = $i
    }
}
NF&gt;p { p = NF }
END {    
    for(j=1; j&lt;=p; j++) {
        str=a[1,j]
        for(i=2; i&lt;=NR; i++){
            str=str&quot; &quot;a[i,j];
        }
        print str
    }
}' &gt; $out">
                     <pre class="notranslate" lang="{bash,"><code>############ ./convertvcftogeno.sh ############
#!/usr/bin/env bash
file=$1
out=${file}.out
bgzip -dc ${file} | awk '$1 ~ "Super" || $1 ~ "#CHROM" {print}' | awk '$1 !~ "##" {print}' | sed 's#0/0#0#g;s#1/1#1#g;s#2/2#2#g;s#3/3#3#g;s#4/4#4#g;s#0/1#-9#g;s#0/2#-9#g;s#0/3#-9#g;s#0/4#-9#g;s#1/2#-9#g;s#1/3#-9#g;s#1/4#-9#g;s#2/3#-9#g;s#2/4#-9#g;s#3/4#-9#g;s#./.#-9#g' | cut -f 2,10- | awk '
{ 
    for (i=1; i&lt;=NF; i++)  {
        a[NR,i] = $i
    }
}
NF&gt;p { p = NF }
END {    
    for(j=1; j&lt;=p; j++) {
        str=a[1,j]
        for(i=2; i&lt;=NR; i++){
            str=str" "a[i,j];
        }
        print str
    }
}' &gt; $out
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      ii. STRUCTURE
                     </h4>
                     <a aria-label="Permalink: ii. STRUCTURE" class="anchor" href="#ii-structure" id="user-content-ii-structure">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Ran STRUCTURE on each subsampling 10 times: burnin = 10,000 and MCMC reps = 50,000. K values = 2-20.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# e.g.
#!/bin/bash
module load structure-console/2.3.4
structure -K 10 -o structure_large10-1-2">
                     <pre class="notranslate" lang="{bash,"><code># e.g.
#!/bin/bash
module load structure-console/2.3.4
structure -K 10 -o structure_large10-1-2
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      iii. STRUCTUREharvester
                     </h4>
                     <a aria-label="Permalink: iii. STRUCTUREharvester" class="anchor" href="#iii-structureharvester" id="user-content-iii-structureharvester">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Using outputs of STRUCTURE, combine them using STRUCTURE_HARVESTER (online web server:
                     <a href="http://taylor0.biology.ucla.edu/structureHarvester/" rel="nofollow">
                      http://taylor0.biology.ucla.edu/structureHarvester/
                     </a>
                     ) to produce .indfiles for CLUMPP.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      iv. CLUMPP
                     </h4>
                     <a aria-label="Permalink: iv. CLUMPP" class="anchor" href="#iv-clumpp" id="user-content-iv-clumpp">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Ran CLUMPP on HPC cluster to collate 10 permutations of each subsample. Used LargeKGreedy algorithm.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#e.g.
#!/bin/bash
module load CLUMPP/1.1.2
CLUMPP K2.paramfile -i K12.indfile -k 12 -c 133 -r 10 -m 3 -w 0 -s 2 -o K12.outfile -j K12.miscfile">
                     <pre class="notranslate" lang="{bash,"><code>#e.g.
#!/bin/bash
module load CLUMPP/1.1.2
CLUMPP K2.paramfile -i K12.indfile -k 12 -c 133 -r 10 -m 3 -w 0 -s 2 -o K12.outfile -j K12.miscfile
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      v. STRUCTURE plotted with R script
                     </h4>
                     <a aria-label="Permalink: v. STRUCTURE plotted with R script" class="anchor" href="#v-structure-plotted-with-r-script" id="user-content-v-structure-plotted-with-r-script">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="######### ./structure_plotting_larger_collection #############">
                     <pre class="notranslate" lang="{bash,"><code>######### ./structure_plotting_larger_collection #############
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      vi. Clonal groups were assigned based on Euclidean distances
                     </h4>
                     <a aria-label="Permalink: vi. Clonal groups were assigned based on Euclidean distances" class="anchor" href="#vi-clonal-groups-were-assigned-based-on-euclidean-distances" id="user-content-vi-clonal-groups-were-assigned-based-on-euclidean-distances">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='######### ./plotting_structure_network.R ##########
./plotting_structure_network.R --list "list of all .outfile paths" --label "list of isolate names (in order)" --title "title of graph (usually K val included)"'>
                     <pre class="notranslate" lang="{bash,"><code>######### ./plotting_structure_network.R ##########
./plotting_structure_network.R --list "list of all .outfile paths" --label "list of isolate names (in order)" --title "title of graph (usually K val included)"
</code></pre>
                    </div>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      <a id="user-content-pop_gen">
                      </a>
                      5. Population Analyses
                     </h1>
                     <a aria-label="Permalink: 5. Population Analyses" class="anchor" href="#5-population-analyses-1" id="user-content-5-population-analyses-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      A. Geographic Analyses
                     </h3>
                     <a aria-label="Permalink: A. Geographic Analyses" class="anchor" href="#a-geographic-analyses" id="user-content-a-geographic-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      i. Plotting onto maps in R
                     </h4>
                     <a aria-label="Permalink: i. Plotting onto maps in R" class="anchor" href="#i-plotting-onto-maps-in-r" id="user-content-i-plotting-onto-maps-in-r">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To create plots of geographic sampling (both chickpea and Fungal sampling)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="######### ./geo_plot_ethiopia.R ###########">
                     <pre class="notranslate" lang="{bash,"><code>######### ./geo_plot_ethiopia.R ###########
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_figure1.gif" rel="noopener noreferrer" target="_blank">
                      <img alt="" data-animated-image="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_figure1.gif" style="max-width: 100%;"/>
                     </a>
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_geog_subgroups.gif" rel="noopener noreferrer" target="_blank">
                      <img alt="" data-animated-image="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_geog_subgroups.gif" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_geog_mat.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_geog_mat.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      ii. Mantel tests
                     </h4>
                     <a aria-label="Permalink: ii. Mantel tests" class="anchor" href="#ii-mantel-tests" id="user-content-ii-mantel-tests">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The Mantel test is a measure of correlation between two distance matrices. Here we want to understand how genetic difference (both sequence and presence/absence variation) is correlated with geography.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='##We prepared the co-ordinates for all infected sample into data frame for mental test
library(ade4)
location = read.delim("location-for-distance-plot-99-isolates.txt")
rownames(location) &lt;- location[, "Sample"]
location &lt;- location[, -1]
x = as.matrix(location)
y &lt;- distm(x, fun = distGeo)
physical_distance &lt;- melt(y)

n &lt;- max(table(physical_distance$Sample)) + 1  # +1,  so we have diagonal of 
res &lt;- lapply(with(physical_distance, split(value, physical_distance$Sample)), function(x) c(rep(NA, n - length(x)), x))
res &lt;- do.call("rbind", res)
res &lt;- rbind(res, rep(NA, n))
res &lt;- as.dist(t(res))
res
##mantel test for accessary genome usimg Jaccarad distance
m &lt;- max(table(pan_distance$X)) + 1  # +1,  so we have diagonal of 
res2 &lt;- lapply(with(pan_distance, split(value, pan_distance$X)), function(x) c(rep(NA, m - length(x)), x))
res2 &lt;- do.call("rbind", res2)
res2 &lt;- rbind(res2, rep(NA, m))
res2 &lt;- as.dist(t(res2))
res2
mantel.rtest(res, res2, nrepet = 9999)
###manetl test for core using Conserved Ortholog genes were also performed the same.'>
                     <pre class="notranslate" lang="{r,"><code>##We prepared the co-ordinates for all infected sample into data frame for mental test
library(ade4)
location = read.delim("location-for-distance-plot-99-isolates.txt")
rownames(location) &lt;- location[, "Sample"]
location &lt;- location[, -1]
x = as.matrix(location)
y &lt;- distm(x, fun = distGeo)
physical_distance &lt;- melt(y)

n &lt;- max(table(physical_distance$Sample)) + 1  # +1,  so we have diagonal of 
res &lt;- lapply(with(physical_distance, split(value, physical_distance$Sample)), function(x) c(rep(NA, n - length(x)), x))
res &lt;- do.call("rbind", res)
res &lt;- rbind(res, rep(NA, n))
res &lt;- as.dist(t(res))
res
##mantel test for accessary genome usimg Jaccarad distance
m &lt;- max(table(pan_distance$X)) + 1  # +1,  so we have diagonal of 
res2 &lt;- lapply(with(pan_distance, split(value, pan_distance$X)), function(x) c(rep(NA, m - length(x)), x))
res2 &lt;- do.call("rbind", res2)
res2 &lt;- rbind(res2, rep(NA, m))
res2 &lt;- as.dist(t(res2))
res2
mantel.rtest(res, res2, nrepet = 9999)
###manetl test for core using Conserved Ortholog genes were also performed the same.
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      B. Linkage disequilibrium
                     </h3>
                     <a aria-label="Permalink: B. Linkage disequilibrium" class="anchor" href="#b-linkage-disequilibrium" id="user-content-b-linkage-disequilibrium">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We calculated linkage disequilibrium using two different methods:
                    </p>
                    <ol dir="auto">
                     <li>
                      Pearson's correlation coefficient (r2) [implemented in PLINK]
                     </li>
                     <li>
                      Index of association (rbarD) [implemented in PoppR]
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      i. Linkage Disequilibrium via r2
                     </h4>
                     <a aria-label="Permalink: i. Linkage Disequilibrium via r2" class="anchor" href="#i-linkage-disequilibrium-via-r2" id="user-content-i-linkage-disequilibrium-via-r2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       Plink v1.90 (Purcell et al, 2007)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       BCFtools v1.16 (Danecek et al, 2021)
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Preparing datasets for "clonally-derived" vs "ancestrally-derived" SNPs (Figure 4 and Table 1)
                     </h4>
                     <a aria-label='Permalink: Preparing datasets for "clonally-derived" vs "ancestrally-derived" SNPs (Figure 4 and Table 1)' class="anchor" href="#preparing-datasets-for-clonally-derived-vs-ancestrally-derived-snps-figure-4-and-table-1" id="user-content-preparing-datasets-for-clonally-derived-vs-ancestrally-derived-snps-figure-4-and-table-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     "Clonally-derived" SNPs are variants that are variable within clonal subgroups.
                    </p>
                    <p dir="auto">
                     "Ancestrally-derived" SNPs are variants that are fixed within clonal subgroups
                    </p>
                    <p dir="auto">
                     To do this, we separated the final VCF file (See Filtering VCF) into the subgroups defined by BEAST (phylogenetics) and STRUCTURE (K clustering).
                    </p>
                    <ol dir="auto">
                     <li>
                      Split the VCf file into groups
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="######## ./splitting_vcfs_retaining.sh ###########

#!/usr/bin/env bash
for file in *.vcf.gz; do
  for sample in `bcftools query -l $file`; do
    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file
  done
done">
                     <pre class="notranslate" lang="{bash,"><code>######## ./splitting_vcfs_retaining.sh ###########

#!/usr/bin/env bash
for file in *.vcf.gz; do
  for sample in `bcftools query -l $file`; do
    bcftools view -Oz -s $sample -o ${file/.vcf*/.$sample.vcf.gz} $file
  done
done
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      Combine into VCF files on a clade basis using BCFtools
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#e.g. This is an example for clonal subgroup E. Note that this was done for all groups E-U
#note that "clade_e.txt" is a text file containing the list of isolates contained in subgroup E
bcftools merge -l clade_e.txt &gt; clade_e.vcf.gz'>
                     <pre class="notranslate" lang="{bash,"><code>#e.g. This is an example for clonal subgroup E. Note that this was done for all groups E-U
#note that "clade_e.txt" is a text file containing the list of isolates contained in subgroup E
bcftools merge -l clade_e.txt &gt; clade_e.vcf.gz
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Filtering step for clonally-derived SNPs
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bgzip -dc  $group.vcf.gz | // #input file lineages
awk '($0 ~ &quot;1/1&quot; &amp;&amp; $0 ~ &quot;0/0&quot;) || ($0 ~ &quot;1/1&quot; &amp;&amp; $0 ~ &quot;2/2&quot;) || //
($0 ~ &quot;2/2&quot; &amp;&amp; $0 ~ &quot;0/0&quot;) || ($0 ~ &quot;3/3&quot; &amp;&amp; $0 ~ &quot;0/0&quot;) || //
($0 ~ &quot;4/4&quot; &amp;&amp; $0 ~ &quot;0/0&quot;) || ($0 ~ &quot;3/3&quot; &amp;&amp; $0 ~ &quot;1/1&quot;) || //
($0 ~ &quot;4/4&quot; &amp;&amp; $0 ~ &quot;1/1&quot;) || ($0 ~ &quot;3/3&quot; &amp;&amp; $0 ~ &quot;2/2&quot;) || //
($0 ~ &quot;4/4&quot; &amp;&amp; $0 ~ &quot;2/2&quot;) || ($0 ~ &quot;3/3&quot; &amp;&amp; $0 ~ &quot;4/4&quot;) || //
$1 ~ /#/ {print $0}' | bgzip -c &gt; $output.vcf.gz #output SNPs variable across each clonal group
#note that these files were used to generate Figure 4C (decay plot for each clonal lineage)

bcftools merge -l $(ls clade*) &gt; var_clones.vcf.gz">
                     <pre class="notranslate" lang="{bash,"><code>bgzip -dc  $group.vcf.gz | // #input file lineages
awk '($0 ~ "1/1" &amp;&amp; $0 ~ "0/0") || ($0 ~ "1/1" &amp;&amp; $0 ~ "2/2") || //
($0 ~ "2/2" &amp;&amp; $0 ~ "0/0") || ($0 ~ "3/3" &amp;&amp; $0 ~ "0/0") || //
($0 ~ "4/4" &amp;&amp; $0 ~ "0/0") || ($0 ~ "3/3" &amp;&amp; $0 ~ "1/1") || //
($0 ~ "4/4" &amp;&amp; $0 ~ "1/1") || ($0 ~ "3/3" &amp;&amp; $0 ~ "2/2") || //
($0 ~ "4/4" &amp;&amp; $0 ~ "2/2") || ($0 ~ "3/3" &amp;&amp; $0 ~ "4/4") || //
$1 ~ /#/ {print $0}' | bgzip -c &gt; $output.vcf.gz #output SNPs variable across each clonal group
#note that these files were used to generate Figure 4C (decay plot for each clonal lineage)

bcftools merge -l $(ls clade*) &gt; var_clones.vcf.gz
</code></pre>
                    </div>
                    <ol dir="auto" start="4">
                     <li>
                      Filtering step for ancestrally-derived SNPs
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#filter across clonal subgroup
for F in *.vcf.gz; do bgzip -dc $F | //
awk '{for(i=11;i&lt;=NF;i++)if($i!=$(i-1)&amp;&amp;$1!~/#/)next}1' | //
bgzip -c &gt; ${F}_filtered.vcf.gz; done #output SNPs fixed across entire clonal group

#choose one sequence from the subgroup VCF files
for F in *_filtered.vcf.gz; do bgzip -dc $F | //
cut -f 1-11 | //
bgzip -c &gt; ${F}_represent.vcf.gz; done

#merge representative isolates
bcftools merge -l $(ls *_represent.vcf.gz) &gt; eth_clonecorrected.vcf.gz">
                     <pre class="notranslate" lang="{bash,"><code>#filter across clonal subgroup
for F in *.vcf.gz; do bgzip -dc $F | //
awk '{for(i=11;i&lt;=NF;i++)if($i!=$(i-1)&amp;&amp;$1!~/#/)next}1' | //
bgzip -c &gt; ${F}_filtered.vcf.gz; done #output SNPs fixed across entire clonal group

#choose one sequence from the subgroup VCF files
for F in *_filtered.vcf.gz; do bgzip -dc $F | //
cut -f 1-11 | //
bgzip -c &gt; ${F}_represent.vcf.gz; done

#merge representative isolates
bcftools merge -l $(ls *_represent.vcf.gz) &gt; eth_clonecorrected.vcf.gz
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Linkage disequilibrium - r2
                     </h4>
                     <a aria-label="Permalink: Linkage disequilibrium - r2" class="anchor" href="#linkage-disequilibrium---r2" id="user-content-linkage-disequilibrium---r2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      Run intra-chromosomal PLINK LD
                     </li>
                    </ol>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#code for clonally-derived SNPs
plink --vcf var_clones.vcf.gz --allow-extra-chr --r2 //
--ld-window-r2 0 --ld-window 99999 --out var_clone

#code for ancestrally-derived SNPs
plink --vcf total_clone.vcf.gz --allow-extra-chr --r2 --ld-window 99999 //
--ld-window-r2 0 --out total_clone

#note that PLINK automatically retains only biallelic SNPs

#calculating bp difference from PLINK LD output
for f in *.ld; do awk -F' ' '{print $1, $2, $5, $7, ($2-$5) }' $f | sed 's/-//g;s/ /\t/g' &gt; diff_calculated/$f.diff; done

#filtering for bp difference &lt;75000 (saves memory to permit R-based analysis)
for f in *diff; do awk '($5 &lt; 75000) {print}' $f &gt; rm_large_diffs/$f.75000.ld; done

#rest of plotting done in R">
                     <pre class="notranslate" lang="{bash,"><code>#code for clonally-derived SNPs
plink --vcf var_clones.vcf.gz --allow-extra-chr --r2 //
--ld-window-r2 0 --ld-window 99999 --out var_clone

#code for ancestrally-derived SNPs
plink --vcf total_clone.vcf.gz --allow-extra-chr --r2 --ld-window 99999 //
--ld-window-r2 0 --out total_clone

#note that PLINK automatically retains only biallelic SNPs

#calculating bp difference from PLINK LD output
for f in *.ld; do awk -F' ' '{print $1, $2, $5, $7, ($2-$5) }' $f | sed 's/-//g;s/ /\t/g' &gt; diff_calculated/$f.diff; done

#filtering for bp difference &lt;75000 (saves memory to permit R-based analysis)
for f in *diff; do awk '($5 &lt; 75000) {print}' $f &gt; rm_large_diffs/$f.75000.ld; done

#rest of plotting done in R
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      Run inter-chromosomal PLINK LD
                     </li>
                    </ol>
                    <p dir="auto">
                     Since inter-chromosomal SNPs can create unreasonably large file sizes (&gt;100GB), we subset the SNP files before running
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#selects ~2.5% of the total SNPs (~2000-3000 SNPs)
#run on both var_clone.vcf.gz and eth_clonecorrected.vcf.gz
awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025 || FNR==1) print $0}' $IN.vcf.gz &gt; $OUT-filtered.vcf.gz

#calculating interchromosomal LD
plink --vcf var_clone.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //
--ld-window-r2 0 --out var_clone_inter

#calculating interchromosomal LD
plink --vcf eth_clonecorrected.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //
--ld-window-r2 0 --out total_clone_inter

#mean of each LD was calculated (background LD)
# file.ld = any inter-chromosomal PLINK LD output
count=0; total=0; for i in $( awk '{ print $7; }' file.ld );\
do total=$(echo $total+$i | bc ); \
((count++)); done; echo &quot;scale=2; $total / $count&quot; | bc">
                     <pre class="notranslate" lang="{bash,"><code>#selects ~2.5% of the total SNPs (~2000-3000 SNPs)
#run on both var_clone.vcf.gz and eth_clonecorrected.vcf.gz
awk 'BEGIN {srand()} !/^$/ { if (rand() &lt;= .025 || FNR==1) print $0}' $IN.vcf.gz &gt; $OUT-filtered.vcf.gz

#calculating interchromosomal LD
plink --vcf var_clone.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //
--ld-window-r2 0 --out var_clone_inter

#calculating interchromosomal LD
plink --vcf eth_clonecorrected.vcf.gz-filtered.vcf.gz --allow-extra-chr --inter-chr --r2 //
--ld-window-r2 0 --out total_clone_inter

#mean of each LD was calculated (background LD)
# file.ld = any inter-chromosomal PLINK LD output
count=0; total=0; for i in $( awk '{ print $7; }' file.ld );\
do total=$(echo $total+$i | bc ); \
((count++)); done; echo "scale=2; $total / $count" | bc
</code></pre>
                    </div>
                    <ol dir="auto" start="3">
                     <li>
                      Plotting r2 decay curves and genome-wide r2
                     </li>
                    </ol>
                    <p dir="auto">
                     r2 decay plots
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="############# ld_decay.R ##################">
                     <pre class="notranslate" lang="{r,"><code>############# ld_decay.R ##################
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_ld_decay_figures.gif" rel="noopener noreferrer" target="_blank">
                      <img alt="" data-animated-image="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_ld_decay_figures.gif" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     r2 genome-wide plots
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="############## making_ld_windows.Rmd ###############">
                     <pre class="notranslate" lang="{r,"><code>############## making_ld_windows.Rmd ###############
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/pipeline_ld_across_window.gif" rel="noopener noreferrer" target="_blank">
                      <img alt="" data-animated-image="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/pipeline_ld_across_window.gif" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      ii. Linkage disequilibrium - rbarD
                     </h4>
                     <a aria-label="Permalink: ii. Linkage disequilibrium - rbarD" class="anchor" href="#ii-linkage-disequilibrium---rbard" id="user-content-ii-linkage-disequilibrium---rbard">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To complement the Pearson's correlation coefficient (r2) LD data, we included the index of association (rbarD)(Agapow and Burt, 2001)
                    </p>
                    <p dir="auto">
                     One issue that we experienced with this metric though is that it is very computationally intensive (at least in PoppR). As a result, most SNP sets had to be downsampled to a size of ~3000 SNPs
                    </p>
                    <p dir="auto">
                     We used the same subsamples as for
                     <a href="#struct">
                      STRUCTURE
                     </a>
                     . As a result, the rbarD calculations were averaged over the 10 subsamples.
                    </p>
                    <p dir="auto">
                     We used PoppR (Kamvar et al, 2014) to run rbarD. The VCF files were converted into geno files using vcfR (Knaus et al, 2017)
                    </p>
                    <p dir="auto">
                     The generalized framework for calculating the index of association (rbarD) in R:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='##########################################################################
## For full code, see ./ia_analysis.Rmd. Here is an example of how the ###
## index of association (rbarD) was calculated using Poppr. This example #
## contains only Subsample 1. ############################################
##########################################################################

#load required libraries
library(tidyverse)
library(RColorBrewer)
library(ggplot2)
library(vcfR)
library(adegenet)
library(poppr)
library(magrittr)

##############################################################################
#function to prep VCF data
file_prep &lt;- function(file) {
  isolate_vcf &lt;- read.vcfR(file = file, verbose = FALSE) #import vcf file (can also be vcf.gz)
  isolate_genind &lt;- vcfR2genind(isolate_vcf) #converts vcf into genind format
  isolate_genind$other$lineage &lt;- c("SING", "SING", "SING", "SING", "SING", "T", "SING", "T", "O", "SING", "Q", "Q", "O", "J", "N", "SING", "N", "M", "M", "N", "SING", "P", "SING", "K", "N", "T", "Q", "SING", "P", "T", "N", "N", "P", "J", "J", "SING", "SING", "S", "S", "SING", "O", "SING", "SING", "T", "P", "T", "P", "R", "SING", "SING", "SING", "O", "R", "SING", "SING", "U", "SING", "SING", "E", "SING", "SING", "Q", "SING", "R", "O", "Q", "R", "SING", "SING", "R", "R", "R", "SING", "Q", "SING", "SING", "R", "O", "I", "M", "R", "E", "H", "SING", "G", "H", "F", "SING", "H", "SING", "L", "L", "O", "H", "H", "P", "R", "SING", "O", "T", "K", "E", "O", "SING", "O", "SING", "F", "G", "G", "M", "O", "Q", "P", "E", "T", "K", "R", "R", "SING", "O", "I", "U", "E", "SING", "P", "P", "R", "SING", "N", "R", "O", "O", "SING") #predetermined lineages from STRUCTURE and phylogenetic analysis
  pop(isolate_genind) &lt;- isolate_genind$other$lineage #assigns lineages as populations
  isolate_genclone &lt;- as.genclone(isolate_genind) #creates genclone file for poppr analysis
}

###################################################################
ss1 &lt;- file_prep("subsample_1.vcf")
#etc.
# All 10 subsamples were loaded into R like this.

# Note
# (Subsample #) : (# of SNPs)
# Subsample 1 : 3658
# Subsample 2 : 3816
# Subsample 3 : 3642
# Subsample 4 : 3754
# Subsample 5 : 3769
# Subsample 6 : 3671
# Subsample 7 : 3737
# Subsample 8 : 3774
# Subsample 9 : 3745
# Subsample 10 : 3744

##Same SNP subsamples as used to delimit clonal groups in STRUCUTURE

###################################################################
# non-clone corrected Ia and Rbar calculations
set.seed(100)
resmpl_ia_ss1 &lt;- resample.ia(ss1, reps = 999, n=132)
#etc.

#combine all subsampled data together
resmpl_ia_all &lt;- rbind(resmpl_ia_ss1,
    resmpl_ia_ss2,
    resmpl_ia_ss3,
    resmpl_ia_ss4,
    resmpl_ia_ss5,
    resmpl_ia_ss6,
    resmpl_ia_ss7,
    resmpl_ia_ss8,
    resmpl_ia_ss9,
    resmpl_ia_ss10)

  write_delim(resmpl_ia_all, file = "results/all_isolates_ia_vals.txt", delim = "\t")
resmpl_ia_all &lt;- read_delim("results/all_isolates_ia_vals.txt", delim = "\t")
std(resmpl_ia_all$rbarD)


t &lt;- ggplot(resmpl_ia_all, aes(x = rbarD)) +
  geom_histogram(bins=100) +
  geom_vline(xintercept = ia(ss1)[2]) +
#  geom_vline(xintercept = 0.0702, linetype=2) +
  geom_vline(xintercept = ia(clonecorrect(ss1))[2], linetype = 2) +
  xlab(expression(bar(r)[d]))
t

######################################################################
#separating populations before index of association calculation
obj1 &lt;- seppop(ss1) #separates the genclone objects into a list of different populations specified earlier
#etc.

####SS1
sing_genclone1 &lt;- obj1$SING
e_genclone1 &lt;- obj1$E
f_genclone1 &lt;- obj1$F
g_genclone1 &lt;- obj1$G
h_genclone1 &lt;- obj1$H
i_genclone1 &lt;- obj1$I
k_genclone1 &lt;- obj1$K
l_genclone1 &lt;- obj1$L
m_genclone1 &lt;- obj1$M
n_genclone1 &lt;- obj1$N
o_genclone1 &lt;- obj1$O
p_genclone1 &lt;- obj1$P
q_genclone1 &lt;- obj1$Q
r_genclone1 &lt;- obj1$R
s_genclone1 &lt;- obj1$S
t_genclone1 &lt;- obj1$T

# calculate Ia and rbarD for subgroup E, subsample 1
resmpl_ia_ss1_e &lt;- resample.ia(e_genclone1, reps = 999, n=(nrow(as.data.frame(e_genclone1))-1))
#etc. 
# calculated for all other subsamples (E) too
#combined all subsamples for subgroup E together

resmpl_ia_e &lt;- rbind(resmpl_ia_ss1_e,
    resmpl_ia_ss2_e,
    resmpl_ia_ss3_e,
    resmpl_ia_ss4_e,
    resmpl_ia_ss5_e,
    resmpl_ia_ss6_e,
    resmpl_ia_ss7_e,
    resmpl_ia_ss8_e,
    resmpl_ia_ss9_e,
    resmpl_ia_ss10_e)

# save data for subgroup E
write_delim(resmpl_ia_e, file = "results/E_isolates_ia_vals.txt", delim = "\t")


###### calculated for all other subgroups too #########

# Averages (Mean) and Standard Deviations were calculated in R #'>
                     <pre class="notranslate" lang="{r,"><code>##########################################################################
## For full code, see ./ia_analysis.Rmd. Here is an example of how the ###
## index of association (rbarD) was calculated using Poppr. This example #
## contains only Subsample 1. ############################################
##########################################################################

#load required libraries
library(tidyverse)
library(RColorBrewer)
library(ggplot2)
library(vcfR)
library(adegenet)
library(poppr)
library(magrittr)

##############################################################################
#function to prep VCF data
file_prep &lt;- function(file) {
  isolate_vcf &lt;- read.vcfR(file = file, verbose = FALSE) #import vcf file (can also be vcf.gz)
  isolate_genind &lt;- vcfR2genind(isolate_vcf) #converts vcf into genind format
  isolate_genind$other$lineage &lt;- c("SING", "SING", "SING", "SING", "SING", "T", "SING", "T", "O", "SING", "Q", "Q", "O", "J", "N", "SING", "N", "M", "M", "N", "SING", "P", "SING", "K", "N", "T", "Q", "SING", "P", "T", "N", "N", "P", "J", "J", "SING", "SING", "S", "S", "SING", "O", "SING", "SING", "T", "P", "T", "P", "R", "SING", "SING", "SING", "O", "R", "SING", "SING", "U", "SING", "SING", "E", "SING", "SING", "Q", "SING", "R", "O", "Q", "R", "SING", "SING", "R", "R", "R", "SING", "Q", "SING", "SING", "R", "O", "I", "M", "R", "E", "H", "SING", "G", "H", "F", "SING", "H", "SING", "L", "L", "O", "H", "H", "P", "R", "SING", "O", "T", "K", "E", "O", "SING", "O", "SING", "F", "G", "G", "M", "O", "Q", "P", "E", "T", "K", "R", "R", "SING", "O", "I", "U", "E", "SING", "P", "P", "R", "SING", "N", "R", "O", "O", "SING") #predetermined lineages from STRUCTURE and phylogenetic analysis
  pop(isolate_genind) &lt;- isolate_genind$other$lineage #assigns lineages as populations
  isolate_genclone &lt;- as.genclone(isolate_genind) #creates genclone file for poppr analysis
}

###################################################################
ss1 &lt;- file_prep("subsample_1.vcf")
#etc.
# All 10 subsamples were loaded into R like this.

# Note
# (Subsample #) : (# of SNPs)
# Subsample 1 : 3658
# Subsample 2 : 3816
# Subsample 3 : 3642
# Subsample 4 : 3754
# Subsample 5 : 3769
# Subsample 6 : 3671
# Subsample 7 : 3737
# Subsample 8 : 3774
# Subsample 9 : 3745
# Subsample 10 : 3744

##Same SNP subsamples as used to delimit clonal groups in STRUCUTURE

###################################################################
# non-clone corrected Ia and Rbar calculations
set.seed(100)
resmpl_ia_ss1 &lt;- resample.ia(ss1, reps = 999, n=132)
#etc.

#combine all subsampled data together
resmpl_ia_all &lt;- rbind(resmpl_ia_ss1,
    resmpl_ia_ss2,
    resmpl_ia_ss3,
    resmpl_ia_ss4,
    resmpl_ia_ss5,
    resmpl_ia_ss6,
    resmpl_ia_ss7,
    resmpl_ia_ss8,
    resmpl_ia_ss9,
    resmpl_ia_ss10)

  write_delim(resmpl_ia_all, file = "results/all_isolates_ia_vals.txt", delim = "\t")
resmpl_ia_all &lt;- read_delim("results/all_isolates_ia_vals.txt", delim = "\t")
std(resmpl_ia_all$rbarD)


t &lt;- ggplot(resmpl_ia_all, aes(x = rbarD)) +
  geom_histogram(bins=100) +
  geom_vline(xintercept = ia(ss1)[2]) +
#  geom_vline(xintercept = 0.0702, linetype=2) +
  geom_vline(xintercept = ia(clonecorrect(ss1))[2], linetype = 2) +
  xlab(expression(bar(r)[d]))
t

######################################################################
#separating populations before index of association calculation
obj1 &lt;- seppop(ss1) #separates the genclone objects into a list of different populations specified earlier
#etc.

####SS1
sing_genclone1 &lt;- obj1$SING
e_genclone1 &lt;- obj1$E
f_genclone1 &lt;- obj1$F
g_genclone1 &lt;- obj1$G
h_genclone1 &lt;- obj1$H
i_genclone1 &lt;- obj1$I
k_genclone1 &lt;- obj1$K
l_genclone1 &lt;- obj1$L
m_genclone1 &lt;- obj1$M
n_genclone1 &lt;- obj1$N
o_genclone1 &lt;- obj1$O
p_genclone1 &lt;- obj1$P
q_genclone1 &lt;- obj1$Q
r_genclone1 &lt;- obj1$R
s_genclone1 &lt;- obj1$S
t_genclone1 &lt;- obj1$T

# calculate Ia and rbarD for subgroup E, subsample 1
resmpl_ia_ss1_e &lt;- resample.ia(e_genclone1, reps = 999, n=(nrow(as.data.frame(e_genclone1))-1))
#etc. 
# calculated for all other subsamples (E) too
#combined all subsamples for subgroup E together

resmpl_ia_e &lt;- rbind(resmpl_ia_ss1_e,
    resmpl_ia_ss2_e,
    resmpl_ia_ss3_e,
    resmpl_ia_ss4_e,
    resmpl_ia_ss5_e,
    resmpl_ia_ss6_e,
    resmpl_ia_ss7_e,
    resmpl_ia_ss8_e,
    resmpl_ia_ss9_e,
    resmpl_ia_ss10_e)

# save data for subgroup E
write_delim(resmpl_ia_e, file = "results/E_isolates_ia_vals.txt", delim = "\t")


###### calculated for all other subgroups too #########

# Averages (Mean) and Standard Deviations were calculated in R #
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      C. Nucleotide Diversity (pi)
                     </h2>
                     <a aria-label="Permalink: C. Nucleotide Diversity (pi)" class="anchor" href="#c-nucleotide-diversity-pi" id="user-content-c-nucleotide-diversity-pi">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      VCFtools v0.1.15 (Danecek et al, 2011)
                     </li>
                    </ul>
                    <p dir="auto">
                     Nucleotide diversity (pi) was calculated from VCF files (both
                     <a href="#read_mapping">
                      read-mapping
                     </a>
                     using VCFtools (Danecek et al, 2011)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="vcftools --gzvcf $IN.vcf.gz --window-pi  10000 --out $OUT ">
                     <pre class="notranslate" lang="{bash,"><code>vcftools --gzvcf $IN.vcf.gz --window-pi  10000 --out $OUT 
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      D. Tajima's D
                     </h2>
                     <a aria-label="Permalink: D. Tajima's D" class="anchor" href="#d-tajimas-d" id="user-content-d-tajimas-d">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      VCFtools v0.1.15 (Danecek et al, 2011)
                     </li>
                    </ul>
                    <p dir="auto">
                     Tajima's D was calculated from VCF files for
                     <a href="#read_mapping">
                      read-mapping
                     </a>
                     directly using VCFtools (Danecek et al, 2011). The Tajima's D analysis calculated for the pangenome required additional steps and so is described above under "
                     <a href="#pangenome">
                      Pangenome
                     </a>
                     ".
                    </p>
                    <p dir="auto">
                     For the readmapping VCF files, Tajima's D was calculated per SNP. These were then averaged across the genome in R.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="vcftools --gzvcf $IN.vcf.gz --TajimaD  1 --out $OUT ">
                     <pre class="notranslate" lang="{bash,"><code>vcftools --gzvcf $IN.vcf.gz --TajimaD  1 --out $OUT 
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      E. Genetic Differentiation (Fst) (Weighted)
                     </h2>
                     <a aria-label="Permalink: E. Genetic Differentiation (Fst) (Weighted)" class="anchor" href="#e-genetic-differentiation-fst-weighted" id="user-content-e-genetic-differentiation-fst-weighted">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Genetic differentiation (Fst) is a commonly used proxy to determine gene flow between multiple populations, because more gene flow = lower genetic differentiation.
                    </p>
                    <p dir="auto">
                     We wanted to understand the levels of genetic differentiation between different clonal subgroups (and therefore estimate the level of gene flow between purely clonal lineages). The idea behind this is to understand whether gene flow is still ongoing even after a transition to clonality. As a result, the Fst of sequence variation (nongenic core SNPs - Figure 6) was compared to presence/absence variation. Hypothesis: If gene flow is ongoing, we expect to see lower genetic differentiation between clonal subgroups that core SNP variation. Furthermore, they should display different demographic histories (i.e. the two Fst sets should be correlated).
                    </p>
                    <p dir="auto">
                     To account for the role that genic selection, we included a supplemental figure (Supplemental Figure 3) displaying sequence variation of conserved orthologs. This should serve as a sandwich with the core gene Fst (under highest purifying selection) and core read-mapped Fst (under no selection) serving as the two extremes. If selection is the only reason for lower genetic differentiation (Figure 6), we expect that the accessory gene presence-absence genetic differentiation to be sandwiched between the Fst values in Figure 6A and Supplemental Figure 3A. Since the accessory genome is less differentiated than both core SNP sets, we can deduce that neither selection or genome-wide recombination is the primary mechanism diversifying the accessory genome. Either gene flow is ongoing or ancestral gene loss is driving diversity in the accessory genome in our Ethiopian F. oxysporum collection.
                    </p>
                    <p dir="auto">
                     We used custom R scripts to calculate weighted Fst values. The input files for each R script are VCF files (haplotype) or presence/absence matrix (coverage) containing only specified subgroups.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#sequence variation (core genes and core SNPs)
# input is any VCF file
############# calculating_haplotype_diversity.R ##################
Rscript ./calculating_haplotype_diversity.R --populationa $clonal_lineageA.vcf --populationb $clonal_lineageB.vcf &gt; $OUT


#presence/absence variation (accessory genes)
############# calculating_coverage_diversity.R ##################
Rscript ./calculating_coverage_diversity.R --populationa $clonal_lineageA.txt --populationb $clonal_lineageB.txt &gt; $OUT
">
                     <pre class="notranslate" lang="{bash,"><code>#sequence variation (core genes and core SNPs)
# input is any VCF file
############# calculating_haplotype_diversity.R ##################
Rscript ./calculating_haplotype_diversity.R --populationa $clonal_lineageA.vcf --populationb $clonal_lineageB.vcf &gt; $OUT


#presence/absence variation (accessory genes)
############# calculating_coverage_diversity.R ##################
Rscript ./calculating_coverage_diversity.R --populationa $clonal_lineageA.txt --populationb $clonal_lineageB.txt &gt; $OUT

</code></pre>
                    </div>
                    <p dir="auto">
                     One can see that this phenomenon is not due to selection exclusively in the boxplot below
                    </p>
                    <p dir="auto">
                     <a href="/gnrobinson/Eth_foc_project/blob/main/Supplemental_Methods/selection_snps_tajD.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/gnrobinson/Eth_foc_project/raw/main/Supplemental_Methods/selection_snps_tajD.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      F. Effector Annotation
                     </h2>
                     <a aria-label="Permalink: F. Effector Annotation" class="anchor" href="#f-effector-annotation" id="user-content-f-effector-annotation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Tools required:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       TMHMM v2.0 (Krogh et al, 2001)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       SignalP v4.1 (Petersen et al, 2011)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       EffectorP (Sperschneider et al, 2022)
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     We relied on EffectorP and previous GO term annotations (see
                     <a id="user-content-gene_annotation">
                     </a>
                     B. Gene Annotation, Part II) to understand the functional annotation and enrichment within the putative effector dataset.
                    </p>
                    <p dir="auto">
                     As recommended, SignalP and TMHMM were run on representative pangenome gene cluster sequences, prior to EffectorP. This was ensure that effectors are predicted from the secretome.
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# running SignalP
for f in *.fa; do signalp -c 0 -f short -t euk $f &gt; ${f}.signalp; done

# running TMHMM
for f in *.fa; do cat $f | cat decodeanhmm.Linux_x86_64 -f /share/apps/tmhmm-2.0c/lib/TMHMM2.0.options -modelfile /share/apps/tmhmm-2.0c/lib/TMHMM2.0.model &gt; ${f}.tmhmm; done

#filtering correct sequences
cat *.signalp | awk '($0 !~ &quot;SignalP-TM&quot;){print}' &gt; putative_secrete.list
for i in $(readline);do cat $i &gt;&gt; putative_secrete.fa; done
while read -r line; do $1=fasta | cat $fasta &gt;&gt; putative_secrete.fa; done &lt; putative_secrete.list

# running effectorP
python EffectorP.py -i putative_secrete.fasta
">
                     <pre class="notranslate" lang="{bash,"><code># running SignalP
for f in *.fa; do signalp -c 0 -f short -t euk $f &gt; ${f}.signalp; done

# running TMHMM
for f in *.fa; do cat $f | cat decodeanhmm.Linux_x86_64 -f /share/apps/tmhmm-2.0c/lib/TMHMM2.0.options -modelfile /share/apps/tmhmm-2.0c/lib/TMHMM2.0.model &gt; ${f}.tmhmm; done

#filtering correct sequences
cat *.signalp | awk '($0 !~ "SignalP-TM"){print}' &gt; putative_secrete.list
for i in $(readline);do cat $i &gt;&gt; putative_secrete.fa; done
while read -r line; do $1=fasta | cat $fasta &gt;&gt; putative_secrete.fa; done &lt; putative_secrete.list

# running effectorP
python EffectorP.py -i putative_secrete.fasta

</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      <a id="user-content-refs">
                      </a>
                      6. References
                     </h1>
                     <a aria-label="Permalink: 6. References" class="anchor" href="#6-references-1" id="user-content-6-references-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Agapow, P.M. and Burt, A., 2001. Indices of multilocus linkage disequilibrium. Molecular Ecology Notes, 1(1‐2), pp.101-102.
                    </p>
                    <p dir="auto">
                     Almagro Armenteros, J.J., Tsirigos, K.D., Sønderby, C.K., Petersen, T.N., Winther, O., Brunak, S., von Heijne, G. and Nielsen, H., 2019. SignalP 5.0 improves signal peptide predictions using deep neural networks. Nature biotechnology, 37(4), pp.420-423.
                    </p>
                    <p dir="auto">
                     Bolger, A.M., Lohse, M. and Usadel, B., 2014. Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics, 30(15), pp.2114-2120.
                    </p>
                    <p dir="auto">
                     Cantarel, B.L., Korf, I., Robb, S.M., Parra, G., Ross, E., Moore, B., Holt, C., Alvarado, A.S. and Yandell, M., 2008. MAKER: an easy-to-use annotation pipeline designed for emerging model organism genomes. Genome research, 18(1), pp.188-196.
                    </p>
                    <p dir="auto">
                     Cheng, H., Concepcion, G.T., Feng, X., Zhang, H. and Li, H. 2021. Haplotype-resolved de novo assembly using phased assembly graphs with Hifiasm. Nat methods, 18: 170-175.
                    </p>
                    <p dir="auto">
                     Danecek, P., Auton, A., Abecasis, G., Albers, C.A., Banks, E., DePristo, M.A., Handsaker, R.E., Lunter, G., Marth, G.T., Sherry, S.T. and McVean, G., 2011. The variant call format and VCFtools. Bioinformatics, 27(15), pp.2156-2158.
                    </p>
                    <p dir="auto">
                     Danecek, P., Bonfield, J.K., Liddle, J., Marshall, J., Ohan, V., Pollard, M.O., Whitwham, A., Keane, T., McCarthy, S.A., Davies, R.M. and Li, H., 2021. Twelve years of SAMtools and BCFtools. Gigascience, 10(2), p.giab008.
                    </p>
                    <p dir="auto">
                     Earl, D.A. and VonHoldt, B.M., 2012. STRUCTURE HARVESTER: a website and program for visualizing STRUCTURE output and implementing the Evanno method. Conservation genetics resources, 4, pp.359-361.
                    </p>
                    <p dir="auto">
                     Emms, D.M., and Kelly, S. 2019. OrthoFinder: phylogenetic orthology inference for comparative genomics. Genome Biol 20, 238.
                    </p>
                    <p dir="auto">
                     Fouts, D.E., Brinkac, L., Beck, E., Inman, J. and Sutton, G., 2012. PanOCT: automated clustering of orthologs using conserved gene neighborhood for pan-genomic analysis of bacterial strains and closely related species. Nucleic acids research, 40(22), pp.e172-e172.
                    </p>
                    <p dir="auto">
                     Gnerre, S., MacCallum, I., Przybylski, D., Ribeiro, F.J., Burton, J.N., Walker, B.J., Sharpe, T., Hall, G., Shea, T.P., Sykes, S. and Berlin, A.M., 2011. High-quality draft assemblies of mammalian genomes from massively parallel sequence data. Proceedings of the National Academy of Sciences, 108(4), pp.1513-1518.
                    </p>
                    <p dir="auto">
                     Huerta-Cepas, J., Serra, F. and Bork, P., 2016. ETE 3: reconstruction, analysis, and visualization of phylogenomic data. Molecular biology and evolution, 33(6), pp.1635-1638.
                    </p>
                    <p dir="auto">
                     Jakobsson, M. and Rosenberg, N.A., 2007. CLUMPP: a cluster matching and permutation program for dealing with label switching and multimodality in analysis of population structure. Bioinformatics, 23(14), pp.1801-1806.
                    </p>
                    <p dir="auto">
                     Jones, P., Binns, D., Chang, H.Y., Fraser, M., Li, W., McAnulla, C., McWilliam, H., Maslen, J., Mitchell, A., Nuka, G. and Pesseat, S., 2014. InterProScan 5: genome-scale protein function classification. Bioinformatics, 30(9), pp.1236-1240.
                    </p>
                    <p dir="auto">
                     Kamvar, Z.N., Tabima, J.F. and Grünwald, N.J., 2014. Poppr: an R package for genetic analysis of populations with clonal, partially clonal, and/or sexual reproduction. PeerJ, 2, p.e281.
                    </p>
                    <p dir="auto">
                     Knaus, B.J. and Grünwald, N.J., 2017. vcfr: a package to manipulate and visualize variant call format data in R. Molecular ecology resources, 17(1), pp.44-53.
                    </p>
                    <p dir="auto">
                     Krogh, A., Larsson, B., Von Heijne, G. and Sonnhammer, E.L., 2001. Predicting transmembrane protein topology with a hidden Markov model: application to complete genomes. Journal of molecular biology, 305(3), pp.567-580.
                    </p>
                    <p dir="auto">
                     Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., Marth, G., Abecasis, G., Durbin, R. and 1000 Genome Project Data Processing Subgroup, 2009. The sequence alignment/map format and SAMtools. bioinformatics, 25(16), pp.2078-2079.
                    </p>
                    <p dir="auto">
                     Li, H., 2013. Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM. arXiv preprint arXiv:1303.3997.
                    </p>
                    <p dir="auto">
                     Marçais, G., Delcher, A.L., Phillippy, A.M., Coston, R., Salzberg, S.L. and Zimin, A., 2018. MUMmer4: A fast and versatile genome alignment system. PLoS computational biology, 14(1), p.e1005944.
                    </p>
                    <p dir="auto">
                     McKenna, A., Hanna, M., Banks, E., Sivachenko, A., Cibulskis, K., Kernytsky, A., Garimella, K., Altshuler, D., Gabriel, S., Daly, M. and DePristo, M.A., 2010. The Genome Analysis Toolkit: a MapReduce framework for analyzing next-generation DNA sequencing data. Genome research, 20(9), pp.1297-1303.
                    </p>
                    <p dir="auto">
                     Petersen, T.N., Brunak, S., Von Heijne, G. and Nielsen, H., 2011. SignalP 4.0: discriminating signal peptides from transmembrane regions. Nature methods, 8(10), pp.785-786.
                    </p>
                    <p dir="auto">
                     Pritchard, J.K., Stephens, M. and Donnelly, P., 2000. Inference of population structure using multilocus genotype data. Genetics, 155(2), pp.945-959.
                    </p>
                    <p dir="auto">
                     Pritchard, L., Glover, R.H., Humphris, S., Elphinstone, J.G. and Toth, I.K., 2016. Genomics and taxonomy in diagnostics for food security: soft-rotting enterobacterial plant pathogens. Analytical Methods, 8(1), pp.12-24.
                    </p>
                    <p dir="auto">
                     Purcell, S., Neale, B., Todd-Brown, K., Thomas, L., Ferreira, M.A., Bender, D., Maller, J., Sklar, P., De Bakker, P.I., Daly, M.J. and Sham, P.C., 2007. PLINK: a tool set for whole-genome association and population-based linkage analyses. The American journal of human genetics, 81(3), pp.559-575.
                    </p>
                    <p dir="auto">
                     Simão, F.A., Waterhouse, R.M., Ioannidis, P., Kriventseva, E.V. and Zdobnov, E.M., 2015. BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs. Bioinformatics, 31(19), pp.3210-3212.
                    </p>
                    <p dir="auto">
                     Sperschneider, J. and Dodds, P.N., 2022. EffectorP 3.0: prediction of apoplastic and cytoplasmic effectors in fungi and oomycetes. Molecular Plant-Microbe Interactions, 35(2), pp.146-156.
                    </p>
                    <p dir="auto">
                     Suchard, M.A., Lemey, P., Baele, G., Ayres, D.L., Drummond, A.J. and Rambaut, A., 2018. Bayesian phylogenetic and phylodynamic data integration using BEAST 1.10. Virus evolution, 4(1), p.vey016.
                    </p>
                    <p dir="auto">
                     Tritt, A., Eisen, J.A., Facciotti, M.T. and Darling, A.E., 2012. An integrated pipeline for de novo assembly of microbial genomes.
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="C1ml5lW7FEgZndWLSa3HbUU7p9ntGinYp2s18kcs0sMcgTpJybBsKBrRL_If-VqjM4GKOg2zUkLWuBFZQo9M4A">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Population genomics of Ethiopian Fusarium spps
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/gnrobinson/Eth_foc_project/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/gnrobinson/Eth_foc_project/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/gnrobinson/Eth_foc_project/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  star
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/gnrobinson/Eth_foc_project/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/gnrobinson/Eth_foc_project/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fgnrobinson%2FEth_foc_project&amp;report=gnrobinson+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/gnrobinson/Eth_foc_project/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/gnrobinson/packages?repo_name=Eth_foc_project">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/gnrobinson/Eth_foc_project/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="R 100.0" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 100.0%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/gnrobinson/Eth_foc_project/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   100.0%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
