<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         hamidghaedi/scRNA_seq-analysis: scRNA analysis of public dataset on bladder cancer
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="90A8:5F4D:2C45F1:3E77B9:6605C70E" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjkwQTg6NUY0RDoyQzQ1RjE6M0U3N0I5OjY2MDVDNzBFIiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="83781e54604d725ff2e4fc78ba22d33b44113a9145dc1a89638f8b0d60df78a7" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:320420414" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="scRNA analysis of public dataset on bladder cancer - hamidghaedi/scRNA_seq-analysis" name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/hamidghaedi/scRNA_seq-analysis" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/a7dc612eecc3bdd0f805525bad0d798a0056e4e41f936c6fe85223e2ce241849/hamidghaedi/scRNA_seq-analysis" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="hamidghaedi/scRNA_seq-analysis: scRNA analysis of public dataset on bladder cancer" name="twitter:title">
                       <meta content="scRNA analysis of public dataset on bladder cancer - hamidghaedi/scRNA_seq-analysis" name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/a7dc612eecc3bdd0f805525bad0d798a0056e4e41f936c6fe85223e2ce241849/hamidghaedi/scRNA_seq-analysis" property="og:image"/>
                        <meta content="scRNA analysis of public dataset on bladder cancer - hamidghaedi/scRNA_seq-analysis" property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="hamidghaedi/scRNA_seq-analysis: scRNA analysis of public dataset on bladder cancer" property="og:title"/>
                        <meta content="https://github.com/hamidghaedi/scRNA_seq-analysis" property="og:url"/>
                        <meta content="scRNA analysis of public dataset on bladder cancer - hamidghaedi/scRNA_seq-analysis" property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjExOTc1MjU2NjcsInQiOjE3MTE2NTQ2NzR9--a59fe76f67f353a50aa57b70e8f43c93d2d2fe8cdddaa7e32b1202fde685699b" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/hamidghaedi/scRNA_seq-analysis git https://github.com/hamidghaedi/scRNA_seq-analysis.git" name="go-import"/>
                        <meta content="69734675" name="octolytics-dimension-user_id">
                         <meta content="hamidghaedi" name="octolytics-dimension-user_login">
                          <meta content="320420414" name="octolytics-dimension-repository_id">
                           <meta content="hamidghaedi/scRNA_seq-analysis" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="320420414" name="octolytics-dimension-repository_network_root_id">
                               <meta content="hamidghaedi/scRNA_seq-analysis" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/hamidghaedi/scRNA_seq-analysis" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e" data-view-component="true" id="dialog-show-dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e-description" aria-labelledby="dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-df9f80cb-52d7-45a2-916a-fa9e8e74b26e-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-8523df82-bc59-473f-b668-e086ef9f44c3">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-5cb37909-ce09-4afc-b576-6da0b0c1c70a">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-79a6a423-025a-4de8-bcc0-25cdf79977aa">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-507b4fb3-d4b2-45c5-9b9e-a23ad4d87d7f">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-5fb9f3be-57bd-4bff-a47c-bf47cf097edd">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-556eb6f1-8561-4a41-9191-f8f4d83100d8">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-2c7ad65b-3db5-43d0-8254-3cfab49a9aa8">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-1374e67c-2976-4963-a282-5c80b91947c4">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               Â© 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: hamidghaedi / scRNA_seq-analysis" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              hamidghaedi
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              scRNA_seq-analysis
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"hamidghaedi","screen_size":"compact"}' data-view-component="true" href="/hamidghaedi">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 hamidghaedi
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"scRNA_seq-analysis","screen_size":"compact"}' data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 scRNA_seq-analysis
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"hamidghaedi","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/hamidghaedi/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/hamidghaedi">
             <span class="AppHeader-context-item-label">
              hamidghaedi
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"scRNA_seq-analysis","screen_size":"full"}' data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis">
             <span class="AppHeader-context-item-label">
              scRNA_seq-analysis
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="gw_1xQqh3g4LAyUllqzua1oL9SGPFvUuvOEi_h6A1RvTLPlNmCwBwbyZfQszOwwwboc1yrs9QL9imlxAs0Ro8w" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="hamidghaedi" data-current-repository="hamidghaedi/scRNA_seq-analysis" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="471qMAWgQxolIwbDsxMWTdfaQ2R4cUX0Q-IzX06-QE-rd30aI-YKb3KPdBpVFuZchX4BzV4WtZsvAHL0qYBbAw" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:hamidghaedi/scRNA_seq-analysis">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump toâ¦" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump toâ¦" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-c7b7b809-fd0d-4dd1-8762-dde91bb8c223" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-34082833-2921-406b-85e6-a4d06d62a5ac" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-34082833-2921-406b-85e6-a4d06d62a5ac">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="WFbj9-PJ3WMeeivWxnfljKZOgkYJem-2ELaAqu4LsvnK5PrPKx_u-a5YJdL4buR9e0VPhVNaieMeNpUQGFq3Qg"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="yvD45cfu-A8dBvyVEixleARQ9zsfXIGOR25hr7vksUCGAppfn2-m1lZvQThY-ahFzQCTLElZw1XkNrAXpKPoFQ">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="PhO-f09W7m9L2yVzBs8VLQsGzOz1bEPJ99Zoo7_VaKQmTvPZGNRHbMjPbPBGnUtHvcHJa9u1o1Gadspxh7q7bA">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="oSJzmEn1s2yBlO15GKVn2WhO31blZ7h4A7oH92kuLIVmdxy6UPJ1DmZPd4TRHhzKuPYHnGDBfkgM2VlS8bpNjw">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-d283e1ea-0125-4793-bf32-14213ac4e023" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-3466c26b-da1b-4bba-bbf3-f715b98d6f1e" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-0fdec68a-3b72-4d95-af4c-ac3b682205eb" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-8d0020f6-673c-4cd5-96e0-d805952870a2" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-3a67a75f-0765-48ef-b96d-da832fb95d04" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-7e23f12c-47b7-4672-834d-2cb48a5980a4" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-03d4d261-9968-4e69-adcb-58c18467d6ba" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-ee849979-8f1e-488f-be89-659c359828dd">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-ee849979-8f1e-488f-be89-659c359828dd" id="tooltip-03d4d261-9968-4e69-adcb-58c18467d6ba" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-e27eebbd-eaa6-4daa-9775-efec0386ec74" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-509ba750-aa01-42f8-9a80-4b5ea41dd3fd">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-509ba750-aa01-42f8-9a80-4b5ea41dd3fd" id="tooltip-e27eebbd-eaa6-4daa-9775-efec0386ec74" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NDY3NH0=--aeba4c45bef63b23f1364063f1b45855d8969b37be54d4dc35df0ff716390e7c" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-6390dcc2-efe4-4f23-9910-d58d367fb4bc" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=320420414">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-b466a417-3cf6-4acd-a67f-28209a861de9" data-view-component="true" id="dialog-show-dialog-b466a417-3cf6-4acd-a67f-28209a861de9" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-b466a417-3cf6-4acd-a67f-28209a861de9-description" aria-labelledby="dialog-b466a417-3cf6-4acd-a67f-28209a861de9-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-b466a417-3cf6-4acd-a67f-28209a861de9">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-b466a417-3cf6-4acd-a67f-28209a861de9-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-b06f9415-b79f-469f-a0fb-8b068323a190" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-19da5132-090a-4111-adbe-338652c0d0e8" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-34752e81-defd-42a3-9a4a-d0f71d69b73c" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-5f5c2306-ba9c-4f18-a94e-0f1db645e4a9" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-8f6abb9a-e8d2-4a70-93ca-f3c0fb76fb3c" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-bd9c4e8e-ba6d-4b30-bf8d-88e2b5ebc165" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-b466a417-3cf6-4acd-a67f-28209a861de9" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-b466a417-3cf6-4acd-a67f-28209a861de9-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-f50b4bc5-4617-467e-88ea-2f0701ce1abb" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-4f9814d0-0975-4e52-84f4-291712751378">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-4fc4070a-7567-48c9-a98d-6ae8c027545c" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-ff89373e-f4ad-4741-b6dc-fbd06ed6c94a">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-63ecfd27-9332-4ccc-89ee-a316c0a60057">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-80e14f1f-85e2-4d49-a244-0e257bbcf286" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-87e38056-0557-4c06-a6af-63bb8271f36d" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-f1847d7c-a279-4789-b729-7df049cc2d24">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-7957dff6-3fde-4a70-8260-5f233afc3aa7">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-9fbe0030-ab51-4cdb-a224-416c5fa57ad7">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-ab8e06bf-9b60-4352-911a-44d96d026c64" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-a910a6a9-274e-4ce0-9697-63355590d6b6" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-463ecf7a-75ae-4158-8656-eb2704aaa232">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-daa81fbb-3f8d-46cd-9832-c13dc896d5d6">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-a1b305d1-75c0-49d0-909d-2ced9e137cc0">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-0993c1ab-2246-47f0-8412-8ee23e3f095f">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /hamidghaedi/scRNA_seq-analysis" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /hamidghaedi/scRNA_seq-analysis/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /hamidghaedi/scRNA_seq-analysis/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /hamidghaedi/scRNA_seq-analysis/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /hamidghaedi/scRNA_seq-analysis/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /hamidghaedi/scRNA_seq-analysis/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/hamidghaedi/scRNA_seq-analysis/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /hamidghaedi/scRNA_seq-analysis/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-list" aria-haspopup="true" aria-labelledby="tooltip-c09675bf-098a-45fa-a6ea-3569118b6e2d" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-button" popovertarget="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-button" id="tooltip-c09675bf-098a-45fa-a6ea-3569118b6e2d" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-button" anchor-offset="normal" data-view-component="true" id="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-79dddaf6-d49e-42da-b03a-987ef3ded5f9-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis" id="item-dc016004-06f0-4017-8313-69aed9ca2ffc" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/issues" id="item-df85bd1d-14d2-44bf-a161-d74ea2efe69c" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/pulls" id="item-498f2d70-c33f-40d0-be08-511a8a03543f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/actions" id="item-efb9737f-7279-4c1b-8301-94f202703dd9" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/projects" id="item-aca65a7e-235e-4ee9-b7eb-dd57e1ed39f0" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/security" id="item-969fed98-6761-436e-9951-841751a3662c" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/pulse" id="item-2fbac92b-73f1-41a0-ac56-2ff4b8477ec3" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-a87f9d28-0742-4d01-bd88-49472508633b" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-3eb56798-f8cd-49df-9c64-9f149532707c" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-3eb56798-f8cd-49df-9c64-9f149532707c" id="tooltip-a87f9d28-0742-4d01-bd88-49472508633b" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NDY3NH0=--aeba4c45bef63b23f1364063f1b45855d8969b37be54d4dc35df0ff716390e7c" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/hamidghaedi/scRNA_seq-analysis" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjY5NzM0Njc1" data-targets="command-palette-scope.tokens" data-text="hamidghaedi" data-type="owner" data-value="hamidghaedi" style="white-space:nowrap;line-height:20px;">
         hamidghaedi
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzMjA0MjA0MTQ=" data-targets="command-palette-scope.tokens" data-text="scRNA_seq-analysis" data-type="repository" data-value="scRNA_seq-analysis" style="white-space:nowrap;line-height:20px;">
         scRNA_seq-analysis
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-0507bf5b-09bc-434d-8fdb-d845dd68e69b" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzMjA0MjA0MTQ=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         Weâve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="hamidghaedi" data-scope-id="MDQ6VXNlcjY5NzM0Njc1" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="scRNA_seq-analysis" data-scope-id="MDEwOlJlcG9zaXRvcnkzMjA0MjA0MTQ=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="hamidghaedi" data-scope-id="MDQ6VXNlcjY5NzM0Njc1" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="scRNA_seq-analysis" data-scope-id="MDEwOlJlcG9zaXRvcnkzMjA0MjA0MTQ=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="4Dsur5KLUwLMDoiHukqyqz24GUtugOJILNbtaOw7eCkwV4SLB1GFpS8zIuApkisNmLvSPuqny3SyfMjdWOpung">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="â­ï¸ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="FzoPZbOY1x4wx40bo6uKnp9vB-Mb4QNsK9bw8W7xQlV3ElOUfLaFeuaMEl9AqU6onoVZkrcTvVaH_Io4jJn2-w">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/69734675?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/hamidghaedi/scRNA_seq-analysis">
            scRNA_seq-analysis
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/320420414/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":320420414,"originating_url":"https://github.com/hamidghaedi/scRNA_seq-analysis","user_id":126681664}}' data-hydro-click-hmac="a8c9ee20be72556f71c367fead7f2d2df70e854507376d728e56a411aa814748" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="3">
              3
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-777aa2d9-c530-4dcc-8b45-8bbaa9c383e9" popover="manual">
              Fork your own copy of hamidghaedi/scRNA_seq-analysis
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-320420414">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/hamidghaedi/scRNA_seq-analysis/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-320420414" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-4afd49">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/hamidghaedi/scRNA_seq-analysis/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="M0xzusM60widw_8ThfBlcJbj5uYhxoiJRajZ1IPmxtcI9QqzXyixpfMZP8-qL70uX3T98BQzDjr9xIQz4y8p2w">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="p8LwfXaETXGsO2sJTSsexrCJXAEHyzIeA8WdKmh_e42ce4l06pYv3MLhq9Vi9MaYeR5HFzI-tK27qcDNCLaUgQ">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (6)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":320420414,"originating_url":"https://github.com/hamidghaedi/scRNA_seq-analysis","user_id":126681664}}' data-hydro-click-hmac="84e2a6afdd21f669f2018975b02de7b0f545981589ee9804c8613cb8a5326b81" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="6 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="6">
                  6
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-320420414-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/hamidghaedi/scRNA_seq-analysis/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-320420414-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-e73838">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/hamidghaedi/scRNA_seq-analysis/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="mX335NtVJ5LDCtkZzvBjVmuNHG36dSY7r18Ahjo9RHWiuOOIOYwF9pqfTRykw-w9gmJuGClWB9rnUBfNXqTu0w">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (6)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":320420414,"originating_url":"https://github.com/hamidghaedi/scRNA_seq-analysis","user_id":126681664}}' data-hydro-click-hmac="17ece72aab88af961b7485dd1a54187f229485f963a0241f7e416470293431ff" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="6 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="6">
                 6
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-320420414-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/hamidghaedi/scRNA_seq-analysis/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-320420414-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-0fdf9a">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":320420414,"originating_url":"https://github.com/hamidghaedi/scRNA_seq-analysis","user_id":126681664}}' data-hydro-click-hmac="44234887b593ceeab8b54ad5a54170dbac159a6a057954ab6509ba74ed029d31" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="On2WXHD4lqNtwA0yJd4OsTjaegzwFEo6Fj67APZ4Bq0ivZombkfjooHTOjj5PojGZ__pGSVO4e40xUn0ocm4NA">
                   <input name="repository_id" type="hidden" value="320420414"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="BZ2_VgHTWSiecca1WWKbKE3s5lFqSAFkWXXsVvLxm80dXbMsH2wsKXJi8b-Fgh1fEsl1RL8SqrB7jh6ipUAlVA">
                   <input name="repository_id" type="hidden" value="320420414"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-4f4b7d44-1d93-418f-95ef-4e5d990b5398" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-4a214026-d797-40d9-a691-99e9fe92ed18" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":320420414,"originating_url":"https://github.com/hamidghaedi/scRNA_seq-analysis","user_id":126681664}}' data-hydro-click-hmac="a8c9ee20be72556f71c367fead7f2d2df70e854507376d728e56a411aa814748" data-view-component="true" group_item="true" href="/hamidghaedi/scRNA_seq-analysis/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-4a214026-d797-40d9-a691-99e9fe92ed18" popover="manual">
              Fork your own copy of hamidghaedi/scRNA_seq-analysis
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/hamidghaedi/scRNA_seq-analysis/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="JcXn2Iq4DkYpA3nwPAPjLDiX4JrF8C6nVLxSRmiOxTsefJ7RFqps60fZuSwT3Dty8QD7jPAFqBTs0A-hCEcqNw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-4b94af93-38a0-4831-86ab-2a20d263bfc1" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":320420414,"originating_url":"https://github.com/hamidghaedi/scRNA_seq-analysis","user_id":126681664}}' data-hydro-click-hmac="84e2a6afdd21f669f2018975b02de7b0f545981589ee9804c8613cb8a5326b81" data-view-component="true" id="icon-button-9e13c29c-5f86-426d-89aa-b70a0dac8a89" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-9e13c29c-5f86-426d-89aa-b70a0dac8a89" id="tooltip-4b94af93-38a0-4831-86ab-2a20d263bfc1" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/hamidghaedi/scRNA_seq-analysis/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="bJY4gYaQ6Yhov03Jfw-CJg3kHuQnD4fH9ku4yos2eM1XUyztZEnL7DEq2cwVPA1N5AtskfQspia-RK-B76_Saw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-988995a3-75c9-477a-b89d-e28e2d27edaa" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":320420414,"originating_url":"https://github.com/hamidghaedi/scRNA_seq-analysis","user_id":126681664}}' data-hydro-click-hmac="17ece72aab88af961b7485dd1a54187f229485f963a0241f7e416470293431ff" data-view-component="true" id="icon-button-fa8f05a4-7515-431b-bfa3-6dc5f95e60bc" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-fa8f05a4-7515-431b-bfa3-6dc5f95e60bc" id="tooltip-988995a3-75c9-477a-b89d-e28e2d27edaa" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           scRNA analysis of public dataset on bladder cancer
          </p>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/hamidghaedi/scRNA_seq-analysis/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              6
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/hamidghaedi/scRNA_seq-analysis/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              3
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              2
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/hamidghaedi/scRNA_seq-analysis/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/hamidghaedi/scRNA_seq-analysis/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/hamidghaedi/scRNA_seq-analysis/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/hamidghaedi/scRNA_seq-analysis/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/hamidghaedi/scRNA_seq-analysis/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/hamidghaedi/scRNA_seq-analysis?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         hamidghaedi/scRNA_seq-analysis
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/hamidghaedi/scRNA_seq-analysis/spoofed_commit_check/75bf23b3ad116c62d55b74eb80b4cd60f924d841">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":320420414,"defaultBranch":"main","name":"scRNA_seq-analysis","ownerLogin":"hamidghaedi","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2020-12-11T08:03:53.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/69734675?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1690344656.0","canEdit":true,"refType":"branch","currentOid":"75bf23b3ad116c62d55b74eb80b4cd60f924d841"},"tree":{"items":[{"name":"images","path":"images","contentType":"directory"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":2,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/hamidghaedi/scRNA_seq-analysis/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/hamidghaedi/scRNA_seq-analysis.git","showCloneWarning":false,"sshUrl":"git@github.com:hamidghaedi/scRNA_seq-analysis.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone hamidghaedi/scRNA_seq-analysis","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/hamidghaedi/scRNA_seq-analysis","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fhamidghaedi%2FscRNA_seq-analysis","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/hamidghaedi/scRNA_seq-analysis/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"102","overviewFiles":[{"displayName":"README.md","repoName":"scRNA_seq-analysis","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003escRNA sequencing analysis\u003c/h1\u003e\u003ca id=\"user-content-scrna-sequencing-analysis\" class=\"anchor\" aria-label=\"Permalink: scRNA sequencing analysis\" href=\"#scrna-sequencing-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis repo presents steps needed to make sense of single-cell RNA sequencing (scRNA) data. I used a scRNA dataset coming from Zhaohui Chen \u003cem\u003eet al.\u003c/em\u003e \u003ca href=\"https://www.nature.com/articles/s41467-020-18916-5#Sec12\" rel=\"nofollow\"\u003epaper\u003c/a\u003e published in Nature Communications 11, Article number: 5077 (2020). The cohort consisted of eight primary bladder tumor tissues (2 low-grade bladder urothelial tumors, and six high-grade bladder urothelial tumors) along with 3 adjacent normal mucosae. In SRA datasets are under BioProject \u003ca href=\"https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA662018\" rel=\"nofollow\"\u003ePRJNA662018\u003c/a\u003e and SRA-explorer can be used to download the data. For practical scRNA-seq analysis, I followed this elegant \u003ca href=\"https://hbctraining.github.io/scRNA-seq_online/schedule/links-to-lessons.html\" rel=\"nofollow\"\u003etutorial\u003c/a\u003e from Harvard Chan Bioinformatics Core.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eNOTE\u003c/strong\u003e If you like to work with \u003ccode\u003ecellranger count\u003c/code\u003e outputs and start analysis using \u003ccode\u003eSeurat\u003c/code\u003e you can download them from this \u003ca href=\"https://figshare.com/articles/dataset/filtered_zip/23834670\" rel=\"nofollow\"\u003elink\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eContents:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#1-data-download-on-computecanada\"\u003eData download on ComputeCanada\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#2-generating-feature-sample-expression-matrix\"\u003eGenerating feature-sample expression matrix\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#4-Background-noise-removal\"\u003eBackground noise removal\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#3-loading-single-cell-rna-seq-count-data\"\u003eLoading single-cell RNA-seq count data\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#5-quality-control\"\u003eQuality control\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#6-normalization-regressing-out-unwanted-variation-and-clustering\"\u003eNormalization, regressing out unwanted variation and clustering\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#7-marker-identification-and-cell-type-assignment\"\u003eMarker identification and cell type assignment\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#8-mibc-vs-nmibc\"\u003eComparing muscle invasive BLCA vs. Non-muscle invasive BLCA\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#9-analyzing-epithelial-epcam--cells\"\u003eAnalyzing epithelial (EPCAM +) cells\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#10-de-and-gsea\"\u003eDE and enrichment analysis\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis#11-trajectory-inference\"\u003eTrajectory inference\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003eRNA velocity analysis. Materials related to RNA velocity analysis have  moved \u003ca href=\"https://github.com/hamidghaedi/scRNA_velocity_analysis\"\u003ehere\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1) Data download on ComputeCanada\u003c/h2\u003e\u003ca id=\"user-content-1-data-download-on-computecanada\" class=\"anchor\" aria-label=\"Permalink: 1) Data download on ComputeCanada\" href=\"#1-data-download-on-computecanada\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n#!/usr/bin/env bash\n\n#!/bin/bash\n#SBATCH --account=#\n#SBATCH --job-name=scRNA__DL\n#SBATCH --qos=privileged\n#SBATCH --nodes=1                # number of Nodes\n#SBATCH --tasks-per-node=4        # number of MPI processes per node\n#SBATCH --mem 8g\n#SBATCH --time 12:00:00\n#SBATCH --output=scRNA_fastq_DL.%J.out\n#SBATCH --error=scRNA_fastq.%J.err\n#SBATCH --mail-type=ALL\n#SBATCH --mail-user=#\n\n\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_1.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_2.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_1.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_2.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_1.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_2.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_1.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_2.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_1.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_2.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_1.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_2.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_1.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_2.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_1.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_2.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_1.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_2.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_1.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_2.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_1.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_2.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_2.fastq.gz\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env bash\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --account=#\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=scRNA__DL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --qos=privileged\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --nodes=1                # number of Nodes\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --tasks-per-node=4        # number of MPI processes per node\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mem 8g\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --time 12:00:00\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --output=scRNA_fastq_DL.%J.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --error=scRNA_fastq.%J.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=ALL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=#\u003c/span\u003e\n\n\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_1.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_2.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_1.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_2.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_1.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_2.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_1.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_2.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_1.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_2.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_1.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_2.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_1.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_2.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_1.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_2.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_1.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_2.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_1.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_2.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_2.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_1.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_1.fastq.gz\ncurl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_2.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_2.fastq.gz\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIt is always recommended to run QC on samples; so using the following code we can run \u003ccode\u003efastqc\u003c/code\u003e on all samples\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n#SBATCH --account= # add your account name \n#SBATCH --job-name=fastqc\n#SBATCH --qos=privileged\n#SBATCH --nodes=12                # number of Nodes\n#SBATCH --tasks-per-node=5        # number of MPI processes per node\n#SBATCH --mem 16g\n#SBATCH --time 24:00:00\n#SBATCH --output=fastqc.%J.out\n#SBATCH --error=fastqc.%J.err\n#SBATCH --mail-type=ALL\n#SBATCH --mail-user=# add your email address \n\n\nmodule load fastqc\n\nfastqc ./fastq/*.fastq.gz --outdir ./qc_raw -f fastq -t 12 --nogroup\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --account= # add your account name \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=fastqc\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --qos=privileged\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --nodes=12                # number of Nodes\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --tasks-per-node=5        # number of MPI processes per node\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mem 16g\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --time 24:00:00\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --output=fastqc.%J.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --error=fastqc.%J.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=ALL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=# add your email address \u003c/span\u003e\n\n\nmodule load fastqc\n\nfastqc ./fastq/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.gz --outdir ./qc_raw -f fastq -t 12 --nogroup\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThere are differences between the sequence length of read 1 and read 2 for each sample; read 1 provides data on Cell barcode \u0026amp; UMI and read 2 has the insert sequence. Files coming from SRA usually come with names like something_1_fastq.gz and something_2_fastq.gz . This \u003ca href=\"https://kb.10xgenomics.com/hc/en-us/articles/115003802691\" rel=\"nofollow\"\u003eblog\u003c/a\u003e is helpful to see what are the naming requirements of fastq files for cellranger tool. Briefly;\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eincompatible file name: SRR9291388_1.fastq.gz\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecompatible file name: SRR9291388_S1_L001_R1_001.fastq.g\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eSo we need to change file names and also set up the directories for \u003ccode\u003ecellranger count\u003c/code\u003e;\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for file in *.fastq.gz\ndo\n  dir=${file%_*}\n  echo $dir\n  mkdir $dir\n  mv $file ./$dir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.gz\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n  dir=\u003cspan class=\"pl-smi\"\u003e${file\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e}\u003c/span\u003e\n  \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$dir\u003c/span\u003e\n  mkdir \u003cspan class=\"pl-smi\"\u003e$dir\u003c/span\u003e\n  mv \u003cspan class=\"pl-smi\"\u003e$file\u003c/span\u003e ./\u003cspan class=\"pl-smi\"\u003e$dir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2) Generating feature-sample expression matrix\u003c/h2\u003e\u003ca id=\"user-content-2-generating-feature-sample-expression-matrix\" class=\"anchor\" aria-label=\"Permalink: 2) Generating feature-sample expression matrix\" href=\"#2-generating-feature-sample-expression-matrix\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe next step is to run \u003ccode\u003ecellranger count\u003c/code\u003e on each sample. There are different types of scRNA libs with different fastq files. This \u003ca href=\"https://www.10xgenomics.com/support/single-cell-gene-expression/documentation/steps/sequencing/sequencing-requirements-for-single-cell-3\" rel=\"nofollow\"\u003eblog\u003c/a\u003e provides details on the type of libs from 10X genomics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRunning \u003ccode\u003ecellranger\u003c/code\u003e on a cluster with \u003ccode\u003eSLURM\u003c/code\u003e as job scheduler is not an easy task. The following is what I came up with working best for me working on ComputeCanada cluster:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n#SBATCH --account=#\n#SBATCH -J test_cellranger\n#SBATCH --export=ALL\n#SBATCH --ntasks=1\n#SBATCH --cpus-per-task=16\n#SBATCH --signal=2\n#SBATCH --no-requeue\n### Alternatively: --ntasks=1 --cpus-per-task={NUM_THREADS}\n###   Consult with your cluster administrators to find the combination that\n###   works best for single-node, multi-threaded applications on your system.\n#SBATCH --mem=250G\n#SBATCH --time 48:00:00\n#SBATCH --output=test_cellranger.%J.out\n#SBATCH --error=test_cellranger.%J.err\n#SBATCH --mail-type=ALL\n#SBATCH --mail-user=#\n\n\nmodule load cellranger\n\ncd /home/ghaedi/projects/def-gooding-ab/ghaedi/sc/raw\nfor d in *\ndo\necho $d\nID=${d:0:11}\ncellranger count --id=$ID \\\n                 --transcriptome=/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/refdata-gex-GRCh38-2020-A \\\n                 --fastqs=$d\n                 --chemistry=SC3Pv3\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --account=#\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -J test_cellranger\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --export=ALL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --ntasks=1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --cpus-per-task=16\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --signal=2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --no-requeue\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Alternatively: --ntasks=1 --cpus-per-task={NUM_THREADS}\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##   Consult with your cluster administrators to find the combination that\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##   works best for single-node, multi-threaded applications on your system.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mem=250G\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --time 48:00:00\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --output=test_cellranger.%J.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --error=test_cellranger.%J.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=ALL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=#\u003c/span\u003e\n\n\nmodule load cellranger\n\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /home/ghaedi/projects/def-gooding-ab/ghaedi/sc/raw\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$d\u003c/span\u003e\nID=\u003cspan class=\"pl-smi\"\u003e${d\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e0\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e11}\u003c/span\u003e\ncellranger count --id=\u003cspan class=\"pl-smi\"\u003e$ID\u003c/span\u003e \\\n                 --transcriptome=/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/refdata-gex-GRCh38-2020-A \\\n                 --fastqs=\u003cspan class=\"pl-smi\"\u003e$d\u003c/span\u003e\n                 --chemistry=SC3Pv3\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe next steps are mainly based on Harvard Chan Bioinformatics Core materials on \u003ca href=\"https://hbctraining.github.io/scRNA-seq_online/schedule/links-to-lessons.html\" rel=\"nofollow\"\u003escRNA-seq analysis\u003c/a\u003e training.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3) Background noise removal\u003c/h2\u003e\u003ca id=\"user-content-3-background-noise-removal\" class=\"anchor\" aria-label=\"Permalink: 3) Background noise removal\" href=\"#3-background-noise-removal\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBackground removal is crucial in single-cell RNA sequencing (scRNA-seq) analysis due to the presence of contaminants such as cell-free \"ambient\" RNA and chimeric cDNA molecules resulting from barcode swapping.\nBarcode swapping can happen when cDNA molecules from different beads, which carry different barcodes, mix together during the amplification step. This can occur due to unremoved oligonucleotides from other beads or incomplete extension of PCR products. As a result, chimeric cDNA molecules are formed, where the barcode and UMI sequences are \"swapped\" between different beads. When these chimeric molecules are sequenced, the assigned barcode does not match the original bead, leading to the incorrect assignment of the cDNA to the corresponding cell.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThese contaminants contribute to background noise, which can negatively impact data analysis. Background noise reduces the separability of cell-type clusters, interferes with differential expression analysis, and confounds comparisons between samples. To address this, algorithms like SoupX, DecontX, and CellBender estimate and correct for background noise using marker genes, empty droplets, and modeling of barcode swapping. Evaluating method performance involves assessing if the model effectively removes expression from other cell types using datasets with known cell type profiles and exclusive marker genes. However, the impact of background correction on expression level shifts induced by background noise remains a challenge in scRNA-seq analysis.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs recommended by \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02978-x\" rel=\"nofollow\"\u003ePhilipp Janssen et al,2023\u003c/a\u003e:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eIn summary, for marker gene analysis, we would always recommend background removal, but for classification, clustering and pseudotime analyses, we would only recommend background removal when background noise levels are high\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn this repo, I won't be using the cellbender filtered matrix, as the main focus of the analysis is cell clustering, rather than marker identification. The following is a fully functional cellbender code for the given samples\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Running cellbender image on co pute canada \n##pulling the image (creating sif file):\nmodule load apptainer\napptainer pull cellbender.sif docker://us.gcr.io/broad-dsde-methods/cellbender:0.2.2\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Running cellbender image on co pute canada \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#pulling the image (creating sif file):\u003c/span\u003e\nmodule load apptainer\napptainer pull cellbender.sif docker://us.gcr.io/broad-dsde-methods/cellbender:0.2.2\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSbatch script to run cellbender on Graham with GPU\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash\n#SBATCH --account=def-gooding-ab\n#SBATCH --job-name=cellBender\n#SBATCH --nodes=1                # number of Nodes\n#SBATCH --gpus-per-node=p100:2\n#SBATCH --ntasks-per-node=32       \n#SBATCH --mem=127000M\n#SBATCH --time 08:00:00\n#SBATCH --output=cellBender.%J.out\n#SBATCH --error=cellBender.%J.err\n#SBATCH --mail-type=ALL\n#SBATCH --mail-user=qaedi.65@gmail.com\n\nmodule load apptainer\n\n# Define the input and output directories\ninput_dir=\u0026quot;/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/filtered\u0026quot;\noutput_dir=\u0026quot;/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/cellBender_filtered\u0026quot;\n\n# Iterate over the sample directories\nfor sample_dir in \u0026quot;$input_dir\u0026quot;/*; do\n    if [[ -d \u0026quot;$sample_dir\u0026quot; ]]; then\n        sample_name=$(basename \u0026quot;$sample_dir\u0026quot;)\n        echo \u0026quot;Processing sample: $sample_name\u0026quot;\n\n        # Create a directory for the current sample within the output directory\n        sample_output_dir=\u0026quot;$output_dir/$sample_name\u0026quot;\n        mkdir -p \u0026quot;$sample_output_dir\u0026quot;\n\n        # Run cellbender remove-background command for the current sample\n        apptainer run -C -W ${SLURM_TMPDIR} --nv --rocm \\\n        --bind \u0026quot;$input_dir\u0026quot;:\u0026quot;/mnt/filtered\u0026quot; \\\n        --bind \u0026quot;$sample_output_dir\u0026quot;:\u0026quot;/mnt/sample_output\u0026quot; cellbender.sif \\\n        cellbender remove-background \\\n        --input \u0026quot;/mnt/filtered/$sample_name\u0026quot; \\\n        --output \u0026quot;/mnt/sample_output/$sample_name.h5\u0026quot; \\\n        --expected-cells 7000 \\\n        --total-droplets-included 20000 \\\n        --fpr 0.01 \\\n        --epochs 150 \\\n        --cuda\n\n        echo \u0026quot;Completed processing sample: $sample_name\u0026quot;\n        echo\n    fi\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --account=def-gooding-ab\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=cellBender\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --nodes=1                # number of Nodes\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --gpus-per-node=p100:2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --ntasks-per-node=32       \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mem=127000M\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --time 08:00:00\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --output=cellBender.%J.out\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --error=cellBender.%J.err\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=ALL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=qaedi.65@gmail.com\u003c/span\u003e\n\nmodule load apptainer\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Define the input and output directories\u003c/span\u003e\ninput_dir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/filtered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\noutput_dir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/cellBender_filtered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Iterate over the sample directories\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esample_dir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$input_dir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e [[ \u003cspan class=\"pl-k\"\u003e-d\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$sample_dir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ]]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\n        sample_name=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ebasename \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$sample_dir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eProcessing sample: \u003cspan class=\"pl-smi\"\u003e$sample_name\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create a directory for the current sample within the output directory\u003c/span\u003e\n        sample_output_dir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$output_dir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$sample_name\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n        mkdir -p \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$sample_output_dir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n        \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Run cellbender remove-background command for the current sample\u003c/span\u003e\n        apptainer run -C -W \u003cspan class=\"pl-smi\"\u003e${SLURM_TMPDIR}\u003c/span\u003e --nv --rocm \\\n        --bind \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$input_dir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e:\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/mnt/filtered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n        --bind \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$sample_output_dir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e:\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/mnt/sample_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e cellbender.sif \\\n        cellbender remove-background \\\n        --input \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/mnt/filtered/\u003cspan class=\"pl-smi\"\u003e$sample_name\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n        --output \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/mnt/sample_output/\u003cspan class=\"pl-smi\"\u003e$sample_name\u003c/span\u003e.h5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n        --expected-cells 7000 \\\n        --total-droplets-included 20000 \\\n        --fpr 0.01 \\\n        --epochs 150 \\\n        --cuda\n\n        \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCompleted processing sample: \u003cspan class=\"pl-smi\"\u003e$sample_name\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n        \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4) Loading single-cell RNA-seq count data\u003c/h2\u003e\u003ca id=\"user-content-4-loading-single-cell-rna-seq-count-data\" class=\"anchor\" aria-label=\"Permalink: 4) Loading single-cell RNA-seq count data\" href=\"#4-loading-single-cell-rna-seq-count-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Setup the Seurat Object\n\nlibrary(tidyverse)\nlibrary(Seurat)\nlibrary(patchwork)\nlibrary(cowplot)\n\n# create list of samples\nsamples \u0026lt;- list.files(\u0026quot;~/scRNA/filtered/\u0026quot;)\n#samples \u0026lt;- samples[grepl('^filtered',samples,perl=T)]\n\n# read files inot Seurat objects\nfor (file in samples){\n  print(paste0(file))\n  seurat_data \u0026lt;- Read10X(data.dir = paste0(\u0026quot;~/scRNA/filtered/\u0026quot;, file))\n  seurat_obj \u0026lt;- CreateSeuratObject(counts = seurat_data, \n                                   min.features = 100, \n                                   project = file)\n  assign(file, seurat_obj)\n}\n\n# now merging all objects inot one Seurat obj\n\nmerged_seurat \u0026lt;- merge(x = SRR12603780, \n                       y = c(SRR12603781,\n                             SRR12603782,\n                             SRR12603783,\n                             SRR12603784,\n                             SRR12603785,\n                             SRR12603786,\n                             SRR12603787,\n                             SRR12603788,\n                             SRR12603789,\n                             SRR12603790),\n                       #Because the same cell IDs can be used for different samples, we add a sample-specific prefix \n                       # to each of our cell IDs using the add.cell.id argument.\n                       add.cell.id = samples)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Setup the Seurat Object\u003c/span\u003e\n\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyverse\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eSeurat\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003epatchwork\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ecowplot\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e create list of samples\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e list.files(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e~/scRNA/filtered/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esamples \u0026lt;- samples[grepl('^filtered',samples,perl=T)]\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e read files inot Seurat objects\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e){\n  print(paste0(\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e))\n  \u003cspan class=\"pl-smi\"\u003eseurat_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Read10X(\u003cspan class=\"pl-v\"\u003edata.dir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e~/scRNA/filtered/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e))\n  \u003cspan class=\"pl-smi\"\u003eseurat_obj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e CreateSeuratObject(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_data\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003emin.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003eproject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e)\n  assign(\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eseurat_obj\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now merging all objects inot one Seurat obj\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSRR12603780\u003c/span\u003e, \n                       \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-smi\"\u003eSRR12603781\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603782\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603783\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603784\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603785\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603786\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603787\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603788\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603789\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603790\u003c/span\u003e),\n                       \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eBecause the same cell IDs can be used for different samples, we add a sample-specific prefix \u003c/span\u003e\n                       \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to each of our cell IDs using the add.cell.id argument.\u003c/span\u003e\n                       \u003cspan class=\"pl-v\"\u003eadd.cell.id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5) Quality control\u003c/h2\u003e\u003ca id=\"user-content-5-quality-control\" class=\"anchor\" aria-label=\"Permalink: 5) Quality control\" href=\"#5-quality-control\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThere are columns in the metadata:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eorig.ident: this column will contain the sample identity if known. It will default to the value we provided for the project argument when loading in the data\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003enCount_RNA: represents the number of UMIs per cell. UMI (unique molecular identifiers) is used to determine whether a read is a biological or technical duplicate (PCR duplicate). Reads with different UMIs mapping to the same transcript were derived from different molecules and are biological duplicates - each read should be counted. Reads with the same UMI originated from the same molecule and are technical duplicates - the UMIs should be collapsed to be counted as a single read.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003enFeature_RNA: represents the number of genes detected per cell\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eRecommended features to add to metadata:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003enumber of genes detected per UMI (or novelty score): more genes detected per UMI, more complex our data\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003emitochondrial ratio: this metric will give us a percentage of cell reads originating from the mitochondrial genes (coming from dying cells)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Explore merged metadata\nView(merged_seurat@meta.data)\n\n\n#Add number of genes per UMI for each cell to metadata\nmerged_seurat$log10GenesPerUMI \u0026lt;- log10(merged_seurat$nFeature_RNA) / log10(merged_seurat$nCount_RNA)\n\n# Compute percent mito ratio\nmerged_seurat$mitoRatio \u0026lt;- PercentageFeatureSet(object = merged_seurat, pattern = \u0026quot;^MT-\u0026quot;)\nmerged_seurat$mitoRatio \u0026lt;- merged_seurat@meta.data$mitoRatio / 100\n\n# Create metadata dataframe\nmetadata \u0026lt;- merged_seurat@meta.data\n# Add cell IDs to metadata\nmetadata$cells \u0026lt;- rownames(metadata)\n# adding sample type to metadata. The orginal file could be download from SRA explorer\n\nSampleType \u0026lt;- c(\u0026quot;BLCA\u0026quot;, \u0026quot;BLCA\u0026quot;, \u0026quot;Normal\u0026quot;, \u0026quot;BLCA\u0026quot;, \u0026quot;BLCA\u0026quot;, \u0026quot;BLCA\u0026quot;, \u0026quot;BLCA\u0026quot;, \u0026quot;BLCA\u0026quot;, \u0026quot;BLCA\u0026quot;, \u0026quot;Normal\u0026quot;, \u0026quot;Normal\u0026quot;)\n# sample type with grade (Not tested)\n#SampleType \u0026lt;- c(\u0026quot;BLCA_LG\u0026quot;, \u0026quot;BLCA_LG\u0026quot;, \u0026quot;Normal\u0026quot;, \u0026quot;BLCA_HG\u0026quot;, \u0026quot;BLCA_HG\u0026quot;, \u0026quot;BLCA_HG\u0026quot;, \u0026quot;BLCA_HG\u0026quot;, \u0026quot;BLCA_HG\u0026quot;, \u0026quot;BLCA_HG\u0026quot;, \u0026quot;Normal\u0026quot;, \u0026quot;Normal\u0026quot;)\n\nnames(SampleType) \u0026lt;- c(\u0026quot;SRR12603789\u0026quot;, \u0026quot;SRR12603790\u0026quot;, \u0026quot;SRR12603788\u0026quot;, \u0026quot;SRR12603787\u0026quot;, \u0026quot;SRR12603786\u0026quot;, \u0026quot;SRR12603785\u0026quot;, \u0026quot;SRR12603784\u0026quot;, \u0026quot;SRR12603783\u0026quot;, \u0026quot;SRR12603782\u0026quot;, \u0026quot;SRR12603781\u0026quot;, \u0026quot;SRR12603780\u0026quot;)\n\nmetadata$sampleType \u0026lt;- stringr::str_replace_all(metadata$orig.ident, SampleType)\n\n\n# Rename columns\nmetadata \u0026lt;- metadata %\u0026gt;%\n  dplyr::rename(seq_folder = orig.ident,\n                nUMI = nCount_RNA,\n                nGene = nFeature_RNA,\n                sample = sampleType)\n\n# Add metadata back to Seurat object\nmerged_seurat@meta.data \u0026lt;- metadata\n\n# Create .RData object to load at any time\nsave(merged_seurat, file=\u0026quot;~/scRNA/merged_filtered_seurat.RData\u0026quot;)\n\n#\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Explore merged metadata\u003c/span\u003e\nView(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eAdd number of genes per UMI for each cell to metadata\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elog10GenesPerUMI\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e log10(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enFeature_RNA\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e log10(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enCount_RNA\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Compute percent mito ratio\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e PercentageFeatureSet(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epattern\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e^MT-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create metadata dataframe\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Add cell IDs to metadata\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e adding sample type to metadata. The orginal file could be download from SRA explorer\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eSampleType\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sample type with grade (Not tested)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSampleType \u0026lt;- c(\"BLCA_LG\", \"BLCA_LG\", \"Normal\", \"BLCA_HG\", \"BLCA_HG\", \"BLCA_HG\", \"BLCA_HG\", \"BLCA_HG\", \"BLCA_HG\", \"Normal\", \"Normal\")\u003c/span\u003e\n\nnames(\u003cspan class=\"pl-smi\"\u003eSampleType\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603789\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603790\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603788\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603787\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603786\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603785\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603784\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603783\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603782\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603781\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603780\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esampleType\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003estringr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003estr_replace_all(\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eorig.ident\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSampleType\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Rename columns\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003erename(\u003cspan class=\"pl-v\"\u003eseq_folder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eorig.ident\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003enUMI\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enCount_RNA\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003enGene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enFeature_RNA\u003c/span\u003e,\n                \u003cspan class=\"pl-v\"\u003esample\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esampleType\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Add metadata back to Seurat object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create .RData object to load at any time\u003c/span\u003e\nsave(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e~/scRNA/merged_filtered_seurat.RData\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCell counts per sample\u003c/h4\u003e\u003ca id=\"user-content-cell-counts-per-sample\" class=\"anchor\" aria-label=\"Permalink: Cell counts per sample\" href=\"#cell-counts-per-sample\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualize the number of cell counts per sample\nbqcc \u0026lt;- metadata %\u0026gt;% \n  ggplot(aes(x=seq_folder, fill=sample)) + \n  geom_bar() +\n  theme_classic() +\n  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +\n  theme(plot.title = element_text(hjust=0.5, face=\u0026quot;bold\u0026quot;)) +\n  ggtitle(\u0026quot;NCells before QC\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualize the number of cell counts per sample\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ebqcc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  geom_bar() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme(\u003cspan class=\"pl-v\"\u003eaxis.text.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003eangle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNCells before QC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cell_count_per_sample_before_QC.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cell_count_per_sample_before_QC.png\" alt=\"cell_count_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eUMI per cell\u003c/h4\u003e\u003ca id=\"user-content-umi-per-cell\" class=\"anchor\" aria-label=\"Permalink: UMI per cell\" href=\"#umi-per-cell\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTypically, we expect the UMI counts per cell to be higher than 500, which is the lower limit of the expected range. If the UMI counts range between 500-1000, the data is still usable, but deeper sequencing may have been beneficial for these cells.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualize the number UMIs/transcripts per cell\nmetadata %\u0026gt;% \n  ggplot(aes(x=nUMI, fill= sample)) + \n  geom_density(alpha = 0.5) + \n  scale_x_log10() + \n  theme_classic() +\n  ylab(\u0026quot;Cell density\u0026quot;) +\n  facet_wrap(~seq_folder) +\n  geom_vline(xintercept = 1000) +\n  labs(fill = \u0026quot;Sample\u0026quot;)\n  \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualize the number UMIs/transcripts per cell\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enUMI\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  geom_density(\u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  scale_x_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  theme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  ylab(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell density\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  facet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_vline(\u003cspan class=\"pl-v\"\u003exintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  labs(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nUMI.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nUMI.png\" alt=\"cell_count_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\nSo the cells have way more than 1K UMI!\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGenes detected per cell\u003c/h4\u003e\u003ca id=\"user-content-genes-detected-per-cell\" class=\"anchor\" aria-label=\"Permalink: Genes detected per cell\" href=\"#genes-detected-per-cell\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIn scRNA-seq, the number of genes detected per cell is a crucial quality metric that we expect to be similar to the UMI detection, albeit slightly lower. For high-quality data, the proportional histogram of genes detected per cell should show a single peak that represents encapsulated cells. However, if there is a small shoulder or a bimodal distribution to the left of the main peak, this could indicate a few things. It could be due to some failed cells or biologically different cell types, such as quiescent cell populations or less complex cells of interest. For instance, larger cells or different cell types may have higher gene counts.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualize the distribution of genes detected per cell via histogram\nmetadata %\u0026gt;% \n  ggplot(aes(x=nGene, fill= sample)) + \n  geom_density(alpha = 0.5) + \n  scale_x_log10() + \n  theme_classic() +\n  ylab(\u0026quot;Cell density\u0026quot;) +\n  facet_wrap(~seq_folder) +\n  geom_vline(xintercept = 500) +\n  labs(fill = \u0026quot;Sample\u0026quot;)\n  \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualize the distribution of genes detected per cell via histogram\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enGene\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  geom_density(\u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  scale_x_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  theme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  ylab(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell density\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  facet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_vline(\u003cspan class=\"pl-v\"\u003exintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e500\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  labs(\u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n  \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nGene_before_QC.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nGene_before_QC.png\" alt=\"gene_count_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eNovelty score\u003c/h4\u003e\u003ca id=\"user-content-novelty-score\" class=\"anchor\" aria-label=\"Permalink: Novelty score\" href=\"#novelty-score\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe novelty score, computed as the ratio of nGenes over nUMI, measures the complexity of RNA species in each cell. A low number of genes detected in a cell with many captured transcripts (high nUMI) indicates low complexity or novelty. This could be due to an artifact, contamination, or a specific cell type (e.g. red blood cells). A good-quality cell typically has a novelty score above 0.80.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualize the overall complexity of the gene expression by visualizing the genes detected per UMI (novelty score)\nmetadata %\u0026gt;%\n  ggplot(aes(x=log10GenesPerUMI, fill=sample)) +\n  geom_density(alpha = 0.5) +\n  theme_classic() +\n  facet_wrap(~seq_folder) +\n  xlab(\u0026quot;Novelty Score\u0026quot;) +\n  geom_vline(xintercept = 0.8)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualize the overall complexity of the gene expression by visualizing the genes detected per UMI (novelty score)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elog10GenesPerUMI\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_density(\u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  facet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  xlab(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNovelty Score\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_vline(\u003cspan class=\"pl-v\"\u003exintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/novelt_score_before_QC.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/novelt_score_before_QC.png\" alt=\"gene_count_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMitochondrial gene expression detected per cell\u003c/h4\u003e\u003ca id=\"user-content-mitochondrial-gene-expression-detected-per-cell\" class=\"anchor\" aria-label=\"Permalink: Mitochondrial gene expression detected per cell\" href=\"#mitochondrial-gene-expression-detected-per-cell\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHigh levels of expression from mitochondria indicate dying or dead cells. Basically, quality samples are those that surpass 0.2 mitochondria ratio mark.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualize the distribution of mitochondrial gene expression detected per cell\nmetadata %\u0026gt;%\n  ggplot(aes(x=mitoRatio, fill=sample)) + \n  geom_density(alpha = 0.5) + \n  scale_x_log10() + \n  scale_x_continuous(labels = function(x) sprintf(\u0026quot;%.1f\u0026quot;, x)) + \n  theme_classic() +\n  facet_wrap(~seq_folder) +\n  geom_vline(xintercept = 0.2)\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualize the distribution of mitochondrial gene expression detected per cell\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  geom_density(\u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  scale_x_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  scale_x_continuous(\u003cspan class=\"pl-en\"\u003elabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e) sprintf(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%.1f\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  theme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  facet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_vline(\u003cspan class=\"pl-v\"\u003exintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e)\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/mitoRatio_before_qc.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/mitoRatio_before_qc.png\" alt=\"mitoRatio_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eJoint filtering: nUMI, nGene and mitoRatio\u003c/h4\u003e\u003ca id=\"user-content-joint-filtering-numi-ngene-and-mitoratio\" class=\"anchor\" aria-label=\"Permalink: Joint filtering: nUMI, nGene and mitoRatio\" href=\"#joint-filtering-numi-ngene-and-mitoratio\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualize the correlation between genes detected and number of UMIs and determine whether strong presence of cells with low numbers of genes/UMIs\n\nmetadata %\u0026gt;% \n  \tggplot(aes(x=nUMI, y=nGene, color=mitoRatio)) + \n  \tgeom_point() + \n  \tscale_colour_gradient(low = \u0026quot;gray90\u0026quot;, high = \u0026quot;black\u0026quot;) +\n  \tstat_smooth(method=lm) +\n  \tscale_x_log10() + \n  \tscale_y_log10() + \n  \ttheme_classic() +\n  \tgeom_vline(xintercept = 1000) +\n  \tgeom_hline(yintercept = 500) +\n  \tfacet_wrap(~seq_folder)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualize the correlation between genes detected and number of UMIs and determine whether strong presence of cells with low numbers of genes/UMIs\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \tggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enUMI\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enGene\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \tgeom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \tscale_colour_gradient(\u003cspan class=\"pl-v\"\u003elow\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egray90\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehigh\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tstat_smooth(\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elm\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tscale_x_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \tscale_y_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \ttheme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tgeom_vline(\u003cspan class=\"pl-v\"\u003exintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tgeom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e500\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tfacet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nUMI_nGene_mitoRatio.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nUMI_nGene_mitoRatio.png\" alt=\"mitoRatio_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThere are samples that shows high-quality cells; high nUMI, high nGene, low number of cells with high mitoRatio and also there are some samples that would clearly benefit from filtering, as they have low-quality cells. We expect to see that dying cells to show high level of mitoRatio and low nUMI and nGene .\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBasically, it is not uncommon to observe cells with high numbers of UMIs and nGene, but also high mitoRatio. These cells may be stressed or damaged, but they could also represent a heterogeneous population of cells with distinct metabolic states.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo investigate the potential cause of high mitochondrial expression ratios, it is important to examine the expression of specific mitochondrial genes and compare them to other genes in the cell. If the expression of mitochondrial genes is elevated relative to other genes, this could suggest mitochondrial dysfunction. Additionally, examining the expression of other stress or damage markers, such as heat shock proteins or cell cycle genes, can also provide insight into the health and state of the cell.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFiltering\u003c/h3\u003e\u003ca id=\"user-content-filtering\" class=\"anchor\" aria-label=\"Permalink: Filtering\" href=\"#filtering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCell-level filtering\u003c/h4\u003e\u003ca id=\"user-content-cell-level-filtering\" class=\"anchor\" aria-label=\"Permalink: Cell-level filtering\" href=\"#cell-level-filtering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e-nUMI \u0026gt; 1000\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e-nGene \u0026gt; 500\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e-log10GenesPerUMI \u0026gt; 0.8\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e-mitoRatio \u0026lt; 0.2\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Filter out low-quality cells using selected thresholds - these will change with the experiment\nfiltered_seurat \u0026lt;- subset(merged_seurat, \n                          subset= nUMI \u0026gt;= 1000 \u0026amp;\n                          nGene \u0026gt;= 500 \u0026amp;\n                          nGene \u0026lt;= 6000 \u0026amp; \n                          log10GenesPerUMI \u0026gt; 0.80 \u0026amp; \n                          mitoRatio \u0026lt; 0.10)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Filter out low-quality cells using selected thresholds - these will change with the experiment\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e subset(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \n                          \u003cspan class=\"pl-v\"\u003esubset\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enUMI\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e\n                          \u003cspan class=\"pl-smi\"\u003enGene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e500\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e\n                          \u003cspan class=\"pl-smi\"\u003enGene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6000\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \n                          \u003cspan class=\"pl-smi\"\u003elog10GenesPerUMI\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.80\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \n                          \u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.10\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGene-level filtering\u003c/h4\u003e\u003ca id=\"user-content-gene-level-filtering\" class=\"anchor\" aria-label=\"Permalink: Gene-level filtering\" href=\"#gene-level-filtering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eKeep only genes that are expressed in 100 or more cells (usually that is 10)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Extract counts\ncounts \u0026lt;- GetAssayData(object = filtered_seurat, slot = \u0026quot;counts\u0026quot;)\n\n# Output a logical matrix specifying for each gene whether or not there are more than zero counts per cell\nnonzero \u0026lt;- counts \u0026gt; 0\n# Sums all TRUE values and returns TRUE if more than 100 TRUE values per gene\nkeep_genes \u0026lt;- Matrix::rowSums(nonzero) \u0026gt;= 100\n\n# Only keeping those genes expressed in more than 100 cells\nfiltered_counts \u0026lt;- counts[keep_genes, ]\n# Reassign to filtered Seurat object\nfiltered_seurat \u0026lt;- CreateSeuratObject(filtered_counts, meta.data = filtered_seurat@meta.data)\n\n# Create.RData object to load at any time\nsave(filtered_seurat, file=\u0026quot;seurat_filtered.RData\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract counts\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e GetAssayData(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eslot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecounts\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Output a logical matrix specifying for each gene whether or not there are more than zero counts per cell\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003enonzero\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Sums all TRUE values and returns TRUE if more than 100 TRUE values per gene\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ekeep_genes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eMatrix\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003erowSums(\u003cspan class=\"pl-smi\"\u003enonzero\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Only keeping those genes expressed in more than 100 cells\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003efiltered_counts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ekeep_genes\u003c/span\u003e, ]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Reassign to filtered Seurat object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e CreateSeuratObject(\u003cspan class=\"pl-smi\"\u003efiltered_counts\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create.RData object to load at any time\u003c/span\u003e\nsave(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eseurat_filtered.RData\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRe-assess QC metrics\u003c/h4\u003e\u003ca id=\"user-content-re-assess-qc-metrics\" class=\"anchor\" aria-label=\"Permalink: Re-assess QC metrics\" href=\"#re-assess-qc-metrics\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n# Save filtered subset to new metadata\nmetadata_clean \u0026lt;- filtered_seurat@meta.data\n\n# to see a drop in filtering cells:\n\nmet_before \u0026lt;- data.frame(unclass(table(metadata$seq_folder)))\nmet_before$QCgroup \u0026lt;- \u0026quot;before\u0026quot;\nmet_before$cell\u0026lt;- rownames(met_before)\nnames(met_before)[1] \u0026lt;- 'count'\n\nmet_after \u0026lt;- data.frame(unclass(table(metadata_clean$seq_folder)))\nmet_after$QCgroup \u0026lt;- \u0026quot;after\u0026quot;\nmet_after$cell\u0026lt;- rownames(met_after)\nnames(met_after)[1] \u0026lt;- 'count'\n# count\ncell_count \u0026lt;- data.frame(rbind(met_before, met_after))\n\n                                \n# visualization :\ncell_count %\u0026gt;% ggplot(aes(x=cell, y=count, fill=QCgroup)) + \n  geom_bar(stat=\u0026quot;identity\u0026quot;, position=position_dodge()) +\n  theme_classic() +\n  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +\n  theme(plot.title = element_text(hjust=0.5, face=\u0026quot;bold\u0026quot;)) +\n  scale_fill_manual(values = c(\u0026quot;#808080\u0026quot;, \u0026quot;#FFBF00\u0026quot;)) +\n  xlab(\u0026quot;samples\u0026quot;) +\n  ggtitle(\u0026quot;nCells count before and after QC\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Save filtered subset to new metadata\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata_clean\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to see a drop in filtering cells:\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003emet_before\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(unclass(table(\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e)))\n\u003cspan class=\"pl-smi\"\u003emet_before\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eQCgroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebefore\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emet_before\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecell\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003emet_before\u003c/span\u003e)\nnames(\u003cspan class=\"pl-smi\"\u003emet_before\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecount\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003emet_after\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(unclass(table(\u003cspan class=\"pl-smi\"\u003emetadata_clean\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e)))\n\u003cspan class=\"pl-smi\"\u003emet_after\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eQCgroup\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eafter\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emet_after\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecell\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003emet_after\u003c/span\u003e)\nnames(\u003cspan class=\"pl-smi\"\u003emet_after\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecount\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e count\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecell_count\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(rbind(\u003cspan class=\"pl-smi\"\u003emet_before\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emet_after\u003c/span\u003e))\n\n                                \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e visualization :\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecell_count\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecell\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecount\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eQCgroup\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  geom_bar(\u003cspan class=\"pl-v\"\u003estat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eidentity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eposition\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003eposition_dodge()) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme(\u003cspan class=\"pl-v\"\u003eaxis.text.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003eangle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  scale_fill_manual(\u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#808080\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e#FFBF00\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  xlab(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esamples\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enCells count before and after QC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/before_after_cell_count.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/before_after_cell_count.png\" alt=\"cell_countbefore_after_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Visualize the correlation between genes detected and the number of UMIs and determine whether the strong presence of cells with low numbers of genes/UMIs\n\nmetadata_clean %\u0026gt;% \n  \tggplot(aes(x=nUMI, y=nGene, color=mitoRatio)) + \n  \tgeom_point() + \n  \tscale_colour_gradient(low = \u0026quot;gray90\u0026quot;, high = \u0026quot;black\u0026quot;) +\n  \tstat_smooth(method=lm) +\n  \tscale_x_log10() + \n  \tscale_y_log10() + \n  \ttheme_classic() +\n  \tgeom_vline(xintercept = 1000) +\n  \tgeom_hline(yintercept = 500) +\n  \tfacet_wrap(~seq_folder)\n  \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualize the correlation between genes detected and the number of UMIs and determine whether the strong presence of cells with low numbers of genes/UMIs\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003emetadata_clean\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \tggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enUMI\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enGene\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \tgeom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \tscale_colour_gradient(\u003cspan class=\"pl-v\"\u003elow\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egray90\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehigh\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tstat_smooth(\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elm\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tscale_x_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \tscale_y_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n  \ttheme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tgeom_vline(\u003cspan class=\"pl-v\"\u003exintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tgeom_hline(\u003cspan class=\"pl-v\"\u003eyintercept\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e500\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \tfacet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e)\n  \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nUMI_nGene_mitoRatio_after.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nUMI_nGene_mitoRatio_after.png\" alt=\"nUMI_nGene_mitoRatio_plot\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e6) Normalization, regressing out unwanted variation and clustering\u003c/h2\u003e\u003ca id=\"user-content-6-normalization-regressing-out-unwanted-variation-and-clustering\" class=\"anchor\" aria-label=\"Permalink: 6) Normalization, regressing out unwanted variation and clustering\" href=\"#6-normalization-regressing-out-unwanted-variation-and-clustering\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe ultimate goal is to define clusters of cells and identify cell types in the samples. To achieve this, there are several steps:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e1-Identify unwanted variability by exploring data and covariates such as cell cycle and mitochondrial gene expression.\nBoth biological sources of variation (e.g. effect of cell cycle on transcriptome) and technical sources should be explored and accounted for. In an early version of Seurat, one needs to normalize data, find variable features, and then scale data while setting a variable like mitochondrial contamination or cell cycle stage to be regressed out. So here is the code for doing these steps to mitigate cell cycle stage effects on the dataset, however, a newer function in Seurat has automated all of these steps (\u003ccode\u003eSCTtansform()\u003c/code\u003e).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e2-Normalize and remove unwanted variability using Seurat's \u003ccode\u003esctransform\u003c/code\u003e function. The normalization step is necessary to make expression counts comparable across genes and/or samples. The counts of mapped reads for each gene are proportional to the expression of RNA (âinterestingâ) in addition to many other factors (âuninterestingâ such as sequencing depth and gene length). Normalization is the process of adjusting raw count values to account for the âuninterestingâ factors.\nFor simplicity, normalization is assumed as two a two-step process: scaling and transforming.\nIn scaling the goal is to multiply each UMI count by a cell-specific factor to get all cells to have the same UMI counts. For transformation, simple approaches like log-transformation showed to be not that useful, especially in the case of genes with high expression but showed decent performance for low/intermediate expressed genes. So we cannot treat all genes the same.\nThe proposed solution for data transformation is Pearson residuals (implemented in Seurat's \u003ccode\u003eSCTransform\u003c/code\u003e function), which applies a gene-specific weight to each measurement based on the evidence of non-uniform expression across cells. This weight is higher for genes expressed in a smaller fraction of cells, making it useful for detecting rare cell populations. The weight takes into account not just the expression level but also the distribution of expression.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e3- Integrate data using Seurat's method to compare cell type expression between groups.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e4-Cluster cells based on similarity of gene expression profiles using Seurat's PCA scores.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e5-Evaluate cluster quality by checking for sources of uninteresting variation, and principal component influence, and exploring cell type identities using known markers.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExploring sources of unwanted variation\u003c/h3\u003e\u003ca id=\"user-content-exploring-sources-of-unwanted-variation\" class=\"anchor\" aria-label=\"Permalink: Exploring sources of unwanted variation\" href=\"#exploring-sources-of-unwanted-variation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eEvaluating effects of cell cycle and mitochondrial expression\u003c/h4\u003e\u003ca id=\"user-content-evaluating-effects-of-cell-cycle-and-mitochondrial-expression\" class=\"anchor\" aria-label=\"Permalink: Evaluating effects of cell cycle and mitochondrial expression\" href=\"#evaluating-effects-of-cell-cycle-and-mitochondrial-expression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe will store the cells for cell cycle genes, and then determine whether cell cycle is a major source of variation in our dataset using PCA.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Normalize the counts\n# This normalization method is solely for the purpose of exploring the sources of variation in our data.\nseurat_phase \u0026lt;- NormalizeData(filtered_seurat, normalization.method = \u0026quot;LogNormalize\u0026quot;, scale.factor = 10000)\n\n# Load cell cycle markers\nload(\u0026quot;C:/Users/qaedi/OneDrive - Queen's University/Documents/scRNA/cycle.rda\u0026quot;)\n\n# Score cells for cell cycle\nseurat_phase \u0026lt;- CellCycleScoring(seurat_phase, \n                                 g2m.features = g2m_genes, \n                                 s.features = s_genes)\n\n# View cell cycle scores and phases assigned to cells                                 \n#View(seurat_phase@meta.data) \ntable(seurat_phase$Phase)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Normalize the counts\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This normalization method is solely for the purpose of exploring the sources of variation in our data.\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e NormalizeData(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enormalization.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLogNormalize\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale.factor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10000\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Load cell cycle markers\u003c/span\u003e\nload(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eC:/Users/qaedi/OneDrive - Queen's University/Documents/scRNA/cycle.rda\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Score cells for cell cycle\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e CellCycleScoring(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e, \n                                 \u003cspan class=\"pl-v\"\u003eg2m.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eg2m_genes\u003c/span\u003e, \n                                 \u003cspan class=\"pl-v\"\u003es.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003es_genes\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e View cell cycle scores and phases assigned to cells                                 \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eView(seurat_phase@meta.data) \u003c/span\u003e\ntable(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePhase\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCells in different cell cycle stages:\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003eG1\u003c/th\u003e\n\u003cth\u003eG2M\u003c/th\u003e\n\u003cth\u003eS\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e54106\u003c/td\u003e\n\u003ctd\u003e12680\u003c/td\u003e\n\u003ctd\u003e25441\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eSo most of the cells are in G1 and then S, which make sense.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Identify the most variable genes and scale them\nseurat_phase \u0026lt;- FindVariableFeatures(seurat_phase, \n                     selection.method = \u0026quot;vst\u0026quot;,\n                     nfeatures = 2000, \n                     verbose = TRUE)\n                     \n# Identify the 10 most highly variable genes\ntop10 \u0026lt;- head(VariableFeatures(seurat_phase), 10)\n\n# plot variable features with and without labels\nplot1 \u0026lt;- VariableFeaturePlot(seurat_phase)\nplot2 \u0026lt;- LabelPoints(plot = plot1, points = top10, repel = TRUE)\nplot1 + plot2\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Identify the most variable genes and scale them\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindVariableFeatures(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003eselection.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                     \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2000\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n                     \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Identify the 10 most highly variable genes\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e head(VariableFeatures(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e), \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e plot variable features with and without labels\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eplot1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e VariableFeaturePlot(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eplot2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e LabelPoints(\u003cspan class=\"pl-v\"\u003eplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eplot1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epoints\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eplot1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eplot2\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/variable_features.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/variable_features.png\" alt=\"variable_feature\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Check quartile values for mitoRatio, we will use this variable later to mitigate the unwanted sources of variation in the dataset\nsummary(seurat_phase@meta.data$mitoRatio)\n\n# Turn mitoRatio into categorical factor vector based on quartile values\nseurat_phase@meta.data$mitoFr \u0026lt;- cut(seurat_phase@meta.data$mitoRatio, \n                   breaks=c(-Inf, 0.015, 0.025, 0.045, Inf), \n                   labels=c(\u0026quot;Low\u0026quot;,\u0026quot;Medium\u0026quot;,\u0026quot;Medium high\u0026quot;, \u0026quot;High\u0026quot;))\n\n\n\n# Scale the counts\n# This step is essential for PCA , clustering and heatmap generation\nseurat_phase \u0026lt;- ScaleData(seurat_phase)\n#saveRDS(seurat_phase, \u0026quot;seurat_phase.rds\u0026quot;)\n\n# Perform PCA\nseurat_phase \u0026lt;- RunPCA(seurat_phase)\n\n# Plot the PCA colored by cell cycle phase\nno_split \u0026lt;- DimPlot(seurat_phase,\n        reduction = \u0026quot;pca\u0026quot;,\n        group.by= \u0026quot;Phase\u0026quot;)\n        \nwith_split \u0026lt;- DimPlot(seurat_phase,\n        reduction = \u0026quot;pca\u0026quot;,\n        group.by= \u0026quot;Phase\u0026quot;,\n        split.by= \u0026quot;Phase\u0026quot;)\n        \nno_split + with_split\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Check quartile values for mitoRatio, we will use this variable later to mitigate the unwanted sources of variation in the dataset\u003c/span\u003e\nsummary(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Turn mitoRatio into categorical factor vector based on quartile values\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoFr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e cut(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e, \n                   \u003cspan class=\"pl-v\"\u003ebreaks\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.015\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.025\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.045\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e), \n                   \u003cspan class=\"pl-v\"\u003elabels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMedium\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMedium high\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHigh\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Scale the counts\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This step is essential for PCA , clustering and heatmap generation\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ScaleData(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esaveRDS(seurat_phase, \"seurat_phase.rds\")\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Perform PCA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunPCA(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the PCA colored by cell cycle phase\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eno_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e DimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePhase\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n        \n\u003cspan class=\"pl-smi\"\u003ewith_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e DimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePhase\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePhase\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n        \n\u003cspan class=\"pl-smi\"\u003eno_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ewith_split\u003c/span\u003e\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cell_cycle_PCA.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cell_cycle_PCA.png\" alt=\"cell_cycle_effect\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor mitochondrial expression:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Plot the PCA colored by mitochondrial expression\nno_split \u0026lt;- DimPlot(seurat_phase,\n        reduction = \u0026quot;pca\u0026quot;,\n        group.by= \u0026quot;mitoFr\u0026quot;)\n        \nwith_split \u0026lt;- DimPlot(seurat_phase,\n        reduction = \u0026quot;pca\u0026quot;,\n        group.by= \u0026quot;mitoFr\u0026quot;,\n        split.by= \u0026quot;mitoFr\u0026quot;)\n        \nno_split + with_split\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the PCA colored by mitochondrial expression\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eno_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e DimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoFr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n        \n\u003cspan class=\"pl-smi\"\u003ewith_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e DimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoFr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoFr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n        \n\u003cspan class=\"pl-smi\"\u003eno_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ewith_split\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/mitoExpression_effect.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/mitoExpression_effect.png\" alt=\"mito_expression_effect\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBased on the above plots, we can see that cells are scattered regardless of their cell cycle phase and mitochondrial genes expression level. So there is no need to regress out the effect of cell cycle and mitochondrial expression in this dataset.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSCTransform\u003c/h3\u003e\u003ca id=\"user-content-sctransform\" class=\"anchor\" aria-label=\"Permalink: SCTransform\" href=\"#sctransform\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis function is useful for  normalization and regressing sources of unwanted variation at the same time. The method constructs a generalized linear model (GLM) for each gene, using UMI counts as the response variable and sequencing depth as the explanatory variable. To handle the fact that different genes have different levels of expression, information is pooled across genes with similar abundances, resulting in more accurate parameter estimates.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis regularization process yields residuals, which represent effectively normalized data values that are no longer correlated with sequencing depth.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis method is a more accurate method of normalizing, estimating the variance of the raw filtered data, and identifying the most variable genes. In practice \u003ccode\u003eSCTransform\u003c/code\u003e single command replaces \u003ccode\u003eNormalizeData()\u003c/code\u003e, \u003ccode\u003eScaleData()\u003c/code\u003e, and \u003ccode\u003eFindVariableFeatures()\u003c/code\u003e. Since we have two groups of samples we will run SCTransform on each group after doing \"integration\".\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIntegration\u003c/h4\u003e\u003ca id=\"user-content-integration\" class=\"anchor\" aria-label=\"Permalink: Integration\" href=\"#integration\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo improve clustering and downstream analyses, it can be beneficial to integrate or align samples across groups using shared highly variable genes. If cells cluster by sample, condition, batch, dataset, or modalities(scRNA, scATAC-seq), integration can help to remove these unwanted sources of variation. For example, if we want to integrate normal samples together and BLCA samples together, we should keep each sample as a separate object and transform them accordingly for integration. This is necessary to ensure that the samples are properly aligned and that downstream analyses are meaningful.\nIf cell types are present in one dataset, but not the other, then the cells will still appear as a separate sample-specific cluster.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# adjust the limit for allowable object sizes within R\noptions(future.globals.maxSize = 4000 * 1024^2)\n\n# Split seurat object by group\nsplit_seurat \u0026lt;- SplitObject(seurat_phase, split.by = \u0026quot;sample\u0026quot;)\n\n# then normalize by SCTansform\nfor (i in 1:length(split_seurat)) {\n    split_seurat[[i]] \u0026lt;- SCTransform(split_seurat[[i]], vars.to.regress = c(\u0026quot;mitoRatio\u0026quot;, \u0026quot;S.Score\u0026quot;, \u0026quot;G2M.Score\u0026quot;))\n    }\n\n# to see what the component of the object are. \n\nsplit_seurat    \n#$Normal\n#An object of class Seurat\n#47302 features across 21519 samples within 2 assays\n#Active assay: SCT (23608 features, 3000 variable features)\n# 1 other assay present: RNA\n# 1 dimensional reduction calculated: pca\n\n#$BLCA\n#An object of class Seurat\n#47388 features across 70708 samples within 2 assays\n#Active assay: SCT (23694 features, 3000 variable features)\n# 1 other assay present: RNA\n# 1 dimensional reduction calculated: pca\n\n    \n# Select the most variable features to use for integration\ninteg_features \u0026lt;- SelectIntegrationFeatures(object.list = split_seurat, \n                                            nfeatures = 3000) \n\n\n# Prepare the SCT list object for integration\nsplit_seurat \u0026lt;- PrepSCTIntegration(object.list = split_seurat, \n                                   anchor.features = integ_features)\n                                   \n# Find best buddies (using canonical correlation analysis or CCA) - can take a while to run\ninteg_anchors \u0026lt;- FindIntegrationAnchors(object.list = split_seurat, \n                                        normalization.method = \u0026quot;SCT\u0026quot;, \n                                        anchor.features = integ_features)\n# Integrate across conditions\nseurat_integrated \u0026lt;- IntegrateData(anchorset = integ_anchors, \n                                   normalization.method = \u0026quot;SCT\u0026quot;)\n                                   \n# Check assays in the object:\nsplit_seurat$Normal@assays\n                                   \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e adjust the limit for allowable object sizes within R\u003c/span\u003e\noptions(\u003cspan class=\"pl-v\"\u003efuture.globals.maxSize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4000\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1024\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e^\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Split seurat object by group\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e SplitObject(\u003cspan class=\"pl-smi\"\u003eseurat_phase\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e then normalize by SCTansform\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e)) {\n    \u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e SCTransform(\u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]], \u003cspan class=\"pl-v\"\u003evars.to.regress\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoRatio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eS.Score\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eG2M.Score\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n    }\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to see what the component of the object are. \u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e    \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e$Normal\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eAn object of class Seurat\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e47302 features across 21519 samples within 2 assays\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eActive assay: SCT (23608 features, 3000 variable features)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 1 other assay present: RNA\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 1 dimensional reduction calculated: pca\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e$BLCA\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eAn object of class Seurat\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e47388 features across 70708 samples within 2 assays\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eActive assay: SCT (23694 features, 3000 variable features)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 1 other assay present: RNA\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 1 dimensional reduction calculated: pca\u003c/span\u003e\n\n    \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Select the most variable features to use for integration\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003einteg_features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e SelectIntegrationFeatures(\u003cspan class=\"pl-v\"\u003eobject.list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e, \n                                            \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3000\u003c/span\u003e) \n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Prepare the SCT list object for integration\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e PrepSCTIntegration(\u003cspan class=\"pl-v\"\u003eobject.list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003eanchor.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003einteg_features\u003c/span\u003e)\n                                   \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Find best buddies (using canonical correlation analysis or CCA) - can take a while to run\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003einteg_anchors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindIntegrationAnchors(\u003cspan class=\"pl-v\"\u003eobject.list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e, \n                                        \u003cspan class=\"pl-v\"\u003enormalization.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                        \u003cspan class=\"pl-v\"\u003eanchor.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003einteg_features\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Integrate across conditions\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e IntegrateData(\u003cspan class=\"pl-v\"\u003eanchorset\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003einteg_anchors\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003enormalization.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n                                   \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Check assays in the object:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eNormal\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\n                                   \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIntegration check\u003c/h3\u003e\u003ca id=\"user-content-integration-check\" class=\"anchor\" aria-label=\"Permalink: Integration check\" href=\"#integration-check\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAfter normalization and integration, we can proceed to PCA and UMAP/t-SNE to see effect of integration.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Run PCA\nseurat_integrated \u0026lt;- RunPCA(object = seurat_integrated, verbose = TRUE)\n# PC_ 1\n# Positive:  IGFBP7, MGP, SPARC, VIM, SPARCL1, CALD1, IFITM3, A2M, COL4A1, COL1A2\n#            COL4A2, TAGLN, COL6A2, BGN, TCF4, COL3A1, CCL2, MT2A, NNMT, COL1A1\n#            MYL9, LGALS1, DCN, SELENOM, CCN1, ADAMTS9, CCN2, TIMP1, IGFBP4, LUM\n# Negative:  KRT19, KRT13, CD52, S100P, RPS19, CLDN4, FXYD3, PSCA, TACSTD2, CSTB\n#            PTPRC, KRT7, RPS18, KRT17, LYPD3, RPL41, CD3D, CCL5, RPS27, SFN\n#            FABP5, SPINK1, GDF15, AQP3, HCST, TRAC, SNCG, ADIRF, ELF3, TRBC2\n# PC_ 2\n# Positive:  PLVAP, CALCRL, PCAT19, MCTP1, AQP1, VWF, CD74, PECAM1, RAMP2, LDB2\n#            RAMP3, FLT1, ZNF385D, TCF4, CLDN5, ACKR1, HSPG2, SPARCL1, EMCN, ADGRL4\n#            SLCO2A1, HLA-DRA, SELE, DOCK4, ECSCR, CCL14, HLA-DRB1, ERG, PCDH17, GNG11\n# Negative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN, LUM, BGN, COL6A2, C1R, SOD3\n#            MFAP4, RARRES2, C1S, MYL9, TPM2, PRKG1, CRYAB, ACTA2, COL6A3, LGALS1\n#            CALD1, COL6A1, TIMP1, SERPINF1, PCOLCE, AEBP1, C11orf96, MEG3, FBLN1, GPC6\n# PC_ 3\n# Positive:  CCL5, CD52, B2M, PTPRC, IL32, SRGN, CD3D, HSPA1A, NKG7, GZMA\n#            TRAC, ARHGAP15, HCST, RGS1, CXCR4, SAMSN1, CCL4, CD7, RGS2, CORO1A\n#            CD2, CRIP1, CST7, CD69, STAT4, FYN, TMSB4X, DNAJB1, PTPN22, CD3E\n# Negative:  ADIRF, SPINK1, IFI27, CSTB, S100P, KRT19, FXYD3, CLDN4, CCT2, CCND1\n#            PSCA, S100A6, SNCG, UCA1, KRT17, YEATS4, KRT7, KRT13, TACSTD2, GDF15\n#            RAB3IP, KRT18, HES1, GAPDH, S100A2, PLVAP, RAMP2, S100A14, FABP5, TM4SF1\n# PC_ 4\n# Positive:  IL32, CD3D, CCL5, CRIP1, TRAC, FYN, COL4A1, COL4A2, CD2, GZMA\n#            CD7, CALD1, IGFBP7, TRBC2, MCAM, NDUFA4L2, CD3E, NKG7, COL18A1, RGS5\n#            SKAP1, MYL9, ITGA1, CYTOR, CD247, SPARC, TRBC1, MAP1B, ACTA2, CACNA1C\n# Negative:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1, FCER1G, HLA-DPA1, AIF1, HLA-DQA1, FTL\n#            LYZ, IFI30, C1QA, HLA-DQB1, C1QB, C1QC, MS4A6A, LST1, APOE, CD14\n#            TMEM176B, S100A9, HLA-DMA, FCGR2A, CXCL8, SPI1, FTH1, CD68, PSAP, CST3\n# PC_ 5\n# Positive:  LUM, MMP2, PTGDS, EMP1, DCN, RARRES2, KRT13, SERPINF1, CXCL1, CXCL8\n#            LSAMP, COL8A1, TM4SF1, AREG, C1S, PDPN, CFD, APOD, SOD2, CTSK\n#            CLMP, KRT17, MFAP4, LYPD3, VCAN, PLAUR, TSHZ2, PLAT, PDGFRA, C1R\n# Negative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9, TAGLN, FRZB, CRIP1, CALD1, GJA4\n#            MCAM, COL18A1, MYH11, TYROBP, TPPP3, COX4I2, PRKG1, IGFBP7, COL4A1, COL4A2\n#            CDH6, HIGD1B, AIF1, TPM2, PTP4A3, FCER1G, WFDC1, HEYL, MYLK, HLA-DRA\n\n\n\n# Plot PCA\npng(filename = \u0026quot;PCA_integrated.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nPCAPlot(seurat_integrated,\n        split.by = \u0026quot;sample\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Run PCA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunPCA(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e PC_ 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Positive:  IGFBP7, MGP, SPARC, VIM, SPARCL1, CALD1, IFITM3, A2M, COL4A1, COL1A2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            COL4A2, TAGLN, COL6A2, BGN, TCF4, COL3A1, CCL2, MT2A, NNMT, COL1A1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            MYL9, LGALS1, DCN, SELENOM, CCN1, ADAMTS9, CCN2, TIMP1, IGFBP4, LUM\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Negative:  KRT19, KRT13, CD52, S100P, RPS19, CLDN4, FXYD3, PSCA, TACSTD2, CSTB\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            PTPRC, KRT7, RPS18, KRT17, LYPD3, RPL41, CD3D, CCL5, RPS27, SFN\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            FABP5, SPINK1, GDF15, AQP3, HCST, TRAC, SNCG, ADIRF, ELF3, TRBC2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e PC_ 2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Positive:  PLVAP, CALCRL, PCAT19, MCTP1, AQP1, VWF, CD74, PECAM1, RAMP2, LDB2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            RAMP3, FLT1, ZNF385D, TCF4, CLDN5, ACKR1, HSPG2, SPARCL1, EMCN, ADGRL4\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            SLCO2A1, HLA-DRA, SELE, DOCK4, ECSCR, CCL14, HLA-DRB1, ERG, PCDH17, GNG11\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Negative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN, LUM, BGN, COL6A2, C1R, SOD3\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            MFAP4, RARRES2, C1S, MYL9, TPM2, PRKG1, CRYAB, ACTA2, COL6A3, LGALS1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            CALD1, COL6A1, TIMP1, SERPINF1, PCOLCE, AEBP1, C11orf96, MEG3, FBLN1, GPC6\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e PC_ 3\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Positive:  CCL5, CD52, B2M, PTPRC, IL32, SRGN, CD3D, HSPA1A, NKG7, GZMA\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            TRAC, ARHGAP15, HCST, RGS1, CXCR4, SAMSN1, CCL4, CD7, RGS2, CORO1A\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            CD2, CRIP1, CST7, CD69, STAT4, FYN, TMSB4X, DNAJB1, PTPN22, CD3E\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Negative:  ADIRF, SPINK1, IFI27, CSTB, S100P, KRT19, FXYD3, CLDN4, CCT2, CCND1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            PSCA, S100A6, SNCG, UCA1, KRT17, YEATS4, KRT7, KRT13, TACSTD2, GDF15\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            RAB3IP, KRT18, HES1, GAPDH, S100A2, PLVAP, RAMP2, S100A14, FABP5, TM4SF1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e PC_ 4\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Positive:  IL32, CD3D, CCL5, CRIP1, TRAC, FYN, COL4A1, COL4A2, CD2, GZMA\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            CD7, CALD1, IGFBP7, TRBC2, MCAM, NDUFA4L2, CD3E, NKG7, COL18A1, RGS5\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            SKAP1, MYL9, ITGA1, CYTOR, CD247, SPARC, TRBC1, MAP1B, ACTA2, CACNA1C\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Negative:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1, FCER1G, HLA-DPA1, AIF1, HLA-DQA1, FTL\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            LYZ, IFI30, C1QA, HLA-DQB1, C1QB, C1QC, MS4A6A, LST1, APOE, CD14\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            TMEM176B, S100A9, HLA-DMA, FCGR2A, CXCL8, SPI1, FTH1, CD68, PSAP, CST3\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e PC_ 5\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Positive:  LUM, MMP2, PTGDS, EMP1, DCN, RARRES2, KRT13, SERPINF1, CXCL1, CXCL8\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            LSAMP, COL8A1, TM4SF1, AREG, C1S, PDPN, CFD, APOD, SOD2, CTSK\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            CLMP, KRT17, MFAP4, LYPD3, VCAN, PLAUR, TSHZ2, PLAT, PDGFRA, C1R\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Negative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9, TAGLN, FRZB, CRIP1, CALD1, GJA4\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            MCAM, COL18A1, MYH11, TYROBP, TPPP3, COX4I2, PRKG1, IGFBP7, COL4A1, COL4A2\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e            CDH6, HIGD1B, AIF1, TPM2, PTP4A3, FCER1G, WFDC1, HEYL, MYLK, HLA-DRA\u003c/span\u003e\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot PCA\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePCA_integrated.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nPCAPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/PCA_integrated.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/PCA_integrated.png\" alt=\"plot-PCA\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Run UMAP\nseurat_integrated \u0026lt;- RunUMAP(seurat_integrated, \n                             dims = 1:40,\n                             reduction = \u0026quot;pca\u0026quot;,\n                             verbose = TRUE)\n\n# Plot UMAP \npng(filename = \u0026quot;UMAP_integrated.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated, split.by = \u0026quot;sample\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Run UMAP\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunUMAP(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                             \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e40\u003c/span\u003e,\n                             \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                             \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot UMAP \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUMAP_integrated.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/UMAP_integrated.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/UMAP_integrated.png\" alt=\"plot-UMAP\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor the future: considering tumor grade and using Harmony for integration\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eClustering cells based on top PCs (metagenes)\u003c/h3\u003e\u003ca id=\"user-content-clustering-cells-based-on-top-pcs-metagenes\" class=\"anchor\" aria-label=\"Permalink: Clustering cells based on top PCs (metagenes)\" href=\"#clustering-cells-based-on-top-pcs-metagenes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentify significant PCs\u003c/h4\u003e\u003ca id=\"user-content-identify-significant-pcs\" class=\"anchor\" aria-label=\"Permalink: Identify significant PCs\" href=\"#identify-significant-pcs\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor new methods like SCTransform it is not needed to calculate the number of PCs for clustering. However, older methods could not efficiently remove technical biases, so using them it was necessary to have some idea about the number of PCs that can capture most of the information in the dataset.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Explore the heatmap of PCs\npng(filename = \u0026quot;heatmap_integrated_2.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimHeatmap(seurat_integrated, \n           dims = 1:9, \n           cells = 500, \n           balanced = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Explore the heatmap of PCs\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eheatmap_integrated_2.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimHeatmap(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n           \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e, \n           \u003cspan class=\"pl-v\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e500\u003c/span\u003e, \n           \u003cspan class=\"pl-v\"\u003ebalanced\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/heatmap_integrated_2.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/heatmap_integrated_2.png\" alt=\"plot-PCA_2\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Printing out the most variable genes driving PCs\nprint(x = seurat_integrated[[\u0026quot;pca\u0026quot;]], \n      dims = 1:10, \n      nfeatures = 5)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Printing out the most variable genes driving PCs\u003c/span\u003e\nprint(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]], \n      \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \n      \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"PC_ 1\nPositive:  KRT19, CD52, KRT13, S100P, RPS19\nNegative:  IGFBP7, MGP, SPARC, VIM, SPARCL1\nPC_ 2\nPositive:  PLVAP, PCAT19, CALCRL, MCTP1, AQP1\nNegative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN\nPC_ 3\nPositive:  CCL5, CD52, PTPRC, B2M, SRGN\nNegative:  ADIRF, SPINK1, CSTB, S100P, KRT19\nPC_ 4\nPositive:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1\nNegative:  CD3D, IL32, CCL5, CRIP1, TRAC\nPC_ 5\nPositive:  LUM, MMP2, PTGDS, RARRES2, DCN\nNegative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9\nPC_ 6\nPositive:  SPINK1, CCT2, LCN15, UCA1, PLA2G2A\nNegative:  KRT13, PLAUR, LYPD3, OLFM4, EMP1\nPC_ 7\nPositive:  FABP5, PLA2G2A, LCN15, FABP4, RPS19\nNegative:  CCT2, ADIRF, SPINK1, HSPA1A, CCND1\nPC_ 8\nPositive:  FOS, HSPA1A, JUN, ZFP36, DNAJB1\nNegative:  SPARC, COL4A1, INSR, CCL5, COL4A2\nPC_ 9\nPositive:  LCN15, PLA2G2A, FABP4, CRTAC1, LINC01088\nNegative:  H19, RPS19, CRH, AP005230.1, PSCA\nPC_ 10\nPositive:  HSPA1A, HSPA1B, DNAJB1, HSP90AA1, MALAT1\nNegative:  CRH, CCL5, LY6D, ACKR1, RPS19\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ePC_ 1\nPositive:  KRT19, CD52, KRT13, S100P, RPS19\nNegative:  IGFBP7, MGP, SPARC, VIM, SPARCL1\nPC_ 2\nPositive:  PLVAP, PCAT19, CALCRL, MCTP1, AQP1\nNegative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN\nPC_ 3\nPositive:  CCL5, CD52, PTPRC, B2M, SRGN\nNegative:  ADIRF, SPINK1, CSTB, S100P, KRT19\nPC_ 4\nPositive:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1\nNegative:  CD3D, IL32, CCL5, CRIP1, TRAC\nPC_ 5\nPositive:  LUM, MMP2, PTGDS, RARRES2, DCN\nNegative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9\nPC_ 6\nPositive:  SPINK1, CCT2, LCN15, UCA1, PLA2G2A\nNegative:  KRT13, PLAUR, LYPD3, OLFM4, EMP1\nPC_ 7\nPositive:  FABP5, PLA2G2A, LCN15, FABP4, RPS19\nNegative:  CCT2, ADIRF, SPINK1, HSPA1A, CCND1\nPC_ 8\nPositive:  FOS, HSPA1A, JUN, ZFP36, DNAJB1\nNegative:  SPARC, COL4A1, INSR, CCL5, COL4A2\nPC_ 9\nPositive:  LCN15, PLA2G2A, FABP4, CRTAC1, LINC01088\nNegative:  H19, RPS19, CRH, AP005230.1, PSCA\nPC_ 10\nPositive:  HSPA1A, HSPA1B, DNAJB1, HSP90AA1, MALAT1\nNegative:  CRH, CCL5, LY6D, ACKR1, RPS19\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# To determine how many Pcs should be considered for clustering:\n# Plot the elbow plot\npng(filename = \u0026quot;elbow.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nElbowPlot(object = seurat_integrated, \n          ndims = 40)\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e To determine how many Pcs should be considered for clustering:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the elbow plot\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eelbow.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nElbowPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n          \u003cspan class=\"pl-v\"\u003endims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e40\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/elbow.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/elbow.png\" alt=\"plot-elbow\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# to make it more quantitative :\n# Determine percent of variation associated with each PC\npct \u0026lt;- seurat_integrated[[\u0026quot;pca\u0026quot;]]@stdev / sum(seurat_integrated[[\u0026quot;pca\u0026quot;]]@stdev) * 100\n\n#pct\n# [1] 7.9623240 6.3833277 5.7806955 4.4000340 3.9930564 3.6021757 3.5527516\n# [8] 3.0283830 2.9108397 2.6331792 2.4437350 2.3524902 2.2441047 2.1753633\n#[15] 2.0485619 2.0020779 1.9396254 1.8168114 1.7548723 1.6539166 1.6410801\n#[22] 1.5884590 1.5223992 1.4950181 1.4627909 1.3921263 1.3815652 1.3046432\n#[29] 1.2605721 1.2382860 1.2051052 1.1972413 1.1821032 1.1741361 1.1577815\n#[36] 1.1294799 1.1190593 1.0968174 1.0591574 1.0519859 1.0396189 1.0211281\n#[43] 1.0034793 0.9976230 0.9786685 0.9497183 0.9415772 0.9171901 0.9103623\n#[50] 0.9025012\n\n\n# Calculate cumulative percents for each PC\ncumu \u0026lt;- cumsum(pct)\n\n#cumu\n# [1]   7.962324  14.345652  20.126347  24.526381  28.519438  32.121613\n# [7]  35.674365  38.702748  41.613588  44.246767  46.690502  49.042992\n#[13]  51.287097  53.462460  55.511022  57.513100  59.452725  61.269537\n#[19]  63.024409  64.678326  66.319406  67.907865  69.430264  70.925282\n#[25]  72.388073  73.780199  75.161764  76.466408  77.726980  78.965266\n#[31]  80.170371  81.367612  82.549715  83.723852  84.881633  86.011113\n#[37]  87.130172  88.226990  89.286147  90.338133  91.377752  92.398880\n#[43]  93.402359  94.399982  95.378651  96.328369  97.269946  98.187137\n\n\n# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5\nco1 \u0026lt;- which(cumu \u0026gt; 90 \u0026amp; pct \u0026lt; 5)[1]\n\nco1\n\n# Determine the difference between variation of PC and subsequent PC\nco2 \u0026lt;- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) \u0026gt; 0.1), decreasing = T)[1] + 1\n\n# last point where change of % of variation is more than 0.1%.\nco2\n\n# Minimum of the two calculation is the optimal number of PC to pick.\npcs \u0026lt;- min(co1, co2)\n\npcs\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to make it more quantitative :\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine percent of variation associated with each PC\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epct\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003estdev\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e sum(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003estdev\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epct\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e [1] 7.9623240 6.3833277 5.7806955 4.4000340 3.9930564 3.6021757 3.5527516\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e [8] 3.0283830 2.9108397 2.6331792 2.4437350 2.3524902 2.2441047 2.1753633\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[15] 2.0485619 2.0020779 1.9396254 1.8168114 1.7548723 1.6539166 1.6410801\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[22] 1.5884590 1.5223992 1.4950181 1.4627909 1.3921263 1.3815652 1.3046432\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[29] 1.2605721 1.2382860 1.2051052 1.1972413 1.1821032 1.1741361 1.1577815\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[36] 1.1294799 1.1190593 1.0968174 1.0591574 1.0519859 1.0396189 1.0211281\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[43] 1.0034793 0.9976230 0.9786685 0.9497183 0.9415772 0.9171901 0.9103623\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[50] 0.9025012\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Calculate cumulative percents for each PC\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecumu\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e cumsum(\u003cspan class=\"pl-smi\"\u003epct\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecumu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e [1]   7.962324  14.345652  20.126347  24.526381  28.519438  32.121613\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e [7]  35.674365  38.702748  41.613588  44.246767  46.690502  49.042992\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[13]  51.287097  53.462460  55.511022  57.513100  59.452725  61.269537\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[19]  63.024409  64.678326  66.319406  67.907865  69.430264  70.925282\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[25]  72.388073  73.780199  75.161764  76.466408  77.726980  78.965266\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[31]  80.170371  81.367612  82.549715  83.723852  84.881633  86.011113\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[37]  87.130172  88.226990  89.286147  90.338133  91.377752  92.398880\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e[43]  93.402359  94.399982  95.378651  96.328369  97.269946  98.187137\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eco1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e which(\u003cspan class=\"pl-smi\"\u003ecumu\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e90\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epct\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]\n\n\u003cspan class=\"pl-smi\"\u003eco1\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine the difference between variation of PC and subsequent PC\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eco2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e sort(which((\u003cspan class=\"pl-smi\"\u003epct\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003epct\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epct\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003epct\u003c/span\u003e)]) \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003edecreasing\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e last point where change of % of variation is more than 0.1%.\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eco2\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Minimum of the two calculation is the optimal number of PC to pick.\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epcs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e min(\u003cspan class=\"pl-smi\"\u003eco1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eco2\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003epcs\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCluster the cells\u003c/h4\u003e\u003ca id=\"user-content-cluster-the-cells\" class=\"anchor\" aria-label=\"Permalink: Cluster the cells\" href=\"#cluster-the-cells\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# to check what is active assay\nDefaultAssay(object = seurat_integrated)\n\n# Determine the K-nearest neighbor graph\nseurat_integrated \u0026lt;- FindNeighbors(object = seurat_integrated, \n                                dims = 1:18)\n                                \nFind clusters\n# Determine the clusters for various resolutions                                \nseurat_integrated \u0026lt;- FindClusters(object = seurat_integrated,\n                               resolution = c(0.4, 0.6, 0.8, 1.0, 1.4))\n                               \n\n# Explore resolutions\nhead(seurat_integrated@meta.data)\n\n# Assign the identity of clusters\nIdents(object = seurat_integrated) \u0026lt;- \u0026quot;integrated_snn_res.0.4\u0026quot;\n\n# Plot the UMAP\npng(filename = \u0026quot;umap_cluster_with_label.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to check what is active assay\u003c/span\u003e\nDefaultAssay(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine the K-nearest neighbor graph\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindNeighbors(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                                \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e18\u003c/span\u003e)\n                                \n\u003cspan class=\"pl-smi\"\u003eFind\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eclusters\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine the clusters for various resolutions                                \u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindClusters(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n                               \u003cspan class=\"pl-v\"\u003eresolution\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1.0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1.4\u003c/span\u003e))\n                               \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Explore resolutions\u003c/span\u003e\nhead(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Assign the identity of clusters\u003c/span\u003e\nIdents(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eintegrated_snn_res.0.4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the UMAP\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_cluster_with_label.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster_with_label.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster_with_label.png\" alt=\"plot-umap_label\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eClustering quality control\u003c/h4\u003e\u003ca id=\"user-content-clustering-quality-control\" class=\"anchor\" aria-label=\"Permalink: Clustering quality control\" href=\"#clustering-quality-control\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAfter clustering, we need to make sure that the assigned clusters are true representative of biological clusters (cell clusters) not due to technical or unwanted source of variation (like cell cycle stages). Also, in this step, we need to identify cell type for each cluster based on the known cell type markers.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSegregation of clusters by sample\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Extract identity and sample information from Seurat object to determine the number of cells per cluster per sample\n\nlibrary(dplyr)\nlibrary(tidyr)\n\n# n_cells \u0026lt;- FetchData(seurat_integrated, \n#                      vars = c(\u0026quot;ident\u0026quot;, \u0026quot;orig.ident\u0026quot;)) %\u0026gt;%\n#         dplyr::count(ident, orig.ident) %\u0026gt;%\n#         tidyr::spread(ident, n)\n\nn_cells \u0026lt;- FetchData(seurat_integrated, \n                      vars = c(\u0026quot;ident\u0026quot;, \u0026quot;orig.ident\u0026quot;))\nn_cells \u0026lt;- dplyr::count(n_cells, ident, orig.ident)\nn_cells \u0026lt;- tidyr::spread(n_cells, ident, n)\n\n\n\n#Ading sample data from paper; we expect to see samples from the same group have more or less similar number of cells in each cluster. \n#So normal samples should show similar patterns: SRR12603780, SRR12603781, and SRR12603788.\n\n\n\nsampleData\u0026lt;- data.frame(tibble::tribble(\n     ~sample_id, ~gender, ~age, ~Grade, ~Invasiveness, ~Surgery_Type, ~Tumor_size_cm,\n  \u0026quot;SRR12603790\u0026quot;,     \u0026quot;M\u0026quot;,  67L,  \u0026quot;low\u0026quot;, \u0026quot;Noninvasive\u0026quot;,       \u0026quot;TURBT\u0026quot;,          \u0026quot;1.9\u0026quot;,\n  \u0026quot;SRR12603789\u0026quot;,     \u0026quot;M\u0026quot;,  70L,  \u0026quot;low\u0026quot;, \u0026quot;Noninvasive\u0026quot;,       \u0026quot;TURBT\u0026quot;,          \u0026quot;2.5\u0026quot;,\n  \u0026quot;SRR12603787\u0026quot;,     \u0026quot;M\u0026quot;,  63L, \u0026quot;high\u0026quot;, \u0026quot;Noninvasive\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,          \u0026quot;3.5\u0026quot;,\n  \u0026quot;SRR12603786\u0026quot;,     \u0026quot;F\u0026quot;,  59L, \u0026quot;high\u0026quot;, \u0026quot;Noninvasive\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,          \u0026quot;4.7\u0026quot;,\n  \u0026quot;SRR12603785\u0026quot;,     \u0026quot;M\u0026quot;,  57L, \u0026quot;high\u0026quot;,    \u0026quot;Invasive\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,          \u0026quot;5.1\u0026quot;,\n  \u0026quot;SRR12603784\u0026quot;,     \u0026quot;M\u0026quot;,  75L, \u0026quot;high\u0026quot;,    \u0026quot;Invasive\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,          \u0026quot;4.3\u0026quot;,\n  \u0026quot;SRR12603783\u0026quot;,     \u0026quot;M\u0026quot;,  77L, \u0026quot;high\u0026quot;,    \u0026quot;Invasive\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,          \u0026quot;4.5\u0026quot;,\n  \u0026quot;SRR12603782\u0026quot;,     \u0026quot;F\u0026quot;,  72L, \u0026quot;high\u0026quot;,    \u0026quot;Invasive\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,          \u0026quot;4.1\u0026quot;,\n  \u0026quot;SRR12603781\u0026quot;,     \u0026quot;M\u0026quot;,  67L, \u0026quot;normal\u0026quot;,   \u0026quot;normal\u0026quot;,       \u0026quot;TURBT\u0026quot;,            \u0026quot;-\u0026quot;,\n  \u0026quot;SRR12603788\u0026quot;,     \u0026quot;M\u0026quot;,  75L, \u0026quot;normal\u0026quot;,   \u0026quot;normal\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,            \u0026quot;-\u0026quot;,\n  \u0026quot;SRR12603780\u0026quot;,     \u0026quot;M\u0026quot;,  63L, \u0026quot;normal\u0026quot;,   \u0026quot;normal\u0026quot;,  \u0026quot;Cystectomy\u0026quot;,            \u0026quot;-\u0026quot;\n  ))\n\n\n# View table\nhead(n_cells)\n# saving objects (to mark where and when we stored the file)\n#seurat_cluster \u0026lt;- seurat_integrated\n#saveRDS(seurat_cluster, \u0026quot;seurat_cluster.RDS\u0026quot;)\n\n\n# UMAP of cells in each cluster by sample\n# This would allow us to see condition specific clusters\npng(filename = \u0026quot;umap_cluster_sample.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated, \n        label = TRUE, \n        split.by = \u0026quot;sample\u0026quot;)  + NoLegend()\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract identity and sample information from Seurat object to determine the number of cells per cluster per sample\u003c/span\u003e\n\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyr\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e n_cells \u0026lt;- FetchData(seurat_integrated, \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                      vars = c(\"ident\", \"orig.ident\")) %\u0026gt;%\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e         dplyr::count(ident, orig.ident) %\u0026gt;%\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e         tidyr::spread(ident, n)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003en_cells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FetchData(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003evars\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003en_cells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003ecount(\u003cspan class=\"pl-smi\"\u003en_cells\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eident\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eorig.ident\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003en_cells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003etidyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003espread(\u003cspan class=\"pl-smi\"\u003en_cells\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eident\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003en\u003c/span\u003e)\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eAding sample data from paper; we expect to see samples from the same group have more or less similar number of cells in each cluster. \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSo normal samples should show similar patterns: SRR12603780, SRR12603781, and SRR12603788.\u003c/span\u003e\n\n\n\n\u003cspan class=\"pl-smi\"\u003esampleData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-e\"\u003etibble\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003etribble(\n     \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample_id\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egender\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eage\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eGrade\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eInvasiveness\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSurgery_Type\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eTumor_size_cm\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603790\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e67L\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNoninvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,       \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTURBT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1.9\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603789\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e70L\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNoninvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,       \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTURBT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e2.5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603787\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e63L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehigh\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNoninvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3.5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603786\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e59L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehigh\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNoninvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4.7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603785\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e57L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehigh\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5.1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603784\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e75L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehigh\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4.3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603783\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e77L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehigh\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4.5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603782\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e72L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehigh\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasive\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4.1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603781\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e67L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,   \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,       \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTURBT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603788\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e75L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,   \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603780\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,     \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-c1\"\u003e63L\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,   \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCystectomy\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n  ))\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e View table\u003c/span\u003e\nhead(\u003cspan class=\"pl-smi\"\u003en_cells\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e saving objects (to mark where and when we stored the file)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eseurat_cluster \u0026lt;- seurat_integrated\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esaveRDS(seurat_cluster, \"seurat_cluster.RDS\")\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e UMAP of cells in each cluster by sample\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This would allow us to see condition specific clusters\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_cluster_sample.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e NoLegend()\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster_sample.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster_sample.png\" alt=\"umap_cluster_sample.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Segregation of clusters by cell cycle phase (unwanted source of variation) \n# Explore whether clusters segregate by cell cycle phase\n\npng(filename = \u0026quot;umap_cluster_cell_cucle.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated,\n        label = TRUE, \n        split.by = \u0026quot;Phase\u0026quot;)  + NoLegend()\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Segregation of clusters by cell cycle phase (unwanted source of variation) \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Explore whether clusters segregate by cell cycle phase\u003c/span\u003e\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_cluster_cell_cucle.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePhase\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e NoLegend()\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster_cell_cucle.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster_cell_cucle.png\" alt=\"umap_cluster_cell_cucle.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSegregation of clusters by various sources of uninteresting variation\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eWe expect to see a uniform coloring for all variables in all clusters. Sometimes this is not the case. For here \u003ccode\u003enUMI\u003c/code\u003e and \u003ccode\u003enGene\u003c/code\u003e show higher values is some clusters. We have to watch these clusters and inspect them in terms of the type of cell therein. So that may explain some of the variation that we are seeing.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Determine metrics to plot present in seurat_integrated@meta.data\nmetrics \u0026lt;-  c(\u0026quot;nUMI\u0026quot;, \u0026quot;nGene\u0026quot;, \u0026quot;S.Score\u0026quot;, \u0026quot;G2M.Score\u0026quot;, \u0026quot;mitoRatio\u0026quot;)\npng(filename = \u0026quot;umap_unwanted_source_clustering.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(seurat_integrated, \n            reduction = \u0026quot;umap\u0026quot;, \n            features = metrics,\n            pt.size = 0.4, \n            order = TRUE,\n            min.cutoff = 'q10',\n            label = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine metrics to plot present in seurat_integrated@meta.data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetrics\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e  c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enUMI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eS.Score\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eG2M.Score\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoRatio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_unwanted_source_clustering.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emetrics\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003ept.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_unwanted_source_clustering.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_unwanted_source_clustering.png\" alt=\"umap_unwanted_source_clustering.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eExploration of the PCs driving the different clusters\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eWe hope that the defined PCs could separate clusters well.We can see how the clusters are represented by the different PCs.Then we could look back at our genes driving this PC to get an idea of what the cell types might be in each cluster.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Defining the information in the seurat object of interest\ncolumns \u0026lt;- c(paste0(\u0026quot;PC_\u0026quot;, 1:18),\n            \u0026quot;ident\u0026quot;,\n            \u0026quot;UMAP_1\u0026quot;, \u0026quot;UMAP_2\u0026quot;)\n\n# Extracting this data from the seurat object\npc_data \u0026lt;- FetchData(seurat_integrated, \n                     vars = columns)\n                     \n                     \n# Adding cluster label to center of cluster on UMAP\numap_label \u0026lt;- FetchData(seurat_integrated, \n                        vars = c(\u0026quot;ident\u0026quot;, \u0026quot;UMAP_1\u0026quot;, \u0026quot;UMAP_2\u0026quot;))  %\u0026gt;%\n  group_by(ident) %\u0026gt;%\n  summarise(x=mean(UMAP_1), y=mean(UMAP_2))\n\n\n\n# Plotting a UMAP plot for each of the PCs\nlibrary(cowplot)\nlibrary(tidyverse)\nlibrary(HGNChelper)\n\npng(filename = \u0026quot;umap_on_pcs.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nmap(paste0(\u0026quot;PC_\u0026quot;, 1:18), function(pc){\n        ggplot(pc_data, \n               aes(UMAP_1, UMAP_2)) +\n                geom_point(aes_string(color=pc), \n                           alpha = 0.7) +\n                scale_color_gradient(guide = FALSE, \n                                     low = \u0026quot;grey90\u0026quot;, \n                                     high = \u0026quot;blue\u0026quot;)  +\n                geom_text(data=umap_label, \n                          aes(label=ident, x, y)) +\n                ggtitle(pc)\n}) %\u0026gt;% \n        plot_grid(plotlist = .)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Defining the information in the seurat object of interest\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecolumns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePC_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e18\u003c/span\u003e),\n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUMAP_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUMAP_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extracting this data from the seurat object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epc_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FetchData(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                     \u003cspan class=\"pl-v\"\u003evars\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecolumns\u003c/span\u003e)\n                     \n                     \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Adding cluster label to center of cluster on UMAP\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eumap_label\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FetchData(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003evars\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUMAP_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUMAP_2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))  %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  group_by(\u003cspan class=\"pl-smi\"\u003eident\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  summarise(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emean(\u003cspan class=\"pl-smi\"\u003eUMAP_1\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003emean(\u003cspan class=\"pl-smi\"\u003eUMAP_2\u003c/span\u003e))\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plotting a UMAP plot for each of the PCs\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003ecowplot\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etidyverse\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eHGNChelper\u003c/span\u003e)\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_on_pcs.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nmap(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePC_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e18\u003c/span\u003e), \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003epc\u003c/span\u003e){\n        ggplot(\u003cspan class=\"pl-smi\"\u003epc_data\u003c/span\u003e, \n               aes(\u003cspan class=\"pl-smi\"\u003eUMAP_1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eUMAP_2\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n                geom_point(aes_string(\u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epc\u003c/span\u003e), \n                           \u003cspan class=\"pl-v\"\u003ealpha\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.7\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n                scale_color_gradient(\u003cspan class=\"pl-v\"\u003eguide\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \n                                     \u003cspan class=\"pl-v\"\u003elow\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egrey90\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                     \u003cspan class=\"pl-v\"\u003ehigh\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n                geom_text(\u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eumap_label\u003c/span\u003e, \n                          aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eident\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ey\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n                ggtitle(\u003cspan class=\"pl-smi\"\u003epc\u003c/span\u003e)\n}) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n        plot_grid(\u003cspan class=\"pl-v\"\u003eplotlist\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_on_pcs.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_on_pcs.png\" alt=\"umap_on_pcs.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Examine PCA results \nprint(seurat_integrated[[\u0026quot;pca\u0026quot;]], dims = 1:5, nfeatures = 5)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Examine PCA results \u003c/span\u003e\nprint(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]], \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Normalize RNA data for visualization purposes\nseurat_integrated \u0026lt;- NormalizeData(seurat_integrated, verbose = FALSE)\n\npng(filename = \u0026quot;umap_fibroblast.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(seurat_integrated, \n            reduction = \u0026quot;umap\u0026quot;, \n            features = c(\u0026quot;IGFBP7\u0026quot;, \u0026quot;MGP\u0026quot;), \n            order = TRUE,\n            min.cutoff = 'q10', \n            label = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Normalize RNA data for visualization purposes\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e NormalizeData(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_fibroblast.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGFBP7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMGP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_fibroblast.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_fibroblast.png\" alt=\"umap_fibroblast.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;umap_endothelial.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(seurat_integrated, \n            reduction = \u0026quot;umap\u0026quot;, \n            features = c(\u0026quot;PLVAP\u0026quot;, \u0026quot;CALCRL\u0026quot;), \n            order = TRUE,\n            min.cutoff = 'q10', \n            label = TRUE)\ndev.off()\n\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_endothelial.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePLVAP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCALCRL\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_endothelial.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_endothelial.png\" alt=\"umap_endothelial.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;umap_t_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(seurat_integrated, \n            reduction = \u0026quot;umap\u0026quot;, \n            features = c(\u0026quot;CCL5\u0026quot;, \u0026quot;CD52\u0026quot;, \u0026quot;IL32\u0026quot;), \n            order = TRUE,\n            min.cutoff = 'q10', \n            label = TRUE)\ndev.off()\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_t_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCCL5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD52\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIL32\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_t_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_t_cells.png\" alt=\"umap_t_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e7) Marker identification and cell type assignment\u003c/h2\u003e\u003ca id=\"user-content-7-marker-identification-and-cell-type-assignment\" class=\"anchor\" aria-label=\"Permalink: 7) Marker identification and cell type assignment\" href=\"#7-marker-identification-and-cell-type-assignment\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis can be the last step in our pipeline which aims to determine the gene markers for each of the clusters and identify cell types of each cluster using markers. Also, this step helps to determine whether there's a need to re-cluster based on cell type markers, or maybe clusters need to be merged or split.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor marker identification, there are three functions in the Seurat package, each with a different application:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eFindAllMarkers()\u003c/code\u003e: It should only be used when comparing a cluster against other clusters belonging to the same group. i.e. this function should be used only when we have one group/condition.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eFindConservedMarkers()\u003c/code\u003e: When we have two groups like tumor vs normal or invasive vs. noninvasive identifying conserved markers is the best approach. In this way, we find DE genes for a given cluster in one condition (e.g. invasive) comparing the cluster against the rest of the cluster in the same condition group. We do the same for that given cluster in the other condition (non-invasive). Finally, the two lists will be merged to give us the conserved marker for a given cluster.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ccode\u003eFindMarkers()\u003c/code\u003e: This is helpful with identifying gene markers for each cluster. In practice sometimes the list of markers returned doesnât sufficiently separate some of the clusters. We can use this function to further differentiate between those clusters.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#______________________________ NOT TO BE RUN________________________________\n# Find markers for every cluster compared to all remaining cells, report only the positive ones\nmarkers \u0026lt;- FindAllMarkers(object = seurat_integrated, \n                          only.pos = TRUE,\n                          logfc.threshold = 0.25)     \n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e______________________________ NOT TO BE RUN________________________________\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Find markers for every cluster compared to all remaining cells, report only the positive ones\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindAllMarkers(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                          \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                          \u003cspan class=\"pl-v\"\u003elogfc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.25\u003c/span\u003e)     \n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWhen we have two groups like tumor vs normal or invasive vs. non-invasive identifying conserved markers is the best approach. In this way, we find DE genes for a given cluster in one condition (e.g. invasive) by comparing the cluster against the rest of the cluster in the same condition group. We do the same for that given cluster in the other condition (non-invasive). Finally, the two lists will be merged to give us the conserved marker for a given cluster.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# explecity set the default object to normalized values\nDefaultAssay(seurat_integrated) \u0026lt;- \u0026quot;RNA\u0026quot;\n\n\ncluster0_conserved_markers \u0026lt;- FindConservedMarkers(seurat_integrated,\n                              ident.1 = 0,\n                              grouping.var = \u0026quot;Invasiveness\u0026quot;,\n                                only.pos = TRUE,\n                             logfc.threshold = 0.60)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e explecity set the default object to normalized values\u003c/span\u003e\nDefaultAssay(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRNA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\n\u003cspan class=\"pl-smi\"\u003ecluster0_conserved_markers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindConservedMarkers(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n                              \u003cspan class=\"pl-v\"\u003eident.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\n                              \u003cspan class=\"pl-v\"\u003egrouping.var\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                             \u003cspan class=\"pl-v\"\u003elogfc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.60\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo add more annotation to the results\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Connect to AnnotationHub\nah \u0026lt;- AnnotationHub()\n\n# Access the Ensembl database for organism\nahDb \u0026lt;- query(ah, \n              pattern = c(\u0026quot;Homo sapiens\u0026quot;, \u0026quot;EnsDb\u0026quot;), \n              ignore.case = TRUE)\n\n# Acquire the latest annotation files\nid \u0026lt;- ahDb %\u0026gt;%\n        mcols() %\u0026gt;%\n        rownames() %\u0026gt;%\n        tail(n = 1)\n\n# Download the appropriate Ensembldb database\nedb \u0026lt;- ah[[id]]\n\n# Extract gene-level information from database\nannotations \u0026lt;- genes(edb, \n                     return.type = \u0026quot;data.frame\u0026quot;)\n\n# Select annotations of interest\nannotations \u0026lt;- annotations %\u0026gt;%\n        dplyr::select(gene_id, gene_name, seq_name, gene_biotype, description)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Connect to AnnotationHub\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e AnnotationHub()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Access the Ensembl database for organism\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eahDb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e query(\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e, \n              \u003cspan class=\"pl-v\"\u003epattern\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHomo sapiens\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEnsDb\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n              \u003cspan class=\"pl-v\"\u003eignore.case\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Acquire the latest annotation files\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eid\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eahDb\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        mcols() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        rownames() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        tail(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Download the appropriate Ensembldb database\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eedb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003eid\u003c/span\u003e]]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract gene-level information from database\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e genes(\u003cspan class=\"pl-smi\"\u003eedb\u003c/span\u003e, \n                     \u003cspan class=\"pl-k\"\u003ereturn\u003c/span\u003e\u003cspan class=\"pl-v\"\u003e.type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edata.frame\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Select annotations of interest\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003egene_id\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egene_name\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eseq_name\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egene_biotype\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edescription\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo find conserved markers for all clusters:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Create function to get conserved markers for any given cluster\nget_conserved \u0026lt;- function(cluster){\n  tryCatch({\n    FindConservedMarkers(seurat_integrated,\n                         ident.1 = cluster,\n                         grouping.var = \u0026quot;Invasiveness\u0026quot;,\n                         only.pos = TRUE,\n                         logfc.threshold = 0.60) %\u0026gt;%\n      rownames_to_column(var = \u0026quot;gene\u0026quot;) %\u0026gt;%\n      left_join(y = unique(annotations[, c(\u0026quot;gene_name\u0026quot;, \u0026quot;description\u0026quot;)]),\n                 by = c(\u0026quot;gene\u0026quot; = \u0026quot;gene_name\u0026quot;)) %\u0026gt;%\n      cbind(cluster_id = cluster, .)\n    },\n    error = function(e) {\n      message(paste0(\u0026quot;Error: \u0026quot;, e$message))\n      return(NULL)\n    }\n  )\n}\n# this function can be an argument for 'map_dfr' function :\n# Iterate function across desired clusters\nconserved_markers \u0026lt;- map_dfr(c(0:20), get_conserved)\n\n# Extract top 10 markers per cluster\ntop10 \u0026lt;- conserved_markers %\u0026gt;% \n  mutate(avg_fc = (Noninvasive_avg_log2FC + Invasive_avg_log2FC) /2) %\u0026gt;% \n  group_by(cluster_id) %\u0026gt;% \n  top_n(n = 10, \n        wt = avg_fc)\n        \nhead(top10)\n\ndata.table::fwrite(top10, \u0026quot;blca_top10_conserved_markers.csv\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create function to get conserved markers for any given cluster\u003c/span\u003e\n\u003cspan class=\"pl-en\"\u003eget_conserved\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e){\n  tryCatch({\n    FindConservedMarkers(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003eident.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003egrouping.var\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003elogfc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.60\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n      rownames_to_column(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n      left_join(\u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e unique(\u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e[, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene_name\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edescription\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)]),\n                 \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene_name\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n      cbind(\u003cspan class=\"pl-v\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e)\n    },\n    \u003cspan class=\"pl-en\"\u003eerror\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ee\u003c/span\u003e) {\n      message(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eError: \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ee\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emessage\u003c/span\u003e))\n      \u003cspan class=\"pl-k\"\u003ereturn\u003c/span\u003e(\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e)\n    }\n  )\n}\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e this function can be an argument for 'map_dfr' function :\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Iterate function across desired clusters\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003econserved_markers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e map_dfr(c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003eget_conserved\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract top 10 markers per cluster\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econserved_markers\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  mutate(\u003cspan class=\"pl-v\"\u003eavg_fc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003eNoninvasive_avg_log2FC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eInvasive_avg_log2FC\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  group_by(\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  top_n(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ewt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eavg_fc\u003c/span\u003e)\n        \nhead(\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efwrite(\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblca_top10_conserved_markers.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThere are a number of tools that one may use to assign cell types to a cluster. However almost none of them at the time of writing can help with bladder tissue. So I have to look up genes in the \u003ca href=\"https://panglaodb.se/index.html\" rel=\"nofollow\"\u003ePanglaoDB\u003c/a\u003e database manually and assign cell type to each cluster. So to do this job in a more efficient way, let's first identify which markers are associated with more clusters, then assign cell type to those clusters.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"top10_mod \u0026lt;- data.frame(unclass(table(top10$gene, top10$cluster_id)))\n\n\ndata.table::fwrite(top10_mod, \u0026quot;top10_mod_conserved_markers.csv\u0026quot;)\n\n# markers with highest frequency\nM \u0026lt;- c(\u0026quot;KRT7\u0026quot;, \u0026quot;KRT19\u0026quot;, \u0026quot;FCER1G\u0026quot;, \u0026quot;AIF1\u0026quot;, \u0026quot;AQP3\u0026quot;, \u0026quot;CCL5\u0026quot;, \u0026quot;CD24\u0026quot;, \u0026quot;CD3D\u0026quot;, \u0026quot;CD52\u0026quot;, \u0026quot;CLDN4\u0026quot;, \u0026quot;COL1A1\u0026quot;, \u0026quot;COL1A2\u0026quot;, \u0026quot;CRTAC1\u0026quot;, \u0026quot;CXCL8\u0026quot;, \u0026quot;DCN\u0026quot;, \u0026quot;FABP4\u0026quot;, \u0026quot;FABP5\u0026quot;, \u0026quot;FXYD3\u0026quot;, \u0026quot;GZMA\u0026quot;, \u0026quot;HLA-DRA\u0026quot;, \u0026quot;IGHA1\u0026quot;, \u0026quot;IGHG1\u0026quot;, \u0026quot;IGHG3\u0026quot;, \u0026quot;IGHG4\u0026quot;, \u0026quot;IGKC\u0026quot;, \u0026quot;IGLC1\u0026quot;, \u0026quot;IGLC2\u0026quot;, \u0026quot;IGLC3\u0026quot;, \u0026quot;JCHAIN\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003etop10_mod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(unclass(table(\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e)))\n\n\n\u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efwrite(\u003cspan class=\"pl-smi\"\u003etop10_mod\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etop10_mod_conserved_markers.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e markers with highest frequency\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eM\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKRT7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKRT19\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFCER1G\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAIF1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAQP3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCCL5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD24\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD3D\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD52\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCLDN4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCOL1A1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCOL1A2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCRTAC1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCXCL8\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDCN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFABP4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFABP5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFXYD3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGZMA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHLA-DRA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHA1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGKC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eJCHAIN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSo cell type for top markers:\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003ecell type\u003c/th\u003e\n\u003cth\u003egenes\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eBasal cells\u003c/td\u003e\n\u003ctd\u003eKRT7, KRT19, AQP3, CD24,CXCL8,FXYD3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDendritic cells\u003c/td\u003e\n\u003ctd\u003eFCER1G,AIF1,FABP4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGamma delta T cells\u003c/td\u003e\n\u003ctd\u003eCCL5,GZMA\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNK cell\u003c/td\u003e\n\u003ctd\u003eCCL5,GZMA\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eT cells\u003c/td\u003e\n\u003ctd\u003eCD3D, CD52\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMacrophages\u003c/td\u003e\n\u003ctd\u003eCD52\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eLuminal epithelial cells\u003c/td\u003e\n\u003ctd\u003eCLDN4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eFibroblasts\u003c/td\u003e\n\u003ctd\u003eCOLA1, COLA2,CXCL8, DCN\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eEpithelial cells\u003c/td\u003e\n\u003ctd\u003eCRTAC1,FXYD3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eEndothelial cells\u003c/td\u003e\n\u003ctd\u003eFABP4, FABP5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePlasma cell\u003c/td\u003e\n\u003ctd\u003eIGHA1,IGHG1,IGHG3,IGHG4,IGKC,IGLC1,IGLC3,IGLC3,JCHAIN\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eAlso, there is a table from \u003ca href=\"https://www.nature.com/articles/s41467-020-16162-3\" rel=\"nofollow\"\u003eJohn P. Sfakianos et al\u003c/a\u003e , which summarizes the tumor cell markers:\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003eCellType\u003c/th\u003e\n\u003cth\u003eGenes\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eLuminal\u003c/td\u003e\n\u003ctd\u003eCYP2J2,ERBB2,ERBB3,FGFR3,FOXA1,GATA3,GPX2,KRT18,KRT19,KRT20,KRT7,KRT8,PPARG,XBP1,UPK1A,UPK2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eEMT and smooth muscle\u003c/td\u003e\n\u003ctd\u003ePGM5,DES,C7,SRFP4,COMP,SGCD\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eEMT and claudin\u003c/td\u003e\n\u003ctd\u003eZEB1,ZEB2,VIM,SNAI1,TWIST1,FOXC2,CDH2,CLDN3,CLDN7,CLDN4,CDH1,SNAI2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eBasal\u003c/td\u003e\n\u003ctd\u003eCD44,CDH3,KRT1,KRT14,KRT16,KRT5,KRT6A,KRT6B,KRT6C\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eP53-like\u003c/td\u003e\n\u003ctd\u003eACTG2,CNN1,MYH11,MFAP4,PGM5,FLNC,ACTC1,DES,PCP4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSquamous\u003c/td\u003e\n\u003ctd\u003eDSC1,DSC2,DSC3,DSG1,DSG2,DSG3,S100A7,S100A8\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eImmune\u003c/td\u003e\n\u003ctd\u003eCD274,PDCD1LG2,IDO1,CXCL11,L1CAM,SAA1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNeuroendocrine\u003c/td\u003e\n\u003ctd\u003eCHGA,CHGB,SCG2,ENO2,SYP,NCAM1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNeuronal differentiation\u003c/td\u003e\n\u003ctd\u003eMSI1,PLEKHG4B,GNG4,PEG10,RND2,APLP1,SOX2,TUBB2B\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDownregulated CIS\u003c/td\u003e\n\u003ctd\u003eCRTAC1,CTSE,PADI3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eUpregulated CIS\u003c/td\u003e\n\u003ctd\u003eMSN,NR3C1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCancer stem cell\u003c/td\u003e\n\u003ctd\u003eCD44,KRT5,RPSA,ALDH1A1\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eLet's visualize some of the genes and see in which cluster they show expression.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n# Plot interesting marker gene expression \npng(filename = \u0026quot;umap_high_freq_basal_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated, \n                        features = c(\u0026quot;KRT7\u0026quot;, \u0026quot;KRT19\u0026quot;, \u0026quot;AQP3\u0026quot;, \u0026quot;CD24\u0026quot;, \u0026quot;FXYD3\u0026quot;, \u0026quot;CXCL8\u0026quot;),\n                         sort.cell = TRUE,\n                         min.cutoff = 'q10', \n                         label = TRUE,\n                         repel = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot interesting marker gene expression \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_high_freq_basal_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKRT7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKRT19\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAQP3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD24\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFXYD3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCXCL8\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n                         \u003cspan class=\"pl-v\"\u003esort.cell\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_high_freq_basal_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_high_freq_basal_cells.png\" alt=\"umap_high_freq_basal_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n# Vln plot - cluster 0\npng(filename = \u0026quot;violin_high_freq_basal_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nVlnPlot(object = seurat_integrated, \n        features = c(\u0026quot;KRT7\u0026quot;, \u0026quot;KRT19\u0026quot;, \u0026quot;AQP3\u0026quot;, \u0026quot;CD24\u0026quot;, \u0026quot;FXYD3\u0026quot;, \u0026quot;CXCL8\u0026quot;))\ndev.off() \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Vln plot - cluster 0\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eviolin_high_freq_basal_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nVlnPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKRT7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKRT19\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAQP3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD24\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFXYD3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCXCL8\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\ndev.off() \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_high_freq_basal_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_high_freq_basal_cells.png\" alt=\"violin_high_freq_basal_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSo according to \"basal cells\" visualization, the following clusters may show clusters of basal cells:\n0, 2,3,9,10,14,17,18,19(?),20 and 21.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Plot interesting marker gene expression \npng(filename = \u0026quot;umap_high_freq_Plasma_cells.png\u0026quot;, width = 26, height = 15.135, units = \u0026quot;in\u0026quot;, res = 600)\nFeaturePlot(object = seurat_integrated, \n                        features = c(\u0026quot;IGHA1\u0026quot;,\u0026quot;IGHG1\u0026quot;,\u0026quot;IGHG3\u0026quot;,\u0026quot;IGHG4\u0026quot;,\u0026quot;IGKC\u0026quot;,\u0026quot;IGLC1\u0026quot;,\u0026quot;IGLC3\u0026quot;,\u0026quot;IGLC3\u0026quot;,\u0026quot;JCHAIN\u0026quot;),\n                         sort.cell = TRUE,\n                         min.cutoff = 'q10', \n                         label = TRUE,\n                         repel = TRUE)\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot interesting marker gene expression \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_high_freq_Plasma_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e15.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e600\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHA1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGKC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eJCHAIN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n                         \u003cspan class=\"pl-v\"\u003esort.cell\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_high_freq_Plasma_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_high_freq_Plasma_cells.png\" alt=\"umap_high_freq_Plasma_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n# Vln plot - cluster 0\npng(filename = \u0026quot;violin_high_freq_basal_cells.png\u0026quot;, width = 26, height = 10.135, units = \u0026quot;in\u0026quot;, res = 600)\nVlnPlot(object = seurat_integrated, \n        features = c(\u0026quot;IGHA1\u0026quot;,\u0026quot;IGHG1\u0026quot;,\u0026quot;IGHG3\u0026quot;,\u0026quot;IGHG4\u0026quot;,\u0026quot;IGKC\u0026quot;,\u0026quot;IGLC1\u0026quot;,\u0026quot;IGLC3\u0026quot;,\u0026quot;IGLC3\u0026quot;,\u0026quot;JCHAIN\u0026quot;))\ndev.off() \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Vln plot - cluster 0\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eviolin_high_freq_basal_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e600\u003c/span\u003e)\nVlnPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHA1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGHG4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGKC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIGLC3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eJCHAIN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\ndev.off() \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_high_freq_basal_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_high_freq_basal_cells.png\" alt=\"violin_high_freq_basal_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCluster 19 looks to be hard to assign it a cell type. Will keep an eye on it.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n# Plot interesting marker gene expression \npng(filename = \u0026quot;umap_high_freq_dc.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated, \n                        features = c(\u0026quot;FCER1G\u0026quot;,\u0026quot;AIF1\u0026quot;,\u0026quot;FABP4\u0026quot;),\n                         sort.cell = TRUE,\n                         min.cutoff = 'q10', \n                         label = TRUE,\n                         repel = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot interesting marker gene expression \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_high_freq_dc.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFCER1G\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAIF1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFABP4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n                         \u003cspan class=\"pl-v\"\u003esort.cell\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_high_freq_dc.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_high_freq_dc.png\" alt=\"umap_high_freq_dc.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Vln plot - cluster 0\npng(filename = \u0026quot;violin_high_freq_dc.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nVlnPlot(object = seurat_integrated, \n        features = c(\u0026quot;FCER1G\u0026quot;,\u0026quot;AIF1\u0026quot;,\u0026quot;FABP4\u0026quot;))\ndev.off() \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Vln plot - cluster 0\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eviolin_high_freq_dc.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nVlnPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFCER1G\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAIF1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFABP4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\ndev.off() \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_high_freq_dc.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_high_freq_dc.png\" alt=\"violin_high_freq_dc.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFor cluster1 and 4\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"d \u0026lt;- data.table::fread(\u0026quot;top10_conserved_markers.csv\u0026quot;)\n\npng(filename = \u0026quot;umap_cluster4_markers.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated, \n                        features = d$gene[d$cluster_id == \u0026quot;4\u0026quot;],\n                         sort.cell = TRUE,\n                         min.cutoff = 'q10', \n                         label = TRUE,\n                         repel = TRUE)\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etop10_conserved_markers.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_cluster4_markers.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e],\n                         \u003cspan class=\"pl-v\"\u003esort.cell\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster4_markers.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster4_markers.png\" alt=\"umap_cluster4_markers.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n# Vln plot - cluster 0\npng(filename = \u0026quot;violin_cluster4_markers.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nVlnPlot(object = seurat_integrated, \n        features = d$gene[d$cluster_id == \u0026quot;4\u0026quot;])\ndev.off() \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Vln plot - cluster 0\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eviolin_cluster4_markers.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nVlnPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e])\ndev.off() \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_cluster4_markers.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_cluster4_markers.png\" alt=\"violin_cluster4_markers.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;umap_cluster12_markers.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated, \n                        features = d$gene[d$cluster_id == \u0026quot;12\u0026quot;],\n                         sort.cell = TRUE,\n                         min.cutoff = 'q10', \n                         label = TRUE,\n                         repel = TRUE)\ndev.off()\n\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_cluster12_markers.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e12\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e],\n                         \u003cspan class=\"pl-v\"\u003esort.cell\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster12_markers.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster12_markers.png\" alt=\"umap_cluster12_markers.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSo it seems clusters 1,4,8,12,13 and 16 are the same cell type with minor subtypes. However cluster 5 and 7 show some level of expression for markers in cluster 1 + 4, but they need separate inspection.\nConserved markers in these two clusters are as follows\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[1] \u0026quot;LDB2\u0026quot;    \u0026quot;SPARCL1\u0026quot; \u0026quot;GNG11\u0026quot;   \u0026quot;FLT1\u0026quot;    \u0026quot;IFI27\u0026quot;   \u0026quot;RAMP2\u0026quot;   \u0026quot;PECAM1\u0026quot;\n[8] \u0026quot;TCF4\u0026quot;    \u0026quot;PLVAP\u0026quot;   \u0026quot;PCAT19\u0026quot;  \u0026quot;ACKR1\u0026quot;   \u0026quot;SELE\u0026quot;    \u0026quot;CALCRL\u0026quot;  \u0026quot;ZNF385D\u0026quot;\n[15] \u0026quot;ADAMTS9\u0026quot; \u0026quot;MCTP1\u0026quot;   \u0026quot;AQP1\u0026quot;    \u0026quot;VWF\u0026quot;     \u0026quot;CCL14\u0026quot;   \u0026quot;PCAT19\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e[1] \"LDB2\"    \"SPARCL1\" \"GNG11\"   \"FLT1\"    \"IFI27\"   \"RAMP2\"   \"PECAM1\"\n[8] \"TCF4\"    \"PLVAP\"   \"PCAT19\"  \"ACKR1\"   \"SELE\"    \"CALCRL\"  \"ZNF385D\"\n[15] \"ADAMTS9\" \"MCTP1\"   \"AQP1\"    \"VWF\"     \"CCL14\"   \"PCAT19\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003ecell type\u003c/th\u003e\n\u003cth\u003egenes\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eEndothelial cells\u003c/td\u003e\n\u003ctd\u003eLDB2, SPARCL1, FLT1,IFI27,RAMP2,PECAM1,TCF4,PLVAP,PCAT19,ACKR1, SELE,CALCRL,ZNF385D,ADAMTS9,MCTP1,AQP1,VWF,\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eFibroblast\u003c/td\u003e\n\u003ctd\u003eSPARCL1,TCF4,PLVAP,ADAMTS9\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;umap_cluster5_7_markers.png\u0026quot;, width = 26, height = 18.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated, \n                        features = d$gene[d$cluster_id %in% c(\u0026quot;5\u0026quot;, \u0026quot;7\u0026quot;)],\n                         sort.cell = TRUE,\n                         min.cutoff = 'q10', \n                         label = TRUE,\n                         repel = TRUE)\ndev.off()\n\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_cluster5_7_markers.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e18.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)],\n                         \u003cspan class=\"pl-v\"\u003esort.cell\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster5_7_markers.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster5_7_markers.png\" alt=\"umap_cluster5_7_markers.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Vln plot - cluster 0\npng(filename = \u0026quot;violin_cluster5_7_markers.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nVlnPlot(object = seurat_integrated, \n        features = d$gene[d$cluster_id == \u0026quot;4\u0026quot;])\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Vln plot - cluster 0\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eviolin_cluster5_7_markers.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nVlnPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e])\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_cluster5_7_markers.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_cluster5_7_markers.png\" alt=\"violin_cluster5_7_markers.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSo as expected clusters 7 and 5 represent the same cell types.\nIn almost all cases cluster 17 showed some level of expression for EC markers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLet's now have a look at markers for clusters 6,15 and 11:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;umap_cluster11_15_6_markers.png\u0026quot;, width = 26, height = 18.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated, \n                        features = unique(d$gene[d$cluster_id %in% c(\u0026quot;6\u0026quot;, \u0026quot;11\u0026quot;,\u0026quot;15\u0026quot;)]),\n                         sort.cell = TRUE,\n                         min.cutoff = 'q10', \n                         label = TRUE,\n                         repel = TRUE)\ndev.off()\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_cluster11_15_6_markers.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e26\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e18.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                        \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e unique(\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e11\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e15\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)]),\n                         \u003cspan class=\"pl-v\"\u003esort.cell\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \n                         \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                         \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster11_15_6_markers.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster11_15_6_markers.png\" alt=\"umap_cluster11_15_6_markers.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAccording to the markers such as different collagen, clusters 11 and 15 seem to be fibroblasts while cluster 6 shows a mix of fibroblast and non-fibroblast markers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNow we can generate UMAP with cell type as labels;\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Rename all identities\nseurat_integrated \u0026lt;- RenameIdents(object = seurat_integrated, \n                               \u0026quot;0\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;1\u0026quot; = \u0026quot;Gamma delta T cell\u0026quot;, # NKG7 and GZMK\n                               \u0026quot;2\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;3\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;4\u0026quot; = \u0026quot;T cells\u0026quot;,\n                               \u0026quot;5\u0026quot; = \u0026quot;Endothelial cells\u0026quot;,\n                               \u0026quot;6\u0026quot; = \u0026quot;Fibroblast\u0026quot;,\n                               \u0026quot;7\u0026quot; = \u0026quot;Endothelial cells\u0026quot;,\n                               \u0026quot;8\u0026quot; = \u0026quot;DC\u0026quot;,\n                               \u0026quot;9\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;10\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;11\u0026quot; = \u0026quot;Fibroblast\u0026quot;,\n                               \u0026quot;12\u0026quot; = \u0026quot;T cells\u0026quot;,\n                               \u0026quot;13\u0026quot; = \u0026quot;T cells\u0026quot;,\n                               \u0026quot;14\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;15\u0026quot; = \u0026quot;Fibroblast\u0026quot;,\n                               \u0026quot;16\u0026quot; = \u0026quot;T cells\u0026quot;,\n                               \u0026quot;17\u0026quot; = \u0026quot;?\u0026quot;, \n                               \u0026quot;18\u0026quot; = \u0026quot;Basal cells\u0026quot;, \n                               \u0026quot;19\u0026quot; = \u0026quot;Plasma cells\u0026quot;,\n                               \u0026quot;20\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;21\u0026quot; = \u0026quot;Basal cells\u0026quot;)\n\n# Plot the UMAP withy new labells\npng(filename = \u0026quot;umap_with_label.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 600)\nDimPlot(object = seurat_integrated, \n        reduction = \u0026quot;umap\u0026quot;, \n        label = TRUE,\n        label.size = 3,\n        repel = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Rename all identities\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RenameIdents(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGamma delta T cell\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e NKG7 and GZMK\u003c/span\u003e\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEndothelial cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFibroblast\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEndothelial cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e9\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e11\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFibroblast\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e12\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e13\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e15\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFibroblast\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e17\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e?\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e19\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePlasma cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e20\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e21\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the UMAP withy new labells\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_with_label.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e600\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_with_label.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_with_label.png\" alt=\"umap_with_label.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOK then it's time to write the Seurat object for later analysis, if any:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Save final R object\nwrite_rds(seurat_integrated,\n          path = \u0026quot;seurat_labelled.rds\u0026quot;)      \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Save final R object\u003c/span\u003e\nwrite_rds(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n          \u003cspan class=\"pl-v\"\u003epath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eseurat_labelled.rds\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)      \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAs a final note I tried to assign cell types to clusters using the scType tool, however, there was no bladder tissue profile in their database. So the generated labels using this approach should be only considered as a guide.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Cell maker assignemnt using scType\n\n# load libraries\nlibrary(HGNChelper)\n\n\n# load gene set preparation function\nsource(\u0026quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R\u0026quot;)\n# load cell type annotation function\nsource(\u0026quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R\u0026quot;)\n\ndb_ = \u0026quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx\u0026quot;;\n#tissue = c(\u0026quot;Immune system\u0026quot;) # e.g. Immune system,Pancreas,Liver,Eye,Kidney,Brain,Lung,Adrenal,Heart,Intestine,Muscle,Placenta,Spleen,Stomach,Thymus \ntissue = c(\u0026quot;Immune system\u0026quot;, \u0026quot;Pancreas\u0026quot;, \u0026quot;Liver\u0026quot;,\u0026quot;Kidney\u0026quot;,\u0026quot;Intestine\u0026quot;,\u0026quot;Placenta\u0026quot;,\u0026quot;Spleen\u0026quot;,\n           \u0026quot;Stomach\u0026quot;) \n\n# prepare gene sets\ngs_list = gene_sets_prepare(db_, tissue)\n\n# get cell-type by cell matrix\nes.max = sctype_score(scRNAseqData = seurat_integrated[[\u0026quot;integrated\u0026quot;]]@scale.data, scaled = TRUE, \n                      gs = gs_list$gs_positive, gs2 = gs_list$gs_negative) \n\n# merge by cluster\ncL_resutls = do.call(\n  \u0026quot;rbind\u0026quot;, lapply(unique(seurat_integrated@meta.data$seurat_clusters), function(cl){\n  es.max.cl = sort(rowSums(es.max[ ,rownames(seurat_integrated@meta.data[seurat_integrated@meta.data$seurat_clusters==cl, ])]), decreasing = !0)\n  head(data.frame(cluster = cl, type = names(es.max.cl), scores = es.max.cl, ncells = sum(seurat_integrated@meta.data$seurat_clusters==cl)), 10)\n}))\n\n\nsctype_scores = cL_resutls %\u0026gt;% group_by(cluster) %\u0026gt;% top_n(n = 1, wt = scores)\n\n# set low-confident (low ScType score) clusters to \u0026quot;unknown\u0026quot;\nsctype_scores$type[as.numeric(as.character(sctype_scores$scores)) \u0026lt; sctype_scores$ncells/4] = \u0026quot;Unknown\u0026quot;\nprint(sctype_scores[,1:3])\n\n\n\n# overlay the identified cell types on UMAP plot:\n\nseurat_integrated@meta.data$customclassif = \u0026quot;\u0026quot;\nfor(j in unique(sctype_scores$cluster)){\n  cl_type = sctype_scores[sctype_scores$cluster==j,]; \n  seurat_integrated@meta.data$customclassif[seurat_integrated@meta.data$seurat_clusters == j] = as.character(cl_type$type[1])\n}\n\npng(filename = \u0026quot;umap_with_label_scType.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 600)\nDimPlot(seurat_integrated, \n        reduction = \u0026quot;umap\u0026quot;, \n        label = TRUE, \n        repel = TRUE, \n        group.by = 'customclassif')        \n\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Cell maker assignemnt using scType\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e load libraries\u003c/span\u003e\nlibrary(\u003cspan class=\"pl-smi\"\u003eHGNChelper\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e load gene set preparation function\u003c/span\u003e\nsource(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e load cell type annotation function\u003c/span\u003e\nsource(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003edb_\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehttps://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e;\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etissue = c(\"Immune system\") # e.g. Immune system,Pancreas,Liver,Eye,Kidney,Brain,Lung,Adrenal,Heart,Intestine,Muscle,Placenta,Spleen,Stomach,Thymus \u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003etissue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eImmune system\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePancreas\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLiver\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKidney\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIntestine\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePlacenta\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSpleen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n           \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eStomach\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e prepare gene sets\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003egs_list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e gene_sets_prepare(\u003cspan class=\"pl-smi\"\u003edb_\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etissue\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e get cell-type by cell matrix\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003ees.max\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sctype_score(\u003cspan class=\"pl-v\"\u003escRNAseqData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eintegrated\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escale.data\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escaled\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n                      \u003cspan class=\"pl-v\"\u003egs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egs_list\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egs_positive\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egs2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egs_list\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egs_negative\u003c/span\u003e) \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e merge by cluster\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003ecL_resutls\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e do.call(\n  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erbind\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, lapply(unique(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseurat_clusters\u003c/span\u003e), \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e){\n  \u003cspan class=\"pl-v\"\u003ees.max.cl\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sort(rowSums(\u003cspan class=\"pl-smi\"\u003ees.max\u003c/span\u003e[ ,rownames(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseurat_clusters\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e, ])]), \u003cspan class=\"pl-v\"\u003edecreasing\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\n  head(\u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-v\"\u003ecluster\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e names(\u003cspan class=\"pl-smi\"\u003ees.max.cl\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003escores\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ees.max.cl\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003encells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sum(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseurat_clusters\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecl\u003c/span\u003e)), \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e)\n}))\n\n\n\u003cspan class=\"pl-v\"\u003esctype_scores\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecL_resutls\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% group_by(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% top_n(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003escores\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e set low-confident (low ScType score) clusters to \"unknown\"\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esctype_scores\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etype\u003c/span\u003e[as.numeric(as.character(\u003cspan class=\"pl-smi\"\u003esctype_scores\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escores\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esctype_scores\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003encells\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUnknown\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nprint(\u003cspan class=\"pl-smi\"\u003esctype_scores\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e])\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e overlay the identified cell types on UMAP plot:\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecustomclassif\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e unique(\u003cspan class=\"pl-smi\"\u003esctype_scores\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e)){\n  \u003cspan class=\"pl-v\"\u003ecl_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esctype_scores\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003esctype_scores\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e,]; \n  \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecustomclassif\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseurat_clusters\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ej\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e as.character(\u003cspan class=\"pl-smi\"\u003ecl_type\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etype\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e])\n}\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_with_label_scType.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e600\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecustomclassif\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)        \n\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_with_label_scType.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_with_label_scType.png\" alt=\"umap_with_label_scType.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e8) MIBC vs. NMIBC\u003c/h1\u003e\u003ca id=\"user-content-8-mibc-vs-nmibc\" class=\"anchor\" aria-label=\"Permalink: 8) MIBC vs. NMIBC\" href=\"#8-mibc-vs-nmibc\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThere are four NMIBC samples(SRR12603790,SRR12603789,SRR12603787,SRR12603786) and four MIBC samples(SRR12603785,SRR12603784,SRR12603783,SRR12603782) in the dataset. Comparing these two group against each other could have given a clue on the invasion cell type signature.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# load object\nload(\u0026quot;~/scRNA/seurat_filtered.RData\u0026quot;)\n\n\n# reading sample information\nsampleInformation \u0026lt;- read.csv(\u0026quot;~/scRNA/sampleInfo.csv\u0026quot;)\n\nfiltered_seurat@meta.data \u0026lt;- left_join(filtered_seurat@meta.data, sampleInformation, by = \u0026quot;orig.ident\u0026quot;)\n# save meta data as a df\nmetaData \u0026lt;- filtered_seurat@meta.data\n\n\n# setting ident\nIdents(filtered_seurat) \u0026lt;- \u0026quot;sample\u0026quot;\n\n# keeping only cancer samples\nblca_seurat \u0026lt;- subset(filtered_seurat, idents = \u0026quot;BLCA\u0026quot;)\nrm(filtered_seurat)\n\n# Repair the meta.data\nblca_seurat@meta.data$cells \u0026lt;- blca_seurat@assays$RNA@data@Dimnames[[2]]\nblca_seurat@meta.data$orig.ident \u0026lt;- substr(blca_seurat@assays$RNA@data@Dimnames[[2]],1,11)\nblca_seurat@meta.data$seq_folder \u0026lt;- substr(blca_seurat@assays$RNA@data@Dimnames[[2]],1,11)\nblca_seurat@meta.data$sample \u0026lt;- \u0026quot;BLCA\u0026quot;\n\n# set rownames\nrownames(blca_seurat@meta.data) \u0026lt;- blca_seurat@meta.data$cells\n\n\n# remove columns fillied by NA\ncol_remove \u0026lt;- c(\u0026quot;nUMI\u0026quot;, \u0026quot;nGene\u0026quot;, \u0026quot;log10GenesPerUMI\u0026quot;, \u0026quot;gender\u0026quot;, \u0026quot;age\u0026quot;, \u0026quot;Grade\u0026quot;, \u0026quot;Invasiveness\u0026quot;, \u0026quot;Surgery_Type\u0026quot;, \u0026quot;Tumor_size_cm\u0026quot;, \u0026quot;mitoRatio\u0026quot;)\n\n# \nblca_seurat@meta.data \u0026lt;- blca_seurat@meta.data[, !(colnames(blca_seurat@meta.data) %in% col_remove)]\n\n# adding more meta.data again\nblca_seurat@meta.data \u0026lt;- left_join(blca_seurat@meta.data, sampleInformation, by = \u0026quot;orig.ident\u0026quot;)\n\n\n# Setting metadata rownames as the column names of expression matrix\nrownames(blca_seurat@meta.data) \u0026lt;- blca_seurat@meta.data$cells\n\n# SCTransform\n\n# Split seurat object by group\n\n# setting ident\nIdents(blca_seurat) \u0026lt;- \u0026quot;Invasiveness\u0026quot;\nblca_split \u0026lt;- SplitObject(blca_seurat)\n\n\nblca_split \n#$Invasive\n#An object of class Seurat \n#29686 features across 41242 samples within 1 assay \n#Active assay: RNA (29686 features, 0 variable features)\n#\n#$Noninvasive\n#An object of class Seurat \n#29686 features across 35891 samples within 1 assay \n#Active assay: RNA (29686 features, 0 variable features)\n\n\n# Repairing meta.data for each object in the splitted obj:\n\nfor(i in 1:length(blca_split)){\n     print(blca_split[[i]])\n     obj = blca_split[[i]]\n     obj@meta.data$cells \u0026lt;- obj@assays$RNA@data@Dimnames[[2]]\n     obj@meta.data \u0026lt;- obj@meta.data[, c(2,3,5)]\n     obj@meta.data$orig.ident \u0026lt;- substr(obj@assays$RNA@data@Dimnames[[2]],1,11)\n     obj@meta.data$seq_folder \u0026lt;- substr(obj@assays$RNA@data@Dimnames[[2]],1,11)\n     obj@meta.data \u0026lt;- left_join(obj@meta.data, sampleInformation, by = \u0026quot;orig.ident\u0026quot;)\n     obj@meta.data \u0026lt;- left_join(obj@meta.data, metaData[,c(8,9)], by = \u0026quot;cells\u0026quot;)\n     # repairng metadata rownames\n     if(all(obj@meta.data$cells == colnames(obj@assays$RNA@counts))){\n     rownames(obj@meta.data) \u0026lt;- obj@meta.data$cells\n     }\n     blca_split[[i]] = obj\n }\n\n#saveRDS(blca_split, \u0026quot;blca_split.rds\u0026quot;)\n# then normalize by SCTansform\n# orig.ident is slected ti be regressed out because each sample was sequenced in seperate batch\n\nfor (i in 1:length(blca_split)) {\n    blca_split[[i]] \u0026lt;- SCTransform(blca_split[[i]], vars.to.regress = c(\u0026quot;orig.ident\u0026quot;), vst.flavor = \u0026quot;v2\u0026quot;)\n    }\n\n# Integration : integrating samples belong to one group\n\n# Select the most variable features to use for integration\ninteg_features \u0026lt;- SelectIntegrationFeatures(object.list = blca_split, \n                                            nfeatures = 3000) \n\n\n# Prepare the SCT list object for integration\nsplit_seurat \u0026lt;- PrepSCTIntegration(object.list = blca_split, \n                                   anchor.features = integ_features)\n                                   \n# Find best buddies (using canonical correlation analysis or CCA) - can take a while to run\ninteg_anchors \u0026lt;- FindIntegrationAnchors(object.list = split_seurat, \n                                        normalization.method = \u0026quot;SCT\u0026quot;, \n                                        anchor.features = integ_features)\n# Integrate across conditions\nseurat_integrated \u0026lt;- IntegrateData(anchorset = integ_anchors, \n                                   normalization.method = \u0026quot;SCT\u0026quot;)\n                                   \n\n# Integration check\n# Run PCA\nseurat_integrated \u0026lt;- RunPCA(object = seurat_integrated, verbose = TRUE)\n\n# Plot PCA\npng(filename = \u0026quot;blca_PCA_integrated.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nPCAPlot(seurat_integrated,\n        split.by = \u0026quot;Invasiveness\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e load object\u003c/span\u003e\nload(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e~/scRNA/seurat_filtered.RData\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e reading sample information\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.csv(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e~/scRNA/sampleInfo.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e left_join(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e save meta data as a df\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetaData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e setting ident\u003c/span\u003e\nIdents(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e keeping only cancer samples\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e subset(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eidents\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nrm(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Repair the meta.data\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDimnames\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]]\n\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eorig.ident\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e substr(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDimnames\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]],\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e substr(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDimnames\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]],\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBLCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e set rownames\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e remove columns fillied by NA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecol_remove\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enUMI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enGene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elog10GenesPerUMI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egender\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eage\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGrade\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSurgery_Type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTumor_size_cm\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoRatio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e \u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e[, \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e(colnames(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecol_remove\u003c/span\u003e)]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e adding more meta.data again\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e left_join(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Setting metadata rownames as the column names of expression matrix\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e SCTransform\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Split seurat object by group\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e setting ident\u003c/span\u003e\nIdents(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e SplitObject(\u003cspan class=\"pl-smi\"\u003eblca_seurat\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e$Invasive\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eAn object of class Seurat \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e29686 features across 41242 samples within 1 assay \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eActive assay: RNA (29686 features, 0 variable features)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e$Noninvasive\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eAn object of class Seurat \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e29686 features across 35891 samples within 1 assay \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eActive assay: RNA (29686 features, 0 variable features)\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Repairing meta.data for each object in the splitted obj:\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e)){\n     print(\u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]])\n     \u003cspan class=\"pl-v\"\u003eobj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]]\n     \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDimnames\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]]\n     \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e[, c(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)]\n     \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eorig.ident\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e substr(\u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDimnames\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]],\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e)\n     \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eseq_folder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e substr(\u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDimnames\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e]],\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e)\n     \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e left_join(\u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n     \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e left_join(\u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emetaData\u003c/span\u003e[,c(\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e)], \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n     \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e repairng metadata rownames\u003c/span\u003e\n     \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e(all(\u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eassays\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRNA\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e))){\n     rownames(\u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e\n     }\n     \u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]] \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eobj\u003c/span\u003e\n }\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esaveRDS(blca_split, \"blca_split.rds\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e then normalize by SCTansform\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e orig.ident is slected ti be regressed out because each sample was sequenced in seperate batch\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003elength(\u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e)) {\n    \u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e SCTransform(\u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e]], \u003cspan class=\"pl-v\"\u003evars.to.regress\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003evst.flavor\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ev2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    }\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Integration : integrating samples belong to one group\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Select the most variable features to use for integration\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003einteg_features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e SelectIntegrationFeatures(\u003cspan class=\"pl-v\"\u003eobject.list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e, \n                                            \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3000\u003c/span\u003e) \n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Prepare the SCT list object for integration\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e PrepSCTIntegration(\u003cspan class=\"pl-v\"\u003eobject.list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eblca_split\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003eanchor.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003einteg_features\u003c/span\u003e)\n                                   \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Find best buddies (using canonical correlation analysis or CCA) - can take a while to run\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003einteg_anchors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindIntegrationAnchors(\u003cspan class=\"pl-v\"\u003eobject.list\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esplit_seurat\u003c/span\u003e, \n                                        \u003cspan class=\"pl-v\"\u003enormalization.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                                        \u003cspan class=\"pl-v\"\u003eanchor.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003einteg_features\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Integrate across conditions\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e IntegrateData(\u003cspan class=\"pl-v\"\u003eanchorset\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003einteg_anchors\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003enormalization.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n                                   \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Integration check\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Run PCA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunPCA(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot PCA\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblca_PCA_integrated.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nPCAPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_PCA_integrated.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_PCA_integrated.png\" alt=\"blca_PCA_integrated.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n# Run UMAP\nseurat_integrated \u0026lt;- RunUMAP(seurat_integrated, \n                             dims = 1:40,\n                             reduction = \u0026quot;pca\u0026quot;,\n                             verbose = TRUE)\n\n# Plot UMAP \npng(filename = \u0026quot;blca_UMAP_integrated.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated, split.by = \u0026quot;Invasiveness\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Run UMAP\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunUMAP(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                             \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e40\u003c/span\u003e,\n                             \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                             \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot UMAP \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblca_UMAP_integrated.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_UMAP_integrated.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_UMAP_integrated.png\" alt=\"blca_UMAP_integrated.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Cluster the cells\n\n# to check what is active assay\nDefaultAssay(object = seurat_integrated)\n\n# Determine the K-nearest neighbor graph\nseurat_integrated \u0026lt;- FindNeighbors(object = seurat_integrated, \n                                dims = 1:18)\n                                \n#Find clusters\n# Determine the clusters for various resolutions                                \nseurat_integrated \u0026lt;- FindClusters(object = seurat_integrated,\n                               resolution = c(0.2,0.4, 0.6, 0.8, 1.0, 1.4))\n                               \n\n# Explore resolutions\nhead(seurat_integrated@meta.data)\n\n# Assign identity of clusters\nIdents(object = seurat_integrated) \u0026lt;- \u0026quot;integrated_snn_res.0.2\u0026quot;\n\n# Plot the UMAP\npng(filename = \u0026quot;blca_umap_cluster_with_label.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Cluster the cells\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to check what is active assay\u003c/span\u003e\nDefaultAssay(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine the K-nearest neighbor graph\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindNeighbors(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                                \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e18\u003c/span\u003e)\n                                \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eFind clusters\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Determine the clusters for various resolutions                                \u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindClusters(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n                               \u003cspan class=\"pl-v\"\u003eresolution\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1.0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1.4\u003c/span\u003e))\n                               \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Explore resolutions\u003c/span\u003e\nhead(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Assign identity of clusters\u003c/span\u003e\nIdents(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eintegrated_snn_res.0.2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the UMAP\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblca_umap_cluster_with_label.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_umap_cluster_with_label.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_umap_cluster_with_label.png\" alt=\"blca_umap_cluster_with_label.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# UMAP of cells in each cluster by invasiveness group\n# This would allow us to see condition specefic clusters\npng(filename = \u0026quot;blca_umap_cluster_sample.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated, \n        label = TRUE, \n        split.by = \u0026quot;Invasiveness\u0026quot;)  + NoLegend()\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e UMAP of cells in each cluster by invasiveness group\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This would allow us to see condition specefic clusters\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblca_umap_cluster_sample.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e NoLegend()\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_umap_cluster_sample.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_umap_cluster_sample.png\" alt=\"blca_umap_cluster_sample.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Explore whether clusters segregate by cell cycle phase\n\npng(filename = \u0026quot;grade_umap_cluster_cell_cucle.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(seurat_integrated,\n        label = TRUE, \n        split.by = \u0026quot;Grade\u0026quot;)  + NoLegend()\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Explore whether clusters segregate by cell cycle phase\u003c/span\u003e\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egrade_umap_cluster_cell_cucle.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGrade\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e NoLegend()\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/grade_umap_cluster_cell_cucle.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/grade_umap_cluster_cell_cucle.png\" alt=\"grade_umap_cluster_cell_cucle.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#Marker identification\n                  \n\n# adding annotation to the genes:\n# Connect to AnnotationHub\nah \u0026lt;- AnnotationHub()\n\n# Access the Ensembl database for organism\nahDb \u0026lt;- query(ah, \n              pattern = c(\u0026quot;Homo sapiens\u0026quot;, \u0026quot;EnsDb\u0026quot;), \n              ignore.case = TRUE)\n\n# Acquire the latest annotation files\nid \u0026lt;- ahDb %\u0026gt;%\n        mcols() %\u0026gt;%\n        rownames() %\u0026gt;%\n        tail(n = 1)\n\n# Download the appropriate Ensembldb database\nedb \u0026lt;- ah[[id]]\n\n# Extract gene-level information from database\nannotations \u0026lt;- genes(edb, \n                     return.type = \u0026quot;data.frame\u0026quot;)\n\n# Select annotations of interest\nannotations \u0026lt;- annotations %\u0026gt;%\n        dplyr::select(gene_id, gene_name, seq_name, gene_biotype, description)\n        \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eMarker identification\u003c/span\u003e\n                  \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e adding annotation to the genes:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Connect to AnnotationHub\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e AnnotationHub()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Access the Ensembl database for organism\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eahDb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e query(\u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e, \n              \u003cspan class=\"pl-v\"\u003epattern\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHomo sapiens\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEnsDb\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n              \u003cspan class=\"pl-v\"\u003eignore.case\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Acquire the latest annotation files\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eid\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eahDb\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        mcols() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        rownames() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        tail(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Download the appropriate Ensembldb database\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eedb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eah\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003eid\u003c/span\u003e]]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract gene-level information from database\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e genes(\u003cspan class=\"pl-smi\"\u003eedb\u003c/span\u003e, \n                     \u003cspan class=\"pl-k\"\u003ereturn\u003c/span\u003e\u003cspan class=\"pl-v\"\u003e.type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edata.frame\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Select annotations of interest\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003egene_id\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egene_name\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eseq_name\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egene_biotype\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edescription\u003c/span\u003e)\n        \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFinding markers for all clusters:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Create function to get conserved markers for any given cluster\nget_conserved \u0026lt;- function(cluster){\n  tryCatch({\n  FindConservedMarkers(seurat_integrated,\n                       ident.1 = cluster,\n                       grouping.var = \u0026quot;Invasiveness\u0026quot;,\n                       only.pos = TRUE,\n                       logfc.threshold = 0.60) %\u0026gt;%\n    rownames_to_column(var = \u0026quot;gene\u0026quot;) %\u0026gt;%\n    left_join(y = unique(annotations[, c(\u0026quot;gene_name\u0026quot;, \u0026quot;description\u0026quot;)]),\n               by = c(\u0026quot;gene\u0026quot; = \u0026quot;gene_name\u0026quot;)) %\u0026gt;%\n    cbind(cluster_id = cluster, .)\n  },\n  error = function(e){\n  message(paste0(\u0026quot;Error: \u0026quot;, e$message))\n    }\n   )\n  }\n  \n# this function can be an argument for 'map_dfr' function :\n# Iterate function across desired clusters\nconserved_markers \u0026lt;- map_dfr(c(0:20), get_conserved)\n\n# Extract top 10 markers per cluster\ntop10 \u0026lt;- conserved_markers %\u0026gt;% \n  mutate(avg_fc = (Normal_avg_log2FC + BLCA_avg_log2FC) /2) %\u0026gt;% \n  group_by(cluster_id) %\u0026gt;% \n  top_n(n = 10, \n        wt = avg_fc)\n        \nhead(top10)\n\n#data.table::fwrite(top10, \u0026quot;top10_conserved_markers.csv\u0026quot;)\n#group based on the clsuter id to define cell types:\ndata_grouped \u0026lt;- top10 %\u0026gt;%\n  group_by(cluster_id) %\u0026gt;%\n  summarize(gene = paste(gene, collapse = \u0026quot;,\u0026quot;))\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Create function to get conserved markers for any given cluster\u003c/span\u003e\n\u003cspan class=\"pl-en\"\u003eget_conserved\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e){\n  tryCatch({\n  FindConservedMarkers(\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n                       \u003cspan class=\"pl-v\"\u003eident.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e,\n                       \u003cspan class=\"pl-v\"\u003egrouping.var\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                       \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                       \u003cspan class=\"pl-v\"\u003elogfc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.60\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rownames_to_column(\u003cspan class=\"pl-v\"\u003evar\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    left_join(\u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e unique(\u003cspan class=\"pl-smi\"\u003eannotations\u003c/span\u003e[, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene_name\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edescription\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)]),\n               \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egene_name\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    cbind(\u003cspan class=\"pl-v\"\u003ecluster_id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e)\n  },\n  \u003cspan class=\"pl-en\"\u003eerror\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ee\u003c/span\u003e){\n  message(paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eError: \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ee\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emessage\u003c/span\u003e))\n    }\n   )\n  }\n  \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e this function can be an argument for 'map_dfr' function :\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Iterate function across desired clusters\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003econserved_markers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e map_dfr(c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003eget_conserved\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract top 10 markers per cluster\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econserved_markers\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  mutate(\u003cspan class=\"pl-v\"\u003eavg_fc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003eNormal_avg_log2FC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBLCA_avg_log2FC\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  group_by(\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  top_n(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ewt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eavg_fc\u003c/span\u003e)\n        \nhead(\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003edata.table::fwrite(top10, \"top10_conserved_markers.csv\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003egroup based on the clsuter id to define cell types:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edata_grouped\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  group_by(\u003cspan class=\"pl-smi\"\u003ecluster_id\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  summarize(\u003cspan class=\"pl-v\"\u003egene\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003c/pre\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003ecluster_id\u003c/th\u003e\n\u003cth\u003egenes\u003c/th\u003e\n\u003cth\u003ecell_type (PanglaoDB + ChatGPT)\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e0\u003c/td\u003e\n\u003ctd\u003eSFN,S100A2,CXCL8,CLDN4,AQP3,PHLDA2,EMP1,KRT13,KRT17,SLPI\u003c/td\u003e\n\u003ctd\u003ebasal cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e1\u003c/td\u003e\n\u003ctd\u003eSPINK1,CD24,UBE2C,UCA1,ADIRF,C15orf48,CSTB,S100A9,CCND1,HIST1H4C\u003c/td\u003e\n\u003ctd\u003eCCND1 + cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e2\u003c/td\u003e\n\u003ctd\u003eCD52,CXCR4,CD3D,CD69,IFNG,TRAC,IL32,CCL5,CCL4,NKG7\u003c/td\u003e\n\u003ctd\u003eT cells/NK cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e3\u003c/td\u003e\n\u003ctd\u003ePSCA,LY6D,KRT7,DHRS2,KRT13,HES1,KRT20,CRH,GCLC,RPS19\u003c/td\u003e\n\u003ctd\u003ebasal/squamous-like cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e4\u003c/td\u003e\n\u003ctd\u003eHSPG2,LDB2,IGFBP7,SPARCL1,GNG11,VWF,FLT1,RAMP2,INSR,PLVAP\u003c/td\u003e\n\u003ctd\u003eND\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e5\u003c/td\u003e\n\u003ctd\u003eCD52,CD2,LINC01871,GZMA,TRBC2,CD3D,TRAC,CORO1A,CCL5,NKG7\u003c/td\u003e\n\u003ctd\u003ecytotoxic T cell\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e6\u003c/td\u003e\n\u003ctd\u003eFABP4,CTSE,AKR1C3,PLA2G2A,FABP5,CRTAC1,GSTM3,KRT7,S100A6,CLU\u003c/td\u003e\n\u003ctd\u003eCAFs\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e7\u003c/td\u003e\n\u003ctd\u003eSNTG1,ZFPM2-AS1,RHEX,MT-ATP6,MT-ND5,CCSER1,FRY,SPINK1\u003c/td\u003e\n\u003ctd\u003eND\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e8\u003c/td\u003e\n\u003ctd\u003eRGS5,RGS5,IGFBP7,CALD1,ACTA2,TAGLN,THY1,NDUFA4L2,MYL9,BGN\u003c/td\u003e\n\u003ctd\u003efibroblasts/myofibroblasts\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e10\u003c/td\u003e\n\u003ctd\u003eC1QA,C1QB,CD74,HLA-DRA,HLA-DRB1,HLA-DPA1,HLA-DPB1,LYZ,TYROBP,APOE\u003c/td\u003e\n\u003ctd\u003eAPCs\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e11\u003c/td\u003e\n\u003ctd\u003eCOL3A1,CXCL14,COL1A2,RARRES2,PTGDS,MGP,LUM,DCN,MT2A,COL1A1\u003c/td\u003e\n\u003ctd\u003eFbroblasts(?)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e13\u003c/td\u003e\n\u003ctd\u003eCPA3,HPGDS,LTC4S,MS4A2,CTSG,TPSB2,TPSAB1,CD69,SRGN,RGS1\u003c/td\u003e\n\u003ctd\u003eMast cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e14\u003c/td\u003e\n\u003ctd\u003eAFF3,BANK1,LTB,HLA-DRA,CD79A,CD37,CD74,CD52,CXCR4,CD83\u003c/td\u003e\n\u003ctd\u003eB cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e17\u003c/td\u003e\n\u003ctd\u003eJCHAIN,IGHG1,IGHG2,IGKC,IGHG3,IGLC3,IGLC2,IGHA1,IGHM,IGHA2\u003c/td\u003e\n\u003ctd\u003eB cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Plot interesting marker gene expression \npng(filename = \u0026quot;cd24_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated,\n            features = c('SPINK1','CD24','UBE2C','UCA1','ADIRF','C15orf48','CSTB','S100A9','CCND1','HIST1H4C'),\n            order = TRUE,\n            min.cutoff = \u0026quot;q10\u0026quot;,\n            label = TRUE,\n            repel = TRUE)\n\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot interesting marker gene expression \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecd24_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSPINK1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCD24\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eUBE2C\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eUCA1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eADIRF\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eC15orf48\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCSTB\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eS100A9\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCCND1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHIST1H4C\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cd24_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cd24_cells.png\" alt=\"cd24_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;lyz_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = seurat_integrated,\n            features = c('C1QA','C1QB','CD74','HLA-DRA','HLA-DRB1','HLA-DPA1','HLA-DPB1','LYZ','TYROBP','APOE'),\n            order = TRUE,\n            min.cutoff = \u0026quot;q10\u0026quot;,\n            label = TRUE,\n            repel = TRUE)\n\ndev.off()\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elyz_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eC1QA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eC1QB\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCD74\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHLA-DRA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHLA-DRB1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHLA-DPA1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHLA-DPB1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eLYZ\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eTYROBP\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eAPOE\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e),\n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/lyz_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/lyz_cells.png\" alt=\"lyz_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Rename all identities\nseurat_integrated \u0026lt;- RenameIdents(object = seurat_integrated, \n                               \u0026quot;0\u0026quot; = \u0026quot;Basal cells\u0026quot;,\n                               \u0026quot;1\u0026quot; = \u0026quot;CCND1 + cells\u0026quot;, # NKG7 and GZMK\n                               \u0026quot;2\u0026quot; = \u0026quot;T cells/NK cells\u0026quot;,\n                               \u0026quot;3\u0026quot; = \u0026quot;basal/squamous-like cells\u0026quot;,\n                               \u0026quot;4\u0026quot; = \u0026quot;ND\u0026quot;,\n                               \u0026quot;5\u0026quot; = \u0026quot;cytotoxic T cell\u0026quot;,\n                               \u0026quot;6\u0026quot; = \u0026quot;CAFs\u0026quot;,\n                               \u0026quot;7\u0026quot; = \u0026quot;ND\u0026quot;,\n                               \u0026quot;8\u0026quot; = \u0026quot;fibroblasts/myofibroblasts\u0026quot;,\n                               \u0026quot;9\u0026quot; = \u0026quot;ND\u0026quot;,\n                               \u0026quot;10\u0026quot; = \u0026quot;fibroblasts/myofibroblasts\u0026quot;,\n                               \u0026quot;11\u0026quot; = \u0026quot;Fibroblast(?)\u0026quot;,\n                               \u0026quot;12\u0026quot; = \u0026quot;ND\u0026quot;,\n                               \u0026quot;13\u0026quot; = \u0026quot;Mast cells\u0026quot;,\n                               \u0026quot;14\u0026quot; = \u0026quot;B cells\u0026quot;,\n                               \u0026quot;15\u0026quot; = \u0026quot;ND\u0026quot;,\n                               \u0026quot;16\u0026quot; = \u0026quot;ND\u0026quot;,\n                               \u0026quot;17\u0026quot; = \u0026quot;B cells\u0026quot;, \n                               \u0026quot;18\u0026quot; = \u0026quot;ND\u0026quot;, \n                               \u0026quot;19\u0026quot; = \u0026quot;ND\u0026quot;,\n                               \u0026quot;20\u0026quot; = \u0026quot;ND\u0026quot;)\n\n# Plot the UMAP withy new labells\npng(filename = \u0026quot;blca_umap_with_label.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 600)\nDimPlot(object = seurat_integrated, \n        reduction = \u0026quot;umap\u0026quot;, \n        label = TRUE,\n        label.size = 3,\n        repel = TRUE,\n        split.by = \u0026quot;Invasiveness\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Rename all identities\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RenameIdents(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBasal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCCND1 + cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e NKG7 and GZMK\u003c/span\u003e\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT cells/NK cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebasal/squamous-like cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecytotoxic T cell\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCAFs\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efibroblasts/myofibroblasts\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e9\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efibroblasts/myofibroblasts\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e11\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFibroblast(?)\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e12\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e13\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMast cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eB cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e15\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e17\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eB cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e19\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e20\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eND\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the UMAP withy new labells\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblca_umap_with_label.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e600\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_integrated\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_umap_with_label.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_umap_with_label.png\" alt=\"blca_umap_with_label.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e9) Analyzing epithelial (EPCAM +) cells\u003c/h1\u003e\u003ca id=\"user-content-9-analyzing-epithelial-epcam--cells\" class=\"anchor\" aria-label=\"Permalink: 9) Analyzing epithelial (EPCAM +) cells\" href=\"#9-analyzing-epithelial-epcam--cells\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis section is based on what I can get from the paper, but I stay focused on cancer samples and will use harmony to do integration. Finally, I do trajectory analysis for all cells which identified as epithelial cells.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Cells with UMI numbersâ\u0026lt;1000 or with over 10% mitochondrial-derived UMI counts were considered low-quality cells and were removed. To eliminate potential doublets, single cells with over 6000 genes detected were also filtered out. Finally, 52721 single cells remained, and they were applied in downstream analyses.\"\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"Since samples from eight patients were processed and sequenced in batches, the patient number was used to remove the potential batch effect.\"\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\"epithelial (EPCAM+) cells; endothelial (CD31+) cells; two types of fibroblasts (COL1A1+)âiCAFs (PDGFRA+) and myo-CAFs (mCAFs) (RGS5+); B cells (CD79A+); myeloid cells (LYZ+); T cells (CD3D+); and mast cells (TPSAB1+)\"\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#________________________Reading the files______________________#\n# create a list of samples\nsamples \u0026lt;- list.files(\u0026quot;./filtered/\u0026quot;)\n#samples \u0026lt;- samples[grepl('^filtered',samples,perl=T)]\n\n# read files into Seurat objects\nfor (file in samples){\n  print(paste0(file))\n  seurat_data \u0026lt;- Read10X(data.dir = paste0(\u0026quot;./filtered/\u0026quot;, file))\n  seurat_obj \u0026lt;- CreateSeuratObject(counts = seurat_data, \n                                   min.features = 100, \n                                   project = file)\n  assign(file, seurat_obj)\n}\n\n# updated sample name \nsamples_blca \u0026lt;- samples[-c(1,2,9)]\n\n# Now merging all objects into one Seurat obj\n\nmerged_seurat \u0026lt;- merge(x = SRR12603782,\n                       y = c(SRR12603783,\n                             SRR12603784,\n                             SRR12603785,\n                             SRR12603786,\n                             SRR12603787,\n                             SRR12603789,\n                             SRR12603790),\n                       add.cell.id = samples_blca)\n                       \n#________________________Filteration____________________________#\n# reading sampleInformation:\nsampleInformation \u0026lt;- read.csv(\u0026quot;./sampleInfo.csv\u0026quot;)\n\n# Compute percent mito ratio\nmerged_seurat$mitoRatio \u0026lt;- PercentageFeatureSet(object = merged_seurat, pattern = \u0026quot;^MT-\u0026quot;)\nmerged_seurat$mitoRatio \u0026lt;- merged_seurat@meta.data$mitoRatio / 100\n\n# adding cell column\nmerged_seurat$cells \u0026lt;- rownames(merged_seurat@meta.data)\n# merging with sample information\nmerged_seurat@meta.data \u0026lt;- merge(merged_seurat@meta.data, sampleInformation)\n# re-setting the rownames\nrownames(merged_seurat@meta.data) \u0026lt;- merged_seurat@meta.data$cells\n\n\n# Filteration\n\nfiltered_seurat \u0026lt;- subset(merged_seurat, \n                          subset= nCount_RNA \u0026gt;= 1000 \u0026amp;\n                          nFeature_RNA \u0026lt;= 6000 \u0026amp; \n                          mitoRatio \u0026lt; 0.10)\n\n#________________________Integration using Harmony____________________________#\n#integration using harmony need several steps to be undertaken:\n\n# Perform log-normalization and feature selection, as well as SCT normalization on the global object\nmerged_seurat \u0026lt;- filtered_seurat %\u0026gt;%\n    NormalizeData() %\u0026gt;%\n    FindVariableFeatures(selection.method = \u0026quot;vst\u0026quot;, nfeatures = 3000) %\u0026gt;% \n    ScaleData() %\u0026gt;%\n    SCTransform(vars.to.regress = c(\u0026quot;mitoRatio\u0026quot;, \u0026quot;orig.ident\u0026quot;))\n\n# Calculate PCs using variable features determined by SCTransform (3000 by default)\nmerged_seurat \u0026lt;- RunPCA(merged_seurat, assay = \u0026quot;SCT\u0026quot;, npcs = 50)\nmerged_seurat \u0026lt;- RunTSNE(merged_seurat, assay = \u0026quot;SCT\u0026quot;, npcs = 50)\n\n# Integration\n#install.packages(\u0026quot;harmony\u0026quot;)\n\nlibrary(harmony)\n\nharmonized_seurat \u0026lt;- RunHarmony(merged_seurat, \n\t\t\t\tgroup.by.vars = c(\u0026quot;orig.ident\u0026quot;, \u0026quot;gender\u0026quot;, \u0026quot;Surgery_Type\u0026quot;), \n\t\t\t\treduction = \u0026quot;pca\u0026quot;, assay.use = \u0026quot;SCT\u0026quot;, reduction.save = \u0026quot;harmony\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________Reading the files______________________#\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e create a list of samples\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e list.files(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./filtered/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esamples \u0026lt;- samples[grepl('^filtered',samples,perl=T)]\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e read files into Seurat objects\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e){\n  print(paste0(\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e))\n  \u003cspan class=\"pl-smi\"\u003eseurat_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Read10X(\u003cspan class=\"pl-v\"\u003edata.dir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./filtered/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e))\n  \u003cspan class=\"pl-smi\"\u003eseurat_obj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e CreateSeuratObject(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_data\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003emin.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003eproject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e)\n  assign(\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eseurat_obj\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e updated sample name \u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esamples_blca\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e[\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e)]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Now merging all objects into one Seurat obj\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSRR12603782\u003c/span\u003e,\n                       \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-smi\"\u003eSRR12603783\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603784\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603785\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603786\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603787\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603789\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603790\u003c/span\u003e),\n                       \u003cspan class=\"pl-v\"\u003eadd.cell.id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples_blca\u003c/span\u003e)\n                       \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________Filteration____________________________#\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e reading sampleInformation:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.csv(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e./sampleInfo.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Compute percent mito ratio\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e PercentageFeatureSet(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epattern\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e^MT-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e adding cell column\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e merging with sample information\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e re-setting the rownames\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Filteration\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e subset(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \n                          \u003cspan class=\"pl-v\"\u003esubset\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enCount_RNA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e\n                          \u003cspan class=\"pl-smi\"\u003enFeature_RNA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6000\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \n                          \u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.10\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________Integration using Harmony____________________________#\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eintegration using harmony need several steps to be undertaken:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Perform log-normalization and feature selection, as well as SCT normalization on the global object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    NormalizeData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    FindVariableFeatures(\u003cspan class=\"pl-v\"\u003eselection.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3000\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    ScaleData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    SCTransform(\u003cspan class=\"pl-v\"\u003evars.to.regress\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoRatio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Calculate PCs using variable features determined by SCTransform (3000 by default)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunPCA(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enpcs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunTSNE(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enpcs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Integration\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"harmony\")\u003c/span\u003e\n\nlibrary(\u003cspan class=\"pl-smi\"\u003eharmony\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunHarmony(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \n\t\t\t\t\u003cspan class=\"pl-v\"\u003egroup.by.vars\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egender\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSurgery_Type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n\t\t\t\t\u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay.use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction.save\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe  above code incorporates an additional reduction of 50 \"harmony components\" (i.e., corrected principal components) to our Seurat object, which is stored in the harmonized_seurat@reductions$harmony variable.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHowever, to ensure that the Harmony integration is accurately represented in the data visualization, we must generate a UMAP that is derived from these harmony embeddings instead of the PCs.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"harmonized_seurat \u0026lt;- RunUMAP(harmonized_seurat, reduction = \u0026quot;harmony\u0026quot;, assay = \u0026quot;SCT\u0026quot;, dims = 1:40)\n#harmonized_seurat \u0026lt;- RunUTSNE(harmonized_seurat, reduction = \u0026quot;harmony\u0026quot;, assay = \u0026quot;SCT\u0026quot;, dims = 1:40)\n\n#________________________Cluster identification and Inspect the effects of Harmony batch removel ____________#\n\n# to set reduction to harmony and find the clusters\nharmonized_seurat \u0026lt;- FindNeighbors(object = harmonized_seurat, reduction = \u0026quot;harmony\u0026quot;)\nharmonized_seurat \u0026lt;- FindClusters(harmonized_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))\n\n# visualization\nIdents(harmonized_seurat) \u0026lt;- harmonized_seurat@meta.data$SCT_snn_res.0.1\n\n# Color cells based on the sample name\n# Plot UMAP \npng(filename = \u0026quot;harmony_UMAP_y_sample.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(harmonized_seurat,\n        group.by = \u0026quot;orig.ident\u0026quot;,\n        reduction = \u0026quot;umap\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunUMAP(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e40\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eharmonized_seurat \u0026lt;- RunUTSNE(harmonized_seurat, reduction = \"harmony\", assay = \"SCT\", dims = 1:40)\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________Cluster identification and Inspect the effects of Harmony batch removel ____________#\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to set reduction to harmony and find the clusters\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindNeighbors(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindClusters(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eresolution\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e visualization\u003c/span\u003e\nIdents(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Color cells based on the sample name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot UMAP \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony_UMAP_y_sample.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_UMAP_y_sample.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/harmony_UMAP_y_sample.png\" alt=\"harmony_UMAP_y_sample.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs the above figure suggests, Harmony did a great job in terms of removing the batch effects.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#________________________SuperCluster Identification____________#\n\npng(filename = \u0026quot;harmony_umap_cluster_with_label.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(harmonized_seurat,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________SuperCluster Identification____________#\u003c/span\u003e\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony_umap_cluster_with_label.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_umap_cluster_with_label.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/harmony_umap_cluster_with_label.png\" alt=\"harmony_umap_cluster_with_label.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# let's visualize cells expressing supercluster markers:\n# CD31: PECAM1\nmarkers \u0026lt;- c(\u0026quot;EPCAM\u0026quot;, \u0026quot;PECAM1\u0026quot;, \u0026quot;COL1A1\u0026quot;, \u0026quot;PDGFRA\u0026quot;, \u0026quot;RGS5\u0026quot;, \u0026quot;CD79A\u0026quot;, \u0026quot;LYZ\u0026quot;, \u0026quot;CD3D\u0026quot;, \u0026quot;TPSAB1\u0026quot;)\n\npng(filename = \u0026quot;umap_superCluster_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = harmonized_seurat,\n            features = markers,\n            order = TRUE,\n            min.cutoff = \u0026quot;q10\u0026quot;,\n            reduction = \u0026quot;umap\u0026quot;,\n            label = TRUE,\n            repel = TRUE)\n\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e let's visualize cells expressing supercluster markers:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e CD31: PECAM1\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEPCAM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePECAM1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCOL1A1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePDGFRA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRGS5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD79A\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLYZ\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD3D\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTPSAB1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap_superCluster_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_superCluster_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_superCluster_cells.png\" alt=\"umap_superCluster_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMarker identification for the 8 superclusters:\u003c/h4\u003e\u003ca id=\"user-content-marker-identification-for-the-8-superclusters\" class=\"anchor\" aria-label=\"Permalink: Marker identification for the 8 superclusters:\" href=\"#marker-identification-for-the-8-superclusters\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#______________________________ All markers________________________________\n# Find markers for every cluster compared to all remaining cells, and report only the positive ones\nmarkers \u0026lt;- FindAllMarkers(object = harmonized_seurat, \n                          only.pos = TRUE,\n                          logfc.threshold = 0.25) \n\nsaveRDS(markers, \u0026quot;harmony_markers.RDS\u0026quot;)\n\n# mutate the markers dataframe\n# Extract top 10 markers per cluster\n\ntop10 \u0026lt;- markers %\u0026gt;%\n  mutate(delta_pct = (pct.1 - pct.2)) %\u0026gt;%\n  #filter(avg_log2FC \u0026gt; 1.5) %\u0026gt;%  # only keep rows where avg_log2FC \u0026gt; 1.5\n  group_by(cluster) %\u0026gt;%\n  top_n(n = 10, wt = delta_pct)\n\ndata.table::fwrite(top10, \u0026quot;harmony_blca_top10_all_markers.csv\u0026quot;)\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e______________________________ All markers________________________________\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Find markers for every cluster compared to all remaining cells, and report only the positive ones\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindAllMarkers(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \n                          \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                          \u003cspan class=\"pl-v\"\u003elogfc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.25\u003c/span\u003e) \n\nsaveRDS(\u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony_markers.RDS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e mutate the markers dataframe\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract top 10 markers per cluster\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  mutate(\u003cspan class=\"pl-v\"\u003edelta_pct\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003epct.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epct.2\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efilter(avg_log2FC \u0026gt; 1.5) %\u0026gt;%  # only keep rows where avg_log2FC \u0026gt; 1.5\u003c/span\u003e\n  group_by(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  top_n(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edelta_pct\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efwrite(\u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony_blca_top10_all_markers.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eVisualization of top markers in each cluster:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCluster markers:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cluster_markers_10 \u0026lt;- top10 %\u0026gt;% \n                   group_by(cluster) %\u0026gt;% \n                   summarize(genes = paste(gene, collapse = \u0026quot;,\u0026quot;))\ndata.table::fwrite(cluster_markers_10, \u0026quot;cluster_markers_10.csv\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecluster_markers_10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etop10\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                   group_by(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                   summarize(\u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efwrite(\u003cspan class=\"pl-smi\"\u003ecluster_markers_10\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecluster_markers_10.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Feature plot for top markers\nplotList = list()\n\nfor(cluster in 1:nrow(cluster_markers_10)){\n    mkr = unlist(strsplit(cluster_markers_10$genes[cluster], \u0026quot;,\u0026quot;))\n    plotList[[cluster]] = FeaturePlot(object = harmonized_seurat,\n            features = mkr,\n            order = TRUE,\n            min.cutoff = \u0026quot;q10\u0026quot;,\n            reduction = \u0026quot;umap\u0026quot;,\n            label = TRUE,\n            repel = TRUE)\n}\n\n# Iterate over all clusters\npng(filename = \u0026quot;harmony_blca_clsuter_markers_cluster8.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nplotList[[9]]\ndev.off()\n#\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Feature plot for top markers\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003eplotList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003elist\u003c/span\u003e()\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003enrow(\u003cspan class=\"pl-smi\"\u003ecluster_markers_10\u003c/span\u003e)){\n    \u003cspan class=\"pl-v\"\u003emkr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e unlist(strsplit(\u003cspan class=\"pl-smi\"\u003ecluster_markers_10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e], \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n    \u003cspan class=\"pl-smi\"\u003eplotList\u003c/span\u003e[[\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e]] \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e FeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emkr\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n}\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Iterate over all clusters\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony_blca_clsuter_markers_cluster8.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eplotList\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e]]\ndev.off()\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eExpression of cluster-specific markers:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster0.png\"\u003ecluster0\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster1.png\"\u003ecluster1\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster2.png\"\u003ecluster2\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster3.png\"\u003ecluster3\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster4.png\"\u003ecluster4\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster5.png\"\u003ecluster5\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster6.png\"\u003ecluster6\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster7.png\"\u003ecluster7\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster8.png\"\u003ecluster8\u003c/a\u003e\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# LYZ cells\npng(filename = \u0026quot;LYZ_harmony_blca_clsuter_marker.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = harmonized_seurat,\n            features = c(\u0026quot;LYZ\u0026quot;),\n            order = TRUE,\n            min.cutoff = \u0026quot;q10\u0026quot;,\n            reduction = \u0026quot;umap\u0026quot;,\n            label = TRUE,\n            repel = TRUE)\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e LYZ cells\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLYZ_harmony_blca_clsuter_marker.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLYZ\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo see where there \u003ca href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/LYZ_harmony_blca_clsuter_marker.png\"\u003emyeloid cells\u003c/a\u003e are.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMarkers and cell types\u003c/h3\u003e\u003ca id=\"user-content-markers-and-cell-types\" class=\"anchor\" aria-label=\"Permalink: Markers and cell types\" href=\"#markers-and-cell-types\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003ecluster_id\u003c/th\u003e\n\u003cth\u003egenes\u003c/th\u003e\n\u003cth\u003ecell_type (PanglaoDB + ChatGPT)\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e0\u003c/td\u003e\n\u003ctd\u003eAGR2,C10orf99,SDC1,SERINC2,SMIM22,CLDN7,TNFRSF21,FAM160A1,CXADR,RAB25\u003c/td\u003e\n\u003ctd\u003eepithelial cell\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e1\u003c/td\u003e\n\u003ctd\u003eCD52,CD3D,PTPRC,TRAC,CD7,CD2,SKAP1,ARHGAP15,HCST,CD3E\u003c/td\u003e\n\u003ctd\u003eT-cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e2\u003c/td\u003e\n\u003ctd\u003ePLVAP,SPARCL1,HSPG2,VWF,TCF4,LDB2,CALCRL,RAMP2,PCAT19,PECAM1\u003c/td\u003e\n\u003ctd\u003eEndothelial cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e3\u003c/td\u003e\n\u003ctd\u003eHLA-DRA,HLA-DPB1,TYROBP,HLA-DQA1,HLA-DRB1,HLA-DPA1,HLA-DQB1,FCER1G\u003c/td\u003e\n\u003ctd\u003eAPCs (Macrophages, B-cells)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e4\u003c/td\u003e\n\u003ctd\u003eIGLC2,IGHG1,IGLC1,IGHG3,IGHA1,IGHG4,JCHAIN,IGHGP,MZB1,DERL3\u003c/td\u003e\n\u003ctd\u003eB-cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e5\u003c/td\u003e\n\u003ctd\u003eRGS5,TAGLN,ACTA2,MYL9,CALD1,PPP1R14A,BGN,PRKG1,SOD3,COL6A2\u003c/td\u003e\n\u003ctd\u003emyo-CAFs\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e6\u003c/td\u003e\n\u003ctd\u003eLUM,DCN,COL1A2,COL3A1,RARRES2,MFAP4,C1S,C1R,COL6A2,COL6A3\u003c/td\u003e\n\u003ctd\u003ei-CAFS\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e7\u003c/td\u003e\n\u003ctd\u003eTPSB2,TPSAB1,CPA3,HPGDS,LTC4S,MS4A2,SAMSN1,RGS13,FCER1G,TYROBP\u003c/td\u003e\n\u003ctd\u003eMast cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e8\u003c/td\u003e\n\u003ctd\u003eMIR205HG,COL4A5,IGFBP2,LINC00511,FOSL1,BCAM,RAP2B,SLC35F1,AQP3,DENND2C\u003c/td\u003e\n\u003ctd\u003eEpithelial cells + Mesenchymal cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# renaming clusters\n\n# Rename all identities\nharmonized_seurat \u0026lt;- RenameIdents(object = harmonized_seurat, \n                               \u0026quot;0\u0026quot; = \u0026quot;Epithelial cells\u0026quot;,\n                               \u0026quot;1\u0026quot; = \u0026quot;T-cells\u0026quot;, # impureity with epithelial cells\n                               \u0026quot;2\u0026quot; = \u0026quot;Endothelial cells\u0026quot;,\n                               \u0026quot;3\u0026quot; = \u0026quot;APCs(Macrophages, B-cells)\u0026quot;,\n                               \u0026quot;4\u0026quot; = \u0026quot;B-cells\u0026quot;,\n                               \u0026quot;5\u0026quot; = \u0026quot;myo-CAF\u0026quot;,\n                               \u0026quot;6\u0026quot; = \u0026quot;i-CAF\u0026quot;,\n                               \u0026quot;7\u0026quot; = \u0026quot;Mast cells\u0026quot;,\n                               \u0026quot;8\u0026quot; = \u0026quot;Epithelial cells+Mesenchymal cells\u0026quot;)\n                               \n                               \n                               \n\n\n# Plot the UMAP with new labels\npng(filename = \u0026quot;harmont_blca_umap_with_label.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 600)\nDimPlot(object = harmonized_seurat, \n        reduction = \u0026quot;umap\u0026quot;, \n        label = TRUE,\n        label.size = 3,\n        repel = TRUE,\n        split.by = \u0026quot;Invasiveness\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e renaming clusters\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Rename all identities\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RenameIdents(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEpithelial cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eT-cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e impureity with epithelial cells\u003c/span\u003e\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEndothelial cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAPCs(Macrophages, B-cells)\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eB-cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emyo-CAF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ei-CAF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e7\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMast cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                               \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEpithelial cells+Mesenchymal cells\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n                               \n                               \n                               \n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the UMAP with new labels\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmont_blca_umap_with_label.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e600\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003esplit.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eInvasiveness\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmont_blca_umap_with_label.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/harmont_blca_umap_with_label.png\" alt=\"harmont_blca_umap_with_label.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCluster 1, T-cells show some level of impurity by cluster 0, epithelial cells. This Impurity is especially evident in invasive samples.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Obtaining clusters with epithelial cells\n\n# To see the number of cells in each cluster\n\ntable(Idents(harmonized_seurat))\n\n#\n#                  epithelial cells                            T-cells\n#                             24135                              19850\n#                 Endothelial cells         APCs(Macrophages, B-cells)\n#                              6699                               5233\n#                           B-cells                            myo-CAF\n#                              4448                               2877\n#                             i-CAF                         Mast cells\n#                              2337                               1126\n#Epithelial cells+Mesenchymal cells\n#                               725\n#\nepi_cell_ids \u0026lt;- rownames(harmonized_seurat@meta.data)[harmonized_seurat@meta.data$SCT_snn_res.0.1 == '0']\n\nepi_cell_ids \u0026lt;- c(epi_cell_ids,rownames(harmonized_seurat@meta.data)[harmonized_seurat@meta.data$SCT_snn_res.0.1 == '8'])\n\n\nepi_seurat \u0026lt;- subset(filtered_seurat, subset = cells %in% epi_cell_ids)\n\n# Perform log-normalization and feature selection, as well as SCT normalization on the global object\nepi_seurat \u0026lt;- epi_seurat %\u0026gt;%\n    NormalizeData() %\u0026gt;%\n    FindVariableFeatures(selection.method = \u0026quot;vst\u0026quot;, nfeatures = 3000) %\u0026gt;% \n    ScaleData() %\u0026gt;%\n    SCTransform(vars.to.regress = c(\u0026quot;mitoRatio\u0026quot;, \u0026quot;orig.ident\u0026quot;))\n\n# Calculate PCs using variable features determined by SCTransform (3000 by default)\nepi_seurat \u0026lt;- RunPCA(epi_seurat, assay = \u0026quot;SCT\u0026quot;, npcs = 50)\n#epi_seurat \u0026lt;- RunTSNE(epi_seurat, assay = \u0026quot;SCT\u0026quot;, npcs = 50)\n\n# Integration\n#install.packages(\u0026quot;harmony\u0026quot;)\n\nlibrary(harmony)\n\nepi_seurat \u0026lt;- RunHarmony(epi_seurat, \n\t\t\t\tgroup.by.vars = c(\u0026quot;orig.ident\u0026quot;, \u0026quot;gender\u0026quot;, \u0026quot;Surgery_Type\u0026quot;), \n\t\t\t\treduction = \u0026quot;pca\u0026quot;, assay.use = \u0026quot;SCT\u0026quot;, reduction.save = \u0026quot;harmony\u0026quot;)\n\nepi_seurat \u0026lt;- RunUMAP(epi_seurat, reduction = \u0026quot;harmony\u0026quot;, assay = \u0026quot;SCT\u0026quot;, dims = 1:40)\n#harmonized_seurat \u0026lt;- RunUTSNE(harmonized_seurat, reduction = \u0026quot;harmony\u0026quot;, assay = \u0026quot;SCT\u0026quot;, dims = 1:40)\n\n#Cluster identification and Inspect the effects of Harmony batch removel#\n\n# to set reduction to harmony and find the clusters\nepi_seurat \u0026lt;- FindNeighbors(object = epi_seurat, reduction = \u0026quot;harmony\u0026quot;)\nepi_seurat \u0026lt;- FindClusters(epi_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))\n\n# visualization\nIdents(epi_seurat) \u0026lt;- epi_seurat@meta.data$SCT_snn_res.0.1\n\n# Color cells based on the sample name\n# Plot UMAP \npng(filename = \u0026quot;epi_harmony_UMAP_y_sample.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(epi_seurat,\n        group.by = \u0026quot;orig.ident\u0026quot;,\n        reduction = \u0026quot;umap\u0026quot;)\ndev.off()\n#\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Obtaining clusters with epithelial cells\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e To see the number of cells in each cluster\u003c/span\u003e\n\ntable(Idents(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                  epithelial cells                            T-cells\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                             24135                              19850\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                 Endothelial cells         APCs(Macrophages, B-cells)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                              6699                               5233\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                           B-cells                            myo-CAF\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                              4448                               2877\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                             i-CAF                         Mast cells\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                              2337                               1126\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eEpithelial cells+Mesenchymal cells\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                               725\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_cell_ids\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n\n\u003cspan class=\"pl-smi\"\u003eepi_cell_ids\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-smi\"\u003eepi_cell_ids\u003c/span\u003e,rownames(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e])\n\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e subset(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esubset\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_cell_ids\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Perform log-normalization and feature selection, as well as SCT normalization on the global object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    NormalizeData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    FindVariableFeatures(\u003cspan class=\"pl-v\"\u003eselection.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3000\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    ScaleData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    SCTransform(\u003cspan class=\"pl-v\"\u003evars.to.regress\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoRatio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Calculate PCs using variable features determined by SCTransform (3000 by default)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunPCA(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enpcs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eepi_seurat \u0026lt;- RunTSNE(epi_seurat, assay = \"SCT\", npcs = 50)\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Integration\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"harmony\")\u003c/span\u003e\n\nlibrary(\u003cspan class=\"pl-smi\"\u003eharmony\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunHarmony(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \n\t\t\t\t\u003cspan class=\"pl-v\"\u003egroup.by.vars\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egender\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSurgery_Type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n\t\t\t\t\u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay.use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction.save\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunUMAP(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e40\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eharmonized_seurat \u0026lt;- RunUTSNE(harmonized_seurat, reduction = \"harmony\", assay = \"SCT\", dims = 1:40)\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCluster identification and Inspect the effects of Harmony batch removel#\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to set reduction to harmony and find the clusters\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindNeighbors(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindClusters(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eresolution\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e visualization\u003c/span\u003e\nIdents(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Color cells based on the sample name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot UMAP \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eepi_harmony_UMAP_y_sample.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/epi_harmony_UMAP_y_sample.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/epi_harmony_UMAP_y_sample.png\" alt=\"epi_harmony_UMAP_y_sample.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# color cells based on the cluster\n# Plot UMAP \npng(filename = \u0026quot;cluster_epi_harmony_UMAP.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(epi_seurat,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e color cells based on the cluster\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot UMAP \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecluster_epi_harmony_UMAP.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cluster_epi_harmony_UMAP.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cluster_epi_harmony_UMAP.png\" alt=\"cluster_epi_harmony_UMAP.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Marker identification\n\nepi_markers \u0026lt;- FindAllMarkers(object = epi_seurat, \n                          only.pos = TRUE,\n                          logfc.threshold = 0.25) \n\n# mutate the markers dataframe\n# Extract top 10 markers per cluster\n\nepi_top10 \u0026lt;- epi_markers %\u0026gt;%\n  mutate(delta_pct = (pct.1 - pct.2)) %\u0026gt;%\n  #filter(avg_log2FC \u0026gt; 1.5) %\u0026gt;%  # only keep rows where avg_log2FC \u0026gt; 1.5\n  group_by(cluster) %\u0026gt;%\n  top_n(n = 10, wt = delta_pct)\n\n\nepi_cluster_markers_10 \u0026lt;- epi_top10 %\u0026gt;% \n                   group_by(cluster) %\u0026gt;% \n                   summarize(genes = paste(gene, collapse = \u0026quot;,\u0026quot;))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Marker identification\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eepi_markers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindAllMarkers(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \n                          \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                          \u003cspan class=\"pl-v\"\u003elogfc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.25\u003c/span\u003e) \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e mutate the markers dataframe\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract top 10 markers per cluster\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eepi_top10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_markers\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  mutate(\u003cspan class=\"pl-v\"\u003edelta_pct\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003epct.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epct.2\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efilter(avg_log2FC \u0026gt; 1.5) %\u0026gt;%  # only keep rows where avg_log2FC \u0026gt; 1.5\u003c/span\u003e\n  group_by(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  top_n(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edelta_pct\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003eepi_cluster_markers_10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_top10\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                   group_by(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                   summarize(\u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003ecluster\u003c/th\u003e\n\u003cth\u003egenes\u003c/th\u003e\n\u003cth\u003ecell_type (PanglaoDB + ChatGPT)\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e0\u003c/td\u003e\n\u003ctd\u003eKRT15,KRT5,BCAM,IGFBP2,IGFBP7,CDH13,FOSL1,LAMB3,COL4A5,SERPINB5,SELENOP\u003c/td\u003e\n\u003ctd\u003ebasal cell\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e1\u003c/td\u003e\n\u003ctd\u003eUCA1,LGALS1,CA9,SPINK1,GJB2,PTPRR,FCRLB,PLA2G2F,GJB6,FBLN1\u003c/td\u003e\n\u003ctd\u003ecancer-associated luminal cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e2\u003c/td\u003e\n\u003ctd\u003eNDUFA4L2,CRH,UPK2,SNX31,KRT20,AC019117.2,UPK1B,LINC02163,H19,TESC\u003c/td\u003e\n\u003ctd\u003eluminal differentiated\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e3\u003c/td\u003e\n\u003ctd\u003eLY6D,TNNT3,IGF2BP2,GMNN,SYT8,AC068587.4,ITM2C,SGPP2,ZNF750,TNFSF10\u003c/td\u003e\n\u003ctd\u003eunique luminal cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e4\u003c/td\u003e\n\u003ctd\u003eS100A9,C15orf48,MYO16,MMP7,MYEOV,AC025159.1,LTO1,IFI27,IRS2,SLPI\u003c/td\u003e\n\u003ctd\u003eimmunomodulatory luminal\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e5\u003c/td\u003e\n\u003ctd\u003eALCAM,ATXN1,PCDH7,COBLL1,LIPH,TRIM31,NAALADL2,ITGA2,SERPINB5,ITGA6,TM4SF1\u003c/td\u003e\n\u003ctd\u003eadhesion and signaling luminal cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"# Reading seurat object\nepi_seurat \u0026lt;- readRDS(\u0026quot;epi_seurat.RDS\u0026quot;)\n\n# Rename all identities\nepi_seurat \u0026lt;- RenameIdents(object = epi_seurat, \n                               \u0026quot;0\u0026quot; = \u0026quot;basal_cell\u0026quot;,\n                               \u0026quot;1\u0026quot; = \u0026quot;cancer_associated_luminal_cell\u0026quot;,\n                               \u0026quot;2\u0026quot; = \u0026quot;differentiated_luminal_cell\u0026quot;,\n                               \u0026quot;3\u0026quot; = \u0026quot;unique_luminal_cell\u0026quot;,\n                               \u0026quot;4\u0026quot; = \u0026quot;immunomodulatory_luminal_cell\u0026quot;,\n                               \u0026quot;5\u0026quot; = \u0026quot;adhesion_signaling_luminal_cell\u0026quot;)\n\n# Plot the UMAP with new labels\npng(filename = \u0026quot;labelled_epi.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 600)\nDimPlot(object = epi_seurat, \n        reduction = \u0026quot;umap\u0026quot;, \n        label = TRUE,\n        label.size = 3,\n        repel = TRUE)\ndev.off()\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e# Reading seurat object\nepi_seurat \u0026lt;- readRDS(\"epi_seurat.RDS\")\n\n# Rename all identities\nepi_seurat \u0026lt;- RenameIdents(object = epi_seurat, \n                               \"0\" = \"basal_cell\",\n                               \"1\" = \"cancer_associated_luminal_cell\",\n                               \"2\" = \"differentiated_luminal_cell\",\n                               \"3\" = \"unique_luminal_cell\",\n                               \"4\" = \"immunomodulatory_luminal_cell\",\n                               \"5\" = \"adhesion_signaling_luminal_cell\")\n\n# Plot the UMAP with new labels\npng(filename = \"labelled_epi.png\", width = 16, height = 8.135, units = \"in\", res = 600)\nDimPlot(object = epi_seurat, \n        reduction = \"umap\", \n        label = TRUE,\n        label.size = 3,\n        repel = TRUE)\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eCluster 5\u003c/strong\u003e can benefit from re-clustering. It consists of several patches of cells. If we increase the clustering resolution we should get each patches as a cluster.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Idents(epi_seurat) \u0026lt;- epi_seurat$SCT_snn_res.0.2\npng(filename = \u0026quot;cluster_epi_harmony_UMAP_ptsize_1_res0.2.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(epi_seurat,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003eIdents(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecluster_epi_harmony_UMAP_ptsize_1_res0.2.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/hamidghaedi/scRNA_seq-analysis/blob/main\"\u003e\u003cimg src=\"/hamidghaedi/scRNA_seq-analysis/raw/main\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\nSo based on the above graph, cells in cluster 5 are now clustered in 8,9,11, and 12. The current cluster 8 can be split into more clusters.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Idents(epi_seurat) \u0026lt;- epi_seurat$SCT_snn_res.0.6\npng(filename = \u0026quot;cluster_epi_harmony_UMAP_ptsize_1_res0.6.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(epi_seurat,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003eIdents(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.6\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecluster_epi_harmony_UMAP_ptsize_1_res0.6.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUsing the above resolution we can get two different clusters of cells for cluster 8. Now let's generate a customized \u003ccode\u003eIdent\u003c/code\u003e for the data set:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"idt \u0026lt;- epi_seurat@meta.data[, c(14,15,17)]\n# convert to character\nidt \u0026lt;- data.frame(apply(idt, 2, as.character))\n# to see what to do:\n# table(epi_seurat$SCT_snn_res.0.2, epi_seurat$SCT_snn_res.0.6)\n\n# Replacing values for cluster 8 with clusters 18 and 19 in res 0.6\nidt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == \u0026quot;15\u0026quot;] \u0026lt;- \u0026quot;8_a\u0026quot;\nidt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == \u0026quot;1\u0026quot;] \u0026lt;- \u0026quot;8_a\u0026quot;\nidt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == \u0026quot;2\u0026quot;] \u0026lt;- \u0026quot;8_a\u0026quot;\nidt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == \u0026quot;16\u0026quot;] \u0026lt;- \u0026quot;8_b\u0026quot;\nidt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == \u0026quot;18\u0026quot;] \u0026lt;- \u0026quot;8_b\u0026quot;\nidt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == \u0026quot;19\u0026quot;] \u0026lt;- \u0026quot;8_b\u0026quot;\n\n# Updating SCT_snn_res.0.1 \n# to see what to do:\n# table(idt$SCT_snn_res.0.1, idt$SCT_snn_res.0.2)\n\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;9\u0026quot;] \u0026lt;- \u0026quot;5_a\u0026quot;\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;8_a\u0026quot;] \u0026lt;- \u0026quot;5_b\u0026quot;\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;3\u0026quot;] \u0026lt;- \u0026quot;5_b\u0026quot;\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;4\u0026quot;] \u0026lt;- \u0026quot;5_b\u0026quot;\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;6\u0026quot;] \u0026lt;- \u0026quot;5_a\u0026quot;\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;8_b\u0026quot;] \u0026lt;- \u0026quot;5_c\u0026quot;\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;11\u0026quot;] \u0026lt;- \u0026quot;5_d\u0026quot;\nidt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == \u0026quot;12\u0026quot;] \u0026lt;- \u0026quot;5_e\u0026quot;\n# adding new idents to Seurat obj\nepi_seurat$new_idents \u0026lt;- idt$SCT_snn_res.0.1\n# setting ident for plotting\nIdents(epi_seurat) \u0026lt;- epi_seurat$new_idents\n# plotting\npng(filename = \u0026quot;cluster_epi_harmony_UMAP_new_idents.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(epi_seurat,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e[, c(\u003cspan class=\"pl-c1\"\u003e14\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e17\u003c/span\u003e)]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e convert to character\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(apply(\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eas.character\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to see what to do:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e table(epi_seurat$SCT_snn_res.0.2, epi_seurat$SCT_snn_res.0.6)\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Replacing values for cluster 8 with clusters 18 and 19 in res 0.6\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e15\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_a\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_a\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_a\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_b\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_b\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.6\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e19\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_b\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Updating SCT_snn_res.0.1 \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to see what to do:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e table(idt$SCT_snn_res.0.1, idt$SCT_snn_res.0.2)\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e9\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_a\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_a\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_b\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_b\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_b\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_a\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e8_b\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_c\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e11\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_d\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e12\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5_e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e adding new idents to Seurat obj\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enew_idents\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eidt\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e setting ident for plotting\u003c/span\u003e\nIdents(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enew_idents\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e plotting\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecluster_epi_harmony_UMAP_new_idents.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cluster_epi_harmony_UMAP_new_idents.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cluster_epi_harmony_UMAP_new_idents.png\" alt=\"cluster_epi_harmony_UMAP_new_idents.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e10) DE and GSEA\u003c/h2\u003e\u003ca id=\"user-content-10-de-and-gsea\" class=\"anchor\" aria-label=\"Permalink: 10) DE and GSEA\" href=\"#10-de-and-gsea\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(SCP)\n\n\n# DE and Enrichement analysis\n\n#Differential expression analysis; this takes a while, to upload the object containing DE result:\nepi_seurat \u0026lt;- readRDS(\u0026quot;updated_epi_seurat.RDS\u0026quot;)\n\nepi_seurat \u0026lt;- RunDEtest(srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, fc.threshold = 1.5, only.pos = FALSE)\n\n#\npng(filename = \u0026quot;vplcano_DE.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nVolcanoPlot(srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eSCP\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e DE and Enrichement analysis\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eDifferential expression analysis; this takes a while, to upload the object containing DE result:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e readRDS(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eupdated_epi_seurat.RDS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunDEtest(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evplcano_DE.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nVolcanoPlot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/vplcano_DE.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/vplcano_DE.png\" alt=\"vplcano_DE.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#\nDEGs \u0026lt;- epi_seurat@tools$DEtest_clusters$AllMarkers_wilcox\nDEGs \u0026lt;- DEGs[with(DEGs, avg_log2FC \u0026gt; 1.5 \u0026amp; p_val_adj \u0026lt; 0.05), ]\n# Annotate features with transcription factors and surface proteins\n#epi_seurat \u0026lt;- AnnotateFeatures(epi_seurat, species = \u0026quot;Homo_sapiens\u0026quot;, db = c(\u0026quot;TF\u0026quot;, \u0026quot;SP\u0026quot;))\n\n# png(filename = \u0026quot;heatmap_DE.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\n# ht \u0026lt;- FeatureHeatmap(\n#   srt = epi_seurat, group.by = \u0026quot;clusters\u0026quot;, features = DEGs$gene, feature_split = DEGs$group1,\n#   species = \u0026quot;Homo_sapiens\u0026quot;, db = c(\u0026quot;GO_BP\u0026quot;, \u0026quot;KEGG\u0026quot;, \u0026quot;WikiPathway\u0026quot;), anno_terms = TRUE)\n# print(ht$plot)\n# dev.off()\n\n\n# Enrichment analysis\n## over-representation analysis\n\nepi_seurat \u0026lt;- RunEnrichment(\n  srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, db = \u0026quot;GO_BP\u0026quot;, species = \u0026quot;Homo_sapiens\u0026quot;,\n  DE_threshold = \u0026quot;avg_log2FC \u0026gt; 1 \u0026amp; p_val_adj \u0026lt; 0.05\u0026quot;\n)\n\npng(filename = \u0026quot;enrichment_1.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nEnrichmentPlot(\n  srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, group_use = c(\u0026quot;basal_cell\u0026quot;, \u0026quot;differentiated_luminal_cell\u0026quot;),\n  plot_type = \u0026quot;bar\u0026quot;\n)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eDEGs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etools\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eDEtest_clusters\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eAllMarkers_wilcox\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eDEGs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eDEGs\u003c/span\u003e[with(\u003cspan class=\"pl-smi\"\u003eDEGs\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eavg_log2FC\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1.5\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ep_val_adj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.05\u003c/span\u003e), ]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Annotate features with transcription factors and surface proteins\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eepi_seurat \u0026lt;- AnnotateFeatures(epi_seurat, species = \"Homo_sapiens\", db = c(\"TF\", \"SP\"))\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e png(filename = \"heatmap_DE.png\", width = 16, height = 8.135, units = \"in\", res = 300)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ht \u0026lt;- FeatureHeatmap(\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e   srt = epi_seurat, group.by = \"clusters\", features = DEGs$gene, feature_split = DEGs$group1,\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e   species = \"Homo_sapiens\", db = c(\"GO_BP\", \"KEGG\", \"WikiPathway\"), anno_terms = TRUE)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e print(ht$plot)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e dev.off()\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Enrichment analysis\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# over-representation analysis\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunEnrichment(\n  \u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGO_BP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003especies\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHomo_sapiens\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003eDE_threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eavg_log2FC \u0026gt; 1 \u0026amp; p_val_adj \u0026lt; 0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n)\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eenrichment_1.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nEnrichmentPlot(\n  \u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebasal_cell\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edifferentiated_luminal_cell\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n  \u003cspan class=\"pl-v\"\u003eplot_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebar\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/enrichment_1.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/enrichment_1.png\" alt=\"enrichment_1.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;enrichment_2.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nEnrichmentPlot(srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, plot_type = \u0026quot;comparison\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eenrichment_2.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nEnrichmentPlot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomparison\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/enrichment_2.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/enrichment_2.png\" alt=\"enrichment_2.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## Enrichment analysis(GSEA)\nepi_seurat \u0026lt;- RunGSEA(\n  srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, db = \u0026quot;GO_BP\u0026quot;, species = \u0026quot;Homo_sapiens\u0026quot;,\n  DE_threshold = \u0026quot;p_val_adj \u0026lt; 0.05\u0026quot;\n)\n\n# plotting for hypoxia\npng(filename = \u0026quot;enrichment_4.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nGSEAPlot(srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, group_use = \u0026quot;differentiated_luminal_cell\u0026quot;, geneSetID = \u0026quot;GO:0001666\u0026quot;)\ndev.off()\n\npng(filename = \u0026quot;enrichment_5.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nGSEAPlot(srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, group_use = \u0026quot;cancer_associated_luminal_cell\u0026quot;, geneSetID = \u0026quot;GO:0001666\u0026quot;)\ndev.off()\n\npng(filename = \u0026quot;enrichment_6.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nGSEAPlot(srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, group_use = \u0026quot;basal_cell\u0026quot;, geneSetID = \u0026quot;GO:0001666\u0026quot;)\ndev.off()\n\n# comparison plot\npng(filename = \u0026quot;enrichment_3.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nGSEAPlot(srt = epi_seurat, group_by = \u0026quot;clusters\u0026quot;, plot_type = \u0026quot;comparison\u0026quot;)\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Enrichment analysis(GSEA)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunGSEA(\n  \u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGO_BP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003especies\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHomo_sapiens\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n  \u003cspan class=\"pl-v\"\u003eDE_threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ep_val_adj \u0026lt; 0.05\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e plotting for hypoxia\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eenrichment_4.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nGSEAPlot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edifferentiated_luminal_cell\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egeneSetID\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGO:0001666\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eenrichment_5.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nGSEAPlot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecancer_associated_luminal_cell\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egeneSetID\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGO:0001666\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eenrichment_6.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nGSEAPlot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebasal_cell\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egeneSetID\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGO:0001666\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e comparison plot\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eenrichment_3.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nGSEAPlot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup_by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecomparison\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/enrichment_3.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/enrichment_3.png\" alt=\"enrichment_3.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e11) Trajectory inference\u003c/h2\u003e\u003ca id=\"user-content-11-trajectory-inference\" class=\"anchor\" aria-label=\"Permalink: 11) Trajectory inference\" href=\"#11-trajectory-inference\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSlingshot is a computational method designed for cell lineage inference and pseudotime estimation in single-cell transcriptomics data. It operates by leveraging gene expression dynamics to reconstruct the developmental trajectories of individual cells and assign them pseudotime values. The method begins by clustering cells based on their gene expression profiles, allowing for the identification of distinct cell populations. It then performs differential expression analysis to identify genes that are dynamically expressed across these populations, capturing the transitions between different cell types or states. Slingshot uses this information to infer lineage relationships between the cell populations, revealing the hierarchical structure of the developmental trajectory. By assigning pseudotime values, Slingshot estimates the relative progression of cells through different stages of development or biological processes. The inferred trajectories and pseudotime values provide insights into the temporal ordering of cellular events and gene expression dynamics. Slingshot's visualization tools facilitate the exploration and interpretation of these results, enabling researchers to gain a deeper understanding of the underlying processes driving cellular differentiation and disease progression in single-cell transcriptomics data.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"png(filename = \u0026quot;trajectory.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nepi_seurat \u0026lt;- RunSlingshot(srt = epi_seurat, group.by = \u0026quot;clusters\u0026quot;, reduction = \u0026quot;UMAP\u0026quot;)\ndev.off()\n\"\u003e\u003cpre\u003epng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etrajectory.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunSlingshot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUMAP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/trajectory.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/trajectory.png\" alt=\"trajectory.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentify temporally dynamic genes\u003c/h3\u003e\u003ca id=\"user-content-identify-temporally-dynamic-genes\" class=\"anchor\" aria-label=\"Permalink: Identify temporally dynamic genes\" href=\"#identify-temporally-dynamic-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAfter running slingshot, I am going to explore gene expression changes during development. I aim to discover genes that undergo significant expression shifts. The method involves fitting a General Additive Model (GAM) for each gene, employing a negative binomial noise distribution to capture potential nonlinear connections between gene expression and pseudotime. Then conduct association tests to unveil meaningful links between expression and pseudotime.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(slingshot)\nlibrary(tradeSeq)\nlibrary(RColorBrewer)\n\n# convert Seurat to sce\nsce \u0026lt;- as.SingleCellExperiment(epi_seurat)\n# Running slingshot\nsce \u0026lt;- slingshot(sce, clusterLabels = 'clusters', reducedDim = 'UMAP')\n\n## The following analysis is computationally intensive; so only a subset of genes will be passed to the function\n# Gene filtration: filter out genes with lower than 5 reads expressed in \u0026lt; 500 cells\ngeneFilter \u0026lt;- apply(assays(sce)$counts,1,function(x){\n  sum(x \u0026gt;= 5) \u0026gt;= 500\n})\ntdg \u0026lt;- rownames(sce[geneFilter, ])\n\n# Variable feature\nvf \u0026lt;- VariableFeatures(epi_seurat)\n# Combining the genes\nselected_feature \u0026lt;- tdg[tdg %in% vf]\n## Using the tradeSeq package\n# Number of knot determined by ....\nicMat \u0026lt;- evaluateK(counts = sce, sds = crv, k = 3:7, nGenes = 50,\n                   verbose = TRUE, plot = TRUE)\n# 5 seems to be the right number for knot\n\n# fit negative binomial GAM\ntradeseq_res \u0026lt;- fitGAM(counts = cse, nknots = 5, genes = selected_feature, verbose = TRUE)\n# test for dynamic expression\nATres \u0026lt;- associationTest(tradeseq_res)\n\n# Visualizing heat map\npst.ord \u0026lt;- order(sce$slingPseudotime_1, na.last = NA)\nheatdata \u0026lt;- as.matrix(assays(sce)$counts[topgenes, pst.ord])\nheatclus \u0026lt;- sce$clusters[pst.ord]\n\npng(\u0026quot;tdg.png\u0026quot;, width=10, height = 10, unit = \u0026quot;in\u0026quot;, res = 300)\nheatmap(log1p(heatdata), Colv = NA,\n        ColSideColors = brewer.pal(9,\u0026quot;Set1\u0026quot;)[heatclus])\ndev.off()\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eslingshot\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003etradeSeq\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eRColorBrewer\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e convert Seurat to sce\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.SingleCellExperiment(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Running slingshot\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e slingshot(\u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eclusterLabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereducedDim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eUMAP\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# The following analysis is computationally intensive; so only a subset of genes will be passed to the function\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Gene filtration: filter out genes with lower than 5 reads expressed in \u0026lt; 500 cells\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003egeneFilter\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e apply(assays(\u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){\n  sum(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e500\u003c/span\u003e\n})\n\u003cspan class=\"pl-smi\"\u003etdg\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003egeneFilter\u003c/span\u003e, ])\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Variable feature\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003evf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e VariableFeatures(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Combining the genes\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eselected_feature\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etdg\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003etdg\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003evf\u003c/span\u003e]\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Using the tradeSeq package\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Number of knot determined by ....\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eicMat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e evaluateK(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esds\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecrv\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e7\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e,\n                   \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e 5 seems to be the right number for knot\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e fit negative binomial GAM\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etradeseq_res\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fitGAM(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecse\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enknots\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eselected_feature\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e test for dynamic expression\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eATres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e associationTest(\u003cspan class=\"pl-smi\"\u003etradeseq_res\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Visualizing heat map\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epst.ord\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e order(\u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eslingPseudotime_1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ena.last\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eheatdata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.matrix(assays(\u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003etopgenes\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003epst.ord\u003c/span\u003e])\n\u003cspan class=\"pl-smi\"\u003eheatclus\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esce\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eclusters\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003epst.ord\u003c/span\u003e]\n\npng(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etdg.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nheatmap(log1p(\u003cspan class=\"pl-smi\"\u003eheatdata\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003eColv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eNA\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003eColSideColors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e brewer.pal(\u003cspan class=\"pl-c1\"\u003e9\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSet1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003eheatclus\u003c/span\u003e])\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/tdg.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/tdg.png\" alt=\"tdg.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExploring normal bladder mouse samples\u003c/h2\u003e\u003ca id=\"user-content-exploring-normal-bladder-mouse-samples\" class=\"anchor\" aria-label=\"Permalink: Exploring normal bladder mouse samples\" href=\"#exploring-normal-bladder-mouse-samples\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#________________________Reading the files______________________#\n\nsamples \u0026lt;- c(\u0026quot;SRR12603780\u0026quot;, \u0026quot;SRR12603781\u0026quot;, \u0026quot;SRR12603788\u0026quot;)\n\n# read files inot Seurat objects\nfor (file in samples){\n  print(paste0(file))\n  seurat_data \u0026lt;- Read10X(data.dir = paste0(\u0026quot;../filtered/\u0026quot;, file))\n  seurat_obj \u0026lt;- CreateSeuratObject(counts = seurat_data, \n                                   min.features = 100, \n                                   project = file)\n  assign(file, seurat_obj)\n}\n\n\n# now merging all objects inot one Seurat obj\n\nmerged_seurat \u0026lt;- merge(x = SRR12603780,\n                       y = c(SRR12603781,\n                             SRR12603788),\n                       add.cell.id = samples)\n                       \n#________________________Filteration____________________________#\n# reading sampleInformation:\nsampleInformation \u0026lt;- read.csv(\u0026quot;../sampleInfo.csv\u0026quot;)\n\n# Compute percent mito ratio\nmerged_seurat$mitoRatio \u0026lt;- PercentageFeatureSet(object = merged_seurat, pattern = \u0026quot;^MT-\u0026quot;)\nmerged_seurat$mitoRatio \u0026lt;- merged_seurat@meta.data$mitoRatio / 100\n\n# adding cell column\nmerged_seurat$cells \u0026lt;- rownames(merged_seurat@meta.data)\n# merging with sample information\nmerged_seurat@meta.data \u0026lt;- merge(merged_seurat@meta.data, sampleInformation)\n# re-setting the rownames\nrownames(merged_seurat@meta.data) \u0026lt;- merged_seurat@meta.data$cells\n\n\n# Filteration\n\nfiltered_seurat \u0026lt;- subset(merged_seurat, \n                          subset= nCount_RNA \u0026gt;= 1000 \u0026amp;\n                          nFeature_RNA \u0026lt;= 6000 \u0026amp; \n                          mitoRatio \u0026lt; 0.10)\n\n#________________________Integration using Harmony____________________________#\n#integration using harmony need sevral steps to be undertaken:\n\n# Perform log-normalization and feature selection, as well as SCT normalization on global object\nmerged_seurat \u0026lt;- filtered_seurat %\u0026gt;%\n    NormalizeData() %\u0026gt;%\n    FindVariableFeatures(selection.method = \u0026quot;vst\u0026quot;, nfeatures = 3000) %\u0026gt;% \n    ScaleData() %\u0026gt;%\n    SCTransform(vars.to.regress = c(\u0026quot;mitoRatio\u0026quot;, \u0026quot;orig.ident\u0026quot;))\n\n# Calculate PCs using variable features determined by SCTransform (3000 by default)\nmerged_seurat \u0026lt;- RunPCA(merged_seurat, assay = \u0026quot;SCT\u0026quot;, npcs = 50)\n\n# Integration\n#install.packages(\u0026quot;harmony\u0026quot;)\n\nlibrary(harmony)\n\nharmonized_seurat \u0026lt;- RunHarmony(merged_seurat, \n\t\t\t\tgroup.by.vars = c(\u0026quot;orig.ident\u0026quot;, \u0026quot;Surgery_Type\u0026quot;), \n\t\t\t\treduction = \u0026quot;pca\u0026quot;, assay.use = \u0026quot;SCT\u0026quot;, reduction.save = \u0026quot;harmony\u0026quot;)\n\nharmonized_seurat \u0026lt;- RunUMAP(harmonized_seurat, reduction = \u0026quot;harmony\u0026quot;, assay = \u0026quot;SCT\u0026quot;, dims = 1:40)\n\n#Cluster identification and Inspect the effects of Harmony batch removel ____________#\n\n# to set reduction to harmony and find the clusters\nharmonized_seurat \u0026lt;- FindNeighbors(object = harmonized_seurat, reduction = \u0026quot;harmony\u0026quot;)\nharmonized_seurat \u0026lt;- FindClusters(harmonized_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))\n\n# visualization\nIdents(harmonized_seurat) \u0026lt;- harmonized_seurat@meta.data$SCT_snn_res.0.1\n\n# Color cells based on the sample name\n# Plot UMAP \npng(filename = \u0026quot;normal_harmony_UMAP_y_sample.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(harmonized_seurat,\n        group.by = \u0026quot;orig.ident\u0026quot;,\n        reduction = \u0026quot;umap\u0026quot;)\ndev.off()\n\n#________________________SuperCluster Identification____________#\n\npng(filename = \u0026quot;normal_harmony_umap_cluster_with_label.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(harmonized_seurat,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\n\n# let's visualize cells expressing supercluster markers:\n# CD31: PECAM1\nmarkers \u0026lt;- c(\u0026quot;EPCAM\u0026quot;, \u0026quot;PECAM1\u0026quot;, \u0026quot;COL1A1\u0026quot;, \u0026quot;PDGFRA\u0026quot;, \u0026quot;RGS5\u0026quot;, \u0026quot;CD79A\u0026quot;, \u0026quot;LYZ\u0026quot;, \u0026quot;CD3D\u0026quot;, \u0026quot;TPSAB1\u0026quot;)\n\npng(filename = \u0026quot;normal_umap_superCluster_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = harmonized_seurat,\n            features = markers,\n            order = TRUE,\n            min.cutoff = \u0026quot;q10\u0026quot;,\n            reduction = \u0026quot;umap\u0026quot;,\n            label = TRUE,\n            repel = TRUE)\n\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________Reading the files______________________#\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603780\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603781\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSRR12603788\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e read files inot Seurat objects\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e){\n  print(paste0(\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e))\n  \u003cspan class=\"pl-smi\"\u003eseurat_data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Read10X(\u003cspan class=\"pl-v\"\u003edata.dir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../filtered/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e))\n  \u003cspan class=\"pl-smi\"\u003eseurat_obj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e CreateSeuratObject(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eseurat_data\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003emin.features\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \n                                   \u003cspan class=\"pl-v\"\u003eproject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e)\n  assign(\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eseurat_obj\u003c/span\u003e)\n}\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now merging all objects inot one Seurat obj\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSRR12603780\u003c/span\u003e,\n                       \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-smi\"\u003eSRR12603781\u003c/span\u003e,\n                             \u003cspan class=\"pl-smi\"\u003eSRR12603788\u003c/span\u003e),\n                       \u003cspan class=\"pl-v\"\u003eadd.cell.id\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esamples\u003c/span\u003e)\n                       \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________Filteration____________________________#\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e reading sampleInformation:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.csv(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e../sampleInfo.csv\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Compute percent mito ratio\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e PercentageFeatureSet(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epattern\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e^MT-\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e adding cell column\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e merging with sample information\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003esampleInformation\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e re-setting the rownames\u003c/span\u003e\nrownames(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Filteration\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e subset(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \n                          \u003cspan class=\"pl-v\"\u003esubset\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enCount_RNA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e\n                          \u003cspan class=\"pl-smi\"\u003enFeature_RNA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6000\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \n                          \u003cspan class=\"pl-smi\"\u003emitoRatio\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.10\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________Integration using Harmony____________________________#\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eintegration using harmony need sevral steps to be undertaken:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Perform log-normalization and feature selection, as well as SCT normalization on global object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    NormalizeData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    FindVariableFeatures(\u003cspan class=\"pl-v\"\u003eselection.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3000\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    ScaleData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    SCTransform(\u003cspan class=\"pl-v\"\u003evars.to.regress\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoRatio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Calculate PCs using variable features determined by SCTransform (3000 by default)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunPCA(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enpcs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Integration\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall.packages(\"harmony\")\u003c/span\u003e\n\nlibrary(\u003cspan class=\"pl-smi\"\u003eharmony\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunHarmony(\u003cspan class=\"pl-smi\"\u003emerged_seurat\u003c/span\u003e, \n\t\t\t\t\u003cspan class=\"pl-v\"\u003egroup.by.vars\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSurgery_Type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n\t\t\t\t\u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay.use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction.save\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunUMAP(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e40\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCluster identification and Inspect the effects of Harmony batch removel ____________#\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to set reduction to harmony and find the clusters\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindNeighbors(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindClusters(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eresolution\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e visualization\u003c/span\u003e\nIdents(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Color cells based on the sample name\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot UMAP \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal_harmony_UMAP_y_sample.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e________________________SuperCluster Identification____________#\u003c/span\u003e\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal_harmony_umap_cluster_with_label.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e let's visualize cells expressing supercluster markers:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e CD31: PECAM1\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eEPCAM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePECAM1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCOL1A1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePDGFRA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRGS5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD79A\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLYZ\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCD3D\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTPSAB1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal_umap_superCluster_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emarkers\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_umap_superCluster_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_umap_superCluster_cells.png\" alt=\"normal_umap_superCluster_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIt seems the epithelial cells are clustered together in cluster 4. So for further analysis, I will be focusing on these cells.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"epi_cell_ids \u0026lt;- rownames(harmonized_seurat@meta.data)[harmonized_seurat@meta.data$SCT_snn_res.0.1 == '4']\n\nepi_seurat \u0026lt;- subset(filtered_seurat, subset = cells %in% epi_cell_ids)\n\n# Perform log-normalization and feature selection, as well as SCT normalization on the global object\nepi_seurat \u0026lt;- epi_seurat %\u0026gt;%\n    NormalizeData() %\u0026gt;%\n    FindVariableFeatures(selection.method = \u0026quot;vst\u0026quot;, nfeatures = 3000) %\u0026gt;% \n    ScaleData() %\u0026gt;%\n    SCTransform(vars.to.regress = c(\u0026quot;mitoRatio\u0026quot;, \u0026quot;orig.ident\u0026quot;))\n\n# Calculate PCs using variable features determined by SCTransform (3000 by default)\nepi_seurat \u0026lt;- RunPCA(epi_seurat, assay = \u0026quot;SCT\u0026quot;, npcs = 50)\n\nepi_seurat \u0026lt;- RunHarmony(epi_seurat, \n\t\t\t\tgroup.by.vars = c(\u0026quot;orig.ident\u0026quot;, \u0026quot;Surgery_Type\u0026quot;), \n\t\t\t\treduction = \u0026quot;pca\u0026quot;, assay.use = \u0026quot;SCT\u0026quot;, reduction.save = \u0026quot;harmony\u0026quot;)\n\nepi_seurat \u0026lt;- RunUMAP(epi_seurat, reduction = \u0026quot;harmony\u0026quot;, assay = \u0026quot;SCT\u0026quot;, dims = 1:40)\n\n\n#Cluster identification and Inspect the effects of Harmony batch removel#\n\n# to set reduction to harmony and find the clusters\nepi_seurat \u0026lt;- FindNeighbors(object = epi_seurat, reduction = \u0026quot;harmony\u0026quot;)\nepi_seurat \u0026lt;- FindClusters(epi_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))\n\n# visualization\nIdents(epi_seurat) \u0026lt;- epi_seurat@meta.data$SCT_snn_res.0.1\n\n\n# color cells based on the cluster\n# Plot UMAP \npng(filename = \u0026quot;normal_cluster_epi_harmony_UMAP.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nDimPlot(epi_seurat,\n        reduction = \u0026quot;umap\u0026quot;,\n        label = TRUE,\n        label.size = 6)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eepi_cell_ids\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e)[\u003cspan class=\"pl-smi\"\u003eharmonized_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e subset(\u003cspan class=\"pl-smi\"\u003efiltered_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esubset\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecells\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e%in%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_cell_ids\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Perform log-normalization and feature selection, as well as SCT normalization on the global object\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    NormalizeData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    FindVariableFeatures(\u003cspan class=\"pl-v\"\u003eselection.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enfeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3000\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    ScaleData() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    SCTransform(\u003cspan class=\"pl-v\"\u003evars.to.regress\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emitoRatio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Calculate PCs using variable features determined by SCTransform (3000 by default)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunPCA(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enpcs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunHarmony(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \n\t\t\t\t\u003cspan class=\"pl-v\"\u003egroup.by.vars\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorig.ident\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSurgery_Type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), \n\t\t\t\t\u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay.use\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction.save\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunUMAP(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eassay\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSCT\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e40\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCluster identification and Inspect the effects of Harmony batch removel#\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e to set reduction to harmony and find the clusters\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindNeighbors(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eharmony\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindClusters(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eresolution\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0.1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.2\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.4\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e visualization\u003c/span\u003e\nIdents(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e@\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003emeta.data\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSCT_snn_res.0.1\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e color cells based on the cluster\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot UMAP \u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal_cluster_epi_harmony_UMAP.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_cluster_epi_harmony_UMAP.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_cluster_epi_harmony_UMAP.png\" alt=\"normal_cluster_epi_harmony_UMAP.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Marker identification\n\nepi_markers \u0026lt;- FindAllMarkers(object = epi_seurat, \n                          only.pos = TRUE,\n                          logfc.threshold = 0.25) \n\n# mutate the markers dataframe\n# Extract top 10 markers per cluster\n\nepi_top10 \u0026lt;- epi_markers %\u0026gt;%\n  mutate(delta_pct = (pct.1 - pct.2)) %\u0026gt;%\n  #filter(avg_log2FC \u0026gt; 1.5) %\u0026gt;%  # only keep rows where avg_log2FC \u0026gt; 1.5\n  group_by(cluster) %\u0026gt;%\n  top_n(n = 10, wt = delta_pct)\n\n\nepi_cluster_markers_10 \u0026lt;- epi_top10 %\u0026gt;% \n                   group_by(cluster) %\u0026gt;% \n                   summarize(genes = paste(gene, collapse = \u0026quot;,\u0026quot;))\n                   \n                   \n# Insepecting the expression patterns\npng(filename = \u0026quot;normal_markers_cells.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nFeaturePlot(object = epi_seurat,\n            features = unlist(strsplit(epi_cluster_markers_10[2,]$genes, split = \u0026quot;,\u0026quot;)),\n            order = TRUE,\n            min.cutoff = \u0026quot;q10\u0026quot;,\n            reduction = \u0026quot;umap\u0026quot;,\n            label = TRUE,\n            repel = TRUE)\n\ndev.off()\n                   \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Marker identification\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eepi_markers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e FindAllMarkers(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \n                          \u003cspan class=\"pl-v\"\u003eonly.pos\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n                          \u003cspan class=\"pl-v\"\u003elogfc.threshold\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.25\u003c/span\u003e) \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e mutate the markers dataframe\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Extract top 10 markers per cluster\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eepi_top10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_markers\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  mutate(\u003cspan class=\"pl-v\"\u003edelta_pct\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003epct.1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epct.2\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efilter(avg_log2FC \u0026gt; 1.5) %\u0026gt;%  # only keep rows where avg_log2FC \u0026gt; 1.5\u003c/span\u003e\n  group_by(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  top_n(\u003cspan class=\"pl-v\"\u003en\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edelta_pct\u003c/span\u003e)\n\n\n\u003cspan class=\"pl-smi\"\u003eepi_cluster_markers_10\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_top10\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                   group_by(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n                   summarize(\u003cspan class=\"pl-v\"\u003egenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e paste(\u003cspan class=\"pl-smi\"\u003egene\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecollapse\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n                   \n                   \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Insepecting the expression patterns\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal_markers_cells.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\nFeaturePlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003efeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e unlist(strsplit(\u003cspan class=\"pl-smi\"\u003eepi_cluster_markers_10\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,]\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esplit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e,\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)),\n            \u003cspan class=\"pl-v\"\u003eorder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003emin.cutoff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eq10\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\ndev.off()\n                   \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_markers_cells.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_markers_cells.png\" alt=\"normal_markers_cells.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003ecluster\u003c/th\u003e\n\u003cth\u003egenes\u003c/th\u003e\n\u003cth\u003ecluster name\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e0\u003c/td\u003e\n\u003ctd\u003eHS6ST3,SLC9A2,GSTM3,CLMN,GRK3,GRHL1,ZNF710,ERBB3,SLC17A5,ASTN2\u003c/td\u003e\n\u003ctd\u003eluminal_intermediate_stage\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e1\u003c/td\u003e\n\u003ctd\u003eGMNN,SLC7A11,ANXA10,NRARP,SLC12A6,TFF1,PTGS2,ODC1,DSP,MIR31HG\u003c/td\u003e\n\u003ctd\u003estromal_associated\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e2\u003c/td\u003e\n\u003ctd\u003eEGR1,DUSP2,DDIT4,DDIT3,KLF10,GATA2,TXNIP,AC005920.2,AL691403.1,PLA2G6\u003c/td\u003e\n\u003ctd\u003eearly_stage_basal_cell\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e3\u003c/td\u003e\n\u003ctd\u003eNBAT1,BICC1,ZNF350-AS1,SULT1E1,CASC15,KRT20,PTN,B4GALT3,RHEX,SCHLAP1\u003c/td\u003e\n\u003ctd\u003eintermediated_luminal_cell\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e4\u003c/td\u003e\n\u003ctd\u003ePALLD,SPOCK3,NRG1,MYO1B,NAV2,ETS1,CAV1,KRT5,FLNA,TUBB6\u003c/td\u003e\n\u003ctd\u003eintermediate_stage_basal_cell\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e5\u003c/td\u003e\n\u003ctd\u003eBCAT1,UPK2,MAL,ARMT1,SDHAF4,PTN,KRT20,SNX31,C4orf48,ISG15\u003c/td\u003e\n\u003ctd\u003eluminal_differentiated\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e6\u003c/td\u003e\n\u003ctd\u003eADAMTS9-AS2,EBF1,SPARCL1,PLCB4,CAVIN1,ZEB1,SLC2A3,CAV1,TCF4,KLF2\u003c/td\u003e\n\u003ctd\u003eearly_stage_basal_cells\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eReason for nomenclature:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecluster 0\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis cluster includes genes that are expressed in the luminal layer of bladder tissue. However, the specific combination of markers suggests a relatively early or intermediate stage of luminal cell differentiation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecluster 1\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBased on the gene expression profile of GMNN, SLC7A11, ANXA10, NRARP, SLC12A6, TFF1, PTGS2, ODC1, DSP, and MIR31HG, a possible name for the group of cells could be \"Stromal-Associated Cluster\" or \"Mesenchymal-Enriched Cluster\". This suggestion is based on the presence of genes like ANXA10 and DSP, which are typically associated with mesenchymal or stromal cell types. Additionally, genes like PTGS2 and SLC7A11 are commonly expressed in stromal or immune cells and may suggest an association with the tumor microenvironment or immune response.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecluster 2\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis cluster consists of genes associated with early differentiation processes and transcriptional regulation. The presence of genes like EGR1 and KLF10 suggests an early stage in the differentiation process.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecluster 3\u003c/strong\u003e\nThis cluster also contains markers associated with both basal and luminal cell characteristics. While KRT20 suggests a luminal presence, the combination of markers does not indicate a well-defined stage of luminal differentiation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecluster 4\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis cluster shows a mix of genes associated with both basal and luminal cell characteristics. While it may not represent the most differentiated luminal cells, it could be positioned in an intermediate stage of differentiation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecluster 5\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eLuminal Differentiation Cluster: This name is suggested based on the presence of markers such as UPK2 and KRT20. UPK2 (Uroplakin 2) is a marker of luminal/umbrella cells in the urothelium, which lines the bladder and urethra. KRT20 is also a luminal marker and is associated with differentiated epithelial cells in various tissues. This suggests that the cluster may represent a group of cells undergoing luminal differentiation or with a mature luminal phenotype.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecluster 6\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eBasal-like: This name emphasizes the presence of markers like ZEB1, TCF4, and KLF2, which have been associated with basal-like features in various epithelial tissues. It suggests that the cluster may represent cells with a basal-like differentiation program.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Rename all identities\nepi_seurat \u0026lt;- RenameIdents(object = epi_seurat,\n                                \u0026quot;0\u0026quot; = \u0026quot;intermediate_luminal\u0026quot;,\n                                \u0026quot;1\u0026quot; = \u0026quot;stromal_associated\u0026quot;,\n                                \u0026quot;2\u0026quot; = \u0026quot;early_basal_1\u0026quot;,\n                                \u0026quot;3\u0026quot; = \u0026quot;luminal[KRT20]_basal\u0026quot;,\n                                \u0026quot;4\u0026quot; = \u0026quot;luminal_basal[KRT5]\u0026quot;,\n                                \u0026quot;5\u0026quot; = \u0026quot;luminal_differentiated\u0026quot;,\n                                \u0026quot;6\u0026quot; = \u0026quot;early_basal_0\u0026quot;)\n# Plot the UMAP withy new labells\npng(filename = \u0026quot;normal_labelled_epi.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 600)\nDimPlot(object = epi_seurat, \n        reduction = \u0026quot;umap\u0026quot;, \n        label = TRUE,\n        label.size = 3,\n        repel = TRUE)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Rename all identities\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RenameIdents(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e,\n                                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eintermediate_luminal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estromal_associated\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eearly_basal_1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eluminal[KRT20]_basal\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eluminal_basal[KRT5]\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e5\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eluminal_differentiated\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n                                \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eearly_basal_0\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Plot the UMAP withy new labells\u003c/span\u003e\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal_labelled_epi.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e600\u003c/span\u003e)\nDimPlot(\u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eumap\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003elabel.size\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003erepel\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_labelled_epi.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_labelled_epi.png\" alt=\"normal_labelled_epi.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTrajectories of normal epithelial cells\u003c/h2\u003e\u003ca id=\"user-content-trajectories-of-normal-epithelial-cells\" class=\"anchor\" aria-label=\"Permalink: Trajectories of normal epithelial cells\" href=\"#trajectories-of-normal-epithelial-cells\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"epi_seurat$clusters \u0026lt;- Idents(epi_seurat)\npng(filename = \u0026quot;normal_trajectory.png\u0026quot;, width = 16, height = 8.135, units = \u0026quot;in\u0026quot;, res = 300)\nepi_seurat \u0026lt;- RunSlingshot(srt = epi_seurat, group.by = \u0026quot;clusters\u0026quot;, reduction = \u0026quot;UMAP\u0026quot;)\ndev.off()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eclusters\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e Idents(\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e)\npng(\u003cspan class=\"pl-v\"\u003efilename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormal_trajectory.png\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e16\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e8.135\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eunits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ein\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eres\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e RunSlingshot(\u003cspan class=\"pl-v\"\u003esrt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eepi_seurat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup.by\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eclusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ereduction\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUMAP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_trajectory.png\"\u003e\u003cimg src=\"https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_trajectory.png\" alt=\"normal_trajectory.png\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"scRNA sequencing analysis","anchor":"scrna-sequencing-analysis","htmlText":"scRNA sequencing analysis"},{"level":2,"text":"1) Data download on ComputeCanada","anchor":"1-data-download-on-computecanada","htmlText":"1) Data download on ComputeCanada"},{"level":2,"text":"2) Generating feature-sample expression matrix","anchor":"2-generating-feature-sample-expression-matrix","htmlText":"2) Generating feature-sample expression matrix"},{"level":2,"text":"3) Background noise removal","anchor":"3-background-noise-removal","htmlText":"3) Background noise removal"},{"level":2,"text":"4) Loading single-cell RNA-seq count data","anchor":"4-loading-single-cell-rna-seq-count-data","htmlText":"4) Loading single-cell RNA-seq count data"},{"level":2,"text":"5) Quality control","anchor":"5-quality-control","htmlText":"5) Quality control"},{"level":4,"text":"Cell counts per sample","anchor":"cell-counts-per-sample","htmlText":"Cell counts per sample"},{"level":4,"text":"UMI per cell","anchor":"umi-per-cell","htmlText":"UMI per cell"},{"level":4,"text":"Genes detected per cell","anchor":"genes-detected-per-cell","htmlText":"Genes detected per cell"},{"level":4,"text":"Novelty score","anchor":"novelty-score","htmlText":"Novelty score"},{"level":4,"text":"Mitochondrial gene expression detected per cell","anchor":"mitochondrial-gene-expression-detected-per-cell","htmlText":"Mitochondrial gene expression detected per cell"},{"level":4,"text":"Joint filtering: nUMI, nGene and mitoRatio","anchor":"joint-filtering-numi-ngene-and-mitoratio","htmlText":"Joint filtering: nUMI, nGene and mitoRatio"},{"level":3,"text":"Filtering","anchor":"filtering","htmlText":"Filtering"},{"level":4,"text":"Cell-level filtering","anchor":"cell-level-filtering","htmlText":"Cell-level filtering"},{"level":4,"text":"Gene-level filtering","anchor":"gene-level-filtering","htmlText":"Gene-level filtering"},{"level":4,"text":"Re-assess QC metrics","anchor":"re-assess-qc-metrics","htmlText":"Re-assess QC metrics"},{"level":2,"text":"6) Normalization, regressing out unwanted variation and clustering","anchor":"6-normalization-regressing-out-unwanted-variation-and-clustering","htmlText":"6) Normalization, regressing out unwanted variation and clustering"},{"level":3,"text":"Exploring sources of unwanted variation","anchor":"exploring-sources-of-unwanted-variation","htmlText":"Exploring sources of unwanted variation"},{"level":4,"text":"Evaluating effects of cell cycle and mitochondrial expression","anchor":"evaluating-effects-of-cell-cycle-and-mitochondrial-expression","htmlText":"Evaluating effects of cell cycle and mitochondrial expression"},{"level":3,"text":"SCTransform","anchor":"sctransform","htmlText":"SCTransform"},{"level":4,"text":"Integration","anchor":"integration","htmlText":"Integration"},{"level":3,"text":"Integration check","anchor":"integration-check","htmlText":"Integration check"},{"level":3,"text":"Clustering cells based on top PCs (metagenes)","anchor":"clustering-cells-based-on-top-pcs-metagenes","htmlText":"Clustering cells based on top PCs (metagenes)"},{"level":4,"text":"Identify significant PCs","anchor":"identify-significant-pcs","htmlText":"Identify significant PCs"},{"level":4,"text":"Cluster the cells","anchor":"cluster-the-cells","htmlText":"Cluster the cells"},{"level":4,"text":"Clustering quality control","anchor":"clustering-quality-control","htmlText":"Clustering quality control"},{"level":2,"text":"7) Marker identification and cell type assignment","anchor":"7-marker-identification-and-cell-type-assignment","htmlText":"7) Marker identification and cell type assignment"},{"level":1,"text":"8) MIBC vs. NMIBC","anchor":"8-mibc-vs-nmibc","htmlText":"8) MIBC vs. NMIBC"},{"level":1,"text":"9) Analyzing epithelial (EPCAM +) cells","anchor":"9-analyzing-epithelial-epcam--cells","htmlText":"9) Analyzing epithelial (EPCAM +) cells"},{"level":4,"text":"Marker identification for the 8 superclusters:","anchor":"marker-identification-for-the-8-superclusters","htmlText":"Marker identification for the 8 superclusters:"},{"level":3,"text":"Markers and cell types","anchor":"markers-and-cell-types","htmlText":"Markers and cell types"},{"level":2,"text":"10) DE and GSEA","anchor":"10-de-and-gsea","htmlText":"10) DE and GSEA"},{"level":2,"text":"11) Trajectory inference","anchor":"11-trajectory-inference","htmlText":"11) Trajectory inference"},{"level":3,"text":"Identify temporally dynamic genes","anchor":"identify-temporally-dynamic-genes","htmlText":"Identify temporally dynamic genes"},{"level":2,"text":"Exploring normal bladder mouse samples","anchor":"exploring-normal-bladder-mouse-samples","htmlText":"Exploring normal bladder mouse samples"},{"level":2,"text":"Trajectories of normal epithelial cells","anchor":"trajectories-of-normal-epithelial-cells","htmlText":"Trajectories of normal epithelial cells"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fhamidghaedi%2FscRNA_seq-analysis"}}],"overviewFilesProcessingTime":71.988812,"createFromTemplatePath":"/new?template_name=scRNA_seq-analysis\u0026template_owner=hamidghaedi"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/hamidghaedi/scRNA_seq-analysis/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/hamidghaedi/scRNA_seq-analysis/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/hamidghaedi/scRNA_seq-analysis/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/hamidghaedi/scRNA_seq-analysis/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/hamidghaedi/scRNA_seq-analysis/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           102 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/hamidghaedi/scRNA_seq-analysis/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/hamidghaedi/scRNA_seq-analysis/tree/main/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/hamidghaedi/scRNA_seq-analysis/tree/main/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/hamidghaedi/scRNA_seq-analysis/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/hamidghaedi/scRNA_seq-analysis/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      scRNA sequencing analysis
                     </h1>
                     <a aria-label="Permalink: scRNA sequencing analysis" class="anchor" href="#scrna-sequencing-analysis" id="user-content-scrna-sequencing-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     This repo presents steps needed to make sense of single-cell RNA sequencing (scRNA) data. I used a scRNA dataset coming from Zhaohui Chen
                     <em>
                      et al.
                     </em>
                     <a href="https://www.nature.com/articles/s41467-020-18916-5#Sec12" rel="nofollow">
                      paper
                     </a>
                     published in Nature Communications 11, Article number: 5077 (2020). The cohort consisted of eight primary bladder tumor tissues (2 low-grade bladder urothelial tumors, and six high-grade bladder urothelial tumors) along with 3 adjacent normal mucosae. In SRA datasets are under BioProject
                     <a href="https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA662018" rel="nofollow">
                      PRJNA662018
                     </a>
                     and SRA-explorer can be used to download the data. For practical scRNA-seq analysis, I followed this elegant
                     <a href="https://hbctraining.github.io/scRNA-seq_online/schedule/links-to-lessons.html" rel="nofollow">
                      tutorial
                     </a>
                     from Harvard Chan Bioinformatics Core.
                    </p>
                    <p dir="auto">
                     <strong>
                      NOTE
                     </strong>
                     If you like to work with
                     <code>
                      cellranger count
                     </code>
                     outputs and start analysis using
                     <code>
                      Seurat
                     </code>
                     you can download them from this
                     <a href="https://figshare.com/articles/dataset/filtered_zip/23834670" rel="nofollow">
                      link
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     Contents:
                    </p>
                    <ol dir="auto">
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#1-data-download-on-computecanada">
                       Data download on ComputeCanada
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#2-generating-feature-sample-expression-matrix">
                       Generating feature-sample expression matrix
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#4-Background-noise-removal">
                       Background noise removal
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#3-loading-single-cell-rna-seq-count-data">
                       Loading single-cell RNA-seq count data
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#5-quality-control">
                       Quality control
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#6-normalization-regressing-out-unwanted-variation-and-clustering">
                       Normalization, regressing out unwanted variation and clustering
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#7-marker-identification-and-cell-type-assignment">
                       Marker identification and cell type assignment
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#8-mibc-vs-nmibc">
                       Comparing muscle invasive BLCA vs. Non-muscle invasive BLCA
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#9-analyzing-epithelial-epcam--cells">
                       Analyzing epithelial (EPCAM +) cells
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#10-de-and-gsea">
                       DE and enrichment analysis
                      </a>
                     </li>
                     <li>
                      <a href="https://github.com/hamidghaedi/scRNA_seq-analysis#11-trajectory-inference">
                       Trajectory inference
                      </a>
                     </li>
                     <li>
                      RNA velocity analysis. Materials related to RNA velocity analysis have  moved
                      <a href="https://github.com/hamidghaedi/scRNA_velocity_analysis">
                       here
                      </a>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      1) Data download on ComputeCanada
                     </h2>
                     <a aria-label="Permalink: 1) Data download on ComputeCanada" class="anchor" href="#1-data-download-on-computecanada" id="user-content-1-data-download-on-computecanada">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
#!/usr/bin/env bash

#!/bin/bash
#SBATCH --account=#
#SBATCH --job-name=scRNA__DL
#SBATCH --qos=privileged
#SBATCH --nodes=1                # number of Nodes
#SBATCH --tasks-per-node=4        # number of MPI processes per node
#SBATCH --mem 8g
#SBATCH --time 12:00:00
#SBATCH --output=scRNA_fastq_DL.%J.out
#SBATCH --error=scRNA_fastq.%J.err
#SBATCH --mail-type=ALL
#SBATCH --mail-user=#


curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_1.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_2.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_1.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_2.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_1.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_2.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_1.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_2.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_1.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_2.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_1.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_2.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_1.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_2.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_1.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_2.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_1.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_2.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_1.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_2.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_1.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_2.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_2.fastq.gz" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env bash</span>

<span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --account=#</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=scRNA__DL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --qos=privileged</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --nodes=1                # number of Nodes</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --tasks-per-node=4        # number of MPI processes per node</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mem 8g</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --time 12:00:00</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --output=scRNA_fastq_DL.%J.out</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --error=scRNA_fastq.%J.err</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=ALL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=#</span>


curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_1.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/089/SRR12603789/SRR12603789_2.fastq.gz -o SRR12603789_urinary_bladder_cancer_fresh_sample_of_patient_number_02_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_1.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/087/SRR12603787/SRR12603787_2.fastq.gz -o SRR12603787_urinary_bladder_cancer_fresh_sample_of_patient_number_03_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_1.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/086/SRR12603786/SRR12603786_2.fastq.gz -o SRR12603786_urinary_bladder_cancer_fresh_sample_of_patient_number_04_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_1.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/090/SRR12603790/SRR12603790_2.fastq.gz -o SRR12603790_urinary_bladder_cancer_fresh_sample_of_patient_number_01_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_1.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/088/SRR12603788/SRR12603788_2.fastq.gz -o SRR12603788_normal_bladder_mucosa_3_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_1.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/084/SRR12603784/SRR12603784_2.fastq.gz -o SRR12603784_urinary_bladder_cancer_fresh_sample_of_patient_number_06_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_1.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/085/SRR12603785/SRR12603785_2.fastq.gz -o SRR12603785_urinary_bladder_cancer_fresh_sample_of_patient_number_05_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_1.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/083/SRR12603783/SRR12603783_2.fastq.gz -o SRR12603783_urinary_bladder_cancer_fresh_sample_of_patient_number_07_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_1.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/082/SRR12603782/SRR12603782_2.fastq.gz -o SRR12603782_urinary_bladder_cancer_fresh_sample_of_patient_number_08_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_1.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/081/SRR12603781/SRR12603781_2.fastq.gz -o SRR12603781_normal_bladder_mucosa_1_2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_1.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR126/080/SRR12603780/SRR12603780_2.fastq.gz -o SRR12603780_normal_bladder_mucosa_2_2.fastq.gz</pre>
                    </div>
                    <p dir="auto">
                     It is always recommended to run QC on samples; so using the following code we can run
                     <code>
                      fastqc
                     </code>
                     on all samples
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash
#SBATCH --account= # add your account name 
#SBATCH --job-name=fastqc
#SBATCH --qos=privileged
#SBATCH --nodes=12                # number of Nodes
#SBATCH --tasks-per-node=5        # number of MPI processes per node
#SBATCH --mem 16g
#SBATCH --time 24:00:00
#SBATCH --output=fastqc.%J.out
#SBATCH --error=fastqc.%J.err
#SBATCH --mail-type=ALL
#SBATCH --mail-user=# add your email address 


module load fastqc

fastqc ./fastq/*.fastq.gz --outdir ./qc_raw -f fastq -t 12 --nogroup
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --account= # add your account name </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=fastqc</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --qos=privileged</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --nodes=12                # number of Nodes</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --tasks-per-node=5        # number of MPI processes per node</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mem 16g</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --time 24:00:00</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --output=fastqc.%J.out</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --error=fastqc.%J.err</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=ALL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=# add your email address </span>


module load fastqc

fastqc ./fastq/<span class="pl-k">*</span>.fastq.gz --outdir ./qc_raw -f fastq -t 12 --nogroup
</pre>
                    </div>
                    <p dir="auto">
                     There are differences between the sequence length of read 1 and read 2 for each sample; read 1 provides data on Cell barcode &amp; UMI and read 2 has the insert sequence. Files coming from SRA usually come with names like something_1_fastq.gz and something_2_fastq.gz . This
                     <a href="https://kb.10xgenomics.com/hc/en-us/articles/115003802691" rel="nofollow">
                      blog
                     </a>
                     is helpful to see what are the naming requirements of fastq files for cellranger tool. Briefly;
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       incompatible file name: SRR9291388_1.fastq.gz
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       compatible file name: SRR9291388_S1_L001_R1_001.fastq.g
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     So we need to change file names and also set up the directories for
                     <code>
                      cellranger count
                     </code>
                     ;
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for file in *.fastq.gz
do
  dir=${file%_*}
  echo $dir
  mkdir $dir
  mv $file ./$dir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">file</span> <span class="pl-k">in</span> <span class="pl-k">*</span>.fastq.gz
<span class="pl-k">do</span>
  dir=<span class="pl-smi">${file<span class="pl-k">%</span>_<span class="pl-k">*</span>}</span>
  <span class="pl-c1">echo</span> <span class="pl-smi">$dir</span>
  mkdir <span class="pl-smi">$dir</span>
  mv <span class="pl-smi">$file</span> ./<span class="pl-smi">$dir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2) Generating feature-sample expression matrix
                     </h2>
                     <a aria-label="Permalink: 2) Generating feature-sample expression matrix" class="anchor" href="#2-generating-feature-sample-expression-matrix" id="user-content-2-generating-feature-sample-expression-matrix">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The next step is to run
                     <code>
                      cellranger count
                     </code>
                     on each sample. There are different types of scRNA libs with different fastq files. This
                     <a href="https://www.10xgenomics.com/support/single-cell-gene-expression/documentation/steps/sequencing/sequencing-requirements-for-single-cell-3" rel="nofollow">
                      blog
                     </a>
                     provides details on the type of libs from 10X genomics.
                    </p>
                    <p dir="auto">
                     Running
                     <code>
                      cellranger
                     </code>
                     on a cluster with
                     <code>
                      SLURM
                     </code>
                     as job scheduler is not an easy task. The following is what I came up with working best for me working on ComputeCanada cluster:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash
#SBATCH --account=#
#SBATCH -J test_cellranger
#SBATCH --export=ALL
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=16
#SBATCH --signal=2
#SBATCH --no-requeue
### Alternatively: --ntasks=1 --cpus-per-task={NUM_THREADS}
###   Consult with your cluster administrators to find the combination that
###   works best for single-node, multi-threaded applications on your system.
#SBATCH --mem=250G
#SBATCH --time 48:00:00
#SBATCH --output=test_cellranger.%J.out
#SBATCH --error=test_cellranger.%J.err
#SBATCH --mail-type=ALL
#SBATCH --mail-user=#


module load cellranger

cd /home/ghaedi/projects/def-gooding-ab/ghaedi/sc/raw
for d in *
do
echo $d
ID=${d:0:11}
cellranger count --id=$ID \
                 --transcriptome=/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/refdata-gex-GRCh38-2020-A \
                 --fastqs=$d
                 --chemistry=SC3Pv3
done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --account=#</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -J test_cellranger</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --export=ALL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --ntasks=1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --cpus-per-task=16</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --signal=2</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --no-requeue</span>
<span class="pl-c"><span class="pl-c">#</span>## Alternatively: --ntasks=1 --cpus-per-task={NUM_THREADS}</span>
<span class="pl-c"><span class="pl-c">#</span>##   Consult with your cluster administrators to find the combination that</span>
<span class="pl-c"><span class="pl-c">#</span>##   works best for single-node, multi-threaded applications on your system.</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mem=250G</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --time 48:00:00</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --output=test_cellranger.%J.out</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --error=test_cellranger.%J.err</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=ALL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=#</span>


module load cellranger

<span class="pl-c1">cd</span> /home/ghaedi/projects/def-gooding-ab/ghaedi/sc/raw
<span class="pl-k">for</span> <span class="pl-smi">d</span> <span class="pl-k">in</span> <span class="pl-k">*</span>
<span class="pl-k">do</span>
<span class="pl-c1">echo</span> <span class="pl-smi">$d</span>
ID=<span class="pl-smi">${d<span class="pl-k">:</span>0<span class="pl-k">:</span>11}</span>
cellranger count --id=<span class="pl-smi">$ID</span> \
                 --transcriptome=/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/refdata-gex-GRCh38-2020-A \
                 --fastqs=<span class="pl-smi">$d</span>
                 --chemistry=SC3Pv3
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The next steps are mainly based on Harvard Chan Bioinformatics Core materials on
                     <a href="https://hbctraining.github.io/scRNA-seq_online/schedule/links-to-lessons.html" rel="nofollow">
                      scRNA-seq analysis
                     </a>
                     training.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      3) Background noise removal
                     </h2>
                     <a aria-label="Permalink: 3) Background noise removal" class="anchor" href="#3-background-noise-removal" id="user-content-3-background-noise-removal">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Background removal is crucial in single-cell RNA sequencing (scRNA-seq) analysis due to the presence of contaminants such as cell-free "ambient" RNA and chimeric cDNA molecules resulting from barcode swapping.
Barcode swapping can happen when cDNA molecules from different beads, which carry different barcodes, mix together during the amplification step. This can occur due to unremoved oligonucleotides from other beads or incomplete extension of PCR products. As a result, chimeric cDNA molecules are formed, where the barcode and UMI sequences are "swapped" between different beads. When these chimeric molecules are sequenced, the assigned barcode does not match the original bead, leading to the incorrect assignment of the cDNA to the corresponding cell.
                    </p>
                    <p dir="auto">
                     These contaminants contribute to background noise, which can negatively impact data analysis. Background noise reduces the separability of cell-type clusters, interferes with differential expression analysis, and confounds comparisons between samples. To address this, algorithms like SoupX, DecontX, and CellBender estimate and correct for background noise using marker genes, empty droplets, and modeling of barcode swapping. Evaluating method performance involves assessing if the model effectively removes expression from other cell types using datasets with known cell type profiles and exclusive marker genes. However, the impact of background correction on expression level shifts induced by background noise remains a challenge in scRNA-seq analysis.
                    </p>
                    <p dir="auto">
                     As recommended by
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02978-x" rel="nofollow">
                      Philipp Janssen et al,2023
                     </a>
                     :
                    </p>
                    <p dir="auto">
                     <code>
                      In summary, for marker gene analysis, we would always recommend background removal, but for classification, clustering and pseudotime analyses, we would only recommend background removal when background noise levels are high
                     </code>
                    </p>
                    <p dir="auto">
                     In this repo, I won't be using the cellbender filtered matrix, as the main focus of the analysis is cell clustering, rather than marker identification. The following is a fully functional cellbender code for the given samples
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Running cellbender image on co pute canada 
##pulling the image (creating sif file):
module load apptainer
apptainer pull cellbender.sif docker://us.gcr.io/broad-dsde-methods/cellbender:0.2.2" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Running cellbender image on co pute canada </span>
<span class="pl-c"><span class="pl-c">#</span>#pulling the image (creating sif file):</span>
module load apptainer
apptainer pull cellbender.sif docker://us.gcr.io/broad-dsde-methods/cellbender:0.2.2</pre>
                    </div>
                    <p dir="auto">
                     Sbatch script to run cellbender on Graham with GPU
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#!/bin/bash
#SBATCH --account=def-gooding-ab
#SBATCH --job-name=cellBender
#SBATCH --nodes=1                # number of Nodes
#SBATCH --gpus-per-node=p100:2
#SBATCH --ntasks-per-node=32       
#SBATCH --mem=127000M
#SBATCH --time 08:00:00
#SBATCH --output=cellBender.%J.out
#SBATCH --error=cellBender.%J.err
#SBATCH --mail-type=ALL
#SBATCH --mail-user=qaedi.65@gmail.com

module load apptainer

# Define the input and output directories
input_dir="/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/filtered"
output_dir="/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/cellBender_filtered"

# Iterate over the sample directories
for sample_dir in "$input_dir"/*; do
    if [[ -d "$sample_dir" ]]; then
        sample_name=$(basename "$sample_dir")
        echo "Processing sample: $sample_name"

        # Create a directory for the current sample within the output directory
        sample_output_dir="$output_dir/$sample_name"
        mkdir -p "$sample_output_dir"

        # Run cellbender remove-background command for the current sample
        apptainer run -C -W ${SLURM_TMPDIR} --nv --rocm \
        --bind "$input_dir":"/mnt/filtered" \
        --bind "$sample_output_dir":"/mnt/sample_output" cellbender.sif \
        cellbender remove-background \
        --input "/mnt/filtered/$sample_name" \
        --output "/mnt/sample_output/$sample_name.h5" \
        --expected-cells 7000 \
        --total-droplets-included 20000 \
        --fpr 0.01 \
        --epochs 150 \
        --cuda

        echo "Completed processing sample: $sample_name"
        echo
    fi
done' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --account=def-gooding-ab</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=cellBender</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --nodes=1                # number of Nodes</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --gpus-per-node=p100:2</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --ntasks-per-node=32       </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mem=127000M</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --time 08:00:00</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --output=cellBender.%J.out</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --error=cellBender.%J.err</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=ALL</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=qaedi.65@gmail.com</span>

module load apptainer

<span class="pl-c"><span class="pl-c">#</span> Define the input and output directories</span>
input_dir=<span class="pl-s"><span class="pl-pds">"</span>/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/filtered<span class="pl-pds">"</span></span>
output_dir=<span class="pl-s"><span class="pl-pds">"</span>/home/ghaedi/projects/def-gooding-ab/ghaedi/sc/cellBender_filtered<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span> Iterate over the sample directories</span>
<span class="pl-k">for</span> <span class="pl-smi">sample_dir</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$input_dir</span><span class="pl-pds">"</span></span>/<span class="pl-k">*</span><span class="pl-k">;</span> <span class="pl-k">do</span>
    <span class="pl-k">if</span> [[ <span class="pl-k">-d</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$sample_dir</span><span class="pl-pds">"</span></span> ]]<span class="pl-k">;</span> <span class="pl-k">then</span>
        sample_name=<span class="pl-s"><span class="pl-pds">$(</span>basename <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$sample_dir</span><span class="pl-pds">"</span></span><span class="pl-pds">)</span></span>
        <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Processing sample: <span class="pl-smi">$sample_name</span><span class="pl-pds">"</span></span>

        <span class="pl-c"><span class="pl-c">#</span> Create a directory for the current sample within the output directory</span>
        sample_output_dir=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$output_dir</span>/<span class="pl-smi">$sample_name</span><span class="pl-pds">"</span></span>
        mkdir -p <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$sample_output_dir</span><span class="pl-pds">"</span></span>

        <span class="pl-c"><span class="pl-c">#</span> Run cellbender remove-background command for the current sample</span>
        apptainer run -C -W <span class="pl-smi">${SLURM_TMPDIR}</span> --nv --rocm \
        --bind <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$input_dir</span><span class="pl-pds">"</span></span>:<span class="pl-s"><span class="pl-pds">"</span>/mnt/filtered<span class="pl-pds">"</span></span> \
        --bind <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$sample_output_dir</span><span class="pl-pds">"</span></span>:<span class="pl-s"><span class="pl-pds">"</span>/mnt/sample_output<span class="pl-pds">"</span></span> cellbender.sif \
        cellbender remove-background \
        --input <span class="pl-s"><span class="pl-pds">"</span>/mnt/filtered/<span class="pl-smi">$sample_name</span><span class="pl-pds">"</span></span> \
        --output <span class="pl-s"><span class="pl-pds">"</span>/mnt/sample_output/<span class="pl-smi">$sample_name</span>.h5<span class="pl-pds">"</span></span> \
        --expected-cells 7000 \
        --total-droplets-included 20000 \
        --fpr 0.01 \
        --epochs 150 \
        --cuda

        <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Completed processing sample: <span class="pl-smi">$sample_name</span><span class="pl-pds">"</span></span>
        <span class="pl-c1">echo</span>
    <span class="pl-k">fi</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4) Loading single-cell RNA-seq count data
                     </h2>
                     <a aria-label="Permalink: 4) Loading single-cell RNA-seq count data" class="anchor" href="#4-loading-single-cell-rna-seq-count-data" id="user-content-4-loading-single-cell-rna-seq-count-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Setup the Seurat Object

library(tidyverse)
library(Seurat)
library(patchwork)
library(cowplot)

# create list of samples
samples &lt;- list.files(&quot;~/scRNA/filtered/&quot;)
#samples &lt;- samples[grepl('^filtered',samples,perl=T)]

# read files inot Seurat objects
for (file in samples){
  print(paste0(file))
  seurat_data &lt;- Read10X(data.dir = paste0(&quot;~/scRNA/filtered/&quot;, file))
  seurat_obj &lt;- CreateSeuratObject(counts = seurat_data, 
                                   min.features = 100, 
                                   project = file)
  assign(file, seurat_obj)
}

# now merging all objects inot one Seurat obj

merged_seurat &lt;- merge(x = SRR12603780, 
                       y = c(SRR12603781,
                             SRR12603782,
                             SRR12603783,
                             SRR12603784,
                             SRR12603785,
                             SRR12603786,
                             SRR12603787,
                             SRR12603788,
                             SRR12603789,
                             SRR12603790),
                       #Because the same cell IDs can be used for different samples, we add a sample-specific prefix 
                       # to each of our cell IDs using the add.cell.id argument.
                       add.cell.id = samples)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Setup the Seurat Object</span>

library(<span class="pl-smi">tidyverse</span>)
library(<span class="pl-smi">Seurat</span>)
library(<span class="pl-smi">patchwork</span>)
library(<span class="pl-smi">cowplot</span>)

<span class="pl-c"><span class="pl-c">#</span> create list of samples</span>
<span class="pl-smi">samples</span> <span class="pl-k">&lt;-</span> list.files(<span class="pl-s"><span class="pl-pds">"</span>~/scRNA/filtered/<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>samples &lt;- samples[grepl('^filtered',samples,perl=T)]</span>

<span class="pl-c"><span class="pl-c">#</span> read files inot Seurat objects</span>
<span class="pl-k">for</span> (<span class="pl-smi">file</span> <span class="pl-k">in</span> <span class="pl-smi">samples</span>){
  print(paste0(<span class="pl-smi">file</span>))
  <span class="pl-smi">seurat_data</span> <span class="pl-k">&lt;-</span> Read10X(<span class="pl-v">data.dir</span> <span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>~/scRNA/filtered/<span class="pl-pds">"</span></span>, <span class="pl-smi">file</span>))
  <span class="pl-smi">seurat_obj</span> <span class="pl-k">&lt;-</span> CreateSeuratObject(<span class="pl-v">counts</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_data</span>, 
                                   <span class="pl-v">min.features</span> <span class="pl-k">=</span> <span class="pl-c1">100</span>, 
                                   <span class="pl-v">project</span> <span class="pl-k">=</span> <span class="pl-smi">file</span>)
  assign(<span class="pl-smi">file</span>, <span class="pl-smi">seurat_obj</span>)
}

<span class="pl-c"><span class="pl-c">#</span> now merging all objects inot one Seurat obj</span>

<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">SRR12603780</span>, 
                       <span class="pl-v">y</span> <span class="pl-k">=</span> c(<span class="pl-smi">SRR12603781</span>,
                             <span class="pl-smi">SRR12603782</span>,
                             <span class="pl-smi">SRR12603783</span>,
                             <span class="pl-smi">SRR12603784</span>,
                             <span class="pl-smi">SRR12603785</span>,
                             <span class="pl-smi">SRR12603786</span>,
                             <span class="pl-smi">SRR12603787</span>,
                             <span class="pl-smi">SRR12603788</span>,
                             <span class="pl-smi">SRR12603789</span>,
                             <span class="pl-smi">SRR12603790</span>),
                       <span class="pl-c"><span class="pl-c">#</span>Because the same cell IDs can be used for different samples, we add a sample-specific prefix </span>
                       <span class="pl-c"><span class="pl-c">#</span> to each of our cell IDs using the add.cell.id argument.</span>
                       <span class="pl-v">add.cell.id</span> <span class="pl-k">=</span> <span class="pl-smi">samples</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5) Quality control
                     </h2>
                     <a aria-label="Permalink: 5) Quality control" class="anchor" href="#5-quality-control" id="user-content-5-quality-control">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     There are columns in the metadata:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       orig.ident: this column will contain the sample identity if known. It will default to the value we provided for the project argument when loading in the data
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       nCount_RNA: represents the number of UMIs per cell. UMI (unique molecular identifiers) is used to determine whether a read is a biological or technical duplicate (PCR duplicate). Reads with different UMIs mapping to the same transcript were derived from different molecules and are biological duplicates - each read should be counted. Reads with the same UMI originated from the same molecule and are technical duplicates - the UMIs should be collapsed to be counted as a single read.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       nFeature_RNA: represents the number of genes detected per cell
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     Recommended features to add to metadata:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       number of genes detected per UMI (or novelty score): more genes detected per UMI, more complex our data
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       mitochondrial ratio: this metric will give us a percentage of cell reads originating from the mitochondrial genes (coming from dying cells)
                      </p>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Explore merged metadata
View(merged_seurat@meta.data)


#Add number of genes per UMI for each cell to metadata
merged_seurat$log10GenesPerUMI &lt;- log10(merged_seurat$nFeature_RNA) / log10(merged_seurat$nCount_RNA)

# Compute percent mito ratio
merged_seurat$mitoRatio &lt;- PercentageFeatureSet(object = merged_seurat, pattern = "^MT-")
merged_seurat$mitoRatio &lt;- merged_seurat@meta.data$mitoRatio / 100

# Create metadata dataframe
metadata &lt;- merged_seurat@meta.data
# Add cell IDs to metadata
metadata$cells &lt;- rownames(metadata)
# adding sample type to metadata. The orginal file could be download from SRA explorer

SampleType &lt;- c("BLCA", "BLCA", "Normal", "BLCA", "BLCA", "BLCA", "BLCA", "BLCA", "BLCA", "Normal", "Normal")
# sample type with grade (Not tested)
#SampleType &lt;- c("BLCA_LG", "BLCA_LG", "Normal", "BLCA_HG", "BLCA_HG", "BLCA_HG", "BLCA_HG", "BLCA_HG", "BLCA_HG", "Normal", "Normal")

names(SampleType) &lt;- c("SRR12603789", "SRR12603790", "SRR12603788", "SRR12603787", "SRR12603786", "SRR12603785", "SRR12603784", "SRR12603783", "SRR12603782", "SRR12603781", "SRR12603780")

metadata$sampleType &lt;- stringr::str_replace_all(metadata$orig.ident, SampleType)


# Rename columns
metadata &lt;- metadata %&gt;%
  dplyr::rename(seq_folder = orig.ident,
                nUMI = nCount_RNA,
                nGene = nFeature_RNA,
                sample = sampleType)

# Add metadata back to Seurat object
merged_seurat@meta.data &lt;- metadata

# Create .RData object to load at any time
save(merged_seurat, file="~/scRNA/merged_filtered_seurat.RData")

#' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Explore merged metadata</span>
View(<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)


<span class="pl-c"><span class="pl-c">#</span>Add number of genes per UMI for each cell to metadata</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">log10GenesPerUMI</span> <span class="pl-k">&lt;-</span> log10(<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">nFeature_RNA</span>) <span class="pl-k">/</span> log10(<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">nCount_RNA</span>)

<span class="pl-c"><span class="pl-c">#</span> Compute percent mito ratio</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;-</span> PercentageFeatureSet(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">merged_seurat</span>, <span class="pl-v">pattern</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>^MT-<span class="pl-pds">"</span></span>)
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">/</span> <span class="pl-c1">100</span>

<span class="pl-c"><span class="pl-c">#</span> Create metadata dataframe</span>
<span class="pl-smi">metadata</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>
<span class="pl-c"><span class="pl-c">#</span> Add cell IDs to metadata</span>
<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">cells</span> <span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">metadata</span>)
<span class="pl-c"><span class="pl-c">#</span> adding sample type to metadata. The orginal file could be download from SRA explorer</span>

<span class="pl-smi">SampleType</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Normal<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Normal<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Normal<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> sample type with grade (Not tested)</span>
<span class="pl-c"><span class="pl-c">#</span>SampleType &lt;- c("BLCA_LG", "BLCA_LG", "Normal", "BLCA_HG", "BLCA_HG", "BLCA_HG", "BLCA_HG", "BLCA_HG", "BLCA_HG", "Normal", "Normal")</span>

names(<span class="pl-smi">SampleType</span>) <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>SRR12603789<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603790<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603788<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603787<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603786<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603785<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603784<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603783<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603782<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603781<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603780<span class="pl-pds">"</span></span>)

<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">sampleType</span> <span class="pl-k">&lt;-</span> <span class="pl-e">stringr</span><span class="pl-k">::</span>str_replace_all(<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">orig.ident</span>, <span class="pl-smi">SampleType</span>)


<span class="pl-c"><span class="pl-c">#</span> Rename columns</span>
<span class="pl-smi">metadata</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>%
  <span class="pl-e">dplyr</span><span class="pl-k">::</span>rename(<span class="pl-v">seq_folder</span> <span class="pl-k">=</span> <span class="pl-smi">orig.ident</span>,
                <span class="pl-v">nUMI</span> <span class="pl-k">=</span> <span class="pl-smi">nCount_RNA</span>,
                <span class="pl-v">nGene</span> <span class="pl-k">=</span> <span class="pl-smi">nFeature_RNA</span>,
                <span class="pl-v">sample</span> <span class="pl-k">=</span> <span class="pl-smi">sampleType</span>)

<span class="pl-c"><span class="pl-c">#</span> Add metadata back to Seurat object</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">metadata</span>

<span class="pl-c"><span class="pl-c">#</span> Create .RData object to load at any time</span>
save(<span class="pl-smi">merged_seurat</span>, <span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>~/scRNA/merged_filtered_seurat.RData<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span></span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Cell counts per sample
                     </h4>
                     <a aria-label="Permalink: Cell counts per sample" class="anchor" href="#cell-counts-per-sample" id="user-content-cell-counts-per-sample">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Visualize the number of cell counts per sample
bqcc &lt;- metadata %&gt;% 
  ggplot(aes(x=seq_folder, fill=sample)) + 
  geom_bar() +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("NCells before QC")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Visualize the number of cell counts per sample</span>
<span class="pl-smi">bqcc</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>% 
  ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">seq_folder</span>, <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">sample</span>)) <span class="pl-k">+</span> 
  geom_bar() <span class="pl-k">+</span>
  theme_classic() <span class="pl-k">+</span>
  theme(<span class="pl-v">axis.text.x</span> <span class="pl-k">=</span> element_text(<span class="pl-v">angle</span> <span class="pl-k">=</span> <span class="pl-c1">45</span>, <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, <span class="pl-v">hjust</span><span class="pl-k">=</span><span class="pl-c1">1</span>)) <span class="pl-k">+</span>
  theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">hjust</span><span class="pl-k">=</span><span class="pl-c1">0.5</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
  ggtitle(<span class="pl-s"><span class="pl-pds">"</span>NCells before QC<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cell_count_per_sample_before_QC.png" rel="noopener noreferrer" target="_blank">
                      <img alt="cell_count_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cell_count_per_sample_before_QC.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      UMI per cell
                     </h4>
                     <a aria-label="Permalink: UMI per cell" class="anchor" href="#umi-per-cell" id="user-content-umi-per-cell">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Typically, we expect the UMI counts per cell to be higher than 500, which is the lower limit of the expected range. If the UMI counts range between 500-1000, the data is still usable, but deeper sequencing may have been beneficial for these cells.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Visualize the number UMIs/transcripts per cell
metadata %&gt;% 
  ggplot(aes(x=nUMI, fill= sample)) + 
  geom_density(alpha = 0.5) + 
  scale_x_log10() + 
  theme_classic() +
  ylab("Cell density") +
  facet_wrap(~seq_folder) +
  geom_vline(xintercept = 1000) +
  labs(fill = "Sample")
  ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Visualize the number UMIs/transcripts per cell</span>
<span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>% 
  ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">nUMI</span>, <span class="pl-v">fill</span><span class="pl-k">=</span> <span class="pl-smi">sample</span>)) <span class="pl-k">+</span> 
  geom_density(<span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>) <span class="pl-k">+</span> 
  scale_x_log10() <span class="pl-k">+</span> 
  theme_classic() <span class="pl-k">+</span>
  ylab(<span class="pl-s"><span class="pl-pds">"</span>Cell density<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  facet_wrap(<span class="pl-k">~</span><span class="pl-smi">seq_folder</span>) <span class="pl-k">+</span>
  geom_vline(<span class="pl-v">xintercept</span> <span class="pl-k">=</span> <span class="pl-c1">1000</span>) <span class="pl-k">+</span>
  labs(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>)
  </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nUMI.png" rel="noopener noreferrer" target="_blank">
                      <img alt="cell_count_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nUMI.png" style="max-width: 100%;"/>
                     </a>
                     So the cells have way more than 1K UMI!
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Genes detected per cell
                     </h4>
                     <a aria-label="Permalink: Genes detected per cell" class="anchor" href="#genes-detected-per-cell" id="user-content-genes-detected-per-cell">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     In scRNA-seq, the number of genes detected per cell is a crucial quality metric that we expect to be similar to the UMI detection, albeit slightly lower. For high-quality data, the proportional histogram of genes detected per cell should show a single peak that represents encapsulated cells. However, if there is a small shoulder or a bimodal distribution to the left of the main peak, this could indicate a few things. It could be due to some failed cells or biologically different cell types, such as quiescent cell populations or less complex cells of interest. For instance, larger cells or different cell types may have higher gene counts.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Visualize the distribution of genes detected per cell via histogram
metadata %&gt;% 
  ggplot(aes(x=nGene, fill= sample)) + 
  geom_density(alpha = 0.5) + 
  scale_x_log10() + 
  theme_classic() +
  ylab("Cell density") +
  facet_wrap(~seq_folder) +
  geom_vline(xintercept = 500) +
  labs(fill = "Sample")
  ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Visualize the distribution of genes detected per cell via histogram</span>
<span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>% 
  ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">nGene</span>, <span class="pl-v">fill</span><span class="pl-k">=</span> <span class="pl-smi">sample</span>)) <span class="pl-k">+</span> 
  geom_density(<span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>) <span class="pl-k">+</span> 
  scale_x_log10() <span class="pl-k">+</span> 
  theme_classic() <span class="pl-k">+</span>
  ylab(<span class="pl-s"><span class="pl-pds">"</span>Cell density<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  facet_wrap(<span class="pl-k">~</span><span class="pl-smi">seq_folder</span>) <span class="pl-k">+</span>
  geom_vline(<span class="pl-v">xintercept</span> <span class="pl-k">=</span> <span class="pl-c1">500</span>) <span class="pl-k">+</span>
  labs(<span class="pl-v">fill</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>)
  </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nGene_before_QC.png" rel="noopener noreferrer" target="_blank">
                      <img alt="gene_count_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nGene_before_QC.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Novelty score
                     </h4>
                     <a aria-label="Permalink: Novelty score" class="anchor" href="#novelty-score" id="user-content-novelty-score">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The novelty score, computed as the ratio of nGenes over nUMI, measures the complexity of RNA species in each cell. A low number of genes detected in a cell with many captured transcripts (high nUMI) indicates low complexity or novelty. This could be due to an artifact, contamination, or a specific cell type (e.g. red blood cells). A good-quality cell typically has a novelty score above 0.80.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Visualize the overall complexity of the gene expression by visualizing the genes detected per UMI (novelty score)
metadata %&gt;%
  ggplot(aes(x=log10GenesPerUMI, fill=sample)) +
  geom_density(alpha = 0.5) +
  theme_classic() +
  facet_wrap(~seq_folder) +
  xlab("Novelty Score") +
  geom_vline(xintercept = 0.8)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Visualize the overall complexity of the gene expression by visualizing the genes detected per UMI (novelty score)</span>
<span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>%
  ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">log10GenesPerUMI</span>, <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">sample</span>)) <span class="pl-k">+</span>
  geom_density(<span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>) <span class="pl-k">+</span>
  theme_classic() <span class="pl-k">+</span>
  facet_wrap(<span class="pl-k">~</span><span class="pl-smi">seq_folder</span>) <span class="pl-k">+</span>
  xlab(<span class="pl-s"><span class="pl-pds">"</span>Novelty Score<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  geom_vline(<span class="pl-v">xintercept</span> <span class="pl-k">=</span> <span class="pl-c1">0.8</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/novelt_score_before_QC.png" rel="noopener noreferrer" target="_blank">
                      <img alt="gene_count_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/novelt_score_before_QC.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Mitochondrial gene expression detected per cell
                     </h4>
                     <a aria-label="Permalink: Mitochondrial gene expression detected per cell" class="anchor" href="#mitochondrial-gene-expression-detected-per-cell" id="user-content-mitochondrial-gene-expression-detected-per-cell">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     High levels of expression from mitochondria indicate dying or dead cells. Basically, quality samples are those that surpass 0.2 mitochondria ratio mark.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Visualize the distribution of mitochondrial gene expression detected per cell
metadata %&gt;%
  ggplot(aes(x=mitoRatio, fill=sample)) + 
  geom_density(alpha = 0.5) + 
  scale_x_log10() + 
  scale_x_continuous(labels = function(x) sprintf("%.1f", x)) + 
  theme_classic() +
  facet_wrap(~seq_folder) +
  geom_vline(xintercept = 0.2)
' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Visualize the distribution of mitochondrial gene expression detected per cell</span>
<span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>%
  ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">mitoRatio</span>, <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">sample</span>)) <span class="pl-k">+</span> 
  geom_density(<span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>) <span class="pl-k">+</span> 
  scale_x_log10() <span class="pl-k">+</span> 
  scale_x_continuous(<span class="pl-en">labels</span> <span class="pl-k">=</span> <span class="pl-k">function</span>(<span class="pl-smi">x</span>) sprintf(<span class="pl-s"><span class="pl-pds">"</span>%.1f<span class="pl-pds">"</span></span>, <span class="pl-smi">x</span>)) <span class="pl-k">+</span> 
  theme_classic() <span class="pl-k">+</span>
  facet_wrap(<span class="pl-k">~</span><span class="pl-smi">seq_folder</span>) <span class="pl-k">+</span>
  geom_vline(<span class="pl-v">xintercept</span> <span class="pl-k">=</span> <span class="pl-c1">0.2</span>)
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/mitoRatio_before_qc.png" rel="noopener noreferrer" target="_blank">
                      <img alt="mitoRatio_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/mitoRatio_before_qc.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Joint filtering: nUMI, nGene and mitoRatio
                     </h4>
                     <a aria-label="Permalink: Joint filtering: nUMI, nGene and mitoRatio" class="anchor" href="#joint-filtering-numi-ngene-and-mitoratio" id="user-content-joint-filtering-numi-ngene-and-mitoratio">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Visualize the correlation between genes detected and number of UMIs and determine whether strong presence of cells with low numbers of genes/UMIs

metadata %&gt;% 
  	ggplot(aes(x=nUMI, y=nGene, color=mitoRatio)) + 
  	geom_point() + 
  	scale_colour_gradient(low = "gray90", high = "black") +
  	stat_smooth(method=lm) +
  	scale_x_log10() + 
  	scale_y_log10() + 
  	theme_classic() +
  	geom_vline(xintercept = 1000) +
  	geom_hline(yintercept = 500) +
  	facet_wrap(~seq_folder)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Visualize the correlation between genes detected and number of UMIs and determine whether strong presence of cells with low numbers of genes/UMIs</span>

<span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>% 
  	ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">nUMI</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">nGene</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">mitoRatio</span>)) <span class="pl-k">+</span> 
  	geom_point() <span class="pl-k">+</span> 
  	scale_colour_gradient(<span class="pl-v">low</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gray90<span class="pl-pds">"</span></span>, <span class="pl-v">high</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  	stat_smooth(<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-smi">lm</span>) <span class="pl-k">+</span>
  	scale_x_log10() <span class="pl-k">+</span> 
  	scale_y_log10() <span class="pl-k">+</span> 
  	theme_classic() <span class="pl-k">+</span>
  	geom_vline(<span class="pl-v">xintercept</span> <span class="pl-k">=</span> <span class="pl-c1">1000</span>) <span class="pl-k">+</span>
  	geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> <span class="pl-c1">500</span>) <span class="pl-k">+</span>
  	facet_wrap(<span class="pl-k">~</span><span class="pl-smi">seq_folder</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nUMI_nGene_mitoRatio.png" rel="noopener noreferrer" target="_blank">
                      <img alt="mitoRatio_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nUMI_nGene_mitoRatio.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     There are samples that shows high-quality cells; high nUMI, high nGene, low number of cells with high mitoRatio and also there are some samples that would clearly benefit from filtering, as they have low-quality cells. We expect to see that dying cells to show high level of mitoRatio and low nUMI and nGene .
                    </p>
                    <p dir="auto">
                     Basically, it is not uncommon to observe cells with high numbers of UMIs and nGene, but also high mitoRatio. These cells may be stressed or damaged, but they could also represent a heterogeneous population of cells with distinct metabolic states.
                    </p>
                    <p dir="auto">
                     To investigate the potential cause of high mitochondrial expression ratios, it is important to examine the expression of specific mitochondrial genes and compare them to other genes in the cell. If the expression of mitochondrial genes is elevated relative to other genes, this could suggest mitochondrial dysfunction. Additionally, examining the expression of other stress or damage markers, such as heat shock proteins or cell cycle genes, can also provide insight into the health and state of the cell.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Filtering
                     </h3>
                     <a aria-label="Permalink: Filtering" class="anchor" href="#filtering" id="user-content-filtering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Cell-level filtering
                     </h4>
                     <a aria-label="Permalink: Cell-level filtering" class="anchor" href="#cell-level-filtering" id="user-content-cell-level-filtering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     -nUMI &gt; 1000
                    </p>
                    <p dir="auto">
                     -nGene &gt; 500
                    </p>
                    <p dir="auto">
                     -log10GenesPerUMI &gt; 0.8
                    </p>
                    <p dir="auto">
                     -mitoRatio &lt; 0.2
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Filter out low-quality cells using selected thresholds - these will change with the experiment
filtered_seurat &lt;- subset(merged_seurat, 
                          subset= nUMI &gt;= 1000 &amp;
                          nGene &gt;= 500 &amp;
                          nGene &lt;= 6000 &amp; 
                          log10GenesPerUMI &gt; 0.80 &amp; 
                          mitoRatio &lt; 0.10)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Filter out low-quality cells using selected thresholds - these will change with the experiment</span>
<span class="pl-smi">filtered_seurat</span> <span class="pl-k">&lt;-</span> subset(<span class="pl-smi">merged_seurat</span>, 
                          <span class="pl-v">subset</span><span class="pl-k">=</span> <span class="pl-smi">nUMI</span> <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">1000</span> <span class="pl-k">&amp;</span>
                          <span class="pl-smi">nGene</span> <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">500</span> <span class="pl-k">&amp;</span>
                          <span class="pl-smi">nGene</span> <span class="pl-k">&lt;</span><span class="pl-k">=</span> <span class="pl-c1">6000</span> <span class="pl-k">&amp;</span> 
                          <span class="pl-smi">log10GenesPerUMI</span> <span class="pl-k">&gt;</span> <span class="pl-c1">0.80</span> <span class="pl-k">&amp;</span> 
                          <span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0.10</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Gene-level filtering
                     </h4>
                     <a aria-label="Permalink: Gene-level filtering" class="anchor" href="#gene-level-filtering" id="user-content-gene-level-filtering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Keep only genes that are expressed in 100 or more cells (usually that is 10)
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Extract counts
counts &lt;- GetAssayData(object = filtered_seurat, slot = "counts")

# Output a logical matrix specifying for each gene whether or not there are more than zero counts per cell
nonzero &lt;- counts &gt; 0
# Sums all TRUE values and returns TRUE if more than 100 TRUE values per gene
keep_genes &lt;- Matrix::rowSums(nonzero) &gt;= 100

# Only keeping those genes expressed in more than 100 cells
filtered_counts &lt;- counts[keep_genes, ]
# Reassign to filtered Seurat object
filtered_seurat &lt;- CreateSeuratObject(filtered_counts, meta.data = filtered_seurat@meta.data)

# Create.RData object to load at any time
save(filtered_seurat, file="seurat_filtered.RData")' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Extract counts</span>
<span class="pl-smi">counts</span> <span class="pl-k">&lt;-</span> GetAssayData(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">filtered_seurat</span>, <span class="pl-v">slot</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>counts<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Output a logical matrix specifying for each gene whether or not there are more than zero counts per cell</span>
<span class="pl-smi">nonzero</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">counts</span> <span class="pl-k">&gt;</span> <span class="pl-c1">0</span>
<span class="pl-c"><span class="pl-c">#</span> Sums all TRUE values and returns TRUE if more than 100 TRUE values per gene</span>
<span class="pl-smi">keep_genes</span> <span class="pl-k">&lt;-</span> <span class="pl-e">Matrix</span><span class="pl-k">::</span>rowSums(<span class="pl-smi">nonzero</span>) <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">100</span>

<span class="pl-c"><span class="pl-c">#</span> Only keeping those genes expressed in more than 100 cells</span>
<span class="pl-smi">filtered_counts</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">counts</span>[<span class="pl-smi">keep_genes</span>, ]
<span class="pl-c"><span class="pl-c">#</span> Reassign to filtered Seurat object</span>
<span class="pl-smi">filtered_seurat</span> <span class="pl-k">&lt;-</span> CreateSeuratObject(<span class="pl-smi">filtered_counts</span>, <span class="pl-v">meta.data</span> <span class="pl-k">=</span> <span class="pl-smi">filtered_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)

<span class="pl-c"><span class="pl-c">#</span> Create.RData object to load at any time</span>
save(<span class="pl-smi">filtered_seurat</span>, <span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>seurat_filtered.RData<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Re-assess QC metrics
                     </h4>
                     <a aria-label="Permalink: Re-assess QC metrics" class="anchor" href="#re-assess-qc-metrics" id="user-content-re-assess-qc-metrics">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
# Save filtered subset to new metadata
metadata_clean &lt;- filtered_seurat@meta.data

# to see a drop in filtering cells:

met_before &lt;- data.frame(unclass(table(metadata$seq_folder)))
met_before$QCgroup &lt;- &quot;before&quot;
met_before$cell&lt;- rownames(met_before)
names(met_before)[1] &lt;- 'count'

met_after &lt;- data.frame(unclass(table(metadata_clean$seq_folder)))
met_after$QCgroup &lt;- &quot;after&quot;
met_after$cell&lt;- rownames(met_after)
names(met_after)[1] &lt;- 'count'
# count
cell_count &lt;- data.frame(rbind(met_before, met_after))

                                
# visualization :
cell_count %&gt;% ggplot(aes(x=cell, y=count, fill=QCgroup)) + 
  geom_bar(stat=&quot;identity&quot;, position=position_dodge()) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  theme(plot.title = element_text(hjust=0.5, face=&quot;bold&quot;)) +
  scale_fill_manual(values = c(&quot;#808080&quot;, &quot;#FFBF00&quot;)) +
  xlab(&quot;samples&quot;) +
  ggtitle(&quot;nCells count before and after QC&quot;)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Save filtered subset to new metadata</span>
<span class="pl-smi">metadata_clean</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">filtered_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>

<span class="pl-c"><span class="pl-c">#</span> to see a drop in filtering cells:</span>

<span class="pl-smi">met_before</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(unclass(table(<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">seq_folder</span>)))
<span class="pl-smi">met_before</span><span class="pl-k">$</span><span class="pl-smi">QCgroup</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>before<span class="pl-pds">"</span></span>
<span class="pl-smi">met_before</span><span class="pl-k">$</span><span class="pl-smi">cell</span><span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">met_before</span>)
names(<span class="pl-smi">met_before</span>)[<span class="pl-c1">1</span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>count<span class="pl-pds">'</span></span>

<span class="pl-smi">met_after</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(unclass(table(<span class="pl-smi">metadata_clean</span><span class="pl-k">$</span><span class="pl-smi">seq_folder</span>)))
<span class="pl-smi">met_after</span><span class="pl-k">$</span><span class="pl-smi">QCgroup</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>after<span class="pl-pds">"</span></span>
<span class="pl-smi">met_after</span><span class="pl-k">$</span><span class="pl-smi">cell</span><span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">met_after</span>)
names(<span class="pl-smi">met_after</span>)[<span class="pl-c1">1</span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">'</span>count<span class="pl-pds">'</span></span>
<span class="pl-c"><span class="pl-c">#</span> count</span>
<span class="pl-smi">cell_count</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(rbind(<span class="pl-smi">met_before</span>, <span class="pl-smi">met_after</span>))

                                
<span class="pl-c"><span class="pl-c">#</span> visualization :</span>
<span class="pl-smi">cell_count</span> %<span class="pl-k">&gt;</span>% ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">cell</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">count</span>, <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">QCgroup</span>)) <span class="pl-k">+</span> 
  geom_bar(<span class="pl-v">stat</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>identity<span class="pl-pds">"</span></span>, <span class="pl-v">position</span><span class="pl-k">=</span>position_dodge()) <span class="pl-k">+</span>
  theme_classic() <span class="pl-k">+</span>
  theme(<span class="pl-v">axis.text.x</span> <span class="pl-k">=</span> element_text(<span class="pl-v">angle</span> <span class="pl-k">=</span> <span class="pl-c1">45</span>, <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, <span class="pl-v">hjust</span><span class="pl-k">=</span><span class="pl-c1">1</span>)) <span class="pl-k">+</span>
  theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">hjust</span><span class="pl-k">=</span><span class="pl-c1">0.5</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
  scale_fill_manual(<span class="pl-v">values</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>#808080<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>#FFBF00<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
  xlab(<span class="pl-s"><span class="pl-pds">"</span>samples<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  ggtitle(<span class="pl-s"><span class="pl-pds">"</span>nCells count before and after QC<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/before_after_cell_count.png" rel="noopener noreferrer" target="_blank">
                      <img alt="cell_countbefore_after_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/before_after_cell_count.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Visualize the correlation between genes detected and the number of UMIs and determine whether the strong presence of cells with low numbers of genes/UMIs

metadata_clean %&gt;% 
  	ggplot(aes(x=nUMI, y=nGene, color=mitoRatio)) + 
  	geom_point() + 
  	scale_colour_gradient(low = "gray90", high = "black") +
  	stat_smooth(method=lm) +
  	scale_x_log10() + 
  	scale_y_log10() + 
  	theme_classic() +
  	geom_vline(xintercept = 1000) +
  	geom_hline(yintercept = 500) +
  	facet_wrap(~seq_folder)
  ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Visualize the correlation between genes detected and the number of UMIs and determine whether the strong presence of cells with low numbers of genes/UMIs</span>

<span class="pl-smi">metadata_clean</span> %<span class="pl-k">&gt;</span>% 
  	ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">nUMI</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">nGene</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">mitoRatio</span>)) <span class="pl-k">+</span> 
  	geom_point() <span class="pl-k">+</span> 
  	scale_colour_gradient(<span class="pl-v">low</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gray90<span class="pl-pds">"</span></span>, <span class="pl-v">high</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
  	stat_smooth(<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-smi">lm</span>) <span class="pl-k">+</span>
  	scale_x_log10() <span class="pl-k">+</span> 
  	scale_y_log10() <span class="pl-k">+</span> 
  	theme_classic() <span class="pl-k">+</span>
  	geom_vline(<span class="pl-v">xintercept</span> <span class="pl-k">=</span> <span class="pl-c1">1000</span>) <span class="pl-k">+</span>
  	geom_hline(<span class="pl-v">yintercept</span> <span class="pl-k">=</span> <span class="pl-c1">500</span>) <span class="pl-k">+</span>
  	facet_wrap(<span class="pl-k">~</span><span class="pl-smi">seq_folder</span>)
  </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/nUMI_nGene_mitoRatio_after.png" rel="noopener noreferrer" target="_blank">
                      <img alt="nUMI_nGene_mitoRatio_plot" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/nUMI_nGene_mitoRatio_after.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      6) Normalization, regressing out unwanted variation and clustering
                     </h2>
                     <a aria-label="Permalink: 6) Normalization, regressing out unwanted variation and clustering" class="anchor" href="#6-normalization-regressing-out-unwanted-variation-and-clustering" id="user-content-6-normalization-regressing-out-unwanted-variation-and-clustering">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The ultimate goal is to define clusters of cells and identify cell types in the samples. To achieve this, there are several steps:
                    </p>
                    <p dir="auto">
                     1-Identify unwanted variability by exploring data and covariates such as cell cycle and mitochondrial gene expression.
Both biological sources of variation (e.g. effect of cell cycle on transcriptome) and technical sources should be explored and accounted for. In an early version of Seurat, one needs to normalize data, find variable features, and then scale data while setting a variable like mitochondrial contamination or cell cycle stage to be regressed out. So here is the code for doing these steps to mitigate cell cycle stage effects on the dataset, however, a newer function in Seurat has automated all of these steps (
                     <code>
                      SCTtansform()
                     </code>
                     ).
                    </p>
                    <p dir="auto">
                     2-Normalize and remove unwanted variability using Seurat's
                     <code>
                      sctransform
                     </code>
                     function. The normalization step is necessary to make expression counts comparable across genes and/or samples. The counts of mapped reads for each gene are proportional to the expression of RNA (âinterestingâ) in addition to many other factors (âuninterestingâ such as sequencing depth and gene length). Normalization is the process of adjusting raw count values to account for the âuninterestingâ factors.
For simplicity, normalization is assumed as two a two-step process: scaling and transforming.
In scaling the goal is to multiply each UMI count by a cell-specific factor to get all cells to have the same UMI counts. For transformation, simple approaches like log-transformation showed to be not that useful, especially in the case of genes with high expression but showed decent performance for low/intermediate expressed genes. So we cannot treat all genes the same.
The proposed solution for data transformation is Pearson residuals (implemented in Seurat's
                     <code>
                      SCTransform
                     </code>
                     function), which applies a gene-specific weight to each measurement based on the evidence of non-uniform expression across cells. This weight is higher for genes expressed in a smaller fraction of cells, making it useful for detecting rare cell populations. The weight takes into account not just the expression level but also the distribution of expression.
                    </p>
                    <p dir="auto">
                     3- Integrate data using Seurat's method to compare cell type expression between groups.
                    </p>
                    <p dir="auto">
                     4-Cluster cells based on similarity of gene expression profiles using Seurat's PCA scores.
                    </p>
                    <p dir="auto">
                     5-Evaluate cluster quality by checking for sources of uninteresting variation, and principal component influence, and exploring cell type identities using known markers.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Exploring sources of unwanted variation
                     </h3>
                     <a aria-label="Permalink: Exploring sources of unwanted variation" class="anchor" href="#exploring-sources-of-unwanted-variation" id="user-content-exploring-sources-of-unwanted-variation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Evaluating effects of cell cycle and mitochondrial expression
                     </h4>
                     <a aria-label="Permalink: Evaluating effects of cell cycle and mitochondrial expression" class="anchor" href="#evaluating-effects-of-cell-cycle-and-mitochondrial-expression" id="user-content-evaluating-effects-of-cell-cycle-and-mitochondrial-expression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We will store the cells for cell cycle genes, and then determine whether cell cycle is a major source of variation in our dataset using PCA.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Normalize the counts
# This normalization method is solely for the purpose of exploring the sources of variation in our data.
seurat_phase &lt;- NormalizeData(filtered_seurat, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)

# Load cell cycle markers
load(&quot;C:/Users/qaedi/OneDrive - Queen's University/Documents/scRNA/cycle.rda&quot;)

# Score cells for cell cycle
seurat_phase &lt;- CellCycleScoring(seurat_phase, 
                                 g2m.features = g2m_genes, 
                                 s.features = s_genes)

# View cell cycle scores and phases assigned to cells                                 
#View(seurat_phase@meta.data) 
table(seurat_phase$Phase)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Normalize the counts</span>
<span class="pl-c"><span class="pl-c">#</span> This normalization method is solely for the purpose of exploring the sources of variation in our data.</span>
<span class="pl-smi">seurat_phase</span> <span class="pl-k">&lt;-</span> NormalizeData(<span class="pl-smi">filtered_seurat</span>, <span class="pl-v">normalization.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LogNormalize<span class="pl-pds">"</span></span>, <span class="pl-v">scale.factor</span> <span class="pl-k">=</span> <span class="pl-c1">10000</span>)

<span class="pl-c"><span class="pl-c">#</span> Load cell cycle markers</span>
load(<span class="pl-s"><span class="pl-pds">"</span>C:/Users/qaedi/OneDrive - Queen's University/Documents/scRNA/cycle.rda<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Score cells for cell cycle</span>
<span class="pl-smi">seurat_phase</span> <span class="pl-k">&lt;-</span> CellCycleScoring(<span class="pl-smi">seurat_phase</span>, 
                                 <span class="pl-v">g2m.features</span> <span class="pl-k">=</span> <span class="pl-smi">g2m_genes</span>, 
                                 <span class="pl-v">s.features</span> <span class="pl-k">=</span> <span class="pl-smi">s_genes</span>)

<span class="pl-c"><span class="pl-c">#</span> View cell cycle scores and phases assigned to cells                                 </span>
<span class="pl-c"><span class="pl-c">#</span>View(seurat_phase@meta.data) </span>
table(<span class="pl-smi">seurat_phase</span><span class="pl-k">$</span><span class="pl-smi">Phase</span>)</pre>
                    </div>
                    <p dir="auto">
                     Cells in different cell cycle stages:
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        G1
                       </th>
                       <th>
                        G2M
                       </th>
                       <th>
                        S
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        54106
                       </td>
                       <td>
                        12680
                       </td>
                       <td>
                        25441
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     So most of the cells are in G1 and then S, which make sense.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Identify the most variable genes and scale them
seurat_phase &lt;- FindVariableFeatures(seurat_phase, 
                     selection.method = "vst",
                     nfeatures = 2000, 
                     verbose = TRUE)
                     
# Identify the 10 most highly variable genes
top10 &lt;- head(VariableFeatures(seurat_phase), 10)

# plot variable features with and without labels
plot1 &lt;- VariableFeaturePlot(seurat_phase)
plot2 &lt;- LabelPoints(plot = plot1, points = top10, repel = TRUE)
plot1 + plot2' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Identify the most variable genes and scale them</span>
<span class="pl-smi">seurat_phase</span> <span class="pl-k">&lt;-</span> FindVariableFeatures(<span class="pl-smi">seurat_phase</span>, 
                     <span class="pl-v">selection.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vst<span class="pl-pds">"</span></span>,
                     <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">2000</span>, 
                     <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
                     
<span class="pl-c"><span class="pl-c">#</span> Identify the 10 most highly variable genes</span>
<span class="pl-smi">top10</span> <span class="pl-k">&lt;-</span> head(VariableFeatures(<span class="pl-smi">seurat_phase</span>), <span class="pl-c1">10</span>)

<span class="pl-c"><span class="pl-c">#</span> plot variable features with and without labels</span>
<span class="pl-smi">plot1</span> <span class="pl-k">&lt;-</span> VariableFeaturePlot(<span class="pl-smi">seurat_phase</span>)
<span class="pl-smi">plot2</span> <span class="pl-k">&lt;-</span> LabelPoints(<span class="pl-v">plot</span> <span class="pl-k">=</span> <span class="pl-smi">plot1</span>, <span class="pl-v">points</span> <span class="pl-k">=</span> <span class="pl-smi">top10</span>, <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-smi">plot1</span> <span class="pl-k">+</span> <span class="pl-smi">plot2</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/variable_features.png" rel="noopener noreferrer" target="_blank">
                      <img alt="variable_feature" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/variable_features.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Check quartile values for mitoRatio, we will use this variable later to mitigate the unwanted sources of variation in the dataset
summary(seurat_phase@meta.data$mitoRatio)

# Turn mitoRatio into categorical factor vector based on quartile values
seurat_phase@meta.data$mitoFr &lt;- cut(seurat_phase@meta.data$mitoRatio, 
                   breaks=c(-Inf, 0.015, 0.025, 0.045, Inf), 
                   labels=c("Low","Medium","Medium high", "High"))



# Scale the counts
# This step is essential for PCA , clustering and heatmap generation
seurat_phase &lt;- ScaleData(seurat_phase)
#saveRDS(seurat_phase, "seurat_phase.rds")

# Perform PCA
seurat_phase &lt;- RunPCA(seurat_phase)

# Plot the PCA colored by cell cycle phase
no_split &lt;- DimPlot(seurat_phase,
        reduction = "pca",
        group.by= "Phase")
        
with_split &lt;- DimPlot(seurat_phase,
        reduction = "pca",
        group.by= "Phase",
        split.by= "Phase")
        
no_split + with_split
' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Check quartile values for mitoRatio, we will use this variable later to mitigate the unwanted sources of variation in the dataset</span>
summary(<span class="pl-smi">seurat_phase</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span>)

<span class="pl-c"><span class="pl-c">#</span> Turn mitoRatio into categorical factor vector based on quartile values</span>
<span class="pl-smi">seurat_phase</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">mitoFr</span> <span class="pl-k">&lt;-</span> cut(<span class="pl-smi">seurat_phase</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span>, 
                   <span class="pl-v">breaks</span><span class="pl-k">=</span>c(<span class="pl-k">-</span><span class="pl-c1">Inf</span>, <span class="pl-c1">0.015</span>, <span class="pl-c1">0.025</span>, <span class="pl-c1">0.045</span>, <span class="pl-c1">Inf</span>), 
                   <span class="pl-v">labels</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>Low<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Medium<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Medium high<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>High<span class="pl-pds">"</span></span>))



<span class="pl-c"><span class="pl-c">#</span> Scale the counts</span>
<span class="pl-c"><span class="pl-c">#</span> This step is essential for PCA , clustering and heatmap generation</span>
<span class="pl-smi">seurat_phase</span> <span class="pl-k">&lt;-</span> ScaleData(<span class="pl-smi">seurat_phase</span>)
<span class="pl-c"><span class="pl-c">#</span>saveRDS(seurat_phase, "seurat_phase.rds")</span>

<span class="pl-c"><span class="pl-c">#</span> Perform PCA</span>
<span class="pl-smi">seurat_phase</span> <span class="pl-k">&lt;-</span> RunPCA(<span class="pl-smi">seurat_phase</span>)

<span class="pl-c"><span class="pl-c">#</span> Plot the PCA colored by cell cycle phase</span>
<span class="pl-smi">no_split</span> <span class="pl-k">&lt;-</span> DimPlot(<span class="pl-smi">seurat_phase</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>,
        <span class="pl-v">group.by</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Phase<span class="pl-pds">"</span></span>)
        
<span class="pl-smi">with_split</span> <span class="pl-k">&lt;-</span> DimPlot(<span class="pl-smi">seurat_phase</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>,
        <span class="pl-v">group.by</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Phase<span class="pl-pds">"</span></span>,
        <span class="pl-v">split.by</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Phase<span class="pl-pds">"</span></span>)
        
<span class="pl-smi">no_split</span> <span class="pl-k">+</span> <span class="pl-smi">with_split</span>
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cell_cycle_PCA.png" rel="noopener noreferrer" target="_blank">
                      <img alt="cell_cycle_effect" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cell_cycle_PCA.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     For mitochondrial expression:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Plot the PCA colored by mitochondrial expression
no_split &lt;- DimPlot(seurat_phase,
        reduction = "pca",
        group.by= "mitoFr")
        
with_split &lt;- DimPlot(seurat_phase,
        reduction = "pca",
        group.by= "mitoFr",
        split.by= "mitoFr")
        
no_split + with_split' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Plot the PCA colored by mitochondrial expression</span>
<span class="pl-smi">no_split</span> <span class="pl-k">&lt;-</span> DimPlot(<span class="pl-smi">seurat_phase</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>,
        <span class="pl-v">group.by</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>mitoFr<span class="pl-pds">"</span></span>)
        
<span class="pl-smi">with_split</span> <span class="pl-k">&lt;-</span> DimPlot(<span class="pl-smi">seurat_phase</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>,
        <span class="pl-v">group.by</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>mitoFr<span class="pl-pds">"</span></span>,
        <span class="pl-v">split.by</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>mitoFr<span class="pl-pds">"</span></span>)
        
<span class="pl-smi">no_split</span> <span class="pl-k">+</span> <span class="pl-smi">with_split</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/mitoExpression_effect.png" rel="noopener noreferrer" target="_blank">
                      <img alt="mito_expression_effect" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/mitoExpression_effect.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Based on the above plots, we can see that cells are scattered regardless of their cell cycle phase and mitochondrial genes expression level. So there is no need to regress out the effect of cell cycle and mitochondrial expression in this dataset.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      SCTransform
                     </h3>
                     <a aria-label="Permalink: SCTransform" class="anchor" href="#sctransform" id="user-content-sctransform">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     This function is useful for  normalization and regressing sources of unwanted variation at the same time. The method constructs a generalized linear model (GLM) for each gene, using UMI counts as the response variable and sequencing depth as the explanatory variable. To handle the fact that different genes have different levels of expression, information is pooled across genes with similar abundances, resulting in more accurate parameter estimates.
                    </p>
                    <p dir="auto">
                     This regularization process yields residuals, which represent effectively normalized data values that are no longer correlated with sequencing depth.
                    </p>
                    <p dir="auto">
                     This method is a more accurate method of normalizing, estimating the variance of the raw filtered data, and identifying the most variable genes. In practice
                     <code>
                      SCTransform
                     </code>
                     single command replaces
                     <code>
                      NormalizeData()
                     </code>
                     ,
                     <code>
                      ScaleData()
                     </code>
                     , and
                     <code>
                      FindVariableFeatures()
                     </code>
                     . Since we have two groups of samples we will run SCTransform on each group after doing "integration".
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Integration
                     </h4>
                     <a aria-label="Permalink: Integration" class="anchor" href="#integration" id="user-content-integration">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To improve clustering and downstream analyses, it can be beneficial to integrate or align samples across groups using shared highly variable genes. If cells cluster by sample, condition, batch, dataset, or modalities(scRNA, scATAC-seq), integration can help to remove these unwanted sources of variation. For example, if we want to integrate normal samples together and BLCA samples together, we should keep each sample as a separate object and transform them accordingly for integration. This is necessary to ensure that the samples are properly aligned and that downstream analyses are meaningful.
If cell types are present in one dataset, but not the other, then the cells will still appear as a separate sample-specific cluster.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# adjust the limit for allowable object sizes within R
options(future.globals.maxSize = 4000 * 1024^2)

# Split seurat object by group
split_seurat &lt;- SplitObject(seurat_phase, split.by = "sample")

# then normalize by SCTansform
for (i in 1:length(split_seurat)) {
    split_seurat[[i]] &lt;- SCTransform(split_seurat[[i]], vars.to.regress = c("mitoRatio", "S.Score", "G2M.Score"))
    }

# to see what the component of the object are. 

split_seurat    
#$Normal
#An object of class Seurat
#47302 features across 21519 samples within 2 assays
#Active assay: SCT (23608 features, 3000 variable features)
# 1 other assay present: RNA
# 1 dimensional reduction calculated: pca

#$BLCA
#An object of class Seurat
#47388 features across 70708 samples within 2 assays
#Active assay: SCT (23694 features, 3000 variable features)
# 1 other assay present: RNA
# 1 dimensional reduction calculated: pca

    
# Select the most variable features to use for integration
integ_features &lt;- SelectIntegrationFeatures(object.list = split_seurat, 
                                            nfeatures = 3000) 


# Prepare the SCT list object for integration
split_seurat &lt;- PrepSCTIntegration(object.list = split_seurat, 
                                   anchor.features = integ_features)
                                   
# Find best buddies (using canonical correlation analysis or CCA) - can take a while to run
integ_anchors &lt;- FindIntegrationAnchors(object.list = split_seurat, 
                                        normalization.method = "SCT", 
                                        anchor.features = integ_features)
# Integrate across conditions
seurat_integrated &lt;- IntegrateData(anchorset = integ_anchors, 
                                   normalization.method = "SCT")
                                   
# Check assays in the object:
split_seurat$Normal@assays
                                   ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> adjust the limit for allowable object sizes within R</span>
options(<span class="pl-v">future.globals.maxSize</span> <span class="pl-k">=</span> <span class="pl-c1">4000</span> <span class="pl-k">*</span> <span class="pl-c1">1024</span><span class="pl-k">^</span><span class="pl-c1">2</span>)

<span class="pl-c"><span class="pl-c">#</span> Split seurat object by group</span>
<span class="pl-smi">split_seurat</span> <span class="pl-k">&lt;-</span> SplitObject(<span class="pl-smi">seurat_phase</span>, <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>sample<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> then normalize by SCTansform</span>
<span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>length(<span class="pl-smi">split_seurat</span>)) {
    <span class="pl-smi">split_seurat</span>[[<span class="pl-smi">i</span>]] <span class="pl-k">&lt;-</span> SCTransform(<span class="pl-smi">split_seurat</span>[[<span class="pl-smi">i</span>]], <span class="pl-v">vars.to.regress</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>mitoRatio<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>S.Score<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>G2M.Score<span class="pl-pds">"</span></span>))
    }

<span class="pl-c"><span class="pl-c">#</span> to see what the component of the object are. </span>

<span class="pl-smi">split_seurat</span>    
<span class="pl-c"><span class="pl-c">#</span>$Normal</span>
<span class="pl-c"><span class="pl-c">#</span>An object of class Seurat</span>
<span class="pl-c"><span class="pl-c">#</span>47302 features across 21519 samples within 2 assays</span>
<span class="pl-c"><span class="pl-c">#</span>Active assay: SCT (23608 features, 3000 variable features)</span>
<span class="pl-c"><span class="pl-c">#</span> 1 other assay present: RNA</span>
<span class="pl-c"><span class="pl-c">#</span> 1 dimensional reduction calculated: pca</span>

<span class="pl-c"><span class="pl-c">#</span>$BLCA</span>
<span class="pl-c"><span class="pl-c">#</span>An object of class Seurat</span>
<span class="pl-c"><span class="pl-c">#</span>47388 features across 70708 samples within 2 assays</span>
<span class="pl-c"><span class="pl-c">#</span>Active assay: SCT (23694 features, 3000 variable features)</span>
<span class="pl-c"><span class="pl-c">#</span> 1 other assay present: RNA</span>
<span class="pl-c"><span class="pl-c">#</span> 1 dimensional reduction calculated: pca</span>

    
<span class="pl-c"><span class="pl-c">#</span> Select the most variable features to use for integration</span>
<span class="pl-smi">integ_features</span> <span class="pl-k">&lt;-</span> SelectIntegrationFeatures(<span class="pl-v">object.list</span> <span class="pl-k">=</span> <span class="pl-smi">split_seurat</span>, 
                                            <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">3000</span>) 


<span class="pl-c"><span class="pl-c">#</span> Prepare the SCT list object for integration</span>
<span class="pl-smi">split_seurat</span> <span class="pl-k">&lt;-</span> PrepSCTIntegration(<span class="pl-v">object.list</span> <span class="pl-k">=</span> <span class="pl-smi">split_seurat</span>, 
                                   <span class="pl-v">anchor.features</span> <span class="pl-k">=</span> <span class="pl-smi">integ_features</span>)
                                   
<span class="pl-c"><span class="pl-c">#</span> Find best buddies (using canonical correlation analysis or CCA) - can take a while to run</span>
<span class="pl-smi">integ_anchors</span> <span class="pl-k">&lt;-</span> FindIntegrationAnchors(<span class="pl-v">object.list</span> <span class="pl-k">=</span> <span class="pl-smi">split_seurat</span>, 
                                        <span class="pl-v">normalization.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, 
                                        <span class="pl-v">anchor.features</span> <span class="pl-k">=</span> <span class="pl-smi">integ_features</span>)
<span class="pl-c"><span class="pl-c">#</span> Integrate across conditions</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> IntegrateData(<span class="pl-v">anchorset</span> <span class="pl-k">=</span> <span class="pl-smi">integ_anchors</span>, 
                                   <span class="pl-v">normalization.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>)
                                   
<span class="pl-c"><span class="pl-c">#</span> Check assays in the object:</span>
<span class="pl-smi">split_seurat</span><span class="pl-k">$</span><span class="pl-smi">Normal</span><span class="pl-k">@</span><span class="pl-smi">assays</span>
                                   </pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Integration check
                     </h3>
                     <a aria-label="Permalink: Integration check" class="anchor" href="#integration-check" id="user-content-integration-check">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     After normalization and integration, we can proceed to PCA and UMAP/t-SNE to see effect of integration.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Run PCA
seurat_integrated &lt;- RunPCA(object = seurat_integrated, verbose = TRUE)
# PC_ 1
# Positive:  IGFBP7, MGP, SPARC, VIM, SPARCL1, CALD1, IFITM3, A2M, COL4A1, COL1A2
#            COL4A2, TAGLN, COL6A2, BGN, TCF4, COL3A1, CCL2, MT2A, NNMT, COL1A1
#            MYL9, LGALS1, DCN, SELENOM, CCN1, ADAMTS9, CCN2, TIMP1, IGFBP4, LUM
# Negative:  KRT19, KRT13, CD52, S100P, RPS19, CLDN4, FXYD3, PSCA, TACSTD2, CSTB
#            PTPRC, KRT7, RPS18, KRT17, LYPD3, RPL41, CD3D, CCL5, RPS27, SFN
#            FABP5, SPINK1, GDF15, AQP3, HCST, TRAC, SNCG, ADIRF, ELF3, TRBC2
# PC_ 2
# Positive:  PLVAP, CALCRL, PCAT19, MCTP1, AQP1, VWF, CD74, PECAM1, RAMP2, LDB2
#            RAMP3, FLT1, ZNF385D, TCF4, CLDN5, ACKR1, HSPG2, SPARCL1, EMCN, ADGRL4
#            SLCO2A1, HLA-DRA, SELE, DOCK4, ECSCR, CCL14, HLA-DRB1, ERG, PCDH17, GNG11
# Negative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN, LUM, BGN, COL6A2, C1R, SOD3
#            MFAP4, RARRES2, C1S, MYL9, TPM2, PRKG1, CRYAB, ACTA2, COL6A3, LGALS1
#            CALD1, COL6A1, TIMP1, SERPINF1, PCOLCE, AEBP1, C11orf96, MEG3, FBLN1, GPC6
# PC_ 3
# Positive:  CCL5, CD52, B2M, PTPRC, IL32, SRGN, CD3D, HSPA1A, NKG7, GZMA
#            TRAC, ARHGAP15, HCST, RGS1, CXCR4, SAMSN1, CCL4, CD7, RGS2, CORO1A
#            CD2, CRIP1, CST7, CD69, STAT4, FYN, TMSB4X, DNAJB1, PTPN22, CD3E
# Negative:  ADIRF, SPINK1, IFI27, CSTB, S100P, KRT19, FXYD3, CLDN4, CCT2, CCND1
#            PSCA, S100A6, SNCG, UCA1, KRT17, YEATS4, KRT7, KRT13, TACSTD2, GDF15
#            RAB3IP, KRT18, HES1, GAPDH, S100A2, PLVAP, RAMP2, S100A14, FABP5, TM4SF1
# PC_ 4
# Positive:  IL32, CD3D, CCL5, CRIP1, TRAC, FYN, COL4A1, COL4A2, CD2, GZMA
#            CD7, CALD1, IGFBP7, TRBC2, MCAM, NDUFA4L2, CD3E, NKG7, COL18A1, RGS5
#            SKAP1, MYL9, ITGA1, CYTOR, CD247, SPARC, TRBC1, MAP1B, ACTA2, CACNA1C
# Negative:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1, FCER1G, HLA-DPA1, AIF1, HLA-DQA1, FTL
#            LYZ, IFI30, C1QA, HLA-DQB1, C1QB, C1QC, MS4A6A, LST1, APOE, CD14
#            TMEM176B, S100A9, HLA-DMA, FCGR2A, CXCL8, SPI1, FTH1, CD68, PSAP, CST3
# PC_ 5
# Positive:  LUM, MMP2, PTGDS, EMP1, DCN, RARRES2, KRT13, SERPINF1, CXCL1, CXCL8
#            LSAMP, COL8A1, TM4SF1, AREG, C1S, PDPN, CFD, APOD, SOD2, CTSK
#            CLMP, KRT17, MFAP4, LYPD3, VCAN, PLAUR, TSHZ2, PLAT, PDGFRA, C1R
# Negative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9, TAGLN, FRZB, CRIP1, CALD1, GJA4
#            MCAM, COL18A1, MYH11, TYROBP, TPPP3, COX4I2, PRKG1, IGFBP7, COL4A1, COL4A2
#            CDH6, HIGD1B, AIF1, TPM2, PTP4A3, FCER1G, WFDC1, HEYL, MYLK, HLA-DRA



# Plot PCA
png(filename = "PCA_integrated.png", width = 16, height = 8.135, units = "in", res = 300)
PCAPlot(seurat_integrated,
        split.by = "sample")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Run PCA</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> RunPCA(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-c"><span class="pl-c">#</span> PC_ 1</span>
<span class="pl-c"><span class="pl-c">#</span> Positive:  IGFBP7, MGP, SPARC, VIM, SPARCL1, CALD1, IFITM3, A2M, COL4A1, COL1A2</span>
<span class="pl-c"><span class="pl-c">#</span>            COL4A2, TAGLN, COL6A2, BGN, TCF4, COL3A1, CCL2, MT2A, NNMT, COL1A1</span>
<span class="pl-c"><span class="pl-c">#</span>            MYL9, LGALS1, DCN, SELENOM, CCN1, ADAMTS9, CCN2, TIMP1, IGFBP4, LUM</span>
<span class="pl-c"><span class="pl-c">#</span> Negative:  KRT19, KRT13, CD52, S100P, RPS19, CLDN4, FXYD3, PSCA, TACSTD2, CSTB</span>
<span class="pl-c"><span class="pl-c">#</span>            PTPRC, KRT7, RPS18, KRT17, LYPD3, RPL41, CD3D, CCL5, RPS27, SFN</span>
<span class="pl-c"><span class="pl-c">#</span>            FABP5, SPINK1, GDF15, AQP3, HCST, TRAC, SNCG, ADIRF, ELF3, TRBC2</span>
<span class="pl-c"><span class="pl-c">#</span> PC_ 2</span>
<span class="pl-c"><span class="pl-c">#</span> Positive:  PLVAP, CALCRL, PCAT19, MCTP1, AQP1, VWF, CD74, PECAM1, RAMP2, LDB2</span>
<span class="pl-c"><span class="pl-c">#</span>            RAMP3, FLT1, ZNF385D, TCF4, CLDN5, ACKR1, HSPG2, SPARCL1, EMCN, ADGRL4</span>
<span class="pl-c"><span class="pl-c">#</span>            SLCO2A1, HLA-DRA, SELE, DOCK4, ECSCR, CCL14, HLA-DRB1, ERG, PCDH17, GNG11</span>
<span class="pl-c"><span class="pl-c">#</span> Negative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN, LUM, BGN, COL6A2, C1R, SOD3</span>
<span class="pl-c"><span class="pl-c">#</span>            MFAP4, RARRES2, C1S, MYL9, TPM2, PRKG1, CRYAB, ACTA2, COL6A3, LGALS1</span>
<span class="pl-c"><span class="pl-c">#</span>            CALD1, COL6A1, TIMP1, SERPINF1, PCOLCE, AEBP1, C11orf96, MEG3, FBLN1, GPC6</span>
<span class="pl-c"><span class="pl-c">#</span> PC_ 3</span>
<span class="pl-c"><span class="pl-c">#</span> Positive:  CCL5, CD52, B2M, PTPRC, IL32, SRGN, CD3D, HSPA1A, NKG7, GZMA</span>
<span class="pl-c"><span class="pl-c">#</span>            TRAC, ARHGAP15, HCST, RGS1, CXCR4, SAMSN1, CCL4, CD7, RGS2, CORO1A</span>
<span class="pl-c"><span class="pl-c">#</span>            CD2, CRIP1, CST7, CD69, STAT4, FYN, TMSB4X, DNAJB1, PTPN22, CD3E</span>
<span class="pl-c"><span class="pl-c">#</span> Negative:  ADIRF, SPINK1, IFI27, CSTB, S100P, KRT19, FXYD3, CLDN4, CCT2, CCND1</span>
<span class="pl-c"><span class="pl-c">#</span>            PSCA, S100A6, SNCG, UCA1, KRT17, YEATS4, KRT7, KRT13, TACSTD2, GDF15</span>
<span class="pl-c"><span class="pl-c">#</span>            RAB3IP, KRT18, HES1, GAPDH, S100A2, PLVAP, RAMP2, S100A14, FABP5, TM4SF1</span>
<span class="pl-c"><span class="pl-c">#</span> PC_ 4</span>
<span class="pl-c"><span class="pl-c">#</span> Positive:  IL32, CD3D, CCL5, CRIP1, TRAC, FYN, COL4A1, COL4A2, CD2, GZMA</span>
<span class="pl-c"><span class="pl-c">#</span>            CD7, CALD1, IGFBP7, TRBC2, MCAM, NDUFA4L2, CD3E, NKG7, COL18A1, RGS5</span>
<span class="pl-c"><span class="pl-c">#</span>            SKAP1, MYL9, ITGA1, CYTOR, CD247, SPARC, TRBC1, MAP1B, ACTA2, CACNA1C</span>
<span class="pl-c"><span class="pl-c">#</span> Negative:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1, FCER1G, HLA-DPA1, AIF1, HLA-DQA1, FTL</span>
<span class="pl-c"><span class="pl-c">#</span>            LYZ, IFI30, C1QA, HLA-DQB1, C1QB, C1QC, MS4A6A, LST1, APOE, CD14</span>
<span class="pl-c"><span class="pl-c">#</span>            TMEM176B, S100A9, HLA-DMA, FCGR2A, CXCL8, SPI1, FTH1, CD68, PSAP, CST3</span>
<span class="pl-c"><span class="pl-c">#</span> PC_ 5</span>
<span class="pl-c"><span class="pl-c">#</span> Positive:  LUM, MMP2, PTGDS, EMP1, DCN, RARRES2, KRT13, SERPINF1, CXCL1, CXCL8</span>
<span class="pl-c"><span class="pl-c">#</span>            LSAMP, COL8A1, TM4SF1, AREG, C1S, PDPN, CFD, APOD, SOD2, CTSK</span>
<span class="pl-c"><span class="pl-c">#</span>            CLMP, KRT17, MFAP4, LYPD3, VCAN, PLAUR, TSHZ2, PLAT, PDGFRA, C1R</span>
<span class="pl-c"><span class="pl-c">#</span> Negative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9, TAGLN, FRZB, CRIP1, CALD1, GJA4</span>
<span class="pl-c"><span class="pl-c">#</span>            MCAM, COL18A1, MYH11, TYROBP, TPPP3, COX4I2, PRKG1, IGFBP7, COL4A1, COL4A2</span>
<span class="pl-c"><span class="pl-c">#</span>            CDH6, HIGD1B, AIF1, TPM2, PTP4A3, FCER1G, WFDC1, HEYL, MYLK, HLA-DRA</span>



<span class="pl-c"><span class="pl-c">#</span> Plot PCA</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>PCA_integrated.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
PCAPlot(<span class="pl-smi">seurat_integrated</span>,
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>sample<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/PCA_integrated.png" rel="noopener noreferrer" target="_blank">
                      <img alt="plot-PCA" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/PCA_integrated.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Run UMAP
seurat_integrated &lt;- RunUMAP(seurat_integrated, 
                             dims = 1:40,
                             reduction = "pca",
                             verbose = TRUE)

# Plot UMAP 
png(filename = "UMAP_integrated.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated, split.by = "sample")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Run UMAP</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> RunUMAP(<span class="pl-smi">seurat_integrated</span>, 
                             <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">40</span>,
                             <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>,
                             <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span> Plot UMAP </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>UMAP_integrated.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>, <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>sample<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/UMAP_integrated.png" rel="noopener noreferrer" target="_blank">
                      <img alt="plot-UMAP" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/UMAP_integrated.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     For the future: considering tumor grade and using Harmony for integration
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Clustering cells based on top PCs (metagenes)
                     </h3>
                     <a aria-label="Permalink: Clustering cells based on top PCs (metagenes)" class="anchor" href="#clustering-cells-based-on-top-pcs-metagenes" id="user-content-clustering-cells-based-on-top-pcs-metagenes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Identify significant PCs
                     </h4>
                     <a aria-label="Permalink: Identify significant PCs" class="anchor" href="#identify-significant-pcs" id="user-content-identify-significant-pcs">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For new methods like SCTransform it is not needed to calculate the number of PCs for clustering. However, older methods could not efficiently remove technical biases, so using them it was necessary to have some idea about the number of PCs that can capture most of the information in the dataset.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Explore the heatmap of PCs
png(filename = "heatmap_integrated_2.png", width = 16, height = 8.135, units = "in", res = 300)
DimHeatmap(seurat_integrated, 
           dims = 1:9, 
           cells = 500, 
           balanced = TRUE)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Explore the heatmap of PCs</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>heatmap_integrated_2.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimHeatmap(<span class="pl-smi">seurat_integrated</span>, 
           <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">9</span>, 
           <span class="pl-v">cells</span> <span class="pl-k">=</span> <span class="pl-c1">500</span>, 
           <span class="pl-v">balanced</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/heatmap_integrated_2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="plot-PCA_2" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/heatmap_integrated_2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Printing out the most variable genes driving PCs
print(x = seurat_integrated[["pca"]], 
      dims = 1:10, 
      nfeatures = 5)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Printing out the most variable genes driving PCs</span>
print(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>[[<span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>]], 
      <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>, 
      <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="PC_ 1
Positive:  KRT19, CD52, KRT13, S100P, RPS19
Negative:  IGFBP7, MGP, SPARC, VIM, SPARCL1
PC_ 2
Positive:  PLVAP, PCAT19, CALCRL, MCTP1, AQP1
Negative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN
PC_ 3
Positive:  CCL5, CD52, PTPRC, B2M, SRGN
Negative:  ADIRF, SPINK1, CSTB, S100P, KRT19
PC_ 4
Positive:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1
Negative:  CD3D, IL32, CCL5, CRIP1, TRAC
PC_ 5
Positive:  LUM, MMP2, PTGDS, RARRES2, DCN
Negative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9
PC_ 6
Positive:  SPINK1, CCT2, LCN15, UCA1, PLA2G2A
Negative:  KRT13, PLAUR, LYPD3, OLFM4, EMP1
PC_ 7
Positive:  FABP5, PLA2G2A, LCN15, FABP4, RPS19
Negative:  CCT2, ADIRF, SPINK1, HSPA1A, CCND1
PC_ 8
Positive:  FOS, HSPA1A, JUN, ZFP36, DNAJB1
Negative:  SPARC, COL4A1, INSR, CCL5, COL4A2
PC_ 9
Positive:  LCN15, PLA2G2A, FABP4, CRTAC1, LINC01088
Negative:  H19, RPS19, CRH, AP005230.1, PSCA
PC_ 10
Positive:  HSPA1A, HSPA1B, DNAJB1, HSP90AA1, MALAT1
Negative:  CRH, CCL5, LY6D, ACKR1, RPS19">
                     <pre class="notranslate"><code>PC_ 1
Positive:  KRT19, CD52, KRT13, S100P, RPS19
Negative:  IGFBP7, MGP, SPARC, VIM, SPARCL1
PC_ 2
Positive:  PLVAP, PCAT19, CALCRL, MCTP1, AQP1
Negative:  COL1A2, COL3A1, COL1A1, TAGLN, DCN
PC_ 3
Positive:  CCL5, CD52, PTPRC, B2M, SRGN
Negative:  ADIRF, SPINK1, CSTB, S100P, KRT19
PC_ 4
Positive:  HLA-DRA, CD74, TYROBP, HLA-DRB1, HLA-DPB1
Negative:  CD3D, IL32, CCL5, CRIP1, TRAC
PC_ 5
Positive:  LUM, MMP2, PTGDS, RARRES2, DCN
Negative:  RGS5, NDUFA4L2, ACTA2, PPP1R14A, MYL9
PC_ 6
Positive:  SPINK1, CCT2, LCN15, UCA1, PLA2G2A
Negative:  KRT13, PLAUR, LYPD3, OLFM4, EMP1
PC_ 7
Positive:  FABP5, PLA2G2A, LCN15, FABP4, RPS19
Negative:  CCT2, ADIRF, SPINK1, HSPA1A, CCND1
PC_ 8
Positive:  FOS, HSPA1A, JUN, ZFP36, DNAJB1
Negative:  SPARC, COL4A1, INSR, CCL5, COL4A2
PC_ 9
Positive:  LCN15, PLA2G2A, FABP4, CRTAC1, LINC01088
Negative:  H19, RPS19, CRH, AP005230.1, PSCA
PC_ 10
Positive:  HSPA1A, HSPA1B, DNAJB1, HSP90AA1, MALAT1
Negative:  CRH, CCL5, LY6D, ACKR1, RPS19
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# To determine how many Pcs should be considered for clustering:
# Plot the elbow plot
png(filename = "elbow.png", width = 16, height = 8.135, units = "in", res = 300)
ElbowPlot(object = seurat_integrated, 
          ndims = 40)
dev.off()
' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> To determine how many Pcs should be considered for clustering:</span>
<span class="pl-c"><span class="pl-c">#</span> Plot the elbow plot</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>elbow.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
ElbowPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
          <span class="pl-v">ndims</span> <span class="pl-k">=</span> <span class="pl-c1">40</span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/elbow.png" rel="noopener noreferrer" target="_blank">
                      <img alt="plot-elbow" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/elbow.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# to make it more quantitative :
# Determine percent of variation associated with each PC
pct &lt;- seurat_integrated[["pca"]]@stdev / sum(seurat_integrated[["pca"]]@stdev) * 100

#pct
# [1] 7.9623240 6.3833277 5.7806955 4.4000340 3.9930564 3.6021757 3.5527516
# [8] 3.0283830 2.9108397 2.6331792 2.4437350 2.3524902 2.2441047 2.1753633
#[15] 2.0485619 2.0020779 1.9396254 1.8168114 1.7548723 1.6539166 1.6410801
#[22] 1.5884590 1.5223992 1.4950181 1.4627909 1.3921263 1.3815652 1.3046432
#[29] 1.2605721 1.2382860 1.2051052 1.1972413 1.1821032 1.1741361 1.1577815
#[36] 1.1294799 1.1190593 1.0968174 1.0591574 1.0519859 1.0396189 1.0211281
#[43] 1.0034793 0.9976230 0.9786685 0.9497183 0.9415772 0.9171901 0.9103623
#[50] 0.9025012


# Calculate cumulative percents for each PC
cumu &lt;- cumsum(pct)

#cumu
# [1]   7.962324  14.345652  20.126347  24.526381  28.519438  32.121613
# [7]  35.674365  38.702748  41.613588  44.246767  46.690502  49.042992
#[13]  51.287097  53.462460  55.511022  57.513100  59.452725  61.269537
#[19]  63.024409  64.678326  66.319406  67.907865  69.430264  70.925282
#[25]  72.388073  73.780199  75.161764  76.466408  77.726980  78.965266
#[31]  80.170371  81.367612  82.549715  83.723852  84.881633  86.011113
#[37]  87.130172  88.226990  89.286147  90.338133  91.377752  92.398880
#[43]  93.402359  94.399982  95.378651  96.328369  97.269946  98.187137


# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 &lt;- which(cumu &gt; 90 &amp; pct &lt; 5)[1]

co1

# Determine the difference between variation of PC and subsequent PC
co2 &lt;- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) &gt; 0.1), decreasing = T)[1] + 1

# last point where change of % of variation is more than 0.1%.
co2

# Minimum of the two calculation is the optimal number of PC to pick.
pcs &lt;- min(co1, co2)

pcs' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> to make it more quantitative :</span>
<span class="pl-c"><span class="pl-c">#</span> Determine percent of variation associated with each PC</span>
<span class="pl-smi">pct</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">seurat_integrated</span>[[<span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>]]<span class="pl-k">@</span><span class="pl-smi">stdev</span> <span class="pl-k">/</span> sum(<span class="pl-smi">seurat_integrated</span>[[<span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>]]<span class="pl-k">@</span><span class="pl-smi">stdev</span>) <span class="pl-k">*</span> <span class="pl-c1">100</span>

<span class="pl-c"><span class="pl-c">#</span>pct</span>
<span class="pl-c"><span class="pl-c">#</span> [1] 7.9623240 6.3833277 5.7806955 4.4000340 3.9930564 3.6021757 3.5527516</span>
<span class="pl-c"><span class="pl-c">#</span> [8] 3.0283830 2.9108397 2.6331792 2.4437350 2.3524902 2.2441047 2.1753633</span>
<span class="pl-c"><span class="pl-c">#</span>[15] 2.0485619 2.0020779 1.9396254 1.8168114 1.7548723 1.6539166 1.6410801</span>
<span class="pl-c"><span class="pl-c">#</span>[22] 1.5884590 1.5223992 1.4950181 1.4627909 1.3921263 1.3815652 1.3046432</span>
<span class="pl-c"><span class="pl-c">#</span>[29] 1.2605721 1.2382860 1.2051052 1.1972413 1.1821032 1.1741361 1.1577815</span>
<span class="pl-c"><span class="pl-c">#</span>[36] 1.1294799 1.1190593 1.0968174 1.0591574 1.0519859 1.0396189 1.0211281</span>
<span class="pl-c"><span class="pl-c">#</span>[43] 1.0034793 0.9976230 0.9786685 0.9497183 0.9415772 0.9171901 0.9103623</span>
<span class="pl-c"><span class="pl-c">#</span>[50] 0.9025012</span>


<span class="pl-c"><span class="pl-c">#</span> Calculate cumulative percents for each PC</span>
<span class="pl-smi">cumu</span> <span class="pl-k">&lt;-</span> cumsum(<span class="pl-smi">pct</span>)

<span class="pl-c"><span class="pl-c">#</span>cumu</span>
<span class="pl-c"><span class="pl-c">#</span> [1]   7.962324  14.345652  20.126347  24.526381  28.519438  32.121613</span>
<span class="pl-c"><span class="pl-c">#</span> [7]  35.674365  38.702748  41.613588  44.246767  46.690502  49.042992</span>
<span class="pl-c"><span class="pl-c">#</span>[13]  51.287097  53.462460  55.511022  57.513100  59.452725  61.269537</span>
<span class="pl-c"><span class="pl-c">#</span>[19]  63.024409  64.678326  66.319406  67.907865  69.430264  70.925282</span>
<span class="pl-c"><span class="pl-c">#</span>[25]  72.388073  73.780199  75.161764  76.466408  77.726980  78.965266</span>
<span class="pl-c"><span class="pl-c">#</span>[31]  80.170371  81.367612  82.549715  83.723852  84.881633  86.011113</span>
<span class="pl-c"><span class="pl-c">#</span>[37]  87.130172  88.226990  89.286147  90.338133  91.377752  92.398880</span>
<span class="pl-c"><span class="pl-c">#</span>[43]  93.402359  94.399982  95.378651  96.328369  97.269946  98.187137</span>


<span class="pl-c"><span class="pl-c">#</span> Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5</span>
<span class="pl-smi">co1</span> <span class="pl-k">&lt;-</span> which(<span class="pl-smi">cumu</span> <span class="pl-k">&gt;</span> <span class="pl-c1">90</span> <span class="pl-k">&amp;</span> <span class="pl-smi">pct</span> <span class="pl-k">&lt;</span> <span class="pl-c1">5</span>)[<span class="pl-c1">1</span>]

<span class="pl-smi">co1</span>

<span class="pl-c"><span class="pl-c">#</span> Determine the difference between variation of PC and subsequent PC</span>
<span class="pl-smi">co2</span> <span class="pl-k">&lt;-</span> sort(which((<span class="pl-smi">pct</span>[<span class="pl-c1">1</span><span class="pl-k">:</span>length(<span class="pl-smi">pct</span>) <span class="pl-k">-</span> <span class="pl-c1">1</span>] <span class="pl-k">-</span> <span class="pl-smi">pct</span>[<span class="pl-c1">2</span><span class="pl-k">:</span>length(<span class="pl-smi">pct</span>)]) <span class="pl-k">&gt;</span> <span class="pl-c1">0.1</span>), <span class="pl-v">decreasing</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>)[<span class="pl-c1">1</span>] <span class="pl-k">+</span> <span class="pl-c1">1</span>

<span class="pl-c"><span class="pl-c">#</span> last point where change of % of variation is more than 0.1%.</span>
<span class="pl-smi">co2</span>

<span class="pl-c"><span class="pl-c">#</span> Minimum of the two calculation is the optimal number of PC to pick.</span>
<span class="pl-smi">pcs</span> <span class="pl-k">&lt;-</span> min(<span class="pl-smi">co1</span>, <span class="pl-smi">co2</span>)

<span class="pl-smi">pcs</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Cluster the cells
                     </h4>
                     <a aria-label="Permalink: Cluster the cells" class="anchor" href="#cluster-the-cells" id="user-content-cluster-the-cells">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# to check what is active assay
DefaultAssay(object = seurat_integrated)

# Determine the K-nearest neighbor graph
seurat_integrated &lt;- FindNeighbors(object = seurat_integrated, 
                                dims = 1:18)
                                
Find clusters
# Determine the clusters for various resolutions                                
seurat_integrated &lt;- FindClusters(object = seurat_integrated,
                               resolution = c(0.4, 0.6, 0.8, 1.0, 1.4))
                               

# Explore resolutions
head(seurat_integrated@meta.data)

# Assign the identity of clusters
Idents(object = seurat_integrated) &lt;- "integrated_snn_res.0.4"

# Plot the UMAP
png(filename = "umap_cluster_with_label.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> to check what is active assay</span>
DefaultAssay(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>)

<span class="pl-c"><span class="pl-c">#</span> Determine the K-nearest neighbor graph</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> FindNeighbors(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                                <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">18</span>)
                                
<span class="pl-smi">Find</span> <span class="pl-smi">clusters</span>
<span class="pl-c"><span class="pl-c">#</span> Determine the clusters for various resolutions                                </span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> FindClusters(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>,
                               <span class="pl-v">resolution</span> <span class="pl-k">=</span> c(<span class="pl-c1">0.4</span>, <span class="pl-c1">0.6</span>, <span class="pl-c1">0.8</span>, <span class="pl-c1">1.0</span>, <span class="pl-c1">1.4</span>))
                               

<span class="pl-c"><span class="pl-c">#</span> Explore resolutions</span>
head(<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)

<span class="pl-c"><span class="pl-c">#</span> Assign the identity of clusters</span>
Idents(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>) <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>integrated_snn_res.0.4<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span> Plot the UMAP</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_cluster_with_label.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster_with_label.png" rel="noopener noreferrer" target="_blank">
                      <img alt="plot-umap_label" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster_with_label.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Clustering quality control
                     </h4>
                     <a aria-label="Permalink: Clustering quality control" class="anchor" href="#clustering-quality-control" id="user-content-clustering-quality-control">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     After clustering, we need to make sure that the assigned clusters are true representative of biological clusters (cell clusters) not due to technical or unwanted source of variation (like cell cycle stages). Also, in this step, we need to identify cell type for each cluster based on the known cell type markers.
                    </p>
                    <ul dir="auto">
                     <li>
                      Segregation of clusters by sample
                     </li>
                    </ul>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Extract identity and sample information from Seurat object to determine the number of cells per cluster per sample

library(dplyr)
library(tidyr)

# n_cells &lt;- FetchData(seurat_integrated, 
#                      vars = c("ident", "orig.ident")) %&gt;%
#         dplyr::count(ident, orig.ident) %&gt;%
#         tidyr::spread(ident, n)

n_cells &lt;- FetchData(seurat_integrated, 
                      vars = c("ident", "orig.ident"))
n_cells &lt;- dplyr::count(n_cells, ident, orig.ident)
n_cells &lt;- tidyr::spread(n_cells, ident, n)



#Ading sample data from paper; we expect to see samples from the same group have more or less similar number of cells in each cluster. 
#So normal samples should show similar patterns: SRR12603780, SRR12603781, and SRR12603788.



sampleData&lt;- data.frame(tibble::tribble(
     ~sample_id, ~gender, ~age, ~Grade, ~Invasiveness, ~Surgery_Type, ~Tumor_size_cm,
  "SRR12603790",     "M",  67L,  "low", "Noninvasive",       "TURBT",          "1.9",
  "SRR12603789",     "M",  70L,  "low", "Noninvasive",       "TURBT",          "2.5",
  "SRR12603787",     "M",  63L, "high", "Noninvasive",  "Cystectomy",          "3.5",
  "SRR12603786",     "F",  59L, "high", "Noninvasive",  "Cystectomy",          "4.7",
  "SRR12603785",     "M",  57L, "high",    "Invasive",  "Cystectomy",          "5.1",
  "SRR12603784",     "M",  75L, "high",    "Invasive",  "Cystectomy",          "4.3",
  "SRR12603783",     "M",  77L, "high",    "Invasive",  "Cystectomy",          "4.5",
  "SRR12603782",     "F",  72L, "high",    "Invasive",  "Cystectomy",          "4.1",
  "SRR12603781",     "M",  67L, "normal",   "normal",       "TURBT",            "-",
  "SRR12603788",     "M",  75L, "normal",   "normal",  "Cystectomy",            "-",
  "SRR12603780",     "M",  63L, "normal",   "normal",  "Cystectomy",            "-"
  ))


# View table
head(n_cells)
# saving objects (to mark where and when we stored the file)
#seurat_cluster &lt;- seurat_integrated
#saveRDS(seurat_cluster, "seurat_cluster.RDS")


# UMAP of cells in each cluster by sample
# This would allow us to see condition specific clusters
png(filename = "umap_cluster_sample.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated, 
        label = TRUE, 
        split.by = "sample")  + NoLegend()
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Extract identity and sample information from Seurat object to determine the number of cells per cluster per sample</span>

library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">tidyr</span>)

<span class="pl-c"><span class="pl-c">#</span> n_cells &lt;- FetchData(seurat_integrated, </span>
<span class="pl-c"><span class="pl-c">#</span>                      vars = c("ident", "orig.ident")) %&gt;%</span>
<span class="pl-c"><span class="pl-c">#</span>         dplyr::count(ident, orig.ident) %&gt;%</span>
<span class="pl-c"><span class="pl-c">#</span>         tidyr::spread(ident, n)</span>

<span class="pl-smi">n_cells</span> <span class="pl-k">&lt;-</span> FetchData(<span class="pl-smi">seurat_integrated</span>, 
                      <span class="pl-v">vars</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>ident<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>))
<span class="pl-smi">n_cells</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>count(<span class="pl-smi">n_cells</span>, <span class="pl-smi">ident</span>, <span class="pl-smi">orig.ident</span>)
<span class="pl-smi">n_cells</span> <span class="pl-k">&lt;-</span> <span class="pl-e">tidyr</span><span class="pl-k">::</span>spread(<span class="pl-smi">n_cells</span>, <span class="pl-smi">ident</span>, <span class="pl-smi">n</span>)



<span class="pl-c"><span class="pl-c">#</span>Ading sample data from paper; we expect to see samples from the same group have more or less similar number of cells in each cluster. </span>
<span class="pl-c"><span class="pl-c">#</span>So normal samples should show similar patterns: SRR12603780, SRR12603781, and SRR12603788.</span>



<span class="pl-smi">sampleData</span><span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-e">tibble</span><span class="pl-k">::</span>tribble(
     <span class="pl-k">~</span><span class="pl-smi">sample_id</span>, <span class="pl-k">~</span><span class="pl-smi">gender</span>, <span class="pl-k">~</span><span class="pl-smi">age</span>, <span class="pl-k">~</span><span class="pl-smi">Grade</span>, <span class="pl-k">~</span><span class="pl-smi">Invasiveness</span>, <span class="pl-k">~</span><span class="pl-smi">Surgery_Type</span>, <span class="pl-k">~</span><span class="pl-smi">Tumor_size_cm</span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603790<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">67L</span>,  <span class="pl-s"><span class="pl-pds">"</span>low<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Noninvasive<span class="pl-pds">"</span></span>,       <span class="pl-s"><span class="pl-pds">"</span>TURBT<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>1.9<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603789<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">70L</span>,  <span class="pl-s"><span class="pl-pds">"</span>low<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Noninvasive<span class="pl-pds">"</span></span>,       <span class="pl-s"><span class="pl-pds">"</span>TURBT<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>2.5<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603787<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">63L</span>, <span class="pl-s"><span class="pl-pds">"</span>high<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Noninvasive<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>3.5<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603786<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>F<span class="pl-pds">"</span></span>,  <span class="pl-c1">59L</span>, <span class="pl-s"><span class="pl-pds">"</span>high<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Noninvasive<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>4.7<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603785<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">57L</span>, <span class="pl-s"><span class="pl-pds">"</span>high<span class="pl-pds">"</span></span>,    <span class="pl-s"><span class="pl-pds">"</span>Invasive<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>5.1<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603784<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">75L</span>, <span class="pl-s"><span class="pl-pds">"</span>high<span class="pl-pds">"</span></span>,    <span class="pl-s"><span class="pl-pds">"</span>Invasive<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>4.3<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603783<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">77L</span>, <span class="pl-s"><span class="pl-pds">"</span>high<span class="pl-pds">"</span></span>,    <span class="pl-s"><span class="pl-pds">"</span>Invasive<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>4.5<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603782<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>F<span class="pl-pds">"</span></span>,  <span class="pl-c1">72L</span>, <span class="pl-s"><span class="pl-pds">"</span>high<span class="pl-pds">"</span></span>,    <span class="pl-s"><span class="pl-pds">"</span>Invasive<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,          <span class="pl-s"><span class="pl-pds">"</span>4.1<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603781<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">67L</span>, <span class="pl-s"><span class="pl-pds">"</span>normal<span class="pl-pds">"</span></span>,   <span class="pl-s"><span class="pl-pds">"</span>normal<span class="pl-pds">"</span></span>,       <span class="pl-s"><span class="pl-pds">"</span>TURBT<span class="pl-pds">"</span></span>,            <span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603788<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">75L</span>, <span class="pl-s"><span class="pl-pds">"</span>normal<span class="pl-pds">"</span></span>,   <span class="pl-s"><span class="pl-pds">"</span>normal<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,            <span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>,
  <span class="pl-s"><span class="pl-pds">"</span>SRR12603780<span class="pl-pds">"</span></span>,     <span class="pl-s"><span class="pl-pds">"</span>M<span class="pl-pds">"</span></span>,  <span class="pl-c1">63L</span>, <span class="pl-s"><span class="pl-pds">"</span>normal<span class="pl-pds">"</span></span>,   <span class="pl-s"><span class="pl-pds">"</span>normal<span class="pl-pds">"</span></span>,  <span class="pl-s"><span class="pl-pds">"</span>Cystectomy<span class="pl-pds">"</span></span>,            <span class="pl-s"><span class="pl-pds">"</span>-<span class="pl-pds">"</span></span>
  ))


<span class="pl-c"><span class="pl-c">#</span> View table</span>
head(<span class="pl-smi">n_cells</span>)
<span class="pl-c"><span class="pl-c">#</span> saving objects (to mark where and when we stored the file)</span>
<span class="pl-c"><span class="pl-c">#</span>seurat_cluster &lt;- seurat_integrated</span>
<span class="pl-c"><span class="pl-c">#</span>saveRDS(seurat_cluster, "seurat_cluster.RDS")</span>


<span class="pl-c"><span class="pl-c">#</span> UMAP of cells in each cluster by sample</span>
<span class="pl-c"><span class="pl-c">#</span> This would allow us to see condition specific clusters</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_cluster_sample.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>sample<span class="pl-pds">"</span></span>)  <span class="pl-k">+</span> NoLegend()
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster_sample.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_cluster_sample.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster_sample.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Segregation of clusters by cell cycle phase (unwanted source of variation) 
# Explore whether clusters segregate by cell cycle phase

png(filename = "umap_cluster_cell_cucle.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated,
        label = TRUE, 
        split.by = "Phase")  + NoLegend()
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Segregation of clusters by cell cycle phase (unwanted source of variation) </span>
<span class="pl-c"><span class="pl-c">#</span> Explore whether clusters segregate by cell cycle phase</span>

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_cluster_cell_cucle.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Phase<span class="pl-pds">"</span></span>)  <span class="pl-k">+</span> NoLegend()
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster_cell_cucle.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_cluster_cell_cucle.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster_cell_cucle.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <ul dir="auto">
                     <li>
                      Segregation of clusters by various sources of uninteresting variation
                     </li>
                    </ul>
                    <p dir="auto">
                     We expect to see a uniform coloring for all variables in all clusters. Sometimes this is not the case. For here
                     <code>
                      nUMI
                     </code>
                     and
                     <code>
                      nGene
                     </code>
                     show higher values is some clusters. We have to watch these clusters and inspect them in terms of the type of cell therein. So that may explain some of the variation that we are seeing.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Determine metrics to plot present in seurat_integrated@meta.data
metrics &lt;-  c(&quot;nUMI&quot;, &quot;nGene&quot;, &quot;S.Score&quot;, &quot;G2M.Score&quot;, &quot;mitoRatio&quot;)
png(filename = &quot;umap_unwanted_source_clustering.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(seurat_integrated, 
            reduction = &quot;umap&quot;, 
            features = metrics,
            pt.size = 0.4, 
            order = TRUE,
            min.cutoff = 'q10',
            label = TRUE)
dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Determine metrics to plot present in seurat_integrated@meta.data</span>
<span class="pl-smi">metrics</span> <span class="pl-k">&lt;-</span>  c(<span class="pl-s"><span class="pl-pds">"</span>nUMI<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>nGene<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>S.Score<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>G2M.Score<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>mitoRatio<span class="pl-pds">"</span></span>)
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_unwanted_source_clustering.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-smi">seurat_integrated</span>, 
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
            <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">metrics</span>,
            <span class="pl-v">pt.size</span> <span class="pl-k">=</span> <span class="pl-c1">0.4</span>, 
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_unwanted_source_clustering.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_unwanted_source_clustering.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_unwanted_source_clustering.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <ul dir="auto">
                     <li>
                      Exploration of the PCs driving the different clusters
                     </li>
                    </ul>
                    <p dir="auto">
                     We hope that the defined PCs could separate clusters well.We can see how the clusters are represented by the different PCs.Then we could look back at our genes driving this PC to get an idea of what the cell types might be in each cluster.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Defining the information in the seurat object of interest
columns &lt;- c(paste0("PC_", 1:18),
            "ident",
            "UMAP_1", "UMAP_2")

# Extracting this data from the seurat object
pc_data &lt;- FetchData(seurat_integrated, 
                     vars = columns)
                     
                     
# Adding cluster label to center of cluster on UMAP
umap_label &lt;- FetchData(seurat_integrated, 
                        vars = c("ident", "UMAP_1", "UMAP_2"))  %&gt;%
  group_by(ident) %&gt;%
  summarise(x=mean(UMAP_1), y=mean(UMAP_2))



# Plotting a UMAP plot for each of the PCs
library(cowplot)
library(tidyverse)
library(HGNChelper)

png(filename = "umap_on_pcs.png", width = 16, height = 8.135, units = "in", res = 300)
map(paste0("PC_", 1:18), function(pc){
        ggplot(pc_data, 
               aes(UMAP_1, UMAP_2)) +
                geom_point(aes_string(color=pc), 
                           alpha = 0.7) +
                scale_color_gradient(guide = FALSE, 
                                     low = "grey90", 
                                     high = "blue")  +
                geom_text(data=umap_label, 
                          aes(label=ident, x, y)) +
                ggtitle(pc)
}) %&gt;% 
        plot_grid(plotlist = .)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Defining the information in the seurat object of interest</span>
<span class="pl-smi">columns</span> <span class="pl-k">&lt;-</span> c(paste0(<span class="pl-s"><span class="pl-pds">"</span>PC_<span class="pl-pds">"</span></span>, <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">18</span>),
            <span class="pl-s"><span class="pl-pds">"</span>ident<span class="pl-pds">"</span></span>,
            <span class="pl-s"><span class="pl-pds">"</span>UMAP_1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>UMAP_2<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Extracting this data from the seurat object</span>
<span class="pl-smi">pc_data</span> <span class="pl-k">&lt;-</span> FetchData(<span class="pl-smi">seurat_integrated</span>, 
                     <span class="pl-v">vars</span> <span class="pl-k">=</span> <span class="pl-smi">columns</span>)
                     
                     
<span class="pl-c"><span class="pl-c">#</span> Adding cluster label to center of cluster on UMAP</span>
<span class="pl-smi">umap_label</span> <span class="pl-k">&lt;-</span> FetchData(<span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">vars</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>ident<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>UMAP_1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>UMAP_2<span class="pl-pds">"</span></span>))  %<span class="pl-k">&gt;</span>%
  group_by(<span class="pl-smi">ident</span>) %<span class="pl-k">&gt;</span>%
  summarise(<span class="pl-v">x</span><span class="pl-k">=</span>mean(<span class="pl-smi">UMAP_1</span>), <span class="pl-v">y</span><span class="pl-k">=</span>mean(<span class="pl-smi">UMAP_2</span>))



<span class="pl-c"><span class="pl-c">#</span> Plotting a UMAP plot for each of the PCs</span>
library(<span class="pl-smi">cowplot</span>)
library(<span class="pl-smi">tidyverse</span>)
library(<span class="pl-smi">HGNChelper</span>)

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_on_pcs.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
map(paste0(<span class="pl-s"><span class="pl-pds">"</span>PC_<span class="pl-pds">"</span></span>, <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">18</span>), <span class="pl-k">function</span>(<span class="pl-smi">pc</span>){
        ggplot(<span class="pl-smi">pc_data</span>, 
               aes(<span class="pl-smi">UMAP_1</span>, <span class="pl-smi">UMAP_2</span>)) <span class="pl-k">+</span>
                geom_point(aes_string(<span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">pc</span>), 
                           <span class="pl-v">alpha</span> <span class="pl-k">=</span> <span class="pl-c1">0.7</span>) <span class="pl-k">+</span>
                scale_color_gradient(<span class="pl-v">guide</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, 
                                     <span class="pl-v">low</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>grey90<span class="pl-pds">"</span></span>, 
                                     <span class="pl-v">high</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>)  <span class="pl-k">+</span>
                geom_text(<span class="pl-v">data</span><span class="pl-k">=</span><span class="pl-smi">umap_label</span>, 
                          aes(<span class="pl-v">label</span><span class="pl-k">=</span><span class="pl-smi">ident</span>, <span class="pl-smi">x</span>, <span class="pl-smi">y</span>)) <span class="pl-k">+</span>
                ggtitle(<span class="pl-smi">pc</span>)
}) %<span class="pl-k">&gt;</span>% 
        plot_grid(<span class="pl-v">plotlist</span> <span class="pl-k">=</span> <span class="pl-smi">.</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_on_pcs.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_on_pcs.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_on_pcs.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Examine PCA results 
print(seurat_integrated[["pca"]], dims = 1:5, nfeatures = 5)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Examine PCA results </span>
print(<span class="pl-smi">seurat_integrated</span>[[<span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>]], <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">5</span>, <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>)</pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Normalize RNA data for visualization purposes
seurat_integrated &lt;- NormalizeData(seurat_integrated, verbose = FALSE)

png(filename = &quot;umap_fibroblast.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(seurat_integrated, 
            reduction = &quot;umap&quot;, 
            features = c(&quot;IGFBP7&quot;, &quot;MGP&quot;), 
            order = TRUE,
            min.cutoff = 'q10', 
            label = TRUE)
dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Normalize RNA data for visualization purposes</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> NormalizeData(<span class="pl-smi">seurat_integrated</span>, <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_fibroblast.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-smi">seurat_integrated</span>, 
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
            <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>IGFBP7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>MGP<span class="pl-pds">"</span></span>), 
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_fibroblast.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_fibroblast.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_fibroblast.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="png(filename = &quot;umap_endothelial.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(seurat_integrated, 
            reduction = &quot;umap&quot;, 
            features = c(&quot;PLVAP&quot;, &quot;CALCRL&quot;), 
            order = TRUE,
            min.cutoff = 'q10', 
            label = TRUE)
dev.off()
" dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_endothelial.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-smi">seurat_integrated</span>, 
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
            <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>PLVAP<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CALCRL<span class="pl-pds">"</span></span>), 
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_endothelial.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_endothelial.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_endothelial.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="png(filename = &quot;umap_t_cells.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(seurat_integrated, 
            reduction = &quot;umap&quot;, 
            features = c(&quot;CCL5&quot;, &quot;CD52&quot;, &quot;IL32&quot;), 
            order = TRUE,
            min.cutoff = 'q10', 
            label = TRUE)
dev.off()" dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_t_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-smi">seurat_integrated</span>, 
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
            <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>CCL5<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD52<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IL32<span class="pl-pds">"</span></span>), 
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_t_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_t_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_t_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      7) Marker identification and cell type assignment
                     </h2>
                     <a aria-label="Permalink: 7) Marker identification and cell type assignment" class="anchor" href="#7-marker-identification-and-cell-type-assignment" id="user-content-7-marker-identification-and-cell-type-assignment">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     This can be the last step in our pipeline which aims to determine the gene markers for each of the clusters and identify cell types of each cluster using markers. Also, this step helps to determine whether there's a need to re-cluster based on cell type markers, or maybe clusters need to be merged or split.
                    </p>
                    <p dir="auto">
                     For marker identification, there are three functions in the Seurat package, each with a different application:
                    </p>
                    <p dir="auto">
                     <code>
                      FindAllMarkers()
                     </code>
                     : It should only be used when comparing a cluster against other clusters belonging to the same group. i.e. this function should be used only when we have one group/condition.
                    </p>
                    <p dir="auto">
                     <code>
                      FindConservedMarkers()
                     </code>
                     : When we have two groups like tumor vs normal or invasive vs. noninvasive identifying conserved markers is the best approach. In this way, we find DE genes for a given cluster in one condition (e.g. invasive) comparing the cluster against the rest of the cluster in the same condition group. We do the same for that given cluster in the other condition (non-invasive). Finally, the two lists will be merged to give us the conserved marker for a given cluster.
                    </p>
                    <p dir="auto">
                     <code>
                      FindMarkers()
                     </code>
                     : This is helpful with identifying gene markers for each cluster. In practice sometimes the list of markers returned doesnât sufficiently separate some of the clusters. We can use this function to further differentiate between those clusters.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#______________________________ NOT TO BE RUN________________________________
# Find markers for every cluster compared to all remaining cells, report only the positive ones
markers &lt;- FindAllMarkers(object = seurat_integrated, 
                          only.pos = TRUE,
                          logfc.threshold = 0.25)     
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>______________________________ NOT TO BE RUN________________________________</span>
<span class="pl-c"><span class="pl-c">#</span> Find markers for every cluster compared to all remaining cells, report only the positive ones</span>
<span class="pl-smi">markers</span> <span class="pl-k">&lt;-</span> FindAllMarkers(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                          <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                          <span class="pl-v">logfc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">0.25</span>)     
</pre>
                    </div>
                    <p dir="auto">
                     When we have two groups like tumor vs normal or invasive vs. non-invasive identifying conserved markers is the best approach. In this way, we find DE genes for a given cluster in one condition (e.g. invasive) by comparing the cluster against the rest of the cluster in the same condition group. We do the same for that given cluster in the other condition (non-invasive). Finally, the two lists will be merged to give us the conserved marker for a given cluster.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# explecity set the default object to normalized values
DefaultAssay(seurat_integrated) &lt;- "RNA"


cluster0_conserved_markers &lt;- FindConservedMarkers(seurat_integrated,
                              ident.1 = 0,
                              grouping.var = "Invasiveness",
                                only.pos = TRUE,
                             logfc.threshold = 0.60)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> explecity set the default object to normalized values</span>
DefaultAssay(<span class="pl-smi">seurat_integrated</span>) <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>RNA<span class="pl-pds">"</span></span>


<span class="pl-smi">cluster0_conserved_markers</span> <span class="pl-k">&lt;-</span> FindConservedMarkers(<span class="pl-smi">seurat_integrated</span>,
                              <span class="pl-v">ident.1</span> <span class="pl-k">=</span> <span class="pl-c1">0</span>,
                              <span class="pl-v">grouping.var</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>,
                                <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                             <span class="pl-v">logfc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">0.60</span>)</pre>
                    </div>
                    <p dir="auto">
                     To add more annotation to the results
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Connect to AnnotationHub
ah &lt;- AnnotationHub()

# Access the Ensembl database for organism
ahDb &lt;- query(ah, 
              pattern = c("Homo sapiens", "EnsDb"), 
              ignore.case = TRUE)

# Acquire the latest annotation files
id &lt;- ahDb %&gt;%
        mcols() %&gt;%
        rownames() %&gt;%
        tail(n = 1)

# Download the appropriate Ensembldb database
edb &lt;- ah[[id]]

# Extract gene-level information from database
annotations &lt;- genes(edb, 
                     return.type = "data.frame")

# Select annotations of interest
annotations &lt;- annotations %&gt;%
        dplyr::select(gene_id, gene_name, seq_name, gene_biotype, description)' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Connect to AnnotationHub</span>
<span class="pl-smi">ah</span> <span class="pl-k">&lt;-</span> AnnotationHub()

<span class="pl-c"><span class="pl-c">#</span> Access the Ensembl database for organism</span>
<span class="pl-smi">ahDb</span> <span class="pl-k">&lt;-</span> query(<span class="pl-smi">ah</span>, 
              <span class="pl-v">pattern</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Homo sapiens<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>EnsDb<span class="pl-pds">"</span></span>), 
              <span class="pl-v">ignore.case</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span> Acquire the latest annotation files</span>
<span class="pl-smi">id</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">ahDb</span> %<span class="pl-k">&gt;</span>%
        mcols() %<span class="pl-k">&gt;</span>%
        rownames() %<span class="pl-k">&gt;</span>%
        tail(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)

<span class="pl-c"><span class="pl-c">#</span> Download the appropriate Ensembldb database</span>
<span class="pl-smi">edb</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">ah</span>[[<span class="pl-smi">id</span>]]

<span class="pl-c"><span class="pl-c">#</span> Extract gene-level information from database</span>
<span class="pl-smi">annotations</span> <span class="pl-k">&lt;-</span> genes(<span class="pl-smi">edb</span>, 
                     <span class="pl-k">return</span><span class="pl-v">.type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>data.frame<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Select annotations of interest</span>
<span class="pl-smi">annotations</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">annotations</span> %<span class="pl-k">&gt;</span>%
        <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">gene_id</span>, <span class="pl-smi">gene_name</span>, <span class="pl-smi">seq_name</span>, <span class="pl-smi">gene_biotype</span>, <span class="pl-smi">description</span>)</pre>
                    </div>
                    <p dir="auto">
                     To find conserved markers for all clusters:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Create function to get conserved markers for any given cluster
get_conserved &lt;- function(cluster){
  tryCatch({
    FindConservedMarkers(seurat_integrated,
                         ident.1 = cluster,
                         grouping.var = &quot;Invasiveness&quot;,
                         only.pos = TRUE,
                         logfc.threshold = 0.60) %&gt;%
      rownames_to_column(var = &quot;gene&quot;) %&gt;%
      left_join(y = unique(annotations[, c(&quot;gene_name&quot;, &quot;description&quot;)]),
                 by = c(&quot;gene&quot; = &quot;gene_name&quot;)) %&gt;%
      cbind(cluster_id = cluster, .)
    },
    error = function(e) {
      message(paste0(&quot;Error: &quot;, e$message))
      return(NULL)
    }
  )
}
# this function can be an argument for 'map_dfr' function :
# Iterate function across desired clusters
conserved_markers &lt;- map_dfr(c(0:20), get_conserved)

# Extract top 10 markers per cluster
top10 &lt;- conserved_markers %&gt;% 
  mutate(avg_fc = (Noninvasive_avg_log2FC + Invasive_avg_log2FC) /2) %&gt;% 
  group_by(cluster_id) %&gt;% 
  top_n(n = 10, 
        wt = avg_fc)
        
head(top10)

data.table::fwrite(top10, &quot;blca_top10_conserved_markers.csv&quot;)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Create function to get conserved markers for any given cluster</span>
<span class="pl-en">get_conserved</span> <span class="pl-k">&lt;-</span> <span class="pl-k">function</span>(<span class="pl-smi">cluster</span>){
  tryCatch({
    FindConservedMarkers(<span class="pl-smi">seurat_integrated</span>,
                         <span class="pl-v">ident.1</span> <span class="pl-k">=</span> <span class="pl-smi">cluster</span>,
                         <span class="pl-v">grouping.var</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>,
                         <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">logfc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">0.60</span>) %<span class="pl-k">&gt;</span>%
      rownames_to_column(<span class="pl-v">var</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gene<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
      left_join(<span class="pl-v">y</span> <span class="pl-k">=</span> unique(<span class="pl-smi">annotations</span>[, c(<span class="pl-s"><span class="pl-pds">"</span>gene_name<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>description<span class="pl-pds">"</span></span>)]),
                 <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>gene<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gene_name<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>%
      cbind(<span class="pl-v">cluster_id</span> <span class="pl-k">=</span> <span class="pl-smi">cluster</span>, <span class="pl-smi">.</span>)
    },
    <span class="pl-en">error</span> <span class="pl-k">=</span> <span class="pl-k">function</span>(<span class="pl-smi">e</span>) {
      message(paste0(<span class="pl-s"><span class="pl-pds">"</span>Error: <span class="pl-pds">"</span></span>, <span class="pl-smi">e</span><span class="pl-k">$</span><span class="pl-smi">message</span>))
      <span class="pl-k">return</span>(<span class="pl-c1">NULL</span>)
    }
  )
}
<span class="pl-c"><span class="pl-c">#</span> this function can be an argument for 'map_dfr' function :</span>
<span class="pl-c"><span class="pl-c">#</span> Iterate function across desired clusters</span>
<span class="pl-smi">conserved_markers</span> <span class="pl-k">&lt;-</span> map_dfr(c(<span class="pl-c1">0</span><span class="pl-k">:</span><span class="pl-c1">20</span>), <span class="pl-smi">get_conserved</span>)

<span class="pl-c"><span class="pl-c">#</span> Extract top 10 markers per cluster</span>
<span class="pl-smi">top10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">conserved_markers</span> %<span class="pl-k">&gt;</span>% 
  mutate(<span class="pl-v">avg_fc</span> <span class="pl-k">=</span> (<span class="pl-smi">Noninvasive_avg_log2FC</span> <span class="pl-k">+</span> <span class="pl-smi">Invasive_avg_log2FC</span>) <span class="pl-k">/</span><span class="pl-c1">2</span>) %<span class="pl-k">&gt;</span>% 
  group_by(<span class="pl-smi">cluster_id</span>) %<span class="pl-k">&gt;</span>% 
  top_n(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, 
        <span class="pl-v">wt</span> <span class="pl-k">=</span> <span class="pl-smi">avg_fc</span>)
        
head(<span class="pl-smi">top10</span>)

<span class="pl-smi">data.table</span><span class="pl-k">::</span>fwrite(<span class="pl-smi">top10</span>, <span class="pl-s"><span class="pl-pds">"</span>blca_top10_conserved_markers.csv<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     There are a number of tools that one may use to assign cell types to a cluster. However almost none of them at the time of writing can help with bladder tissue. So I have to look up genes in the
                     <a href="https://panglaodb.se/index.html" rel="nofollow">
                      PanglaoDB
                     </a>
                     database manually and assign cell type to each cluster. So to do this job in a more efficient way, let's first identify which markers are associated with more clusters, then assign cell type to those clusters.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='top10_mod &lt;- data.frame(unclass(table(top10$gene, top10$cluster_id)))


data.table::fwrite(top10_mod, "top10_mod_conserved_markers.csv")

# markers with highest frequency
M &lt;- c("KRT7", "KRT19", "FCER1G", "AIF1", "AQP3", "CCL5", "CD24", "CD3D", "CD52", "CLDN4", "COL1A1", "COL1A2", "CRTAC1", "CXCL8", "DCN", "FABP4", "FABP5", "FXYD3", "GZMA", "HLA-DRA", "IGHA1", "IGHG1", "IGHG3", "IGHG4", "IGKC", "IGLC1", "IGLC2", "IGLC3", "JCHAIN")' dir="auto">
                     <pre><span class="pl-smi">top10_mod</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(unclass(table(<span class="pl-smi">top10</span><span class="pl-k">$</span><span class="pl-smi">gene</span>, <span class="pl-smi">top10</span><span class="pl-k">$</span><span class="pl-smi">cluster_id</span>)))


<span class="pl-smi">data.table</span><span class="pl-k">::</span>fwrite(<span class="pl-smi">top10_mod</span>, <span class="pl-s"><span class="pl-pds">"</span>top10_mod_conserved_markers.csv<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> markers with highest frequency</span>
<span class="pl-smi">M</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>KRT7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>KRT19<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>FCER1G<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>AIF1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>AQP3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CCL5<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD24<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD3D<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD52<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CLDN4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>COL1A1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>COL1A2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CRTAC1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CXCL8<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>DCN<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>FABP4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>FABP5<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>FXYD3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>GZMA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>HLA-DRA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGHA1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGHG1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGHG3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGHG4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGKC<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGLC1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGLC2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>IGLC3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>JCHAIN<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     So cell type for top markers:
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        cell type
                       </th>
                       <th>
                        genes
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        Basal cells
                       </td>
                       <td>
                        KRT7, KRT19, AQP3, CD24,CXCL8,FXYD3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Dendritic cells
                       </td>
                       <td>
                        FCER1G,AIF1,FABP4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Gamma delta T cells
                       </td>
                       <td>
                        CCL5,GZMA
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NK cell
                       </td>
                       <td>
                        CCL5,GZMA
                       </td>
                      </tr>
                      <tr>
                       <td>
                        T cells
                       </td>
                       <td>
                        CD3D, CD52
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Macrophages
                       </td>
                       <td>
                        CD52
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Luminal epithelial cells
                       </td>
                       <td>
                        CLDN4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Fibroblasts
                       </td>
                       <td>
                        COLA1, COLA2,CXCL8, DCN
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Epithelial cells
                       </td>
                       <td>
                        CRTAC1,FXYD3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Endothelial cells
                       </td>
                       <td>
                        FABP4, FABP5
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Plasma cell
                       </td>
                       <td>
                        IGHA1,IGHG1,IGHG3,IGHG4,IGKC,IGLC1,IGLC3,IGLC3,JCHAIN
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     Also, there is a table from
                     <a href="https://www.nature.com/articles/s41467-020-16162-3" rel="nofollow">
                      John P. Sfakianos et al
                     </a>
                     , which summarizes the tumor cell markers:
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        CellType
                       </th>
                       <th>
                        Genes
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        Luminal
                       </td>
                       <td>
                        CYP2J2,ERBB2,ERBB3,FGFR3,FOXA1,GATA3,GPX2,KRT18,KRT19,KRT20,KRT7,KRT8,PPARG,XBP1,UPK1A,UPK2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        EMT and smooth muscle
                       </td>
                       <td>
                        PGM5,DES,C7,SRFP4,COMP,SGCD
                       </td>
                      </tr>
                      <tr>
                       <td>
                        EMT and claudin
                       </td>
                       <td>
                        ZEB1,ZEB2,VIM,SNAI1,TWIST1,FOXC2,CDH2,CLDN3,CLDN7,CLDN4,CDH1,SNAI2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Basal
                       </td>
                       <td>
                        CD44,CDH3,KRT1,KRT14,KRT16,KRT5,KRT6A,KRT6B,KRT6C
                       </td>
                      </tr>
                      <tr>
                       <td>
                        P53-like
                       </td>
                       <td>
                        ACTG2,CNN1,MYH11,MFAP4,PGM5,FLNC,ACTC1,DES,PCP4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Squamous
                       </td>
                       <td>
                        DSC1,DSC2,DSC3,DSG1,DSG2,DSG3,S100A7,S100A8
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Immune
                       </td>
                       <td>
                        CD274,PDCD1LG2,IDO1,CXCL11,L1CAM,SAA1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Neuroendocrine
                       </td>
                       <td>
                        CHGA,CHGB,SCG2,ENO2,SYP,NCAM1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Neuronal differentiation
                       </td>
                       <td>
                        MSI1,PLEKHG4B,GNG4,PEG10,RND2,APLP1,SOX2,TUBB2B
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Downregulated CIS
                       </td>
                       <td>
                        CRTAC1,CTSE,PADI3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Upregulated CIS
                       </td>
                       <td>
                        MSN,NR3C1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Cancer stem cell
                       </td>
                       <td>
                        CD44,KRT5,RPSA,ALDH1A1
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     Let's visualize some of the genes and see in which cluster they show expression.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
# Plot interesting marker gene expression 
png(filename = &quot;umap_high_freq_basal_cells.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated, 
                        features = c(&quot;KRT7&quot;, &quot;KRT19&quot;, &quot;AQP3&quot;, &quot;CD24&quot;, &quot;FXYD3&quot;, &quot;CXCL8&quot;),
                         sort.cell = TRUE,
                         min.cutoff = 'q10', 
                         label = TRUE,
                         repel = TRUE)
dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Plot interesting marker gene expression </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_high_freq_basal_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>KRT7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>KRT19<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>AQP3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD24<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>FXYD3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CXCL8<span class="pl-pds">"</span></span>),
                         <span class="pl-v">sort.cell</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
                         <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_high_freq_basal_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_high_freq_basal_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_high_freq_basal_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
# Vln plot - cluster 0
png(filename = "violin_high_freq_basal_cells.png", width = 16, height = 8.135, units = "in", res = 300)
VlnPlot(object = seurat_integrated, 
        features = c("KRT7", "KRT19", "AQP3", "CD24", "FXYD3", "CXCL8"))
dev.off() ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Vln plot - cluster 0</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>violin_high_freq_basal_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
VlnPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>KRT7<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>KRT19<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>AQP3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD24<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>FXYD3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CXCL8<span class="pl-pds">"</span></span>))
dev.off() </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_high_freq_basal_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="violin_high_freq_basal_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_high_freq_basal_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     So according to "basal cells" visualization, the following clusters may show clusters of basal cells:
0, 2,3,9,10,14,17,18,19(?),20 and 21.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Plot interesting marker gene expression 
png(filename = &quot;umap_high_freq_Plasma_cells.png&quot;, width = 26, height = 15.135, units = &quot;in&quot;, res = 600)
FeaturePlot(object = seurat_integrated, 
                        features = c(&quot;IGHA1&quot;,&quot;IGHG1&quot;,&quot;IGHG3&quot;,&quot;IGHG4&quot;,&quot;IGKC&quot;,&quot;IGLC1&quot;,&quot;IGLC3&quot;,&quot;IGLC3&quot;,&quot;JCHAIN&quot;),
                         sort.cell = TRUE,
                         min.cutoff = 'q10', 
                         label = TRUE,
                         repel = TRUE)
dev.off()
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Plot interesting marker gene expression </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_high_freq_Plasma_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">26</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">15.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">600</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>IGHA1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGHG1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGHG3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGHG4<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGKC<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGLC1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGLC3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGLC3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>JCHAIN<span class="pl-pds">"</span></span>),
                         <span class="pl-v">sort.cell</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
                         <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_high_freq_Plasma_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_high_freq_Plasma_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_high_freq_Plasma_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
# Vln plot - cluster 0
png(filename = "violin_high_freq_basal_cells.png", width = 26, height = 10.135, units = "in", res = 600)
VlnPlot(object = seurat_integrated, 
        features = c("IGHA1","IGHG1","IGHG3","IGHG4","IGKC","IGLC1","IGLC3","IGLC3","JCHAIN"))
dev.off() ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Vln plot - cluster 0</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>violin_high_freq_basal_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">26</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">10.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">600</span>)
VlnPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>IGHA1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGHG1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGHG3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGHG4<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGKC<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGLC1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGLC3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>IGLC3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>JCHAIN<span class="pl-pds">"</span></span>))
dev.off() </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_high_freq_basal_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="violin_high_freq_basal_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_high_freq_basal_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Cluster 19 looks to be hard to assign it a cell type. Will keep an eye on it.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
# Plot interesting marker gene expression 
png(filename = &quot;umap_high_freq_dc.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated, 
                        features = c(&quot;FCER1G&quot;,&quot;AIF1&quot;,&quot;FABP4&quot;),
                         sort.cell = TRUE,
                         min.cutoff = 'q10', 
                         label = TRUE,
                         repel = TRUE)
dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Plot interesting marker gene expression </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_high_freq_dc.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>FCER1G<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>AIF1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>FABP4<span class="pl-pds">"</span></span>),
                         <span class="pl-v">sort.cell</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
                         <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_high_freq_dc.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_high_freq_dc.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_high_freq_dc.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Vln plot - cluster 0
png(filename = "violin_high_freq_dc.png", width = 16, height = 8.135, units = "in", res = 300)
VlnPlot(object = seurat_integrated, 
        features = c("FCER1G","AIF1","FABP4"))
dev.off() ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Vln plot - cluster 0</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>violin_high_freq_dc.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
VlnPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>FCER1G<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>AIF1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>FABP4<span class="pl-pds">"</span></span>))
dev.off() </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_high_freq_dc.png" rel="noopener noreferrer" target="_blank">
                      <img alt="violin_high_freq_dc.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_high_freq_dc.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     For cluster1 and 4
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="d &lt;- data.table::fread(&quot;top10_conserved_markers.csv&quot;)

png(filename = &quot;umap_cluster4_markers.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated, 
                        features = d$gene[d$cluster_id == &quot;4&quot;],
                         sort.cell = TRUE,
                         min.cutoff = 'q10', 
                         label = TRUE,
                         repel = TRUE)
dev.off()
" dir="auto">
                     <pre><span class="pl-smi">d</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">data.table</span><span class="pl-k">::</span>fread(<span class="pl-s"><span class="pl-pds">"</span>top10_conserved_markers.csv<span class="pl-pds">"</span></span>)

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_cluster4_markers.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">cluster_id</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span>],
                         <span class="pl-v">sort.cell</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
                         <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster4_markers.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_cluster4_markers.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster4_markers.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
# Vln plot - cluster 0
png(filename = "violin_cluster4_markers.png", width = 16, height = 8.135, units = "in", res = 300)
VlnPlot(object = seurat_integrated, 
        features = d$gene[d$cluster_id == "4"])
dev.off() ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Vln plot - cluster 0</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>violin_cluster4_markers.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
VlnPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">cluster_id</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span>])
dev.off() </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_cluster4_markers.png" rel="noopener noreferrer" target="_blank">
                      <img alt="violin_cluster4_markers.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_cluster4_markers.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="png(filename = &quot;umap_cluster12_markers.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated, 
                        features = d$gene[d$cluster_id == &quot;12&quot;],
                         sort.cell = TRUE,
                         min.cutoff = 'q10', 
                         label = TRUE,
                         repel = TRUE)
dev.off()
" dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_cluster12_markers.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">cluster_id</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>12<span class="pl-pds">"</span></span>],
                         <span class="pl-v">sort.cell</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
                         <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster12_markers.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_cluster12_markers.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster12_markers.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     So it seems clusters 1,4,8,12,13 and 16 are the same cell type with minor subtypes. However cluster 5 and 7 show some level of expression for markers in cluster 1 + 4, but they need separate inspection.
Conserved markers in these two clusters are as follows
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[1] "LDB2"    "SPARCL1" "GNG11"   "FLT1"    "IFI27"   "RAMP2"   "PECAM1"
[8] "TCF4"    "PLVAP"   "PCAT19"  "ACKR1"   "SELE"    "CALCRL"  "ZNF385D"
[15] "ADAMTS9" "MCTP1"   "AQP1"    "VWF"     "CCL14"   "PCAT19"'>
                     <pre class="notranslate"><code>[1] "LDB2"    "SPARCL1" "GNG11"   "FLT1"    "IFI27"   "RAMP2"   "PECAM1"
[8] "TCF4"    "PLVAP"   "PCAT19"  "ACKR1"   "SELE"    "CALCRL"  "ZNF385D"
[15] "ADAMTS9" "MCTP1"   "AQP1"    "VWF"     "CCL14"   "PCAT19"
</code></pre>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        cell type
                       </th>
                       <th>
                        genes
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        Endothelial cells
                       </td>
                       <td>
                        LDB2, SPARCL1, FLT1,IFI27,RAMP2,PECAM1,TCF4,PLVAP,PCAT19,ACKR1, SELE,CALCRL,ZNF385D,ADAMTS9,MCTP1,AQP1,VWF,
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Fibroblast
                       </td>
                       <td>
                        SPARCL1,TCF4,PLVAP,ADAMTS9
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="png(filename = &quot;umap_cluster5_7_markers.png&quot;, width = 26, height = 18.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated, 
                        features = d$gene[d$cluster_id %in% c(&quot;5&quot;, &quot;7&quot;)],
                         sort.cell = TRUE,
                         min.cutoff = 'q10', 
                         label = TRUE,
                         repel = TRUE)
dev.off()
" dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_cluster5_7_markers.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">26</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">18.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">cluster_id</span> <span class="pl-k">%in%</span> c(<span class="pl-s"><span class="pl-pds">"</span>5<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>7<span class="pl-pds">"</span></span>)],
                         <span class="pl-v">sort.cell</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
                         <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster5_7_markers.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_cluster5_7_markers.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster5_7_markers.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Vln plot - cluster 0
png(filename = "violin_cluster5_7_markers.png", width = 16, height = 8.135, units = "in", res = 300)
VlnPlot(object = seurat_integrated, 
        features = d$gene[d$cluster_id == "4"])
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Vln plot - cluster 0</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>violin_cluster5_7_markers.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
VlnPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">cluster_id</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span>])
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/violin_cluster5_7_markers.png" rel="noopener noreferrer" target="_blank">
                      <img alt="violin_cluster5_7_markers.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/violin_cluster5_7_markers.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     So as expected clusters 7 and 5 represent the same cell types.
In almost all cases cluster 17 showed some level of expression for EC markers.
                    </p>
                    <p dir="auto">
                     Let's now have a look at markers for clusters 6,15 and 11:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="png(filename = &quot;umap_cluster11_15_6_markers.png&quot;, width = 26, height = 18.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated, 
                        features = unique(d$gene[d$cluster_id %in% c(&quot;6&quot;, &quot;11&quot;,&quot;15&quot;)]),
                         sort.cell = TRUE,
                         min.cutoff = 'q10', 
                         label = TRUE,
                         repel = TRUE)
dev.off()" dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_cluster11_15_6_markers.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">26</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">18.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                        <span class="pl-v">features</span> <span class="pl-k">=</span> unique(<span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">gene</span>[<span class="pl-smi">d</span><span class="pl-k">$</span><span class="pl-smi">cluster_id</span> <span class="pl-k">%in%</span> c(<span class="pl-s"><span class="pl-pds">"</span>6<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>11<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>15<span class="pl-pds">"</span></span>)]),
                         <span class="pl-v">sort.cell</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>q10<span class="pl-pds">'</span></span>, 
                         <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                         <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_cluster11_15_6_markers.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_cluster11_15_6_markers.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_cluster11_15_6_markers.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     According to the markers such as different collagen, clusters 11 and 15 seem to be fibroblasts while cluster 6 shows a mix of fibroblast and non-fibroblast markers.
                    </p>
                    <p dir="auto">
                     Now we can generate UMAP with cell type as labels;
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Rename all identities
seurat_integrated &lt;- RenameIdents(object = seurat_integrated, 
                               "0" = "Basal cells",
                               "1" = "Gamma delta T cell", # NKG7 and GZMK
                               "2" = "Basal cells",
                               "3" = "Basal cells",
                               "4" = "T cells",
                               "5" = "Endothelial cells",
                               "6" = "Fibroblast",
                               "7" = "Endothelial cells",
                               "8" = "DC",
                               "9" = "Basal cells",
                               "10" = "Basal cells",
                               "11" = "Fibroblast",
                               "12" = "T cells",
                               "13" = "T cells",
                               "14" = "Basal cells",
                               "15" = "Fibroblast",
                               "16" = "T cells",
                               "17" = "?", 
                               "18" = "Basal cells", 
                               "19" = "Plasma cells",
                               "20" = "Basal cells",
                               "21" = "Basal cells")

# Plot the UMAP withy new labells
png(filename = "umap_with_label.png", width = 16, height = 8.135, units = "in", res = 600)
DimPlot(object = seurat_integrated, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Rename all identities</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> RenameIdents(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                               <span class="pl-s"><span class="pl-pds">"</span>0<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>1<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Gamma delta T cell<span class="pl-pds">"</span></span>, <span class="pl-c"><span class="pl-c">#</span> NKG7 and GZMK</span>
                               <span class="pl-s"><span class="pl-pds">"</span>2<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>3<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>T cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>5<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Endothelial cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>6<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Fibroblast<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>7<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Endothelial cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>8<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>DC<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>9<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>10<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>11<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Fibroblast<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>12<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>T cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>13<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>T cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>14<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>15<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Fibroblast<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>16<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>T cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>17<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>?<span class="pl-pds">"</span></span>, 
                               <span class="pl-s"><span class="pl-pds">"</span>18<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>, 
                               <span class="pl-s"><span class="pl-pds">"</span>19<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Plasma cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>20<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>21<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Plot the UMAP withy new labells</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_with_label.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">600</span>)
DimPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>,
        <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_with_label.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_with_label.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_with_label.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     OK then it's time to write the Seurat object for later analysis, if any:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Save final R object
write_rds(seurat_integrated,
          path = "seurat_labelled.rds")      ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Save final R object</span>
write_rds(<span class="pl-smi">seurat_integrated</span>,
          <span class="pl-v">path</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>seurat_labelled.rds<span class="pl-pds">"</span></span>)      </pre>
                    </div>
                    <p dir="auto">
                     As a final note I tried to assign cell types to clusters using the scType tool, however, there was no bladder tissue profile in their database. So the generated labels using this approach should be only considered as a guide.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Cell maker assignemnt using scType

# load libraries
library(HGNChelper)


# load gene set preparation function
source(&quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R&quot;)
# load cell type annotation function
source(&quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R&quot;)

db_ = &quot;https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx&quot;;
#tissue = c(&quot;Immune system&quot;) # e.g. Immune system,Pancreas,Liver,Eye,Kidney,Brain,Lung,Adrenal,Heart,Intestine,Muscle,Placenta,Spleen,Stomach,Thymus 
tissue = c(&quot;Immune system&quot;, &quot;Pancreas&quot;, &quot;Liver&quot;,&quot;Kidney&quot;,&quot;Intestine&quot;,&quot;Placenta&quot;,&quot;Spleen&quot;,
           &quot;Stomach&quot;) 

# prepare gene sets
gs_list = gene_sets_prepare(db_, tissue)

# get cell-type by cell matrix
es.max = sctype_score(scRNAseqData = seurat_integrated[[&quot;integrated&quot;]]@scale.data, scaled = TRUE, 
                      gs = gs_list$gs_positive, gs2 = gs_list$gs_negative) 

# merge by cluster
cL_resutls = do.call(
  &quot;rbind&quot;, lapply(unique(seurat_integrated@meta.data$seurat_clusters), function(cl){
  es.max.cl = sort(rowSums(es.max[ ,rownames(seurat_integrated@meta.data[seurat_integrated@meta.data$seurat_clusters==cl, ])]), decreasing = !0)
  head(data.frame(cluster = cl, type = names(es.max.cl), scores = es.max.cl, ncells = sum(seurat_integrated@meta.data$seurat_clusters==cl)), 10)
}))


sctype_scores = cL_resutls %&gt;% group_by(cluster) %&gt;% top_n(n = 1, wt = scores)

# set low-confident (low ScType score) clusters to &quot;unknown&quot;
sctype_scores$type[as.numeric(as.character(sctype_scores$scores)) &lt; sctype_scores$ncells/4] = &quot;Unknown&quot;
print(sctype_scores[,1:3])



# overlay the identified cell types on UMAP plot:

seurat_integrated@meta.data$customclassif = &quot;&quot;
for(j in unique(sctype_scores$cluster)){
  cl_type = sctype_scores[sctype_scores$cluster==j,]; 
  seurat_integrated@meta.data$customclassif[seurat_integrated@meta.data$seurat_clusters == j] = as.character(cl_type$type[1])
}

png(filename = &quot;umap_with_label_scType.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 600)
DimPlot(seurat_integrated, 
        reduction = &quot;umap&quot;, 
        label = TRUE, 
        repel = TRUE, 
        group.by = 'customclassif')        

dev.off()
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Cell maker assignemnt using scType</span>

<span class="pl-c"><span class="pl-c">#</span> load libraries</span>
library(<span class="pl-smi">HGNChelper</span>)


<span class="pl-c"><span class="pl-c">#</span> load gene set preparation function</span>
source(<span class="pl-s"><span class="pl-pds">"</span>https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> load cell type annotation function</span>
source(<span class="pl-s"><span class="pl-pds">"</span>https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R<span class="pl-pds">"</span></span>)

<span class="pl-v">db_</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx<span class="pl-pds">"</span></span>;
<span class="pl-c"><span class="pl-c">#</span>tissue = c("Immune system") # e.g. Immune system,Pancreas,Liver,Eye,Kidney,Brain,Lung,Adrenal,Heart,Intestine,Muscle,Placenta,Spleen,Stomach,Thymus </span>
<span class="pl-v">tissue</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Immune system<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Pancreas<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Liver<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Kidney<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Intestine<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Placenta<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Spleen<span class="pl-pds">"</span></span>,
           <span class="pl-s"><span class="pl-pds">"</span>Stomach<span class="pl-pds">"</span></span>) 

<span class="pl-c"><span class="pl-c">#</span> prepare gene sets</span>
<span class="pl-v">gs_list</span> <span class="pl-k">=</span> gene_sets_prepare(<span class="pl-smi">db_</span>, <span class="pl-smi">tissue</span>)

<span class="pl-c"><span class="pl-c">#</span> get cell-type by cell matrix</span>
<span class="pl-v">es.max</span> <span class="pl-k">=</span> sctype_score(<span class="pl-v">scRNAseqData</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>[[<span class="pl-s"><span class="pl-pds">"</span>integrated<span class="pl-pds">"</span></span>]]<span class="pl-k">@</span><span class="pl-smi">scale.data</span>, <span class="pl-v">scaled</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
                      <span class="pl-v">gs</span> <span class="pl-k">=</span> <span class="pl-smi">gs_list</span><span class="pl-k">$</span><span class="pl-smi">gs_positive</span>, <span class="pl-v">gs2</span> <span class="pl-k">=</span> <span class="pl-smi">gs_list</span><span class="pl-k">$</span><span class="pl-smi">gs_negative</span>) 

<span class="pl-c"><span class="pl-c">#</span> merge by cluster</span>
<span class="pl-v">cL_resutls</span> <span class="pl-k">=</span> do.call(
  <span class="pl-s"><span class="pl-pds">"</span>rbind<span class="pl-pds">"</span></span>, lapply(unique(<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">seurat_clusters</span>), <span class="pl-k">function</span>(<span class="pl-smi">cl</span>){
  <span class="pl-v">es.max.cl</span> <span class="pl-k">=</span> sort(rowSums(<span class="pl-smi">es.max</span>[ ,rownames(<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>[<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">seurat_clusters</span><span class="pl-k">==</span><span class="pl-smi">cl</span>, ])]), <span class="pl-v">decreasing</span> <span class="pl-k">=</span> <span class="pl-k">!</span><span class="pl-c1">0</span>)
  head(<span class="pl-k">data.frame</span>(<span class="pl-v">cluster</span> <span class="pl-k">=</span> <span class="pl-smi">cl</span>, <span class="pl-v">type</span> <span class="pl-k">=</span> names(<span class="pl-smi">es.max.cl</span>), <span class="pl-v">scores</span> <span class="pl-k">=</span> <span class="pl-smi">es.max.cl</span>, <span class="pl-v">ncells</span> <span class="pl-k">=</span> sum(<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">seurat_clusters</span><span class="pl-k">==</span><span class="pl-smi">cl</span>)), <span class="pl-c1">10</span>)
}))


<span class="pl-v">sctype_scores</span> <span class="pl-k">=</span> <span class="pl-smi">cL_resutls</span> %<span class="pl-k">&gt;</span>% group_by(<span class="pl-smi">cluster</span>) %<span class="pl-k">&gt;</span>% top_n(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, <span class="pl-v">wt</span> <span class="pl-k">=</span> <span class="pl-smi">scores</span>)

<span class="pl-c"><span class="pl-c">#</span> set low-confident (low ScType score) clusters to "unknown"</span>
<span class="pl-smi">sctype_scores</span><span class="pl-k">$</span><span class="pl-smi">type</span>[as.numeric(as.character(<span class="pl-smi">sctype_scores</span><span class="pl-k">$</span><span class="pl-smi">scores</span>)) <span class="pl-k">&lt;</span> <span class="pl-smi">sctype_scores</span><span class="pl-k">$</span><span class="pl-smi">ncells</span><span class="pl-k">/</span><span class="pl-c1">4</span>] <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Unknown<span class="pl-pds">"</span></span>
print(<span class="pl-smi">sctype_scores</span>[,<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">3</span>])



<span class="pl-c"><span class="pl-c">#</span> overlay the identified cell types on UMAP plot:</span>

<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-v">customclassif</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>
<span class="pl-k">for</span>(<span class="pl-smi">j</span> <span class="pl-k">in</span> unique(<span class="pl-smi">sctype_scores</span><span class="pl-k">$</span><span class="pl-smi">cluster</span>)){
  <span class="pl-v">cl_type</span> <span class="pl-k">=</span> <span class="pl-smi">sctype_scores</span>[<span class="pl-smi">sctype_scores</span><span class="pl-k">$</span><span class="pl-smi">cluster</span><span class="pl-k">==</span><span class="pl-smi">j</span>,]; 
  <span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">customclassif</span>[<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">seurat_clusters</span> <span class="pl-k">==</span> <span class="pl-smi">j</span>] <span class="pl-k">=</span> as.character(<span class="pl-smi">cl_type</span><span class="pl-k">$</span><span class="pl-smi">type</span>[<span class="pl-c1">1</span>])
}

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_with_label_scType.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">600</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
        <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
        <span class="pl-v">group.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>customclassif<span class="pl-pds">'</span></span>)        

dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_with_label_scType.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_with_label_scType.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_with_label_scType.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      8) MIBC vs. NMIBC
                     </h1>
                     <a aria-label="Permalink: 8) MIBC vs. NMIBC" class="anchor" href="#8-mibc-vs-nmibc" id="user-content-8-mibc-vs-nmibc">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     There are four NMIBC samples(SRR12603790,SRR12603789,SRR12603787,SRR12603786) and four MIBC samples(SRR12603785,SRR12603784,SRR12603783,SRR12603782) in the dataset. Comparing these two group against each other could have given a clue on the invasion cell type signature.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# load object
load("~/scRNA/seurat_filtered.RData")


# reading sample information
sampleInformation &lt;- read.csv("~/scRNA/sampleInfo.csv")

filtered_seurat@meta.data &lt;- left_join(filtered_seurat@meta.data, sampleInformation, by = "orig.ident")
# save meta data as a df
metaData &lt;- filtered_seurat@meta.data


# setting ident
Idents(filtered_seurat) &lt;- "sample"

# keeping only cancer samples
blca_seurat &lt;- subset(filtered_seurat, idents = "BLCA")
rm(filtered_seurat)

# Repair the meta.data
blca_seurat@meta.data$cells &lt;- blca_seurat@assays$RNA@data@Dimnames[[2]]
blca_seurat@meta.data$orig.ident &lt;- substr(blca_seurat@assays$RNA@data@Dimnames[[2]],1,11)
blca_seurat@meta.data$seq_folder &lt;- substr(blca_seurat@assays$RNA@data@Dimnames[[2]],1,11)
blca_seurat@meta.data$sample &lt;- "BLCA"

# set rownames
rownames(blca_seurat@meta.data) &lt;- blca_seurat@meta.data$cells


# remove columns fillied by NA
col_remove &lt;- c("nUMI", "nGene", "log10GenesPerUMI", "gender", "age", "Grade", "Invasiveness", "Surgery_Type", "Tumor_size_cm", "mitoRatio")

# 
blca_seurat@meta.data &lt;- blca_seurat@meta.data[, !(colnames(blca_seurat@meta.data) %in% col_remove)]

# adding more meta.data again
blca_seurat@meta.data &lt;- left_join(blca_seurat@meta.data, sampleInformation, by = "orig.ident")


# Setting metadata rownames as the column names of expression matrix
rownames(blca_seurat@meta.data) &lt;- blca_seurat@meta.data$cells

# SCTransform

# Split seurat object by group

# setting ident
Idents(blca_seurat) &lt;- "Invasiveness"
blca_split &lt;- SplitObject(blca_seurat)


blca_split 
#$Invasive
#An object of class Seurat 
#29686 features across 41242 samples within 1 assay 
#Active assay: RNA (29686 features, 0 variable features)
#
#$Noninvasive
#An object of class Seurat 
#29686 features across 35891 samples within 1 assay 
#Active assay: RNA (29686 features, 0 variable features)


# Repairing meta.data for each object in the splitted obj:

for(i in 1:length(blca_split)){
     print(blca_split[[i]])
     obj = blca_split[[i]]
     obj@meta.data$cells &lt;- obj@assays$RNA@data@Dimnames[[2]]
     obj@meta.data &lt;- obj@meta.data[, c(2,3,5)]
     obj@meta.data$orig.ident &lt;- substr(obj@assays$RNA@data@Dimnames[[2]],1,11)
     obj@meta.data$seq_folder &lt;- substr(obj@assays$RNA@data@Dimnames[[2]],1,11)
     obj@meta.data &lt;- left_join(obj@meta.data, sampleInformation, by = "orig.ident")
     obj@meta.data &lt;- left_join(obj@meta.data, metaData[,c(8,9)], by = "cells")
     # repairng metadata rownames
     if(all(obj@meta.data$cells == colnames(obj@assays$RNA@counts))){
     rownames(obj@meta.data) &lt;- obj@meta.data$cells
     }
     blca_split[[i]] = obj
 }

#saveRDS(blca_split, "blca_split.rds")
# then normalize by SCTansform
# orig.ident is slected ti be regressed out because each sample was sequenced in seperate batch

for (i in 1:length(blca_split)) {
    blca_split[[i]] &lt;- SCTransform(blca_split[[i]], vars.to.regress = c("orig.ident"), vst.flavor = "v2")
    }

# Integration : integrating samples belong to one group

# Select the most variable features to use for integration
integ_features &lt;- SelectIntegrationFeatures(object.list = blca_split, 
                                            nfeatures = 3000) 


# Prepare the SCT list object for integration
split_seurat &lt;- PrepSCTIntegration(object.list = blca_split, 
                                   anchor.features = integ_features)
                                   
# Find best buddies (using canonical correlation analysis or CCA) - can take a while to run
integ_anchors &lt;- FindIntegrationAnchors(object.list = split_seurat, 
                                        normalization.method = "SCT", 
                                        anchor.features = integ_features)
# Integrate across conditions
seurat_integrated &lt;- IntegrateData(anchorset = integ_anchors, 
                                   normalization.method = "SCT")
                                   

# Integration check
# Run PCA
seurat_integrated &lt;- RunPCA(object = seurat_integrated, verbose = TRUE)

# Plot PCA
png(filename = "blca_PCA_integrated.png", width = 16, height = 8.135, units = "in", res = 300)
PCAPlot(seurat_integrated,
        split.by = "Invasiveness")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> load object</span>
load(<span class="pl-s"><span class="pl-pds">"</span>~/scRNA/seurat_filtered.RData<span class="pl-pds">"</span></span>)


<span class="pl-c"><span class="pl-c">#</span> reading sample information</span>
<span class="pl-smi">sampleInformation</span> <span class="pl-k">&lt;-</span> read.csv(<span class="pl-s"><span class="pl-pds">"</span>~/scRNA/sampleInfo.csv<span class="pl-pds">"</span></span>)

<span class="pl-smi">filtered_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> left_join(<span class="pl-smi">filtered_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>, <span class="pl-smi">sampleInformation</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> save meta data as a df</span>
<span class="pl-smi">metaData</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">filtered_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>


<span class="pl-c"><span class="pl-c">#</span> setting ident</span>
Idents(<span class="pl-smi">filtered_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>sample<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span> keeping only cancer samples</span>
<span class="pl-smi">blca_seurat</span> <span class="pl-k">&lt;-</span> subset(<span class="pl-smi">filtered_seurat</span>, <span class="pl-v">idents</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>)
rm(<span class="pl-smi">filtered_seurat</span>)

<span class="pl-c"><span class="pl-c">#</span> Repair the meta.data</span>
<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">assays</span><span class="pl-k">$</span><span class="pl-smi">RNA</span><span class="pl-k">@</span><span class="pl-smi">data</span><span class="pl-k">@</span><span class="pl-smi">Dimnames</span>[[<span class="pl-c1">2</span>]]
<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">orig.ident</span> <span class="pl-k">&lt;-</span> substr(<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">assays</span><span class="pl-k">$</span><span class="pl-smi">RNA</span><span class="pl-k">@</span><span class="pl-smi">data</span><span class="pl-k">@</span><span class="pl-smi">Dimnames</span>[[<span class="pl-c1">2</span>]],<span class="pl-c1">1</span>,<span class="pl-c1">11</span>)
<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">seq_folder</span> <span class="pl-k">&lt;-</span> substr(<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">assays</span><span class="pl-k">$</span><span class="pl-smi">RNA</span><span class="pl-k">@</span><span class="pl-smi">data</span><span class="pl-k">@</span><span class="pl-smi">Dimnames</span>[[<span class="pl-c1">2</span>]],<span class="pl-c1">1</span>,<span class="pl-c1">11</span>)
<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">sample</span> <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>BLCA<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span> set rownames</span>
rownames(<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span>


<span class="pl-c"><span class="pl-c">#</span> remove columns fillied by NA</span>
<span class="pl-smi">col_remove</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>nUMI<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>nGene<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>log10GenesPerUMI<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gender<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>age<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Grade<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Surgery_Type<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Tumor_size_cm<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>mitoRatio<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> </span>
<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>[, <span class="pl-k">!</span>(colnames(<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>) <span class="pl-k">%in%</span> <span class="pl-smi">col_remove</span>)]

<span class="pl-c"><span class="pl-c">#</span> adding more meta.data again</span>
<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> left_join(<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>, <span class="pl-smi">sampleInformation</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>)


<span class="pl-c"><span class="pl-c">#</span> Setting metadata rownames as the column names of expression matrix</span>
rownames(<span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">blca_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span>

<span class="pl-c"><span class="pl-c">#</span> SCTransform</span>

<span class="pl-c"><span class="pl-c">#</span> Split seurat object by group</span>

<span class="pl-c"><span class="pl-c">#</span> setting ident</span>
Idents(<span class="pl-smi">blca_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>
<span class="pl-smi">blca_split</span> <span class="pl-k">&lt;-</span> SplitObject(<span class="pl-smi">blca_seurat</span>)


<span class="pl-smi">blca_split</span> 
<span class="pl-c"><span class="pl-c">#</span>$Invasive</span>
<span class="pl-c"><span class="pl-c">#</span>An object of class Seurat </span>
<span class="pl-c"><span class="pl-c">#</span>29686 features across 41242 samples within 1 assay </span>
<span class="pl-c"><span class="pl-c">#</span>Active assay: RNA (29686 features, 0 variable features)</span>
<span class="pl-c"><span class="pl-c">#</span></span>
<span class="pl-c"><span class="pl-c">#</span>$Noninvasive</span>
<span class="pl-c"><span class="pl-c">#</span>An object of class Seurat </span>
<span class="pl-c"><span class="pl-c">#</span>29686 features across 35891 samples within 1 assay </span>
<span class="pl-c"><span class="pl-c">#</span>Active assay: RNA (29686 features, 0 variable features)</span>


<span class="pl-c"><span class="pl-c">#</span> Repairing meta.data for each object in the splitted obj:</span>

<span class="pl-k">for</span>(<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>length(<span class="pl-smi">blca_split</span>)){
     print(<span class="pl-smi">blca_split</span>[[<span class="pl-smi">i</span>]])
     <span class="pl-v">obj</span> <span class="pl-k">=</span> <span class="pl-smi">blca_split</span>[[<span class="pl-smi">i</span>]]
     <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">assays</span><span class="pl-k">$</span><span class="pl-smi">RNA</span><span class="pl-k">@</span><span class="pl-smi">data</span><span class="pl-k">@</span><span class="pl-smi">Dimnames</span>[[<span class="pl-c1">2</span>]]
     <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>[, c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>,<span class="pl-c1">5</span>)]
     <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">orig.ident</span> <span class="pl-k">&lt;-</span> substr(<span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">assays</span><span class="pl-k">$</span><span class="pl-smi">RNA</span><span class="pl-k">@</span><span class="pl-smi">data</span><span class="pl-k">@</span><span class="pl-smi">Dimnames</span>[[<span class="pl-c1">2</span>]],<span class="pl-c1">1</span>,<span class="pl-c1">11</span>)
     <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">seq_folder</span> <span class="pl-k">&lt;-</span> substr(<span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">assays</span><span class="pl-k">$</span><span class="pl-smi">RNA</span><span class="pl-k">@</span><span class="pl-smi">data</span><span class="pl-k">@</span><span class="pl-smi">Dimnames</span>[[<span class="pl-c1">2</span>]],<span class="pl-c1">1</span>,<span class="pl-c1">11</span>)
     <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> left_join(<span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>, <span class="pl-smi">sampleInformation</span>, <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>)
     <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> left_join(<span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>, <span class="pl-smi">metaData</span>[,c(<span class="pl-c1">8</span>,<span class="pl-c1">9</span>)], <span class="pl-v">by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cells<span class="pl-pds">"</span></span>)
     <span class="pl-c"><span class="pl-c">#</span> repairng metadata rownames</span>
     <span class="pl-k">if</span>(all(<span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span> <span class="pl-k">==</span> colnames(<span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">assays</span><span class="pl-k">$</span><span class="pl-smi">RNA</span><span class="pl-k">@</span><span class="pl-smi">counts</span>))){
     rownames(<span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">obj</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span>
     }
     <span class="pl-smi">blca_split</span>[[<span class="pl-smi">i</span>]] <span class="pl-k">=</span> <span class="pl-smi">obj</span>
 }

<span class="pl-c"><span class="pl-c">#</span>saveRDS(blca_split, "blca_split.rds")</span>
<span class="pl-c"><span class="pl-c">#</span> then normalize by SCTansform</span>
<span class="pl-c"><span class="pl-c">#</span> orig.ident is slected ti be regressed out because each sample was sequenced in seperate batch</span>

<span class="pl-k">for</span> (<span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>length(<span class="pl-smi">blca_split</span>)) {
    <span class="pl-smi">blca_split</span>[[<span class="pl-smi">i</span>]] <span class="pl-k">&lt;-</span> SCTransform(<span class="pl-smi">blca_split</span>[[<span class="pl-smi">i</span>]], <span class="pl-v">vars.to.regress</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>), <span class="pl-v">vst.flavor</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>v2<span class="pl-pds">"</span></span>)
    }

<span class="pl-c"><span class="pl-c">#</span> Integration : integrating samples belong to one group</span>

<span class="pl-c"><span class="pl-c">#</span> Select the most variable features to use for integration</span>
<span class="pl-smi">integ_features</span> <span class="pl-k">&lt;-</span> SelectIntegrationFeatures(<span class="pl-v">object.list</span> <span class="pl-k">=</span> <span class="pl-smi">blca_split</span>, 
                                            <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">3000</span>) 


<span class="pl-c"><span class="pl-c">#</span> Prepare the SCT list object for integration</span>
<span class="pl-smi">split_seurat</span> <span class="pl-k">&lt;-</span> PrepSCTIntegration(<span class="pl-v">object.list</span> <span class="pl-k">=</span> <span class="pl-smi">blca_split</span>, 
                                   <span class="pl-v">anchor.features</span> <span class="pl-k">=</span> <span class="pl-smi">integ_features</span>)
                                   
<span class="pl-c"><span class="pl-c">#</span> Find best buddies (using canonical correlation analysis or CCA) - can take a while to run</span>
<span class="pl-smi">integ_anchors</span> <span class="pl-k">&lt;-</span> FindIntegrationAnchors(<span class="pl-v">object.list</span> <span class="pl-k">=</span> <span class="pl-smi">split_seurat</span>, 
                                        <span class="pl-v">normalization.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, 
                                        <span class="pl-v">anchor.features</span> <span class="pl-k">=</span> <span class="pl-smi">integ_features</span>)
<span class="pl-c"><span class="pl-c">#</span> Integrate across conditions</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> IntegrateData(<span class="pl-v">anchorset</span> <span class="pl-k">=</span> <span class="pl-smi">integ_anchors</span>, 
                                   <span class="pl-v">normalization.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>)
                                   

<span class="pl-c"><span class="pl-c">#</span> Integration check</span>
<span class="pl-c"><span class="pl-c">#</span> Run PCA</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> RunPCA(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span> Plot PCA</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>blca_PCA_integrated.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
PCAPlot(<span class="pl-smi">seurat_integrated</span>,
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_PCA_integrated.png" rel="noopener noreferrer" target="_blank">
                      <img alt="blca_PCA_integrated.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_PCA_integrated.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
# Run UMAP
seurat_integrated &lt;- RunUMAP(seurat_integrated, 
                             dims = 1:40,
                             reduction = "pca",
                             verbose = TRUE)

# Plot UMAP 
png(filename = "blca_UMAP_integrated.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated, split.by = "Invasiveness")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Run UMAP</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> RunUMAP(<span class="pl-smi">seurat_integrated</span>, 
                             <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">40</span>,
                             <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>,
                             <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span> Plot UMAP </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>blca_UMAP_integrated.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>, <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_UMAP_integrated.png" rel="noopener noreferrer" target="_blank">
                      <img alt="blca_UMAP_integrated.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_UMAP_integrated.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Cluster the cells

# to check what is active assay
DefaultAssay(object = seurat_integrated)

# Determine the K-nearest neighbor graph
seurat_integrated &lt;- FindNeighbors(object = seurat_integrated, 
                                dims = 1:18)
                                
#Find clusters
# Determine the clusters for various resolutions                                
seurat_integrated &lt;- FindClusters(object = seurat_integrated,
                               resolution = c(0.2,0.4, 0.6, 0.8, 1.0, 1.4))
                               

# Explore resolutions
head(seurat_integrated@meta.data)

# Assign identity of clusters
Idents(object = seurat_integrated) &lt;- "integrated_snn_res.0.2"

# Plot the UMAP
png(filename = "blca_umap_cluster_with_label.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated,
        reduction = "umap",
        label = TRUE,
        label.size = 6)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Cluster the cells</span>

<span class="pl-c"><span class="pl-c">#</span> to check what is active assay</span>
DefaultAssay(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>)

<span class="pl-c"><span class="pl-c">#</span> Determine the K-nearest neighbor graph</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> FindNeighbors(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                                <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">18</span>)
                                
<span class="pl-c"><span class="pl-c">#</span>Find clusters</span>
<span class="pl-c"><span class="pl-c">#</span> Determine the clusters for various resolutions                                </span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> FindClusters(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>,
                               <span class="pl-v">resolution</span> <span class="pl-k">=</span> c(<span class="pl-c1">0.2</span>,<span class="pl-c1">0.4</span>, <span class="pl-c1">0.6</span>, <span class="pl-c1">0.8</span>, <span class="pl-c1">1.0</span>, <span class="pl-c1">1.4</span>))
                               

<span class="pl-c"><span class="pl-c">#</span> Explore resolutions</span>
head(<span class="pl-smi">seurat_integrated</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)

<span class="pl-c"><span class="pl-c">#</span> Assign identity of clusters</span>
Idents(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>) <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>integrated_snn_res.0.2<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span> Plot the UMAP</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>blca_umap_cluster_with_label.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_umap_cluster_with_label.png" rel="noopener noreferrer" target="_blank">
                      <img alt="blca_umap_cluster_with_label.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_umap_cluster_with_label.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# UMAP of cells in each cluster by invasiveness group
# This would allow us to see condition specefic clusters
png(filename = "blca_umap_cluster_sample.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated, 
        label = TRUE, 
        split.by = "Invasiveness")  + NoLegend()
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> UMAP of cells in each cluster by invasiveness group</span>
<span class="pl-c"><span class="pl-c">#</span> This would allow us to see condition specefic clusters</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>blca_umap_cluster_sample.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>)  <span class="pl-k">+</span> NoLegend()
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_umap_cluster_sample.png" rel="noopener noreferrer" target="_blank">
                      <img alt="blca_umap_cluster_sample.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_umap_cluster_sample.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Explore whether clusters segregate by cell cycle phase

png(filename = "grade_umap_cluster_cell_cucle.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(seurat_integrated,
        label = TRUE, 
        split.by = "Grade")  + NoLegend()
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Explore whether clusters segregate by cell cycle phase</span>

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>grade_umap_cluster_cell_cucle.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">seurat_integrated</span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, 
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Grade<span class="pl-pds">"</span></span>)  <span class="pl-k">+</span> NoLegend()
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/grade_umap_cluster_cell_cucle.png" rel="noopener noreferrer" target="_blank">
                      <img alt="grade_umap_cluster_cell_cucle.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/grade_umap_cluster_cell_cucle.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#Marker identification
                  

# adding annotation to the genes:
# Connect to AnnotationHub
ah &lt;- AnnotationHub()

# Access the Ensembl database for organism
ahDb &lt;- query(ah, 
              pattern = c("Homo sapiens", "EnsDb"), 
              ignore.case = TRUE)

# Acquire the latest annotation files
id &lt;- ahDb %&gt;%
        mcols() %&gt;%
        rownames() %&gt;%
        tail(n = 1)

# Download the appropriate Ensembldb database
edb &lt;- ah[[id]]

# Extract gene-level information from database
annotations &lt;- genes(edb, 
                     return.type = "data.frame")

# Select annotations of interest
annotations &lt;- annotations %&gt;%
        dplyr::select(gene_id, gene_name, seq_name, gene_biotype, description)
        ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>Marker identification</span>
                  

<span class="pl-c"><span class="pl-c">#</span> adding annotation to the genes:</span>
<span class="pl-c"><span class="pl-c">#</span> Connect to AnnotationHub</span>
<span class="pl-smi">ah</span> <span class="pl-k">&lt;-</span> AnnotationHub()

<span class="pl-c"><span class="pl-c">#</span> Access the Ensembl database for organism</span>
<span class="pl-smi">ahDb</span> <span class="pl-k">&lt;-</span> query(<span class="pl-smi">ah</span>, 
              <span class="pl-v">pattern</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>Homo sapiens<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>EnsDb<span class="pl-pds">"</span></span>), 
              <span class="pl-v">ignore.case</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

<span class="pl-c"><span class="pl-c">#</span> Acquire the latest annotation files</span>
<span class="pl-smi">id</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">ahDb</span> %<span class="pl-k">&gt;</span>%
        mcols() %<span class="pl-k">&gt;</span>%
        rownames() %<span class="pl-k">&gt;</span>%
        tail(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>)

<span class="pl-c"><span class="pl-c">#</span> Download the appropriate Ensembldb database</span>
<span class="pl-smi">edb</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">ah</span>[[<span class="pl-smi">id</span>]]

<span class="pl-c"><span class="pl-c">#</span> Extract gene-level information from database</span>
<span class="pl-smi">annotations</span> <span class="pl-k">&lt;-</span> genes(<span class="pl-smi">edb</span>, 
                     <span class="pl-k">return</span><span class="pl-v">.type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>data.frame<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Select annotations of interest</span>
<span class="pl-smi">annotations</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">annotations</span> %<span class="pl-k">&gt;</span>%
        <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">gene_id</span>, <span class="pl-smi">gene_name</span>, <span class="pl-smi">seq_name</span>, <span class="pl-smi">gene_biotype</span>, <span class="pl-smi">description</span>)
        </pre>
                    </div>
                    <p dir="auto">
                     Finding markers for all clusters:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Create function to get conserved markers for any given cluster
get_conserved &lt;- function(cluster){
  tryCatch({
  FindConservedMarkers(seurat_integrated,
                       ident.1 = cluster,
                       grouping.var = &quot;Invasiveness&quot;,
                       only.pos = TRUE,
                       logfc.threshold = 0.60) %&gt;%
    rownames_to_column(var = &quot;gene&quot;) %&gt;%
    left_join(y = unique(annotations[, c(&quot;gene_name&quot;, &quot;description&quot;)]),
               by = c(&quot;gene&quot; = &quot;gene_name&quot;)) %&gt;%
    cbind(cluster_id = cluster, .)
  },
  error = function(e){
  message(paste0(&quot;Error: &quot;, e$message))
    }
   )
  }
  
# this function can be an argument for 'map_dfr' function :
# Iterate function across desired clusters
conserved_markers &lt;- map_dfr(c(0:20), get_conserved)

# Extract top 10 markers per cluster
top10 &lt;- conserved_markers %&gt;% 
  mutate(avg_fc = (Normal_avg_log2FC + BLCA_avg_log2FC) /2) %&gt;% 
  group_by(cluster_id) %&gt;% 
  top_n(n = 10, 
        wt = avg_fc)
        
head(top10)

#data.table::fwrite(top10, &quot;top10_conserved_markers.csv&quot;)
#group based on the clsuter id to define cell types:
data_grouped &lt;- top10 %&gt;%
  group_by(cluster_id) %&gt;%
  summarize(gene = paste(gene, collapse = &quot;,&quot;))
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Create function to get conserved markers for any given cluster</span>
<span class="pl-en">get_conserved</span> <span class="pl-k">&lt;-</span> <span class="pl-k">function</span>(<span class="pl-smi">cluster</span>){
  tryCatch({
  FindConservedMarkers(<span class="pl-smi">seurat_integrated</span>,
                       <span class="pl-v">ident.1</span> <span class="pl-k">=</span> <span class="pl-smi">cluster</span>,
                       <span class="pl-v">grouping.var</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>,
                       <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                       <span class="pl-v">logfc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">0.60</span>) %<span class="pl-k">&gt;</span>%
    rownames_to_column(<span class="pl-v">var</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gene<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
    left_join(<span class="pl-v">y</span> <span class="pl-k">=</span> unique(<span class="pl-smi">annotations</span>[, c(<span class="pl-s"><span class="pl-pds">"</span>gene_name<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>description<span class="pl-pds">"</span></span>)]),
               <span class="pl-v">by</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>gene<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>gene_name<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>%
    cbind(<span class="pl-v">cluster_id</span> <span class="pl-k">=</span> <span class="pl-smi">cluster</span>, <span class="pl-smi">.</span>)
  },
  <span class="pl-en">error</span> <span class="pl-k">=</span> <span class="pl-k">function</span>(<span class="pl-smi">e</span>){
  message(paste0(<span class="pl-s"><span class="pl-pds">"</span>Error: <span class="pl-pds">"</span></span>, <span class="pl-smi">e</span><span class="pl-k">$</span><span class="pl-smi">message</span>))
    }
   )
  }
  
<span class="pl-c"><span class="pl-c">#</span> this function can be an argument for 'map_dfr' function :</span>
<span class="pl-c"><span class="pl-c">#</span> Iterate function across desired clusters</span>
<span class="pl-smi">conserved_markers</span> <span class="pl-k">&lt;-</span> map_dfr(c(<span class="pl-c1">0</span><span class="pl-k">:</span><span class="pl-c1">20</span>), <span class="pl-smi">get_conserved</span>)

<span class="pl-c"><span class="pl-c">#</span> Extract top 10 markers per cluster</span>
<span class="pl-smi">top10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">conserved_markers</span> %<span class="pl-k">&gt;</span>% 
  mutate(<span class="pl-v">avg_fc</span> <span class="pl-k">=</span> (<span class="pl-smi">Normal_avg_log2FC</span> <span class="pl-k">+</span> <span class="pl-smi">BLCA_avg_log2FC</span>) <span class="pl-k">/</span><span class="pl-c1">2</span>) %<span class="pl-k">&gt;</span>% 
  group_by(<span class="pl-smi">cluster_id</span>) %<span class="pl-k">&gt;</span>% 
  top_n(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, 
        <span class="pl-v">wt</span> <span class="pl-k">=</span> <span class="pl-smi">avg_fc</span>)
        
head(<span class="pl-smi">top10</span>)

<span class="pl-c"><span class="pl-c">#</span>data.table::fwrite(top10, "top10_conserved_markers.csv")</span>
<span class="pl-c"><span class="pl-c">#</span>group based on the clsuter id to define cell types:</span>
<span class="pl-smi">data_grouped</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">top10</span> %<span class="pl-k">&gt;</span>%
  group_by(<span class="pl-smi">cluster_id</span>) %<span class="pl-k">&gt;</span>%
  summarize(<span class="pl-v">gene</span> <span class="pl-k">=</span> paste(<span class="pl-smi">gene</span>, <span class="pl-v">collapse</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>))
</pre>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        cluster_id
                       </th>
                       <th>
                        genes
                       </th>
                       <th>
                        cell_type (PanglaoDB + ChatGPT)
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        0
                       </td>
                       <td>
                        SFN,S100A2,CXCL8,CLDN4,AQP3,PHLDA2,EMP1,KRT13,KRT17,SLPI
                       </td>
                       <td>
                        basal cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        1
                       </td>
                       <td>
                        SPINK1,CD24,UBE2C,UCA1,ADIRF,C15orf48,CSTB,S100A9,CCND1,HIST1H4C
                       </td>
                       <td>
                        CCND1 + cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        2
                       </td>
                       <td>
                        CD52,CXCR4,CD3D,CD69,IFNG,TRAC,IL32,CCL5,CCL4,NKG7
                       </td>
                       <td>
                        T cells/NK cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        3
                       </td>
                       <td>
                        PSCA,LY6D,KRT7,DHRS2,KRT13,HES1,KRT20,CRH,GCLC,RPS19
                       </td>
                       <td>
                        basal/squamous-like cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        4
                       </td>
                       <td>
                        HSPG2,LDB2,IGFBP7,SPARCL1,GNG11,VWF,FLT1,RAMP2,INSR,PLVAP
                       </td>
                       <td>
                        ND
                       </td>
                      </tr>
                      <tr>
                       <td>
                        5
                       </td>
                       <td>
                        CD52,CD2,LINC01871,GZMA,TRBC2,CD3D,TRAC,CORO1A,CCL5,NKG7
                       </td>
                       <td>
                        cytotoxic T cell
                       </td>
                      </tr>
                      <tr>
                       <td>
                        6
                       </td>
                       <td>
                        FABP4,CTSE,AKR1C3,PLA2G2A,FABP5,CRTAC1,GSTM3,KRT7,S100A6,CLU
                       </td>
                       <td>
                        CAFs
                       </td>
                      </tr>
                      <tr>
                       <td>
                        7
                       </td>
                       <td>
                        SNTG1,ZFPM2-AS1,RHEX,MT-ATP6,MT-ND5,CCSER1,FRY,SPINK1
                       </td>
                       <td>
                        ND
                       </td>
                      </tr>
                      <tr>
                       <td>
                        8
                       </td>
                       <td>
                        RGS5,RGS5,IGFBP7,CALD1,ACTA2,TAGLN,THY1,NDUFA4L2,MYL9,BGN
                       </td>
                       <td>
                        fibroblasts/myofibroblasts
                       </td>
                      </tr>
                      <tr>
                       <td>
                        10
                       </td>
                       <td>
                        C1QA,C1QB,CD74,HLA-DRA,HLA-DRB1,HLA-DPA1,HLA-DPB1,LYZ,TYROBP,APOE
                       </td>
                       <td>
                        APCs
                       </td>
                      </tr>
                      <tr>
                       <td>
                        11
                       </td>
                       <td>
                        COL3A1,CXCL14,COL1A2,RARRES2,PTGDS,MGP,LUM,DCN,MT2A,COL1A1
                       </td>
                       <td>
                        Fbroblasts(?)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        13
                       </td>
                       <td>
                        CPA3,HPGDS,LTC4S,MS4A2,CTSG,TPSB2,TPSAB1,CD69,SRGN,RGS1
                       </td>
                       <td>
                        Mast cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        14
                       </td>
                       <td>
                        AFF3,BANK1,LTB,HLA-DRA,CD79A,CD37,CD74,CD52,CXCR4,CD83
                       </td>
                       <td>
                        B cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        17
                       </td>
                       <td>
                        JCHAIN,IGHG1,IGHG2,IGKC,IGHG3,IGLC3,IGLC2,IGHA1,IGHM,IGHA2
                       </td>
                       <td>
                        B cells
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Plot interesting marker gene expression 
png(filename = &quot;cd24_cells.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated,
            features = c('SPINK1','CD24','UBE2C','UCA1','ADIRF','C15orf48','CSTB','S100A9','CCND1','HIST1H4C'),
            order = TRUE,
            min.cutoff = &quot;q10&quot;,
            label = TRUE,
            repel = TRUE)

dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Plot interesting marker gene expression </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cd24_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>,
            <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">'</span>SPINK1<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>CD24<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>UBE2C<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>UCA1<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>ADIRF<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>C15orf48<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>CSTB<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>S100A9<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>CCND1<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>HIST1H4C<span class="pl-pds">'</span></span>),
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>q10<span class="pl-pds">"</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cd24_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="cd24_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cd24_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="png(filename = &quot;lyz_cells.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = seurat_integrated,
            features = c('C1QA','C1QB','CD74','HLA-DRA','HLA-DRB1','HLA-DPA1','HLA-DPB1','LYZ','TYROBP','APOE'),
            order = TRUE,
            min.cutoff = &quot;q10&quot;,
            label = TRUE,
            repel = TRUE)

dev.off()" dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>lyz_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>,
            <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">'</span>C1QA<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>C1QB<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>CD74<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>HLA-DRA<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>HLA-DRB1<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>HLA-DPA1<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>HLA-DPB1<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>LYZ<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>TYROBP<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>APOE<span class="pl-pds">'</span></span>),
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>q10<span class="pl-pds">"</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/lyz_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="lyz_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/lyz_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Rename all identities
seurat_integrated &lt;- RenameIdents(object = seurat_integrated, 
                               "0" = "Basal cells",
                               "1" = "CCND1 + cells", # NKG7 and GZMK
                               "2" = "T cells/NK cells",
                               "3" = "basal/squamous-like cells",
                               "4" = "ND",
                               "5" = "cytotoxic T cell",
                               "6" = "CAFs",
                               "7" = "ND",
                               "8" = "fibroblasts/myofibroblasts",
                               "9" = "ND",
                               "10" = "fibroblasts/myofibroblasts",
                               "11" = "Fibroblast(?)",
                               "12" = "ND",
                               "13" = "Mast cells",
                               "14" = "B cells",
                               "15" = "ND",
                               "16" = "ND",
                               "17" = "B cells", 
                               "18" = "ND", 
                               "19" = "ND",
                               "20" = "ND")

# Plot the UMAP withy new labells
png(filename = "blca_umap_with_label.png", width = 16, height = 8.135, units = "in", res = 600)
DimPlot(object = seurat_integrated, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE,
        split.by = "Invasiveness")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Rename all identities</span>
<span class="pl-smi">seurat_integrated</span> <span class="pl-k">&lt;-</span> RenameIdents(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
                               <span class="pl-s"><span class="pl-pds">"</span>0<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Basal cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>1<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>CCND1 + cells<span class="pl-pds">"</span></span>, <span class="pl-c"><span class="pl-c">#</span> NKG7 and GZMK</span>
                               <span class="pl-s"><span class="pl-pds">"</span>2<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>T cells/NK cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>3<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>basal/squamous-like cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>5<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cytotoxic T cell<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>6<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>CAFs<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>7<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>8<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>fibroblasts/myofibroblasts<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>9<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>10<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>fibroblasts/myofibroblasts<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>11<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Fibroblast(?)<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>12<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>13<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Mast cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>14<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>B cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>15<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>16<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>17<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>B cells<span class="pl-pds">"</span></span>, 
                               <span class="pl-s"><span class="pl-pds">"</span>18<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>, 
                               <span class="pl-s"><span class="pl-pds">"</span>19<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>20<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>ND<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Plot the UMAP withy new labells</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>blca_umap_with_label.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">600</span>)
DimPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_integrated</span>, 
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>,
        <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/blca_umap_with_label.png" rel="noopener noreferrer" target="_blank">
                      <img alt="blca_umap_with_label.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/blca_umap_with_label.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      9) Analyzing epithelial (EPCAM +) cells
                     </h1>
                     <a aria-label="Permalink: 9) Analyzing epithelial (EPCAM +) cells" class="anchor" href="#9-analyzing-epithelial-epcam--cells" id="user-content-9-analyzing-epithelial-epcam--cells">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     This section is based on what I can get from the paper, but I stay focused on cancer samples and will use harmony to do integration. Finally, I do trajectory analysis for all cells which identified as epithelial cells.
                    </p>
                    <p dir="auto">
                     "Cells with UMI numbersâ&lt;1000 or with over 10% mitochondrial-derived UMI counts were considered low-quality cells and were removed. To eliminate potential doublets, single cells with over 6000 genes detected were also filtered out. Finally, 52721 single cells remained, and they were applied in downstream analyses."
                    </p>
                    <p dir="auto">
                     "Since samples from eight patients were processed and sequenced in batches, the patient number was used to remove the potential batch effect."
                    </p>
                    <p dir="auto">
                     "epithelial (EPCAM+) cells; endothelial (CD31+) cells; two types of fibroblasts (COL1A1+)âiCAFs (PDGFRA+) and myo-CAFs (mCAFs) (RGS5+); B cells (CD79A+); myeloid cells (LYZ+); T cells (CD3D+); and mast cells (TPSAB1+)"
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#________________________Reading the files______________________#
# create a list of samples
samples &lt;- list.files(&quot;./filtered/&quot;)
#samples &lt;- samples[grepl('^filtered',samples,perl=T)]

# read files into Seurat objects
for (file in samples){
  print(paste0(file))
  seurat_data &lt;- Read10X(data.dir = paste0(&quot;./filtered/&quot;, file))
  seurat_obj &lt;- CreateSeuratObject(counts = seurat_data, 
                                   min.features = 100, 
                                   project = file)
  assign(file, seurat_obj)
}

# updated sample name 
samples_blca &lt;- samples[-c(1,2,9)]

# Now merging all objects into one Seurat obj

merged_seurat &lt;- merge(x = SRR12603782,
                       y = c(SRR12603783,
                             SRR12603784,
                             SRR12603785,
                             SRR12603786,
                             SRR12603787,
                             SRR12603789,
                             SRR12603790),
                       add.cell.id = samples_blca)
                       
#________________________Filteration____________________________#
# reading sampleInformation:
sampleInformation &lt;- read.csv(&quot;./sampleInfo.csv&quot;)

# Compute percent mito ratio
merged_seurat$mitoRatio &lt;- PercentageFeatureSet(object = merged_seurat, pattern = &quot;^MT-&quot;)
merged_seurat$mitoRatio &lt;- merged_seurat@meta.data$mitoRatio / 100

# adding cell column
merged_seurat$cells &lt;- rownames(merged_seurat@meta.data)
# merging with sample information
merged_seurat@meta.data &lt;- merge(merged_seurat@meta.data, sampleInformation)
# re-setting the rownames
rownames(merged_seurat@meta.data) &lt;- merged_seurat@meta.data$cells


# Filteration

filtered_seurat &lt;- subset(merged_seurat, 
                          subset= nCount_RNA &gt;= 1000 &amp;
                          nFeature_RNA &lt;= 6000 &amp; 
                          mitoRatio &lt; 0.10)

#________________________Integration using Harmony____________________________#
#integration using harmony need several steps to be undertaken:

# Perform log-normalization and feature selection, as well as SCT normalization on the global object
merged_seurat &lt;- filtered_seurat %&gt;%
    NormalizeData() %&gt;%
    FindVariableFeatures(selection.method = &quot;vst&quot;, nfeatures = 3000) %&gt;% 
    ScaleData() %&gt;%
    SCTransform(vars.to.regress = c(&quot;mitoRatio&quot;, &quot;orig.ident&quot;))

# Calculate PCs using variable features determined by SCTransform (3000 by default)
merged_seurat &lt;- RunPCA(merged_seurat, assay = &quot;SCT&quot;, npcs = 50)
merged_seurat &lt;- RunTSNE(merged_seurat, assay = &quot;SCT&quot;, npcs = 50)

# Integration
#install.packages(&quot;harmony&quot;)

library(harmony)

harmonized_seurat &lt;- RunHarmony(merged_seurat, 
				group.by.vars = c(&quot;orig.ident&quot;, &quot;gender&quot;, &quot;Surgery_Type&quot;), 
				reduction = &quot;pca&quot;, assay.use = &quot;SCT&quot;, reduction.save = &quot;harmony&quot;)" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>________________________Reading the files______________________#</span>
<span class="pl-c"><span class="pl-c">#</span> create a list of samples</span>
<span class="pl-smi">samples</span> <span class="pl-k">&lt;-</span> list.files(<span class="pl-s"><span class="pl-pds">"</span>./filtered/<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>samples &lt;- samples[grepl('^filtered',samples,perl=T)]</span>

<span class="pl-c"><span class="pl-c">#</span> read files into Seurat objects</span>
<span class="pl-k">for</span> (<span class="pl-smi">file</span> <span class="pl-k">in</span> <span class="pl-smi">samples</span>){
  print(paste0(<span class="pl-smi">file</span>))
  <span class="pl-smi">seurat_data</span> <span class="pl-k">&lt;-</span> Read10X(<span class="pl-v">data.dir</span> <span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>./filtered/<span class="pl-pds">"</span></span>, <span class="pl-smi">file</span>))
  <span class="pl-smi">seurat_obj</span> <span class="pl-k">&lt;-</span> CreateSeuratObject(<span class="pl-v">counts</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_data</span>, 
                                   <span class="pl-v">min.features</span> <span class="pl-k">=</span> <span class="pl-c1">100</span>, 
                                   <span class="pl-v">project</span> <span class="pl-k">=</span> <span class="pl-smi">file</span>)
  assign(<span class="pl-smi">file</span>, <span class="pl-smi">seurat_obj</span>)
}

<span class="pl-c"><span class="pl-c">#</span> updated sample name </span>
<span class="pl-smi">samples_blca</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">samples</span>[<span class="pl-k">-</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">2</span>,<span class="pl-c1">9</span>)]

<span class="pl-c"><span class="pl-c">#</span> Now merging all objects into one Seurat obj</span>

<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">SRR12603782</span>,
                       <span class="pl-v">y</span> <span class="pl-k">=</span> c(<span class="pl-smi">SRR12603783</span>,
                             <span class="pl-smi">SRR12603784</span>,
                             <span class="pl-smi">SRR12603785</span>,
                             <span class="pl-smi">SRR12603786</span>,
                             <span class="pl-smi">SRR12603787</span>,
                             <span class="pl-smi">SRR12603789</span>,
                             <span class="pl-smi">SRR12603790</span>),
                       <span class="pl-v">add.cell.id</span> <span class="pl-k">=</span> <span class="pl-smi">samples_blca</span>)
                       
<span class="pl-c"><span class="pl-c">#</span>________________________Filteration____________________________#</span>
<span class="pl-c"><span class="pl-c">#</span> reading sampleInformation:</span>
<span class="pl-smi">sampleInformation</span> <span class="pl-k">&lt;-</span> read.csv(<span class="pl-s"><span class="pl-pds">"</span>./sampleInfo.csv<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Compute percent mito ratio</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;-</span> PercentageFeatureSet(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">merged_seurat</span>, <span class="pl-v">pattern</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>^MT-<span class="pl-pds">"</span></span>)
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">/</span> <span class="pl-c1">100</span>

<span class="pl-c"><span class="pl-c">#</span> adding cell column</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">cells</span> <span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)
<span class="pl-c"><span class="pl-c">#</span> merging with sample information</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>, <span class="pl-smi">sampleInformation</span>)
<span class="pl-c"><span class="pl-c">#</span> re-setting the rownames</span>
rownames(<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span>


<span class="pl-c"><span class="pl-c">#</span> Filteration</span>

<span class="pl-smi">filtered_seurat</span> <span class="pl-k">&lt;-</span> subset(<span class="pl-smi">merged_seurat</span>, 
                          <span class="pl-v">subset</span><span class="pl-k">=</span> <span class="pl-smi">nCount_RNA</span> <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">1000</span> <span class="pl-k">&amp;</span>
                          <span class="pl-smi">nFeature_RNA</span> <span class="pl-k">&lt;</span><span class="pl-k">=</span> <span class="pl-c1">6000</span> <span class="pl-k">&amp;</span> 
                          <span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0.10</span>)

<span class="pl-c"><span class="pl-c">#</span>________________________Integration using Harmony____________________________#</span>
<span class="pl-c"><span class="pl-c">#</span>integration using harmony need several steps to be undertaken:</span>

<span class="pl-c"><span class="pl-c">#</span> Perform log-normalization and feature selection, as well as SCT normalization on the global object</span>
<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">filtered_seurat</span> %<span class="pl-k">&gt;</span>%
    NormalizeData() %<span class="pl-k">&gt;</span>%
    FindVariableFeatures(<span class="pl-v">selection.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vst<span class="pl-pds">"</span></span>, <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">3000</span>) %<span class="pl-k">&gt;</span>% 
    ScaleData() %<span class="pl-k">&gt;</span>%
    SCTransform(<span class="pl-v">vars.to.regress</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>mitoRatio<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>))

<span class="pl-c"><span class="pl-c">#</span> Calculate PCs using variable features determined by SCTransform (3000 by default)</span>
<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> RunPCA(<span class="pl-smi">merged_seurat</span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">npcs</span> <span class="pl-k">=</span> <span class="pl-c1">50</span>)
<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> RunTSNE(<span class="pl-smi">merged_seurat</span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">npcs</span> <span class="pl-k">=</span> <span class="pl-c1">50</span>)

<span class="pl-c"><span class="pl-c">#</span> Integration</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("harmony")</span>

library(<span class="pl-smi">harmony</span>)

<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> RunHarmony(<span class="pl-smi">merged_seurat</span>, 
				<span class="pl-v">group.by.vars</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gender<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Surgery_Type<span class="pl-pds">"</span></span>), 
				<span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>, <span class="pl-v">assay.use</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">reduction.save</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     The  above code incorporates an additional reduction of 50 "harmony components" (i.e., corrected principal components) to our Seurat object, which is stored in the harmonized_seurat@reductions$harmony variable.
                    </p>
                    <p dir="auto">
                     However, to ensure that the Harmony integration is accurately represented in the data visualization, we must generate a UMAP that is derived from these harmony embeddings instead of the PCs.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='harmonized_seurat &lt;- RunUMAP(harmonized_seurat, reduction = "harmony", assay = "SCT", dims = 1:40)
#harmonized_seurat &lt;- RunUTSNE(harmonized_seurat, reduction = "harmony", assay = "SCT", dims = 1:40)

#________________________Cluster identification and Inspect the effects of Harmony batch removel ____________#

# to set reduction to harmony and find the clusters
harmonized_seurat &lt;- FindNeighbors(object = harmonized_seurat, reduction = "harmony")
harmonized_seurat &lt;- FindClusters(harmonized_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))

# visualization
Idents(harmonized_seurat) &lt;- harmonized_seurat@meta.data$SCT_snn_res.0.1

# Color cells based on the sample name
# Plot UMAP 
png(filename = "harmony_UMAP_y_sample.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(harmonized_seurat,
        group.by = "orig.ident",
        reduction = "umap")
dev.off()' dir="auto">
                     <pre><span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> RunUMAP(<span class="pl-smi">harmonized_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">40</span>)
<span class="pl-c"><span class="pl-c">#</span>harmonized_seurat &lt;- RunUTSNE(harmonized_seurat, reduction = "harmony", assay = "SCT", dims = 1:40)</span>

<span class="pl-c"><span class="pl-c">#</span>________________________Cluster identification and Inspect the effects of Harmony batch removel ____________#</span>

<span class="pl-c"><span class="pl-c">#</span> to set reduction to harmony and find the clusters</span>
<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> FindNeighbors(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)
<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> FindClusters(<span class="pl-smi">harmonized_seurat</span>, <span class="pl-v">resolution</span> <span class="pl-k">=</span> c(<span class="pl-c1">0.1</span>, <span class="pl-c1">0.2</span>, <span class="pl-c1">0.4</span>, <span class="pl-c1">0.6</span>, <span class="pl-c1">0.8</span>))

<span class="pl-c"><span class="pl-c">#</span> visualization</span>
Idents(<span class="pl-smi">harmonized_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>

<span class="pl-c"><span class="pl-c">#</span> Color cells based on the sample name</span>
<span class="pl-c"><span class="pl-c">#</span> Plot UMAP </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony_UMAP_y_sample.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">harmonized_seurat</span>,
        <span class="pl-v">group.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_UMAP_y_sample.png" rel="noopener noreferrer" target="_blank">
                      <img alt="harmony_UMAP_y_sample.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/harmony_UMAP_y_sample.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     As the above figure suggests, Harmony did a great job in terms of removing the batch effects.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#________________________SuperCluster Identification____________#

png(filename = "harmony_umap_cluster_with_label.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(harmonized_seurat,
        reduction = "umap",
        label = TRUE,
        label.size = 6)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>________________________SuperCluster Identification____________#</span>

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony_umap_cluster_with_label.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">harmonized_seurat</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_umap_cluster_with_label.png" rel="noopener noreferrer" target="_blank">
                      <img alt="harmony_umap_cluster_with_label.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/harmony_umap_cluster_with_label.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# let's visualize cells expressing supercluster markers:
# CD31: PECAM1
markers &lt;- c(&quot;EPCAM&quot;, &quot;PECAM1&quot;, &quot;COL1A1&quot;, &quot;PDGFRA&quot;, &quot;RGS5&quot;, &quot;CD79A&quot;, &quot;LYZ&quot;, &quot;CD3D&quot;, &quot;TPSAB1&quot;)

png(filename = &quot;umap_superCluster_cells.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = harmonized_seurat,
            features = markers,
            order = TRUE,
            min.cutoff = &quot;q10&quot;,
            reduction = &quot;umap&quot;,
            label = TRUE,
            repel = TRUE)

dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> let's visualize cells expressing supercluster markers:</span>
<span class="pl-c"><span class="pl-c">#</span> CD31: PECAM1</span>
<span class="pl-smi">markers</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>EPCAM<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>PECAM1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>COL1A1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>PDGFRA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RGS5<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD79A<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LYZ<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD3D<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>TPSAB1<span class="pl-pds">"</span></span>)

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap_superCluster_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>,
            <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">markers</span>,
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>q10<span class="pl-pds">"</span></span>,
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/umap_superCluster_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="umap_superCluster_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/umap_superCluster_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Marker identification for the 8 superclusters:
                     </h4>
                     <a aria-label="Permalink: Marker identification for the 8 superclusters:" class="anchor" href="#marker-identification-for-the-8-superclusters" id="user-content-marker-identification-for-the-8-superclusters">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#______________________________ All markers________________________________
# Find markers for every cluster compared to all remaining cells, and report only the positive ones
markers &lt;- FindAllMarkers(object = harmonized_seurat, 
                          only.pos = TRUE,
                          logfc.threshold = 0.25) 

saveRDS(markers, "harmony_markers.RDS")

# mutate the markers dataframe
# Extract top 10 markers per cluster

top10 &lt;- markers %&gt;%
  mutate(delta_pct = (pct.1 - pct.2)) %&gt;%
  #filter(avg_log2FC &gt; 1.5) %&gt;%  # only keep rows where avg_log2FC &gt; 1.5
  group_by(cluster) %&gt;%
  top_n(n = 10, wt = delta_pct)

data.table::fwrite(top10, "harmony_blca_top10_all_markers.csv")
' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>______________________________ All markers________________________________</span>
<span class="pl-c"><span class="pl-c">#</span> Find markers for every cluster compared to all remaining cells, and report only the positive ones</span>
<span class="pl-smi">markers</span> <span class="pl-k">&lt;-</span> FindAllMarkers(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>, 
                          <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                          <span class="pl-v">logfc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">0.25</span>) 

saveRDS(<span class="pl-smi">markers</span>, <span class="pl-s"><span class="pl-pds">"</span>harmony_markers.RDS<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> mutate the markers dataframe</span>
<span class="pl-c"><span class="pl-c">#</span> Extract top 10 markers per cluster</span>

<span class="pl-smi">top10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">markers</span> %<span class="pl-k">&gt;</span>%
  mutate(<span class="pl-v">delta_pct</span> <span class="pl-k">=</span> (<span class="pl-smi">pct.1</span> <span class="pl-k">-</span> <span class="pl-smi">pct.2</span>)) %<span class="pl-k">&gt;</span>%
  <span class="pl-c"><span class="pl-c">#</span>filter(avg_log2FC &gt; 1.5) %&gt;%  # only keep rows where avg_log2FC &gt; 1.5</span>
  group_by(<span class="pl-smi">cluster</span>) %<span class="pl-k">&gt;</span>%
  top_n(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, <span class="pl-v">wt</span> <span class="pl-k">=</span> <span class="pl-smi">delta_pct</span>)

<span class="pl-smi">data.table</span><span class="pl-k">::</span>fwrite(<span class="pl-smi">top10</span>, <span class="pl-s"><span class="pl-pds">"</span>harmony_blca_top10_all_markers.csv<span class="pl-pds">"</span></span>)
</pre>
                    </div>
                    <p dir="auto">
                     Visualization of top markers in each cluster:
                    </p>
                    <p dir="auto">
                     Cluster markers:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='cluster_markers_10 &lt;- top10 %&gt;% 
                   group_by(cluster) %&gt;% 
                   summarize(genes = paste(gene, collapse = ","))
data.table::fwrite(cluster_markers_10, "cluster_markers_10.csv")' dir="auto">
                     <pre><span class="pl-smi">cluster_markers_10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">top10</span> %<span class="pl-k">&gt;</span>% 
                   group_by(<span class="pl-smi">cluster</span>) %<span class="pl-k">&gt;</span>% 
                   summarize(<span class="pl-v">genes</span> <span class="pl-k">=</span> paste(<span class="pl-smi">gene</span>, <span class="pl-v">collapse</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>))
<span class="pl-smi">data.table</span><span class="pl-k">::</span>fwrite(<span class="pl-smi">cluster_markers_10</span>, <span class="pl-s"><span class="pl-pds">"</span>cluster_markers_10.csv<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Feature plot for top markers
plotList = list()

for(cluster in 1:nrow(cluster_markers_10)){
    mkr = unlist(strsplit(cluster_markers_10$genes[cluster], ","))
    plotList[[cluster]] = FeaturePlot(object = harmonized_seurat,
            features = mkr,
            order = TRUE,
            min.cutoff = "q10",
            reduction = "umap",
            label = TRUE,
            repel = TRUE)
}

# Iterate over all clusters
png(filename = "harmony_blca_clsuter_markers_cluster8.png", width = 16, height = 8.135, units = "in", res = 300)
plotList[[9]]
dev.off()
#' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Feature plot for top markers</span>
<span class="pl-v">plotList</span> <span class="pl-k">=</span> <span class="pl-k">list</span>()

<span class="pl-k">for</span>(<span class="pl-smi">cluster</span> <span class="pl-k">in</span> <span class="pl-c1">1</span><span class="pl-k">:</span>nrow(<span class="pl-smi">cluster_markers_10</span>)){
    <span class="pl-v">mkr</span> <span class="pl-k">=</span> unlist(strsplit(<span class="pl-smi">cluster_markers_10</span><span class="pl-k">$</span><span class="pl-smi">genes</span>[<span class="pl-smi">cluster</span>], <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>))
    <span class="pl-smi">plotList</span>[[<span class="pl-smi">cluster</span>]] <span class="pl-k">=</span> FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>,
            <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">mkr</span>,
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>q10<span class="pl-pds">"</span></span>,
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
}

<span class="pl-c"><span class="pl-c">#</span> Iterate over all clusters</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony_blca_clsuter_markers_cluster8.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
<span class="pl-smi">plotList</span>[[<span class="pl-c1">9</span>]]
dev.off()
<span class="pl-c"><span class="pl-c">#</span></span></pre>
                    </div>
                    <p dir="auto">
                     Expression of cluster-specific markers:
                    </p>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster0.png">
                        cluster0
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster1.png">
                        cluster1
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster2.png">
                        cluster2
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster3.png">
                        cluster3
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster4.png">
                        cluster4
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster5.png">
                        cluster5
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster6.png">
                        cluster6
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster7.png">
                        cluster7
                       </a>
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmony_blca_clsuter_markers_cluster8.png">
                        cluster8
                       </a>
                      </p>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# LYZ cells
png(filename = "LYZ_harmony_blca_clsuter_marker.png", width = 16, height = 8.135, units = "in", res = 300)
FeaturePlot(object = harmonized_seurat,
            features = c("LYZ"),
            order = TRUE,
            min.cutoff = "q10",
            reduction = "umap",
            label = TRUE,
            repel = TRUE)
dev.off()
' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> LYZ cells</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>LYZ_harmony_blca_clsuter_marker.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>,
            <span class="pl-v">features</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>LYZ<span class="pl-pds">"</span></span>),
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>q10<span class="pl-pds">"</span></span>,
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     To see where there
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/LYZ_harmony_blca_clsuter_marker.png">
                      myeloid cells
                     </a>
                     are.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Markers and cell types
                     </h3>
                     <a aria-label="Permalink: Markers and cell types" class="anchor" href="#markers-and-cell-types" id="user-content-markers-and-cell-types">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        cluster_id
                       </th>
                       <th>
                        genes
                       </th>
                       <th>
                        cell_type (PanglaoDB + ChatGPT)
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        0
                       </td>
                       <td>
                        AGR2,C10orf99,SDC1,SERINC2,SMIM22,CLDN7,TNFRSF21,FAM160A1,CXADR,RAB25
                       </td>
                       <td>
                        epithelial cell
                       </td>
                      </tr>
                      <tr>
                       <td>
                        1
                       </td>
                       <td>
                        CD52,CD3D,PTPRC,TRAC,CD7,CD2,SKAP1,ARHGAP15,HCST,CD3E
                       </td>
                       <td>
                        T-cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        2
                       </td>
                       <td>
                        PLVAP,SPARCL1,HSPG2,VWF,TCF4,LDB2,CALCRL,RAMP2,PCAT19,PECAM1
                       </td>
                       <td>
                        Endothelial cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        3
                       </td>
                       <td>
                        HLA-DRA,HLA-DPB1,TYROBP,HLA-DQA1,HLA-DRB1,HLA-DPA1,HLA-DQB1,FCER1G
                       </td>
                       <td>
                        APCs (Macrophages, B-cells)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        4
                       </td>
                       <td>
                        IGLC2,IGHG1,IGLC1,IGHG3,IGHA1,IGHG4,JCHAIN,IGHGP,MZB1,DERL3
                       </td>
                       <td>
                        B-cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        5
                       </td>
                       <td>
                        RGS5,TAGLN,ACTA2,MYL9,CALD1,PPP1R14A,BGN,PRKG1,SOD3,COL6A2
                       </td>
                       <td>
                        myo-CAFs
                       </td>
                      </tr>
                      <tr>
                       <td>
                        6
                       </td>
                       <td>
                        LUM,DCN,COL1A2,COL3A1,RARRES2,MFAP4,C1S,C1R,COL6A2,COL6A3
                       </td>
                       <td>
                        i-CAFS
                       </td>
                      </tr>
                      <tr>
                       <td>
                        7
                       </td>
                       <td>
                        TPSB2,TPSAB1,CPA3,HPGDS,LTC4S,MS4A2,SAMSN1,RGS13,FCER1G,TYROBP
                       </td>
                       <td>
                        Mast cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        8
                       </td>
                       <td>
                        MIR205HG,COL4A5,IGFBP2,LINC00511,FOSL1,BCAM,RAP2B,SLC35F1,AQP3,DENND2C
                       </td>
                       <td>
                        Epithelial cells + Mesenchymal cells
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# renaming clusters

# Rename all identities
harmonized_seurat &lt;- RenameIdents(object = harmonized_seurat, 
                               "0" = "Epithelial cells",
                               "1" = "T-cells", # impureity with epithelial cells
                               "2" = "Endothelial cells",
                               "3" = "APCs(Macrophages, B-cells)",
                               "4" = "B-cells",
                               "5" = "myo-CAF",
                               "6" = "i-CAF",
                               "7" = "Mast cells",
                               "8" = "Epithelial cells+Mesenchymal cells")
                               
                               
                               


# Plot the UMAP with new labels
png(filename = "harmont_blca_umap_with_label.png", width = 16, height = 8.135, units = "in", res = 600)
DimPlot(object = harmonized_seurat, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE,
        split.by = "Invasiveness")
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> renaming clusters</span>

<span class="pl-c"><span class="pl-c">#</span> Rename all identities</span>
<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> RenameIdents(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>, 
                               <span class="pl-s"><span class="pl-pds">"</span>0<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Epithelial cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>1<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>T-cells<span class="pl-pds">"</span></span>, <span class="pl-c"><span class="pl-c">#</span> impureity with epithelial cells</span>
                               <span class="pl-s"><span class="pl-pds">"</span>2<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Endothelial cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>3<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>APCs(Macrophages, B-cells)<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>B-cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>5<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>myo-CAF<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>6<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>i-CAF<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>7<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Mast cells<span class="pl-pds">"</span></span>,
                               <span class="pl-s"><span class="pl-pds">"</span>8<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Epithelial cells+Mesenchymal cells<span class="pl-pds">"</span></span>)
                               
                               
                               


<span class="pl-c"><span class="pl-c">#</span> Plot the UMAP with new labels</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmont_blca_umap_with_label.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">600</span>)
DimPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>, 
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>,
        <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">split.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Invasiveness<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/harmont_blca_umap_with_label.png" rel="noopener noreferrer" target="_blank">
                      <img alt="harmont_blca_umap_with_label.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/harmont_blca_umap_with_label.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Cluster 1, T-cells show some level of impurity by cluster 0, epithelial cells. This Impurity is especially evident in invasive samples.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Obtaining clusters with epithelial cells

# To see the number of cells in each cluster

table(Idents(harmonized_seurat))

#
#                  epithelial cells                            T-cells
#                             24135                              19850
#                 Endothelial cells         APCs(Macrophages, B-cells)
#                              6699                               5233
#                           B-cells                            myo-CAF
#                              4448                               2877
#                             i-CAF                         Mast cells
#                              2337                               1126
#Epithelial cells+Mesenchymal cells
#                               725
#
epi_cell_ids &lt;- rownames(harmonized_seurat@meta.data)[harmonized_seurat@meta.data$SCT_snn_res.0.1 == '0']

epi_cell_ids &lt;- c(epi_cell_ids,rownames(harmonized_seurat@meta.data)[harmonized_seurat@meta.data$SCT_snn_res.0.1 == '8'])


epi_seurat &lt;- subset(filtered_seurat, subset = cells %in% epi_cell_ids)

# Perform log-normalization and feature selection, as well as SCT normalization on the global object
epi_seurat &lt;- epi_seurat %&gt;%
    NormalizeData() %&gt;%
    FindVariableFeatures(selection.method = &quot;vst&quot;, nfeatures = 3000) %&gt;% 
    ScaleData() %&gt;%
    SCTransform(vars.to.regress = c(&quot;mitoRatio&quot;, &quot;orig.ident&quot;))

# Calculate PCs using variable features determined by SCTransform (3000 by default)
epi_seurat &lt;- RunPCA(epi_seurat, assay = &quot;SCT&quot;, npcs = 50)
#epi_seurat &lt;- RunTSNE(epi_seurat, assay = &quot;SCT&quot;, npcs = 50)

# Integration
#install.packages(&quot;harmony&quot;)

library(harmony)

epi_seurat &lt;- RunHarmony(epi_seurat, 
				group.by.vars = c(&quot;orig.ident&quot;, &quot;gender&quot;, &quot;Surgery_Type&quot;), 
				reduction = &quot;pca&quot;, assay.use = &quot;SCT&quot;, reduction.save = &quot;harmony&quot;)

epi_seurat &lt;- RunUMAP(epi_seurat, reduction = &quot;harmony&quot;, assay = &quot;SCT&quot;, dims = 1:40)
#harmonized_seurat &lt;- RunUTSNE(harmonized_seurat, reduction = &quot;harmony&quot;, assay = &quot;SCT&quot;, dims = 1:40)

#Cluster identification and Inspect the effects of Harmony batch removel#

# to set reduction to harmony and find the clusters
epi_seurat &lt;- FindNeighbors(object = epi_seurat, reduction = &quot;harmony&quot;)
epi_seurat &lt;- FindClusters(epi_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))

# visualization
Idents(epi_seurat) &lt;- epi_seurat@meta.data$SCT_snn_res.0.1

# Color cells based on the sample name
# Plot UMAP 
png(filename = &quot;epi_harmony_UMAP_y_sample.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
DimPlot(epi_seurat,
        group.by = &quot;orig.ident&quot;,
        reduction = &quot;umap&quot;)
dev.off()
#" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Obtaining clusters with epithelial cells</span>

<span class="pl-c"><span class="pl-c">#</span> To see the number of cells in each cluster</span>

table(Idents(<span class="pl-smi">harmonized_seurat</span>))

<span class="pl-c"><span class="pl-c">#</span></span>
<span class="pl-c"><span class="pl-c">#</span>                  epithelial cells                            T-cells</span>
<span class="pl-c"><span class="pl-c">#</span>                             24135                              19850</span>
<span class="pl-c"><span class="pl-c">#</span>                 Endothelial cells         APCs(Macrophages, B-cells)</span>
<span class="pl-c"><span class="pl-c">#</span>                              6699                               5233</span>
<span class="pl-c"><span class="pl-c">#</span>                           B-cells                            myo-CAF</span>
<span class="pl-c"><span class="pl-c">#</span>                              4448                               2877</span>
<span class="pl-c"><span class="pl-c">#</span>                             i-CAF                         Mast cells</span>
<span class="pl-c"><span class="pl-c">#</span>                              2337                               1126</span>
<span class="pl-c"><span class="pl-c">#</span>Epithelial cells+Mesenchymal cells</span>
<span class="pl-c"><span class="pl-c">#</span>                               725</span>
<span class="pl-c"><span class="pl-c">#</span></span>
<span class="pl-smi">epi_cell_ids</span> <span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)[<span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>]

<span class="pl-smi">epi_cell_ids</span> <span class="pl-k">&lt;-</span> c(<span class="pl-smi">epi_cell_ids</span>,rownames(<span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)[<span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">'</span>8<span class="pl-pds">'</span></span>])


<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> subset(<span class="pl-smi">filtered_seurat</span>, <span class="pl-v">subset</span> <span class="pl-k">=</span> <span class="pl-smi">cells</span> <span class="pl-k">%in%</span> <span class="pl-smi">epi_cell_ids</span>)

<span class="pl-c"><span class="pl-c">#</span> Perform log-normalization and feature selection, as well as SCT normalization on the global object</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span> %<span class="pl-k">&gt;</span>%
    NormalizeData() %<span class="pl-k">&gt;</span>%
    FindVariableFeatures(<span class="pl-v">selection.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vst<span class="pl-pds">"</span></span>, <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">3000</span>) %<span class="pl-k">&gt;</span>% 
    ScaleData() %<span class="pl-k">&gt;</span>%
    SCTransform(<span class="pl-v">vars.to.regress</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>mitoRatio<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>))

<span class="pl-c"><span class="pl-c">#</span> Calculate PCs using variable features determined by SCTransform (3000 by default)</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunPCA(<span class="pl-smi">epi_seurat</span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">npcs</span> <span class="pl-k">=</span> <span class="pl-c1">50</span>)
<span class="pl-c"><span class="pl-c">#</span>epi_seurat &lt;- RunTSNE(epi_seurat, assay = "SCT", npcs = 50)</span>

<span class="pl-c"><span class="pl-c">#</span> Integration</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("harmony")</span>

library(<span class="pl-smi">harmony</span>)

<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunHarmony(<span class="pl-smi">epi_seurat</span>, 
				<span class="pl-v">group.by.vars</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>gender<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Surgery_Type<span class="pl-pds">"</span></span>), 
				<span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>, <span class="pl-v">assay.use</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">reduction.save</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)

<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunUMAP(<span class="pl-smi">epi_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">40</span>)
<span class="pl-c"><span class="pl-c">#</span>harmonized_seurat &lt;- RunUTSNE(harmonized_seurat, reduction = "harmony", assay = "SCT", dims = 1:40)</span>

<span class="pl-c"><span class="pl-c">#</span>Cluster identification and Inspect the effects of Harmony batch removel#</span>

<span class="pl-c"><span class="pl-c">#</span> to set reduction to harmony and find the clusters</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> FindNeighbors(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> FindClusters(<span class="pl-smi">epi_seurat</span>, <span class="pl-v">resolution</span> <span class="pl-k">=</span> c(<span class="pl-c1">0.1</span>, <span class="pl-c1">0.2</span>, <span class="pl-c1">0.4</span>, <span class="pl-c1">0.6</span>, <span class="pl-c1">0.8</span>))

<span class="pl-c"><span class="pl-c">#</span> visualization</span>
Idents(<span class="pl-smi">epi_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>

<span class="pl-c"><span class="pl-c">#</span> Color cells based on the sample name</span>
<span class="pl-c"><span class="pl-c">#</span> Plot UMAP </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>epi_harmony_UMAP_y_sample.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">epi_seurat</span>,
        <span class="pl-v">group.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>)
dev.off()
<span class="pl-c"><span class="pl-c">#</span></span></pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/epi_harmony_UMAP_y_sample.png" rel="noopener noreferrer" target="_blank">
                      <img alt="epi_harmony_UMAP_y_sample.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/epi_harmony_UMAP_y_sample.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# color cells based on the cluster
# Plot UMAP 
png(filename = "cluster_epi_harmony_UMAP.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(epi_seurat,
        reduction = "umap",
        label = TRUE,
        label.size = 6)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> color cells based on the cluster</span>
<span class="pl-c"><span class="pl-c">#</span> Plot UMAP </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cluster_epi_harmony_UMAP.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">epi_seurat</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cluster_epi_harmony_UMAP.png" rel="noopener noreferrer" target="_blank">
                      <img alt="cluster_epi_harmony_UMAP.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cluster_epi_harmony_UMAP.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Marker identification

epi_markers &lt;- FindAllMarkers(object = epi_seurat, 
                          only.pos = TRUE,
                          logfc.threshold = 0.25) 

# mutate the markers dataframe
# Extract top 10 markers per cluster

epi_top10 &lt;- epi_markers %&gt;%
  mutate(delta_pct = (pct.1 - pct.2)) %&gt;%
  #filter(avg_log2FC &gt; 1.5) %&gt;%  # only keep rows where avg_log2FC &gt; 1.5
  group_by(cluster) %&gt;%
  top_n(n = 10, wt = delta_pct)


epi_cluster_markers_10 &lt;- epi_top10 %&gt;% 
                   group_by(cluster) %&gt;% 
                   summarize(genes = paste(gene, collapse = ","))' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Marker identification</span>

<span class="pl-smi">epi_markers</span> <span class="pl-k">&lt;-</span> FindAllMarkers(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, 
                          <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                          <span class="pl-v">logfc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">0.25</span>) 

<span class="pl-c"><span class="pl-c">#</span> mutate the markers dataframe</span>
<span class="pl-c"><span class="pl-c">#</span> Extract top 10 markers per cluster</span>

<span class="pl-smi">epi_top10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_markers</span> %<span class="pl-k">&gt;</span>%
  mutate(<span class="pl-v">delta_pct</span> <span class="pl-k">=</span> (<span class="pl-smi">pct.1</span> <span class="pl-k">-</span> <span class="pl-smi">pct.2</span>)) %<span class="pl-k">&gt;</span>%
  <span class="pl-c"><span class="pl-c">#</span>filter(avg_log2FC &gt; 1.5) %&gt;%  # only keep rows where avg_log2FC &gt; 1.5</span>
  group_by(<span class="pl-smi">cluster</span>) %<span class="pl-k">&gt;</span>%
  top_n(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, <span class="pl-v">wt</span> <span class="pl-k">=</span> <span class="pl-smi">delta_pct</span>)


<span class="pl-smi">epi_cluster_markers_10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_top10</span> %<span class="pl-k">&gt;</span>% 
                   group_by(<span class="pl-smi">cluster</span>) %<span class="pl-k">&gt;</span>% 
                   summarize(<span class="pl-v">genes</span> <span class="pl-k">=</span> paste(<span class="pl-smi">gene</span>, <span class="pl-v">collapse</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>))</pre>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        cluster
                       </th>
                       <th>
                        genes
                       </th>
                       <th>
                        cell_type (PanglaoDB + ChatGPT)
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        0
                       </td>
                       <td>
                        KRT15,KRT5,BCAM,IGFBP2,IGFBP7,CDH13,FOSL1,LAMB3,COL4A5,SERPINB5,SELENOP
                       </td>
                       <td>
                        basal cell
                       </td>
                      </tr>
                      <tr>
                       <td>
                        1
                       </td>
                       <td>
                        UCA1,LGALS1,CA9,SPINK1,GJB2,PTPRR,FCRLB,PLA2G2F,GJB6,FBLN1
                       </td>
                       <td>
                        cancer-associated luminal cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        2
                       </td>
                       <td>
                        NDUFA4L2,CRH,UPK2,SNX31,KRT20,AC019117.2,UPK1B,LINC02163,H19,TESC
                       </td>
                       <td>
                        luminal differentiated
                       </td>
                      </tr>
                      <tr>
                       <td>
                        3
                       </td>
                       <td>
                        LY6D,TNNT3,IGF2BP2,GMNN,SYT8,AC068587.4,ITM2C,SGPP2,ZNF750,TNFSF10
                       </td>
                       <td>
                        unique luminal cells
                       </td>
                      </tr>
                      <tr>
                       <td>
                        4
                       </td>
                       <td>
                        S100A9,C15orf48,MYO16,MMP7,MYEOV,AC025159.1,LTO1,IFI27,IRS2,SLPI
                       </td>
                       <td>
                        immunomodulatory luminal
                       </td>
                      </tr>
                      <tr>
                       <td>
                        5
                       </td>
                       <td>
                        ALCAM,ATXN1,PCDH7,COBLL1,LIPH,TRIM31,NAALADL2,ITGA2,SERPINB5,ITGA6,TM4SF1
                       </td>
                       <td>
                        adhesion and signaling luminal cells
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Reading seurat object
epi_seurat &lt;- readRDS("epi_seurat.RDS")

# Rename all identities
epi_seurat &lt;- RenameIdents(object = epi_seurat, 
                               "0" = "basal_cell",
                               "1" = "cancer_associated_luminal_cell",
                               "2" = "differentiated_luminal_cell",
                               "3" = "unique_luminal_cell",
                               "4" = "immunomodulatory_luminal_cell",
                               "5" = "adhesion_signaling_luminal_cell")

# Plot the UMAP with new labels
png(filename = "labelled_epi.png", width = 16, height = 8.135, units = "in", res = 600)
DimPlot(object = epi_seurat, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE)
dev.off()'>
                     <pre class="notranslate"><code># Reading seurat object
epi_seurat &lt;- readRDS("epi_seurat.RDS")

# Rename all identities
epi_seurat &lt;- RenameIdents(object = epi_seurat, 
                               "0" = "basal_cell",
                               "1" = "cancer_associated_luminal_cell",
                               "2" = "differentiated_luminal_cell",
                               "3" = "unique_luminal_cell",
                               "4" = "immunomodulatory_luminal_cell",
                               "5" = "adhesion_signaling_luminal_cell")

# Plot the UMAP with new labels
png(filename = "labelled_epi.png", width = 16, height = 8.135, units = "in", res = 600)
DimPlot(object = epi_seurat, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE)
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Cluster 5
                     </strong>
                     can benefit from re-clustering. It consists of several patches of cells. If we increase the clustering resolution we should get each patches as a cluster.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='Idents(epi_seurat) &lt;- epi_seurat$SCT_snn_res.0.2
png(filename = "cluster_epi_harmony_UMAP_ptsize_1_res0.2.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(epi_seurat,
        reduction = "umap",
        label = TRUE,
        label.size = 6)
dev.off()' dir="auto">
                     <pre>Idents(<span class="pl-smi">epi_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cluster_epi_harmony_UMAP_ptsize_1_res0.2.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">epi_seurat</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="/hamidghaedi/scRNA_seq-analysis/blob/main" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/hamidghaedi/scRNA_seq-analysis/raw/main" style="max-width: 100%;"/>
                     </a>
                     So based on the above graph, cells in cluster 5 are now clustered in 8,9,11, and 12. The current cluster 8 can be split into more clusters.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='Idents(epi_seurat) &lt;- epi_seurat$SCT_snn_res.0.6
png(filename = "cluster_epi_harmony_UMAP_ptsize_1_res0.6.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(epi_seurat,
        reduction = "umap",
        label = TRUE,
        label.size = 6)
dev.off()' dir="auto">
                     <pre>Idents(<span class="pl-smi">epi_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.6</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cluster_epi_harmony_UMAP_ptsize_1_res0.6.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">epi_seurat</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     Using the above resolution we can get two different clusters of cells for cluster 8. Now let's generate a customized
                     <code>
                      Ident
                     </code>
                     for the data set:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='idt &lt;- epi_seurat@meta.data[, c(14,15,17)]
# convert to character
idt &lt;- data.frame(apply(idt, 2, as.character))
# to see what to do:
# table(epi_seurat$SCT_snn_res.0.2, epi_seurat$SCT_snn_res.0.6)

# Replacing values for cluster 8 with clusters 18 and 19 in res 0.6
idt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == "15"] &lt;- "8_a"
idt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == "1"] &lt;- "8_a"
idt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == "2"] &lt;- "8_a"
idt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == "16"] &lt;- "8_b"
idt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == "18"] &lt;- "8_b"
idt$SCT_snn_res.0.2[idt$SCT_snn_res.0.6 == "19"] &lt;- "8_b"

# Updating SCT_snn_res.0.1 
# to see what to do:
# table(idt$SCT_snn_res.0.1, idt$SCT_snn_res.0.2)

idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "9"] &lt;- "5_a"
idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "8_a"] &lt;- "5_b"
idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "3"] &lt;- "5_b"
idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "4"] &lt;- "5_b"
idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "6"] &lt;- "5_a"
idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "8_b"] &lt;- "5_c"
idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "11"] &lt;- "5_d"
idt$SCT_snn_res.0.1[idt$SCT_snn_res.0.2 == "12"] &lt;- "5_e"
# adding new idents to Seurat obj
epi_seurat$new_idents &lt;- idt$SCT_snn_res.0.1
# setting ident for plotting
Idents(epi_seurat) &lt;- epi_seurat$new_idents
# plotting
png(filename = "cluster_epi_harmony_UMAP_new_idents.png", width = 16, height = 8.135, units = "in", res = 300)
DimPlot(epi_seurat,
        reduction = "umap",
        label = TRUE,
        label.size = 6)
dev.off()' dir="auto">
                     <pre><span class="pl-smi">idt</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>[, c(<span class="pl-c1">14</span>,<span class="pl-c1">15</span>,<span class="pl-c1">17</span>)]
<span class="pl-c"><span class="pl-c">#</span> convert to character</span>
<span class="pl-smi">idt</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(apply(<span class="pl-smi">idt</span>, <span class="pl-c1">2</span>, <span class="pl-smi">as.character</span>))
<span class="pl-c"><span class="pl-c">#</span> to see what to do:</span>
<span class="pl-c"><span class="pl-c">#</span> table(epi_seurat$SCT_snn_res.0.2, epi_seurat$SCT_snn_res.0.6)</span>

<span class="pl-c"><span class="pl-c">#</span> Replacing values for cluster 8 with clusters 18 and 19 in res 0.6</span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.6</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>15<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>8_a<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.6</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>1<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>8_a<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.6</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>2<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>8_a<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.6</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>16<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>8_b<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.6</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>18<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>8_b<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.6</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>19<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>8_b<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span> Updating SCT_snn_res.0.1 </span>
<span class="pl-c"><span class="pl-c">#</span> to see what to do:</span>
<span class="pl-c"><span class="pl-c">#</span> table(idt$SCT_snn_res.0.1, idt$SCT_snn_res.0.2)</span>

<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>9<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_a<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>8_a<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_b<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>3<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_b<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_b<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>6<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_a<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>8_b<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_c<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>11<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_d<span class="pl-pds">"</span></span>
<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>[<span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.2</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">"</span>12<span class="pl-pds">"</span></span>] <span class="pl-k">&lt;-</span> <span class="pl-s"><span class="pl-pds">"</span>5_e<span class="pl-pds">"</span></span>
<span class="pl-c"><span class="pl-c">#</span> adding new idents to Seurat obj</span>
<span class="pl-smi">epi_seurat</span><span class="pl-k">$</span><span class="pl-smi">new_idents</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">idt</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>
<span class="pl-c"><span class="pl-c">#</span> setting ident for plotting</span>
Idents(<span class="pl-smi">epi_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span><span class="pl-k">$</span><span class="pl-smi">new_idents</span>
<span class="pl-c"><span class="pl-c">#</span> plotting</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cluster_epi_harmony_UMAP_new_idents.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">epi_seurat</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/cluster_epi_harmony_UMAP_new_idents.png" rel="noopener noreferrer" target="_blank">
                      <img alt="cluster_epi_harmony_UMAP_new_idents.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/cluster_epi_harmony_UMAP_new_idents.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      10) DE and GSEA
                     </h2>
                     <a aria-label="Permalink: 10) DE and GSEA" class="anchor" href="#10-de-and-gsea" id="user-content-10-de-and-gsea">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(SCP)


# DE and Enrichement analysis

#Differential expression analysis; this takes a while, to upload the object containing DE result:
epi_seurat &lt;- readRDS("updated_epi_seurat.RDS")

epi_seurat &lt;- RunDEtest(srt = epi_seurat, group_by = "clusters", fc.threshold = 1.5, only.pos = FALSE)

#
png(filename = "vplcano_DE.png", width = 16, height = 8.135, units = "in", res = 300)
VolcanoPlot(srt = epi_seurat, group_by = "clusters")
dev.off()' dir="auto">
                     <pre>library(<span class="pl-smi">SCP</span>)


<span class="pl-c"><span class="pl-c">#</span> DE and Enrichement analysis</span>

<span class="pl-c"><span class="pl-c">#</span>Differential expression analysis; this takes a while, to upload the object containing DE result:</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> readRDS(<span class="pl-s"><span class="pl-pds">"</span>updated_epi_seurat.RDS<span class="pl-pds">"</span></span>)

<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunDEtest(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">fc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">1.5</span>, <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

<span class="pl-c"><span class="pl-c">#</span></span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vplcano_DE.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
VolcanoPlot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/vplcano_DE.png" rel="noopener noreferrer" target="_blank">
                      <img alt="vplcano_DE.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/vplcano_DE.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#
DEGs &lt;- epi_seurat@tools$DEtest_clusters$AllMarkers_wilcox
DEGs &lt;- DEGs[with(DEGs, avg_log2FC &gt; 1.5 &amp; p_val_adj &lt; 0.05), ]
# Annotate features with transcription factors and surface proteins
#epi_seurat &lt;- AnnotateFeatures(epi_seurat, species = "Homo_sapiens", db = c("TF", "SP"))

# png(filename = "heatmap_DE.png", width = 16, height = 8.135, units = "in", res = 300)
# ht &lt;- FeatureHeatmap(
#   srt = epi_seurat, group.by = "clusters", features = DEGs$gene, feature_split = DEGs$group1,
#   species = "Homo_sapiens", db = c("GO_BP", "KEGG", "WikiPathway"), anno_terms = TRUE)
# print(ht$plot)
# dev.off()


# Enrichment analysis
## over-representation analysis

epi_seurat &lt;- RunEnrichment(
  srt = epi_seurat, group_by = "clusters", db = "GO_BP", species = "Homo_sapiens",
  DE_threshold = "avg_log2FC &gt; 1 &amp; p_val_adj &lt; 0.05"
)

png(filename = "enrichment_1.png", width = 16, height = 8.135, units = "in", res = 300)
EnrichmentPlot(
  srt = epi_seurat, group_by = "clusters", group_use = c("basal_cell", "differentiated_luminal_cell"),
  plot_type = "bar"
)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span></span>
<span class="pl-smi">DEGs</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span><span class="pl-k">@</span><span class="pl-smi">tools</span><span class="pl-k">$</span><span class="pl-smi">DEtest_clusters</span><span class="pl-k">$</span><span class="pl-smi">AllMarkers_wilcox</span>
<span class="pl-smi">DEGs</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">DEGs</span>[with(<span class="pl-smi">DEGs</span>, <span class="pl-smi">avg_log2FC</span> <span class="pl-k">&gt;</span> <span class="pl-c1">1.5</span> <span class="pl-k">&amp;</span> <span class="pl-smi">p_val_adj</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0.05</span>), ]
<span class="pl-c"><span class="pl-c">#</span> Annotate features with transcription factors and surface proteins</span>
<span class="pl-c"><span class="pl-c">#</span>epi_seurat &lt;- AnnotateFeatures(epi_seurat, species = "Homo_sapiens", db = c("TF", "SP"))</span>

<span class="pl-c"><span class="pl-c">#</span> png(filename = "heatmap_DE.png", width = 16, height = 8.135, units = "in", res = 300)</span>
<span class="pl-c"><span class="pl-c">#</span> ht &lt;- FeatureHeatmap(</span>
<span class="pl-c"><span class="pl-c">#</span>   srt = epi_seurat, group.by = "clusters", features = DEGs$gene, feature_split = DEGs$group1,</span>
<span class="pl-c"><span class="pl-c">#</span>   species = "Homo_sapiens", db = c("GO_BP", "KEGG", "WikiPathway"), anno_terms = TRUE)</span>
<span class="pl-c"><span class="pl-c">#</span> print(ht$plot)</span>
<span class="pl-c"><span class="pl-c">#</span> dev.off()</span>


<span class="pl-c"><span class="pl-c">#</span> Enrichment analysis</span>
<span class="pl-c"><span class="pl-c">#</span># over-representation analysis</span>

<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunEnrichment(
  <span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">db</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GO_BP<span class="pl-pds">"</span></span>, <span class="pl-v">species</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Homo_sapiens<span class="pl-pds">"</span></span>,
  <span class="pl-v">DE_threshold</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>avg_log2FC &gt; 1 &amp; p_val_adj &lt; 0.05<span class="pl-pds">"</span></span>
)

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>enrichment_1.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
EnrichmentPlot(
  <span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">group_use</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>basal_cell<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>differentiated_luminal_cell<span class="pl-pds">"</span></span>),
  <span class="pl-v">plot_type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>bar<span class="pl-pds">"</span></span>
)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/enrichment_1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="enrichment_1.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/enrichment_1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='png(filename = "enrichment_2.png", width = 16, height = 8.135, units = "in", res = 300)
EnrichmentPlot(srt = epi_seurat, group_by = "clusters", plot_type = "comparison")
dev.off()' dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>enrichment_2.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
EnrichmentPlot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">plot_type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>comparison<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/enrichment_2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="enrichment_2.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/enrichment_2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## Enrichment analysis(GSEA)
epi_seurat &lt;- RunGSEA(
  srt = epi_seurat, group_by = "clusters", db = "GO_BP", species = "Homo_sapiens",
  DE_threshold = "p_val_adj &lt; 0.05"
)

# plotting for hypoxia
png(filename = "enrichment_4.png", width = 16, height = 8.135, units = "in", res = 300)
GSEAPlot(srt = epi_seurat, group_by = "clusters", group_use = "differentiated_luminal_cell", geneSetID = "GO:0001666")
dev.off()

png(filename = "enrichment_5.png", width = 16, height = 8.135, units = "in", res = 300)
GSEAPlot(srt = epi_seurat, group_by = "clusters", group_use = "cancer_associated_luminal_cell", geneSetID = "GO:0001666")
dev.off()

png(filename = "enrichment_6.png", width = 16, height = 8.135, units = "in", res = 300)
GSEAPlot(srt = epi_seurat, group_by = "clusters", group_use = "basal_cell", geneSetID = "GO:0001666")
dev.off()

# comparison plot
png(filename = "enrichment_3.png", width = 16, height = 8.135, units = "in", res = 300)
GSEAPlot(srt = epi_seurat, group_by = "clusters", plot_type = "comparison")
dev.off()
' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># Enrichment analysis(GSEA)</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunGSEA(
  <span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">db</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GO_BP<span class="pl-pds">"</span></span>, <span class="pl-v">species</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Homo_sapiens<span class="pl-pds">"</span></span>,
  <span class="pl-v">DE_threshold</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>p_val_adj &lt; 0.05<span class="pl-pds">"</span></span>
)

<span class="pl-c"><span class="pl-c">#</span> plotting for hypoxia</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>enrichment_4.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
GSEAPlot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">group_use</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>differentiated_luminal_cell<span class="pl-pds">"</span></span>, <span class="pl-v">geneSetID</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GO:0001666<span class="pl-pds">"</span></span>)
dev.off()

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>enrichment_5.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
GSEAPlot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">group_use</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cancer_associated_luminal_cell<span class="pl-pds">"</span></span>, <span class="pl-v">geneSetID</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GO:0001666<span class="pl-pds">"</span></span>)
dev.off()

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>enrichment_6.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
GSEAPlot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">group_use</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>basal_cell<span class="pl-pds">"</span></span>, <span class="pl-v">geneSetID</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>GO:0001666<span class="pl-pds">"</span></span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span> comparison plot</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>enrichment_3.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
GSEAPlot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group_by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">plot_type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>comparison<span class="pl-pds">"</span></span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/enrichment_3.png" rel="noopener noreferrer" target="_blank">
                      <img alt="enrichment_3.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/enrichment_3.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      11) Trajectory inference
                     </h2>
                     <a aria-label="Permalink: 11) Trajectory inference" class="anchor" href="#11-trajectory-inference" id="user-content-11-trajectory-inference">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Slingshot is a computational method designed for cell lineage inference and pseudotime estimation in single-cell transcriptomics data. It operates by leveraging gene expression dynamics to reconstruct the developmental trajectories of individual cells and assign them pseudotime values. The method begins by clustering cells based on their gene expression profiles, allowing for the identification of distinct cell populations. It then performs differential expression analysis to identify genes that are dynamically expressed across these populations, capturing the transitions between different cell types or states. Slingshot uses this information to infer lineage relationships between the cell populations, revealing the hierarchical structure of the developmental trajectory. By assigning pseudotime values, Slingshot estimates the relative progression of cells through different stages of development or biological processes. The inferred trajectories and pseudotime values provide insights into the temporal ordering of cellular events and gene expression dynamics. Slingshot's visualization tools facilitate the exploration and interpretation of these results, enabling researchers to gain a deeper understanding of the underlying processes driving cellular differentiation and disease progression in single-cell transcriptomics data.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='png(filename = "trajectory.png", width = 16, height = 8.135, units = "in", res = 300)
epi_seurat &lt;- RunSlingshot(srt = epi_seurat, group.by = "clusters", reduction = "UMAP")
dev.off()
' dir="auto">
                     <pre>png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>trajectory.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunSlingshot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>UMAP<span class="pl-pds">"</span></span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/trajectory.png" rel="noopener noreferrer" target="_blank">
                      <img alt="trajectory.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/trajectory.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Identify temporally dynamic genes
                     </h3>
                     <a aria-label="Permalink: Identify temporally dynamic genes" class="anchor" href="#identify-temporally-dynamic-genes" id="user-content-identify-temporally-dynamic-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     After running slingshot, I am going to explore gene expression changes during development. I aim to discover genes that undergo significant expression shifts. The method involves fitting a General Additive Model (GAM) for each gene, employing a negative binomial noise distribution to capture potential nonlinear connections between gene expression and pseudotime. Then conduct association tests to unveil meaningful links between expression and pseudotime.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(slingshot)
library(tradeSeq)
library(RColorBrewer)

# convert Seurat to sce
sce &lt;- as.SingleCellExperiment(epi_seurat)
# Running slingshot
sce &lt;- slingshot(sce, clusterLabels = 'clusters', reducedDim = 'UMAP')

## The following analysis is computationally intensive; so only a subset of genes will be passed to the function
# Gene filtration: filter out genes with lower than 5 reads expressed in &lt; 500 cells
geneFilter &lt;- apply(assays(sce)$counts,1,function(x){
  sum(x &gt;= 5) &gt;= 500
})
tdg &lt;- rownames(sce[geneFilter, ])

# Variable feature
vf &lt;- VariableFeatures(epi_seurat)
# Combining the genes
selected_feature &lt;- tdg[tdg %in% vf]
## Using the tradeSeq package
# Number of knot determined by ....
icMat &lt;- evaluateK(counts = sce, sds = crv, k = 3:7, nGenes = 50,
                   verbose = TRUE, plot = TRUE)
# 5 seems to be the right number for knot

# fit negative binomial GAM
tradeseq_res &lt;- fitGAM(counts = cse, nknots = 5, genes = selected_feature, verbose = TRUE)
# test for dynamic expression
ATres &lt;- associationTest(tradeseq_res)

# Visualizing heat map
pst.ord &lt;- order(sce$slingPseudotime_1, na.last = NA)
heatdata &lt;- as.matrix(assays(sce)$counts[topgenes, pst.ord])
heatclus &lt;- sce$clusters[pst.ord]

png(&quot;tdg.png&quot;, width=10, height = 10, unit = &quot;in&quot;, res = 300)
heatmap(log1p(heatdata), Colv = NA,
        ColSideColors = brewer.pal(9,&quot;Set1&quot;)[heatclus])
dev.off()" dir="auto">
                     <pre>library(<span class="pl-smi">slingshot</span>)
library(<span class="pl-smi">tradeSeq</span>)
library(<span class="pl-smi">RColorBrewer</span>)

<span class="pl-c"><span class="pl-c">#</span> convert Seurat to sce</span>
<span class="pl-smi">sce</span> <span class="pl-k">&lt;-</span> as.SingleCellExperiment(<span class="pl-smi">epi_seurat</span>)
<span class="pl-c"><span class="pl-c">#</span> Running slingshot</span>
<span class="pl-smi">sce</span> <span class="pl-k">&lt;-</span> slingshot(<span class="pl-smi">sce</span>, <span class="pl-v">clusterLabels</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>clusters<span class="pl-pds">'</span></span>, <span class="pl-v">reducedDim</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">'</span>UMAP<span class="pl-pds">'</span></span>)

<span class="pl-c"><span class="pl-c">#</span># The following analysis is computationally intensive; so only a subset of genes will be passed to the function</span>
<span class="pl-c"><span class="pl-c">#</span> Gene filtration: filter out genes with lower than 5 reads expressed in &lt; 500 cells</span>
<span class="pl-smi">geneFilter</span> <span class="pl-k">&lt;-</span> apply(assays(<span class="pl-smi">sce</span>)<span class="pl-k">$</span><span class="pl-smi">counts</span>,<span class="pl-c1">1</span>,<span class="pl-k">function</span>(<span class="pl-smi">x</span>){
  sum(<span class="pl-smi">x</span> <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">5</span>) <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">500</span>
})
<span class="pl-smi">tdg</span> <span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">sce</span>[<span class="pl-smi">geneFilter</span>, ])

<span class="pl-c"><span class="pl-c">#</span> Variable feature</span>
<span class="pl-smi">vf</span> <span class="pl-k">&lt;-</span> VariableFeatures(<span class="pl-smi">epi_seurat</span>)
<span class="pl-c"><span class="pl-c">#</span> Combining the genes</span>
<span class="pl-smi">selected_feature</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">tdg</span>[<span class="pl-smi">tdg</span> <span class="pl-k">%in%</span> <span class="pl-smi">vf</span>]
<span class="pl-c"><span class="pl-c">#</span># Using the tradeSeq package</span>
<span class="pl-c"><span class="pl-c">#</span> Number of knot determined by ....</span>
<span class="pl-smi">icMat</span> <span class="pl-k">&lt;-</span> evaluateK(<span class="pl-v">counts</span> <span class="pl-k">=</span> <span class="pl-smi">sce</span>, <span class="pl-v">sds</span> <span class="pl-k">=</span> <span class="pl-smi">crv</span>, <span class="pl-v">k</span> <span class="pl-k">=</span> <span class="pl-c1">3</span><span class="pl-k">:</span><span class="pl-c1">7</span>, <span class="pl-v">nGenes</span> <span class="pl-k">=</span> <span class="pl-c1">50</span>,
                   <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">plot</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-c"><span class="pl-c">#</span> 5 seems to be the right number for knot</span>

<span class="pl-c"><span class="pl-c">#</span> fit negative binomial GAM</span>
<span class="pl-smi">tradeseq_res</span> <span class="pl-k">&lt;-</span> fitGAM(<span class="pl-v">counts</span> <span class="pl-k">=</span> <span class="pl-smi">cse</span>, <span class="pl-v">nknots</span> <span class="pl-k">=</span> <span class="pl-c1">5</span>, <span class="pl-v">genes</span> <span class="pl-k">=</span> <span class="pl-smi">selected_feature</span>, <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-c"><span class="pl-c">#</span> test for dynamic expression</span>
<span class="pl-smi">ATres</span> <span class="pl-k">&lt;-</span> associationTest(<span class="pl-smi">tradeseq_res</span>)

<span class="pl-c"><span class="pl-c">#</span> Visualizing heat map</span>
<span class="pl-smi">pst.ord</span> <span class="pl-k">&lt;-</span> order(<span class="pl-smi">sce</span><span class="pl-k">$</span><span class="pl-smi">slingPseudotime_1</span>, <span class="pl-v">na.last</span> <span class="pl-k">=</span> <span class="pl-c1">NA</span>)
<span class="pl-smi">heatdata</span> <span class="pl-k">&lt;-</span> as.matrix(assays(<span class="pl-smi">sce</span>)<span class="pl-k">$</span><span class="pl-smi">counts</span>[<span class="pl-smi">topgenes</span>, <span class="pl-smi">pst.ord</span>])
<span class="pl-smi">heatclus</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">sce</span><span class="pl-k">$</span><span class="pl-smi">clusters</span>[<span class="pl-smi">pst.ord</span>]

png(<span class="pl-s"><span class="pl-pds">"</span>tdg.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">10</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, <span class="pl-v">unit</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
heatmap(log1p(<span class="pl-smi">heatdata</span>), <span class="pl-v">Colv</span> <span class="pl-k">=</span> <span class="pl-c1">NA</span>,
        <span class="pl-v">ColSideColors</span> <span class="pl-k">=</span> brewer.pal(<span class="pl-c1">9</span>,<span class="pl-s"><span class="pl-pds">"</span>Set1<span class="pl-pds">"</span></span>)[<span class="pl-smi">heatclus</span>])
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/tdg.png" rel="noopener noreferrer" target="_blank">
                      <img alt="tdg.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/tdg.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Exploring normal bladder mouse samples
                     </h2>
                     <a aria-label="Permalink: Exploring normal bladder mouse samples" class="anchor" href="#exploring-normal-bladder-mouse-samples" id="user-content-exploring-normal-bladder-mouse-samples">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#________________________Reading the files______________________#

samples &lt;- c(&quot;SRR12603780&quot;, &quot;SRR12603781&quot;, &quot;SRR12603788&quot;)

# read files inot Seurat objects
for (file in samples){
  print(paste0(file))
  seurat_data &lt;- Read10X(data.dir = paste0(&quot;../filtered/&quot;, file))
  seurat_obj &lt;- CreateSeuratObject(counts = seurat_data, 
                                   min.features = 100, 
                                   project = file)
  assign(file, seurat_obj)
}


# now merging all objects inot one Seurat obj

merged_seurat &lt;- merge(x = SRR12603780,
                       y = c(SRR12603781,
                             SRR12603788),
                       add.cell.id = samples)
                       
#________________________Filteration____________________________#
# reading sampleInformation:
sampleInformation &lt;- read.csv(&quot;../sampleInfo.csv&quot;)

# Compute percent mito ratio
merged_seurat$mitoRatio &lt;- PercentageFeatureSet(object = merged_seurat, pattern = &quot;^MT-&quot;)
merged_seurat$mitoRatio &lt;- merged_seurat@meta.data$mitoRatio / 100

# adding cell column
merged_seurat$cells &lt;- rownames(merged_seurat@meta.data)
# merging with sample information
merged_seurat@meta.data &lt;- merge(merged_seurat@meta.data, sampleInformation)
# re-setting the rownames
rownames(merged_seurat@meta.data) &lt;- merged_seurat@meta.data$cells


# Filteration

filtered_seurat &lt;- subset(merged_seurat, 
                          subset= nCount_RNA &gt;= 1000 &amp;
                          nFeature_RNA &lt;= 6000 &amp; 
                          mitoRatio &lt; 0.10)

#________________________Integration using Harmony____________________________#
#integration using harmony need sevral steps to be undertaken:

# Perform log-normalization and feature selection, as well as SCT normalization on global object
merged_seurat &lt;- filtered_seurat %&gt;%
    NormalizeData() %&gt;%
    FindVariableFeatures(selection.method = &quot;vst&quot;, nfeatures = 3000) %&gt;% 
    ScaleData() %&gt;%
    SCTransform(vars.to.regress = c(&quot;mitoRatio&quot;, &quot;orig.ident&quot;))

# Calculate PCs using variable features determined by SCTransform (3000 by default)
merged_seurat &lt;- RunPCA(merged_seurat, assay = &quot;SCT&quot;, npcs = 50)

# Integration
#install.packages(&quot;harmony&quot;)

library(harmony)

harmonized_seurat &lt;- RunHarmony(merged_seurat, 
				group.by.vars = c(&quot;orig.ident&quot;, &quot;Surgery_Type&quot;), 
				reduction = &quot;pca&quot;, assay.use = &quot;SCT&quot;, reduction.save = &quot;harmony&quot;)

harmonized_seurat &lt;- RunUMAP(harmonized_seurat, reduction = &quot;harmony&quot;, assay = &quot;SCT&quot;, dims = 1:40)

#Cluster identification and Inspect the effects of Harmony batch removel ____________#

# to set reduction to harmony and find the clusters
harmonized_seurat &lt;- FindNeighbors(object = harmonized_seurat, reduction = &quot;harmony&quot;)
harmonized_seurat &lt;- FindClusters(harmonized_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))

# visualization
Idents(harmonized_seurat) &lt;- harmonized_seurat@meta.data$SCT_snn_res.0.1

# Color cells based on the sample name
# Plot UMAP 
png(filename = &quot;normal_harmony_UMAP_y_sample.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
DimPlot(harmonized_seurat,
        group.by = &quot;orig.ident&quot;,
        reduction = &quot;umap&quot;)
dev.off()

#________________________SuperCluster Identification____________#

png(filename = &quot;normal_harmony_umap_cluster_with_label.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
DimPlot(harmonized_seurat,
        reduction = &quot;umap&quot;,
        label = TRUE,
        label.size = 6)
dev.off()

# let's visualize cells expressing supercluster markers:
# CD31: PECAM1
markers &lt;- c(&quot;EPCAM&quot;, &quot;PECAM1&quot;, &quot;COL1A1&quot;, &quot;PDGFRA&quot;, &quot;RGS5&quot;, &quot;CD79A&quot;, &quot;LYZ&quot;, &quot;CD3D&quot;, &quot;TPSAB1&quot;)

png(filename = &quot;normal_umap_superCluster_cells.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
FeaturePlot(object = harmonized_seurat,
            features = markers,
            order = TRUE,
            min.cutoff = &quot;q10&quot;,
            reduction = &quot;umap&quot;,
            label = TRUE,
            repel = TRUE)

dev.off()" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>________________________Reading the files______________________#</span>

<span class="pl-smi">samples</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>SRR12603780<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603781<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>SRR12603788<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> read files inot Seurat objects</span>
<span class="pl-k">for</span> (<span class="pl-smi">file</span> <span class="pl-k">in</span> <span class="pl-smi">samples</span>){
  print(paste0(<span class="pl-smi">file</span>))
  <span class="pl-smi">seurat_data</span> <span class="pl-k">&lt;-</span> Read10X(<span class="pl-v">data.dir</span> <span class="pl-k">=</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>../filtered/<span class="pl-pds">"</span></span>, <span class="pl-smi">file</span>))
  <span class="pl-smi">seurat_obj</span> <span class="pl-k">&lt;-</span> CreateSeuratObject(<span class="pl-v">counts</span> <span class="pl-k">=</span> <span class="pl-smi">seurat_data</span>, 
                                   <span class="pl-v">min.features</span> <span class="pl-k">=</span> <span class="pl-c1">100</span>, 
                                   <span class="pl-v">project</span> <span class="pl-k">=</span> <span class="pl-smi">file</span>)
  assign(<span class="pl-smi">file</span>, <span class="pl-smi">seurat_obj</span>)
}


<span class="pl-c"><span class="pl-c">#</span> now merging all objects inot one Seurat obj</span>

<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">SRR12603780</span>,
                       <span class="pl-v">y</span> <span class="pl-k">=</span> c(<span class="pl-smi">SRR12603781</span>,
                             <span class="pl-smi">SRR12603788</span>),
                       <span class="pl-v">add.cell.id</span> <span class="pl-k">=</span> <span class="pl-smi">samples</span>)
                       
<span class="pl-c"><span class="pl-c">#</span>________________________Filteration____________________________#</span>
<span class="pl-c"><span class="pl-c">#</span> reading sampleInformation:</span>
<span class="pl-smi">sampleInformation</span> <span class="pl-k">&lt;-</span> read.csv(<span class="pl-s"><span class="pl-pds">"</span>../sampleInfo.csv<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span> Compute percent mito ratio</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;-</span> PercentageFeatureSet(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">merged_seurat</span>, <span class="pl-v">pattern</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>^MT-<span class="pl-pds">"</span></span>)
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">mitoRatio</span> <span class="pl-k">/</span> <span class="pl-c1">100</span>

<span class="pl-c"><span class="pl-c">#</span> adding cell column</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">$</span><span class="pl-smi">cells</span> <span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)
<span class="pl-c"><span class="pl-c">#</span> merging with sample information</span>
<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>, <span class="pl-smi">sampleInformation</span>)
<span class="pl-c"><span class="pl-c">#</span> re-setting the rownames</span>
rownames(<span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">merged_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">cells</span>


<span class="pl-c"><span class="pl-c">#</span> Filteration</span>

<span class="pl-smi">filtered_seurat</span> <span class="pl-k">&lt;-</span> subset(<span class="pl-smi">merged_seurat</span>, 
                          <span class="pl-v">subset</span><span class="pl-k">=</span> <span class="pl-smi">nCount_RNA</span> <span class="pl-k">&gt;</span><span class="pl-k">=</span> <span class="pl-c1">1000</span> <span class="pl-k">&amp;</span>
                          <span class="pl-smi">nFeature_RNA</span> <span class="pl-k">&lt;</span><span class="pl-k">=</span> <span class="pl-c1">6000</span> <span class="pl-k">&amp;</span> 
                          <span class="pl-smi">mitoRatio</span> <span class="pl-k">&lt;</span> <span class="pl-c1">0.10</span>)

<span class="pl-c"><span class="pl-c">#</span>________________________Integration using Harmony____________________________#</span>
<span class="pl-c"><span class="pl-c">#</span>integration using harmony need sevral steps to be undertaken:</span>

<span class="pl-c"><span class="pl-c">#</span> Perform log-normalization and feature selection, as well as SCT normalization on global object</span>
<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">filtered_seurat</span> %<span class="pl-k">&gt;</span>%
    NormalizeData() %<span class="pl-k">&gt;</span>%
    FindVariableFeatures(<span class="pl-v">selection.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vst<span class="pl-pds">"</span></span>, <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">3000</span>) %<span class="pl-k">&gt;</span>% 
    ScaleData() %<span class="pl-k">&gt;</span>%
    SCTransform(<span class="pl-v">vars.to.regress</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>mitoRatio<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>))

<span class="pl-c"><span class="pl-c">#</span> Calculate PCs using variable features determined by SCTransform (3000 by default)</span>
<span class="pl-smi">merged_seurat</span> <span class="pl-k">&lt;-</span> RunPCA(<span class="pl-smi">merged_seurat</span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">npcs</span> <span class="pl-k">=</span> <span class="pl-c1">50</span>)

<span class="pl-c"><span class="pl-c">#</span> Integration</span>
<span class="pl-c"><span class="pl-c">#</span>install.packages("harmony")</span>

library(<span class="pl-smi">harmony</span>)

<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> RunHarmony(<span class="pl-smi">merged_seurat</span>, 
				<span class="pl-v">group.by.vars</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Surgery_Type<span class="pl-pds">"</span></span>), 
				<span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>, <span class="pl-v">assay.use</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">reduction.save</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)

<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> RunUMAP(<span class="pl-smi">harmonized_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">40</span>)

<span class="pl-c"><span class="pl-c">#</span>Cluster identification and Inspect the effects of Harmony batch removel ____________#</span>

<span class="pl-c"><span class="pl-c">#</span> to set reduction to harmony and find the clusters</span>
<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> FindNeighbors(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)
<span class="pl-smi">harmonized_seurat</span> <span class="pl-k">&lt;-</span> FindClusters(<span class="pl-smi">harmonized_seurat</span>, <span class="pl-v">resolution</span> <span class="pl-k">=</span> c(<span class="pl-c1">0.1</span>, <span class="pl-c1">0.2</span>, <span class="pl-c1">0.4</span>, <span class="pl-c1">0.6</span>, <span class="pl-c1">0.8</span>))

<span class="pl-c"><span class="pl-c">#</span> visualization</span>
Idents(<span class="pl-smi">harmonized_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>

<span class="pl-c"><span class="pl-c">#</span> Color cells based on the sample name</span>
<span class="pl-c"><span class="pl-c">#</span> Plot UMAP </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>normal_harmony_UMAP_y_sample.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">harmonized_seurat</span>,
        <span class="pl-v">group.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span>________________________SuperCluster Identification____________#</span>

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>normal_harmony_umap_cluster_with_label.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">harmonized_seurat</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span> let's visualize cells expressing supercluster markers:</span>
<span class="pl-c"><span class="pl-c">#</span> CD31: PECAM1</span>
<span class="pl-smi">markers</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>EPCAM<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>PECAM1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>COL1A1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>PDGFRA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RGS5<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD79A<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>LYZ<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>CD3D<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>TPSAB1<span class="pl-pds">"</span></span>)

png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>normal_umap_superCluster_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">harmonized_seurat</span>,
            <span class="pl-v">features</span> <span class="pl-k">=</span> <span class="pl-smi">markers</span>,
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>q10<span class="pl-pds">"</span></span>,
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_umap_superCluster_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="normal_umap_superCluster_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_umap_superCluster_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     It seems the epithelial cells are clustered together in cluster 4. So for further analysis, I will be focusing on these cells.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="epi_cell_ids &lt;- rownames(harmonized_seurat@meta.data)[harmonized_seurat@meta.data$SCT_snn_res.0.1 == '4']

epi_seurat &lt;- subset(filtered_seurat, subset = cells %in% epi_cell_ids)

# Perform log-normalization and feature selection, as well as SCT normalization on the global object
epi_seurat &lt;- epi_seurat %&gt;%
    NormalizeData() %&gt;%
    FindVariableFeatures(selection.method = &quot;vst&quot;, nfeatures = 3000) %&gt;% 
    ScaleData() %&gt;%
    SCTransform(vars.to.regress = c(&quot;mitoRatio&quot;, &quot;orig.ident&quot;))

# Calculate PCs using variable features determined by SCTransform (3000 by default)
epi_seurat &lt;- RunPCA(epi_seurat, assay = &quot;SCT&quot;, npcs = 50)

epi_seurat &lt;- RunHarmony(epi_seurat, 
				group.by.vars = c(&quot;orig.ident&quot;, &quot;Surgery_Type&quot;), 
				reduction = &quot;pca&quot;, assay.use = &quot;SCT&quot;, reduction.save = &quot;harmony&quot;)

epi_seurat &lt;- RunUMAP(epi_seurat, reduction = &quot;harmony&quot;, assay = &quot;SCT&quot;, dims = 1:40)


#Cluster identification and Inspect the effects of Harmony batch removel#

# to set reduction to harmony and find the clusters
epi_seurat &lt;- FindNeighbors(object = epi_seurat, reduction = &quot;harmony&quot;)
epi_seurat &lt;- FindClusters(epi_seurat, resolution = c(0.1, 0.2, 0.4, 0.6, 0.8))

# visualization
Idents(epi_seurat) &lt;- epi_seurat@meta.data$SCT_snn_res.0.1


# color cells based on the cluster
# Plot UMAP 
png(filename = &quot;normal_cluster_epi_harmony_UMAP.png&quot;, width = 16, height = 8.135, units = &quot;in&quot;, res = 300)
DimPlot(epi_seurat,
        reduction = &quot;umap&quot;,
        label = TRUE,
        label.size = 6)
dev.off()" dir="auto">
                     <pre><span class="pl-smi">epi_cell_ids</span> <span class="pl-k">&lt;-</span> rownames(<span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span>)[<span class="pl-smi">harmonized_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">'</span>4<span class="pl-pds">'</span></span>]

<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> subset(<span class="pl-smi">filtered_seurat</span>, <span class="pl-v">subset</span> <span class="pl-k">=</span> <span class="pl-smi">cells</span> <span class="pl-k">%in%</span> <span class="pl-smi">epi_cell_ids</span>)

<span class="pl-c"><span class="pl-c">#</span> Perform log-normalization and feature selection, as well as SCT normalization on the global object</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span> %<span class="pl-k">&gt;</span>%
    NormalizeData() %<span class="pl-k">&gt;</span>%
    FindVariableFeatures(<span class="pl-v">selection.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vst<span class="pl-pds">"</span></span>, <span class="pl-v">nfeatures</span> <span class="pl-k">=</span> <span class="pl-c1">3000</span>) %<span class="pl-k">&gt;</span>% 
    ScaleData() %<span class="pl-k">&gt;</span>%
    SCTransform(<span class="pl-v">vars.to.regress</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>mitoRatio<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>))

<span class="pl-c"><span class="pl-c">#</span> Calculate PCs using variable features determined by SCTransform (3000 by default)</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunPCA(<span class="pl-smi">epi_seurat</span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">npcs</span> <span class="pl-k">=</span> <span class="pl-c1">50</span>)

<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunHarmony(<span class="pl-smi">epi_seurat</span>, 
				<span class="pl-v">group.by.vars</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>orig.ident<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Surgery_Type<span class="pl-pds">"</span></span>), 
				<span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>pca<span class="pl-pds">"</span></span>, <span class="pl-v">assay.use</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">reduction.save</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)

<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunUMAP(<span class="pl-smi">epi_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>, <span class="pl-v">assay</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SCT<span class="pl-pds">"</span></span>, <span class="pl-v">dims</span> <span class="pl-k">=</span> <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">40</span>)


<span class="pl-c"><span class="pl-c">#</span>Cluster identification and Inspect the effects of Harmony batch removel#</span>

<span class="pl-c"><span class="pl-c">#</span> to set reduction to harmony and find the clusters</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> FindNeighbors(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>harmony<span class="pl-pds">"</span></span>)
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> FindClusters(<span class="pl-smi">epi_seurat</span>, <span class="pl-v">resolution</span> <span class="pl-k">=</span> c(<span class="pl-c1">0.1</span>, <span class="pl-c1">0.2</span>, <span class="pl-c1">0.4</span>, <span class="pl-c1">0.6</span>, <span class="pl-c1">0.8</span>))

<span class="pl-c"><span class="pl-c">#</span> visualization</span>
Idents(<span class="pl-smi">epi_seurat</span>) <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_seurat</span><span class="pl-k">@</span><span class="pl-smi">meta.data</span><span class="pl-k">$</span><span class="pl-smi">SCT_snn_res.0.1</span>


<span class="pl-c"><span class="pl-c">#</span> color cells based on the cluster</span>
<span class="pl-c"><span class="pl-c">#</span> Plot UMAP </span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>normal_cluster_epi_harmony_UMAP.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
DimPlot(<span class="pl-smi">epi_seurat</span>,
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_cluster_epi_harmony_UMAP.png" rel="noopener noreferrer" target="_blank">
                      <img alt="normal_cluster_epi_harmony_UMAP.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_cluster_epi_harmony_UMAP.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Marker identification

epi_markers &lt;- FindAllMarkers(object = epi_seurat, 
                          only.pos = TRUE,
                          logfc.threshold = 0.25) 

# mutate the markers dataframe
# Extract top 10 markers per cluster

epi_top10 &lt;- epi_markers %&gt;%
  mutate(delta_pct = (pct.1 - pct.2)) %&gt;%
  #filter(avg_log2FC &gt; 1.5) %&gt;%  # only keep rows where avg_log2FC &gt; 1.5
  group_by(cluster) %&gt;%
  top_n(n = 10, wt = delta_pct)


epi_cluster_markers_10 &lt;- epi_top10 %&gt;% 
                   group_by(cluster) %&gt;% 
                   summarize(genes = paste(gene, collapse = ","))
                   
                   
# Insepecting the expression patterns
png(filename = "normal_markers_cells.png", width = 16, height = 8.135, units = "in", res = 300)
FeaturePlot(object = epi_seurat,
            features = unlist(strsplit(epi_cluster_markers_10[2,]$genes, split = ",")),
            order = TRUE,
            min.cutoff = "q10",
            reduction = "umap",
            label = TRUE,
            repel = TRUE)

dev.off()
                   ' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Marker identification</span>

<span class="pl-smi">epi_markers</span> <span class="pl-k">&lt;-</span> FindAllMarkers(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, 
                          <span class="pl-v">only.pos</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
                          <span class="pl-v">logfc.threshold</span> <span class="pl-k">=</span> <span class="pl-c1">0.25</span>) 

<span class="pl-c"><span class="pl-c">#</span> mutate the markers dataframe</span>
<span class="pl-c"><span class="pl-c">#</span> Extract top 10 markers per cluster</span>

<span class="pl-smi">epi_top10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_markers</span> %<span class="pl-k">&gt;</span>%
  mutate(<span class="pl-v">delta_pct</span> <span class="pl-k">=</span> (<span class="pl-smi">pct.1</span> <span class="pl-k">-</span> <span class="pl-smi">pct.2</span>)) %<span class="pl-k">&gt;</span>%
  <span class="pl-c"><span class="pl-c">#</span>filter(avg_log2FC &gt; 1.5) %&gt;%  # only keep rows where avg_log2FC &gt; 1.5</span>
  group_by(<span class="pl-smi">cluster</span>) %<span class="pl-k">&gt;</span>%
  top_n(<span class="pl-v">n</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, <span class="pl-v">wt</span> <span class="pl-k">=</span> <span class="pl-smi">delta_pct</span>)


<span class="pl-smi">epi_cluster_markers_10</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">epi_top10</span> %<span class="pl-k">&gt;</span>% 
                   group_by(<span class="pl-smi">cluster</span>) %<span class="pl-k">&gt;</span>% 
                   summarize(<span class="pl-v">genes</span> <span class="pl-k">=</span> paste(<span class="pl-smi">gene</span>, <span class="pl-v">collapse</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>))
                   
                   
<span class="pl-c"><span class="pl-c">#</span> Insepecting the expression patterns</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>normal_markers_cells.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
FeaturePlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>,
            <span class="pl-v">features</span> <span class="pl-k">=</span> unlist(strsplit(<span class="pl-smi">epi_cluster_markers_10</span>[<span class="pl-c1">2</span>,]<span class="pl-k">$</span><span class="pl-smi">genes</span>, <span class="pl-v">split</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>,<span class="pl-pds">"</span></span>)),
            <span class="pl-v">order</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">min.cutoff</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>q10<span class="pl-pds">"</span></span>,
            <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>,
            <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
            <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)

dev.off()
                   </pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_markers_cells.png" rel="noopener noreferrer" target="_blank">
                      <img alt="normal_markers_cells.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_markers_cells.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        cluster
                       </th>
                       <th>
                        genes
                       </th>
                       <th>
                        cluster name
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        0
                       </td>
                       <td>
                        HS6ST3,SLC9A2,GSTM3,CLMN,GRK3,GRHL1,ZNF710,ERBB3,SLC17A5,ASTN2
                       </td>
                       <td>
                        luminal_intermediate_stage
                       </td>
                      </tr>
                      <tr>
                       <td>
                        1
                       </td>
                       <td>
                        GMNN,SLC7A11,ANXA10,NRARP,SLC12A6,TFF1,PTGS2,ODC1,DSP,MIR31HG
                       </td>
                       <td>
                        stromal_associated
                       </td>
                      </tr>
                      <tr>
                       <td>
                        2
                       </td>
                       <td>
                        EGR1,DUSP2,DDIT4,DDIT3,KLF10,GATA2,TXNIP,AC005920.2,AL691403.1,PLA2G6
                       </td>
                       <td>
                        early_stage_basal_cell
                       </td>
                      </tr>
                      <tr>
                       <td>
                        3
                       </td>
                       <td>
                        NBAT1,BICC1,ZNF350-AS1,SULT1E1,CASC15,KRT20,PTN,B4GALT3,RHEX,SCHLAP1
                       </td>
                       <td>
                        intermediated_luminal_cell
                       </td>
                      </tr>
                      <tr>
                       <td>
                        4
                       </td>
                       <td>
                        PALLD,SPOCK3,NRG1,MYO1B,NAV2,ETS1,CAV1,KRT5,FLNA,TUBB6
                       </td>
                       <td>
                        intermediate_stage_basal_cell
                       </td>
                      </tr>
                      <tr>
                       <td>
                        5
                       </td>
                       <td>
                        BCAT1,UPK2,MAL,ARMT1,SDHAF4,PTN,KRT20,SNX31,C4orf48,ISG15
                       </td>
                       <td>
                        luminal_differentiated
                       </td>
                      </tr>
                      <tr>
                       <td>
                        6
                       </td>
                       <td>
                        ADAMTS9-AS2,EBF1,SPARCL1,PLCB4,CAVIN1,ZEB1,SLC2A3,CAV1,TCF4,KLF2
                       </td>
                       <td>
                        early_stage_basal_cells
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     Reason for nomenclature:
                    </p>
                    <p dir="auto">
                     <strong>
                      cluster 0
                     </strong>
                    </p>
                    <p dir="auto">
                     This cluster includes genes that are expressed in the luminal layer of bladder tissue. However, the specific combination of markers suggests a relatively early or intermediate stage of luminal cell differentiation.
                    </p>
                    <p dir="auto">
                     <strong>
                      cluster 1
                     </strong>
                    </p>
                    <p dir="auto">
                     Based on the gene expression profile of GMNN, SLC7A11, ANXA10, NRARP, SLC12A6, TFF1, PTGS2, ODC1, DSP, and MIR31HG, a possible name for the group of cells could be "Stromal-Associated Cluster" or "Mesenchymal-Enriched Cluster". This suggestion is based on the presence of genes like ANXA10 and DSP, which are typically associated with mesenchymal or stromal cell types. Additionally, genes like PTGS2 and SLC7A11 are commonly expressed in stromal or immune cells and may suggest an association with the tumor microenvironment or immune response.
                    </p>
                    <p dir="auto">
                     <strong>
                      cluster 2
                     </strong>
                    </p>
                    <p dir="auto">
                     This cluster consists of genes associated with early differentiation processes and transcriptional regulation. The presence of genes like EGR1 and KLF10 suggests an early stage in the differentiation process.
                    </p>
                    <p dir="auto">
                     <strong>
                      cluster 3
                     </strong>
                     This cluster also contains markers associated with both basal and luminal cell characteristics. While KRT20 suggests a luminal presence, the combination of markers does not indicate a well-defined stage of luminal differentiation.
                    </p>
                    <p dir="auto">
                     <strong>
                      cluster 4
                     </strong>
                    </p>
                    <p dir="auto">
                     This cluster shows a mix of genes associated with both basal and luminal cell characteristics. While it may not represent the most differentiated luminal cells, it could be positioned in an intermediate stage of differentiation.
                    </p>
                    <p dir="auto">
                     <strong>
                      cluster 5
                     </strong>
                    </p>
                    <p dir="auto">
                     Luminal Differentiation Cluster: This name is suggested based on the presence of markers such as UPK2 and KRT20. UPK2 (Uroplakin 2) is a marker of luminal/umbrella cells in the urothelium, which lines the bladder and urethra. KRT20 is also a luminal marker and is associated with differentiated epithelial cells in various tissues. This suggests that the cluster may represent a group of cells undergoing luminal differentiation or with a mature luminal phenotype.
                    </p>
                    <p dir="auto">
                     <strong>
                      cluster 6
                     </strong>
                    </p>
                    <p dir="auto">
                     Basal-like: This name emphasizes the presence of markers like ZEB1, TCF4, and KLF2, which have been associated with basal-like features in various epithelial tissues. It suggests that the cluster may represent cells with a basal-like differentiation program.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='# Rename all identities
epi_seurat &lt;- RenameIdents(object = epi_seurat,
                                "0" = "intermediate_luminal",
                                "1" = "stromal_associated",
                                "2" = "early_basal_1",
                                "3" = "luminal[KRT20]_basal",
                                "4" = "luminal_basal[KRT5]",
                                "5" = "luminal_differentiated",
                                "6" = "early_basal_0")
# Plot the UMAP withy new labells
png(filename = "normal_labelled_epi.png", width = 16, height = 8.135, units = "in", res = 600)
DimPlot(object = epi_seurat, 
        reduction = "umap", 
        label = TRUE,
        label.size = 3,
        repel = TRUE)
dev.off()' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Rename all identities</span>
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RenameIdents(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>,
                                <span class="pl-s"><span class="pl-pds">"</span>0<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>intermediate_luminal<span class="pl-pds">"</span></span>,
                                <span class="pl-s"><span class="pl-pds">"</span>1<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>stromal_associated<span class="pl-pds">"</span></span>,
                                <span class="pl-s"><span class="pl-pds">"</span>2<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>early_basal_1<span class="pl-pds">"</span></span>,
                                <span class="pl-s"><span class="pl-pds">"</span>3<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>luminal[KRT20]_basal<span class="pl-pds">"</span></span>,
                                <span class="pl-s"><span class="pl-pds">"</span>4<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>luminal_basal[KRT5]<span class="pl-pds">"</span></span>,
                                <span class="pl-s"><span class="pl-pds">"</span>5<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>luminal_differentiated<span class="pl-pds">"</span></span>,
                                <span class="pl-s"><span class="pl-pds">"</span>6<span class="pl-pds">"</span></span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>early_basal_0<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span> Plot the UMAP withy new labells</span>
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>normal_labelled_epi.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">600</span>)
DimPlot(<span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, 
        <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>umap<span class="pl-pds">"</span></span>, 
        <span class="pl-v">label</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">label.size</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>,
        <span class="pl-v">repel</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_labelled_epi.png" rel="noopener noreferrer" target="_blank">
                      <img alt="normal_labelled_epi.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_labelled_epi.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Trajectories of normal epithelial cells
                     </h2>
                     <a aria-label="Permalink: Trajectories of normal epithelial cells" class="anchor" href="#trajectories-of-normal-epithelial-cells" id="user-content-trajectories-of-normal-epithelial-cells">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='epi_seurat$clusters &lt;- Idents(epi_seurat)
png(filename = "normal_trajectory.png", width = 16, height = 8.135, units = "in", res = 300)
epi_seurat &lt;- RunSlingshot(srt = epi_seurat, group.by = "clusters", reduction = "UMAP")
dev.off()' dir="auto">
                     <pre><span class="pl-smi">epi_seurat</span><span class="pl-k">$</span><span class="pl-smi">clusters</span> <span class="pl-k">&lt;-</span> Idents(<span class="pl-smi">epi_seurat</span>)
png(<span class="pl-v">filename</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>normal_trajectory.png<span class="pl-pds">"</span></span>, <span class="pl-v">width</span> <span class="pl-k">=</span> <span class="pl-c1">16</span>, <span class="pl-v">height</span> <span class="pl-k">=</span> <span class="pl-c1">8.135</span>, <span class="pl-v">units</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>in<span class="pl-pds">"</span></span>, <span class="pl-v">res</span> <span class="pl-k">=</span> <span class="pl-c1">300</span>)
<span class="pl-smi">epi_seurat</span> <span class="pl-k">&lt;-</span> RunSlingshot(<span class="pl-v">srt</span> <span class="pl-k">=</span> <span class="pl-smi">epi_seurat</span>, <span class="pl-v">group.by</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>clusters<span class="pl-pds">"</span></span>, <span class="pl-v">reduction</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>UMAP<span class="pl-pds">"</span></span>)
dev.off()</pre>
                    </div>
                    <p dir="auto">
                     <a href="https://github.com/hamidghaedi/scRNA_seq-analysis/blob/main/images/normal_trajectory.png" rel="noopener noreferrer" target="_blank">
                      <img alt="normal_trajectory.png" src="https://github.com/hamidghaedi/scRNA_seq-analysis/raw/main/images/normal_trajectory.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="UidnoeBwp_0UMRm_YacJ_N3dh7qjTs2ivXq-coslG4RF__gOfHvfnRd948Y385Qyq2eqWUPntjjMqZrZjoaFpw">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 scRNA analysis of public dataset on bladder cancer
                </p>
                <h3 class="sr-only">
                 Topics
                </h3>
                <div class="my-3">
                 <div class="f6">
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:cancer" data-view-component="true" href="/topics/cancer" title="Topic: cancer">
                   cancer
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:transcriptomics" data-view-component="true" href="/topics/transcriptomics" title="Topic: transcriptomics">
                   transcriptomics
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:scrna-seq-analysis" data-view-component="true" href="/topics/scrna-seq-analysis" title="Topic: scrna-seq-analysis">
                   scrna-seq-analysis
                  </a>
                 </div>
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/hamidghaedi/scRNA_seq-analysis/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   6
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   2
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   3
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fhamidghaedi%2FscRNA_seq-analysis&amp;report=hamidghaedi+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/hamidghaedi/packages?repo_name=scRNA_seq-analysis">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/hamidghaedi/scRNA_seq-analysis/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/hamidghaedi/scRNA_seq-analysis/graphs/contributors">
                 Contributors
                 <span class="Counter ml-1" data-view-component="true" title="2">
                  2
                 </span>
                </a>
               </h2>
               <include-fragment aria-busy="true" aria-label="Loading contributors" src="/hamidghaedi/scRNA_seq-analysis/contributors_list?count=2&amp;current_repository=scRNA_seq-analysis&amp;items_to_show=2">
                <ul class="list-style-none">
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                </ul>
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       Â© 2024 GitHub,Â Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You canât perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
