<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-96df017957ee.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-27057bd9ed0b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-369bd99876f6.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-fb4b8d40f206.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-ee3fc84d7fb0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-119a649bdc24.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-1a8626dd714a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-b481106ce166.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         harrisonlab/fusarium: Fusarium Genomics
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="D0A2:3DD321:B95F8B:C9F18B:6604F475" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IkQwQTI6M0REMzIxOkI5NUY4QjpDOUYxOEI6NjYwNEY0NzUiLCJ2aXNpdG9yX2lkIjoiNTU2MzY4NTk5MTA4NDgyOTI4NiIsInJlZ2lvbl9lZGdlIjoic291dGhlYXN0YXNpYSIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="8a1b6fdb621fa65ef8a637c5f576257e5752743e5b66ff4ff4c3760d79fd1522" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:19213040" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Fusarium Genomics. Contribute to harrisonlab/fusarium development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/harrisonlab/fusarium" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/3c0075cfb1f6d2d3accf9613c78d18c492c3dd413519af0a5f5afd61ab65cb8a/harrisonlab/fusarium" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="harrisonlab/fusarium: Fusarium Genomics" name="twitter:title">
                       <meta content="Fusarium Genomics. Contribute to harrisonlab/fusarium development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/3c0075cfb1f6d2d3accf9613c78d18c492c3dd413519af0a5f5afd61ab65cb8a/harrisonlab/fusarium" property="og:image"/>
                        <meta content="Fusarium Genomics. Contribute to harrisonlab/fusarium development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="harrisonlab/fusarium: Fusarium Genomics" property="og:title"/>
                        <meta content="https://github.com/harrisonlab/fusarium" property="og:url"/>
                        <meta content="Fusarium Genomics. Contribute to harrisonlab/fusarium development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjEyMzYyOTcxNiwidCI6MTcxMTYwMDc2MX0=--6246fe40bbf37bbe39ab474450352558f4a8859c722e18201177581cc6da9b54" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="5ce037db78b05f1253451cd9cc9f2fa41350c4f00042e4672de9acd5aa9800d3" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="b80a03d97b3d017a291b86d18cd3df7764386850635dc33cd29a90c2aaa22e18" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/harrisonlab/fusarium git https://github.com/harrisonlab/fusarium.git" name="go-import"/>
                        <meta content="7422262" name="octolytics-dimension-user_id">
                         <meta content="harrisonlab" name="octolytics-dimension-user_login">
                          <meta content="19213040" name="octolytics-dimension-repository_id">
                           <meta content="harrisonlab/fusarium" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="19213040" name="octolytics-dimension-repository_network_root_id">
                               <meta content="harrisonlab/fusarium" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/harrisonlab/fusarium" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-9041b775dcfb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-32fb838a-ead1-47f7-a72f-76f243927482" data-view-component="true" id="dialog-show-dialog-32fb838a-ead1-47f7-a72f-76f243927482" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-32fb838a-ead1-47f7-a72f-76f243927482-description" aria-labelledby="dialog-32fb838a-ead1-47f7-a72f-76f243927482-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-32fb838a-ead1-47f7-a72f-76f243927482">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-32fb838a-ead1-47f7-a72f-76f243927482-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-32fb838a-ead1-47f7-a72f-76f243927482" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-32fb838a-ead1-47f7-a72f-76f243927482-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-aac126d2-1773-4cff-a5cc-51ef4e5871f6">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-780e67f7-9379-4a4f-9501-d83711d516bd">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-85e757bc-1eeb-4e8d-a7cb-183893767f73">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-5c5798eb-3702-45b7-ab82-c891afde17c7">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-112b5a81-ee72-406e-a91c-e1185eeb2a4c">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-f8624cda-726a-420a-ac29-4a8339e6d82f">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-da5e72c8-9ce0-4f09-8abe-790278094e89">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-bae2625a-2b2f-4ad6-b004-018e3b678eac">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               Â© 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: harrisonlab / fusarium" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              harrisonlab
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              fusarium
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"harrisonlab","screen_size":"compact"}' data-view-component="true" href="/harrisonlab">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-organization mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                  </path>
                 </svg>
                 harrisonlab
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"fusarium","screen_size":"compact"}' data-view-component="true" href="/harrisonlab/fusarium">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 fusarium
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"harrisonlab","screen_size":"full"}' data-hovercard-type="organization" data-hovercard-url="/orgs/harrisonlab/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/harrisonlab">
             <span class="AppHeader-context-item-label">
              harrisonlab
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"fusarium","screen_size":"full"}' data-view-component="true" href="/harrisonlab/fusarium">
             <span class="AppHeader-context-item-label">
              fusarium
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="lJ58xdBpICljnvTsGg8siBVUWdQ7RwMyWmu1l1ZaI0PEvXBNQuT_5tQErMK_mM7TIdiZPw9stqOEEMsp-56eqw" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="harrisonlab" data-current-owner="" data-current-repository="harrisonlab/fusarium" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="hSpCdfVAC0KsNHEEgppj8pioBAmsNwCmVFWKIwZ5A0LN4FVf0wZCN_uYA91kn5PjygxGoIpQ8Mk4t8uI4UcYDg" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:harrisonlab/fusarium">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump toâ¦" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump toâ¦" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-2a77336e-5f25-45fd-9b5e-f66ff6004fd6" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-49eb12df-5f69-48a5-bf52-19d6a0156e4d" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-49eb12df-5f69-48a5-bf52-19d6a0156e4d">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="0dBpwmzaTLmIGW6c36oiLkY3yxgAFkUC70IHJqybTmVDYnD6pAx_Izg7YJjhsyPfmzwG21o2o1fhwhKcWspL3g"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="nH09a9o6jJZ4NAPQHPtqfTYjKvGeno5Bsaf30ubflgrQj1_RgrvSTzNdvn1WLqdA_3NO5sibzJoS_yZq-ZjPXw">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="d1qVmXWvf5D2YcuFAFoHRIY3S9Mb2pzyucULfJh9gTRvB9g_Ii3Wk3V1ggZACFkuMPBOVDUDfGrUZamuoBJS_A">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="ZP4dSjBo05L-EMSyLQGVtNWCTjxJyjl4iCMOnjlZF36jq3JoKW8V8BnLXk_kuu6nBTqW9sxs_0iHQFA7oc12dA">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-cf1829d6-4f7a-489e-b0ee-d93f28e3418b" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-76b75131-0ec3-4bae-a584-5407123fc508" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-609d0e3f-a5c9-49a6-b8ab-8f22cd9baa22" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-fe210398-c0dc-479f-9a86-2e37b53fc29a" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-bf97f501-5682-4545-8a44-39a605e1f6a1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-c3304c91-18dd-4279-a772-79796535da9f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-689cc72b-fea4-4763-8068-fc13bfc75527" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-d9c3bc81-442f-41a1-8682-9980ab919388">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-d9c3bc81-442f-41a1-8682-9980ab919388" id="tooltip-689cc72b-fea4-4763-8068-fc13bfc75527" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-0e086d4e-7127-4941-996a-68cf649d5c1b" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-21da4d1d-415f-47bc-8a2d-84dff2381cad">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-21da4d1d-415f-47bc-8a2d-84dff2381cad" id="tooltip-0e086d4e-7127-4941-996a-68cf649d5c1b" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTYwMDc2MX0=--c0cafb030ae0843dfaccbae0d14e999c86918c1e42adfb40c70480163331758d" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-0f52d68e-7862-46e1-9ef5-5793a47fc9fa" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=19213040">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9" data-view-component="true" id="dialog-show-dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9-description" aria-labelledby="dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-bf6206f5-f8c9-4d31-abe4-12924d36dc9e" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-71cdc121-9a4e-418c-8fb0-5eb657bc6482" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-777ab05b-66c6-47d1-acdf-9bd04c5a6384" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-95a49377-8507-4220-ab55-0223f372d117" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-3816a981-8096-4fdb-9d29-8a6cc47efefc" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-c521e285-7947-467c-9b7b-cb2fda41ece6" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-89cf9ca0-cd06-400b-9210-ba1241b1cdc9-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-54910724-151f-4b5f-8618-d7483032f9e9" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-a6e0f28f-309e-4176-b77a-482ccb52cebd">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-c913f388-9446-4048-939b-55cd53e74854" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-a66877c8-b34d-49b7-b10a-584ca4c0422a">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-47382089-cf34-4731-8603-c1e50d204855">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-86f3aa6d-e39a-4df8-9329-fdb5f0d9ba92" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-c3bbab0c-6a01-494f-a783-febaa64fd89e" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-b5cb8e43-43b1-40b0-9be6-4c0297054eef">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-e93db7cc-5cd7-488a-9031-eb7caf2cccad">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-c54b95e1-274c-4305-926b-29238a0344b6">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-e7097735-f354-4fa7-8137-0494191c2114" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-46e558c9-f49b-400b-bad4-1ddf95a1210f" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-02756ccc-6431-451f-b036-e2e60aef5813">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-a5269d2f-7630-4337-b4b7-0ab8b35c84bb">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-80fe2edc-fad4-4b98-b400-a92ede182a0f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-5dfdf15b-9911-4cc7-92c5-57e12a1f860b">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /harrisonlab/fusarium" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /harrisonlab/fusarium/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /harrisonlab/fusarium/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /harrisonlab/fusarium/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /harrisonlab/fusarium/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Wiki","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g w" data-pjax="#repo-content-pjax-container" data-selected-links="repo_wiki /harrisonlab/fusarium/wiki" data-tab-item="i5wiki-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium/wiki" id="wiki-tab">
          <svg aria-hidden="true" class="octicon octicon-book UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
           </path>
          </svg>
          <span data-content="Wiki">
           Wiki
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="wiki-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /harrisonlab/fusarium/security" data-tab-item="i6security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/harrisonlab/fusarium/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /harrisonlab/fusarium/pulse" data-tab-item="i7insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/harrisonlab/fusarium/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-list" aria-haspopup="true" aria-labelledby="tooltip-b54838f0-8df4-4474-96ea-991917e43c11" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-button" popovertarget="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-button" id="tooltip-b54838f0-8df4-4474-96ea-991917e43c11" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-button" anchor-offset="normal" data-view-component="true" id="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-1dba4967-70af-4448-b8ce-2dd4138faae4-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium" id="item-dda8190f-6047-4c7d-bf4d-513b26fc4d92" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium/issues" id="item-8fa313d3-ee51-4fbf-a998-581ea6b575fe" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium/pulls" id="item-c81f0c12-44d5-4a47-ad39-dbe2a4504409" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium/actions" id="item-93d89c98-61dd-420a-8b81-110b4f2e3428" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium/projects" id="item-282fee23-1376-426f-9da2-e0bca51e12a5" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5wiki-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium/wiki" id="item-4dd8673d-2d6d-4d75-9682-234389041234" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Wiki
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium/security" id="item-9a1e7118-55d2-4038-850d-d23bbdddb3f7" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i7insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/harrisonlab/fusarium/pulse" id="item-83fdeb2c-5952-49ce-8c41-ed9d5e8ea731" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-cb4dd0e2-1b30-4980-bbe2-fd6a100aa662" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-e6c8e166-e815-413f-be6b-18bd7d139588" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-e6c8e166-e815-413f-be6b-18bd7d139588" id="tooltip-cb4dd0e2-1b30-4980-bbe2-fd6a100aa662" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTYwMDc2MX0=--c0cafb030ae0843dfaccbae0d14e999c86918c1e42adfb40c70480163331758d" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/harrisonlab/fusarium" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEyOk9yZ2FuaXphdGlvbjc0MjIyNjI=" data-targets="command-palette-scope.tokens" data-text="harrisonlab" data-type="owner" data-value="harrisonlab" style="white-space:nowrap;line-height:20px;">
         harrisonlab
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkxOTIxMzA0MA==" data-targets="command-palette-scope.tokens" data-text="fusarium" data-type="repository" data-value="fusarium" style="white-space:nowrap;line-height:20px;">
         fusarium
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-5ea84d0b-c44b-4a01-9441-30967fc2a5a5" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkxOTIxMzA0MA==" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         Weâve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="harrisonlab" data-scope-id="MDEyOk9yZ2FuaXphdGlvbjc0MjIyNjI=" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="fusarium" data-scope-id="MDEwOlJlcG9zaXRvcnkxOTIxMzA0MA==" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="harrisonlab" data-scope-id="MDEyOk9yZ2FuaXphdGlvbjc0MjIyNjI=" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="fusarium" data-scope-id="MDEwOlJlcG9zaXRvcnkxOTIxMzA0MA==" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="js-NHitX8EqELVAhYcFhEE8ipPOZWmsGuxUQjelUiodeoyc6vo0m7WcQ-kbyGfi26iFvhh19QjolvzU4XYWcMA">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="â­ï¸ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="F8BM1zcDpv5JM1APi9FfE3hvpJ6wnw-JD-C_AIfeSyt36BAm-C30mp94z0to05sleYX67xxtsbOjysXJZbb_hQ">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block" height="24" src="https://avatars.githubusercontent.com/u/7422262?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/harrisonlab/fusarium">
            fusarium
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/19213040/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":19213040,"originating_url":"https://github.com/harrisonlab/fusarium","user_id":126681664}}' data-hydro-click-hmac="8d45750a54c5d52cf381217f3b2fedd92dbc6a770d5fbe76ebb7328aab118d3e" data-view-component="true" href="/harrisonlab/fusarium/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="2">
              2
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-8ded7f70-ff93-42f4-bd37-7dd181863089" popover="manual">
              Fork your own copy of harrisonlab/fusarium
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-19213040">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/harrisonlab/fusarium/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-19213040" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-13ffc9">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/harrisonlab/fusarium/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="pve6MNlk31anZ6OmMSdE6d9L2ZQasen-BXdtM1uGJrEut8CkpXQ_kMarvyWr-DeQau6Uiv2us2PA13ZoEeWbeQ">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="lwk_ozEcL-oJX2Pyz-iPIfVLjvBuaVvFYFthSKro3uAfSUU3TQzPLGiTf3FVN_xYQO7D7ol2AVil-3oT4ItjKA">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (2)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":19213040,"originating_url":"https://github.com/harrisonlab/fusarium","user_id":126681664}}' data-hydro-click-hmac="b3ab2b8d27d3ab3571d6cf8576b2ba23b78d35435d787daf598d34326d505fb1" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="2 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="2">
                  2
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-19213040-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/harrisonlab/fusarium/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-19213040-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-a873a1">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/harrisonlab/fusarium/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="PV-muve1GBB7Wds4EsGaWu9wnSxlClX9IUtaLq3sC7hqxZVhKFCQlhxn2SP2iHyeTWtMlS3IGPEOXerWFeNorQ">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (2)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":19213040,"originating_url":"https://github.com/harrisonlab/fusarium","user_id":126681664}}' data-hydro-click-hmac="f9c7bcad9fd67c8ceb58d7797e3cf06f3a7e19cc101a2570847468e27ad53143" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="2 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="2">
                 2
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-19213040-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/harrisonlab/fusarium/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-19213040-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-153557">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":19213040,"originating_url":"https://github.com/harrisonlab/fusarium","user_id":126681664}}' data-hydro-click-hmac="47393209b87c76b16b7fbe568487ee7b81bae3d6dea4dc938505d9850d371ed5" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="h9KOzV72_2sdO4C-iPhFlb0xxtb_bTm-7Sq8QQRZibOfEoK3QEmKavEot7RUGMPi4hRVwyo3kmrP0U61U-g3Kg">
                   <input name="repository_id" type="hidden" value="19213040"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="Ch793JhsS-o3xu7_McF0rn3JCsOVt6UYjD2fXYuk3eoS3vGmhtM-69vV2fXtIfLZIuyZ1kDtDsyuxm2p3BVjcw">
                   <input name="repository_id" type="hidden" value="19213040"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-e889eeb1-3204-4eef-81d8-d426d79b8fcd" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-944bea6e-5bb7-430b-bc60-4648de3ec37a" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":19213040,"originating_url":"https://github.com/harrisonlab/fusarium","user_id":126681664}}' data-hydro-click-hmac="8d45750a54c5d52cf381217f3b2fedd92dbc6a770d5fbe76ebb7328aab118d3e" data-view-component="true" group_item="true" href="/harrisonlab/fusarium/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-944bea6e-5bb7-430b-bc60-4648de3ec37a" popover="manual">
              Fork your own copy of harrisonlab/fusarium
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/harrisonlab/fusarium/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="cmFl9oVyc-7pywdrNn4ZgRGaGOJ0Bnjnvnjyw1JIEVv6IR9i-WKTKIgHG-isoWr4pD9V_JMZInp72OmYGCuskw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-7bf9c285-f940-4273-ad73-9fe1489a162c" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":19213040,"originating_url":"https://github.com/harrisonlab/fusarium","user_id":126681664}}' data-hydro-click-hmac="b3ab2b8d27d3ab3571d6cf8576b2ba23b78d35435d787daf598d34326d505fb1" data-view-component="true" id="icon-button-e146a27f-8c3d-4201-a11d-50e745f6b5e8" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-e146a27f-8c3d-4201-a11d-50e745f6b5e8" id="tooltip-7bf9c285-f940-4273-ad73-9fe1489a162c" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/harrisonlab/fusarium/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="-wik-So_hdlv19yx01zPqZ_lZFa95Kh08TGEklGd8VSskpci9doNXwjp3qo3FSltPf617_Um5XjeJzRq6ZKSQQ">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-538e142a-b977-4402-8e9b-0f6470f3000e" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":19213040,"originating_url":"https://github.com/harrisonlab/fusarium","user_id":126681664}}' data-hydro-click-hmac="f9c7bcad9fd67c8ceb58d7797e3cf06f3a7e19cc101a2570847468e27ad53143" data-view-component="true" id="icon-button-4688d685-8cec-444a-b6c7-698a572e8cb6" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-4688d685-8cec-444a-b6c7-698a572e8cb6" id="tooltip-538e142a-b977-4402-8e9b-0f6470f3000e" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           Fusarium Genomics
          </p>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/harrisonlab/fusarium/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              2
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/harrisonlab/fusarium/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              2
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              24
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/harrisonlab/fusarium/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/harrisonlab/fusarium/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/harrisonlab/fusarium/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/harrisonlab/fusarium/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/harrisonlab/fusarium/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/harrisonlab/fusarium/custom-properties" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-note mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25Zm1.75-.25a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25ZM3.5 6.25a.75.75 0 0 1 .75-.75h7a.75.75 0 0 1 0 1.5h-7a.75.75 0 0 1-.75-.75Zm.75 2.25h4a.75.75 0 0 1 0 1.5h-4a.75.75 0 0 1 0-1.5Z">
              </path>
             </svg>
             <span>
              Custom properties
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/harrisonlab/fusarium?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         harrisonlab/fusarium
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/harrisonlab/fusarium/spoofed_commit_check/44508d0a57b4f920f5fca77df7b1dfb5cf544b97">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-a58d7c11e858.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-654469d743cd.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-009ec6ff31b8.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":19213040,"defaultBranch":"master","name":"fusarium","ownerLogin":"harrisonlab","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2014-04-28T02:31:41.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/7422262?v=4","public":true,"private":false,"isOrgOwned":true},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"master","listCacheKey":"v0:1613741166.2985861","canEdit":true,"refType":"branch","currentOid":"44508d0a57b4f920f5fca77df7b1dfb5cf544b97"},"tree":{"items":[{"name":"AHDB_project","path":"AHDB_project","contentType":"directory"},{"name":"FoN_FoP_papers","path":"FoN_FoP_papers","contentType":"directory"},{"name":"RNAseq","path":"RNAseq","contentType":"directory"},{"name":"analysis","path":"analysis","contentType":"directory"},{"name":"assembly","path":"assembly","contentType":"directory"},{"name":"gene_annotation","path":"gene_annotation","contentType":"directory"},{"name":"gene_prediction","path":"gene_prediction","contentType":"directory"},{"name":"genome_submisison","path":"genome_submisison","contentType":"directory"},{"name":"minion_assemblies","path":"minion_assemblies","contentType":"directory"},{"name":"pathogen","path":"pathogen","contentType":"directory"},{"name":"repeat_masking","path":"repeat_masking","contentType":"directory"},{"name":"tools","path":"tools","contentType":"directory"},{"name":"Fus2_assembly.md","path":"Fus2_assembly.md","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"downloaded_genomes.md","path":"downloaded_genomes.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":15,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/harrisonlab/fusarium/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/harrisonlab/fusarium.git","showCloneWarning":false,"sshUrl":"git@github.com:harrisonlab/fusarium.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone harrisonlab/fusarium","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/harrisonlab/fusarium","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fharrisonlab%2Ffusarium","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/harrisonlab/fusarium/archive/refs/heads/master.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":null},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"614","overviewFiles":[{"displayName":"README.md","repoName":"fusarium","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e#HAPI\u003c/h1\u003e\u003ca id=\"user-content-hapi\" class=\"anchor\" aria-label=\"Permalink: #HAPI\" href=\"#hapi\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eScripts used for the analysis of Fusarium genomes\nNote - all this work was performed in the directory:\n/home/groups/harrisonlab/project_files/fusarium\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe following is a summary of the work presented in this Readme.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe following processes were applied to Fusarium genomes prior to analysis:\nData qc\nGenome assembly\nRepeatmasking\nGene prediction\nFunctional annotation\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAnalyses performed on these genomes involved BLAST searching for:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003els contigs were identified using:\nAlignment of raw reads to assembled genomes\nAssembly of remaining reads\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Building of directory structure\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3/\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tmkdir -p $ProjectDir/raw_dna/paired/F.proliferatum/A8/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.proliferatum/A8/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.avenaceum/PG8/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.avenaceum/PG8/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F\n\tmkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R\"\u003e\u003cpre\u003e\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3/\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.proliferatum/A8/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.proliferatum/A8/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.avenaceum/PG8/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.avenaceum/PG8/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSequence data was moved into the appropriate directories\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp $RawDatDir/Fproliferatum_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.proliferatum/A8/F/.\n\tcp $RawDatDir/Fproliferatum_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.proliferatum/A8/R/.\n\tcp $RawDatDir/FoxysporumN139_S2_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F/.\n\tcp $RawDatDir/FoxysporumN139_S2_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R/.\n\tcp $RawDatDir/FoxysporumPG3_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F/.\n\tcp $RawDatDir/FoxysporumPG3_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R/.\n\tcp $RawDatDir/FoxysporumPG18_S4_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F/.\n\tcp $RawDatDir/FoxysporumPG18_S4_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R/.\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp $RawDatDir/FOP1_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F/.\n\tcp $RawDatDir/FOP1_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R/.\n\tcp $RawDatDir/FOP5_S2_L001_R1_001.fastq.gz  $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F/.\n\tcp $RawDatDir/FOP5_S2_L001_R2_001.fastq.gz  $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R/.\n\tcp $RawDatDir/L5_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F/.\n\tcp $RawDatDir/L5_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R/.\n\tcp $RawDatDir/HB6_S4_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.\n\tcp $RawDatDir/HB6_S4_L001_R2_001.fastq.gz  $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp $RawDatDir/A12_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F/.\n\tcp $RawDatDir/A12_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R/.\n\tcp $RawDatDir/A13_S2_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F/.\n\tcp $RawDatDir/A13_S2_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R/.\n\tcp $RawDatDir/CB3_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F/.\n\tcp $RawDatDir/CB3_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R/.\n\tcp $RawDatDir/HB6_S5_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.\n\tcp $RawDatDir/HB6_S5_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.\n\tcp $RawDatDir/PG8_S4_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.avenaceum/PG8/F/.\n\tcp $RawDatDir/PG8_S4_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.avenaceum/PG8/R/.\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp $RawDatDir/FOP2_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F/.\n\tcp $RawDatDir/FOP2_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R/.\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0\n\tcp $RawDatDir/AF064_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F/.\n\tcp $RawDatDir/AF064_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R/.\n\tcp $RawDatDir/CH5-2_S2_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F/.\n\tcp $RawDatDir/CH5-2_S2_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R/.\"\u003e\u003cpre\u003e\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/Fproliferatum_S1_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.proliferatum/A8/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/Fproliferatum_S1_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.proliferatum/A8/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FoxysporumN139_S2_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FoxysporumN139_S2_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FoxysporumPG3_S3_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FoxysporumPG3_S3_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FoxysporumPG18_S4_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FoxysporumPG18_S4_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R/.\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FOP1_S1_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FOP1_S1_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FOP5_S2_L001_R1_001.fastq.gz  \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FOP5_S2_L001_R2_001.fastq.gz  \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/L5_S3_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/L5_S3_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/HB6_S4_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/HB6_S4_L001_R2_001.fastq.gz  \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/A12_S1_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/A12_S1_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/A13_S2_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/A13_S2_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/CB3_S3_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/CB3_S3_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/HB6_S5_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/HB6_S5_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/PG8_S4_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.avenaceum/PG8/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/PG8_S4_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.avenaceum/PG8/R/.\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FOP2_S1_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/FOP2_S1_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R/.\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/AF064_S3_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/AF064_S3_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/CH5-2_S2_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/CH5-2_S2_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R/.\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis process was repeated for RNAseq data:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/rna_seq\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F\n\tmkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R\n\n\tcp $RawDatDir/4_S1_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F/.\n\tcp $RawDatDir/4_S1_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R/.\n\tcp $RawDatDir/6_S2_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F/.\n\tcp $RawDatDir/6_S2_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R/.\n\tcp $RawDatDir/7_S3_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F/.\n\tcp $RawDatDir/7_S3_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R/.\n\tcp $RawDatDir/9_S4_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F/.\n\tcp $RawDatDir/9_S4_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R/.\"\u003e\u003cpre\u003e\tRawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/rna_seq\n\tProjectDir=/home/groups/harrisonlab/project_files/fusarium\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R\n\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/4_S1_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/4_S1_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/6_S2_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/6_S2_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/7_S3_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/7_S3_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/9_S4_L001_R1_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F/.\n\tcp \u003cspan class=\"pl-smi\"\u003e$RawDatDir\u003c/span\u003e/9_S4_L001_R2_001.fastq.gz \u003cspan class=\"pl-smi\"\u003e$ProjectDir\u003c/span\u003e/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R/.\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e#Data qc\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eprograms:\nfastqc\nfastq-mcf\nkmc\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eData quality was visualised using fastqc:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor RawData in $(ls raw_dna/paired/*/*/*/*.fastq.gz); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\techo $RawData;\n\t\tqsub $ProgDir/run_fastqc.sh $RawData\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRawData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els raw_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_fastqc.sh \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTrimming was performed on data to trim adapters from\nsequences and remove poor quality data. This was done with fastq-mcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFirstly, those strains with more than one run were identified:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor Strain in $(ls -d raw_dna/paired/*/*); do\n\tNumReads=$(ls $Strain/F/*.gz | wc -l);\n\t\tif [ $NumReads -gt 1 ]; then\n\t\t\techo \u0026quot;$Strain\u0026quot;;\n\t\t\techo \u0026quot;$NumReads\u0026quot;;\n\t\tfi;\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eStrain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d raw_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tNumReads=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.gz \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\t\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e [ \u003cspan class=\"pl-smi\"\u003e$NumReads\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-gt\u003c/span\u003e 1 ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\n\t\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$NumReads\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\t\u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\traw_dna/paired/F.oxysporum_fsp_cepae/Fus2\n\t2\n\traw_dna/paired/F.oxysporum_fsp_cepae/HB6\n\t2\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\traw_dna/paired/F.oxysporum_fsp_cepae/Fus2\n\t2\n\traw_dna/paired/F.oxysporum_fsp_cepae/HB6\n\t2\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTrimming was first performed on all strains that had a single run of data:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor StrainPath in $(ls -d raw_dna/paired/*/* | grep -v -e 'Fus2' -e 'HB6'); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc\n\t\tIlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa\n\t\tReadsF=$(ls $StrainPath/F/*.fastq*)\n\t\tReadsR=$(ls $StrainPath/R/*.fastq*)\n\t\techo $ReadsF\n\t\techo $ReadsR\n\t\tqsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eStrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d raw_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB6\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc\n\t\tIlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa\n\t\tReadsF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tReadsR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ReadsF\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ReadsR\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rna_qc_fastq-mcf.sh \u003cspan class=\"pl-smi\"\u003e$ReadsF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ReadsR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$IlluminaAdapters\u003c/span\u003e DNA\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTrimming was then performed for strains with multiple runs of data\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc\n\tIlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa\n\techo \u0026quot;Fus2\u0026quot;\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2\n\tReadsF=$(ls $StrainPath/F/s_6_1_sequence.fastq.gz)\n\tReadsR=$(ls $StrainPath/R/s_6_2_sequence.fastq.gz)\n\tqsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2\n\tReadsF=$(ls $StrainPath/F/FUS2_S2_L001_R1_001.fastq.gz)\n\tReadsR=$(ls $StrainPath/R/FUS2_S2_L001_R2_001.fastq.gz)\n\tqsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA\n\techo \u0026quot;HB6\u0026quot;\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6\n\tReadsF=$(ls $StrainPath/F/HB6_S4_L001_R1_001.fastq.gz)\n\tReadsR=$(ls $StrainPath/R/HB6_S4_L001_R2_001.fastq.gz)\n\tqsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6\n\tReadsF=$(ls $StrainPath/F/HB6_S5_L001_R1_001.fastq.gz)\n\tReadsR=$(ls $StrainPath/R/HB6_S5_L001_R2_001.fastq.gz)\n\tqsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA\"\u003e\u003cpre\u003e\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc\n\tIlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2\n\tReadsF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/s_6_1_sequence.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tReadsR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/s_6_2_sequence.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rna_qc_fastq-mcf.sh \u003cspan class=\"pl-smi\"\u003e$ReadsF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ReadsR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$IlluminaAdapters\u003c/span\u003e DNA\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2\n\tReadsF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/FUS2_S2_L001_R1_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tReadsR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/FUS2_S2_L001_R2_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rna_qc_fastq-mcf.sh \u003cspan class=\"pl-smi\"\u003e$ReadsF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ReadsR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$IlluminaAdapters\u003c/span\u003e DNA\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHB6\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6\n\tReadsF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/HB6_S4_L001_R1_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tReadsR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/HB6_S4_L001_R2_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rna_qc_fastq-mcf.sh \u003cspan class=\"pl-smi\"\u003e$ReadsF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ReadsR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$IlluminaAdapters\u003c/span\u003e DNA\n\tStrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6\n\tReadsF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/HB6_S5_L001_R1_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tReadsR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/HB6_S5_L001_R2_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rna_qc_fastq-mcf.sh \u003cspan class=\"pl-smi\"\u003e$ReadsF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ReadsR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$IlluminaAdapters\u003c/span\u003e DNA\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eData quality was visualised once again following trimming:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor RawData in $(ls qc_dna/paired/*/*/*/*.fq.gz | grep 'FOP2'); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\techo $RawData;\n\t\tqsub $ProgDir/run_fastqc.sh $RawData\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRawData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els qc_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_fastqc.sh \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ekmer counting was performed using kmc\nThis allowed estimation of sequencing depth and total genome size\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis was performed for strains with single runs of data\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor TrimPath in $(ls -d raw_dna/paired/*/* | grep -v -e 'Fus2' -e 'HB6' | grep 'FOP2'); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\tTrimF=$(ls $TrimPath/F/*.fastq*)\n\t\tTrimR=$(ls $TrimPath/R/*.fastq*)\n\t\techo $TrimF\n\t\techo $TrimR\n\t\tqsub $ProgDir/kmc_kmer_counting.sh $TrimF $TrimR\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTrimPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d raw_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB6\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\tTrimF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tTrimR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/R/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/kmc_kmer_counting.sh \u003cspan class=\"pl-smi\"\u003e$TrimF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eand for strains with muiltiple runs of data:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor TrimPath in $(ls -d raw_dna/paired/*/Fus2); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\tTrimF1=$(ls $TrimPath/F/s_6_1_sequence.fastq.gz)\n\t\tTrimR1=$(ls $TrimPath/R/s_6_2_sequence.fastq.gz)\n\t\techo $TrimF1\n\t\techo $TrimR1\n\t\tTrimF2=$(ls $TrimPath/F/FUS2_S2_L001_R1_001.fastq.gz)\n\t\tTrimR2=$(ls $TrimPath/R/FUS2_S2_L001_R2_001.fastq.gz)\n\t\techo $TrimF2\n\t\techo $TrimR2\n\t\tqsub $ProgDir/kmc_kmer_counting.sh $TrimF1 $TrimR1 $TrimF2 $TrimR2\n\tdone\n\tfor TrimPath in $(ls -d raw_dna/paired/*/HB6); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\tTrimF1=$(ls $TrimPath/F/HB6_S4_L001_R1_001.fastq.gz)\n\t\tTrimR1=$(ls $TrimPath/R/HB6_S4_L001_R2_001.fastq.gz)\n\t\techo $TrimF1\n\t\techo $TrimR1\n\t\tTrimF2=$(ls $TrimPath/F/HB6_S5_L001_R1_001.fastq.gz)\n\t\tTrimR2=$(ls $TrimPath/R/HB6_S5_L001_R2_001.fastq.gz)\n\t\techo $TrimF2\n\t\techo $TrimR2\n\t\tqsub $ProgDir/kmc_kmer_counting.sh $TrimF1 $TrimR1 $TrimF2 $TrimR2\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTrimPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d raw_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/Fus2\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\tTrimF1=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/F/s_6_1_sequence.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tTrimR1=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/R/s_6_2_sequence.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF1\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1\u003c/span\u003e\n\t\tTrimF2=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/F/FUS2_S2_L001_R1_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tTrimR2=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/R/FUS2_S2_L001_R2_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/kmc_kmer_counting.sh \u003cspan class=\"pl-smi\"\u003e$TrimF1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTrimPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d raw_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/HB6\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\tTrimF1=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/F/HB6_S4_L001_R1_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tTrimR1=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/R/HB6_S4_L001_R2_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF1\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1\u003c/span\u003e\n\t\tTrimF2=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/F/HB6_S5_L001_R1_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tTrimR2=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$TrimPath\u003c/span\u003e/R/HB6_S5_L001_R2_001.fastq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/kmc_kmer_counting.sh \u003cspan class=\"pl-smi\"\u003e$TrimF1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003emode kmer abundance prior to error correction was reported using the following\ncommands:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls qc_dna/kmc/*/*/*_true_kmer_summary.txt); do\nbasename $File;\ntail -n3 $File | head -n1 ;\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els qc_dna/kmc/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_true_kmer_summary.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nbasename \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\ntail -n3 \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e head -n1 \u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\tPG8_true_kmer_summary.txt\n\tThe mode kmer abundance is:  53 \u0026lt; fail\n\t125_true_kmer_summary.txt\n\tThe mode kmer abundance is:  46 \u0026lt; pass\n\t55_true_kmer_summary.txt\n\tThe mode kmer abundance is:  29 \u0026lt; pass\n\tA1-2_true_kmer_summary.txt\n\tThe mode kmer abundance is:  16 \u0026lt; pass\n\tA13_true_kmer_summary.txt\n\tThe mode kmer abundance is:  22 \u0026lt; pass\n\tA23_true_kmer_summary.txt\n\tThe mode kmer abundance is:  33 \u0026lt; pass\n\tA28_true_kmer_summary.txt\n\tThe mode kmer abundance is:  36 \u0026lt; pass\n\tCB3_true_kmer_summary.txt\n\tThe mode kmer abundance is:  21 \u0026lt; pass\n\tD2_true_kmer_summary.txt\n\tThe mode kmer abundance is:  11 \u0026lt; pass\n\tFus2_true_kmer_summary.txt\n\tThe mode kmer abundance is:  109 \u0026lt; 2lib\n\tHB17_true_kmer_summary.txt\n\tThe mode kmer abundance is:  27 \u0026lt; pass\n\tHB6_true_kmer_summary.txt\n\tThe mode kmer abundance is:  91 \u0026lt; 2 lib\n\tPG_true_kmer_summary.txt\n\tThe mode kmer abundance is:  58 \u0026lt; pass\n\tN139_true_kmer_summary.txt\n\tThe mode kmer abundance is:  26 \u0026lt; pass\n\tFOP1_true_kmer_summary.txt\n\tThe mode kmer abundance is:  32 \u0026lt;- fail\n\tFOP5_true_kmer_summary.txt\n\tThe mode kmer abundance is:  62 \u0026lt;- fail\n\tL5_true_kmer_summary.txt\n\tThe mode kmer abundance is:  35 \u0026lt;- pass\n\tPG18_true_kmer_summary.txt\n\tThe mode kmer abundance is:  24 \u0026lt;- pass\n\tPG3_true_kmer_summary.txt\n\tThe mode kmer abundance is:  28 \u0026lt;- pass\n\tA8_true_kmer_summary.txt\n\tThe mode kmer abundance is:  21 \u0026lt;- pass\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\tPG8_true_kmer_summary.txt\n\tThe mode kmer abundance is:  53 \u0026lt; fail\n\t125_true_kmer_summary.txt\n\tThe mode kmer abundance is:  46 \u0026lt; pass\n\t55_true_kmer_summary.txt\n\tThe mode kmer abundance is:  29 \u0026lt; pass\n\tA1-2_true_kmer_summary.txt\n\tThe mode kmer abundance is:  16 \u0026lt; pass\n\tA13_true_kmer_summary.txt\n\tThe mode kmer abundance is:  22 \u0026lt; pass\n\tA23_true_kmer_summary.txt\n\tThe mode kmer abundance is:  33 \u0026lt; pass\n\tA28_true_kmer_summary.txt\n\tThe mode kmer abundance is:  36 \u0026lt; pass\n\tCB3_true_kmer_summary.txt\n\tThe mode kmer abundance is:  21 \u0026lt; pass\n\tD2_true_kmer_summary.txt\n\tThe mode kmer abundance is:  11 \u0026lt; pass\n\tFus2_true_kmer_summary.txt\n\tThe mode kmer abundance is:  109 \u0026lt; 2lib\n\tHB17_true_kmer_summary.txt\n\tThe mode kmer abundance is:  27 \u0026lt; pass\n\tHB6_true_kmer_summary.txt\n\tThe mode kmer abundance is:  91 \u0026lt; 2 lib\n\tPG_true_kmer_summary.txt\n\tThe mode kmer abundance is:  58 \u0026lt; pass\n\tN139_true_kmer_summary.txt\n\tThe mode kmer abundance is:  26 \u0026lt; pass\n\tFOP1_true_kmer_summary.txt\n\tThe mode kmer abundance is:  32 \u0026lt;- fail\n\tFOP5_true_kmer_summary.txt\n\tThe mode kmer abundance is:  62 \u0026lt;- fail\n\tL5_true_kmer_summary.txt\n\tThe mode kmer abundance is:  35 \u0026lt;- pass\n\tPG18_true_kmer_summary.txt\n\tThe mode kmer abundance is:  24 \u0026lt;- pass\n\tPG3_true_kmer_summary.txt\n\tThe mode kmer abundance is:  28 \u0026lt;- pass\n\tA8_true_kmer_summary.txt\n\tThe mode kmer abundance is:  21 \u0026lt;- pass\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eEstimate coverage\u003c/h2\u003e\u003ca id=\"user-content-estimate-coverage\" class=\"anchor\" aria-label=\"Permalink: Estimate coverage\" href=\"#estimate-coverage\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor Miseq data:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for RawData in $(ls qc_dna/paired/*/*/*/*.fq.gz | grep -e 'N139'); do\necho $RawData;\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc;\nqsub $ProgDir/run_fastqc.sh $RawData;\nGenomeSz=57\nOutDir=$(dirname $RawData)\nqsub $ProgDir/sub_count_nuc.sh $GenomeSz $RawData $OutDir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRawData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els qc_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_fastqc.sh \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\nGenomeSz=57\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_count_nuc.sh \u003cspan class=\"pl-smi\"\u003e$GenomeSz\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  for StrainDir in $(ls -d qc_dna/paired/*/* | grep -w -e 'N139'); do\n    Strain=$(basename $StrainDir)\n    printf \u0026quot;$Strain\\t\u0026quot;\n    for File in $(ls qc_dna/paired/*/\u0026quot;$Strain\u0026quot;/*/*.txt); do\n      echo $(basename $File);\n      cat $File | tail -n1 | rev | cut -f2 -d ' ' | rev;\n    done | grep -v '.txt' | awk '{ SUM += $1} END { print SUM }'\n  done\"\u003e\u003cpre\u003e  \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eStrainDir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d qc_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n    Strain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ebasename \u003cspan class=\"pl-smi\"\u003e$StrainDir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els qc_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n      \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ebasename \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n      cat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n    \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{ SUM += $1} END { print SUM }\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n  \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e#Assembly\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAssembly was performed with:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSpades\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSpades Assembly\u003c/h2\u003e\u003ca id=\"user-content-spades-assembly\" class=\"anchor\" aria-label=\"Permalink: Spades Assembly\" href=\"#spades-assembly\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor StrainPath in $(ls -d qc_dna/paired/*/* | grep -v -e 'Fus2' -e 'HB6' | grep 'FOP2'); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades\n\t\tStrain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)\n\t\tOrganism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)\n\t\tF_Read=$(ls $StrainPath/F/*.fq.gz)\n\t\tR_Read=$(ls $StrainPath/R/*.fq.gz)\n\t\tOutDir=assembly/spades/$Organism/$Strain\n\t\tJobs=$(qstat | grep 'submit_SPA' | grep 'qw' | wc -l)\n\t\twhile [ $Jobs -gt 1 ]; do\n\t\t\tsleep 5m\n\t\t\tprintf \u0026quot;.\u0026quot;\n\t\t\tJobs=$(qstat | grep 'submit_SPA' | grep 'qw' | wc -l)\n\t\tdone\t\t\n\t\tprintf \u0026quot;\\n\u0026quot;\n\t\techo $F_Read\n\t\techo $R_Read\n\t\tqsub $ProgDir/submit_SPAdes.sh $F_Read $R_Read $OutDir correct 10\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eStrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d qc_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB6\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tF_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tR_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\n\t\tJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esubmit_SPA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eqw\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e [ \u003cspan class=\"pl-smi\"\u003e$Jobs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-gt\u003c/span\u003e 1 ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\t\tsleep 5m\n\t\t\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\t\tJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esubmit_SPA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eqw\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\t\t\n\t\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$F_Read\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$R_Read\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/submit_SPAdes.sh \u003cspan class=\"pl-smi\"\u003e$F_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$R_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e correct 10\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAssembly for PG8, FOP1 and FOP5 failed due to a lack of memory, as such the assembly was\nresubmitted with more RAM.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor StrainPath in $(ls -d qc_dna/paired/*/* | grep -e 'FOP5' -e 'PG8' -e 'FOP1'); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades\n\t\tStrain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)\n\t\tOrganism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)\n\t\tF_Read=$(ls $StrainPath/F/*.fq.gz)\n\t\tR_Read=$(ls $StrainPath/R/*.fq.gz)\n\t\tOutDir=assembly/spades/$Organism/$Strain\n\t\techo $F_Read\n\t\techo $R_Read\n\t\tqsub $ProgDir/submit_SPAdes_HiMem.sh $F_Read $R_Read $OutDir correct 10\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eStrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d qc_dna/paired/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP5\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tF_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tR_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$F_Read\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$R_Read\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/submit_SPAdes_HiMem.sh \u003cspan class=\"pl-smi\"\u003e$F_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$R_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e correct 10\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAssemblies were submitted for genomes with data from multiple sequencing runs:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for StrainPath in $(ls -d qc_dna/paired/F.*/HB6); do\n  echo $StrainPath\n    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries\n    Strain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)\n    Organism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)\n    echo $Strain\n    echo $Organism\n    TrimF1_Read=$(ls $StrainPath/F/HB6_S4_L001_R1_001_trim.fq.gz);\n    TrimR1_Read=$(ls $StrainPath/R/HB6_S4_L001_R2_001_trim.fq.gz);\n    TrimF2_Read=$(ls $StrainPath/F/HB6_S5_L001_R1_001_trim.fq.gz);\n    TrimR2_Read=$(ls $StrainPath/R/HB6_S5_L001_R2_001_trim.fq.gz);\n    echo $TrimF1_Read\n    echo $TrimR1_Read\n    echo $TrimF2_Read\n    echo $TrimR2_Read\n    OutDir=assembly/spades/$Organism/$Strain\n    qsub $ProgDir/subSpades_2lib.sh $TrimF1_Read $TrimR1_Read $TrimF2_Read $TrimR2_Read $OutDir correct 10\n  done\n\tfor StrainPath in $(ls -d qc_dna/paired/F.*/Fus2); do\n\t  echo $StrainPath\n\t    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries\n\t    Strain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)\n\t    Organism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)\n\t    echo $Strain\n\t    echo $Organism\n\t    TrimF1_Read=$(ls $StrainPath/F/s_6_1_sequence_trim.fq.gz);\n\t    TrimR1_Read=$(ls $StrainPath/R/s_6_2_sequence_trim.fq.gz);\n\t    TrimF2_Read=$(ls $StrainPath/F/FUS2_S2_L001_R1_001_trim.fq.gz);\n\t    TrimR2_Read=$(ls $StrainPath/R/FUS2_S2_L001_R2_001_trim.fq.gz);\n\t    echo $TrimF1_Read\n\t    echo $TrimR1_Read\n\t    echo $TrimF2_Read\n\t    echo $TrimR2_Read\n\t    OutDir=assembly/spades/$Organism/$Strain\n\t    qsub $ProgDir/subSpades_2lib_HiMem.sh $TrimF1_Read $TrimR1_Read $TrimF2_Read $TrimR2_Read $OutDir correct 10\n\t  done\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eStrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d qc_dna/paired/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/HB6\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n  \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e\n    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries\n    Strain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n    Organism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\n    TrimF1_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/HB6_S4_L001_R1_001_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n    TrimR1_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/HB6_S4_L001_R2_001_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n    TrimF2_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/HB6_S5_L001_R1_001_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n    TrimR2_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/HB6_S5_L001_R2_001_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF1_Read\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1_Read\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2_Read\u003c/span\u003e\n    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2_Read\u003c/span\u003e\n    OutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\n    qsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/subSpades_2lib.sh \u003cspan class=\"pl-smi\"\u003e$TrimF1_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e correct 10\n  \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eStrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d qc_dna/paired/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/Fus2\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t  \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e\n\t    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries\n\t    Strain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t    Organism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\n\t    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\n\t    TrimF1_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/s_6_1_sequence_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t    TrimR1_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/s_6_2_sequence_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t    TrimF2_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/F/FUS2_S2_L001_R1_001_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t    TrimR2_Read=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$StrainPath\u003c/span\u003e/R/FUS2_S2_L001_R2_001_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF1_Read\u003c/span\u003e\n\t    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1_Read\u003c/span\u003e\n\t    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2_Read\u003c/span\u003e\n\t    \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2_Read\u003c/span\u003e\n\t    OutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\n\t    qsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/subSpades_2lib_HiMem.sh \u003cspan class=\"pl-smi\"\u003e$TrimF1_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR1_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimF2_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TrimR2_Read\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e correct 10\n\t  \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eQuast\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast\n  for Assembly in $(ls assembly/spades/*/*/filtered_contigs/contigs_min_500bp.fasta | grep 'FOP2'); do\n\t# for Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep 'Fus2_edited_v2'); do\n  # for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do\n    Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\n    Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  \n    OutDir=assembly/spades/$Organism/$Strain/filtered_contigs\n    qsub $ProgDir/sub_quast.sh $Assembly $OutDir\n  done\"\u003e\u003cpre\u003e  ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast\n  \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/filtered_contigs/contigs_min_500bp.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep 'Fus2_edited_v2'); do\u003c/span\u003e\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do\u003c/span\u003e\n    Strain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n    Organism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e  \n    OutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/filtered_contigs\n    qsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_quast.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n  \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\n\u003cp dir=\"auto\"\u003eContigs were renamed in accordance with ncbi recomendations.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"  ProgDir=~/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants\n  touch tmp.csv\n\tfor Assembly in $(ls assembly/spades/*/*/filtered_contigs/contigs_min_500bp.fasta); do\n    Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\n    Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  \n    OutDir=assembly/spades/$Organism/$Strain/filtered_contigs\n    $ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/contigs_min_500bp_renamed.fasta --coord_file tmp.csv\n  done\n  rm tmp.csv\"\u003e\u003cpre\u003e  ProgDir=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants\n  touch tmp.csv\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/filtered_contigs/contigs_min_500bp.fasta\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n    Strain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n    Organism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e  \n    OutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/filtered_contigs\n    \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/remove_contaminants.py --inp \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e --out \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/contigs_min_500bp_renamed.fasta --coord_file tmp.csv\n  \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n  rm tmp.csv\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA Bioproject and Biosample was made with NCBI genbank for submission of genomes.\nFollowing the creation of these submissions, the .fasta assembly was uploaded\nthrough the submission portal. A note was provided requesting that the assembly\nbe run through the contamination screen to aid a more detailed resubmission in\nfuture. The returned FCSreport.txt was downloaded from the NCBI webportal and\nused to correct the assembly to NCBI standards.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNCBI reports (FCSreport.txt) were manually downloaded to the following loactions:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep -v 'ncbi_edits' | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do\n\t\tStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\n\t\tOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  \n\t\tNCBI_report_dir=genome_submission/$Organism/$Strain/initial_submission\n\t\tmkdir -p $NCBI_report_dir\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/contigs_min_500bp_renamed.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi_edits\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e  \n\t\tNCBI_report_dir=genome_submission/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/initial_submission\n\t\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$NCBI_report_dir\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThese downloaded files were used to correct assemblies:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls assembly/spades/*/*/filtered_contigs/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'N139'); do\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nNCBI_report=$(ls genome_submission/$Organism/$Strain/initial_submission/FCSreport.txt)\nOutDir=assembly/spades/$Organism/$Strain/ncbi_edits\nmkdir -p $OutDir\nProgDir=~/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants\n$ProgDir/remove_contaminants.py --keep_mitochondria --inp $Assembly --out $OutDir/contigs_min_500bp_renamed.fasta --coord_file $NCBI_report \u0026gt; $OutDir/log.txt\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/filtered_contigs/contigs_min_500bp_renamed.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nNCBI_report=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els genome_submission/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/initial_submission/FCSreport.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/ncbi_edits\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nProgDir=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/remove_contaminants.py --keep_mitochondria --inp \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e --out \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/contigs_min_500bp_renamed.fasta --coord_file \u003cspan class=\"pl-smi\"\u003e$NCBI_report\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/log.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA second submission of assemblies to ncbi gave a second round of edits to make.\nThese reports (FCSreport.txt)  were stored in the following location:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep -w 'ncbi_edits' | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  \nNCBI_report_dir=genome_submission/$Organism/\u0026quot;$Strain\u0026quot;_ncbi/submission\nmkdir -p $NCBI_report_dir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/contigs_min_500bp_renamed.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi_edits\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e  \nNCBI_report_dir=genome_submission/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_ncbi/submission\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$NCBI_report_dir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThese downloaded files were used to correct assemblies:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls assembly/spades/*/*/ncbi_edits/contigs_min_500bp_renamed.fasta| grep 'ncbi_edits' | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'A8' -e 'PG' -e 'A23'); do\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nNCBI_report=$(ls genome_submission/$Organism/\u0026quot;$Strain\u0026quot;_ncbi/submission/FCSreport.txt)\nOutDir=assembly/spades/$Organism/$Strain/ncbi_edits2\nmkdir -p $OutDir\ntouch $NCBI_report\nProgDir=~/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants\n$ProgDir/remove_contaminants.py --keep_mitochondria --inp $Assembly --out $OutDir/contigs_min_500bp_renamed.fasta --coord_file $NCBI_report \u0026gt; $OutDir/log.txt\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/ncbi_edits/contigs_min_500bp_renamed.fasta\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi_edits\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nNCBI_report=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els genome_submission/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_ncbi/submission/FCSreport.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=assembly/spades/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/ncbi_edits2\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ntouch \u003cspan class=\"pl-smi\"\u003e$NCBI_report\u003c/span\u003e\nProgDir=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/remove_contaminants.py --keep_mitochondria --inp \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e --out \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/contigs_min_500bp_renamed.fasta --coord_file \u003cspan class=\"pl-smi\"\u003e$NCBI_report\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/log.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eQuast was used to collect details on these assemblies\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast\nfor Assembly in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'A8' -e 'PG' -e 'A23'); do\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  \necho \u0026quot;$Organism - $Strain\u0026quot;\nOutDir=$(dirname $Assembly)\nqsub $ProgDir/sub_quast.sh $Assembly $OutDir\ndone\"\u003e\u003cpre\u003eProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/ncbi_edits2/contigs_min_500bp_renamed.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e  \n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_quast.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRepeatmasking\u003c/h1\u003e\u003ca id=\"user-content-repeatmasking\" class=\"anchor\" aria-label=\"Permalink: Repeatmasking\" href=\"#repeatmasking\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRepeat masking was performed and used the following programs:\nRepeatmasker\nRepeatmodeler\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe best assemblies were used to perform repeatmasking\u003c/p\u003e\n\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking\nfor BestAss in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'A8' -e 'PG' -e 'A23'); do\n# for BestAss in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do\nStrain=$(echo $BestAss | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $BestAss | rev | cut -f4 -d '/' | rev)\nOutDir=repeat_masked/$Organism/\u0026quot;$Strain\u0026quot;_ncbi/ncbi_submission\nqsub $ProgDir/rep_modeling.sh $BestAss $OutDir\nqsub $ProgDir/transposonPSI.sh $BestAss $OutDir\ndone\"\u003e\u003cpre\u003eProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBestAss\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els assembly/spades/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/ncbi_edits2/contigs_min_500bp_renamed.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for BestAss in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BestAss\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BestAss\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=repeat_masked/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_ncbi/ncbi_submission\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rep_modeling.sh \u003cspan class=\"pl-smi\"\u003e$BestAss\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/transposonPSI.sh \u003cspan class=\"pl-smi\"\u003e$BestAss\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe published non-pathogen genome for isolate FO47 was also repeatmasked as\nthis isolate was also used in experimental work.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking\n\tFo47Ass=assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta\n\tfor BestAss in $(ls $Fo47Ass); do\n\t\tqsub $ProgDir/rep_modeling.sh $BestAss\n\t\tqsub $ProgDir/transposonPSI.sh $BestAss\n\tdone\"\u003e\u003cpre\u003e\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking\n\tFo47Ass=assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBestAss\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$Fo47Ass\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rep_modeling.sh \u003cspan class=\"pl-smi\"\u003e$BestAss\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/transposonPSI.sh \u003cspan class=\"pl-smi\"\u003e$BestAss\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFo47 masked contigs were noted to still contain additional text following spaces.\nThis disrupted downstream programs.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls repeat_masked/F.oxysporum/fo47/*/fo47_contigs_*masked.fa); do\nsed -i 's/ of Fusarium oxysporum Fo47//g' $File\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/F.oxysporum/fo47/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/fo47_contigs_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emasked.fa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nsed -i \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/ of Fusarium oxysporum Fo47//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe TransposonPSI masked bases were used to mask additional bases from the\nrepeatmasker / repeatmodeller softmasked and hardmasked files.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)\n# while [ $Jobs -gt 0 ]; do\n# sleep 10\n# printf \u0026quot;.\u0026quot;\n# Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)\n# done\n# printf \u0026quot;\\n\u0026quot;\nfor File in $(ls repeat_masked/*/*/*/*_contigs_softmasked.fa | grep 'ncbi' | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -v 'old' | grep 'N139'); do\nOutDir=$(dirname $File)\nTPSI=$(ls $OutDir/*_contigs_unmasked.fa.TPSI.allHits.chains.gff3)\nOutFile=$(echo $File | sed 's/_contigs_softmasked.fa/_contigs_softmasked_repeatmasker_TPSI_appended.fa/g')\necho \u0026quot;$OutFile\u0026quot;\nbedtools maskfasta -soft -fi $File -bed $TPSI -fo $OutFile\necho \u0026quot;Number of masked bases:\u0026quot;\ncat $OutFile | grep -v '\u0026gt;' | tr -d '\\n' | awk '{print $0, gsub(\u0026quot;[a-z]\u0026quot;, \u0026quot;.\u0026quot;)}' | cut -f2 -d ' '\ndone\n# The number of N's in hardmasked sequence are not counted as some may be present within the assembly and were therefore not repeatmasked.\nfor File in $(ls repeat_masked/*/*/*/*_contigs_hardmasked.fa | grep 'ncbi' | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -v 'old' | grep 'N139'); do\nOutDir=$(dirname $File)\nTPSI=$(ls $OutDir/*_contigs_unmasked.fa.TPSI.allHits.chains.gff3)\nOutFile=$(echo $File | sed 's/_contigs_hardmasked.fa/_contigs_hardmasked_repeatmasker_TPSI_appended.fa/g')\necho \u0026quot;$OutFile\u0026quot;\nbedtools maskfasta -fi $File -bed $TPSI -fo $OutFile\n# echo \u0026quot;Number of masked bases:\u0026quot;\n# cat $OutFile | grep -v '\u0026gt;' | tr -d '\\n' | awk '{print $0, gsub(\u0026quot;[N]\u0026quot;, \u0026quot;.\u0026quot;)}' | cut -f2 -d ' '\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e while [ $Jobs -gt 0 ]; do\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sleep 10\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e printf \".\"\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e done\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e printf \"\\n\"\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nTPSI=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa.TPSI.allHits.chains.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutFile=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_contigs_softmasked.fa/_contigs_softmasked_repeatmasker_TPSI_appended.fa/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nbedtools maskfasta -soft -fi \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e -bed \u003cspan class=\"pl-smi\"\u003e$TPSI\u003c/span\u003e -fo \u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of masked bases:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\\n\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{print $0, gsub(\"[a-z]\", \".\")}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e The number of N's in hardmasked sequence are not counted as some may be present within the assembly and were therefore not repeatmasked.\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_hardmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nTPSI=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa.TPSI.allHits.chains.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutFile=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_contigs_hardmasked.fa/_contigs_hardmasked_repeatmasker_TPSI_appended.fa/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nbedtools maskfasta -fi \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e -bed \u003cspan class=\"pl-smi\"\u003e$TPSI\u003c/span\u003e -fo \u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e echo \"Number of masked bases:\"\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e cat $OutFile | grep -v '\u0026gt;' | tr -d '\\n' | awk '{print $0, gsub(\"[N]\", \".\")}' | cut -f2 -d ' '\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"repeat_masked/F.oxysporum_fsp_cepae/125_ncbi/ncbi_submission/125_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n3781475\nrepeat_masked/F.oxysporum_fsp_cepae/A13_ncbi/ncbi_submission/A13_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n4946858\nrepeat_masked/F.oxysporum_fsp_cepae/A23_ncbi/ncbi_submission/A23_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n3553622\nrepeat_masked/F.oxysporum_fsp_cepae/A28_ncbi/ncbi_submission/A28_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n4429343\nrepeat_masked/F.oxysporum_fsp_cepae/CB3_ncbi/ncbi_submission/CB3_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n2863123\nrepeat_masked/F.oxysporum_fsp_cepae/PG_ncbi/ncbi_submission/PG_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n3070545\nrepeat_masked/F.proliferatum/A8_ncbi/ncbi_submission/A8_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n1209471\nrepeat_masked/F.oxysporum_fsp_narcissi/N139_ncbi/ncbi_submission/N139_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n5689747\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003erepeat_masked/F.oxysporum_fsp_cepae/125_ncbi/ncbi_submission/125_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n3781475\nrepeat_masked/F.oxysporum_fsp_cepae/A13_ncbi/ncbi_submission/A13_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n4946858\nrepeat_masked/F.oxysporum_fsp_cepae/A23_ncbi/ncbi_submission/A23_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n3553622\nrepeat_masked/F.oxysporum_fsp_cepae/A28_ncbi/ncbi_submission/A28_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n4429343\nrepeat_masked/F.oxysporum_fsp_cepae/CB3_ncbi/ncbi_submission/CB3_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n2863123\nrepeat_masked/F.oxysporum_fsp_cepae/PG_ncbi/ncbi_submission/PG_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n3070545\nrepeat_masked/F.proliferatum/A8_ncbi/ncbi_submission/A8_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n1209471\nrepeat_masked/F.oxysporum_fsp_narcissi/N139_ncbi/ncbi_submission/N139_contigs_softmasked_repeatmasker_TPSI_appended.fa\nNumber of masked bases:\n5689747\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGene Prediction\u003c/h1\u003e\u003ca id=\"user-content-gene-prediction\" class=\"anchor\" aria-label=\"Permalink: Gene Prediction\" href=\"#gene-prediction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGene prediction followed three steps:\nPre-gene prediction\n- Quality of genome assemblies were assessed using Cegma to see how many core eukaryotic genes can be identified.\nGene model training\n- Gene models were trained using assembled RNAseq data as part of the Braker1 pipeline\nGene prediction\n- Gene models were used to predict genes in genomes as part of the the Braker1 pipeline. This used RNAseq data as hints for gene models.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePre-gene prediction\u003c/h1\u003e\u003ca id=\"user-content-pre-gene-prediction\" class=\"anchor\" aria-label=\"Permalink: Pre-gene prediction\" href=\"#pre-gene-prediction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eQuality of genome assemblies was assessed by looking for the gene space in the assemblies.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/cegma\ncd /home/groups/harrisonlab/project_files/fusarium\nfor Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi' | grep -v 'old'); do\n#for Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -w 'fo47'); do\necho $Genome;\nqsub $ProgDir/sub_cegma.sh $Genome dna;\ndone\"\u003e\u003cpre\u003eProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/cegma\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /home/groups/harrisonlab/project_files/fusarium\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eGenome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efor Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -w 'fo47'); do\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Genome\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_cegma.sh \u003cspan class=\"pl-smi\"\u003e$Genome\u003c/span\u003e dna\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOutputs were summarised using the commands:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor File in $(ls gene_pred/cegma/F*/*/*_dna_cegma.completeness_report | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi'); do\n\t\tStrain=$(echo $File | rev | cut -f2 -d '/' | rev);\n\t\tSpecies=$(echo $File | rev | cut -f3 -d '/' | rev);\n\t\tprintf \u0026quot;$Species\\t$Strain\\n\u0026quot;;\n\t\tcat $File | head -n18 | tail -n+4;printf \u0026quot;\\n\u0026quot;;\n\tdone \u0026gt; gene_pred/cegma/cegma_results_dna_summary.txt\n\n\tless gene_pred/cegma/cegma_results_dna_summary.txt\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/cegma/F\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_dna_cegma.completeness_report \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\tSpecies=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Species\u003c/span\u003e\\t\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\tcat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e head -n18 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+4\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e gene_pred/cegma/cegma_results_dna_summary.txt\n\n\tless gene_pred/cegma/cegma_results_dna_summary.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBusco has replaced CEGMA and was run to check gene space in assemblies\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -e 'Fus2' -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' -e 'canu_new' | grep -v 'old'); do\n\t\tStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\n\t\tOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\n\t\techo \u0026quot;$Organism - $Strain\u0026quot;\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco\n\t\t# BuscoDB=\u0026quot;Fungal\u0026quot;\n\t\tBuscoDB=$(ls -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9)\n\t\tOutDir=gene_pred/busco/$Organism/$Strain/assembly\n\t\tqsub $ProgDir/sub_busco2.sh $Assembly $BuscoDB $OutDir\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecanu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco\n\t\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BuscoDB=\"Fungal\"\u003c/span\u003e\n\t\tBuscoDB=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOutDir=gene_pred/busco/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/assembly\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_busco2.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$BuscoDB\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor File in $(ls gene_pred/busco/F*/*/assembly/*/short_summary_*.txt); do  \n\t\techo $File;\n\t\tcat $File | grep -e '(C)' -e 'Total';\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/busco/F\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/assembly/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/short_summary_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e  \n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\tcat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e(C)\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eTotal\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e#Gene prediction\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGene prediction was performed for Fusarium genomes. Two gene prediction\napproaches were used:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGene prediction using Braker1\nPrediction of all putative ORFs in the genome using the ORF finder (atg.pl)\napproach.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGene prediction 1 - Braker1 gene model training and prediction\u003c/h2\u003e\u003ca id=\"user-content-gene-prediction-1---braker1-gene-model-training-and-prediction\" class=\"anchor\" aria-label=\"Permalink: Gene prediction 1 - Braker1 gene model training and prediction\" href=\"#gene-prediction-1---braker1-gene-model-training-and-prediction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGene prediction was performed using Braker1.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFirst, RNAseq data was aligned to Fusarium genomes.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eqc of RNA seq data is detailed below:\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" for Folder in $(ls -d raw_rna/paired/F.oxysporum_fsp_cepae/*); do\n\t FolderName=$(echo $Folder | rev | cut -f1 -d '/' | rev);\n\t echo $FolderName;\n\t ls -lh $Folder/F;\n\t ls -lh $Folder/R;\n\tdone\"\u003e\u003cpre\u003e \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFolder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d raw_rna/paired/F.oxysporum_fsp_cepae/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t FolderName=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Folder\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FolderName\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t ls -lh \u003cspan class=\"pl-smi\"\u003e$Folder\u003c/span\u003e/F\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t ls -lh \u003cspan class=\"pl-smi\"\u003e$Folder\u003c/span\u003e/R\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis contained the following data:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\t55_72hrs_rep1\n\tsample013_1.combined.fastq.gz\n\tsample013_2.combined.fastq.gz\n\t55_72hrs_rep2\n\tsample014_1.combined.fastq.gz\n\tsample014_2.combined.fastq.gz\n\t55_72hrs_rep3\n\tsample015_1.combined.fastq.gz\n\tsample015_2.combined.fastq.gz\n\tcontrol_72hrs_rep1\n\tsample002_1.combined.fastq.gz\n\tsample002_2.combined.fastq.gz\n\tcontrol_72hrs_rep2\n\tsample004_1.combined.fastq.gz\n\tsample004_2.combined.fastq.gz\n\tcontrol_72hrs_rep3\n\tsample005_1.combined.fastq.gz\n\tsample005_2.combined.fastq.gz\n\tFO47_72hrs_rep1\n\tsample006_1.combined.fastq.gz\n\tsample006_2.combined.fastq.gz\n\tFO47_72hrs_rep2\n\tsample007_1.combined.fastq.gz\n\tsample007_2.combined.fastq.gz\n\tFO47_72hrs_rep3\n\tsample012_1.combined.fastq.gz\n\tsample012_2.combined.fastq.gz\n\tFus2_0hrs_prelim\n\t1_S1_L001_R1_001.fastq.gz\n\t1_S1_L001_R2_001.fastq.gz\n\tFus2_16hrs_prelim\n\t3_S2_L001_R1_001.fastq.gz\n\t3_S2_L001_R2_001.fastq.gz\n\tFus2_24hrs_prelim_rep1\n\t4_S3_L001_R1_001.fastq.gz\n\t4_S3_L001_R2_001.fastq.gz\n\tFus2_24hrs_prelim_rep2\n\tFus2_36hrs_prelim\n\t36hr-root_S10_L001_R1_001.fastq.gz\n\t36hr-root_S10_L001_R2_001.fastq.gz\n\tFus2_48hrs_prelim\n\t6_S4_L001_R1_001.fastq.gz\n\t6_S4_L001_R2_001.fastq.gz\n\tFus2_4hrs_prelim\n\t4hr-root_S7_L001_R1_001.fastq.gz\n\t4hr-root_S7_L001_R2_001.fastq.gz\n\tFus2_72hrs_prelim\n\t7_S5_L001_R1_001.fastq.gz\n\t7_S5_L001_R2_001.fastq.gz\n\tFus2_72hrs_rep1\n\tsample016_1.combined.fastq.gz\n\tsample016_2.combined.fastq.gz\n\tFus2_72hrs_rep2\n\tsample018_1.combined.fastq.gz\n\tsample018_2.combined.fastq.gz\n\tFus2_72hrs_rep3\n\tsample019_1.combined.fastq.gz\n\tsample019_2.combined.fastq.gz\n\tFus2_8hrs_prelim\n\t8hr-root_S8_L001_R1_001.fastq.gz\n\t8hr-root_S8_L001_R2_001.fastq.gz\n\tFus2_96hrs_prelim\n\t8_S6_L001_R1_001.fastq.gz\n\t8_S6_L001_R2_001.fastq.gz\n\tFus2_CzapekDox\n\t6_S2_L001_R1_001_fastqc  6_S2_L001_R1_001.fastq.gz\n\t6_S2_L001_R2_001_fastqc  6_S2_L001_R2_001.fastq.gz\n\tFus2_GlucosePeptone\n\t7_S3_L001_R1_001_fastqc  7_S3_L001_R1_001.fastq.gz\n\t7_S3_L001_R2_001_fastqc  7_S3_L001_R2_001.fastq.gz\n\tFus2_PDA\n\t9_S4_L001_R1_001_fastqc  9_S4_L001_R1_001.fastq.gz\n\t9_S4_L001_R2_001_fastqc  9_S4_L001_R2_001.fastq.gz\n\tFus2_PDB\n\t4_S1_L001_R1_001_fastqc  4_S1_L001_R1_001.fastq.gz\n\t4_S1_L001_R2_001_fastqc  4_S1_L001_R2_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\t55_72hrs_rep1\n\tsample013_1.combined.fastq.gz\n\tsample013_2.combined.fastq.gz\n\t55_72hrs_rep2\n\tsample014_1.combined.fastq.gz\n\tsample014_2.combined.fastq.gz\n\t55_72hrs_rep3\n\tsample015_1.combined.fastq.gz\n\tsample015_2.combined.fastq.gz\n\tcontrol_72hrs_rep1\n\tsample002_1.combined.fastq.gz\n\tsample002_2.combined.fastq.gz\n\tcontrol_72hrs_rep2\n\tsample004_1.combined.fastq.gz\n\tsample004_2.combined.fastq.gz\n\tcontrol_72hrs_rep3\n\tsample005_1.combined.fastq.gz\n\tsample005_2.combined.fastq.gz\n\tFO47_72hrs_rep1\n\tsample006_1.combined.fastq.gz\n\tsample006_2.combined.fastq.gz\n\tFO47_72hrs_rep2\n\tsample007_1.combined.fastq.gz\n\tsample007_2.combined.fastq.gz\n\tFO47_72hrs_rep3\n\tsample012_1.combined.fastq.gz\n\tsample012_2.combined.fastq.gz\n\tFus2_0hrs_prelim\n\t1_S1_L001_R1_001.fastq.gz\n\t1_S1_L001_R2_001.fastq.gz\n\tFus2_16hrs_prelim\n\t3_S2_L001_R1_001.fastq.gz\n\t3_S2_L001_R2_001.fastq.gz\n\tFus2_24hrs_prelim_rep1\n\t4_S3_L001_R1_001.fastq.gz\n\t4_S3_L001_R2_001.fastq.gz\n\tFus2_24hrs_prelim_rep2\n\tFus2_36hrs_prelim\n\t36hr-root_S10_L001_R1_001.fastq.gz\n\t36hr-root_S10_L001_R2_001.fastq.gz\n\tFus2_48hrs_prelim\n\t6_S4_L001_R1_001.fastq.gz\n\t6_S4_L001_R2_001.fastq.gz\n\tFus2_4hrs_prelim\n\t4hr-root_S7_L001_R1_001.fastq.gz\n\t4hr-root_S7_L001_R2_001.fastq.gz\n\tFus2_72hrs_prelim\n\t7_S5_L001_R1_001.fastq.gz\n\t7_S5_L001_R2_001.fastq.gz\n\tFus2_72hrs_rep1\n\tsample016_1.combined.fastq.gz\n\tsample016_2.combined.fastq.gz\n\tFus2_72hrs_rep2\n\tsample018_1.combined.fastq.gz\n\tsample018_2.combined.fastq.gz\n\tFus2_72hrs_rep3\n\tsample019_1.combined.fastq.gz\n\tsample019_2.combined.fastq.gz\n\tFus2_8hrs_prelim\n\t8hr-root_S8_L001_R1_001.fastq.gz\n\t8hr-root_S8_L001_R2_001.fastq.gz\n\tFus2_96hrs_prelim\n\t8_S6_L001_R1_001.fastq.gz\n\t8_S6_L001_R2_001.fastq.gz\n\tFus2_CzapekDox\n\t6_S2_L001_R1_001_fastqc  6_S2_L001_R1_001.fastq.gz\n\t6_S2_L001_R2_001_fastqc  6_S2_L001_R2_001.fastq.gz\n\tFus2_GlucosePeptone\n\t7_S3_L001_R1_001_fastqc  7_S3_L001_R1_001.fastq.gz\n\t7_S3_L001_R2_001_fastqc  7_S3_L001_R2_001.fastq.gz\n\tFus2_PDA\n\t9_S4_L001_R1_001_fastqc  9_S4_L001_R1_001.fastq.gz\n\t9_S4_L001_R2_001_fastqc  9_S4_L001_R2_001.fastq.gz\n\tFus2_PDB\n\t4_S1_L001_R1_001_fastqc  4_S1_L001_R1_001.fastq.gz\n\t4_S1_L001_R2_001_fastqc  4_S1_L001_R2_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePerform qc of RNAseq timecourse data\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor num in $(qstat | grep 'rna_qc' | cut -f1 -d ' '); do\n\t\tTreatment=$(head -n1 rna_qc_fastq-mcf.sh.o$num | cut -f10 -d '/')\n\t\tfor FilePath in $(raw_rna/paired/F.oxysporum_fsp_cepae/55_72hrs_rep1); do\n\t\t\techo $FilePath\n\t\t\tFileF=$(ls $FilePath/F/*.gz)\n\t\t\tFileR=$(ls $FilePath/R/*.gz)\n\t\t\tIlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa\n\t\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc\n\t\t\tqsub $ProgDir/rna_qc_fastq-mcf.sh $FileF $FileR $IlluminaAdapters RNA\n\t\t\tsleep 10\n\t\tdone\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003erna_qc\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tTreatment=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ehead -n1 rna_qc_fastq-mcf.sh.o\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f10 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFilePath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eraw_rna/paired/F.oxysporum_fsp_cepae/55_72hrs_rep1\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FilePath\u003c/span\u003e\n\t\t\tFileF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$FilePath\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\tFileR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$FilePath\u003c/span\u003e/R/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\tIlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa\n\t\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc\n\t\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/rna_qc_fastq-mcf.sh \u003cspan class=\"pl-smi\"\u003e$FileF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FileR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$IlluminaAdapters\u003c/span\u003e RNA\n\t\t\tsleep 10\n\t\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eData quality was visualised using fastqc:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor RawData in $(ls qc_rna/paired/F.oxysporum_fsp_cepae/*/*/*.fq.gz); do\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\techo $RawData;\n\t\tqsub $ProgDir/run_fastqc.sh $RawData\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRawData\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els qc_rna/paired/F.oxysporum_fsp_cepae/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_fastqc.sh \u003cspan class=\"pl-smi\"\u003e$RawData\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAligning\u003c/h4\u003e\u003ca id=\"user-content-aligning\" class=\"anchor\" aria-label=\"Permalink: Aligning\" href=\"#aligning\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInsert sizes of the RNA seq library were unknown until a draft alignment could\nbe made. To do this tophat and cufflinks were run, aligning the reads against a\nsingle genome. The fragment length and stdev were printed to stdout while\ncufflinks was running.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep 'FOP2'); do\n\t\tStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\n\t\tOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\n\t\techo \u0026quot;$Organism - $Strain\u0026quot;\n\t\tfor RNADir in $(ls -d qc_rna/paired/F.oxysporum_fsp_cepae/*); do\n\t\t\tTimepoint=$(echo $RNADir | rev | cut -f1 -d '/' | rev)\n\t\t\techo \u0026quot;$Timepoint\u0026quot;\n\t\t\tFileF=$(ls $RNADir/F/*_trim.fq.gz)\n\t\t\tFileR=$(ls $RNADir/R/*_trim.fq.gz)\n\t\t\tOutDir=alignment/$Organism/$Strain/$Timepoint\n\t\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq\n\t\t\tqsub $ProgDir/tophat_alignment.sh $Assembly $FileF $FileR $OutDir\n\t\tdone\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRNADir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d qc_rna/paired/F.oxysporum_fsp_cepae/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\t\tTimepoint=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$RNADir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Timepoint\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\t\tFileF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$RNADir\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\tFileR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$RNADir\u003c/span\u003e/R/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\tOutDir=alignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Timepoint\u003c/span\u003e\n\t\t\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq\n\t\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/tophat_alignment.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FileF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FileR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAlignments were concatenated prior to running cufflinks:\nCufflinks was run to produce the fragment length and stdev statistics:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor Assembly in $(ls repeat_masked/*/Fo47/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep 'FOP2'); do\n\t\tStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\n\t\tOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\n\t\tAcceptedHits=alignment/$Organism/$Strain/concatenated/concatenated.bam\n\t\tOutDir=gene_pred/cufflinks/$Organism/$Strain/concatenated_prelim\n\t\techo \u0026quot;$Organism - $Strain\u0026quot;\n\t\tmkdir -p $OutDir\n\t\tsamtools merge -f $AcceptedHits \\\n\t\talignment/$Organism/$Strain/55_72hrs_rep1/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/55_72hrs_rep2/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/55_72hrs_rep3/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/FO47_72hrs_rep1/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/FO47_72hrs_rep2/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/FO47_72hrs_rep3/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_0hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_16hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_24hrs_prelim_rep1/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_36hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_48hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_4hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_72hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_72hrs_rep1/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_72hrs_rep2/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_72hrs_rep3/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_8hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_96hrs_prelim/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_CzapekDox/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_GlucosePeptone/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_PDA/accepted_hits.bam \\\n\t\talignment/$Organism/$Strain/Fus2_PDB/accepted_hits.bam\n\t\tcufflinks -o $OutDir/cufflinks -p 8 --max-intron-length 4000 $AcceptedHits 2\u0026gt;\u0026amp;1 | tee $OutDir/cufflinks/cufflinks.log\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/Fo47/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFOP2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tAcceptedHits=alignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated/concatenated.bam\n\t\tOutDir=gene_pred/cufflinks/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated_prelim\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\tsamtools merge -f \u003cspan class=\"pl-smi\"\u003e$AcceptedHits\u003c/span\u003e \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/55_72hrs_rep1/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/55_72hrs_rep2/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/55_72hrs_rep3/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/FO47_72hrs_rep1/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/FO47_72hrs_rep2/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/FO47_72hrs_rep3/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_0hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_16hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_24hrs_prelim_rep1/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_36hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_48hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_4hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_rep1/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_rep2/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_rep3/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_8hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_96hrs_prelim/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_CzapekDox/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_GlucosePeptone/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_PDA/accepted_hits.bam \\\n\t\talignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_PDB/accepted_hits.bam\n\t\tcufflinks -o \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/cufflinks -p 8 --max-intron-length 4000 \u003cspan class=\"pl-smi\"\u003e$AcceptedHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u0026amp;1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tee \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/cufflinks/cufflinks.log\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOutput from stdout included:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\tProcessed 22484 loci.                        [*************************] 100%\n\tMap Properties:\n\t     Normalized Map Mass: 50507412.55\n\t     Raw Map Mass: 50507412.55\n\t     Fragment Length Distribution: Empirical (learned)\n\t                   Estimated Mean: 181.98\n\t                Estimated Std Dev: 78.39\n\t[13:02:48] Assembling transcripts and estimating abundances.\n\tProcessed 22506 loci.                        [*************************] 100%\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\tProcessed 22484 loci.                        [*************************] 100%\n\tMap Properties:\n\t     Normalized Map Mass: 50507412.55\n\t     Raw Map Mass: 50507412.55\n\t     Fragment Length Distribution: Empirical (learned)\n\t                   Estimated Mean: 181.98\n\t                Estimated Std Dev: 78.39\n\t[13:02:48] Assembling transcripts and estimating abundances.\n\tProcessed 22506 loci.                        [*************************] 100%\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe Estimated Mean: 181.98 allowed calculation of of the mean insert gap to be\n-20bp 182-(97*2) where 97 was the mean read length. This was provided to tophat\non a second run (as the -r option) along with the fragment length stdev to\nincrease the accuracy of mapping.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThen Rnaseq data was aligned to each genome assembly:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi' | grep -v 'old'); do\n# for Assembly in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do\n# for Assembly in $(ls assembly/merged_canu_spades/*/Fus2/filtered_contigs/Fus2_contigs_renamed.fasta); do\n# for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do\nStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\nOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nfor RNADir in $(ls -d qc_rna/paired/F.oxysporum_fsp_cepae/*); do\nTimepoint=$(echo $RNADir | rev | cut -f1 -d '/' | rev)\necho \u0026quot;$Timepoint\u0026quot;\nFileF=$(ls $RNADir/F/*_trim.fq.gz)\nFileR=$(ls $RNADir/R/*_trim.fq.gz)\nOutDir=alignment/$Organism/$Strain/$Timepoint\nInsertGap='-20'\nInsertStdDev='78'\nJobs=$(qstat | grep 'tophat' | grep 'qw' | wc -l)\nwhile [ $Jobs -gt 1 ]; do\nsleep 10\nprintf \u0026quot;.\u0026quot;\nJobs=$(qstat | grep 'tophat' | grep 'qw' | wc -l)\ndone\nprintf \u0026quot;\\n\u0026quot;\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq\nqsub $ProgDir/tophat_alignment.sh $Assembly $FileF $FileR $OutDir $InsertGap $InsertStdDev\ndone\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Assembly in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Assembly in $(ls assembly/merged_canu_spades/*/Fus2/filtered_contigs/Fus2_contigs_renamed.fasta); do\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRNADir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d qc_rna/paired/F.oxysporum_fsp_cepae/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nTimepoint=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$RNADir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Timepoint\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nFileF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$RNADir\u003c/span\u003e/F/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nFileR=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$RNADir\u003c/span\u003e/R/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_trim.fq.gz\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=alignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Timepoint\u003c/span\u003e\nInsertGap=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e-20\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nInsertStdDev=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e78\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etophat\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eqw\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e [ \u003cspan class=\"pl-smi\"\u003e$Jobs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-gt\u003c/span\u003e 1 ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nsleep 10\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etophat\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eqw\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/tophat_alignment.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FileF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FileR\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$InsertGap\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$InsertStdDev\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBraker prediction\u003c/h4\u003e\u003ca id=\"user-content-braker-prediction\" class=\"anchor\" aria-label=\"Permalink: Braker prediction\" href=\"#braker-prediction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBefore braker predictiction was performed, I double checked that I had the\ngenemark key in my user area and copied it over from the genemark install\ndirectory:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tls ~/.gm_key\n\tcp /home/armita/prog/genemark/gm_key_64 ~/.gm_key\"\u003e\u003cpre\u003e\tls \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/.gm_key\n\tcp /home/armita/prog/genemark/gm_key_64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/.gm_key\u003c/pre\u003e\u003c/div\u003e\n\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\t# for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep 'Fus2' | grep -e 'Fus2_canu_new' -e 'Fus2_merged' | grep 'cepae' | grep 'Fus2_merged'); do\n\t# for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep 'proliferatum'); do\nfor Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi' | grep -v 'old'); do\nJobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)\nwhile [ $Jobs -gt 1 ]; do\nsleep 10\nprintf \u0026quot;.\u0026quot;\nJobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)\ndone\nprintf \u0026quot;\\n\u0026quot;\nStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\nOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nmkdir -p alignment/$Organism/$Strain/concatenated\nsamtools merge -f alignment/$Organism/$Strain/concatenated/concatenated.bam \\\nalignment/$Organism/$Strain/55_72hrs_rep1/accepted_hits.bam \\\nalignment/$Organism/$Strain/55_72hrs_rep2/accepted_hits.bam \\\nalignment/$Organism/$Strain/55_72hrs_rep3/accepted_hits.bam \\\nalignment/$Organism/$Strain/FO47_72hrs_rep1/accepted_hits.bam \\\nalignment/$Organism/$Strain/FO47_72hrs_rep2/accepted_hits.bam \\\nalignment/$Organism/$Strain/FO47_72hrs_rep3/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_0hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_16hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_24hrs_prelim_rep1/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_36hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_48hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_4hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_72hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_72hrs_rep1/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_72hrs_rep2/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_72hrs_rep3/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_8hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_96hrs_prelim/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_CzapekDox/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_GlucosePeptone/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_PDA/accepted_hits.bam \\\nalignment/$Organism/$Strain/Fus2_PDB/accepted_hits.bam\nOutDir=gene_pred/braker/$Organism/\u0026quot;$Strain\u0026quot;_braker\nAcceptedHits=alignment/$Organism/$Strain/concatenated/concatenated.bam\nGeneModelName=\u0026quot;$Organism\u0026quot;_\u0026quot;$Strain\u0026quot;_braker_new\nrm -r /home/armita/prog/augustus-3.1/config/species/\u0026quot;$Organism\u0026quot;_\u0026quot;$Strain\u0026quot;_braker_new\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/braker1\nqsub $ProgDir/sub_braker_fungi.sh $Assembly $OutDir $AcceptedHits $GeneModelName\ndone \u0026gt; log.txt\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep 'Fus2' | grep -e 'Fus2_canu_new' -e 'Fus2_merged' | grep 'cepae' | grep 'Fus2_merged'); do\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep 'proliferatum'); do\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etophat\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003er\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e [ \u003cspan class=\"pl-smi\"\u003e$Jobs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-gt\u003c/span\u003e 1 ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nsleep 10\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etophat\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003er\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nmkdir -p alignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated\nsamtools merge -f alignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated/concatenated.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/55_72hrs_rep1/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/55_72hrs_rep2/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/55_72hrs_rep3/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/FO47_72hrs_rep1/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/FO47_72hrs_rep2/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/FO47_72hrs_rep3/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_0hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_16hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_24hrs_prelim_rep1/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_36hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_48hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_4hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_rep1/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_rep2/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_72hrs_rep3/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_8hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_96hrs_prelim/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_CzapekDox/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_GlucosePeptone/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_PDA/accepted_hits.bam \\\nalignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/Fus2_PDB/accepted_hits.bam\nOutDir=gene_pred/braker/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_braker\nAcceptedHits=alignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated/concatenated.bam\nGeneModelName=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_braker_new\nrm -r /home/armita/prog/augustus-3.1/config/species/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_braker_new\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/braker1\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_braker_fungi.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AcceptedHits\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$GeneModelName\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e log.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFasta and gff files were extracted from Braker1 output.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls gene_pred/braker/F.*/*_braker/*/augustus.gff | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'canu_new' -e 'ncbi' | grep -v 'canu_new'); do\ngetAnnoFasta.pl $File\nOutDir=$(dirname $File)\necho \u0026quot;##gff-version 3\u0026quot; \u0026gt; $OutDir/augustus_extracted.gff\ncat $File | grep -v '#' \u0026gt;\u0026gt; $OutDir/augustus_extracted.gff\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/braker/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_braker/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/augustus.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecanu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecanu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\ngetAnnoFasta.pl \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e##gff-version 3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/augustus_extracted.gff\ncat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e#\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/augustus_extracted.gff\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe relationship between gene models and aligned reads was investigated. To do\nthis aligned reads needed to be sorted and indexed:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNote - IGV was used to view aligned reads against the Fus2 genome on my local\nmachine.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tInBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated.bam\n\tViewBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_view.bam\n\tSortBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_sorted\n\tsamtools view -b $InBam \u0026gt; $ViewBam\n\tsamtools sort $ViewBam $SortBam\n\tsamtools index $SortBam.bam\"\u003e\u003cpre\u003e\tInBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated.bam\n\tViewBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_view.bam\n\tSortBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_sorted\n\tsamtools view -b \u003cspan class=\"pl-smi\"\u003e$InBam\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ViewBam\u003c/span\u003e\n\tsamtools sort \u003cspan class=\"pl-smi\"\u003e$ViewBam\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$SortBam\u003c/span\u003e\n\tsamtools index \u003cspan class=\"pl-smi\"\u003e$SortBam\u003c/span\u003e.bam\u003c/pre\u003e\u003c/div\u003e\n\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSupplimenting Braker gene models with CodingQuary genes\u003c/h2\u003e\u003ca id=\"user-content-supplimenting-braker-gene-models-with-codingquary-genes\" class=\"anchor\" aria-label=\"Permalink: Supplimenting Braker gene models with CodingQuary genes\" href=\"#supplimenting-braker-gene-models-with-codingquary-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAdditional genes were added to Braker gene predictions, using CodingQuary in\npathogen mode to predict additional regions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFistly, aligned RNAseq data was assembled into transcripts using Cufflinks.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNote - cufflinks doesn't always predict direction of a transcript and\ntherefore features can not be restricted by strand when they are intersected.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)\n# while [ $Jobs -gt 1 ]; do\n# sleep 10\n# printf \u0026quot;.\u0026quot;\n# Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)\n# done\n# sleep 2h\n# Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)\n# while [ $Jobs -gt 1 ]; do\n# sleep 10\n# printf \u0026quot;.\u0026quot;\n# Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)\n# done\nfor Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' | grep -v 'old'); do\nStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\nOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nOutDir=gene_pred/cufflinks/$Organism/$Strain/concatenated\nmkdir -p $OutDir\nAcceptedHits=$(ls alignment/$Organism/$Strain/concatenated/concatenated.bam)\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq\nqsub $ProgDir/sub_cufflinks.sh $AcceptedHits $OutDir\ndone \u0026gt; log2.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e while [ $Jobs -gt 1 ]; do\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sleep 10\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e printf \".\"\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e done\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sleep 2h\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e while [ $Jobs -gt 1 ]; do\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sleep 10\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e printf \".\"\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e done\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutDir=gene_pred/cufflinks/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nAcceptedHits=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els alignment/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated/concatenated.bam\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_cufflinks.sh \u003cspan class=\"pl-smi\"\u003e$AcceptedHits\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e log2.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSecondly, genes were predicted using CodingQuary:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# Jobs=$(qstat | grep 'armita' | wc -l)\n# while [ $Jobs -ge 1 ]; do\n# sleep 10m\n# printf \u0026quot;.\u0026quot;\n# Jobs=$(qstat | grep 'armita' | wc -l)\n# done\nfor Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -v 'old'); do\nJobs=$(qstat | grep 'sub_cuffli' | wc -l)\nwhile [ $Jobs -ge 1 ]; do\nsleep 10\nprintf \u0026quot;.\u0026quot;\nJobs=$(qstat | grep 'sub_cuffli' | wc -l)\ndone\nStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\nOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nOutDir=gene_pred/codingquary/$Organism/$Strain\nCufflinksGTF=$(ls gene_pred/cufflinks/$Organism/$Strain/concatenated/transcripts.gtf)\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\nqsub $ProgDir/sub_CodingQuary.sh $Assembly $CufflinksGTF $OutDir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'armita' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e while [ $Jobs -ge 1 ]; do\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e sleep 10m\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e printf \".\"\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Jobs=$(qstat | grep 'armita' | wc -l)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e done\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esub_cuffli\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e [ \u003cspan class=\"pl-smi\"\u003e$Jobs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-ge\u003c/span\u003e 1 ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nsleep 10\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003esub_cuffli\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutDir=gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nCufflinksGTF=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/cufflinks/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/concatenated/transcripts.gtf\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_CodingQuary.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$CufflinksGTF\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen, additional transcripts were added to Braker gene models, when CodingQuary\ngenes were predicted in regions of the genome, not containing Braker gene\nmodels:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\t# for BrakerGff in $(ls gene_pred/braker/F.*/*_braker_new/*/augustus.gff3 | grep -w -e 'Fus2'); do\nfor BrakerGff in $(ls gene_pred/braker/F.*/*/*/augustus.gff3 | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' | grep 'N139'); do\nStrain=$(echo $BrakerGff| rev | cut -d '/' -f3 | rev | sed 's/_braker_pacbio//g'| sed 's/_braker//g')\nOrganism=$(echo $BrakerGff | rev | cut -d '/' -f4 | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nAssembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)\nCodingQuaryGff=gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3\nPGNGff=gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3\nAddDir=gene_pred/codingquary/$Organism/$Strain/additional\nFinalDir=gene_pred/final_genes/$Organism/$Strain/final\nAddGenesList=$AddDir/additional_genes.txt\nAddGenesGff=$AddDir/additional_genes.gff\nFinalGff=$AddDir/combined_genes.gff\nmkdir -p $AddDir\nmkdir -p $FinalDir\n\nbedtools intersect -v -a $CodingQuaryGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' \u0026gt; $AddGenesList\nbedtools intersect -v -a $PGNGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' \u0026gt;\u0026gt; $AddGenesList\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation\n$ProgDir/gene_list_to_gff.pl $AddGenesList $CodingQuaryGff CodingQuarry_v2.0 ID CodingQuary \u0026gt; $AddGenesGff\n$ProgDir/gene_list_to_gff.pl $AddGenesList $PGNGff PGNCodingQuarry_v2.0 ID CodingQuary \u0026gt;\u0026gt; $AddGenesGff\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\n\n$ProgDir/add_CodingQuary_features.pl $AddGenesGff $Assembly \u0026gt; $FinalDir/final_genes_CodingQuary.gff3\n$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_CodingQuary.gff3 $FinalDir/final_genes_CodingQuary\ncp $BrakerGff $FinalDir/final_genes_Braker.gff3\n$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_Braker.gff3 $FinalDir/final_genes_Braker\ncat $FinalDir/final_genes_Braker.pep.fasta $FinalDir/final_genes_CodingQuary.pep.fasta | sed -r 's/\\*/X/g' \u0026gt; $FinalDir/final_genes_combined.pep.fasta\ncat $FinalDir/final_genes_Braker.cdna.fasta $FinalDir/final_genes_CodingQuary.cdna.fasta \u0026gt; $FinalDir/final_genes_combined.cdna.fasta\ncat $FinalDir/final_genes_Braker.gene.fasta $FinalDir/final_genes_CodingQuary.gene.fasta \u0026gt; $FinalDir/final_genes_combined.gene.fasta\ncat $FinalDir/final_genes_Braker.upstream3000.fasta $FinalDir/final_genes_CodingQuary.upstream3000.fasta \u0026gt; $FinalDir/final_genes_combined.upstream3000.fasta\n\nGffBraker=$FinalDir/final_genes_CodingQuary.gff3\nGffQuary=$FinalDir/final_genes_Braker.gff3\nGffAppended=$FinalDir/final_genes_appended.gff3\ncat $GffBraker $GffQuary \u0026gt; $GffAppended\n\ndone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for BrakerGff in $(ls gene_pred/braker/F.*/*_braker_new/*/augustus.gff3 | grep -w -e 'Fus2'); do\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBrakerGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/braker/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/augustus.gff3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_braker_pacbio//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_braker//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nAssembly=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nCodingQuaryGff=gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PredictedPass.gff3\nPGNGff=gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PGN_predictedPass.gff3\nAddDir=gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/additional\nFinalDir=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/final\nAddGenesList=\u003cspan class=\"pl-smi\"\u003e$AddDir\u003c/span\u003e/additional_genes.txt\nAddGenesGff=\u003cspan class=\"pl-smi\"\u003e$AddDir\u003c/span\u003e/additional_genes.gff\nFinalGff=\u003cspan class=\"pl-smi\"\u003e$AddDir\u003c/span\u003e/combined_genes.gff\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$AddDir\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e\n\nbedtools intersect -v -a \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e\nbedtools intersect -v -a \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gene_list_to_gff.pl \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e CodingQuarry_v2.0 ID CodingQuary \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesGff\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gene_list_to_gff.pl \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e PGNCodingQuarry_v2.0 ID CodingQuary \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesGff\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\n\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/add_CodingQuary_features.pl \u003cspan class=\"pl-smi\"\u003e$AddGenesGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gff3\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gff2fasta.pl \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gff3 \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary\ncp \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gff3\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gff2fasta.pl \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gff3 \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.pep.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed -r \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/\\*/X/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.pep.fasta\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.cdna.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.cdna.fasta \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.cdna.fasta\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gene.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gene.fasta \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.gene.fasta\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.upstream3000.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.upstream3000.fasta \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.upstream3000.fasta\n\nGffBraker=\u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gff3\nGffQuary=\u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gff3\nGffAppended=\u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_appended.gff3\ncat \u003cspan class=\"pl-smi\"\u003e$GffBraker\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$GffQuary\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$GffAppended\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe final number of genes per isolate was observed using:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for DirPath in $(ls -d gene_pred/final_genes/F.*/*/final | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'  | grep -e 'ncbi'); do\necho $DirPath;\ncat $DirPath/final_genes_Braker.pep.fasta | grep '\u0026gt;' | wc -l;\ncat $DirPath/final_genes_CodingQuary.pep.fasta | grep '\u0026gt;' | wc -l;\ncat $DirPath/final_genes_combined.pep.fasta | grep '\u0026gt;' | wc -l;\necho \u0026quot;\u0026quot;;\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eDirPath\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$DirPath\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$DirPath\u003c/span\u003e/final_genes_Braker.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$DirPath\u003c/span\u003e/final_genes_CodingQuary.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$DirPath\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gene_pred/final_genes/F.oxysporum_fsp_cepae/125_ncbi/final\n17197\n1546\n18743\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/A13_ncbi/final\n17986\n948\n18934\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/A23_ncbi/final\n17006\n1551\n18557\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/A28_ncbi/final\n17426\n1448\n18874\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/CB3_ncbi/final\n16833\n1345\n18178\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/PG_ncbi/final\n16811\n1290\n18101\n\ngene_pred/final_genes/F.oxysporum_fsp_narcissi/N139_ncbi/final\n19059\n1642\n20701\n\ngene_pred/final_genes/F.proliferatum/A8_ncbi/final\n15421\n37\n15458\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egene_pred/final_genes/F.oxysporum_fsp_cepae/125_ncbi/final\n17197\n1546\n18743\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/A13_ncbi/final\n17986\n948\n18934\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/A23_ncbi/final\n17006\n1551\n18557\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/A28_ncbi/final\n17426\n1448\n18874\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/CB3_ncbi/final\n16833\n1345\n18178\n\ngene_pred/final_genes/F.oxysporum_fsp_cepae/PG_ncbi/final\n16811\n1290\n18101\n\ngene_pred/final_genes/F.oxysporum_fsp_narcissi/N139_ncbi/final\n19059\n1642\n20701\n\ngene_pred/final_genes/F.proliferatum/A8_ncbi/final\n15421\n37\n15458\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Gff in $(ls gene_pred/final_genes/*/*/final/final_genes_appended.gff3 | grep -e 'canu_new' -e 'ncbi'); do\n\tStrain=$(echo $Gff | rev | cut -d '/' -f3 | rev)\n\tOrganism=$(echo $Gff | rev | cut -d '/' -f4 | rev)\n\techo \u0026quot;$Strain - $Organism\u0026quot;\n\tcat $Gff | grep -w 'gene' | wc -l\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final/final_genes_appended.gff3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecanu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"125_ncbi - F.oxysporum_fsp_cepae\n18505\nA13_ncbi - F.oxysporum_fsp_cepae\n18790\nA23_ncbi - F.oxysporum_fsp_cepae\n18323\nA28_ncbi - F.oxysporum_fsp_cepae\n18629\nCB3_ncbi - F.oxysporum_fsp_cepae\n17943\nFus2_canu_new - F.oxysporum_fsp_cepae\n18855\nPG_ncbi - F.oxysporum_fsp_cepae\n17830\nN139_ncbi - F.oxysporum_fsp_narcissi\n20493\nA8_ncbi - F.proliferatum\n15418\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e125_ncbi - F.oxysporum_fsp_cepae\n18505\nA13_ncbi - F.oxysporum_fsp_cepae\n18790\nA23_ncbi - F.oxysporum_fsp_cepae\n18323\nA28_ncbi - F.oxysporum_fsp_cepae\n18629\nCB3_ncbi - F.oxysporum_fsp_cepae\n17943\nFus2_canu_new - F.oxysporum_fsp_cepae\n18855\nPG_ncbi - F.oxysporum_fsp_cepae\n17830\nN139_ncbi - F.oxysporum_fsp_narcissi\n20493\nA8_ncbi - F.proliferatum\n15418\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentification of duplicated genes in additional CodingQuary gene models\u003c/h2\u003e\u003ca id=\"user-content-identification-of-duplicated-genes-in-additional-codingquary-gene-models\" class=\"anchor\" aria-label=\"Permalink: Identification of duplicated genes in additional CodingQuary gene models\" href=\"#identification-of-duplicated-genes-in-additional-codingquary-gene-models\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for AddGenes in $(ls gene_pred/codingquary/F.*/*/additional/additional_genes.gff | grep 'ncbi' | grep -v 'ncbi_braker' | grep 'A23'); do\nStrain=$(echo $AddGenes| rev | cut -d '/' -f3 | rev)\nOrganism=$(echo $AddGenes | rev | cut -d '/' -f4 | rev)\nOutDir=$(dirname $AddGenes)\necho \u0026quot;$Organism - $Strain\u0026quot; \u0026gt; $OutDir/duplicated_genes.txt\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\n$ProgDir/remove_dup_features.py --inp_gff $AddGenes \u0026gt;\u0026gt; $OutDir/duplicated_genes.txt\ncat $OutDir/duplicated_genes.txt\necho \u0026quot;\u0026quot;\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAddGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/additional/additional_genes.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi_braker\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$AddGenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$AddGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$AddGenes\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/duplicated_genes.txt\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/remove_dup_features.py --inp_gff \u003cspan class=\"pl-smi\"\u003e$AddGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/duplicated_genes.txt\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eManual editing of gene models\u003c/h2\u003e\u003ca id=\"user-content-manual-editing-of-gene-models\" class=\"anchor\" aria-label=\"Permalink: Manual editing of gene models\" href=\"#manual-editing-of-gene-models\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFollowing annotation of gene models and orthology analysis documented below,\nsome gene models were noted to require manual editing. This was performed and\ngene models were re-annotated. In some cases this required removal of genes\nannotated by Braker so that the gene models predicted by coding queary could be\nfavoured. In other cases, gene models were manually edited based upon results\nfrom other Fusarium spp. genomes:\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFus2:\u003c/h3\u003e\u003ca id=\"user-content-fus2\" class=\"anchor\" aria-label=\"Permalink: Fus2:\" href=\"#fus2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eGene models requiring editing\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eFTF1 g16934 - delete gene, replace with CodingQuary model.\u003c/li\u003e\n\u003cli\u003eSIX3 g16847, g16190 - A23 one copy\u003c/li\u003e\n\u003cli\u003eSIX5 g16849 - A23 missing intron\u003c/li\u003e\n\u003cli\u003eSIX7 CUFF_4344_1_33 - Gene model good\u003c/li\u003e\n\u003cli\u003eSIX9 g13574, g16273 - premature stop in A23 (end of contig) \u0026amp; one copy in A23 \u0026amp; 125\u003c/li\u003e\n\u003cli\u003eSIX10 g17153 - Gene model good\u003c/li\u003e\n\u003cli\u003eSIX12 PGN_CUFF_4345_1_2 - checked start of Fus2, FoC sequence is too long but FoL reference matches this length - Bring A23 and 125 in line with Fus2 \u0026amp; FoL\u003c/li\u003e\n\u003cli\u003eNo gene model\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tOrganism=F.oxysporum_fsp_cepae\n\tStrain=Fus2_canu_new\n\tOutDir=gene_pred/final_genes/$Organism/$Strain/edited\n\tmkdir -p $OutDir\n\n\tAssembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)\n\tSourceDir=gene_pred/codingquary/$Organism/$Strain\n\tBrakerGff=$(ls $SourceDir/final/final_genes_Braker.gff3)\n\tCodingQuaryGff=$(ls $SourceDir/out/PredictedPass.gff3)\n\tPGNGff=$(ls $SourceDir/out/PGN_predictedPass.gff3)\n\n\tcat $BrakerGff \\\n\t| grep -v -w 'g16934' \\\n\t| grep -v -w 'g13345' \\\n\t\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\tprintf \\\n\t\u0026quot;contig_19_pilon\\tManual_annotation\\tgene\\t445791\\t445964\\t0\\t+\\t.\\tID=man_1;\n\tcontig_19_pilon\\tManual_annotation\\tmRNA\\t445791\\t445964\\t.\\t+\\t.\\tID=man_1.t1;Parent=man_1\n\tcontig_19_pilon\\tManual_annotation\\tstart_codon\\t445791\\t445793\\t.\\t+\\t.\\tParent=man_1.t1;\n\tcontig_19_pilon\\tManual_annotation\\tCDS\\t445791\\t445964\\t.\\t+\\t0\\tID=man_1.t1.CDS1;Parent=man_1.t1\n\tcontig_19_pilon\\tManual_annotation\\texon\\t445791\\t445964\\t.\\t+\\t.\\tID=man_1.t1.exon1;Parent=man_1.t1;\n\tcontig_19_pilon\\tManual_annotation\\tstop_codon\\t445962\\t445964\\t.\\t+\\t.\\tParent=man_1.t1;\\n\u0026quot; \\\n\t\u0026gt;\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\n\tcat $CodingQuaryGff \\\n\t| grep -v -w -e 'NS.00001' -e 'NS.00002' -e 'NS.00003' -e 'NS.04463' -e 'CUFF.11065.1.82' \\\n\t\u0026gt; $OutDir/PredictedPass_ed.gff3\n\tcp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\n\n\ttouch $OutDir/manual_annotations.gff3\n\t#\n\t# printf \\\n\t# \u0026quot;contig_10_pilon\\tManual_annotation\\tgene\\t1569\\t5055\\t0\\t+\\t.\\tID=g13345_ps;\n\t# contig_10_pilon\\tManual_annotation\\tmRNA\\t1569\\t5055\\t.\\t+\\t.\\tID=g13345_ps.t1;Parent=g13345_ps\n\t# contig_10_pilon\\tManual_annotation\\tstart_codon\\t1569\\t1571\\t.\\t+\\t.\\tParent=g13345_ps.t1;\n\t# contig_10_pilon\\tManual_annotation\\tCDS\\t1569\\t2856\\t.\\t+\\t0\\tID=g13345_ps.t1.CDS1;Parent=g13345_ps.t1\n\t# contig_10_pilon\\tManual_annotation\\texon\\t1569\\t2856\\t.\\t+\\t.\\tParent=g13345_ps.t1\n\t# contig_10_pilon\\tManual_annotation\\tintron\\t2857\\t2905\\t.\\t+\\t.\\tParent=g13345_ps.t1\n\t# contig_10_pilon\\tManual_annotation\\tCDS\\t2906\\t5055\\t.\\t+\\t2\\tID=g13345_ps.t1.CDS2;Parent=g13345_ps.t1\n\t# contig_10_pilon\\tManual_annotation\\texon\\t2906\\t5055\\t.\\t+\\t.\\tID=g13345_ps.t1.exon1;Parent=g13345_ps.t1;\n\t# contig_10_pilon\\tManual_annotation\\tstop_codon\\t5053\\t5055\\t.\\t+\\t.\\tParent=g13345_ps.t1;\\n\u0026quot; \\\n\t# \u0026gt; $OutDir/pseudogenes.gff3\"\u003e\u003cpre\u003e\tOrganism=F.oxysporum_fsp_cepae\n\tStrain=Fus2_canu_new\n\tOutDir=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/edited\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\n\tAssembly=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tSourceDir=gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\n\tBrakerGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$SourceDir\u003c/span\u003e/final/final_genes_Braker.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tCodingQuaryGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$SourceDir\u003c/span\u003e/out/PredictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tPGNGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$SourceDir\u003c/span\u003e/out/PGN_predictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\n\tcat \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16934\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg13345\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_19_pilon\\tManual_annotation\\tgene\\t445791\\t445964\\t0\\t+\\t.\\tID=man_1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_19_pilon\\tManual_annotation\\tmRNA\\t445791\\t445964\\t.\\t+\\t.\\tID=man_1.t1;Parent=man_1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_19_pilon\\tManual_annotation\\tstart_codon\\t445791\\t445793\\t.\\t+\\t.\\tParent=man_1.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_19_pilon\\tManual_annotation\\tCDS\\t445791\\t445964\\t.\\t+\\t0\\tID=man_1.t1.CDS1;Parent=man_1.t1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_19_pilon\\tManual_annotation\\texon\\t445791\\t445964\\t.\\t+\\t.\\tID=man_1.t1.exon1;Parent=man_1.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_19_pilon\\tManual_annotation\\tstop_codon\\t445962\\t445964\\t.\\t+\\t.\\tParent=man_1.t1;\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\n\tcat \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNS.00001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNS.00002\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNS.00003\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eNS.04463\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.11065.1.82\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\n\tcp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\n\n\ttouch \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/manual_annotations.gff3\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e printf \\\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e \"contig_10_pilon\\tManual_annotation\\tgene\\t1569\\t5055\\t0\\t+\\t.\\tID=g13345_ps;\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\tmRNA\\t1569\\t5055\\t.\\t+\\t.\\tID=g13345_ps.t1;Parent=g13345_ps\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\tstart_codon\\t1569\\t1571\\t.\\t+\\t.\\tParent=g13345_ps.t1;\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\tCDS\\t1569\\t2856\\t.\\t+\\t0\\tID=g13345_ps.t1.CDS1;Parent=g13345_ps.t1\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\texon\\t1569\\t2856\\t.\\t+\\t.\\tParent=g13345_ps.t1\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\tintron\\t2857\\t2905\\t.\\t+\\t.\\tParent=g13345_ps.t1\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\tCDS\\t2906\\t5055\\t.\\t+\\t2\\tID=g13345_ps.t1.CDS2;Parent=g13345_ps.t1\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\texon\\t2906\\t5055\\t.\\t+\\t.\\tID=g13345_ps.t1.exon1;Parent=g13345_ps.t1;\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e contig_10_pilon\\tManual_annotation\\tstop_codon\\t5053\\t5055\\t.\\t+\\t.\\tParent=g13345_ps.t1;\\n\" \\\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e \u0026gt; $OutDir/pseudogenes.gff3\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA23\u003c/h2\u003e\u003ca id=\"user-content-a23\" class=\"anchor\" aria-label=\"Permalink: A23\" href=\"#a23\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that at this stage all codingquary genes contain . characters rather than _ characters\ng16038 - SIX12 - use alternative transcript\nno gene - SIX14 - manual annotation\ng16100 - SIX5 - shorten intron and final CDS\ng15699 - SIX7 - remove intron and final CDS.\nno gene - SIX9  - manual annotation\nDuplicated genes:\nCUFF.7204.2.14\nCUFF.7961.1.0\nCUFF.11924.2.127\nCUFF.6792.1.4\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Organism=F.oxysporum_fsp_cepae\nStrain=A23_ncbi\nOutDir=gene_pred/final_genes/$Organism/$Strain/edited\nmkdir -p $OutDir\n\nAssembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)\nBrakerGff=$(ls gene_pred/braker/$Organism/\u0026quot;$Strain\u0026quot;_braker/*/augustus.gff3)\nCodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)\nPGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)\n\ncat $BrakerGff \\\n| grep -w -v 'g7779' \\\n| grep -w -v 'g13351' \\\n| grep -v 'g16038' \\\n| sed \\\n's/contig_819\\tAUGUSTUS\\tgene\\t655\\t1273\\t0.73\\t+\\t.\\tID=g16100;/contig_819\\tAUGUSTUS\\tgene\\t655\\t1188\\t0.73\\t+\\t.\\tID=g16100;/g' \\\n| sed \\\n's/contig_819\\tAUGUSTUS\\tmRNA\\t655\\t1273\\t0.73\\t+\\t.\\tID=g16100.t1;Parent=g16100/contig_819\\tAUGUSTUS\\tmRNA\\t655\\t1188\\t0.73\\t+\\t.\\tID=g16100.t1;Parent=g16100/g' \\\n| sed \\\n's/contig_819\\tAUGUSTUS\\tintron\\t1002\\t1057\\t1\\t+\\t.\\tParent=g16100.t1;/contig_819\\tAUGUSTUS\\tintron\\t1002\\t1050\\t1\\t+\\t.\\tParent=g16100.t1;/g' \\\n| sed \\\n's/contig_819\\tAUGUSTUS\\tCDS\\t1058\\t1273\\t0.74\\t+\\t0\\tID=g16100.t1.CDS4;Parent=g16100.t1/contig_819\\tAUGUSTUS\\tCDS\\t1051\\t1188\\t0.74\\t+\\t0\\tID=g16100.t1.CDS4;Parent=g16100.t1/g' \\\n| sed \\\n's/contig_819\\tAUGUSTUS\\texon\\t1058\\t1273\\t.\\t+\\t.\\tID=g16100.t1.exon4;Parent=g16100.t1;/contig_819\\tAUGUSTUS\\texon\\t1051\\t1188\\t.\\t+\\t.\\tID=g16100.t1.exon4;Parent=g16100.t1;/g' \\\n| sed \\\n's/contig_819\\tAUGUSTUS\\tstop_codon\\t1271\\t1273\\t.\\t+\\t0\\tParent=g16100.t1;/contig_819\\tAUGUSTUS\\tstop_codon\\t1186\\t1188\\t.\\t+\\t0\\tParent=g16100.t1;/g' \\\n| sed \\\n's/contig_592\\tAUGUSTUS\\tgene\\t1046\\t1683\\t0.68\\t+\\t.\\tID=g15699;/contig_592\\tAUGUSTUS\\tgene\\t1046\\t1543\\t0.68\\t+\\t.\\tID=g15699;/g' \\\n| sed \\\n's/contig_592\\tAUGUSTUS\\tmRNA\\t1046\\t1683\\t0.68\\t+\\t.\\tID=g15699.t1;Parent=g15699/contig_592\\tAUGUSTUS\\tmRNA\\t1046\\t1543\\t0.68\\t+\\t.\\tID=g15699.t1;Parent=g15699/g' \\\n| sed \\\n's/contig_592\\tAUGUSTUS\\tCDS\\t1046\\t1539\\t0.68\\t+\\t0\\tID=g15699.t1.CDS1;Parent=g15699.t1/contig_592\\tAUGUSTUS\\tCDS\\t1046\\t1543\\t0.68\\t+\\t0\\tID=g15699.t1.CDS1;Parent=g15699.t1/g' \\\n| sed \\\n's/contig_592\\tAUGUSTUS\\texon\\t1046\\t1539\\t.\\t+\\t.\\tID=g15699.t1.exon1;Parent=g15699.t1;/contig_592\\tAUGUSTUS\\texon\\t1046\\t1543\\t.\\t+\\t.\\tID=g15699.t1.exon1;Parent=g15699.t1;/g' \\\n| grep -v 'contig_592.AUGUSTUS.intron.1540.1676.1.+...Parent=g15699.t1;' \\\n| grep -v 'g15699.t1.CDS2' | grep -v 'g15699.t1.exon2' \\\n| sed \\\n's/contig_592\\tAUGUSTUS\\tstop_codon\\t1681\\t1683\\t.\\t+\\t0\\tParent=g15699.t1;/contig_592\\tAUGUSTUS\\tstop_codon\\t1541\\t1543\\t.\\t+\\t0\\tParent=g15699.t1;/g' \\\n\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\n\nprintf \\\n\u0026quot;contig_1792\\tManual_annotation\\tgene\\t1\\t219\\t1\\t+\\t.\\tID=man_1;\ncontig_1792\\tManual_annotation\\tmRNA\\t1\\t219\\t1\\t+\\t.\\tID=man_1.t1;Parent=man_1\ncontig_1792\\tManual_annotation\\tCDS\\t1\\t219\\t1.0\\t+\\t0\\tID=man_1.t1.CDS1;Parent=man_1.t1\ncontig_1792\\tManual_annotation\\texon\\t1\\t219\\t.\\t+\\t.\\tID=man_1.t1.exon1;Parent=man_1.t1;\ncontig_1792\\tManual_annotation\\tstop_codon\\t217\\t219\\t.\\t+\\t.\\tParent=man_1.t1;\\n\u0026quot; \\\n\u0026gt;\u0026gt; $OutDir/final_genes_Braker_ed.gff3\nprintf \\\n\u0026quot;contig_1016\\tManual_annotation\\tgene\\t2103\\t2276\\t1\\t+\\t.\\tID=man_2;\ncontig_1016\\tManual_annotation\\tmRNA\\t2103\\t2276\\t1\\t+\\t.\\tID=man_2.t1;Parent=man_2\ncontig_1016\\tManual_annotation\\tstart_codon\\t2103\\t2105\\t.\\t+\\t.\\tParent=man_2.t1;\ncontig_1016\\tManual_annotation\\tCDS\\t2103\\t2276\\t1.0\\t+\\t0\\tID=man_2.t1.CDS1;Parent=man_2.t1\ncontig_1016\\tManual_annotation\\texon\\t2103\\t2276\\t.\\t+\\t.\\tID=man_2.t1.exon1;Parent=man_2.t1;\ncontig_1016\\tManual_annotation\\tstop_codon\\t2274\\t2276\\t.\\t+\\t.\\tParent=man_2.t1;\\n\u0026quot; \\\n\u0026gt;\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\ncat $CodingQuaryGff | grep -w -v -e 'CUFF.7204.2.14' -e 'CUFF.7961.1.0' -e 'CUFF.11924.2.127' -e 'CUFF.6792.1.4' \u0026gt; $OutDir/PredictedPass_ed.gff3\ncp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\"\u003e\u003cpre\u003eOrganism=F.oxysporum_fsp_cepae\nStrain=A23_ncbi\nOutDir=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/edited\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\nAssembly=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nBrakerGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/braker/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_braker/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/augustus.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nCodingQuaryGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PredictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nPGNGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PGN_predictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\ncat \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg7779\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg13351\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16038\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_819\\tAUGUSTUS\\tgene\\t655\\t1273\\t0.73\\t+\\t.\\tID=g16100;/contig_819\\tAUGUSTUS\\tgene\\t655\\t1188\\t0.73\\t+\\t.\\tID=g16100;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_819\\tAUGUSTUS\\tmRNA\\t655\\t1273\\t0.73\\t+\\t.\\tID=g16100.t1;Parent=g16100/contig_819\\tAUGUSTUS\\tmRNA\\t655\\t1188\\t0.73\\t+\\t.\\tID=g16100.t1;Parent=g16100/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_819\\tAUGUSTUS\\tintron\\t1002\\t1057\\t1\\t+\\t.\\tParent=g16100.t1;/contig_819\\tAUGUSTUS\\tintron\\t1002\\t1050\\t1\\t+\\t.\\tParent=g16100.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_819\\tAUGUSTUS\\tCDS\\t1058\\t1273\\t0.74\\t+\\t0\\tID=g16100.t1.CDS4;Parent=g16100.t1/contig_819\\tAUGUSTUS\\tCDS\\t1051\\t1188\\t0.74\\t+\\t0\\tID=g16100.t1.CDS4;Parent=g16100.t1/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_819\\tAUGUSTUS\\texon\\t1058\\t1273\\t.\\t+\\t.\\tID=g16100.t1.exon4;Parent=g16100.t1;/contig_819\\tAUGUSTUS\\texon\\t1051\\t1188\\t.\\t+\\t.\\tID=g16100.t1.exon4;Parent=g16100.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_819\\tAUGUSTUS\\tstop_codon\\t1271\\t1273\\t.\\t+\\t0\\tParent=g16100.t1;/contig_819\\tAUGUSTUS\\tstop_codon\\t1186\\t1188\\t.\\t+\\t0\\tParent=g16100.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_592\\tAUGUSTUS\\tgene\\t1046\\t1683\\t0.68\\t+\\t.\\tID=g15699;/contig_592\\tAUGUSTUS\\tgene\\t1046\\t1543\\t0.68\\t+\\t.\\tID=g15699;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_592\\tAUGUSTUS\\tmRNA\\t1046\\t1683\\t0.68\\t+\\t.\\tID=g15699.t1;Parent=g15699/contig_592\\tAUGUSTUS\\tmRNA\\t1046\\t1543\\t0.68\\t+\\t.\\tID=g15699.t1;Parent=g15699/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_592\\tAUGUSTUS\\tCDS\\t1046\\t1539\\t0.68\\t+\\t0\\tID=g15699.t1.CDS1;Parent=g15699.t1/contig_592\\tAUGUSTUS\\tCDS\\t1046\\t1543\\t0.68\\t+\\t0\\tID=g15699.t1.CDS1;Parent=g15699.t1/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_592\\tAUGUSTUS\\texon\\t1046\\t1539\\t.\\t+\\t.\\tID=g15699.t1.exon1;Parent=g15699.t1;/contig_592\\tAUGUSTUS\\texon\\t1046\\t1543\\t.\\t+\\t.\\tID=g15699.t1.exon1;Parent=g15699.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_592.AUGUSTUS.intron.1540.1676.1.+...Parent=g15699.t1;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg15699.t1.CDS2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg15699.t1.exon2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_592\\tAUGUSTUS\\tstop_codon\\t1681\\t1683\\t.\\t+\\t0\\tParent=g15699.t1;/contig_592\\tAUGUSTUS\\tstop_codon\\t1541\\t1543\\t.\\t+\\t0\\tParent=g15699.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\n\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_1792\\tManual_annotation\\tgene\\t1\\t219\\t1\\t+\\t.\\tID=man_1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1792\\tManual_annotation\\tmRNA\\t1\\t219\\t1\\t+\\t.\\tID=man_1.t1;Parent=man_1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1792\\tManual_annotation\\tCDS\\t1\\t219\\t1.0\\t+\\t0\\tID=man_1.t1.CDS1;Parent=man_1.t1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1792\\tManual_annotation\\texon\\t1\\t219\\t.\\t+\\t.\\tID=man_1.t1.exon1;Parent=man_1.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1792\\tManual_annotation\\tstop_codon\\t217\\t219\\t.\\t+\\t.\\tParent=man_1.t1;\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \\\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_1016\\tManual_annotation\\tgene\\t2103\\t2276\\t1\\t+\\t.\\tID=man_2;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1016\\tManual_annotation\\tmRNA\\t2103\\t2276\\t1\\t+\\t.\\tID=man_2.t1;Parent=man_2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1016\\tManual_annotation\\tstart_codon\\t2103\\t2105\\t.\\t+\\t.\\tParent=man_2.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1016\\tManual_annotation\\tCDS\\t2103\\t2276\\t1.0\\t+\\t0\\tID=man_2.t1.CDS1;Parent=man_2.t1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1016\\tManual_annotation\\texon\\t2103\\t2276\\t.\\t+\\t.\\tID=man_2.t1.exon1;Parent=man_2.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003econtig_1016\\tManual_annotation\\tstop_codon\\t2274\\t2276\\t.\\t+\\t.\\tParent=man_2.t1;\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\ncat \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.7204.2.14\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.7961.1.0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.11924.2.127\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.6792.1.4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\ncp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFoC 125\u003c/h2\u003e\u003ca id=\"user-content-foc-125\" class=\"anchor\" aria-label=\"Permalink: FoC 125\" href=\"#foc-125\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that at this stage all codingquary genes contain . characters rather than _ characters\ng16001 - SIX12 - remove gene and use PGN_predictedPass gene\ng16002 - SIX7 - remove intron and make stop codon earlier\nCUFF_14695 - SIX14 - Replace with manual annotation in correct orientation\nDuplicated genes:\nCUFF.14630.2.11\nCUFF.7213.2.7\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tOrganism=F.oxysporum_fsp_cepae\n\tStrain=125_ncbi\n\tOutDir=gene_pred/final_genes/$Organism/$Strain/edited\n\tmkdir -p $OutDir\n\tAssembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)\n\tBrakerGff=$(ls gene_pred/braker/$Organism/\u0026quot;$Strain\u0026quot;_braker/*/augustus.gff3)\n\tCodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)\n\tPGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)\n\tcat $BrakerGff \\\n\t| grep -v -w 'g16001' \\\n\t| sed \\\n\t's/contig_714\\tAUGUSTUS\\tgene\\t6151\\t6788\\t0.93\\t+\\t.\\tID=g16002/contig_714\\tAUGUSTUS\\tgene\\t6151\\t6648\\t0.93\\t+\\t.\\tID=g16002/g' \\\n\t| sed \\\n\t's/contig_714\\tAUGUSTUS\\tmRNA\\t6151\\t6788\\t0.93\\t+\\t.\\tID=g16002.t1;Parent=g16002/contig_714\\tAUGUSTUS\\tmRNA\\t6151\\t6648\\t0.93\\t+\\t.\\tID=g16002.t1;Parent=g16002/g' \\\n\t| sed \\\n\t's/contig_714\\tAUGUSTUS\\tCDS\\t6151\\t6644\\t0.93\\t+\\t0\\tID=g16002.t1.CDS1;Parent=g16002.t1/contig_714\\tAUGUSTUS\\tCDS\\t6151\\t6648\\t0.93\\t+\\t0\\tID=g16002.t1.CDS1;Parent=g16002.t1/g' \\\n\t| sed \\\n\t's/contig_714\\tAUGUSTUS\\texon\\t6151\\t6644\\t.\\t+\\t.\\tID=g16002.t1.exon1;Parent=g16002.t1;/contig_714\\tAUGUSTUS\\texon\\t6151\\t6648\\t.\\t+\\t.\\tID=g16002.t1.exon1;Parent=g16002.t1;/g' \\\n\t| grep -v -e 'g16002.t1.CDS2' -e 'g16002.t1.exon2' \\\n\t| grep -v 'contig_714.AUGUSTUS.intron.6645.6781.1.+...Parent=g16002.t1;' \\\n\t| sed \\\n\t's/contig_714\\tAUGUSTUS\\tstop_codon\\t6786\\t6788\\t.\\t+\\t0\\tParent=g16002.t1;/contig_714\\tAUGUSTUS\\tstop_codon\\t6646\\t6648\\t.\\t+\\t0\\tParent=g16002.t1;/g' \\\n\t\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\n\tprintf \\\n\t\u0026quot;contig_2082\\tManual_annotation\\tgene\\t261\\t525\\t1\\t+\\t.\\tID=man_1;\n\tcontig_2082\\tManual_annotation\\tmRNA\\t261\\t525\\t1\\t+\\t.\\tID=man_1.t1;Parent=man_1\n\tcontig_2082\\tManual_annotation\\tstart_codon\\t261\\t263\\t.\\t+\\t.\\tParent=man_1.t1;\n\tcontig_2082\\tManual_annotation\\tCDS\\t261\\t525\\t1.0\\t+\\t0\\tID=man_1.t1.CDS1;Parent=man_1.t1\n\tcontig_2082\\tManual_annotation\\texon\\t261\\t525\\t.\\t+\\t.\\tID=man_1.t1.exon1;Parent=man_1.t1;\\n\u0026quot; \\\n\t\u0026gt;\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\tprintf \\\n\t\u0026quot;contig_971\\tManual_annotation\\tgene\\t333\\t506\\t1\\t-\\t.\\tID=man_2;\n\tcontig_971\\tManual_annotation\\tmRNA\\t333\\t506\\t1\\t-\\t.\\tID=man_2.t1;Parent=man_2\n\tcontig_971\\tManual_annotation\\tstop_codon\\t333\\t335\\t.\\t-\\t.\\tParent=man_2.t1;\n\tcontig_971\\tManual_annotation\\tCDS\\t333\\t506\\t1.0\\t-\\t0\\tID=man_2.t1.CDS1;Parent=man_2.t1\n\tcontig_971\\tManual_annotation\\texon\\t333\\t506\\t.\\t-\\t.\\tID=man_2.t1.exon1;Parent=man_2.t1;\n\tcontig_971\\tManual_annotation\\tstart_codon\\t504\\t506\\t.\\t-\\t.\\tParent=man_2.t1;\\n\u0026quot; \\\n\t\u0026gt;\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\n\tcat $CodingQuaryGff | grep -v -w -e 'CUFF.14630.2.11' -e\t'CUFF.7213.2.7' \u0026gt; $OutDir/PredictedPass_ed.gff3\n\tcp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\"\u003e\u003cpre\u003e\tOrganism=F.oxysporum_fsp_cepae\n\tStrain=125_ncbi\n\tOutDir=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/edited\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\tAssembly=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tBrakerGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/braker/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_braker/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/augustus.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tCodingQuaryGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PredictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tPGNGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PGN_predictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16001\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_714\\tAUGUSTUS\\tgene\\t6151\\t6788\\t0.93\\t+\\t.\\tID=g16002/contig_714\\tAUGUSTUS\\tgene\\t6151\\t6648\\t0.93\\t+\\t.\\tID=g16002/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_714\\tAUGUSTUS\\tmRNA\\t6151\\t6788\\t0.93\\t+\\t.\\tID=g16002.t1;Parent=g16002/contig_714\\tAUGUSTUS\\tmRNA\\t6151\\t6648\\t0.93\\t+\\t.\\tID=g16002.t1;Parent=g16002/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_714\\tAUGUSTUS\\tCDS\\t6151\\t6644\\t0.93\\t+\\t0\\tID=g16002.t1.CDS1;Parent=g16002.t1/contig_714\\tAUGUSTUS\\tCDS\\t6151\\t6648\\t0.93\\t+\\t0\\tID=g16002.t1.CDS1;Parent=g16002.t1/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_714\\tAUGUSTUS\\texon\\t6151\\t6644\\t.\\t+\\t.\\tID=g16002.t1.exon1;Parent=g16002.t1;/contig_714\\tAUGUSTUS\\texon\\t6151\\t6648\\t.\\t+\\t.\\tID=g16002.t1.exon1;Parent=g16002.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16002.t1.CDS2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16002.t1.exon2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_714.AUGUSTUS.intron.6645.6781.1.+...Parent=g16002.t1;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_714\\tAUGUSTUS\\tstop_codon\\t6786\\t6788\\t.\\t+\\t0\\tParent=g16002.t1;/contig_714\\tAUGUSTUS\\tstop_codon\\t6646\\t6648\\t.\\t+\\t0\\tParent=g16002.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_2082\\tManual_annotation\\tgene\\t261\\t525\\t1\\t+\\t.\\tID=man_1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_2082\\tManual_annotation\\tmRNA\\t261\\t525\\t1\\t+\\t.\\tID=man_1.t1;Parent=man_1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_2082\\tManual_annotation\\tstart_codon\\t261\\t263\\t.\\t+\\t.\\tParent=man_1.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_2082\\tManual_annotation\\tCDS\\t261\\t525\\t1.0\\t+\\t0\\tID=man_1.t1.CDS1;Parent=man_1.t1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_2082\\tManual_annotation\\texon\\t261\\t525\\t.\\t+\\t.\\tID=man_1.t1.exon1;Parent=man_1.t1;\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_971\\tManual_annotation\\tgene\\t333\\t506\\t1\\t-\\t.\\tID=man_2;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_971\\tManual_annotation\\tmRNA\\t333\\t506\\t1\\t-\\t.\\tID=man_2.t1;Parent=man_2\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_971\\tManual_annotation\\tstop_codon\\t333\\t335\\t.\\t-\\t.\\tParent=man_2.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_971\\tManual_annotation\\tCDS\\t333\\t506\\t1.0\\t-\\t0\\tID=man_2.t1.CDS1;Parent=man_2.t1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_971\\tManual_annotation\\texon\\t333\\t506\\t.\\t-\\t.\\tID=man_2.t1.exon1;Parent=man_2.t1;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tcontig_971\\tManual_annotation\\tstart_codon\\t504\\t506\\t.\\t-\\t.\\tParent=man_2.t1;\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\n\tcat \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.14630.2.11\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.7213.2.7\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\n\tcp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFoN N139\u003c/h2\u003e\u003ca id=\"user-content-fon-n139\" class=\"anchor\" aria-label=\"Permalink: FoN N139\" href=\"#fon-n139\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that at this stage all codingquary genes contain . characters rather than _ characters\ng16592 - SIX10 - remove gene and use PGN_predictedPass gene\ng16591 - SIX12 - shorten intron and final CDS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDuplicated genes:\nCUFF.639.2.10\nCUFF.7442.1.80 (identified by ncbi)\nCUFF.2100.1.11 (identified by ncbi)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tOrganism=F.oxysporum_fsp_narcissi\n\tStrain=N139_ncbi\n\tOutDir=gene_pred/final_genes/$Organism/$Strain/edited\n\tmkdir -p $OutDir\n\n\tAssembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)\n\tBrakerGff=$(ls gene_pred/braker/$Organism/\u0026quot;$Strain\u0026quot;_braker/*/augustus.gff3)\n\tCodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)\n\tPGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)\n\n\tcat $BrakerGff \\\n\t| grep -v -w 'g16592' \\\n\t| sed \\\n\t's/contig_728\\tAUGUSTUS\\tgene\\t3227\\t3658\\t0.99\\t+\\t.\\tID=g16591;/contig_728\\tAUGUSTUS\\tgene\\t3317\\t3658\\t0.99\\t+\\t.\\tID=g16591;/g' \\\n\t| sed \\\n\t's/contig_728\\tAUGUSTUS\\tmRNA\\t3227\\t3658\\t0.99\\t+\\t.\\tID=g16591.t1;Parent=g16591/contig_728\\tAUGUSTUS\\tmRNA\\t3317\\t3658\\t0.99\\t+\\t.\\tID=g16591.t1;Parent=g16591/g' \\\n\t| sed \\\n\t's/contig_728\\tAUGUSTUS\\tstart_codon\\t3227\\t3229\\t.\\t+\\t0\\tParent=g16591.t1;/contig_728\\tAUGUSTUS\\tstart_codon\\t3317\\t3319\\t.\\t+\\t0\\tParent=g16591.t1;/g' \\\n\t| sed \\\n\t's/contig_728\\tAUGUSTUS\\tCDS\\t3227\\t3248\\t1\\t+\\t0\\tID=g16591.t1.CDS1;Parent=g16591.t1/contig_728\\tAUGUSTUS\\tCDS\\t3317\\t3658\\t1\\t+\\t0\\tID=g16591.t1.CDS1;Parent=g16591.t1/g' \\\n\t| sed \\\n\t's/contig_728\\tAUGUSTUS\\texon\\t3227\\t3248\\t.\\t+\\t.\\tID=g16591.t1.exon1;Parent=g16591.t1;/contig_728\\tAUGUSTUS\\texon\\t3317\\t3658\\t.\\t+\\t.\\tID=g16591.t1.exon1;Parent=g16591.t1;/g' \\\n\t| grep -v -e 'intron.3249.3299.1.+...Parent=g16591.t1;' \\\n\t| grep -v -e 'g16591.t1.CDS2' -e 'g16591.t1.exon2' \\\n\t\u0026gt; $OutDir/final_genes_Braker_ed.gff3\n\n\tcat $CodingQuaryGff | grep -v -w -e 'CUFF.636.2.10' -e 'CUFF.7442.1.80' -e 'CUFF.2100.2.11' \u0026gt; $OutDir/PredictedPass_ed.gff3\n\tcp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\"\u003e\u003cpre\u003e\tOrganism=F.oxysporum_fsp_narcissi\n\tStrain=N139_ncbi\n\tOutDir=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/edited\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\n\tAssembly=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tBrakerGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/braker/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_braker/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/augustus.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tCodingQuaryGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PredictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tPGNGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PGN_predictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\n\tcat \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16592\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_728\\tAUGUSTUS\\tgene\\t3227\\t3658\\t0.99\\t+\\t.\\tID=g16591;/contig_728\\tAUGUSTUS\\tgene\\t3317\\t3658\\t0.99\\t+\\t.\\tID=g16591;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_728\\tAUGUSTUS\\tmRNA\\t3227\\t3658\\t0.99\\t+\\t.\\tID=g16591.t1;Parent=g16591/contig_728\\tAUGUSTUS\\tmRNA\\t3317\\t3658\\t0.99\\t+\\t.\\tID=g16591.t1;Parent=g16591/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_728\\tAUGUSTUS\\tstart_codon\\t3227\\t3229\\t.\\t+\\t0\\tParent=g16591.t1;/contig_728\\tAUGUSTUS\\tstart_codon\\t3317\\t3319\\t.\\t+\\t0\\tParent=g16591.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_728\\tAUGUSTUS\\tCDS\\t3227\\t3248\\t1\\t+\\t0\\tID=g16591.t1.CDS1;Parent=g16591.t1/contig_728\\tAUGUSTUS\\tCDS\\t3317\\t3658\\t1\\t+\\t0\\tID=g16591.t1.CDS1;Parent=g16591.t1/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \\\n\t\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/contig_728\\tAUGUSTUS\\texon\\t3227\\t3248\\t.\\t+\\t.\\tID=g16591.t1.exon1;Parent=g16591.t1;/contig_728\\tAUGUSTUS\\texon\\t3317\\t3658\\t.\\t+\\t.\\tID=g16591.t1.exon1;Parent=g16591.t1;/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eintron.3249.3299.1.+...Parent=g16591.t1;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16591.t1.CDS2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eg16591.t1.exon2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \\\n\t\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\n\n\tcat \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.636.2.10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.7442.1.80\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.2100.2.11\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\n\tcp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFor remaining Fo and Fp isolates\u003c/h2\u003e\u003ca id=\"user-content-for-remaining-fo-and-fp-isolates\" class=\"anchor\" aria-label=\"Permalink: For remaining Fo and Fp isolates\" href=\"#for-remaining-fo-and-fp-isolates\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote that at this stage all codingquary genes contain . characters rather than _ characters\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'ncbi' -e 'Fus2_canu_new' | grep -v -e 'Fus2_canu_new' -e 'A23' -e '125' -e 'N139' | grep -v 'old' | grep 'A8'); do\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\nOutDir=gene_pred/final_genes/$Organism/$Strain/edited\nmkdir -p $OutDir\n\nBrakerGff=$(ls gene_pred/braker/$Organism/\u0026quot;$Strain\u0026quot;_braker/*/augustus.gff3)\nCodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)\nPGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)\nif [[ $Strain == 'A28_ncbi' ]]; then\ncp $BrakerGff $OutDir/final_genes_Braker_ed.gff3\ncat $CodingQuaryGff | grep -v -w -e 'CUFF.7109.1.50' \u0026gt; $OutDir/PredictedPass_ed.gff3\ncp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\nelif [[ $Strain == 'CB3_ncbi' ]]; then\ncp $BrakerGff $OutDir/final_genes_Braker_ed.gff3\ncat $CodingQuaryGff | grep -v -w -e 'CUFF.10490.1.97' \u0026gt; $OutDir/PredictedPass_ed.gff3\ncp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\nelif [[ $Strain == 'PG_ncbi' ]]; then\ncp $BrakerGff $OutDir/final_genes_Braker_ed.gff3\ncat $CodingQuaryGff | grep -v -w -e 'CUFF.7109.1.38' -e 'CUFF.11875.1.103' \u0026gt; $OutDir/PredictedPass_ed.gff3\ncp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\nelse\ncp $BrakerGff $OutDir/final_genes_Braker_ed.gff3\ncp $CodingQuaryGff $OutDir/PredictedPass_ed.gff3\ncp $PGNGff $OutDir/PGN_predictedPass_ed.gff3\nfi\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/edited\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\nBrakerGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/braker/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_braker/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/augustus.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nCodingQuaryGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PredictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nPGNGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/codingquary/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/out/PGN_predictedPass.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e [[ \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28_ncbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e ]]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\ncp \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\ncat \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.7109.1.50\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\ncp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\n\u003cspan class=\"pl-k\"\u003eelif\u003c/span\u003e [[ \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3_ncbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e ]]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\ncp \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\ncat \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.10490.1.97\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\ncp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\n\u003cspan class=\"pl-k\"\u003eelif\u003c/span\u003e [[ \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG_ncbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e ]]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\ncp \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\ncat \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.7109.1.38\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCUFF.11875.1.103\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\ncp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\n\u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e\ncp \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/final_genes_Braker_ed.gff3\ncp \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PredictedPass_ed.gff3\ncp \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\n\u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for EditDir in $(ls -d gene_pred/final_genes/*/*/edited | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'Fus2_canu_new' | grep -e '125' -e 'A23' -e 'N139' | grep 'N139'); do\nStrain=$(echo $EditDir | rev | cut -d '/' -f2 | rev)\nOrganism=$(echo $EditDir | rev | cut -d '/' -f3 | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nAssembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)\nBrakerGff=$EditDir/final_genes_Braker_ed.gff3\nCodingQuaryGff=$EditDir/PredictedPass_ed.gff3\nPGNGff=$EditDir/PGN_predictedPass_ed.gff3\n# ManGff=$EditDir/manual_annotations.gff3\nAddDir=$EditDir/additional\nFinalDir=gene_pred/final_genes/$Organism/$Strain/final\nAddGenesList=$AddDir/additional_genes.txt\nAddGenesGff=$AddDir/additional_genes.gff\n# FinalGff=$AddDir/combined_genes.gff\nmkdir -p $AddDir\nmkdir -p $FinalDir\n\nbedtools intersect -v -a $CodingQuaryGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' \u0026gt; $AddGenesList\nbedtools intersect -v -a $PGNGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' \u0026gt;\u0026gt; $AddGenesList\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation\n$ProgDir/gene_list_to_gff.pl $AddGenesList $CodingQuaryGff CodingQuarry_v2.0 ID CodingQuary \u0026gt; $AddGenesGff\n$ProgDir/gene_list_to_gff.pl $AddGenesList $PGNGff PGNCodingQuarry_v2.0 ID CodingQuary \u0026gt;\u0026gt; $AddGenesGff\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\n\n$ProgDir/add_CodingQuary_features.pl $AddGenesGff $Assembly \u0026gt; $FinalDir/final_genes_CodingQuary.gff3\n$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_CodingQuary.gff3 $FinalDir/final_genes_CodingQuary\ncp $BrakerGff $FinalDir/final_genes_Braker.gff3\n$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_Braker.gff3 $FinalDir/final_genes_Braker\n# cp $ManGff $FinalDir/final_genes_manual.gff3\n# $ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_manual.gff3 $FinalDir/final_genes_manual\ncat $FinalDir/final_genes_Braker.pep.fasta $FinalDir/final_genes_CodingQuary.pep.fasta | sed -r 's/\\*/X/g' \u0026gt; $FinalDir/final_genes_combined.pep.fasta\ncat $FinalDir/final_genes_Braker.cdna.fasta $FinalDir/final_genes_CodingQuary.cdna.fasta \u0026gt; $FinalDir/final_genes_combined.cdna.fasta\ncat $FinalDir/final_genes_Braker.gene.fasta $FinalDir/final_genes_CodingQuary.gene.fasta \u0026gt; $FinalDir/final_genes_combined.gene.fasta\ncat $FinalDir/final_genes_Braker.upstream3000.fasta $FinalDir/final_genes_CodingQuary.upstream3000.fasta \u0026gt; $FinalDir/final_genes_combined.upstream3000.fasta\n\nGffBraker=$FinalDir/final_genes_CodingQuary.gff3\nGffQuary=$FinalDir/final_genes_Braker.gff3\n# GffManual=$FinalDir/final_genes_manual.gff3\nGffAppended=$FinalDir/final_genes_appended.gff3\ncat $GffBraker $GffQuary \u0026gt; $GffAppended\n# cat $GffBraker $GffQuary $GffManual \u0026gt; $GffAppended\n\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eEditDir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d gene_pred/final_genes/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/edited \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$EditDir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$EditDir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nAssembly=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nBrakerGff=\u003cspan class=\"pl-smi\"\u003e$EditDir\u003c/span\u003e/final_genes_Braker_ed.gff3\nCodingQuaryGff=\u003cspan class=\"pl-smi\"\u003e$EditDir\u003c/span\u003e/PredictedPass_ed.gff3\nPGNGff=\u003cspan class=\"pl-smi\"\u003e$EditDir\u003c/span\u003e/PGN_predictedPass_ed.gff3\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ManGff=$EditDir/manual_annotations.gff3\u003c/span\u003e\nAddDir=\u003cspan class=\"pl-smi\"\u003e$EditDir\u003c/span\u003e/additional\nFinalDir=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/final\nAddGenesList=\u003cspan class=\"pl-smi\"\u003e$AddDir\u003c/span\u003e/additional_genes.txt\nAddGenesGff=\u003cspan class=\"pl-smi\"\u003e$AddDir\u003c/span\u003e/additional_genes.gff\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e FinalGff=$AddDir/combined_genes.gff\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$AddDir\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e\n\nbedtools intersect -v -a \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e\nbedtools intersect -v -a \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gene_list_to_gff.pl \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$CodingQuaryGff\u003c/span\u003e CodingQuarry_v2.0 ID CodingQuary \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesGff\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gene_list_to_gff.pl \u003cspan class=\"pl-smi\"\u003e$AddGenesList\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$PGNGff\u003c/span\u003e PGNCodingQuarry_v2.0 ID CodingQuary \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$AddGenesGff\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary\n\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/add_CodingQuary_features.pl \u003cspan class=\"pl-smi\"\u003e$AddGenesGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gff3\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gff2fasta.pl \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gff3 \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary\ncp \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gff3\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gff2fasta.pl \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gff3 \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e cp $ManGff $FinalDir/final_genes_manual.gff3\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e $ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_manual.gff3 $FinalDir/final_genes_manual\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.pep.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed -r \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/\\*/X/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.pep.fasta\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.cdna.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.cdna.fasta \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.cdna.fasta\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gene.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gene.fasta \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.gene.fasta\ncat \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.upstream3000.fasta \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.upstream3000.fasta \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_combined.upstream3000.fasta\n\nGffBraker=\u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_CodingQuary.gff3\nGffQuary=\u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_Braker.gff3\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e GffManual=$FinalDir/final_genes_manual.gff3\u003c/span\u003e\nGffAppended=\u003cspan class=\"pl-smi\"\u003e$FinalDir\u003c/span\u003e/final_genes_appended.gff3\ncat \u003cspan class=\"pl-smi\"\u003e$GffBraker\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$GffQuary\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$GffAppended\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e cat $GffBraker $GffQuary $GffManual \u0026gt; $GffAppended\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAssessing the Gene space in predicted transcriptomes:\u003c/h2\u003e\u003ca id=\"user-content-assessing-the-gene-space-in-predicted-transcriptomes\" class=\"anchor\" aria-label=\"Permalink: Assessing the Gene space in predicted transcriptomes:\" href=\"#assessing-the-gene-space-in-predicted-transcriptomes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor Assembly in $(ls gene_pred/final_genes/*/*/final/final_genes_combined.gene.fasta | grep -e 'Fus2' -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' -e 'canu_new' | grep -v 'old'); do\n\t\tStrain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)\n\t\tOrganism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)\n\t\techo \u0026quot;$Organism - $Strain\u0026quot;\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco\n\t\t# BuscoDB=\u0026quot;Fungal\u0026quot;\n\t\tBuscoDB=$(ls -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9)\n\t\tOutDir=gene_pred/busco/$Organism/$Strain/genes\n\t\tqsub $ProgDir/sub_busco2.sh $Assembly $BuscoDB $OutDir\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final/final_genes_combined.gene.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA13\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA28\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCB3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ePG\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecanu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\tProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco\n\t\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e BuscoDB=\"Fungal\"\u003c/span\u003e\n\t\tBuscoDB=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\tOutDir=gene_pred/busco/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/genes\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_busco2.sh \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$BuscoDB\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tfor File in $(ls gene_pred/busco/*/*/genes/*/short_summary_*.txt); do  \n\t\techo $File;\n\t\tcat $File | grep -e '(C)' -e 'Total';\n\tdone\"\u003e\u003cpre\u003e\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/busco/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/genes/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/short_summary_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e  \n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\tcat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e(C)\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eTotal\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eORF finder\u003c/h2\u003e\u003ca id=\"user-content-orf-finder\" class=\"anchor\" aria-label=\"Permalink: ORF finder\" href=\"#orf-finder\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe genome was searched in six reading frames for any start codon and following\ntranslated identification of a start codon translating sequence until a stop\ncodon was found. This is based upon the atg.pl script used in paper describing\nthe P. infestans genome. Additional functionality was added to this script by\nalso printing ORFs in .gff format.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n\tfor Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -w -e 'Fus2'); do\n\t\tqsub $ProgDir/run_ORF_finder.sh $Genome\n\tdone\"\u003e\u003cpre\u003e\tProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eGenome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_ORF_finder.sh \u003cspan class=\"pl-smi\"\u003e$Genome\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe Gff files from the the ORF finder are not in true Gff3 format. These were\ncorrected using the following commands:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tProgDir=~/git_repos/emr_repos/tools/seq_tools/feature_annotation\n\tfor ORF_Gff in $(ls gene_pred/ORF_finder/*/*/*_ORF.gff | grep -v '_F_atg_' | grep -v '_R_atg_'); do\n\t\tORF_Gff_mod=$(echo $ORF_Gff | sed 's/_ORF.gff/_ORF_corrected.gff3/g')\n\t\techo \u0026quot;\u0026quot;\n\t\techo \u0026quot;Correcting the following file:\u0026quot;\n\t\techo $ORF_Gff\n\t\techo \u0026quot;Redirecting to:\u0026quot;\n\t\techo $ORF_Gff_mod\n\t\t$ProgDir/gff_corrector.pl $ORF_Gff \u0026gt; $ORF_Gff_mod\n\tdone\"\u003e\u003cpre\u003e\tProgDir=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/git_repos/emr_repos/tools/seq_tools/feature_annotation\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eORF_Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/ORF_finder/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_ORF.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_F_atg_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_R_atg_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\tORF_Gff_mod=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$ORF_Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_ORF.gff/_ORF_corrected.gff3/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCorrecting the following file:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ORF_Gff\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRedirecting to:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ORF_Gff_mod\u003c/span\u003e\n\t\t\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gff_corrector.pl \u003cspan class=\"pl-smi\"\u003e$ORF_Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$ORF_Gff_mod\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e#Functional annotation\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA) Interproscan\u003c/h2\u003e\u003ca id=\"user-content-a-interproscan\" class=\"anchor\" aria-label=\"Permalink: A) Interproscan\" href=\"#a-interproscan\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInterproscan was used to give gene models functional annotations.\nAnnotation was run using the commands below:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNote: This is a long-running script. As such, these commands were run using\n'screen' to allow jobs to be submitted and monitored in the background.\nThis allows the session to be disconnected and reconnected over time.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eScreen ouput detailing the progress of submission of interporscan jobs\nwas redirected to a temporary output file named interproscan_submission.log .\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tscreen -a\n\tcd /home/groups/harrisonlab/project_files/fusarium\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan\n\tfor Genes in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139' ); do\n\techo $Genes\n\t$ProgDir/sub_interproscan.sh $Genes\n\tdone 2\u0026gt;\u0026amp;1 | tee -a interproscan_submisison.log\"\u003e\u003cpre\u003e\tscreen -a\n\t\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /home/groups/harrisonlab/project_files/fusarium\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Genes\u003c/span\u003e\n\t\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_interproscan.sh \u003cspan class=\"pl-smi\"\u003e$Genes\u003c/span\u003e\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e2\u0026gt;\u0026amp;1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tee -a interproscan_submisison.log\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFollowing interproscan annotation split files were combined using the following\ncommands:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan\nfor Proteins in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139'); do\nStrain=$(echo $Proteins | rev | cut -d '/' -f3 | rev)\nOrganism=$(echo $Proteins | rev | cut -d '/' -f4 | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\necho $Strain\nInterProRaw=gene_pred/interproscan/$Organism/$Strain/raw\n$ProgDir/append_interpro.sh $Proteins $InterProRaw\ndone\"\u003e\u003cpre\u003eProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eProteins\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteins\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteins\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nInterProRaw=gene_pred/interproscan/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/raw\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/append_interpro.sh \u003cspan class=\"pl-smi\"\u003e$Proteins\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$InterProRaw\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB) SwissProt\u003c/h2\u003e\u003ca id=\"user-content-b-swissprot\" class=\"anchor\" aria-label=\"Permalink: B) SwissProt\" href=\"#b-swissprot\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139'); do\nStrain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)\nOutDir=gene_pred/swissprot/$Organism/$Strain\nSwissDbDir=../../uniprot/swissprot\nSwissDbName=uniprot_sprot\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot\nqsub $ProgDir/sub_swissprot.sh $Proteome $OutDir $SwissDbDir $SwissDbName\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eProteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=gene_pred/swissprot/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nSwissDbDir=../../uniprot/swissprot\nSwissDbName=uniprot_sprot\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_swissprot.sh \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$SwissDbDir\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$SwissDbName\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for SwissTable in $(ls gene_pred/swissprot/*/*/swissprot_vJul2016_10_hits.tbl | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139'); do\n# SwissTable=gene_pred/swissprot/Fus2/swissprot_v2015_10_hits.tbl\nStrain=$(echo $SwissTable | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $SwissTable | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nOutTable=gene_pred/swissprot/$Organism/$Strain/swissprot_vJul2016_tophit_parsed.tbl\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot\n$ProgDir/swissprot_parser.py --blast_tbl $SwissTable --blast_db_fasta ../../uniprot/swissprot/uniprot_sprot.fasta \u0026gt; $OutTable\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSwissTable\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/swissprot/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/swissprot_vJul2016_10_hits.tbl \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e125\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e SwissTable=gene_pred/swissprot/Fus2/swissprot_v2015_10_hits.tbl\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$SwissTable\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$SwissTable\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutTable=gene_pred/swissprot/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/swissprot_vJul2016_tophit_parsed.tbl\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/swissprot_parser.py --blast_tbl \u003cspan class=\"pl-smi\"\u003e$SwissTable\u003c/span\u003e --blast_db_fasta ../../uniprot/swissprot/uniprot_sprot.fasta \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutTable\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC) CAZY proteins\u003c/h2\u003e\u003ca id=\"user-content-c-cazy-proteins\" class=\"anchor\" aria-label=\"Permalink: C) CAZY proteins\" href=\"#c-cazy-proteins\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCarbohydrte active enzymes were idnetified using CAZYfollowing recomendations\nat \u003ca href=\"http://csbl.bmb.uga.edu/dbCAN/download/readme.txt\" rel=\"nofollow\"\u003ehttp://csbl.bmb.uga.edu/dbCAN/download/readme.txt\u003c/a\u003e :\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do\nStrain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)\nOutDir=gene_pred/CAZY/$Organism/$Strain\nmkdir -p $OutDir\nPrefix=\u0026quot;$Strain\u0026quot;_CAZY\nCazyHmm=../../dbCAN/dbCAN-fam-HMMs.txt\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/HMMER\nqsub $ProgDir/sub_hmmscan.sh $CazyHmm $Proteome $Prefix $OutDir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eProteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=gene_pred/CAZY/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nPrefix=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY\nCazyHmm=../../dbCAN/dbCAN-fam-HMMs.txt\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/HMMER\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sub_hmmscan.sh \u003cspan class=\"pl-smi\"\u003e$CazyHmm\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Prefix\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe Hmm parser was used to filter hits by an E-value of E1x10-5 or E1x10-e3 if they had a hit over a length of X %.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThose proteins with a signal peptide were extracted from the list and gff files\nrepresenting these proteins made.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls gene_pred/CAZY/*/*/*CAZY.out.dm | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi'); do\nStrain=$(echo $File | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $File | rev | cut -f3 -d '/' | rev)\nOutDir=$(dirname $File)\necho \u0026quot;$Organism - $Strain\u0026quot;\nProgDir=/home/groups/harrisonlab/dbCAN\n$ProgDir/hmmscan-parser.sh $OutDir/\u0026quot;$Strain\u0026quot;_CAZY.out.dm \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_CAZY.out.dm.ps\nCazyHeaders=$(echo $File | sed 's/.out.dm/_headers.txt/g')\ncat $OutDir/\u0026quot;$Strain\u0026quot;_CAZY.out.dm.ps | cut -f3 | sort | uniq \u0026gt; $CazyHeaders\necho \u0026quot;number of CAZY proteins identified:\u0026quot;\ncat $CazyHeaders | wc -l\n# Gff=$(ls gene_pred/codingquary/$Organism/$Strain/final/final_genes_appended.gff3)\nGff=$(ls gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3)\nCazyGff=$OutDir/\u0026quot;$Strain\u0026quot;_CAZY.gff\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n$ProgDir/extract_gff_for_sigP_hits.pl $CazyHeaders $Gff CAZyme ID \u0026gt; $CazyGff\necho \u0026quot;number of CAZY genes identified:\u0026quot;\ncat $CazyGff | grep -w 'gene' | wc -l\n\nSecretedProts=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/\u0026quot;$Strain\u0026quot;_final_sp_no_trans_mem.aa)\nSecretedHeaders=$(echo $SecretedProts | sed 's/.aa/_headers.txt/g')\ncat $SecretedProts | grep '\u0026gt;' | tr -d '\u0026gt;' \u0026gt; $SecretedHeaders\nCazyGffSecreted=$OutDir/\u0026quot;$Strain\u0026quot;_CAZY_secreted.gff\n$ProgDir/extract_gff_for_sigP_hits.pl $SecretedHeaders $CazyGff Secreted_CAZyme ID \u0026gt; $CazyGffSecreted\necho \u0026quot;number of Secreted CAZY proteins identified:\u0026quot;\ncat $CazyGffSecreted | grep -w 'mRNA' | cut -f9 | tr -d 'ID=' | cut -f1 -d ';' \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_CAZY_secreted_headers.txt\ncat $OutDir/\u0026quot;$Strain\u0026quot;_CAZY_secreted_headers.txt | wc -l\necho \u0026quot;number of Secreted CAZY genes identified:\u0026quot;\ncat $CazyGffSecreted | grep -w 'gene' | wc -l\n# cat $OutDir/\u0026quot;$Strain\u0026quot;_CAZY_secreted_headers.txt | cut -f1 -d '.' | sort | uniq | wc -l\ndone \u0026gt; tmp.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/CAZY/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eCAZY.out.dm \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/groups/harrisonlab/dbCAN\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/hmmscan-parser.sh \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY.out.dm \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY.out.dm.ps\nCazyHeaders=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.out.dm/_headers.txt/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY.out.dm.ps \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$CazyHeaders\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enumber of CAZY proteins identified:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$CazyHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Gff=$(ls gene_pred/codingquary/$Organism/$Strain/final/final_genes_appended.gff3)\u003c/span\u003e\nGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/final/final_genes_appended.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nCazyGff=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY.gff\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/extract_gff_for_sigP_hits.pl \u003cspan class=\"pl-smi\"\u003e$CazyHeaders\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Gff\u003c/span\u003e CAZyme ID \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$CazyGff\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enumber of CAZY genes identified:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$CazyGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\nSecretedProts=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes_signalp-4.1/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_sp_no_trans_mem.aa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nSecretedHeaders=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$SecretedProts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.aa/_headers.txt/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$SecretedProts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$SecretedHeaders\u003c/span\u003e\nCazyGffSecreted=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY_secreted.gff\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/extract_gff_for_sigP_hits.pl \u003cspan class=\"pl-smi\"\u003e$SecretedHeaders\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$CazyGff\u003c/span\u003e Secreted_CAZyme ID \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$CazyGffSecreted\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enumber of Secreted CAZY proteins identified:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$CazyGffSecreted\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f9 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eID=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY_secreted_headers.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_CAZY_secreted_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enumber of Secreted CAZY genes identified:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$CazyGffSecreted\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e cat $OutDir/\"$Strain\"_CAZY_secreted_headers.txt | cut -f1 -d '.' | sort | uniq | wc -l\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - 125_ncbi\nnumber of CAZY proteins identified:\n925\nnumber of CAZY genes identified:\n925\nnumber of Secreted CAZY proteins identified:\n386\nnumber of Secreted CAZY genes identified:\n386\nF.oxysporum_fsp_cepae - A13_ncbi\nnumber of CAZY proteins identified:\n965\nnumber of CAZY genes identified:\n965\nnumber of Secreted CAZY proteins identified:\n397\nnumber of Secreted CAZY genes identified:\n397\nF.oxysporum_fsp_cepae - A23_ncbi\nnumber of CAZY proteins identified:\n930\nnumber of CAZY genes identified:\n930\nnumber of Secreted CAZY proteins identified:\n387\nnumber of Secreted CAZY genes identified:\n387\nF.oxysporum_fsp_cepae - A28_ncbi\nnumber of CAZY proteins identified:\n919\nnumber of CAZY genes identified:\n919\nnumber of Secreted CAZY proteins identified:\n376\nnumber of Secreted CAZY genes identified:\n376\nF.oxysporum_fsp_cepae - CB3_ncbi\nnumber of CAZY proteins identified:\n909\nnumber of CAZY genes identified:\n909\nnumber of Secreted CAZY proteins identified:\n383\nnumber of Secreted CAZY genes identified:\n383\nF.oxysporum_fsp_cepae - Fus2_canu_new\nnumber of CAZY proteins identified:\n940\nnumber of CAZY genes identified:\n940\nnumber of Secreted CAZY proteins identified:\n386\nnumber of Secreted CAZY genes identified:\n386\nF.oxysporum_fsp_cepae - PG_ncbi\nnumber of CAZY proteins identified:\n914\nnumber of CAZY genes identified:\n914\nnumber of Secreted CAZY proteins identified:\n381\nnumber of Secreted CAZY genes identified:\n381\nF.oxysporum_fsp_narcissi - N139_ncbi\nnumber of CAZY proteins identified:\n961\nnumber of CAZY genes identified:\n961\nnumber of Secreted CAZY proteins identified:\n399\nnumber of Secreted CAZY genes identified:\n399\nF.proliferatum - A8_ncbi\nnumber of CAZY proteins identified:\n819\nnumber of CAZY genes identified:\n819\nnumber of Secreted CAZY proteins identified:\n341\nnumber of Secreted CAZY genes identified:\n341\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - 125_ncbi\nnumber of CAZY proteins identified:\n925\nnumber of CAZY genes identified:\n925\nnumber of Secreted CAZY proteins identified:\n386\nnumber of Secreted CAZY genes identified:\n386\nF.oxysporum_fsp_cepae - A13_ncbi\nnumber of CAZY proteins identified:\n965\nnumber of CAZY genes identified:\n965\nnumber of Secreted CAZY proteins identified:\n397\nnumber of Secreted CAZY genes identified:\n397\nF.oxysporum_fsp_cepae - A23_ncbi\nnumber of CAZY proteins identified:\n930\nnumber of CAZY genes identified:\n930\nnumber of Secreted CAZY proteins identified:\n387\nnumber of Secreted CAZY genes identified:\n387\nF.oxysporum_fsp_cepae - A28_ncbi\nnumber of CAZY proteins identified:\n919\nnumber of CAZY genes identified:\n919\nnumber of Secreted CAZY proteins identified:\n376\nnumber of Secreted CAZY genes identified:\n376\nF.oxysporum_fsp_cepae - CB3_ncbi\nnumber of CAZY proteins identified:\n909\nnumber of CAZY genes identified:\n909\nnumber of Secreted CAZY proteins identified:\n383\nnumber of Secreted CAZY genes identified:\n383\nF.oxysporum_fsp_cepae - Fus2_canu_new\nnumber of CAZY proteins identified:\n940\nnumber of CAZY genes identified:\n940\nnumber of Secreted CAZY proteins identified:\n386\nnumber of Secreted CAZY genes identified:\n386\nF.oxysporum_fsp_cepae - PG_ncbi\nnumber of CAZY proteins identified:\n914\nnumber of CAZY genes identified:\n914\nnumber of Secreted CAZY proteins identified:\n381\nnumber of Secreted CAZY genes identified:\n381\nF.oxysporum_fsp_narcissi - N139_ncbi\nnumber of CAZY proteins identified:\n961\nnumber of CAZY genes identified:\n961\nnumber of Secreted CAZY proteins identified:\n399\nnumber of Secreted CAZY genes identified:\n399\nF.proliferatum - A8_ncbi\nnumber of CAZY proteins identified:\n819\nnumber of CAZY genes identified:\n819\nnumber of Secreted CAZY proteins identified:\n341\nnumber of Secreted CAZY genes identified:\n341\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNote - the CAZY genes identified may need further filtering based on e value and\ncuttoff length - see below:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCols in yourfile.out.dm.ps:\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eFamily HMM\u003c/li\u003e\n\u003cli\u003eHMM length\u003c/li\u003e\n\u003cli\u003eQuery ID\u003c/li\u003e\n\u003cli\u003eQuery length\u003c/li\u003e\n\u003cli\u003eE-value (how similar to the family HMM)\u003c/li\u003e\n\u003cli\u003eHMM start\u003c/li\u003e\n\u003cli\u003eHMM end\u003c/li\u003e\n\u003cli\u003eQuery start\u003c/li\u003e\n\u003cli\u003eQuery end\u003c/li\u003e\n\u003cli\u003eCoverage\u003c/li\u003e\n\u003c/ol\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFor fungi, use E-value \u0026lt; 1e-17 and coverage \u0026gt; 0.45\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eThe best threshold varies for different CAZyme classes (please see \u003ca href=\"http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4132414/\" rel=\"nofollow\"\u003ehttp://www.ncbi.nlm.nih.gov/pmc/articles/PMC4132414/\u003c/a\u003e for details). Basically to annotate GH proteins, one should use a very relax coverage cutoff or the sensitivity will be low (Supplementary Tables S4 and S9); (ii) to annotate CE families a very stringent E-value cutoff and coverage cutoff should be used; otherwise the precision will be very low due to a very high false positive rate (Supplementary Tables S5 and S10)\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSummary of CAZY families by organism\u003c/h3\u003e\u003ca id=\"user-content-summary-of-cazy-families-by-organism\" class=\"anchor\" aria-label=\"Permalink: Summary of CAZY families by organism\" href=\"#summary-of-cazy-families-by-organism\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for CAZY in $(ls gene_pred/CAZY/*/*/*_CAZY.out.dm.ps | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -e 'Fus2' -e 'N139'); do\n  Strain=$(echo $CAZY | rev | cut -f2 -d '/' | rev)\n  Organism=$(echo $CAZY | rev | cut -f3 -d '/' | rev)\n  OutDir=$(dirname $CAZY)\n  echo \u0026quot;$Organism - $Strain\u0026quot;\n  Secreted=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp_no_trans_mem_headers.txt)\n  Gff=gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3\n  ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/CAZY\n  $ProgDir/summarise_CAZY.py --cazy $CAZY --inp_secreted $Secreted --inp_gff $Gff --summarise_family --trim_gene_id 2 --kubicek_2014\ndone | less -S\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eCAZY\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/CAZY/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_CAZY.out.dm.ps \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eN139\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n  Strain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$CAZY\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n  Organism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$CAZY\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n  OutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$CAZY\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n  \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n  Secreted=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes_signalp-4.1/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_sp_no_trans_mem_headers.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n  Gff=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/final/final_genes_appended.gff3\n  ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/CAZY\n  \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/summarise_CAZY.py --cazy \u003cspan class=\"pl-smi\"\u003e$CAZY\u003c/span\u003e --inp_secreted \u003cspan class=\"pl-smi\"\u003e$Secreted\u003c/span\u003e --inp_gff \u003cspan class=\"pl-smi\"\u003e$Gff\u003c/span\u003e --summarise_family --trim_gene_id 2 --kubicek_2014\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e less -S\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - 125_ncbi\nAA - 70\nGT - 3\nCE - 64\nCBM - 46\nPL - 25\nGH - 177\nF.oxysporum_fsp_cepae - A13_ncbi\nAA - 67\nGT - 2\nCE - 71\nCBM - 51\nPL - 24\nGH - 182\nF.oxysporum_fsp_cepae - A23_ncbi\nAA - 69\nGT - 3\nCE - 67\nCBM - 48\nPL - 25\nGH - 175\nF.oxysporum_fsp_cepae - A28_ncbi\nAA - 63\nGT - 2\nCE - 64\nCBM - 49\nPL - 22\nGH - 176\nF.oxysporum_fsp_cepae - CB3_ncbi\nAA - 64\nGT - 3\nCE - 69\nCBM - 42\nPL - 25\nGH - 179\nF.oxysporum_fsp_cepae - Fus2_canu_new\nAA - 69\nGT - 3\nCE - 66\nCBM - 47\nPL - 26\nGH - 174\nF.oxysporum_fsp_cepae - PG_ncbi\nAA - 66\nGT - 2\nCE - 65\nCBM - 48\nPL - 27\nGH - 173\nF.oxysporum_fsp_narcissi - N139_ncbi\nAA - 68\nGT - 1\nCE - 66\nCBM - 41\nPL - 27\nGH - 195\nF.proliferatum - A8_ncbi\nAA - 64\nGT - 3\nCE - 64\nCBM - 28\nPL - 26\nGH - 156\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - 125_ncbi\nAA - 70\nGT - 3\nCE - 64\nCBM - 46\nPL - 25\nGH - 177\nF.oxysporum_fsp_cepae - A13_ncbi\nAA - 67\nGT - 2\nCE - 71\nCBM - 51\nPL - 24\nGH - 182\nF.oxysporum_fsp_cepae - A23_ncbi\nAA - 69\nGT - 3\nCE - 67\nCBM - 48\nPL - 25\nGH - 175\nF.oxysporum_fsp_cepae - A28_ncbi\nAA - 63\nGT - 2\nCE - 64\nCBM - 49\nPL - 22\nGH - 176\nF.oxysporum_fsp_cepae - CB3_ncbi\nAA - 64\nGT - 3\nCE - 69\nCBM - 42\nPL - 25\nGH - 179\nF.oxysporum_fsp_cepae - Fus2_canu_new\nAA - 69\nGT - 3\nCE - 66\nCBM - 47\nPL - 26\nGH - 174\nF.oxysporum_fsp_cepae - PG_ncbi\nAA - 66\nGT - 2\nCE - 65\nCBM - 48\nPL - 27\nGH - 173\nF.oxysporum_fsp_narcissi - N139_ncbi\nAA - 68\nGT - 1\nCE - 66\nCBM - 41\nPL - 27\nGH - 195\nF.proliferatum - A8_ncbi\nAA - 64\nGT - 3\nCE - 64\nCBM - 28\nPL - 26\nGH - 156\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - 125_ncbi\nB-Galactosidases - 3\nA-Galactosidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 26\nXylanases - 11\nPolygalacturonate lyases - 26\nB-Glucuronidases - 4\nB-Glycosidases - 13\nCellulases - 20\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - A13_ncbi\nB-Galactosidases - 3\nA-Galactosidases - 3\nPolygalacturonase - 14\nA-Arabinosidases - 28\nXylanases - 11\nPolygalacturonate lyases - 26\nB-Glucuronidases - 3\nB-Glycosidases - 13\nCellulases - 22\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - A23_ncbi\nB-Galactosidases - 3\nB-Glucuronidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 26\nXylanases - 11\nPolygalacturonate lyases - 26\nA-Galactosidases - 4\nB-Glycosidases - 13\nCellulases - 21\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - A28_ncbi\nB-Galactosidases - 2\nA-Galactosidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 29\nXylanases - 11\nPolygalacturonate lyases - 24\nB-Glucuronidases - 4\nB-Glycosidases - 13\nCellulases - 21\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - CB3_ncbi\nB-Galactosidases - 2\nB-Glucuronidases - 4\nPolygalacturonase - 12\nA-Arabinosidases - 28\nXylanases - 10\nPolygalacturonate lyases - 26\nA-Galactosidases - 4\nB-Glycosidases - 14\nCellulases - 21\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - Fus2_canu_new\nB-Galactosidases - 2\nA-Galactosidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 26\nXylanases - 11\nPolygalacturonate lyases - 27\nB-Glucuronidases - 4\nB-Glycosidases - 13\nCellulases - 20\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - PG_ncbi\nB-Galactosidases - 2\nA-Galactosidases - 4\nPolygalacturonase - 12\nA-Arabinosidases - 28\nXylanases - 10\nPolygalacturonate lyases - 26\nB-Glucuronidases - 4\nB-Glycosidases - 12\nCellulases - 20\nXyloglucanases - 1\nF.oxysporum_fsp_narcissi - N139_ncbi\nB-Galactosidases - 2\nB-Glucuronidases - 4\nPolygalacturonase - 12\nA-Arabinosidases - 28\nXylanases - 11\nPolygalacturonate lyases - 27\nA-Galactosidases - 4\nB-Glycosidases - 17\nCellulases - 20\nXyloglucanases - 1\nF.proliferatum - A8_ncbi\nB-Galactosidases - 2\nA-Galactosidases - 3\nPolygalacturonase - 11\nA-Arabinosidases - 20\nXylanases - 9\nPolygalacturonate lyases - 26\nB-Glucuronidases - 2\nB-Glycosidases - 10\nCellulases - 21\nXyloglucanases - 1\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - 125_ncbi\nB-Galactosidases - 3\nA-Galactosidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 26\nXylanases - 11\nPolygalacturonate lyases - 26\nB-Glucuronidases - 4\nB-Glycosidases - 13\nCellulases - 20\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - A13_ncbi\nB-Galactosidases - 3\nA-Galactosidases - 3\nPolygalacturonase - 14\nA-Arabinosidases - 28\nXylanases - 11\nPolygalacturonate lyases - 26\nB-Glucuronidases - 3\nB-Glycosidases - 13\nCellulases - 22\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - A23_ncbi\nB-Galactosidases - 3\nB-Glucuronidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 26\nXylanases - 11\nPolygalacturonate lyases - 26\nA-Galactosidases - 4\nB-Glycosidases - 13\nCellulases - 21\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - A28_ncbi\nB-Galactosidases - 2\nA-Galactosidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 29\nXylanases - 11\nPolygalacturonate lyases - 24\nB-Glucuronidases - 4\nB-Glycosidases - 13\nCellulases - 21\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - CB3_ncbi\nB-Galactosidases - 2\nB-Glucuronidases - 4\nPolygalacturonase - 12\nA-Arabinosidases - 28\nXylanases - 10\nPolygalacturonate lyases - 26\nA-Galactosidases - 4\nB-Glycosidases - 14\nCellulases - 21\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - Fus2_canu_new\nB-Galactosidases - 2\nA-Galactosidases - 4\nPolygalacturonase - 13\nA-Arabinosidases - 26\nXylanases - 11\nPolygalacturonate lyases - 27\nB-Glucuronidases - 4\nB-Glycosidases - 13\nCellulases - 20\nXyloglucanases - 1\nF.oxysporum_fsp_cepae - PG_ncbi\nB-Galactosidases - 2\nA-Galactosidases - 4\nPolygalacturonase - 12\nA-Arabinosidases - 28\nXylanases - 10\nPolygalacturonate lyases - 26\nB-Glucuronidases - 4\nB-Glycosidases - 12\nCellulases - 20\nXyloglucanases - 1\nF.oxysporum_fsp_narcissi - N139_ncbi\nB-Galactosidases - 2\nB-Glucuronidases - 4\nPolygalacturonase - 12\nA-Arabinosidases - 28\nXylanases - 11\nPolygalacturonate lyases - 27\nA-Galactosidases - 4\nB-Glycosidases - 17\nCellulases - 20\nXyloglucanases - 1\nF.proliferatum - A8_ncbi\nB-Galactosidases - 2\nA-Galactosidases - 3\nPolygalacturonase - 11\nA-Arabinosidases - 20\nXylanases - 9\nPolygalacturonate lyases - 26\nB-Glucuronidases - 2\nB-Glycosidases - 10\nCellulases - 21\nXyloglucanases - 1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eD) AntiSMASH\u003c/h2\u003e\u003ca id=\"user-content-d-antismash\" class=\"anchor\" aria-label=\"Permalink: D) AntiSMASH\" href=\"#d-antismash\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAntismash was run to identify clusters of secondary metabolite genes within\nthe genome. Antismash was run using the weserver at:\n\u003ca href=\"http://antismash.secondarymetabolites.org\" rel=\"nofollow\"\u003ehttp://antismash.secondarymetabolites.org\u003c/a\u003e\u003c/p\u003e\n\n\n\u003cp dir=\"auto\"\u003eResults of web-annotation of gene clusters within the assembly were downloaded to\nthe following directories:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'canu_new' -e 'ncbi'); do\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\nOutDir=analysis/antismash/$Organism/$Strain\nmkdir -p $OutDir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_softmasked_repeatmasker_TPSI_appended.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecanu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=analysis/antismash/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Zip in $(ls analysis/antismash/*/*/*.zip | grep -e 'ncbi' | grep 'A23'); do\nOutDir=$(dirname $Zip)\nunzip -d $OutDir $Zip\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eZip\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/antismash/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.zip \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$Zip\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nunzip -d \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Zip\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"# AntiSmash=analysis/antismash/79c1471f-4a2b-41f7-ba36-18ba94675f59/contig_1_pilon.final.gbk\nfor AntiSmash in $(ls analysis/antismash/*/*/*/*.final.gbk | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'A23'); do\nOrganism=$(echo $AntiSmash | rev | cut -f4 -d '/' | rev)\nStrain=$(echo $AntiSmash | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nOutDir=analysis/antismash/$Organism/$Strain\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/secondary_metabolites\n$ProgDir/antismash2gff.py --inp_antismash $AntiSmash \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_secondary_metabolite_regions.gff\nprintf \u0026quot;Number of clusters detected:\\t\u0026quot;\ncat $OutDir/\u0026quot;$Strain\u0026quot;_secondary_metabolite_regions.gff | grep 'antismash_cluster' | wc -l\n# GeneGff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3\nGeneGff=gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3\nbedtools intersect -u -a $GeneGff -b $OutDir/\u0026quot;$Strain\u0026quot;_secondary_metabolite_regions.gff \u0026gt; $OutDir/metabolite_cluster_genes.gff\ncat $OutDir/metabolite_cluster_genes.gff | grep -w 'mRNA' | cut -f9 | cut -f2 -d '=' | cut -f1 -d ';' \u0026gt; $OutDir/metabolite_cluster_gene_headers.txt\nprintf \u0026quot;Number of predicted proteins in clusters:\\t\u0026quot;\ncat $OutDir/metabolite_cluster_gene_headers.txt | wc -l\nprintf \u0026quot;Number of predicted genes in clusters:\\t\u0026quot;\ncat $OutDir/metabolite_cluster_genes.gff | grep -w 'gene' | wc -l\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e AntiSmash=analysis/antismash/79c1471f-4a2b-41f7-ba36-18ba94675f59/contig_1_pilon.final.gbk\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAntiSmash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/antismash/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.final.gbk \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA23\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$AntiSmash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$AntiSmash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutDir=analysis/antismash/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/secondary_metabolites\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/antismash2gff.py --inp_antismash \u003cspan class=\"pl-smi\"\u003e$AntiSmash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_secondary_metabolite_regions.gff\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of clusters detected:\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_secondary_metabolite_regions.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eantismash_cluster\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e GeneGff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3\u003c/span\u003e\nGeneGff=gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/final/final_genes_appended.gff3\nbedtools intersect -u -a \u003cspan class=\"pl-smi\"\u003e$GeneGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_secondary_metabolite_regions.gff \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/metabolite_cluster_genes.gff\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/metabolite_cluster_genes.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f9 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/metabolite_cluster_gene_headers.txt\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of predicted proteins in clusters:\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/metabolite_cluster_gene_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of predicted genes in clusters:\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/metabolite_cluster_genes.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThese clusters represented the following genes. Note that these numbers just\nshow the number of intersected genes with gff clusters and are not confirmed by\nfunction\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - 125_ncbi\nNumber of clusters detected:\t46\nNumber of predicted proteins in clusters:\t568\nNumber of predicted genes in clusters:\t559\nF.oxysporum_fsp_cepae - A13_ncbi\nNumber of clusters detected:\t49\nNumber of predicted proteins in clusters:\t626\nNumber of predicted genes in clusters:\t620\nF.oxysporum_fsp_cepae - A23_ncbi\nNumber of clusters detected:\t35\nNumber of predicted proteins in clusters:\t436\nNumber of predicted genes in clusters:\t432\nF.oxysporum_fsp_cepae - A28_ncbi\nNumber of clusters detected:\t46\nNumber of predicted proteins in clusters:\t668\nNumber of predicted genes in clusters:\t656\nF.oxysporum_fsp_cepae - CB3_ncbi\nNumber of clusters detected:\t44\nNumber of predicted proteins in clusters:\t613\nNumber of predicted genes in clusters:\t604\nF.oxysporum_fsp_cepae - Fus2_canu_new\nNumber of clusters detected:\t50\nNumber of predicted proteins in clusters:\t720\nNumber of predicted genes in clusters:\t703\nF.oxysporum_fsp_cepae - PG_ncbi\nNumber of clusters detected:\t45\nNumber of predicted proteins in clusters:\t646\nNumber of predicted genes in clusters:\t638\nF.oxysporum_fsp_narcissi - N139_ncbi\nNumber of clusters detected:\t46\nNumber of predicted proteins in clusters:\t606\nNumber of predicted genes in clusters:\t601\nF.proliferatum - A8_ncbi\nNumber of clusters detected:\t58\nNumber of predicted proteins in clusters:\t762\nNumber of predicted genes in clusters:\t762\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - 125_ncbi\nNumber of clusters detected:\t46\nNumber of predicted proteins in clusters:\t568\nNumber of predicted genes in clusters:\t559\nF.oxysporum_fsp_cepae - A13_ncbi\nNumber of clusters detected:\t49\nNumber of predicted proteins in clusters:\t626\nNumber of predicted genes in clusters:\t620\nF.oxysporum_fsp_cepae - A23_ncbi\nNumber of clusters detected:\t35\nNumber of predicted proteins in clusters:\t436\nNumber of predicted genes in clusters:\t432\nF.oxysporum_fsp_cepae - A28_ncbi\nNumber of clusters detected:\t46\nNumber of predicted proteins in clusters:\t668\nNumber of predicted genes in clusters:\t656\nF.oxysporum_fsp_cepae - CB3_ncbi\nNumber of clusters detected:\t44\nNumber of predicted proteins in clusters:\t613\nNumber of predicted genes in clusters:\t604\nF.oxysporum_fsp_cepae - Fus2_canu_new\nNumber of clusters detected:\t50\nNumber of predicted proteins in clusters:\t720\nNumber of predicted genes in clusters:\t703\nF.oxysporum_fsp_cepae - PG_ncbi\nNumber of clusters detected:\t45\nNumber of predicted proteins in clusters:\t646\nNumber of predicted genes in clusters:\t638\nF.oxysporum_fsp_narcissi - N139_ncbi\nNumber of clusters detected:\t46\nNumber of predicted proteins in clusters:\t606\nNumber of predicted genes in clusters:\t601\nF.proliferatum - A8_ncbi\nNumber of clusters detected:\t58\nNumber of predicted proteins in clusters:\t762\nNumber of predicted genes in clusters:\t762\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Antismash in $(ls analysis/antismash/F.*/*/*_secondary_metabolite_regions.gff | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'Fus2'); do\n\tOrganism=$(echo $Antismash | rev | cut -f3 -d '/' | rev)\n\tStrain=$(echo $Antismash | rev | cut -f2 -d '/' | rev)\n\techo \u0026quot;$Organism - $Strain\u0026quot;\n\tcat $Antismash | cut -f3 | sort | uniq -c\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAntismash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/antismash/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_secondary_metabolite_regions.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Antismash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\tStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Antismash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$Antismash\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -c\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - 125_ncbi\n      3 indole\n     11 nrps\n      1 nrps-indole\n      4 nrps-t1pks\n     10 other\n      6 t1pks\n      1 t3pks\n     10 terpene\nF.oxysporum_fsp_cepae - A13_ncbi\n      3 indole\n     12 nrps\n      3 nrps-t1pks\n     10 other\n      9 t1pks\n      1 t1pks-nrps\n      1 t3pks\n     10 terpene\nF.oxysporum_fsp_cepae - A23_ncbi\n      3 indole\n     11 nrps\n      1 nrps-indole\n      4 nrps-t1pks\n     10 other\n      6 t1pks\n      1 t3pks\n     10 terpene\nF.oxysporum_fsp_cepae - A28_ncbi\n      3 indole\n      1 indole-nrps\n     11 nrps\n      2 nrps-t1pks\n     10 other\n      8 t1pks\n      1 t1pks-nrps\n      1 t3pks\n      9 terpene\nF.oxysporum_fsp_cepae - CB3_ncbi\n      1 bacteriocin\n      3 indole\n      1 indole-nrps\n     11 nrps\n      3 nrps-t1pks\n      9 other\n      6 t1pks\n      1 t1pks-indole\n      1 t3pks\n      8 terpene\nF.oxysporum_fsp_cepae - Fus2_canu_new\n      3 indole\n      1 indole-nrps\n     11 nrps\n      3 nrps-t1pks\n     11 other\n      7 t1pks\n      1 t1pks-nrps\n      1 t3pks\n     12 terpene\nF.oxysporum_fsp_cepae - PG_ncbi\n      3 indole\n      1 indole-nrps\n     11 nrps\n      3 nrps-t1pks\n     10 other\n      5 t1pks\n      1 t1pks-indole\n      1 t3pks\n      9 terpene\n      1 terpene-t1pks\nF.oxysporum_fsp_narcissi - N139_ncbi\n      3 indole\n      1 indole-nrps\n     13 nrps\n      3 nrps-t1pks\n     10 other\n      6 t1pks\n      1 t1pks-terpene\n      1 t3pks\n      8 terpene\nF.proliferatum - A8_ncbi\n      1 hserlactone\n      3 indole\n     13 nrps\n      1 nrps-indole\n      6 nrps-t1pks\n      9 other\n     14 t1pks\n      1 t1pks-terpene\n      1 t3pks\n      9 terpene\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - 125_ncbi\n      3 indole\n     11 nrps\n      1 nrps-indole\n      4 nrps-t1pks\n     10 other\n      6 t1pks\n      1 t3pks\n     10 terpene\nF.oxysporum_fsp_cepae - A13_ncbi\n      3 indole\n     12 nrps\n      3 nrps-t1pks\n     10 other\n      9 t1pks\n      1 t1pks-nrps\n      1 t3pks\n     10 terpene\nF.oxysporum_fsp_cepae - A23_ncbi\n      3 indole\n     11 nrps\n      1 nrps-indole\n      4 nrps-t1pks\n     10 other\n      6 t1pks\n      1 t3pks\n     10 terpene\nF.oxysporum_fsp_cepae - A28_ncbi\n      3 indole\n      1 indole-nrps\n     11 nrps\n      2 nrps-t1pks\n     10 other\n      8 t1pks\n      1 t1pks-nrps\n      1 t3pks\n      9 terpene\nF.oxysporum_fsp_cepae - CB3_ncbi\n      1 bacteriocin\n      3 indole\n      1 indole-nrps\n     11 nrps\n      3 nrps-t1pks\n      9 other\n      6 t1pks\n      1 t1pks-indole\n      1 t3pks\n      8 terpene\nF.oxysporum_fsp_cepae - Fus2_canu_new\n      3 indole\n      1 indole-nrps\n     11 nrps\n      3 nrps-t1pks\n     11 other\n      7 t1pks\n      1 t1pks-nrps\n      1 t3pks\n     12 terpene\nF.oxysporum_fsp_cepae - PG_ncbi\n      3 indole\n      1 indole-nrps\n     11 nrps\n      3 nrps-t1pks\n     10 other\n      5 t1pks\n      1 t1pks-indole\n      1 t3pks\n      9 terpene\n      1 terpene-t1pks\nF.oxysporum_fsp_narcissi - N139_ncbi\n      3 indole\n      1 indole-nrps\n     13 nrps\n      3 nrps-t1pks\n     10 other\n      6 t1pks\n      1 t1pks-terpene\n      1 t3pks\n      8 terpene\nF.proliferatum - A8_ncbi\n      1 hserlactone\n      3 indole\n     13 nrps\n      1 nrps-indole\n      6 nrps-t1pks\n      9 other\n     14 t1pks\n      1 t1pks-terpene\n      1 t3pks\n      9 terpene\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eE) SSCP\u003c/h2\u003e\u003ca id=\"user-content-e-sscp\" class=\"anchor\" aria-label=\"Permalink: E) SSCP\" href=\"#e-sscp\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSmall secreted cysteine rich proteins were identified within secretomes. These\nproteins may be identified by EffectorP, but this approach allows direct control\nover what constitutes a SSCP.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nfor Secretome in $(ls gene_pred/final_genes_signalp-4.1/*/*/*_final_sp_no_trans_mem.aa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi'); do\nStrain=$(echo $Secretome| rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $Secretome | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nOutDir=analysis/sscp/$Organism/$Strain\nmkdir -p $OutDir\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/sscp\n$ProgDir/sscp_filter.py --inp_fasta $Secretome --max_length 300 --threshold 3 --out_fasta $OutDir/\u0026quot;$Strain\u0026quot;_sscp_all_results.fa\ncat $OutDir/\u0026quot;$Strain\u0026quot;_sscp_all_results.fa | grep 'Yes' \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_sscp.fa\nprintf \u0026quot;number of SSC-rich genes:\\t\u0026quot;\ncat $OutDir/\u0026quot;$Strain\u0026quot;_sscp.fa | grep '\u0026gt;' | tr -d '\u0026gt;' | cut -f1 -d '.' | sort | uniq | wc -l\nprintf \u0026quot;Number of effectors predicted by EffectorP:\\t\u0026quot;\nEffectorP=$(ls analysis/effectorP/$Organism/$Strain/*_EffectorP_secreted_headers.txt)\ncat $EffectorP | wc -l\nprintf \u0026quot;Number of SSCPs predicted by both effectorP and this approach: \\t\u0026quot;\ncat $OutDir/\u0026quot;$Strain\u0026quot;_sscp.fa | grep '\u0026gt;' | tr -d '\u0026gt;' \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_sscp_headers.txt\ncat $OutDir/\u0026quot;$Strain\u0026quot;_sscp_headers.txt $EffectorP | cut -f1 | sort | uniq -d | wc -l\necho \u0026quot;\u0026quot;\ndone \u0026gt; tmp.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSecretome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes_signalp-4.1/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_sp_no_trans_mem.aa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Secretome\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Secretome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutDir=analysis/sscp/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/sscp\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/sscp_filter.py --inp_fasta \u003cspan class=\"pl-smi\"\u003e$Secretome\u003c/span\u003e --max_length 300 --threshold 3 --out_fasta \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_sscp_all_results.fa\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_sscp_all_results.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eYes\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_sscp.fa\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enumber of SSC-rich genes:\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_sscp.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of effectors predicted by EffectorP:\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nEffectorP=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/effectorP/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_EffectorP_secreted_headers.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$EffectorP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of SSCPs predicted by both effectorP and this approach: \\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_sscp.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_sscp_headers.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_sscp_headers.txt \u003cspan class=\"pl-smi\"\u003e$EffectorP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -d \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - 125_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        289\nnumber of SSC-rich genes:       287\nNumber of effectors predicted by EffectorP:     357\nNumber of SSCPs predicted by both effectorP and this approach:  221\n\nF.oxysporum_fsp_cepae - A13_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        318\nnumber of SSC-rich genes:       316\nNumber of effectors predicted by EffectorP:     364\nNumber of SSCPs predicted by both effectorP and this approach:  237\n\nF.oxysporum_fsp_cepae - A23_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        286\nnumber of SSC-rich genes:       284\nNumber of effectors predicted by EffectorP:     355\nNumber of SSCPs predicted by both effectorP and this approach:  218\n\nF.oxysporum_fsp_cepae - A28_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        283\nnumber of SSC-rich genes:       281\nNumber of effectors predicted by EffectorP:     346\nNumber of SSCPs predicted by both effectorP and this approach:  214\n\nF.oxysporum_fsp_cepae - CB3_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        279\nnumber of SSC-rich genes:       277\nNumber of effectors predicted by EffectorP:     357\nNumber of SSCPs predicted by both effectorP and this approach:  213\nF.oxysporum_fsp_cepae - Fus2_canu_new\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        288\nnumber of SSC-rich genes:       286\nNumber of effectors predicted by EffectorP:     355\nNumber of SSCPs predicted by both effectorP and this approach:  218\n\nF.oxysporum_fsp_cepae - PG_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        287\nnumber of SSC-rich genes:       285\nNumber of effectors predicted by EffectorP:     337\nNumber of SSCPs predicted by both effectorP and this approach:  214\n\nF.oxysporum_fsp_narcissi - N139_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        344\nnumber of SSC-rich genes:       342\nNumber of effectors predicted by EffectorP:     407\nNumber of SSCPs predicted by both effectorP and this approach:  263\n\nF.proliferatum - A8_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        197\nnumber of SSC-rich genes:       197\nNumber of effectors predicted by EffectorP:     258\nNumber of SSCPs predicted by both effectorP and this approach:  147\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - 125_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        289\nnumber of SSC-rich genes:       287\nNumber of effectors predicted by EffectorP:     357\nNumber of SSCPs predicted by both effectorP and this approach:  221\n\nF.oxysporum_fsp_cepae - A13_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        318\nnumber of SSC-rich genes:       316\nNumber of effectors predicted by EffectorP:     364\nNumber of SSCPs predicted by both effectorP and this approach:  237\n\nF.oxysporum_fsp_cepae - A23_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        286\nnumber of SSC-rich genes:       284\nNumber of effectors predicted by EffectorP:     355\nNumber of SSCPs predicted by both effectorP and this approach:  218\n\nF.oxysporum_fsp_cepae - A28_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        283\nnumber of SSC-rich genes:       281\nNumber of effectors predicted by EffectorP:     346\nNumber of SSCPs predicted by both effectorP and this approach:  214\n\nF.oxysporum_fsp_cepae - CB3_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        279\nnumber of SSC-rich genes:       277\nNumber of effectors predicted by EffectorP:     357\nNumber of SSCPs predicted by both effectorP and this approach:  213\nF.oxysporum_fsp_cepae - Fus2_canu_new\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        288\nnumber of SSC-rich genes:       286\nNumber of effectors predicted by EffectorP:     355\nNumber of SSCPs predicted by both effectorP and this approach:  218\n\nF.oxysporum_fsp_cepae - PG_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        287\nnumber of SSC-rich genes:       285\nNumber of effectors predicted by EffectorP:     337\nNumber of SSCPs predicted by both effectorP and this approach:  214\n\nF.oxysporum_fsp_narcissi - N139_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        344\nnumber of SSC-rich genes:       342\nNumber of effectors predicted by EffectorP:     407\nNumber of SSCPs predicted by both effectorP and this approach:  263\n\nF.proliferatum - A8_ncbi\n% cysteine content threshold set to:    3\nmaximum length set to:  300\nNo. short-cysteine rich proteins in input fasta:        197\nnumber of SSC-rich genes:       197\nNumber of effectors predicted by EffectorP:     258\nNumber of SSCPs predicted by both effectorP and this approach:  147\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e#Genomic analysis\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eComparison to FoL 4287\u003c/h2\u003e\u003ca id=\"user-content-comparison-to-fol-4287\" class=\"anchor\" aria-label=\"Permalink: Comparison to FoL 4287\" href=\"#comparison-to-fol-4287\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBLast searches were used to identify which genes had homologs on which\nchromosomes of the Fusarium lycopersici genome.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"FoLGenomeFa=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum_chromosome_and_additional_contigs.fa\nfor Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'ncbi'); do\n# for Proteome in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_proteins.fasta); do\n# for Proteome in $(ls gene_pred/external_group/F.oxysporum_fsp_lycopersici/4287/Fusox1/Fusox1_GeneCatalog_proteins_20110522_parsed.fa); do\nStrain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nOutDir=analysis/blast_homology/$Organism/$Strain\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub $ProgDir/run_blast2csv.sh $Proteome protein $FoLGenomeFa $OutDir\ndone\"\u003e\u003cpre\u003eFoLGenomeFa=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum_chromosome_and_additional_contigs.fa\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eProteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Proteome in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_proteins.fasta); do\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e for Proteome in $(ls gene_pred/external_group/F.oxysporum_fsp_lycopersici/4287/Fusox1/Fusox1_GeneCatalog_proteins_20110522_parsed.fa); do\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutDir=analysis/blast_homology/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_blast2csv.sh \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e protein \u003cspan class=\"pl-smi\"\u003e$FoLGenomeFa\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eConvert top blast hits into gff annotations\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for BlastHitsCsv in $(ls analysis/blast_homology/F.*/*/4287_chromosomal_final_genes_combined.pep.fasta_hits.csv | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'ncbi'); do\nOrganism=$(echo $BlastHitsCsv | rev | cut -f3 -d '/' | rev)\nStrain=$(echo $BlastHitsCsv | rev | cut -f2 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nHitsGff=$(echo $BlastHitsCsv | sed  's/.csv/.gff/g')\nColumn2=\u0026quot;$Strain\u0026quot;_gene_homolog\nNumHits=1\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\n$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHitsCsv \u0026gt; $HitsGff\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBlastHitsCsv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/blast_homology/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/4287_chromosomal_final_genes_combined.pep.fasta_hits.csv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHitsCsv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHitsCsv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nHitsGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHitsCsv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.csv/.gff/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nColumn2=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_gene_homolog\nNumHits=1\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/blast2gff.pl \u003cspan class=\"pl-smi\"\u003e$Column2\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$NumHits\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$BlastHitsCsv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIntersecting blast hits with genes from FoL\u003c/h4\u003e\u003ca id=\"user-content-intersecting-blast-hits-with-genes-from-fol\" class=\"anchor\" aria-label=\"Permalink: Intersecting blast hits with genes from FoL\" href=\"#intersecting-blast-hits-with-genes-from-fol\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for HitsGff in $(ls analysis/blast_homology/F.*/*/4287_chromosomal_final_genes_combined.pep.fasta_hits.gff | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'ncbi'); do\nOrganism=$(echo $HitsGff | rev | cut -f3 -d '/' | rev)\nStrain=$(echo $HitsGff| rev | cut -f2 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nHitsDir=$(dirname $HitsGff)\nFoLGenes=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum.FO2.31.chr.gff3\nFoLIntersect=$HitsDir/4287_chromosomal_final_genes_combined_intersect.bed\nbedtools intersect -wo -a $HitsGff -b $FoLGenes \u0026gt; $FoLIntersect\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eHitsGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/blast_homology/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/4287_chromosomal_final_genes_combined.pep.fasta_hits.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nHitsDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nFoLGenes=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum.FO2.31.chr.gff3\nFoLIntersect=\u003cspan class=\"pl-smi\"\u003e$HitsDir\u003c/span\u003e/4287_chromosomal_final_genes_combined_intersect.bed\nbedtools intersect -wo -a \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$FoLGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$FoLIntersect\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eEffector genes\u003c/h2\u003e\u003ca id=\"user-content-effector-genes\" class=\"anchor\" aria-label=\"Permalink: Effector genes\" href=\"#effector-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePutative pathogenicity and effector related genes were identified within Braker\ngene models using a number of approaches:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eA) From Augustus gene models - Identifying secreted proteins\u003c/li\u003e\n\u003cli\u003eB) From Augustus gene models - Effector identification using EffectorP\u003c/li\u003e\n\u003cli\u003eD) From ORF fragments - Signal peptide \u0026amp; RxLR motif\u003c/li\u003e\n\u003cli\u003eE) From ORF fragments - Hmm evidence of WY domains\u003c/li\u003e\n\u003cli\u003eF) From ORF fragments - Hmm evidence of RxLR effectors\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA) From Augustus gene models - Identifying secreted proteins\u003c/h3\u003e\u003ca id=\"user-content-a-from-augustus-gene-models---identifying-secreted-proteins\" class=\"anchor\" aria-label=\"Permalink: A) From Augustus gene models - Identifying secreted proteins\" href=\"#a-from-augustus-gene-models---identifying-secreted-proteins\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRequired programs:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSignalP-4.1\u003c/li\u003e\n\u003cli\u003eTMHMM\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eProteins that were predicted to contain signal peptides were identified using\nthe following commands:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"SplitfileDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides\nCurPath=$PWD\nfor Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do\nStrain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)\nSplitDir=gene_pred/final_genes_split/$Organism/$Strain\nmkdir -p $SplitDir\nBaseName=\u0026quot;$Organism\u0026quot;\u0026quot;_$Strain\u0026quot;_final_preds\n$SplitfileDir/splitfile_500.py --inp_fasta $Proteome --out_dir $SplitDir --out_base $BaseName\nfor File in $(ls $SplitDir/*_final_preds_*); do\nJobs=$(qstat | grep 'pred_sigP' | wc -l)\nwhile [ $Jobs -gt 20 ]; do\nsleep 10\nprintf \u0026quot;.\u0026quot;\nJobs=$(qstat | grep 'pred_sigP' | wc -l)\ndone\nprintf \u0026quot;\\n\u0026quot;\necho $File\nqsub $ProgDir/pred_sigP.sh $File signalp-4.1\ndone\ndone\"\u003e\u003cpre\u003eSplitfileDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides\nCurPath=\u003cspan class=\"pl-smi\"\u003e$PWD\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eProteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nSplitDir=gene_pred/final_genes_split/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$SplitDir\u003c/span\u003e\nBaseName=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_preds\n\u003cspan class=\"pl-smi\"\u003e$SplitfileDir\u003c/span\u003e/splitfile_500.py --inp_fasta \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e --out_dir \u003cspan class=\"pl-smi\"\u003e$SplitDir\u003c/span\u003e --out_base \u003cspan class=\"pl-smi\"\u003e$BaseName\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els \u003cspan class=\"pl-smi\"\u003e$SplitDir\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_preds_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003epred_sigP\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003ewhile\u003c/span\u003e [ \u003cspan class=\"pl-smi\"\u003e$Jobs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-gt\u003c/span\u003e 20 ]\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nsleep 10\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nJobs=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eqstat \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003epred_sigP\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/pred_sigP.sh \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e signalp-4.1\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe batch files of predicted secreted proteins needed to be combined into a\nsingle file for each strain. This was done with the following commands:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for SplitDir in $(ls -d gene_pred/final_genes_split/*/* | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do\nStrain=$(echo $SplitDir | rev |cut -d '/' -f1 | rev)\nOrganism=$(echo $SplitDir | rev |cut -d '/' -f2 | rev)\nInStringAA=''\nInStringNeg=''\nInStringTab=''\nInStringTxt=''\nSigpDir=final_genes_signalp-4.1\nfor GRP in $(ls -l $SplitDir/*_final_preds_*.fa | rev | cut -d '_' -f1 | rev | sort -n); do  \nInStringAA=\u0026quot;$InStringAA gene_pred/$SigpDir/$Organism/$Strain/split/\u0026quot;$Organism\u0026quot;_\u0026quot;$Strain\u0026quot;_final_preds_$GRP\u0026quot;\u0026quot;_sp.aa\u0026quot;;  \nInStringNeg=\u0026quot;$InStringNeg gene_pred/$SigpDir/$Organism/$Strain/split/\u0026quot;$Organism\u0026quot;_\u0026quot;$Strain\u0026quot;_final_preds_$GRP\u0026quot;\u0026quot;_sp_neg.aa\u0026quot;;  \nInStringTab=\u0026quot;$InStringTab gene_pred/$SigpDir/$Organism/$Strain/split/\u0026quot;$Organism\u0026quot;_\u0026quot;$Strain\u0026quot;_final_preds_$GRP\u0026quot;\u0026quot;_sp.tab\u0026quot;;\nInStringTxt=\u0026quot;$InStringTxt gene_pred/$SigpDir/$Organism/$Strain/split/\u0026quot;$Organism\u0026quot;_\u0026quot;$Strain\u0026quot;_final_preds_$GRP\u0026quot;\u0026quot;_sp.txt\u0026quot;;  \ndone\ncat $InStringAA \u0026gt; gene_pred/$SigpDir/$Organism/$Strain/\u0026quot;$Strain\u0026quot;_final_sp.aa\ncat $InStringNeg \u0026gt; gene_pred/$SigpDir/$Organism/$Strain/\u0026quot;$Strain\u0026quot;_final_neg_sp.aa\ntail -n +2 -q $InStringTab \u0026gt; gene_pred/$SigpDir/$Organism/$Strain/\u0026quot;$Strain\u0026quot;_final_sp.tab\ncat $InStringTxt \u0026gt; gene_pred/$SigpDir/$Organism/$Strain/\u0026quot;$Strain\u0026quot;_final_sp.txt\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eSplitDir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -d gene_pred/final_genes_split/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$SplitDir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ecut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$SplitDir\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ecut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nInStringAA=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nInStringNeg=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nInStringTab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nInStringTxt=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nSigpDir=final_genes_signalp-4.1\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eGRP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els -l \u003cspan class=\"pl-smi\"\u003e$SplitDir\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_preds_\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -n\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e  \nInStringAA=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$InStringAA\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/split/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_final_preds_\u003cspan class=\"pl-smi\"\u003e$GRP\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_sp.aa\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e  \nInStringNeg=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$InStringNeg\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/split/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_final_preds_\u003cspan class=\"pl-smi\"\u003e$GRP\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_sp_neg.aa\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e  \nInStringTab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$InStringTab\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/split/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_final_preds_\u003cspan class=\"pl-smi\"\u003e$GRP\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_sp.tab\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\nInStringTxt=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$InStringTxt\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/split/\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_final_preds_\u003cspan class=\"pl-smi\"\u003e$GRP\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_sp.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e  \n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$InStringAA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_sp.aa\ncat \u003cspan class=\"pl-smi\"\u003e$InStringNeg\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_neg_sp.aa\ntail -n +2 -q \u003cspan class=\"pl-smi\"\u003e$InStringTab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_sp.tab\ncat \u003cspan class=\"pl-smi\"\u003e$InStringTxt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e gene_pred/\u003cspan class=\"pl-smi\"\u003e$SigpDir\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_sp.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSome proteins that are incorporated into the cell membrane require secretion.\nTherefore proteins with a transmembrane domain are not likely to represent\ncytoplasmic or apoplastic effectors.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eProteins containing a transmembrane domain were identified:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do\nStrain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/transmembrane_helices\nqsub $ProgDir/submit_TMHMM.sh $Proteome\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eProteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/transmembrane_helices\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/submit_TMHMM.sh \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThose proteins with transmembrane domains were removed from lists of Signal\npeptide containing proteins\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls gene_pred/trans_mem/*/*/*_TM_genes_neg.txt | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi' ); do\nStrain=$(echo $File | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $File | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nTmHeaders=$(echo \u0026quot;$File\u0026quot; | sed 's/neg.txt/neg_headers.txt/g')\ncat $File | cut -f1 \u0026gt; $TmHeaders\nSigP=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp.aa)\nOutDir=$(dirname $SigP)\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n$ProgDir/extract_from_fasta.py --fasta $SigP --headers $TmHeaders \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_final_sp_no_trans_mem.aa\necho \u0026quot;Number of SigP proteins:\u0026quot;\ncat $SigP | grep '\u0026gt;' | wc -l\necho \u0026quot;Number without transmembrane domains:\u0026quot;\ncat $OutDir/\u0026quot;$Strain\u0026quot;_final_sp_no_trans_mem.aa | grep '\u0026gt;' | wc -l\necho \u0026quot;Number of gene models:\u0026quot;\ncat $OutDir/\u0026quot;$Strain\u0026quot;_final_sp_no_trans_mem.aa | grep '\u0026gt;' | cut -f1 -d't' | sort | uniq |wc -l\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/trans_mem/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_TM_genes_neg.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nTmHeaders=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/neg.txt/neg_headers.txt/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TmHeaders\u003c/span\u003e\nSigP=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes_signalp-4.1/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_sp.aa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$SigP\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/extract_from_fasta.py --fasta \u003cspan class=\"pl-smi\"\u003e$SigP\u003c/span\u003e --headers \u003cspan class=\"pl-smi\"\u003e$TmHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_sp_no_trans_mem.aa\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of SigP proteins:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$SigP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber without transmembrane domains:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_sp_no_trans_mem.aa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eNumber of gene models:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_final_sp_no_trans_mem.aa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003et\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ewc -l\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - Fus2_canu_new\nNumber of SigP proteins:\n1793\nNumber without transmembrane domains:\n1475\nNumber of gene models:\n1449\nF.oxysporum_fsp_cepae - A13_ncbi\nNumber of SigP proteins:\n1826\nNumber without transmembrane domains:\n1508\nF.oxysporum_fsp_cepae - A23_ncbi\nNumber of SigP proteins:\n1752\nNumber without transmembrane domains:\n1439\nF.oxysporum_fsp_cepae - A28_ncbi\nNumber of SigP proteins:\n1772\nNumber without transmembrane domains:\n1450\nF.oxysporum_fsp_cepae - CB3_ncbi\nNumber of SigP proteins:\n1763\nNumber without transmembrane domains:\n1439\nF.oxysporum_fsp_cepae - PG_ncbi\nNumber of SigP proteins:\n1760\nNumber without transmembrane domains:\n1440\nF.oxysporum_fsp_narcissi - N139_ncbi\nNumber of SigP proteins:\n1901\nNumber without transmembrane domains:\n1566\nF.proliferatum - A8_ncbi\nNumber of SigP proteins:\n1553\nNumber without transmembrane domains:\n1254\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - Fus2_canu_new\nNumber of SigP proteins:\n1793\nNumber without transmembrane domains:\n1475\nNumber of gene models:\n1449\nF.oxysporum_fsp_cepae - A13_ncbi\nNumber of SigP proteins:\n1826\nNumber without transmembrane domains:\n1508\nF.oxysporum_fsp_cepae - A23_ncbi\nNumber of SigP proteins:\n1752\nNumber without transmembrane domains:\n1439\nF.oxysporum_fsp_cepae - A28_ncbi\nNumber of SigP proteins:\n1772\nNumber without transmembrane domains:\n1450\nF.oxysporum_fsp_cepae - CB3_ncbi\nNumber of SigP proteins:\n1763\nNumber without transmembrane domains:\n1439\nF.oxysporum_fsp_cepae - PG_ncbi\nNumber of SigP proteins:\n1760\nNumber without transmembrane domains:\n1440\nF.oxysporum_fsp_narcissi - N139_ncbi\nNumber of SigP proteins:\n1901\nNumber without transmembrane domains:\n1566\nF.proliferatum - A8_ncbi\nNumber of SigP proteins:\n1553\nNumber without transmembrane domains:\n1254\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eB) From Augustus gene models - Effector identification using EffectorP\u003c/h3\u003e\u003ca id=\"user-content-b-from-augustus-gene-models---effector-identification-using-effectorp\" class=\"anchor\" aria-label=\"Permalink: B) From Augustus gene models - Effector identification using EffectorP\" href=\"#b-from-augustus-gene-models---effector-identification-using-effectorp\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRequired programs:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eEffectorP.py\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'Fus2_canu_new'); do\nStrain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)\nOrganism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)\nBaseName=\u0026quot;$Organism\u0026quot;_\u0026quot;$Strain\u0026quot;_EffectorP\nOutDir=analysis/effectorP/$Organism/$Strain\nProgDir=~/git_repos/emr_repos/tools/seq_tools/feature_annotation/fungal_effectors\nqsub $ProgDir/pred_effectorP.sh $Proteome $BaseName $OutDir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eProteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_combined.pep.fasta \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nBaseName=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_EffectorP\nOutDir=analysis/effectorP/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nProgDir=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/git_repos/emr_repos/tools/seq_tools/feature_annotation/fungal_effectors\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/pred_effectorP.sh \u003cspan class=\"pl-smi\"\u003e$Proteome\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$BaseName\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThose genes that were predicted as secreted and tested positive by effectorP\nwere identified:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls analysis/effectorP/*/*/*_EffectorP.txt | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do\nStrain=$(echo $File | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $File | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nHeaders=$(echo \u0026quot;$File\u0026quot; | sed 's/_EffectorP.txt/_EffectorP_headers.txt/g')\ncat $File | grep 'Effector' | cut -f1 \u0026gt; $Headers\nSecretome=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp_no_trans_mem.aa)\nOutFile=$(echo \u0026quot;$File\u0026quot; | sed 's/_EffectorP.txt/_EffectorP_secreted.aa/g')\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n$ProgDir/extract_from_fasta.py --fasta $Secretome --headers $Headers \u0026gt; $OutFile\nOutFileHeaders=$(echo \u0026quot;$File\u0026quot; | sed 's/_EffectorP.txt/_EffectorP_secreted_headers.txt/g')\ncat $OutFile | grep '\u0026gt;' | tr -d '\u0026gt;' \u0026gt; $OutFileHeaders\ncat $OutFileHeaders | wc -l\nGff=$(ls gene_pred/final_genes/$Organism/$Strain/*/final_genes_appended.gff3)\nEffectorP_Gff=$(echo \u0026quot;$File\u0026quot; | sed 's/_EffectorP.txt/_EffectorP_secreted.gff/g')\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n$ProgDir/extract_gff_for_sigP_hits.pl $OutFileHeaders $Gff effectorP ID \u0026gt; $EffectorP_Gff\ncat $EffectorP_Gff | grep -w 'gene' | wc -l\ndone \u0026gt; tmp.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/effectorP/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_EffectorP.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nHeaders=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_EffectorP.txt/_EffectorP_headers.txt/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eEffector\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Headers\u003c/span\u003e\nSecretome=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes_signalp-4.1/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_sp_no_trans_mem.aa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutFile=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_EffectorP.txt/_EffectorP_secreted.aa/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/extract_from_fasta.py --fasta \u003cspan class=\"pl-smi\"\u003e$Secretome\u003c/span\u003e --headers \u003cspan class=\"pl-smi\"\u003e$Headers\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e\nOutFileHeaders=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_EffectorP.txt/_EffectorP_secreted_headers.txt/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutFileHeaders\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutFileHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\nGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final_genes_appended.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nEffectorP_Gff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_EffectorP.txt/_EffectorP_secreted.gff/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/extract_gff_for_sigP_hits.pl \u003cspan class=\"pl-smi\"\u003e$OutFileHeaders\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Gff\u003c/span\u003e effectorP ID \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$EffectorP_Gff\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$EffectorP_Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - Fus2_canu_new\n355\n355\nF.oxysporum_fsp_cepae - A13_ncbi\n364\n364\nF.oxysporum_fsp_cepae - A23_ncbi\n355\n355\nF.oxysporum_fsp_cepae - A28_ncbi\n346\n346\nF.oxysporum_fsp_cepae - CB3_ncbi\n357\n357\nF.oxysporum_fsp_cepae - PG_ncbi\n337\n337\nF.oxysporum_fsp_narcissi - N139_ncbi\n407\n407\nF.proliferatum - A8_ncbi\n258\n258\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - Fus2_canu_new\n355\n355\nF.oxysporum_fsp_cepae - A13_ncbi\n364\n364\nF.oxysporum_fsp_cepae - A23_ncbi\n355\n355\nF.oxysporum_fsp_cepae - A28_ncbi\n346\n346\nF.oxysporum_fsp_cepae - CB3_ncbi\n357\n357\nF.oxysporum_fsp_cepae - PG_ncbi\n337\n337\nF.oxysporum_fsp_narcissi - N139_ncbi\n407\n407\nF.proliferatum - A8_ncbi\n258\n258\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC) Identification of MIMP-flanking genes\u003c/h3\u003e\u003ca id=\"user-content-c-identification-of-mimp-flanking-genes\" class=\"anchor\" aria-label=\"Permalink: C) Identification of MIMP-flanking genes\" href=\"#c-identification-of-mimp-flanking-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'old' | grep -v 'ncbi' ); do\nOrganism=$(echo \u0026quot;$Genome\u0026quot; | rev | cut -d '/' -f4 | rev)\nStrain=$(echo \u0026quot;$Genome\u0026quot; | rev | cut -d '/' -f3 | rev)\nBrakerGff=$(ls gene_pred/final_genes/$Organism/\u0026quot;$Strain\u0026quot;/final/final_genes_CodingQuary.gff3)\nQuaryGff=$(ls gene_pred/final_genes/$Organism/\u0026quot;$Strain\u0026quot;/final/final_genes_Braker.gff3)\nOutDir=analysis/mimps/$Organism/$Strain\nmkdir -p \u0026quot;$OutDir\u0026quot;\necho \u0026quot;$Organism - $Strain\u0026quot;\nProgDir=\u0026quot;/home/armita/git_repos/emr_repos/tools/pathogen/mimp_finder\u0026quot;\n$ProgDir/mimp_finder.pl $Genome $OutDir/\u0026quot;$Strain\u0026quot;_mimps.fa $OutDir/\u0026quot;$Strain\u0026quot;_mimps.gff \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_mimps.log\n$ProgDir/gffexpander.pl +- 2000 $OutDir/\u0026quot;$Strain\u0026quot;_mimps.gff \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_mimps_exp.gff\necho \u0026quot;The number of mimps identified:\u0026quot;\ncat $OutDir/\u0026quot;$Strain\u0026quot;_mimps.fa | grep '\u0026gt;' | wc -l\nbedtools intersect -u -a $BrakerGff -b $OutDir/\u0026quot;$Strain\u0026quot;_mimps_exp.gff \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_genes_in_2kb_mimp.gff\nbedtools intersect -u -a $QuaryGff -b $OutDir/\u0026quot;$Strain\u0026quot;_mimps_exp.gff \u0026gt;\u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_genes_in_2kb_mimp.gff\necho \u0026quot;The following transcripts intersect mimps:\u0026quot;\nMimpProtsTxt=$OutDir/\u0026quot;$Strain\u0026quot;_prots_in_2kb_mimp.txt\nMimpGenesTxt=$OutDir/\u0026quot;$Strain\u0026quot;_genes_in_2kb_mimp.txt\ncat $OutDir/\u0026quot;$Strain\u0026quot;_genes_in_2kb_mimp.gff | grep -w 'mRNA' | cut -f9 | cut -f1 -d';' | cut -f2 -d'=' | sort | uniq \u0026gt; $MimpProtsTxt\ncat $OutDir/\u0026quot;$Strain\u0026quot;_genes_in_2kb_mimp.gff | grep -w 'mRNA' | cut -f9 | cut -f1 -d';' | cut -f2 -d'=' | cut -f1 -d '.'| sort | uniq \u0026gt; $MimpGenesTxt\ncat $MimpProtsTxt | wc -l\ncat $MimpGenesTxt | wc -l\necho \u0026quot;\u0026quot;\ndone \u0026gt; tmp.txt\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eGenome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Genome\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Genome\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -f3 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nBrakerGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e/final/final_genes_CodingQuary.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nQuaryGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e/final/final_genes_Braker.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=analysis/mimps/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nProgDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/home/armita/git_repos/emr_repos/tools/pathogen/mimp_finder\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/mimp_finder.pl \u003cspan class=\"pl-smi\"\u003e$Genome\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps.fa \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps.gff \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps.log\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/gffexpander.pl +- 2000 \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps.gff \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps_exp.gff\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eThe number of mimps identified:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\nbedtools intersect -u -a \u003cspan class=\"pl-smi\"\u003e$BrakerGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps_exp.gff \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_genes_in_2kb_mimp.gff\nbedtools intersect -u -a \u003cspan class=\"pl-smi\"\u003e$QuaryGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_mimps_exp.gff \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_genes_in_2kb_mimp.gff\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eThe following transcripts intersect mimps:\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nMimpProtsTxt=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_prots_in_2kb_mimp.txt\nMimpGenesTxt=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_genes_in_2kb_mimp.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_genes_in_2kb_mimp.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f9 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$MimpProtsTxt\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_genes_in_2kb_mimp.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f9 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$MimpGenesTxt\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$MimpProtsTxt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\ncat \u003cspan class=\"pl-smi\"\u003e$MimpGenesTxt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\tF.oxysporum_fsp_cepae - Fus2_canu_new\n\tThe number of mimps identified:\n\t153\n\tThe following transcripts intersect mimps:\n\t157\n\t155\n\n\tF.oxysporum_fsp_cepae - 125_ncbi\n\tThe number of mimps identified:\n\t140\n\tThe following transcripts intersect mimps:\n\t95\n\t95\n\n\tF.oxysporum_fsp_cepae - A13_ncbi\n\tThe number of mimps identified:\n\t55\n\tThe following transcripts intersect mimps:\n\t49\n\t49\n\n\tF.oxysporum_fsp_cepae - A23_ncbi\n\tThe number of mimps identified:\n\t136\n\tThe following transcripts intersect mimps:\n\t88\n\t88\n\n\tF.oxysporum_fsp_cepae - A28_ncbi\n\tThe number of mimps identified:\n\t35\n\tThe following transcripts intersect mimps:\n\t36\n\t36\n\n\tF.oxysporum_fsp_cepae - CB3_ncbi\n\tThe number of mimps identified:\n\t30\n\tThe following transcripts intersect mimps:\n\t32\n\t32\n\n\tF.oxysporum_fsp_cepae - PG_ncbi\n\tThe number of mimps identified:\n\t51\n\tThe following transcripts intersect mimps:\n\t43\n\t43\n\n\tF.oxysporum_fsp_narcissi - N139_ncbi\n\tThe number of mimps identified:\n\t207\n\tThe following transcripts intersect mimps:\n\t120\n\t119\n\n\tF.proliferatum - A8_ncbi\n\tThe number of mimps identified:\n\t6\n\tThe following transcripts intersect mimps:\n\t11\n\t11\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\tF.oxysporum_fsp_cepae - Fus2_canu_new\n\tThe number of mimps identified:\n\t153\n\tThe following transcripts intersect mimps:\n\t157\n\t155\n\n\tF.oxysporum_fsp_cepae - 125_ncbi\n\tThe number of mimps identified:\n\t140\n\tThe following transcripts intersect mimps:\n\t95\n\t95\n\n\tF.oxysporum_fsp_cepae - A13_ncbi\n\tThe number of mimps identified:\n\t55\n\tThe following transcripts intersect mimps:\n\t49\n\t49\n\n\tF.oxysporum_fsp_cepae - A23_ncbi\n\tThe number of mimps identified:\n\t136\n\tThe following transcripts intersect mimps:\n\t88\n\t88\n\n\tF.oxysporum_fsp_cepae - A28_ncbi\n\tThe number of mimps identified:\n\t35\n\tThe following transcripts intersect mimps:\n\t36\n\t36\n\n\tF.oxysporum_fsp_cepae - CB3_ncbi\n\tThe number of mimps identified:\n\t30\n\tThe following transcripts intersect mimps:\n\t32\n\t32\n\n\tF.oxysporum_fsp_cepae - PG_ncbi\n\tThe number of mimps identified:\n\t51\n\tThe following transcripts intersect mimps:\n\t43\n\t43\n\n\tF.oxysporum_fsp_narcissi - N139_ncbi\n\tThe number of mimps identified:\n\t207\n\tThe following transcripts intersect mimps:\n\t120\n\t119\n\n\tF.proliferatum - A8_ncbi\n\tThe number of mimps identified:\n\t6\n\tThe following transcripts intersect mimps:\n\t11\n\t11\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\n\u003cp dir=\"auto\"\u003eThose genes that were predicted as secreted and within 2Kb of a MIMP\nwere identified:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls analysis/mimps/*/*/*_genes_in_2kb_mimp.txt | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'old' | grep 'ncbi' ); do\nStrain=$(echo $File | rev | cut -f2 -d '/' | rev | sed 's/_chromosomal//g')\nOrganism=$(echo $File | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nProtsFile=$(echo $File | sed 's/genes/prots/g')\nSecretome=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp_no_trans_mem.aa)\nOutFile=$(echo \u0026quot;$File\u0026quot; | sed 's/.gff/_secreted.gff/g')\nSecretedHeaders=$(echo \u0026quot;$Secretome\u0026quot; | sed 's/.aa/_headers.txt/g')\ncat $Secretome | grep '\u0026gt;' | tr -d '\u0026gt;' | sed 's/-p.//g' \u0026gt; $SecretedHeaders\ncat $ProtsFile $SecretedHeaders | cut -f1 | sort | uniq -d | wc -l\ncat $SecretedHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $File | wc -l\n# ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\n# $ProgDir/extract_gff_for_sigP_hits.pl $SecretedHeaders $File secreted_mimp ID \u0026gt; $OutFile\n# cat $OutFile | grep -w 'mRNA' | wc -l\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/mimps/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_genes_in_2kb_mimp.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_chromosomal//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nProtsFile=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/genes/prots/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nSecretome=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes_signalp-4.1/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_sp_no_trans_mem.aa\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutFile=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.gff/_secreted.gff/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nSecretedHeaders=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Secretome\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.aa/_headers.txt/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$Secretome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/-p.//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$SecretedHeaders\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ProtsFile\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$SecretedHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -d \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\ncat \u003cspan class=\"pl-smi\"\u003e$SecretedHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e $ProgDir/extract_gff_for_sigP_hits.pl $SecretedHeaders $File secreted_mimp ID \u0026gt; $OutFile\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e cat $OutFile | grep -w 'mRNA' | wc -l\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - Fus2_canu_new\n31\n31\nF.oxysporum_fsp_cepae - 125_ncbi\n24\n26\nF.oxysporum_fsp_cepae - A13_ncbi\n9\n9\nF.oxysporum_fsp_cepae - A23_ncbi\n20\n20\nF.oxysporum_fsp_cepae - A28_ncbi\n5\n5\nF.oxysporum_fsp_cepae - CB3_ncbi\n1\n1\nF.oxysporum_fsp_cepae - PG_ncbi\n3\n3\nF.oxysporum_fsp_narcissi - N139_ncbi\n31\n31\nF.proliferatum - A8_ncbi\n1\n1\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - Fus2_canu_new\n31\n31\nF.oxysporum_fsp_cepae - 125_ncbi\n24\n26\nF.oxysporum_fsp_cepae - A13_ncbi\n9\n9\nF.oxysporum_fsp_cepae - A23_ncbi\n20\n20\nF.oxysporum_fsp_cepae - A28_ncbi\n5\n5\nF.oxysporum_fsp_cepae - CB3_ncbi\n1\n1\nF.oxysporum_fsp_cepae - PG_ncbi\n3\n3\nF.oxysporum_fsp_narcissi - N139_ncbi\n31\n31\nF.proliferatum - A8_ncbi\n1\n1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eC)\u003c/h2\u003e\u003ca id=\"user-content-c\" class=\"anchor\" aria-label=\"Permalink: C)\" href=\"#c\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSecreted proteins with a N-terminal [SG]-P-C-[KR]-P motif\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for File in $(ls gene_pred/final_genes_signalp-4.1/F.*/*/*_final_sp_no_trans_mem.aa | grep -e 'Fus2_canu_new' -e 'ncbi'); do\necho $File;\ncat $File | grep -B1 \u0026quot;[SG]PC[KR]P\u0026quot; | grep '\u0026gt;' | cut -f1 | tr -d '\u0026gt;' | wc -l;\ndone\ncat gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa | grep -B1 \u0026quot;[SG]PC[KR]P\u0026quot; | grep '\u0026gt;' | cut -f1 | tr -d '\u0026gt;' \u0026gt; tmp.txt\ncat analysis/expression/Fus2_expressed_genes.tsv | grep -w -f 'tmp.txt' \u0026gt; tmp.tsv\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eFile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes_signalp-4.1/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_final_sp_no_trans_mem.aa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$File\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -B1 \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e[SG]PC[KR]P\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\ncat gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -B1 \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e[SG]PC[KR]P\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.txt\ncat analysis/expression/Fus2_expressed_genes.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -f \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etmp.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.tsv\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/125_ncbi/125_ncbi_final_sp_no_trans_mem.aa\n43\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A13_ncbi/A13_ncbi_final_sp_no_trans_mem.aa\n50\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A23_ncbi/A23_ncbi_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A28_ncbi/A28_ncbi_final_sp_no_trans_mem.aa\n45\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/CB3_ncbi/CB3_ncbi_final_sp_no_trans_mem.aa\n46\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/PG_ncbi/PG_ncbi_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_narcissi/N139_ncbi/N139_ncbi_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.proliferatum/A8_ncbi/A8_ncbi_final_sp_no_trans_mem.aa\n40\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003egene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/125_ncbi/125_ncbi_final_sp_no_trans_mem.aa\n43\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A13_ncbi/A13_ncbi_final_sp_no_trans_mem.aa\n50\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A23_ncbi/A23_ncbi_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A28_ncbi/A28_ncbi_final_sp_no_trans_mem.aa\n45\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/CB3_ncbi/CB3_ncbi_final_sp_no_trans_mem.aa\n46\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/PG_ncbi/PG_ncbi_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_narcissi/N139_ncbi/N139_ncbi_final_sp_no_trans_mem.aa\n42\ngene_pred/final_genes_signalp-4.1/F.proliferatum/A8_ncbi/A8_ncbi_final_sp_no_trans_mem.aa\n40\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4. Genomic analysis\u003c/h1\u003e\u003ca id=\"user-content-4-genomic-analysis\" class=\"anchor\" aria-label=\"Permalink: 4. Genomic analysis\" href=\"#4-genomic-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.1 Chracterisation of LS regions\u003c/h2\u003e\u003ca id=\"user-content-41-chracterisation-of-ls-regions\" class=\"anchor\" aria-label=\"Permalink: 4.1 Chracterisation of LS regions\" href=\"#41-chracterisation-of-ls-regions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSee pathogen/identifying_LS_chromosomes/characterising_LS_contigs.md\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2 Orthology\u003c/h2\u003e\u003ca id=\"user-content-42-orthology\" class=\"anchor\" aria-label=\"Permalink: 4.2 Orthology\" href=\"#42-orthology\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOrthomcl was used to identify orthologous groups between Fusarium spp. genomes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eGenomes were grouped by subspecies and orthology was determined within each\nsubspecies group. Orthology was also determined between subspecies groups.\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003ePathogenic\u003c/th\u003e\n\u003cth\u003enon-pathogenic\u003c/th\u003e\n\u003cth\u003eIntermediate\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e125\u003c/td\u003e\n\u003ctd\u003eA28\u003c/td\u003e\n\u003ctd\u003e55\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eA23\u003c/td\u003e\n\u003ctd\u003eD2\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eFus2\u003c/td\u003e\n\u003ctd\u003ePG\u003c/td\u003e\n\u003ctd\u003e\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4.2.c) Orthology between pathogenic and non-pathogenic isolates\u003c/h3\u003e\u003ca id=\"user-content-42c-orthology-between-pathogenic-and-non-pathogenic-isolates\" class=\"anchor\" aria-label=\"Permalink: 4.2.c) Orthology between pathogenic and non-pathogenic isolates\" href=\"#42c-orthology-between-pathogenic-and-non-pathogenic-isolates\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe Commands used to run this analysis are shown in\npathogen/orthology/F.oxysporum_fsp.cepae_pathogen_vs_non-pathogen_orthology.md\u003c/p\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.  Searches\u003c/h2\u003e\u003ca id=\"user-content-5--searches\" class=\"anchor\" aria-label=\"Permalink: 5.  Searches\" href=\"#5--searches\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.1.A) Identifying SIX genes\u003c/h2\u003e\u003ca id=\"user-content-51a-identifying-six-genes\" class=\"anchor\" aria-label=\"Permalink: 5.1.A) Identifying SIX genes\" href=\"#51a-identifying-six-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eProtein sequence of previously characterised SIX genes used to BLAST against\nassemblies.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nfor Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -w -v 'Fus2_canu_new' | grep 'ncbi' | grep -v 'old'); do\necho $Assembly\nQuery=analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX.fa\nqsub $ProgDir/blast_pipe.sh $Query dna $Assembly\ndone\"\u003e\u003cpre\u003eProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\nQuery=analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX.fa\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/blast_pipe.sh \u003cspan class=\"pl-smi\"\u003e$Query\u003c/span\u003e dna \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOnce blast searches had completed, the BLAST hits were converted to GFF\nannotations:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for BlastHits in $(ls analysis/blast_homology/*/*/*Fo_path_genes_CRX.fa_homologs.csv | grep -v -w 'Fus2_canu_new' | grep 'ncbi'); do\nStrain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nHitsGff=$(echo $BlastHits | sed  's/.csv/.gff/g')\nColumn2=SIX_homolog\nNumHits=12\n$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHits \u0026gt; $HitsGff\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/blast_homology/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eFo_path_genes_CRX.fa_homologs.csv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nHitsGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.csv/.gff/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nColumn2=SIX_homolog\nNumHits=12\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/blast2gff.pl \u003cspan class=\"pl-smi\"\u003e$Column2\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$NumHits\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"The blast hits were summarised in a single table for all the genomes. The top\nidentity of the top blast hit in relation to the enquire query sequence was\npresented for each blast hit.\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eThe blast hits were summarised in a single table for all the genomes. The top\nidentity of the top blast hit in relation to the enquire query sequence was\npresented for each blast hit.\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"OutFile=analysis/blast_homology/Fo_path_genes_CRX_summary.tab\necho \u0026quot;Organism\u0026quot; \u0026gt; tmp2.tab\ncat analysis/blast_homology/F.proliferatum/A8/A8_Fo_path_genes_CRX.fa_homologs.csv | cut -f1 \u0026gt;\u0026gt; tmp2.tab\nfor BlastHits in $(ls analysis/blast_homology/*/*/*Fo_path_genes_CRX.fa_homologs.csv | grep -v -w -e 'Fus2' -e 'Fus2_edited_v1' -e 'Fus2_edited_v2' -e 'Fus2_canu' -e 'Fus2_3' -e '4287' | grep -e 'Fus2_canu_new' -e 'ncbi'); do\nStrain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism\u0026quot; \u0026gt; tmp.tab\necho \u0026quot;$Strain\u0026quot; \u0026gt;\u0026gt; tmp.tab\ncat $BlastHits | cut -f10,19 | tail -n +2 | sed -r \u0026quot;s/\\t/,/g\u0026quot; \u0026gt;\u0026gt; tmp.tab\npaste tmp2.tab tmp.tab \u0026gt; $OutFile\ncp $OutFile tmp2.tab\ndone\nrm tmp.tab\nrm tmp2.tab\"\u003e\u003cpre\u003eOutFile=analysis/blast_homology/Fo_path_genes_CRX_summary.tab\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrganism\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp2.tab\ncat analysis/blast_homology/F.proliferatum/A8/A8_Fo_path_genes_CRX.fa_homologs.csv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e tmp2.tab\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/blast_homology/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eFo_path_genes_CRX.fa_homologs.csv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_edited_v1\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_edited_v2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e4287\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e tmp.tab\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e tmp.tab\ncat \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f10,19 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n +2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed -r \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003es/\\t/,/g\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e tmp.tab\npaste tmp2.tab tmp.tab \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e\ncp \u003cspan class=\"pl-smi\"\u003e$OutFile\u003c/span\u003e tmp2.tab\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nrm tmp.tab\nrm tmp2.tab\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for HitsGff in $(ls analysis/blast_homology/*/*/*Fo_path_genes_CRX.fa_homologs.gff | grep -v 'trinity' | grep -e 'Fus2_canu_new' -e 'ncbi'); do\nStrain=$(echo $HitsGff | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $HitsGff | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nGffAppended=$(ls gene_pred/final_genes/$Organism/\u0026quot;$Strain\u0026quot;*/final/final_genes_appended.gff3)\nOutDir=$(dirname $HitsGff)\nSixIntersect=$OutDir/\u0026quot;$Strain\u0026quot;_Fo_path_genes_CRX.fa_hit_genes.bed\nbedtools intersect -wao -a $HitsGff -b $GffAppended \u0026gt; $SixIntersect\nbedtools intersect -wao -a $HitsGff -b $GffAppended | cut -f9,18 | grep -v 'Parent'\necho \u0026quot;\u0026quot;\ndone | tr -d ';' | tr -d '\u0026quot;' | sed 's/ID=//g'\u0026gt; analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX_hit_genes_summary.tab\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eHitsGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/blast_homology/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eFo_path_genes_CRX.fa_homologs.gff \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003etrinity\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nGffAppended=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/final/final_genes_appended.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003edirname \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nSixIntersect=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_Fo_path_genes_CRX.fa_hit_genes.bed\nbedtools intersect -wao -a \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$GffAppended\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$SixIntersect\u003c/span\u003e\nbedtools intersect -wao -a \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$GffAppended\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f9,18 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eParent\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\"\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/ID=//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX_hit_genes_summary.tab\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"F.oxysporum_fsp_cepae - Fus2_canu_new\nC5_Seq45_BlastHit_1     g13425\nCRX1_Seq49_BlastHit_1   g15739\nCRX1_Seq49_BlastHit_2   g11056\nCRX1_Seq49_BlastHit_3   g59\nCRX1_Seq49_BlastHit_4   g16971\nCRX2_Seq50_BlastHit_1   g11056\nCRX2_Seq50_BlastHit_2   g15739\nCRX2_Seq50_BlastHit_3   g59\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_1  g15724\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_2  g17045\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_3  CUFF_4289_1_31\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_4  CUFF_327_1_65\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_5  g12255\nMIMP_g29_ex.19GJ26_H12-1.seq_BlastHit_1 g16905\nMIMP_g16_ex.19GJ26_B07-1.seq_BlastHit_1 g16179\nMIMP_g41_ex.45DG31-1.seq_BlastHit_1     g17144\nMIMP_g41_ex.45DG31-1.seq_BlastHit_2     NS_02766\nFusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_1 g16847\nFusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_2 g16190\nFusarium_oxysporum_f._sp._lycopersici_isolate_BFOL-51_secreted_in_xylem_5_(SIX5)_gene,_partial_cds_BlastHit_1   g16849\nFusarium_oxysporum_f._sp._lycopersici_secreted_in_xylem_Six7_(SIX7)_mRNA,_complete_cds_BlastHit_1       CUFF_4344_1_33\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_1   g16273\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_2   g13574\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six10_(SIX10)_mRNA,_complete_cds_BlastHit_1 g17153\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six12_(SIX12)_mRNA,_complete_cds_BlastHit_1 PGN_CUFF_4345_1_2\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six14_(SIX14)_mRNA,_complete_cds_BlastHit_1 man_1\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eF.oxysporum_fsp_cepae - Fus2_canu_new\nC5_Seq45_BlastHit_1     g13425\nCRX1_Seq49_BlastHit_1   g15739\nCRX1_Seq49_BlastHit_2   g11056\nCRX1_Seq49_BlastHit_3   g59\nCRX1_Seq49_BlastHit_4   g16971\nCRX2_Seq50_BlastHit_1   g11056\nCRX2_Seq50_BlastHit_2   g15739\nCRX2_Seq50_BlastHit_3   g59\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_1  g15724\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_2  g17045\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_3  CUFF_4289_1_31\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_4  CUFF_327_1_65\nMIMP_39_ex.19GJ26_B10-1.seq_BlastHit_5  g12255\nMIMP_g29_ex.19GJ26_H12-1.seq_BlastHit_1 g16905\nMIMP_g16_ex.19GJ26_B07-1.seq_BlastHit_1 g16179\nMIMP_g41_ex.45DG31-1.seq_BlastHit_1     g17144\nMIMP_g41_ex.45DG31-1.seq_BlastHit_2     NS_02766\nFusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_1 g16847\nFusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_2 g16190\nFusarium_oxysporum_f._sp._lycopersici_isolate_BFOL-51_secreted_in_xylem_5_(SIX5)_gene,_partial_cds_BlastHit_1   g16849\nFusarium_oxysporum_f._sp._lycopersici_secreted_in_xylem_Six7_(SIX7)_mRNA,_complete_cds_BlastHit_1       CUFF_4344_1_33\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_1   g16273\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_2   g13574\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six10_(SIX10)_mRNA,_complete_cds_BlastHit_1 g17153\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six12_(SIX12)_mRNA,_complete_cds_BlastHit_1 PGN_CUFF_4345_1_2\nFusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six14_(SIX14)_mRNA,_complete_cds_BlastHit_1 man_1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.1.B) Identifying FTF genes\u003c/h2\u003e\u003ca id=\"user-content-51b-identifying-ftf-genes\" class=\"anchor\" aria-label=\"Permalink: 5.1.B) Identifying FTF genes\" href=\"#51b-identifying-ftf-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePreviously published FTF genes from Sanchez et al 2016 were blasted against\nFusarium genomes.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -v -e 'HB17' | grep -v -w 'Fus2_canu_new' | grep 'A8_ncbi'); do\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\necho $Assembly\nQuery=analysis/blast_homology/Fo_path_genes/FTF_cds_Sanchez_et_al_2016.fasta\nOutDir=analysis/FTF/$Organism/$Strain\nmkdir -p $OutDir\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub $ProgDir/run_blast2csv.sh $Query dna $Assembly $OutDir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8_ncbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\nQuery=analysis/blast_homology/Fo_path_genes/FTF_cds_Sanchez_et_al_2016.fasta\nOutDir=analysis/FTF/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_blast2csv.sh \u003cspan class=\"pl-smi\"\u003e$Query\u003c/span\u003e dna \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBLAST hits were converted to Gff annotations and intersected with gene models:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for BlastHits in $(ls analysis/FTF/*/*/*_FTF_cds_Sanchez_et_al_2016.fasta_hits.csv | grep -v -w 'Fus2_canu_new' | grep 'A8_ncbi'); do\nStrain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)\nOutDir=analysis/FTF/$Organism/$Strain\nHitsGff=$(echo $BlastHits | sed  's/.csv/.gff/g')\nColumn2=FTF_homolog\nNumHits=1\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\n$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHits \u0026gt; $HitsGff\n\nGffAppended=$(ls gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3)\nbedtools intersect -wao -a $HitsGff -b $GffAppended \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_FTF_hits_intersected.bed\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/FTF/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_FTF_cds_Sanchez_et_al_2016.fasta_hits.csv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8_ncbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOutDir=analysis/FTF/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nHitsGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.csv/.gff/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nColumn2=FTF_homolog\nNumHits=1\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/blast2gff.pl \u003cspan class=\"pl-smi\"\u003e$Column2\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$NumHits\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e\n\nGffAppended=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/final/final_genes_appended.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nbedtools intersect -wao -a \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$GffAppended\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_FTF_hits_intersected.bed\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFus2 genes g16859 and g10474 were identified as FTF1 and FTF2 homologs.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cat $OutDir/\u0026quot;$Strain\u0026quot;_FTF_hits_intersected.bed | grep -w 'gene' | cut -f18 | cut -f1 -d ';' | sort | uniq\"\u003e\u003cpre\u003ecat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_FTF_hits_intersected.bed \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f18 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe orthogorups that these genes belonged to was identified:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tcat analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt | grep -e 'Fus2|g16859.t' -e 'Fus2|g10474.t'\"\u003e\u003cpre\u003e\tcat analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2|g16859.t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2|g10474.t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"orthogroup730: 4287|FOXG_17458-t26_1-p1 4287|FOXG_14422-t26_1-p1 4287|FOXG_15059-t26_1-p1 A13|g9338.t1 4287|FOXG_09390-t26_1-p1 A28|g1088.t1 fo47|FOZG_12687T0 PG|g9482.t1 CB3|g6501.t1 Fus2|g10474.t1 A23|g3017.t1 fo47|FOZG_18011T0 PG|g14993.t1 125|g2599.t1 125|g16201.t1 Fus2|g16859.t1 Fus2|NS_02884.t1 A23|g16099.t1 4287|FOXG_17123-t26_1-p1 4287|FOXG_17084-t26_1-p1 4287|FOXG_14000-t26_1-p1 4287|FOXG_12589-t26_1-p1 4287|FOXG_12539-t26_1-p1 4287|FOXG_16414-t26_1-p1 4287|FOXG_14257-t26_1-p1 A13|g15237.t1\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eorthogroup730: 4287|FOXG_17458-t26_1-p1 4287|FOXG_14422-t26_1-p1 4287|FOXG_15059-t26_1-p1 A13|g9338.t1 4287|FOXG_09390-t26_1-p1 A28|g1088.t1 fo47|FOZG_12687T0 PG|g9482.t1 CB3|g6501.t1 Fus2|g10474.t1 A23|g3017.t1 fo47|FOZG_18011T0 PG|g14993.t1 125|g2599.t1 125|g16201.t1 Fus2|g16859.t1 Fus2|NS_02884.t1 A23|g16099.t1 4287|FOXG_17123-t26_1-p1 4287|FOXG_17084-t26_1-p1 4287|FOXG_14000-t26_1-p1 4287|FOXG_12589-t26_1-p1 4287|FOXG_12539-t26_1-p1 4287|FOXG_16414-t26_1-p1 4287|FOXG_14257-t26_1-p1 A13|g15237.t1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis identified orthogroup730 as containing FTF1 \u0026amp; FTF2.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e9 Transcription factor genes (TF) have been identified in FoL that potentially\nregulate SIX gene expression. The genes in FoL can be used to identify orthologous\nFoC genes:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"TF_list=\u0026quot;FOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274\u0026quot;\nnum=0\nOrthoTxt=$(ls analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt)\nAnnotTab=$(ls gene_pred/annotations/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_gene_annotations.tab)\nfor TF in $TF_list; do\nnum=$(($num +1))\necho \u0026quot;TF\u0026quot;$num\nOrthogroup=$(cat $OrthoTxt | grep -w \u0026quot;$TF\u0026quot; | cut -f1 -d ':')\ncat $AnnotTab | grep -w \u0026quot;$Orthogroup\u0026quot;\ndone \u0026gt; analysis/transcription_factors/Fus2_canu_new_TF1-9.tsv\"\u003e\u003cpre\u003eTF_list=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nnum=0\nOrthoTxt=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nAnnotTab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/annotations/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_gene_annotations.tab\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTF\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TF_list\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nnum=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$((\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e))\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\nOrthogroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ecat \u003cspan class=\"pl-smi\"\u003e$OrthoTxt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$TF\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e:\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$AnnotTab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Orthogroup\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e analysis/transcription_factors/Fus2_canu_new_TF1-9.tsv\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"TF_list=\u0026quot;FOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274\u0026quot;\nnum=0\nOrthoTxt=$(ls analysis/orthology/orthomcl/Fp_Fv_FoC_FoL_Fo/Fp_Fv_FoC_FoL_Fo_orthogroups.txt)\nAnnotTab=$(ls gene_pred/annotations/F.proliferatum/A8_ncbi/A8_ncbi_gene_annotations.tab)\nfor TF in $TF_list; do\nnum=$(($num +1))\necho \u0026quot;TF\u0026quot;$num\nOrthogroup=$(cat $OrthoTxt | grep -w \u0026quot;$TF\u0026quot; | cut -f1 -d ':')\ncat $AnnotTab | grep -w \u0026quot;$Orthogroup\u0026quot;\ndone \u0026gt; analysis/transcription_factors/Fp_TF1-9.tsv\"\u003e\u003cpre\u003eTF_list=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nnum=0\nOrthoTxt=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/orthology/orthomcl/Fp_Fv_FoC_FoL_Fo/Fp_Fv_FoC_FoL_Fo_orthogroups.txt\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nAnnotTab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/annotations/F.proliferatum/A8_ncbi/A8_ncbi_gene_annotations.tab\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eTF\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$TF_list\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nnum=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$((\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e))\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\nOrthogroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003ecat \u003cspan class=\"pl-smi\"\u003e$OrthoTxt\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$TF\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e:\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$AnnotTab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Orthogroup\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e analysis/transcription_factors/Fp_TF1-9.tsv\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.1.C) Identifying Mitochondrial genes in assemblies\u003c/h2\u003e\u003ca id=\"user-content-51c-identifying-mitochondrial-genes-in-assemblies\" class=\"anchor\" aria-label=\"Permalink: 5.1.C) Identifying Mitochondrial genes in assemblies\" href=\"#51c-identifying-mitochondrial-genes-in-assemblies\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePreviously published Fusarium spp. Mitcochondial genes from Al-Reedy 2012 were blasted against\nassembled genomes.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -v -e 'HB17' | grep -w 'Fus2_canu_new'); do\nOrganism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)\nStrain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)\necho $Assembly\nQuery=analysis/blast_homology/Mt_genes/F.spp._mt_prots_Al-Reedy_et_al._2012.fasta\nOutDir=analysis/Mt_genes/$Organism/$Strain\nmkdir -p $OutDir\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub $ProgDir/run_blast2csv.sh $Query protein $Assembly $OutDir\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -w \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f4 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\nQuery=analysis/blast_homology/Mt_genes/F.spp._mt_prots_Al-Reedy_et_al._2012.fasta\nOutDir=analysis/Mt_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/run_blast2csv.sh \u003cspan class=\"pl-smi\"\u003e$Query\u003c/span\u003e protein \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eBLAST hits were converted to Gff annotations and intersected with gene models:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for BlastHits in $(ls analysis/Mt_genes/*/*/*_F.spp._mt_prots_Al-Reedy_et_al._2012.fasta_hits.csv | grep -v -e 'HB17' | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'A8'); do\nStrain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)\nOrganism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)\necho \u0026quot;$Organism - $Strain\u0026quot;\nOutDir=analysis/Mt_genes/$Organism/$Strain\nHitsGff=$(echo $BlastHits | sed  's/.csv/.gff/g')\nColumn2=Mt_gene_homolog\nNumHits=1\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\n$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHits \u0026gt; $HitsGff\n\nGffAppended=$(ls gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3)\nbedtools intersect -wao -a $HitsGff -b $GffAppended \u0026gt; $OutDir/\u0026quot;$Strain\u0026quot;_Mt_hits_intersected.bed\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eBlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els analysis/Mt_genes/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_F.spp._mt_prots_Al-Reedy_et_al._2012.fasta_hits.csv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eA8\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nStrain=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nOrganism=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f3 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e rev\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e - \u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\nOutDir=analysis/Mt_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\nHitsGff=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eecho \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed  \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/.csv/.gff/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nColumn2=Mt_gene_homolog\nNumHits=1\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/blast2gff.pl \u003cspan class=\"pl-smi\"\u003e$Column2\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$NumHits\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$BlastHits\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e\n\nGffAppended=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els gene_pred/final_genes/\u003cspan class=\"pl-smi\"\u003e$Organism\u003c/span\u003e/\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e/final/final_genes_appended.gff3\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\nbedtools intersect -wao -a \u003cspan class=\"pl-smi\"\u003e$HitsGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$GffAppended\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Strain\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_Mt_hits_intersected.bed\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.2 Identifying PHIbase homologs\u003c/h2\u003e\u003ca id=\"user-content-52-identifying-phibase-homologs\" class=\"anchor\" aria-label=\"Permalink: 5.2 Identifying PHIbase homologs\" href=\"#52-identifying-phibase-homologs\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe PHIbase database was searched against the assembled genomes using tBLASTx.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for Assembly in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa| grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'old'); do\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub $ProgDir/blast_pipe.sh analysis/blast_homology/PHIbase/PHI_36_accessions.fa protein $Assembly\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eAssembly\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003els repeat_masked/F.\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_contigs_unmasked.fa\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eHB17\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecepae\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eproliferatum\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003enarcissi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2_canu_new\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003encbi\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eold\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast\nqsub \u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/blast_pipe.sh analysis/blast_homology/PHIbase/PHI_36_accessions.fa protein \u003cspan class=\"pl-smi\"\u003e$Assembly\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003efollowing blasting PHIbase to the genome, the hits were filtered by effect on\nvirulence.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eFirst the a tab seperated file was made in the clusters core directory containing\nPHIbase. These commands were run as part of previous projects but have been\nincluded here for completeness.\u003c/p\u003e\n\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExpressed Genes\u003c/h3\u003e\u003ca id=\"user-content-expressed-genes\" class=\"anchor\" aria-label=\"Permalink: Expressed Genes\" href=\"#expressed-genes\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAs a preliminary analysis of the RNAseq data, highly expressed genes at 72hrs\npost infection were identified in Fus2.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis process is described in RNAseq/Fus2_expressed_path_genes.md and RNAseq/fpkm_analysis.md\u003c/p\u003e\n\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.4 Enrichment of genes on LS contigs\u003c/h3\u003e\u003ca id=\"user-content-54-enrichment-of-genes-on-ls-contigs\" class=\"anchor\" aria-label=\"Permalink: 5.4 Enrichment of genes on LS contigs\" href=\"#54-enrichment-of-genes-on-ls-contigs\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFus2 contigs 10, 14, 16, 19, 20, 21 and 22 were identified as lineage specific.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAs such interproscan annotation were extracted for these contigs and Functional\nenrichments were compared to the entire genome.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eEnrichment of effector families by contig\u003c/h4\u003e\u003ca id=\"user-content-enrichment-of-effector-families-by-contig\" class=\"anchor\" aria-label=\"Permalink: Enrichment of effector families by contig\" href=\"#enrichment-of-effector-families-by-contig\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"GffGenes=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3\ncat $GffGenes | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'\nGffEffP=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted.gff\ncat $GffEffP | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'\nGffCAZY=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted.gff\ncat $GffCAZY | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'\nGffSecMet=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_secondary_metabolite_regions.gff\ncat $GffSecMet | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'\nGffMimp=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff\ncat $GffMimp | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'\nGffMimpSec=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff\ncat $GffMimpSec | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'\"\u003e\u003cpre\u003eGffGenes=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3\ncat \u003cspan class=\"pl-smi\"\u003e$GffGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -c \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -n -k2 -t \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nGffEffP=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted.gff\ncat \u003cspan class=\"pl-smi\"\u003e$GffEffP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -c \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -n -k2 -t \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nGffCAZY=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted.gff\ncat \u003cspan class=\"pl-smi\"\u003e$GffCAZY\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -c \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -n -k2 -t \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nGffSecMet=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_secondary_metabolite_regions.gff\ncat \u003cspan class=\"pl-smi\"\u003e$GffSecMet\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -c \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -n -k2 -t \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nGffMimp=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff\ncat \u003cspan class=\"pl-smi\"\u003e$GffMimp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -c \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -n -k2 -t \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\nGffMimpSec=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff\ncat \u003cspan class=\"pl-smi\"\u003e$GffMimpSec\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq -c \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort -n -k2 -t \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFrom this it was clear that the genes within 2kb of a mimp were primarily associated\nwith the pathogen specific contigs. INterproscan information on these genes\nwas extracted\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"MimpGeneHeaders=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp.txt\nMimpInterpro=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp_interpro.txt\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/filter_by_gene.py --inp_interpro $InterPro --inp_txt $MimpGeneHeaders \u0026gt; $MimpInterpro\"\u003e\u003cpre\u003eMimpGeneHeaders=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp.txt\nMimpInterpro=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp_interpro.txt\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/filter_by_gene.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$InterPro\u003c/span\u003e --inp_txt \u003cspan class=\"pl-smi\"\u003e$MimpGeneHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$MimpInterpro\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003emore information of number of mimps in LS and PS regions was extracted:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"TotalMimps=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps.gff\n# Total in Genome\ncat $TotalMimps |wc -l\n# In all LS regions\ncat $TotalMimps | grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | wc -l\n# In PS regions\ncat $TotalMimps | grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' | wc -l\nGenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff\n# Total in Genome\ncat $GenesIn2Kb | grep 'gene'| wc -l\n# In all LS regions\ncat $GenesIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | wc -l\n# In PS regions\ncat $GenesIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' | wc -l\nSecretedIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff\n# Total in Genome\ncat $SecretedIn2Kb | grep 'gene'| wc -l\n# In all LS regions\ncat $SecretedIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | wc -l\n# In PS regions\ncat $SecretedIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21'  | wc -l\"\u003e\u003cpre\u003eTotalMimps=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps.gff\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Total in Genome\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$TotalMimps\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ewc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e In all LS regions\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$TotalMimps\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_14\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_20\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_22\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e In PS regions\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$TotalMimps\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\nGenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Total in Genome\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$GenesIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e In all LS regions\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$GenesIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_14\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_20\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_22\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e In PS regions\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$GenesIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\nSecretedIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Total in Genome\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$SecretedIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e In all LS regions\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$SecretedIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_14\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_20\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_22\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e In PS regions\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$SecretedIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eEffector candidates within 2Kb of a MIMP were identified:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"GenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff\nOutDir=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new\ncat $GenesIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' | cut -f2 -d '=' | tr -d ';' \u0026gt; $OutDir/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt\ncat $GenesIn2Kb | grep 'gene'| grep -e 'contig_14' -e 'contig_20' -e 'contig_22' | cut -f2 -d '=' | tr -d ';' \u0026gt; $OutDir/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt\ncat $GenesIn2Kb | grep 'gene'| grep -v -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | cut -f2 -d '=' | tr -d ';' \u0026gt; $OutDir/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt\n\nEffectorPHeaders=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted_headers.txt\ncat $EffectorPHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt | wc -l\ncat $EffectorPHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt | wc -l\ncat $EffectorPHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt | wc -l\n\nCazyHeaders=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted_headers.txt\ncat $CazyHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt | wc -l\ncat $CazyHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt | wc -l\ncat $CazyHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt | wc -l\n\n\n\nAntismashGff=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_secondary_metabolite_regions.gff\nMimpPlus2KbGff=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps_exp.gff\nbedtools intersect -u -a $AntismashGff -b $MimpPlus2KbGff | less -S\n\"\u003e\u003cpre\u003eGenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff\nOutDir=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new\ncat \u003cspan class=\"pl-smi\"\u003e$GenesIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt\ncat \u003cspan class=\"pl-smi\"\u003e$GenesIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_14\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_20\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_22\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt\ncat \u003cspan class=\"pl-smi\"\u003e$GenesIn2Kb\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egene\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_10\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_19\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_21\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_14\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_20\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003econtig_22\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tr -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt\n\nEffectorPHeaders=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted_headers.txt\ncat \u003cspan class=\"pl-smi\"\u003e$EffectorPHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\ncat \u003cspan class=\"pl-smi\"\u003e$EffectorPHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\ncat \u003cspan class=\"pl-smi\"\u003e$EffectorPHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\nCazyHeaders=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted_headers.txt\ncat \u003cspan class=\"pl-smi\"\u003e$CazyHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\ncat \u003cspan class=\"pl-smi\"\u003e$CazyHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\ncat \u003cspan class=\"pl-smi\"\u003e$CazyHeaders\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e.\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e wc -l\n\n\n\nAntismashGff=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_secondary_metabolite_regions.gff\nMimpPlus2KbGff=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps_exp.gff\nbedtools intersect -u -a \u003cspan class=\"pl-smi\"\u003e$AntismashGff\u003c/span\u003e -b \u003cspan class=\"pl-smi\"\u003e$MimpPlus2KbGff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e less -S\n\u003c/pre\u003e\u003c/div\u003e\n\n\u003cp dir=\"auto\"\u003eEnrichment was performed for each contig:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"for num in $(seq 1 34); do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\nGff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3\nGeneList=$OutDir/\u0026quot;$Contig\u0026quot;_interproscan.txt\ncat $Gff | grep \u0026quot;$Contig\u0026quot; | grep 'mRNA' | cut -f9 | cut -f1 -d ';' | cut -f2 -d '='  \u0026gt; $GeneList\nInterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv\nInterPro_contig=$OutDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/filter_by_gene.py --inp_interpro $InterPro --inp_txt $GeneList \u0026gt; $InterPro_contig\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $InterPro_contig \u0026gt; $OutDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt\ndone\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e$(\u003c/span\u003eseq 1 34\u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nGff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3\nGeneList=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.txt\ncat \u003cspan class=\"pl-smi\"\u003e$Gff\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emRNA\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f9 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f2 -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e=\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$GeneList\u003c/span\u003e\nInterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv\nInterPro_contig=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/filter_by_gene.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$InterPro\u003c/span\u003e --inp_txt \u003cspan class=\"pl-smi\"\u003e$GeneList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$InterPro_contig\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$InterPro_contig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe WEGO wesite was used to generate summaries of genes on core and LS regions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eA similar process was then repeated for duplicated genes:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes\nmkdir -p $OutDir\nDupGenesTab=/home/sobczm/popgen/codon/blast/dagchainer/testing/no_transposon/non-transposon_duplications_summaryf_ann\ncat $DupGenesTab | tail -n+2 | cut -f1,3,5 | sed 's/\\t/\\n/g' | sed 's/,/\\n/g' | grep -v \u0026quot;^$\u0026quot; | grep 'Fus2' | sed 's/Fus2_//g' | sort | uniq \u0026gt; $OutDir/duplicated_genes.txt\nInterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv\nInterPro_dups=$OutDir/duplicated_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/filter_by_gene.py --inp_interpro $InterPro --inp_txt $OutDir/duplicated_genes.txt \u0026gt; $InterPro_dups\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $InterPro_dups \u0026gt; $OutDir/duplicated_genes_GO_WEGO.txt\"\u003e\u003cpre\u003eOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\nDupGenesTab=/home/sobczm/popgen/codon/blast/dagchainer/testing/no_transposon/non-transposon_duplications_summaryf_ann\ncat \u003cspan class=\"pl-smi\"\u003e$DupGenesTab\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f1,3,5 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/\\t/\\n/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/,/\\n/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e^$\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eFus2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/Fus2_//g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e sort \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e uniq \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/duplicated_genes.txt\nInterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv\nInterPro_dups=\u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/duplicated_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/filter_by_gene.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$InterPro\u003c/span\u003e --inp_txt \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/duplicated_genes.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$InterPro_dups\u003c/span\u003e\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$InterPro_dups\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/duplicated_genes_GO_WEGO.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAnnotations by contig were grouped by chromosome:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr1\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\n\tmkdir -p $OutDir\n\tprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\techo \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tfor num in 1; do\n\tContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\n\techo $Contig\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\n\tmkdir -p $OutDir\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tdone\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr2\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\n\tmkdir -p $OutDir\n\tprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\techo \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tfor num in 17 5; do\n\tContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\n\techo $Contig\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\n\tmkdir -p $OutDir\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tdone\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr4\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\n\tmkdir -p $OutDir\n\tprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\techo \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tfor num in 2; do\n\tContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\n\techo $Contig\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\n\tmkdir -p $OutDir\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tdone\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr5\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\n\tmkdir -p $OutDir\n\tprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\techo \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tfor num in 3; do\n\tContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\n\techo $Contig\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\n\tmkdir -p $OutDir\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tdone\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr7\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\n\tmkdir -p $OutDir\n\tprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\techo \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tfor num in 4; do\n\tContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\n\techo $Contig\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\n\tmkdir -p $OutDir\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\n\tcat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\n\tdone\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr8\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 6; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr9\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 7; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr10\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 8; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr11\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 9; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr12\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 13 15 18; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr13\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 12 11; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS14\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 14; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS10\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 10; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS16\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 16; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS19\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 19; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS20\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 20; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS21\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 21; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS22\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr\nmkdir -p $OutDir\nprintf \u0026quot;\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\necho \u0026quot;!interpro2wego_output\u0026quot; \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\nfor num in 22; do\nContig=\u0026quot;contig_\u0026quot;$num\u0026quot;_pilon\u0026quot;\necho $Contig\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig\nmkdir -p $OutDir\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_interproscan.tsv \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv\ncat $ContigDir/\u0026quot;$Contig\u0026quot;_GO_WEGO.txt | tail -n+2 \u0026gt;\u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_GO_WEGO.txt\ndone\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat $OutDir/\u0026quot;$Chr\u0026quot;_interproscan.tsv | grep -f $DupGenes \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n$ProgDir/interpro2wego.py --inp_interpro $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_interproscan.tsv \u0026gt; $OutDir/\u0026quot;$Chr\u0026quot;_dup_genes_GO_WEGO.txt\"\u003e\u003cpre\u003e\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr1\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 1\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr2\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 17 5\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr4\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 2\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr5\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 3\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\n\tChr=Chr7\n\tOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 4\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\n\tContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\n\tmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\tcat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\t\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\tDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\n\tcat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\n\tProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\t\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr8\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 6\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr9\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 7\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr10\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 8\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr11\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 9\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr12\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 13 15 18\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=Chr13\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 12 11\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS14\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 14\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS10\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 10\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS16\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 16\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS19\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 19\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS20\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 20\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS21\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 21\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\n\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome\nChr=LS22\nOutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eprintf\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e!interpro2wego_output\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enum\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 22\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e\nContig=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econtig_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$num\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_pilon\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\nmkdir -p \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv\ncat \u003cspan class=\"pl-smi\"\u003e$ContigDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Contig\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n+2 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_GO_WEGO.txt\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\nDupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt\ncat \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_interproscan.tsv \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -f \u003cspan class=\"pl-smi\"\u003e$DupGenes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv\nProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction\n\u003cspan class=\"pl-smi\"\u003e$ProgDir\u003c/span\u003e/interpro2wego.py --inp_interpro \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_interproscan.tsv \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$OutDir\u003c/span\u003e/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$Chr\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e_dup_genes_GO_WEGO.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e6. Summarising the Fusarium Proteome\u003c/h2\u003e\u003ca id=\"user-content-6-summarising-the-fusarium-proteome\" class=\"anchor\" aria-label=\"Permalink: 6. Summarising the Fusarium Proteome\" href=\"#6-summarising-the-fusarium-proteome\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\"\u003e\u003cpre\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"#HAPI","anchor":"hapi","htmlText":"#HAPI"},{"level":2,"text":"Estimate coverage","anchor":"estimate-coverage","htmlText":"Estimate coverage"},{"level":2,"text":"Spades Assembly","anchor":"spades-assembly","htmlText":"Spades Assembly"},{"level":1,"text":"Repeatmasking","anchor":"repeatmasking","htmlText":"Repeatmasking"},{"level":1,"text":"Gene Prediction","anchor":"gene-prediction","htmlText":"Gene Prediction"},{"level":1,"text":"Pre-gene prediction","anchor":"pre-gene-prediction","htmlText":"Pre-gene prediction"},{"level":2,"text":"Gene prediction 1 - Braker1 gene model training and prediction","anchor":"gene-prediction-1---braker1-gene-model-training-and-prediction","htmlText":"Gene prediction 1 - Braker1 gene model training and prediction"},{"level":4,"text":"Aligning","anchor":"aligning","htmlText":"Aligning"},{"level":4,"text":"Braker prediction","anchor":"braker-prediction","htmlText":"Braker prediction"},{"level":2,"text":"Supplimenting Braker gene models with CodingQuary genes","anchor":"supplimenting-braker-gene-models-with-codingquary-genes","htmlText":"Supplimenting Braker gene models with CodingQuary genes"},{"level":2,"text":"Identification of duplicated genes in additional CodingQuary gene models","anchor":"identification-of-duplicated-genes-in-additional-codingquary-gene-models","htmlText":"Identification of duplicated genes in additional CodingQuary gene models"},{"level":2,"text":"Manual editing of gene models","anchor":"manual-editing-of-gene-models","htmlText":"Manual editing of gene models"},{"level":3,"text":"Fus2:","anchor":"fus2","htmlText":"Fus2:"},{"level":2,"text":"A23","anchor":"a23","htmlText":"A23"},{"level":2,"text":"FoC 125","anchor":"foc-125","htmlText":"FoC 125"},{"level":2,"text":"FoN N139","anchor":"fon-n139","htmlText":"FoN N139"},{"level":2,"text":"For remaining Fo and Fp isolates","anchor":"for-remaining-fo-and-fp-isolates","htmlText":"For remaining Fo and Fp isolates"},{"level":2,"text":"Assessing the Gene space in predicted transcriptomes:","anchor":"assessing-the-gene-space-in-predicted-transcriptomes","htmlText":"Assessing the Gene space in predicted transcriptomes:"},{"level":2,"text":"ORF finder","anchor":"orf-finder","htmlText":"ORF finder"},{"level":2,"text":"A) Interproscan","anchor":"a-interproscan","htmlText":"A) Interproscan"},{"level":2,"text":"B) SwissProt","anchor":"b-swissprot","htmlText":"B) SwissProt"},{"level":2,"text":"C) CAZY proteins","anchor":"c-cazy-proteins","htmlText":"C) CAZY proteins"},{"level":3,"text":"Summary of CAZY families by organism","anchor":"summary-of-cazy-families-by-organism","htmlText":"Summary of CAZY families by organism"},{"level":2,"text":"D) AntiSMASH","anchor":"d-antismash","htmlText":"D) AntiSMASH"},{"level":2,"text":"E) SSCP","anchor":"e-sscp","htmlText":"E) SSCP"},{"level":2,"text":"Comparison to FoL 4287","anchor":"comparison-to-fol-4287","htmlText":"Comparison to FoL 4287"},{"level":4,"text":"Intersecting blast hits with genes from FoL","anchor":"intersecting-blast-hits-with-genes-from-fol","htmlText":"Intersecting blast hits with genes from FoL"},{"level":2,"text":"Effector genes","anchor":"effector-genes","htmlText":"Effector genes"},{"level":3,"text":"A) From Augustus gene models - Identifying secreted proteins","anchor":"a-from-augustus-gene-models---identifying-secreted-proteins","htmlText":"A) From Augustus gene models - Identifying secreted proteins"},{"level":3,"text":"B) From Augustus gene models - Effector identification using EffectorP","anchor":"b-from-augustus-gene-models---effector-identification-using-effectorp","htmlText":"B) From Augustus gene models - Effector identification using EffectorP"},{"level":3,"text":"C) Identification of MIMP-flanking genes","anchor":"c-identification-of-mimp-flanking-genes","htmlText":"C) Identification of MIMP-flanking genes"},{"level":2,"text":"C)","anchor":"c","htmlText":"C)"},{"level":1,"text":"4. Genomic analysis","anchor":"4-genomic-analysis","htmlText":"4. Genomic analysis"},{"level":2,"text":"4.1 Chracterisation of LS regions","anchor":"41-chracterisation-of-ls-regions","htmlText":"4.1 Chracterisation of LS regions"},{"level":2,"text":"4.2 Orthology","anchor":"42-orthology","htmlText":"4.2 Orthology"},{"level":3,"text":"4.2.c) Orthology between pathogenic and non-pathogenic isolates","anchor":"42c-orthology-between-pathogenic-and-non-pathogenic-isolates","htmlText":"4.2.c) Orthology between pathogenic and non-pathogenic isolates"},{"level":2,"text":"5.  Searches","anchor":"5--searches","htmlText":"5.  Searches"},{"level":2,"text":"5.1.A) Identifying SIX genes","anchor":"51a-identifying-six-genes","htmlText":"5.1.A) Identifying SIX genes"},{"level":2,"text":"5.1.B) Identifying FTF genes","anchor":"51b-identifying-ftf-genes","htmlText":"5.1.B) Identifying FTF genes"},{"level":2,"text":"5.1.C) Identifying Mitochondrial genes in assemblies","anchor":"51c-identifying-mitochondrial-genes-in-assemblies","htmlText":"5.1.C) Identifying Mitochondrial genes in assemblies"},{"level":2,"text":"5.2 Identifying PHIbase homologs","anchor":"52-identifying-phibase-homologs","htmlText":"5.2 Identifying PHIbase homologs"},{"level":3,"text":"Expressed Genes","anchor":"expressed-genes","htmlText":"Expressed Genes"},{"level":3,"text":"5.4 Enrichment of genes on LS contigs","anchor":"54-enrichment-of-genes-on-ls-contigs","htmlText":"5.4 Enrichment of genes on LS contigs"},{"level":4,"text":"Enrichment of effector families by contig","anchor":"enrichment-of-effector-families-by-contig","htmlText":"Enrichment of effector families by contig"},{"level":2,"text":"6. Summarising the Fusarium Proteome","anchor":"6-summarising-the-fusarium-proteome","htmlText":"6. Summarising the Fusarium Proteome"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fharrisonlab%2Ffusarium"}}],"overviewFilesProcessingTime":75.163601,"createFromTemplatePath":"/new?template_name=fusarium\u0026template_owner=harrisonlab"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/harrisonlab/fusarium/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/harrisonlab/fusarium/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/harrisonlab/fusarium/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/harrisonlab/fusarium/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/harrisonlab/fusarium/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           614 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/harrisonlab/fusarium/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="AHDB_project, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/AHDB_project" title="AHDB_project">
                          AHDB_project
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="AHDB_project, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/AHDB_project" title="AHDB_project">
                          AHDB_project
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="FoN_FoP_papers, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/FoN_FoP_papers" title="FoN_FoP_papers">
                          FoN_FoP_papers
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="FoN_FoP_papers, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/FoN_FoP_papers" title="FoN_FoP_papers">
                          FoN_FoP_papers
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="RNAseq, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/RNAseq" title="RNAseq">
                          RNAseq
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="RNAseq, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/RNAseq" title="RNAseq">
                          RNAseq
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="analysis, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/analysis" title="analysis">
                          analysis
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="analysis, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/analysis" title="analysis">
                          analysis
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="assembly, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/assembly" title="assembly">
                          assembly
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="assembly, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/assembly" title="assembly">
                          assembly
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="gene_annotation, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/gene_annotation" title="gene_annotation">
                          gene_annotation
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="gene_annotation, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/gene_annotation" title="gene_annotation">
                          gene_annotation
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="gene_prediction, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/gene_prediction" title="gene_prediction">
                          gene_prediction
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="gene_prediction, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/gene_prediction" title="gene_prediction">
                          gene_prediction
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="genome_submisison, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/genome_submisison" title="genome_submisison">
                          genome_submisison
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="genome_submisison, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/genome_submisison" title="genome_submisison">
                          genome_submisison
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="minion_assemblies, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/minion_assemblies" title="minion_assemblies">
                          minion_assemblies
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="minion_assemblies, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/minion_assemblies" title="minion_assemblies">
                          minion_assemblies
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-9">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pathogen, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/pathogen" title="pathogen">
                          pathogen
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pathogen, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/pathogen" title="pathogen">
                          pathogen
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-10">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="repeat_masking, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/repeat_masking" title="repeat_masking">
                          repeat_masking
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="repeat_masking, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/repeat_masking" title="repeat_masking">
                          repeat_masking
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-11">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tools, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/tools" title="tools">
                          tools
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tools, (Directory)" class="Link--primary" href="/harrisonlab/fusarium/tree/master/tools" title="tools">
                          tools
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-12">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Fus2_assembly.md, (File)" class="Link--primary" href="/harrisonlab/fusarium/blob/master/Fus2_assembly.md" title="Fus2_assembly.md">
                          Fus2_assembly.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="Fus2_assembly.md, (File)" class="Link--primary" href="/harrisonlab/fusarium/blob/master/Fus2_assembly.md" title="Fus2_assembly.md">
                          Fus2_assembly.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-13">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/harrisonlab/fusarium/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/harrisonlab/fusarium/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-14">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="downloaded_genomes.md, (File)" class="Link--primary" href="/harrisonlab/fusarium/blob/master/downloaded_genomes.md" title="downloaded_genomes.md">
                          downloaded_genomes.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="downloaded_genomes.md, (File)" class="Link--primary" href="/harrisonlab/fusarium/blob/master/downloaded_genomes.md" title="downloaded_genomes.md">
                          downloaded_genomes.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd show-for-mobile" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      #HAPI
                     </h1>
                     <a aria-label="Permalink: #HAPI" class="anchor" href="#hapi" id="user-content-hapi">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Scripts used for the analysis of Fusarium genomes
Note - all this work was performed in the directory:
/home/groups/harrisonlab/project_files/fusarium
                    </p>
                    <p dir="auto">
                     The following is a summary of the work presented in this Readme.
                    </p>
                    <p dir="auto">
                     The following processes were applied to Fusarium genomes prior to analysis:
Data qc
Genome assembly
Repeatmasking
Gene prediction
Functional annotation
                    </p>
                    <p dir="auto">
                     Analyses performed on these genomes involved BLAST searching for:
                    </p>
                    <p dir="auto">
                     ls contigs were identified using:
Alignment of raw reads to assembled genomes
Assembly of remaining reads
                    </p>
                    <p dir="auto">
                     #Building of directory structure
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3/
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	mkdir -p $ProjectDir/raw_dna/paired/F.proliferatum/A8/F
	mkdir -p $ProjectDir/raw_dna/paired/F.proliferatum/A8/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R
	mkdir -p $ProjectDir/raw_dna/paired/F.avenaceum/PG8/F
	mkdir -p $ProjectDir/raw_dna/paired/F.avenaceum/PG8/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F
	mkdir -p $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R" dir="auto">
                     <pre>	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3/
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.proliferatum/A8/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.proliferatum/A8/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.avenaceum/PG8/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.avenaceum/PG8/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R</pre>
                    </div>
                    <p dir="auto">
                     Sequence data was moved into the appropriate directories
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp $RawDatDir/Fproliferatum_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.proliferatum/A8/F/.
	cp $RawDatDir/Fproliferatum_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.proliferatum/A8/R/.
	cp $RawDatDir/FoxysporumN139_S2_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F/.
	cp $RawDatDir/FoxysporumN139_S2_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R/.
	cp $RawDatDir/FoxysporumPG3_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F/.
	cp $RawDatDir/FoxysporumPG3_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R/.
	cp $RawDatDir/FoxysporumPG18_S4_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F/.
	cp $RawDatDir/FoxysporumPG18_S4_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R/.
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp $RawDatDir/FOP1_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F/.
	cp $RawDatDir/FOP1_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R/.
	cp $RawDatDir/FOP5_S2_L001_R1_001.fastq.gz  $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F/.
	cp $RawDatDir/FOP5_S2_L001_R2_001.fastq.gz  $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R/.
	cp $RawDatDir/L5_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F/.
	cp $RawDatDir/L5_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R/.
	cp $RawDatDir/HB6_S4_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.
	cp $RawDatDir/HB6_S4_L001_R2_001.fastq.gz  $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp $RawDatDir/A12_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F/.
	cp $RawDatDir/A12_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R/.
	cp $RawDatDir/A13_S2_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F/.
	cp $RawDatDir/A13_S2_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R/.
	cp $RawDatDir/CB3_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F/.
	cp $RawDatDir/CB3_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R/.
	cp $RawDatDir/HB6_S5_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.
	cp $RawDatDir/HB6_S5_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.
	cp $RawDatDir/PG8_S4_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.avenaceum/PG8/F/.
	cp $RawDatDir/PG8_S4_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.avenaceum/PG8/R/.
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp $RawDatDir/FOP2_S1_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F/.
	cp $RawDatDir/FOP2_S1_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R/.
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0
	cp $RawDatDir/AF064_S3_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F/.
	cp $RawDatDir/AF064_S3_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R/.
	cp $RawDatDir/CH5-2_S2_L001_R1_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F/.
	cp $RawDatDir/CH5-2_S2_L001_R2_001.fastq.gz $ProjectDir/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R/." dir="auto">
                     <pre>	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/HAPI_seq_3
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp <span class="pl-smi">$RawDatDir</span>/Fproliferatum_S1_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.proliferatum/A8/F/.
	cp <span class="pl-smi">$RawDatDir</span>/Fproliferatum_S1_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.proliferatum/A8/R/.
	cp <span class="pl-smi">$RawDatDir</span>/FoxysporumN139_S2_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/F/.
	cp <span class="pl-smi">$RawDatDir</span>/FoxysporumN139_S2_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_narcissi/N139/R/.
	cp <span class="pl-smi">$RawDatDir</span>/FoxysporumPG3_S3_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/F/.
	cp <span class="pl-smi">$RawDatDir</span>/FoxysporumPG3_S3_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG3/R/.
	cp <span class="pl-smi">$RawDatDir</span>/FoxysporumPG18_S4_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/F/.
	cp <span class="pl-smi">$RawDatDir</span>/FoxysporumPG18_S4_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/PG18/R/.
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160115_M01678_0041_000000000-AEMMK
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp <span class="pl-smi">$RawDatDir</span>/FOP1_S1_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/F/.
	cp <span class="pl-smi">$RawDatDir</span>/FOP1_S1_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP1/R/.
	cp <span class="pl-smi">$RawDatDir</span>/FOP5_S2_L001_R1_001.fastq.gz  <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/F/.
	cp <span class="pl-smi">$RawDatDir</span>/FOP5_S2_L001_R2_001.fastq.gz  <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP5/R/.
	cp <span class="pl-smi">$RawDatDir</span>/L5_S3_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/L5/F/.
	cp <span class="pl-smi">$RawDatDir</span>/L5_S3_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/L5/R/.
	cp <span class="pl-smi">$RawDatDir</span>/HB6_S4_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.
	cp <span class="pl-smi">$RawDatDir</span>/HB6_S4_L001_R2_001.fastq.gz  <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160119_M016878_0042-AGDN7
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp <span class="pl-smi">$RawDatDir</span>/A12_S1_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/F/.
	cp <span class="pl-smi">$RawDatDir</span>/A12_S1_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A1-2/R/.
	cp <span class="pl-smi">$RawDatDir</span>/A13_S2_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A13/F/.
	cp <span class="pl-smi">$RawDatDir</span>/A13_S2_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/A13/R/.
	cp <span class="pl-smi">$RawDatDir</span>/CB3_S3_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/F/.
	cp <span class="pl-smi">$RawDatDir</span>/CB3_S3_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CB3/R/.
	cp <span class="pl-smi">$RawDatDir</span>/HB6_S5_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/F/.
	cp <span class="pl-smi">$RawDatDir</span>/HB6_S5_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/HB6/R/.
	cp <span class="pl-smi">$RawDatDir</span>/PG8_S4_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.avenaceum/PG8/F/.
	cp <span class="pl-smi">$RawDatDir</span>/PG8_S4_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.avenaceum/PG8/R/.
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160401_M004465_0007-AGKF2
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	cp <span class="pl-smi">$RawDatDir</span>/FOP2_S1_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/F/.
	cp <span class="pl-smi">$RawDatDir</span>/FOP2_S1_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_pisi/FOP2/R/.
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/raw_reads/160725_M04465_0020-AP1N0
	cp <span class="pl-smi">$RawDatDir</span>/AF064_S3_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/F/.
	cp <span class="pl-smi">$RawDatDir</span>/AF064_S3_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/AF064/R/.
	cp <span class="pl-smi">$RawDatDir</span>/CH5-2_S2_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/F/.
	cp <span class="pl-smi">$RawDatDir</span>/CH5-2_S2_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_dna/paired/F.oxysporum_fsp_cepae/CH5-2/R/.</pre>
                    </div>
                    <p dir="auto">
                     This process was repeated for RNAseq data:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/rna_seq
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F
	mkdir -p $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R

	cp $RawDatDir/4_S1_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F/.
	cp $RawDatDir/4_S1_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R/.
	cp $RawDatDir/6_S2_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F/.
	cp $RawDatDir/6_S2_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R/.
	cp $RawDatDir/7_S3_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F/.
	cp $RawDatDir/7_S3_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R/.
	cp $RawDatDir/9_S4_L001_R1_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F/.
	cp $RawDatDir/9_S4_L001_R2_001.fastq.gz $ProjectDir/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R/." dir="auto">
                     <pre>	RawDatDir=/home/groups/harrisonlab/raw_data/raw_seq/fusarium/rna_seq
	ProjectDir=/home/groups/harrisonlab/project_files/fusarium
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F
	mkdir -p <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R

	cp <span class="pl-smi">$RawDatDir</span>/4_S1_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/F/.
	cp <span class="pl-smi">$RawDatDir</span>/4_S1_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDB/R/.
	cp <span class="pl-smi">$RawDatDir</span>/6_S2_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/F/.
	cp <span class="pl-smi">$RawDatDir</span>/6_S2_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_CzapekDox/R/.
	cp <span class="pl-smi">$RawDatDir</span>/7_S3_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/F/.
	cp <span class="pl-smi">$RawDatDir</span>/7_S3_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_GlucosePeptone/R/.
	cp <span class="pl-smi">$RawDatDir</span>/9_S4_L001_R1_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/F/.
	cp <span class="pl-smi">$RawDatDir</span>/9_S4_L001_R2_001.fastq.gz <span class="pl-smi">$ProjectDir</span>/raw_rna/paired/F.oxysporum_fsp_cepae/Fus2_PDA/R/.</pre>
                    </div>
                    <p dir="auto">
                     #Data qc
                    </p>
                    <p dir="auto">
                     programs:
fastqc
fastq-mcf
kmc
                    </p>
                    <p dir="auto">
                     Data quality was visualised using fastqc:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for RawData in $(ls raw_dna/paired/*/*/*/*.fastq.gz); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		echo $RawData;
		qsub $ProgDir/run_fastqc.sh $RawData
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">RawData</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls raw_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>.fastq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		<span class="pl-c1">echo</span> <span class="pl-smi">$RawData</span><span class="pl-k">;</span>
		qsub <span class="pl-smi">$ProgDir</span>/run_fastqc.sh <span class="pl-smi">$RawData</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Trimming was performed on data to trim adapters from
sequences and remove poor quality data. This was done with fastq-mcf
                    </p>
                    <p dir="auto">
                     Firstly, those strains with more than one run were identified:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='	for Strain in $(ls -d raw_dna/paired/*/*); do
	NumReads=$(ls $Strain/F/*.gz | wc -l);
		if [ $NumReads -gt 1 ]; then
			echo "$Strain";
			echo "$NumReads";
		fi;
	done' dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">Strain</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d raw_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
	NumReads=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$Strain</span>/F/<span class="pl-k">*</span>.gz <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span><span class="pl-k">;</span>
		<span class="pl-k">if</span> [ <span class="pl-smi">$NumReads</span> <span class="pl-k">-gt</span> 1 ]<span class="pl-k">;</span> <span class="pl-k">then</span>
			<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span><span class="pl-k">;</span>
			<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$NumReads</span><span class="pl-pds">"</span></span><span class="pl-k">;</span>
		<span class="pl-k">fi</span><span class="pl-k">;</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	raw_dna/paired/F.oxysporum_fsp_cepae/Fus2
	2
	raw_dna/paired/F.oxysporum_fsp_cepae/HB6
	2">
                     <pre class="notranslate"><code>	raw_dna/paired/F.oxysporum_fsp_cepae/Fus2
	2
	raw_dna/paired/F.oxysporum_fsp_cepae/HB6
	2
</code></pre>
                    </div>
                    <p dir="auto">
                     Trimming was first performed on all strains that had a single run of data:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for StrainPath in $(ls -d raw_dna/paired/*/* | grep -v -e 'Fus2' -e 'HB6'); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc
		IlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa
		ReadsF=$(ls $StrainPath/F/*.fastq*)
		ReadsR=$(ls $StrainPath/R/*.fastq*)
		echo $ReadsF
		echo $ReadsR
		qsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">StrainPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d raw_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span> <span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>HB6<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc
		IlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa
		ReadsF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/<span class="pl-k">*</span>.fastq<span class="pl-k">*</span><span class="pl-pds">)</span></span>
		ReadsR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/<span class="pl-k">*</span>.fastq<span class="pl-k">*</span><span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$ReadsF</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$ReadsR</span>
		qsub <span class="pl-smi">$ProgDir</span>/rna_qc_fastq-mcf.sh <span class="pl-smi">$ReadsF</span> <span class="pl-smi">$ReadsR</span> <span class="pl-smi">$IlluminaAdapters</span> DNA
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Trimming was then performed for strains with multiple runs of data
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc
	IlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa
	echo "Fus2"
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2
	ReadsF=$(ls $StrainPath/F/s_6_1_sequence.fastq.gz)
	ReadsR=$(ls $StrainPath/R/s_6_2_sequence.fastq.gz)
	qsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2
	ReadsF=$(ls $StrainPath/F/FUS2_S2_L001_R1_001.fastq.gz)
	ReadsR=$(ls $StrainPath/R/FUS2_S2_L001_R2_001.fastq.gz)
	qsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA
	echo "HB6"
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6
	ReadsF=$(ls $StrainPath/F/HB6_S4_L001_R1_001.fastq.gz)
	ReadsR=$(ls $StrainPath/R/HB6_S4_L001_R2_001.fastq.gz)
	qsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6
	ReadsF=$(ls $StrainPath/F/HB6_S5_L001_R1_001.fastq.gz)
	ReadsR=$(ls $StrainPath/R/HB6_S5_L001_R2_001.fastq.gz)
	qsub $ProgDir/rna_qc_fastq-mcf.sh $ReadsF $ReadsR $IlluminaAdapters DNA' dir="auto">
                     <pre>	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc
	IlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Fus2<span class="pl-pds">"</span></span>
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2
	ReadsF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/s_6_1_sequence.fastq.gz<span class="pl-pds">)</span></span>
	ReadsR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/s_6_2_sequence.fastq.gz<span class="pl-pds">)</span></span>
	qsub <span class="pl-smi">$ProgDir</span>/rna_qc_fastq-mcf.sh <span class="pl-smi">$ReadsF</span> <span class="pl-smi">$ReadsR</span> <span class="pl-smi">$IlluminaAdapters</span> DNA
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/Fus2
	ReadsF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/FUS2_S2_L001_R1_001.fastq.gz<span class="pl-pds">)</span></span>
	ReadsR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/FUS2_S2_L001_R2_001.fastq.gz<span class="pl-pds">)</span></span>
	qsub <span class="pl-smi">$ProgDir</span>/rna_qc_fastq-mcf.sh <span class="pl-smi">$ReadsF</span> <span class="pl-smi">$ReadsR</span> <span class="pl-smi">$IlluminaAdapters</span> DNA
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>HB6<span class="pl-pds">"</span></span>
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6
	ReadsF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/HB6_S4_L001_R1_001.fastq.gz<span class="pl-pds">)</span></span>
	ReadsR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/HB6_S4_L001_R2_001.fastq.gz<span class="pl-pds">)</span></span>
	qsub <span class="pl-smi">$ProgDir</span>/rna_qc_fastq-mcf.sh <span class="pl-smi">$ReadsF</span> <span class="pl-smi">$ReadsR</span> <span class="pl-smi">$IlluminaAdapters</span> DNA
	StrainPath=raw_dna/paired/F.oxysporum_fsp_cepae/HB6
	ReadsF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/HB6_S5_L001_R1_001.fastq.gz<span class="pl-pds">)</span></span>
	ReadsR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/HB6_S5_L001_R2_001.fastq.gz<span class="pl-pds">)</span></span>
	qsub <span class="pl-smi">$ProgDir</span>/rna_qc_fastq-mcf.sh <span class="pl-smi">$ReadsF</span> <span class="pl-smi">$ReadsR</span> <span class="pl-smi">$IlluminaAdapters</span> DNA</pre>
                    </div>
                    <p dir="auto">
                     Data quality was visualised once again following trimming:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for RawData in $(ls qc_dna/paired/*/*/*/*.fq.gz | grep 'FOP2'); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		echo $RawData;
		qsub $ProgDir/run_fastqc.sh $RawData
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">RawData</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls qc_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>.fq.gz <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>FOP2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		<span class="pl-c1">echo</span> <span class="pl-smi">$RawData</span><span class="pl-k">;</span>
		qsub <span class="pl-smi">$ProgDir</span>/run_fastqc.sh <span class="pl-smi">$RawData</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     kmer counting was performed using kmc
This allowed estimation of sequencing depth and total genome size
                    </p>
                    <p dir="auto">
                     This was performed for strains with single runs of data
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for TrimPath in $(ls -d raw_dna/paired/*/* | grep -v -e 'Fus2' -e 'HB6' | grep 'FOP2'); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		TrimF=$(ls $TrimPath/F/*.fastq*)
		TrimR=$(ls $TrimPath/R/*.fastq*)
		echo $TrimF
		echo $TrimR
		qsub $ProgDir/kmc_kmer_counting.sh $TrimF $TrimR
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">TrimPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d raw_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span> <span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>HB6<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>FOP2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		TrimF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/F/<span class="pl-k">*</span>.fastq<span class="pl-k">*</span><span class="pl-pds">)</span></span>
		TrimR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/R/<span class="pl-k">*</span>.fastq<span class="pl-k">*</span><span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimF</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimR</span>
		qsub <span class="pl-smi">$ProgDir</span>/kmc_kmer_counting.sh <span class="pl-smi">$TrimF</span> <span class="pl-smi">$TrimR</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     and for strains with muiltiple runs of data:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for TrimPath in $(ls -d raw_dna/paired/*/Fus2); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		TrimF1=$(ls $TrimPath/F/s_6_1_sequence.fastq.gz)
		TrimR1=$(ls $TrimPath/R/s_6_2_sequence.fastq.gz)
		echo $TrimF1
		echo $TrimR1
		TrimF2=$(ls $TrimPath/F/FUS2_S2_L001_R1_001.fastq.gz)
		TrimR2=$(ls $TrimPath/R/FUS2_S2_L001_R2_001.fastq.gz)
		echo $TrimF2
		echo $TrimR2
		qsub $ProgDir/kmc_kmer_counting.sh $TrimF1 $TrimR1 $TrimF2 $TrimR2
	done
	for TrimPath in $(ls -d raw_dna/paired/*/HB6); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		TrimF1=$(ls $TrimPath/F/HB6_S4_L001_R1_001.fastq.gz)
		TrimR1=$(ls $TrimPath/R/HB6_S4_L001_R2_001.fastq.gz)
		echo $TrimF1
		echo $TrimR1
		TrimF2=$(ls $TrimPath/F/HB6_S5_L001_R1_001.fastq.gz)
		TrimR2=$(ls $TrimPath/R/HB6_S5_L001_R2_001.fastq.gz)
		echo $TrimF2
		echo $TrimR2
		qsub $ProgDir/kmc_kmer_counting.sh $TrimF1 $TrimR1 $TrimF2 $TrimR2
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">TrimPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d raw_dna/paired/<span class="pl-k">*</span>/Fus2<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		TrimF1=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/F/s_6_1_sequence.fastq.gz<span class="pl-pds">)</span></span>
		TrimR1=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/R/s_6_2_sequence.fastq.gz<span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimF1</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimR1</span>
		TrimF2=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/F/FUS2_S2_L001_R1_001.fastq.gz<span class="pl-pds">)</span></span>
		TrimR2=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/R/FUS2_S2_L001_R2_001.fastq.gz<span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimF2</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimR2</span>
		qsub <span class="pl-smi">$ProgDir</span>/kmc_kmer_counting.sh <span class="pl-smi">$TrimF1</span> <span class="pl-smi">$TrimR1</span> <span class="pl-smi">$TrimF2</span> <span class="pl-smi">$TrimR2</span>
	<span class="pl-k">done</span>
	<span class="pl-k">for</span> <span class="pl-smi">TrimPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d raw_dna/paired/<span class="pl-k">*</span>/HB6<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		TrimF1=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/F/HB6_S4_L001_R1_001.fastq.gz<span class="pl-pds">)</span></span>
		TrimR1=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/R/HB6_S4_L001_R2_001.fastq.gz<span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimF1</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimR1</span>
		TrimF2=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/F/HB6_S5_L001_R1_001.fastq.gz<span class="pl-pds">)</span></span>
		TrimR2=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$TrimPath</span>/R/HB6_S5_L001_R2_001.fastq.gz<span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimF2</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$TrimR2</span>
		qsub <span class="pl-smi">$ProgDir</span>/kmc_kmer_counting.sh <span class="pl-smi">$TrimF1</span> <span class="pl-smi">$TrimR1</span> <span class="pl-smi">$TrimF2</span> <span class="pl-smi">$TrimR2</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     mode kmer abundance prior to error correction was reported using the following
commands:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls qc_dna/kmc/*/*/*_true_kmer_summary.txt); do
basename $File;
tail -n3 $File | head -n1 ;
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls qc_dna/kmc/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_true_kmer_summary.txt<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
basename <span class="pl-smi">$File</span><span class="pl-k">;</span>
tail -n3 <span class="pl-smi">$File</span> <span class="pl-k">|</span> head -n1 <span class="pl-k">;</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	PG8_true_kmer_summary.txt
	The mode kmer abundance is:  53 &lt; fail
	125_true_kmer_summary.txt
	The mode kmer abundance is:  46 &lt; pass
	55_true_kmer_summary.txt
	The mode kmer abundance is:  29 &lt; pass
	A1-2_true_kmer_summary.txt
	The mode kmer abundance is:  16 &lt; pass
	A13_true_kmer_summary.txt
	The mode kmer abundance is:  22 &lt; pass
	A23_true_kmer_summary.txt
	The mode kmer abundance is:  33 &lt; pass
	A28_true_kmer_summary.txt
	The mode kmer abundance is:  36 &lt; pass
	CB3_true_kmer_summary.txt
	The mode kmer abundance is:  21 &lt; pass
	D2_true_kmer_summary.txt
	The mode kmer abundance is:  11 &lt; pass
	Fus2_true_kmer_summary.txt
	The mode kmer abundance is:  109 &lt; 2lib
	HB17_true_kmer_summary.txt
	The mode kmer abundance is:  27 &lt; pass
	HB6_true_kmer_summary.txt
	The mode kmer abundance is:  91 &lt; 2 lib
	PG_true_kmer_summary.txt
	The mode kmer abundance is:  58 &lt; pass
	N139_true_kmer_summary.txt
	The mode kmer abundance is:  26 &lt; pass
	FOP1_true_kmer_summary.txt
	The mode kmer abundance is:  32 &lt;- fail
	FOP5_true_kmer_summary.txt
	The mode kmer abundance is:  62 &lt;- fail
	L5_true_kmer_summary.txt
	The mode kmer abundance is:  35 &lt;- pass
	PG18_true_kmer_summary.txt
	The mode kmer abundance is:  24 &lt;- pass
	PG3_true_kmer_summary.txt
	The mode kmer abundance is:  28 &lt;- pass
	A8_true_kmer_summary.txt
	The mode kmer abundance is:  21 &lt;- pass">
                     <pre class="notranslate"><code>	PG8_true_kmer_summary.txt
	The mode kmer abundance is:  53 &lt; fail
	125_true_kmer_summary.txt
	The mode kmer abundance is:  46 &lt; pass
	55_true_kmer_summary.txt
	The mode kmer abundance is:  29 &lt; pass
	A1-2_true_kmer_summary.txt
	The mode kmer abundance is:  16 &lt; pass
	A13_true_kmer_summary.txt
	The mode kmer abundance is:  22 &lt; pass
	A23_true_kmer_summary.txt
	The mode kmer abundance is:  33 &lt; pass
	A28_true_kmer_summary.txt
	The mode kmer abundance is:  36 &lt; pass
	CB3_true_kmer_summary.txt
	The mode kmer abundance is:  21 &lt; pass
	D2_true_kmer_summary.txt
	The mode kmer abundance is:  11 &lt; pass
	Fus2_true_kmer_summary.txt
	The mode kmer abundance is:  109 &lt; 2lib
	HB17_true_kmer_summary.txt
	The mode kmer abundance is:  27 &lt; pass
	HB6_true_kmer_summary.txt
	The mode kmer abundance is:  91 &lt; 2 lib
	PG_true_kmer_summary.txt
	The mode kmer abundance is:  58 &lt; pass
	N139_true_kmer_summary.txt
	The mode kmer abundance is:  26 &lt; pass
	FOP1_true_kmer_summary.txt
	The mode kmer abundance is:  32 &lt;- fail
	FOP5_true_kmer_summary.txt
	The mode kmer abundance is:  62 &lt;- fail
	L5_true_kmer_summary.txt
	The mode kmer abundance is:  35 &lt;- pass
	PG18_true_kmer_summary.txt
	The mode kmer abundance is:  24 &lt;- pass
	PG3_true_kmer_summary.txt
	The mode kmer abundance is:  28 &lt;- pass
	A8_true_kmer_summary.txt
	The mode kmer abundance is:  21 &lt;- pass
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Estimate coverage
                     </h2>
                     <a aria-label="Permalink: Estimate coverage" class="anchor" href="#estimate-coverage" id="user-content-estimate-coverage">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For Miseq data:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for RawData in $(ls qc_dna/paired/*/*/*/*.fq.gz | grep -e 'N139'); do
echo $RawData;
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc;
qsub $ProgDir/run_fastqc.sh $RawData;
GenomeSz=57
OutDir=$(dirname $RawData)
qsub $ProgDir/sub_count_nuc.sh $GenomeSz $RawData $OutDir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">RawData</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls qc_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>.fq.gz <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c1">echo</span> <span class="pl-smi">$RawData</span><span class="pl-k">;</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc<span class="pl-k">;</span>
qsub <span class="pl-smi">$ProgDir</span>/run_fastqc.sh <span class="pl-smi">$RawData</span><span class="pl-k">;</span>
GenomeSz=57
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$RawData</span><span class="pl-pds">)</span></span>
qsub <span class="pl-smi">$ProgDir</span>/sub_count_nuc.sh <span class="pl-smi">$GenomeSz</span> <span class="pl-smi">$RawData</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  for StrainDir in $(ls -d qc_dna/paired/*/* | grep -w -e 'N139'); do
    Strain=$(basename $StrainDir)
    printf &quot;$Strain\t&quot;
    for File in $(ls qc_dna/paired/*/&quot;$Strain&quot;/*/*.txt); do
      echo $(basename $File);
      cat $File | tail -n1 | rev | cut -f2 -d ' ' | rev;
    done | grep -v '.txt' | awk '{ SUM += $1} END { print SUM }'
  done" dir="auto">
                     <pre>  <span class="pl-k">for</span> <span class="pl-smi">StrainDir</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d qc_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span> <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
    Strain=<span class="pl-s"><span class="pl-pds">$(</span>basename <span class="pl-smi">$StrainDir</span><span class="pl-pds">)</span></span>
    <span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span>\t<span class="pl-pds">"</span></span>
    <span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls qc_dna/paired/<span class="pl-k">*</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>.txt<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
      <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">$(</span>basename <span class="pl-smi">$File</span><span class="pl-pds">)</span></span><span class="pl-k">;</span>
      cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> tail -n1 <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span> <span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-k">;</span>
    <span class="pl-k">done</span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>.txt<span class="pl-pds">'</span></span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{ SUM += $1} END { print SUM }<span class="pl-pds">'</span></span>
  <span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="">
                     <pre class="notranslate"><code>
</code></pre>
                    </div>
                    <p dir="auto">
                     #Assembly
                    </p>
                    <p dir="auto">
                     Assembly was performed with:
                    </p>
                    <ul dir="auto">
                     <li>
                      Spades
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Spades Assembly
                     </h2>
                     <a aria-label="Permalink: Spades Assembly" class="anchor" href="#spades-assembly" id="user-content-spades-assembly">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for StrainPath in $(ls -d qc_dna/paired/*/* | grep -v -e 'Fus2' -e 'HB6' | grep 'FOP2'); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades
		Strain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)
		Organism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)
		F_Read=$(ls $StrainPath/F/*.fq.gz)
		R_Read=$(ls $StrainPath/R/*.fq.gz)
		OutDir=assembly/spades/$Organism/$Strain
		Jobs=$(qstat | grep 'submit_SPA' | grep 'qw' | wc -l)
		while [ $Jobs -gt 1 ]; do
			sleep 5m
			printf &quot;.&quot;
			Jobs=$(qstat | grep 'submit_SPA' | grep 'qw' | wc -l)
		done		
		printf &quot;\n&quot;
		echo $F_Read
		echo $R_Read
		qsub $ProgDir/submit_SPAdes.sh $F_Read $R_Read $OutDir correct 10
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">StrainPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d qc_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span> <span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>HB6<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>FOP2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		F_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/<span class="pl-k">*</span>.fq.gz<span class="pl-pds">)</span></span>
		R_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/<span class="pl-k">*</span>.fq.gz<span class="pl-pds">)</span></span>
		OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
		Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>submit_SPA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>qw<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
		<span class="pl-k">while</span> [ <span class="pl-smi">$Jobs</span> <span class="pl-k">-gt</span> 1 ]<span class="pl-k">;</span> <span class="pl-k">do</span>
			sleep 5m
			<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>
			Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>submit_SPA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>qw<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
		<span class="pl-k">done</span>		
		<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>\n<span class="pl-pds">"</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$F_Read</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$R_Read</span>
		qsub <span class="pl-smi">$ProgDir</span>/submit_SPAdes.sh <span class="pl-smi">$F_Read</span> <span class="pl-smi">$R_Read</span> <span class="pl-smi">$OutDir</span> correct 10
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Assembly for PG8, FOP1 and FOP5 failed due to a lack of memory, as such the assembly was
resubmitted with more RAM.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for StrainPath in $(ls -d qc_dna/paired/*/* | grep -e 'FOP5' -e 'PG8' -e 'FOP1'); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades
		Strain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)
		Organism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)
		F_Read=$(ls $StrainPath/F/*.fq.gz)
		R_Read=$(ls $StrainPath/R/*.fq.gz)
		OutDir=assembly/spades/$Organism/$Strain
		echo $F_Read
		echo $R_Read
		qsub $ProgDir/submit_SPAdes_HiMem.sh $F_Read $R_Read $OutDir correct 10
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">StrainPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d qc_dna/paired/<span class="pl-k">*</span>/<span class="pl-k">*</span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>FOP5<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>FOP1<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/spades
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		F_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/<span class="pl-k">*</span>.fq.gz<span class="pl-pds">)</span></span>
		R_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/<span class="pl-k">*</span>.fq.gz<span class="pl-pds">)</span></span>
		OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$F_Read</span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$R_Read</span>
		qsub <span class="pl-smi">$ProgDir</span>/submit_SPAdes_HiMem.sh <span class="pl-smi">$F_Read</span> <span class="pl-smi">$R_Read</span> <span class="pl-smi">$OutDir</span> correct 10
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Assemblies were submitted for genomes with data from multiple sequencing runs:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for StrainPath in $(ls -d qc_dna/paired/F.*/HB6); do
  echo $StrainPath
    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries
    Strain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)
    Organism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)
    echo $Strain
    echo $Organism
    TrimF1_Read=$(ls $StrainPath/F/HB6_S4_L001_R1_001_trim.fq.gz);
    TrimR1_Read=$(ls $StrainPath/R/HB6_S4_L001_R2_001_trim.fq.gz);
    TrimF2_Read=$(ls $StrainPath/F/HB6_S5_L001_R1_001_trim.fq.gz);
    TrimR2_Read=$(ls $StrainPath/R/HB6_S5_L001_R2_001_trim.fq.gz);
    echo $TrimF1_Read
    echo $TrimR1_Read
    echo $TrimF2_Read
    echo $TrimR2_Read
    OutDir=assembly/spades/$Organism/$Strain
    qsub $ProgDir/subSpades_2lib.sh $TrimF1_Read $TrimR1_Read $TrimF2_Read $TrimR2_Read $OutDir correct 10
  done
	for StrainPath in $(ls -d qc_dna/paired/F.*/Fus2); do
	  echo $StrainPath
	    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries
	    Strain=$(echo $StrainPath | rev | cut -f1 -d '/' | rev)
	    Organism=$(echo $StrainPath | rev | cut -f2 -d '/' | rev)
	    echo $Strain
	    echo $Organism
	    TrimF1_Read=$(ls $StrainPath/F/s_6_1_sequence_trim.fq.gz);
	    TrimR1_Read=$(ls $StrainPath/R/s_6_2_sequence_trim.fq.gz);
	    TrimF2_Read=$(ls $StrainPath/F/FUS2_S2_L001_R1_001_trim.fq.gz);
	    TrimR2_Read=$(ls $StrainPath/R/FUS2_S2_L001_R2_001_trim.fq.gz);
	    echo $TrimF1_Read
	    echo $TrimR1_Read
	    echo $TrimF2_Read
	    echo $TrimR2_Read
	    OutDir=assembly/spades/$Organism/$Strain
	    qsub $ProgDir/subSpades_2lib_HiMem.sh $TrimF1_Read $TrimR1_Read $TrimF2_Read $TrimR2_Read $OutDir correct 10
	  done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">StrainPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d qc_dna/paired/F.<span class="pl-k">*</span>/HB6<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
  <span class="pl-c1">echo</span> <span class="pl-smi">$StrainPath</span>
    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries
    Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
    Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
    <span class="pl-c1">echo</span> <span class="pl-smi">$Strain</span>
    <span class="pl-c1">echo</span> <span class="pl-smi">$Organism</span>
    TrimF1_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/HB6_S4_L001_R1_001_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
    TrimR1_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/HB6_S4_L001_R2_001_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
    TrimF2_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/HB6_S5_L001_R1_001_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
    TrimR2_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/HB6_S5_L001_R2_001_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimF1_Read</span>
    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimR1_Read</span>
    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimF2_Read</span>
    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimR2_Read</span>
    OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
    qsub <span class="pl-smi">$ProgDir</span>/subSpades_2lib.sh <span class="pl-smi">$TrimF1_Read</span> <span class="pl-smi">$TrimR1_Read</span> <span class="pl-smi">$TrimF2_Read</span> <span class="pl-smi">$TrimR2_Read</span> <span class="pl-smi">$OutDir</span> correct 10
  <span class="pl-k">done</span>
	<span class="pl-k">for</span> <span class="pl-smi">StrainPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d qc_dna/paired/F.<span class="pl-k">*</span>/Fus2<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
	  <span class="pl-c1">echo</span> <span class="pl-smi">$StrainPath</span>
	    ProgDir=/home/ransoe/git_repos/tools/seq_tools/assemblers/spades/multiple_libraries
	    Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
	    Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$StrainPath</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
	    <span class="pl-c1">echo</span> <span class="pl-smi">$Strain</span>
	    <span class="pl-c1">echo</span> <span class="pl-smi">$Organism</span>
	    TrimF1_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/s_6_1_sequence_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
	    TrimR1_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/s_6_2_sequence_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
	    TrimF2_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/F/FUS2_S2_L001_R1_001_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
	    TrimR2_Read=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$StrainPath</span>/R/FUS2_S2_L001_R2_001_trim.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span>
	    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimF1_Read</span>
	    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimR1_Read</span>
	    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimF2_Read</span>
	    <span class="pl-c1">echo</span> <span class="pl-smi">$TrimR2_Read</span>
	    OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
	    qsub <span class="pl-smi">$ProgDir</span>/subSpades_2lib_HiMem.sh <span class="pl-smi">$TrimF1_Read</span> <span class="pl-smi">$TrimR1_Read</span> <span class="pl-smi">$TrimF2_Read</span> <span class="pl-smi">$TrimR2_Read</span> <span class="pl-smi">$OutDir</span> correct 10
	  <span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Quast
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast
  for Assembly in $(ls assembly/spades/*/*/filtered_contigs/contigs_min_500bp.fasta | grep 'FOP2'); do
	# for Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep 'Fus2_edited_v2'); do
  # for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do
    Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
    Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  
    OutDir=assembly/spades/$Organism/$Strain/filtered_contigs
    qsub $ProgDir/sub_quast.sh $Assembly $OutDir
  done" dir="auto">
                     <pre>  ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast
  <span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/filtered_contigs/contigs_min_500bp.fasta <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>FOP2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
	<span class="pl-c"><span class="pl-c">#</span> for Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep 'Fus2_edited_v2'); do</span>
  <span class="pl-c"><span class="pl-c">#</span> for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do</span>
    Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
    Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>  
    OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/filtered_contigs
    qsub <span class="pl-smi">$ProgDir</span>/sub_quast.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$OutDir</span>
  <span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Contigs were renamed in accordance with ncbi recomendations.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="  ProgDir=~/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
  touch tmp.csv
	for Assembly in $(ls assembly/spades/*/*/filtered_contigs/contigs_min_500bp.fasta); do
    Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
    Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  
    OutDir=assembly/spades/$Organism/$Strain/filtered_contigs
    $ProgDir/remove_contaminants.py --inp $Assembly --out $OutDir/contigs_min_500bp_renamed.fasta --coord_file tmp.csv
  done
  rm tmp.csv" dir="auto">
                     <pre>  ProgDir=<span class="pl-k">~</span>/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
  touch tmp.csv
	<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/filtered_contigs/contigs_min_500bp.fasta<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
    Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
    Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>  
    OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/filtered_contigs
    <span class="pl-smi">$ProgDir</span>/remove_contaminants.py --inp <span class="pl-smi">$Assembly</span> --out <span class="pl-smi">$OutDir</span>/contigs_min_500bp_renamed.fasta --coord_file tmp.csv
  <span class="pl-k">done</span>
  rm tmp.csv</pre>
                    </div>
                    <p dir="auto">
                     A Bioproject and Biosample was made with NCBI genbank for submission of genomes.
Following the creation of these submissions, the .fasta assembly was uploaded
through the submission portal. A note was provided requesting that the assembly
be run through the contamination screen to aid a more detailed resubmission in
future. The returned FCSreport.txt was downloaded from the NCBI webportal and
used to correct the assembly to NCBI standards.
                    </p>
                    <p dir="auto">
                     NCBI reports (FCSreport.txt) were manually downloaded to the following loactions:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep -v 'ncbi_edits' | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do
		Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
		Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  
		NCBI_report_dir=genome_submission/$Organism/$Strain/initial_submission
		mkdir -p $NCBI_report_dir
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/contigs_min_500bp_renamed.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi_edits<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>  
		NCBI_report_dir=genome_submission/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/initial_submission
		mkdir -p <span class="pl-smi">$NCBI_report_dir</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     These downloaded files were used to correct assemblies:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls assembly/spades/*/*/filtered_contigs/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'N139'); do
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
NCBI_report=$(ls genome_submission/$Organism/$Strain/initial_submission/FCSreport.txt)
OutDir=assembly/spades/$Organism/$Strain/ncbi_edits
mkdir -p $OutDir
ProgDir=~/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
$ProgDir/remove_contaminants.py --keep_mitochondria --inp $Assembly --out $OutDir/contigs_min_500bp_renamed.fasta --coord_file $NCBI_report &gt; $OutDir/log.txt
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/filtered_contigs/contigs_min_500bp_renamed.fasta <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
NCBI_report=<span class="pl-s"><span class="pl-pds">$(</span>ls genome_submission/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/initial_submission/FCSreport.txt<span class="pl-pds">)</span></span>
OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/ncbi_edits
mkdir -p <span class="pl-smi">$OutDir</span>
ProgDir=<span class="pl-k">~</span>/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
<span class="pl-smi">$ProgDir</span>/remove_contaminants.py --keep_mitochondria --inp <span class="pl-smi">$Assembly</span> --out <span class="pl-smi">$OutDir</span>/contigs_min_500bp_renamed.fasta --coord_file <span class="pl-smi">$NCBI_report</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/log.txt
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     A second submission of assemblies to ncbi gave a second round of edits to make.
These reports (FCSreport.txt)  were stored in the following location:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls assembly/spades/*/*/*/contigs_min_500bp_renamed.fasta | grep -w 'ncbi_edits' | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  
NCBI_report_dir=genome_submission/$Organism/&quot;$Strain&quot;_ncbi/submission
mkdir -p $NCBI_report_dir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/contigs_min_500bp_renamed.fasta <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>ncbi_edits<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>  
NCBI_report_dir=genome_submission/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_ncbi/submission
mkdir -p <span class="pl-smi">$NCBI_report_dir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     These downloaded files were used to correct assemblies:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls assembly/spades/*/*/ncbi_edits/contigs_min_500bp_renamed.fasta| grep 'ncbi_edits' | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'A8' -e 'PG' -e 'A23'); do
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
NCBI_report=$(ls genome_submission/$Organism/&quot;$Strain&quot;_ncbi/submission/FCSreport.txt)
OutDir=assembly/spades/$Organism/$Strain/ncbi_edits2
mkdir -p $OutDir
touch $NCBI_report
ProgDir=~/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
$ProgDir/remove_contaminants.py --keep_mitochondria --inp $Assembly --out $OutDir/contigs_min_500bp_renamed.fasta --coord_file $NCBI_report &gt; $OutDir/log.txt
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/ncbi_edits/contigs_min_500bp_renamed.fasta<span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi_edits<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
NCBI_report=<span class="pl-s"><span class="pl-pds">$(</span>ls genome_submission/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_ncbi/submission/FCSreport.txt<span class="pl-pds">)</span></span>
OutDir=assembly/spades/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/ncbi_edits2
mkdir -p <span class="pl-smi">$OutDir</span>
touch <span class="pl-smi">$NCBI_report</span>
ProgDir=<span class="pl-k">~</span>/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/remove_contaminants
<span class="pl-smi">$ProgDir</span>/remove_contaminants.py --keep_mitochondria --inp <span class="pl-smi">$Assembly</span> --out <span class="pl-smi">$OutDir</span>/contigs_min_500bp_renamed.fasta --coord_file <span class="pl-smi">$NCBI_report</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/log.txt
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Quast was used to collect details on these assemblies
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast
for Assembly in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'A8' -e 'PG' -e 'A23'); do
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)  
echo &quot;$Organism - $Strain&quot;
OutDir=$(dirname $Assembly)
qsub $ProgDir/sub_quast.sh $Assembly $OutDir
done" dir="auto">
                     <pre>ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/assemblers/assembly_qc/quast
<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/ncbi_edits2/contigs_min_500bp_renamed.fasta <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>  
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$Assembly</span><span class="pl-pds">)</span></span>
qsub <span class="pl-smi">$ProgDir</span>/sub_quast.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Repeatmasking
                     </h1>
                     <a aria-label="Permalink: Repeatmasking" class="anchor" href="#repeatmasking" id="user-content-repeatmasking">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Repeat masking was performed and used the following programs:
Repeatmasker
Repeatmodeler
                    </p>
                    <p dir="auto">
                     The best assemblies were used to perform repeatmasking
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking
for BestAss in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'A8' -e 'PG' -e 'A23'); do
# for BestAss in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do
Strain=$(echo $BestAss | rev | cut -f3 -d '/' | rev)
Organism=$(echo $BestAss | rev | cut -f4 -d '/' | rev)
OutDir=repeat_masked/$Organism/&quot;$Strain&quot;_ncbi/ncbi_submission
qsub $ProgDir/rep_modeling.sh $BestAss $OutDir
qsub $ProgDir/transposonPSI.sh $BestAss $OutDir
done" dir="auto">
                     <pre>ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking
<span class="pl-k">for</span> <span class="pl-smi">BestAss</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls assembly/spades/<span class="pl-k">*</span>/<span class="pl-k">*</span>/ncbi_edits2/contigs_min_500bp_renamed.fasta <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c"><span class="pl-c">#</span> for BestAss in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BestAss</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BestAss</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=repeat_masked/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_ncbi/ncbi_submission
qsub <span class="pl-smi">$ProgDir</span>/rep_modeling.sh <span class="pl-smi">$BestAss</span> <span class="pl-smi">$OutDir</span>
qsub <span class="pl-smi">$ProgDir</span>/transposonPSI.sh <span class="pl-smi">$BestAss</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The published non-pathogen genome for isolate FO47 was also repeatmasked as
this isolate was also used in experimental work.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking
	Fo47Ass=assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta
	for BestAss in $(ls $Fo47Ass); do
		qsub $ProgDir/rep_modeling.sh $BestAss
		qsub $ProgDir/transposonPSI.sh $BestAss
	done" dir="auto">
                     <pre>	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/repeat_masking
	Fo47Ass=assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta
	<span class="pl-k">for</span> <span class="pl-smi">BestAss</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$Fo47Ass</span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		qsub <span class="pl-smi">$ProgDir</span>/rep_modeling.sh <span class="pl-smi">$BestAss</span>
		qsub <span class="pl-smi">$ProgDir</span>/transposonPSI.sh <span class="pl-smi">$BestAss</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Fo47 masked contigs were noted to still contain additional text following spaces.
This disrupted downstream programs.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls repeat_masked/F.oxysporum/fo47/*/fo47_contigs_*masked.fa); do
sed -i 's/ of Fusarium oxysporum Fo47//g' $File
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/F.oxysporum/fo47/<span class="pl-k">*</span>/fo47_contigs_<span class="pl-k">*</span>masked.fa<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
sed -i <span class="pl-s"><span class="pl-pds">'</span>s/ of Fusarium oxysporum Fo47//g<span class="pl-pds">'</span></span> <span class="pl-smi">$File</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The TransposonPSI masked bases were used to mask additional bases from the
repeatmasker / repeatmodeller softmasked and hardmasked files.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)
# while [ $Jobs -gt 0 ]; do
# sleep 10
# printf &quot;.&quot;
# Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)
# done
# printf &quot;\n&quot;
for File in $(ls repeat_masked/*/*/*/*_contigs_softmasked.fa | grep 'ncbi' | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -v 'old' | grep 'N139'); do
OutDir=$(dirname $File)
TPSI=$(ls $OutDir/*_contigs_unmasked.fa.TPSI.allHits.chains.gff3)
OutFile=$(echo $File | sed 's/_contigs_softmasked.fa/_contigs_softmasked_repeatmasker_TPSI_appended.fa/g')
echo &quot;$OutFile&quot;
bedtools maskfasta -soft -fi $File -bed $TPSI -fo $OutFile
echo &quot;Number of masked bases:&quot;
cat $OutFile | grep -v '&gt;' | tr -d '\n' | awk '{print $0, gsub(&quot;[a-z]&quot;, &quot;.&quot;)}' | cut -f2 -d ' '
done
# The number of N's in hardmasked sequence are not counted as some may be present within the assembly and were therefore not repeatmasked.
for File in $(ls repeat_masked/*/*/*/*_contigs_hardmasked.fa | grep 'ncbi' | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -v 'old' | grep 'N139'); do
OutDir=$(dirname $File)
TPSI=$(ls $OutDir/*_contigs_unmasked.fa.TPSI.allHits.chains.gff3)
OutFile=$(echo $File | sed 's/_contigs_hardmasked.fa/_contigs_hardmasked_repeatmasker_TPSI_appended.fa/g')
echo &quot;$OutFile&quot;
bedtools maskfasta -fi $File -bed $TPSI -fo $OutFile
# echo &quot;Number of masked bases:&quot;
# cat $OutFile | grep -v '&gt;' | tr -d '\n' | awk '{print $0, gsub(&quot;[N]&quot;, &quot;.&quot;)}' | cut -f2 -d ' '
done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> while [ $Jobs -gt 0 ]; do</span>
<span class="pl-c"><span class="pl-c">#</span> sleep 10</span>
<span class="pl-c"><span class="pl-c">#</span> printf "."</span>
<span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'rep_modeli' | grep -w 'r' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> done</span>
<span class="pl-c"><span class="pl-c">#</span> printf "\n"</span>
<span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$File</span><span class="pl-pds">)</span></span>
TPSI=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$OutDir</span>/<span class="pl-k">*</span>_contigs_unmasked.fa.TPSI.allHits.chains.gff3<span class="pl-pds">)</span></span>
OutFile=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_contigs_softmasked.fa/_contigs_softmasked_repeatmasker_TPSI_appended.fa/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$OutFile</span><span class="pl-pds">"</span></span>
bedtools maskfasta -soft -fi <span class="pl-smi">$File</span> -bed <span class="pl-smi">$TPSI</span> -fo <span class="pl-smi">$OutFile</span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Number of masked bases:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutFile</span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>\n<span class="pl-pds">'</span></span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $0, gsub("[a-z]", ".")}<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span> <span class="pl-pds">'</span></span>
<span class="pl-k">done</span>
<span class="pl-c"><span class="pl-c">#</span> The number of N's in hardmasked sequence are not counted as some may be present within the assembly and were therefore not repeatmasked.</span>
<span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_hardmasked.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$File</span><span class="pl-pds">)</span></span>
TPSI=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$OutDir</span>/<span class="pl-k">*</span>_contigs_unmasked.fa.TPSI.allHits.chains.gff3<span class="pl-pds">)</span></span>
OutFile=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_contigs_hardmasked.fa/_contigs_hardmasked_repeatmasker_TPSI_appended.fa/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$OutFile</span><span class="pl-pds">"</span></span>
bedtools maskfasta -fi <span class="pl-smi">$File</span> -bed <span class="pl-smi">$TPSI</span> -fo <span class="pl-smi">$OutFile</span>
<span class="pl-c"><span class="pl-c">#</span> echo "Number of masked bases:"</span>
<span class="pl-c"><span class="pl-c">#</span> cat $OutFile | grep -v '&gt;' | tr -d '\n' | awk '{print $0, gsub("[N]", ".")}' | cut -f2 -d ' '</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="repeat_masked/F.oxysporum_fsp_cepae/125_ncbi/ncbi_submission/125_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
3781475
repeat_masked/F.oxysporum_fsp_cepae/A13_ncbi/ncbi_submission/A13_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
4946858
repeat_masked/F.oxysporum_fsp_cepae/A23_ncbi/ncbi_submission/A23_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
3553622
repeat_masked/F.oxysporum_fsp_cepae/A28_ncbi/ncbi_submission/A28_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
4429343
repeat_masked/F.oxysporum_fsp_cepae/CB3_ncbi/ncbi_submission/CB3_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
2863123
repeat_masked/F.oxysporum_fsp_cepae/PG_ncbi/ncbi_submission/PG_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
3070545
repeat_masked/F.proliferatum/A8_ncbi/ncbi_submission/A8_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
1209471
repeat_masked/F.oxysporum_fsp_narcissi/N139_ncbi/ncbi_submission/N139_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
5689747">
                     <pre class="notranslate"><code>repeat_masked/F.oxysporum_fsp_cepae/125_ncbi/ncbi_submission/125_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
3781475
repeat_masked/F.oxysporum_fsp_cepae/A13_ncbi/ncbi_submission/A13_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
4946858
repeat_masked/F.oxysporum_fsp_cepae/A23_ncbi/ncbi_submission/A23_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
3553622
repeat_masked/F.oxysporum_fsp_cepae/A28_ncbi/ncbi_submission/A28_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
4429343
repeat_masked/F.oxysporum_fsp_cepae/CB3_ncbi/ncbi_submission/CB3_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
2863123
repeat_masked/F.oxysporum_fsp_cepae/PG_ncbi/ncbi_submission/PG_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
3070545
repeat_masked/F.proliferatum/A8_ncbi/ncbi_submission/A8_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
1209471
repeat_masked/F.oxysporum_fsp_narcissi/N139_ncbi/ncbi_submission/N139_contigs_softmasked_repeatmasker_TPSI_appended.fa
Number of masked bases:
5689747
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Gene Prediction
                     </h1>
                     <a aria-label="Permalink: Gene Prediction" class="anchor" href="#gene-prediction" id="user-content-gene-prediction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Gene prediction followed three steps:
Pre-gene prediction
- Quality of genome assemblies were assessed using Cegma to see how many core eukaryotic genes can be identified.
Gene model training
- Gene models were trained using assembled RNAseq data as part of the Braker1 pipeline
Gene prediction
- Gene models were used to predict genes in genomes as part of the the Braker1 pipeline. This used RNAseq data as hints for gene models.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Pre-gene prediction
                     </h1>
                     <a aria-label="Permalink: Pre-gene prediction" class="anchor" href="#pre-gene-prediction" id="user-content-pre-gene-prediction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Quality of genome assemblies was assessed by looking for the gene space in the assemblies.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/cegma
cd /home/groups/harrisonlab/project_files/fusarium
for Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi' | grep -v 'old'); do
#for Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -w 'fo47'); do
echo $Genome;
qsub $ProgDir/sub_cegma.sh $Genome dna;
done" dir="auto">
                     <pre>ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/cegma
<span class="pl-c1">cd</span> /home/groups/harrisonlab/project_files/fusarium
<span class="pl-k">for</span> <span class="pl-smi">Genome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c"><span class="pl-c">#</span>for Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -w 'fo47'); do</span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Genome</span><span class="pl-k">;</span>
qsub <span class="pl-smi">$ProgDir</span>/sub_cegma.sh <span class="pl-smi">$Genome</span> dna<span class="pl-k">;</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Outputs were summarised using the commands:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for File in $(ls gene_pred/cegma/F*/*/*_dna_cegma.completeness_report | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi'); do
		Strain=$(echo $File | rev | cut -f2 -d '/' | rev);
		Species=$(echo $File | rev | cut -f3 -d '/' | rev);
		printf &quot;$Species\t$Strain\n&quot;;
		cat $File | head -n18 | tail -n+4;printf &quot;\n&quot;;
	done &gt; gene_pred/cegma/cegma_results_dna_summary.txt

	less gene_pred/cegma/cegma_results_dna_summary.txt" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/cegma/F<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_dna_cegma.completeness_report <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span><span class="pl-k">;</span>
		Species=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span><span class="pl-k">;</span>
		<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Species</span>\t<span class="pl-smi">$Strain</span>\n<span class="pl-pds">"</span></span><span class="pl-k">;</span>
		cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> head -n18 <span class="pl-k">|</span> tail -n+4<span class="pl-k">;</span><span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>\n<span class="pl-pds">"</span></span><span class="pl-k">;</span>
	<span class="pl-k">done</span> <span class="pl-k">&gt;</span> gene_pred/cegma/cegma_results_dna_summary.txt

	less gene_pred/cegma/cegma_results_dna_summary.txt</pre>
                    </div>
                    <p dir="auto">
                     Busco has replaced CEGMA and was run to check gene space in assemblies
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -e 'Fus2' -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' -e 'canu_new' | grep -v 'old'); do
		Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
		Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
		echo &quot;$Organism - $Strain&quot;
		ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco
		# BuscoDB=&quot;Fungal&quot;
		BuscoDB=$(ls -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
		OutDir=gene_pred/busco/$Organism/$Strain/assembly
		qsub $ProgDir/sub_busco2.sh $Assembly $BuscoDB $OutDir
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco
		<span class="pl-c"><span class="pl-c">#</span> BuscoDB="Fungal"</span>
		BuscoDB=<span class="pl-s"><span class="pl-pds">$(</span>ls -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9<span class="pl-pds">)</span></span>
		OutDir=gene_pred/busco/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/assembly
		qsub <span class="pl-smi">$ProgDir</span>/sub_busco2.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$BuscoDB</span> <span class="pl-smi">$OutDir</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for File in $(ls gene_pred/busco/F*/*/assembly/*/short_summary_*.txt); do  
		echo $File;
		cat $File | grep -e '(C)' -e 'Total';
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/busco/F<span class="pl-k">*</span>/<span class="pl-k">*</span>/assembly/<span class="pl-k">*</span>/short_summary_<span class="pl-k">*</span>.txt<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>  
		<span class="pl-c1">echo</span> <span class="pl-smi">$File</span><span class="pl-k">;</span>
		cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>(C)<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Total<span class="pl-pds">'</span></span><span class="pl-k">;</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     #Gene prediction
                    </p>
                    <p dir="auto">
                     Gene prediction was performed for Fusarium genomes. Two gene prediction
approaches were used:
                    </p>
                    <p dir="auto">
                     Gene prediction using Braker1
Prediction of all putative ORFs in the genome using the ORF finder (atg.pl)
approach.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Gene prediction 1 - Braker1 gene model training and prediction
                     </h2>
                     <a aria-label="Permalink: Gene prediction 1 - Braker1 gene model training and prediction" class="anchor" href="#gene-prediction-1---braker1-gene-model-training-and-prediction" id="user-content-gene-prediction-1---braker1-gene-model-training-and-prediction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Gene prediction was performed using Braker1.
                    </p>
                    <p dir="auto">
                     First, RNAseq data was aligned to Fusarium genomes.
                    </p>
                    <ul dir="auto">
                     <li>
                      qc of RNA seq data is detailed below:
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" for Folder in $(ls -d raw_rna/paired/F.oxysporum_fsp_cepae/*); do
	 FolderName=$(echo $Folder | rev | cut -f1 -d '/' | rev);
	 echo $FolderName;
	 ls -lh $Folder/F;
	 ls -lh $Folder/R;
	done" dir="auto">
                     <pre> <span class="pl-k">for</span> <span class="pl-smi">Folder</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d raw_rna/paired/F.oxysporum_fsp_cepae/<span class="pl-k">*</span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
	 FolderName=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Folder</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span><span class="pl-k">;</span>
	 <span class="pl-c1">echo</span> <span class="pl-smi">$FolderName</span><span class="pl-k">;</span>
	 ls -lh <span class="pl-smi">$Folder</span>/F<span class="pl-k">;</span>
	 ls -lh <span class="pl-smi">$Folder</span>/R<span class="pl-k">;</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     This contained the following data:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	55_72hrs_rep1
	sample013_1.combined.fastq.gz
	sample013_2.combined.fastq.gz
	55_72hrs_rep2
	sample014_1.combined.fastq.gz
	sample014_2.combined.fastq.gz
	55_72hrs_rep3
	sample015_1.combined.fastq.gz
	sample015_2.combined.fastq.gz
	control_72hrs_rep1
	sample002_1.combined.fastq.gz
	sample002_2.combined.fastq.gz
	control_72hrs_rep2
	sample004_1.combined.fastq.gz
	sample004_2.combined.fastq.gz
	control_72hrs_rep3
	sample005_1.combined.fastq.gz
	sample005_2.combined.fastq.gz
	FO47_72hrs_rep1
	sample006_1.combined.fastq.gz
	sample006_2.combined.fastq.gz
	FO47_72hrs_rep2
	sample007_1.combined.fastq.gz
	sample007_2.combined.fastq.gz
	FO47_72hrs_rep3
	sample012_1.combined.fastq.gz
	sample012_2.combined.fastq.gz
	Fus2_0hrs_prelim
	1_S1_L001_R1_001.fastq.gz
	1_S1_L001_R2_001.fastq.gz
	Fus2_16hrs_prelim
	3_S2_L001_R1_001.fastq.gz
	3_S2_L001_R2_001.fastq.gz
	Fus2_24hrs_prelim_rep1
	4_S3_L001_R1_001.fastq.gz
	4_S3_L001_R2_001.fastq.gz
	Fus2_24hrs_prelim_rep2
	Fus2_36hrs_prelim
	36hr-root_S10_L001_R1_001.fastq.gz
	36hr-root_S10_L001_R2_001.fastq.gz
	Fus2_48hrs_prelim
	6_S4_L001_R1_001.fastq.gz
	6_S4_L001_R2_001.fastq.gz
	Fus2_4hrs_prelim
	4hr-root_S7_L001_R1_001.fastq.gz
	4hr-root_S7_L001_R2_001.fastq.gz
	Fus2_72hrs_prelim
	7_S5_L001_R1_001.fastq.gz
	7_S5_L001_R2_001.fastq.gz
	Fus2_72hrs_rep1
	sample016_1.combined.fastq.gz
	sample016_2.combined.fastq.gz
	Fus2_72hrs_rep2
	sample018_1.combined.fastq.gz
	sample018_2.combined.fastq.gz
	Fus2_72hrs_rep3
	sample019_1.combined.fastq.gz
	sample019_2.combined.fastq.gz
	Fus2_8hrs_prelim
	8hr-root_S8_L001_R1_001.fastq.gz
	8hr-root_S8_L001_R2_001.fastq.gz
	Fus2_96hrs_prelim
	8_S6_L001_R1_001.fastq.gz
	8_S6_L001_R2_001.fastq.gz
	Fus2_CzapekDox
	6_S2_L001_R1_001_fastqc  6_S2_L001_R1_001.fastq.gz
	6_S2_L001_R2_001_fastqc  6_S2_L001_R2_001.fastq.gz
	Fus2_GlucosePeptone
	7_S3_L001_R1_001_fastqc  7_S3_L001_R1_001.fastq.gz
	7_S3_L001_R2_001_fastqc  7_S3_L001_R2_001.fastq.gz
	Fus2_PDA
	9_S4_L001_R1_001_fastqc  9_S4_L001_R1_001.fastq.gz
	9_S4_L001_R2_001_fastqc  9_S4_L001_R2_001.fastq.gz
	Fus2_PDB
	4_S1_L001_R1_001_fastqc  4_S1_L001_R1_001.fastq.gz
	4_S1_L001_R2_001_fastqc  4_S1_L001_R2_001.fastq.gz">
                     <pre class="notranslate"><code>	55_72hrs_rep1
	sample013_1.combined.fastq.gz
	sample013_2.combined.fastq.gz
	55_72hrs_rep2
	sample014_1.combined.fastq.gz
	sample014_2.combined.fastq.gz
	55_72hrs_rep3
	sample015_1.combined.fastq.gz
	sample015_2.combined.fastq.gz
	control_72hrs_rep1
	sample002_1.combined.fastq.gz
	sample002_2.combined.fastq.gz
	control_72hrs_rep2
	sample004_1.combined.fastq.gz
	sample004_2.combined.fastq.gz
	control_72hrs_rep3
	sample005_1.combined.fastq.gz
	sample005_2.combined.fastq.gz
	FO47_72hrs_rep1
	sample006_1.combined.fastq.gz
	sample006_2.combined.fastq.gz
	FO47_72hrs_rep2
	sample007_1.combined.fastq.gz
	sample007_2.combined.fastq.gz
	FO47_72hrs_rep3
	sample012_1.combined.fastq.gz
	sample012_2.combined.fastq.gz
	Fus2_0hrs_prelim
	1_S1_L001_R1_001.fastq.gz
	1_S1_L001_R2_001.fastq.gz
	Fus2_16hrs_prelim
	3_S2_L001_R1_001.fastq.gz
	3_S2_L001_R2_001.fastq.gz
	Fus2_24hrs_prelim_rep1
	4_S3_L001_R1_001.fastq.gz
	4_S3_L001_R2_001.fastq.gz
	Fus2_24hrs_prelim_rep2
	Fus2_36hrs_prelim
	36hr-root_S10_L001_R1_001.fastq.gz
	36hr-root_S10_L001_R2_001.fastq.gz
	Fus2_48hrs_prelim
	6_S4_L001_R1_001.fastq.gz
	6_S4_L001_R2_001.fastq.gz
	Fus2_4hrs_prelim
	4hr-root_S7_L001_R1_001.fastq.gz
	4hr-root_S7_L001_R2_001.fastq.gz
	Fus2_72hrs_prelim
	7_S5_L001_R1_001.fastq.gz
	7_S5_L001_R2_001.fastq.gz
	Fus2_72hrs_rep1
	sample016_1.combined.fastq.gz
	sample016_2.combined.fastq.gz
	Fus2_72hrs_rep2
	sample018_1.combined.fastq.gz
	sample018_2.combined.fastq.gz
	Fus2_72hrs_rep3
	sample019_1.combined.fastq.gz
	sample019_2.combined.fastq.gz
	Fus2_8hrs_prelim
	8hr-root_S8_L001_R1_001.fastq.gz
	8hr-root_S8_L001_R2_001.fastq.gz
	Fus2_96hrs_prelim
	8_S6_L001_R1_001.fastq.gz
	8_S6_L001_R2_001.fastq.gz
	Fus2_CzapekDox
	6_S2_L001_R1_001_fastqc  6_S2_L001_R1_001.fastq.gz
	6_S2_L001_R2_001_fastqc  6_S2_L001_R2_001.fastq.gz
	Fus2_GlucosePeptone
	7_S3_L001_R1_001_fastqc  7_S3_L001_R1_001.fastq.gz
	7_S3_L001_R2_001_fastqc  7_S3_L001_R2_001.fastq.gz
	Fus2_PDA
	9_S4_L001_R1_001_fastqc  9_S4_L001_R1_001.fastq.gz
	9_S4_L001_R2_001_fastqc  9_S4_L001_R2_001.fastq.gz
	Fus2_PDB
	4_S1_L001_R1_001_fastqc  4_S1_L001_R1_001.fastq.gz
	4_S1_L001_R2_001_fastqc  4_S1_L001_R2_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     Perform qc of RNAseq timecourse data
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for num in $(qstat | grep 'rna_qc' | cut -f1 -d ' '); do
		Treatment=$(head -n1 rna_qc_fastq-mcf.sh.o$num | cut -f10 -d '/')
		for FilePath in $(raw_rna/paired/F.oxysporum_fsp_cepae/55_72hrs_rep1); do
			echo $FilePath
			FileF=$(ls $FilePath/F/*.gz)
			FileR=$(ls $FilePath/R/*.gz)
			IlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa
			ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc
			qsub $ProgDir/rna_qc_fastq-mcf.sh $FileF $FileR $IlluminaAdapters RNA
			sleep 10
		done
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>rna_qc<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span> <span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		Treatment=<span class="pl-s"><span class="pl-pds">$(</span>head -n1 rna_qc_fastq-mcf.sh.o<span class="pl-smi">$num</span> <span class="pl-k">|</span> cut -f10 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
		<span class="pl-k">for</span> <span class="pl-smi">FilePath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>raw_rna/paired/F.oxysporum_fsp_cepae/55_72hrs_rep1<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
			<span class="pl-c1">echo</span> <span class="pl-smi">$FilePath</span>
			FileF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$FilePath</span>/F/<span class="pl-k">*</span>.gz<span class="pl-pds">)</span></span>
			FileR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$FilePath</span>/R/<span class="pl-k">*</span>.gz<span class="pl-pds">)</span></span>
			IlluminaAdapters=/home/armita/git_repos/emr_repos/tools/seq_tools/ncbi_adapters.fa
			ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/rna_qc
			qsub <span class="pl-smi">$ProgDir</span>/rna_qc_fastq-mcf.sh <span class="pl-smi">$FileF</span> <span class="pl-smi">$FileR</span> <span class="pl-smi">$IlluminaAdapters</span> RNA
			sleep 10
		<span class="pl-k">done</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Data quality was visualised using fastqc:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for RawData in $(ls qc_rna/paired/F.oxysporum_fsp_cepae/*/*/*.fq.gz); do
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		echo $RawData;
		qsub $ProgDir/run_fastqc.sh $RawData
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">RawData</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls qc_rna/paired/F.oxysporum_fsp_cepae/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>.fq.gz<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/dna_qc
		<span class="pl-c1">echo</span> <span class="pl-smi">$RawData</span><span class="pl-k">;</span>
		qsub <span class="pl-smi">$ProgDir</span>/run_fastqc.sh <span class="pl-smi">$RawData</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Aligning
                     </h4>
                     <a aria-label="Permalink: Aligning" class="anchor" href="#aligning" id="user-content-aligning">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Insert sizes of the RNA seq library were unknown until a draft alignment could
be made. To do this tophat and cufflinks were run, aligning the reads against a
single genome. The fragment length and stdev were printed to stdout while
cufflinks was running.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep 'FOP2'); do
		Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
		Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
		echo &quot;$Organism - $Strain&quot;
		for RNADir in $(ls -d qc_rna/paired/F.oxysporum_fsp_cepae/*); do
			Timepoint=$(echo $RNADir | rev | cut -f1 -d '/' | rev)
			echo &quot;$Timepoint&quot;
			FileF=$(ls $RNADir/F/*_trim.fq.gz)
			FileR=$(ls $RNADir/R/*_trim.fq.gz)
			OutDir=alignment/$Organism/$Strain/$Timepoint
			ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq
			qsub $ProgDir/tophat_alignment.sh $Assembly $FileF $FileR $OutDir
		done
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>FOP2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
		<span class="pl-k">for</span> <span class="pl-smi">RNADir</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d qc_rna/paired/F.oxysporum_fsp_cepae/<span class="pl-k">*</span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
			Timepoint=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$RNADir</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
			<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Timepoint</span><span class="pl-pds">"</span></span>
			FileF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$RNADir</span>/F/<span class="pl-k">*</span>_trim.fq.gz<span class="pl-pds">)</span></span>
			FileR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$RNADir</span>/R/<span class="pl-k">*</span>_trim.fq.gz<span class="pl-pds">)</span></span>
			OutDir=alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-smi">$Timepoint</span>
			ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq
			qsub <span class="pl-smi">$ProgDir</span>/tophat_alignment.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$FileF</span> <span class="pl-smi">$FileR</span> <span class="pl-smi">$OutDir</span>
		<span class="pl-k">done</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Alignments were concatenated prior to running cufflinks:
Cufflinks was run to produce the fragment length and stdev statistics:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for Assembly in $(ls repeat_masked/*/Fo47/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep 'FOP2'); do
		Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
		Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
		AcceptedHits=alignment/$Organism/$Strain/concatenated/concatenated.bam
		OutDir=gene_pred/cufflinks/$Organism/$Strain/concatenated_prelim
		echo &quot;$Organism - $Strain&quot;
		mkdir -p $OutDir
		samtools merge -f $AcceptedHits \
		alignment/$Organism/$Strain/55_72hrs_rep1/accepted_hits.bam \
		alignment/$Organism/$Strain/55_72hrs_rep2/accepted_hits.bam \
		alignment/$Organism/$Strain/55_72hrs_rep3/accepted_hits.bam \
		alignment/$Organism/$Strain/FO47_72hrs_rep1/accepted_hits.bam \
		alignment/$Organism/$Strain/FO47_72hrs_rep2/accepted_hits.bam \
		alignment/$Organism/$Strain/FO47_72hrs_rep3/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_0hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_16hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_24hrs_prelim_rep1/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_36hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_48hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_4hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_72hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_72hrs_rep1/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_72hrs_rep2/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_72hrs_rep3/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_8hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_96hrs_prelim/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_CzapekDox/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_GlucosePeptone/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_PDA/accepted_hits.bam \
		alignment/$Organism/$Strain/Fus2_PDB/accepted_hits.bam
		cufflinks -o $OutDir/cufflinks -p 8 --max-intron-length 4000 $AcceptedHits 2&gt;&amp;1 | tee $OutDir/cufflinks/cufflinks.log
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/Fo47/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>FOP2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		AcceptedHits=alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated/concatenated.bam
		OutDir=gene_pred/cufflinks/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated_prelim
		<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
		mkdir -p <span class="pl-smi">$OutDir</span>
		samtools merge -f <span class="pl-smi">$AcceptedHits</span> \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/55_72hrs_rep1/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/55_72hrs_rep2/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/55_72hrs_rep3/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/FO47_72hrs_rep1/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/FO47_72hrs_rep2/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/FO47_72hrs_rep3/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_0hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_16hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_24hrs_prelim_rep1/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_36hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_48hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_4hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_rep1/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_rep2/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_rep3/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_8hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_96hrs_prelim/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_CzapekDox/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_GlucosePeptone/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_PDA/accepted_hits.bam \
		alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_PDB/accepted_hits.bam
		cufflinks -o <span class="pl-smi">$OutDir</span>/cufflinks -p 8 --max-intron-length 4000 <span class="pl-smi">$AcceptedHits</span> <span class="pl-k">2&gt;&amp;1</span> <span class="pl-k">|</span> tee <span class="pl-smi">$OutDir</span>/cufflinks/cufflinks.log
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Output from stdout included:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	Processed 22484 loci.                        [*************************] 100%
	Map Properties:
	     Normalized Map Mass: 50507412.55
	     Raw Map Mass: 50507412.55
	     Fragment Length Distribution: Empirical (learned)
	                   Estimated Mean: 181.98
	                Estimated Std Dev: 78.39
	[13:02:48] Assembling transcripts and estimating abundances.
	Processed 22506 loci.                        [*************************] 100%">
                     <pre class="notranslate"><code>	Processed 22484 loci.                        [*************************] 100%
	Map Properties:
	     Normalized Map Mass: 50507412.55
	     Raw Map Mass: 50507412.55
	     Fragment Length Distribution: Empirical (learned)
	                   Estimated Mean: 181.98
	                Estimated Std Dev: 78.39
	[13:02:48] Assembling transcripts and estimating abundances.
	Processed 22506 loci.                        [*************************] 100%
</code></pre>
                    </div>
                    <p dir="auto">
                     The Estimated Mean: 181.98 allowed calculation of of the mean insert gap to be
-20bp 182-(97*2) where 97 was the mean read length. This was provided to tophat
on a second run (as the -r option) along with the fragment length stdev to
increase the accuracy of mapping.
                    </p>
                    <p dir="auto">
                     Then Rnaseq data was aligned to each genome assembly:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi' | grep -v 'old'); do
# for Assembly in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do
# for Assembly in $(ls assembly/merged_canu_spades/*/Fus2/filtered_contigs/Fus2_contigs_renamed.fasta); do
# for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do
Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
echo &quot;$Organism - $Strain&quot;
for RNADir in $(ls -d qc_rna/paired/F.oxysporum_fsp_cepae/*); do
Timepoint=$(echo $RNADir | rev | cut -f1 -d '/' | rev)
echo &quot;$Timepoint&quot;
FileF=$(ls $RNADir/F/*_trim.fq.gz)
FileR=$(ls $RNADir/R/*_trim.fq.gz)
OutDir=alignment/$Organism/$Strain/$Timepoint
InsertGap='-20'
InsertStdDev='78'
Jobs=$(qstat | grep 'tophat' | grep 'qw' | wc -l)
while [ $Jobs -gt 1 ]; do
sleep 10
printf &quot;.&quot;
Jobs=$(qstat | grep 'tophat' | grep 'qw' | wc -l)
done
printf &quot;\n&quot;
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq
qsub $ProgDir/tophat_alignment.sh $Assembly $FileF $FileR $OutDir $InsertGap $InsertStdDev
done
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c"><span class="pl-c">#</span> for Assembly in $(ls assembly/spades/*/*/ncbi_edits2/contigs_min_500bp_renamed.fasta | grep -w -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'); do</span>
<span class="pl-c"><span class="pl-c">#</span> for Assembly in $(ls assembly/merged_canu_spades/*/Fus2/filtered_contigs/Fus2_contigs_renamed.fasta); do</span>
<span class="pl-c"><span class="pl-c">#</span> for Assembly in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_supercontigs.fasta); do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
<span class="pl-k">for</span> <span class="pl-smi">RNADir</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d qc_rna/paired/F.oxysporum_fsp_cepae/<span class="pl-k">*</span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Timepoint=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$RNADir</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Timepoint</span><span class="pl-pds">"</span></span>
FileF=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$RNADir</span>/F/<span class="pl-k">*</span>_trim.fq.gz<span class="pl-pds">)</span></span>
FileR=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$RNADir</span>/R/<span class="pl-k">*</span>_trim.fq.gz<span class="pl-pds">)</span></span>
OutDir=alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-smi">$Timepoint</span>
InsertGap=<span class="pl-s"><span class="pl-pds">'</span>-20<span class="pl-pds">'</span></span>
InsertStdDev=<span class="pl-s"><span class="pl-pds">'</span>78<span class="pl-pds">'</span></span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>tophat<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>qw<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">while</span> [ <span class="pl-smi">$Jobs</span> <span class="pl-k">-gt</span> 1 ]<span class="pl-k">;</span> <span class="pl-k">do</span>
sleep 10
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>tophat<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>qw<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">done</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>\n<span class="pl-pds">"</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq
qsub <span class="pl-smi">$ProgDir</span>/tophat_alignment.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$FileF</span> <span class="pl-smi">$FileR</span> <span class="pl-smi">$OutDir</span> <span class="pl-smi">$InsertGap</span> <span class="pl-smi">$InsertStdDev</span>
<span class="pl-k">done</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Braker prediction
                     </h4>
                     <a aria-label="Permalink: Braker prediction" class="anchor" href="#braker-prediction" id="user-content-braker-prediction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Before braker predictiction was performed, I double checked that I had the
genemark key in my user area and copied it over from the genemark install
directory:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	ls ~/.gm_key
	cp /home/armita/prog/genemark/gm_key_64 ~/.gm_key" dir="auto">
                     <pre>	ls <span class="pl-k">~</span>/.gm_key
	cp /home/armita/prog/genemark/gm_key_64 <span class="pl-k">~</span>/.gm_key</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	# for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep 'Fus2' | grep -e 'Fus2_canu_new' -e 'Fus2_merged' | grep 'cepae' | grep 'Fus2_merged'); do
	# for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep 'proliferatum'); do
for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep 'ncbi' | grep -v 'old'); do
Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)
while [ $Jobs -gt 1 ]; do
sleep 10
printf &quot;.&quot;
Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)
done
printf &quot;\n&quot;
Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
echo &quot;$Organism - $Strain&quot;
mkdir -p alignment/$Organism/$Strain/concatenated
samtools merge -f alignment/$Organism/$Strain/concatenated/concatenated.bam \
alignment/$Organism/$Strain/55_72hrs_rep1/accepted_hits.bam \
alignment/$Organism/$Strain/55_72hrs_rep2/accepted_hits.bam \
alignment/$Organism/$Strain/55_72hrs_rep3/accepted_hits.bam \
alignment/$Organism/$Strain/FO47_72hrs_rep1/accepted_hits.bam \
alignment/$Organism/$Strain/FO47_72hrs_rep2/accepted_hits.bam \
alignment/$Organism/$Strain/FO47_72hrs_rep3/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_0hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_16hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_24hrs_prelim_rep1/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_36hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_48hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_4hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_72hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_72hrs_rep1/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_72hrs_rep2/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_72hrs_rep3/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_8hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_96hrs_prelim/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_CzapekDox/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_GlucosePeptone/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_PDA/accepted_hits.bam \
alignment/$Organism/$Strain/Fus2_PDB/accepted_hits.bam
OutDir=gene_pred/braker/$Organism/&quot;$Strain&quot;_braker
AcceptedHits=alignment/$Organism/$Strain/concatenated/concatenated.bam
GeneModelName=&quot;$Organism&quot;_&quot;$Strain&quot;_braker_new
rm -r /home/armita/prog/augustus-3.1/config/species/&quot;$Organism&quot;_&quot;$Strain&quot;_braker_new
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/braker1
qsub $ProgDir/sub_braker_fungi.sh $Assembly $OutDir $AcceptedHits $GeneModelName
done &gt; log.txt" dir="auto">
                     <pre>	<span class="pl-c"><span class="pl-c">#</span> for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep 'Fus2' | grep -e 'Fus2_canu_new' -e 'Fus2_merged' | grep 'cepae' | grep 'Fus2_merged'); do</span>
	<span class="pl-c"><span class="pl-c">#</span> for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep 'proliferatum'); do</span>
<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>tophat<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>r<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">while</span> [ <span class="pl-smi">$Jobs</span> <span class="pl-k">-gt</span> 1 ]<span class="pl-k">;</span> <span class="pl-k">do</span>
sleep 10
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>tophat<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>r<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">done</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>\n<span class="pl-pds">"</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
mkdir -p alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated
samtools merge -f alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated/concatenated.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/55_72hrs_rep1/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/55_72hrs_rep2/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/55_72hrs_rep3/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/FO47_72hrs_rep1/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/FO47_72hrs_rep2/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/FO47_72hrs_rep3/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_0hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_16hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_24hrs_prelim_rep1/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_36hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_48hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_4hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_rep1/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_rep2/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_72hrs_rep3/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_8hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_96hrs_prelim/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_CzapekDox/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_GlucosePeptone/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_PDA/accepted_hits.bam \
alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/Fus2_PDB/accepted_hits.bam
OutDir=gene_pred/braker/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_braker
AcceptedHits=alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated/concatenated.bam
GeneModelName=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span><span class="pl-pds">"</span></span>_<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_braker_new
rm -r /home/armita/prog/augustus-3.1/config/species/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span><span class="pl-pds">"</span></span>_<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_braker_new
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/braker1
qsub <span class="pl-smi">$ProgDir</span>/sub_braker_fungi.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$OutDir</span> <span class="pl-smi">$AcceptedHits</span> <span class="pl-smi">$GeneModelName</span>
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> log.txt</pre>
                    </div>
                    <p dir="auto">
                     Fasta and gff files were extracted from Braker1 output.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls gene_pred/braker/F.*/*_braker/*/augustus.gff | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'canu_new' -e 'ncbi' | grep -v 'canu_new'); do
getAnnoFasta.pl $File
OutDir=$(dirname $File)
echo &quot;##gff-version 3&quot; &gt; $OutDir/augustus_extracted.gff
cat $File | grep -v '#' &gt;&gt; $OutDir/augustus_extracted.gff
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/braker/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>_braker/<span class="pl-k">*</span>/augustus.gff <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>canu_new<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
getAnnoFasta.pl <span class="pl-smi">$File</span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$File</span><span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>##gff-version 3<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/augustus_extracted.gff
cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>#<span class="pl-pds">'</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/augustus_extracted.gff
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The relationship between gene models and aligned reads was investigated. To do
this aligned reads needed to be sorted and indexed:
                    </p>
                    <p dir="auto">
                     Note - IGV was used to view aligned reads against the Fus2 genome on my local
machine.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	InBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated.bam
	ViewBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_view.bam
	SortBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_sorted
	samtools view -b $InBam &gt; $ViewBam
	samtools sort $ViewBam $SortBam
	samtools index $SortBam.bam" dir="auto">
                     <pre>	InBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated.bam
	ViewBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_view.bam
	SortBam=alignment/F.oxysporum_fsp_cepae/Fus2_edited_v2/concatenated/concatenated_sorted
	samtools view -b <span class="pl-smi">$InBam</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$ViewBam</span>
	samtools sort <span class="pl-smi">$ViewBam</span> <span class="pl-smi">$SortBam</span>
	samtools index <span class="pl-smi">$SortBam</span>.bam</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Supplimenting Braker gene models with CodingQuary genes
                     </h2>
                     <a aria-label="Permalink: Supplimenting Braker gene models with CodingQuary genes" class="anchor" href="#supplimenting-braker-gene-models-with-codingquary-genes" id="user-content-supplimenting-braker-gene-models-with-codingquary-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Additional genes were added to Braker gene predictions, using CodingQuary in
pathogen mode to predict additional regions.
                    </p>
                    <p dir="auto">
                     Fistly, aligned RNAseq data was assembled into transcripts using Cufflinks.
                    </p>
                    <p dir="auto">
                     Note - cufflinks doesn't always predict direction of a transcript and
therefore features can not be restricted by strand when they are intersected.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)
# while [ $Jobs -gt 1 ]; do
# sleep 10
# printf &quot;.&quot;
# Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)
# done
# sleep 2h
# Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)
# while [ $Jobs -gt 1 ]; do
# sleep 10
# printf &quot;.&quot;
# Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)
# done
for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' | grep -v 'old'); do
Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
echo &quot;$Organism - $Strain&quot;
OutDir=gene_pred/cufflinks/$Organism/$Strain/concatenated
mkdir -p $OutDir
AcceptedHits=$(ls alignment/$Organism/$Strain/concatenated/concatenated.bam)
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq
qsub $ProgDir/sub_cufflinks.sh $AcceptedHits $OutDir
done &gt; log2.txt" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> while [ $Jobs -gt 1 ]; do</span>
<span class="pl-c"><span class="pl-c">#</span> sleep 10</span>
<span class="pl-c"><span class="pl-c">#</span> printf "."</span>
<span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'tophat' | grep -w 'r' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> done</span>
<span class="pl-c"><span class="pl-c">#</span> sleep 2h</span>
<span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> while [ $Jobs -gt 1 ]; do</span>
<span class="pl-c"><span class="pl-c">#</span> sleep 10</span>
<span class="pl-c"><span class="pl-c">#</span> printf "."</span>
<span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'sub_br' | grep -w 'r' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> done</span>
<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutDir=gene_pred/cufflinks/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated
mkdir -p <span class="pl-smi">$OutDir</span>
AcceptedHits=<span class="pl-s"><span class="pl-pds">$(</span>ls alignment/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated/concatenated.bam<span class="pl-pds">)</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/RNAseq
qsub <span class="pl-smi">$ProgDir</span>/sub_cufflinks.sh <span class="pl-smi">$AcceptedHits</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> log2.txt</pre>
                    </div>
                    <p dir="auto">
                     Secondly, genes were predicted using CodingQuary:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# Jobs=$(qstat | grep 'armita' | wc -l)
# while [ $Jobs -ge 1 ]; do
# sleep 10m
# printf &quot;.&quot;
# Jobs=$(qstat | grep 'armita' | wc -l)
# done
for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -v 'old'); do
Jobs=$(qstat | grep 'sub_cuffli' | wc -l)
while [ $Jobs -ge 1 ]; do
sleep 10
printf &quot;.&quot;
Jobs=$(qstat | grep 'sub_cuffli' | wc -l)
done
Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
echo &quot;$Organism - $Strain&quot;
OutDir=gene_pred/codingquary/$Organism/$Strain
CufflinksGTF=$(ls gene_pred/cufflinks/$Organism/$Strain/concatenated/transcripts.gtf)
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary
qsub $ProgDir/sub_CodingQuary.sh $Assembly $CufflinksGTF $OutDir
done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'armita' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> while [ $Jobs -ge 1 ]; do</span>
<span class="pl-c"><span class="pl-c">#</span> sleep 10m</span>
<span class="pl-c"><span class="pl-c">#</span> printf "."</span>
<span class="pl-c"><span class="pl-c">#</span> Jobs=$(qstat | grep 'armita' | wc -l)</span>
<span class="pl-c"><span class="pl-c">#</span> done</span>
<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>sub_cuffli<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">while</span> [ <span class="pl-smi">$Jobs</span> <span class="pl-k">-ge</span> 1 ]<span class="pl-k">;</span> <span class="pl-k">do</span>
sleep 10
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>sub_cuffli<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">done</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutDir=gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
CufflinksGTF=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/cufflinks/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/concatenated/transcripts.gtf<span class="pl-pds">)</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary
qsub <span class="pl-smi">$ProgDir</span>/sub_CodingQuary.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$CufflinksGTF</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Then, additional transcripts were added to Braker gene models, when CodingQuary
genes were predicted in regions of the genome, not containing Braker gene
models:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	# for BrakerGff in $(ls gene_pred/braker/F.*/*_braker_new/*/augustus.gff3 | grep -w -e 'Fus2'); do
for BrakerGff in $(ls gene_pred/braker/F.*/*/*/augustus.gff3 | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' | grep 'N139'); do
Strain=$(echo $BrakerGff| rev | cut -d '/' -f3 | rev | sed 's/_braker_pacbio//g'| sed 's/_braker//g')
Organism=$(echo $BrakerGff | rev | cut -d '/' -f4 | rev)
echo &quot;$Organism - $Strain&quot;
Assembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)
CodingQuaryGff=gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3
PGNGff=gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3
AddDir=gene_pred/codingquary/$Organism/$Strain/additional
FinalDir=gene_pred/final_genes/$Organism/$Strain/final
AddGenesList=$AddDir/additional_genes.txt
AddGenesGff=$AddDir/additional_genes.gff
FinalGff=$AddDir/combined_genes.gff
mkdir -p $AddDir
mkdir -p $FinalDir

bedtools intersect -v -a $CodingQuaryGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' &gt; $AddGenesList
bedtools intersect -v -a $PGNGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' &gt;&gt; $AddGenesList
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation
$ProgDir/gene_list_to_gff.pl $AddGenesList $CodingQuaryGff CodingQuarry_v2.0 ID CodingQuary &gt; $AddGenesGff
$ProgDir/gene_list_to_gff.pl $AddGenesList $PGNGff PGNCodingQuarry_v2.0 ID CodingQuary &gt;&gt; $AddGenesGff
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary

$ProgDir/add_CodingQuary_features.pl $AddGenesGff $Assembly &gt; $FinalDir/final_genes_CodingQuary.gff3
$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_CodingQuary.gff3 $FinalDir/final_genes_CodingQuary
cp $BrakerGff $FinalDir/final_genes_Braker.gff3
$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_Braker.gff3 $FinalDir/final_genes_Braker
cat $FinalDir/final_genes_Braker.pep.fasta $FinalDir/final_genes_CodingQuary.pep.fasta | sed -r 's/\*/X/g' &gt; $FinalDir/final_genes_combined.pep.fasta
cat $FinalDir/final_genes_Braker.cdna.fasta $FinalDir/final_genes_CodingQuary.cdna.fasta &gt; $FinalDir/final_genes_combined.cdna.fasta
cat $FinalDir/final_genes_Braker.gene.fasta $FinalDir/final_genes_CodingQuary.gene.fasta &gt; $FinalDir/final_genes_combined.gene.fasta
cat $FinalDir/final_genes_Braker.upstream3000.fasta $FinalDir/final_genes_CodingQuary.upstream3000.fasta &gt; $FinalDir/final_genes_combined.upstream3000.fasta

GffBraker=$FinalDir/final_genes_CodingQuary.gff3
GffQuary=$FinalDir/final_genes_Braker.gff3
GffAppended=$FinalDir/final_genes_appended.gff3
cat $GffBraker $GffQuary &gt; $GffAppended

done" dir="auto">
                     <pre>	<span class="pl-c"><span class="pl-c">#</span> for BrakerGff in $(ls gene_pred/braker/F.*/*_braker_new/*/augustus.gff3 | grep -w -e 'Fus2'); do</span>
<span class="pl-k">for</span> <span class="pl-smi">BrakerGff</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/braker/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/augustus.gff3 <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BrakerGff</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_braker_pacbio//g<span class="pl-pds">'</span></span><span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_braker//g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BrakerGff</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
Assembly=<span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa<span class="pl-pds">)</span></span>
CodingQuaryGff=gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PredictedPass.gff3
PGNGff=gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PGN_predictedPass.gff3
AddDir=gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/additional
FinalDir=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/final
AddGenesList=<span class="pl-smi">$AddDir</span>/additional_genes.txt
AddGenesGff=<span class="pl-smi">$AddDir</span>/additional_genes.gff
FinalGff=<span class="pl-smi">$AddDir</span>/combined_genes.gff
mkdir -p <span class="pl-smi">$AddDir</span>
mkdir -p <span class="pl-smi">$FinalDir</span>

bedtools intersect -v -a <span class="pl-smi">$CodingQuaryGff</span> -b <span class="pl-smi">$BrakerGff</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> cut -f2 -d<span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d<span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$AddGenesList</span>
bedtools intersect -v -a <span class="pl-smi">$PGNGff</span> -b <span class="pl-smi">$BrakerGff</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> cut -f2 -d<span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d<span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$AddGenesList</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation
<span class="pl-smi">$ProgDir</span>/gene_list_to_gff.pl <span class="pl-smi">$AddGenesList</span> <span class="pl-smi">$CodingQuaryGff</span> CodingQuarry_v2.0 ID CodingQuary <span class="pl-k">&gt;</span> <span class="pl-smi">$AddGenesGff</span>
<span class="pl-smi">$ProgDir</span>/gene_list_to_gff.pl <span class="pl-smi">$AddGenesList</span> <span class="pl-smi">$PGNGff</span> PGNCodingQuarry_v2.0 ID CodingQuary <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$AddGenesGff</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary

<span class="pl-smi">$ProgDir</span>/add_CodingQuary_features.pl <span class="pl-smi">$AddGenesGff</span> <span class="pl-smi">$Assembly</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gff3
<span class="pl-smi">$ProgDir</span>/gff2fasta.pl <span class="pl-smi">$Assembly</span> <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gff3 <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary
cp <span class="pl-smi">$BrakerGff</span> <span class="pl-smi">$FinalDir</span>/final_genes_Braker.gff3
<span class="pl-smi">$ProgDir</span>/gff2fasta.pl <span class="pl-smi">$Assembly</span> <span class="pl-smi">$FinalDir</span>/final_genes_Braker.gff3 <span class="pl-smi">$FinalDir</span>/final_genes_Braker
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.pep.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.pep.fasta <span class="pl-k">|</span> sed -r <span class="pl-s"><span class="pl-pds">'</span>s/\*/X/g<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.pep.fasta
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.cdna.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.cdna.fasta <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.cdna.fasta
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.gene.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gene.fasta <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.gene.fasta
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.upstream3000.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.upstream3000.fasta <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.upstream3000.fasta

GffBraker=<span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gff3
GffQuary=<span class="pl-smi">$FinalDir</span>/final_genes_Braker.gff3
GffAppended=<span class="pl-smi">$FinalDir</span>/final_genes_appended.gff3
cat <span class="pl-smi">$GffBraker</span> <span class="pl-smi">$GffQuary</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$GffAppended</span>

<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The final number of genes per isolate was observed using:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for DirPath in $(ls -d gene_pred/final_genes/F.*/*/final | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139'  | grep -e 'ncbi'); do
echo $DirPath;
cat $DirPath/final_genes_Braker.pep.fasta | grep '&gt;' | wc -l;
cat $DirPath/final_genes_CodingQuary.pep.fasta | grep '&gt;' | wc -l;
cat $DirPath/final_genes_combined.pep.fasta | grep '&gt;' | wc -l;
echo &quot;&quot;;
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">DirPath</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/final <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span>  <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c1">echo</span> <span class="pl-smi">$DirPath</span><span class="pl-k">;</span>
cat <span class="pl-smi">$DirPath</span>/final_genes_Braker.pep.fasta <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-k">;</span>
cat <span class="pl-smi">$DirPath</span>/final_genes_CodingQuary.pep.fasta <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-k">;</span>
cat <span class="pl-smi">$DirPath</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-k">;</span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span><span class="pl-k">;</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gene_pred/final_genes/F.oxysporum_fsp_cepae/125_ncbi/final
17197
1546
18743

gene_pred/final_genes/F.oxysporum_fsp_cepae/A13_ncbi/final
17986
948
18934

gene_pred/final_genes/F.oxysporum_fsp_cepae/A23_ncbi/final
17006
1551
18557

gene_pred/final_genes/F.oxysporum_fsp_cepae/A28_ncbi/final
17426
1448
18874

gene_pred/final_genes/F.oxysporum_fsp_cepae/CB3_ncbi/final
16833
1345
18178

gene_pred/final_genes/F.oxysporum_fsp_cepae/PG_ncbi/final
16811
1290
18101

gene_pred/final_genes/F.oxysporum_fsp_narcissi/N139_ncbi/final
19059
1642
20701

gene_pred/final_genes/F.proliferatum/A8_ncbi/final
15421
37
15458">
                     <pre class="notranslate"><code>gene_pred/final_genes/F.oxysporum_fsp_cepae/125_ncbi/final
17197
1546
18743

gene_pred/final_genes/F.oxysporum_fsp_cepae/A13_ncbi/final
17986
948
18934

gene_pred/final_genes/F.oxysporum_fsp_cepae/A23_ncbi/final
17006
1551
18557

gene_pred/final_genes/F.oxysporum_fsp_cepae/A28_ncbi/final
17426
1448
18874

gene_pred/final_genes/F.oxysporum_fsp_cepae/CB3_ncbi/final
16833
1345
18178

gene_pred/final_genes/F.oxysporum_fsp_cepae/PG_ncbi/final
16811
1290
18101

gene_pred/final_genes/F.oxysporum_fsp_narcissi/N139_ncbi/final
19059
1642
20701

gene_pred/final_genes/F.proliferatum/A8_ncbi/final
15421
37
15458
</code></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Gff in $(ls gene_pred/final_genes/*/*/final/final_genes_appended.gff3 | grep -e 'canu_new' -e 'ncbi'); do
	Strain=$(echo $Gff | rev | cut -d '/' -f3 | rev)
	Organism=$(echo $Gff | rev | cut -d '/' -f4 | rev)
	echo &quot;$Strain - $Organism&quot;
	cat $Gff | grep -w 'gene' | wc -l
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Gff</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final/final_genes_appended.gff3 <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
	Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Gff</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
	Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Gff</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span> - <span class="pl-smi">$Organism</span><span class="pl-pds">"</span></span>
	cat <span class="pl-smi">$Gff</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="125_ncbi - F.oxysporum_fsp_cepae
18505
A13_ncbi - F.oxysporum_fsp_cepae
18790
A23_ncbi - F.oxysporum_fsp_cepae
18323
A28_ncbi - F.oxysporum_fsp_cepae
18629
CB3_ncbi - F.oxysporum_fsp_cepae
17943
Fus2_canu_new - F.oxysporum_fsp_cepae
18855
PG_ncbi - F.oxysporum_fsp_cepae
17830
N139_ncbi - F.oxysporum_fsp_narcissi
20493
A8_ncbi - F.proliferatum
15418">
                     <pre class="notranslate"><code>125_ncbi - F.oxysporum_fsp_cepae
18505
A13_ncbi - F.oxysporum_fsp_cepae
18790
A23_ncbi - F.oxysporum_fsp_cepae
18323
A28_ncbi - F.oxysporum_fsp_cepae
18629
CB3_ncbi - F.oxysporum_fsp_cepae
17943
Fus2_canu_new - F.oxysporum_fsp_cepae
18855
PG_ncbi - F.oxysporum_fsp_cepae
17830
N139_ncbi - F.oxysporum_fsp_narcissi
20493
A8_ncbi - F.proliferatum
15418
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Identification of duplicated genes in additional CodingQuary gene models
                     </h2>
                     <a aria-label="Permalink: Identification of duplicated genes in additional CodingQuary gene models" class="anchor" href="#identification-of-duplicated-genes-in-additional-codingquary-gene-models" id="user-content-identification-of-duplicated-genes-in-additional-codingquary-gene-models">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for AddGenes in $(ls gene_pred/codingquary/F.*/*/additional/additional_genes.gff | grep 'ncbi' | grep -v 'ncbi_braker' | grep 'A23'); do
Strain=$(echo $AddGenes| rev | cut -d '/' -f3 | rev)
Organism=$(echo $AddGenes | rev | cut -d '/' -f4 | rev)
OutDir=$(dirname $AddGenes)
echo &quot;$Organism - $Strain&quot; &gt; $OutDir/duplicated_genes.txt
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary
$ProgDir/remove_dup_features.py --inp_gff $AddGenes &gt;&gt; $OutDir/duplicated_genes.txt
cat $OutDir/duplicated_genes.txt
echo &quot;&quot;
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">AddGenes</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/additional/additional_genes.gff <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi_braker<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$AddGenes</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$AddGenes</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$AddGenes</span><span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/duplicated_genes.txt
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary
<span class="pl-smi">$ProgDir</span>/remove_dup_features.py --inp_gff <span class="pl-smi">$AddGenes</span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/duplicated_genes.txt
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Manual editing of gene models
                     </h2>
                     <a aria-label="Permalink: Manual editing of gene models" class="anchor" href="#manual-editing-of-gene-models" id="user-content-manual-editing-of-gene-models">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Following annotation of gene models and orthology analysis documented below,
some gene models were noted to require manual editing. This was performed and
gene models were re-annotated. In some cases this required removal of genes
annotated by Braker so that the gene models predicted by coding queary could be
favoured. In other cases, gene models were manually edited based upon results
from other Fusarium spp. genomes:
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Fus2:
                     </h3>
                     <a aria-label="Permalink: Fus2:" class="anchor" href="#fus2" id="user-content-fus2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Gene models requiring editing
                    </p>
                    <ul dir="auto">
                     <li>
                      FTF1 g16934 - delete gene, replace with CodingQuary model.
                     </li>
                     <li>
                      SIX3 g16847, g16190 - A23 one copy
                     </li>
                     <li>
                      SIX5 g16849 - A23 missing intron
                     </li>
                     <li>
                      SIX7 CUFF_4344_1_33 - Gene model good
                     </li>
                     <li>
                      SIX9 g13574, g16273 - premature stop in A23 (end of contig) &amp; one copy in A23 &amp; 125
                     </li>
                     <li>
                      SIX10 g17153 - Gene model good
                     </li>
                     <li>
                      SIX12 PGN_CUFF_4345_1_2 - checked start of Fus2, FoC sequence is too long but FoL reference matches this length - Bring A23 and 125 in line with Fus2 &amp; FoL
                     </li>
                     <li>
                      No gene model
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	Organism=F.oxysporum_fsp_cepae
	Strain=Fus2_canu_new
	OutDir=gene_pred/final_genes/$Organism/$Strain/edited
	mkdir -p $OutDir

	Assembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)
	SourceDir=gene_pred/codingquary/$Organism/$Strain
	BrakerGff=$(ls $SourceDir/final/final_genes_Braker.gff3)
	CodingQuaryGff=$(ls $SourceDir/out/PredictedPass.gff3)
	PGNGff=$(ls $SourceDir/out/PGN_predictedPass.gff3)

	cat $BrakerGff \
	| grep -v -w 'g16934' \
	| grep -v -w 'g13345' \
	&gt; $OutDir/final_genes_Braker_ed.gff3
	printf \
	&quot;contig_19_pilon\tManual_annotation\tgene\t445791\t445964\t0\t+\t.\tID=man_1;
	contig_19_pilon\tManual_annotation\tmRNA\t445791\t445964\t.\t+\t.\tID=man_1.t1;Parent=man_1
	contig_19_pilon\tManual_annotation\tstart_codon\t445791\t445793\t.\t+\t.\tParent=man_1.t1;
	contig_19_pilon\tManual_annotation\tCDS\t445791\t445964\t.\t+\t0\tID=man_1.t1.CDS1;Parent=man_1.t1
	contig_19_pilon\tManual_annotation\texon\t445791\t445964\t.\t+\t.\tID=man_1.t1.exon1;Parent=man_1.t1;
	contig_19_pilon\tManual_annotation\tstop_codon\t445962\t445964\t.\t+\t.\tParent=man_1.t1;\n&quot; \
	&gt;&gt; $OutDir/final_genes_Braker_ed.gff3

	cat $CodingQuaryGff \
	| grep -v -w -e 'NS.00001' -e 'NS.00002' -e 'NS.00003' -e 'NS.04463' -e 'CUFF.11065.1.82' \
	&gt; $OutDir/PredictedPass_ed.gff3
	cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3

	touch $OutDir/manual_annotations.gff3
	#
	# printf \
	# &quot;contig_10_pilon\tManual_annotation\tgene\t1569\t5055\t0\t+\t.\tID=g13345_ps;
	# contig_10_pilon\tManual_annotation\tmRNA\t1569\t5055\t.\t+\t.\tID=g13345_ps.t1;Parent=g13345_ps
	# contig_10_pilon\tManual_annotation\tstart_codon\t1569\t1571\t.\t+\t.\tParent=g13345_ps.t1;
	# contig_10_pilon\tManual_annotation\tCDS\t1569\t2856\t.\t+\t0\tID=g13345_ps.t1.CDS1;Parent=g13345_ps.t1
	# contig_10_pilon\tManual_annotation\texon\t1569\t2856\t.\t+\t.\tParent=g13345_ps.t1
	# contig_10_pilon\tManual_annotation\tintron\t2857\t2905\t.\t+\t.\tParent=g13345_ps.t1
	# contig_10_pilon\tManual_annotation\tCDS\t2906\t5055\t.\t+\t2\tID=g13345_ps.t1.CDS2;Parent=g13345_ps.t1
	# contig_10_pilon\tManual_annotation\texon\t2906\t5055\t.\t+\t.\tID=g13345_ps.t1.exon1;Parent=g13345_ps.t1;
	# contig_10_pilon\tManual_annotation\tstop_codon\t5053\t5055\t.\t+\t.\tParent=g13345_ps.t1;\n&quot; \
	# &gt; $OutDir/pseudogenes.gff3" dir="auto">
                     <pre>	Organism=F.oxysporum_fsp_cepae
	Strain=Fus2_canu_new
	OutDir=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/edited
	mkdir -p <span class="pl-smi">$OutDir</span>

	Assembly=<span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa<span class="pl-pds">)</span></span>
	SourceDir=gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
	BrakerGff=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$SourceDir</span>/final/final_genes_Braker.gff3<span class="pl-pds">)</span></span>
	CodingQuaryGff=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$SourceDir</span>/out/PredictedPass.gff3<span class="pl-pds">)</span></span>
	PGNGff=<span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$SourceDir</span>/out/PGN_predictedPass.gff3<span class="pl-pds">)</span></span>

	cat <span class="pl-smi">$BrakerGff</span> \
	<span class="pl-k">|</span> grep -v -w <span class="pl-s"><span class="pl-pds">'</span>g16934<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> grep -v -w <span class="pl-s"><span class="pl-pds">'</span>g13345<span class="pl-pds">'</span></span> \
	<span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3
	<span class="pl-c1">printf</span> \
	<span class="pl-s"><span class="pl-pds">"</span>contig_19_pilon\tManual_annotation\tgene\t445791\t445964\t0\t+\t.\tID=man_1;</span>
<span class="pl-s">	contig_19_pilon\tManual_annotation\tmRNA\t445791\t445964\t.\t+\t.\tID=man_1.t1;Parent=man_1</span>
<span class="pl-s">	contig_19_pilon\tManual_annotation\tstart_codon\t445791\t445793\t.\t+\t.\tParent=man_1.t1;</span>
<span class="pl-s">	contig_19_pilon\tManual_annotation\tCDS\t445791\t445964\t.\t+\t0\tID=man_1.t1.CDS1;Parent=man_1.t1</span>
<span class="pl-s">	contig_19_pilon\tManual_annotation\texon\t445791\t445964\t.\t+\t.\tID=man_1.t1.exon1;Parent=man_1.t1;</span>
<span class="pl-s">	contig_19_pilon\tManual_annotation\tstop_codon\t445962\t445964\t.\t+\t.\tParent=man_1.t1;\n<span class="pl-pds">"</span></span> \
	<span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3

	cat <span class="pl-smi">$CodingQuaryGff</span> \
	<span class="pl-k">|</span> grep -v -w -e <span class="pl-s"><span class="pl-pds">'</span>NS.00001<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>NS.00002<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>NS.00003<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>NS.04463<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.11065.1.82<span class="pl-pds">'</span></span> \
	<span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
	cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3

	touch <span class="pl-smi">$OutDir</span>/manual_annotations.gff3
	<span class="pl-c"><span class="pl-c">#</span></span>
	<span class="pl-c"><span class="pl-c">#</span> printf \</span>
	<span class="pl-c"><span class="pl-c">#</span> "contig_10_pilon\tManual_annotation\tgene\t1569\t5055\t0\t+\t.\tID=g13345_ps;</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\tmRNA\t1569\t5055\t.\t+\t.\tID=g13345_ps.t1;Parent=g13345_ps</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\tstart_codon\t1569\t1571\t.\t+\t.\tParent=g13345_ps.t1;</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\tCDS\t1569\t2856\t.\t+\t0\tID=g13345_ps.t1.CDS1;Parent=g13345_ps.t1</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\texon\t1569\t2856\t.\t+\t.\tParent=g13345_ps.t1</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\tintron\t2857\t2905\t.\t+\t.\tParent=g13345_ps.t1</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\tCDS\t2906\t5055\t.\t+\t2\tID=g13345_ps.t1.CDS2;Parent=g13345_ps.t1</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\texon\t2906\t5055\t.\t+\t.\tID=g13345_ps.t1.exon1;Parent=g13345_ps.t1;</span>
	<span class="pl-c"><span class="pl-c">#</span> contig_10_pilon\tManual_annotation\tstop_codon\t5053\t5055\t.\t+\t.\tParent=g13345_ps.t1;\n" \</span>
	<span class="pl-c"><span class="pl-c">#</span> &gt; $OutDir/pseudogenes.gff3</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      A23
                     </h2>
                     <a aria-label="Permalink: A23" class="anchor" href="#a23" id="user-content-a23">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Note that at this stage all codingquary genes contain . characters rather than _ characters
g16038 - SIX12 - use alternative transcript
no gene - SIX14 - manual annotation
g16100 - SIX5 - shorten intron and final CDS
g15699 - SIX7 - remove intron and final CDS.
no gene - SIX9  - manual annotation
Duplicated genes:
CUFF.7204.2.14
CUFF.7961.1.0
CUFF.11924.2.127
CUFF.6792.1.4
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Organism=F.oxysporum_fsp_cepae
Strain=A23_ncbi
OutDir=gene_pred/final_genes/$Organism/$Strain/edited
mkdir -p $OutDir

Assembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)
BrakerGff=$(ls gene_pred/braker/$Organism/&quot;$Strain&quot;_braker/*/augustus.gff3)
CodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)
PGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)

cat $BrakerGff \
| grep -w -v 'g7779' \
| grep -w -v 'g13351' \
| grep -v 'g16038' \
| sed \
's/contig_819\tAUGUSTUS\tgene\t655\t1273\t0.73\t+\t.\tID=g16100;/contig_819\tAUGUSTUS\tgene\t655\t1188\t0.73\t+\t.\tID=g16100;/g' \
| sed \
's/contig_819\tAUGUSTUS\tmRNA\t655\t1273\t0.73\t+\t.\tID=g16100.t1;Parent=g16100/contig_819\tAUGUSTUS\tmRNA\t655\t1188\t0.73\t+\t.\tID=g16100.t1;Parent=g16100/g' \
| sed \
's/contig_819\tAUGUSTUS\tintron\t1002\t1057\t1\t+\t.\tParent=g16100.t1;/contig_819\tAUGUSTUS\tintron\t1002\t1050\t1\t+\t.\tParent=g16100.t1;/g' \
| sed \
's/contig_819\tAUGUSTUS\tCDS\t1058\t1273\t0.74\t+\t0\tID=g16100.t1.CDS4;Parent=g16100.t1/contig_819\tAUGUSTUS\tCDS\t1051\t1188\t0.74\t+\t0\tID=g16100.t1.CDS4;Parent=g16100.t1/g' \
| sed \
's/contig_819\tAUGUSTUS\texon\t1058\t1273\t.\t+\t.\tID=g16100.t1.exon4;Parent=g16100.t1;/contig_819\tAUGUSTUS\texon\t1051\t1188\t.\t+\t.\tID=g16100.t1.exon4;Parent=g16100.t1;/g' \
| sed \
's/contig_819\tAUGUSTUS\tstop_codon\t1271\t1273\t.\t+\t0\tParent=g16100.t1;/contig_819\tAUGUSTUS\tstop_codon\t1186\t1188\t.\t+\t0\tParent=g16100.t1;/g' \
| sed \
's/contig_592\tAUGUSTUS\tgene\t1046\t1683\t0.68\t+\t.\tID=g15699;/contig_592\tAUGUSTUS\tgene\t1046\t1543\t0.68\t+\t.\tID=g15699;/g' \
| sed \
's/contig_592\tAUGUSTUS\tmRNA\t1046\t1683\t0.68\t+\t.\tID=g15699.t1;Parent=g15699/contig_592\tAUGUSTUS\tmRNA\t1046\t1543\t0.68\t+\t.\tID=g15699.t1;Parent=g15699/g' \
| sed \
's/contig_592\tAUGUSTUS\tCDS\t1046\t1539\t0.68\t+\t0\tID=g15699.t1.CDS1;Parent=g15699.t1/contig_592\tAUGUSTUS\tCDS\t1046\t1543\t0.68\t+\t0\tID=g15699.t1.CDS1;Parent=g15699.t1/g' \
| sed \
's/contig_592\tAUGUSTUS\texon\t1046\t1539\t.\t+\t.\tID=g15699.t1.exon1;Parent=g15699.t1;/contig_592\tAUGUSTUS\texon\t1046\t1543\t.\t+\t.\tID=g15699.t1.exon1;Parent=g15699.t1;/g' \
| grep -v 'contig_592.AUGUSTUS.intron.1540.1676.1.+...Parent=g15699.t1;' \
| grep -v 'g15699.t1.CDS2' | grep -v 'g15699.t1.exon2' \
| sed \
's/contig_592\tAUGUSTUS\tstop_codon\t1681\t1683\t.\t+\t0\tParent=g15699.t1;/contig_592\tAUGUSTUS\tstop_codon\t1541\t1543\t.\t+\t0\tParent=g15699.t1;/g' \
&gt; $OutDir/final_genes_Braker_ed.gff3


printf \
&quot;contig_1792\tManual_annotation\tgene\t1\t219\t1\t+\t.\tID=man_1;
contig_1792\tManual_annotation\tmRNA\t1\t219\t1\t+\t.\tID=man_1.t1;Parent=man_1
contig_1792\tManual_annotation\tCDS\t1\t219\t1.0\t+\t0\tID=man_1.t1.CDS1;Parent=man_1.t1
contig_1792\tManual_annotation\texon\t1\t219\t.\t+\t.\tID=man_1.t1.exon1;Parent=man_1.t1;
contig_1792\tManual_annotation\tstop_codon\t217\t219\t.\t+\t.\tParent=man_1.t1;\n&quot; \
&gt;&gt; $OutDir/final_genes_Braker_ed.gff3
printf \
&quot;contig_1016\tManual_annotation\tgene\t2103\t2276\t1\t+\t.\tID=man_2;
contig_1016\tManual_annotation\tmRNA\t2103\t2276\t1\t+\t.\tID=man_2.t1;Parent=man_2
contig_1016\tManual_annotation\tstart_codon\t2103\t2105\t.\t+\t.\tParent=man_2.t1;
contig_1016\tManual_annotation\tCDS\t2103\t2276\t1.0\t+\t0\tID=man_2.t1.CDS1;Parent=man_2.t1
contig_1016\tManual_annotation\texon\t2103\t2276\t.\t+\t.\tID=man_2.t1.exon1;Parent=man_2.t1;
contig_1016\tManual_annotation\tstop_codon\t2274\t2276\t.\t+\t.\tParent=man_2.t1;\n&quot; \
&gt;&gt; $OutDir/final_genes_Braker_ed.gff3

cat $CodingQuaryGff | grep -w -v -e 'CUFF.7204.2.14' -e 'CUFF.7961.1.0' -e 'CUFF.11924.2.127' -e 'CUFF.6792.1.4' &gt; $OutDir/PredictedPass_ed.gff3
cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3" dir="auto">
                     <pre>Organism=F.oxysporum_fsp_cepae
Strain=A23_ncbi
OutDir=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/edited
mkdir -p <span class="pl-smi">$OutDir</span>

Assembly=<span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa<span class="pl-pds">)</span></span>
BrakerGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/braker/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_braker/<span class="pl-k">*</span>/augustus.gff3<span class="pl-pds">)</span></span>
CodingQuaryGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PredictedPass.gff3<span class="pl-pds">)</span></span>
PGNGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PGN_predictedPass.gff3<span class="pl-pds">)</span></span>

cat <span class="pl-smi">$BrakerGff</span> \
<span class="pl-k">|</span> grep -w -v <span class="pl-s"><span class="pl-pds">'</span>g7779<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> grep -w -v <span class="pl-s"><span class="pl-pds">'</span>g13351<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>g16038<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_819\tAUGUSTUS\tgene\t655\t1273\t0.73\t+\t.\tID=g16100;/contig_819\tAUGUSTUS\tgene\t655\t1188\t0.73\t+\t.\tID=g16100;/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_819\tAUGUSTUS\tmRNA\t655\t1273\t0.73\t+\t.\tID=g16100.t1;Parent=g16100/contig_819\tAUGUSTUS\tmRNA\t655\t1188\t0.73\t+\t.\tID=g16100.t1;Parent=g16100/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_819\tAUGUSTUS\tintron\t1002\t1057\t1\t+\t.\tParent=g16100.t1;/contig_819\tAUGUSTUS\tintron\t1002\t1050\t1\t+\t.\tParent=g16100.t1;/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_819\tAUGUSTUS\tCDS\t1058\t1273\t0.74\t+\t0\tID=g16100.t1.CDS4;Parent=g16100.t1/contig_819\tAUGUSTUS\tCDS\t1051\t1188\t0.74\t+\t0\tID=g16100.t1.CDS4;Parent=g16100.t1/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_819\tAUGUSTUS\texon\t1058\t1273\t.\t+\t.\tID=g16100.t1.exon4;Parent=g16100.t1;/contig_819\tAUGUSTUS\texon\t1051\t1188\t.\t+\t.\tID=g16100.t1.exon4;Parent=g16100.t1;/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_819\tAUGUSTUS\tstop_codon\t1271\t1273\t.\t+\t0\tParent=g16100.t1;/contig_819\tAUGUSTUS\tstop_codon\t1186\t1188\t.\t+\t0\tParent=g16100.t1;/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_592\tAUGUSTUS\tgene\t1046\t1683\t0.68\t+\t.\tID=g15699;/contig_592\tAUGUSTUS\tgene\t1046\t1543\t0.68\t+\t.\tID=g15699;/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_592\tAUGUSTUS\tmRNA\t1046\t1683\t0.68\t+\t.\tID=g15699.t1;Parent=g15699/contig_592\tAUGUSTUS\tmRNA\t1046\t1543\t0.68\t+\t.\tID=g15699.t1;Parent=g15699/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_592\tAUGUSTUS\tCDS\t1046\t1539\t0.68\t+\t0\tID=g15699.t1.CDS1;Parent=g15699.t1/contig_592\tAUGUSTUS\tCDS\t1046\t1543\t0.68\t+\t0\tID=g15699.t1.CDS1;Parent=g15699.t1/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_592\tAUGUSTUS\texon\t1046\t1539\t.\t+\t.\tID=g15699.t1.exon1;Parent=g15699.t1;/contig_592\tAUGUSTUS\texon\t1046\t1543\t.\t+\t.\tID=g15699.t1.exon1;Parent=g15699.t1;/g<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>contig_592.AUGUSTUS.intron.1540.1676.1.+...Parent=g15699.t1;<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>g15699.t1.CDS2<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>g15699.t1.exon2<span class="pl-pds">'</span></span> \
<span class="pl-k">|</span> sed \
<span class="pl-s"><span class="pl-pds">'</span>s/contig_592\tAUGUSTUS\tstop_codon\t1681\t1683\t.\t+\t0\tParent=g15699.t1;/contig_592\tAUGUSTUS\tstop_codon\t1541\t1543\t.\t+\t0\tParent=g15699.t1;/g<span class="pl-pds">'</span></span> \
<span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3


<span class="pl-c1">printf</span> \
<span class="pl-s"><span class="pl-pds">"</span>contig_1792\tManual_annotation\tgene\t1\t219\t1\t+\t.\tID=man_1;</span>
<span class="pl-s">contig_1792\tManual_annotation\tmRNA\t1\t219\t1\t+\t.\tID=man_1.t1;Parent=man_1</span>
<span class="pl-s">contig_1792\tManual_annotation\tCDS\t1\t219\t1.0\t+\t0\tID=man_1.t1.CDS1;Parent=man_1.t1</span>
<span class="pl-s">contig_1792\tManual_annotation\texon\t1\t219\t.\t+\t.\tID=man_1.t1.exon1;Parent=man_1.t1;</span>
<span class="pl-s">contig_1792\tManual_annotation\tstop_codon\t217\t219\t.\t+\t.\tParent=man_1.t1;\n<span class="pl-pds">"</span></span> \
<span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3
<span class="pl-c1">printf</span> \
<span class="pl-s"><span class="pl-pds">"</span>contig_1016\tManual_annotation\tgene\t2103\t2276\t1\t+\t.\tID=man_2;</span>
<span class="pl-s">contig_1016\tManual_annotation\tmRNA\t2103\t2276\t1\t+\t.\tID=man_2.t1;Parent=man_2</span>
<span class="pl-s">contig_1016\tManual_annotation\tstart_codon\t2103\t2105\t.\t+\t.\tParent=man_2.t1;</span>
<span class="pl-s">contig_1016\tManual_annotation\tCDS\t2103\t2276\t1.0\t+\t0\tID=man_2.t1.CDS1;Parent=man_2.t1</span>
<span class="pl-s">contig_1016\tManual_annotation\texon\t2103\t2276\t.\t+\t.\tID=man_2.t1.exon1;Parent=man_2.t1;</span>
<span class="pl-s">contig_1016\tManual_annotation\tstop_codon\t2274\t2276\t.\t+\t.\tParent=man_2.t1;\n<span class="pl-pds">"</span></span> \
<span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3

cat <span class="pl-smi">$CodingQuaryGff</span> <span class="pl-k">|</span> grep -w -v -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.7204.2.14<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.7961.1.0<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.11924.2.127<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.6792.1.4<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      FoC 125
                     </h2>
                     <a aria-label="Permalink: FoC 125" class="anchor" href="#foc-125" id="user-content-foc-125">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Note that at this stage all codingquary genes contain . characters rather than _ characters
g16001 - SIX12 - remove gene and use PGN_predictedPass gene
g16002 - SIX7 - remove intron and make stop codon earlier
CUFF_14695 - SIX14 - Replace with manual annotation in correct orientation
Duplicated genes:
CUFF.14630.2.11
CUFF.7213.2.7
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	Organism=F.oxysporum_fsp_cepae
	Strain=125_ncbi
	OutDir=gene_pred/final_genes/$Organism/$Strain/edited
	mkdir -p $OutDir
	Assembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)
	BrakerGff=$(ls gene_pred/braker/$Organism/&quot;$Strain&quot;_braker/*/augustus.gff3)
	CodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)
	PGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)
	cat $BrakerGff \
	| grep -v -w 'g16001' \
	| sed \
	's/contig_714\tAUGUSTUS\tgene\t6151\t6788\t0.93\t+\t.\tID=g16002/contig_714\tAUGUSTUS\tgene\t6151\t6648\t0.93\t+\t.\tID=g16002/g' \
	| sed \
	's/contig_714\tAUGUSTUS\tmRNA\t6151\t6788\t0.93\t+\t.\tID=g16002.t1;Parent=g16002/contig_714\tAUGUSTUS\tmRNA\t6151\t6648\t0.93\t+\t.\tID=g16002.t1;Parent=g16002/g' \
	| sed \
	's/contig_714\tAUGUSTUS\tCDS\t6151\t6644\t0.93\t+\t0\tID=g16002.t1.CDS1;Parent=g16002.t1/contig_714\tAUGUSTUS\tCDS\t6151\t6648\t0.93\t+\t0\tID=g16002.t1.CDS1;Parent=g16002.t1/g' \
	| sed \
	's/contig_714\tAUGUSTUS\texon\t6151\t6644\t.\t+\t.\tID=g16002.t1.exon1;Parent=g16002.t1;/contig_714\tAUGUSTUS\texon\t6151\t6648\t.\t+\t.\tID=g16002.t1.exon1;Parent=g16002.t1;/g' \
	| grep -v -e 'g16002.t1.CDS2' -e 'g16002.t1.exon2' \
	| grep -v 'contig_714.AUGUSTUS.intron.6645.6781.1.+...Parent=g16002.t1;' \
	| sed \
	's/contig_714\tAUGUSTUS\tstop_codon\t6786\t6788\t.\t+\t0\tParent=g16002.t1;/contig_714\tAUGUSTUS\tstop_codon\t6646\t6648\t.\t+\t0\tParent=g16002.t1;/g' \
	&gt; $OutDir/final_genes_Braker_ed.gff3

	printf \
	&quot;contig_2082\tManual_annotation\tgene\t261\t525\t1\t+\t.\tID=man_1;
	contig_2082\tManual_annotation\tmRNA\t261\t525\t1\t+\t.\tID=man_1.t1;Parent=man_1
	contig_2082\tManual_annotation\tstart_codon\t261\t263\t.\t+\t.\tParent=man_1.t1;
	contig_2082\tManual_annotation\tCDS\t261\t525\t1.0\t+\t0\tID=man_1.t1.CDS1;Parent=man_1.t1
	contig_2082\tManual_annotation\texon\t261\t525\t.\t+\t.\tID=man_1.t1.exon1;Parent=man_1.t1;\n&quot; \
	&gt;&gt; $OutDir/final_genes_Braker_ed.gff3
	printf \
	&quot;contig_971\tManual_annotation\tgene\t333\t506\t1\t-\t.\tID=man_2;
	contig_971\tManual_annotation\tmRNA\t333\t506\t1\t-\t.\tID=man_2.t1;Parent=man_2
	contig_971\tManual_annotation\tstop_codon\t333\t335\t.\t-\t.\tParent=man_2.t1;
	contig_971\tManual_annotation\tCDS\t333\t506\t1.0\t-\t0\tID=man_2.t1.CDS1;Parent=man_2.t1
	contig_971\tManual_annotation\texon\t333\t506\t.\t-\t.\tID=man_2.t1.exon1;Parent=man_2.t1;
	contig_971\tManual_annotation\tstart_codon\t504\t506\t.\t-\t.\tParent=man_2.t1;\n&quot; \
	&gt;&gt; $OutDir/final_genes_Braker_ed.gff3

	cat $CodingQuaryGff | grep -v -w -e 'CUFF.14630.2.11' -e	'CUFF.7213.2.7' &gt; $OutDir/PredictedPass_ed.gff3
	cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3" dir="auto">
                     <pre>	Organism=F.oxysporum_fsp_cepae
	Strain=125_ncbi
	OutDir=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/edited
	mkdir -p <span class="pl-smi">$OutDir</span>
	Assembly=<span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa<span class="pl-pds">)</span></span>
	BrakerGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/braker/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_braker/<span class="pl-k">*</span>/augustus.gff3<span class="pl-pds">)</span></span>
	CodingQuaryGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PredictedPass.gff3<span class="pl-pds">)</span></span>
	PGNGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PGN_predictedPass.gff3<span class="pl-pds">)</span></span>
	cat <span class="pl-smi">$BrakerGff</span> \
	<span class="pl-k">|</span> grep -v -w <span class="pl-s"><span class="pl-pds">'</span>g16001<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_714\tAUGUSTUS\tgene\t6151\t6788\t0.93\t+\t.\tID=g16002/contig_714\tAUGUSTUS\tgene\t6151\t6648\t0.93\t+\t.\tID=g16002/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_714\tAUGUSTUS\tmRNA\t6151\t6788\t0.93\t+\t.\tID=g16002.t1;Parent=g16002/contig_714\tAUGUSTUS\tmRNA\t6151\t6648\t0.93\t+\t.\tID=g16002.t1;Parent=g16002/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_714\tAUGUSTUS\tCDS\t6151\t6644\t0.93\t+\t0\tID=g16002.t1.CDS1;Parent=g16002.t1/contig_714\tAUGUSTUS\tCDS\t6151\t6648\t0.93\t+\t0\tID=g16002.t1.CDS1;Parent=g16002.t1/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_714\tAUGUSTUS\texon\t6151\t6644\t.\t+\t.\tID=g16002.t1.exon1;Parent=g16002.t1;/contig_714\tAUGUSTUS\texon\t6151\t6648\t.\t+\t.\tID=g16002.t1.exon1;Parent=g16002.t1;/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>g16002.t1.CDS2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>g16002.t1.exon2<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>contig_714.AUGUSTUS.intron.6645.6781.1.+...Parent=g16002.t1;<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_714\tAUGUSTUS\tstop_codon\t6786\t6788\t.\t+\t0\tParent=g16002.t1;/contig_714\tAUGUSTUS\tstop_codon\t6646\t6648\t.\t+\t0\tParent=g16002.t1;/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3

	<span class="pl-c1">printf</span> \
	<span class="pl-s"><span class="pl-pds">"</span>contig_2082\tManual_annotation\tgene\t261\t525\t1\t+\t.\tID=man_1;</span>
<span class="pl-s">	contig_2082\tManual_annotation\tmRNA\t261\t525\t1\t+\t.\tID=man_1.t1;Parent=man_1</span>
<span class="pl-s">	contig_2082\tManual_annotation\tstart_codon\t261\t263\t.\t+\t.\tParent=man_1.t1;</span>
<span class="pl-s">	contig_2082\tManual_annotation\tCDS\t261\t525\t1.0\t+\t0\tID=man_1.t1.CDS1;Parent=man_1.t1</span>
<span class="pl-s">	contig_2082\tManual_annotation\texon\t261\t525\t.\t+\t.\tID=man_1.t1.exon1;Parent=man_1.t1;\n<span class="pl-pds">"</span></span> \
	<span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3
	<span class="pl-c1">printf</span> \
	<span class="pl-s"><span class="pl-pds">"</span>contig_971\tManual_annotation\tgene\t333\t506\t1\t-\t.\tID=man_2;</span>
<span class="pl-s">	contig_971\tManual_annotation\tmRNA\t333\t506\t1\t-\t.\tID=man_2.t1;Parent=man_2</span>
<span class="pl-s">	contig_971\tManual_annotation\tstop_codon\t333\t335\t.\t-\t.\tParent=man_2.t1;</span>
<span class="pl-s">	contig_971\tManual_annotation\tCDS\t333\t506\t1.0\t-\t0\tID=man_2.t1.CDS1;Parent=man_2.t1</span>
<span class="pl-s">	contig_971\tManual_annotation\texon\t333\t506\t.\t-\t.\tID=man_2.t1.exon1;Parent=man_2.t1;</span>
<span class="pl-s">	contig_971\tManual_annotation\tstart_codon\t504\t506\t.\t-\t.\tParent=man_2.t1;\n<span class="pl-pds">"</span></span> \
	<span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3

	cat <span class="pl-smi">$CodingQuaryGff</span> <span class="pl-k">|</span> grep -v -w -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.14630.2.11<span class="pl-pds">'</span></span> -e	<span class="pl-s"><span class="pl-pds">'</span>CUFF.7213.2.7<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
	cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      FoN N139
                     </h2>
                     <a aria-label="Permalink: FoN N139" class="anchor" href="#fon-n139" id="user-content-fon-n139">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Note that at this stage all codingquary genes contain . characters rather than _ characters
g16592 - SIX10 - remove gene and use PGN_predictedPass gene
g16591 - SIX12 - shorten intron and final CDS
                    </p>
                    <p dir="auto">
                     Duplicated genes:
CUFF.639.2.10
CUFF.7442.1.80 (identified by ncbi)
CUFF.2100.1.11 (identified by ncbi)
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	Organism=F.oxysporum_fsp_narcissi
	Strain=N139_ncbi
	OutDir=gene_pred/final_genes/$Organism/$Strain/edited
	mkdir -p $OutDir

	Assembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)
	BrakerGff=$(ls gene_pred/braker/$Organism/&quot;$Strain&quot;_braker/*/augustus.gff3)
	CodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)
	PGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)

	cat $BrakerGff \
	| grep -v -w 'g16592' \
	| sed \
	's/contig_728\tAUGUSTUS\tgene\t3227\t3658\t0.99\t+\t.\tID=g16591;/contig_728\tAUGUSTUS\tgene\t3317\t3658\t0.99\t+\t.\tID=g16591;/g' \
	| sed \
	's/contig_728\tAUGUSTUS\tmRNA\t3227\t3658\t0.99\t+\t.\tID=g16591.t1;Parent=g16591/contig_728\tAUGUSTUS\tmRNA\t3317\t3658\t0.99\t+\t.\tID=g16591.t1;Parent=g16591/g' \
	| sed \
	's/contig_728\tAUGUSTUS\tstart_codon\t3227\t3229\t.\t+\t0\tParent=g16591.t1;/contig_728\tAUGUSTUS\tstart_codon\t3317\t3319\t.\t+\t0\tParent=g16591.t1;/g' \
	| sed \
	's/contig_728\tAUGUSTUS\tCDS\t3227\t3248\t1\t+\t0\tID=g16591.t1.CDS1;Parent=g16591.t1/contig_728\tAUGUSTUS\tCDS\t3317\t3658\t1\t+\t0\tID=g16591.t1.CDS1;Parent=g16591.t1/g' \
	| sed \
	's/contig_728\tAUGUSTUS\texon\t3227\t3248\t.\t+\t.\tID=g16591.t1.exon1;Parent=g16591.t1;/contig_728\tAUGUSTUS\texon\t3317\t3658\t.\t+\t.\tID=g16591.t1.exon1;Parent=g16591.t1;/g' \
	| grep -v -e 'intron.3249.3299.1.+...Parent=g16591.t1;' \
	| grep -v -e 'g16591.t1.CDS2' -e 'g16591.t1.exon2' \
	&gt; $OutDir/final_genes_Braker_ed.gff3

	cat $CodingQuaryGff | grep -v -w -e 'CUFF.636.2.10' -e 'CUFF.7442.1.80' -e 'CUFF.2100.2.11' &gt; $OutDir/PredictedPass_ed.gff3
	cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3" dir="auto">
                     <pre>	Organism=F.oxysporum_fsp_narcissi
	Strain=N139_ncbi
	OutDir=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/edited
	mkdir -p <span class="pl-smi">$OutDir</span>

	Assembly=<span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa<span class="pl-pds">)</span></span>
	BrakerGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/braker/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_braker/<span class="pl-k">*</span>/augustus.gff3<span class="pl-pds">)</span></span>
	CodingQuaryGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PredictedPass.gff3<span class="pl-pds">)</span></span>
	PGNGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PGN_predictedPass.gff3<span class="pl-pds">)</span></span>

	cat <span class="pl-smi">$BrakerGff</span> \
	<span class="pl-k">|</span> grep -v -w <span class="pl-s"><span class="pl-pds">'</span>g16592<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_728\tAUGUSTUS\tgene\t3227\t3658\t0.99\t+\t.\tID=g16591;/contig_728\tAUGUSTUS\tgene\t3317\t3658\t0.99\t+\t.\tID=g16591;/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_728\tAUGUSTUS\tmRNA\t3227\t3658\t0.99\t+\t.\tID=g16591.t1;Parent=g16591/contig_728\tAUGUSTUS\tmRNA\t3317\t3658\t0.99\t+\t.\tID=g16591.t1;Parent=g16591/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_728\tAUGUSTUS\tstart_codon\t3227\t3229\t.\t+\t0\tParent=g16591.t1;/contig_728\tAUGUSTUS\tstart_codon\t3317\t3319\t.\t+\t0\tParent=g16591.t1;/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_728\tAUGUSTUS\tCDS\t3227\t3248\t1\t+\t0\tID=g16591.t1.CDS1;Parent=g16591.t1/contig_728\tAUGUSTUS\tCDS\t3317\t3658\t1\t+\t0\tID=g16591.t1.CDS1;Parent=g16591.t1/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> sed \
	<span class="pl-s"><span class="pl-pds">'</span>s/contig_728\tAUGUSTUS\texon\t3227\t3248\t.\t+\t.\tID=g16591.t1.exon1;Parent=g16591.t1;/contig_728\tAUGUSTUS\texon\t3317\t3658\t.\t+\t.\tID=g16591.t1.exon1;Parent=g16591.t1;/g<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>intron.3249.3299.1.+...Parent=g16591.t1;<span class="pl-pds">'</span></span> \
	<span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>g16591.t1.CDS2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>g16591.t1.exon2<span class="pl-pds">'</span></span> \
	<span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3

	cat <span class="pl-smi">$CodingQuaryGff</span> <span class="pl-k">|</span> grep -v -w -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.636.2.10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.7442.1.80<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.2100.2.11<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
	cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      For remaining Fo and Fp isolates
                     </h2>
                     <a aria-label="Permalink: For remaining Fo and Fp isolates" class="anchor" href="#for-remaining-fo-and-fp-isolates" id="user-content-for-remaining-fo-and-fp-isolates">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Note that at this stage all codingquary genes contain . characters rather than _ characters
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'ncbi' -e 'Fus2_canu_new' | grep -v -e 'Fus2_canu_new' -e 'A23' -e '125' -e 'N139' | grep -v 'old' | grep 'A8'); do
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
OutDir=gene_pred/final_genes/$Organism/$Strain/edited
mkdir -p $OutDir

BrakerGff=$(ls gene_pred/braker/$Organism/&quot;$Strain&quot;_braker/*/augustus.gff3)
CodingQuaryGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PredictedPass.gff3)
PGNGff=$(ls gene_pred/codingquary/$Organism/$Strain/out/PGN_predictedPass.gff3)
if [[ $Strain == 'A28_ncbi' ]]; then
cp $BrakerGff $OutDir/final_genes_Braker_ed.gff3
cat $CodingQuaryGff | grep -v -w -e 'CUFF.7109.1.50' &gt; $OutDir/PredictedPass_ed.gff3
cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3
elif [[ $Strain == 'CB3_ncbi' ]]; then
cp $BrakerGff $OutDir/final_genes_Braker_ed.gff3
cat $CodingQuaryGff | grep -v -w -e 'CUFF.10490.1.97' &gt; $OutDir/PredictedPass_ed.gff3
cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3
elif [[ $Strain == 'PG_ncbi' ]]; then
cp $BrakerGff $OutDir/final_genes_Braker_ed.gff3
cat $CodingQuaryGff | grep -v -w -e 'CUFF.7109.1.38' -e 'CUFF.11875.1.103' &gt; $OutDir/PredictedPass_ed.gff3
cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3
else
cp $BrakerGff $OutDir/final_genes_Braker_ed.gff3
cp $CodingQuaryGff $OutDir/PredictedPass_ed.gff3
cp $PGNGff $OutDir/PGN_predictedPass_ed.gff3
fi
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/edited
mkdir -p <span class="pl-smi">$OutDir</span>

BrakerGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/braker/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_braker/<span class="pl-k">*</span>/augustus.gff3<span class="pl-pds">)</span></span>
CodingQuaryGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PredictedPass.gff3<span class="pl-pds">)</span></span>
PGNGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/codingquary/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/out/PGN_predictedPass.gff3<span class="pl-pds">)</span></span>
<span class="pl-k">if</span> [[ <span class="pl-smi">$Strain</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">'</span>A28_ncbi<span class="pl-pds">'</span></span> ]]<span class="pl-k">;</span> <span class="pl-k">then</span>
cp <span class="pl-smi">$BrakerGff</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3
cat <span class="pl-smi">$CodingQuaryGff</span> <span class="pl-k">|</span> grep -v -w -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.7109.1.50<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3
<span class="pl-k">elif</span> [[ <span class="pl-smi">$Strain</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">'</span>CB3_ncbi<span class="pl-pds">'</span></span> ]]<span class="pl-k">;</span> <span class="pl-k">then</span>
cp <span class="pl-smi">$BrakerGff</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3
cat <span class="pl-smi">$CodingQuaryGff</span> <span class="pl-k">|</span> grep -v -w -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.10490.1.97<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3
<span class="pl-k">elif</span> [[ <span class="pl-smi">$Strain</span> <span class="pl-k">==</span> <span class="pl-s"><span class="pl-pds">'</span>PG_ncbi<span class="pl-pds">'</span></span> ]]<span class="pl-k">;</span> <span class="pl-k">then</span>
cp <span class="pl-smi">$BrakerGff</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3
cat <span class="pl-smi">$CodingQuaryGff</span> <span class="pl-k">|</span> grep -v -w -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.7109.1.38<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CUFF.11875.1.103<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3
<span class="pl-k">else</span>
cp <span class="pl-smi">$BrakerGff</span> <span class="pl-smi">$OutDir</span>/final_genes_Braker_ed.gff3
cp <span class="pl-smi">$CodingQuaryGff</span> <span class="pl-smi">$OutDir</span>/PredictedPass_ed.gff3
cp <span class="pl-smi">$PGNGff</span> <span class="pl-smi">$OutDir</span>/PGN_predictedPass_ed.gff3
<span class="pl-k">fi</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for EditDir in $(ls -d gene_pred/final_genes/*/*/edited | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'Fus2_canu_new' | grep -e '125' -e 'A23' -e 'N139' | grep 'N139'); do
Strain=$(echo $EditDir | rev | cut -d '/' -f2 | rev)
Organism=$(echo $EditDir | rev | cut -d '/' -f3 | rev)
echo &quot;$Organism - $Strain&quot;
Assembly=$(ls repeat_masked/$Organism/$Strain/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa)
BrakerGff=$EditDir/final_genes_Braker_ed.gff3
CodingQuaryGff=$EditDir/PredictedPass_ed.gff3
PGNGff=$EditDir/PGN_predictedPass_ed.gff3
# ManGff=$EditDir/manual_annotations.gff3
AddDir=$EditDir/additional
FinalDir=gene_pred/final_genes/$Organism/$Strain/final
AddGenesList=$AddDir/additional_genes.txt
AddGenesGff=$AddDir/additional_genes.gff
# FinalGff=$AddDir/combined_genes.gff
mkdir -p $AddDir
mkdir -p $FinalDir

bedtools intersect -v -a $CodingQuaryGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' &gt; $AddGenesList
bedtools intersect -v -a $PGNGff -b $BrakerGff | grep 'gene'| cut -f2 -d'=' | cut -f1 -d';' &gt;&gt; $AddGenesList
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation
$ProgDir/gene_list_to_gff.pl $AddGenesList $CodingQuaryGff CodingQuarry_v2.0 ID CodingQuary &gt; $AddGenesGff
$ProgDir/gene_list_to_gff.pl $AddGenesList $PGNGff PGNCodingQuarry_v2.0 ID CodingQuary &gt;&gt; $AddGenesGff
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary

$ProgDir/add_CodingQuary_features.pl $AddGenesGff $Assembly &gt; $FinalDir/final_genes_CodingQuary.gff3
$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_CodingQuary.gff3 $FinalDir/final_genes_CodingQuary
cp $BrakerGff $FinalDir/final_genes_Braker.gff3
$ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_Braker.gff3 $FinalDir/final_genes_Braker
# cp $ManGff $FinalDir/final_genes_manual.gff3
# $ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_manual.gff3 $FinalDir/final_genes_manual
cat $FinalDir/final_genes_Braker.pep.fasta $FinalDir/final_genes_CodingQuary.pep.fasta | sed -r 's/\*/X/g' &gt; $FinalDir/final_genes_combined.pep.fasta
cat $FinalDir/final_genes_Braker.cdna.fasta $FinalDir/final_genes_CodingQuary.cdna.fasta &gt; $FinalDir/final_genes_combined.cdna.fasta
cat $FinalDir/final_genes_Braker.gene.fasta $FinalDir/final_genes_CodingQuary.gene.fasta &gt; $FinalDir/final_genes_combined.gene.fasta
cat $FinalDir/final_genes_Braker.upstream3000.fasta $FinalDir/final_genes_CodingQuary.upstream3000.fasta &gt; $FinalDir/final_genes_combined.upstream3000.fasta

GffBraker=$FinalDir/final_genes_CodingQuary.gff3
GffQuary=$FinalDir/final_genes_Braker.gff3
# GffManual=$FinalDir/final_genes_manual.gff3
GffAppended=$FinalDir/final_genes_appended.gff3
cat $GffBraker $GffQuary &gt; $GffAppended
# cat $GffBraker $GffQuary $GffManual &gt; $GffAppended

done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">EditDir</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d gene_pred/final_genes/<span class="pl-k">*</span>/<span class="pl-k">*</span>/edited <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$EditDir</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f2 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$EditDir</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
Assembly=<span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa<span class="pl-pds">)</span></span>
BrakerGff=<span class="pl-smi">$EditDir</span>/final_genes_Braker_ed.gff3
CodingQuaryGff=<span class="pl-smi">$EditDir</span>/PredictedPass_ed.gff3
PGNGff=<span class="pl-smi">$EditDir</span>/PGN_predictedPass_ed.gff3
<span class="pl-c"><span class="pl-c">#</span> ManGff=$EditDir/manual_annotations.gff3</span>
AddDir=<span class="pl-smi">$EditDir</span>/additional
FinalDir=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/final
AddGenesList=<span class="pl-smi">$AddDir</span>/additional_genes.txt
AddGenesGff=<span class="pl-smi">$AddDir</span>/additional_genes.gff
<span class="pl-c"><span class="pl-c">#</span> FinalGff=$AddDir/combined_genes.gff</span>
mkdir -p <span class="pl-smi">$AddDir</span>
mkdir -p <span class="pl-smi">$FinalDir</span>

bedtools intersect -v -a <span class="pl-smi">$CodingQuaryGff</span> -b <span class="pl-smi">$BrakerGff</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> cut -f2 -d<span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d<span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$AddGenesList</span>
bedtools intersect -v -a <span class="pl-smi">$PGNGff</span> -b <span class="pl-smi">$BrakerGff</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> cut -f2 -d<span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d<span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$AddGenesList</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation
<span class="pl-smi">$ProgDir</span>/gene_list_to_gff.pl <span class="pl-smi">$AddGenesList</span> <span class="pl-smi">$CodingQuaryGff</span> CodingQuarry_v2.0 ID CodingQuary <span class="pl-k">&gt;</span> <span class="pl-smi">$AddGenesGff</span>
<span class="pl-smi">$ProgDir</span>/gene_list_to_gff.pl <span class="pl-smi">$AddGenesList</span> <span class="pl-smi">$PGNGff</span> PGNCodingQuarry_v2.0 ID CodingQuary <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$AddGenesGff</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/codingquary

<span class="pl-smi">$ProgDir</span>/add_CodingQuary_features.pl <span class="pl-smi">$AddGenesGff</span> <span class="pl-smi">$Assembly</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gff3
<span class="pl-smi">$ProgDir</span>/gff2fasta.pl <span class="pl-smi">$Assembly</span> <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gff3 <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary
cp <span class="pl-smi">$BrakerGff</span> <span class="pl-smi">$FinalDir</span>/final_genes_Braker.gff3
<span class="pl-smi">$ProgDir</span>/gff2fasta.pl <span class="pl-smi">$Assembly</span> <span class="pl-smi">$FinalDir</span>/final_genes_Braker.gff3 <span class="pl-smi">$FinalDir</span>/final_genes_Braker
<span class="pl-c"><span class="pl-c">#</span> cp $ManGff $FinalDir/final_genes_manual.gff3</span>
<span class="pl-c"><span class="pl-c">#</span> $ProgDir/gff2fasta.pl $Assembly $FinalDir/final_genes_manual.gff3 $FinalDir/final_genes_manual</span>
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.pep.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.pep.fasta <span class="pl-k">|</span> sed -r <span class="pl-s"><span class="pl-pds">'</span>s/\*/X/g<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.pep.fasta
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.cdna.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.cdna.fasta <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.cdna.fasta
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.gene.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gene.fasta <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.gene.fasta
cat <span class="pl-smi">$FinalDir</span>/final_genes_Braker.upstream3000.fasta <span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.upstream3000.fasta <span class="pl-k">&gt;</span> <span class="pl-smi">$FinalDir</span>/final_genes_combined.upstream3000.fasta

GffBraker=<span class="pl-smi">$FinalDir</span>/final_genes_CodingQuary.gff3
GffQuary=<span class="pl-smi">$FinalDir</span>/final_genes_Braker.gff3
<span class="pl-c"><span class="pl-c">#</span> GffManual=$FinalDir/final_genes_manual.gff3</span>
GffAppended=<span class="pl-smi">$FinalDir</span>/final_genes_appended.gff3
cat <span class="pl-smi">$GffBraker</span> <span class="pl-smi">$GffQuary</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$GffAppended</span>
<span class="pl-c"><span class="pl-c">#</span> cat $GffBraker $GffQuary $GffManual &gt; $GffAppended</span>

<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Assessing the Gene space in predicted transcriptomes:
                     </h2>
                     <a aria-label="Permalink: Assessing the Gene space in predicted transcriptomes:" class="anchor" href="#assessing-the-gene-space-in-predicted-transcriptomes" id="user-content-assessing-the-gene-space-in-predicted-transcriptomes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for Assembly in $(ls gene_pred/final_genes/*/*/final/final_genes_combined.gene.fasta | grep -e 'Fus2' -e '125' -e 'A23' -e 'A13' -e 'A28' -e 'CB3' -e 'PG' -e 'A8' -e 'N139' | grep -e 'ncbi' -e 'canu_new' | grep -v 'old'); do
		Strain=$(echo $Assembly| rev | cut -d '/' -f3 | rev)
		Organism=$(echo $Assembly | rev | cut -d '/' -f4 | rev)
		echo &quot;$Organism - $Strain&quot;
		ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco
		# BuscoDB=&quot;Fungal&quot;
		BuscoDB=$(ls -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9)
		OutDir=gene_pred/busco/$Organism/$Strain/genes
		qsub $ProgDir/sub_busco2.sh $Assembly $BuscoDB $OutDir
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final/final_genes_combined.gene.fasta <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A13<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A28<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>CB3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>PG<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
		ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/busco
		<span class="pl-c"><span class="pl-c">#</span> BuscoDB="Fungal"</span>
		BuscoDB=<span class="pl-s"><span class="pl-pds">$(</span>ls -d /home/groups/harrisonlab/dbBusco/sordariomyceta_odb9<span class="pl-pds">)</span></span>
		OutDir=gene_pred/busco/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/genes
		qsub <span class="pl-smi">$ProgDir</span>/sub_busco2.sh <span class="pl-smi">$Assembly</span> <span class="pl-smi">$BuscoDB</span> <span class="pl-smi">$OutDir</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	for File in $(ls gene_pred/busco/*/*/genes/*/short_summary_*.txt); do  
		echo $File;
		cat $File | grep -e '(C)' -e 'Total';
	done" dir="auto">
                     <pre>	<span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/busco/<span class="pl-k">*</span>/<span class="pl-k">*</span>/genes/<span class="pl-k">*</span>/short_summary_<span class="pl-k">*</span>.txt<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>  
		<span class="pl-c1">echo</span> <span class="pl-smi">$File</span><span class="pl-k">;</span>
		cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>(C)<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Total<span class="pl-pds">'</span></span><span class="pl-k">;</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      ORF finder
                     </h2>
                     <a aria-label="Permalink: ORF finder" class="anchor" href="#orf-finder" id="user-content-orf-finder">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The genome was searched in six reading frames for any start codon and following
translated identification of a start codon translating sequence until a stop
codon was found. This is based upon the atg.pl script used in paper describing
the P. infestans genome. Additional functionality was added to this script by
also printing ORFs in .gff format.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
	for Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -w -e 'Fus2'); do
		qsub $ProgDir/run_ORF_finder.sh $Genome
	done" dir="auto">
                     <pre>	ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
	<span class="pl-k">for</span> <span class="pl-smi">Genome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -w -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		qsub <span class="pl-smi">$ProgDir</span>/run_ORF_finder.sh <span class="pl-smi">$Genome</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The Gff files from the the ORF finder are not in true Gff3 format. These were
corrected using the following commands:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	ProgDir=~/git_repos/emr_repos/tools/seq_tools/feature_annotation
	for ORF_Gff in $(ls gene_pred/ORF_finder/*/*/*_ORF.gff | grep -v '_F_atg_' | grep -v '_R_atg_'); do
		ORF_Gff_mod=$(echo $ORF_Gff | sed 's/_ORF.gff/_ORF_corrected.gff3/g')
		echo &quot;&quot;
		echo &quot;Correcting the following file:&quot;
		echo $ORF_Gff
		echo &quot;Redirecting to:&quot;
		echo $ORF_Gff_mod
		$ProgDir/gff_corrector.pl $ORF_Gff &gt; $ORF_Gff_mod
	done" dir="auto">
                     <pre>	ProgDir=<span class="pl-k">~</span>/git_repos/emr_repos/tools/seq_tools/feature_annotation
	<span class="pl-k">for</span> <span class="pl-smi">ORF_Gff</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/ORF_finder/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_ORF.gff <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>_F_atg_<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>_R_atg_<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
		ORF_Gff_mod=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$ORF_Gff</span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_ORF.gff/_ORF_corrected.gff3/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
		<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>
		<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Correcting the following file:<span class="pl-pds">"</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$ORF_Gff</span>
		<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Redirecting to:<span class="pl-pds">"</span></span>
		<span class="pl-c1">echo</span> <span class="pl-smi">$ORF_Gff_mod</span>
		<span class="pl-smi">$ProgDir</span>/gff_corrector.pl <span class="pl-smi">$ORF_Gff</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$ORF_Gff_mod</span>
	<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     #Functional annotation
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      A) Interproscan
                     </h2>
                     <a aria-label="Permalink: A) Interproscan" class="anchor" href="#a-interproscan" id="user-content-a-interproscan">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Interproscan was used to give gene models functional annotations.
Annotation was run using the commands below:
                    </p>
                    <p dir="auto">
                     Note: This is a long-running script. As such, these commands were run using
'screen' to allow jobs to be submitted and monitored in the background.
This allows the session to be disconnected and reconnected over time.
                    </p>
                    <p dir="auto">
                     Screen ouput detailing the progress of submission of interporscan jobs
was redirected to a temporary output file named interproscan_submission.log .
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	screen -a
	cd /home/groups/harrisonlab/project_files/fusarium
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan
	for Genes in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139' ); do
	echo $Genes
	$ProgDir/sub_interproscan.sh $Genes
	done 2&gt;&amp;1 | tee -a interproscan_submisison.log" dir="auto">
                     <pre>	screen -a
	<span class="pl-c1">cd</span> /home/groups/harrisonlab/project_files/fusarium
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan
	<span class="pl-k">for</span> <span class="pl-smi">Genes</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span> <span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
	<span class="pl-c1">echo</span> <span class="pl-smi">$Genes</span>
	<span class="pl-smi">$ProgDir</span>/sub_interproscan.sh <span class="pl-smi">$Genes</span>
	<span class="pl-k">done</span> <span class="pl-k">2&gt;&amp;1</span> <span class="pl-k">|</span> tee -a interproscan_submisison.log</pre>
                    </div>
                    <p dir="auto">
                     Following interproscan annotation split files were combined using the following
commands:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan
for Proteins in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139'); do
Strain=$(echo $Proteins | rev | cut -d '/' -f3 | rev)
Organism=$(echo $Proteins | rev | cut -d '/' -f4 | rev)
echo &quot;$Organism - $Strain&quot;
echo $Strain
InterProRaw=gene_pred/interproscan/$Organism/$Strain/raw
$ProgDir/append_interpro.sh $Proteins $InterProRaw
done" dir="auto">
                     <pre>ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan
<span class="pl-k">for</span> <span class="pl-smi">Proteins</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteins</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteins</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Strain</span>
InterProRaw=gene_pred/interproscan/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/raw
<span class="pl-smi">$ProgDir</span>/append_interpro.sh <span class="pl-smi">$Proteins</span> <span class="pl-smi">$InterProRaw</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      B) SwissProt
                     </h2>
                     <a aria-label="Permalink: B) SwissProt" class="anchor" href="#b-swissprot" id="user-content-b-swissprot">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139'); do
Strain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)
OutDir=gene_pred/swissprot/$Organism/$Strain
SwissDbDir=../../uniprot/swissprot
SwissDbName=uniprot_sprot
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot
qsub $ProgDir/sub_swissprot.sh $Proteome $OutDir $SwissDbDir $SwissDbName
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Proteome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=gene_pred/swissprot/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
SwissDbDir=../../uniprot/swissprot
SwissDbName=uniprot_sprot
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot
qsub <span class="pl-smi">$ProgDir</span>/sub_swissprot.sh <span class="pl-smi">$Proteome</span> <span class="pl-smi">$OutDir</span> <span class="pl-smi">$SwissDbDir</span> <span class="pl-smi">$SwissDbName</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for SwissTable in $(ls gene_pred/swissprot/*/*/swissprot_vJul2016_10_hits.tbl | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'ncbi' | grep -e '125' -e 'A23' -e 'N139'); do
# SwissTable=gene_pred/swissprot/Fus2/swissprot_v2015_10_hits.tbl
Strain=$(echo $SwissTable | rev | cut -f2 -d '/' | rev)
Organism=$(echo $SwissTable | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
OutTable=gene_pred/swissprot/$Organism/$Strain/swissprot_vJul2016_tophit_parsed.tbl
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot
$ProgDir/swissprot_parser.py --blast_tbl $SwissTable --blast_db_fasta ../../uniprot/swissprot/uniprot_sprot.fasta &gt; $OutTable
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">SwissTable</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/swissprot/<span class="pl-k">*</span>/<span class="pl-k">*</span>/swissprot_vJul2016_10_hits.tbl <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>125<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c"><span class="pl-c">#</span> SwissTable=gene_pred/swissprot/Fus2/swissprot_v2015_10_hits.tbl</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$SwissTable</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$SwissTable</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutTable=gene_pred/swissprot/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/swissprot_vJul2016_tophit_parsed.tbl
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/swissprot
<span class="pl-smi">$ProgDir</span>/swissprot_parser.py --blast_tbl <span class="pl-smi">$SwissTable</span> --blast_db_fasta ../../uniprot/swissprot/uniprot_sprot.fasta <span class="pl-k">&gt;</span> <span class="pl-smi">$OutTable</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      C) CAZY proteins
                     </h2>
                     <a aria-label="Permalink: C) CAZY proteins" class="anchor" href="#c-cazy-proteins" id="user-content-c-cazy-proteins">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Carbohydrte active enzymes were idnetified using CAZYfollowing recomendations
at
                     <a href="http://csbl.bmb.uga.edu/dbCAN/download/readme.txt" rel="nofollow">
                      http://csbl.bmb.uga.edu/dbCAN/download/readme.txt
                     </a>
                     :
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do
Strain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)
OutDir=gene_pred/CAZY/$Organism/$Strain
mkdir -p $OutDir
Prefix=&quot;$Strain&quot;_CAZY
CazyHmm=../../dbCAN/dbCAN-fam-HMMs.txt
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/HMMER
qsub $ProgDir/sub_hmmscan.sh $CazyHmm $Proteome $Prefix $OutDir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Proteome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=gene_pred/CAZY/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
mkdir -p <span class="pl-smi">$OutDir</span>
Prefix=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY
CazyHmm=../../dbCAN/dbCAN-fam-HMMs.txt
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/HMMER
qsub <span class="pl-smi">$ProgDir</span>/sub_hmmscan.sh <span class="pl-smi">$CazyHmm</span> <span class="pl-smi">$Proteome</span> <span class="pl-smi">$Prefix</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The Hmm parser was used to filter hits by an E-value of E1x10-5 or E1x10-e3 if they had a hit over a length of X %.
                    </p>
                    <p dir="auto">
                     Those proteins with a signal peptide were extracted from the list and gff files
representing these proteins made.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls gene_pred/CAZY/*/*/*CAZY.out.dm | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi'); do
Strain=$(echo $File | rev | cut -f2 -d '/' | rev)
Organism=$(echo $File | rev | cut -f3 -d '/' | rev)
OutDir=$(dirname $File)
echo &quot;$Organism - $Strain&quot;
ProgDir=/home/groups/harrisonlab/dbCAN
$ProgDir/hmmscan-parser.sh $OutDir/&quot;$Strain&quot;_CAZY.out.dm &gt; $OutDir/&quot;$Strain&quot;_CAZY.out.dm.ps
CazyHeaders=$(echo $File | sed 's/.out.dm/_headers.txt/g')
cat $OutDir/&quot;$Strain&quot;_CAZY.out.dm.ps | cut -f3 | sort | uniq &gt; $CazyHeaders
echo &quot;number of CAZY proteins identified:&quot;
cat $CazyHeaders | wc -l
# Gff=$(ls gene_pred/codingquary/$Organism/$Strain/final/final_genes_appended.gff3)
Gff=$(ls gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3)
CazyGff=$OutDir/&quot;$Strain&quot;_CAZY.gff
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
$ProgDir/extract_gff_for_sigP_hits.pl $CazyHeaders $Gff CAZyme ID &gt; $CazyGff
echo &quot;number of CAZY genes identified:&quot;
cat $CazyGff | grep -w 'gene' | wc -l

SecretedProts=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/&quot;$Strain&quot;_final_sp_no_trans_mem.aa)
SecretedHeaders=$(echo $SecretedProts | sed 's/.aa/_headers.txt/g')
cat $SecretedProts | grep '&gt;' | tr -d '&gt;' &gt; $SecretedHeaders
CazyGffSecreted=$OutDir/&quot;$Strain&quot;_CAZY_secreted.gff
$ProgDir/extract_gff_for_sigP_hits.pl $SecretedHeaders $CazyGff Secreted_CAZyme ID &gt; $CazyGffSecreted
echo &quot;number of Secreted CAZY proteins identified:&quot;
cat $CazyGffSecreted | grep -w 'mRNA' | cut -f9 | tr -d 'ID=' | cut -f1 -d ';' &gt; $OutDir/&quot;$Strain&quot;_CAZY_secreted_headers.txt
cat $OutDir/&quot;$Strain&quot;_CAZY_secreted_headers.txt | wc -l
echo &quot;number of Secreted CAZY genes identified:&quot;
cat $CazyGffSecreted | grep -w 'gene' | wc -l
# cat $OutDir/&quot;$Strain&quot;_CAZY_secreted_headers.txt | cut -f1 -d '.' | sort | uniq | wc -l
done &gt; tmp.txt" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/CAZY/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>CAZY.out.dm <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$File</span><span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
ProgDir=/home/groups/harrisonlab/dbCAN
<span class="pl-smi">$ProgDir</span>/hmmscan-parser.sh <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY.out.dm <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY.out.dm.ps
CazyHeaders=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/.out.dm/_headers.txt/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY.out.dm.ps <span class="pl-k">|</span> cut -f3 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">&gt;</span> <span class="pl-smi">$CazyHeaders</span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>number of CAZY proteins identified:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$CazyHeaders</span> <span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> Gff=$(ls gene_pred/codingquary/$Organism/$Strain/final/final_genes_appended.gff3)</span>
Gff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/final/final_genes_appended.gff3<span class="pl-pds">)</span></span>
CazyGff=<span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY.gff
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
<span class="pl-smi">$ProgDir</span>/extract_gff_for_sigP_hits.pl <span class="pl-smi">$CazyHeaders</span> <span class="pl-smi">$Gff</span> CAZyme ID <span class="pl-k">&gt;</span> <span class="pl-smi">$CazyGff</span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>number of CAZY genes identified:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$CazyGff</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l

SecretedProts=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes_signalp-4.1/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_sp_no_trans_mem.aa<span class="pl-pds">)</span></span>
SecretedHeaders=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$SecretedProts</span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/.aa/_headers.txt/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$SecretedProts</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$SecretedHeaders</span>
CazyGffSecreted=<span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY_secreted.gff
<span class="pl-smi">$ProgDir</span>/extract_gff_for_sigP_hits.pl <span class="pl-smi">$SecretedHeaders</span> <span class="pl-smi">$CazyGff</span> Secreted_CAZyme ID <span class="pl-k">&gt;</span> <span class="pl-smi">$CazyGffSecreted</span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>number of Secreted CAZY proteins identified:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$CazyGffSecreted</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f9 <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>ID=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY_secreted_headers.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_CAZY_secreted_headers.txt <span class="pl-k">|</span> wc -l
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>number of Secreted CAZY genes identified:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$CazyGffSecreted</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> cat $OutDir/"$Strain"_CAZY_secreted_headers.txt | cut -f1 -d '.' | sort | uniq | wc -l</span>
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> tmp.txt</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - 125_ncbi
number of CAZY proteins identified:
925
number of CAZY genes identified:
925
number of Secreted CAZY proteins identified:
386
number of Secreted CAZY genes identified:
386
F.oxysporum_fsp_cepae - A13_ncbi
number of CAZY proteins identified:
965
number of CAZY genes identified:
965
number of Secreted CAZY proteins identified:
397
number of Secreted CAZY genes identified:
397
F.oxysporum_fsp_cepae - A23_ncbi
number of CAZY proteins identified:
930
number of CAZY genes identified:
930
number of Secreted CAZY proteins identified:
387
number of Secreted CAZY genes identified:
387
F.oxysporum_fsp_cepae - A28_ncbi
number of CAZY proteins identified:
919
number of CAZY genes identified:
919
number of Secreted CAZY proteins identified:
376
number of Secreted CAZY genes identified:
376
F.oxysporum_fsp_cepae - CB3_ncbi
number of CAZY proteins identified:
909
number of CAZY genes identified:
909
number of Secreted CAZY proteins identified:
383
number of Secreted CAZY genes identified:
383
F.oxysporum_fsp_cepae - Fus2_canu_new
number of CAZY proteins identified:
940
number of CAZY genes identified:
940
number of Secreted CAZY proteins identified:
386
number of Secreted CAZY genes identified:
386
F.oxysporum_fsp_cepae - PG_ncbi
number of CAZY proteins identified:
914
number of CAZY genes identified:
914
number of Secreted CAZY proteins identified:
381
number of Secreted CAZY genes identified:
381
F.oxysporum_fsp_narcissi - N139_ncbi
number of CAZY proteins identified:
961
number of CAZY genes identified:
961
number of Secreted CAZY proteins identified:
399
number of Secreted CAZY genes identified:
399
F.proliferatum - A8_ncbi
number of CAZY proteins identified:
819
number of CAZY genes identified:
819
number of Secreted CAZY proteins identified:
341
number of Secreted CAZY genes identified:
341">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - 125_ncbi
number of CAZY proteins identified:
925
number of CAZY genes identified:
925
number of Secreted CAZY proteins identified:
386
number of Secreted CAZY genes identified:
386
F.oxysporum_fsp_cepae - A13_ncbi
number of CAZY proteins identified:
965
number of CAZY genes identified:
965
number of Secreted CAZY proteins identified:
397
number of Secreted CAZY genes identified:
397
F.oxysporum_fsp_cepae - A23_ncbi
number of CAZY proteins identified:
930
number of CAZY genes identified:
930
number of Secreted CAZY proteins identified:
387
number of Secreted CAZY genes identified:
387
F.oxysporum_fsp_cepae - A28_ncbi
number of CAZY proteins identified:
919
number of CAZY genes identified:
919
number of Secreted CAZY proteins identified:
376
number of Secreted CAZY genes identified:
376
F.oxysporum_fsp_cepae - CB3_ncbi
number of CAZY proteins identified:
909
number of CAZY genes identified:
909
number of Secreted CAZY proteins identified:
383
number of Secreted CAZY genes identified:
383
F.oxysporum_fsp_cepae - Fus2_canu_new
number of CAZY proteins identified:
940
number of CAZY genes identified:
940
number of Secreted CAZY proteins identified:
386
number of Secreted CAZY genes identified:
386
F.oxysporum_fsp_cepae - PG_ncbi
number of CAZY proteins identified:
914
number of CAZY genes identified:
914
number of Secreted CAZY proteins identified:
381
number of Secreted CAZY genes identified:
381
F.oxysporum_fsp_narcissi - N139_ncbi
number of CAZY proteins identified:
961
number of CAZY genes identified:
961
number of Secreted CAZY proteins identified:
399
number of Secreted CAZY genes identified:
399
F.proliferatum - A8_ncbi
number of CAZY proteins identified:
819
number of CAZY genes identified:
819
number of Secreted CAZY proteins identified:
341
number of Secreted CAZY genes identified:
341
</code></pre>
                    </div>
                    <p dir="auto">
                     Note - the CAZY genes identified may need further filtering based on e value and
cuttoff length - see below:
                    </p>
                    <p dir="auto">
                     Cols in yourfile.out.dm.ps:
                    </p>
                    <ol dir="auto">
                     <li>
                      Family HMM
                     </li>
                     <li>
                      HMM length
                     </li>
                     <li>
                      Query ID
                     </li>
                     <li>
                      Query length
                     </li>
                     <li>
                      E-value (how similar to the family HMM)
                     </li>
                     <li>
                      HMM start
                     </li>
                     <li>
                      HMM end
                     </li>
                     <li>
                      Query start
                     </li>
                     <li>
                      Query end
                     </li>
                     <li>
                      Coverage
                     </li>
                    </ol>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       For fungi, use E-value &lt; 1e-17 and coverage &gt; 0.45
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       The best threshold varies for different CAZyme classes (please see
                       <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4132414/" rel="nofollow">
                        http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4132414/
                       </a>
                       for details). Basically to annotate GH proteins, one should use a very relax coverage cutoff or the sensitivity will be low (Supplementary Tables S4 and S9); (ii) to annotate CE families a very stringent E-value cutoff and coverage cutoff should be used; otherwise the precision will be very low due to a very high false positive rate (Supplementary Tables S5 and S10)
                      </p>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Summary of CAZY families by organism
                     </h3>
                     <a aria-label="Permalink: Summary of CAZY families by organism" class="anchor" href="#summary-of-cazy-families-by-organism" id="user-content-summary-of-cazy-families-by-organism">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for CAZY in $(ls gene_pred/CAZY/*/*/*_CAZY.out.dm.ps | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -e 'Fus2' -e 'N139'); do
  Strain=$(echo $CAZY | rev | cut -f2 -d '/' | rev)
  Organism=$(echo $CAZY | rev | cut -f3 -d '/' | rev)
  OutDir=$(dirname $CAZY)
  echo &quot;$Organism - $Strain&quot;
  Secreted=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp_no_trans_mem_headers.txt)
  Gff=gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3
  ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/CAZY
  $ProgDir/summarise_CAZY.py --cazy $CAZY --inp_secreted $Secreted --inp_gff $Gff --summarise_family --trim_gene_id 2 --kubicek_2014
done | less -S" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">CAZY</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/CAZY/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_CAZY.out.dm.ps <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>N139<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
  Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$CAZY</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
  Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$CAZY</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
  OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$CAZY</span><span class="pl-pds">)</span></span>
  <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
  Secreted=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes_signalp-4.1/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>_final_sp_no_trans_mem_headers.txt<span class="pl-pds">)</span></span>
  Gff=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/final/final_genes_appended.gff3
  ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/CAZY
  <span class="pl-smi">$ProgDir</span>/summarise_CAZY.py --cazy <span class="pl-smi">$CAZY</span> --inp_secreted <span class="pl-smi">$Secreted</span> --inp_gff <span class="pl-smi">$Gff</span> --summarise_family --trim_gene_id 2 --kubicek_2014
<span class="pl-k">done</span> <span class="pl-k">|</span> less -S</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - 125_ncbi
AA - 70
GT - 3
CE - 64
CBM - 46
PL - 25
GH - 177
F.oxysporum_fsp_cepae - A13_ncbi
AA - 67
GT - 2
CE - 71
CBM - 51
PL - 24
GH - 182
F.oxysporum_fsp_cepae - A23_ncbi
AA - 69
GT - 3
CE - 67
CBM - 48
PL - 25
GH - 175
F.oxysporum_fsp_cepae - A28_ncbi
AA - 63
GT - 2
CE - 64
CBM - 49
PL - 22
GH - 176
F.oxysporum_fsp_cepae - CB3_ncbi
AA - 64
GT - 3
CE - 69
CBM - 42
PL - 25
GH - 179
F.oxysporum_fsp_cepae - Fus2_canu_new
AA - 69
GT - 3
CE - 66
CBM - 47
PL - 26
GH - 174
F.oxysporum_fsp_cepae - PG_ncbi
AA - 66
GT - 2
CE - 65
CBM - 48
PL - 27
GH - 173
F.oxysporum_fsp_narcissi - N139_ncbi
AA - 68
GT - 1
CE - 66
CBM - 41
PL - 27
GH - 195
F.proliferatum - A8_ncbi
AA - 64
GT - 3
CE - 64
CBM - 28
PL - 26
GH - 156">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - 125_ncbi
AA - 70
GT - 3
CE - 64
CBM - 46
PL - 25
GH - 177
F.oxysporum_fsp_cepae - A13_ncbi
AA - 67
GT - 2
CE - 71
CBM - 51
PL - 24
GH - 182
F.oxysporum_fsp_cepae - A23_ncbi
AA - 69
GT - 3
CE - 67
CBM - 48
PL - 25
GH - 175
F.oxysporum_fsp_cepae - A28_ncbi
AA - 63
GT - 2
CE - 64
CBM - 49
PL - 22
GH - 176
F.oxysporum_fsp_cepae - CB3_ncbi
AA - 64
GT - 3
CE - 69
CBM - 42
PL - 25
GH - 179
F.oxysporum_fsp_cepae - Fus2_canu_new
AA - 69
GT - 3
CE - 66
CBM - 47
PL - 26
GH - 174
F.oxysporum_fsp_cepae - PG_ncbi
AA - 66
GT - 2
CE - 65
CBM - 48
PL - 27
GH - 173
F.oxysporum_fsp_narcissi - N139_ncbi
AA - 68
GT - 1
CE - 66
CBM - 41
PL - 27
GH - 195
F.proliferatum - A8_ncbi
AA - 64
GT - 3
CE - 64
CBM - 28
PL - 26
GH - 156
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - 125_ncbi
B-Galactosidases - 3
A-Galactosidases - 4
Polygalacturonase - 13
A-Arabinosidases - 26
Xylanases - 11
Polygalacturonate lyases - 26
B-Glucuronidases - 4
B-Glycosidases - 13
Cellulases - 20
Xyloglucanases - 1
F.oxysporum_fsp_cepae - A13_ncbi
B-Galactosidases - 3
A-Galactosidases - 3
Polygalacturonase - 14
A-Arabinosidases - 28
Xylanases - 11
Polygalacturonate lyases - 26
B-Glucuronidases - 3
B-Glycosidases - 13
Cellulases - 22
Xyloglucanases - 1
F.oxysporum_fsp_cepae - A23_ncbi
B-Galactosidases - 3
B-Glucuronidases - 4
Polygalacturonase - 13
A-Arabinosidases - 26
Xylanases - 11
Polygalacturonate lyases - 26
A-Galactosidases - 4
B-Glycosidases - 13
Cellulases - 21
Xyloglucanases - 1
F.oxysporum_fsp_cepae - A28_ncbi
B-Galactosidases - 2
A-Galactosidases - 4
Polygalacturonase - 13
A-Arabinosidases - 29
Xylanases - 11
Polygalacturonate lyases - 24
B-Glucuronidases - 4
B-Glycosidases - 13
Cellulases - 21
Xyloglucanases - 1
F.oxysporum_fsp_cepae - CB3_ncbi
B-Galactosidases - 2
B-Glucuronidases - 4
Polygalacturonase - 12
A-Arabinosidases - 28
Xylanases - 10
Polygalacturonate lyases - 26
A-Galactosidases - 4
B-Glycosidases - 14
Cellulases - 21
Xyloglucanases - 1
F.oxysporum_fsp_cepae - Fus2_canu_new
B-Galactosidases - 2
A-Galactosidases - 4
Polygalacturonase - 13
A-Arabinosidases - 26
Xylanases - 11
Polygalacturonate lyases - 27
B-Glucuronidases - 4
B-Glycosidases - 13
Cellulases - 20
Xyloglucanases - 1
F.oxysporum_fsp_cepae - PG_ncbi
B-Galactosidases - 2
A-Galactosidases - 4
Polygalacturonase - 12
A-Arabinosidases - 28
Xylanases - 10
Polygalacturonate lyases - 26
B-Glucuronidases - 4
B-Glycosidases - 12
Cellulases - 20
Xyloglucanases - 1
F.oxysporum_fsp_narcissi - N139_ncbi
B-Galactosidases - 2
B-Glucuronidases - 4
Polygalacturonase - 12
A-Arabinosidases - 28
Xylanases - 11
Polygalacturonate lyases - 27
A-Galactosidases - 4
B-Glycosidases - 17
Cellulases - 20
Xyloglucanases - 1
F.proliferatum - A8_ncbi
B-Galactosidases - 2
A-Galactosidases - 3
Polygalacturonase - 11
A-Arabinosidases - 20
Xylanases - 9
Polygalacturonate lyases - 26
B-Glucuronidases - 2
B-Glycosidases - 10
Cellulases - 21
Xyloglucanases - 1">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - 125_ncbi
B-Galactosidases - 3
A-Galactosidases - 4
Polygalacturonase - 13
A-Arabinosidases - 26
Xylanases - 11
Polygalacturonate lyases - 26
B-Glucuronidases - 4
B-Glycosidases - 13
Cellulases - 20
Xyloglucanases - 1
F.oxysporum_fsp_cepae - A13_ncbi
B-Galactosidases - 3
A-Galactosidases - 3
Polygalacturonase - 14
A-Arabinosidases - 28
Xylanases - 11
Polygalacturonate lyases - 26
B-Glucuronidases - 3
B-Glycosidases - 13
Cellulases - 22
Xyloglucanases - 1
F.oxysporum_fsp_cepae - A23_ncbi
B-Galactosidases - 3
B-Glucuronidases - 4
Polygalacturonase - 13
A-Arabinosidases - 26
Xylanases - 11
Polygalacturonate lyases - 26
A-Galactosidases - 4
B-Glycosidases - 13
Cellulases - 21
Xyloglucanases - 1
F.oxysporum_fsp_cepae - A28_ncbi
B-Galactosidases - 2
A-Galactosidases - 4
Polygalacturonase - 13
A-Arabinosidases - 29
Xylanases - 11
Polygalacturonate lyases - 24
B-Glucuronidases - 4
B-Glycosidases - 13
Cellulases - 21
Xyloglucanases - 1
F.oxysporum_fsp_cepae - CB3_ncbi
B-Galactosidases - 2
B-Glucuronidases - 4
Polygalacturonase - 12
A-Arabinosidases - 28
Xylanases - 10
Polygalacturonate lyases - 26
A-Galactosidases - 4
B-Glycosidases - 14
Cellulases - 21
Xyloglucanases - 1
F.oxysporum_fsp_cepae - Fus2_canu_new
B-Galactosidases - 2
A-Galactosidases - 4
Polygalacturonase - 13
A-Arabinosidases - 26
Xylanases - 11
Polygalacturonate lyases - 27
B-Glucuronidases - 4
B-Glycosidases - 13
Cellulases - 20
Xyloglucanases - 1
F.oxysporum_fsp_cepae - PG_ncbi
B-Galactosidases - 2
A-Galactosidases - 4
Polygalacturonase - 12
A-Arabinosidases - 28
Xylanases - 10
Polygalacturonate lyases - 26
B-Glucuronidases - 4
B-Glycosidases - 12
Cellulases - 20
Xyloglucanases - 1
F.oxysporum_fsp_narcissi - N139_ncbi
B-Galactosidases - 2
B-Glucuronidases - 4
Polygalacturonase - 12
A-Arabinosidases - 28
Xylanases - 11
Polygalacturonate lyases - 27
A-Galactosidases - 4
B-Glycosidases - 17
Cellulases - 20
Xyloglucanases - 1
F.proliferatum - A8_ncbi
B-Galactosidases - 2
A-Galactosidases - 3
Polygalacturonase - 11
A-Arabinosidases - 20
Xylanases - 9
Polygalacturonate lyases - 26
B-Glucuronidases - 2
B-Glycosidases - 10
Cellulases - 21
Xyloglucanases - 1
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      D) AntiSMASH
                     </h2>
                     <a aria-label="Permalink: D) AntiSMASH" class="anchor" href="#d-antismash" id="user-content-d-antismash">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Antismash was run to identify clusters of secondary metabolite genes within
the genome. Antismash was run using the weserver at:
                     <a href="http://antismash.secondarymetabolites.org" rel="nofollow">
                      http://antismash.secondarymetabolites.org
                     </a>
                    </p>
                    <p dir="auto">
                     Results of web-annotation of gene clusters within the assembly were downloaded to
the following directories:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls repeat_masked/*/*/*/*_contigs_softmasked_repeatmasker_TPSI_appended.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'canu_new' -e 'ncbi'); do
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
OutDir=analysis/antismash/$Organism/$Strain
mkdir -p $OutDir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_softmasked_repeatmasker_TPSI_appended.fa <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=analysis/antismash/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Zip in $(ls analysis/antismash/*/*/*.zip | grep -e 'ncbi' | grep 'A23'); do
OutDir=$(dirname $Zip)
unzip -d $OutDir $Zip
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Zip</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/antismash/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>.zip <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$Zip</span><span class="pl-pds">)</span></span>
unzip -d <span class="pl-smi">$OutDir</span> <span class="pl-smi">$Zip</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="# AntiSmash=analysis/antismash/79c1471f-4a2b-41f7-ba36-18ba94675f59/contig_1_pilon.final.gbk
for AntiSmash in $(ls analysis/antismash/*/*/*/*.final.gbk | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'A23'); do
Organism=$(echo $AntiSmash | rev | cut -f4 -d '/' | rev)
Strain=$(echo $AntiSmash | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
OutDir=analysis/antismash/$Organism/$Strain
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/secondary_metabolites
$ProgDir/antismash2gff.py --inp_antismash $AntiSmash &gt; $OutDir/&quot;$Strain&quot;_secondary_metabolite_regions.gff
printf &quot;Number of clusters detected:\t&quot;
cat $OutDir/&quot;$Strain&quot;_secondary_metabolite_regions.gff | grep 'antismash_cluster' | wc -l
# GeneGff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3
GeneGff=gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3
bedtools intersect -u -a $GeneGff -b $OutDir/&quot;$Strain&quot;_secondary_metabolite_regions.gff &gt; $OutDir/metabolite_cluster_genes.gff
cat $OutDir/metabolite_cluster_genes.gff | grep -w 'mRNA' | cut -f9 | cut -f2 -d '=' | cut -f1 -d ';' &gt; $OutDir/metabolite_cluster_gene_headers.txt
printf &quot;Number of predicted proteins in clusters:\t&quot;
cat $OutDir/metabolite_cluster_gene_headers.txt | wc -l
printf &quot;Number of predicted genes in clusters:\t&quot;
cat $OutDir/metabolite_cluster_genes.gff | grep -w 'gene' | wc -l
done" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span> AntiSmash=analysis/antismash/79c1471f-4a2b-41f7-ba36-18ba94675f59/contig_1_pilon.final.gbk</span>
<span class="pl-k">for</span> <span class="pl-smi">AntiSmash</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/antismash/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>.final.gbk <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>A23<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$AntiSmash</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$AntiSmash</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutDir=analysis/antismash/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/secondary_metabolites
<span class="pl-smi">$ProgDir</span>/antismash2gff.py --inp_antismash <span class="pl-smi">$AntiSmash</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_secondary_metabolite_regions.gff
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>Number of clusters detected:\t<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_secondary_metabolite_regions.gff <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>antismash_cluster<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> GeneGff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3</span>
GeneGff=gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/final/final_genes_appended.gff3
bedtools intersect -u -a <span class="pl-smi">$GeneGff</span> -b <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_secondary_metabolite_regions.gff <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/metabolite_cluster_genes.gff
cat <span class="pl-smi">$OutDir</span>/metabolite_cluster_genes.gff <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f9 <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/metabolite_cluster_gene_headers.txt
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>Number of predicted proteins in clusters:\t<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/metabolite_cluster_gene_headers.txt <span class="pl-k">|</span> wc -l
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>Number of predicted genes in clusters:\t<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/metabolite_cluster_genes.gff <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     These clusters represented the following genes. Note that these numbers just
show the number of intersected genes with gff clusters and are not confirmed by
function
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - 125_ncbi
Number of clusters detected:	46
Number of predicted proteins in clusters:	568
Number of predicted genes in clusters:	559
F.oxysporum_fsp_cepae - A13_ncbi
Number of clusters detected:	49
Number of predicted proteins in clusters:	626
Number of predicted genes in clusters:	620
F.oxysporum_fsp_cepae - A23_ncbi
Number of clusters detected:	35
Number of predicted proteins in clusters:	436
Number of predicted genes in clusters:	432
F.oxysporum_fsp_cepae - A28_ncbi
Number of clusters detected:	46
Number of predicted proteins in clusters:	668
Number of predicted genes in clusters:	656
F.oxysporum_fsp_cepae - CB3_ncbi
Number of clusters detected:	44
Number of predicted proteins in clusters:	613
Number of predicted genes in clusters:	604
F.oxysporum_fsp_cepae - Fus2_canu_new
Number of clusters detected:	50
Number of predicted proteins in clusters:	720
Number of predicted genes in clusters:	703
F.oxysporum_fsp_cepae - PG_ncbi
Number of clusters detected:	45
Number of predicted proteins in clusters:	646
Number of predicted genes in clusters:	638
F.oxysporum_fsp_narcissi - N139_ncbi
Number of clusters detected:	46
Number of predicted proteins in clusters:	606
Number of predicted genes in clusters:	601
F.proliferatum - A8_ncbi
Number of clusters detected:	58
Number of predicted proteins in clusters:	762
Number of predicted genes in clusters:	762">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - 125_ncbi
Number of clusters detected:	46
Number of predicted proteins in clusters:	568
Number of predicted genes in clusters:	559
F.oxysporum_fsp_cepae - A13_ncbi
Number of clusters detected:	49
Number of predicted proteins in clusters:	626
Number of predicted genes in clusters:	620
F.oxysporum_fsp_cepae - A23_ncbi
Number of clusters detected:	35
Number of predicted proteins in clusters:	436
Number of predicted genes in clusters:	432
F.oxysporum_fsp_cepae - A28_ncbi
Number of clusters detected:	46
Number of predicted proteins in clusters:	668
Number of predicted genes in clusters:	656
F.oxysporum_fsp_cepae - CB3_ncbi
Number of clusters detected:	44
Number of predicted proteins in clusters:	613
Number of predicted genes in clusters:	604
F.oxysporum_fsp_cepae - Fus2_canu_new
Number of clusters detected:	50
Number of predicted proteins in clusters:	720
Number of predicted genes in clusters:	703
F.oxysporum_fsp_cepae - PG_ncbi
Number of clusters detected:	45
Number of predicted proteins in clusters:	646
Number of predicted genes in clusters:	638
F.oxysporum_fsp_narcissi - N139_ncbi
Number of clusters detected:	46
Number of predicted proteins in clusters:	606
Number of predicted genes in clusters:	601
F.proliferatum - A8_ncbi
Number of clusters detected:	58
Number of predicted proteins in clusters:	762
Number of predicted genes in clusters:	762
</code></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Antismash in $(ls analysis/antismash/F.*/*/*_secondary_metabolite_regions.gff | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'Fus2'); do
	Organism=$(echo $Antismash | rev | cut -f3 -d '/' | rev)
	Strain=$(echo $Antismash | rev | cut -f2 -d '/' | rev)
	echo &quot;$Organism - $Strain&quot;
	cat $Antismash | cut -f3 | sort | uniq -c
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Antismash</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/antismash/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_secondary_metabolite_regions.gff <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
	Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Antismash</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
	Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Antismash</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
	cat <span class="pl-smi">$Antismash</span> <span class="pl-k">|</span> cut -f3 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - 125_ncbi
      3 indole
     11 nrps
      1 nrps-indole
      4 nrps-t1pks
     10 other
      6 t1pks
      1 t3pks
     10 terpene
F.oxysporum_fsp_cepae - A13_ncbi
      3 indole
     12 nrps
      3 nrps-t1pks
     10 other
      9 t1pks
      1 t1pks-nrps
      1 t3pks
     10 terpene
F.oxysporum_fsp_cepae - A23_ncbi
      3 indole
     11 nrps
      1 nrps-indole
      4 nrps-t1pks
     10 other
      6 t1pks
      1 t3pks
     10 terpene
F.oxysporum_fsp_cepae - A28_ncbi
      3 indole
      1 indole-nrps
     11 nrps
      2 nrps-t1pks
     10 other
      8 t1pks
      1 t1pks-nrps
      1 t3pks
      9 terpene
F.oxysporum_fsp_cepae - CB3_ncbi
      1 bacteriocin
      3 indole
      1 indole-nrps
     11 nrps
      3 nrps-t1pks
      9 other
      6 t1pks
      1 t1pks-indole
      1 t3pks
      8 terpene
F.oxysporum_fsp_cepae - Fus2_canu_new
      3 indole
      1 indole-nrps
     11 nrps
      3 nrps-t1pks
     11 other
      7 t1pks
      1 t1pks-nrps
      1 t3pks
     12 terpene
F.oxysporum_fsp_cepae - PG_ncbi
      3 indole
      1 indole-nrps
     11 nrps
      3 nrps-t1pks
     10 other
      5 t1pks
      1 t1pks-indole
      1 t3pks
      9 terpene
      1 terpene-t1pks
F.oxysporum_fsp_narcissi - N139_ncbi
      3 indole
      1 indole-nrps
     13 nrps
      3 nrps-t1pks
     10 other
      6 t1pks
      1 t1pks-terpene
      1 t3pks
      8 terpene
F.proliferatum - A8_ncbi
      1 hserlactone
      3 indole
     13 nrps
      1 nrps-indole
      6 nrps-t1pks
      9 other
     14 t1pks
      1 t1pks-terpene
      1 t3pks
      9 terpene">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - 125_ncbi
      3 indole
     11 nrps
      1 nrps-indole
      4 nrps-t1pks
     10 other
      6 t1pks
      1 t3pks
     10 terpene
F.oxysporum_fsp_cepae - A13_ncbi
      3 indole
     12 nrps
      3 nrps-t1pks
     10 other
      9 t1pks
      1 t1pks-nrps
      1 t3pks
     10 terpene
F.oxysporum_fsp_cepae - A23_ncbi
      3 indole
     11 nrps
      1 nrps-indole
      4 nrps-t1pks
     10 other
      6 t1pks
      1 t3pks
     10 terpene
F.oxysporum_fsp_cepae - A28_ncbi
      3 indole
      1 indole-nrps
     11 nrps
      2 nrps-t1pks
     10 other
      8 t1pks
      1 t1pks-nrps
      1 t3pks
      9 terpene
F.oxysporum_fsp_cepae - CB3_ncbi
      1 bacteriocin
      3 indole
      1 indole-nrps
     11 nrps
      3 nrps-t1pks
      9 other
      6 t1pks
      1 t1pks-indole
      1 t3pks
      8 terpene
F.oxysporum_fsp_cepae - Fus2_canu_new
      3 indole
      1 indole-nrps
     11 nrps
      3 nrps-t1pks
     11 other
      7 t1pks
      1 t1pks-nrps
      1 t3pks
     12 terpene
F.oxysporum_fsp_cepae - PG_ncbi
      3 indole
      1 indole-nrps
     11 nrps
      3 nrps-t1pks
     10 other
      5 t1pks
      1 t1pks-indole
      1 t3pks
      9 terpene
      1 terpene-t1pks
F.oxysporum_fsp_narcissi - N139_ncbi
      3 indole
      1 indole-nrps
     13 nrps
      3 nrps-t1pks
     10 other
      6 t1pks
      1 t1pks-terpene
      1 t3pks
      8 terpene
F.proliferatum - A8_ncbi
      1 hserlactone
      3 indole
     13 nrps
      1 nrps-indole
      6 nrps-t1pks
      9 other
     14 t1pks
      1 t1pks-terpene
      1 t3pks
      9 terpene
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      E) SSCP
                     </h2>
                     <a aria-label="Permalink: E) SSCP" class="anchor" href="#e-sscp" id="user-content-e-sscp">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Small secreted cysteine rich proteins were identified within secretomes. These
proteins may be identified by EffectorP, but this approach allows direct control
over what constitutes a SSCP.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
for Secretome in $(ls gene_pred/final_genes_signalp-4.1/*/*/*_final_sp_no_trans_mem.aa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi'); do
Strain=$(echo $Secretome| rev | cut -f2 -d '/' | rev)
Organism=$(echo $Secretome | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
OutDir=analysis/sscp/$Organism/$Strain
mkdir -p $OutDir
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/sscp
$ProgDir/sscp_filter.py --inp_fasta $Secretome --max_length 300 --threshold 3 --out_fasta $OutDir/&quot;$Strain&quot;_sscp_all_results.fa
cat $OutDir/&quot;$Strain&quot;_sscp_all_results.fa | grep 'Yes' &gt; $OutDir/&quot;$Strain&quot;_sscp.fa
printf &quot;number of SSC-rich genes:\t&quot;
cat $OutDir/&quot;$Strain&quot;_sscp.fa | grep '&gt;' | tr -d '&gt;' | cut -f1 -d '.' | sort | uniq | wc -l
printf &quot;Number of effectors predicted by EffectorP:\t&quot;
EffectorP=$(ls analysis/effectorP/$Organism/$Strain/*_EffectorP_secreted_headers.txt)
cat $EffectorP | wc -l
printf &quot;Number of SSCPs predicted by both effectorP and this approach: \t&quot;
cat $OutDir/&quot;$Strain&quot;_sscp.fa | grep '&gt;' | tr -d '&gt;' &gt; $OutDir/&quot;$Strain&quot;_sscp_headers.txt
cat $OutDir/&quot;$Strain&quot;_sscp_headers.txt $EffectorP | cut -f1 | sort | uniq -d | wc -l
echo &quot;&quot;
done &gt; tmp.txt" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Secretome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes_signalp-4.1/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_final_sp_no_trans_mem.aa <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Secretome</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Secretome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutDir=analysis/sscp/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
mkdir -p <span class="pl-smi">$OutDir</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/sscp
<span class="pl-smi">$ProgDir</span>/sscp_filter.py --inp_fasta <span class="pl-smi">$Secretome</span> --max_length 300 --threshold 3 --out_fasta <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_sscp_all_results.fa
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_sscp_all_results.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>Yes<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_sscp.fa
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>number of SSC-rich genes:\t<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_sscp.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> wc -l
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>Number of effectors predicted by EffectorP:\t<span class="pl-pds">"</span></span>
EffectorP=<span class="pl-s"><span class="pl-pds">$(</span>ls analysis/effectorP/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>_EffectorP_secreted_headers.txt<span class="pl-pds">)</span></span>
cat <span class="pl-smi">$EffectorP</span> <span class="pl-k">|</span> wc -l
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>Number of SSCPs predicted by both effectorP and this approach: \t<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_sscp.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_sscp_headers.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_sscp_headers.txt <span class="pl-smi">$EffectorP</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -d <span class="pl-k">|</span> wc -l
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> tmp.txt</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - 125_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        289
number of SSC-rich genes:       287
Number of effectors predicted by EffectorP:     357
Number of SSCPs predicted by both effectorP and this approach:  221

F.oxysporum_fsp_cepae - A13_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        318
number of SSC-rich genes:       316
Number of effectors predicted by EffectorP:     364
Number of SSCPs predicted by both effectorP and this approach:  237

F.oxysporum_fsp_cepae - A23_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        286
number of SSC-rich genes:       284
Number of effectors predicted by EffectorP:     355
Number of SSCPs predicted by both effectorP and this approach:  218

F.oxysporum_fsp_cepae - A28_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        283
number of SSC-rich genes:       281
Number of effectors predicted by EffectorP:     346
Number of SSCPs predicted by both effectorP and this approach:  214

F.oxysporum_fsp_cepae - CB3_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        279
number of SSC-rich genes:       277
Number of effectors predicted by EffectorP:     357
Number of SSCPs predicted by both effectorP and this approach:  213
F.oxysporum_fsp_cepae - Fus2_canu_new
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        288
number of SSC-rich genes:       286
Number of effectors predicted by EffectorP:     355
Number of SSCPs predicted by both effectorP and this approach:  218

F.oxysporum_fsp_cepae - PG_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        287
number of SSC-rich genes:       285
Number of effectors predicted by EffectorP:     337
Number of SSCPs predicted by both effectorP and this approach:  214

F.oxysporum_fsp_narcissi - N139_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        344
number of SSC-rich genes:       342
Number of effectors predicted by EffectorP:     407
Number of SSCPs predicted by both effectorP and this approach:  263

F.proliferatum - A8_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        197
number of SSC-rich genes:       197
Number of effectors predicted by EffectorP:     258
Number of SSCPs predicted by both effectorP and this approach:  147">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - 125_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        289
number of SSC-rich genes:       287
Number of effectors predicted by EffectorP:     357
Number of SSCPs predicted by both effectorP and this approach:  221

F.oxysporum_fsp_cepae - A13_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        318
number of SSC-rich genes:       316
Number of effectors predicted by EffectorP:     364
Number of SSCPs predicted by both effectorP and this approach:  237

F.oxysporum_fsp_cepae - A23_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        286
number of SSC-rich genes:       284
Number of effectors predicted by EffectorP:     355
Number of SSCPs predicted by both effectorP and this approach:  218

F.oxysporum_fsp_cepae - A28_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        283
number of SSC-rich genes:       281
Number of effectors predicted by EffectorP:     346
Number of SSCPs predicted by both effectorP and this approach:  214

F.oxysporum_fsp_cepae - CB3_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        279
number of SSC-rich genes:       277
Number of effectors predicted by EffectorP:     357
Number of SSCPs predicted by both effectorP and this approach:  213
F.oxysporum_fsp_cepae - Fus2_canu_new
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        288
number of SSC-rich genes:       286
Number of effectors predicted by EffectorP:     355
Number of SSCPs predicted by both effectorP and this approach:  218

F.oxysporum_fsp_cepae - PG_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        287
number of SSC-rich genes:       285
Number of effectors predicted by EffectorP:     337
Number of SSCPs predicted by both effectorP and this approach:  214

F.oxysporum_fsp_narcissi - N139_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        344
number of SSC-rich genes:       342
Number of effectors predicted by EffectorP:     407
Number of SSCPs predicted by both effectorP and this approach:  263

F.proliferatum - A8_ncbi
% cysteine content threshold set to:    3
maximum length set to:  300
No. short-cysteine rich proteins in input fasta:        197
number of SSC-rich genes:       197
Number of effectors predicted by EffectorP:     258
Number of SSCPs predicted by both effectorP and this approach:  147
</code></pre>
                    </div>
                    <p dir="auto">
                     #Genomic analysis
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Comparison to FoL 4287
                     </h2>
                     <a aria-label="Permalink: Comparison to FoL 4287" class="anchor" href="#comparison-to-fol-4287" id="user-content-comparison-to-fol-4287">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     BLast searches were used to identify which genes had homologs on which
chromosomes of the Fusarium lycopersici genome.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="FoLGenomeFa=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum_chromosome_and_additional_contigs.fa
for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'ncbi'); do
# for Proteome in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_proteins.fasta); do
# for Proteome in $(ls gene_pred/external_group/F.oxysporum_fsp_lycopersici/4287/Fusox1/Fusox1_GeneCatalog_proteins_20110522_parsed.fa); do
Strain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
OutDir=analysis/blast_homology/$Organism/$Strain
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub $ProgDir/run_blast2csv.sh $Proteome protein $FoLGenomeFa $OutDir
done" dir="auto">
                     <pre>FoLGenomeFa=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum_chromosome_and_additional_contigs.fa
<span class="pl-k">for</span> <span class="pl-smi">Proteome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c"><span class="pl-c">#</span> for Proteome in $(ls assembly/external_group/F.oxysporum/fo47/broad/fusarium_oxysporum_fo47_1_proteins.fasta); do</span>
<span class="pl-c"><span class="pl-c">#</span> for Proteome in $(ls gene_pred/external_group/F.oxysporum_fsp_lycopersici/4287/Fusox1/Fusox1_GeneCatalog_proteins_20110522_parsed.fa); do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutDir=analysis/blast_homology/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub <span class="pl-smi">$ProgDir</span>/run_blast2csv.sh <span class="pl-smi">$Proteome</span> protein <span class="pl-smi">$FoLGenomeFa</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Convert top blast hits into gff annotations
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for BlastHitsCsv in $(ls analysis/blast_homology/F.*/*/4287_chromosomal_final_genes_combined.pep.fasta_hits.csv | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'ncbi'); do
Organism=$(echo $BlastHitsCsv | rev | cut -f3 -d '/' | rev)
Strain=$(echo $BlastHitsCsv | rev | cut -f2 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
HitsGff=$(echo $BlastHitsCsv | sed  's/.csv/.gff/g')
Column2=&quot;$Strain&quot;_gene_homolog
NumHits=1
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHitsCsv &gt; $HitsGff
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">BlastHitsCsv</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/blast_homology/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/4287_chromosomal_final_genes_combined.pep.fasta_hits.csv <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHitsCsv</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHitsCsv</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
HitsGff=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHitsCsv</span> <span class="pl-k">|</span> sed  <span class="pl-s"><span class="pl-pds">'</span>s/.csv/.gff/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
Column2=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_gene_homolog
NumHits=1
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
<span class="pl-smi">$ProgDir</span>/blast2gff.pl <span class="pl-smi">$Column2</span> <span class="pl-smi">$NumHits</span> <span class="pl-smi">$BlastHitsCsv</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$HitsGff</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Intersecting blast hits with genes from FoL
                     </h4>
                     <a aria-label="Permalink: Intersecting blast hits with genes from FoL" class="anchor" href="#intersecting-blast-hits-with-genes-from-fol" id="user-content-intersecting-blast-hits-with-genes-from-fol">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for HitsGff in $(ls analysis/blast_homology/F.*/*/4287_chromosomal_final_genes_combined.pep.fasta_hits.gff | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'ncbi'); do
Organism=$(echo $HitsGff | rev | cut -f3 -d '/' | rev)
Strain=$(echo $HitsGff| rev | cut -f2 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
HitsDir=$(dirname $HitsGff)
FoLGenes=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum.FO2.31.chr.gff3
FoLIntersect=$HitsDir/4287_chromosomal_final_genes_combined_intersect.bed
bedtools intersect -wo -a $HitsGff -b $FoLGenes &gt; $FoLIntersect
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">HitsGff</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/blast_homology/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/4287_chromosomal_final_genes_combined.pep.fasta_hits.gff <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$HitsGff</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$HitsGff</span><span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
HitsDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$HitsGff</span><span class="pl-pds">)</span></span>
FoLGenes=assembly/external_group/F.oxysporum_fsp_lycopersici/4287_chromosomal/ensembl/Fusarium_oxysporum.FO2.31.chr.gff3
FoLIntersect=<span class="pl-smi">$HitsDir</span>/4287_chromosomal_final_genes_combined_intersect.bed
bedtools intersect -wo -a <span class="pl-smi">$HitsGff</span> -b <span class="pl-smi">$FoLGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$FoLIntersect</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Effector genes
                     </h2>
                     <a aria-label="Permalink: Effector genes" class="anchor" href="#effector-genes" id="user-content-effector-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Putative pathogenicity and effector related genes were identified within Braker
gene models using a number of approaches:
                    </p>
                    <ul dir="auto">
                     <li>
                      A) From Augustus gene models - Identifying secreted proteins
                     </li>
                     <li>
                      B) From Augustus gene models - Effector identification using EffectorP
                     </li>
                     <li>
                      D) From ORF fragments - Signal peptide &amp; RxLR motif
                     </li>
                     <li>
                      E) From ORF fragments - Hmm evidence of WY domains
                     </li>
                     <li>
                      F) From ORF fragments - Hmm evidence of RxLR effectors
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      A) From Augustus gene models - Identifying secreted proteins
                     </h3>
                     <a aria-label="Permalink: A) From Augustus gene models - Identifying secreted proteins" class="anchor" href="#a-from-augustus-gene-models---identifying-secreted-proteins" id="user-content-a-from-augustus-gene-models---identifying-secreted-proteins">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Required programs:
                    </p>
                    <ul dir="auto">
                     <li>
                      SignalP-4.1
                     </li>
                     <li>
                      TMHMM
                     </li>
                    </ul>
                    <p dir="auto">
                     Proteins that were predicted to contain signal peptides were identified using
the following commands:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="SplitfileDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides
CurPath=$PWD
for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do
Strain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)
SplitDir=gene_pred/final_genes_split/$Organism/$Strain
mkdir -p $SplitDir
BaseName=&quot;$Organism&quot;&quot;_$Strain&quot;_final_preds
$SplitfileDir/splitfile_500.py --inp_fasta $Proteome --out_dir $SplitDir --out_base $BaseName
for File in $(ls $SplitDir/*_final_preds_*); do
Jobs=$(qstat | grep 'pred_sigP' | wc -l)
while [ $Jobs -gt 20 ]; do
sleep 10
printf &quot;.&quot;
Jobs=$(qstat | grep 'pred_sigP' | wc -l)
done
printf &quot;\n&quot;
echo $File
qsub $ProgDir/pred_sigP.sh $File signalp-4.1
done
done" dir="auto">
                     <pre>SplitfileDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/signal_peptides
CurPath=<span class="pl-smi">$PWD</span>
<span class="pl-k">for</span> <span class="pl-smi">Proteome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
SplitDir=gene_pred/final_genes_split/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
mkdir -p <span class="pl-smi">$SplitDir</span>
BaseName=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span><span class="pl-pds">"</span><span class="pl-pds">"</span>_<span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_preds
<span class="pl-smi">$SplitfileDir</span>/splitfile_500.py --inp_fasta <span class="pl-smi">$Proteome</span> --out_dir <span class="pl-smi">$SplitDir</span> --out_base <span class="pl-smi">$BaseName</span>
<span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls <span class="pl-smi">$SplitDir</span>/<span class="pl-k">*</span>_final_preds_<span class="pl-k">*</span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>pred_sigP<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">while</span> [ <span class="pl-smi">$Jobs</span> <span class="pl-k">-gt</span> 20 ]<span class="pl-k">;</span> <span class="pl-k">do</span>
sleep 10
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>.<span class="pl-pds">"</span></span>
Jobs=<span class="pl-s"><span class="pl-pds">$(</span>qstat <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>pred_sigP<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-pds">)</span></span>
<span class="pl-k">done</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span>\n<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$File</span>
qsub <span class="pl-smi">$ProgDir</span>/pred_sigP.sh <span class="pl-smi">$File</span> signalp-4.1
<span class="pl-k">done</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The batch files of predicted secreted proteins needed to be combined into a
single file for each strain. This was done with the following commands:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for SplitDir in $(ls -d gene_pred/final_genes_split/*/* | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do
Strain=$(echo $SplitDir | rev |cut -d '/' -f1 | rev)
Organism=$(echo $SplitDir | rev |cut -d '/' -f2 | rev)
InStringAA=''
InStringNeg=''
InStringTab=''
InStringTxt=''
SigpDir=final_genes_signalp-4.1
for GRP in $(ls -l $SplitDir/*_final_preds_*.fa | rev | cut -d '_' -f1 | rev | sort -n); do  
InStringAA=&quot;$InStringAA gene_pred/$SigpDir/$Organism/$Strain/split/&quot;$Organism&quot;_&quot;$Strain&quot;_final_preds_$GRP&quot;&quot;_sp.aa&quot;;  
InStringNeg=&quot;$InStringNeg gene_pred/$SigpDir/$Organism/$Strain/split/&quot;$Organism&quot;_&quot;$Strain&quot;_final_preds_$GRP&quot;&quot;_sp_neg.aa&quot;;  
InStringTab=&quot;$InStringTab gene_pred/$SigpDir/$Organism/$Strain/split/&quot;$Organism&quot;_&quot;$Strain&quot;_final_preds_$GRP&quot;&quot;_sp.tab&quot;;
InStringTxt=&quot;$InStringTxt gene_pred/$SigpDir/$Organism/$Strain/split/&quot;$Organism&quot;_&quot;$Strain&quot;_final_preds_$GRP&quot;&quot;_sp.txt&quot;;  
done
cat $InStringAA &gt; gene_pred/$SigpDir/$Organism/$Strain/&quot;$Strain&quot;_final_sp.aa
cat $InStringNeg &gt; gene_pred/$SigpDir/$Organism/$Strain/&quot;$Strain&quot;_final_neg_sp.aa
tail -n +2 -q $InStringTab &gt; gene_pred/$SigpDir/$Organism/$Strain/&quot;$Strain&quot;_final_sp.tab
cat $InStringTxt &gt; gene_pred/$SigpDir/$Organism/$Strain/&quot;$Strain&quot;_final_sp.txt
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">SplitDir</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -d gene_pred/final_genes_split/<span class="pl-k">*</span>/<span class="pl-k">*</span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$SplitDir</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span>cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f1 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$SplitDir</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span>cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f2 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
InStringAA=<span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>
InStringNeg=<span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>
InStringTab=<span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>
InStringTxt=<span class="pl-s"><span class="pl-pds">'</span><span class="pl-pds">'</span></span>
SigpDir=final_genes_signalp-4.1
<span class="pl-k">for</span> <span class="pl-smi">GRP</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls -l <span class="pl-smi">$SplitDir</span>/<span class="pl-k">*</span>_final_preds_<span class="pl-k">*</span>.fa <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span> -f1 <span class="pl-k">|</span> rev <span class="pl-k">|</span> sort -n<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>  
InStringAA=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$InStringAA</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/split/<span class="pl-pds">"</span></span><span class="pl-smi">$Organism</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span><span class="pl-smi">$Strain</span><span class="pl-s"><span class="pl-pds">"</span>_final_preds_<span class="pl-smi">$GRP</span><span class="pl-pds">"</span><span class="pl-pds">"</span>_sp.aa<span class="pl-pds">"</span></span><span class="pl-k">;</span>  
InStringNeg=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$InStringNeg</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/split/<span class="pl-pds">"</span></span><span class="pl-smi">$Organism</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span><span class="pl-smi">$Strain</span><span class="pl-s"><span class="pl-pds">"</span>_final_preds_<span class="pl-smi">$GRP</span><span class="pl-pds">"</span><span class="pl-pds">"</span>_sp_neg.aa<span class="pl-pds">"</span></span><span class="pl-k">;</span>  
InStringTab=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$InStringTab</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/split/<span class="pl-pds">"</span></span><span class="pl-smi">$Organism</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span><span class="pl-smi">$Strain</span><span class="pl-s"><span class="pl-pds">"</span>_final_preds_<span class="pl-smi">$GRP</span><span class="pl-pds">"</span><span class="pl-pds">"</span>_sp.tab<span class="pl-pds">"</span></span><span class="pl-k">;</span>
InStringTxt=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$InStringTxt</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/split/<span class="pl-pds">"</span></span><span class="pl-smi">$Organism</span><span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span><span class="pl-smi">$Strain</span><span class="pl-s"><span class="pl-pds">"</span>_final_preds_<span class="pl-smi">$GRP</span><span class="pl-pds">"</span><span class="pl-pds">"</span>_sp.txt<span class="pl-pds">"</span></span><span class="pl-k">;</span>  
<span class="pl-k">done</span>
cat <span class="pl-smi">$InStringAA</span> <span class="pl-k">&gt;</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_sp.aa
cat <span class="pl-smi">$InStringNeg</span> <span class="pl-k">&gt;</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_neg_sp.aa
tail -n +2 -q <span class="pl-smi">$InStringTab</span> <span class="pl-k">&gt;</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_sp.tab
cat <span class="pl-smi">$InStringTxt</span> <span class="pl-k">&gt;</span> gene_pred/<span class="pl-smi">$SigpDir</span>/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_sp.txt
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Some proteins that are incorporated into the cell membrane require secretion.
Therefore proteins with a transmembrane domain are not likely to represent
cytoplasmic or apoplastic effectors.
                    </p>
                    <p dir="auto">
                     Proteins containing a transmembrane domain were identified:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do
Strain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/transmembrane_helices
qsub $ProgDir/submit_TMHMM.sh $Proteome
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Proteome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/transmembrane_helices
qsub <span class="pl-smi">$ProgDir</span>/submit_TMHMM.sh <span class="pl-smi">$Proteome</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Those proteins with transmembrane domains were removed from lists of Signal
peptide containing proteins
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls gene_pred/trans_mem/*/*/*_TM_genes_neg.txt | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi' ); do
Strain=$(echo $File | rev | cut -f2 -d '/' | rev)
Organism=$(echo $File | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
TmHeaders=$(echo &quot;$File&quot; | sed 's/neg.txt/neg_headers.txt/g')
cat $File | cut -f1 &gt; $TmHeaders
SigP=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp.aa)
OutDir=$(dirname $SigP)
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
$ProgDir/extract_from_fasta.py --fasta $SigP --headers $TmHeaders &gt; $OutDir/&quot;$Strain&quot;_final_sp_no_trans_mem.aa
echo &quot;Number of SigP proteins:&quot;
cat $SigP | grep '&gt;' | wc -l
echo &quot;Number without transmembrane domains:&quot;
cat $OutDir/&quot;$Strain&quot;_final_sp_no_trans_mem.aa | grep '&gt;' | wc -l
echo &quot;Number of gene models:&quot;
cat $OutDir/&quot;$Strain&quot;_final_sp_no_trans_mem.aa | grep '&gt;' | cut -f1 -d't' | sort | uniq |wc -l
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/trans_mem/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_TM_genes_neg.txt <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
TmHeaders=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$File</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/neg.txt/neg_headers.txt/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">&gt;</span> <span class="pl-smi">$TmHeaders</span>
SigP=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes_signalp-4.1/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>_final_sp.aa<span class="pl-pds">)</span></span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$SigP</span><span class="pl-pds">)</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
<span class="pl-smi">$ProgDir</span>/extract_from_fasta.py --fasta <span class="pl-smi">$SigP</span> --headers <span class="pl-smi">$TmHeaders</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_sp_no_trans_mem.aa
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Number of SigP proteins:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$SigP</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Number without transmembrane domains:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_sp_no_trans_mem.aa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Number of gene models:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_final_sp_no_trans_mem.aa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d<span class="pl-s"><span class="pl-pds">'</span>t<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span>wc -l
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - Fus2_canu_new
Number of SigP proteins:
1793
Number without transmembrane domains:
1475
Number of gene models:
1449
F.oxysporum_fsp_cepae - A13_ncbi
Number of SigP proteins:
1826
Number without transmembrane domains:
1508
F.oxysporum_fsp_cepae - A23_ncbi
Number of SigP proteins:
1752
Number without transmembrane domains:
1439
F.oxysporum_fsp_cepae - A28_ncbi
Number of SigP proteins:
1772
Number without transmembrane domains:
1450
F.oxysporum_fsp_cepae - CB3_ncbi
Number of SigP proteins:
1763
Number without transmembrane domains:
1439
F.oxysporum_fsp_cepae - PG_ncbi
Number of SigP proteins:
1760
Number without transmembrane domains:
1440
F.oxysporum_fsp_narcissi - N139_ncbi
Number of SigP proteins:
1901
Number without transmembrane domains:
1566
F.proliferatum - A8_ncbi
Number of SigP proteins:
1553
Number without transmembrane domains:
1254">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - Fus2_canu_new
Number of SigP proteins:
1793
Number without transmembrane domains:
1475
Number of gene models:
1449
F.oxysporum_fsp_cepae - A13_ncbi
Number of SigP proteins:
1826
Number without transmembrane domains:
1508
F.oxysporum_fsp_cepae - A23_ncbi
Number of SigP proteins:
1752
Number without transmembrane domains:
1439
F.oxysporum_fsp_cepae - A28_ncbi
Number of SigP proteins:
1772
Number without transmembrane domains:
1450
F.oxysporum_fsp_cepae - CB3_ncbi
Number of SigP proteins:
1763
Number without transmembrane domains:
1439
F.oxysporum_fsp_cepae - PG_ncbi
Number of SigP proteins:
1760
Number without transmembrane domains:
1440
F.oxysporum_fsp_narcissi - N139_ncbi
Number of SigP proteins:
1901
Number without transmembrane domains:
1566
F.proliferatum - A8_ncbi
Number of SigP proteins:
1553
Number without transmembrane domains:
1254
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      B) From Augustus gene models - Effector identification using EffectorP
                     </h3>
                     <a aria-label="Permalink: B) From Augustus gene models - Effector identification using EffectorP" class="anchor" href="#b-from-augustus-gene-models---effector-identification-using-effectorp" id="user-content-b-from-augustus-gene-models---effector-identification-using-effectorp">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Required programs:
                    </p>
                    <ul dir="auto">
                     <li>
                      EffectorP.py
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Proteome in $(ls gene_pred/final_genes/F.*/*/*/final_genes_combined.pep.fasta | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep 'Fus2_canu_new'); do
Strain=$(echo $Proteome | rev | cut -f3 -d '/' | rev)
Organism=$(echo $Proteome | rev | cut -f4 -d '/' | rev)
BaseName=&quot;$Organism&quot;_&quot;$Strain&quot;_EffectorP
OutDir=analysis/effectorP/$Organism/$Strain
ProgDir=~/git_repos/emr_repos/tools/seq_tools/feature_annotation/fungal_effectors
qsub $ProgDir/pred_effectorP.sh $Proteome $BaseName $OutDir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Proteome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/final_genes_combined.pep.fasta <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Proteome</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
BaseName=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span><span class="pl-pds">"</span></span>_<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_EffectorP
OutDir=analysis/effectorP/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
ProgDir=<span class="pl-k">~</span>/git_repos/emr_repos/tools/seq_tools/feature_annotation/fungal_effectors
qsub <span class="pl-smi">$ProgDir</span>/pred_effectorP.sh <span class="pl-smi">$Proteome</span> <span class="pl-smi">$BaseName</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Those genes that were predicted as secreted and tested positive by effectorP
were identified:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls analysis/effectorP/*/*/*_EffectorP.txt | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'ncbi'); do
Strain=$(echo $File | rev | cut -f2 -d '/' | rev)
Organism=$(echo $File | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
Headers=$(echo &quot;$File&quot; | sed 's/_EffectorP.txt/_EffectorP_headers.txt/g')
cat $File | grep 'Effector' | cut -f1 &gt; $Headers
Secretome=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp_no_trans_mem.aa)
OutFile=$(echo &quot;$File&quot; | sed 's/_EffectorP.txt/_EffectorP_secreted.aa/g')
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
$ProgDir/extract_from_fasta.py --fasta $Secretome --headers $Headers &gt; $OutFile
OutFileHeaders=$(echo &quot;$File&quot; | sed 's/_EffectorP.txt/_EffectorP_secreted_headers.txt/g')
cat $OutFile | grep '&gt;' | tr -d '&gt;' &gt; $OutFileHeaders
cat $OutFileHeaders | wc -l
Gff=$(ls gene_pred/final_genes/$Organism/$Strain/*/final_genes_appended.gff3)
EffectorP_Gff=$(echo &quot;$File&quot; | sed 's/_EffectorP.txt/_EffectorP_secreted.gff/g')
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
$ProgDir/extract_gff_for_sigP_hits.pl $OutFileHeaders $Gff effectorP ID &gt; $EffectorP_Gff
cat $EffectorP_Gff | grep -w 'gene' | wc -l
done &gt; tmp.txt" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/effectorP/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_EffectorP.txt <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
Headers=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$File</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_EffectorP.txt/_EffectorP_headers.txt/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>Effector<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">&gt;</span> <span class="pl-smi">$Headers</span>
Secretome=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes_signalp-4.1/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>_final_sp_no_trans_mem.aa<span class="pl-pds">)</span></span>
OutFile=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$File</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_EffectorP.txt/_EffectorP_secreted.aa/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
<span class="pl-smi">$ProgDir</span>/extract_from_fasta.py --fasta <span class="pl-smi">$Secretome</span> --headers <span class="pl-smi">$Headers</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutFile</span>
OutFileHeaders=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$File</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_EffectorP.txt/_EffectorP_secreted_headers.txt/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$OutFile</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutFileHeaders</span>
cat <span class="pl-smi">$OutFileHeaders</span> <span class="pl-k">|</span> wc -l
Gff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>/final_genes_appended.gff3<span class="pl-pds">)</span></span>
EffectorP_Gff=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$File</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_EffectorP.txt/_EffectorP_secreted.gff/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
<span class="pl-smi">$ProgDir</span>/extract_gff_for_sigP_hits.pl <span class="pl-smi">$OutFileHeaders</span> <span class="pl-smi">$Gff</span> effectorP ID <span class="pl-k">&gt;</span> <span class="pl-smi">$EffectorP_Gff</span>
cat <span class="pl-smi">$EffectorP_Gff</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> tmp.txt</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - Fus2_canu_new
355
355
F.oxysporum_fsp_cepae - A13_ncbi
364
364
F.oxysporum_fsp_cepae - A23_ncbi
355
355
F.oxysporum_fsp_cepae - A28_ncbi
346
346
F.oxysporum_fsp_cepae - CB3_ncbi
357
357
F.oxysporum_fsp_cepae - PG_ncbi
337
337
F.oxysporum_fsp_narcissi - N139_ncbi
407
407
F.proliferatum - A8_ncbi
258
258">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - Fus2_canu_new
355
355
F.oxysporum_fsp_cepae - A13_ncbi
364
364
F.oxysporum_fsp_cepae - A23_ncbi
355
355
F.oxysporum_fsp_cepae - A28_ncbi
346
346
F.oxysporum_fsp_cepae - CB3_ncbi
357
357
F.oxysporum_fsp_cepae - PG_ncbi
337
337
F.oxysporum_fsp_narcissi - N139_ncbi
407
407
F.proliferatum - A8_ncbi
258
258
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      C) Identification of MIMP-flanking genes
                     </h3>
                     <a aria-label="Permalink: C) Identification of MIMP-flanking genes" class="anchor" href="#c-identification-of-mimp-flanking-genes" id="user-content-c-identification-of-mimp-flanking-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Genome in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'old' | grep -v 'ncbi' ); do
Organism=$(echo &quot;$Genome&quot; | rev | cut -d '/' -f4 | rev)
Strain=$(echo &quot;$Genome&quot; | rev | cut -d '/' -f3 | rev)
BrakerGff=$(ls gene_pred/final_genes/$Organism/&quot;$Strain&quot;/final/final_genes_CodingQuary.gff3)
QuaryGff=$(ls gene_pred/final_genes/$Organism/&quot;$Strain&quot;/final/final_genes_Braker.gff3)
OutDir=analysis/mimps/$Organism/$Strain
mkdir -p &quot;$OutDir&quot;
echo &quot;$Organism - $Strain&quot;
ProgDir=&quot;/home/armita/git_repos/emr_repos/tools/pathogen/mimp_finder&quot;
$ProgDir/mimp_finder.pl $Genome $OutDir/&quot;$Strain&quot;_mimps.fa $OutDir/&quot;$Strain&quot;_mimps.gff &gt; $OutDir/&quot;$Strain&quot;_mimps.log
$ProgDir/gffexpander.pl +- 2000 $OutDir/&quot;$Strain&quot;_mimps.gff &gt; $OutDir/&quot;$Strain&quot;_mimps_exp.gff
echo &quot;The number of mimps identified:&quot;
cat $OutDir/&quot;$Strain&quot;_mimps.fa | grep '&gt;' | wc -l
bedtools intersect -u -a $BrakerGff -b $OutDir/&quot;$Strain&quot;_mimps_exp.gff &gt; $OutDir/&quot;$Strain&quot;_genes_in_2kb_mimp.gff
bedtools intersect -u -a $QuaryGff -b $OutDir/&quot;$Strain&quot;_mimps_exp.gff &gt;&gt; $OutDir/&quot;$Strain&quot;_genes_in_2kb_mimp.gff
echo &quot;The following transcripts intersect mimps:&quot;
MimpProtsTxt=$OutDir/&quot;$Strain&quot;_prots_in_2kb_mimp.txt
MimpGenesTxt=$OutDir/&quot;$Strain&quot;_genes_in_2kb_mimp.txt
cat $OutDir/&quot;$Strain&quot;_genes_in_2kb_mimp.gff | grep -w 'mRNA' | cut -f9 | cut -f1 -d';' | cut -f2 -d'=' | sort | uniq &gt; $MimpProtsTxt
cat $OutDir/&quot;$Strain&quot;_genes_in_2kb_mimp.gff | grep -w 'mRNA' | cut -f9 | cut -f1 -d';' | cut -f2 -d'=' | cut -f1 -d '.'| sort | uniq &gt; $MimpGenesTxt
cat $MimpProtsTxt | wc -l
cat $MimpGenesTxt | wc -l
echo &quot;&quot;
done &gt; tmp.txt" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Genome</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Genome</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f4 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Genome</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> -f3 <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
BrakerGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>/final/final_genes_CodingQuary.gff3<span class="pl-pds">)</span></span>
QuaryGff=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>/final/final_genes_Braker.gff3<span class="pl-pds">)</span></span>
OutDir=analysis/mimps/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
mkdir -p <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$OutDir</span><span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
ProgDir=<span class="pl-s"><span class="pl-pds">"</span>/home/armita/git_repos/emr_repos/tools/pathogen/mimp_finder<span class="pl-pds">"</span></span>
<span class="pl-smi">$ProgDir</span>/mimp_finder.pl <span class="pl-smi">$Genome</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps.fa <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps.gff <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps.log
<span class="pl-smi">$ProgDir</span>/gffexpander.pl +- 2000 <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps.gff <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps_exp.gff
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>The number of mimps identified:<span class="pl-pds">"</span></span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps.fa <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
bedtools intersect -u -a <span class="pl-smi">$BrakerGff</span> -b <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps_exp.gff <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_genes_in_2kb_mimp.gff
bedtools intersect -u -a <span class="pl-smi">$QuaryGff</span> -b <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_mimps_exp.gff <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_genes_in_2kb_mimp.gff
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>The following transcripts intersect mimps:<span class="pl-pds">"</span></span>
MimpProtsTxt=<span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_prots_in_2kb_mimp.txt
MimpGenesTxt=<span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_genes_in_2kb_mimp.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_genes_in_2kb_mimp.gff <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f9 <span class="pl-k">|</span> cut -f1 -d<span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f2 -d<span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">&gt;</span> <span class="pl-smi">$MimpProtsTxt</span>
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_genes_in_2kb_mimp.gff <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f9 <span class="pl-k">|</span> cut -f1 -d<span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f2 -d<span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span><span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">&gt;</span> <span class="pl-smi">$MimpGenesTxt</span>
cat <span class="pl-smi">$MimpProtsTxt</span> <span class="pl-k">|</span> wc -l
cat <span class="pl-smi">$MimpGenesTxt</span> <span class="pl-k">|</span> wc -l
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> tmp.txt</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	F.oxysporum_fsp_cepae - Fus2_canu_new
	The number of mimps identified:
	153
	The following transcripts intersect mimps:
	157
	155

	F.oxysporum_fsp_cepae - 125_ncbi
	The number of mimps identified:
	140
	The following transcripts intersect mimps:
	95
	95

	F.oxysporum_fsp_cepae - A13_ncbi
	The number of mimps identified:
	55
	The following transcripts intersect mimps:
	49
	49

	F.oxysporum_fsp_cepae - A23_ncbi
	The number of mimps identified:
	136
	The following transcripts intersect mimps:
	88
	88

	F.oxysporum_fsp_cepae - A28_ncbi
	The number of mimps identified:
	35
	The following transcripts intersect mimps:
	36
	36

	F.oxysporum_fsp_cepae - CB3_ncbi
	The number of mimps identified:
	30
	The following transcripts intersect mimps:
	32
	32

	F.oxysporum_fsp_cepae - PG_ncbi
	The number of mimps identified:
	51
	The following transcripts intersect mimps:
	43
	43

	F.oxysporum_fsp_narcissi - N139_ncbi
	The number of mimps identified:
	207
	The following transcripts intersect mimps:
	120
	119

	F.proliferatum - A8_ncbi
	The number of mimps identified:
	6
	The following transcripts intersect mimps:
	11
	11">
                     <pre class="notranslate"><code>	F.oxysporum_fsp_cepae - Fus2_canu_new
	The number of mimps identified:
	153
	The following transcripts intersect mimps:
	157
	155

	F.oxysporum_fsp_cepae - 125_ncbi
	The number of mimps identified:
	140
	The following transcripts intersect mimps:
	95
	95

	F.oxysporum_fsp_cepae - A13_ncbi
	The number of mimps identified:
	55
	The following transcripts intersect mimps:
	49
	49

	F.oxysporum_fsp_cepae - A23_ncbi
	The number of mimps identified:
	136
	The following transcripts intersect mimps:
	88
	88

	F.oxysporum_fsp_cepae - A28_ncbi
	The number of mimps identified:
	35
	The following transcripts intersect mimps:
	36
	36

	F.oxysporum_fsp_cepae - CB3_ncbi
	The number of mimps identified:
	30
	The following transcripts intersect mimps:
	32
	32

	F.oxysporum_fsp_cepae - PG_ncbi
	The number of mimps identified:
	51
	The following transcripts intersect mimps:
	43
	43

	F.oxysporum_fsp_narcissi - N139_ncbi
	The number of mimps identified:
	207
	The following transcripts intersect mimps:
	120
	119

	F.proliferatum - A8_ncbi
	The number of mimps identified:
	6
	The following transcripts intersect mimps:
	11
	11
</code></pre>
                    </div>
                    <p dir="auto">
                     Those genes that were predicted as secreted and within 2Kb of a MIMP
were identified:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls analysis/mimps/*/*/*_genes_in_2kb_mimp.txt | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi'| grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'old' | grep 'ncbi' ); do
Strain=$(echo $File | rev | cut -f2 -d '/' | rev | sed 's/_chromosomal//g')
Organism=$(echo $File | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
ProtsFile=$(echo $File | sed 's/genes/prots/g')
Secretome=$(ls gene_pred/final_genes_signalp-4.1/$Organism/$Strain/*_final_sp_no_trans_mem.aa)
OutFile=$(echo &quot;$File&quot; | sed 's/.gff/_secreted.gff/g')
SecretedHeaders=$(echo &quot;$Secretome&quot; | sed 's/.aa/_headers.txt/g')
cat $Secretome | grep '&gt;' | tr -d '&gt;' | sed 's/-p.//g' &gt; $SecretedHeaders
cat $ProtsFile $SecretedHeaders | cut -f1 | sort | uniq -d | wc -l
cat $SecretedHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $File | wc -l
# ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder
# $ProgDir/extract_gff_for_sigP_hits.pl $SecretedHeaders $File secreted_mimp ID &gt; $OutFile
# cat $OutFile | grep -w 'mRNA' | wc -l
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/mimps/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_genes_in_2kb_mimp.txt <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/_chromosomal//g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
ProtsFile=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$File</span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/genes/prots/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
Secretome=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes_signalp-4.1/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/<span class="pl-k">*</span>_final_sp_no_trans_mem.aa<span class="pl-pds">)</span></span>
OutFile=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$File</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/.gff/_secreted.gff/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
SecretedHeaders=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Secretome</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/.aa/_headers.txt/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$Secretome</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/-p.//g<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$SecretedHeaders</span>
cat <span class="pl-smi">$ProtsFile</span> <span class="pl-smi">$SecretedHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -d <span class="pl-k">|</span> wc -l
cat <span class="pl-smi">$SecretedHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> grep -f <span class="pl-smi">$File</span> <span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> ProgDir=/home/armita/git_repos/emr_repos/tools/gene_prediction/ORF_finder</span>
<span class="pl-c"><span class="pl-c">#</span> $ProgDir/extract_gff_for_sigP_hits.pl $SecretedHeaders $File secreted_mimp ID &gt; $OutFile</span>
<span class="pl-c"><span class="pl-c">#</span> cat $OutFile | grep -w 'mRNA' | wc -l</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - Fus2_canu_new
31
31
F.oxysporum_fsp_cepae - 125_ncbi
24
26
F.oxysporum_fsp_cepae - A13_ncbi
9
9
F.oxysporum_fsp_cepae - A23_ncbi
20
20
F.oxysporum_fsp_cepae - A28_ncbi
5
5
F.oxysporum_fsp_cepae - CB3_ncbi
1
1
F.oxysporum_fsp_cepae - PG_ncbi
3
3
F.oxysporum_fsp_narcissi - N139_ncbi
31
31
F.proliferatum - A8_ncbi
1
1">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - Fus2_canu_new
31
31
F.oxysporum_fsp_cepae - 125_ncbi
24
26
F.oxysporum_fsp_cepae - A13_ncbi
9
9
F.oxysporum_fsp_cepae - A23_ncbi
20
20
F.oxysporum_fsp_cepae - A28_ncbi
5
5
F.oxysporum_fsp_cepae - CB3_ncbi
1
1
F.oxysporum_fsp_cepae - PG_ncbi
3
3
F.oxysporum_fsp_narcissi - N139_ncbi
31
31
F.proliferatum - A8_ncbi
1
1
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      C)
                     </h2>
                     <a aria-label="Permalink: C)" class="anchor" href="#c" id="user-content-c">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Secreted proteins with a N-terminal [SG]-P-C-[KR]-P motif
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for File in $(ls gene_pred/final_genes_signalp-4.1/F.*/*/*_final_sp_no_trans_mem.aa | grep -e 'Fus2_canu_new' -e 'ncbi'); do
echo $File;
cat $File | grep -B1 &quot;[SG]PC[KR]P&quot; | grep '&gt;' | cut -f1 | tr -d '&gt;' | wc -l;
done
cat gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa | grep -B1 &quot;[SG]PC[KR]P&quot; | grep '&gt;' | cut -f1 | tr -d '&gt;' &gt; tmp.txt
cat analysis/expression/Fus2_expressed_genes.tsv | grep -w -f 'tmp.txt' &gt; tmp.tsv" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">File</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes_signalp-4.1/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_final_sp_no_trans_mem.aa <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c1">echo</span> <span class="pl-smi">$File</span><span class="pl-k">;</span>
cat <span class="pl-smi">$File</span> <span class="pl-k">|</span> grep -B1 <span class="pl-s"><span class="pl-pds">"</span>[SG]PC[KR]P<span class="pl-pds">"</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l<span class="pl-k">;</span>
<span class="pl-k">done</span>
cat gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa <span class="pl-k">|</span> grep -B1 <span class="pl-s"><span class="pl-pds">"</span>[SG]PC[KR]P<span class="pl-pds">"</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> tmp.txt
cat analysis/expression/Fus2_expressed_genes.tsv <span class="pl-k">|</span> grep -w -f <span class="pl-s"><span class="pl-pds">'</span>tmp.txt<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> tmp.tsv</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/125_ncbi/125_ncbi_final_sp_no_trans_mem.aa
43
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A13_ncbi/A13_ncbi_final_sp_no_trans_mem.aa
50
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A23_ncbi/A23_ncbi_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A28_ncbi/A28_ncbi_final_sp_no_trans_mem.aa
45
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/CB3_ncbi/CB3_ncbi_final_sp_no_trans_mem.aa
46
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/PG_ncbi/PG_ncbi_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_narcissi/N139_ncbi/N139_ncbi_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.proliferatum/A8_ncbi/A8_ncbi_final_sp_no_trans_mem.aa
40">
                     <pre class="notranslate"><code>gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/125_ncbi/125_ncbi_final_sp_no_trans_mem.aa
43
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A13_ncbi/A13_ncbi_final_sp_no_trans_mem.aa
50
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A23_ncbi/A23_ncbi_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/A28_ncbi/A28_ncbi_final_sp_no_trans_mem.aa
45
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/CB3_ncbi/CB3_ncbi_final_sp_no_trans_mem.aa
46
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_cepae/PG_ncbi/PG_ncbi_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.oxysporum_fsp_narcissi/N139_ncbi/N139_ncbi_final_sp_no_trans_mem.aa
42
gene_pred/final_genes_signalp-4.1/F.proliferatum/A8_ncbi/A8_ncbi_final_sp_no_trans_mem.aa
40
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      4. Genomic analysis
                     </h1>
                     <a aria-label="Permalink: 4. Genomic analysis" class="anchor" href="#4-genomic-analysis" id="user-content-4-genomic-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.1 Chracterisation of LS regions
                     </h2>
                     <a aria-label="Permalink: 4.1 Chracterisation of LS regions" class="anchor" href="#41-chracterisation-of-ls-regions" id="user-content-41-chracterisation-of-ls-regions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     See pathogen/identifying_LS_chromosomes/characterising_LS_contigs.md
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4.2 Orthology
                     </h2>
                     <a aria-label="Permalink: 4.2 Orthology" class="anchor" href="#42-orthology" id="user-content-42-orthology">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Orthomcl was used to identify orthologous groups between Fusarium spp. genomes
                    </p>
                    <p dir="auto">
                     Genomes were grouped by subspecies and orthology was determined within each
subspecies group. Orthology was also determined between subspecies groups.
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        Pathogenic
                       </th>
                       <th>
                        non-pathogenic
                       </th>
                       <th>
                        Intermediate
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        125
                       </td>
                       <td>
                        A28
                       </td>
                       <td>
                        55
                       </td>
                      </tr>
                      <tr>
                       <td>
                        A23
                       </td>
                       <td>
                        D2
                       </td>
                       <td>
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Fus2
                       </td>
                       <td>
                        PG
                       </td>
                       <td>
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4.2.c) Orthology between pathogenic and non-pathogenic isolates
                     </h3>
                     <a aria-label="Permalink: 4.2.c) Orthology between pathogenic and non-pathogenic isolates" class="anchor" href="#42c-orthology-between-pathogenic-and-non-pathogenic-isolates" id="user-content-42c-orthology-between-pathogenic-and-non-pathogenic-isolates">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The Commands used to run this analysis are shown in
pathogen/orthology/F.oxysporum_fsp.cepae_pathogen_vs_non-pathogen_orthology.md
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.  Searches
                     </h2>
                     <a aria-label="Permalink: 5.  Searches" class="anchor" href="#5--searches" id="user-content-5--searches">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.1.A) Identifying SIX genes
                     </h2>
                     <a aria-label="Permalink: 5.1.A) Identifying SIX genes" class="anchor" href="#51a-identifying-six-genes" id="user-content-51a-identifying-six-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Protein sequence of previously characterised SIX genes used to BLAST against
assemblies.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -w -v 'Fus2_canu_new' | grep 'ncbi' | grep -v 'old'); do
echo $Assembly
Query=analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX.fa
qsub $ProgDir/blast_pipe.sh $Query dna $Assembly
done" dir="auto">
                     <pre>ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
<span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -w -v <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Assembly</span>
Query=analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX.fa
qsub <span class="pl-smi">$ProgDir</span>/blast_pipe.sh <span class="pl-smi">$Query</span> dna <span class="pl-smi">$Assembly</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Once blast searches had completed, the BLAST hits were converted to GFF
annotations:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for BlastHits in $(ls analysis/blast_homology/*/*/*Fo_path_genes_CRX.fa_homologs.csv | grep -v -w 'Fus2_canu_new' | grep 'ncbi'); do
Strain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)
Organism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
HitsGff=$(echo $BlastHits | sed  's/.csv/.gff/g')
Column2=SIX_homolog
NumHits=12
$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHits &gt; $HitsGff
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">BlastHits</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/blast_homology/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>Fo_path_genes_CRX.fa_homologs.csv <span class="pl-k">|</span> grep -v -w <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
HitsGff=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> sed  <span class="pl-s"><span class="pl-pds">'</span>s/.csv/.gff/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
Column2=SIX_homolog
NumHits=12
<span class="pl-smi">$ProgDir</span>/blast2gff.pl <span class="pl-smi">$Column2</span> <span class="pl-smi">$NumHits</span> <span class="pl-smi">$BlastHits</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$HitsGff</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="The blast hits were summarised in a single table for all the genomes. The top
identity of the top blast hit in relation to the enquire query sequence was
presented for each blast hit.">
                     <pre class="notranslate"><code>The blast hits were summarised in a single table for all the genomes. The top
identity of the top blast hit in relation to the enquire query sequence was
presented for each blast hit.
</code></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="OutFile=analysis/blast_homology/Fo_path_genes_CRX_summary.tab
echo &quot;Organism&quot; &gt; tmp2.tab
cat analysis/blast_homology/F.proliferatum/A8/A8_Fo_path_genes_CRX.fa_homologs.csv | cut -f1 &gt;&gt; tmp2.tab
for BlastHits in $(ls analysis/blast_homology/*/*/*Fo_path_genes_CRX.fa_homologs.csv | grep -v -w -e 'Fus2' -e 'Fus2_edited_v1' -e 'Fus2_edited_v2' -e 'Fus2_canu' -e 'Fus2_3' -e '4287' | grep -e 'Fus2_canu_new' -e 'ncbi'); do
Strain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)
Organism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism&quot; &gt; tmp.tab
echo &quot;$Strain&quot; &gt;&gt; tmp.tab
cat $BlastHits | cut -f10,19 | tail -n +2 | sed -r &quot;s/\t/,/g&quot; &gt;&gt; tmp.tab
paste tmp2.tab tmp.tab &gt; $OutFile
cp $OutFile tmp2.tab
done
rm tmp.tab
rm tmp2.tab" dir="auto">
                     <pre>OutFile=analysis/blast_homology/Fo_path_genes_CRX_summary.tab
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>Organism<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> tmp2.tab
cat analysis/blast_homology/F.proliferatum/A8/A8_Fo_path_genes_CRX.fa_homologs.csv <span class="pl-k">|</span> cut -f1 <span class="pl-k">&gt;&gt;</span> tmp2.tab
<span class="pl-k">for</span> <span class="pl-smi">BlastHits</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/blast_homology/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>Fo_path_genes_CRX.fa_homologs.csv <span class="pl-k">|</span> grep -v -w -e <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_edited_v1<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_edited_v2<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_3<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>4287<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> tmp.tab
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;&gt;</span> tmp.tab
cat <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> cut -f10,19 <span class="pl-k">|</span> tail -n +2 <span class="pl-k">|</span> sed -r <span class="pl-s"><span class="pl-pds">"</span>s/\t/,/g<span class="pl-pds">"</span></span> <span class="pl-k">&gt;&gt;</span> tmp.tab
paste tmp2.tab tmp.tab <span class="pl-k">&gt;</span> <span class="pl-smi">$OutFile</span>
cp <span class="pl-smi">$OutFile</span> tmp2.tab
<span class="pl-k">done</span>
rm tmp.tab
rm tmp2.tab</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for HitsGff in $(ls analysis/blast_homology/*/*/*Fo_path_genes_CRX.fa_homologs.gff | grep -v 'trinity' | grep -e 'Fus2_canu_new' -e 'ncbi'); do
Strain=$(echo $HitsGff | rev | cut -f2 -d '/' | rev)
Organism=$(echo $HitsGff | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
GffAppended=$(ls gene_pred/final_genes/$Organism/&quot;$Strain&quot;*/final/final_genes_appended.gff3)
OutDir=$(dirname $HitsGff)
SixIntersect=$OutDir/&quot;$Strain&quot;_Fo_path_genes_CRX.fa_hit_genes.bed
bedtools intersect -wao -a $HitsGff -b $GffAppended &gt; $SixIntersect
bedtools intersect -wao -a $HitsGff -b $GffAppended | cut -f9,18 | grep -v 'Parent'
echo &quot;&quot;
done | tr -d ';' | tr -d '&quot;' | sed 's/ID=//g'&gt; analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX_hit_genes_summary.tab" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">HitsGff</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/blast_homology/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>Fo_path_genes_CRX.fa_homologs.gff <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>trinity<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$HitsGff</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$HitsGff</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
GffAppended=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span><span class="pl-k">*</span>/final/final_genes_appended.gff3<span class="pl-pds">)</span></span>
OutDir=<span class="pl-s"><span class="pl-pds">$(</span>dirname <span class="pl-smi">$HitsGff</span><span class="pl-pds">)</span></span>
SixIntersect=<span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_Fo_path_genes_CRX.fa_hit_genes.bed
bedtools intersect -wao -a <span class="pl-smi">$HitsGff</span> -b <span class="pl-smi">$GffAppended</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$SixIntersect</span>
bedtools intersect -wao -a <span class="pl-smi">$HitsGff</span> -b <span class="pl-smi">$GffAppended</span> <span class="pl-k">|</span> cut -f9,18 <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>Parent<span class="pl-pds">'</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>
<span class="pl-k">done</span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>"<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/ID=//g<span class="pl-pds">'</span></span><span class="pl-k">&gt;</span> analysis/blast_homology/Fo_path_genes/Fo_path_genes_CRX_hit_genes_summary.tab</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="F.oxysporum_fsp_cepae - Fus2_canu_new
C5_Seq45_BlastHit_1     g13425
CRX1_Seq49_BlastHit_1   g15739
CRX1_Seq49_BlastHit_2   g11056
CRX1_Seq49_BlastHit_3   g59
CRX1_Seq49_BlastHit_4   g16971
CRX2_Seq50_BlastHit_1   g11056
CRX2_Seq50_BlastHit_2   g15739
CRX2_Seq50_BlastHit_3   g59
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_1  g15724
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_2  g17045
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_3  CUFF_4289_1_31
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_4  CUFF_327_1_65
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_5  g12255
MIMP_g29_ex.19GJ26_H12-1.seq_BlastHit_1 g16905
MIMP_g16_ex.19GJ26_B07-1.seq_BlastHit_1 g16179
MIMP_g41_ex.45DG31-1.seq_BlastHit_1     g17144
MIMP_g41_ex.45DG31-1.seq_BlastHit_2     NS_02766
Fusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_1 g16847
Fusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_2 g16190
Fusarium_oxysporum_f._sp._lycopersici_isolate_BFOL-51_secreted_in_xylem_5_(SIX5)_gene,_partial_cds_BlastHit_1   g16849
Fusarium_oxysporum_f._sp._lycopersici_secreted_in_xylem_Six7_(SIX7)_mRNA,_complete_cds_BlastHit_1       CUFF_4344_1_33
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_1   g16273
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_2   g13574
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six10_(SIX10)_mRNA,_complete_cds_BlastHit_1 g17153
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six12_(SIX12)_mRNA,_complete_cds_BlastHit_1 PGN_CUFF_4345_1_2
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six14_(SIX14)_mRNA,_complete_cds_BlastHit_1 man_1">
                     <pre class="notranslate"><code>F.oxysporum_fsp_cepae - Fus2_canu_new
C5_Seq45_BlastHit_1     g13425
CRX1_Seq49_BlastHit_1   g15739
CRX1_Seq49_BlastHit_2   g11056
CRX1_Seq49_BlastHit_3   g59
CRX1_Seq49_BlastHit_4   g16971
CRX2_Seq50_BlastHit_1   g11056
CRX2_Seq50_BlastHit_2   g15739
CRX2_Seq50_BlastHit_3   g59
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_1  g15724
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_2  g17045
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_3  CUFF_4289_1_31
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_4  CUFF_327_1_65
MIMP_39_ex.19GJ26_B10-1.seq_BlastHit_5  g12255
MIMP_g29_ex.19GJ26_H12-1.seq_BlastHit_1 g16905
MIMP_g16_ex.19GJ26_B07-1.seq_BlastHit_1 g16179
MIMP_g41_ex.45DG31-1.seq_BlastHit_1     g17144
MIMP_g41_ex.45DG31-1.seq_BlastHit_2     NS_02766
Fusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_1 g16847
Fusarium_oxysporum_f._sp._lycopersici_isolate_FOL-MM10_secreted_in_xylem_3_(SIX3)_gene,_complete_cds_BlastHit_2 g16190
Fusarium_oxysporum_f._sp._lycopersici_isolate_BFOL-51_secreted_in_xylem_5_(SIX5)_gene,_partial_cds_BlastHit_1   g16849
Fusarium_oxysporum_f._sp._lycopersici_secreted_in_xylem_Six7_(SIX7)_mRNA,_complete_cds_BlastHit_1       CUFF_4344_1_33
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_1   g16273
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six9_(SIX9)_mRNA,_complete_cds_BlastHit_2   g13574
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six10_(SIX10)_mRNA,_complete_cds_BlastHit_1 g17153
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six12_(SIX12)_mRNA,_complete_cds_BlastHit_1 PGN_CUFF_4345_1_2
Fusarium_oxysporum_f._sp._lycopersici_strain_Fol007_Six14_(SIX14)_mRNA,_complete_cds_BlastHit_1 man_1
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.1.B) Identifying FTF genes
                     </h2>
                     <a aria-label="Permalink: 5.1.B) Identifying FTF genes" class="anchor" href="#51b-identifying-ftf-genes" id="user-content-51b-identifying-ftf-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Previously published FTF genes from Sanchez et al 2016 were blasted against
Fusarium genomes.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -v -e 'HB17' | grep -v -w 'Fus2_canu_new' | grep 'A8_ncbi'); do
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
echo $Assembly
Query=analysis/blast_homology/Fo_path_genes/FTF_cds_Sanchez_et_al_2016.fasta
OutDir=analysis/FTF/$Organism/$Strain
mkdir -p $OutDir
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub $ProgDir/run_blast2csv.sh $Query dna $Assembly $OutDir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v -w <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>A8_ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Assembly</span>
Query=analysis/blast_homology/Fo_path_genes/FTF_cds_Sanchez_et_al_2016.fasta
OutDir=analysis/FTF/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
mkdir -p <span class="pl-smi">$OutDir</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub <span class="pl-smi">$ProgDir</span>/run_blast2csv.sh <span class="pl-smi">$Query</span> dna <span class="pl-smi">$Assembly</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     BLAST hits were converted to Gff annotations and intersected with gene models:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for BlastHits in $(ls analysis/FTF/*/*/*_FTF_cds_Sanchez_et_al_2016.fasta_hits.csv | grep -v -w 'Fus2_canu_new' | grep 'A8_ncbi'); do
Strain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)
Organism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)
OutDir=analysis/FTF/$Organism/$Strain
HitsGff=$(echo $BlastHits | sed  's/.csv/.gff/g')
Column2=FTF_homolog
NumHits=1
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHits &gt; $HitsGff

GffAppended=$(ls gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3)
bedtools intersect -wao -a $HitsGff -b $GffAppended &gt; $OutDir/&quot;$Strain&quot;_FTF_hits_intersected.bed
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">BlastHits</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/FTF/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_FTF_cds_Sanchez_et_al_2016.fasta_hits.csv <span class="pl-k">|</span> grep -v -w <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>A8_ncbi<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
OutDir=analysis/FTF/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
HitsGff=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> sed  <span class="pl-s"><span class="pl-pds">'</span>s/.csv/.gff/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
Column2=FTF_homolog
NumHits=1
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
<span class="pl-smi">$ProgDir</span>/blast2gff.pl <span class="pl-smi">$Column2</span> <span class="pl-smi">$NumHits</span> <span class="pl-smi">$BlastHits</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$HitsGff</span>

GffAppended=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/final/final_genes_appended.gff3<span class="pl-pds">)</span></span>
bedtools intersect -wao -a <span class="pl-smi">$HitsGff</span> -b <span class="pl-smi">$GffAppended</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_FTF_hits_intersected.bed
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     Fus2 genes g16859 and g10474 were identified as FTF1 and FTF2 homologs.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat $OutDir/&quot;$Strain&quot;_FTF_hits_intersected.bed | grep -w 'gene' | cut -f18 | cut -f1 -d ';' | sort | uniq" dir="auto">
                     <pre>cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_FTF_hits_intersected.bed <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f18 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq</pre>
                    </div>
                    <p dir="auto">
                     The orthogorups that these genes belonged to was identified:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	cat analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt | grep -e 'Fus2|g16859.t' -e 'Fus2|g10474.t'" dir="auto">
                     <pre>	cat analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2|g16859.t<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>Fus2|g10474.t<span class="pl-pds">'</span></span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="orthogroup730: 4287|FOXG_17458-t26_1-p1 4287|FOXG_14422-t26_1-p1 4287|FOXG_15059-t26_1-p1 A13|g9338.t1 4287|FOXG_09390-t26_1-p1 A28|g1088.t1 fo47|FOZG_12687T0 PG|g9482.t1 CB3|g6501.t1 Fus2|g10474.t1 A23|g3017.t1 fo47|FOZG_18011T0 PG|g14993.t1 125|g2599.t1 125|g16201.t1 Fus2|g16859.t1 Fus2|NS_02884.t1 A23|g16099.t1 4287|FOXG_17123-t26_1-p1 4287|FOXG_17084-t26_1-p1 4287|FOXG_14000-t26_1-p1 4287|FOXG_12589-t26_1-p1 4287|FOXG_12539-t26_1-p1 4287|FOXG_16414-t26_1-p1 4287|FOXG_14257-t26_1-p1 A13|g15237.t1">
                     <pre class="notranslate"><code>orthogroup730: 4287|FOXG_17458-t26_1-p1 4287|FOXG_14422-t26_1-p1 4287|FOXG_15059-t26_1-p1 A13|g9338.t1 4287|FOXG_09390-t26_1-p1 A28|g1088.t1 fo47|FOZG_12687T0 PG|g9482.t1 CB3|g6501.t1 Fus2|g10474.t1 A23|g3017.t1 fo47|FOZG_18011T0 PG|g14993.t1 125|g2599.t1 125|g16201.t1 Fus2|g16859.t1 Fus2|NS_02884.t1 A23|g16099.t1 4287|FOXG_17123-t26_1-p1 4287|FOXG_17084-t26_1-p1 4287|FOXG_14000-t26_1-p1 4287|FOXG_12589-t26_1-p1 4287|FOXG_12539-t26_1-p1 4287|FOXG_16414-t26_1-p1 4287|FOXG_14257-t26_1-p1 A13|g15237.t1
</code></pre>
                    </div>
                    <p dir="auto">
                     This identified orthogroup730 as containing FTF1 &amp; FTF2.
                    </p>
                    <p dir="auto">
                     9 Transcription factor genes (TF) have been identified in FoL that potentially
regulate SIX gene expression. The genes in FoL can be used to identify orthologous
FoC genes:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="TF_list=&quot;FOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274&quot;
num=0
OrthoTxt=$(ls analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt)
AnnotTab=$(ls gene_pred/annotations/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_gene_annotations.tab)
for TF in $TF_list; do
num=$(($num +1))
echo &quot;TF&quot;$num
Orthogroup=$(cat $OrthoTxt | grep -w &quot;$TF&quot; | cut -f1 -d ':')
cat $AnnotTab | grep -w &quot;$Orthogroup&quot;
done &gt; analysis/transcription_factors/Fus2_canu_new_TF1-9.tsv" dir="auto">
                     <pre>TF_list=<span class="pl-s"><span class="pl-pds">"</span>FOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274<span class="pl-pds">"</span></span>
num=0
OrthoTxt=<span class="pl-s"><span class="pl-pds">$(</span>ls analysis/orthology/orthomcl/FoC_vs_Fo_vs_FoL_publication_ncbi/FoC_vs_Fo_vs_FoL_publication_ncbi_orthogroups.txt<span class="pl-pds">)</span></span>
AnnotTab=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/annotations/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_gene_annotations.tab<span class="pl-pds">)</span></span>
<span class="pl-k">for</span> <span class="pl-smi">TF</span> <span class="pl-k">in</span> <span class="pl-smi">$TF_list</span><span class="pl-k">;</span> <span class="pl-k">do</span>
num=<span class="pl-s"><span class="pl-pds">$((</span><span class="pl-smi">$num</span> <span class="pl-k">+</span><span class="pl-c1">1</span><span class="pl-pds">))</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>TF<span class="pl-pds">"</span></span><span class="pl-smi">$num</span>
Orthogroup=<span class="pl-s"><span class="pl-pds">$(</span>cat <span class="pl-smi">$OrthoTxt</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$TF</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>:<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$AnnotTab</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Orthogroup</span><span class="pl-pds">"</span></span>
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> analysis/transcription_factors/Fus2_canu_new_TF1-9.tsv</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="TF_list=&quot;FOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274&quot;
num=0
OrthoTxt=$(ls analysis/orthology/orthomcl/Fp_Fv_FoC_FoL_Fo/Fp_Fv_FoC_FoL_Fo_orthogroups.txt)
AnnotTab=$(ls gene_pred/annotations/F.proliferatum/A8_ncbi/A8_ncbi_gene_annotations.tab)
for TF in $TF_list; do
num=$(($num +1))
echo &quot;TF&quot;$num
Orthogroup=$(cat $OrthoTxt | grep -w &quot;$TF&quot; | cut -f1 -d ':')
cat $AnnotTab | grep -w &quot;$Orthogroup&quot;
done &gt; analysis/transcription_factors/Fp_TF1-9.tsv" dir="auto">
                     <pre>TF_list=<span class="pl-s"><span class="pl-pds">"</span>FOXG_14257 FOXG_17260 FOXG_17266 FOXG_14201 FOXG_14230 FOXG_14211 FOXG_14275 FOXG_14277 FOXG_14274<span class="pl-pds">"</span></span>
num=0
OrthoTxt=<span class="pl-s"><span class="pl-pds">$(</span>ls analysis/orthology/orthomcl/Fp_Fv_FoC_FoL_Fo/Fp_Fv_FoC_FoL_Fo_orthogroups.txt<span class="pl-pds">)</span></span>
AnnotTab=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/annotations/F.proliferatum/A8_ncbi/A8_ncbi_gene_annotations.tab<span class="pl-pds">)</span></span>
<span class="pl-k">for</span> <span class="pl-smi">TF</span> <span class="pl-k">in</span> <span class="pl-smi">$TF_list</span><span class="pl-k">;</span> <span class="pl-k">do</span>
num=<span class="pl-s"><span class="pl-pds">$((</span><span class="pl-smi">$num</span> <span class="pl-k">+</span><span class="pl-c1">1</span><span class="pl-pds">))</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>TF<span class="pl-pds">"</span></span><span class="pl-smi">$num</span>
Orthogroup=<span class="pl-s"><span class="pl-pds">$(</span>cat <span class="pl-smi">$OrthoTxt</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$TF</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>:<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
cat <span class="pl-smi">$AnnotTab</span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Orthogroup</span><span class="pl-pds">"</span></span>
<span class="pl-k">done</span> <span class="pl-k">&gt;</span> analysis/transcription_factors/Fp_TF1-9.tsv</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.1.C) Identifying Mitochondrial genes in assemblies
                     </h2>
                     <a aria-label="Permalink: 5.1.C) Identifying Mitochondrial genes in assemblies" class="anchor" href="#51c-identifying-mitochondrial-genes-in-assemblies" id="user-content-51c-identifying-mitochondrial-genes-in-assemblies">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Previously published Fusarium spp. Mitcochondial genes from Al-Reedy 2012 were blasted against
assembled genomes.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls repeat_masked/*/*/*/*_contigs_unmasked.fa | grep -v -e 'HB17' | grep -w 'Fus2_canu_new'); do
Organism=$(echo $Assembly | rev | cut -f4 -d '/' | rev)
Strain=$(echo $Assembly | rev | cut -f3 -d '/' | rev)
echo $Assembly
Query=analysis/blast_homology/Mt_genes/F.spp._mt_prots_Al-Reedy_et_al._2012.fasta
OutDir=analysis/Mt_genes/$Organism/$Strain
mkdir -p $OutDir
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub $ProgDir/run_blast2csv.sh $Query protein $Assembly $OutDir
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa <span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -w <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f4 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$Assembly</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Assembly</span>
Query=analysis/blast_homology/Mt_genes/F.spp._mt_prots_Al-Reedy_et_al._2012.fasta
OutDir=analysis/Mt_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
mkdir -p <span class="pl-smi">$OutDir</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub <span class="pl-smi">$ProgDir</span>/run_blast2csv.sh <span class="pl-smi">$Query</span> protein <span class="pl-smi">$Assembly</span> <span class="pl-smi">$OutDir</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     BLAST hits were converted to Gff annotations and intersected with gene models:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for BlastHits in $(ls analysis/Mt_genes/*/*/*_F.spp._mt_prots_Al-Reedy_et_al._2012.fasta_hits.csv | grep -v -e 'HB17' | grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep 'A8'); do
Strain=$(echo $BlastHits | rev | cut -f2 -d '/' | rev)
Organism=$(echo $BlastHits | rev | cut -f3 -d '/' | rev)
echo &quot;$Organism - $Strain&quot;
OutDir=analysis/Mt_genes/$Organism/$Strain
HitsGff=$(echo $BlastHits | sed  's/.csv/.gff/g')
Column2=Mt_gene_homolog
NumHits=1
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
$ProgDir/blast2gff.pl $Column2 $NumHits $BlastHits &gt; $HitsGff

GffAppended=$(ls gene_pred/final_genes/$Organism/$Strain/final/final_genes_appended.gff3)
bedtools intersect -wao -a $HitsGff -b $GffAppended &gt; $OutDir/&quot;$Strain&quot;_Mt_hits_intersected.bed
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">BlastHits</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls analysis/Mt_genes/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_F.spp._mt_prots_Al-Reedy_et_al._2012.fasta_hits.csv <span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>A8<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Strain=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
Organism=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> rev <span class="pl-k">|</span> cut -f3 -d <span class="pl-s"><span class="pl-pds">'</span>/<span class="pl-pds">'</span></span> <span class="pl-k">|</span> rev<span class="pl-pds">)</span></span>
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Organism</span> - <span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>
OutDir=analysis/Mt_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>
HitsGff=<span class="pl-s"><span class="pl-pds">$(</span>echo <span class="pl-smi">$BlastHits</span> <span class="pl-k">|</span> sed  <span class="pl-s"><span class="pl-pds">'</span>s/.csv/.gff/g<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span>
Column2=Mt_gene_homolog
NumHits=1
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
<span class="pl-smi">$ProgDir</span>/blast2gff.pl <span class="pl-smi">$Column2</span> <span class="pl-smi">$NumHits</span> <span class="pl-smi">$BlastHits</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$HitsGff</span>

GffAppended=<span class="pl-s"><span class="pl-pds">$(</span>ls gene_pred/final_genes/<span class="pl-smi">$Organism</span>/<span class="pl-smi">$Strain</span>/final/final_genes_appended.gff3<span class="pl-pds">)</span></span>
bedtools intersect -wao -a <span class="pl-smi">$HitsGff</span> -b <span class="pl-smi">$GffAppended</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Strain</span><span class="pl-pds">"</span></span>_Mt_hits_intersected.bed
<span class="pl-k">done</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.2 Identifying PHIbase homologs
                     </h2>
                     <a aria-label="Permalink: 5.2 Identifying PHIbase homologs" class="anchor" href="#52-identifying-phibase-homologs" id="user-content-52-identifying-phibase-homologs">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The PHIbase database was searched against the assembled genomes using tBLASTx.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for Assembly in $(ls repeat_masked/F.*/*/*/*_contigs_unmasked.fa| grep -v 'HB17' | grep -e 'cepae' -e 'proliferatum' -e 'narcissi' | grep -e 'Fus2_canu_new' -e 'ncbi' | grep -v 'old'); do
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub $ProgDir/blast_pipe.sh analysis/blast_homology/PHIbase/PHI_36_accessions.fa protein $Assembly
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">Assembly</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>ls repeat_masked/F.<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>/<span class="pl-k">*</span>_contigs_unmasked.fa<span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>HB17<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>cepae<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>proliferatum<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>narcissi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>Fus2_canu_new<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>ncbi<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">'</span>old<span class="pl-pds">'</span></span><span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/pathogen/blast
qsub <span class="pl-smi">$ProgDir</span>/blast_pipe.sh analysis/blast_homology/PHIbase/PHI_36_accessions.fa protein <span class="pl-smi">$Assembly</span>
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     following blasting PHIbase to the genome, the hits were filtered by effect on
virulence.
                    </p>
                    <p dir="auto">
                     First the a tab seperated file was made in the clusters core directory containing
PHIbase. These commands were run as part of previous projects but have been
included here for completeness.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Expressed Genes
                     </h3>
                     <a aria-label="Permalink: Expressed Genes" class="anchor" href="#expressed-genes" id="user-content-expressed-genes">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     As a preliminary analysis of the RNAseq data, highly expressed genes at 72hrs
post infection were identified in Fus2.
                    </p>
                    <p dir="auto">
                     This process is described in RNAseq/Fus2_expressed_path_genes.md and RNAseq/fpkm_analysis.md
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      5.4 Enrichment of genes on LS contigs
                     </h3>
                     <a aria-label="Permalink: 5.4 Enrichment of genes on LS contigs" class="anchor" href="#54-enrichment-of-genes-on-ls-contigs" id="user-content-54-enrichment-of-genes-on-ls-contigs">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Fus2 contigs 10, 14, 16, 19, 20, 21 and 22 were identified as lineage specific.
                    </p>
                    <p dir="auto">
                     As such interproscan annotation were extracted for these contigs and Functional
enrichments were compared to the entire genome.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Enrichment of effector families by contig
                     </h4>
                     <a aria-label="Permalink: Enrichment of effector families by contig" class="anchor" href="#enrichment-of-effector-families-by-contig" id="user-content-enrichment-of-effector-families-by-contig">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="GffGenes=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3
cat $GffGenes | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'
GffEffP=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted.gff
cat $GffEffP | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'
GffCAZY=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted.gff
cat $GffCAZY | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'
GffSecMet=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_secondary_metabolite_regions.gff
cat $GffSecMet | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'
GffMimp=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff
cat $GffMimp | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'
GffMimpSec=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff
cat $GffMimpSec | grep 'mRNA' | cut -f1 | sort | uniq -c | sort -n -k2 -t '_'" dir="auto">
                     <pre>GffGenes=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3
cat <span class="pl-smi">$GffGenes</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c <span class="pl-k">|</span> sort -n -k2 -t <span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>
GffEffP=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted.gff
cat <span class="pl-smi">$GffEffP</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c <span class="pl-k">|</span> sort -n -k2 -t <span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>
GffCAZY=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted.gff
cat <span class="pl-smi">$GffCAZY</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c <span class="pl-k">|</span> sort -n -k2 -t <span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>
GffSecMet=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_secondary_metabolite_regions.gff
cat <span class="pl-smi">$GffSecMet</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c <span class="pl-k">|</span> sort -n -k2 -t <span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>
GffMimp=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff
cat <span class="pl-smi">$GffMimp</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c <span class="pl-k">|</span> sort -n -k2 -t <span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span>
GffMimpSec=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff
cat <span class="pl-smi">$GffMimpSec</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq -c <span class="pl-k">|</span> sort -n -k2 -t <span class="pl-s"><span class="pl-pds">'</span>_<span class="pl-pds">'</span></span></pre>
                    </div>
                    <p dir="auto">
                     From this it was clear that the genes within 2kb of a mimp were primarily associated
with the pathogen specific contigs. INterproscan information on these genes
was extracted
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="MimpGeneHeaders=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp.txt
MimpInterpro=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp_interpro.txt
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/filter_by_gene.py --inp_interpro $InterPro --inp_txt $MimpGeneHeaders &gt; $MimpInterpro" dir="auto">
                     <pre>MimpGeneHeaders=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp.txt
MimpInterpro=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_prots_in_2kb_mimp_interpro.txt
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/filter_by_gene.py --inp_interpro <span class="pl-smi">$InterPro</span> --inp_txt <span class="pl-smi">$MimpGeneHeaders</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$MimpInterpro</span></pre>
                    </div>
                    <p dir="auto">
                     more information of number of mimps in LS and PS regions was extracted:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="TotalMimps=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps.gff
# Total in Genome
cat $TotalMimps |wc -l
# In all LS regions
cat $TotalMimps | grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | wc -l
# In PS regions
cat $TotalMimps | grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' | wc -l
GenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff
# Total in Genome
cat $GenesIn2Kb | grep 'gene'| wc -l
# In all LS regions
cat $GenesIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | wc -l
# In PS regions
cat $GenesIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' | wc -l
SecretedIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff
# Total in Genome
cat $SecretedIn2Kb | grep 'gene'| wc -l
# In all LS regions
cat $SecretedIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | wc -l
# In PS regions
cat $SecretedIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21'  | wc -l" dir="auto">
                     <pre>TotalMimps=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps.gff
<span class="pl-c"><span class="pl-c">#</span> Total in Genome</span>
cat <span class="pl-smi">$TotalMimps</span> <span class="pl-k">|</span>wc -l
<span class="pl-c"><span class="pl-c">#</span> In all LS regions</span>
cat <span class="pl-smi">$TotalMimps</span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_14<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_20<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_22<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> In PS regions</span>
cat <span class="pl-smi">$TotalMimps</span> <span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
GenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff
<span class="pl-c"><span class="pl-c">#</span> Total in Genome</span>
cat <span class="pl-smi">$GenesIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> In all LS regions</span>
cat <span class="pl-smi">$GenesIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_14<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_20<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_22<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> In PS regions</span>
cat <span class="pl-smi">$GenesIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
SecretedIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp_secreted.gff
<span class="pl-c"><span class="pl-c">#</span> Total in Genome</span>
cat <span class="pl-smi">$SecretedIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> In all LS regions</span>
cat <span class="pl-smi">$SecretedIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_14<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_20<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_22<span class="pl-pds">'</span></span> <span class="pl-k">|</span> wc -l
<span class="pl-c"><span class="pl-c">#</span> In PS regions</span>
cat <span class="pl-smi">$SecretedIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span>  <span class="pl-k">|</span> wc -l</pre>
                    </div>
                    <p dir="auto">
                     Effector candidates within 2Kb of a MIMP were identified:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="GenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff
OutDir=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new
cat $GenesIn2Kb | grep 'gene'| grep -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' | cut -f2 -d '=' | tr -d ';' &gt; $OutDir/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt
cat $GenesIn2Kb | grep 'gene'| grep -e 'contig_14' -e 'contig_20' -e 'contig_22' | cut -f2 -d '=' | tr -d ';' &gt; $OutDir/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt
cat $GenesIn2Kb | grep 'gene'| grep -v -e 'contig_10' -e 'contig_16' -e 'contig_19' -e 'contig_21' -e 'contig_14' -e 'contig_20' -e 'contig_22' | cut -f2 -d '=' | tr -d ';' &gt; $OutDir/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt

EffectorPHeaders=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted_headers.txt
cat $EffectorPHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt | wc -l
cat $EffectorPHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt | wc -l
cat $EffectorPHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt | wc -l

CazyHeaders=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted_headers.txt
cat $CazyHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt | wc -l
cat $CazyHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt | wc -l
cat $CazyHeaders | cut -f1 | cut -f1 -d '.' | sort | uniq | grep -f $OutDir/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt | wc -l



AntismashGff=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_secondary_metabolite_regions.gff
MimpPlus2KbGff=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps_exp.gff
bedtools intersect -u -a $AntismashGff -b $MimpPlus2KbGff | less -S
" dir="auto">
                     <pre>GenesIn2Kb=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_genes_in_2kb_mimp.gff
OutDir=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new
cat <span class="pl-smi">$GenesIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt
cat <span class="pl-smi">$GenesIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -e <span class="pl-s"><span class="pl-pds">'</span>contig_14<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_20<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_22<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt
cat <span class="pl-smi">$GenesIn2Kb</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>gene<span class="pl-pds">'</span></span><span class="pl-k">|</span> grep -v -e <span class="pl-s"><span class="pl-pds">'</span>contig_10<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_16<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_19<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_21<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_14<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_20<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>contig_22<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span> <span class="pl-k">|</span> tr -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt

EffectorPHeaders=analysis/effectorP/F.oxysporum_fsp_cepae/Fus2_canu_new/F.oxysporum_fsp_cepae_Fus2_canu_new_EffectorP_secreted_headers.txt
cat <span class="pl-smi">$EffectorPHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> grep -f <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt <span class="pl-k">|</span> wc -l
cat <span class="pl-smi">$EffectorPHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> grep -f <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt <span class="pl-k">|</span> wc -l
cat <span class="pl-smi">$EffectorPHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> grep -f <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt <span class="pl-k">|</span> wc -l

CazyHeaders=gene_pred/CAZY/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_CAZY_secreted_headers.txt
cat <span class="pl-smi">$CazyHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> grep -f <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_PS_headers.txt <span class="pl-k">|</span> wc -l
cat <span class="pl-smi">$CazyHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> grep -f <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_non-PS_LS_headers.txt <span class="pl-k">|</span> wc -l
cat <span class="pl-smi">$CazyHeaders</span> <span class="pl-k">|</span> cut -f1 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>.<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">|</span> grep -f <span class="pl-smi">$OutDir</span>/Fus2_canu_new_genes_in_2kb_mimp_core_headers.txt <span class="pl-k">|</span> wc -l



AntismashGff=analysis/antismash/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_secondary_metabolite_regions.gff
MimpPlus2KbGff=analysis/mimps/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_mimps_exp.gff
bedtools intersect -u -a <span class="pl-smi">$AntismashGff</span> -b <span class="pl-smi">$MimpPlus2KbGff</span> <span class="pl-k">|</span> less -S
</pre>
                    </div>
                    <p dir="auto">
                     Enrichment was performed for each contig:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for num in $(seq 1 34); do
Contig=&quot;contig_&quot;$num&quot;_pilon&quot;
echo $Contig
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
Gff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3
GeneList=$OutDir/&quot;$Contig&quot;_interproscan.txt
cat $Gff | grep &quot;$Contig&quot; | grep 'mRNA' | cut -f9 | cut -f1 -d ';' | cut -f2 -d '='  &gt; $GeneList
InterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv
InterPro_contig=$OutDir/&quot;$Contig&quot;_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/filter_by_gene.py --inp_interpro $InterPro --inp_txt $GeneList &gt; $InterPro_contig
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $InterPro_contig &gt; $OutDir/&quot;$Contig&quot;_GO_WEGO.txt
done" dir="auto">
                     <pre><span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> <span class="pl-s"><span class="pl-pds">$(</span>seq 1 34<span class="pl-pds">)</span></span><span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
Gff=gene_pred/final_genes/F.oxysporum_fsp_cepae/Fus2_canu_new/final/final_genes_appended.gff3
GeneList=<span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.txt
cat <span class="pl-smi">$Gff</span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>mRNA<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f9 <span class="pl-k">|</span> cut -f1 -d <span class="pl-s"><span class="pl-pds">'</span>;<span class="pl-pds">'</span></span> <span class="pl-k">|</span> cut -f2 -d <span class="pl-s"><span class="pl-pds">'</span>=<span class="pl-pds">'</span></span>  <span class="pl-k">&gt;</span> <span class="pl-smi">$GeneList</span>
InterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv
InterPro_contig=<span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/filter_by_gene.py --inp_interpro <span class="pl-smi">$InterPro</span> --inp_txt <span class="pl-smi">$GeneList</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$InterPro_contig</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$InterPro_contig</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span></pre>
                    </div>
                    <p dir="auto">
                     The WEGO wesite was used to generate summaries of genes on core and LS regions.
                    </p>
                    <p dir="auto">
                     A similar process was then repeated for duplicated genes:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes
mkdir -p $OutDir
DupGenesTab=/home/sobczm/popgen/codon/blast/dagchainer/testing/no_transposon/non-transposon_duplications_summaryf_ann
cat $DupGenesTab | tail -n+2 | cut -f1,3,5 | sed 's/\t/\n/g' | sed 's/,/\n/g' | grep -v &quot;^$&quot; | grep 'Fus2' | sed 's/Fus2_//g' | sort | uniq &gt; $OutDir/duplicated_genes.txt
InterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv
InterPro_dups=$OutDir/duplicated_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/filter_by_gene.py --inp_interpro $InterPro --inp_txt $OutDir/duplicated_genes.txt &gt; $InterPro_dups
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $InterPro_dups &gt; $OutDir/duplicated_genes_GO_WEGO.txt" dir="auto">
                     <pre>OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes
mkdir -p <span class="pl-smi">$OutDir</span>
DupGenesTab=/home/sobczm/popgen/codon/blast/dagchainer/testing/no_transposon/non-transposon_duplications_summaryf_ann
cat <span class="pl-smi">$DupGenesTab</span> <span class="pl-k">|</span> tail -n+2 <span class="pl-k">|</span> cut -f1,3,5 <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/\t/\n/g<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/,/\n/g<span class="pl-pds">'</span></span> <span class="pl-k">|</span> grep -v <span class="pl-s"><span class="pl-pds">"</span>^$<span class="pl-pds">"</span></span> <span class="pl-k">|</span> grep <span class="pl-s"><span class="pl-pds">'</span>Fus2<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sed <span class="pl-s"><span class="pl-pds">'</span>s/Fus2_//g<span class="pl-pds">'</span></span> <span class="pl-k">|</span> sort <span class="pl-k">|</span> uniq <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/duplicated_genes.txt
InterPro=gene_pred/interproscan/F.oxysporum_fsp_cepae/Fus2_canu_new/Fus2_canu_new_interproscan.tsv
InterPro_dups=<span class="pl-smi">$OutDir</span>/duplicated_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/filter_by_gene.py --inp_interpro <span class="pl-smi">$InterPro</span> --inp_txt <span class="pl-smi">$OutDir</span>/duplicated_genes.txt <span class="pl-k">&gt;</span> <span class="pl-smi">$InterPro_dups</span>
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$InterPro_dups</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/duplicated_genes_GO_WEGO.txt</pre>
                    </div>
                    <p dir="auto">
                     Annotations by contig were grouped by chromosome:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr1
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
	mkdir -p $OutDir
	printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
	echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
	for num in 1; do
	Contig="contig_"$num"_pilon"
	echo $Contig
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
	mkdir -p $OutDir
	cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
	cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
	done
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr2
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
	mkdir -p $OutDir
	printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
	echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
	for num in 17 5; do
	Contig="contig_"$num"_pilon"
	echo $Contig
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
	mkdir -p $OutDir
	cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
	cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
	done
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr4
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
	mkdir -p $OutDir
	printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
	echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
	for num in 2; do
	Contig="contig_"$num"_pilon"
	echo $Contig
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
	mkdir -p $OutDir
	cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
	cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
	done
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr5
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
	mkdir -p $OutDir
	printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
	echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
	for num in 3; do
	Contig="contig_"$num"_pilon"
	echo $Contig
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
	mkdir -p $OutDir
	cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
	cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
	done
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr7
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
	mkdir -p $OutDir
	printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
	echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
	for num in 4; do
	Contig="contig_"$num"_pilon"
	echo $Contig
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
	mkdir -p $OutDir
	cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
	cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
	done
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr8
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 6; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr9
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 7; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr10
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 8; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr11
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 9; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr12
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 13 15 18; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr13
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 12 11; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS14
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 14; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS10
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 10; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS16
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 16; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS19
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 19; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS20
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 20; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS21
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 21; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS22
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/$Chr
mkdir -p $OutDir
printf "" &gt; $OutDir/"$Chr"_interproscan.tsv
echo "!interpro2wego_output" &gt; $OutDir/"$Chr"_GO_WEGO.txt
for num in 22; do
Contig="contig_"$num"_pilon"
echo $Contig
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/$Contig
mkdir -p $OutDir
cat $ContigDir/"$Contig"_interproscan.tsv &gt;&gt; $OutDir/"$Chr"_interproscan.tsv
cat $ContigDir/"$Contig"_GO_WEGO.txt | tail -n+2 &gt;&gt; $OutDir/"$Chr"_GO_WEGO.txt
done
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat $OutDir/"$Chr"_interproscan.tsv | grep -f $DupGenes &gt; $OutDir/"$Chr"_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
$ProgDir/interpro2wego.py --inp_interpro $OutDir/"$Chr"_dup_genes_interproscan.tsv &gt; $OutDir/"$Chr"_dup_genes_GO_WEGO.txt' dir="auto">
                     <pre>	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr1
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 1<span class="pl-k">;</span> <span class="pl-k">do</span>
	Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
	<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">done</span>
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr2
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 17 5<span class="pl-k">;</span> <span class="pl-k">do</span>
	Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
	<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">done</span>
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr4
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 2<span class="pl-k">;</span> <span class="pl-k">do</span>
	Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
	<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">done</span>
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr5
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 3<span class="pl-k">;</span> <span class="pl-k">do</span>
	Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
	<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">done</span>
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
	Chr=Chr7
	OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 4<span class="pl-k">;</span> <span class="pl-k">do</span>
	Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
	<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
	ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
	mkdir -p <span class="pl-smi">$OutDir</span>
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
	cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
	<span class="pl-k">done</span>
	DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
	cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
	ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
	<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr8
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 6<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr9
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 7<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr10
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 8<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr11
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 9<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr12
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 13 15 18<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=Chr13
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 12 11<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS14
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 14<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS10
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 10<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS16
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 16<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS19
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 19<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS20
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 20<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS21
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 21<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt

OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome
Chr=LS22
OutDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_chromosome/<span class="pl-smi">$Chr</span>
mkdir -p <span class="pl-smi">$OutDir</span>
<span class="pl-c1">printf</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>!interpro2wego_output<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">for</span> <span class="pl-smi">num</span> <span class="pl-k">in</span> 22<span class="pl-k">;</span> <span class="pl-k">do</span>
Contig=<span class="pl-s"><span class="pl-pds">"</span>contig_<span class="pl-pds">"</span></span><span class="pl-smi">$num</span><span class="pl-s"><span class="pl-pds">"</span>_pilon<span class="pl-pds">"</span></span>
<span class="pl-c1">echo</span> <span class="pl-smi">$Contig</span>
ContigDir=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/by_contig/<span class="pl-smi">$Contig</span>
mkdir -p <span class="pl-smi">$OutDir</span>
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv
cat <span class="pl-smi">$ContigDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Contig</span><span class="pl-pds">"</span></span>_GO_WEGO.txt <span class="pl-k">|</span> tail -n+2 <span class="pl-k">&gt;&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_GO_WEGO.txt
<span class="pl-k">done</span>
DupGenes=analysis/enrichment/LS_regions/F.oxysporum_fsp_cepae/Fus2_canu_new/duplicated_genes/duplicated_genes.txt
cat <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_interproscan.tsv <span class="pl-k">|</span> grep -f <span class="pl-smi">$DupGenes</span> <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv
ProgDir=/home/armita/git_repos/emr_repos/tools/seq_tools/feature_annotation/interproscan/feature_extraction
<span class="pl-smi">$ProgDir</span>/interpro2wego.py --inp_interpro <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_interproscan.tsv <span class="pl-k">&gt;</span> <span class="pl-smi">$OutDir</span>/<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$Chr</span><span class="pl-pds">"</span></span>_dup_genes_GO_WEGO.txt</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      6. Summarising the Fusarium Proteome
                     </h2>
                     <a aria-label="Permalink: 6. Summarising the Fusarium Proteome" class="anchor" href="#6-summarising-the-fusarium-proteome" id="user-content-6-summarising-the-fusarium-proteome">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="" dir="auto">
                     <pre></pre>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="08Fz9IjrLZMoXuzsQo-ondwJ-9z6HeInXxQEBIPYecTEGexbFOBV8ysSFpUU2zVTqrPWPxq0mb0uxyCvhnvn5w">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Fusarium Genomics
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/harrisonlab/fusarium/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/harrisonlab/fusarium/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/harrisonlab/fusarium/custom-properties">
                  <svg aria-hidden="true" class="octicon octicon-note mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25Zm1.75-.25a.25.25 0 0 0-.25.25v8.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25ZM3.5 6.25a.75.75 0 0 1 .75-.75h7a.75.75 0 0 1 0 1.5h-7a.75.75 0 0 1-.75-.75Zm.75 2.25h4a.75.75 0 0 1 0 1.5h-4a.75.75 0 0 1 0-1.5Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Custom properties
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/harrisonlab/fusarium/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   2
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/harrisonlab/fusarium/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   24
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/harrisonlab/fusarium/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   2
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fharrisonlab%2Ffusarium&amp;report=harrisonlab+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/harrisonlab/fusarium/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/orgs/harrisonlab/packages?repo_name=fusarium">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/harrisonlab/fusarium/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/harrisonlab/fusarium/graphs/contributors">
                 Contributors
                 <span class="Counter ml-1" data-view-component="true" title="2">
                  2
                 </span>
                </a>
               </h2>
               <include-fragment aria-busy="true" aria-label="Loading contributors" src="/harrisonlab/fusarium/contributors_list?count=2&amp;current_repository=fusarium&amp;items_to_show=2">
                <ul class="list-style-none">
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                 <li class="mb-2 d-flex">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                  <div class="Skeleton Skeleton--text flex-1 flex-self-center f4">
                  </div>
                 </li>
                </ul>
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="Python 49.1" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#3572A5 !important;;width: 49.1%;">
                 </span>
                 <span aria-label="Shell 36.5" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#89e051 !important;;width: 36.5%;">
                 </span>
                 <span aria-label="R 13.8" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 13.8%;">
                 </span>
                 <span aria-label="Perl 0.6" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#0298c3 !important;;width: 0.6%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/harrisonlab/fusarium/search?l=python">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#3572A5;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Python
                  </span>
                  <span>
                   49.1%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/harrisonlab/fusarium/search?l=shell">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#89e051;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Shell
                  </span>
                  <span>
                   36.5%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/harrisonlab/fusarium/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   13.8%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/harrisonlab/fusarium/search?l=perl">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#0298c3;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Perl
                  </span>
                  <span>
                   0.6%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       Â© 2024 GitHub,Â Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="true" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You canât perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
