<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-bf4f9425a973.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-340b2bef927d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         j-andrews7/awesome-bioinformatics-benchmarks: A curated list of bioinformatics bench-marking papers and resources.
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="FB70:6E100:5B79F85:816B991:660599E4" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IkZCNzA6NkUxMDA6NUI3OUY4NTo4MTZCOTkxOjY2MDU5OUU0IiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="1237c4003b574446de4b854659582d3a58b8183876f82b30ed8ea0ddbc952714" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:223818793" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="A curated list of bioinformatics bench-marking papers and resources. - j-andrews7/awesome-bioinformatics-benchmarks" name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/j-andrews7/awesome-bioinformatics-benchmarks" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/ed43b8e3bb040bf491dd8b3ca17968fb341dbbae15a50cbb67dbcd576994f34e/j-andrews7/awesome-bioinformatics-benchmarks" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="j-andrews7/awesome-bioinformatics-benchmarks: A curated list of bioinformatics bench-marking papers and resources." name="twitter:title">
                       <meta content="A curated list of bioinformatics bench-marking papers and resources. - j-andrews7/awesome-bioinformatics-benchmarks" name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/ed43b8e3bb040bf491dd8b3ca17968fb341dbbae15a50cbb67dbcd576994f34e/j-andrews7/awesome-bioinformatics-benchmarks" property="og:image"/>
                        <meta content="A curated list of bioinformatics bench-marking papers and resources. - j-andrews7/awesome-bioinformatics-benchmarks" property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="j-andrews7/awesome-bioinformatics-benchmarks: A curated list of bioinformatics bench-marking papers and resources." property="og:title"/>
                        <meta content="https://github.com/j-andrews7/awesome-bioinformatics-benchmarks" property="og:url"/>
                        <meta content="A curated list of bioinformatics bench-marking papers and resources. - j-andrews7/awesome-bioinformatics-benchmarks" property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjI2OTY0OTU1MzAsInQiOjE3MTE2NDMxMTN9--d252ae71ab417d8a3af4d4c5789f070ccc14f81a6758bad347284a3315bcda57" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="a81c045767350c9f2e551b272402db2e99346c1d2710552f41e25d405666b5eb" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="3037d78571ecdebb88047af8e128a53bf5dc96ac8c533579876941ff0be8ae32" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/j-andrews7/awesome-bioinformatics-benchmarks git https://github.com/j-andrews7/awesome-bioinformatics-benchmarks.git" name="go-import"/>
                        <meta content="10225716" name="octolytics-dimension-user_id">
                         <meta content="j-andrews7" name="octolytics-dimension-user_login">
                          <meta content="223818793" name="octolytics-dimension-repository_id">
                           <meta content="j-andrews7/awesome-bioinformatics-benchmarks" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="223818793" name="octolytics-dimension-repository_network_root_id">
                               <meta content="j-andrews7/awesome-bioinformatics-benchmarks" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/j-andrews7/awesome-bioinformatics-benchmarks" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd" data-view-component="true" id="dialog-show-dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd-description" aria-labelledby="dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-70a2154a-eccf-493b-9bb3-4ae39ae4e6fd-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-fb3f75c4-4b27-4a29-8965-53cd487043a4">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-d5126e5c-b883-44b7-990c-a2a0396538a4">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-71c8abaa-6649-4b74-91e3-400dfde06562">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-5b92239d-5f49-49bf-a03c-4513ab90c07a">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-9f1a54fa-7e6a-4d2e-9a62-9925adb2d25f">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-d9736b4e-0c31-4246-9d0a-739e754896ad">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-3a03a8d6-0ebc-4974-983a-faf418c1757d">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-adebf430-3105-46cb-b4e3-8802c48cf77b">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               Â© 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: j-andrews7 / awesome-bioinformatics-benchmarks" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              j-andrews7
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              awesome-bioinformatics-benchmarks
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"j-andrews7","screen_size":"compact"}' data-view-component="true" href="/j-andrews7">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 j-andrews7
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"awesome-bioinformatics-benchmarks","screen_size":"compact"}' data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 awesome-bioinformatics-benchmarks
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"j-andrews7","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/j-andrews7/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/j-andrews7">
             <span class="AppHeader-context-item-label">
              j-andrews7
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"awesome-bioinformatics-benchmarks","screen_size":"full"}' data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks">
             <span class="AppHeader-context-item-label">
              awesome-bioinformatics-benchmarks
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="o67Yyb4MIURKvcfUgFYoEZnyC0X8Q4cq2VdrTI_OcUvzjdRBLIH-i_0nn_olwcpKrX7LrshoMrsHLBXyIgrMow" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="j-andrews7" data-current-repository="j-andrews7/awesome-bioinformatics-benchmarks" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="nxawsSSjmy9QVs7PEH0apA14ZjA6t-BhnaV23pVUQcPX3KebAuXSWgf6vBb2eOq1X9wkmRzQEA7xRzd1cmpajw" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:j-andrews7/awesome-bioinformatics-benchmarks">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump toâ¦" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump toâ¦" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-cfbea7b4-9202-49c4-87e1-e61d6b8fd87d" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-f8d4ccdc-3a67-4728-897e-aebbb68fc8f0" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-f8d4ccdc-3a67-4728-897e-aebbb68fc8f0">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="aixK1ogDfXi0p2uHZSwyZCHyns7Zgqz8yg72L--maxX4nlPuQNVO4gSFZYNbNTOV_PlTDYOiSqnEjuOVGfdurg"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="EP-2qkzBvw7Uc1Y6lE5V9sjh_8p94-rKBgm0iuMHp5dcDdQQFEDh158a65fem5jLAbGb3SvmqBGlUWUy_ED-wg">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="-08LMpw476zqwvaSobvbS1B41E4-_YmyBDA_E73JUBHjEkaUy7pGr2nWvxHh6YUh5r_RyRAkaSppkJ3BhaaD2Q">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="7H9t1z-bsLgt2sPhkJO07eo_U-ndUDI4tUWPF68d6-4rKgL1Jpx22soBWRxZKM_-OoeLI1j29Ai6JtGyN4mK5A">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-505a0f9a-0b8d-4996-86d1-45516128919b" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-c1cac2be-5d3b-4816-8414-0c6593d7714b" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-3d8b6e29-17d0-48e0-8605-9300fdb48200" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-74b3e0b6-6bb8-4e53-86de-fc02cb9c4e89" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-1465197a-2416-4a0c-9d1f-c058d06afca9" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-5fc3778c-eab3-4f8d-9225-f9f0c1bbc306" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-458820b1-3ec2-40b5-92cd-94df0ea8acb2" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-c5872de6-6309-46fe-833c-bbf75665bcb4">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-c5872de6-6309-46fe-833c-bbf75665bcb4" id="tooltip-458820b1-3ec2-40b5-92cd-94df0ea8acb2" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-301b937d-0d36-4308-a92a-a124191fe6a6" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-c1c14f96-ed9c-41ba-bdf4-c24121cb43b1">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-c1c14f96-ed9c-41ba-bdf4-c24121cb43b1" id="tooltip-301b937d-0d36-4308-a92a-a124191fe6a6" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY0MzExM30=--c18e66099cc46b267b989aa9eb81c5796525b7389a6f7c83a42ae650dba77cb8" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-722f5b27-d66f-47d2-b761-e986124af264" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=223818793">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb" data-view-component="true" id="dialog-show-dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb-description" aria-labelledby="dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-1c529b54-b8bb-4516-9c40-de81f638711c" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-5144ac74-b2b5-4122-9e0c-f9fce9fb6add" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-8c2effee-140b-4497-85f9-2398b6a5ce3d" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-f58b5fd1-732e-41f0-8e07-6aeb941c8c5d" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-2be2e7ca-39c1-4e35-af1c-6600dd1effc0" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-db9bd0c3-3d4d-4ad8-8122-47eca72b685f" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-d28a356f-f956-45bf-8ca9-e27bfa892dcb-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-d4ef3e4e-4964-4889-bca0-c524b8b4ac4d" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-82df30fa-fb26-4bae-aa1a-00b1a582e2ff">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-e50c367e-2dfa-48b2-8e62-6b172c082cf1" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-de1ab9ee-0ee9-4863-b813-53c2cbac832e">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-bda8b603-fc9b-42c9-86e5-7025822f03c7">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-420de9dc-0345-4ccf-b805-e43d94c86763" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-16ddc389-f148-45ec-bc1a-e3ed68aa41ba" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-4d71585e-7876-4815-8a70-85b169869ee8">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-96c8916c-5c19-4980-aedf-2db06b738ee5">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-83e71098-a48a-487e-9e27-5ca3accdc319">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-fed7bf0c-9633-4712-8763-a754e87eccb2" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-c63f1c81-495b-4b6e-9eb3-05af0971288e" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-539280d9-1698-479c-8472-7c55afc0ab49">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-c170a8f7-df8d-4a75-8a93-7de46722b4d3">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-832d1553-5257-4ce5-8024-eac0da777e10">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-aedeb208-f7ab-4afd-ad8a-eb64f47aa3cf">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /j-andrews7/awesome-bioinformatics-benchmarks" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /j-andrews7/awesome-bioinformatics-benchmarks/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="issues-repo-tab-count" title="20">
           20
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /j-andrews7/awesome-bioinformatics-benchmarks/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /j-andrews7/awesome-bioinformatics-benchmarks/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /j-andrews7/awesome-bioinformatics-benchmarks/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /j-andrews7/awesome-bioinformatics-benchmarks/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/j-andrews7/awesome-bioinformatics-benchmarks/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /j-andrews7/awesome-bioinformatics-benchmarks/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-list" aria-haspopup="true" aria-labelledby="tooltip-8d7ce32c-2881-440a-9784-c47f671043ea" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-button" popovertarget="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-button" id="tooltip-8d7ce32c-2881-440a-9784-c47f671043ea" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-button" anchor-offset="normal" data-view-component="true" id="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-b464346a-605f-4179-9d8d-d0d6b01f49e1-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks" id="item-792f65d8-8533-4af9-8516-dc81302e755f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/issues" id="item-6f4ebc19-2c7d-4edc-8937-1e39d1d14082" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/pulls" id="item-5380b93c-ebc1-4096-96b5-ac36c5dd1d82" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/actions" id="item-48c063b0-7a41-49d7-8483-7a9c6688680b" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/projects" id="item-ceb43fc4-a143-43bf-931d-3586e6b5a84f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/security" id="item-02bb00de-f005-46f5-9eae-f665eef32c91" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/pulse" id="item-ff07db4f-beab-4057-a31e-b83dc557c405" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-4ef42a75-9456-4e70-8f5a-fdd6e487b230" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-0cb0b4e1-b5de-4225-9070-d91e9d6fa9e3" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-0cb0b4e1-b5de-4225-9070-d91e9d6fa9e3" id="tooltip-4ef42a75-9456-4e70-8f5a-fdd6e487b230" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY0MzExM30=--c18e66099cc46b267b989aa9eb81c5796525b7389a6f7c83a42ae650dba77cb8" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/j-andrews7/awesome-bioinformatics-benchmarks" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjEwMjI1NzE2" data-targets="command-palette-scope.tokens" data-text="j-andrews7" data-type="owner" data-value="j-andrews7" style="white-space:nowrap;line-height:20px;">
         j-andrews7
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkyMjM4MTg3OTM=" data-targets="command-palette-scope.tokens" data-text="awesome-bioinformatics-benchmarks" data-type="repository" data-value="awesome-bioinformatics-benchmarks" style="white-space:nowrap;line-height:20px;">
         awesome-bioinform...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-8d73633b-b0a6-4f15-a133-93f4dd795b76" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkyMjM4MTg3OTM=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         Weâve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="j-andrews7" data-scope-id="MDQ6VXNlcjEwMjI1NzE2" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="awesome-bioinformatics-benchmarks" data-scope-id="MDEwOlJlcG9zaXRvcnkyMjM4MTg3OTM=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="j-andrews7" data-scope-id="MDQ6VXNlcjEwMjI1NzE2" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="awesome-bioinformatics-benchmarks" data-scope-id="MDEwOlJlcG9zaXRvcnkyMjM4MTg3OTM=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="V5eDoL8rrpIF39Tt5A6cutYtVRkluG-a4HBPxTTzEcaH-ymEKvF4Nebifop31gUccy6ebKGfRqZ-2mpwgCIHcQ">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="â­ï¸ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="5SlKQ3ZueR_6J_0AI8Z_VWWsLCEDkalx8AZK_hUV4S-FARayuUAreyxsYkTAxLtjZEZyUK9jF0tcLDA3931VgQ">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/10225716?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/j-andrews7/awesome-bioinformatics-benchmarks">
            awesome-bioinformatics-benchmarks
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/223818793/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":223818793,"originating_url":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","user_id":126681664}}' data-hydro-click-hmac="766a3a15c1c3b1146802f11b822e01574013f98354c2b4aaf3ce262476112b98" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="55">
              55
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-8546c09c-f170-4ac8-b541-dde0268da1e5" popover="manual">
              Fork your own copy of j-andrews7/awesome-bioinformatics-benchmarks
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-223818793">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/j-andrews7/awesome-bioinformatics-benchmarks/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-223818793" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-fa8b63">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/j-andrews7/awesome-bioinformatics-benchmarks/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="FuXMfvpEktBJUOvP7XrelcVb9k6fbZmeuxuhh_wG6QMKEOkhozfV8OSXzZTuJh0Kud2rnHahJfkEzNepQvmvjA">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="oPyqT-hKWqE98BxSAZYjK7Q288OeVyJPGOh06cCDDte8CY8QsTkdgZA3OgkCyuC0yLCuEXebniinPwLHfnxIWA">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (289)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":223818793,"originating_url":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","user_id":126681664}}' data-hydro-click-hmac="b9ff88303544cf4cc57ae6f897e76a98ff51ecbe6aea662adfa2644e5a2c0942" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="289 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="289">
                  289
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-223818793-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/j-andrews7/awesome-bioinformatics-benchmarks/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-223818793-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-66d5d6">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/j-andrews7/awesome-bioinformatics-benchmarks/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="AZaGIvDnb_0f9YijaQuwZnW7f7Nd6pog6szohJb9WRRGfXEydp9A_HtFudIG1a5KNbP9t81eqhtqs4fF_6Jpmg">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (289)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":223818793,"originating_url":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","user_id":126681664}}' data-hydro-click-hmac="84a041e9646f9c0d91cf1deace457e1610a61cc15cc2417aa818fddb26d038c4" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="289 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="289">
                 289
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-223818793-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/j-andrews7/awesome-bioinformatics-benchmarks/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-223818793-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-e354b8">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":223818793,"originating_url":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","user_id":126681664}}' data-hydro-click-hmac="9f584a1efaf6242a528ae844fde81f8f059daee4ed365a2faac00219e2f89fc4" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="E8cjNYyqd_vx46JM44Tj6FIDjCZPXO2OD_WGMEVxPY0LBy9PkhUC-h3wlUY_ZGWfDSYfM5oGRlotDnTEEsCDFA">
                   <input name="repository_id" type="hidden" value="223818793"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="XeO8fQ_w6xYbzzSkYRgucmk5aPVEaDGK8XfZ7LXcpTdFI7AHEU-eF_fcA669-KgFNhz74JEyml7TjCsY4m0brg">
                   <input name="repository_id" type="hidden" value="223818793"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-908e72bb-b4be-47e5-8893-58ebc4142bab" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-bb24d058-75e7-4ab4-a61b-f94a275034ef" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":223818793,"originating_url":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","user_id":126681664}}' data-hydro-click-hmac="766a3a15c1c3b1146802f11b822e01574013f98354c2b4aaf3ce262476112b98" data-view-component="true" group_item="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-bb24d058-75e7-4ab4-a61b-f94a275034ef" popover="manual">
              Fork your own copy of j-andrews7/awesome-bioinformatics-benchmarks
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/j-andrews7/awesome-bioinformatics-benchmarks/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="3_9AsDZBhLTUkxWRxBLro9A_O6oFHFO11ITkSKlZkoXDCmXvbzLDlHlUM8rHTig8rLlmeOzQ79JrU5JmF6bUCg">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-dabcab6d-860a-4a57-8fe1-ae3bff5c40e5" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":223818793,"originating_url":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","user_id":126681664}}' data-hydro-click-hmac="b9ff88303544cf4cc57ae6f897e76a98ff51ecbe6aea662adfa2644e5a2c0942" data-view-component="true" id="icon-button-bcc2055d-b403-4fe5-b122-1b152424c745" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-bcc2055d-b403-4fe5-b122-1b152424c745" id="tooltip-dabcab6d-860a-4a57-8fe1-ae3bff5c40e5" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/j-andrews7/awesome-bioinformatics-benchmarks/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="u8V0XKCUanxo7oJJPTveeLTpkW24iPbarR4DAppqqHv8LoNMJuxFfQxeszhS5cBU9OETaSg8xuEtYWxD8zWY9Q">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-6a37ad59-6ee0-4cee-b721-8af50e30ce25" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":223818793,"originating_url":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","user_id":126681664}}' data-hydro-click-hmac="84a041e9646f9c0d91cf1deace457e1610a61cc15cc2417aa818fddb26d038c4" data-view-component="true" id="icon-button-2615e3d8-dfe5-4115-9a95-8aafc706ef76" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-2615e3d8-dfe5-4115-9a95-8aafc706ef76" id="tooltip-6a37ad59-6ee0-4cee-b721-8af50e30ce25" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           A curated list of bioinformatics bench-marking papers and resources.
          </p>
          <h3 class="sr-only">
           License
          </h3>
          <div class="mb-2">
           <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/LICENSE">
            <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
             </path>
            </svg>
            GPL-3.0 license
           </a>
          </div>
          <div class="mb-3">
           <h3 class="sr-only">
            Code of conduct
           </h3>
           <div class="mb-2">
            <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:code of conduct"}' href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/CODE_OF_CONDUCT.md">
             <svg aria-hidden="true" class="octicon octicon-code-of-conduct mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8.048 2.241c.964-.709 2.079-1.238 3.325-1.241a4.616 4.616 0 0 1 3.282 1.355c.41.408.757.86.996 1.428.238.568.348 1.206.347 1.968 0 2.193-1.505 4.254-3.081 5.862-1.496 1.526-3.213 2.796-4.249 3.563l-.22.163a.749.749 0 0 1-.895 0l-.221-.163c-1.036-.767-2.753-2.037-4.249-3.563C1.51 10.008.007 7.952.002 5.762a4.614 4.614 0 0 1 1.353-3.407C3.123.585 6.223.537 8.048 2.24Zm-1.153.983c-1.25-1.033-3.321-.967-4.48.191a3.115 3.115 0 0 0-.913 2.335c0 1.556 1.109 3.24 2.652 4.813C5.463 11.898 6.96 13.032 8 13.805c.353-.262.758-.565 1.191-.905l-1.326-1.223a.75.75 0 0 1 1.018-1.102l1.48 1.366c.328-.281.659-.577.984-.887L9.99 9.802a.75.75 0 1 1 1.019-1.103l1.384 1.28c.295-.329.566-.661.81-.995L12.92 8.7l-1.167-1.168c-.674-.671-1.78-.664-2.474.03-.268.269-.538.537-.802.797-.893.882-2.319.843-3.185-.032-.346-.35-.693-.697-1.043-1.047a.75.75 0 0 1-.04-1.016c.162-.191.336-.401.52-.623.62-.748 1.356-1.637 2.166-2.417Zm7.112 4.442c.313-.65.491-1.293.491-1.916v-.001c0-.614-.088-1.045-.23-1.385-.143-.339-.357-.633-.673-.949a3.111 3.111 0 0 0-2.218-.915c-1.092.003-2.165.627-3.226 1.602-.823.755-1.554 1.637-2.228 2.45l-.127.154.562.566a.755.755 0 0 0 1.066.02l.794-.79c1.258-1.258 3.312-1.31 4.594-.032.396.394.792.791 1.173 1.173Z">
              </path>
             </svg>
             Code of conduct
            </a>
           </div>
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/j-andrews7/awesome-bioinformatics-benchmarks/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              289
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/j-andrews7/awesome-bioinformatics-benchmarks/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              55
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              18
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/j-andrews7/awesome-bioinformatics-benchmarks/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/j-andrews7/awesome-bioinformatics-benchmarks/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/j-andrews7/awesome-bioinformatics-benchmarks/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/j-andrews7/awesome-bioinformatics-benchmarks/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/j-andrews7/awesome-bioinformatics-benchmarks/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/j-andrews7/awesome-bioinformatics-benchmarks?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         j-andrews7/awesome-bioinformatics-benchmarks
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/j-andrews7/awesome-bioinformatics-benchmarks/spoofed_commit_check/5423fa862af5eb37c9f5923a1b43fa0dff247502">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":223818793,"defaultBranch":"master","name":"awesome-bioinformatics-benchmarks","ownerLogin":"j-andrews7","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2019-11-25T06:20:17.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/10225716?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"master","listCacheKey":"v0:1703025208.0","canEdit":true,"refType":"branch","currentOid":"5423fa862af5eb37c9f5923a1b43fa0dff247502"},"tree":{"items":[{"name":".github","path":".github","contentType":"directory"},{"name":"CODE_OF_CONDUCT.md","path":"CODE_OF_CONDUCT.md","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":4,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/j-andrews7/awesome-bioinformatics-benchmarks/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks.git","showCloneWarning":false,"sshUrl":"git@github.com:j-andrews7/awesome-bioinformatics-benchmarks.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone j-andrews7/awesome-bioinformatics-benchmarks","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/j-andrews7/awesome-bioinformatics-benchmarks","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fj-andrews7%2Fawesome-bioinformatics-benchmarks","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/j-andrews7/awesome-bioinformatics-benchmarks/archive/refs/heads/master.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"124","overviewFiles":[{"displayName":"README.md","repoName":"awesome-bioinformatics-benchmarks","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAwesome Bioinformatics Benchmarks \u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks/actions/workflows/action.yml/badge.svg\"\u003e\u003cimg src=\"https://github.com/j-andrews7/awesome-bioinformatics-benchmarks/actions/workflows/action.yml/badge.svg\" alt=\"Build Status\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/h1\u003e\u003ca id=\"user-content-awesome-bioinformatics-benchmarks-\" class=\"anchor\" aria-label=\"Permalink: Awesome Bioinformatics Benchmarks \" href=\"#awesome-bioinformatics-benchmarks-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA curated list of bioinformatics benchmarking papers and resources.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe credit for this format goes to Sean Davis for his \u003ca href=\"https://github.com/seandavi/awesome-single-cell\"\u003eawesome-single-cell\u003c/a\u003e repository and Ming Tang for his \u003ca href=\"https://github.com/crazyhottommy/ChIP-seq-analysis\"\u003eChIP-seq-analysis\u003c/a\u003e repository.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIf you have a benchmarking study that is not yet included on this list, please make a \u003ca href=\"https://github.com/j-andrews7/Awesome-Bioinformatics-Benchmarks/pulls\"\u003ePull Request\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eContents\u003c/h2\u003e\u003ca id=\"user-content-contents\" class=\"anchor\" aria-label=\"Permalink: Contents\" href=\"#contents\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#awesome-bioinformatics-benchmarks-\"\u003eAwesome Bioinformatics Benchmarks \u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#contents\"\u003eContents\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#rules-for-included-papers\"\u003eRules for Included Papers\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#format--organization\"\u003eFormat \u0026amp; Organization\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#benchmarking-theory\"\u003eBenchmarking Theory\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#toolmethod-sections\"\u003eTool/Method Sections\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#dnase-atac-and-chip-seq\"\u003eDNase, ATAC, and ChIP-seq\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#peak-callers\"\u003ePeak Callers\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#normalization-methods\"\u003eNormalization Methods\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#rna-seq\"\u003eRNA-seq\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#alignmentquantification-methods\"\u003eAlignment/Quantification Methods\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#normalisation-methods\"\u003eNormalisation Methods\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#differential-gene-expression\"\u003eDifferential Gene Expression\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#gene-set-enrichment-analysis\"\u003eGene Set Enrichment Analysis\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#differential-splicing\"\u003eDifferential Splicing\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#de-novo-assembly-and-quantification\"\u003e\u003cem\u003ede novo\u003c/em\u003e Assembly and Quantification\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#cell-type-deconvolution\"\u003eCell-Type Deconvolution\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#crispr-screens\"\u003eCRISPR Screens\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#dna-methylation\"\u003eDNA Methylation\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#platforms-and-library-prep-methods\"\u003ePlatforms and Library Prep Methods\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#cpg-methylation-from-nanopore-data\"\u003eCpg Methylation from Nanopore Data\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#variant-callers\"\u003eVariant Callers\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#germline-snpindel-callers\"\u003eGermline SNP/Indel Callers\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#somatic-snvindel-callers\"\u003eSomatic SNV/Indel callers\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#cnv-callers\"\u003eCNV Callers\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#sv-callers\"\u003eSV callers\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#single-cell\"\u003eSingle Cell\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#scrna-transformationsnormalization\"\u003escRNA Transformation/Normalization\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#gene-signature-scoring\"\u003eGene Signature Scoring\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#scrna-sequencing-protocols\"\u003escRNA Sequencing Protocols\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#scrna-analysis-pipelines\"\u003escRNA Analysis Pipelines\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#scrna-imputation-methods\"\u003escRNA Imputation Methods\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#scrna-differential-gene-expression\"\u003escRNA Differential Gene Expression\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#trajectory-inference\"\u003eTrajectory Inference\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#gene-regulatory-network-inference\"\u003eGene Regulatory Network Inference\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#integrationbatch-correction\"\u003eIntegration/Batch Correction\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#dimensionality-reduction\"\u003eDimensionality Reduction\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#cell-annotationinference\"\u003eCell Annotation/Inference\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#variant-calling\"\u003eVariant Calling\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#atac-seq\"\u003eATAC-seq\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#statistics\"\u003eStatistics\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#false-discovery-rates\"\u003eFalse Discovery Rates\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#microbiome\"\u003eMicrobiome\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#diversity-analysis\"\u003eDiversity analysis\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#hi-chi-chip\"\u003eHi-C/Hi-ChIP\u003c/a\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#tad-calling\"\u003eTAD Calling\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#normalization-methods-1\"\u003eNormalization Methods\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#contributors\"\u003eContributors\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRules for Included Papers\u003c/h2\u003e\u003ca id=\"user-content-rules-for-included-papers\" class=\"anchor\" aria-label=\"Permalink: Rules for Included Papers\" href=\"#rules-for-included-papers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ePapers must be objective comparisons of 3 or more tools/methods.\u003c/li\u003e\n\u003cli\u003ePapers must be \u003cstrong\u003eawesome\u003c/strong\u003e. This list isn't meant to chronicle every benchmarking study ever performed, only those that are particularly expansive, well done, and/or provide unique insights.\u003c/li\u003e\n\u003cli\u003ePapers should generally \u003cstrong\u003enot\u003c/strong\u003e be from authors showing why their tool/method is better than others.\u003c/li\u003e\n\u003cli\u003eBenchmarking data should be publicly available or simulation code/methods must be well-documented and reproducible.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eAdditional guidelines/rules may be added as necessary.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFormat \u0026amp; Organization\u003c/h2\u003e\u003ca id=\"user-content-format--organization\" class=\"anchor\" aria-label=\"Permalink: Format \u0026amp; Organization\" href=\"#format--organization\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePlease include the following information when adding papers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links (optional):\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePapers within each section should be ordered by publication date, with more recent papers listed first.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eBenchmarking Theory\u003c/h1\u003e\u003ca id=\"user-content-benchmarking-theory\" class=\"anchor\" aria-label=\"Permalink: Benchmarking Theory\" href=\"#benchmarking-theory\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1738-8\" rel=\"nofollow\"\u003eEssential guidelines for computational method benchmarking\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Lukas Weber, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, June 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper presents 10 main guidelines for conducting and writing benchmark papers covering necessary data, methods and metric choices, reproducible research, and documentation.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/s41467-019-09406-4\" rel=\"nofollow\"\u003eSystematic benchmarking of omics computational tools\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Sergei Mangul, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Communications, March 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e A survey of 25 benchmarking studies published between 2011 and 2017 in terms of design, methods, and information types. Discusses overfitting, sharing, incentives.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTool/Method Sections\u003c/h1\u003e\u003ca id=\"user-content-toolmethod-sections\" class=\"anchor\" aria-label=\"Permalink: Tool/Method Sections\" href=\"#toolmethod-sections\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAdditional sections/sub-sections can be added as needed.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDNase, ATAC, and ChIP-seq\u003c/h2\u003e\u003ca id=\"user-content-dnase-atac-and-chip-seq\" class=\"anchor\" aria-label=\"Permalink: DNase, ATAC, and ChIP-seq\" href=\"#dnase-atac-and-chip-seq\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePeak Callers\u003c/h3\u003e\u003ca id=\"user-content-peak-callers\" class=\"anchor\" aria-label=\"Permalink: Peak Callers\" href=\"#peak-callers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://academic.oup.com/bib/article/18/3/441/2453291\" rel=\"nofollow\"\u003eFeatures that define the best ChIP-seq peak calling algorithms\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Reuben Thomas, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Briefings in Bioinformatics, May 2017\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared six peak calling methods on 300 simulated and three real ChIP-seq data sets across a range of significance values.\nMethods were scored by sensitivity, precision, and F-score.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eGEM\u003c/code\u003e, \u003ccode\u003eMACS2\u003c/code\u003e, \u003ccode\u003eMUSIC\u003c/code\u003e, \u003ccode\u003eBCP\u003c/code\u003e, \u003ccode\u003eThreshold-based method (TM)\u003c/code\u003e, \u003ccode\u003eZINBA\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Varies. \u003ca href=\"http://ranger.sourceforge.net/manual1.18.html\" rel=\"nofollow\"\u003eBCP\u003c/a\u003e and \u003ca href=\"https://github.com/taoliu/MACS\"\u003eMACS2\u003c/a\u003e performed the best across all metrics on the simulated data.\nFor Tbx5 ChIP-seq, \u003ca href=\"http://groups.csail.mit.edu/cgs/gem/\" rel=\"nofollow\"\u003eGEM\u003c/a\u003e performed the best, with BCP also scoring highly.\nFor histone H3K36me3 and H3K4me3 data, all methods performed relatively comparably with the exception of ZINBA, which the authors could not get to run properly.\n\u003ca href=\"https://github.com/gersteinlab/MUSIC\"\u003eMUSIC\u003c/a\u003e and BCP had a slight edge over the others for the histone data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMore generally, they found that methods that utilize variable window sizes and Poisson test to rank peaks are more powerful than those that use a Binomial test.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0096303\" rel=\"nofollow\"\u003eA Comparison of Peak Callers Used for DNase-Seq Data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Hashem Koohy, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e PLoS ONE, May 2014\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compares four peak callers specificity and sensitivity on DNase-seq data from two publications composed of three cell types, using ENCODE data for the same cell types as a benchmark.\nThe authors tested multiple parameters for each caller to determine the best settings for DNase-seq data for each.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eF-seq\u003c/code\u003e, \u003ccode\u003eHotspot\u003c/code\u003e, \u003ccode\u003eMACS2\u003c/code\u003e, \u003ccode\u003eZINBA\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e \u003ca href=\"https://github.com/aboyle/F-seq\"\u003eF-seq\u003c/a\u003e was the most sensitive, though \u003ca href=\"https://github.com/taoliu/MACS\"\u003eMACS2\u003c/a\u003e and \u003ca href=\"https://github.com/rthurman/hotspot\"\u003eHotspot\u003c/a\u003e both performed competitively as well.\nZINBA was the least performant by a massive margin, requiring much more time to run, and was also the least sensitive.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eNormalization Methods\u003c/h3\u003e\u003ca id=\"user-content-normalization-methods\" class=\"anchor\" aria-label=\"Permalink: Normalization Methods\" href=\"#normalization-methods\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-020-00342-y\" rel=\"nofollow\"\u003eATAC-seq normalization method can significantly affect differential accessibility analysis and interpretation\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Jake J. Reske, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Epigenetics \u0026amp; Chromatin, April 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e The study examines how different ATAC-seq data normalization methods impact the analysis and interpretation of differential chromatin accessibility (DA). Using both in vivo and published yeast ATAC-seq datasets, the authors demonstrate that the choice of normalization method can significantly alter the identification of differentially accessible regions in the genome.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e  \u003ccode\u003eMACS2\u003c/code\u003e, \u003ccode\u003eDiffBind\u003c/code\u003e, \u003ccode\u003ecsaw\u003c/code\u003e, \u003ccode\u003evoom\u003c/code\u003e, \u003ccode\u003eDEseq2\u003c/code\u003e, \u003ccode\u003eedgeR\u003c/code\u003e, \u003ccode\u003elimma\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e  The authors recommend systematic comparison of multiple normalization methods before proceeding with differential accessibility analysis. They stress the importance of understanding the potential biases and assumptions of each method. They also propose a generalized workflow for differential ATAC-seq data analysis, emphasizing the need for standardizing molecular complexity before quantification. This workflow is based on the standardized ENCODE pipeline with modifications for ATAC-seq.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe authors also proposed a generalized workflow for differential accessibility analysis, which can be found in \u003ca href=\"https://github.com/reskejak/ATAC-seq\"\u003eGithub\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e  For ATAC-Seq data anlysis, there is another paper: \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1929-3\" rel=\"nofollow\"\u003eFrom reads to insight: a hitchhikerâs guide to ATAC-seq data analysis\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRNA-seq\u003c/h2\u003e\u003ca id=\"user-content-rna-seq\" class=\"anchor\" aria-label=\"Permalink: RNA-seq\" href=\"#rna-seq\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAlignment/Quantification Methods\u003c/h3\u003e\u003ca id=\"user-content-alignmentquantification-methods\" class=\"anchor\" aria-label=\"Permalink: Alignment/Quantification Methods\" href=\"#alignmentquantification-methods\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.biorxiv.org/content/10.1101/657874v2\" rel=\"nofollow\"\u003eAlignment and mapping methodology influence transcript abundance estimation\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Avi Srivastava* \u0026amp;  Laraib Malik*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e bioRXiv, October 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compares the influence of mapping and alignment on the accuracy of transcript quantification in both simulated and experimental data, as well as the effect on subsequent differential expression analysis.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eBowtie2\u003c/code\u003e, \u003ccode\u003eSTAR\u003c/code\u003e, \u003ccode\u003equasi-mapping\u003c/code\u003e, \u003ccode\u003eSelective Alignment\u003c/code\u003e, \u003ccode\u003eRSEM\u003c/code\u003e, \u003ccode\u003eSalmon\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e When trying to choose an approach, a choice can be made by the user performing the analysis based on any time-accuracy tradeoff they wish to make. In terms of speed, quasi-mapping is the fastest approach, followed by \u003ca href=\"https://github.com/COMBINE-lab/salmon\"\u003eSelective Alignment\u003c/a\u003e (SA) then \u003ca href=\"https://github.com/alexdobin/STAR\"\u003eSTAR\u003c/a\u003e. \u003ca href=\"http://bowtie-bio.sourceforge.net/bowtie2/index.shtml\" rel=\"nofollow\"\u003eBowtie2\u003c/a\u003e was considerably slower than all three of these approaches. However, in terms of accuracy, SA yielded the best results, followed by alignment to the genome (with subsequent transcriptomic projection) using STAR and SA (using carefully selected decoy sequences). Bowtie2 generally performed similarly to SA, but without the benefit of decoy sequences, seemed to admit more spurious mappings. Finally, lightweight mapping of sequencing reads to the transcriptome showed the lowest overall consistency with quantifications derived from the oracle alignments. Note: Both Selective Alignment and quasi-mapping are part of the \u003ca href=\"https://combine-lab.github.io/salmon/\" rel=\"nofollow\"\u003esalmon\u003c/a\u003e codebase.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1308-8\" rel=\"nofollow\"\u003eA comprehensive evaluation of ensembl, RefSeq, and UCSC annotations in the context of RNA-seq read mapping and gene quantification\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Shanrong Zhao* \u0026amp; Baohong Zhang\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e BMC Genomics, February 2015\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compares the effect of different gene annotations in the context of RNA-seq mapping and gene quantification using data from the Human Body Map 2.0 Project.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eEnsembl\u003c/code\u003e, \u003ccode\u003eRefseq\u003c/code\u003e, \u003ccode\u003eUCSC\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Though the authors warn there is no \"best\" set of annotations to use, they do emphasize the impact that annotation choice can have on downstream analyses such as differential gene expression, as genes with identical gene symbols can map to completely different regions in different annotations. Though Ensembl annotations are much more comprehensive than the others, the authors recommend a less complex genome annotation, such as the Refseq annotation, if the RNA-seq is being used as a replacement for microarrays. Conversely, the Ensembl annotations are preferrable if non-coding RNAs are of particular interest.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eNormalisation Methods\u003c/h3\u003e\u003ca id=\"user-content-normalisation-methods\" class=\"anchor\" aria-label=\"Permalink: Normalisation Methods\" href=\"#normalisation-methods\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://academic.oup.com/bib/article/19/5/776/3056951\" rel=\"nofollow\"\u003eSelecting between-sample RNA-Seq normalization methods from the perspective of their assumptions\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Ciaran Evans, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Briefings in Bioinformatics, February 2017\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study underscores the importance of selecting appropriate normalization methods in RNA-Seq studies based on their underlying assumptions. It highlights how normalization choices impact gene behavior analysis under various biological conditions. The paper emphasizes the critical role of assumptions in normalization methods and their effects on the accuracy of downstream analyses, such as detecting differential expression. The study utilized simulations to evaluate normalization methods under varying conditions of mRNA/cell levels and asymmetry in gene expression, assessing error in fold change estimates and empirical error rates in detecting differential expression.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e The paper provides a general discussion on RNA-Seq normalization methods, focusing on the types (e.g., normalization by library size, normalization by distribution, normalization by testing) rather than specific tools. The simulations used various methods, including DEGES, DESeq, Oracle, PoissonSeq, TMM, and Total Count.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The authors suggest selecting normalization methods based on the specific conditions of the biological experiment. Library size normalization is effective when total mRNA/cell is consistent across conditions, whereas normalization by distribution/testing is better suited for conditions with symmetry in differential expression, regardless of mRNA/cell differences. In scenarios with both asymmetry and varying mRNA/cell levels, both methods are expected to perform poorly.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors' simulation code is available on \u003ca href=\"https://github.com/ciaranlevans/rnaSeqAssumptions\"\u003eGithub\u003c/a\u003e.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://academic.oup.com/bioinformatics/article/31/22/3625/240923\" rel=\"nofollow\"\u003eStatistical models for RNA-seq data derived from a two-condition 48-replicate experiment\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Marek GierliÅski*, Christian Cole*, PietÃ¡ Schofield*, Nicholas J. Schurch*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Bioinformatics, November 2015\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compares the effect of normal, log-normal, and negative binomial distribution assumptions on RNA-seq gene read-counts from 48 RNA-seq replicates.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003enormal\u003c/code\u003e, \u003ccode\u003elog-normal\u003c/code\u003e, \u003ccode\u003enegative binomial\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Assuming a normal distribution leads to a large number of false positives during differential gene expression. A log-normal distribution model works well unless a sample contains zero counts. Use tools that assume a negative binomial distribution (\u003ccode\u003eedgeR\u003c/code\u003e, \u003ccode\u003eDESeq\u003c/code\u003e, \u003ccode\u003eDESeq2\u003c/code\u003e, etc).\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://academic.oup.com/bib/article/14/6/671/189645\" rel=\"nofollow\"\u003eA comprehensive evaluation of normalization methods for Illumina high-throughput RNA sequencing data analysis\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Marie-AgnÃ¨s Dillies, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Briefings in Bioinformatics, November 2013\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared seven RNA-seq normalization methods in the context of differential expression analysis on four real datasets and thousands of simulations.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eTotal Count (TC)\u003c/code\u003e, \u003ccode\u003eUpper Quartile (UQ)\u003c/code\u003e, \u003ccode\u003eMedian (Med),\u003c/code\u003e \u003ccode\u003eDESeq\u003c/code\u003e, \u003ccode\u003eedgeR\u003c/code\u003e, \u003ccode\u003eQuantile (Q)\u003c/code\u003e, \u003ccode\u003eRPKM\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The authors recommend DESeq (now deprecated and replaced by \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/DESeq2.html\" rel=\"nofollow\"\u003eDESeq2\u003c/a\u003e) or \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/edgeR.html\" rel=\"nofollow\"\u003eedgeR\u003c/a\u003e, as those methods are robust to the presence of different library sizes and compositions, whereas the (still common) Total Count and RPKM methods are ineffective and should be abandoned.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDifferential Gene Expression\u003c/h3\u003e\u003ca id=\"user-content-differential-gene-expression\" class=\"anchor\" aria-label=\"Permalink: Differential Gene Expression\" href=\"#differential-gene-expression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pubmed/30726870\" rel=\"nofollow\"\u003eHow well do RNA-Seq differential gene expression tools perform in a complex eukaryote? A case study in Arabidopsis thaliana.\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Kimon Froussios*, Nick J Schurch*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Bioinformatics, February 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared nine differential gene expression tools (and their underlying model distributions) in 17 RNA-seq replicates of \u003cem\u003eArabidopsis thaliana\u003c/em\u003e.\nHandling of inter-replicate variability and false positive fraction were the benchmarking metrics used.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003ebaySeq\u003c/code\u003e, \u003ccode\u003eDEGseq\u003c/code\u003e, \u003ccode\u003eDESeq\u003c/code\u003e, \u003ccode\u003eDESeq2\u003c/code\u003e, \u003ccode\u003eEBSeq\u003c/code\u003e, \u003ccode\u003eedgeR\u003c/code\u003e, \u003ccode\u003elimma\u003c/code\u003e, \u003ccode\u003ePoisson-Seq\u003c/code\u003e, \u003ccode\u003eSAM-Seq\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Six of the tools that utilize negative binomial or log-normal distributions (\u003ca href=\"https://bioconductor.org/packages/release/bioc/html/edgeR.html\" rel=\"nofollow\"\u003eedgeR\u003c/a\u003e, \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/DESeq2.html\" rel=\"nofollow\"\u003eDESeq2\u003c/a\u003e, DESeq (now deprecated and replaced by \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/DESeq2.html\" rel=\"nofollow\"\u003eDESeq2\u003c/a\u003e, \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/baySeq.html\" rel=\"nofollow\"\u003ebaySeq\u003c/a\u003e, \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/limma.html\" rel=\"nofollow\"\u003elimma\u003c/a\u003e, and \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/EBSeq.html\" rel=\"nofollow\"\u003eEBseq\u003c/a\u003e control their identification of false positives well.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors released their benchmarking scripts on \u003ca href=\"https://github.com/bartongroup/KF_arabidopsis-GRNA\"\u003eGithub\u003c/a\u003e.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pubmed/27022035\" rel=\"nofollow\"\u003eHow many biological replicates are needed in an RNA-seq experiment and which differential expression tool should you use?\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Nicholas J. Schurch*, PietÃ¡ Schofield*, Marek GierliÅski*, Christian Cole*, Alexander Sherstnev*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e RNA, March 2016\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared 11 differential expression tools on varying numbers of RNA-seq biological replicates (3-42) between two conditions.\nEach tool was compared against itself as a standard (using all replicates) and against the other tools.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003ebaySeq\u003c/code\u003e, \u003ccode\u003ecuffdiff\u003c/code\u003e, \u003ccode\u003eDEGSeq\u003c/code\u003e, \u003ccode\u003eDESeq\u003c/code\u003e, \u003ccode\u003eDESeq2\u003c/code\u003e, \u003ccode\u003eEBSeq\u003c/code\u003e, \u003ccode\u003eedgeR (exact and glm modes)\u003c/code\u003e, \u003ccode\u003elimma\u003c/code\u003e, \u003ccode\u003eNOISeq\u003c/code\u003e, \u003ccode\u003ePoissonSeq\u003c/code\u003e, \u003ccode\u003eSAMSeq\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e With fewer than 12 biological replicates, \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/edgeR.html\" rel=\"nofollow\"\u003eedgeR\u003c/a\u003e and \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/DESeq2.html\" rel=\"nofollow\"\u003eDESeq2\u003c/a\u003e were the top performers.\nAs replicates increased, DESeq (now deprecated and replaced by \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/DESeq2.html\" rel=\"nofollow\"\u003eDESeq2\u003c/a\u003e did a better job minimizing false positives than other tools.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAdditionally, the authors recommend at least six biological replicates should be used, rising to at least 12 if users want to identify all significantly differentially expressed genes no matter the fold change magnitude.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-9-r95\" rel=\"nofollow\"\u003eComprehensive evaluation of differential gene expression analysis methods for RNA-seq data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Franck Rapaport, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, September 2013\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared six differential expression methods on three cell line data sets from ENCODE (GM12878, H1-hESC, and MCF-7) and two samples from the SEQC study, which had a large fraction of differentially expressed genes validated by qRT-PCR.\nSpecificity, sensitivity, and false positive rate were the main benchmarking metrics used.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eCuffdiff\u003c/code\u003e, \u003ccode\u003eedgeR\u003c/code\u003e, \u003ccode\u003eDESeq\u003c/code\u003e, \u003ccode\u003ePoissonSeq\u003c/code\u003e, \u003ccode\u003ebaySeq\u003c/code\u003e, \u003ccode\u003elimma\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Though no method emerged as favorable in all conditions, those that used negative binomial modeling (DESeq (now deprecated and replaced by \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/DESeq2.html\" rel=\"nofollow\"\u003eDESeq2\u003c/a\u003e), \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/edgeR.html\" rel=\"nofollow\"\u003eedgeR\u003c/a\u003e, \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/baySeq.html\" rel=\"nofollow\"\u003ebaySeq\u003c/a\u003e) generally performed best.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe more replicates, the better. Replicate numbers (both biological and technical) have a greater impact on differential detection accuracy than sequencing depth.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGene Set Enrichment Analysis\u003c/h3\u003e\u003ca id=\"user-content-gene-set-enrichment-analysis\" class=\"anchor\" aria-label=\"Permalink: Gene Set Enrichment Analysis\" href=\"#gene-set-enrichment-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1093/bib/bbz158\" rel=\"nofollow\"\u003eToward a gold standard for benchmarking gene set enrichment analysis\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Ludwig Geistlinger, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Briefings in Bioinformatics, February 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper developed a Bioconductor package for reproducible GSEA benchmarking, and used the package to assess 10 widely used enrichment methods with regard to runtime and applicability to RNA-seq data, fraction of enriched gene sets depending on the null hypothesis tested, and recovery of biologically relevant gene sets. The framework can be extended to additional methods, datasets, and benchmark criteria, and should serve as a gold standard for future GSEA benchmarking studies.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e The paper quantitatively asseses the performance of 10 enrichment methods (\u003ccode\u003eORA\u003c/code\u003e, \u003ccode\u003eGSEA\u003c/code\u003e, \u003ccode\u003eGSA\u003c/code\u003e, \u003ccode\u003ePADOG\u003c/code\u003e, \u003ccode\u003eSAFE\u003c/code\u003e, \u003ccode\u003eCAMERA\u003c/code\u003e, \u003ccode\u003eROAST\u003c/code\u003e, \u003ccode\u003eGSVA\u003c/code\u003e, \u003ccode\u003eGLOBALTEST\u003c/code\u003e, \u003ccode\u003eSAMGS\u003c/code\u003e). The paper also compares 10 frequently used enrichment tools implementing these methods (\u003ccode\u003eDAVID\u003c/code\u003e, \u003ccode\u003eENRICHR\u003c/code\u003e, \u003ccode\u003eCLUSTER-PROFILER\u003c/code\u003e, \u003ccode\u003eGOSTATS\u003c/code\u003e, \u003ccode\u003eWEBGESTALT\u003c/code\u003e, \u003ccode\u003eG:PROFILER\u003c/code\u003e, \u003ccode\u003eGENETRAIL\u003c/code\u003e, \u003ccode\u003eGORILLA\u003c/code\u003e, \u003ccode\u003eTOPPGENE\u003c/code\u003e, \u003ccode\u003ePANTHER\u003c/code\u003e).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ccode\u003eORA\u003c/code\u003e for the exploratory analysis of simple gene lists, pre-ranked \u003ccode\u003eGSEA\u003c/code\u003e or pre-ranked \u003ccode\u003eCAMERA\u003c/code\u003e for the analysis of pre-ranked gene lists accompanied by gene scores such as fold changes, \u003cbr\u003e\u003c/li\u003e\n\u003cli\u003eFor enrichment analysis on the full expression matrix (genes x samples), the paper recommends to provide normalized log2 intensities for microarray data and logTPMs (or logRPKMs/logFPKMs) for RNA-seq data; when given raw read counts, the paper recommends to apply a variance-stabilizing transformation such as \u003ccode\u003evoom\u003c/code\u003e to arrive at library-size normalized logCPMs, \u003cbr\u003e\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003eROAST\u003c/code\u003e (sample group comparisons) or \u003ccode\u003eGSVA\u003c/code\u003e (single sample) if the question of interest is to test for association of any gene in the set with the phenotype (self-contained null hypothesis), \u003cbr\u003e\u003c/li\u003e\n\u003cli\u003e\u003ccode\u003ePADOG\u003c/code\u003e (simple experimental designs) or \u003ccode\u003eSAFE\u003c/code\u003e (extended experimental designs) if the question of interest is to test for excess of differential expression in a gene set relative to genes outside the set (competitive null hypothesis).\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e \u003ca href=\"http://bioconductor.org/packages/GSEABenchmarkeR\" rel=\"nofollow\"\u003ehttp://bioconductor.org/packages/GSEABenchmarkeR\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDifferential Splicing\u003c/h3\u003e\u003ca id=\"user-content-differential-splicing\" class=\"anchor\" aria-label=\"Permalink: Differential Splicing\" href=\"#differential-splicing\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://humgenomics.biomedcentral.com/articles/10.1186/1479-7364-8-3\" rel=\"nofollow\"\u003eA survey of software for genome-wide discovery of differential splicing in RNA-Seq data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Joan E Hooper\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Human Genomics, January 2014\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compares the methodologies, advantages, and disadvantages of eight differential splicing analysis tools, detailing use-cases and features for each.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eCuffdiff2\u003c/code\u003e, \u003ccode\u003eMISO\u003c/code\u003e, \u003ccode\u003eDEXSeq\u003c/code\u003e, \u003ccode\u003eDSGseq\u003c/code\u003e, \u003ccode\u003eMATS\u003c/code\u003e, \u003ccode\u003eDiffSplice\u003c/code\u003e, \u003ccode\u003eSplicing compass\u003c/code\u003e, \u003ccode\u003eAltAnalyze\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e This is a true breakdown of each tools' advantages and disadvantages.\nThe author makes no recommendation due to the performance reliance on experimental setup, data type (e.g. \u003ccode\u003eAltAnalyze\u003c/code\u003e works best on junction + exon microarrays), and user objectives.\nTable 1 provides a good comparison of the features and methodology of each method.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://academic.oup.com/bib/article/20/2/471/4524048\" rel=\"nofollow\"\u003eA benchmarking of workflows for detecting differential splicing and differential expression at isoform level in human RNA-seq studies\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Gabriela A Merino\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Briefings in Bioinformatics, March 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compares nine most commonly used workflows to detect differential isoform expression and splicing.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eEBSeq\u003c/code\u003e, \u003ccode\u003eDESeq2\u003c/code\u003e, \u003ccode\u003eNOISeq\u003c/code\u003e, \u003ccode\u003eLimma\u003c/code\u003e, \u003ccode\u003eLimmaDS\u003c/code\u003e, \u003ccode\u003eDEXSeq\u003c/code\u003e, \u003ccode\u003eCufflinks\u003c/code\u003e, \u003ccode\u003eCufflinksDS\u003c/code\u003e, \u003ccode\u003eSplicingCompass\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e DESeq2, Limma and NOISeq for differential isoform expression(DIE) analysis and DEXSeq and LimmaDS for differential splicing (DS) testing.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e\u003cem\u003ede novo\u003c/em\u003e Assembly and Quantification\u003c/h3\u003e\u003ca id=\"user-content-de-novo-assembly-and-quantification\" class=\"anchor\" aria-label=\"Permalink: de novo Assembly and Quantification\" href=\"#de-novo-assembly-and-quantification\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1093/bioinformatics/btv488\" rel=\"nofollow\"\u003eBenchmark analysis of algorithms for determining and quantifying full-length mRNA splice forms from RNA-seq data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Katharina E. Hayer et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Bioinformatics, Dec 2015\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared both guided and de novo transcript reconstruction algorithms using simulated and in vitro transcription (IVT) generated libraries. Precision/recall metrics were obtained by comparing the reconstructed transcripts to their true models.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eCufflinks\u003c/code\u003e, \u003ccode\u003eCLASS\u003c/code\u003e, \u003ccode\u003eFlipFlop\u003c/code\u003e, \u003ccode\u003eIReckon\u003c/code\u003e, \u003ccode\u003eIsoLasso\u003c/code\u003e, \u003ccode\u003eMiTie\u003c/code\u003e, \u003ccode\u003eStringTie\u003c/code\u003e, \u003ccode\u003eStringTie-SR\u003c/code\u003e, \u003ccode\u003eAUGUSTUS\u003c/code\u003e, \u003ccode\u003eTrinity\u003c/code\u003e, \u003ccode\u003eSOAP\u003c/code\u003e, \u003ccode\u003eTrans-ABySS\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\nAll tools measured produced less than ideal precision-recall (both \u0026lt;90%) when using imperfect simulated or IVT data and genes producing mulitple isoforms. \u003ca href=\"https://github.com/cole-trapnell-lab/cufflinks\"\u003eCufflinks\u003c/a\u003e and \u003ca href=\"https://github.com/gpertea/stringtie\"\u003eStringTie\u003c/a\u003e are among the best performers.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://academic.oup.com/gigascience/article/8/5/giz039/5488105\" rel=\"nofollow\"\u003eDe novo transcriptome assembly: A comprehensive cross-species comparison of short-read RNA-Seq assemblers\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Martin Holzer \u0026amp; Manja Marz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e GigaScience, May 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compares 10 \u003cem\u003ede novo\u003c/em\u003e assembly tools across 9 RNA-seq datasets spanning multiple species and kingdoms for 20 biological-based and reference-free metrics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eTrinity\u003c/code\u003e, \u003ccode\u003eOases\u003c/code\u003e, \u003ccode\u003eTrans-ABySS\u003c/code\u003e, \u003ccode\u003eSOAPdenovo-Trans\u003c/code\u003e, \u003ccode\u003eIDBA-Tran\u003c/code\u003e, \u003ccode\u003eBridger\u003c/code\u003e, \u003ccode\u003eBinPacker\u003c/code\u003e, \u003ccode\u003eShannon\u003c/code\u003e, \u003ccode\u003eSPAdes-sc\u003c/code\u003e, \u003ccode\u003eSPAdes-rna\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The authors found that no tool's performance was dominant for all data sets, but \u003ca href=\"https://github.com/trinityrnaseq/trinityrnaseq\"\u003eTrinity\u003c/a\u003e, \u003ca href=\"https://cab.spbu.ru/software/spades/\" rel=\"nofollow\"\u003eSPAdes\u003c/a\u003e, and \u003ca href=\"https://github.com/bcgsc/transabyss\"\u003eTrans-ABySS\u003c/a\u003e were typically among the best. For assembly evaluation, the authors recommend a hybrid approach combining both biological-based (BUSCO, # of full length transcripts) and reference-free metric (e.g. \u003ccode\u003eTransRate\u003c/code\u003e, \u003ccode\u003eDETONATE\u003c/code\u003e).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors provide a comprehensive \u003ca href=\"https://www.rna.uni-jena.de/supplements/assembly/index.html\" rel=\"nofollow\"\u003eelectronic supplement website\u003c/a\u003e containing all metrics and assembly commands in addition to many supplementary figures.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCell-Type Deconvolution\u003c/h3\u003e\u003ca id=\"user-content-cell-type-deconvolution\" class=\"anchor\" aria-label=\"Permalink: Cell-Type Deconvolution\" href=\"#cell-type-deconvolution\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://academic.oup.com/bioinformatics/article/35/14/i436/5529146\" rel=\"nofollow\"\u003eComprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Markus List*, Tatsiana Aneichyk*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Bioinformatics, July 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper benchmarks and compares seven methods for computational deconvolution of cell-type abundance in bulk RNA-seq samples. Each method was tested on both simulated and true bulk RNA-seq samples validated by FACS.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003equanTIseq\u003c/code\u003e, \u003ccode\u003eTIMER\u003c/code\u003e, \u003ccode\u003eCIBERSORT\u003c/code\u003e, \u003ccode\u003eCIBERSORT abs. mode\u003c/code\u003e, \u003ccode\u003eMCPCounter\u003c/code\u003e, \u003ccode\u003exCell\u003c/code\u003e, \u003ccode\u003eEPIC\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Varies. In general, the authors recommend \u003ca href=\"https://github.com/GfellerLab/EPIC\"\u003eEPIC\u003c/a\u003e and \u003ca href=\"http://icbi.at/software/quantiseq/doc/index.html\" rel=\"nofollow\"\u003equanTIseq\u003c/a\u003e due to their overall robustness and absolute (rather than relative) scoring, though \u003ca href=\"https://comphealth.ucsf.edu/app/xcell\" rel=\"nofollow\"\u003exCell\u003c/a\u003e is recommended for binary presence/absence of cell types and \u003ca href=\"https://github.com/ebecht/MCPcounter\"\u003eMCPcounter\u003c/a\u003e was their recommended relative scoring method.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors created an \u003ca href=\"https://github.com/icbi-lab/immunedeconv\"\u003eR package called immunedeconv\u003c/a\u003e for easy installation and use of all these methods. For developers, they have made their \u003ca href=\"https://github.com/icbi-lab/immune_deconvolution_benchmark\"\u003ebenchmarking pipeline\u003c/a\u003e available so that others can reproduce/extend it to test their own tools/methods.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.biorxiv.org/content/10.1101/2020.01.10.897116v1.full\" rel=\"nofollow\"\u003eComprehensive benchmarking of computational deconvolution of transcriptomics data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Francisco Avila Cobos, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e bioRxiv, January 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared the effects of transformation, scaling/normalization, marker selection, cell type composition, and deconvolution methods on computing cell type proportions in mixed bulk RNA-seq samples. Performance is assessed by means of Pearson correlation and root-mean-square-error (RMSE) between the cell type proportions computed by the different deconvolution methods and known compositions of 1000 pseudo-bulk mixtures from 4 different single cell RNA-seq datasets with varying numbers of cells.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e Transformation methods: \u003ccode\u003elinear (none)\u003c/code\u003e, \u003ccode\u003elog\u003c/code\u003e, \u003ccode\u003eVST (DESeq2)\u003c/code\u003e, \u003ccode\u003esqrt\u003c/code\u003e. Scaling/normalization methods (bulk): \u003ccode\u003ecolumn-wise\u003c/code\u003e, \u003ccode\u003emin-max\u003c/code\u003e, \u003ccode\u003ez-score\u003c/code\u003e, \u003ccode\u003eQN\u003c/code\u003e, \u003ccode\u003eUQ\u003c/code\u003e, \u003ccode\u003erow-wise\u003c/code\u003e, \u003ccode\u003eglobal min-max\u003c/code\u003e, \u003ccode\u003eglobal z-score\u003c/code\u003e, \u003ccode\u003eTPM\u003c/code\u003e, \u003ccode\u003eTMM\u003c/code\u003e, \u003ccode\u003emedian ratios\u003c/code\u003e, \u003ccode\u003eLogNormalize\u003c/code\u003e. Scaling/normalization methods (single cell): \u003ccode\u003eRNBR\u003c/code\u003e, \u003ccode\u003escran\u003c/code\u003e, \u003ccode\u003escater\u003c/code\u003e, \u003ccode\u003eLinnorm\u003c/code\u003e. Deconvolution methods (bulk): \u003ccode\u003eOLS\u003c/code\u003e, \u003ccode\u003eNNLS\u003c/code\u003e, \u003ccode\u003eFARDEEP\u003c/code\u003e, \u003ccode\u003eRLR\u003c/code\u003e, \u003ccode\u003elasso\u003c/code\u003e, \u003ccode\u003eridge\u003c/code\u003e, \u003ccode\u003eDCQ\u003c/code\u003e, \u003ccode\u003eelastic net\u003c/code\u003e, \u003ccode\u003eDSA\u003c/code\u003e, \u003ccode\u003eEPIC\u003c/code\u003e, \u003ccode\u003eCIBERSORT\u003c/code\u003e, \u003ccode\u003edtangle\u003c/code\u003e, \u003ccode\u003essFrobenius\u003c/code\u003e, \u003ccode\u003essKL\u003c/code\u003e, \u003ccode\u003eDeconRNASeq\u003c/code\u003e. Deconvolution methods (scRNA-seq reference): \u003ccode\u003eBisqueRNA\u003c/code\u003e, \u003ccode\u003edeconvSeq\u003c/code\u003e, \u003ccode\u003eDWLS\u003c/code\u003e, \u003ccode\u003eMuSiC\u003c/code\u003e, \u003ccode\u003eSCDC\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The authors strongly recommend keeping data in the linear scale for deconvolution, avoiding the use of \u003ccode\u003ecolumn min-max\u003c/code\u003e, \u003ccode\u003ecolumn z-score\u003c/code\u003e, and \u003ccode\u003eQN\u003c/code\u003e for normalization/scaling of bulk RNA-seq data, avoiding the use of \u003ccode\u003erow-normalization\u003c/code\u003e, \u003ccode\u003ecolumn min-max\u003c/code\u003e, and \u003ccode\u003eTPM\u003c/code\u003e for normalization/scaling of single cell RNA-seq data, use all possible cell markers, ensure that all possible cell types are represented in the reference matrix, and use one of the top performing deconvolution methods - \u003ccode\u003eOLS\u003c/code\u003e, \u003ccode\u003ennls\u003c/code\u003e, \u003ccode\u003eRLR\u003c/code\u003e, \u003ccode\u003eFARDEEP\u003c/code\u003e, \u003ccode\u003eCIBERSORT\u003c/code\u003e, \u003ccode\u003eDWLS\u003c/code\u003e, \u003ccode\u003eMuSiC\u003c/code\u003e, or \u003ccode\u003eSCDC\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors provide their benchmarking code \u003ca href=\"https://github.com/favilaco/deconv_benchmark\"\u003eon Github\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCRISPR Screens\u003c/h2\u003e\u003ca id=\"user-content-crispr-screens\" class=\"anchor\" aria-label=\"Permalink: CRISPR Screens\" href=\"#crispr-screens\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHit/Dependency Identification\u003c/h3\u003e\u003ca id=\"user-content-hitdependency-identification\" class=\"anchor\" aria-label=\"Permalink: Hit/Dependency Identification\" href=\"#hitdependency-identification\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1186/s13059-020-01972-x\" rel=\"nofollow\"\u003eA benchmark of algorithms for the analysis of pooled CRISPR screens\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Sunil Bodapati*, Timothy P. Daley*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, March 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study evaluates and compares various algorithms used for analyzing data from pooled CRISPR screens, using a comprehensive simulation framework and real datasets. The algorithms were benchmarked for their ability to accurately identify essential genes in CRISPR knockout (CRISPRko), CRISPR interference (CRISPRi), and CRISPR activation (CRISPRa) screens. Key parameters such as the number of guides per gene, guide binding efficiency, sequencing depth, and control guides were varied to assess the robustness of these algorithms under different conditions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e The study compared several algorithms, including Redundant siRNA Activity (RSA), MAGeCK Robust Ranking Algorithm (RRA), HiTSelect, MAGeCK Maximum Likelihood Estimation (MLE), CRISPhieRmix, CERES, JACKS, and a standard t test. BAGEL was also discussed, but not included in the testing due to its requirement of a priori knowledge.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eMAGeCK RRA is recommended for general use due to its robustness and consistent performance across various conditions.\u003c/li\u003e\n\u003cli\u003eCRISPhieRmix is suggested for screens with high variable guide efficiency, such as in CRISPRi or CRISPRa screens.\u003c/li\u003e\n\u003cli\u003eFor multiple screens across different cell types or lines, MAGeCK MLE, JACKS, and CERES are good options.\u003c/li\u003e\n\u003cli\u003eThe study also highlights that a simple t test can be effective when suitable control guides are used.\u003c/li\u003e\n\u003cli\u003eThe number of guides per gene is crucial, with four being the minimum recommended for reliable results. More guides may be necessary for low-signal phenotypes.\u003c/li\u003e\n\u003cli\u003eSequencing depth is less critical than previously thought, with the performance of most algorithms plateauing at 25 reads per guide.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The simulation framework and scripts used in the study are available at \u003ca href=\"https://github.com/sbodapati/CRISPR_Benchmarking_Algorithms\"\u003eGitHub - CRISPR Benchmarking Algorithms\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDNA Methylation\u003c/h2\u003e\u003ca id=\"user-content-dna-methylation\" class=\"anchor\" aria-label=\"Permalink: DNA Methylation\" href=\"#dna-methylation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlatforms and Library Prep Methods\u003c/h3\u003e\u003ca id=\"user-content-platforms-and-library-prep-methods\" class=\"anchor\" aria-label=\"Permalink: Platforms and Library Prep Methods\" href=\"#platforms-and-library-prep-methods\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/s41598-019-46875-5\" rel=\"nofollow\"\u003eSystematic evaluation of library preparation methods and sequencing platforms for high-throughput whole genome bisulfite sequencing\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Li Zhou, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Scientific Reports, 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study focuses on improving strategies for whole genome bisulfite sequencing (WGBS) used in large-scale epidemiological studies. It systematically compares three WGBS library preparation methods (Swift Biosciences Accel-NGS, Illumina TruSeq, and QIAGEN QIAseq) across two Illumina sequencing platforms (NovaSeq and HiSeq X). The study also examines the concordance between WGBS and methylation array data. The study assayed quality metrics (Q20 and Q30 fractions), insert size, adaptor contamination, overlapping bases, read duplication rate, alignment rate, coverage depth, and genome coverage uniformity across the platforms.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eLibrary Preparation Methods: Swift Biosciences Accel-NGS, Illumina TruSeq, QIAGEN QIAseq.\u003c/li\u003e\n\u003cli\u003eSequencing Platforms: Illumina NovaSeq, HiSeq X.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eSwift Biosciences Accel-NGS achieved the highest proportion of CpG sites assayed and effective coverage, making it the most recommended method.\u003c/li\u003e\n\u003cli\u003eIllumina TruSeq had a high proportion of PCR duplicates, while QIAGEN QIAseq generally underperformed across all quality metrics.\u003c/li\u003e\n\u003cli\u003eNovaSeq and HiSeq X platforms showed similar performance, except for a higher read duplication rate in NovaSeq.\u003c/li\u003e\n\u003cli\u003eWGBS was less precise than methylation arrays, requiring a minimum coverage of 100x for comparable precision.\u003c/li\u003e\n\u003cli\u003eSwift outperformed other methods in terms of uniform genome coverage and CpG site coverage at lower depths.\u003c/li\u003e\n\u003cli\u003eNo significant differences in nucleotide amplification bias were observed between NovaSeq and HiSeq X.\u003c/li\u003e\n\u003cli\u003eFor quantifying DNA methylation, WGBS with Swift on HiSeq X outperformed methylation arrays in CpG coverage.\u003c/li\u003e\n\u003c/ul\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1038/s41587-022-01336-9\" rel=\"nofollow\"\u003eComparison and imputation-aided integration of five commercial platforms for targeted DNA methylome analysis\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Miljana TaniÄ, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Biotechnology, October 2022.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study benchmarks five commercial Targeted Bisulfite Sequencing (TBS) platforms for analyzing human DNA methylomes at base-pair resolution. The platforms include three hybridization capture-based (Agilent, Roche, and Illumina) and two reduced-representation-based (Diagenode and NuGen). Eleven sample types were analyzed, including cell lines and DNA methylation standards. Two samples were also compared with whole-genome DNA methylation sequencing using Illumina and Oxford Nanopore platforms. Key assessment parameters included workflow complexity, on/off-target performance, coverage accuracy, and reproducibility.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eHybridization capture-based platforms: Agilent SureSelect Methyl-Seq, Roche NimbleGen SeqCap EpiGiant, and Illumina TruSeq Methyl Capture EPIC.\u003c/li\u003e\n\u003cli\u003eReduced-representation-based platforms: Diagenode Premium RRBS and NuGen Ovation RRBS Methyl-Seq.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eEach platform exhibited strengths and limitations in coverage, reproducibility, and concordance of DNA methylation levels.\u003c/li\u003e\n\u003cli\u003eRRBS platforms (Diagenode and NuGen) showed more uniform CpG coverage compared to hybridization capture-based methods (Agilent, Roche, Illumina).\u003c/li\u003e\n\u003cli\u003eIlluminaâs platform had the highest on-target capture efficiency (~90.6%), followed by Agilent's (~78.2%) and Roche's (~61.5%).\u003c/li\u003e\n\u003cli\u003eDifferences in CpG coverage between platforms indicated varying suitability for specific genomic features. For instance, RRBS platforms had higher coverage in CpG islands and shores, while Roche excelled in covering open-sea regions and Illumina in enhancer regions.\u003c/li\u003e\n\u003cli\u003eHigh intra-platform and inter-platform correlation in DNA methylation levels were observed, except for Diagenode, which slightly underperformed.\u003c/li\u003e\n\u003cli\u003eAll TBS platforms showed a strong correlation to whole-genome bisulfite sequencing (WGBS) and Oxford Nanopore data, suggesting their reliability for DNA methylome analysis.\u003c/li\u003e\n\u003cli\u003eThe study suggests considering specific genomic feature coverage and desired analysis depth when choosing a TBS platform.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors provide their analysis code on \u003ca href=\"https://github.com/ucl-medical-genomics/EpiCapture\"\u003eGithub\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCpG Methylation from Nanopore Data\u003c/h3\u003e\u003ca id=\"user-content-cpg-methylation-from-nanopore-data\" class=\"anchor\" aria-label=\"Permalink: CpG Methylation from Nanopore Data\" href=\"#cpg-methylation-from-nanopore-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1038/s41467-021-23778-6\" rel=\"nofollow\"\u003eSystematic benchmarking of tools for CpG methylation detection from nanopore sequencing\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Zaka Wing-Sze Yuen, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Communications, 2021\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study systematically benchmarks six tools for detecting 5-methylcytosine (5mC) from nanopore sequencing using individual reads, controlled methylation mixtures, Cas9-targeted sequencing, and whole-genome bisulfite sequencing (WGBS). The research highlights a trade-off between true positives and false positives among these tools and a general high dispersion in predicting methylation frequencies. Metrics include accuracy (true positive rate \u0026amp; true negative rate) at the individual read level and per controlled mixture. The authors also tested a consensus approach combining the results of pairs of callers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eNanopolish\u003c/code\u003e, \u003ccode\u003eMegalodon\u003c/code\u003e, \u003ccode\u003eDeepSignal\u003c/code\u003e, \u003ccode\u003eGuppy\u003c/code\u003e, \u003ccode\u003eTombo\u003c/code\u003e, and \u003ccode\u003eDeepMod\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eNo single method accurately predicts across all methylation frequency ranges.\u003c/li\u003e\n\u003cli\u003eGuppy excels in identifying unmethylated sites but fails at fully methylated sites.\u003c/li\u003e\n\u003cli\u003eNanopolish and Tombo accurately recover fully methylated sites but have high false positives at unmethylated sites.\u003c/li\u003e\n\u003cli\u003eMegalodon showed the best overall performance but requires GPU support.\u003c/li\u003e\n\u003cli\u003eThe consensus approach METEORE, combining predictions from two or more tools (specifically Megalodon and DeepSignal), improved accuracy over individual methods. It balances accuracy and running times, although it also requires a GPU for efficiency.\u003c/li\u003e\n\u003cli\u003eOn a CPU, the combination of Nanopolish and DeepSignal can match Megalodon's accuracy and be time-competitive.\u003c/li\u003e\n\u003cli\u003eReassessing score cutoffs for individual reads and removing sites with uncertain methylation status can further enhance accuracy.\u003c/li\u003e\n\u003cli\u003eThese methods showed good consistency with WGBS data, suggesting potential for sensitive diagnostic and forensic tests without high coverage.\u003c/li\u003e\n\u003cli\u003eThe study also notes that the highest discrepancy with WGBS occurred at CG sites in AT-rich sequences, particularly for DeepMod and DeepSignal.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e\nThe authors provide their method for consensus calling (METEORE) on \u003ca href=\"https://github.com/comprna/METEORE\"\u003eGitHub\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVariant Callers\u003c/h2\u003e\u003ca id=\"user-content-variant-callers\" class=\"anchor\" aria-label=\"Permalink: Variant Callers\" href=\"#variant-callers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGermline SNP/Indel Callers\u003c/h3\u003e\u003ca id=\"user-content-germline-snpindel-callers\" class=\"anchor\" aria-label=\"Permalink: Germline SNP/Indel Callers\" href=\"#germline-snpindel-callers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1093/bib/bbaa148\" rel=\"nofollow\"\u003eBenchmarking variant callers in next-generation and third-generation sequencing analysis\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Surui Pei, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Briefings in Bioinformatics, July 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared evaluated 11 modes among 6 variant callers on 12 NGS and TGS datasets on germline and somatic variant calling.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eSentieon\u003c/code\u003e (\u003ccode\u003eTNscope\u003c/code\u003e, \u003ccode\u003eTNseq\u003c/code\u003e, \u003ccode\u003eDNAseq\u003c/code\u003e), \u003ccode\u003eDeepVariant\u003c/code\u003e (\u003ccode\u003eWGS\u003c/code\u003e), \u003ccode\u003eGATK\u003c/code\u003e (\u003ccode\u003eHC\u003c/code\u003e \u0026amp; \u003ccode\u003eMuTect2\u003c/code\u003e), \u003ccode\u003eNeuSomatic\u003c/code\u003e, \u003ccode\u003eVarScan2\u003c/code\u003e, \u003ccode\u003eStrelka2\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e All the four germline callers had comparable performance on NGS data. For TGS data, all the three callers had similar performance in SNP calling, while \u003ca href=\"https://github.com/google/deepvariant\"\u003eDeepVariant\u003c/a\u003e outperformed the others in InDel calling. For somatic variant calling on NGS, Sentieon TNscope and \u003ca href=\"https://github.com/broadinstitute/gatk\"\u003eGATK Mutect2\u003c/a\u003e outperformed the other callers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e--\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6597787/\" rel=\"nofollow\"\u003eSystematic comparison of germline variant calling pipelines cross multiple next-generation sequencers\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Jiayun Chen, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Scientific Reports, June 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared three variant callers for WGS and WES samples from NA12878 across five next-gen sequencing platforms\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eGATK\u003c/code\u003e, \u003ccode\u003eStrelka2\u003c/code\u003e, \u003ccode\u003eSamtools-Varscan2\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Though all methods tested generally scored well, \u003ca href=\"https://github.com/Illumina/strelka\"\u003eStrelka2\u003c/a\u003e had the highest F-scores for both SNP and indel calling in addition to being the most computationally performant.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6294778/\" rel=\"nofollow\"\u003eComparison of three variant callers for human whole genome sequencing\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Anna Supernat, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Scientific Reports, December 2018\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e The paper compared three variant callers for WGS samples from NA12878 at 10x, 15x, and 30x coverage.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eDeepVariant\u003c/code\u003e, \u003ccode\u003eGATK\u003c/code\u003e, \u003ccode\u003eSpeedSeq\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e All methods had similar F-scores, precision, and recall for SNP calling, but \u003ca href=\"https://github.com/google/deepvariant\"\u003eDeepVariant\u003c/a\u003e scored higher across all metrics for indels at all coverages.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4619817/\" rel=\"nofollow\"\u003eA Comparison of Variant Calling Pipelines Using Genome in a Bottle as a Reference\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Adam Cornish, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e BioMed Research International, October 2015\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared 30 variant calling pipelines composed of six different variant callers and five different aligners on NA12878 WES data from the \"Genome in a Bottle\" consortium.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eVariant callers: \u003ccode\u003eFreeBayes\u003c/code\u003e, \u003ccode\u003eGATK-HaplotypeCaller\u003c/code\u003e, \u003ccode\u003eGATK-UnifiedGenotyper\u003c/code\u003e, \u003ccode\u003eSAMtools mpileup\u003c/code\u003e, \u003ccode\u003eSNPSVM\u003c/code\u003e\u003c/li\u003e\n\u003cli\u003eAligners: \u003ccode\u003ebowtie2\u003c/code\u003e, \u003ccode\u003eBWA-mem\u003c/code\u003e, \u003ccode\u003eBWA-sampe\u003c/code\u003e, \u003ccode\u003eCUSHAW3\u003c/code\u003e, \u003ccode\u003eMOSAIK\u003c/code\u003e, \u003ccode\u003eNovoalign\u003c/code\u003e.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e \u003ca href=\"http://www.novocraft.com/products/novoalign/\" rel=\"nofollow\"\u003eNovoalign\u003c/a\u003e with \u003ca href=\"https://github.com/broadinstitute/gatk\"\u003eGATK-UnifiedGenotyper\u003c/a\u003e exhibited the highest sensitivity while producing few false positives.\nIn general, \u003ca href=\"https://github.com/lh3/bwa\"\u003eBWA-mem\u003c/a\u003e was the most consistent aligner, and \u003ccode\u003eGATK-UnifiedGenotyper\u003c/code\u003e performed well across the top aligners (BWA, bowtie2, and Novoalign).\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSomatic SNV/Indel callers\u003c/h3\u003e\u003ca id=\"user-content-somatic-snvindel-callers\" class=\"anchor\" aria-label=\"Permalink: Somatic SNV/Indel callers\" href=\"#somatic-snvindel-callers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0151664\" rel=\"nofollow\"\u003eEvaluation of Nine Somatic Variant Callers for Detection of Somatic Mutations in Exome and Targeted Deep Sequencing Data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Anne Bruun KrÃ¸igÃ¥rd, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e PLoS ONE, March 2016\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper performed comparisons between nine somatic variant callers on five paired tumor-normal samples from breast cancer patients subjected to WES and targeted deep sequencing.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eEBCall\u003c/code\u003e, \u003ccode\u003eMutect\u003c/code\u003e, \u003ccode\u003eSeurat\u003c/code\u003e, \u003ccode\u003eShimmer\u003c/code\u003e, \u003ccode\u003eIndelocator\u003c/code\u003e, \u003ccode\u003eSomaticSniper\u003c/code\u003e, \u003ccode\u003eStrelka\u003c/code\u003e, \u003ccode\u003eVarScan2\u003c/code\u003e, \u003ccode\u003eVirmid\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e \u003ca href=\"https://github.com/friend1ws/EBCall\"\u003eEBCall\u003c/a\u003e, \u003ca href=\"https://github.com/broadinstitute/mutect\"\u003eMutect\u003c/a\u003e, \u003ca href=\"https://sourceforge.net/p/virmid/wiki/Home/\" rel=\"nofollow\"\u003eVirmid\u003c/a\u003e, and \u003ca href=\"https://github.com/Illumina/strelka\"\u003eStrelka\u003c/a\u003e (now Strelka2) were most reliable for both WES and targeted deep sequencing.\n\u003ca href=\"https://github.com/friend1ws/EBCall\"\u003eEBCall\u003c/a\u003e was superior for indel calling due to high sensitivity and robustness to changes in sequencing depths.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3986649/\" rel=\"nofollow\"\u003eComparison of somatic mutation calling methods in amplicon and whole exome sequence data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Huilei Xu, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e BMC Genomics, March 2014\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e Using the \"Genome in a Bottle\" gold standard variant set, this paper compared five somatic mutation calling methods on matched tumor-normal amplicon and WES data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e  \u003ccode\u003eGATK-UnifiedGenotyper followed by subtraction\u003c/code\u003e, \u003ccode\u003eMuTect\u003c/code\u003e, \u003ccode\u003eStrelka\u003c/code\u003e, \u003ccode\u003eSomaticSniper\u003c/code\u003e, \u003ccode\u003eVarScan2\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e \u003ca href=\"https://github.com/broadinstitute/mutect\"\u003eMuTect\u003c/a\u003e and \u003ca href=\"https://github.com/Illumina/strelka\"\u003eStrelka\u003c/a\u003e (now Strelka2) had the highest sensitivity, particularly at low frequency alleles, in addition to the highest specificity.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCNV Callers\u003c/h3\u003e\u003ca id=\"user-content-cnv-callers\" class=\"anchor\" aria-label=\"Permalink: CNV Callers\" href=\"#cnv-callers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.biorxiv.org/content/10.1101/850958v1\" rel=\"nofollow\"\u003eBenchmark of tools for CNV detection from NGS panel data in a genetic diagnostics context\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e JosÃ© Marcos Moreno-Cabrera, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e bioRxiv, November 2019.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared five germline copy number variation callers against four genetic diagnostics datasets (495 samples, 231 CNVs validated by MLPA) using both default and optimized parameters.\nSensitivity, specificity, positive predictive value, negative predictive value, F1 score, and various correlation coefficients were used as benchmarking metrics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eDECoN\u003c/code\u003e, \u003ccode\u003eCoNVaDING\u003c/code\u003e, \u003ccode\u003epanelcn.MOPS\u003c/code\u003e, \u003ccode\u003eExomeDepth\u003c/code\u003e, \u003ccode\u003eCODEX2\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Most tools performed well, but varied based on datasets.\nThe authors felt \u003ca href=\"https://github.com/RahmanTeam/DECoN\"\u003eDECoN\u003c/a\u003e and \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/panelcn.mops.html\" rel=\"nofollow\"\u003epanelcn.MOPS\u003c/a\u003e with optimized parameters were sensitive enough to be used as screening methods in genetic dianostics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors have made their benchmarking code (\u003ca href=\"https://github.com/TranslationalBioinformaticsIGTP/CNVbenchmarkeR\"\u003eCNVbenchmarkeR\u003c/a\u003e) available, which can be run to determine optimal parameters for each algorithm for a given user's data.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5452530/\" rel=\"nofollow\"\u003eAn evaluation of copy number variation detection tools for cancer using whole exome sequencing data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Fatima Zare, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e BMC Bioinformatics, May 2017\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared six copy number variation callers on ten TCGA breast cancer tumor-normal pair WES datasets in addition to simulated datasets from VarSimLab.\nSensitivity, specificity, and false-discovery rate were used as the benchmarking metrics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eADTEx\u003c/code\u003e, \u003ccode\u003eCONTRA\u003c/code\u003e, \u003ccode\u003ecn.MOPS\u003c/code\u003e, \u003ccode\u003eExomeCNV\u003c/code\u003e, \u003ccode\u003eVarScan2\u003c/code\u003e, \u003ccode\u003eCoNVEX\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e All tools suffered from high FDRs (~30-60%), but \u003ca href=\"https://github.com/cran/ExomeCNV\"\u003eExomeCNV\u003c/a\u003e (a now defunct R package) had the highest overall sensitivity.\n\u003ca href=\"http://dkoboldt.github.io/varscan/\" rel=\"nofollow\"\u003eVarScan2\u003c/a\u003e had moderate sensitivity and specificity for both amplifications and deletions.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSV callers\u003c/h3\u003e\u003ca id=\"user-content-sv-callers\" class=\"anchor\" aria-label=\"Permalink: SV callers\" href=\"#sv-callers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/s41467-019-11146-4\" rel=\"nofollow\"\u003eComprehensive evaluation and characterisation of short read general-purpose structural variant calling software\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Daniel L. Cameron, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Communications, July 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared 10 structural variant callers on four cell line WGS datasets (NA12878, HG002, CHM1, and CHM13) with orthogonal validation data.\nPrecision and recall were the benchmarking metrics used.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eBreakDancer\u003c/code\u003e, \u003ccode\u003ecortex\u003c/code\u003e, \u003ccode\u003eCREST\u003c/code\u003e, \u003ccode\u003eDELLY\u003c/code\u003e, \u003ccode\u003eGRIDSS\u003c/code\u003e, \u003ccode\u003eHydra\u003c/code\u003e, \u003ccode\u003eLUMPY\u003c/code\u003e, \u003ccode\u003emanta\u003c/code\u003e, \u003ccode\u003ePindel\u003c/code\u003e, \u003ccode\u003eSOCRATES\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The authors found \u003ca href=\"https://github.com/PapenfussLab/gridss\"\u003eGRIDSS\u003c/a\u003e and \u003ca href=\"https://github.com/Illumina/manta\"\u003emanta\u003c/a\u003e consistently performed well, but also provide more general guidelines for both users and developers.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eUse a caller that utilizes multiple sources of evidence and assembly.\u003c/li\u003e\n\u003cli\u003eUse a caller that can call all events you care about.\u003c/li\u003e\n\u003cli\u003eEnsemble calling is not a cure-all and generally don't outperform the best individual callers (at least on these datasets).\u003c/li\u003e\n\u003cli\u003eDo not use callers that rely only on paired-end data.\u003c/li\u003e\n\u003cli\u003eCalls with high read counts are typically artefacts.\u003c/li\u003e\n\u003cli\u003eSimulations aren't real - benchmarking solely on simulations is a bad idea.\u003c/li\u003e\n\u003cli\u003eDevelopers - be wary of incomplete trust sets and the potential for overfitting. Test tools on multiple datasets.\u003c/li\u003e\n\u003cli\u003eDevelopers - make your tool easy to use with basic sanity checks to protect against invalid inputs. Use standard file formats.\u003c/li\u003e\n\u003cli\u003eDevelopers - use all available evidence and produce meaningful quality scores.\u003c/li\u003e\n\u003c/ul\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1720-5\" rel=\"nofollow\"\u003eComprehensive evaluation of structural variation detection algorithms for whole genome sequencing\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Shunichi Kosugi, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, June 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study compared 69 structural variation callers on simulated and real (NA12878, HG002, and HG00514) datasets.\nF-scores, precision, and recall were the main benchmarking metrics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003e1-2-3-SV\u003c/code\u003e, \u003ccode\u003eAS-GENESENG\u003c/code\u003e, \u003ccode\u003eBASIL-ANISE,\u003c/code\u003e \u003ccode\u003eBatVI\u003c/code\u003e, \u003ccode\u003eBICseq2\u003c/code\u003e, \u003ccode\u003eBreakDancer\u003c/code\u003e, \u003ccode\u003eBreakSeek\u003c/code\u003e, \u003ccode\u003eBreakSeq2\u003c/code\u003e, \u003ccode\u003eBreakway\u003c/code\u003e, \u003ccode\u003eCLEVER\u003c/code\u003e, \u003ccode\u003eCNVnator\u003c/code\u003e,\n\u003ccode\u003eControl-FREEC\u003c/code\u003e, \u003ccode\u003eCREST\u003c/code\u003e, \u003ccode\u003eDELLY\u003c/code\u003e, \u003ccode\u003eDINUMT\u003c/code\u003e, \u003ccode\u003eERDS\u003c/code\u003e, \u003ccode\u003eFermiKit\u003c/code\u003e, \u003ccode\u003eforestSV\u003c/code\u003e, \u003ccode\u003eGASVPro\u003c/code\u003e, \u003ccode\u003eGenomeSTRiP\u003c/code\u003e, \u003ccode\u003eGRIDSS\u003c/code\u003e, \u003ccode\u003eHGT-ID\u003c/code\u003e, \u003ccode\u003eHydra-sv\u003c/code\u003e, \u003ccode\u003eiCopyDAV\u003c/code\u003e, \u003ccode\u003einGAP-sv\u003c/code\u003e, \u003ccode\u003eITIS\u003c/code\u003e,\n\u003ccode\u003elaSV\u003c/code\u003e, \u003ccode\u003eLumpy\u003c/code\u003e, \u003ccode\u003eManta\u003c/code\u003e, \u003ccode\u003eMATCHCLIP\u003c/code\u003e, \u003ccode\u003eMeerkat\u003c/code\u003e, \u003ccode\u003eMELT\u003c/code\u003e, \u003ccode\u003eMELT-numt\u003c/code\u003e, \u003ccode\u003eMetaSV\u003c/code\u003e, \u003ccode\u003eMindTheGap\u003c/code\u003e, \u003ccode\u003eMobster\u003c/code\u003e, \u003ccode\u003eMobster-numt\u003c/code\u003e, \u003ccode\u003eMobster-vei\u003c/code\u003e, \u003ccode\u003eOncoSNP-SEQ\u003c/code\u003e, \u003ccode\u003ePamir\u003c/code\u003e, \u003ccode\u003ePBHoney\u003c/code\u003e,\n\u003ccode\u003ePBHoney-NGM\u003c/code\u003e, \u003ccode\u003epbsv\u003c/code\u003e, \u003ccode\u003ePennCNV-Seq\u003c/code\u003e, \u003ccode\u003ePindel\u003c/code\u003e, \u003ccode\u003ePopIns\u003c/code\u003e, \u003ccode\u003ePRISM\u003c/code\u003e, \u003ccode\u003eRAPTR\u003c/code\u003e, \u003ccode\u003ereadDepth\u003c/code\u003e, \u003ccode\u003eRetroSeq\u003c/code\u003e, \u003ccode\u003eSniffles\u003c/code\u003e, \u003ccode\u003eSocrates\u003c/code\u003e, \u003ccode\u003eSoftSearch\u003c/code\u003e, \u003ccode\u003eSoftSV\u003c/code\u003e, \u003ccode\u003eSoloDel\u003c/code\u003e, \u003ccode\u003eSprites\u003c/code\u003e,\n\u003ccode\u003eSvABA\u003c/code\u003e, \u003ccode\u003eSVDetect\u003c/code\u003e, \u003ccode\u003eSvelter\u003c/code\u003e, \u003ccode\u003eSVfinder\u003c/code\u003e, \u003ccode\u003eSVseq2\u003c/code\u003e, \u003ccode\u003eTangram\u003c/code\u003e, \u003ccode\u003eTangram-numt\u003c/code\u003e, \u003ccode\u003eTangram-vei\u003c/code\u003e, \u003ccode\u003eTea\u003c/code\u003e, \u003ccode\u003eTEMP\u003c/code\u003e, \u003ccode\u003eTIDDIT\u003c/code\u003e, \u003ccode\u003eUlysses\u003c/code\u003e, \u003ccode\u003eVariationHunter\u003c/code\u003e, \u003ccode\u003eVirusFinder\u003c/code\u003e, \u003ccode\u003eVirusSeq\u003c/code\u003e, \u003ccode\u003eWham\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Varies greatly depending on type and size of the structural variant in addition to read length.\n\u003ccode\u003eGRIDSS\u003c/code\u003e, \u003ccode\u003eLumpy\u003c/code\u003e, \u003ccode\u003eSVseq2\u003c/code\u003e, \u003ccode\u003eSoftSV\u003c/code\u003e, and \u003ccode\u003eManta\u003c/code\u003e performed well calling deletions of diverse sizes.\n\u003ccode\u003eTIDDIT\u003c/code\u003e, \u003ccode\u003eforestSV\u003c/code\u003e, \u003ccode\u003eERDS\u003c/code\u003e, and \u003ccode\u003eCNVnator\u003c/code\u003e called large deletions well, while \u003ccode\u003epbsv\u003c/code\u003e, \u003ccode\u003eSniffles\u003c/code\u003e, and \u003ccode\u003ePBHoney\u003c/code\u003e were the best performers for small deletions.\nFor duplications, good choices included \u003ccode\u003eWham\u003c/code\u003e, \u003ccode\u003eSoftSV\u003c/code\u003e, \u003ccode\u003eMATCHCLIP\u003c/code\u003e, and \u003ccode\u003eGRIDSS\u003c/code\u003e, while \u003ccode\u003eCNVnator\u003c/code\u003e, \u003ccode\u003eERDS\u003c/code\u003e, and \u003ccode\u003eiCopyDAV\u003c/code\u003e excelled calling large duplications.\nFor insertions, \u003ccode\u003eMELT\u003c/code\u003e, \u003ccode\u003eMobster\u003c/code\u003e, \u003ccode\u003einGAP-sv\u003c/code\u003e, and methods using long read data were most effective.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1858-1\" rel=\"nofollow\"\u003eEvaluating nanopore sequencing data processing pipelines for structural variation identification\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Anbo Zhou, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, November 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper evaluated four alignment tools and three SV detection tools on four nanopore datasets (both simulated and real).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003cem\u003ealigners\u003c/em\u003e - \u003ccode\u003eminimap2\u003c/code\u003e, \u003ccode\u003eNGMLR\u003c/code\u003e, \u003ccode\u003eGraphMap\u003c/code\u003e, \u003ccode\u003eLAST\u003c/code\u003e. SV Callers - \u003ccode\u003eSniffles\u003c/code\u003e, \u003ccode\u003eNanoSV\u003c/code\u003e, \u003ccode\u003ePicky\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The authors recommend using the \u003ca href=\"https://github.com/lh3/minimap2\"\u003eminimap2\u003c/a\u003e aligner in combination with the SV caller \u003ca href=\"https://github.com/fritzsedlazeck/Sniffles\"\u003eSniffles\u003c/a\u003e because of their speed and relatively balanced performance.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors provide \u003ca href=\"https://github.com/JXing-Lab/nanopore-sv-evaluation\"\u003eall code used in the study\u003c/a\u003e as well as a singularity package containing pre-installed programs and all seven pipeline.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSingle Cell\u003c/h2\u003e\u003ca id=\"user-content-single-cell\" class=\"anchor\" aria-label=\"Permalink: Single Cell\" href=\"#single-cell\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003escRNA Transformations/Normalization\u003c/h3\u003e\u003ca id=\"user-content-scrna-transformationsnormalization\" class=\"anchor\" aria-label=\"Permalink: scRNA Transformations/Normalization\" href=\"#scrna-transformationsnormalization\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1038/s41592-023-01814-1\" rel=\"nofollow\"\u003eComparison of transformations for single-cell RNA-seq data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Constantin Ahlmann-Eltze \u0026amp; Wolfgang Huber\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Methods, May 2023\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study evaluates 22 transformations for preprocessing single-cell RNA-sequencing data. These transformations are aimed at adjusting counts for variable sampling efficiency and making variance similar across the dynamic range. The research focuses on understanding cell types and states in terms of lower-dimensional mathematical structures, with benchmarks designed to assess the ability of transformations to recover latent cell structures.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eDelta method-based variance-stabilizing transformations:\u003c/strong\u003e acosh transformation, shifted logarithm (with pseudo-count y0 = 1 or y0 = 1/4Î±), shifted logarithm with CPM, and additional variations including HVG selection, z scoring, and output rescaling.\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eResiduals-based variance-stabilizing transformations:\u003c/strong\u003e clipped and unclipped Pearson residuals (implemented by sctransform and transformGamPoi), randomized quantile residuals, clipped Pearson residuals with HVG selection, z scoring, and an analytical approximation to the Pearson residuals.\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eLatent gene expression-based transformations (Lat Expr):\u003c/strong\u003e Sanity Distance, Sanity MAP, Dino, Normalisr.\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eCount-based factor analysis models (Count):\u003c/strong\u003e GLM PCA, NewWave.\u003c/li\u003e\n\u003cli\u003e\u003cstrong\u003eNegative controls:\u003c/strong\u003e raw untransformed counts (y) and raw counts scaled by the size factor (y/s).\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eThe shifted logarithm transformation (log(y/s + y0) with y0 = 1) followed by PCA performed well, often better than more sophisticated alternatives.\u003c/li\u003e\n\u003cli\u003eThe study advises against using CPM as input for the shifted logarithm due to unrealistic large overdispersion assumptions.\u003c/li\u003e\n\u003cli\u003ePearson residuals-based transformation has good theoretical properties and performed similarly to the shifted logarithm. However, it has limitations in handling genes with large dynamic range across cells.\u003c/li\u003e\n\u003cli\u003eLatent expression state transformations and count-based latent factor models did not outperform the shifted logarithm and were computationally expensive.\u003c/li\u003e\n\u003cli\u003eThe delta method-based transformation produced more consistent results on the 10x datasets.\u003c/li\u003e\n\u003cli\u003eOverall, despite extensive research in preprocessing methods for single-cell RNA-seq data, the shifted logarithm still ranks among the best, underlining the utility of lower-dimensional embeddings of the transformed count matrix for noise reduction and fidelity increase.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e An \u003ca href=\"https://shiny-portal.embl.de/shinyapps/app/08_single-cell_transformation_benchmark\" rel=\"nofollow\"\u003einteractive website\u003c/a\u003e with all results for all tested parameter combinations is provided.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGene Signature Scoring\u003c/h3\u003e\u003ca id=\"user-content-gene-signature-scoring\" class=\"anchor\" aria-label=\"Permalink: Gene Signature Scoring\" href=\"#gene-signature-scoring\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://elifesciences.org/articles/71994\" rel=\"nofollow\"\u003eSignature-scoring methods developed for bulk samples are not adequate for cancer single-cell RNA sequencing data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Nighat Noureen, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e eLife, February 2022\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study benchmarks five signature-scoring methods in the context of cancer single-cell RNA sequencing data. The authors highlight that methods developed for bulk sample analysis, specifically single sample gene set enrichment analysis (ssGSEA) and Gene Set Variation Analysis (GSVA), show biases and inaccuracies when applied to single-cell data. This is attributed to the higher gene counts in cancer cells compared to normal cells, which affects the performance of ssGSEA and GSVA. The study emphasizes the importance of considering cellular context in signature scoring, especially the effect of dropouts in single-cell data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003essGSEA\u003c/code\u003e, \u003ccode\u003eGSVA\u003c/code\u003e, \u003ccode\u003eAUCell\u003c/code\u003e, \u003ccode\u003eSingle Cell Signature Explorer (SCSE)\u003c/code\u003e, \u003ccode\u003eJointly Assessing Signature Mean and Inferring Enrichment (JASMINE)\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The study recommends caution when using bulk-sample-based methods like ssGSEA and GSVA for single-cell RNA sequencing data due to their susceptibility to biases caused by high gene counts and dropouts in cancer cells. Single-cell-based methods, particularly JASMINE and SCSE, are more robust in this context. JASMINE, a new method developed in this study, showed particular effectiveness in accounting for dropouts and evaluating average expression levels of expressed signature genes. Typically, I'd avoid adding a paper in which the author's are touting their new tool, but I feel the contextual information therein is invaluable in this case.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The GitHub repository for JASMINE is available \u003ca href=\"https://github.com/NNoureen/JASMINE\"\u003ehere\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003escRNA Sequencing Protocols\u003c/h3\u003e\u003ca id=\"user-content-scrna-sequencing-protocols\" class=\"anchor\" aria-label=\"Permalink: scRNA Sequencing Protocols\" href=\"#scrna-sequencing-protocols\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/s41587-020-0469-4\" rel=\"nofollow\"\u003eBenchmarking single-cell RNA-sequencing protocols for cell atlas projects\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Elisabetta Mereu*, Atefeh Lafzi*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Biotechnology, April 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper evaluated 13 single cell/nuclei RNA-seq protocols to evaluate their aptitude for use in cell atlas-like projects. Using a single cell, multi-species mixture, the authors measured each protocol's ability to capture cell markers, gene detection power, clusterability (with and without integration with other protocols), mappability, and mixability.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eQuartz-seq2\u003c/code\u003e, \u003ccode\u003eChromium\u003c/code\u003e, \u003ccode\u003eSmart-seq2\u003c/code\u003e, \u003ccode\u003eCEL-seq2\u003c/code\u003e, \u003ccode\u003eC1HT-medium\u003c/code\u003e, \u003ccode\u003eC1HT-small\u003c/code\u003e, \u003ccode\u003eddSEQ\u003c/code\u003e, \u003ccode\u003eChromium (single nuclei)\u003c/code\u003e, \u003ccode\u003eDrop-seq\u003c/code\u003e, \u003ccode\u003einDrop\u003c/code\u003e, \u003ccode\u003eICELL8\u003c/code\u003e, \u003ccode\u003eMARS-seq\u003c/code\u003e, \u003ccode\u003egmcSCRB-seq\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e See figure 6 for a summary of benchmarking results for each method. \u003ccode\u003eQuartz-seq2\u003c/code\u003e was the overall best performing, yielding superior results for gene detection and marker expression over other methods, though \u003ccode\u003eChromium\u003c/code\u003e, \u003ccode\u003eSmart-seq2\u003c/code\u003e, and \u003ccode\u003eCEL-seq2\u003c/code\u003e were also strong performers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors provide benchmarking code and analysis code in two different Github repositories - \u003ca href=\"https://github.com/ati-lz/HCA_Benchmarking\"\u003ehere\u003c/a\u003e and \u003ca href=\"https://github.com/elimereu/matchSCore2\"\u003ehere\u003c/a\u003e.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/s41587-020-0465-8\" rel=\"nofollow\"\u003eSystematic comparison of single-cell and single-nucleus RNA-sequencing methods\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Jiarui Ding, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Biotechnology, April 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study evaluated seven methods for single-cell and/or single-nucleus RNA-sequencing on three types of samples: cell lines, PBMCs, and brain tissue. Evaluation metrics included the structure and alignment of reads, number of multiplets and detection sensitivity, and ability to recover known biological information.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eSmart-seq2\u003c/code\u003e, \u003ccode\u003eCEL-Seq2\u003c/code\u003e, \u003ccode\u003e3' 10X Chromium\u003c/code\u003e, \u003ccode\u003eDrop-Seq\u003c/code\u003e, \u003ccode\u003eSeq-Well\u003c/code\u003e, \u003ccode\u003einDrops\u003c/code\u003e, \u003ccode\u003esci-RNA-seq\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Overall, the authors found \u003ccode\u003e3' 10X Chromium\u003c/code\u003e to have the strongest consistent performance among the high-throughput methods, yielding the highest sensitivity, though it did not perform any better for cell type classification. When greater sensitivity is required, the authors recommend \u003ccode\u003eSmart-seq2\u003c/code\u003e or \u003ccode\u003eCEL-Seq2\u003c/code\u003e, which both performed similarly. Supplementary table 7 includes an overview of each method's relative merits.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors made their unified analysis pipeline (scumi) available as a \u003ca href=\"https://bitbucket.org/jerry00/scumi-dev/src/master/\" rel=\"nofollow\"\u003epython package\u003c/a\u003e, the repo of which also includes their R scripts used for cell filtering and cell type assignment.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003escRNA Analysis Pipelines\u003c/h3\u003e\u003ca id=\"user-content-scrna-analysis-pipelines\" class=\"anchor\" aria-label=\"Permalink: scRNA Analysis Pipelines\" href=\"#scrna-analysis-pipelines\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1093/nargab/lqaa052\" rel=\"nofollow\"\u003eComparison of visualization tools for single-cell RNAseq data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Batuhan Cakir, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e NAR Genomics and Bioinformatics, September 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study evaluated 13 scRNA-seq visualization platforms based on their features, performance, cloud and web support, containerization, and inter-operability between analysis platforms and data formats (loom, h5ad, SingleCellExperiment, Seurat, raw txt/csv.) with varying numbers of cells (5k to 2 million).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eASAP\u003c/code\u003e, \u003ccode\u003eBbrowser\u003c/code\u003e, \u003ccode\u003ecellxgene\u003c/code\u003e, \u003ccode\u003eGranatum\u003c/code\u003e, \u003ccode\u003eiSEE\u003c/code\u003e, \u003ccode\u003eLoom viewer\u003c/code\u003e, \u003ccode\u003eLoupe Cell Browser\u003c/code\u003e, \u003ccode\u003eSCope\u003c/code\u003e, \u003ccode\u003escSVA\u003c/code\u003e, \u003ccode\u003escVI\u003c/code\u003e, \u003ccode\u003eSingle Cell Explorer\u003c/code\u003e, \u003ccode\u003eSPRING\u003c/code\u003e, and \u003ccode\u003eUCSC Cell Browser\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e \u003cstrong\u003eTable 1\u003c/strong\u003e provides an overview of the tool capabilities and current support. The authors further compared tools that could be used for web sharing - \u003ccode\u003ecellxgene\u003c/code\u003e, \u003ccode\u003eiSEE\u003c/code\u003e with SCE files, \u003ccode\u003eiSEE\u003c/code\u003e with loom files, \u003ccode\u003eloom-viewer\u003c/code\u003e, \u003ccode\u003eSCope\u003c/code\u003e, \u003ccode\u003eSingle Cell Explorer\u003c/code\u003e, \u003ccode\u003escSVA\u003c/code\u003e, and \u003ccode\u003eUCSC Cell Browser\u003c/code\u003e for preprocessing memory and time requirements with varying numbers of cells. \u003ccode\u003eiSEE-loom\u003c/code\u003e, \u003ccode\u003eSCope\u003c/code\u003e, \u003ccode\u003escSVA\u003c/code\u003e and \u003ccode\u003eloom-viewer\u003c/code\u003e all enable efficient integration with the hierarchical data format (HDF5) from which loom and h5ad formats are derived, and as such, have the lowest preprocessing time requirements. \u003ccode\u003eiSEE-SCE\u003c/code\u003e performed poorly with large numbers of cells (\u0026gt;50k) with the default number of panels (8) but performed better with a more limited set of visualizations. Some of the tools had unexplainable resource spikes with high numbers of cells, e.g. loom-viewer dramatically increasing in both time and memory requirements with 2M cells versus 1.5M. \u003ccode\u003ecellxgene\u003c/code\u003e was generally recommended due to its ease of use, expansive community support and active maintenance, and ability to handle large datasets. \u003ccode\u003eSingle Cell Explorer\u003c/code\u003e and \u003ccode\u003eUCSC Cell Browser\u003c/code\u003e both were among the poorest performers resource-wise, with their usage increasing linearly with cell number. \u003ccode\u003eiSEE\u003c/code\u003e was generally commended for its flexibility and support of custom visualizations.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors made their package (sceasy) to convert between Seurat, SCE, Loom, and AnnData objects available via \u003ca href=\"https://github.com/cellgeni/sceasy\"\u003eGitHub\u003c/a\u003e.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1093/bib/bbaa116\" rel=\"nofollow\"\u003eComparison of high-throughput single-cell RNA sequencing data processing pipelines\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Mingxuan Gao, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Briefings in Bioinformatics, July 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study evaluated 7 scRNA-seq pipelines on 8 data sets.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eDrop-seq-tools version-2.3.0\u003c/code\u003e, \u003ccode\u003eCell Ranger version-3.0.2\u003c/code\u003e, \u003ccode\u003escPipe version-1.4.1\u003c/code\u003e, \u003ccode\u003ezUMIs version-2.4.5b\u003c/code\u003e, \u003ccode\u003eUMI-tools version-1.0.0\u003c/code\u003e, \u003ccode\u003eumis version-1.0.3\u003c/code\u003e, \u003ccode\u003edropEst version-0.8.6\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Cell Ranger shows the highest algorithm complexity and parallelization, whereas scPipe, umis and zUMIs have lower complexity that is suitable for large-scale scRNA-seq integration analysis. UMI-tools show the highest transcript quantification accuracy on ERCC datasets from three scRNA-seq platforms. Integration of expression matrices from different pipelines will introduce confounding factors akin to batch effect. zUMIs and dropEst have higher sensitivity to detectmore genes for single cells, which may also bring unwanted factors. For most downstream analysis, Drop-seq-tools, Cell Ranger and UMI-tools show high consistency, whereas umis and zUMIs show inconsistent results compared with the other pipelines.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/s41467-019-12266-7\" rel=\"nofollow\"\u003eA systematic evaluation of single cell RNA-seq analysis pipelines\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Beate Veith, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Communications, October 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study evaluated ~3000 pipeline combinations based on three mapping, three annotation, four imputation, seven normalization, and four differential expression testing approaches with five scRNA-seq library protocols on simulated data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e scRNA-seq library prep protocols - \u003ccode\u003eSCRB-seq\u003c/code\u003e, \u003ccode\u003eSmart-seq2\u003c/code\u003e, \u003ccode\u003eCEL-seq2\u003c/code\u003e, \u003ccode\u003eDrop-seq\u003c/code\u003e, \u003ccode\u003e10X Genomics\u003c/code\u003e. Mapping - \u003ccode\u003ebwa\u003c/code\u003e, \u003ccode\u003eSTAR\u003c/code\u003e, \u003ccode\u003ekallisto\u003c/code\u003e. Annotation - \u003ccode\u003egencode\u003c/code\u003e, \u003ccode\u003erefseq\u003c/code\u003e, \u003ccode\u003evega\u003c/code\u003e. Imputation - \u003ccode\u003efiltering\u003c/code\u003e, \u003ccode\u003eDrImpute\u003c/code\u003e, \u003ccode\u003escone\u003c/code\u003e, \u003ccode\u003eSAVER\u003c/code\u003e. Normalization - \u003ccode\u003escran\u003c/code\u003e, \u003ccode\u003eSCnorm\u003c/code\u003e, \u003ccode\u003eLinnorm\u003c/code\u003e, \u003ccode\u003eCensus\u003c/code\u003e, \u003ccode\u003eMR\u003c/code\u003e, \u003ccode\u003eTMM\u003c/code\u003e. Differential testing - \u003ccode\u003eedgeR-zingeR\u003c/code\u003e, \u003ccode\u003elimma\u003c/code\u003e, \u003ccode\u003eMAST\u003c/code\u003e, \u003ccode\u003eT-test\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e \u003cstrong\u003eFigure 5F\u003c/strong\u003e contains a flowchart with the authors' recommendations. For alignment, \u003ccode\u003eSTAR\u003c/code\u003e with Gencode annotations generally had the highest mapping and assignment rates. All mappers performed best with Gencode annotations. For normalization, \u003ccode\u003escran\u003c/code\u003e was found to best handle potential assymetric differential expression and large numbers of differentially expressed genes. They also note that normalization is overall the most influential step, particularly if asymmetric DE is present (\u003cstrong\u003eFigure 5\u003c/strong\u003e). For Smart-seq2 data without spike-ins, the authors suggest \u003ccode\u003eCensus\u003c/code\u003e may be the best choice. The authors found little benefit to imputation in most scenarios, particularly if one of the better normalization methods (e.g. \u003ccode\u003escran\u003c/code\u003e) was used. The authors found library prep and normalization strategies to have a stronger effect on pipeline performance than the choice of differential expression tool, but generally found \u003ccode\u003elimma-trend\u003c/code\u003e to have the most robust performance.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors made their simulation tool (powsimR) available on \u003ca href=\"https://github.com/bvieth/powsimR\"\u003eGithub\u003c/a\u003e along with their \u003ca href=\"https://github.com/bvieth/scRNA-seq-pipelines\"\u003epipeline scripts\u003c/a\u003e to reproduce their analyses.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003escRNA Imputation Methods\u003c/h3\u003e\u003ca id=\"user-content-scrna-imputation-methods\" class=\"anchor\" aria-label=\"Permalink: scRNA Imputation Methods\" href=\"#scrna-imputation-methods\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.biorxiv.org/content/10.1101/2020.01.29.925974v1.full\" rel=\"nofollow\"\u003eA Systematic Evaluation of Single-cell RNA-sequencing Imputation Methods\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Wenpin Hou, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e bioRxiv, January 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper evaluated 18 scRNA-seq imputation methods using seven datasets containing cell line and tissue data from several experimental protocols. The authors assessed the similarity of imputed cell profiles to bulk samples and investigated whether imputation improves signal recovery or introduces noise in three downstream applications - differential expression, unsupervised clustering, and trajectory inference.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003escVI\u003c/code\u003e, \u003ccode\u003eDCA\u003c/code\u003e, \u003ccode\u003eMAGIC\u003c/code\u003e, \u003ccode\u003escImpute\u003c/code\u003e, \u003ccode\u003ekNN-smoothing\u003c/code\u003e, \u003ccode\u003emcImpute\u003c/code\u003e, \u003ccode\u003eSAUCIE\u003c/code\u003e, \u003ccode\u003eDrImpute\u003c/code\u003e, \u003ccode\u003ePBLR\u003c/code\u003e, \u003ccode\u003eSAVER\u003c/code\u003e, \u003ccode\u003eVIPER\u003c/code\u003e, \u003ccode\u003eSAVERX\u003c/code\u003e, \u003ccode\u003eDeepImpute\u003c/code\u003e, \u003ccode\u003escRecover\u003c/code\u003e, \u003ccode\u003eALRA\u003c/code\u003e, \u003ccode\u003ebayNorm\u003c/code\u003e, \u003ccode\u003eAutoImpute\u003c/code\u003e, \u003ccode\u003escScope\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Figure 6 provides a performance summary of the tested methods. In general, the authors recommend caution using any of these methods, as they can introduce significant variability and noise into downstream analyses. Of the methods tested, \u003ccode\u003eMAGIC\u003c/code\u003e, \u003ccode\u003ekNN-smoothing\u003c/code\u003e, and \u003ccode\u003eSAVER\u003c/code\u003e outperformed the other methods most consistently, though this varied widely across evaluation criteria, protocols, datasets, and downstream analysis. Many methods show no clear improvement over no imputation, and in some cases, perform significantly worse in downstream analyses.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links (optional):\u003c/strong\u003e The authors placed all of their benchmaking code on \u003ca href=\"https://github.com/Winnie09/imputationBenchmark\"\u003eGithub\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003escRNA Differential Gene Expression\u003c/h3\u003e\u003ca id=\"user-content-scrna-differential-gene-expression\" class=\"anchor\" aria-label=\"Permalink: scRNA Differential Gene Expression\" href=\"#scrna-differential-gene-expression\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/nmeth.4612\" rel=\"nofollow\"\u003eBias, robustness and scalability in single-cell differential expression analysis\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Charlotte Soneson* \u0026amp; Mark D Robinson*\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Methods, February 2018\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper evaluated 36 approaches for determining differential gene expression from both synthetic and 36 real scRNA-seq datasets. The authors assess type I error control, FDR control and power, computational efficiency, and consistency.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eedgeRQLFDetRate\u003c/code\u003e, \u003ccode\u003eMASTcpmDetRate\u003c/code\u003e, \u003ccode\u003elimmatrend\u003c/code\u003e, \u003ccode\u003eMASTtpmDetRate\u003c/code\u003e, \u003ccode\u003eedgeRQLF\u003c/code\u003e, \u003ccode\u003ettest\u003c/code\u003e, \u003ccode\u003evoomlimma\u003c/code\u003e, \u003ccode\u003eWilcoxon\u003c/code\u003e, \u003ccode\u003eMASTcpm\u003c/code\u003e, \u003ccode\u003eMASTtpm\u003c/code\u003e, \u003ccode\u003eSAMseq\u003c/code\u003e, \u003ccode\u003eD3E\u003c/code\u003e, \u003ccode\u003eedgeRLRT\u003c/code\u003e, \u003ccode\u003emetagenomeSeq\u003c/code\u003e, \u003ccode\u003eedgeRLRTcensus\u003c/code\u003e, \u003ccode\u003eedgeRLRTdeconv\u003c/code\u003e, \u003ccode\u003emonoclecensus\u003c/code\u003e, \u003ccode\u003eROTStpm\u003c/code\u003e, \u003ccode\u003eROTSvoom\u003c/code\u003e, \u003ccode\u003eDESeq2betapFALSE\u003c/code\u003e, \u003ccode\u003eedgeRLRTrobust\u003c/code\u003e, \u003ccode\u003emonoclecount\u003c/code\u003e, \u003ccode\u003eDESeq2\u003c/code\u003e, \u003ccode\u003eDESeq2nofilt\u003c/code\u003e, \u003ccode\u003eROTScpm\u003c/code\u003e, \u003ccode\u003eSeuratTobit\u003c/code\u003e, \u003ccode\u003eNODES\u003c/code\u003e, \u003ccode\u003eDESeq2census\u003c/code\u003e, \u003ccode\u003escDD\u003c/code\u003e, \u003ccode\u003eBPSC\u003c/code\u003e, \u003ccode\u003eSCDE\u003c/code\u003e, \u003ccode\u003eDEsingle\u003c/code\u003e, \u003ccode\u003emonocle\u003c/code\u003e, \u003ccode\u003eSeuratBimodnofilt\u003c/code\u003e, \u003ccode\u003eSeuratBimodlsExpr2\u003c/code\u003e, \u003ccode\u003eSeuratBimod\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e In general, the authors found that gene prefiltering was essential for good, robust performance from many methods. They note high variability between methods and summarize general performance across all metrics in Figure 5. They do not make recommendations as to a specific method/tool. Of note is that Seurat switched to using the wilcoxon test by default after this study was released, as it performed much better than their previously available methods.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors make their benchmarking pipeline, \u003ca href=\"https://github.com/markrobinsonuzh/conquer\"\u003econquer\u003c/a\u003e, available on Github.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTrajectory Inference\u003c/h3\u003e\u003ca id=\"user-content-trajectory-inference\" class=\"anchor\" aria-label=\"Permalink: Trajectory Inference\" href=\"#trajectory-inference\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1038/s41587-019-0071-9\" rel=\"nofollow\"\u003eA comparison of single-cell trajectory inference methods\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Wouter Saelens*, Robrecht Cannoodt*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nat Biotech, April 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e A comprehensive evaluation of 45 trajectory inference methods, this paper provides an unmatched comparison of the rapidly evolving field of single-cell trajectory inference. Each method was scored on accuracy, scalability, stability, and usability. Should be considered a gold-standard for other benchmarking studies.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003ePAGA\u003c/code\u003e, \u003ccode\u003eRaceID/StemID\u003c/code\u003e, \u003ccode\u003eSLICER\u003c/code\u003e, \u003ccode\u003eSlingshot\u003c/code\u003e, \u003ccode\u003ePAGA Tree\u003c/code\u003e, \u003ccode\u003eMST\u003c/code\u003e, \u003ccode\u003epCreode\u003c/code\u003e, \u003ccode\u003eSCUBA\u003c/code\u003e, \u003ccode\u003eMonocle DDRTree\u003c/code\u003e, \u003ccode\u003eMonocle ICA\u003c/code\u003e, \u003ccode\u003ecellTree maptpx\u003c/code\u003e, \u003ccode\u003eSLICE\u003c/code\u003e, \u003ccode\u003ecellTree VEM\u003c/code\u003e, \u003ccode\u003eEIPiGraph\u003c/code\u003e, \u003ccode\u003eSincell\u003c/code\u003e, \u003ccode\u003eURD\u003c/code\u003e, \u003ccode\u003eCellTrails\u003c/code\u003e, \u003ccode\u003eMpath\u003c/code\u003e, \u003ccode\u003eCellRouter\u003c/code\u003e, \u003ccode\u003eSTEMNET\u003c/code\u003e, \u003ccode\u003eFateID\u003c/code\u003e, \u003ccode\u003eMFA\u003c/code\u003e, \u003ccode\u003eGPfates\u003c/code\u003e, \u003ccode\u003eDPT\u003c/code\u003e, \u003ccode\u003eWishbone\u003c/code\u003e, \u003ccode\u003eSCORPIUS\u003c/code\u003e, \u003ccode\u003eComponent 1\u003c/code\u003e, \u003ccode\u003eEmbeddr\u003c/code\u003e, \u003ccode\u003eMATCHER\u003c/code\u003e, \u003ccode\u003eTSCAN\u003c/code\u003e, \u003ccode\u003eWanderlust\u003c/code\u003e, \u003ccode\u003ePhenoPath\u003c/code\u003e, \u003ccode\u003etopslam\u003c/code\u003e, \u003ccode\u003eWaterfall\u003c/code\u003e, \u003ccode\u003eEIPiGraph linear\u003c/code\u003e, \u003ccode\u003eouijaflow\u003c/code\u003e, \u003ccode\u003eFORKS\u003c/code\u003e, \u003ccode\u003eAngle\u003c/code\u003e, \u003ccode\u003eEIPiGraph cycle\u003c/code\u003e, \u003ccode\u003ereCAT\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Varies depending on dataset and expected trajectory type, though \u003ca href=\"https://icb-scanpy.readthedocs-hosted.com/en/stable/tutorials.html#trajectory-inference\" rel=\"nofollow\"\u003ePAGA, PAGA Tree\u003c/a\u003e, \u003ca href=\"https://github.com/rcannood/SCORPIUS\"\u003eSCORPIUS\u003c/a\u003e, and \u003ca href=\"https://bioconductor.org/packages/release/bioc/html/slingshot.html\" rel=\"nofollow\"\u003eSlingshot\u003c/a\u003e all scored highly across all metrics.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAuthors wrote an \u003ca href=\"https://dynverse.org/users/3-user-guide/2-guidelines/\" rel=\"nofollow\"\u003einteractive Shiny app\u003c/a\u003e to help users choose the best methods for their data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The \u003ca href=\"https://dynverse.org/\" rel=\"nofollow\"\u003edynverse site\u003c/a\u003e contains numerous packages for users to run and compare results from different trajectory methods on their own data without installing each individually by using Docker. Additionally, they provide several tools for developers to wrap and benchmark their own method against those included in the study.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGene Regulatory Network Inference\u003c/h3\u003e\u003ca id=\"user-content-gene-regulatory-network-inference\" class=\"anchor\" aria-label=\"Permalink: Gene Regulatory Network Inference\" href=\"#gene-regulatory-network-inference\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.nature.com/articles/s41592-019-0690-6\" rel=\"nofollow\"\u003eBenchmarking algorithms for gene regulatory network inference from single-cell transcriptomic data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Aditya Pratapa, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Nature Methods, January 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e The authors compared 12 gene regulatory network (GRN) inference techniques to assess the accuracy, robustness, and efficiency of each method on simulated data from synthetic networks, simulated data from curated models, and real scRNA-seq datasets.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eGENIE3\u003c/code\u003e, \u003ccode\u003ePPCOR\u003c/code\u003e, \u003ccode\u003eLEAP\u003c/code\u003e, \u003ccode\u003eSCODE\u003c/code\u003e, \u003ccode\u003ePIDC\u003c/code\u003e, \u003ccode\u003eSINCERITIES\u003c/code\u003e, \u003ccode\u003eSCNS\u003c/code\u003e, \u003ccode\u003eGRNVBEM\u003c/code\u003e, \u003ccode\u003eSCRIBE\u003c/code\u003e, \u003ccode\u003eGRNBoost2\u003c/code\u003e, \u003ccode\u003eGRISLI\u003c/code\u003e, \u003ccode\u003eSINGE\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e In general, the authors recommend \u003ccode\u003ePIDC\u003c/code\u003e, \u003ccode\u003eGENIE3\u003c/code\u003e, or \u003ccode\u003eGRNBoost2\u003c/code\u003e, as they were the leading and consistent performers for both curated models and experimental datasets in terms of accuracy as well as having multithreaded implementations. Both the \u003ccode\u003eGENIE3\u003c/code\u003e and \u003ccode\u003eGRNBoost2\u003c/code\u003e methods can be used from the \u003ca href=\"https://aertslab.org/#scenic\" rel=\"nofollow\"\u003eSCENIC\u003c/a\u003e workflows in either R or the (much faster) python implementation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors provide their benchmarking framework, \u003ca href=\"https://github.com/murali-group/BEELINE\"\u003eBEELINE\u003c/a\u003e on Github, which also provides an easy-to-use and uniform interface to each method in the form of a Docker image.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2217-z\" rel=\"nofollow\"\u003eEvaluating methods of inferring gene regulatory networks highlights their lack of performance for single cell gene expression data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Shounan Chen* \u0026amp; Jessica C. Mar*\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e BMC Bioinformatics, June 2018\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study compared 8 gene regulatory network inference methods (5 bulk RNA-seq, 3 specific to scRNA-seq) for scRNA-seq data for precision and recall.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003ePartial correlation (Pcorr)\u003c/code\u003e, \u003ccode\u003eBayesian networks\u003c/code\u003e, \u003ccode\u003eGENIE3\u003c/code\u003e, \u003ccode\u003eARACNE\u003c/code\u003e, \u003ccode\u003eCLR\u003c/code\u003e, \u003ccode\u003eSCENIC\u003c/code\u003e, \u003ccode\u003eSCODE\u003c/code\u003e, \u003ccode\u003ePIDC\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Generally, the authors found relatively poor performance across all methods both for simulated and real data. The results from each method had few similarities with other methods and high false positive rates, and the authors recommend caution when interpreting the networks reconstructed with these methods. The authors showed that many of these methods were dramatically affected by dropout events.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIntegration/Batch Correction\u003c/h3\u003e\u003ca id=\"user-content-integrationbatch-correction\" class=\"anchor\" aria-label=\"Permalink: Integration/Batch Correction\" href=\"#integrationbatch-correction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.biorxiv.org/content/10.1101/2023.11.15.564963v1.full\" rel=\"nofollow\"\u003eBenchmarking multi-omics integration algorithms across single-cell RNA and ATAC data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Chuxi Xiao, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e bioRxiv, November 2023\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study benchmarks 12 multi-omics integration methods across three integration tasks, assessing their performance in combining single-cell RNA (scRNA-seq) and ATAC (scATAC-seq) data. The evaluation considers aspects such as the extent of mixing between different omics, cell type conservation, single-cell level alignment accuracy, trajectory preservation, scalability, and ease of use.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003escMVP\u003c/code\u003e, \u003ccode\u003eMOFA+\u003c/code\u003e, \u003ccode\u003eMultiVI\u003c/code\u003e, \u003ccode\u003eCobolt\u003c/code\u003e, \u003ccode\u003escDART\u003c/code\u003e, \u003ccode\u003eUnionCom\u003c/code\u003e, \u003ccode\u003eMMD-MA\u003c/code\u003e, \u003ccode\u003escJoint\u003c/code\u003e, \u003ccode\u003eHarmony\u003c/code\u003e, \u003ccode\u003eSeurat (v4.3)\u003c/code\u003e, \u003ccode\u003eLIGER\u003c/code\u003e, \u003ccode\u003eGLUE\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e The study recommends different methods based on dataset type and size. For unpaired datasets, \u003ccode\u003eGLUE\u003c/code\u003e is preferred. In paired tasks, \u003ccode\u003eGLUE\u003c/code\u003e and \u003ccode\u003eMultiVI\u003c/code\u003e are top choices, with the latter excelling in trajectory conservation. For omics mixing, \u003ccode\u003escDART\u003c/code\u003e, \u003ccode\u003eLIGER\u003c/code\u003e, and \u003ccode\u003eSeurat\u003c/code\u003e are recommended. For cell type conservation, \u003ccode\u003eMOFA+\u003c/code\u003e and \u003ccode\u003escMVP\u003c/code\u003e are viable options. In terms of scalability, \u003ccode\u003eSeurat\u003c/code\u003e, \u003ccode\u003eLIGER\u003c/code\u003e, and \u003ccode\u003eMOFA+\u003c/code\u003e are efficient. For ease of use, \u003ccode\u003escDART\u003c/code\u003e, \u003ccode\u003escJoint\u003c/code\u003e, and \u003ccode\u003eSeurat\u003c/code\u003e are highlighted for their detailed guidance.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1850-9\" rel=\"nofollow\"\u003eA benchmark of batch-effect correction methods for single-cell RNA sequencing data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Hoa Thi Nhu Tran, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, January 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e The authors compared 14 methods in terms of computational runtime, the ability to handle large datasets, and batch-effect correction efficacy while preserving cell type purity.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e\n\u003ccode\u003eSeurat2\u003c/code\u003e, \u003ccode\u003eSeurat3\u003c/code\u003e, \u003ccode\u003eHarmony\u003c/code\u003e, \u003ccode\u003efastMNN\u003c/code\u003e, \u003ccode\u003eMNN Correct\u003c/code\u003e, \u003ccode\u003eComBat\u003c/code\u003e, \u003ccode\u003eLimma\u003c/code\u003e, \u003ccode\u003escGen\u003c/code\u003e, \u003ccode\u003eScanorama\u003c/code\u003e, \u003ccode\u003eMMD-ResNet\u003c/code\u003e, \u003ccode\u003eZINB-WaVe\u003c/code\u003e, \u003ccode\u003escMerge\u003c/code\u003e, \u003ccode\u003eLIGER\u003c/code\u003e, \u003ccode\u003eBBKNN\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Based on the benchmarking results authors suggest \u003ca href=\"https://github.com/immunogenomics/harmony\"\u003eHarmony\u003c/a\u003e, \u003ca href=\"https://github.com/MacoskoLab/liger\"\u003eLIGER\u003c/a\u003e, and \u003ca href=\"https://satijalab.org/seurat/\" rel=\"nofollow\"\u003eSeurat3\u003c/a\u003e as best methods for batch integration.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDimensionality Reduction\u003c/h3\u003e\u003ca id=\"user-content-dimensionality-reduction\" class=\"anchor\" aria-label=\"Permalink: Dimensionality Reduction\" href=\"#dimensionality-reduction\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://www.biorxiv.org/content/10.1101/641142v2\" rel=\"nofollow\"\u003eAccuracy, Robustness and Scalability of Dimensionality Reduction Methods for Single Cell RNAseq Analysis\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Shiquan Sun, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e BioRxiv, October 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e A mammoth comparison of 18 different dimension reduction methods on 30 publicly available scRNAseq data sets in addition to 2 simulated datasets for a variety of purposes ranging from cell clustering to trajectory inference to neighborhood preservation.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e\n\u003ccode\u003efactor analysis (FA)\u003c/code\u003e, \u003ccode\u003eprincipal component analysis (PCA)\u003c/code\u003e, \u003ccode\u003eindependent component analysis (ICA)\u003c/code\u003e, \u003ccode\u003eDiffusion Map\u003c/code\u003e, \u003ccode\u003enonnegative matrix factorization (NMF)\u003c/code\u003e, \u003ccode\u003ePoisson NMF\u003c/code\u003e, \u003ccode\u003ezero-inflated factor analysis (ZIFA)\u003c/code\u003e, \u003ccode\u003ezero-inflated negative binomial based wanted variation extraction (ZINB-WaVE)\u003c/code\u003e, \u003ccode\u003eprobabilistic count matrix factorization (pCMF)\u003c/code\u003e, \u003ccode\u003edeep count autoencoder network (DCA)\u003c/code\u003e, \u003ccode\u003escScope\u003c/code\u003e, \u003ccode\u003egeneralized linear model principal component analysis (GLMPCA)\u003c/code\u003e, \u003ccode\u003emultidimensional scaling (MDS)\u003c/code\u003e, \u003ccode\u003elocally linear embedding (LLE)\u003c/code\u003e, \u003ccode\u003elocal tangent space alignment (LTSA)\u003c/code\u003e, \u003ccode\u003eIsomap\u003c/code\u003e, \u003ccode\u003euniform manifold approximation and projection (UMAP)\u003c/code\u003e, \u003ccode\u003et-distributed stochastic neighbor embedding (tSNE)\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Varies depending on use case.\n\u003ca href=\"https://cran.r-project.org/web/packages/psych/index.html\" rel=\"nofollow\"\u003eFactor Analysis\u003c/a\u003e and \u003ca href=\"https://www.rdocumentation.org/packages/stats/versions/3.6.1/topics/prcomp\" rel=\"nofollow\"\u003eprincipal component analysis\u003c/a\u003e performed well for most use cases.\nSee figure 5 for pratical guidelines.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors have made their benchmarking code available \u003ca href=\"https://github.com/xzhoulab/DRComparison\"\u003eon Github\u003c/a\u003e.\u003c/p\u003e\n\u003chr\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1900-3\" rel=\"nofollow\"\u003eBenchmarking principal component analysis for large-scale single-cell RNA-sequencing\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Koki Tsuyuzaki, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, January 2020\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study compared 21 implementations of 10 algorithms across Python, R, and Julia for principal component analysis for scRNA-seq data, measuring scalability, computational efficiency, outlier robustness, t-SNE/UMAP replication, ease of use, and more using both synthetic and real datasets.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003ePCA (sklearn, full)\u003c/code\u003e, \u003ccode\u003efit (MultiVariateStats.jl)\u003c/code\u003e, \u003ccode\u003eDownsampling\u003c/code\u003e, \u003ccode\u003eIncrementalPCA (sklearn)\u003c/code\u003e, \u003ccode\u003eirlba (irlba)\u003c/code\u003e, \u003ccode\u003esvds (RSpectra)\u003c/code\u003e, \u003ccode\u003epropack.svd (svd)\u003c/code\u003e, \u003ccode\u003ePCA (sklearn, arpack)\u003c/code\u003e, \u003ccode\u003eirlb (Cell Ranger)\u003c/code\u003e, \u003ccode\u003esvds (Arpack.jl)\u003c/code\u003e, \u003ccode\u003eorthiter (OnlinePCA.jl)\u003c/code\u003e, \u003ccode\u003egd (OnlinePCA.jl)\u003c/code\u003e, \u003ccode\u003esgd (OnlinePCA.jl)\u003c/code\u003e, \u003ccode\u003ersvd (rsvd)\u003c/code\u003e, \u003ccode\u003eoocPCA_CSV (oocRPCA)\u003c/code\u003e, \u003ccode\u003ePCA (sklearn, randomized)\u003c/code\u003e, \u003ccode\u003erandomized_svd (sklearn)\u003c/code\u003e, \u003ccode\u003ePCA (dask-ml)\u003c/code\u003e, \u003ccode\u003ehalko (OnlinePCA.jl)\u003c/code\u003e, \u003ccode\u003ealgorithm971 (OnlinePCA.jl)\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Author recommendations vary based on the language being used and matrix size. See figure 8 for recommendations along with recommended parameter settings.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors published their benchmarking scripts \u003ca href=\"https://github.com/rikenbit/onlinePCA-experiments\"\u003eon Github\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCell Annotation/Inference\u003c/h3\u003e\u003ca id=\"user-content-cell-annotationinference\" class=\"anchor\" aria-label=\"Permalink: Cell Annotation/Inference\" href=\"#cell-annotationinference\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1795-z\" rel=\"nofollow\"\u003eA comparison of automatic cell identification methods for single-cell RNA sequencing data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Tamim Abdelaal*, Lieke Michielsen*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, September 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e The authors benchmarked 22 classification methods that automatically assign cell identities including \u003cem\u003esingle-cell-specific\u003c/em\u003e and \u003cem\u003egeneral-purpose\u003c/em\u003e classifiers across 27 publicly available single-cell RNA sequencing datasets of different sizes, technologies, species, and levels of complexity. Two types of experimental setups were used evaluate the performance of each method for within dataset predictions (intra-dataset) and across datasets (inter-dataset) based on accuracy, percentage of unclassified cells, and computation time.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eGarnett\u003c/code\u003e, \u003ccode\u003eMoana\u003c/code\u003e, \u003ccode\u003eDigitalCellSorter\u003c/code\u003e, \u003ccode\u003eSCINA\u003c/code\u003e, \u003ccode\u003escVI\u003c/code\u003e, \u003ccode\u003eCell-BLAST\u003c/code\u003e, \u003ccode\u003eACTINN\u003c/code\u003e, \u003ccode\u003eLAmbDA\u003c/code\u003e, \u003ccode\u003escmapcluster\u003c/code\u003e, \u003ccode\u003escmapcell\u003c/code\u003e, \u003ccode\u003escPred\u003c/code\u003e, \u003ccode\u003eCHETAH\u003c/code\u003e, \u003ccode\u003eCaSTLe\u003c/code\u003e, \u003ccode\u003eSingleR\u003c/code\u003e, \u003ccode\u003escID\u003c/code\u003e, \u003ccode\u003esingleCellNet\u003c/code\u003e, \u003ccode\u003eLDA\u003c/code\u003e, \u003ccode\u003eNMC\u003c/code\u003e, \u003ccode\u003eRF\u003c/code\u003e, \u003ccode\u003eSVM\u003c/code\u003e, \u003ccode\u003eSVM\u0026lt;sub\u0026gt;rejection\u0026lt;/sub\u0026gt;\u003c/code\u003e, \u003ccode\u003ekNN\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e All classifiers performed well. The authors recommended SVM\u003csub\u003erejection\u003c/sub\u003e classifier (with a linear kernel). Other classifiers include SVM , singleCellNet, scmapcell, and scPred were also of high performances.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn their experiments, incorporating prior knowledge in the form of marker genes does not improve the performance.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e A Snakemake workflow, \u003ca href=\"https://github.com/tabdelaal/scRNAseq_Benchmark/\"\u003escRNAseq_Benchmark\u003c/a\u003e, was provided to automate the benchmarking analyses.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVariant Calling\u003c/h3\u003e\u003ca id=\"user-content-variant-calling\" class=\"anchor\" aria-label=\"Permalink: Variant Calling\" href=\"#variant-calling\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1863-4\" rel=\"nofollow\"\u003eSystematic comparative analysis of single-nucleotide variant detection methods from single-cell RNA sequencing data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Fenglin Liu*, Yuanyuan Zhang*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, November 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This paper compared seven variant callers using both simulation and real scRNA-seq datasets and identified several elements influencing their performance, including read depth, variant allele frequency, and specific genomic contexts. Sensitivity and specificity were the benchmarking metrics used.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eSAMtools\u003c/code\u003e, \u003ccode\u003eGATK\u003c/code\u003e, \u003ccode\u003eCTAT\u003c/code\u003e, \u003ccode\u003eFreeBayes\u003c/code\u003e, \u003ccode\u003eMuTect2\u003c/code\u003e, \u003ccode\u003eStrelka2\u003c/code\u003e, \u003ccode\u003eVarScan2\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Varies, see figure 7 for a flowchart breakdown. Generally, \u003ca href=\"http://www.htslib.org/\" rel=\"nofollow\"\u003eSAMtools\u003c/a\u003e (most sensitive, lower specificity in intronic or high-identity regions), \u003ca href=\"https://github.com/Illumina/strelka\"\u003eStrelka2\u003c/a\u003e (good performance when read depth \u0026gt;5), \u003ca href=\"https://github.com/ekg/freebayes\"\u003eFreeBayes\u003c/a\u003e (good specificity/sensitivity in cases with high variant allele frequencies), and CTAT (no alignment step necessary) were top performers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors made their benchmarking code available \u003ca href=\"https://github.com/fenglin0/benchmarking_variant_callers\"\u003eon Github\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eATAC-seq\u003c/h3\u003e\u003ca id=\"user-content-atac-seq\" class=\"anchor\" aria-label=\"Permalink: ATAC-seq\" href=\"#atac-seq\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1854-5\" rel=\"nofollow\"\u003eAssessment of computational methods for the analysis of single-cell ATAC-seq data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Caleb Lareau*, Tommaso Andreani*, Micheal E. Vinyard*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, November 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e This study compares 10 methods for scATAC-seq processing and featurizing using 13 synthetic and real datasets from diverse tissues and organisms.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eBROCKMAN\u003c/code\u003e, \u003ccode\u003echromVAR\u003c/code\u003e, \u003ccode\u003ecisTopic\u003c/code\u003e, \u003ccode\u003eCicero\u003c/code\u003e, \u003ccode\u003eGene Scoring\u003c/code\u003e, \u003ccode\u003eCusanovich2018\u003c/code\u003e, \u003ccode\u003escABC\u003c/code\u003e, \u003ccode\u003eScasat\u003c/code\u003e, \u003ccode\u003eSCRAT\u003c/code\u003e, \u003ccode\u003eSnapATAC\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e \u003ca href=\"https://github.com/r3fang/SnapATAC\"\u003eSnapATAC\u003c/a\u003e, \u003ca href=\"https://www.ncbi.nlm.nih.gov/pubmed/30078704\" rel=\"nofollow\"\u003eCusanovich2018\u003c/a\u003e, and \u003ca href=\"https://github.com/aertslab/cisTopic\"\u003ecisTopic\u003c/a\u003e were the top performers for separating cell populations of different coverages and noise levels. SnapATAC was the only method capable of analyzing a large dataset (\u0026gt;80k cells).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors have made their benchmarking code available \u003ca href=\"https://github.com/pinellolab/scATAC-benchmarking/\"\u003eon Github\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eStatistics\u003c/h2\u003e\u003ca id=\"user-content-statistics\" class=\"anchor\" aria-label=\"Permalink: Statistics\" href=\"#statistics\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFalse Discovery Rates\u003c/h3\u003e\u003ca id=\"user-content-false-discovery-rates\" class=\"anchor\" aria-label=\"Permalink: False Discovery Rates\" href=\"#false-discovery-rates\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1716-1\" rel=\"nofollow\"\u003eA practical guide to methods controlling false discoveries in computational biology\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Keegan Korthauer*, Patrick K. Kimes*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, June 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e An benchmark comparison of the accuracy, applicability, and ease of use of two classic and six modern methods that control for the false discovery rate (FDR). Used simulation studies as well as six case studies in computational biology (specifically differential expression testing in bulk RNA-seq, differential expression testing in single-cell RNA-seq, differential abundance testing and correlation analysis in 16S microbiome data, differential binding testing in ChIP-seq, genome-wide association testing, and gene set analysis).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e Benjamini-Hochberg, Storeyâs q-value, conditional local FDR (LFDR), FDR regression (FDRreg), independent hypothesis weighting (IHW), adaptive shrinkage (ASH), Boca and Leekâs FDR regression (BL), and adaptive p-value thresholding (AdaPT).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e Modern FDR methods that use an informative covariate (as opposed to only \u003cem\u003ep\u003c/em\u003e-values) leads to more power while controlling the FDR over classic methods. The improvement of the modern FDR methods over the classic methods increases with the informativeness of the covariate, total number of hypothesis tests, and proportion of truly non-null hypotheses.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e Full analyses of the in silico experiments, simulations, and case studies are provided in Additional files 2â41 at \u003ca href=\"https://pkimes.github.io/benchmark-fdr-html/\" rel=\"nofollow\"\u003ehttps://pkimes.github.io/benchmark-fdr-html/\u003c/a\u003e. The source code to reproduce all results in the manuscript and additional files, as well as all figures, is \u003ca href=\"https://github.com/pkimes/benchmark-fdr\"\u003eavailable on GitHub\u003c/a\u003e. An \u003ccode\u003eExperimentHub\u003c/code\u003e package containing the full set of results objects is available through the Bioconductor project, and a Shiny application for interactive exploration of these results is also \u003ca href=\"https://github.com/kdkorthauer/benchmarkfdr-shiny\"\u003eavailable on GitHub\u003c/a\u003e. The source code, ExperimentHub package, and Shiny application are all made available under the MIT license.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eMicrobiome\u003c/h2\u003e\u003ca id=\"user-content-microbiome\" class=\"anchor\" aria-label=\"Permalink: Microbiome\" href=\"#microbiome\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDiversity analysis\u003c/h3\u003e\u003ca id=\"user-content-diversity-analysis\" class=\"anchor\" aria-label=\"Permalink: Diversity analysis\" href=\"#diversity-analysis\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://pubmed.ncbi.nlm.nih.gov/31657871/\" rel=\"nofollow\"\u003eEvaluating Bioinformatic Pipeline Performance for Forensic Microbiome Analysis\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Sierra F. Kaszubinski*, Jennifer L. Pechal*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Journal of Forensic Sciences, 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e Sequence reads from postmortem microbiome samples were analyzed with \u003ca href=\"https://mothur.org\" rel=\"nofollow\"\u003emothur v1.39.5\u003c/a\u003e, \u003ca href=\"https://qiime.org\" rel=\"nofollow\"\u003eQIIME2 v2018.11\u003c/a\u003e, and \u003ca href=\"https://mg-rast.org\" rel=\"nofollow\"\u003eMG-RAST v4.0.3\u003c/a\u003e. For postmortem data, MG-RAST had a much smaller effect size than mothur and QIIME2 due to the twofold reduction in samples. QIIME2 and Mothur returned similar results, with Mothur showing inflated richness due to unclassified taxa. Adjusting minimum library size had significant effects on microbial community structure, sample size less so except for low abundant taxa.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003emothur\u003c/code\u003e, \u003ccode\u003eQIIME2\u003c/code\u003e, \u003ccode\u003eMG-RAST\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e QIIME2 was deemed the most appropriate choice for forensic analysis in this study.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e Sequence data are archived through the European Bioinformatics Institute European Nucleotide Archive (\u003ca href=\"http://www.ebi.ac.uk/ena\" rel=\"nofollow\"\u003ewww.ebi.ac.uk/ena\u003c/a\u003e) under accession number: PRJEB22642. Pipeline parameters and microbial community analyses are available on \u003ca href=\"https://github.com/sierrakasz/postmortem-analysis\"\u003eGitHub\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHi-C/Hi-ChIP\u003c/h2\u003e\u003ca id=\"user-content-hi-chi-chip\" class=\"anchor\" aria-label=\"Permalink: Hi-C/Hi-ChIP\" href=\"#hi-chi-chip\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTAD Calling\u003c/h3\u003e\u003ca id=\"user-content-tad-calling\" class=\"anchor\" aria-label=\"Permalink: TAD Calling\" href=\"#tad-calling\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.1186/s13059-018-1596-9\" rel=\"nofollow\"\u003eComparison of computational methods for the identification of topologically associating domains\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Marie Zufferey*, Daniele Tavernari*, et al.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Genome Biology, December 2018\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e In this study, the authors compared the performance of 22 TAD callers, each on 20 different conditions (4 map resolutions each normalized with 2 independent strategies, plus 12 additional contact maps with variable sequencing depth) and assessed their performance via concordance, robustness to data resolution and normalization method, and ability to recapitulate biological features typically associated with TADs and TAD boundaries. Assessments were performed on high-resolution Hi-C data from GM12878 and validated in other datasets for select callers.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003e3DNetMod\u003c/code\u003e, \u003ccode\u003earmatus\u003c/code\u003e, \u003ccode\u003earrowhead\u003c/code\u003e, \u003ccode\u003eCaTCH\u003c/code\u003e, \u003ccode\u003eCHDF\u003c/code\u003e, \u003ccode\u003echromoR\u003c/code\u003e, \u003ccode\u003eClusterTAD\u003c/code\u003e, \u003ccode\u003eDI\u003c/code\u003e, \u003ccode\u003eEAST\u003c/code\u003e, \u003ccode\u003eGMAP\u003c/code\u003e, \u003ccode\u003eHiCExplorer\u003c/code\u003e, \u003ccode\u003eHiCseq\u003c/code\u003e, \u003ccode\u003eHiTAD\u003c/code\u003e, \u003ccode\u003eICFinder\u003c/code\u003e, \u003ccode\u003eIS\u003c/code\u003e, \u003ccode\u003ematryoshka\u003c/code\u003e, \u003ccode\u003eMrTADFinder\u003c/code\u003e, \u003ccode\u003ePSYCHIC\u003c/code\u003e, \u003ccode\u003espectral\u003c/code\u003e, \u003ccode\u003eTADbit\u003c/code\u003e, \u003ccode\u003eTADtree\u003c/code\u003e, and \u003ccode\u003eTopDom\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e See \u003cstrong\u003eTable 2\u003c/strong\u003e for a succinct results summary. In general, the authors found that \u003ccode\u003eTopDom\u003c/code\u003e, \u003ccode\u003eHiCseg\u003c/code\u003e, and \u003ccode\u003eCaTCH\u003c/code\u003e satisfied at least four out of five criteria: robustness with respect to bin size (resolution) and normalization strategy (ICE and LGF); cost-effective performance based on the ability of the caller to identify concordant TADs with \u0026lt;1% of reads; reproducibility of TADs called by other callers; computational efficiency; and biological relevance based on previosly reported TAD-associated features. The authors do, however, note that callers that attempt to call hierarchical TAD structures generally performed worse than those that do not, potentially due to requiring higher data resolution.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eNormalization Methods\u003c/h3\u003e\u003ca id=\"user-content-normalization-methods-1\" class=\"anchor\" aria-label=\"Permalink: Normalization Methods\" href=\"#normalization-methods-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTitle:\u003c/strong\u003e \u003ca href=\"https://doi.org/10.2144/btn-2019-0105\" rel=\"nofollow\"\u003eComparison of normalization methods for Hi-C data\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAuthors:\u003c/strong\u003e Hongqiang Lyu, Erhu Liu, Zhifang Wu\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eJournal Info:\u003c/strong\u003e Biotechniques, October 2019\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eDescription:\u003c/strong\u003e In this study, the authors compared the performance of 6 Hi-C normalization methods at 8 different resolution levels - 2.5M, 1M, 500k, 250k, 100k, 50k, 10k, and 5k from 4 different Hi-C studies. These methods were compared for heat map texture, statistical quality, influence of resolution, consistency of distance stratum, and reproducibility of TAD architecture. The authors assessed the quality of statistics by comparing the distribution of interaction frequency, correlation of replicates, and comparability of replicates between contexts.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eTools/methods compared:\u003c/strong\u003e \u003ccode\u003eSCN\u003c/code\u003e, \u003ccode\u003eHiCNorm\u003c/code\u003e, \u003ccode\u003eICE\u003c/code\u003e, \u003ccode\u003eKR\u003c/code\u003e, \u003ccode\u003echromoR\u003c/code\u003e, and \u003ccode\u003emultiHiCcompare\u003c/code\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRecommendation(s):\u003c/strong\u003e See \u003cstrong\u003eTable 2\u003c/strong\u003e for a succinct results summary. In summary, the authors found that use-case determines the best tool. All of these methods except \u003ccode\u003emultiHiCcompare\u003c/code\u003e are single sample methods, which led to \u003ccode\u003emultiHiCcompare\u003c/code\u003e performing the best in many considerations, including distribution of interaction frequency, comparability between contexts, and consistency of distance stratum. However, \u003ccode\u003emultiHiCcompare\u003c/code\u003e is by far the most computationally expensive and requires a large number of samples to be useful. \u003ccode\u003eSCN\u003c/code\u003e and \u003ccode\u003eKR\u003c/code\u003e show the best reproducibility of TAD architecture across various resolutions. \u003ccode\u003echromoR\u003c/code\u003e was found to blur interaction heatmaps at lower resolutions due to its de-noising procedure, though it was also found the achieve the best correlation of replicates at mid-resolutions (500k, 250k).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAdditional links:\u003c/strong\u003e The authors have made their implementations to run the various methods available on \u003ca href=\"https://github.com/lhqxinghun/bioinformatics/tree/master/Hi-C/NormCompare\"\u003eGitHub\u003c/a\u003e.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eContributors\u003c/h1\u003e\u003ca id=\"user-content-contributors\" class=\"anchor\" aria-label=\"Permalink: Contributors\" href=\"#contributors\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eJared Andrews (\u003ca href=\"https://github.com/j-andrews7/\"\u003e@j-andrews7\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eKevin Blighe (\u003ca href=\"https://github.com/kevinblighe\"\u003e@kevinblighe\u003c/a\u003e, \u003ca href=\"https://www.biostars.org/u/41557/\" rel=\"nofollow\"\u003ebiostars\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eLudwig Geistlinger (\u003ca href=\"https://github.com/lgeistlinger\"\u003e@lgeistlinger\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eJeremy Leipzig (\u003ca href=\"https://github.com/leipzig\"\u003e@leipzig\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eAvi Srivastava (\u003ca href=\"https://github.com/k3yavi\"\u003e@k3yavi\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eStephanie Hicks (\u003ca href=\"https://github.com/stephaniehicks\"\u003e@stephaniehicks\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eSridhar N Srivatsan (\u003ca href=\"https://github.com/sridhar0605\"\u003e@sridhar0605\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eQingzhou Zhang (\u003ca href=\"https://github.com/zqzneptune\"\u003e@zqzneptune\u003c/a\u003e)\u003c/li\u003e\n\u003cli\u003eGuandong Shang (\u003ca href=\"https://github.com/shangguandong1996\"\u003e@shangguandong1996\u003c/a\u003e, \u003ca href=\"https://twitter.com/GuandongS\" rel=\"nofollow\"\u003e@GuandongS\u003c/a\u003e)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Awesome Bioinformatics Benchmarks ","anchor":"awesome-bioinformatics-benchmarks-","htmlText":"Awesome Bioinformatics Benchmarks "},{"level":2,"text":"Contents","anchor":"contents","htmlText":"Contents"},{"level":2,"text":"Rules for Included Papers","anchor":"rules-for-included-papers","htmlText":"Rules for Included Papers"},{"level":2,"text":"Format \u0026 Organization","anchor":"format--organization","htmlText":"Format \u0026amp; Organization"},{"level":1,"text":"Benchmarking Theory","anchor":"benchmarking-theory","htmlText":"Benchmarking Theory"},{"level":1,"text":"Tool/Method Sections","anchor":"toolmethod-sections","htmlText":"Tool/Method Sections"},{"level":2,"text":"DNase, ATAC, and ChIP-seq","anchor":"dnase-atac-and-chip-seq","htmlText":"DNase, ATAC, and ChIP-seq"},{"level":3,"text":"Peak Callers","anchor":"peak-callers","htmlText":"Peak Callers"},{"level":3,"text":"Normalization Methods","anchor":"normalization-methods","htmlText":"Normalization Methods"},{"level":2,"text":"RNA-seq","anchor":"rna-seq","htmlText":"RNA-seq"},{"level":3,"text":"Alignment/Quantification Methods","anchor":"alignmentquantification-methods","htmlText":"Alignment/Quantification Methods"},{"level":3,"text":"Normalisation Methods","anchor":"normalisation-methods","htmlText":"Normalisation Methods"},{"level":3,"text":"Differential Gene Expression","anchor":"differential-gene-expression","htmlText":"Differential Gene Expression"},{"level":3,"text":"Gene Set Enrichment Analysis","anchor":"gene-set-enrichment-analysis","htmlText":"Gene Set Enrichment Analysis"},{"level":3,"text":"Differential Splicing","anchor":"differential-splicing","htmlText":"Differential Splicing"},{"level":3,"text":"de novo Assembly and Quantification","anchor":"de-novo-assembly-and-quantification","htmlText":"de novo Assembly and Quantification"},{"level":3,"text":"Cell-Type Deconvolution","anchor":"cell-type-deconvolution","htmlText":"Cell-Type Deconvolution"},{"level":2,"text":"CRISPR Screens","anchor":"crispr-screens","htmlText":"CRISPR Screens"},{"level":3,"text":"Hit/Dependency Identification","anchor":"hitdependency-identification","htmlText":"Hit/Dependency Identification"},{"level":2,"text":"DNA Methylation","anchor":"dna-methylation","htmlText":"DNA Methylation"},{"level":3,"text":"Platforms and Library Prep Methods","anchor":"platforms-and-library-prep-methods","htmlText":"Platforms and Library Prep Methods"},{"level":3,"text":"CpG Methylation from Nanopore Data","anchor":"cpg-methylation-from-nanopore-data","htmlText":"CpG Methylation from Nanopore Data"},{"level":2,"text":"Variant Callers","anchor":"variant-callers","htmlText":"Variant Callers"},{"level":3,"text":"Germline SNP/Indel Callers","anchor":"germline-snpindel-callers","htmlText":"Germline SNP/Indel Callers"},{"level":3,"text":"Somatic SNV/Indel callers","anchor":"somatic-snvindel-callers","htmlText":"Somatic SNV/Indel callers"},{"level":3,"text":"CNV Callers","anchor":"cnv-callers","htmlText":"CNV Callers"},{"level":3,"text":"SV callers","anchor":"sv-callers","htmlText":"SV callers"},{"level":2,"text":"Single Cell","anchor":"single-cell","htmlText":"Single Cell"},{"level":3,"text":"scRNA Transformations/Normalization","anchor":"scrna-transformationsnormalization","htmlText":"scRNA Transformations/Normalization"},{"level":3,"text":"Gene Signature Scoring","anchor":"gene-signature-scoring","htmlText":"Gene Signature Scoring"},{"level":3,"text":"scRNA Sequencing Protocols","anchor":"scrna-sequencing-protocols","htmlText":"scRNA Sequencing Protocols"},{"level":3,"text":"scRNA Analysis Pipelines","anchor":"scrna-analysis-pipelines","htmlText":"scRNA Analysis Pipelines"},{"level":3,"text":"scRNA Imputation Methods","anchor":"scrna-imputation-methods","htmlText":"scRNA Imputation Methods"},{"level":3,"text":"scRNA Differential Gene Expression","anchor":"scrna-differential-gene-expression","htmlText":"scRNA Differential Gene Expression"},{"level":3,"text":"Trajectory Inference","anchor":"trajectory-inference","htmlText":"Trajectory Inference"},{"level":3,"text":"Gene Regulatory Network Inference","anchor":"gene-regulatory-network-inference","htmlText":"Gene Regulatory Network Inference"},{"level":3,"text":"Integration/Batch Correction","anchor":"integrationbatch-correction","htmlText":"Integration/Batch Correction"},{"level":3,"text":"Dimensionality Reduction","anchor":"dimensionality-reduction","htmlText":"Dimensionality Reduction"},{"level":3,"text":"Cell Annotation/Inference","anchor":"cell-annotationinference","htmlText":"Cell Annotation/Inference"},{"level":3,"text":"Variant Calling","anchor":"variant-calling","htmlText":"Variant Calling"},{"level":3,"text":"ATAC-seq","anchor":"atac-seq","htmlText":"ATAC-seq"},{"level":2,"text":"Statistics","anchor":"statistics","htmlText":"Statistics"},{"level":3,"text":"False Discovery Rates","anchor":"false-discovery-rates","htmlText":"False Discovery Rates"},{"level":2,"text":"Microbiome","anchor":"microbiome","htmlText":"Microbiome"},{"level":3,"text":"Diversity analysis","anchor":"diversity-analysis","htmlText":"Diversity analysis"},{"level":2,"text":"Hi-C/Hi-ChIP","anchor":"hi-chi-chip","htmlText":"Hi-C/Hi-ChIP"},{"level":3,"text":"TAD Calling","anchor":"tad-calling","htmlText":"TAD Calling"},{"level":3,"text":"Normalization Methods","anchor":"normalization-methods-1","htmlText":"Normalization Methods"},{"level":1,"text":"Contributors","anchor":"contributors","htmlText":"Contributors"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fj-andrews7%2Fawesome-bioinformatics-benchmarks"}},{"displayName":"CODE_OF_CONDUCT.md","repoName":"awesome-bioinformatics-benchmarks","refName":"master","path":"CODE_OF_CONDUCT.md","preferredFileType":"code_of_conduct","tabName":"Code of conduct","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fj-andrews7%2Fawesome-bioinformatics-benchmarks"}},{"displayName":"LICENSE","repoName":"awesome-bioinformatics-benchmarks","refName":"master","path":"LICENSE","preferredFileType":"license","tabName":"GPL-3.0","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fj-andrews7%2Fawesome-bioinformatics-benchmarks"}}],"overviewFilesProcessingTime":63.71273600000001,"createFromTemplatePath":"/new?template_name=awesome-bioinformatics-benchmarks\u0026template_owner=j-andrews7"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/j-andrews7/awesome-bioinformatics-benchmarks/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/j-andrews7/awesome-bioinformatics-benchmarks/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/j-andrews7/awesome-bioinformatics-benchmarks/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           124 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/j-andrews7/awesome-bioinformatics-benchmarks/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".github, (Directory)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/tree/master/.github" title=".github">
                          .github
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".github, (Directory)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/tree/master/.github" title=".github">
                          .github
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="CODE_OF_CONDUCT.md, (File)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/CODE_OF_CONDUCT.md" title="CODE_OF_CONDUCT.md">
                          CODE_OF_CONDUCT.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="CODE_OF_CONDUCT.md, (File)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/CODE_OF_CONDUCT.md" title="CODE_OF_CONDUCT.md">
                          CODE_OF_CONDUCT.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/j-andrews7/awesome-bioinformatics-benchmarks/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-code-of-conduct" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.048 2.241c.964-.709 2.079-1.238 3.325-1.241a4.616 4.616 0 0 1 3.282 1.355c.41.408.757.86.996 1.428.238.568.348 1.206.347 1.968 0 2.193-1.505 4.254-3.081 5.862-1.496 1.526-3.213 2.796-4.249 3.563l-.22.163a.749.749 0 0 1-.895 0l-.221-.163c-1.036-.767-2.753-2.037-4.249-3.563C1.51 10.008.007 7.952.002 5.762a4.614 4.614 0 0 1 1.353-3.407C3.123.585 6.223.537 8.048 2.24Zm-1.153.983c-1.25-1.033-3.321-.967-4.48.191a3.115 3.115 0 0 0-.913 2.335c0 1.556 1.109 3.24 2.652 4.813C5.463 11.898 6.96 13.032 8 13.805c.353-.262.758-.565 1.191-.905l-1.326-1.223a.75.75 0 0 1 1.018-1.102l1.48 1.366c.328-.281.659-.577.984-.887L9.99 9.802a.75.75 0 1 1 1.019-1.103l1.384 1.28c.295-.329.566-.661.81-.995L12.92 8.7l-1.167-1.168c-.674-.671-1.78-.664-2.474.03-.268.269-.538.537-.802.797-.893.882-2.319.843-3.185-.032-.346-.35-.693-.697-1.043-1.047a.75.75 0 0 1-.04-1.016c.162-.191.336-.401.52-.623.62-.748 1.356-1.637 2.166-2.417Zm7.112 4.442c.313-.65.491-1.293.491-1.916v-.001c0-.614-.088-1.045-.23-1.385-.143-.339-.357-.633-.673-.949a3.111 3.111 0 0 0-2.218-.915c-1.092.003-2.165.627-3.226 1.602-.823.755-1.554 1.637-2.228 2.45l-.127.154.562.566a.755.755 0 0 0 1.066.02l.794-.79c1.258-1.258 3.312-1.31 4.594-.032.396.394.792.791 1.173 1.173Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="Code of conduct">
                        Code of conduct
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="GPL-3.0 license">
                        GPL-3.0 license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Awesome Bioinformatics Benchmarks
                      <a href="https://github.com/j-andrews7/awesome-bioinformatics-benchmarks/actions/workflows/action.yml/badge.svg" rel="noopener noreferrer" target="_blank">
                       <img alt="Build Status" src="https://github.com/j-andrews7/awesome-bioinformatics-benchmarks/actions/workflows/action.yml/badge.svg" style="max-width: 100%;"/>
                      </a>
                     </h1>
                     <a aria-label="Permalink: Awesome Bioinformatics Benchmarks " class="anchor" href="#awesome-bioinformatics-benchmarks-" id="user-content-awesome-bioinformatics-benchmarks-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     A curated list of bioinformatics benchmarking papers and resources.
                    </p>
                    <p dir="auto">
                     The credit for this format goes to Sean Davis for his
                     <a href="https://github.com/seandavi/awesome-single-cell">
                      awesome-single-cell
                     </a>
                     repository and Ming Tang for his
                     <a href="https://github.com/crazyhottommy/ChIP-seq-analysis">
                      ChIP-seq-analysis
                     </a>
                     repository.
                    </p>
                    <p dir="auto">
                     If you have a benchmarking study that is not yet included on this list, please make a
                     <a href="https://github.com/j-andrews7/Awesome-Bioinformatics-Benchmarks/pulls">
                      Pull Request
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Contents
                     </h2>
                     <a aria-label="Permalink: Contents" class="anchor" href="#contents" id="user-content-contents">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <a href="#awesome-bioinformatics-benchmarks-">
                       Awesome Bioinformatics Benchmarks
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#contents">
                         Contents
                        </a>
                       </li>
                       <li>
                        <a href="#rules-for-included-papers">
                         Rules for Included Papers
                        </a>
                       </li>
                       <li>
                        <a href="#format--organization">
                         Format &amp; Organization
                        </a>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#benchmarking-theory">
                       Benchmarking Theory
                      </a>
                     </li>
                     <li>
                      <a href="#toolmethod-sections">
                       Tool/Method Sections
                      </a>
                      <ul dir="auto">
                       <li>
                        <a href="#dnase-atac-and-chip-seq">
                         DNase, ATAC, and ChIP-seq
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#peak-callers">
                           Peak Callers
                          </a>
                         </li>
                         <li>
                          <a href="#normalization-methods">
                           Normalization Methods
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#rna-seq">
                         RNA-seq
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#alignmentquantification-methods">
                           Alignment/Quantification Methods
                          </a>
                         </li>
                         <li>
                          <a href="#normalisation-methods">
                           Normalisation Methods
                          </a>
                         </li>
                         <li>
                          <a href="#differential-gene-expression">
                           Differential Gene Expression
                          </a>
                         </li>
                         <li>
                          <a href="#gene-set-enrichment-analysis">
                           Gene Set Enrichment Analysis
                          </a>
                         </li>
                         <li>
                          <a href="#differential-splicing">
                           Differential Splicing
                          </a>
                         </li>
                         <li>
                          <a href="#de-novo-assembly-and-quantification">
                           <em>
                            de novo
                           </em>
                           Assembly and Quantification
                          </a>
                         </li>
                         <li>
                          <a href="#cell-type-deconvolution">
                           Cell-Type Deconvolution
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#crispr-screens">
                         CRISPR Screens
                        </a>
                       </li>
                       <li>
                        <a href="#dna-methylation">
                         DNA Methylation
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#platforms-and-library-prep-methods">
                           Platforms and Library Prep Methods
                          </a>
                         </li>
                         <li>
                          <a href="#cpg-methylation-from-nanopore-data">
                           Cpg Methylation from Nanopore Data
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#variant-callers">
                         Variant Callers
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#germline-snpindel-callers">
                           Germline SNP/Indel Callers
                          </a>
                         </li>
                         <li>
                          <a href="#somatic-snvindel-callers">
                           Somatic SNV/Indel callers
                          </a>
                         </li>
                         <li>
                          <a href="#cnv-callers">
                           CNV Callers
                          </a>
                         </li>
                         <li>
                          <a href="#sv-callers">
                           SV callers
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#single-cell">
                         Single Cell
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#scrna-transformationsnormalization">
                           scRNA Transformation/Normalization
                          </a>
                         </li>
                         <li>
                          <a href="#gene-signature-scoring">
                           Gene Signature Scoring
                          </a>
                         </li>
                         <li>
                          <a href="#scrna-sequencing-protocols">
                           scRNA Sequencing Protocols
                          </a>
                         </li>
                         <li>
                          <a href="#scrna-analysis-pipelines">
                           scRNA Analysis Pipelines
                          </a>
                         </li>
                         <li>
                          <a href="#scrna-imputation-methods">
                           scRNA Imputation Methods
                          </a>
                         </li>
                         <li>
                          <a href="#scrna-differential-gene-expression">
                           scRNA Differential Gene Expression
                          </a>
                         </li>
                         <li>
                          <a href="#trajectory-inference">
                           Trajectory Inference
                          </a>
                         </li>
                         <li>
                          <a href="#gene-regulatory-network-inference">
                           Gene Regulatory Network Inference
                          </a>
                         </li>
                         <li>
                          <a href="#integrationbatch-correction">
                           Integration/Batch Correction
                          </a>
                         </li>
                         <li>
                          <a href="#dimensionality-reduction">
                           Dimensionality Reduction
                          </a>
                         </li>
                         <li>
                          <a href="#cell-annotationinference">
                           Cell Annotation/Inference
                          </a>
                         </li>
                         <li>
                          <a href="#variant-calling">
                           Variant Calling
                          </a>
                         </li>
                         <li>
                          <a href="#atac-seq">
                           ATAC-seq
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#statistics">
                         Statistics
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#false-discovery-rates">
                           False Discovery Rates
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#microbiome">
                         Microbiome
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#diversity-analysis">
                           Diversity analysis
                          </a>
                         </li>
                        </ul>
                       </li>
                       <li>
                        <a href="#hi-chi-chip">
                         Hi-C/Hi-ChIP
                        </a>
                        <ul dir="auto">
                         <li>
                          <a href="#tad-calling">
                           TAD Calling
                          </a>
                         </li>
                         <li>
                          <a href="#normalization-methods-1">
                           Normalization Methods
                          </a>
                         </li>
                        </ul>
                       </li>
                      </ul>
                     </li>
                     <li>
                      <a href="#contributors">
                       Contributors
                      </a>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Rules for Included Papers
                     </h2>
                     <a aria-label="Permalink: Rules for Included Papers" class="anchor" href="#rules-for-included-papers" id="user-content-rules-for-included-papers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Papers must be objective comparisons of 3 or more tools/methods.
                     </li>
                     <li>
                      Papers must be
                      <strong>
                       awesome
                      </strong>
                      . This list isn't meant to chronicle every benchmarking study ever performed, only those that are particularly expansive, well done, and/or provide unique insights.
                     </li>
                     <li>
                      Papers should generally
                      <strong>
                       not
                      </strong>
                      be from authors showing why their tool/method is better than others.
                     </li>
                     <li>
                      Benchmarking data should be publicly available or simulation code/methods must be well-documented and reproducible.
                     </li>
                    </ul>
                    <p dir="auto">
                     Additional guidelines/rules may be added as necessary.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Format &amp; Organization
                     </h2>
                     <a aria-label="Permalink: Format &amp; Organization" class="anchor" href="#format--organization" id="user-content-format--organization">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Please include the following information when adding papers.
                    </p>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links (optional):
                     </strong>
                    </p>
                    <p dir="auto">
                     Papers within each section should be ordered by publication date, with more recent papers listed first.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Benchmarking Theory
                     </h1>
                     <a aria-label="Permalink: Benchmarking Theory" class="anchor" href="#benchmarking-theory" id="user-content-benchmarking-theory">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1738-8" rel="nofollow">
                      Essential guidelines for computational method benchmarking
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Lukas Weber, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, June 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper presents 10 main guidelines for conducting and writing benchmark papers covering necessary data, methods and metric choices, reproducible research, and documentation.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/s41467-019-09406-4" rel="nofollow">
                      Systematic benchmarking of omics computational tools
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Sergei Mangul, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Communications, March 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     A survey of 25 benchmarking studies published between 2011 and 2017 in terms of design, methods, and information types. Discusses overfitting, sharing, incentives.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Tool/Method Sections
                     </h1>
                     <a aria-label="Permalink: Tool/Method Sections" class="anchor" href="#toolmethod-sections" id="user-content-toolmethod-sections">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Additional sections/sub-sections can be added as needed.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      DNase, ATAC, and ChIP-seq
                     </h2>
                     <a aria-label="Permalink: DNase, ATAC, and ChIP-seq" class="anchor" href="#dnase-atac-and-chip-seq" id="user-content-dnase-atac-and-chip-seq">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Peak Callers
                     </h3>
                     <a aria-label="Permalink: Peak Callers" class="anchor" href="#peak-callers" id="user-content-peak-callers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://academic.oup.com/bib/article/18/3/441/2453291" rel="nofollow">
                      Features that define the best ChIP-seq peak calling algorithms
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Reuben Thomas, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Briefings in Bioinformatics, May 2017
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared six peak calling methods on 300 simulated and three real ChIP-seq data sets across a range of significance values.
Methods were scored by sensitivity, precision, and F-score.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      GEM
                     </code>
                     ,
                     <code>
                      MACS2
                     </code>
                     ,
                     <code>
                      MUSIC
                     </code>
                     ,
                     <code>
                      BCP
                     </code>
                     ,
                     <code>
                      Threshold-based method (TM)
                     </code>
                     ,
                     <code>
                      ZINBA
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Varies.
                     <a href="http://ranger.sourceforge.net/manual1.18.html" rel="nofollow">
                      BCP
                     </a>
                     and
                     <a href="https://github.com/taoliu/MACS">
                      MACS2
                     </a>
                     performed the best across all metrics on the simulated data.
For Tbx5 ChIP-seq,
                     <a href="http://groups.csail.mit.edu/cgs/gem/" rel="nofollow">
                      GEM
                     </a>
                     performed the best, with BCP also scoring highly.
For histone H3K36me3 and H3K4me3 data, all methods performed relatively comparably with the exception of ZINBA, which the authors could not get to run properly.
                     <a href="https://github.com/gersteinlab/MUSIC">
                      MUSIC
                     </a>
                     and BCP had a slight edge over the others for the histone data.
                    </p>
                    <p dir="auto">
                     More generally, they found that methods that utilize variable window sizes and Poisson test to rank peaks are more powerful than those that use a Binomial test.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0096303" rel="nofollow">
                      A Comparison of Peak Callers Used for DNase-Seq Data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Hashem Koohy, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     PLoS ONE, May 2014
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compares four peak callers specificity and sensitivity on DNase-seq data from two publications composed of three cell types, using ENCODE data for the same cell types as a benchmark.
The authors tested multiple parameters for each caller to determine the best settings for DNase-seq data for each.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      F-seq
                     </code>
                     ,
                     <code>
                      Hotspot
                     </code>
                     ,
                     <code>
                      MACS2
                     </code>
                     ,
                     <code>
                      ZINBA
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     <a href="https://github.com/aboyle/F-seq">
                      F-seq
                     </a>
                     was the most sensitive, though
                     <a href="https://github.com/taoliu/MACS">
                      MACS2
                     </a>
                     and
                     <a href="https://github.com/rthurman/hotspot">
                      Hotspot
                     </a>
                     both performed competitively as well.
ZINBA was the least performant by a massive margin, requiring much more time to run, and was also the least sensitive.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Normalization Methods
                     </h3>
                     <a aria-label="Permalink: Normalization Methods" class="anchor" href="#normalization-methods" id="user-content-normalization-methods">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-020-00342-y" rel="nofollow">
                      ATAC-seq normalization method can significantly affect differential accessibility analysis and interpretation
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Jake J. Reske, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Epigenetics &amp; Chromatin, April 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     The study examines how different ATAC-seq data normalization methods impact the analysis and interpretation of differential chromatin accessibility (DA). Using both in vivo and published yeast ATAC-seq datasets, the authors demonstrate that the choice of normalization method can significantly alter the identification of differentially accessible regions in the genome.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      MACS2
                     </code>
                     ,
                     <code>
                      DiffBind
                     </code>
                     ,
                     <code>
                      csaw
                     </code>
                     ,
                     <code>
                      voom
                     </code>
                     ,
                     <code>
                      DEseq2
                     </code>
                     ,
                     <code>
                      edgeR
                     </code>
                     ,
                     <code>
                      limma
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The authors recommend systematic comparison of multiple normalization methods before proceeding with differential accessibility analysis. They stress the importance of understanding the potential biases and assumptions of each method. They also propose a generalized workflow for differential ATAC-seq data analysis, emphasizing the need for standardizing molecular complexity before quantification. This workflow is based on the standardized ENCODE pipeline with modifications for ATAC-seq.
                    </p>
                    <p dir="auto">
                     The authors also proposed a generalized workflow for differential accessibility analysis, which can be found in
                     <a href="https://github.com/reskejak/ATAC-seq">
                      Github
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     For ATAC-Seq data anlysis, there is another paper:
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1929-3" rel="nofollow">
                      From reads to insight: a hitchhikerâs guide to ATAC-seq data analysis
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      RNA-seq
                     </h2>
                     <a aria-label="Permalink: RNA-seq" class="anchor" href="#rna-seq" id="user-content-rna-seq">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Alignment/Quantification Methods
                     </h3>
                     <a aria-label="Permalink: Alignment/Quantification Methods" class="anchor" href="#alignmentquantification-methods" id="user-content-alignmentquantification-methods">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.biorxiv.org/content/10.1101/657874v2" rel="nofollow">
                      Alignment and mapping methodology influence transcript abundance estimation
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Avi Srivastava* &amp;  Laraib Malik*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     bioRXiv, October 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compares the influence of mapping and alignment on the accuracy of transcript quantification in both simulated and experimental data, as well as the effect on subsequent differential expression analysis.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Bowtie2
                     </code>
                     ,
                     <code>
                      STAR
                     </code>
                     ,
                     <code>
                      quasi-mapping
                     </code>
                     ,
                     <code>
                      Selective Alignment
                     </code>
                     ,
                     <code>
                      RSEM
                     </code>
                     ,
                     <code>
                      Salmon
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     When trying to choose an approach, a choice can be made by the user performing the analysis based on any time-accuracy tradeoff they wish to make. In terms of speed, quasi-mapping is the fastest approach, followed by
                     <a href="https://github.com/COMBINE-lab/salmon">
                      Selective Alignment
                     </a>
                     (SA) then
                     <a href="https://github.com/alexdobin/STAR">
                      STAR
                     </a>
                     .
                     <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" rel="nofollow">
                      Bowtie2
                     </a>
                     was considerably slower than all three of these approaches. However, in terms of accuracy, SA yielded the best results, followed by alignment to the genome (with subsequent transcriptomic projection) using STAR and SA (using carefully selected decoy sequences). Bowtie2 generally performed similarly to SA, but without the benefit of decoy sequences, seemed to admit more spurious mappings. Finally, lightweight mapping of sequencing reads to the transcriptome showed the lowest overall consistency with quantifications derived from the oracle alignments. Note: Both Selective Alignment and quasi-mapping are part of the
                     <a href="https://combine-lab.github.io/salmon/" rel="nofollow">
                      salmon
                     </a>
                     codebase.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1308-8" rel="nofollow">
                      A comprehensive evaluation of ensembl, RefSeq, and UCSC annotations in the context of RNA-seq read mapping and gene quantification
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Shanrong Zhao* &amp; Baohong Zhang
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     BMC Genomics, February 2015
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compares the effect of different gene annotations in the context of RNA-seq mapping and gene quantification using data from the Human Body Map 2.0 Project.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Ensembl
                     </code>
                     ,
                     <code>
                      Refseq
                     </code>
                     ,
                     <code>
                      UCSC
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Though the authors warn there is no "best" set of annotations to use, they do emphasize the impact that annotation choice can have on downstream analyses such as differential gene expression, as genes with identical gene symbols can map to completely different regions in different annotations. Though Ensembl annotations are much more comprehensive than the others, the authors recommend a less complex genome annotation, such as the Refseq annotation, if the RNA-seq is being used as a replacement for microarrays. Conversely, the Ensembl annotations are preferrable if non-coding RNAs are of particular interest.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Normalisation Methods
                     </h3>
                     <a aria-label="Permalink: Normalisation Methods" class="anchor" href="#normalisation-methods" id="user-content-normalisation-methods">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://academic.oup.com/bib/article/19/5/776/3056951" rel="nofollow">
                      Selecting between-sample RNA-Seq normalization methods from the perspective of their assumptions
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Ciaran Evans, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Briefings in Bioinformatics, February 2017
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study underscores the importance of selecting appropriate normalization methods in RNA-Seq studies based on their underlying assumptions. It highlights how normalization choices impact gene behavior analysis under various biological conditions. The paper emphasizes the critical role of assumptions in normalization methods and their effects on the accuracy of downstream analyses, such as detecting differential expression. The study utilized simulations to evaluate normalization methods under varying conditions of mRNA/cell levels and asymmetry in gene expression, assessing error in fold change estimates and empirical error rates in detecting differential expression.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     The paper provides a general discussion on RNA-Seq normalization methods, focusing on the types (e.g., normalization by library size, normalization by distribution, normalization by testing) rather than specific tools. The simulations used various methods, including DEGES, DESeq, Oracle, PoissonSeq, TMM, and Total Count.
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The authors suggest selecting normalization methods based on the specific conditions of the biological experiment. Library size normalization is effective when total mRNA/cell is consistent across conditions, whereas normalization by distribution/testing is better suited for conditions with symmetry in differential expression, regardless of mRNA/cell differences. In scenarios with both asymmetry and varying mRNA/cell levels, both methods are expected to perform poorly.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors' simulation code is available on
                     <a href="https://github.com/ciaranlevans/rnaSeqAssumptions">
                      Github
                     </a>
                     .
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://academic.oup.com/bioinformatics/article/31/22/3625/240923" rel="nofollow">
                      Statistical models for RNA-seq data derived from a two-condition 48-replicate experiment
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Marek GierliÅski*, Christian Cole*, PietÃ¡ Schofield*, Nicholas J. Schurch*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Bioinformatics, November 2015
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compares the effect of normal, log-normal, and negative binomial distribution assumptions on RNA-seq gene read-counts from 48 RNA-seq replicates.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      normal
                     </code>
                     ,
                     <code>
                      log-normal
                     </code>
                     ,
                     <code>
                      negative binomial
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Assuming a normal distribution leads to a large number of false positives during differential gene expression. A log-normal distribution model works well unless a sample contains zero counts. Use tools that assume a negative binomial distribution (
                     <code>
                      edgeR
                     </code>
                     ,
                     <code>
                      DESeq
                     </code>
                     ,
                     <code>
                      DESeq2
                     </code>
                     , etc).
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://academic.oup.com/bib/article/14/6/671/189645" rel="nofollow">
                      A comprehensive evaluation of normalization methods for Illumina high-throughput RNA sequencing data analysis
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Marie-AgnÃ¨s Dillies, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Briefings in Bioinformatics, November 2013
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared seven RNA-seq normalization methods in the context of differential expression analysis on four real datasets and thousands of simulations.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Total Count (TC)
                     </code>
                     ,
                     <code>
                      Upper Quartile (UQ)
                     </code>
                     ,
                     <code>
                      Median (Med),
                     </code>
                     <code>
                      DESeq
                     </code>
                     ,
                     <code>
                      edgeR
                     </code>
                     ,
                     <code>
                      Quantile (Q)
                     </code>
                     ,
                     <code>
                      RPKM
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The authors recommend DESeq (now deprecated and replaced by
                     <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" rel="nofollow">
                      DESeq2
                     </a>
                     ) or
                     <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" rel="nofollow">
                      edgeR
                     </a>
                     , as those methods are robust to the presence of different library sizes and compositions, whereas the (still common) Total Count and RPKM methods are ineffective and should be abandoned.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Differential Gene Expression
                     </h3>
                     <a aria-label="Permalink: Differential Gene Expression" class="anchor" href="#differential-gene-expression" id="user-content-differential-gene-expression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.ncbi.nlm.nih.gov/pubmed/30726870" rel="nofollow">
                      How well do RNA-Seq differential gene expression tools perform in a complex eukaryote? A case study in Arabidopsis thaliana.
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Kimon Froussios*, Nick J Schurch*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Bioinformatics, February 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared nine differential gene expression tools (and their underlying model distributions) in 17 RNA-seq replicates of
                     <em>
                      Arabidopsis thaliana
                     </em>
                     .
Handling of inter-replicate variability and false positive fraction were the benchmarking metrics used.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      baySeq
                     </code>
                     ,
                     <code>
                      DEGseq
                     </code>
                     ,
                     <code>
                      DESeq
                     </code>
                     ,
                     <code>
                      DESeq2
                     </code>
                     ,
                     <code>
                      EBSeq
                     </code>
                     ,
                     <code>
                      edgeR
                     </code>
                     ,
                     <code>
                      limma
                     </code>
                     ,
                     <code>
                      Poisson-Seq
                     </code>
                     ,
                     <code>
                      SAM-Seq
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Six of the tools that utilize negative binomial or log-normal distributions (
                     <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" rel="nofollow">
                      edgeR
                     </a>
                     ,
                     <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" rel="nofollow">
                      DESeq2
                     </a>
                     , DESeq (now deprecated and replaced by
                     <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" rel="nofollow">
                      DESeq2
                     </a>
                     ,
                     <a href="https://bioconductor.org/packages/release/bioc/html/baySeq.html" rel="nofollow">
                      baySeq
                     </a>
                     ,
                     <a href="https://bioconductor.org/packages/release/bioc/html/limma.html" rel="nofollow">
                      limma
                     </a>
                     , and
                     <a href="https://bioconductor.org/packages/release/bioc/html/EBSeq.html" rel="nofollow">
                      EBseq
                     </a>
                     control their identification of false positives well.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors released their benchmarking scripts on
                     <a href="https://github.com/bartongroup/KF_arabidopsis-GRNA">
                      Github
                     </a>
                     .
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.ncbi.nlm.nih.gov/pubmed/27022035" rel="nofollow">
                      How many biological replicates are needed in an RNA-seq experiment and which differential expression tool should you use?
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Nicholas J. Schurch*, PietÃ¡ Schofield*, Marek GierliÅski*, Christian Cole*, Alexander Sherstnev*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     RNA, March 2016
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared 11 differential expression tools on varying numbers of RNA-seq biological replicates (3-42) between two conditions.
Each tool was compared against itself as a standard (using all replicates) and against the other tools.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      baySeq
                     </code>
                     ,
                     <code>
                      cuffdiff
                     </code>
                     ,
                     <code>
                      DEGSeq
                     </code>
                     ,
                     <code>
                      DESeq
                     </code>
                     ,
                     <code>
                      DESeq2
                     </code>
                     ,
                     <code>
                      EBSeq
                     </code>
                     ,
                     <code>
                      edgeR (exact and glm modes)
                     </code>
                     ,
                     <code>
                      limma
                     </code>
                     ,
                     <code>
                      NOISeq
                     </code>
                     ,
                     <code>
                      PoissonSeq
                     </code>
                     ,
                     <code>
                      SAMSeq
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     With fewer than 12 biological replicates,
                     <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" rel="nofollow">
                      edgeR
                     </a>
                     and
                     <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" rel="nofollow">
                      DESeq2
                     </a>
                     were the top performers.
As replicates increased, DESeq (now deprecated and replaced by
                     <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" rel="nofollow">
                      DESeq2
                     </a>
                     did a better job minimizing false positives than other tools.
                    </p>
                    <p dir="auto">
                     Additionally, the authors recommend at least six biological replicates should be used, rising to at least 12 if users want to identify all significantly differentially expressed genes no matter the fold change magnitude.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-9-r95" rel="nofollow">
                      Comprehensive evaluation of differential gene expression analysis methods for RNA-seq data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Franck Rapaport, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, September 2013
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared six differential expression methods on three cell line data sets from ENCODE (GM12878, H1-hESC, and MCF-7) and two samples from the SEQC study, which had a large fraction of differentially expressed genes validated by qRT-PCR.
Specificity, sensitivity, and false positive rate were the main benchmarking metrics used.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Cuffdiff
                     </code>
                     ,
                     <code>
                      edgeR
                     </code>
                     ,
                     <code>
                      DESeq
                     </code>
                     ,
                     <code>
                      PoissonSeq
                     </code>
                     ,
                     <code>
                      baySeq
                     </code>
                     ,
                     <code>
                      limma
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Though no method emerged as favorable in all conditions, those that used negative binomial modeling (DESeq (now deprecated and replaced by
                     <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" rel="nofollow">
                      DESeq2
                     </a>
                     ),
                     <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html" rel="nofollow">
                      edgeR
                     </a>
                     ,
                     <a href="https://bioconductor.org/packages/release/bioc/html/baySeq.html" rel="nofollow">
                      baySeq
                     </a>
                     ) generally performed best.
                    </p>
                    <p dir="auto">
                     The more replicates, the better. Replicate numbers (both biological and technical) have a greater impact on differential detection accuracy than sequencing depth.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Gene Set Enrichment Analysis
                     </h3>
                     <a aria-label="Permalink: Gene Set Enrichment Analysis" class="anchor" href="#gene-set-enrichment-analysis" id="user-content-gene-set-enrichment-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1093/bib/bbz158" rel="nofollow">
                      Toward a gold standard for benchmarking gene set enrichment analysis
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Ludwig Geistlinger, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Briefings in Bioinformatics, February 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper developed a Bioconductor package for reproducible GSEA benchmarking, and used the package to assess 10 widely used enrichment methods with regard to runtime and applicability to RNA-seq data, fraction of enriched gene sets depending on the null hypothesis tested, and recovery of biologically relevant gene sets. The framework can be extended to additional methods, datasets, and benchmark criteria, and should serve as a gold standard for future GSEA benchmarking studies.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     The paper quantitatively asseses the performance of 10 enrichment methods (
                     <code>
                      ORA
                     </code>
                     ,
                     <code>
                      GSEA
                     </code>
                     ,
                     <code>
                      GSA
                     </code>
                     ,
                     <code>
                      PADOG
                     </code>
                     ,
                     <code>
                      SAFE
                     </code>
                     ,
                     <code>
                      CAMERA
                     </code>
                     ,
                     <code>
                      ROAST
                     </code>
                     ,
                     <code>
                      GSVA
                     </code>
                     ,
                     <code>
                      GLOBALTEST
                     </code>
                     ,
                     <code>
                      SAMGS
                     </code>
                     ). The paper also compares 10 frequently used enrichment tools implementing these methods (
                     <code>
                      DAVID
                     </code>
                     ,
                     <code>
                      ENRICHR
                     </code>
                     ,
                     <code>
                      CLUSTER-PROFILER
                     </code>
                     ,
                     <code>
                      GOSTATS
                     </code>
                     ,
                     <code>
                      WEBGESTALT
                     </code>
                     ,
                     <code>
                      G:PROFILER
                     </code>
                     ,
                     <code>
                      GENETRAIL
                     </code>
                     ,
                     <code>
                      GORILLA
                     </code>
                     ,
                     <code>
                      TOPPGENE
                     </code>
                     ,
                     <code>
                      PANTHER
                     </code>
                     ).
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      <code>
                       ORA
                      </code>
                      for the exploratory analysis of simple gene lists, pre-ranked
                      <code>
                       GSEA
                      </code>
                      or pre-ranked
                      <code>
                       CAMERA
                      </code>
                      for the analysis of pre-ranked gene lists accompanied by gene scores such as fold changes,
                      <br/>
                     </li>
                     <li>
                      For enrichment analysis on the full expression matrix (genes x samples), the paper recommends to provide normalized log2 intensities for microarray data and logTPMs (or logRPKMs/logFPKMs) for RNA-seq data; when given raw read counts, the paper recommends to apply a variance-stabilizing transformation such as
                      <code>
                       voom
                      </code>
                      to arrive at library-size normalized logCPMs,
                      <br/>
                     </li>
                     <li>
                      <code>
                       ROAST
                      </code>
                      (sample group comparisons) or
                      <code>
                       GSVA
                      </code>
                      (single sample) if the question of interest is to test for association of any gene in the set with the phenotype (self-contained null hypothesis),
                      <br/>
                     </li>
                     <li>
                      <code>
                       PADOG
                      </code>
                      (simple experimental designs) or
                      <code>
                       SAFE
                      </code>
                      (extended experimental designs) if the question of interest is to test for excess of differential expression in a gene set relative to genes outside the set (competitive null hypothesis).
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     <a href="http://bioconductor.org/packages/GSEABenchmarkeR" rel="nofollow">
                      http://bioconductor.org/packages/GSEABenchmarkeR
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Differential Splicing
                     </h3>
                     <a aria-label="Permalink: Differential Splicing" class="anchor" href="#differential-splicing" id="user-content-differential-splicing">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://humgenomics.biomedcentral.com/articles/10.1186/1479-7364-8-3" rel="nofollow">
                      A survey of software for genome-wide discovery of differential splicing in RNA-Seq data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Joan E Hooper
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Human Genomics, January 2014
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compares the methodologies, advantages, and disadvantages of eight differential splicing analysis tools, detailing use-cases and features for each.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Cuffdiff2
                     </code>
                     ,
                     <code>
                      MISO
                     </code>
                     ,
                     <code>
                      DEXSeq
                     </code>
                     ,
                     <code>
                      DSGseq
                     </code>
                     ,
                     <code>
                      MATS
                     </code>
                     ,
                     <code>
                      DiffSplice
                     </code>
                     ,
                     <code>
                      Splicing compass
                     </code>
                     ,
                     <code>
                      AltAnalyze
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     This is a true breakdown of each tools' advantages and disadvantages.
The author makes no recommendation due to the performance reliance on experimental setup, data type (e.g.
                     <code>
                      AltAnalyze
                     </code>
                     works best on junction + exon microarrays), and user objectives.
Table 1 provides a good comparison of the features and methodology of each method.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://academic.oup.com/bib/article/20/2/471/4524048" rel="nofollow">
                      A benchmarking of workflows for detecting differential splicing and differential expression at isoform level in human RNA-seq studies
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Gabriela A Merino
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Briefings in Bioinformatics, March 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compares nine most commonly used workflows to detect differential isoform expression and splicing.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      EBSeq
                     </code>
                     ,
                     <code>
                      DESeq2
                     </code>
                     ,
                     <code>
                      NOISeq
                     </code>
                     ,
                     <code>
                      Limma
                     </code>
                     ,
                     <code>
                      LimmaDS
                     </code>
                     ,
                     <code>
                      DEXSeq
                     </code>
                     ,
                     <code>
                      Cufflinks
                     </code>
                     ,
                     <code>
                      CufflinksDS
                     </code>
                     ,
                     <code>
                      SplicingCompass
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     DESeq2, Limma and NOISeq for differential isoform expression(DIE) analysis and DEXSeq and LimmaDS for differential splicing (DS) testing.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      <em>
                       de novo
                      </em>
                      Assembly and Quantification
                     </h3>
                     <a aria-label="Permalink: de novo Assembly and Quantification" class="anchor" href="#de-novo-assembly-and-quantification" id="user-content-de-novo-assembly-and-quantification">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1093/bioinformatics/btv488" rel="nofollow">
                      Benchmark analysis of algorithms for determining and quantifying full-length mRNA splice forms from RNA-seq data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Katharina E. Hayer et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Bioinformatics, Dec 2015
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared both guided and de novo transcript reconstruction algorithms using simulated and in vitro transcription (IVT) generated libraries. Precision/recall metrics were obtained by comparing the reconstructed transcripts to their true models.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Cufflinks
                     </code>
                     ,
                     <code>
                      CLASS
                     </code>
                     ,
                     <code>
                      FlipFlop
                     </code>
                     ,
                     <code>
                      IReckon
                     </code>
                     ,
                     <code>
                      IsoLasso
                     </code>
                     ,
                     <code>
                      MiTie
                     </code>
                     ,
                     <code>
                      StringTie
                     </code>
                     ,
                     <code>
                      StringTie-SR
                     </code>
                     ,
                     <code>
                      AUGUSTUS
                     </code>
                     ,
                     <code>
                      Trinity
                     </code>
                     ,
                     <code>
                      SOAP
                     </code>
                     ,
                     <code>
                      Trans-ABySS
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     All tools measured produced less than ideal precision-recall (both &lt;90%) when using imperfect simulated or IVT data and genes producing mulitple isoforms.
                     <a href="https://github.com/cole-trapnell-lab/cufflinks">
                      Cufflinks
                     </a>
                     and
                     <a href="https://github.com/gpertea/stringtie">
                      StringTie
                     </a>
                     are among the best performers.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://academic.oup.com/gigascience/article/8/5/giz039/5488105" rel="nofollow">
                      De novo transcriptome assembly: A comprehensive cross-species comparison of short-read RNA-Seq assemblers
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Martin Holzer &amp; Manja Marz
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     GigaScience, May 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compares 10
                     <em>
                      de novo
                     </em>
                     assembly tools across 9 RNA-seq datasets spanning multiple species and kingdoms for 20 biological-based and reference-free metrics.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Trinity
                     </code>
                     ,
                     <code>
                      Oases
                     </code>
                     ,
                     <code>
                      Trans-ABySS
                     </code>
                     ,
                     <code>
                      SOAPdenovo-Trans
                     </code>
                     ,
                     <code>
                      IDBA-Tran
                     </code>
                     ,
                     <code>
                      Bridger
                     </code>
                     ,
                     <code>
                      BinPacker
                     </code>
                     ,
                     <code>
                      Shannon
                     </code>
                     ,
                     <code>
                      SPAdes-sc
                     </code>
                     ,
                     <code>
                      SPAdes-rna
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The authors found that no tool's performance was dominant for all data sets, but
                     <a href="https://github.com/trinityrnaseq/trinityrnaseq">
                      Trinity
                     </a>
                     ,
                     <a href="https://cab.spbu.ru/software/spades/" rel="nofollow">
                      SPAdes
                     </a>
                     , and
                     <a href="https://github.com/bcgsc/transabyss">
                      Trans-ABySS
                     </a>
                     were typically among the best. For assembly evaluation, the authors recommend a hybrid approach combining both biological-based (BUSCO, # of full length transcripts) and reference-free metric (e.g.
                     <code>
                      TransRate
                     </code>
                     ,
                     <code>
                      DETONATE
                     </code>
                     ).
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors provide a comprehensive
                     <a href="https://www.rna.uni-jena.de/supplements/assembly/index.html" rel="nofollow">
                      electronic supplement website
                     </a>
                     containing all metrics and assembly commands in addition to many supplementary figures.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Cell-Type Deconvolution
                     </h3>
                     <a aria-label="Permalink: Cell-Type Deconvolution" class="anchor" href="#cell-type-deconvolution" id="user-content-cell-type-deconvolution">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://academic.oup.com/bioinformatics/article/35/14/i436/5529146" rel="nofollow">
                      Comprehensive evaluation of transcriptome-based cell-type quantification methods for immuno-oncology
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Markus List*, Tatsiana Aneichyk*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Bioinformatics, July 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper benchmarks and compares seven methods for computational deconvolution of cell-type abundance in bulk RNA-seq samples. Each method was tested on both simulated and true bulk RNA-seq samples validated by FACS.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      quanTIseq
                     </code>
                     ,
                     <code>
                      TIMER
                     </code>
                     ,
                     <code>
                      CIBERSORT
                     </code>
                     ,
                     <code>
                      CIBERSORT abs. mode
                     </code>
                     ,
                     <code>
                      MCPCounter
                     </code>
                     ,
                     <code>
                      xCell
                     </code>
                     ,
                     <code>
                      EPIC
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Varies. In general, the authors recommend
                     <a href="https://github.com/GfellerLab/EPIC">
                      EPIC
                     </a>
                     and
                     <a href="http://icbi.at/software/quantiseq/doc/index.html" rel="nofollow">
                      quanTIseq
                     </a>
                     due to their overall robustness and absolute (rather than relative) scoring, though
                     <a href="https://comphealth.ucsf.edu/app/xcell" rel="nofollow">
                      xCell
                     </a>
                     is recommended for binary presence/absence of cell types and
                     <a href="https://github.com/ebecht/MCPcounter">
                      MCPcounter
                     </a>
                     was their recommended relative scoring method.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors created an
                     <a href="https://github.com/icbi-lab/immunedeconv">
                      R package called immunedeconv
                     </a>
                     for easy installation and use of all these methods. For developers, they have made their
                     <a href="https://github.com/icbi-lab/immune_deconvolution_benchmark">
                      benchmarking pipeline
                     </a>
                     available so that others can reproduce/extend it to test their own tools/methods.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.biorxiv.org/content/10.1101/2020.01.10.897116v1.full" rel="nofollow">
                      Comprehensive benchmarking of computational deconvolution of transcriptomics data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Francisco Avila Cobos, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     bioRxiv, January 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared the effects of transformation, scaling/normalization, marker selection, cell type composition, and deconvolution methods on computing cell type proportions in mixed bulk RNA-seq samples. Performance is assessed by means of Pearson correlation and root-mean-square-error (RMSE) between the cell type proportions computed by the different deconvolution methods and known compositions of 1000 pseudo-bulk mixtures from 4 different single cell RNA-seq datasets with varying numbers of cells.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     Transformation methods:
                     <code>
                      linear (none)
                     </code>
                     ,
                     <code>
                      log
                     </code>
                     ,
                     <code>
                      VST (DESeq2)
                     </code>
                     ,
                     <code>
                      sqrt
                     </code>
                     . Scaling/normalization methods (bulk):
                     <code>
                      column-wise
                     </code>
                     ,
                     <code>
                      min-max
                     </code>
                     ,
                     <code>
                      z-score
                     </code>
                     ,
                     <code>
                      QN
                     </code>
                     ,
                     <code>
                      UQ
                     </code>
                     ,
                     <code>
                      row-wise
                     </code>
                     ,
                     <code>
                      global min-max
                     </code>
                     ,
                     <code>
                      global z-score
                     </code>
                     ,
                     <code>
                      TPM
                     </code>
                     ,
                     <code>
                      TMM
                     </code>
                     ,
                     <code>
                      median ratios
                     </code>
                     ,
                     <code>
                      LogNormalize
                     </code>
                     . Scaling/normalization methods (single cell):
                     <code>
                      RNBR
                     </code>
                     ,
                     <code>
                      scran
                     </code>
                     ,
                     <code>
                      scater
                     </code>
                     ,
                     <code>
                      Linnorm
                     </code>
                     . Deconvolution methods (bulk):
                     <code>
                      OLS
                     </code>
                     ,
                     <code>
                      NNLS
                     </code>
                     ,
                     <code>
                      FARDEEP
                     </code>
                     ,
                     <code>
                      RLR
                     </code>
                     ,
                     <code>
                      lasso
                     </code>
                     ,
                     <code>
                      ridge
                     </code>
                     ,
                     <code>
                      DCQ
                     </code>
                     ,
                     <code>
                      elastic net
                     </code>
                     ,
                     <code>
                      DSA
                     </code>
                     ,
                     <code>
                      EPIC
                     </code>
                     ,
                     <code>
                      CIBERSORT
                     </code>
                     ,
                     <code>
                      dtangle
                     </code>
                     ,
                     <code>
                      ssFrobenius
                     </code>
                     ,
                     <code>
                      ssKL
                     </code>
                     ,
                     <code>
                      DeconRNASeq
                     </code>
                     . Deconvolution methods (scRNA-seq reference):
                     <code>
                      BisqueRNA
                     </code>
                     ,
                     <code>
                      deconvSeq
                     </code>
                     ,
                     <code>
                      DWLS
                     </code>
                     ,
                     <code>
                      MuSiC
                     </code>
                     ,
                     <code>
                      SCDC
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The authors strongly recommend keeping data in the linear scale for deconvolution, avoiding the use of
                     <code>
                      column min-max
                     </code>
                     ,
                     <code>
                      column z-score
                     </code>
                     , and
                     <code>
                      QN
                     </code>
                     for normalization/scaling of bulk RNA-seq data, avoiding the use of
                     <code>
                      row-normalization
                     </code>
                     ,
                     <code>
                      column min-max
                     </code>
                     , and
                     <code>
                      TPM
                     </code>
                     for normalization/scaling of single cell RNA-seq data, use all possible cell markers, ensure that all possible cell types are represented in the reference matrix, and use one of the top performing deconvolution methods -
                     <code>
                      OLS
                     </code>
                     ,
                     <code>
                      nnls
                     </code>
                     ,
                     <code>
                      RLR
                     </code>
                     ,
                     <code>
                      FARDEEP
                     </code>
                     ,
                     <code>
                      CIBERSORT
                     </code>
                     ,
                     <code>
                      DWLS
                     </code>
                     ,
                     <code>
                      MuSiC
                     </code>
                     , or
                     <code>
                      SCDC
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors provide their benchmarking code
                     <a href="https://github.com/favilaco/deconv_benchmark">
                      on Github
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      CRISPR Screens
                     </h2>
                     <a aria-label="Permalink: CRISPR Screens" class="anchor" href="#crispr-screens" id="user-content-crispr-screens">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Hit/Dependency Identification
                     </h3>
                     <a aria-label="Permalink: Hit/Dependency Identification" class="anchor" href="#hitdependency-identification" id="user-content-hitdependency-identification">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1186/s13059-020-01972-x" rel="nofollow">
                      A benchmark of algorithms for the analysis of pooled CRISPR screens
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Sunil Bodapati*, Timothy P. Daley*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, March 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study evaluates and compares various algorithms used for analyzing data from pooled CRISPR screens, using a comprehensive simulation framework and real datasets. The algorithms were benchmarked for their ability to accurately identify essential genes in CRISPR knockout (CRISPRko), CRISPR interference (CRISPRi), and CRISPR activation (CRISPRa) screens. Key parameters such as the number of guides per gene, guide binding efficiency, sequencing depth, and control guides were varied to assess the robustness of these algorithms under different conditions.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     The study compared several algorithms, including Redundant siRNA Activity (RSA), MAGeCK Robust Ranking Algorithm (RRA), HiTSelect, MAGeCK Maximum Likelihood Estimation (MLE), CRISPhieRmix, CERES, JACKS, and a standard t test. BAGEL was also discussed, but not included in the testing due to its requirement of a priori knowledge.
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                    </p>
                    <ol dir="auto">
                     <li>
                      MAGeCK RRA is recommended for general use due to its robustness and consistent performance across various conditions.
                     </li>
                     <li>
                      CRISPhieRmix is suggested for screens with high variable guide efficiency, such as in CRISPRi or CRISPRa screens.
                     </li>
                     <li>
                      For multiple screens across different cell types or lines, MAGeCK MLE, JACKS, and CERES are good options.
                     </li>
                     <li>
                      The study also highlights that a simple t test can be effective when suitable control guides are used.
                     </li>
                     <li>
                      The number of guides per gene is crucial, with four being the minimum recommended for reliable results. More guides may be necessary for low-signal phenotypes.
                     </li>
                     <li>
                      Sequencing depth is less critical than previously thought, with the performance of most algorithms plateauing at 25 reads per guide.
                     </li>
                    </ol>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The simulation framework and scripts used in the study are available at
                     <a href="https://github.com/sbodapati/CRISPR_Benchmarking_Algorithms">
                      GitHub - CRISPR Benchmarking Algorithms
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      DNA Methylation
                     </h2>
                     <a aria-label="Permalink: DNA Methylation" class="anchor" href="#dna-methylation" id="user-content-dna-methylation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Platforms and Library Prep Methods
                     </h3>
                     <a aria-label="Permalink: Platforms and Library Prep Methods" class="anchor" href="#platforms-and-library-prep-methods" id="user-content-platforms-and-library-prep-methods">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/s41598-019-46875-5" rel="nofollow">
                      Systematic evaluation of library preparation methods and sequencing platforms for high-throughput whole genome bisulfite sequencing
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Li Zhou, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Scientific Reports, 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study focuses on improving strategies for whole genome bisulfite sequencing (WGBS) used in large-scale epidemiological studies. It systematically compares three WGBS library preparation methods (Swift Biosciences Accel-NGS, Illumina TruSeq, and QIAGEN QIAseq) across two Illumina sequencing platforms (NovaSeq and HiSeq X). The study also examines the concordance between WGBS and methylation array data. The study assayed quality metrics (Q20 and Q30 fractions), insert size, adaptor contamination, overlapping bases, read duplication rate, alignment rate, coverage depth, and genome coverage uniformity across the platforms.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Library Preparation Methods: Swift Biosciences Accel-NGS, Illumina TruSeq, QIAGEN QIAseq.
                     </li>
                     <li>
                      Sequencing Platforms: Illumina NovaSeq, HiSeq X.
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Swift Biosciences Accel-NGS achieved the highest proportion of CpG sites assayed and effective coverage, making it the most recommended method.
                     </li>
                     <li>
                      Illumina TruSeq had a high proportion of PCR duplicates, while QIAGEN QIAseq generally underperformed across all quality metrics.
                     </li>
                     <li>
                      NovaSeq and HiSeq X platforms showed similar performance, except for a higher read duplication rate in NovaSeq.
                     </li>
                     <li>
                      WGBS was less precise than methylation arrays, requiring a minimum coverage of 100x for comparable precision.
                     </li>
                     <li>
                      Swift outperformed other methods in terms of uniform genome coverage and CpG site coverage at lower depths.
                     </li>
                     <li>
                      No significant differences in nucleotide amplification bias were observed between NovaSeq and HiSeq X.
                     </li>
                     <li>
                      For quantifying DNA methylation, WGBS with Swift on HiSeq X outperformed methylation arrays in CpG coverage.
                     </li>
                    </ul>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1038/s41587-022-01336-9" rel="nofollow">
                      Comparison and imputation-aided integration of five commercial platforms for targeted DNA methylome analysis
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Miljana TaniÄ, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Biotechnology, October 2022.
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study benchmarks five commercial Targeted Bisulfite Sequencing (TBS) platforms for analyzing human DNA methylomes at base-pair resolution. The platforms include three hybridization capture-based (Agilent, Roche, and Illumina) and two reduced-representation-based (Diagenode and NuGen). Eleven sample types were analyzed, including cell lines and DNA methylation standards. Two samples were also compared with whole-genome DNA methylation sequencing using Illumina and Oxford Nanopore platforms. Key assessment parameters included workflow complexity, on/off-target performance, coverage accuracy, and reproducibility.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Hybridization capture-based platforms: Agilent SureSelect Methyl-Seq, Roche NimbleGen SeqCap EpiGiant, and Illumina TruSeq Methyl Capture EPIC.
                     </li>
                     <li>
                      Reduced-representation-based platforms: Diagenode Premium RRBS and NuGen Ovation RRBS Methyl-Seq.
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Each platform exhibited strengths and limitations in coverage, reproducibility, and concordance of DNA methylation levels.
                     </li>
                     <li>
                      RRBS platforms (Diagenode and NuGen) showed more uniform CpG coverage compared to hybridization capture-based methods (Agilent, Roche, Illumina).
                     </li>
                     <li>
                      Illuminaâs platform had the highest on-target capture efficiency (~90.6%), followed by Agilent's (~78.2%) and Roche's (~61.5%).
                     </li>
                     <li>
                      Differences in CpG coverage between platforms indicated varying suitability for specific genomic features. For instance, RRBS platforms had higher coverage in CpG islands and shores, while Roche excelled in covering open-sea regions and Illumina in enhancer regions.
                     </li>
                     <li>
                      High intra-platform and inter-platform correlation in DNA methylation levels were observed, except for Diagenode, which slightly underperformed.
                     </li>
                     <li>
                      All TBS platforms showed a strong correlation to whole-genome bisulfite sequencing (WGBS) and Oxford Nanopore data, suggesting their reliability for DNA methylome analysis.
                     </li>
                     <li>
                      The study suggests considering specific genomic feature coverage and desired analysis depth when choosing a TBS platform.
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors provide their analysis code on
                     <a href="https://github.com/ucl-medical-genomics/EpiCapture">
                      Github
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      CpG Methylation from Nanopore Data
                     </h3>
                     <a aria-label="Permalink: CpG Methylation from Nanopore Data" class="anchor" href="#cpg-methylation-from-nanopore-data" id="user-content-cpg-methylation-from-nanopore-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1038/s41467-021-23778-6" rel="nofollow">
                      Systematic benchmarking of tools for CpG methylation detection from nanopore sequencing
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Zaka Wing-Sze Yuen, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Communications, 2021
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study systematically benchmarks six tools for detecting 5-methylcytosine (5mC) from nanopore sequencing using individual reads, controlled methylation mixtures, Cas9-targeted sequencing, and whole-genome bisulfite sequencing (WGBS). The research highlights a trade-off between true positives and false positives among these tools and a general high dispersion in predicting methylation frequencies. Metrics include accuracy (true positive rate &amp; true negative rate) at the individual read level and per controlled mixture. The authors also tested a consensus approach combining the results of pairs of callers.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Nanopolish
                     </code>
                     ,
                     <code>
                      Megalodon
                     </code>
                     ,
                     <code>
                      DeepSignal
                     </code>
                     ,
                     <code>
                      Guppy
                     </code>
                     ,
                     <code>
                      Tombo
                     </code>
                     , and
                     <code>
                      DeepMod
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      No single method accurately predicts across all methylation frequency ranges.
                     </li>
                     <li>
                      Guppy excels in identifying unmethylated sites but fails at fully methylated sites.
                     </li>
                     <li>
                      Nanopolish and Tombo accurately recover fully methylated sites but have high false positives at unmethylated sites.
                     </li>
                     <li>
                      Megalodon showed the best overall performance but requires GPU support.
                     </li>
                     <li>
                      The consensus approach METEORE, combining predictions from two or more tools (specifically Megalodon and DeepSignal), improved accuracy over individual methods. It balances accuracy and running times, although it also requires a GPU for efficiency.
                     </li>
                     <li>
                      On a CPU, the combination of Nanopolish and DeepSignal can match Megalodon's accuracy and be time-competitive.
                     </li>
                     <li>
                      Reassessing score cutoffs for individual reads and removing sites with uncertain methylation status can further enhance accuracy.
                     </li>
                     <li>
                      These methods showed good consistency with WGBS data, suggesting potential for sensitive diagnostic and forensic tests without high coverage.
                     </li>
                     <li>
                      The study also notes that the highest discrepancy with WGBS occurred at CG sites in AT-rich sequences, particularly for DeepMod and DeepSignal.
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors provide their method for consensus calling (METEORE) on
                     <a href="https://github.com/comprna/METEORE">
                      GitHub
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Variant Callers
                     </h2>
                     <a aria-label="Permalink: Variant Callers" class="anchor" href="#variant-callers" id="user-content-variant-callers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Germline SNP/Indel Callers
                     </h3>
                     <a aria-label="Permalink: Germline SNP/Indel Callers" class="anchor" href="#germline-snpindel-callers" id="user-content-germline-snpindel-callers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1093/bib/bbaa148" rel="nofollow">
                      Benchmarking variant callers in next-generation and third-generation sequencing analysis
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Surui Pei, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Briefings in Bioinformatics, July 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared evaluated 11 modes among 6 variant callers on 12 NGS and TGS datasets on germline and somatic variant calling.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Sentieon
                     </code>
                     (
                     <code>
                      TNscope
                     </code>
                     ,
                     <code>
                      TNseq
                     </code>
                     ,
                     <code>
                      DNAseq
                     </code>
                     ),
                     <code>
                      DeepVariant
                     </code>
                     (
                     <code>
                      WGS
                     </code>
                     ),
                     <code>
                      GATK
                     </code>
                     (
                     <code>
                      HC
                     </code>
                     &amp;
                     <code>
                      MuTect2
                     </code>
                     ),
                     <code>
                      NeuSomatic
                     </code>
                     ,
                     <code>
                      VarScan2
                     </code>
                     ,
                     <code>
                      Strelka2
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     All the four germline callers had comparable performance on NGS data. For TGS data, all the three callers had similar performance in SNP calling, while
                     <a href="https://github.com/google/deepvariant">
                      DeepVariant
                     </a>
                     outperformed the others in InDel calling. For somatic variant calling on NGS, Sentieon TNscope and
                     <a href="https://github.com/broadinstitute/gatk">
                      GATK Mutect2
                     </a>
                     outperformed the other callers.
                    </p>
                    <p dir="auto">
                     --
                    </p>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6597787/" rel="nofollow">
                      Systematic comparison of germline variant calling pipelines cross multiple next-generation sequencers
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Jiayun Chen, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Scientific Reports, June 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared three variant callers for WGS and WES samples from NA12878 across five next-gen sequencing platforms
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      GATK
                     </code>
                     ,
                     <code>
                      Strelka2
                     </code>
                     ,
                     <code>
                      Samtools-Varscan2
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Though all methods tested generally scored well,
                     <a href="https://github.com/Illumina/strelka">
                      Strelka2
                     </a>
                     had the highest F-scores for both SNP and indel calling in addition to being the most computationally performant.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6294778/" rel="nofollow">
                      Comparison of three variant callers for human whole genome sequencing
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Anna Supernat, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Scientific Reports, December 2018
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     The paper compared three variant callers for WGS samples from NA12878 at 10x, 15x, and 30x coverage.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      DeepVariant
                     </code>
                     ,
                     <code>
                      GATK
                     </code>
                     ,
                     <code>
                      SpeedSeq
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     All methods had similar F-scores, precision, and recall for SNP calling, but
                     <a href="https://github.com/google/deepvariant">
                      DeepVariant
                     </a>
                     scored higher across all metrics for indels at all coverages.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4619817/" rel="nofollow">
                      A Comparison of Variant Calling Pipelines Using Genome in a Bottle as a Reference
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Adam Cornish, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     BioMed Research International, October 2015
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared 30 variant calling pipelines composed of six different variant callers and five different aligners on NA12878 WES data from the "Genome in a Bottle" consortium.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      Variant callers:
                      <code>
                       FreeBayes
                      </code>
                      ,
                      <code>
                       GATK-HaplotypeCaller
                      </code>
                      ,
                      <code>
                       GATK-UnifiedGenotyper
                      </code>
                      ,
                      <code>
                       SAMtools mpileup
                      </code>
                      ,
                      <code>
                       SNPSVM
                      </code>
                     </li>
                     <li>
                      Aligners:
                      <code>
                       bowtie2
                      </code>
                      ,
                      <code>
                       BWA-mem
                      </code>
                      ,
                      <code>
                       BWA-sampe
                      </code>
                      ,
                      <code>
                       CUSHAW3
                      </code>
                      ,
                      <code>
                       MOSAIK
                      </code>
                      ,
                      <code>
                       Novoalign
                      </code>
                      .
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     <a href="http://www.novocraft.com/products/novoalign/" rel="nofollow">
                      Novoalign
                     </a>
                     with
                     <a href="https://github.com/broadinstitute/gatk">
                      GATK-UnifiedGenotyper
                     </a>
                     exhibited the highest sensitivity while producing few false positives.
In general,
                     <a href="https://github.com/lh3/bwa">
                      BWA-mem
                     </a>
                     was the most consistent aligner, and
                     <code>
                      GATK-UnifiedGenotyper
                     </code>
                     performed well across the top aligners (BWA, bowtie2, and Novoalign).
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Somatic SNV/Indel callers
                     </h3>
                     <a aria-label="Permalink: Somatic SNV/Indel callers" class="anchor" href="#somatic-snvindel-callers" id="user-content-somatic-snvindel-callers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0151664" rel="nofollow">
                      Evaluation of Nine Somatic Variant Callers for Detection of Somatic Mutations in Exome and Targeted Deep Sequencing Data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Anne Bruun KrÃ¸igÃ¥rd, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     PLoS ONE, March 2016
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper performed comparisons between nine somatic variant callers on five paired tumor-normal samples from breast cancer patients subjected to WES and targeted deep sequencing.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      EBCall
                     </code>
                     ,
                     <code>
                      Mutect
                     </code>
                     ,
                     <code>
                      Seurat
                     </code>
                     ,
                     <code>
                      Shimmer
                     </code>
                     ,
                     <code>
                      Indelocator
                     </code>
                     ,
                     <code>
                      SomaticSniper
                     </code>
                     ,
                     <code>
                      Strelka
                     </code>
                     ,
                     <code>
                      VarScan2
                     </code>
                     ,
                     <code>
                      Virmid
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     <a href="https://github.com/friend1ws/EBCall">
                      EBCall
                     </a>
                     ,
                     <a href="https://github.com/broadinstitute/mutect">
                      Mutect
                     </a>
                     ,
                     <a href="https://sourceforge.net/p/virmid/wiki/Home/" rel="nofollow">
                      Virmid
                     </a>
                     , and
                     <a href="https://github.com/Illumina/strelka">
                      Strelka
                     </a>
                     (now Strelka2) were most reliable for both WES and targeted deep sequencing.
                     <a href="https://github.com/friend1ws/EBCall">
                      EBCall
                     </a>
                     was superior for indel calling due to high sensitivity and robustness to changes in sequencing depths.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3986649/" rel="nofollow">
                      Comparison of somatic mutation calling methods in amplicon and whole exome sequence data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Huilei Xu, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     BMC Genomics, March 2014
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     Using the "Genome in a Bottle" gold standard variant set, this paper compared five somatic mutation calling methods on matched tumor-normal amplicon and WES data.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      GATK-UnifiedGenotyper followed by subtraction
                     </code>
                     ,
                     <code>
                      MuTect
                     </code>
                     ,
                     <code>
                      Strelka
                     </code>
                     ,
                     <code>
                      SomaticSniper
                     </code>
                     ,
                     <code>
                      VarScan2
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     <a href="https://github.com/broadinstitute/mutect">
                      MuTect
                     </a>
                     and
                     <a href="https://github.com/Illumina/strelka">
                      Strelka
                     </a>
                     (now Strelka2) had the highest sensitivity, particularly at low frequency alleles, in addition to the highest specificity.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      CNV Callers
                     </h3>
                     <a aria-label="Permalink: CNV Callers" class="anchor" href="#cnv-callers" id="user-content-cnv-callers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.biorxiv.org/content/10.1101/850958v1" rel="nofollow">
                      Benchmark of tools for CNV detection from NGS panel data in a genetic diagnostics context
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     JosÃ© Marcos Moreno-Cabrera, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     bioRxiv, November 2019.
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared five germline copy number variation callers against four genetic diagnostics datasets (495 samples, 231 CNVs validated by MLPA) using both default and optimized parameters.
Sensitivity, specificity, positive predictive value, negative predictive value, F1 score, and various correlation coefficients were used as benchmarking metrics.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      DECoN
                     </code>
                     ,
                     <code>
                      CoNVaDING
                     </code>
                     ,
                     <code>
                      panelcn.MOPS
                     </code>
                     ,
                     <code>
                      ExomeDepth
                     </code>
                     ,
                     <code>
                      CODEX2
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Most tools performed well, but varied based on datasets.
The authors felt
                     <a href="https://github.com/RahmanTeam/DECoN">
                      DECoN
                     </a>
                     and
                     <a href="https://bioconductor.org/packages/release/bioc/html/panelcn.mops.html" rel="nofollow">
                      panelcn.MOPS
                     </a>
                     with optimized parameters were sensitive enough to be used as screening methods in genetic dianostics.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors have made their benchmarking code (
                     <a href="https://github.com/TranslationalBioinformaticsIGTP/CNVbenchmarkeR">
                      CNVbenchmarkeR
                     </a>
                     ) available, which can be run to determine optimal parameters for each algorithm for a given user's data.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5452530/" rel="nofollow">
                      An evaluation of copy number variation detection tools for cancer using whole exome sequencing data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Fatima Zare, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     BMC Bioinformatics, May 2017
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared six copy number variation callers on ten TCGA breast cancer tumor-normal pair WES datasets in addition to simulated datasets from VarSimLab.
Sensitivity, specificity, and false-discovery rate were used as the benchmarking metrics.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      ADTEx
                     </code>
                     ,
                     <code>
                      CONTRA
                     </code>
                     ,
                     <code>
                      cn.MOPS
                     </code>
                     ,
                     <code>
                      ExomeCNV
                     </code>
                     ,
                     <code>
                      VarScan2
                     </code>
                     ,
                     <code>
                      CoNVEX
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     All tools suffered from high FDRs (~30-60%), but
                     <a href="https://github.com/cran/ExomeCNV">
                      ExomeCNV
                     </a>
                     (a now defunct R package) had the highest overall sensitivity.
                     <a href="http://dkoboldt.github.io/varscan/" rel="nofollow">
                      VarScan2
                     </a>
                     had moderate sensitivity and specificity for both amplifications and deletions.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      SV callers
                     </h3>
                     <a aria-label="Permalink: SV callers" class="anchor" href="#sv-callers" id="user-content-sv-callers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/s41467-019-11146-4" rel="nofollow">
                      Comprehensive evaluation and characterisation of short read general-purpose structural variant calling software
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Daniel L. Cameron, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Communications, July 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared 10 structural variant callers on four cell line WGS datasets (NA12878, HG002, CHM1, and CHM13) with orthogonal validation data.
Precision and recall were the benchmarking metrics used.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      BreakDancer
                     </code>
                     ,
                     <code>
                      cortex
                     </code>
                     ,
                     <code>
                      CREST
                     </code>
                     ,
                     <code>
                      DELLY
                     </code>
                     ,
                     <code>
                      GRIDSS
                     </code>
                     ,
                     <code>
                      Hydra
                     </code>
                     ,
                     <code>
                      LUMPY
                     </code>
                     ,
                     <code>
                      manta
                     </code>
                     ,
                     <code>
                      Pindel
                     </code>
                     ,
                     <code>
                      SOCRATES
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The authors found
                     <a href="https://github.com/PapenfussLab/gridss">
                      GRIDSS
                     </a>
                     and
                     <a href="https://github.com/Illumina/manta">
                      manta
                     </a>
                     consistently performed well, but also provide more general guidelines for both users and developers.
                    </p>
                    <ul dir="auto">
                     <li>
                      Use a caller that utilizes multiple sources of evidence and assembly.
                     </li>
                     <li>
                      Use a caller that can call all events you care about.
                     </li>
                     <li>
                      Ensemble calling is not a cure-all and generally don't outperform the best individual callers (at least on these datasets).
                     </li>
                     <li>
                      Do not use callers that rely only on paired-end data.
                     </li>
                     <li>
                      Calls with high read counts are typically artefacts.
                     </li>
                     <li>
                      Simulations aren't real - benchmarking solely on simulations is a bad idea.
                     </li>
                     <li>
                      Developers - be wary of incomplete trust sets and the potential for overfitting. Test tools on multiple datasets.
                     </li>
                     <li>
                      Developers - make your tool easy to use with basic sanity checks to protect against invalid inputs. Use standard file formats.
                     </li>
                     <li>
                      Developers - use all available evidence and produce meaningful quality scores.
                     </li>
                    </ul>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1720-5" rel="nofollow">
                      Comprehensive evaluation of structural variation detection algorithms for whole genome sequencing
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Shunichi Kosugi, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, June 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study compared 69 structural variation callers on simulated and real (NA12878, HG002, and HG00514) datasets.
F-scores, precision, and recall were the main benchmarking metrics.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      1-2-3-SV
                     </code>
                     ,
                     <code>
                      AS-GENESENG
                     </code>
                     ,
                     <code>
                      BASIL-ANISE,
                     </code>
                     <code>
                      BatVI
                     </code>
                     ,
                     <code>
                      BICseq2
                     </code>
                     ,
                     <code>
                      BreakDancer
                     </code>
                     ,
                     <code>
                      BreakSeek
                     </code>
                     ,
                     <code>
                      BreakSeq2
                     </code>
                     ,
                     <code>
                      Breakway
                     </code>
                     ,
                     <code>
                      CLEVER
                     </code>
                     ,
                     <code>
                      CNVnator
                     </code>
                     ,
                     <code>
                      Control-FREEC
                     </code>
                     ,
                     <code>
                      CREST
                     </code>
                     ,
                     <code>
                      DELLY
                     </code>
                     ,
                     <code>
                      DINUMT
                     </code>
                     ,
                     <code>
                      ERDS
                     </code>
                     ,
                     <code>
                      FermiKit
                     </code>
                     ,
                     <code>
                      forestSV
                     </code>
                     ,
                     <code>
                      GASVPro
                     </code>
                     ,
                     <code>
                      GenomeSTRiP
                     </code>
                     ,
                     <code>
                      GRIDSS
                     </code>
                     ,
                     <code>
                      HGT-ID
                     </code>
                     ,
                     <code>
                      Hydra-sv
                     </code>
                     ,
                     <code>
                      iCopyDAV
                     </code>
                     ,
                     <code>
                      inGAP-sv
                     </code>
                     ,
                     <code>
                      ITIS
                     </code>
                     ,
                     <code>
                      laSV
                     </code>
                     ,
                     <code>
                      Lumpy
                     </code>
                     ,
                     <code>
                      Manta
                     </code>
                     ,
                     <code>
                      MATCHCLIP
                     </code>
                     ,
                     <code>
                      Meerkat
                     </code>
                     ,
                     <code>
                      MELT
                     </code>
                     ,
                     <code>
                      MELT-numt
                     </code>
                     ,
                     <code>
                      MetaSV
                     </code>
                     ,
                     <code>
                      MindTheGap
                     </code>
                     ,
                     <code>
                      Mobster
                     </code>
                     ,
                     <code>
                      Mobster-numt
                     </code>
                     ,
                     <code>
                      Mobster-vei
                     </code>
                     ,
                     <code>
                      OncoSNP-SEQ
                     </code>
                     ,
                     <code>
                      Pamir
                     </code>
                     ,
                     <code>
                      PBHoney
                     </code>
                     ,
                     <code>
                      PBHoney-NGM
                     </code>
                     ,
                     <code>
                      pbsv
                     </code>
                     ,
                     <code>
                      PennCNV-Seq
                     </code>
                     ,
                     <code>
                      Pindel
                     </code>
                     ,
                     <code>
                      PopIns
                     </code>
                     ,
                     <code>
                      PRISM
                     </code>
                     ,
                     <code>
                      RAPTR
                     </code>
                     ,
                     <code>
                      readDepth
                     </code>
                     ,
                     <code>
                      RetroSeq
                     </code>
                     ,
                     <code>
                      Sniffles
                     </code>
                     ,
                     <code>
                      Socrates
                     </code>
                     ,
                     <code>
                      SoftSearch
                     </code>
                     ,
                     <code>
                      SoftSV
                     </code>
                     ,
                     <code>
                      SoloDel
                     </code>
                     ,
                     <code>
                      Sprites
                     </code>
                     ,
                     <code>
                      SvABA
                     </code>
                     ,
                     <code>
                      SVDetect
                     </code>
                     ,
                     <code>
                      Svelter
                     </code>
                     ,
                     <code>
                      SVfinder
                     </code>
                     ,
                     <code>
                      SVseq2
                     </code>
                     ,
                     <code>
                      Tangram
                     </code>
                     ,
                     <code>
                      Tangram-numt
                     </code>
                     ,
                     <code>
                      Tangram-vei
                     </code>
                     ,
                     <code>
                      Tea
                     </code>
                     ,
                     <code>
                      TEMP
                     </code>
                     ,
                     <code>
                      TIDDIT
                     </code>
                     ,
                     <code>
                      Ulysses
                     </code>
                     ,
                     <code>
                      VariationHunter
                     </code>
                     ,
                     <code>
                      VirusFinder
                     </code>
                     ,
                     <code>
                      VirusSeq
                     </code>
                     ,
                     <code>
                      Wham
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Varies greatly depending on type and size of the structural variant in addition to read length.
                     <code>
                      GRIDSS
                     </code>
                     ,
                     <code>
                      Lumpy
                     </code>
                     ,
                     <code>
                      SVseq2
                     </code>
                     ,
                     <code>
                      SoftSV
                     </code>
                     , and
                     <code>
                      Manta
                     </code>
                     performed well calling deletions of diverse sizes.
                     <code>
                      TIDDIT
                     </code>
                     ,
                     <code>
                      forestSV
                     </code>
                     ,
                     <code>
                      ERDS
                     </code>
                     , and
                     <code>
                      CNVnator
                     </code>
                     called large deletions well, while
                     <code>
                      pbsv
                     </code>
                     ,
                     <code>
                      Sniffles
                     </code>
                     , and
                     <code>
                      PBHoney
                     </code>
                     were the best performers for small deletions.
For duplications, good choices included
                     <code>
                      Wham
                     </code>
                     ,
                     <code>
                      SoftSV
                     </code>
                     ,
                     <code>
                      MATCHCLIP
                     </code>
                     , and
                     <code>
                      GRIDSS
                     </code>
                     , while
                     <code>
                      CNVnator
                     </code>
                     ,
                     <code>
                      ERDS
                     </code>
                     , and
                     <code>
                      iCopyDAV
                     </code>
                     excelled calling large duplications.
For insertions,
                     <code>
                      MELT
                     </code>
                     ,
                     <code>
                      Mobster
                     </code>
                     ,
                     <code>
                      inGAP-sv
                     </code>
                     , and methods using long read data were most effective.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1858-1" rel="nofollow">
                      Evaluating nanopore sequencing data processing pipelines for structural variation identification
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Anbo Zhou, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, November 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper evaluated four alignment tools and three SV detection tools on four nanopore datasets (both simulated and real).
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <em>
                      aligners
                     </em>
                     -
                     <code>
                      minimap2
                     </code>
                     ,
                     <code>
                      NGMLR
                     </code>
                     ,
                     <code>
                      GraphMap
                     </code>
                     ,
                     <code>
                      LAST
                     </code>
                     . SV Callers -
                     <code>
                      Sniffles
                     </code>
                     ,
                     <code>
                      NanoSV
                     </code>
                     ,
                     <code>
                      Picky
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The authors recommend using the
                     <a href="https://github.com/lh3/minimap2">
                      minimap2
                     </a>
                     aligner in combination with the SV caller
                     <a href="https://github.com/fritzsedlazeck/Sniffles">
                      Sniffles
                     </a>
                     because of their speed and relatively balanced performance.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors provide
                     <a href="https://github.com/JXing-Lab/nanopore-sv-evaluation">
                      all code used in the study
                     </a>
                     as well as a singularity package containing pre-installed programs and all seven pipeline.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Single Cell
                     </h2>
                     <a aria-label="Permalink: Single Cell" class="anchor" href="#single-cell" id="user-content-single-cell">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      scRNA Transformations/Normalization
                     </h3>
                     <a aria-label="Permalink: scRNA Transformations/Normalization" class="anchor" href="#scrna-transformationsnormalization" id="user-content-scrna-transformationsnormalization">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1038/s41592-023-01814-1" rel="nofollow">
                      Comparison of transformations for single-cell RNA-seq data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Constantin Ahlmann-Eltze &amp; Wolfgang Huber
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Methods, May 2023
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study evaluates 22 transformations for preprocessing single-cell RNA-sequencing data. These transformations are aimed at adjusting counts for variable sampling efficiency and making variance similar across the dynamic range. The research focuses on understanding cell types and states in terms of lower-dimensional mathematical structures, with benchmarks designed to assess the ability of transformations to recover latent cell structures.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                    </p>
                    <ol dir="auto">
                     <li>
                      <strong>
                       Delta method-based variance-stabilizing transformations:
                      </strong>
                      acosh transformation, shifted logarithm (with pseudo-count y0 = 1 or y0 = 1/4Î±), shifted logarithm with CPM, and additional variations including HVG selection, z scoring, and output rescaling.
                     </li>
                     <li>
                      <strong>
                       Residuals-based variance-stabilizing transformations:
                      </strong>
                      clipped and unclipped Pearson residuals (implemented by sctransform and transformGamPoi), randomized quantile residuals, clipped Pearson residuals with HVG selection, z scoring, and an analytical approximation to the Pearson residuals.
                     </li>
                     <li>
                      <strong>
                       Latent gene expression-based transformations (Lat Expr):
                      </strong>
                      Sanity Distance, Sanity MAP, Dino, Normalisr.
                     </li>
                     <li>
                      <strong>
                       Count-based factor analysis models (Count):
                      </strong>
                      GLM PCA, NewWave.
                     </li>
                     <li>
                      <strong>
                       Negative controls:
                      </strong>
                      raw untransformed counts (y) and raw counts scaled by the size factor (y/s).
                     </li>
                    </ol>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                    </p>
                    <ul dir="auto">
                     <li>
                      The shifted logarithm transformation (log(y/s + y0) with y0 = 1) followed by PCA performed well, often better than more sophisticated alternatives.
                     </li>
                     <li>
                      The study advises against using CPM as input for the shifted logarithm due to unrealistic large overdispersion assumptions.
                     </li>
                     <li>
                      Pearson residuals-based transformation has good theoretical properties and performed similarly to the shifted logarithm. However, it has limitations in handling genes with large dynamic range across cells.
                     </li>
                     <li>
                      Latent expression state transformations and count-based latent factor models did not outperform the shifted logarithm and were computationally expensive.
                     </li>
                     <li>
                      The delta method-based transformation produced more consistent results on the 10x datasets.
                     </li>
                     <li>
                      Overall, despite extensive research in preprocessing methods for single-cell RNA-seq data, the shifted logarithm still ranks among the best, underlining the utility of lower-dimensional embeddings of the transformed count matrix for noise reduction and fidelity increase.
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     An
                     <a href="https://shiny-portal.embl.de/shinyapps/app/08_single-cell_transformation_benchmark" rel="nofollow">
                      interactive website
                     </a>
                     with all results for all tested parameter combinations is provided.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Gene Signature Scoring
                     </h3>
                     <a aria-label="Permalink: Gene Signature Scoring" class="anchor" href="#gene-signature-scoring" id="user-content-gene-signature-scoring">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://elifesciences.org/articles/71994" rel="nofollow">
                      Signature-scoring methods developed for bulk samples are not adequate for cancer single-cell RNA sequencing data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Nighat Noureen, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     eLife, February 2022
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study benchmarks five signature-scoring methods in the context of cancer single-cell RNA sequencing data. The authors highlight that methods developed for bulk sample analysis, specifically single sample gene set enrichment analysis (ssGSEA) and Gene Set Variation Analysis (GSVA), show biases and inaccuracies when applied to single-cell data. This is attributed to the higher gene counts in cancer cells compared to normal cells, which affects the performance of ssGSEA and GSVA. The study emphasizes the importance of considering cellular context in signature scoring, especially the effect of dropouts in single-cell data.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      ssGSEA
                     </code>
                     ,
                     <code>
                      GSVA
                     </code>
                     ,
                     <code>
                      AUCell
                     </code>
                     ,
                     <code>
                      Single Cell Signature Explorer (SCSE)
                     </code>
                     ,
                     <code>
                      Jointly Assessing Signature Mean and Inferring Enrichment (JASMINE)
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The study recommends caution when using bulk-sample-based methods like ssGSEA and GSVA for single-cell RNA sequencing data due to their susceptibility to biases caused by high gene counts and dropouts in cancer cells. Single-cell-based methods, particularly JASMINE and SCSE, are more robust in this context. JASMINE, a new method developed in this study, showed particular effectiveness in accounting for dropouts and evaluating average expression levels of expressed signature genes. Typically, I'd avoid adding a paper in which the author's are touting their new tool, but I feel the contextual information therein is invaluable in this case.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The GitHub repository for JASMINE is available
                     <a href="https://github.com/NNoureen/JASMINE">
                      here
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      scRNA Sequencing Protocols
                     </h3>
                     <a aria-label="Permalink: scRNA Sequencing Protocols" class="anchor" href="#scrna-sequencing-protocols" id="user-content-scrna-sequencing-protocols">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/s41587-020-0469-4" rel="nofollow">
                      Benchmarking single-cell RNA-sequencing protocols for cell atlas projects
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Elisabetta Mereu*, Atefeh Lafzi*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Biotechnology, April 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper evaluated 13 single cell/nuclei RNA-seq protocols to evaluate their aptitude for use in cell atlas-like projects. Using a single cell, multi-species mixture, the authors measured each protocol's ability to capture cell markers, gene detection power, clusterability (with and without integration with other protocols), mappability, and mixability.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Quartz-seq2
                     </code>
                     ,
                     <code>
                      Chromium
                     </code>
                     ,
                     <code>
                      Smart-seq2
                     </code>
                     ,
                     <code>
                      CEL-seq2
                     </code>
                     ,
                     <code>
                      C1HT-medium
                     </code>
                     ,
                     <code>
                      C1HT-small
                     </code>
                     ,
                     <code>
                      ddSEQ
                     </code>
                     ,
                     <code>
                      Chromium (single nuclei)
                     </code>
                     ,
                     <code>
                      Drop-seq
                     </code>
                     ,
                     <code>
                      inDrop
                     </code>
                     ,
                     <code>
                      ICELL8
                     </code>
                     ,
                     <code>
                      MARS-seq
                     </code>
                     ,
                     <code>
                      gmcSCRB-seq
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     See figure 6 for a summary of benchmarking results for each method.
                     <code>
                      Quartz-seq2
                     </code>
                     was the overall best performing, yielding superior results for gene detection and marker expression over other methods, though
                     <code>
                      Chromium
                     </code>
                     ,
                     <code>
                      Smart-seq2
                     </code>
                     , and
                     <code>
                      CEL-seq2
                     </code>
                     were also strong performers.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors provide benchmarking code and analysis code in two different Github repositories -
                     <a href="https://github.com/ati-lz/HCA_Benchmarking">
                      here
                     </a>
                     and
                     <a href="https://github.com/elimereu/matchSCore2">
                      here
                     </a>
                     .
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/s41587-020-0465-8" rel="nofollow">
                      Systematic comparison of single-cell and single-nucleus RNA-sequencing methods
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Jiarui Ding, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Biotechnology, April 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study evaluated seven methods for single-cell and/or single-nucleus RNA-sequencing on three types of samples: cell lines, PBMCs, and brain tissue. Evaluation metrics included the structure and alignment of reads, number of multiplets and detection sensitivity, and ability to recover known biological information.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Smart-seq2
                     </code>
                     ,
                     <code>
                      CEL-Seq2
                     </code>
                     ,
                     <code>
                      3' 10X Chromium
                     </code>
                     ,
                     <code>
                      Drop-Seq
                     </code>
                     ,
                     <code>
                      Seq-Well
                     </code>
                     ,
                     <code>
                      inDrops
                     </code>
                     ,
                     <code>
                      sci-RNA-seq
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Overall, the authors found
                     <code>
                      3' 10X Chromium
                     </code>
                     to have the strongest consistent performance among the high-throughput methods, yielding the highest sensitivity, though it did not perform any better for cell type classification. When greater sensitivity is required, the authors recommend
                     <code>
                      Smart-seq2
                     </code>
                     or
                     <code>
                      CEL-Seq2
                     </code>
                     , which both performed similarly. Supplementary table 7 includes an overview of each method's relative merits.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors made their unified analysis pipeline (scumi) available as a
                     <a href="https://bitbucket.org/jerry00/scumi-dev/src/master/" rel="nofollow">
                      python package
                     </a>
                     , the repo of which also includes their R scripts used for cell filtering and cell type assignment.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      scRNA Analysis Pipelines
                     </h3>
                     <a aria-label="Permalink: scRNA Analysis Pipelines" class="anchor" href="#scrna-analysis-pipelines" id="user-content-scrna-analysis-pipelines">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1093/nargab/lqaa052" rel="nofollow">
                      Comparison of visualization tools for single-cell RNAseq data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Batuhan Cakir, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     NAR Genomics and Bioinformatics, September 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study evaluated 13 scRNA-seq visualization platforms based on their features, performance, cloud and web support, containerization, and inter-operability between analysis platforms and data formats (loom, h5ad, SingleCellExperiment, Seurat, raw txt/csv.) with varying numbers of cells (5k to 2 million).
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      ASAP
                     </code>
                     ,
                     <code>
                      Bbrowser
                     </code>
                     ,
                     <code>
                      cellxgene
                     </code>
                     ,
                     <code>
                      Granatum
                     </code>
                     ,
                     <code>
                      iSEE
                     </code>
                     ,
                     <code>
                      Loom viewer
                     </code>
                     ,
                     <code>
                      Loupe Cell Browser
                     </code>
                     ,
                     <code>
                      SCope
                     </code>
                     ,
                     <code>
                      scSVA
                     </code>
                     ,
                     <code>
                      scVI
                     </code>
                     ,
                     <code>
                      Single Cell Explorer
                     </code>
                     ,
                     <code>
                      SPRING
                     </code>
                     , and
                     <code>
                      UCSC Cell Browser
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     <strong>
                      Table 1
                     </strong>
                     provides an overview of the tool capabilities and current support. The authors further compared tools that could be used for web sharing -
                     <code>
                      cellxgene
                     </code>
                     ,
                     <code>
                      iSEE
                     </code>
                     with SCE files,
                     <code>
                      iSEE
                     </code>
                     with loom files,
                     <code>
                      loom-viewer
                     </code>
                     ,
                     <code>
                      SCope
                     </code>
                     ,
                     <code>
                      Single Cell Explorer
                     </code>
                     ,
                     <code>
                      scSVA
                     </code>
                     , and
                     <code>
                      UCSC Cell Browser
                     </code>
                     for preprocessing memory and time requirements with varying numbers of cells.
                     <code>
                      iSEE-loom
                     </code>
                     ,
                     <code>
                      SCope
                     </code>
                     ,
                     <code>
                      scSVA
                     </code>
                     and
                     <code>
                      loom-viewer
                     </code>
                     all enable efficient integration with the hierarchical data format (HDF5) from which loom and h5ad formats are derived, and as such, have the lowest preprocessing time requirements.
                     <code>
                      iSEE-SCE
                     </code>
                     performed poorly with large numbers of cells (&gt;50k) with the default number of panels (8) but performed better with a more limited set of visualizations. Some of the tools had unexplainable resource spikes with high numbers of cells, e.g. loom-viewer dramatically increasing in both time and memory requirements with 2M cells versus 1.5M.
                     <code>
                      cellxgene
                     </code>
                     was generally recommended due to its ease of use, expansive community support and active maintenance, and ability to handle large datasets.
                     <code>
                      Single Cell Explorer
                     </code>
                     and
                     <code>
                      UCSC Cell Browser
                     </code>
                     both were among the poorest performers resource-wise, with their usage increasing linearly with cell number.
                     <code>
                      iSEE
                     </code>
                     was generally commended for its flexibility and support of custom visualizations.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors made their package (sceasy) to convert between Seurat, SCE, Loom, and AnnData objects available via
                     <a href="https://github.com/cellgeni/sceasy">
                      GitHub
                     </a>
                     .
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1093/bib/bbaa116" rel="nofollow">
                      Comparison of high-throughput single-cell RNA sequencing data processing pipelines
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Mingxuan Gao, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Briefings in Bioinformatics, July 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study evaluated 7 scRNA-seq pipelines on 8 data sets.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Drop-seq-tools version-2.3.0
                     </code>
                     ,
                     <code>
                      Cell Ranger version-3.0.2
                     </code>
                     ,
                     <code>
                      scPipe version-1.4.1
                     </code>
                     ,
                     <code>
                      zUMIs version-2.4.5b
                     </code>
                     ,
                     <code>
                      UMI-tools version-1.0.0
                     </code>
                     ,
                     <code>
                      umis version-1.0.3
                     </code>
                     ,
                     <code>
                      dropEst version-0.8.6
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Cell Ranger shows the highest algorithm complexity and parallelization, whereas scPipe, umis and zUMIs have lower complexity that is suitable for large-scale scRNA-seq integration analysis. UMI-tools show the highest transcript quantification accuracy on ERCC datasets from three scRNA-seq platforms. Integration of expression matrices from different pipelines will introduce confounding factors akin to batch effect. zUMIs and dropEst have higher sensitivity to detectmore genes for single cells, which may also bring unwanted factors. For most downstream analysis, Drop-seq-tools, Cell Ranger and UMI-tools show high consistency, whereas umis and zUMIs show inconsistent results compared with the other pipelines.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/s41467-019-12266-7" rel="nofollow">
                      A systematic evaluation of single cell RNA-seq analysis pipelines
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Beate Veith, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Communications, October 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study evaluated ~3000 pipeline combinations based on three mapping, three annotation, four imputation, seven normalization, and four differential expression testing approaches with five scRNA-seq library protocols on simulated data.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     scRNA-seq library prep protocols -
                     <code>
                      SCRB-seq
                     </code>
                     ,
                     <code>
                      Smart-seq2
                     </code>
                     ,
                     <code>
                      CEL-seq2
                     </code>
                     ,
                     <code>
                      Drop-seq
                     </code>
                     ,
                     <code>
                      10X Genomics
                     </code>
                     . Mapping -
                     <code>
                      bwa
                     </code>
                     ,
                     <code>
                      STAR
                     </code>
                     ,
                     <code>
                      kallisto
                     </code>
                     . Annotation -
                     <code>
                      gencode
                     </code>
                     ,
                     <code>
                      refseq
                     </code>
                     ,
                     <code>
                      vega
                     </code>
                     . Imputation -
                     <code>
                      filtering
                     </code>
                     ,
                     <code>
                      DrImpute
                     </code>
                     ,
                     <code>
                      scone
                     </code>
                     ,
                     <code>
                      SAVER
                     </code>
                     . Normalization -
                     <code>
                      scran
                     </code>
                     ,
                     <code>
                      SCnorm
                     </code>
                     ,
                     <code>
                      Linnorm
                     </code>
                     ,
                     <code>
                      Census
                     </code>
                     ,
                     <code>
                      MR
                     </code>
                     ,
                     <code>
                      TMM
                     </code>
                     . Differential testing -
                     <code>
                      edgeR-zingeR
                     </code>
                     ,
                     <code>
                      limma
                     </code>
                     ,
                     <code>
                      MAST
                     </code>
                     ,
                     <code>
                      T-test
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     <strong>
                      Figure 5F
                     </strong>
                     contains a flowchart with the authors' recommendations. For alignment,
                     <code>
                      STAR
                     </code>
                     with Gencode annotations generally had the highest mapping and assignment rates. All mappers performed best with Gencode annotations. For normalization,
                     <code>
                      scran
                     </code>
                     was found to best handle potential assymetric differential expression and large numbers of differentially expressed genes. They also note that normalization is overall the most influential step, particularly if asymmetric DE is present (
                     <strong>
                      Figure 5
                     </strong>
                     ). For Smart-seq2 data without spike-ins, the authors suggest
                     <code>
                      Census
                     </code>
                     may be the best choice. The authors found little benefit to imputation in most scenarios, particularly if one of the better normalization methods (e.g.
                     <code>
                      scran
                     </code>
                     ) was used. The authors found library prep and normalization strategies to have a stronger effect on pipeline performance than the choice of differential expression tool, but generally found
                     <code>
                      limma-trend
                     </code>
                     to have the most robust performance.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors made their simulation tool (powsimR) available on
                     <a href="https://github.com/bvieth/powsimR">
                      Github
                     </a>
                     along with their
                     <a href="https://github.com/bvieth/scRNA-seq-pipelines">
                      pipeline scripts
                     </a>
                     to reproduce their analyses.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      scRNA Imputation Methods
                     </h3>
                     <a aria-label="Permalink: scRNA Imputation Methods" class="anchor" href="#scrna-imputation-methods" id="user-content-scrna-imputation-methods">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.biorxiv.org/content/10.1101/2020.01.29.925974v1.full" rel="nofollow">
                      A Systematic Evaluation of Single-cell RNA-sequencing Imputation Methods
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Wenpin Hou, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     bioRxiv, January 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper evaluated 18 scRNA-seq imputation methods using seven datasets containing cell line and tissue data from several experimental protocols. The authors assessed the similarity of imputed cell profiles to bulk samples and investigated whether imputation improves signal recovery or introduces noise in three downstream applications - differential expression, unsupervised clustering, and trajectory inference.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      scVI
                     </code>
                     ,
                     <code>
                      DCA
                     </code>
                     ,
                     <code>
                      MAGIC
                     </code>
                     ,
                     <code>
                      scImpute
                     </code>
                     ,
                     <code>
                      kNN-smoothing
                     </code>
                     ,
                     <code>
                      mcImpute
                     </code>
                     ,
                     <code>
                      SAUCIE
                     </code>
                     ,
                     <code>
                      DrImpute
                     </code>
                     ,
                     <code>
                      PBLR
                     </code>
                     ,
                     <code>
                      SAVER
                     </code>
                     ,
                     <code>
                      VIPER
                     </code>
                     ,
                     <code>
                      SAVERX
                     </code>
                     ,
                     <code>
                      DeepImpute
                     </code>
                     ,
                     <code>
                      scRecover
                     </code>
                     ,
                     <code>
                      ALRA
                     </code>
                     ,
                     <code>
                      bayNorm
                     </code>
                     ,
                     <code>
                      AutoImpute
                     </code>
                     ,
                     <code>
                      scScope
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Figure 6 provides a performance summary of the tested methods. In general, the authors recommend caution using any of these methods, as they can introduce significant variability and noise into downstream analyses. Of the methods tested,
                     <code>
                      MAGIC
                     </code>
                     ,
                     <code>
                      kNN-smoothing
                     </code>
                     , and
                     <code>
                      SAVER
                     </code>
                     outperformed the other methods most consistently, though this varied widely across evaluation criteria, protocols, datasets, and downstream analysis. Many methods show no clear improvement over no imputation, and in some cases, perform significantly worse in downstream analyses.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links (optional):
                     </strong>
                     The authors placed all of their benchmaking code on
                     <a href="https://github.com/Winnie09/imputationBenchmark">
                      Github
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      scRNA Differential Gene Expression
                     </h3>
                     <a aria-label="Permalink: scRNA Differential Gene Expression" class="anchor" href="#scrna-differential-gene-expression" id="user-content-scrna-differential-gene-expression">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/nmeth.4612" rel="nofollow">
                      Bias, robustness and scalability in single-cell differential expression analysis
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Charlotte Soneson* &amp; Mark D Robinson*
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Methods, February 2018
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper evaluated 36 approaches for determining differential gene expression from both synthetic and 36 real scRNA-seq datasets. The authors assess type I error control, FDR control and power, computational efficiency, and consistency.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      edgeRQLFDetRate
                     </code>
                     ,
                     <code>
                      MASTcpmDetRate
                     </code>
                     ,
                     <code>
                      limmatrend
                     </code>
                     ,
                     <code>
                      MASTtpmDetRate
                     </code>
                     ,
                     <code>
                      edgeRQLF
                     </code>
                     ,
                     <code>
                      ttest
                     </code>
                     ,
                     <code>
                      voomlimma
                     </code>
                     ,
                     <code>
                      Wilcoxon
                     </code>
                     ,
                     <code>
                      MASTcpm
                     </code>
                     ,
                     <code>
                      MASTtpm
                     </code>
                     ,
                     <code>
                      SAMseq
                     </code>
                     ,
                     <code>
                      D3E
                     </code>
                     ,
                     <code>
                      edgeRLRT
                     </code>
                     ,
                     <code>
                      metagenomeSeq
                     </code>
                     ,
                     <code>
                      edgeRLRTcensus
                     </code>
                     ,
                     <code>
                      edgeRLRTdeconv
                     </code>
                     ,
                     <code>
                      monoclecensus
                     </code>
                     ,
                     <code>
                      ROTStpm
                     </code>
                     ,
                     <code>
                      ROTSvoom
                     </code>
                     ,
                     <code>
                      DESeq2betapFALSE
                     </code>
                     ,
                     <code>
                      edgeRLRTrobust
                     </code>
                     ,
                     <code>
                      monoclecount
                     </code>
                     ,
                     <code>
                      DESeq2
                     </code>
                     ,
                     <code>
                      DESeq2nofilt
                     </code>
                     ,
                     <code>
                      ROTScpm
                     </code>
                     ,
                     <code>
                      SeuratTobit
                     </code>
                     ,
                     <code>
                      NODES
                     </code>
                     ,
                     <code>
                      DESeq2census
                     </code>
                     ,
                     <code>
                      scDD
                     </code>
                     ,
                     <code>
                      BPSC
                     </code>
                     ,
                     <code>
                      SCDE
                     </code>
                     ,
                     <code>
                      DEsingle
                     </code>
                     ,
                     <code>
                      monocle
                     </code>
                     ,
                     <code>
                      SeuratBimodnofilt
                     </code>
                     ,
                     <code>
                      SeuratBimodlsExpr2
                     </code>
                     ,
                     <code>
                      SeuratBimod
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     In general, the authors found that gene prefiltering was essential for good, robust performance from many methods. They note high variability between methods and summarize general performance across all metrics in Figure 5. They do not make recommendations as to a specific method/tool. Of note is that Seurat switched to using the wilcoxon test by default after this study was released, as it performed much better than their previously available methods.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors make their benchmarking pipeline,
                     <a href="https://github.com/markrobinsonuzh/conquer">
                      conquer
                     </a>
                     , available on Github.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Trajectory Inference
                     </h3>
                     <a aria-label="Permalink: Trajectory Inference" class="anchor" href="#trajectory-inference" id="user-content-trajectory-inference">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1038/s41587-019-0071-9" rel="nofollow">
                      A comparison of single-cell trajectory inference methods
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Wouter Saelens*, Robrecht Cannoodt*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nat Biotech, April 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     A comprehensive evaluation of 45 trajectory inference methods, this paper provides an unmatched comparison of the rapidly evolving field of single-cell trajectory inference. Each method was scored on accuracy, scalability, stability, and usability. Should be considered a gold-standard for other benchmarking studies.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      PAGA
                     </code>
                     ,
                     <code>
                      RaceID/StemID
                     </code>
                     ,
                     <code>
                      SLICER
                     </code>
                     ,
                     <code>
                      Slingshot
                     </code>
                     ,
                     <code>
                      PAGA Tree
                     </code>
                     ,
                     <code>
                      MST
                     </code>
                     ,
                     <code>
                      pCreode
                     </code>
                     ,
                     <code>
                      SCUBA
                     </code>
                     ,
                     <code>
                      Monocle DDRTree
                     </code>
                     ,
                     <code>
                      Monocle ICA
                     </code>
                     ,
                     <code>
                      cellTree maptpx
                     </code>
                     ,
                     <code>
                      SLICE
                     </code>
                     ,
                     <code>
                      cellTree VEM
                     </code>
                     ,
                     <code>
                      EIPiGraph
                     </code>
                     ,
                     <code>
                      Sincell
                     </code>
                     ,
                     <code>
                      URD
                     </code>
                     ,
                     <code>
                      CellTrails
                     </code>
                     ,
                     <code>
                      Mpath
                     </code>
                     ,
                     <code>
                      CellRouter
                     </code>
                     ,
                     <code>
                      STEMNET
                     </code>
                     ,
                     <code>
                      FateID
                     </code>
                     ,
                     <code>
                      MFA
                     </code>
                     ,
                     <code>
                      GPfates
                     </code>
                     ,
                     <code>
                      DPT
                     </code>
                     ,
                     <code>
                      Wishbone
                     </code>
                     ,
                     <code>
                      SCORPIUS
                     </code>
                     ,
                     <code>
                      Component 1
                     </code>
                     ,
                     <code>
                      Embeddr
                     </code>
                     ,
                     <code>
                      MATCHER
                     </code>
                     ,
                     <code>
                      TSCAN
                     </code>
                     ,
                     <code>
                      Wanderlust
                     </code>
                     ,
                     <code>
                      PhenoPath
                     </code>
                     ,
                     <code>
                      topslam
                     </code>
                     ,
                     <code>
                      Waterfall
                     </code>
                     ,
                     <code>
                      EIPiGraph linear
                     </code>
                     ,
                     <code>
                      ouijaflow
                     </code>
                     ,
                     <code>
                      FORKS
                     </code>
                     ,
                     <code>
                      Angle
                     </code>
                     ,
                     <code>
                      EIPiGraph cycle
                     </code>
                     ,
                     <code>
                      reCAT
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Varies depending on dataset and expected trajectory type, though
                     <a href="https://icb-scanpy.readthedocs-hosted.com/en/stable/tutorials.html#trajectory-inference" rel="nofollow">
                      PAGA, PAGA Tree
                     </a>
                     ,
                     <a href="https://github.com/rcannood/SCORPIUS">
                      SCORPIUS
                     </a>
                     , and
                     <a href="https://bioconductor.org/packages/release/bioc/html/slingshot.html" rel="nofollow">
                      Slingshot
                     </a>
                     all scored highly across all metrics.
                    </p>
                    <p dir="auto">
                     Authors wrote an
                     <a href="https://dynverse.org/users/3-user-guide/2-guidelines/" rel="nofollow">
                      interactive Shiny app
                     </a>
                     to help users choose the best methods for their data.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The
                     <a href="https://dynverse.org/" rel="nofollow">
                      dynverse site
                     </a>
                     contains numerous packages for users to run and compare results from different trajectory methods on their own data without installing each individually by using Docker. Additionally, they provide several tools for developers to wrap and benchmark their own method against those included in the study.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Gene Regulatory Network Inference
                     </h3>
                     <a aria-label="Permalink: Gene Regulatory Network Inference" class="anchor" href="#gene-regulatory-network-inference" id="user-content-gene-regulatory-network-inference">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.nature.com/articles/s41592-019-0690-6" rel="nofollow">
                      Benchmarking algorithms for gene regulatory network inference from single-cell transcriptomic data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Aditya Pratapa, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Nature Methods, January 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     The authors compared 12 gene regulatory network (GRN) inference techniques to assess the accuracy, robustness, and efficiency of each method on simulated data from synthetic networks, simulated data from curated models, and real scRNA-seq datasets.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      GENIE3
                     </code>
                     ,
                     <code>
                      PPCOR
                     </code>
                     ,
                     <code>
                      LEAP
                     </code>
                     ,
                     <code>
                      SCODE
                     </code>
                     ,
                     <code>
                      PIDC
                     </code>
                     ,
                     <code>
                      SINCERITIES
                     </code>
                     ,
                     <code>
                      SCNS
                     </code>
                     ,
                     <code>
                      GRNVBEM
                     </code>
                     ,
                     <code>
                      SCRIBE
                     </code>
                     ,
                     <code>
                      GRNBoost2
                     </code>
                     ,
                     <code>
                      GRISLI
                     </code>
                     ,
                     <code>
                      SINGE
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     In general, the authors recommend
                     <code>
                      PIDC
                     </code>
                     ,
                     <code>
                      GENIE3
                     </code>
                     , or
                     <code>
                      GRNBoost2
                     </code>
                     , as they were the leading and consistent performers for both curated models and experimental datasets in terms of accuracy as well as having multithreaded implementations. Both the
                     <code>
                      GENIE3
                     </code>
                     and
                     <code>
                      GRNBoost2
                     </code>
                     methods can be used from the
                     <a href="https://aertslab.org/#scenic" rel="nofollow">
                      SCENIC
                     </a>
                     workflows in either R or the (much faster) python implementation.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors provide their benchmarking framework,
                     <a href="https://github.com/murali-group/BEELINE">
                      BEELINE
                     </a>
                     on Github, which also provides an easy-to-use and uniform interface to each method in the form of a Docker image.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2217-z" rel="nofollow">
                      Evaluating methods of inferring gene regulatory networks highlights their lack of performance for single cell gene expression data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Shounan Chen* &amp; Jessica C. Mar*
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     BMC Bioinformatics, June 2018
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study compared 8 gene regulatory network inference methods (5 bulk RNA-seq, 3 specific to scRNA-seq) for scRNA-seq data for precision and recall.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Partial correlation (Pcorr)
                     </code>
                     ,
                     <code>
                      Bayesian networks
                     </code>
                     ,
                     <code>
                      GENIE3
                     </code>
                     ,
                     <code>
                      ARACNE
                     </code>
                     ,
                     <code>
                      CLR
                     </code>
                     ,
                     <code>
                      SCENIC
                     </code>
                     ,
                     <code>
                      SCODE
                     </code>
                     ,
                     <code>
                      PIDC
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Generally, the authors found relatively poor performance across all methods both for simulated and real data. The results from each method had few similarities with other methods and high false positive rates, and the authors recommend caution when interpreting the networks reconstructed with these methods. The authors showed that many of these methods were dramatically affected by dropout events.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Integration/Batch Correction
                     </h3>
                     <a aria-label="Permalink: Integration/Batch Correction" class="anchor" href="#integrationbatch-correction" id="user-content-integrationbatch-correction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.biorxiv.org/content/10.1101/2023.11.15.564963v1.full" rel="nofollow">
                      Benchmarking multi-omics integration algorithms across single-cell RNA and ATAC data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Chuxi Xiao, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     bioRxiv, November 2023
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study benchmarks 12 multi-omics integration methods across three integration tasks, assessing their performance in combining single-cell RNA (scRNA-seq) and ATAC (scATAC-seq) data. The evaluation considers aspects such as the extent of mixing between different omics, cell type conservation, single-cell level alignment accuracy, trajectory preservation, scalability, and ease of use.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      scMVP
                     </code>
                     ,
                     <code>
                      MOFA+
                     </code>
                     ,
                     <code>
                      MultiVI
                     </code>
                     ,
                     <code>
                      Cobolt
                     </code>
                     ,
                     <code>
                      scDART
                     </code>
                     ,
                     <code>
                      UnionCom
                     </code>
                     ,
                     <code>
                      MMD-MA
                     </code>
                     ,
                     <code>
                      scJoint
                     </code>
                     ,
                     <code>
                      Harmony
                     </code>
                     ,
                     <code>
                      Seurat (v4.3)
                     </code>
                     ,
                     <code>
                      LIGER
                     </code>
                     ,
                     <code>
                      GLUE
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     The study recommends different methods based on dataset type and size. For unpaired datasets,
                     <code>
                      GLUE
                     </code>
                     is preferred. In paired tasks,
                     <code>
                      GLUE
                     </code>
                     and
                     <code>
                      MultiVI
                     </code>
                     are top choices, with the latter excelling in trajectory conservation. For omics mixing,
                     <code>
                      scDART
                     </code>
                     ,
                     <code>
                      LIGER
                     </code>
                     , and
                     <code>
                      Seurat
                     </code>
                     are recommended. For cell type conservation,
                     <code>
                      MOFA+
                     </code>
                     and
                     <code>
                      scMVP
                     </code>
                     are viable options. In terms of scalability,
                     <code>
                      Seurat
                     </code>
                     ,
                     <code>
                      LIGER
                     </code>
                     , and
                     <code>
                      MOFA+
                     </code>
                     are efficient. For ease of use,
                     <code>
                      scDART
                     </code>
                     ,
                     <code>
                      scJoint
                     </code>
                     , and
                     <code>
                      Seurat
                     </code>
                     are highlighted for their detailed guidance.
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1850-9" rel="nofollow">
                      A benchmark of batch-effect correction methods for single-cell RNA sequencing data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Hoa Thi Nhu Tran, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, January 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     The authors compared 14 methods in terms of computational runtime, the ability to handle large datasets, and batch-effect correction efficacy while preserving cell type purity.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Seurat2
                     </code>
                     ,
                     <code>
                      Seurat3
                     </code>
                     ,
                     <code>
                      Harmony
                     </code>
                     ,
                     <code>
                      fastMNN
                     </code>
                     ,
                     <code>
                      MNN Correct
                     </code>
                     ,
                     <code>
                      ComBat
                     </code>
                     ,
                     <code>
                      Limma
                     </code>
                     ,
                     <code>
                      scGen
                     </code>
                     ,
                     <code>
                      Scanorama
                     </code>
                     ,
                     <code>
                      MMD-ResNet
                     </code>
                     ,
                     <code>
                      ZINB-WaVe
                     </code>
                     ,
                     <code>
                      scMerge
                     </code>
                     ,
                     <code>
                      LIGER
                     </code>
                     ,
                     <code>
                      BBKNN
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Based on the benchmarking results authors suggest
                     <a href="https://github.com/immunogenomics/harmony">
                      Harmony
                     </a>
                     ,
                     <a href="https://github.com/MacoskoLab/liger">
                      LIGER
                     </a>
                     , and
                     <a href="https://satijalab.org/seurat/" rel="nofollow">
                      Seurat3
                     </a>
                     as best methods for batch integration.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Dimensionality Reduction
                     </h3>
                     <a aria-label="Permalink: Dimensionality Reduction" class="anchor" href="#dimensionality-reduction" id="user-content-dimensionality-reduction">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://www.biorxiv.org/content/10.1101/641142v2" rel="nofollow">
                      Accuracy, Robustness and Scalability of Dimensionality Reduction Methods for Single Cell RNAseq Analysis
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Shiquan Sun, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     BioRxiv, October 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     A mammoth comparison of 18 different dimension reduction methods on 30 publicly available scRNAseq data sets in addition to 2 simulated datasets for a variety of purposes ranging from cell clustering to trajectory inference to neighborhood preservation.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      factor analysis (FA)
                     </code>
                     ,
                     <code>
                      principal component analysis (PCA)
                     </code>
                     ,
                     <code>
                      independent component analysis (ICA)
                     </code>
                     ,
                     <code>
                      Diffusion Map
                     </code>
                     ,
                     <code>
                      nonnegative matrix factorization (NMF)
                     </code>
                     ,
                     <code>
                      Poisson NMF
                     </code>
                     ,
                     <code>
                      zero-inflated factor analysis (ZIFA)
                     </code>
                     ,
                     <code>
                      zero-inflated negative binomial based wanted variation extraction (ZINB-WaVE)
                     </code>
                     ,
                     <code>
                      probabilistic count matrix factorization (pCMF)
                     </code>
                     ,
                     <code>
                      deep count autoencoder network (DCA)
                     </code>
                     ,
                     <code>
                      scScope
                     </code>
                     ,
                     <code>
                      generalized linear model principal component analysis (GLMPCA)
                     </code>
                     ,
                     <code>
                      multidimensional scaling (MDS)
                     </code>
                     ,
                     <code>
                      locally linear embedding (LLE)
                     </code>
                     ,
                     <code>
                      local tangent space alignment (LTSA)
                     </code>
                     ,
                     <code>
                      Isomap
                     </code>
                     ,
                     <code>
                      uniform manifold approximation and projection (UMAP)
                     </code>
                     ,
                     <code>
                      t-distributed stochastic neighbor embedding (tSNE)
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Varies depending on use case.
                     <a href="https://cran.r-project.org/web/packages/psych/index.html" rel="nofollow">
                      Factor Analysis
                     </a>
                     and
                     <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.1/topics/prcomp" rel="nofollow">
                      principal component analysis
                     </a>
                     performed well for most use cases.
See figure 5 for pratical guidelines.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors have made their benchmarking code available
                     <a href="https://github.com/xzhoulab/DRComparison">
                      on Github
                     </a>
                     .
                    </p>
                    <hr/>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1900-3" rel="nofollow">
                      Benchmarking principal component analysis for large-scale single-cell RNA-sequencing
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Koki Tsuyuzaki, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, January 2020
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study compared 21 implementations of 10 algorithms across Python, R, and Julia for principal component analysis for scRNA-seq data, measuring scalability, computational efficiency, outlier robustness, t-SNE/UMAP replication, ease of use, and more using both synthetic and real datasets.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      PCA (sklearn, full)
                     </code>
                     ,
                     <code>
                      fit (MultiVariateStats.jl)
                     </code>
                     ,
                     <code>
                      Downsampling
                     </code>
                     ,
                     <code>
                      IncrementalPCA (sklearn)
                     </code>
                     ,
                     <code>
                      irlba (irlba)
                     </code>
                     ,
                     <code>
                      svds (RSpectra)
                     </code>
                     ,
                     <code>
                      propack.svd (svd)
                     </code>
                     ,
                     <code>
                      PCA (sklearn, arpack)
                     </code>
                     ,
                     <code>
                      irlb (Cell Ranger)
                     </code>
                     ,
                     <code>
                      svds (Arpack.jl)
                     </code>
                     ,
                     <code>
                      orthiter (OnlinePCA.jl)
                     </code>
                     ,
                     <code>
                      gd (OnlinePCA.jl)
                     </code>
                     ,
                     <code>
                      sgd (OnlinePCA.jl)
                     </code>
                     ,
                     <code>
                      rsvd (rsvd)
                     </code>
                     ,
                     <code>
                      oocPCA_CSV (oocRPCA)
                     </code>
                     ,
                     <code>
                      PCA (sklearn, randomized)
                     </code>
                     ,
                     <code>
                      randomized_svd (sklearn)
                     </code>
                     ,
                     <code>
                      PCA (dask-ml)
                     </code>
                     ,
                     <code>
                      halko (OnlinePCA.jl)
                     </code>
                     ,
                     <code>
                      algorithm971 (OnlinePCA.jl)
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Author recommendations vary based on the language being used and matrix size. See figure 8 for recommendations along with recommended parameter settings.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors published their benchmarking scripts
                     <a href="https://github.com/rikenbit/onlinePCA-experiments">
                      on Github
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Cell Annotation/Inference
                     </h3>
                     <a aria-label="Permalink: Cell Annotation/Inference" class="anchor" href="#cell-annotationinference" id="user-content-cell-annotationinference">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1795-z" rel="nofollow">
                      A comparison of automatic cell identification methods for single-cell RNA sequencing data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Tamim Abdelaal*, Lieke Michielsen*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, September 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     The authors benchmarked 22 classification methods that automatically assign cell identities including
                     <em>
                      single-cell-specific
                     </em>
                     and
                     <em>
                      general-purpose
                     </em>
                     classifiers across 27 publicly available single-cell RNA sequencing datasets of different sizes, technologies, species, and levels of complexity. Two types of experimental setups were used evaluate the performance of each method for within dataset predictions (intra-dataset) and across datasets (inter-dataset) based on accuracy, percentage of unclassified cells, and computation time.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      Garnett
                     </code>
                     ,
                     <code>
                      Moana
                     </code>
                     ,
                     <code>
                      DigitalCellSorter
                     </code>
                     ,
                     <code>
                      SCINA
                     </code>
                     ,
                     <code>
                      scVI
                     </code>
                     ,
                     <code>
                      Cell-BLAST
                     </code>
                     ,
                     <code>
                      ACTINN
                     </code>
                     ,
                     <code>
                      LAmbDA
                     </code>
                     ,
                     <code>
                      scmapcluster
                     </code>
                     ,
                     <code>
                      scmapcell
                     </code>
                     ,
                     <code>
                      scPred
                     </code>
                     ,
                     <code>
                      CHETAH
                     </code>
                     ,
                     <code>
                      CaSTLe
                     </code>
                     ,
                     <code>
                      SingleR
                     </code>
                     ,
                     <code>
                      scID
                     </code>
                     ,
                     <code>
                      singleCellNet
                     </code>
                     ,
                     <code>
                      LDA
                     </code>
                     ,
                     <code>
                      NMC
                     </code>
                     ,
                     <code>
                      RF
                     </code>
                     ,
                     <code>
                      SVM
                     </code>
                     ,
                     <code>
                      SVM&lt;sub&gt;rejection&lt;/sub&gt;
                     </code>
                     ,
                     <code>
                      kNN
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     All classifiers performed well. The authors recommended SVM
                     <sub>
                      rejection
                     </sub>
                     classifier (with a linear kernel). Other classifiers include SVM , singleCellNet, scmapcell, and scPred were also of high performances.
                    </p>
                    <p dir="auto">
                     In their experiments, incorporating prior knowledge in the form of marker genes does not improve the performance.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     A Snakemake workflow,
                     <a href="https://github.com/tabdelaal/scRNAseq_Benchmark/">
                      scRNAseq_Benchmark
                     </a>
                     , was provided to automate the benchmarking analyses.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Variant Calling
                     </h3>
                     <a aria-label="Permalink: Variant Calling" class="anchor" href="#variant-calling" id="user-content-variant-calling">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1863-4" rel="nofollow">
                      Systematic comparative analysis of single-nucleotide variant detection methods from single-cell RNA sequencing data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Fenglin Liu*, Yuanyuan Zhang*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, November 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This paper compared seven variant callers using both simulation and real scRNA-seq datasets and identified several elements influencing their performance, including read depth, variant allele frequency, and specific genomic contexts. Sensitivity and specificity were the benchmarking metrics used.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      SAMtools
                     </code>
                     ,
                     <code>
                      GATK
                     </code>
                     ,
                     <code>
                      CTAT
                     </code>
                     ,
                     <code>
                      FreeBayes
                     </code>
                     ,
                     <code>
                      MuTect2
                     </code>
                     ,
                     <code>
                      Strelka2
                     </code>
                     ,
                     <code>
                      VarScan2
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Varies, see figure 7 for a flowchart breakdown. Generally,
                     <a href="http://www.htslib.org/" rel="nofollow">
                      SAMtools
                     </a>
                     (most sensitive, lower specificity in intronic or high-identity regions),
                     <a href="https://github.com/Illumina/strelka">
                      Strelka2
                     </a>
                     (good performance when read depth &gt;5),
                     <a href="https://github.com/ekg/freebayes">
                      FreeBayes
                     </a>
                     (good specificity/sensitivity in cases with high variant allele frequencies), and CTAT (no alignment step necessary) were top performers.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors made their benchmarking code available
                     <a href="https://github.com/fenglin0/benchmarking_variant_callers">
                      on Github
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      ATAC-seq
                     </h3>
                     <a aria-label="Permalink: ATAC-seq" class="anchor" href="#atac-seq" id="user-content-atac-seq">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1854-5" rel="nofollow">
                      Assessment of computational methods for the analysis of single-cell ATAC-seq data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Caleb Lareau*, Tommaso Andreani*, Micheal E. Vinyard*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, November 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     This study compares 10 methods for scATAC-seq processing and featurizing using 13 synthetic and real datasets from diverse tissues and organisms.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      BROCKMAN
                     </code>
                     ,
                     <code>
                      chromVAR
                     </code>
                     ,
                     <code>
                      cisTopic
                     </code>
                     ,
                     <code>
                      Cicero
                     </code>
                     ,
                     <code>
                      Gene Scoring
                     </code>
                     ,
                     <code>
                      Cusanovich2018
                     </code>
                     ,
                     <code>
                      scABC
                     </code>
                     ,
                     <code>
                      Scasat
                     </code>
                     ,
                     <code>
                      SCRAT
                     </code>
                     ,
                     <code>
                      SnapATAC
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     <a href="https://github.com/r3fang/SnapATAC">
                      SnapATAC
                     </a>
                     ,
                     <a href="https://www.ncbi.nlm.nih.gov/pubmed/30078704" rel="nofollow">
                      Cusanovich2018
                     </a>
                     , and
                     <a href="https://github.com/aertslab/cisTopic">
                      cisTopic
                     </a>
                     were the top performers for separating cell populations of different coverages and noise levels. SnapATAC was the only method capable of analyzing a large dataset (&gt;80k cells).
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors have made their benchmarking code available
                     <a href="https://github.com/pinellolab/scATAC-benchmarking/">
                      on Github
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Statistics
                     </h2>
                     <a aria-label="Permalink: Statistics" class="anchor" href="#statistics" id="user-content-statistics">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      False Discovery Rates
                     </h3>
                     <a aria-label="Permalink: False Discovery Rates" class="anchor" href="#false-discovery-rates" id="user-content-false-discovery-rates">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1716-1" rel="nofollow">
                      A practical guide to methods controlling false discoveries in computational biology
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Keegan Korthauer*, Patrick K. Kimes*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, June 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     An benchmark comparison of the accuracy, applicability, and ease of use of two classic and six modern methods that control for the false discovery rate (FDR). Used simulation studies as well as six case studies in computational biology (specifically differential expression testing in bulk RNA-seq, differential expression testing in single-cell RNA-seq, differential abundance testing and correlation analysis in 16S microbiome data, differential binding testing in ChIP-seq, genome-wide association testing, and gene set analysis).
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     Benjamini-Hochberg, Storeyâs q-value, conditional local FDR (LFDR), FDR regression (FDRreg), independent hypothesis weighting (IHW), adaptive shrinkage (ASH), Boca and Leekâs FDR regression (BL), and adaptive p-value thresholding (AdaPT).
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     Modern FDR methods that use an informative covariate (as opposed to only
                     <em>
                      p
                     </em>
                     -values) leads to more power while controlling the FDR over classic methods. The improvement of the modern FDR methods over the classic methods increases with the informativeness of the covariate, total number of hypothesis tests, and proportion of truly non-null hypotheses.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     Full analyses of the in silico experiments, simulations, and case studies are provided in Additional files 2â41 at
                     <a href="https://pkimes.github.io/benchmark-fdr-html/" rel="nofollow">
                      https://pkimes.github.io/benchmark-fdr-html/
                     </a>
                     . The source code to reproduce all results in the manuscript and additional files, as well as all figures, is
                     <a href="https://github.com/pkimes/benchmark-fdr">
                      available on GitHub
                     </a>
                     . An
                     <code>
                      ExperimentHub
                     </code>
                     package containing the full set of results objects is available through the Bioconductor project, and a Shiny application for interactive exploration of these results is also
                     <a href="https://github.com/kdkorthauer/benchmarkfdr-shiny">
                      available on GitHub
                     </a>
                     . The source code, ExperimentHub package, and Shiny application are all made available under the MIT license.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Microbiome
                     </h2>
                     <a aria-label="Permalink: Microbiome" class="anchor" href="#microbiome" id="user-content-microbiome">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Diversity analysis
                     </h3>
                     <a aria-label="Permalink: Diversity analysis" class="anchor" href="#diversity-analysis" id="user-content-diversity-analysis">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://pubmed.ncbi.nlm.nih.gov/31657871/" rel="nofollow">
                      Evaluating Bioinformatic Pipeline Performance for Forensic Microbiome Analysis
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Sierra F. Kaszubinski*, Jennifer L. Pechal*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Journal of Forensic Sciences, 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     Sequence reads from postmortem microbiome samples were analyzed with
                     <a href="https://mothur.org" rel="nofollow">
                      mothur v1.39.5
                     </a>
                     ,
                     <a href="https://qiime.org" rel="nofollow">
                      QIIME2 v2018.11
                     </a>
                     , and
                     <a href="https://mg-rast.org" rel="nofollow">
                      MG-RAST v4.0.3
                     </a>
                     . For postmortem data, MG-RAST had a much smaller effect size than mothur and QIIME2 due to the twofold reduction in samples. QIIME2 and Mothur returned similar results, with Mothur showing inflated richness due to unclassified taxa. Adjusting minimum library size had significant effects on microbial community structure, sample size less so except for low abundant taxa.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      mothur
                     </code>
                     ,
                     <code>
                      QIIME2
                     </code>
                     ,
                     <code>
                      MG-RAST
                     </code>
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     QIIME2 was deemed the most appropriate choice for forensic analysis in this study.
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     Sequence data are archived through the European Bioinformatics Institute European Nucleotide Archive (
                     <a href="http://www.ebi.ac.uk/ena" rel="nofollow">
                      www.ebi.ac.uk/ena
                     </a>
                     ) under accession number: PRJEB22642. Pipeline parameters and microbial community analyses are available on
                     <a href="https://github.com/sierrakasz/postmortem-analysis">
                      GitHub
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Hi-C/Hi-ChIP
                     </h2>
                     <a aria-label="Permalink: Hi-C/Hi-ChIP" class="anchor" href="#hi-chi-chip" id="user-content-hi-chi-chip">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      TAD Calling
                     </h3>
                     <a aria-label="Permalink: TAD Calling" class="anchor" href="#tad-calling" id="user-content-tad-calling">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.1186/s13059-018-1596-9" rel="nofollow">
                      Comparison of computational methods for the identification of topologically associating domains
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Marie Zufferey*, Daniele Tavernari*, et al.
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Genome Biology, December 2018
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     In this study, the authors compared the performance of 22 TAD callers, each on 20 different conditions (4 map resolutions each normalized with 2 independent strategies, plus 12 additional contact maps with variable sequencing depth) and assessed their performance via concordance, robustness to data resolution and normalization method, and ability to recapitulate biological features typically associated with TADs and TAD boundaries. Assessments were performed on high-resolution Hi-C data from GM12878 and validated in other datasets for select callers.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      3DNetMod
                     </code>
                     ,
                     <code>
                      armatus
                     </code>
                     ,
                     <code>
                      arrowhead
                     </code>
                     ,
                     <code>
                      CaTCH
                     </code>
                     ,
                     <code>
                      CHDF
                     </code>
                     ,
                     <code>
                      chromoR
                     </code>
                     ,
                     <code>
                      ClusterTAD
                     </code>
                     ,
                     <code>
                      DI
                     </code>
                     ,
                     <code>
                      EAST
                     </code>
                     ,
                     <code>
                      GMAP
                     </code>
                     ,
                     <code>
                      HiCExplorer
                     </code>
                     ,
                     <code>
                      HiCseq
                     </code>
                     ,
                     <code>
                      HiTAD
                     </code>
                     ,
                     <code>
                      ICFinder
                     </code>
                     ,
                     <code>
                      IS
                     </code>
                     ,
                     <code>
                      matryoshka
                     </code>
                     ,
                     <code>
                      MrTADFinder
                     </code>
                     ,
                     <code>
                      PSYCHIC
                     </code>
                     ,
                     <code>
                      spectral
                     </code>
                     ,
                     <code>
                      TADbit
                     </code>
                     ,
                     <code>
                      TADtree
                     </code>
                     , and
                     <code>
                      TopDom
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     See
                     <strong>
                      Table 2
                     </strong>
                     for a succinct results summary. In general, the authors found that
                     <code>
                      TopDom
                     </code>
                     ,
                     <code>
                      HiCseg
                     </code>
                     , and
                     <code>
                      CaTCH
                     </code>
                     satisfied at least four out of five criteria: robustness with respect to bin size (resolution) and normalization strategy (ICE and LGF); cost-effective performance based on the ability of the caller to identify concordant TADs with &lt;1% of reads; reproducibility of TADs called by other callers; computational efficiency; and biological relevance based on previosly reported TAD-associated features. The authors do, however, note that callers that attempt to call hierarchical TAD structures generally performed worse than those that do not, potentially due to requiring higher data resolution.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Normalization Methods
                     </h3>
                     <a aria-label="Permalink: Normalization Methods" class="anchor" href="#normalization-methods-1" id="user-content-normalization-methods-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <strong>
                      Title:
                     </strong>
                     <a href="https://doi.org/10.2144/btn-2019-0105" rel="nofollow">
                      Comparison of normalization methods for Hi-C data
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Authors:
                     </strong>
                     Hongqiang Lyu, Erhu Liu, Zhifang Wu
                    </p>
                    <p dir="auto">
                     <strong>
                      Journal Info:
                     </strong>
                     Biotechniques, October 2019
                    </p>
                    <p dir="auto">
                     <strong>
                      Description:
                     </strong>
                     In this study, the authors compared the performance of 6 Hi-C normalization methods at 8 different resolution levels - 2.5M, 1M, 500k, 250k, 100k, 50k, 10k, and 5k from 4 different Hi-C studies. These methods were compared for heat map texture, statistical quality, influence of resolution, consistency of distance stratum, and reproducibility of TAD architecture. The authors assessed the quality of statistics by comparing the distribution of interaction frequency, correlation of replicates, and comparability of replicates between contexts.
                    </p>
                    <p dir="auto">
                     <strong>
                      Tools/methods compared:
                     </strong>
                     <code>
                      SCN
                     </code>
                     ,
                     <code>
                      HiCNorm
                     </code>
                     ,
                     <code>
                      ICE
                     </code>
                     ,
                     <code>
                      KR
                     </code>
                     ,
                     <code>
                      chromoR
                     </code>
                     , and
                     <code>
                      multiHiCcompare
                     </code>
                     .
                    </p>
                    <p dir="auto">
                     <strong>
                      Recommendation(s):
                     </strong>
                     See
                     <strong>
                      Table 2
                     </strong>
                     for a succinct results summary. In summary, the authors found that use-case determines the best tool. All of these methods except
                     <code>
                      multiHiCcompare
                     </code>
                     are single sample methods, which led to
                     <code>
                      multiHiCcompare
                     </code>
                     performing the best in many considerations, including distribution of interaction frequency, comparability between contexts, and consistency of distance stratum. However,
                     <code>
                      multiHiCcompare
                     </code>
                     is by far the most computationally expensive and requires a large number of samples to be useful.
                     <code>
                      SCN
                     </code>
                     and
                     <code>
                      KR
                     </code>
                     show the best reproducibility of TAD architecture across various resolutions.
                     <code>
                      chromoR
                     </code>
                     was found to blur interaction heatmaps at lower resolutions due to its de-noising procedure, though it was also found the achieve the best correlation of replicates at mid-resolutions (500k, 250k).
                    </p>
                    <p dir="auto">
                     <strong>
                      Additional links:
                     </strong>
                     The authors have made their implementations to run the various methods available on
                     <a href="https://github.com/lhqxinghun/bioinformatics/tree/master/Hi-C/NormCompare">
                      GitHub
                     </a>
                     .
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Contributors
                     </h1>
                     <a aria-label="Permalink: Contributors" class="anchor" href="#contributors" id="user-content-contributors">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Jared Andrews (
                      <a href="https://github.com/j-andrews7/">
                       @j-andrews7
                      </a>
                      )
                     </li>
                     <li>
                      Kevin Blighe (
                      <a href="https://github.com/kevinblighe">
                       @kevinblighe
                      </a>
                      ,
                      <a href="https://www.biostars.org/u/41557/" rel="nofollow">
                       biostars
                      </a>
                      )
                     </li>
                     <li>
                      Ludwig Geistlinger (
                      <a href="https://github.com/lgeistlinger">
                       @lgeistlinger
                      </a>
                      )
                     </li>
                     <li>
                      Jeremy Leipzig (
                      <a href="https://github.com/leipzig">
                       @leipzig
                      </a>
                      )
                     </li>
                     <li>
                      Avi Srivastava (
                      <a href="https://github.com/k3yavi">
                       @k3yavi
                      </a>
                      )
                     </li>
                     <li>
                      Stephanie Hicks (
                      <a href="https://github.com/stephaniehicks">
                       @stephaniehicks
                      </a>
                      )
                     </li>
                     <li>
                      Sridhar N Srivatsan (
                      <a href="https://github.com/sridhar0605">
                       @sridhar0605
                      </a>
                      )
                     </li>
                     <li>
                      Qingzhou Zhang (
                      <a href="https://github.com/zqzneptune">
                       @zqzneptune
                      </a>
                      )
                     </li>
                     <li>
                      Guandong Shang (
                      <a href="https://github.com/shangguandong1996">
                       @shangguandong1996
                      </a>
                      ,
                      <a href="https://twitter.com/GuandongS" rel="nofollow">
                       @GuandongS
                      </a>
                      )
                     </li>
                    </ul>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="VRCiuRIy3ZlGXLSjZNMtuRa2IAeP6-jYMenoDfG-8TBCyD0Wjjml-UUQTtoyh7B3YAwN5G9Ck0JAOsym9B1vEw">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 A curated list of bioinformatics bench-marking papers and resources.
                </p>
                <h3 class="sr-only">
                 Topics
                </h3>
                <div class="my-3">
                 <div class="f6">
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:bioinformatics" data-view-component="true" href="/topics/bioinformatics" title="Topic: bioinformatics">
                   bioinformatics
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:collection" data-view-component="true" href="/topics/collection" title="Topic: collection">
                   collection
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:genomics" data-view-component="true" href="/topics/genomics" title="Topic: genomics">
                   genomics
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:bioinformatics-benchmarks" data-view-component="true" href="/topics/bioinformatics-benchmarks" title="Topic: bioinformatics-benchmarks">
                   bioinformatics-benchmarks
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:benchmarking-study" data-view-component="true" href="/topics/benchmarking-study" title="Topic: benchmarking-study">
                   benchmarking-study
                  </a>
                 </div>
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#GPL-3.0-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  GPL-3.0 license
                 </a>
                </div>
                <h3 class="sr-only">
                 Code of conduct
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:code of conduct"}' href="#coc-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-code-of-conduct mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.048 2.241c.964-.709 2.079-1.238 3.325-1.241a4.616 4.616 0 0 1 3.282 1.355c.41.408.757.86.996 1.428.238.568.348 1.206.347 1.968 0 2.193-1.505 4.254-3.081 5.862-1.496 1.526-3.213 2.796-4.249 3.563l-.22.163a.749.749 0 0 1-.895 0l-.221-.163c-1.036-.767-2.753-2.037-4.249-3.563C1.51 10.008.007 7.952.002 5.762a4.614 4.614 0 0 1 1.353-3.407C3.123.585 6.223.537 8.048 2.24Zm-1.153.983c-1.25-1.033-3.321-.967-4.48.191a3.115 3.115 0 0 0-.913 2.335c0 1.556 1.109 3.24 2.652 4.813C5.463 11.898 6.96 13.032 8 13.805c.353-.262.758-.565 1.191-.905l-1.326-1.223a.75.75 0 0 1 1.018-1.102l1.48 1.366c.328-.281.659-.577.984-.887L9.99 9.802a.75.75 0 1 1 1.019-1.103l1.384 1.28c.295-.329.566-.661.81-.995L12.92 8.7l-1.167-1.168c-.674-.671-1.78-.664-2.474.03-.268.269-.538.537-.802.797-.893.882-2.319.843-3.185-.032-.346-.35-.693-.697-1.043-1.047a.75.75 0 0 1-.04-1.016c.162-.191.336-.401.52-.623.62-.748 1.356-1.637 2.166-2.417Zm7.112 4.442c.313-.65.491-1.293.491-1.916v-.001c0-.614-.088-1.045-.23-1.385-.143-.339-.357-.633-.673-.949a3.111 3.111 0 0 0-2.218-.915c-1.092.003-2.165.627-3.226 1.602-.823.755-1.554 1.637-2.228 2.45l-.127.154.562.566a.755.755 0 0 0 1.066.02l.794-.79c1.258-1.258 3.312-1.31 4.594-.032.396.394.792.791 1.173 1.173Z">
                   </path>
                  </svg>
                  Code of conduct
                 </a>
                </div>
                <include-fragment src="/j-andrews7/awesome-bioinformatics-benchmarks/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   289
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   18
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   55
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fj-andrews7%2Fawesome-bioinformatics-benchmarks&amp;report=j-andrews7+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/j-andrews7/packages?repo_name=awesome-bioinformatics-benchmarks">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/j-andrews7/awesome-bioinformatics-benchmarks/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/j-andrews7/awesome-bioinformatics-benchmarks/graphs/contributors">
                 Contributors
                 <span class="Counter ml-1" data-view-component="true" title="10">
                  10
                 </span>
                </a>
               </h2>
               <include-fragment aria-busy="true" aria-label="Loading contributors" src="/j-andrews7/awesome-bioinformatics-benchmarks/contributors_list?count=10&amp;current_repository=awesome-bioinformatics-benchmarks&amp;items_to_show=10">
                <ul class="list-style-none d-flex flex-wrap mb-n2">
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                </ul>
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       Â© 2024 GitHub,Â Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You canât perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
