<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         kconf001/21SpKristinaCAdvancedGenomicsLog: Notebook for Advanced Genomics
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="90A8:5F4D:355F31:4B7320:6605CB7B" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjkwQTg6NUY0RDozNTVGMzE6NEI3MzIwOjY2MDVDQjdCIiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="19d250821b58a78f9682d0f5e78e24e9a4d51479abb89e0bb88a9cba6bf1c2c6" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:331983717" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Notebook for Advanced Genomics. Contribute to kconf001/21SpKristinaCAdvancedGenomicsLog development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/6391a6dd03704a42fdca2cf2d6789a257dc1ad946ad3c097b44c217d85a628a0/kconf001/21SpKristinaCAdvancedGenomicsLog" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="kconf001/21SpKristinaCAdvancedGenomicsLog: Notebook for Advanced Genomics" name="twitter:title">
                       <meta content="Notebook for Advanced Genomics. Contribute to kconf001/21SpKristinaCAdvancedGenomicsLog development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/6391a6dd03704a42fdca2cf2d6789a257dc1ad946ad3c097b44c217d85a628a0/kconf001/21SpKristinaCAdvancedGenomicsLog" property="og:image"/>
                        <meta content="Notebook for Advanced Genomics. Contribute to kconf001/21SpKristinaCAdvancedGenomicsLog development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="kconf001/21SpKristinaCAdvancedGenomicsLog: Notebook for Advanced Genomics" property="og:title"/>
                        <meta content="https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog" property="og:url"/>
                        <meta content="Notebook for Advanced Genomics. Contribute to kconf001/21SpKristinaCAdvancedGenomicsLog development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjI0NjA4NjA3OTQsInQiOjE3MTE2NTU4MDd9--a916c32c904adecbedeca63fb259d5e0bccfc9317a54303286f174ae26819fc0" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/kconf001/21SpKristinaCAdvancedGenomicsLog git https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git" name="go-import"/>
                        <meta content="77846529" name="octolytics-dimension-user_id">
                         <meta content="kconf001" name="octolytics-dimension-user_login">
                          <meta content="331983717" name="octolytics-dimension-repository_id">
                           <meta content="kconf001/21SpKristinaCAdvancedGenomicsLog" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="331983717" name="octolytics-dimension-repository_network_root_id">
                               <meta content="kconf001/21SpKristinaCAdvancedGenomicsLog" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-a77bec96-de34-4065-9a26-ebc3f4328b53" data-view-component="true" id="dialog-show-dialog-a77bec96-de34-4065-9a26-ebc3f4328b53" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-a77bec96-de34-4065-9a26-ebc3f4328b53-description" aria-labelledby="dialog-a77bec96-de34-4065-9a26-ebc3f4328b53-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-a77bec96-de34-4065-9a26-ebc3f4328b53">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-a77bec96-de34-4065-9a26-ebc3f4328b53-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-a77bec96-de34-4065-9a26-ebc3f4328b53" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-a77bec96-de34-4065-9a26-ebc3f4328b53-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-2505dbc3-eda8-4500-bd6e-2aaa22bd22b5">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-0be5baf5-76e0-4478-8492-abeb6f0c052e">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-49562590-252d-4b81-a695-a1d207cb03b8">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-f9ed9008-9467-447a-881f-7daa6c1196e4">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-c4f5bb23-46fe-495a-9bd7-6bef0dc6c2bb">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-9d2136b0-d62a-4400-b066-b0dc0df0bf0a">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-dfea35b7-de0f-4ef4-b62c-a262c066f993">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-fe377b5f-da54-4b4a-8ca8-eae36d6d424c">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               Â© 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: kconf001 / 21SpKristinaCAdvancedGenomicsLog" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              kconf001
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              21SpKristinaCAdvancedGenomicsLog
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"kconf001","screen_size":"compact"}' data-view-component="true" href="/kconf001">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 kconf001
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"21SpKristinaCAdvancedGenomicsLog","screen_size":"compact"}' data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 21SpKristinaCAdvancedGenomicsLog
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"kconf001","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/kconf001/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/kconf001">
             <span class="AppHeader-context-item-label">
              kconf001
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"21SpKristinaCAdvancedGenomicsLog","screen_size":"full"}' data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog">
             <span class="AppHeader-context-item-label">
              21SpKristinaCAdvancedGenomicsLog
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="ae8d2bjAPC50WtlkQOaF0dTJbrExiGzuoF1cL_n-8os5zBFRKk3j4cPAgUrlcWeK4EWuWgWj2X9-JiKRVDpPYw" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="kconf001" data-current-repository="kconf001/21SpKristinaCAdvancedGenomicsLog" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="ZnCYM7g2bnHsjrJnF86s9EPHHqINy48IbGvDY0C9j5Euuo8ZnnAnBLsiwL7xy1zlEWNcCyusf2cAiYLIp4OU3Q" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:kconf001/21SpKristinaCAdvancedGenomicsLog">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump toâ¦" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump toâ¦" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-0319a1b2-d4f5-4e32-8e7d-6d98a0d4f53c" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-6856f4c6-889d-4492-b47e-a50d23598df4" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-6856f4c6-889d-4492-b47e-a50d23598df4">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="1yimwOkOfefo4VlbDRd1Jv48u4-fH8A6FSKKzeYLNkFFmr_4IdhOfVjDV18zDnTXIzd2TMU_Jm8bop93EFoz-g"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="BGoSk8GoLnGBs8mr9ratu9Dg2oTm8hfDlVPz3KbjatpImHApmSlwqMradAa8Y2CGGbC-k7D3VRg2CyJkuaQzjw">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="ZXoxohIzMUCAYYlfTeiprllkPUsxcFLY2DaSr4qKAjV9J3wERbGYQwN1wNwNuvfE76M4zB-pskC1ljB9suXR_Q">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="ecmmxCY9rSnwyd8sdqhTb68B2nfr-tEHRdN8eBRpxb6-nMnmPzprSxcSRdG_Eyh8f7kCvW5cFzdKsCLdjP2ktA">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-e5660067-cf8b-49d5-9d00-8fe5099d394f" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-22e58dc8-5eed-4607-9535-e5b69a41a366" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-3a1a2fcb-e44b-4806-b86d-d255f5e0bf46" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-b582cf68-f64c-4cf7-85c6-6ffc2dcce9c4" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-6e5dbebf-cb2b-4396-b8c1-65f896390214" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-df1a5d15-8b00-472a-92dc-500c7a748230" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-9fee9acc-bb56-47a9-8080-cdfc4052ffc5" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-9a98db1e-f1c3-46e7-9899-46b3467ac3a4">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-9a98db1e-f1c3-46e7-9899-46b3467ac3a4" id="tooltip-9fee9acc-bb56-47a9-8080-cdfc4052ffc5" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-2139d322-1855-4df0-a70a-9906d996c69c" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-866eb026-1126-4269-a5b3-6f02d0e795df">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-866eb026-1126-4269-a5b3-6f02d0e795df" id="tooltip-2139d322-1855-4df0-a70a-9906d996c69c" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTgwN30=--bd48b93326354e9d0faaa6eae427303eb0c3bce625b72c7b22ebac82a1fb06b5" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-46e7318a-258d-443b-bf3f-c30330bcce47" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=331983717">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857" data-view-component="true" id="dialog-show-dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857-description" aria-labelledby="dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-d4eb90a1-293a-4bd6-b07e-6ffd45c20cbc" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-6ed0f14b-e27d-4eb2-b7b8-0221d4de38f5" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-336c567b-a775-47f3-8b43-892411a5bb20" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-a5ef5627-60c5-446a-b7e8-17367c834df5" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-ea93eca1-adbd-49ee-a4e9-4aa8cfc2f9fe" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-85bb4c81-090d-4f54-9beb-f524a6e5c4f4" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-3c1a6436-5937-4154-b4cd-f5a7eab83857-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-07b0e42a-e87e-473e-8755-3619a7b4a932" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-e9eb3247-5bd8-41c7-97a0-f681e71a5527">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-6eced994-ad04-44bb-ab44-cbe77e741cce" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-5353e7d2-134c-48d4-8ca7-59a82f55b7a4">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-5e5196e4-f1b4-42be-b990-1c6d15c573af">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-abf9ff06-1f11-4e36-8ded-373091fbc179" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-fa442731-8dc5-4b3e-86ac-3a1d7ce20fc2" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-c1a67511-80c0-44ae-ba4f-3b09c4855b6e">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-99e1c965-1d72-4d75-b261-4b1d8fca04da">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-acc9ced2-b0f4-49f1-a489-953b42409d9f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-c7e617c1-a3ee-4a9a-9075-d19e9e00ef83" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-624fb5bc-a1f7-4a44-8fd1-ba968e923a6d" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-bb78a412-2f17-49f7-b392-e8e45b0a553f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-442ca4c2-f756-4aee-ad72-fbb63c1ef16b">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-f1455867-210d-4a1c-9282-705b924ac940">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-d5483c70-7f7b-4840-8439-03635a0aec64">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /kconf001/21SpKristinaCAdvancedGenomicsLog" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /kconf001/21SpKristinaCAdvancedGenomicsLog/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /kconf001/21SpKristinaCAdvancedGenomicsLog/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /kconf001/21SpKristinaCAdvancedGenomicsLog/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /kconf001/21SpKristinaCAdvancedGenomicsLog/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /kconf001/21SpKristinaCAdvancedGenomicsLog/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/kconf001/21SpKristinaCAdvancedGenomicsLog/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /kconf001/21SpKristinaCAdvancedGenomicsLog/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-list" aria-haspopup="true" aria-labelledby="tooltip-8339cba1-9d70-43c4-8931-4bc9764950f4" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-button" popovertarget="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-button" id="tooltip-8339cba1-9d70-43c4-8931-4bc9764950f4" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-button" anchor-offset="normal" data-view-component="true" id="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-f8b72879-56af-438b-a148-b1b1b2b51403-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog" id="item-382779b2-d138-41e6-8b03-e5c9e15a303f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/issues" id="item-65de0d5b-9859-4a46-81ee-8eb05db15a39" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/pulls" id="item-1ff5777f-87f0-4ca1-aa6d-e90a37102fe4" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/actions" id="item-fa6844cc-d79f-4426-98c2-bef07a273953" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/projects" id="item-02ac889a-e33f-43ff-a51b-6a94fff75fd1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/security" id="item-ee638d84-ed0a-4f4a-9d59-1bd1c54ca342" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/pulse" id="item-28b04ab7-2b11-4778-85f5-c48d649c2669" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-0df8ad84-0c42-4fa1-ab6f-29a8571ea32b" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-d74fa449-3324-427d-b717-58887afa2a0b" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-d74fa449-3324-427d-b717-58887afa2a0b" id="tooltip-0df8ad84-0c42-4fa1-ab6f-29a8571ea32b" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTgwN30=--bd48b93326354e9d0faaa6eae427303eb0c3bce625b72c7b22ebac82a1fb06b5" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/kconf001/21SpKristinaCAdvancedGenomicsLog" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjc3ODQ2NTI5" data-targets="command-palette-scope.tokens" data-text="kconf001" data-type="owner" data-value="kconf001" style="white-space:nowrap;line-height:20px;">
         kconf001
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzMzE5ODM3MTc=" data-targets="command-palette-scope.tokens" data-text="21SpKristinaCAdvancedGenomicsLog" data-type="repository" data-value="21SpKristinaCAdvancedGenomicsLog" style="white-space:nowrap;line-height:20px;">
         21SpKristinaCAdva...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-f6006c83-63d9-4757-b531-df42088d4b81" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5ODM3MTc=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         Weâve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="kconf001" data-scope-id="MDQ6VXNlcjc3ODQ2NTI5" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="21SpKristinaCAdvancedGenomicsLog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5ODM3MTc=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="kconf001" data-scope-id="MDQ6VXNlcjc3ODQ2NTI5" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="21SpKristinaCAdvancedGenomicsLog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5ODM3MTc=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="M__B26xrnp9ctaovo575DPkUcvzMD8RkM8ObYmaRxn3jk2v_ObFIOL-IAEgwRmCqXBe5iUgo7Vitab7X0kDQyg">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="â­ï¸ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="fVJOfNnL-ltpv1Pk_nt-ZGZ957Su6Yol5hO0oSx0KukdehKNFuWoP7_0zKAdebpSZ5e5xQIbNB9KOc5ozhyeRw">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/77846529?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/kconf001/21SpKristinaCAdvancedGenomicsLog">
            21SpKristinaCAdvancedGenomicsLog
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/331983717/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":331983717,"originating_url":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="7c400eeaa9ff7d59cc59b39aa3b796d235bda794716ee5631051a4b552b7e49e" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-72645af1-4f74-4aa6-8d3f-5378f4080d95" popover="manual">
              Fork your own copy of kconf001/21SpKristinaCAdvancedGenomicsLog
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-331983717">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/kconf001/21SpKristinaCAdvancedGenomicsLog/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-331983717" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-e728bf">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kconf001/21SpKristinaCAdvancedGenomicsLog/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="L4wAdQsM2qCXv-yfgeqyVVhU1pzG2ytJK06o31XM_5HEkw2neyXh09IQFgzNerqosmKvtPB_JWYY3Orw8XBAig">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="0Tm7fakrJQK7U7UZpaSKElhXgrVBMU0lD8LjxmZgEH86Jrav2QIecf78T4rpNILvsmH7nXeVQwo8UKHpwtyvZA">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (0)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":331983717,"originating_url":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="061fb1bc489712ffb7fc53f36030cc9beb865288085dd9608c46a5161829be60" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="0">
                  0
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-331983717-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/kconf001/21SpKristinaCAdvancedGenomicsLog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-331983717-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-e9b18b">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kconf001/21SpKristinaCAdvancedGenomicsLog/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="SiQWMtnf-vD7TKBu5HN503RNX0XcXWxBaF3QgTJjCLzrRvkT3zmzSC0Yro8zUXnGlnAmzqnRYbUBFtK1nYjANg">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (0)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":331983717,"originating_url":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="d07a79aa531292da4798f6fd11a7930151b3e93e78ed0e81272062e9920c58dc" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
                 0
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-331983717-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/kconf001/21SpKristinaCAdvancedGenomicsLog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-331983717-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-143221">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":331983717,"originating_url":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="bdaeebe99582a624e84148de7ab03b9149d6291760cd001e1441ffa133583272" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="u3QEfTaUasdq5Wo8v35E83Z-ItWDiSAdNCuSiA6oVmajtAgHKCsfxob2XTZjnsKEKVuxwFbTi8kW0GB8WRno_w">
                   <input name="repository_id" type="hidden" value="331983717"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="twbxKKms1DOlMDOMJe6PordmZ6QoBUhUrj-apMa5ubKvxv1StxOhMkkjBIb5DgnV6EP0sf1f44CMxGhQkQgHKw">
                   <input name="repository_id" type="hidden" value="331983717"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-f9d326b4-7100-4463-ab3f-f9f3a9a9c816" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-7ed22537-5174-4f73-b491-372b8ecbcd24" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":331983717,"originating_url":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="7c400eeaa9ff7d59cc59b39aa3b796d235bda794716ee5631051a4b552b7e49e" data-view-component="true" group_item="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-7ed22537-5174-4f73-b491-372b8ecbcd24" popover="manual">
              Fork your own copy of kconf001/21SpKristinaCAdvancedGenomicsLog
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kconf001/21SpKristinaCAdvancedGenomicsLog/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="h9SHAEfn_PBDfKuJq7G2VSMX2eQgpvYN-q0A5p7oCsFsy4rSN87HgwbTURrnIb6oySGgzBYC-CLJP0LJOlS12g">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-8aa66bbb-fda6-41fc-bf22-cd79e647ef1a" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":331983717,"originating_url":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="061fb1bc489712ffb7fc53f36030cc9beb865288085dd9608c46a5161829be60" data-view-component="true" id="icon-button-010c3845-b29f-4a92-995c-3a6d32962b90" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-010c3845-b29f-4a92-995c-3a6d32962b90" id="tooltip-8aa66bbb-fda6-41fc-bf22-cd79e647ef1a" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kconf001/21SpKristinaCAdvancedGenomicsLog/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="J94W2emjg2_nXIx58LLts9j6F8NkTvdJIWYhlBy2rVuGvPn470XK1zEIgpgnkO2mOsduSBHC-r1ILSOgs11l0Q">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-91215ac1-4dfd-4528-9c74-b1a23c57cca6" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":331983717,"originating_url":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="d07a79aa531292da4798f6fd11a7930151b3e93e78ed0e81272062e9920c58dc" data-view-component="true" id="icon-button-ecbaf665-2457-49e0-a923-1a08ffb5f04e" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-ecbaf665-2457-49e0-a923-1a08ffb5f04e" id="tooltip-91215ac1-4dfd-4528-9c74-b1a23c57cca6" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           Notebook for Advanced Genomics
          </p>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/kconf001/21SpKristinaCAdvancedGenomicsLog/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/kconf001/21SpKristinaCAdvancedGenomicsLog/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/kconf001/21SpKristinaCAdvancedGenomicsLog?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         kconf001/21SpKristinaCAdvancedGenomicsLog
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/kconf001/21SpKristinaCAdvancedGenomicsLog/spoofed_commit_check/a225f382d253bacf7a0362bd3ecb81660b4891f4">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":331983717,"defaultBranch":"main","name":"21SpKristinaCAdvancedGenomicsLog","ownerLogin":"kconf001","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-01-22T23:21:19.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/77846529?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1611329669.0","canEdit":true,"refType":"branch","currentOid":"a225f382d253bacf7a0362bd3ecb81660b4891f4"},"tree":{"items":[{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":1,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/kconf001/21SpKristinaCAdvancedGenomicsLog/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git","showCloneWarning":false,"sshUrl":"git@github.com:kconf001/21SpKristinaCAdvancedGenomicsLog.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone kconf001/21SpKristinaCAdvancedGenomicsLog","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fkconf001%2F21SpKristinaCAdvancedGenomicsLog","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/kconf001/21SpKristinaCAdvancedGenomicsLog/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"39","overviewFiles":[{"displayName":"README.md","repoName":"21SpKristinaCAdvancedGenomicsLog","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e01/22/2021\u003c/h1\u003e\u003ca id=\"user-content-01222021\" class=\"anchor\" aria-label=\"Permalink: 01/22/2021\" href=\"#01222021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExercises Day 02\u003c/h2\u003e\u003ca id=\"user-content-exercises-day-02\" class=\"anchor\" aria-label=\"Permalink: Exercises Day 02\" href=\"#exercises-day-02\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eLog into the Cluster\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~\n$ ssh kconf001@turing.hpc.odu.edu\nkconf001@turing.hpc.odu.edu's password:\nLast login: Fri Jan 22 10:12:23 2021 from ip70-160-48-140.hr.hr.cox.net\"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\n$ ssh kconf001@turing.hpc.odu.edu\nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eLast login: Fri Jan 22 10:12:23 2021 from ip70-160-48-140.hr.hr.cox.net\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eMake directory in course workspace called data\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC\n[kconf001@turing1 KristinaC]$ ls\ngroundrules.txt  KristinaC_Exercise1.txt  scripts\n[kconf001@turing1 KristinaC]$ mkdir data\n[kconf001@turing1 KristinaC]$ ls\ndata  groundrules.txt  KristinaC_Exercise1.txt  scripts\"\u003e\u003cpre\u003e[kconf001@turing1 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC\n[kconf001@turing1 KristinaC]$ ls\ngroundrules.txt  KristinaC_Exercise1.txt  scripts\n[kconf001@turing1 KristinaC]$ mkdir data\n[kconf001@turing1 KristinaC]$ ls\ndata  groundrules.txt  KristinaC_Exercise1.txt  scripts\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eMake directory called exercises in data directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 KristinaC]$ cd data/\n[kconf001@turing1 data]$ mkdir exercises\n[kconf001@turing1 data]$ ls\nexercises \n[kconf001@turing1 exercises]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises  \"\u003e\u003cpre\u003e[kconf001@turing1 KristinaC]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data/\n[kconf001@turing1 data]$ mkdir exercises\n[kconf001@turing1 data]$ ls\nexercises \n[kconf001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises  \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eExecute pwd command \u0026amp; start a log of commands with a header for today's date in README.md github logfile in local workspace\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~\n$ pwd        \n/c/Users/lyka\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~\n$ cd Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)\n$ ls                                                                                                                             \n21sp_advgenomics/\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)\n$ pwd \n/c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis \nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)\n$ mkdir 21SpKristinaCAdvancedGenomicsLog\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis\n$ ls\n21sp_advgenomics/  21SpKristinaCAdvancedGenomicsLog/\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis\n$ cd ./21SpKristinaCAdvancedGenomicsLog\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ nano     \"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\n$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e        \n/c/Users/lyka\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\n$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)\n$ ls                                                                                                                             \n21sp_advgenomics/\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)\n$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e \n/c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis \nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)\n$ mkdir 21SpKristinaCAdvancedGenomicsLog\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis\n$ ls\n21sp_advgenomics/  21SpKristinaCAdvancedGenomicsLog/\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis\n$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ./21SpKristinaCAdvancedGenomicsLog\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ nano     \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eWith nano command, README.md file is created with Exercise 02 Header \u0026amp; logfile.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eLogfile is filled at the end of each part of the exercises completed.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eCopy the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your course exercises directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003efrom the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./\n[kconf001@turing1 exercises]$ ls\nExercise2.fasta.gz  Exercise2.fastq.tar.gz\n[kconf001@turing1 exercises]$ ls -alh\ntotal 13M \ndrwxrwxrwx 2 kconf001 users   76 Jan 22 14:25 .\ndrwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz \"\u003e\u003cpre\u003e[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e ./\n[kconf001@turing1 exercises]$ ls\nExercise2.fasta.gz  Exercise2.fastq.tar.gz\n[kconf001@turing1 exercises]$ ls -alh\ntotal 13M \ndrwxrwxrwx 2 kconf001 users   76 Jan 22 14:25 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eUnzip \u0026amp; untar the files\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 exercises]$ gunzip -c Exercise2.fasta.gz \u0026gt; Exercise2.fasta\n[kconf001@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq.tar.g\n[kconf001@turing1 exercises]$ ls -alh\ntotal 37M\ndrwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 .\ndrwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..\n-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz\n[kconf001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz\nExercise2.fastq\n[kconf001@turing1 exercises]$ ls -alh\ntotal 62M\ndrwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 .                                                                   \ndrwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 .. \n-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta                                                     \n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz\n-rw-r--r-- 1 kconf001 users  18M Sep  2  2015 Exercise2.fastq                                                 \n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz \n[kconf001@turing1 exercises]$ pwd \n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises                    \n[kconf001@turing1 exercises]$ ls                      \nExercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq  Exercise2.fastq.tar.gz\"\u003e\u003cpre\u003e[kconf001@turing1 exercises]$ gunzip -c Exercise2.fasta.gz \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e Exercise2.fasta\n[kconf001@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq.tar.g\n[kconf001@turing1 exercises]$ ls -alh\ntotal 37M\ndrwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..\n-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz\n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz\n[kconf001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz\nExercise2.fastq\n[kconf001@turing1 exercises]$ ls -alh\ntotal 62M\ndrwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e                                                                   \ndrwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 .. \n-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta                                                     \n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz\n-rw-r--r-- 1 kconf001 users  18M Sep  2  2015 Exercise2.fastq                                                 \n-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz \n[kconf001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e \n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises                    \n[kconf001@turing1 exercises]$ ls                      \nExercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq  Exercise2.fastq.tar.gz\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003eAdd commands to notebook file (see above)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003eCalculate how many sequences are in each file\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFasta file: Sequences start with '\u0026gt;'\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFastq file: Sequences start with '@'; Each fastq read includes 4 lines.\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFirst line- Identifier\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSecond line- Sequence\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eThird line- Blank line (starts with +, sometimes have the same description as first line)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eFourth line- Quality for each base in Sequence (Second line)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003egrep -c = Used to find patterns (find a string in a file or string)\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCan also divide total lines by 4 to find total sequence reads\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 exercises]$ head -1 Exercise2.fasta\n\u0026gt;scaffold10078|size20675\n[kconf001@turing1 exercises]$ grep -c '\u0026gt;' Exercise2.fasta\n138\n[kconf001@turing1 exercises]$ head -1 Exercise2.fastq\n@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC\n[kconf001@turing1 exercises]$ grep -c '@HS3' Exercise2.fastq\n61304\n[kconf001@turing1 exercises]$ wc Exercise2.fastq\n245216   306520 18459769 Exercise2.fastq\n[kconf001@turing1 exercises]$ echo 245216/4 | bc -l\n61304.00000000000000000000  \"\u003e\u003cpre\u003e[kconf001@turing1 exercises]$ head -1 Exercise2.fasta\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003escaffold10078\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003esize20675\n[kconf001@turing1 exercises]$ grep -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e Exercise2.fasta\n138\n[kconf001@turing1 exercises]$ head -1 Exercise2.fastq\n@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC\n[kconf001@turing1 exercises]$ grep -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e@HS3\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e Exercise2.fastq\n61304\n[kconf001@turing1 exercises]$ wc Exercise2.fastq\n245216   306520 18459769 Exercise2.fastq\n[kconf001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e 245216/4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e bc -l\n61304.00000000000000000000  \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003eCopy the python file from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003einto the class scripts directory\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/\n[kconf001@turing1 exercises]$ cd ../\n[kconf001@turing1 data]$ cd ../\n[kconf001@turing1 KristinaC]$ cd scripts/\n[kconf001@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py\"\u003e\u003cpre\u003e[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/\n[kconf001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kconf001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kconf001@turing1 KristinaC]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e scripts/\n[kconf001@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"10\" dir=\"auto\"\u003e\n\u003cli\u003eStart interactive compute session and re-navigate to exercises directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 scripts]$ cd ../\n[kconf001@turing1 KristinaC]$ cd data/\n[kconf001@turing1 data]$ cd exercises/\n[kconf001@turing1 exercises]$ salloc\nsalloc: Pending job allocation 9269144\nsalloc: job 9269144 queued and waiting for resources\nsalloc: job 9269144 has been allocated resources\nsalloc: Granted job allocation 9269517\nThis session will be terminated in 7 days. If your application requires\na longer excution time, please use command \u0026quot;salloc -t N-0\u0026quot; where N is the\nnumber of days that you need.\n[kconf001@coreV1-22-016 exercises]$\"\u003e\u003cpre\u003e[kconf001@turing1 scripts]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kconf001@turing1 KristinaC]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data/\n[kconf001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e exercises/\n[kconf001@turing1 exercises]$ salloc\nsalloc: Pending job allocation 9269144\nsalloc: job 9269144 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9269144 has been allocated resources\nsalloc: Granted job allocation 9269517\nThis session will be terminated \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 7 days. If your application requires\na longer excution time, please use \u003cspan class=\"pl-c1\"\u003ecommand\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esalloc -t N-0\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e where N is the\nnumber of days that you need.\n[kconf001@coreV1-22-016 exercises]$\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"11\" dir=\"auto\"\u003e\n\u003cli\u003eRun the python script on the Exercise2.fasta file\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003eby typing the path to the script followed by the Exercise2.fasta file name\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV1-22-016 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 138\ncontigs \u0026gt;= 1000bp = 138\ncontigs \u0026gt;= 2000bp = 135\n[kconf001@coreV1-22-016 exercises]$ exit\nlogout\nsalloc: Relinquishing job allocation 9269517\"\u003e\u003cpre\u003e[kconf001@coreV1-22-016 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 150bp = 0\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 500bp = 138\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 1000bp = 138\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 2000bp = 135\n[kconf001@coreV1-22-016 exercises]$ \u003cspan class=\"pl-c1\"\u003eexit\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003elogout\u003c/span\u003e\nsalloc: Relinquishing job allocation 9269517\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"12\" dir=\"auto\"\u003e\n\u003cli\u003eAdd all parts to notebook README.md file including the results of the script\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSee above\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"13\" dir=\"auto\"\u003e\n\u003cli\u003ePush notebook file to github page\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git add README.md\nwarning: LF will be replaced by CRLF in README.md.\nThe file will have its original line endings in your working directory\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git commit -m 'updating readme'\n[main c32c255] updating readme\n1 file changed, 161 insertions(+), 4 deletions(-)\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git push -u origin main\nEnumerating objects: 5, done.\nCounting objects: 100% (5/5), done.\nDelta compression using up to 8 threads\nCompressing objects: 100% (2/2), done.\nWriting objects: 100% (3/3), 2.11 KiB | 2.11 MiB/s, done.\nTotal 3 (delta 0), reused 0 (delta 0), pack-reused 0\nTo https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git\nd677019..c32c255  main -\u0026gt; main\nBranch 'main' set up to track remote branch 'main' from 'origin'.\"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git add README.md\nwarning: LF will be replaced by CRLF \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e README.md.\nThe file will have its original line endings \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e your working directory\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n[main c32c255] updating readme\n1 file changed, 161 insertions(+), 4 deletions(-)\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git push -u origin main\nEnumerating objects: 5, done.\nCounting objects: 100% (5/5), done.\nDelta compression using up to 8 threads\nCompressing objects: 100% (2/2), done.\nWriting objects: 100% (3/3), 2.11 KiB \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e 2.11 MiB/s, done.\nTotal 3 (delta 0), reused 0 (delta 0), pack-reused 0\nTo https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git\nd677019..c32c255  main -\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e main\nBranch \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emain\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eset\u003c/span\u003e up to track remote branch \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emain\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e from \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eorigin\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e.\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e01/27/2021\u003c/h1\u003e\u003ca id=\"user-content-01272021\" class=\"anchor\" aria-label=\"Permalink: 01/27/2021\" href=\"#01272021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 02 (continuation of Exercise 02)\u003c/h2\u003e\u003ca id=\"user-content-homework-day-02-continuation-of-exercise-02\" class=\"anchor\" aria-label=\"Permalink: Homework Day 02 (continuation of Exercise 02)\" href=\"#homework-day-02-continuation-of-exercise-02\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eMake sure logged into turing node \u0026amp; in personal data directory.\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eAlso input salloc command (see previous day for details).\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eWrite an sbatch script to cp the files\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003einto your own data directory\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eCan use nano or scp\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n#SBATCH -o KristinaCCopylane01.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCCopylane01\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01*.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCCopylane01.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCCopylane01\u003c/span\u003e\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eAdd sbatch script to data directory (see above)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"nano method (in turing cluster)\n[kconf001@coreV2-25-072 data] nano KristinaCCopy1.sh \"\u003e\u003cpre\u003enano method (in turing cluster)\n[kconf001@coreV2-25-072 data] nano KristinaCCopy1.sh \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"scp method (in local directory files)\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis \n$ nano KristinaCCopy1.sh\n$ scp KristinaCCopylane01.sh kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/\nkconf001@turing.hpc.odu.edu's password:\nKristinaCCopylane01.sh                                                                           100%  341     8.7KB/s   00:00\n\nInput sbatch script into .sh file\"\u003e\u003cpre\u003escp method (in \u003cspan class=\"pl-k\"\u003elocal\u003c/span\u003e directory files)\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis \n$ nano KristinaCCopy1.sh\n$ scp KristinaCCopylane01.sh kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/\nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKristinaCCopylane01.sh                                                                           100%  341     8.7KB/s   00:00\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eInput sbatch script into .sh file\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eSubmit slurm script (sbatch KristinaCCopyl.sh) and verify that it's working (squeue -u kconf001 and/or ls -alh)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-072 data] $ sbatch KristinaCCopy1.sh\nSubmitted batch job 9270444\n[kconf001@coreV2-25-072 data] $ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)  \n9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072  \n920444      main       sh kconf001  R      01:25      1 coreV2-25-072\"\u003e\u003cpre\u003e[kconf001@coreV2-25-072 data] $ sbatch KristinaCCopy1.sh\nSubmitted batch job 9270444\n[kconf001@coreV2-25-072 data] $ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)  \n9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072  \n920444      main       sh kconf001  R      01:25      1 coreV2-25-072\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eDocument in README.md / github notebook (see #13 in previous day)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eWrite sbatch script to gunzip all the fastaq.gz files once copying is finished.\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Nano method\n#!/bin/bash -l\n#SBATCH -o KristinaCCopylane01.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCGunZiplane01\ngunzip ./*fastq.gz\"\u003e\u003cpre\u003eNano method\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCCopylane01.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCGunZiplane01\u003c/span\u003e\ngunzip ./\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003efastq.gz\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eVerify that it is working\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-072 data] $ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072 \n9270462      main       sh kconf001  R      25:25      1 coreV2-25-072\n\n[kconf001@coreV2-25-072 data]$ ls -alh\ntotal 58G \ndrwxrwxrwx 3 kconf001 users  1017 Jan 27 14:37 . \ndrwxrwxrwx 4 kconf001 users   121 Jan 22 11:31 .. \ndrwxrwxrwx 2 kconf001 users   142 Jan 25 09:57 exercises\n-rw-r--r-- 1 kconf001 users  2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq  \n-rw-r--r-- 1 kconf001 users  981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users   13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq\n-rw------- 1 kconf001 users  1.2G Jan 27 14:37 HADB01-M_S29_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  657M Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq.gz \n-rw-r--r-- 1 kconf001 users  671M Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq.gz\n-rw-r--r-- 1 kconf001 users 1000M Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq.gz\n-rw-r--r-- 1 kconf001 users  538M Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq.gz\n-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:29 KristinaCCopylane01.txt\n-rwxrwxrwx 1 kconf001 users   189 Jan 27 13:49 KristinaCGunZiplane01.sh\n-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:52 KristinaCGunZiplane01.txt\n-rwxrwxrwx 1 kconf001 users   341 Jan 27 13:20 KristinaCLane01.sh\"\u003e\u003cpre\u003e[kconf001@coreV2-25-072 data] $ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072 \n9270462      main       sh kconf001  R      25:25      1 coreV2-25-072\n\n[kconf001@coreV2-25-072 data]$ ls -alh\ntotal 58G \ndrwxrwxrwx 3 kconf001 users  1017 Jan 27 14:37 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e \ndrwxrwxrwx 4 kconf001 users   121 Jan 22 11:31 .. \ndrwxrwxrwx 2 kconf001 users   142 Jan 25 09:57 exercises\n-rw-r--r-- 1 kconf001 users  2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq  \n-rw-r--r-- 1 kconf001 users  981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users   13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq\n-rw------- 1 kconf001 users  1.2G Jan 27 14:37 HADB01-M_S29_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users  657M Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq.gz \n-rw-r--r-- 1 kconf001 users  671M Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq.gz\n-rw-r--r-- 1 kconf001 users 1000M Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq.gz\n-rw-r--r-- 1 kconf001 users  538M Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq.gz\n-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:29 KristinaCCopylane01.txt\n-rwxrwxrwx 1 kconf001 users   189 Jan 27 13:49 KristinaCGunZiplane01.sh\n-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:52 KristinaCGunZiplane01.txt\n-rwxrwxrwx 1 kconf001 users   341 Jan 27 13:20 KristinaCLane01.sh\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003ePush updated README.md log to github page (see #13 in previous day).\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e01/27/2021\u003c/h1\u003e\u003ca id=\"user-content-01272021-1\" class=\"anchor\" aria-label=\"Permalink: 01/27/2021\" href=\"#01272021-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 03\u003c/h2\u003e\u003ca id=\"user-content-homework-day-03\" class=\"anchor\" aria-label=\"Permalink: Homework Day 03\" href=\"#homework-day-03\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ecp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory \u0026amp; files to personal sandbox\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-072 data]$ cd ../\n[kconf001@coreV2-25-072 KristinaC]$  cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./ ls\n[kconf001@coreV2-25-072 KristinaC]$ ls\ndata  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts\"\u003e\u003cpre\u003e[kconf001@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kconf001@coreV2-25-072 KristinaC]$  cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./ ls\n[kconf001@coreV2-25-072 KristinaC]$ ls\ndata  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eMake a new fastq directory (mkdir fastq) in directory and move (mv) all the .fastq files into the fastq directory\nMake sure all files have finished gunzipping (ls -alh)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-072 KristinaC]$ cd data\n[kconf001@coreV2-25-072 data]$ mkdir fastq\n[kconf001@coreV2-25-072 data]$ pwd                                                                                   \n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data\n[kconf001@coreV2-25-072 data]$ ls -alh                                                                               \ntotal 79G\ndrwxrwxrwx 4 kconf001 users  980 Jan 27 14:52 .\ndrwxrwxrwx 5 kconf001 users  144 Jan 27 14:49 ..\ndrwxrwxrwx 2 kconf001 users  142 Jan 25 09:57 exercises\ndrwxrwxrwx 2 kconf001 users    0 Jan 27 14:50 fastq\n-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq             \n-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq                                         \n-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:29 KristinaCCopylane01.txt                                                \n-rwxrwxrwx 1 kconf001 users  189 Jan 27 13:49 KristinaCGunZiplane01.sh                                               \n-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:52 KristinaCGunZiplane01.txt                                              \n-rwxrwxrwx 1 kconf001 users  341 Jan 27 13:20 KristinaCLane01.sh   \n[kconf001@coreV2-25-072 data]$ mv *.fastq ./fastq/\n[kconf001@coreV2-25-072 data]$ ls\nexercises  fastq  KristinaCCopylane01.txt  KristinaCGunZiplane01.sh  KristinaCGunZiplane01.txt  KristinaCLane01.sh\"\u003e\u003cpre\u003e[kconf001@coreV2-25-072 KristinaC]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data\n[kconf001@coreV2-25-072 data]$ mkdir fastq\n[kconf001@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e                                                                                   \n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data\n[kconf001@coreV2-25-072 data]$ ls -alh                                                                               \ntotal 79G\ndrwxrwxrwx 4 kconf001 users  980 Jan 27 14:52 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 5 kconf001 users  144 Jan 27 14:49 ..\ndrwxrwxrwx 2 kconf001 users  142 Jan 25 09:57 exercises\ndrwxrwxrwx 2 kconf001 users    0 Jan 27 14:50 fastq\n-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq\n-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq             \n-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq                                         \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq                                         \n-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:29 KristinaCCopylane01.txt                                                \n-rwxrwxrwx 1 kconf001 users  189 Jan 27 13:49 KristinaCGunZiplane01.sh                                               \n-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:52 KristinaCGunZiplane01.txt                                              \n-rwxrwxrwx 1 kconf001 users  341 Jan 27 13:20 KristinaCLane01.sh   \n[kconf001@coreV2-25-072 data]$ mv \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq ./fastq/\n[kconf001@coreV2-25-072 data]$ ls\nexercises  fastq  KristinaCCopylane01.txt  KristinaCGunZiplane01.sh  KristinaCGunZiplane01.txt  KristinaCLane01.sh\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003ecp the renamingtable_complete.txt from the day03 directory into your fastq directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003eand the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder\u003c/li\u003e\n\u003cli\u003eand less the new script and check out the usage statement.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cp the renamingtable_complete.txt from the day03 directory into your fastq directory \n[kconf001@coreV2-25-072 data]$ cp ../day03/renamingtable_complete.txt ./fastq/\n[kconf001@coreV2-25-072 data]$ cd fastq/\n[kconf001@coreV2-25-072 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq\nHADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq\nHADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt\nHADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq\"\u003e\u003cpre\u003ecp the renamingtable_complete.txt from the day03 directory into your fastq directory \n[kconf001@coreV2-25-072 data]$ cp ../day03/renamingtable_complete.txt ./fastq/\n[kconf001@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e fastq/\n[kconf001@coreV2-25-072 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq\nHADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq\nHADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt\nHADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder\n[kconf001@coreV2-25-072 data]$ cd ../../\n[kconf001@coreV2-25-072 KristinaC]$ ls                                                                               \ndata  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts                                                       \n[kconf001@coreV2-25-072 KristinaC]$ cd scripts/\n[kconf001@coreV2-25-072 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py  /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/\n[kconf001@coreV2-25-072 scripts]$ ls \navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py\"\u003e\u003cpre\u003ecp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder\n[kconf001@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../../\n[kconf001@coreV2-25-072 KristinaC]$ ls                                                                               \ndata  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts                                                       \n[kconf001@coreV2-25-072 KristinaC]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e scripts/\n[kconf001@coreV2-25-072 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py  /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/\n[kconf001@coreV2-25-072 scripts]$ ls \navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"less (head) new script\n[kconf001@coreV2-25-072 scripts]$ head renamer_advbioinf.py \n#!/usr/bin/env python \n####usage renamer.py renamingtable \n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column \nimport sys \nimport os\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\nlinecount+=1\"\u003e\u003cpre\u003eless (head) new script\n[kconf001@coreV2-25-072 scripts]$ head renamer_advbioinf.py \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###usage renamer.py renamingtable \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column \u003c/span\u003e\nimport sys \nimport os\nfin=open(sys.argv[1],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003er\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\nlinecount=0\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eline\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e fin:\nlinecount+=1\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eRun renamer_advbioinf.py script in fastq folter using renamingtable_complete.txt as practice \u0026amp; verify output to screen by hand.\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~                                                                                                             \n$ ssh kconf001@turing.hpc.odu.edu  \nkconf001@turing.hpc.odu.edu's password:                                                                         \nLast login: Wed Jan 27 15:37:12 2021 from ip70-160-48-140.hr.hr.cox.net\n[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/\n[kconf001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\n[kconf001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\nmv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\nmv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\nmv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\nmv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\nmv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\nmv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\nmv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\nmv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\nmv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\nmv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\nmv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\nmv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\nmv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\nmv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\nmv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\nmv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\nmv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\nmv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\nmv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\nmv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\nmv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\nmv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\nmv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\nmv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\nmv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\nmv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\nmv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\nmv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\nmv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\nmv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\nmv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\nmv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\nmv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\nmv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\nmv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\nmv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\nmv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\nmv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\nmv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\nmv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\nmv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\nmv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\nmv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\nmv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\nmv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\nmv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\nmv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\nmv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\nmv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\nmv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\nmv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\nmv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\nmv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\nmv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\nmv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\nmv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\nmv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\nmv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\nmv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\nmv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\nmv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\nmv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\nmv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\nmv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\nmv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\nmv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\nmv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\nmv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\nmv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\nmv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\nmv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\nmv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\nmv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\nmv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\nmv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\nmv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\nmv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\nmv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\nmv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\nmv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\nmv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\nmv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\nmv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\nmv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\nmv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\nmv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\nmv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\nmv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\nmv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\nmv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\nmv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\nmv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\nmv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\nmv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\nmv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\nmv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e                                                                                                             \n$ ssh kconf001@turing.hpc.odu.edu  \nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:                                                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eLast login: Wed Jan 27 15:37:12 2021 from ip70-160-48-140.hr.hr.cox.net\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@turing1 fastq]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eUncomment (remove the #) in renaming python script thaht starts with os.open and comment out the print line\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 fastq]$ pwd \n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\n[kconf001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py\n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\nlinecount+=1\nif linecount\u0026gt;=2:\ncols=line.rstrip().split('\\t')\n#print 'mv %s %s' %(cols[0], cols[1])\nos.popen('mv %s %s' %(cols[0], cols[1]))\"\u003e\u003cpre\u003e[kconf001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e \n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\n[kconf001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###usage renamer.py renamingtable\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\u003c/span\u003e\nimport sys\nimport os\nlinecount+=1\n\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e linecount\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e=2:\ncols=line.rstrip().split(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eprint 'mv %s %s' %(cols[0], cols[1])\u003c/span\u003e\nos.popen(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emv %s %s\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e %(cols[0], cols[1]))\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eWrite sbatch script \u0026amp; submit it to rename all the .fastq files according to renaming table using renamer python script\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 fastq]$ nano KristinaCRenamer.sh\n[kconf001@turing1 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-D_S20_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-E_S21_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq  KristinaCRenamer.sh\nHADB01-C_S19_L002_R1_001.fastq  HADB01-F_S22_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq  renamingtable_complete.txt\n[kconf001@turing1 fastq]$ salloc\nsalloc: Pending job allocation 9270528\nsalloc: job 9270528 queued and waiting for resources \nsalloc: job 9270528 has been allocated resources \nsalloc: Granted job allocation 9270528\n[kconf001@coreV2-25-072 fastq]$ cat KristinaCRenamer.sh \n#!/bin/bash -l \n#SBATCH -o KristinaCRenamer.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCRenamerFastq\n[kconf001@coreV2-25-072 fastq]$ sbatch ./KristinaCRenamer.sh\nSubmitted batch job 9270529\n[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9270529      main Kristina kconf001 PD       0:00      1 (Priority)\n9270528      main       sh kconf001  R       0:29      1 coreV2-25-072\nOnce job is done running: checking to see if renaming is completed. \n[kconf001@coreV2-25-072 fastq]$ ls -alh\ntotal 79G\ndrwxrwxrwx 2 kconf001 users  667 Jan 27 16:02 .\ndrwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..\n-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh\n-rwxrwxrwx 1 kconf001 users  11K Jan 27 16:02 KristinaCRenamer.txt\n-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt\n-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq\n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq\n-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq\n-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq\n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq\n-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq\n-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq      \"\u003e\u003cpre\u003e[kconf001@turing1 fastq]$ nano KristinaCRenamer.sh\n[kconf001@turing1 fastq]$ ls\nHADB01-A_S17_L002_R1_001.fastq  HADB01-D_S20_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq\nHADB01-B_S18_L002_R1_001.fastq  HADB01-E_S21_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq  KristinaCRenamer.sh\nHADB01-C_S19_L002_R1_001.fastq  HADB01-F_S22_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq  renamingtable_complete.txt\n[kconf001@turing1 fastq]$ salloc\nsalloc: Pending job allocation 9270528\nsalloc: job 9270528 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources \nsalloc: job 9270528 has been allocated resources \nsalloc: Granted job allocation 9270528\n[kconf001@coreV2-25-072 fastq]$ cat KristinaCRenamer.sh \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCRenamer.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCRenamerFastq\u003c/span\u003e\n[kconf001@coreV2-25-072 fastq]$ sbatch ./KristinaCRenamer.sh\nSubmitted batch job 9270529\n[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9270529      main Kristina kconf001 PD       0:00      1 (Priority)\n9270528      main       sh kconf001  R       0:29      1 coreV2-25-072\nOnce job is \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e running: checking to see \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e renaming is completed. \n[kconf001@coreV2-25-072 fastq]$ ls -alh\ntotal 79G\ndrwxrwxrwx 2 kconf001 users  667 Jan 27 16:02 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..\n-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh\n-rwxrwxrwx 1 kconf001 users  11K Jan 27 16:02 KristinaCRenamer.txt\n-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt\n-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq\n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq\n-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq\n-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq\n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq\n-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq\n-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq      \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003eUpdate github (see #13 on Exercises Day 02 for similar workflow)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git add README.md\nwarning: LF will be replaced by CRLF in README.md.\nThe file will have its original line endings in your working directory\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git commit -m 'updating readme'\n[main 964b72c] updating readme\n1 file changed, 235 insertions(+), 2 deletions(-)\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git push -u origin main\nEnumerating objects: 5, done.\nCounting objects: 100% (5/5), done.\nDelta compression using up to 8 threads\nCompressing objects: 100% (2/2), done. \nWriting objects: 100% (3/3), 2.99 KiB | 2.99 MiB/s, done.\nTotal 3 (delta 1), reused 0 (delta 0), pack-reused 0\nremote: Resolving deltas: 100% (1/1), completed with 1 local object.\nTo https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git  \nb869eb3..964b72c  main -\u0026gt; main \nBranch 'main' set up to track remote branch 'main' from 'origin'.\"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git add README.md\nwarning: LF will be replaced by CRLF \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e README.md.\nThe file will have its original line endings \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e your working directory\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating readme\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n[main 964b72c] updating readme\n1 file changed, 235 insertions(+), 2 deletions(-)\nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ git push -u origin main\nEnumerating objects: 5, done.\nCounting objects: 100% (5/5), done.\nDelta compression using up to 8 threads\nCompressing objects: 100% (2/2), done. \nWriting objects: 100% (3/3), 2.99 KiB \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e 2.99 MiB/s, done.\nTotal 3 (delta 1), reused 0 (delta 0), pack-reused 0\nremote: Resolving deltas: 100% (1/1), completed with 1 \u003cspan class=\"pl-k\"\u003elocal\u003c/span\u003e object.\nTo https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git  \nb869eb3..964b72c  main -\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e main \nBranch \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emain\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eset\u003c/span\u003e up to track remote branch \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emain\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e from \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eorigin\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e.\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003eNaming convention:\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eSOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eThere are 2 sources: Virginia and Rhode Island\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eThere are 2 symbiotic states: Brown and White\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003eStart process of adapter clipping \u0026amp; quality trimming all the renamed .fastq files in batches, by lane.\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"10\" dir=\"auto\"\u003e\n\u003cli\u003ecp script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory.\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-072 fastq]$ exit\nlogout\nsalloc: Relinquishing job allocation 9270528\n[kconf001@turing1 fastq]$ cd ../../\n[kconf001@turing1 KristinaC]$ ls\ndata  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts\n[kconf001@turing1 KristinaC]$ cd scripts/\n[kconf001@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts\n[kconf001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\"\u003e\u003cpre\u003e[kconf001@coreV2-25-072 fastq]$ \u003cspan class=\"pl-c1\"\u003eexit\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003elogout\u003c/span\u003e\nsalloc: Relinquishing job allocation 9270528\n[kconf001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../../\n[kconf001@turing1 KristinaC]$ ls\ndata  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts\n[kconf001@turing1 KristinaC]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e scripts/\n[kconf001@turing1 scripts]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts\n[kconf001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"11\" dir=\"auto\"\u003e\n\u003cli\u003eless/head new script \u0026amp; check the usage statement\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\n[kconf001@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py\n#!/usr/bin/env python\n# Written by Dan Barshis\nimport sys, os\n########### Usage #############\n# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\n# This should be run from within the folder with all of your original .fastqstrim\n# Will quality trim multiple SINGLE-END fastq files\n# Things to customize for your particular platform:\n[kconf001@turing1 scripts]$  \"\u003e\u003cpre\u003e[kconf001@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py\n[kconf001@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Written by Dan Barshis\u003c/span\u003e\nimport sys, os\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########## Usage #############\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This should be run from within the folder with all of your original .fastqstrim\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Will quality trim multiple SINGLE-END fastq files\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Things to customize for your particular platform:\u003c/span\u003e\n[kconf001@turing1 scripts]$  \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"12\" dir=\"auto\"\u003e\n\u003cli\u003ecp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 scripts]$ cd ../\n[kconf001@turing1 KristinaC]$ cd data/\n[kconf001@turing1 data]$ cd fastq/\n[kconf001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\n[kconf001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\n[kconf001@turing1 fastq]$ ls \nadapterlist_advbioinf.txt   RI_B_01_18.fastq   RI_W_01_22.fastq   VA_B_09_SNP.fastq\nKristinaCRenamer.sh         RI_B_01_22.fastq   RI_W_08_SNP.fastq  VA_W_01_14.fastq\nKristinaCRenamer.txt        RI_B_08_SNP.fastq  VA_B_01_14.fastq   VA_W_01_18.fastq\nrenamingtable_complete.txt  RI_W_01_14.fastq   VA_B_01_18.fastq   VA_W_01_22.fastq\nRI_B_01_14.fastq            RI_W_01_18.fastq   VA_B_01_22.fastq   VA_W_08_SNP.fastq\"\u003e\u003cpre\u003e[kconf001@turing1 scripts]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kconf001@turing1 KristinaC]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data/\n[kconf001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e fastq/\n[kconf001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\n[kconf001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\n[kconf001@turing1 fastq]$ ls \nadapterlist_advbioinf.txt   RI_B_01_18.fastq   RI_W_01_22.fastq   VA_B_09_SNP.fastq\nKristinaCRenamer.sh         RI_B_01_22.fastq   RI_W_08_SNP.fastq  VA_W_01_14.fastq\nKristinaCRenamer.txt        RI_B_08_SNP.fastq  VA_B_01_14.fastq   VA_W_01_18.fastq\nrenamingtable_complete.txt  RI_W_01_14.fastq   VA_B_01_18.fastq   VA_W_01_22.fastq\nRI_B_01_14.fastq            RI_W_01_18.fastq   VA_B_01_22.fastq   VA_W_08_SNP.fastq\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"13\" dir=\"auto\"\u003e\n\u003cli\u003eMake sbatch script for Trimclipfilter, then script \u0026amp; run on fastq files\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n[kconf001@turing1 fastq]$ nano KristinaCTestTrimClip.sh\n[kconf001@turing1 fastq]$ cat KristinaCTestTrimClip.sh\n\n#!/bin/bash -l\n\n#SBATCH -o KristinaCTestTrimClip.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCTestTrimClip\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\n\n[kconf001@turing1 fastq]$ salloc\nsalloc: Pending job allocation 9270528 \nsalloc: job 9270528 queued and waiting for resources\nsalloc: job 9270528 has been allocated resources\nsalloc: Granted job allocation 9270542\n[kconf001@coreV2-25-072 fastq]$ sbatch KristinaCTestTrimClip.sh\nSubmitted batch job 9270546\n[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9270542      main       sh kconf001  R      10:03      1 coreV2-25-072\n9270546      main Kristina kconf001  R       0:03      1 coreV2-25-005\n[kconf001@coreV2-25-072 fastq]$ ls -alh \ndrwxrwxrwx 2 kconf001 users  872 Jan 27 17:35 . \ndrwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 .. \n-rwxr-xr-x 1 kconf001 users 7.7K Jan 27 16:46 adapterlist_advbioinf.txt\n-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh \n-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt\n-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh\n-rwxrwxrwx 1 kconf001 users  123 Jan 27 17:22 KristinaCTestTrimClip.txt\n-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt\n-rwxrwxrwx 1 kconf001 users 959M Jan 27 17:35 RI_B_01_14_clipped.fastq \n-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq\n-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq\n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq\n-rwxrwxrwx 1 kconf001 users   45 Jan 27 17:27 trimclipstats.txt \n-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq\n-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq \n-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq\n-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq \"\u003e\u003cpre\u003e[kconf001@turing1 fastq]$ nano KristinaCTestTrimClip.sh\n[kconf001@turing1 fastq]$ cat KristinaCTestTrimClip.sh\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCTestTrimClip.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCTestTrimClip\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq\n\n[kconf001@turing1 fastq]$ salloc\nsalloc: Pending job allocation 9270528 \nsalloc: job 9270528 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9270528 has been allocated resources\nsalloc: Granted job allocation 9270542\n[kconf001@coreV2-25-072 fastq]$ sbatch KristinaCTestTrimClip.sh\nSubmitted batch job 9270546\n[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001\nJOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9270542      main       sh kconf001  R      10:03      1 coreV2-25-072\n9270546      main Kristina kconf001  R       0:03      1 coreV2-25-005\n[kconf001@coreV2-25-072 fastq]$ ls -alh \ndrwxrwxrwx 2 kconf001 users  872 Jan 27 17:35 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e \ndrwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 .. \n-rwxr-xr-x 1 kconf001 users 7.7K Jan 27 16:46 adapterlist_advbioinf.txt\n-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh \n-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt\n-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh\n-rwxrwxrwx 1 kconf001 users  123 Jan 27 17:22 KristinaCTestTrimClip.txt\n-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt\n-rwxrwxrwx 1 kconf001 users 959M Jan 27 17:35 RI_B_01_14_clipped.fastq \n-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq \n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq\n-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq\n-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq\n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq\n-rwxrwxrwx 1 kconf001 users   45 Jan 27 17:27 trimclipstats.txt \n-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq \n-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq\n-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq\n-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq \n-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq\n-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq\n-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"14\" dir=\"auto\"\u003e\n\u003cli\u003eJob has finished\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 fastq]$ ls -alh                                                                                                                                                                                      \ntotal 584K                                                                                                                                                                                                             \ndrwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 .\ndrwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..\ndrwxrwxrwx 2 kconf001 users 2.2K Jan 28 01:59 filteringstats\n-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh\n-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt\n-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh\n-rwxrwxrwx 1 kconf001 users 3.8K Jan 28 01:59 KristinaCTestTrimClip.txt\ndrwxrwxrwx 2 kconf001 users  591 Jan 28 01:59 originalfastqs\ndrwxrwxrwx 2 kconf001 users  788 Jan 28 01:59 QCFastqs\n-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt\"\u003e\u003cpre\u003e[kconf001@turing1 fastq]$ ls -alh                                                                                                                                                                                      \ntotal 584K                                                                                                                                                                                                             \ndrwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..\ndrwxrwxrwx 2 kconf001 users 2.2K Jan 28 01:59 filteringstats\n-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh\n-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt\n-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh\n-rwxrwxrwx 1 kconf001 users 3.8K Jan 28 01:59 KristinaCTestTrimClip.txt\ndrwxrwxrwx 2 kconf001 users  591 Jan 28 01:59 originalfastqs\ndrwxrwxrwx 2 kconf001 users  788 Jan 28 01:59 QCFastqs\n-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"15\" dir=\"auto\"\u003e\n\u003cli\u003eUpdated Github \u0026amp; README.md (See #7 on Homework Day 03 for similar workflow)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e01/29/2021\u003c/h1\u003e\u003ca id=\"user-content-01292021\" class=\"anchor\" aria-label=\"Permalink: 01/29/2021\" href=\"#01292021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 04\u003c/h2\u003e\u003ca id=\"user-content-homework-day-04\" class=\"anchor\" aria-label=\"Permalink: Homework Day 04\" href=\"#homework-day-04\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eAdd trimclipstats.txt output to full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e1a. Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\nThis script takes a stats output file from fastx_clipper and converts it into a table.\nUsage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt\nOptions (-c and -v must be listed separately to run together):\n-h      Display this help message\n-c      Use comma delimiter instead of tabs\n-v      Verbose mode (print steps to stdout)\n[kconf001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\"\u003e\u003cpre\u003e[kconf001@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\nThis script takes a stats output file from fastx_clipper and converts it into a table.\nUsage: Schafran_trimstatstable.py [-c, -v, \u003cspan class=\"pl-k\"\u003e-h\u003c/span\u003e] inputfile.txt outputfile.txt\nOptions (-c and -v must be listed separately to run together):\n-h      Display this \u003cspan class=\"pl-c1\"\u003ehelp\u003c/span\u003e message\n-c      Use comma delimiter instead of tabs\n-v      Verbose mode (print steps to stdout)\n[kconf001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e1b. Run script on data with outputfilename YOURNAME_trimclipstatsout.txt\u003c/li\u003e\n\u003cli\u003eHad to switch to filteringstats directory because this is where the trimclipstats.txt is located.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 fastq]$ cd filteringstats/ \n[kconf001@turing1 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KristinaC_trimclipstatsout.txt \n[kconf001@turing1 filteringstats]$ ls  \nKristinaC_trimclipstatsout.txt        RI_B_08_SNP_clipped_trimmed_stats.txt  RI_W_01_22_nucdist.png                 VA_B_01_18_nucdist.png                 VA_W_01_14_qualbox.png \nRI_B_01_14_clipped_trimmed_stats.txt  RI_B_08_SNP_nucdist.png                RI_W_01_22_qualbox.png                 VA_B_01_18_qualbox.png                 VA_W_01_18_clipped_trimmed_stats.txt \nRI_B_01_14_nucdist.png                RI_B_08_SNP_qualbox.png                RI_W_08_SNP_clipped_trimmed_stats.txt  VA_B_01_22_clipped_trimmed_stats.txt   VA_W_01_18_nucdist.png \nRI_B_01_14_qualbox.png                RI_W_01_14_clipped_trimmed_stats.txt   RI_W_08_SNP_nucdist.png                VA_B_01_22_nucdist.png                 VA_W_01_18_qualbox.png \nRI_B_01_18_clipped_trimmed_stats.txt  RI_W_01_14_nucdist.png                 RI_W_08_SNP_qualbox.png                VA_B_01_22_qualbox.png                 VA_W_01_22_clipped_trimmed_stats.txt \nRI_B_01_18_nucdist.png                RI_W_01_14_qualbox.png                 trimclipstats.txt                      VA_B_09_SNP_clipped_trimmed_stats.txt  VA_W_01_22_nucdist.png \nRI_B_01_18_qualbox.png                RI_W_01_18_clipped_trimmed_stats.txt   VA_B_01_14_clipped_trimmed_stats.txt   VA_B_09_SNP_nucdist.png                VA_W_01_22_qualbox.png         \nRI_B_01_22_clipped_trimmed_stats.txt  RI_W_01_18_nucdist.png                 VA_B_01_14_nucdist.png                 VA_B_09_SNP_qualbox.png                VA_W_08_SNP_clipped_trimmed_stats.txt \nRI_B_01_22_nucdist.png                RI_W_01_18_qualbox.png                 VA_B_01_14_qualbox.png                 VA_W_01_14_clipped_trimmed_stats.txt   VA_W_08_SNP_nucdist.png \nRI_B_01_22_qualbox.png                RI_W_01_22_clipped_trimmed_stats.txt   VA_B_01_18_clipped_trimmed_stats.txt   VA_W_01_14_nucdist.png                 VA_W_08_SNP_qualbox.png \"\u003e\u003cpre\u003e[kconf001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e filteringstats/ \n[kconf001@turing1 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KristinaC_trimclipstatsout.txt \n[kconf001@turing1 filteringstats]$ ls  \nKristinaC_trimclipstatsout.txt        RI_B_08_SNP_clipped_trimmed_stats.txt  RI_W_01_22_nucdist.png                 VA_B_01_18_nucdist.png                 VA_W_01_14_qualbox.png \nRI_B_01_14_clipped_trimmed_stats.txt  RI_B_08_SNP_nucdist.png                RI_W_01_22_qualbox.png                 VA_B_01_18_qualbox.png                 VA_W_01_18_clipped_trimmed_stats.txt \nRI_B_01_14_nucdist.png                RI_B_08_SNP_qualbox.png                RI_W_08_SNP_clipped_trimmed_stats.txt  VA_B_01_22_clipped_trimmed_stats.txt   VA_W_01_18_nucdist.png \nRI_B_01_14_qualbox.png                RI_W_01_14_clipped_trimmed_stats.txt   RI_W_08_SNP_nucdist.png                VA_B_01_22_nucdist.png                 VA_W_01_18_qualbox.png \nRI_B_01_18_clipped_trimmed_stats.txt  RI_W_01_14_nucdist.png                 RI_W_08_SNP_qualbox.png                VA_B_01_22_qualbox.png                 VA_W_01_22_clipped_trimmed_stats.txt \nRI_B_01_18_nucdist.png                RI_W_01_14_qualbox.png                 trimclipstats.txt                      VA_B_09_SNP_clipped_trimmed_stats.txt  VA_W_01_22_nucdist.png \nRI_B_01_18_qualbox.png                RI_W_01_18_clipped_trimmed_stats.txt   VA_B_01_14_clipped_trimmed_stats.txt   VA_B_09_SNP_nucdist.png                VA_W_01_22_qualbox.png         \nRI_B_01_22_clipped_trimmed_stats.txt  RI_W_01_18_nucdist.png                 VA_B_01_14_nucdist.png                 VA_B_09_SNP_qualbox.png                VA_W_08_SNP_clipped_trimmed_stats.txt \nRI_B_01_22_nucdist.png                RI_W_01_18_qualbox.png                 VA_B_01_14_qualbox.png                 VA_W_01_14_clipped_trimmed_stats.txt   VA_W_08_SNP_nucdist.png \nRI_B_01_22_qualbox.png                RI_W_01_22_clipped_trimmed_stats.txt   VA_B_01_18_clipped_trimmed_stats.txt   VA_W_01_14_nucdist.png                 VA_W_08_SNP_qualbox.png \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e1c. Add KristinaC_trimclipstatsout.txt to the class file (Fulltrimclipstatstable.txt).\u003c/li\u003e\n\u003cli\u003echeck to make sure things are in the correct directory first!\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 filteringstats]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/\n[kconf001@turing1 Astrangia_poculata]$ ls -alh \ntotal 320K\ndrwxrwxrwx 4 dbarshis users  105 Jan 29 01:17 .\ndrwxrwxrwx 3 dbarshis users   36 Jan 21 23:36 ..\n-rwxr-xr-x 1 dbarshis users  29K Jan 29 01:18 Fulltrimclipstatstable.txt \ndrwxrwxrwx 2 dbarshis users 4.8K Jan 21 23:37 originalfastqs \ndrwxrwxrwx 2 dbarshis users  337 Jan 29 00:52 refassembly \n[kconf001@turing1 Astrangia_poculata]$ cd /cm/shared/courses//dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats/\n[kconf001@turing1 filteringstats]$ salloc\nsalloc: Pending job allocation 9271098\nsalloc: job 9271098 queued and waiting for resources\nsalloc: job 9271098 has been allocated resources\nsalloc: Granted job allocation 9271098 \n[kconf001@coreV2-25-072 filteringstats]$ tail -n +2 KristinaC_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\"\u003e\u003cpre\u003e[kconf001@turing1 filteringstats]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/\n[kconf001@turing1 Astrangia_poculata]$ ls -alh \ntotal 320K\ndrwxrwxrwx 4 dbarshis users  105 Jan 29 01:17 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 3 dbarshis users   36 Jan 21 23:36 ..\n-rwxr-xr-x 1 dbarshis users  29K Jan 29 01:18 Fulltrimclipstatstable.txt \ndrwxrwxrwx 2 dbarshis users 4.8K Jan 21 23:37 originalfastqs \ndrwxrwxrwx 2 dbarshis users  337 Jan 29 00:52 refassembly \n[kconf001@turing1 Astrangia_poculata]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /cm/shared/courses//dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats/\n[kconf001@turing1 filteringstats]$ salloc\nsalloc: Pending job allocation 9271098\nsalloc: job 9271098 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9271098 has been allocated resources\nsalloc: Granted job allocation 9271098 \n[kconf001@coreV2-25-072 filteringstats]$ tail -n +2 KristinaC_trimclipstatsout.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eMap reads back to assembly using Bowtie2 alignment algorithm\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eWrite sbatch (in QCFastqs directory) script to do Bowtie2/2.4 on _clippedtrimmedfiltered.fastq data files from data in personal sandbox\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-072 filteringstats]$ cd ../\n[kconf001@coreV2-25-072 fastq]$ cd QCFastqs/\n[kconf001@coreV2-25-072 QCFastqs]$ nano KristinaCBowtieln.sh\n[kconf001@coreV2-25-072 QCFastqs]$ cat KristinaCBowtieln.sh\n\n#!/bin/bash -l\n\n#SBATCH -o KristinaCbowtieln.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCBowtieln\n\nmodule load bowtie2/2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \\\n--rg SM:${i%_clippedtrimmed.fastq} \\\n--very-sensitive -x /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/refassembly/ast_hostsym -U $i \\\n\u0026gt; ${i%_clippedtrimmedfilterd.fastq}.sam; done\n\n[kconf001@coreV2-25-072 QCFastqs]$ sbatch KristinaCBowtieln.sh \nSubmitted batch job 9271110 \n[kconf001@coreV2-25-072 QCFastqs]$ squeue -u kconf001 \n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271110      main Kristina kconf001  R       0:01      1 coreV2-25-007\n           9271098      main       sh kconf001  R      35:31      1 coreV2-25-072 \"\u003e\u003cpre\u003e[kconf001@coreV2-25-072 filteringstats]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kconf001@coreV2-25-072 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e QCFastqs/\n[kconf001@coreV2-25-072 QCFastqs]$ nano KristinaCBowtieln.sh\n[kconf001@coreV2-25-072 QCFastqs]$ cat KristinaCBowtieln.sh\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCbowtieln.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCBowtieln\u003c/span\u003e\n\nmodule load bowtie2/2.4\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_clippedtrimmed.fastq\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e bowtie2 --rg-id \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmed.fastq}\u003c/span\u003e \\\n--rg SM:\u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmed.fastq}\u003c/span\u003e \\\n--very-sensitive -x /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/refassembly/ast_hostsym -U \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmedfilterd.fastq}\u003c/span\u003e.sam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n[kconf001@coreV2-25-072 QCFastqs]$ sbatch KristinaCBowtieln.sh \nSubmitted batch job 9271110 \n[kconf001@coreV2-25-072 QCFastqs]$ squeue -u kconf001 \n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n           9271110      main Kristina kconf001  R       0:01      1 coreV2-25-007\n           9271098      main       sh kconf001  R      35:31      1 coreV2-25-072 \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eJob is complete; should get _clippedtrimmed.fastq.sam files\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 QCFastqs]$ ls -alh\ntotal 184G\ndrwxrwxrwx 2 kconf001 users 1.7K Jan 29 21:18 .\ndrwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 ..\n-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh \n-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt\n-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt\n-rwxrwxrwx 1 kconf001 users 1.1G Jan 28 00:45 RI_B_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 1.8G Jan 29 15:25 RI_B_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:47 RI_B_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 15:54 RI_B_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:50 RI_B_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 16:23 RI_B_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.2G Jan 28 00:52 RI_B_08_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.2G Jan 29 16:51 RI_B_08_SNP_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 886M Jan 28 00:53 RI_W_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 1.4G Jan 29 16:59 RI_W_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.6G Jan 28 00:56 RI_W_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.9G Jan 29 17:29 RI_W_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 00:58 RI_W_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 17:53 RI_W_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 5.9G Jan 28 01:02 RI_W_08_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 9.7G Jan 29 18:41 RI_W_08_SNP_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 01:05 VA_B_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 19:04 VA_B_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.3G Jan 28 01:07 VA_B_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.1G Jan 29 19:25 VA_B_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 4.2G Jan 28 01:10 VA_B_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.9G Jan 29 19:58 VA_B_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:13 VA_B_09_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.5G Jan 29 20:29 VA_B_09_SNP_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 2.3G Jan 28 01:15 VA_W_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 3.7G Jan 29 20:46 VA_W_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 4.1G Jan 28 01:18 VA_W_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.7G Jan 29 21:18 VA_W_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users  11M Jan 28 01:18 VA_W_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users  22M Jan 29 21:18 VA_W_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:21 VA_W_08_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.4G Jan 29 21:48 VA_W_08_SNP_clippedtrimmed.fastq.sam\"\u003e\u003cpre\u003e[kconf001@turing1 QCFastqs]$ ls -alh\ntotal 184G\ndrwxrwxrwx 2 kconf001 users 1.7K Jan 29 21:18 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 ..\n-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh \n-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt\n-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt\n-rwxrwxrwx 1 kconf001 users 1.1G Jan 28 00:45 RI_B_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 1.8G Jan 29 15:25 RI_B_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:47 RI_B_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 15:54 RI_B_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:50 RI_B_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 16:23 RI_B_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.2G Jan 28 00:52 RI_B_08_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.2G Jan 29 16:51 RI_B_08_SNP_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 886M Jan 28 00:53 RI_W_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 1.4G Jan 29 16:59 RI_W_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.6G Jan 28 00:56 RI_W_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.9G Jan 29 17:29 RI_W_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 00:58 RI_W_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 17:53 RI_W_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 5.9G Jan 28 01:02 RI_W_08_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 9.7G Jan 29 18:41 RI_W_08_SNP_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 01:05 VA_B_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 19:04 VA_B_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.3G Jan 28 01:07 VA_B_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 5.1G Jan 29 19:25 VA_B_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 4.2G Jan 28 01:10 VA_B_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.9G Jan 29 19:58 VA_B_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:13 VA_B_09_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.5G Jan 29 20:29 VA_B_09_SNP_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 2.3G Jan 28 01:15 VA_W_01_14_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 3.7G Jan 29 20:46 VA_W_01_14_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 4.1G Jan 28 01:18 VA_W_01_18_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.7G Jan 29 21:18 VA_W_01_18_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users  11M Jan 28 01:18 VA_W_01_22_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users  22M Jan 29 21:18 VA_W_01_22_clippedtrimmed.fastq.sam\n-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:21 VA_W_08_SNP_clippedtrimmed.fastq\n-rwxrwxrwx 1 kconf001 users 6.4G Jan 29 21:48 VA_W_08_SNP_clippedtrimmed.fastq.sam\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eUpdated README.md \u0026amp; github (See #7 on Homework Day 03 for similar workflow)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e02/03/2021\u003c/h1\u003e\u003ca id=\"user-content-02032021\" class=\"anchor\" aria-label=\"Permalink: 02/03/2021\" href=\"#02032021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 05\u003c/h2\u003e\u003ca id=\"user-content-homework-day-05\" class=\"anchor\" aria-label=\"Permalink: Homework Day 05\" href=\"#homework-day-05\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eCombine data with partner (KCKC directory- with Katie Crider)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs *.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\"\u003e\u003cpre\u003ecp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eRun Triniry denovo assembler for two the two combined lanes\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eModify sbatch script below\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#!/bin/bash -l\n\n#SBATCH -o KCKCTrinitydenovo.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCKCTrinitydenovo\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCKCTrinitydenovo.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 32\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -p himem\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCKCTrinitydenovo\u003c/span\u003e\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eCheck \u003ca href=\"https://trinityrnaseq.github.io/\" rel=\"nofollow\"\u003ehttps://trinityrnaseq.github.io/\u003c/a\u003e for usage info\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eSubmit Trinity script\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-072 QCFastqs]$ sbatch KCKCTrinity.sh \"\u003e\u003cpre\u003e[kconf001@coreV2-25-072 QCFastqs]$ sbatch KCKCTrinity.sh \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eWe were experiencing some problems with the job input, had to change some parts of the .sbatch script.\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e02/05/2021\u003c/h1\u003e\u003ca id=\"user-content-02052021\" class=\"anchor\" aria-label=\"Permalink: 02/05/2021\" href=\"#02052021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 6\u003c/h2\u003e\u003ca id=\"user-content-homework-day-6\" class=\"anchor\" aria-label=\"Permalink: Homework Day 6\" href=\"#homework-day-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eStart interactive session via salloc \u0026amp; run cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~ $ ssh kconf001@turing.hpc.odu.edu\nkconf001@turing.hpc.odu.edu's password:\nLast login: Wed Feb  3 15:02:28 2021 from ip70-160-48-140.hr.hr.cox.net\n[kconf001@turing1 ~]$ salloc\nsalloc: Pending job allocation 9272854\nsalloc: job 9272854 queued and waiting for resources\nsalloc: job 9272854 has been allocated resources\nsalloc: Granted job allocation 9272854\n[kconf001@coreV3-23-024 ~]$cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir\n[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta\n################################\n## Counts of transcripts, etc.\n################################\nTotal trinity 'genes':  39219\nTotal trinity transcripts: 41303\nPercent GC: 44.54\n########################################\nStats based on ALL transcript contigs:\n########################################\nContig N10: 1323\nContig N20: 783\nContig N30: 562\nContig N40: 440\nContig N50: 364\nMedian contig length: 277\nAverage contig: 371.08\nTotal assembled bases: 15326666\n#####################################################\n## Stats based on ONLY LONGEST ISOFORM per 'GENE':\n#####################################################\nContig N10: 1094\nContig N20: 682\nContig N30: 511\nContig N40: 410\nContig N50: 347\nMedian contig length: 274\nAverage contig: 355.69\nTotal assembled bases: 13949619\"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e $ ssh kconf001@turing.hpc.odu.edu\nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eLast login: Wed Feb  3 15:02:28 2021 from ip70-160-48-140.hr.hr.cox.net\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@turing1 ~]$ salloc\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: Pending job allocation 9272854\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: job 9272854 queued and waiting for resources\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: job 9272854 has been allocated resources\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: Granted job allocation 9272854\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@coreV3-23-024 ~]$cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e################################\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e## Counts of transcripts, etc.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e################################\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eTotal trinity \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003egenes\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e:  39219\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eTotal trinity transcripts: 41303\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ePercent GC: 44.54\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e########################################\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eStats based on ALL transcript contigs:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e########################################\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N10: 1323\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N20: 783\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N30: 562\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N40: 440\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N50: 364\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eMedian contig length: 277\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAverage contig: 371.08\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eTotal assembled bases: 15326666\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#####################################################\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e## Stats based on ONLY LONGEST ISOFORM per \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003eGENE\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#####################################################\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N10: 1094\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N20: 682\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N30: 511\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N40: 410\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eContig N50: 347\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eMedian contig length: 274\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAverage contig: 355.69\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eTotal assembled bases: 13949619\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eCompare this with the output from avg_cov_len_fasta_advbioinf.py\non our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\nThe total number of sequences is 15079\nThe average sequence length is 876\nThe total number of bases is 13210470\nThe minimum sequence length is 500\nThe maximum sequence length is 10795\nThe N50 is 881\nMedian Length = 578\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 15079\ncontigs \u0026gt;= 1000bp = 3660\ncontigs \u0026gt;= 2000bp = 536\"\u003e\u003cpre\u003e[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\nThe total number of sequences is 15079\nThe average sequence length is 876\nThe total number of bases is 13210470\nThe minimum sequence length is 500\nThe maximum sequence length is 10795\nThe N50 is 881\nMedian Length = 578\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 150bp = 0\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 500bp = 15079\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 1000bp = 3660\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 2000bp = 536\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eLess or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e3a. Mean % \"Overall alignment rate\"\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-024 trinirt_our_dir]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs \n[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;overall alignment rate\u0026quot; KristinaCbowtieln.txt\n89.71% overall alignment rate\n91.95% overall alignment rate\n91.31% overall alignment rate\n92.53% overall alignment rate\n65.88% overall alignment rate\n93.03% overall alignment rate\n93.80% overall alignment rate\n93.07% overall alignment rate\n90.51% overall alignment rate\n79.87% overall alignment rate\n91.41% overall alignment rate\n92.94% overall alignment rate\n93.42% overall alignment rate\n93.54% overall alignment rate\n95.85% overall alignment rate\n92.74% overall alignment rate\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;overall alignment rate\u0026quot; KristinaCbowtieln.txt | cut -d \u0026quot; \u0026quot; -f 1 | cut -d \u0026quot;%\u0026quot; -f 1\n89.71\n91.95\n91.31\n92.53\n65.88\n93.03\n93.80\n93.07\n90.51\n79.87\n91.41\n92.94\n93.42\n93.54\n95.85\n92.74 \nCalculate mean on Excel, mean % overall alignment rate is 89.89933\"\u003e\u003cpre\u003e[kconf001@coreV3-23-024 trinirt_our_dir]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs \n[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt\n89.71% overall alignment rate\n91.95% overall alignment rate\n91.31% overall alignment rate\n92.53% overall alignment rate\n65.88% overall alignment rate\n93.03% overall alignment rate\n93.80% overall alignment rate\n93.07% overall alignment rate\n90.51% overall alignment rate\n79.87% overall alignment rate\n91.41% overall alignment rate\n92.94% overall alignment rate\n93.42% overall alignment rate\n93.54% overall alignment rate\n95.85% overall alignment rate\n92.74% overall alignment rate\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1\n89.71\n91.95\n91.31\n92.53\n65.88\n93.03\n93.80\n93.07\n90.51\n79.87\n91.41\n92.94\n93.42\n93.54\n95.85\n92.74 \nCalculate mean on Excel, mean % overall alignment rate is 89.89933\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e3b. Mean % reads \"aligned exactly 1 time\"\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; KristinaCbowtieln.txt\n425341 (5.89%) aligned exactly 1 time\n1336595 (5.57%) aligned exactly 1 time\n1309089 (5.42%) aligned exactly 1 time\n1148628 (5.28%) aligned exactly 1 time\n313795 (5.25%) aligned exactly 1 time\n1058240 (4.28%) aligned exactly 1 time\n894094 (4.34%) aligned exactly 1 time\n1825387 (4.49%) aligned exactly 1 time\n1191456 (5.74%) aligned exactly 1 time\n1757128 (7.79%) aligned exactly 1 time\n1659193 (5.71%) aligned exactly 1 time\n1277294 (4.72%) aligned exactly 1 time\n626232 (4.08%) aligned exactly 1 time\n1217713 (4.36%) aligned exactly 1 time\n1865 (2.60%) aligned exactly 1 time\n1229632 (4.58%) aligned exactly 1 time\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; KristinaCbowtieln.txt | cut -d \u0026quot;(\u0026quot; -f 2 | cut -d \u0026quot;%\u0026quot; -f 1\n5.89\n5.57\n5.42\n5.28\n5.25\n4.28\n4.34\n4.49\n5.74\n7.79\n5.71\n4.72\n4.08\n4.36\n2.60\n4.58\nCalculate mean on Excel, mean % reads aligned exactly 1 time is 5.00625\"\u003e\u003cpre\u003e[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt\n425341 (5.89%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1336595 (5.57%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1309089 (5.42%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1148628 (5.28%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n313795 (5.25%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1058240 (4.28%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n894094 (4.34%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1825387 (4.49%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1191456 (5.74%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1757128 (7.79%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1659193 (5.71%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1277294 (4.72%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n626232 (4.08%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1217713 (4.36%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1865 (2.60%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1229632 (4.58%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e(\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1\n5.89\n5.57\n5.42\n5.28\n5.25\n4.28\n4.34\n4.49\n5.74\n7.79\n5.71\n4.72\n4.08\n4.36\n2.60\n4.58\nCalculate mean on Excel, mean % reads aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e is 5.00625\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e3c. Mean # of reads \"aligned exactly 1 time\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; KristinaCbowtieln.txt\n425341 (5.89%) aligned exactly 1 time\n1336595 (5.57%) aligned exactly 1 time\n1309089 (5.42%) aligned exactly 1 time\n1148628 (5.28%) aligned exactly 1 time\n313795 (5.25%) aligned exactly 1 time\n1058240 (4.28%) aligned exactly 1 time\n894094 (4.34%) aligned exactly 1 time\n1825387 (4.49%) aligned exactly 1 time\n1191456 (5.74%) aligned exactly 1 time\n1757128 (7.79%) aligned exactly 1 time\n1659193 (5.71%) aligned exactly 1 time\n1277294 (4.72%) aligned exactly 1 time\n626232 (4.08%) aligned exactly 1 time\n1217713 (4.36%) aligned exactly 1 time\n1865 (2.60%) aligned exactly 1 time\n1229632 (4.58%) aligned exactly 1 time\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; KristinaCbowtieln.txt | cut -d \u0026quot; \u0026quot; -f 5\n425341\n1336595\n1309089\n1148628\n313795\n1058240\n894094\n1825387\n1191456\n1757128\n1659193\n1277294\n626232\n1217713\n1865\n1229632\nCalculate mean on Excel, mean # of reads alligned exactly 1 time is 1079480\"\u003e\u003cpre\u003e[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt\n425341 (5.89%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1336595 (5.57%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1309089 (5.42%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1148628 (5.28%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n313795 (5.25%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1058240 (4.28%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n894094 (4.34%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1825387 (4.49%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1191456 (5.74%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1757128 (7.79%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1659193 (5.71%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1277294 (4.72%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n626232 (4.08%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1217713 (4.36%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1865 (2.60%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n1229632 (4.58%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 5\n425341\n1336595\n1309089\n1148628\n313795\n1058240\n894094\n1825387\n1191456\n1757128\n1659193\n1277294\n626232\n1217713\n1865\n1229632\nCalculate mean on Excel, mean \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e of reads alligned exactly 1 time is 1079480\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e3d. Mean % reads \"aligned \u0026gt;1 times\"\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;aligned \u0026gt;1 times\u0026quot; KristinaCbowtieln.txt\n6056902 (83.83%) aligned \u0026gt;1 times\n20712542 (86.38%) aligned \u0026gt;1 times\n20741901 (85.89%) aligned \u0026gt;1 times\n18972847 (87.25%) aligned \u0026gt;1 times\n3627297 (60.63%) aligned \u0026gt;1 times\n21931999 (88.74%) aligned \u0026gt;1 times\n18425413 (89.46%) aligned \u0026gt;1 times\n35983989 (88.58%) aligned \u0026gt;1 times\n17579848 (84.76%) aligned \u0026gt;1 times\n16251697 (72.08%) aligned \u0026gt;1 times\n24925544 (85.71%) aligned \u0026gt;1 times\n23877047 (88.22%) aligned \u0026gt;1 times\n13722839 (89.34%) aligned \u0026gt;1 times\n24899339 (89.18%) aligned \u0026gt;1 times\n66899 (93.25%) aligned \u0026gt;1 times\n23648840 (88.15%) aligned \u0026gt;1 times\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u0026quot;aligned \u0026gt;1 times\u0026quot; KristinaCbowtieln.txt | cut -d \u0026quot;(\u0026quot; -f 2 | cut -d \u0026quot;%\u0026quot; -f 1\n83.83\n86.38\n85.89\n87.25\n60.63\n88.74\n89.46\n88.58\n84.76\n72.08\n85.71\n88.22\n89.34\n89.18\n93.25\n88.15\nCalculate mean on Excel, mean % reads aligned \u0026gt;1 times is 85.09063.\"\u003e\u003cpre\u003e[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt\n6056902 (83.83%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n20712542 (86.38%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n20741901 (85.89%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n18972847 (87.25%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n3627297 (60.63%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n21931999 (88.74%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n18425413 (89.46%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n35983989 (88.58%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n17579848 (84.76%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n16251697 (72.08%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n24925544 (85.71%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n23877047 (88.22%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n13722839 (89.34%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n24899339 (89.18%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n66899 (93.25%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n23648840 (88.15%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n[kconf001@coreV3-23-024 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KristinaCbowtieln.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e(\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1\n83.83\n86.38\n85.89\n87.25\n60.63\n88.74\n89.46\n88.58\n84.76\n72.08\n85.71\n88.22\n89.34\n89.18\n93.25\n88.15\nCalculate mean on Excel, mean % reads aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e is 85.09063.\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eAdd statistics as single rows to shared table\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-024 QCFastqs]$ nano KristinaCalignstats.txt                                                                                                            \n[kconf001@coreV3-23-024 QCFastqs]$ cat KristinaCalignstats.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\"\u003e\u003cpre\u003e[kconf001@coreV3-23-024 QCFastqs]$ nano KristinaCalignstats.txt                                                                                                            \n[kconf001@coreV3-23-024 QCFastqs]$ cat KristinaCalignstats.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e5 \u0026amp; 6: Having issues accessing Trinity.fasta files (file \u0026amp; directory not found)- managed to find Trinity.fasta files, currently sbatch is running\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003eData cleanup \u0026amp; archiving\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e5a. Move Trinity.fasta file into new testassembly directory\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 testassembly]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/       \n[kconf001@coreV3-23-040 testassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/testassembly\"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 testassembly]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/       \n[kconf001@coreV3-23-040 testassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/testassembly\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e5b. sbatch script to rm -r original fastqs * filteringstats (Could not remove -trinity_out_dir since it is already gone- forgot to cp to sandbox\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 testassembly]$ nano Cleanup.sh\n[kconf001@coreV3-23-040 testassembly]$ cat Cleanup.sh\n#!/bin/bash -l\n\n#SBATCH -o KristinaCCleanup.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCCleanup\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats\n[kconf001@coreV3-23-040 testassembly]$ sbatch Cleanup.sh\nSubmitted batch job 9276432\"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 testassembly]$ nano Cleanup.sh\n[kconf001@coreV3-23-040 testassembly]$ cat Cleanup.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCCleanup.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCCleanup\u003c/span\u003e\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats\n[kconf001@coreV3-23-040 testassembly]$ sbatch Cleanup.sh\nSubmitted batch job 9276432\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eSubmit blast job from testassembly directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@turing1 testassembly]$ nano KristinaCBlast.sh\n[kconf001@turing1 testassembly]$ cat KristinaCBlast.sh\n#!/bin/bash -l\n#SBATCH -o KristinaCBlast.txt\n#SBATCH -n 6\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCBlast\nenable_lmod container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n-evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n[kconf001@turing1 testassembly]$ sbatch KristinaCBlast.sh\nSubmitted batch job 9276434\n[kconf001@turing1 testassembly]$ squeue -u kconf001\nJOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9276434      main Kristina kconf001 PD       0:00      1 (Priority)\n9276428      main       sh kconf001  R      45:55      1 coreV3-23-040\n9276430      main Kristina kconf001  R      29:25      1 coreV3-23-040                                              \"\u003e\u003cpre\u003e[kconf001@turing1 testassembly]$ nano KristinaCBlast.sh\n[kconf001@turing1 testassembly]$ cat KristinaCBlast.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCBlast.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 6\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCBlast\u003c/span\u003e\nenable_lmod container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n-evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n[kconf001@turing1 testassembly]$ sbatch KristinaCBlast.sh\nSubmitted batch job 9276434\n[kconf001@turing1 testassembly]$ squeue -u kconf001\nJOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9276434      main Kristina kconf001 PD       0:00      1 (Priority)\n9276428      main       sh kconf001  R      45:55      1 coreV3-23-040\n9276430      main Kristina kconf001  R      29:25      1 coreV3-23-040                                              \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003eHead one of the .sam files to look at the header\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 QCFastqs]$ head VA_W_08_SNP_clippedtrimmed.fastq.sam\n@HD     VN:1.0  SO:unsorted\n@SQ     SN:AstrangiaT1FW_1      LN:4823\n@SQ     SN:AstrangiaT1FW_10     LN:1096\n@SQ     SN:AstrangiaT1FW_11     LN:2560\n@SQ     SN:AstrangiaT1FW_12     LN:2686\n@SQ     SN:AstrangiaT1FW_13     LN:629\n@SQ     SN:AstrangiaT1FW_15     LN:582\n@SQ     SN:AstrangiaT1FW_16     LN:502\n@SQ     SN:AstrangiaT1FW_17     LN:1492\n@SQ     SN:AstrangiaT1FW_18     LN:1197\"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 QCFastqs]$ head VA_W_08_SNP_clippedtrimmed.fastq.sam\n@HD     VN:1.0  SO:unsorted\n@SQ     SN:AstrangiaT1FW_1      LN:4823\n@SQ     SN:AstrangiaT1FW_10     LN:1096\n@SQ     SN:AstrangiaT1FW_11     LN:2560\n@SQ     SN:AstrangiaT1FW_12     LN:2686\n@SQ     SN:AstrangiaT1FW_13     LN:629\n@SQ     SN:AstrangiaT1FW_15     LN:582\n@SQ     SN:AstrangiaT1FW_16     LN:502\n@SQ     SN:AstrangiaT1FW_17     LN:1492\n@SQ     SN:AstrangiaT1FW_18     LN:1197\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003egrep -v '@' sam | head to look at sequence read files, this works because '@' denotes the start of a sequence.\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 QCFastqs]$ grep -v \u0026quot;@\u0026quot; VA_W_08_SNP_clippedtrimmed.fastq.sam | head                                                                                                                                                     \nK00188:59:HMTFHBBXX:2:1101:4026:1648    0       AstrangiaT1FW_65377     610     1       51M     *       0       0       CNCGTTAATCCATTCATGCGCGTCACTAATTAGATGACGAGGCATTTGGCT     A#AAF-A-AFFJAJFJJA\u0026lt;7-AJJA7JJJ\u0026lt;FJJJJ\u0026lt;F\u0026lt;\u0026lt;JFFJJJJ-AJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:4614:1648    0       AstrangiaT1FW_488087    41      1       51M     *       0       0       GNTCTTGATTAATGAAAACATTCTTGGCAAATGCTTTCGCAGTAGTTCGTC     A#AFFJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJJFJJJJJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5690:1648    16      AstrangiaT1FW_65728     10184   1       51M     *       0       0       AGACGAACTACTGCGAAAGCATTTGCCAAGAATGTTTTCATTAATCAAGNG     FJFAJJJAJJJJJFJJJJJJJJJJF-AJFF\u0026lt;JJJJJJJJJJJJAJJFFA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49A0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       \nK00188:59:HMTFHBBXX:2:1101:3772:1666    16      AstrangiaT1FW_65718     4457    1       51M     *       0       0       TTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTGGACATGNG     7JJFFJJAJJF\u0026lt;JJJJFJJJJJJJJ7JFJJ\u0026lt;FFA\u0026lt;JJJFJJJJJJJFFA#\u0026lt;    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:49G1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5051:1666    16      AstrangiaT1FW_131671    3       1       51M     *       0       0       ACAGATGTGCCGCCCCAGCCAAACTCCCAACCTGACGGTGTCTTGGACTNG     JJFJJFJJFJJJJJJFJJJJJFJJJJJJFFJFFJAJJJJJJJJJJJFAA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49C0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5172:1666    0       AstrangiaT1FW_65662     1901    1       51M     *       0       0       TNCGACGGGCGGTGTGTACAAAGGGCAGGGACGTAATCAACGCGAGCTGAT     A#AFFJAJJJJJJJFJFFJJJJFF-AJJJJJJJJJJJJJJJFJJJJJAJJ7    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:0A0G49     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       \nK00188:59:HMTFHBBXX:2:1101:5680:1666    0       AstrangiaT1FW_65668     593     1       51M     *       0       0       GNGCATTGCTGAGTCCATTCGTTTCCCTTTCAACGGTTTCACGTACTTTTT     A#-AFFJFJJ7JJJJJJFFJ\u0026lt;FFFF-FJJFJJJF\u0026lt;AFF7FJJAJFJJ7FJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:6674:1666    16      AstrangiaT1FW_65718     1086    1       23M     *       0       0       TCCCAGGAGCATGTCTGTCTGNG FF\u0026lt;JJJJAJJJJJJJJJJFFA#A AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0 NM:i:1  MD:Z:21A1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                                                                                       \nK00188:59:HMTFHBBXX:2:1101:7060:1666    0       AstrangiaT1FW_65712     6946    1       51M     *       0       0       CNAAATTTGACGATCGATTTGCACGTCAGAATCGCTACGAGCCTCCACCAG     A#AFFJFFJJJJJFJJJJJJJJJJ\u0026lt;-FJJJJFJJJJJJJJJJJJJJJJJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:7080:1666    0       AstrangiaT1FW_65718     10568   0       51M     *       0       0       TNTTGCAATAGTAATTCTGCTCAGTACGAGAGGAACCGCAGATTCAGACAA     A#AFFJJJFJJJJJJJJJJJ-FJAAAJJJFJJJJJJJJJJJJJJJJJJJJJ    AS:i:-13        XS:i:-19        XN:i:0  XM:i:3  XO:i:0  XG:i:0  NM:i:3  MD:Z:1G16T3G28  YT:Z:UU RG:Z:VA_W_08_SNP     \"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 QCFastqs]$ grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e@\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e VA_W_08_SNP_clippedtrimmed.fastq.sam \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e head                                                                                                                                                     \nK00188:59:HMTFHBBXX:2:1101:4026:1648    0       AstrangiaT1FW_65377     610     1       51M     *       0       0       CNCGTTAATCCATTCATGCGCGTCACTAATTAGATGACGAGGCATTTGGCT     A#AAF-A-AFFJAJFJJA\u0026lt;7-AJJA7JJJ\u0026lt;FJJJJ\u0026lt;F\u0026lt;\u0026lt;JFFJJJJ-AJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:4614:1648    0       AstrangiaT1FW_488087    41      1       51M     *       0       0       GNTCTTGATTAATGAAAACATTCTTGGCAAATGCTTTCGCAGTAGTTCGTC     A#AFFJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJJFJJJJJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5690:1648    16      AstrangiaT1FW_65728     10184   1       51M     *       0       0       AGACGAACTACTGCGAAAGCATTTGCCAAGAATGTTTTCATTAATCAAGNG     FJFAJJJAJJJJJFJJJJJJJJJJF-AJFF\u0026lt;JJJJJJJJJJJJAJJFFA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49A0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       \nK00188:59:HMTFHBBXX:2:1101:3772:1666    16      AstrangiaT1FW_65718     4457    1       51M     *       0       0       TTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTGGACATGNG     7JJFFJJAJJF\u0026lt;JJJJFJJJJJJJJ7JFJJ\u0026lt;FFA\u0026lt;JJJFJJJJJJJFFA#\u0026lt;    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:49G1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5051:1666    16      AstrangiaT1FW_131671    3       1       51M     *       0       0       ACAGATGTGCCGCCCCAGCCAAACTCCCAACCTGACGGTGTCTTGGACTNG     JJFJJFJJFJJJJJJFJJJJJFJJJJJJFFJFFJAJJJJJJJJJJJFAA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49C0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5172:1666    0       AstrangiaT1FW_65662     1901    1       51M     *       0       0       TNCGACGGGCGGTGTGTACAAAGGGCAGGGACGTAATCAACGCGAGCTGAT     A#AFFJAJJJJJJJFJFFJJJJFF-AJJJJJJJJJJJJJJJFJJJJJAJJ7    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:0A0G49     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       \nK00188:59:HMTFHBBXX:2:1101:5680:1666    0       AstrangiaT1FW_65668     593     1       51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       GNGCATTGCTGAGTCCATTCGTTTCCCTTTCAACGGTTTCACGTACTTTTT     A#-AFFJFJJ7JJJJJJFFJ\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003eFFFF-FJJFJJJF\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003eAFF7FJJAJFJJ7FJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:6674:1666    16      AstrangiaT1FW_65718     1086    1       23M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       TCCCAGGAGCATGTCTGTCTGNG FF\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003eJJJJAJJJJJJJJJJFFA#A AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0 NM:i:1  MD:Z:21A1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                                                                                       \nK00188:59:HMTFHBBXX:2:1101:7060:1666    0       AstrangiaT1FW_65712     6946    1       51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       CNAAATTTGACGATCGATTTGCACGTCAGAATCGCTACGAGCCTCCACCAG     A#AFFJFFJJJJJFJJJJJJJJJJ\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-FJJJJFJJJJJJJJJJJJJJJJJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:7080:1666    0       AstrangiaT1FW_65718     10568   0       51M     \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e       0       0       TNTTGCAATAGTAATTCTGCTCAGTACGAGAGGAACCGCAGATTCAGACAA     A#AFFJJJFJJJJJJJJJJJ-FJAAAJJJFJJJJJJJJJJJJJJJJJJJJJ    AS:i:-13        XS:i:-19        XN:i:0  XM:i:3  XO:i:0  XG:i:0  NM:i:3  MD:Z:1G16T3G28  YT:Z:UU RG:Z:VA_W_08_SNP     \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"9\" dir=\"auto\"\u003e\n\u003cli\u003eRun get_explain_sam_flags phython script on .sam files\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n[kconf001@coreV3-23-040 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py VA_W_01_22_clippedtrimmed.fastq.sam RI_B_01_18_clippedtrimmed.fastq.sam RI_W_01_14_clippedtrimmed.fastq.sam RI_B_01_22_clippedtrimmed.fastq.sam                                                                                                                                                                       \nVA_W_01_22_clippedtrimmed.fastq.sam\n['0', '4', '16']\n0 :\n4 :\nread unmapped\n16 :\nread reverse strand\nRI_B_01_18_clippedtrimmed.fastq.sam\n['0', '4', '16']\n0 :\n4 :\nread unmapped\n16 :\nread reverse strand\nRI_W_01_14_clippedtrimmed.fastq.sam\n['0', '4', '16']\n0 :\n4 :\nread unmapped\n16 :\nread reverse strand\nRI_B_01_22_clippedtrimmed.fastq.sam\n['0', '4', '16']\n0 :\n4 :\nread unmapped\n16 :        \"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py VA_W_01_22_clippedtrimmed.fastq.sam RI_B_01_18_clippedtrimmed.fastq.sam RI_W_01_14_clippedtrimmed.fastq.sam RI_B_01_22_clippedtrimmed.fastq.sam                                                                                                                                                                       \nVA_W_01_22_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n4 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e unmapped\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\nRI_B_01_18_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n4 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e unmapped\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\nRI_W_01_14_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n4 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e unmapped\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\nRI_B_01_22_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n4 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e unmapped\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e        \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"10\" dir=\"auto\"\u003e\n\u003cli\u003eRun read sorting step fo SNP calling\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCbamsort.sh \n[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCbamsort.sh\n#!/bin/bash -l\n#SBATCH -o KristinaCBamSort.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCBamSort\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\n[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCbamsort.sh\nSubmitted batch job 9276430\"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCbamsort.sh \n[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCbamsort.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCBamSort.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCBamSort\u003c/span\u003e\nenable_lmod\nmodule load samtools/1\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esamtools view -bS \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.sam}\u003c/span\u003e_UNSORTED.bam\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e samtools sort \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\nsamtools index \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCbamsort.sh\nSubmitted batch job 9276430\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e02/12/2021\u003c/h1\u003e\u003ca id=\"user-content-02122021\" class=\"anchor\" aria-label=\"Permalink: 02/12/2021\" href=\"#02122021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 8\u003c/h2\u003e\u003ca id=\"user-content-homework-day-8\" class=\"anchor\" aria-label=\"Permalink: Homework Day 8\" href=\"#homework-day-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eRun the following command on your sprot output file to process into the contig length/match format that trinity examines\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 testassembly]$ nano KristinaCBlastparse.sh\n[kconf001@coreV3-23-040 testassembly]$ cat KristinaCBlastparse.sh\n#!/bin/bash -l\n#SBATCH -o KristinaCBlastparse.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001.odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCBlastparse\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n[kconf001@coreV3-23-040 testassembly]$ sbatch KristinaCBlastparse.sh \nSubmitted batch job 9276467\"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 testassembly]$ nano KristinaCBlastparse.sh\n[kconf001@coreV3-23-040 testassembly]$ cat KristinaCBlastparse.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCBlastparse.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001.odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCBlastparse\u003c/span\u003e\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n[kconf001@coreV3-23-040 testassembly]$ sbatch KristinaCBlastparse.sh \nSubmitted batch job 9276467\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eOutput of sprot\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist\n#hit_pct_cov_bin        count_in_bin    \u0026gt;bin_below\n100     212     212\n90      98      310\n80      104     414\n70      137     551\n60      182     733\n50      256     989\n40      413     1402\n30      719     2121\n20      1440    3561\n10      984     4545\n[kconf001@coreV3-23-040 testassembly]$ grep -c '\u0026gt;' Trinity.fasta\n41303\n[kconf001@coreV3-23-040 testassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta\nThe total number of sequences is 41303\nThe average sequence length is 371\nThe total number of bases is 15326666\nThe minimum sequence length is 184\nThe maximum sequence length is 10771\nThe N50 is 364\nMedian Length = 627\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 5936\"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ehit_pct_cov_bin        count_in_bin    \u0026gt;bin_below\u003c/span\u003e\n100     212     212\n90      98      310\n80      104     414\n70      137     551\n60      182     733\n50      256     989\n40      413     1402\n30      719     2121\n20      1440    3561\n10      984     4545\n[kconf001@coreV3-23-040 testassembly]$ grep -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u0026gt;\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e Trinity.fasta\n41303\n[kconf001@coreV3-23-040 testassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta\nThe total number of sequences is 41303\nThe average sequence length is 371\nThe total number of bases is 15326666\nThe minimum sequence length is 184\nThe maximum sequence length is 10771\nThe N50 is 364\nMedian Length = 627\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 150bp = 0\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 500bp = 5936\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eRm UNSORTED.bam's from your QCFastqs directory (or wherever your .bams and .sams are)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 QCFastqs]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs\n[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCrmsortbam.sh\n[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCrmsortbam.sh\n#!/bin/bash -l\n#SBATCH -o KristinaCrmunsortbam.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCrmunsortbam\nrm *UNSORTED.bam\n[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCrmsortbam.sh\nSubmitted batch job 9276468\"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs\n[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCrmsortbam.sh\n[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCrmsortbam.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCrmunsortbam.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCrmunsortbam\u003c/span\u003e\nrm \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\n[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCrmsortbam.sh\nSubmitted batch job 9276468\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003eRun the following to start genotyping your SNPs for filtering next week\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCfreebayessubref.sh\n[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCfreebayessubref.sh\n#!/bin/bash -l\n#SBATCH -o KristinaCfreebayessubref.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCfreebayessubref\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam \u0026gt; KristinaCmergedfastqs.vcf\n[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCfreebayessubref.sh\nSubmitted batch job 9276565 \"\u003e\u003cpre\u003e[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCfreebayessubref.sh\n[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCfreebayessubref.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCfreebayessubref.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCfreebayessubref\u003c/span\u003e\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.bam \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e KristinaCmergedfastqs.vcf\n[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCfreebayessubref.sh\nSubmitted batch job 9276565 \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e[kconf001@coreV3-23-050 QCFastqs]$ mv *.sam ../../sams/\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\t-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\n```sh\n[kconf001@coreV3-23-050 QCFastqs]$ cd ../../\n[kconf001@coreV3-23-050 data]$ mkdir bams\n[kconf001@coreV3-23-050 data]$ cd fastq/QCFastqs/\n[kconf001@coreV3-23-050 QCFastqs]$ mv *.bam *.bam.bai ../../bams/\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\t-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\n```sh\n[kconf001@coreV3-23-050 QCFastqs]$ cd ../../\n[kconf001@coreV3-23-050 data]$ mkdir bams\n[kconf001@coreV3-23-050 data]$ cd fastq/QCFastqs/\n[kconf001@coreV3-23-050 QCFastqs]$ mv *.bam *.bam.bai ../../bams/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"-rm your _unfiltered.vcf files and .fastq files if you have any\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e-rm your _unfiltered.vcf files and .fastq files if you have any\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 QCFastqs]$ rm *_unfiltered.vcf\n[kconf001@coreV3-23-050 QCFastqs]$ rm *.fastq \n[kconf001@coreV3-23-050 QCFastqs]$ ls -alh\ntotal 8.2M\ndrwxrwxrwx 2 kconf001 users  408 Feb 12 12:56 .\ndrwxrwxrwx 3 kconf001 users  230 Feb 10 08:52 ..\n-rwxrwxrwx 1 kconf001 users   37 Feb  5 14:50 KristinaCalignstats.txt\n-rwxrwxrwx 1 kconf001 users  389 Feb 10 08:26 KristinaCbamsort.sh\n-rwxrwxrwx 1 kconf001 users  946 Feb 10 09:48 KristinaCBamSort.txt\n-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh\n-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt\n-rwxrwxrwx 1 kconf001 users  398 Feb 10 14:20 KristinaCfreebayessubref.sh\n-rwxrwxrwx 1 kconf001 users 1.2K Feb 11 19:14 KristinaCfreebayessubref.txt\n-rwxrwxrwx 1 kconf001 users 5.3M Feb 11 19:14 KristinaCmergedfastqs.vcf\n-rwxrwxrwx 1 kconf001 users  193 Feb 10 12:48 KristinaCrmsortbam.sh\n-rwxrwxrwx 1 kconf001 users    0 Feb 10 12:48 KristinaCrmunsortbam.txt \"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 QCFastqs]$ rm \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_unfiltered.vcf\n[kconf001@coreV3-23-050 QCFastqs]$ rm \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq \n[kconf001@coreV3-23-050 QCFastqs]$ ls -alh\ntotal 8.2M\ndrwxrwxrwx 2 kconf001 users  408 Feb 12 12:56 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 3 kconf001 users  230 Feb 10 08:52 ..\n-rwxrwxrwx 1 kconf001 users   37 Feb  5 14:50 KristinaCalignstats.txt\n-rwxrwxrwx 1 kconf001 users  389 Feb 10 08:26 KristinaCbamsort.sh\n-rwxrwxrwx 1 kconf001 users  946 Feb 10 09:48 KristinaCBamSort.txt\n-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh\n-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt\n-rwxrwxrwx 1 kconf001 users  398 Feb 10 14:20 KristinaCfreebayessubref.sh\n-rwxrwxrwx 1 kconf001 users 1.2K Feb 11 19:14 KristinaCfreebayessubref.txt\n-rwxrwxrwx 1 kconf001 users 5.3M Feb 11 19:14 KristinaCmergedfastqs.vcf\n-rwxrwxrwx 1 kconf001 users  193 Feb 10 12:48 KristinaCrmsortbam.sh\n-rwxrwxrwx 1 kconf001 users    0 Feb 10 12:48 KristinaCrmunsortbam.txt \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e-Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 QCFastqs]$ mkdir ../../vcf\n[kconf001@coreV3-23-050 QCFastqs]$ cd ../../vcf  \"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 QCFastqs]$ mkdir ../../vcf\n[kconf001@coreV3-23-050 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../../vcf  \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003eStart an interactive session via salloc (already done after logging in, see previous assignments for similar workflows)\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003ecp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf\n[kconf001@coreV3-23-050 vcf]$ ls -alh\ntotal 2.1G\ndrwxrwxrwx 2 kconf001 users   54 Feb 12 12:59 .\ndrwxrwxrwx 8 kconf001 users  307 Feb 12 12:57 ..\n-rwxr-xr-x 1 kconf001 users 1.5G Feb 12 13:00 mergedfastq_HEAAstrangiaAssembly.vcf\"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf\n[kconf001@coreV3-23-050 vcf]$ ls -alh\ntotal 2.1G\ndrwxrwxrwx 2 kconf001 users   54 Feb 12 12:59 \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\ndrwxrwxrwx 8 kconf001 users  307 Feb 12 12:57 ..\n-rwxr-xr-x 1 kconf001 users 1.5G Feb 12 13:00 mergedfastq_HEAAstrangiaAssembly.vcf\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003eDetermine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 5.00 seconds\"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 5.00 seconds\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003ecp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf \"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003eRun our host vcf extractor on your merged vcf file using the following syntax:\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvBioinf/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf                               \nRead in ContigList\"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf                               \nRead \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e ContigList\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e7 Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 432676 out of a possible 432676 Sites\nRun Time = 6.00 seconds\n[kconf001@coreV3-23-050 vcf]$\n[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nRun Time = 5.00 seconds   ```sh\n \"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 432676 out of a possible 432676 Sites\nRun Time = 6.00 seconds\n[kconf001@coreV3-23-050 vcf]$\n[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nRun Time = 5.00 seconds   \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e \u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003eWork through the VCF filtering tutorial until the following step:\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 5.00 seconds\n[kconf001@coreV3-23-050 vcf]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--recode-INFO-all\n--mac 3\n--minQ 30\n--max-missing 0.5\n--out raw.g5mac3\n--recode\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 432676 Sites\nRun Time = 26.00 seconds\n[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf raw.g5mac3.recode.vcf\n--recode-INFO-all\n--minDP 3\n--out raw.g5mac3dp3\n--recode\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 24.00 seconds\n[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf raw.g5mac3dp3.recode.vcf\n--missing-indv\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 3.00 seconds\n[kconf001@coreV3-23-050 vcf]$ bash\nkconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ mawk '!/IN/' out.imiss | cut -f5 \u0026gt; totalmissing\nkconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ gnuplot \u0026lt;\u0026lt; \\EOF\n\u0026gt; set terminal dumb size 120, 30\n\u0026gt; set autoscale\n\u0026gt; unset label\n\u0026gt; set title \u0026quot;Histogram of % missing data per individual\u0026quot;\n\u0026gt; set ylabel \u0026quot;Number of Occurrences\u0026quot;\n\u0026gt; set xlabel \u0026quot;% of missing data\u0026quot;\n\u0026gt; #set yr [0:100000]\n\u0026gt; binwidth=0.01\n\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\n\u0026gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes\n\u0026gt; pause -1\n\u0026gt; EOF\nâ \nHistogram of % missing data per individual\nNumber of Occurrences\n3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++ \n+           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +\n|                                                                * *\n|                                                                                          \n|                                                                * *                                         \n|                                                                                          \n|                                                                * *                                         \n|                                                                                          \n|                                                                * *                                         \n|                                                                                      \n2.5 ++                                                           * *                                          ++\n|                                                                * *                                         \n|                                                                                          \n|                                                                * *                                         \n|                                                                                          \n|                                                                * *                                         \n|                                                                                          \n|                                                                * *                                         \n|                                                                                          \n|                                                                * *                                         \n|                                                                                        \n2 ++                                                     **      * * ***  ******    ** ****   ****             ++\n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                      \n1.5 ++                                                   **      * * * *  * *  *    ** ** *   *  *            ++\n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n|                                                        **      * * * *  * *  *    ** ** *   *  *           \n|                                                                                          \n+           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *             +\n1 *********************************************************************************************************---++\n0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\n% of missing data\"\u003e\u003cpre\u003e[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 5.00 seconds\n[kconf001@coreV3-23-050 vcf]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--recode-INFO-all\n--mac 3\n--minQ 30\n--max-missing 0.5\n--out raw.g5mac3\n--recode\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 432676 Sites\nRun Time = 26.00 seconds\n[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf raw.g5mac3.recode.vcf\n--recode-INFO-all\n--minDP 3\n--out raw.g5mac3dp3\n--recode\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 24.00 seconds\n[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf raw.g5mac3dp3.recode.vcf\n--missing-indv\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 3.00 seconds\n[kconf001@coreV3-23-050 vcf]$ bash\nkconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ mawk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e!/IN/\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e out.imiss \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -f5 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e totalmissing\nkconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ gnuplot \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u0026lt;\u003c/span\u003e \\\u003cspan class=\"pl-k\"\u003eEOF\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set terminal dumb size 120, 30\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set autoscale\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; unset label\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set title \"Histogram of % missing data per individual\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set ylabel \"Number of Occurrences\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set xlabel \"% of missing data\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; #set yr [0:100000]\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; binwidth=0.01\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; plot 'totalmissing' using (bin(\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e,binwidth)):(1.0) smooth freq with boxes\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; pause -1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; EOF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eâ \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eHistogram of % missing data per individual\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eNumber of Occurrences\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++ \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e+           +           +           +           +       'totalmissing' using (bin(\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e,binwidth)):(1.0) ****** +\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                      \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e2.5 ++                                                           * *                                          ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                * *                                         \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                        \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e2 ++                                                     **      * * ***  ******    ** ****   ****             ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                      \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e1.5 ++                                                   **      * * * *  * *  *    ** ** *   *  *            ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                        **      * * * *  * *  *    ** ** *   *  *           \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e|                                                                                          \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e+           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *             +\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e1 *********************************************************************************************************---++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e% of missing data\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNow that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003evcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e02/17/2021\u003c/h1\u003e\u003ca id=\"user-content-02172021\" class=\"anchor\" aria-label=\"Permalink: 02/17/2021\" href=\"#02172021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 09\u003c/h2\u003e\u003ca id=\"user-content-homework-day-09\" class=\"anchor\" aria-label=\"Permalink: Homework Day 09\" href=\"#homework-day-09\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e1-  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters\n##Half the class run the following /cm/shared/apps/vcftools/0.1.12b/bin/vcftools\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--maf 0.015\n--max-alleles 2\n--max-missing 0.5\n--minQ 30\n--minGQ 20\n--minDP 3\n--remove-indels\n--hwe 0.01\n--recode\n--recode-INFO-all\n--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf \n--max-missing 0.5 \n--mac 3 \n--minQ 30 \n--minDP 10 \n--max-alleles 2 \n--maf 0.015 \n--remove-indels \n--recode \n--recode-INFO-all \n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 18718 out of a possible 432676 Sites\nRun Time = 11.00 seconds \"\u003e\u003cpre\u003e[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf \n--max-missing 0.5 \n--mac 3 \n--minQ 30 \n--minDP 10 \n--max-alleles 2 \n--maf 0.015 \n--remove-indels \n--recode \n--recode-INFO-all \n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 18718 out of a possible 432676 Sites\nRun Time = 11.00 seconds \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e##the other half run the filters we used from the paper\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--max-missing 0.5\n--mac 3\n--minQ 30\n--minDP 10\n--max-alleles 2\n--maf 0.015\n--remove-indels\n--recode\n--recode-INFO-all\n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf \n--max-missing 0.5 \n--mac 3 \n--minQ 30 \n--minDP 10 \n--max-alleles 2 \n--maf 0.015 \n--remove-indels \n--recode \n--recode-INFO-all \n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--recode-INFO-all\n--mac 3\n--maf 0.015\n--max-alleles 2\n--minDP 10\n--minQ 30\n--max-missing 0.5\n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n--recode\n--remove-indels\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 1578 out of a possible 432676 Sites\nRun Time = 5.00 seconds  \"\u003e\u003cpre\u003e[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf \n--max-missing 0.5 \n--mac 3 \n--minQ 30 \n--minDP 10 \n--max-alleles 2 \n--maf 0.015 \n--remove-indels \n--recode \n--recode-INFO-all \n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\nParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--recode-INFO-all\n--mac 3\n--maf 0.015\n--max-alleles 2\n--minDP 10\n--minQ 30\n--max-missing 0.5\n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n--recode\n--remove-indels\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 1578 out of a possible 432676 Sites\nRun Time = 5.00 seconds  \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e2- Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-002 vcf]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf  \u0026gt; samplenames.txt\n[kconf001@coreV3-23-002 vcf]$ cat samplenames.txt\n#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  \nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n[kconf001@coreV3-23-002 vcf]$ nano samplenames.txt \"\u003e\u003cpre\u003e[kconf001@coreV3-23-002 vcf]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e#CHROM\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e mergedfastq_HEAAstrangiaAssembly_subset.vcf  \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e samplenames.txt\n[kconf001@coreV3-23-002 vcf]$ cat samplenames.txt\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  \u003c/span\u003e\nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n[kconf001@coreV3-23-002 vcf]$ nano samplenames.txt \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNeed to edit the file on excel or use sed 's/\\s+/\\n/g' samplenames. txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eor use .imiss file (the one I used)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-002 vcf]$ cat out.imiss\nINDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS\nRI_W_06_merged  99853   0       67075   0.671737\nRI_W_07_merged  99853   0       63917   0.640111\nRI_W_02_merged  99853   0       73105   0.732126\nRI_W_04_merged  99853   0       71159   0.712638\nVA_W_09_SNP_clipped     99853   0       19459   0.194876\nRI_B_08_SNP_clipped     99853   0       88110   0.882397\nVA_W_08_SNP_clipped     99853   0       83344   0.834667\nVA_B_08_SNP_clipped     99853   0       94221   0.943597\nVA_W_02_merged  99853   0       71050   0.711546\nVA_B_07_merged  99853   0       62687   0.627793\nRI_B_05_merged  99853   0       43945   0.440097\nVA_W_06_merged  99853   0       65164   0.652599\nVA_W_04_merged  99853   0       50911   0.509859\nVA_W_01_merged  99853   0       67150   0.672489\nVA_B_10_SNP_clipped     99853   0       82831   0.829529\nVA_B_06_merged  99853   0       57724   0.57809\nVA_W_05_merged  99853   0       66297   0.663946\nRI_B_09_SNP_clipped     99853   0       85107   0.852323\nVA_W_10_SNP_clipped     99853   0       83234   0.833565\nRI_W_08_SNP_clipped     99853   0       77802   0.779165\nRI_B_06_merged  99853   0       79766   0.798834\nRI_W_10_SNP_clipped     99853   0       91323   0.914574\nRI_B_04_merged  99853   0       46717   0.467858\nVA_W_03_merged  99853   0       64016   0.641102\nRI_B_07_merged  99853   0       70798   0.709022\nRI_W_05_merged  99853   0       57117   0.572011\nRI_W_09_SNP_clipped     99853   0       88492   0.886223\nVA_B_01_merged  99853   0       47217   0.472865\nVA_B_09_SNP_clipped     99853   0       79187   0.793036\nRI_B_10_SNP_clipped     99853   0       82779   0.829009\nRI_W_01_merged  99853   0       73169   0.732767\nRI_B_01_merged  99853   0       69619   0.697215\nVA_B_04_merged  99853   0       58190   0.582757\nRI_B_02_merged  99853   0       87439   0.875677\nRI_W_03_merged  99853   0       38760   0.388171\nVA_B_02_merged  99853   0       78381   0.784964\nVA_W_07_merged  99853   0       64592   0.646871\nVA_B_05_merged  99853   0       54401   0.544811\nRI_B_03_merged  99853   0       80369   0.804873\n\n[kconf001@coreV3-23-002 vcf]$ cut -f 1 out.imiss | tail -n +2\nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\"\u003e\u003cpre\u003e[kconf001@coreV3-23-002 vcf]$ cat out.imiss\nINDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS\nRI_W_06_merged  99853   0       67075   0.671737\nRI_W_07_merged  99853   0       63917   0.640111\nRI_W_02_merged  99853   0       73105   0.732126\nRI_W_04_merged  99853   0       71159   0.712638\nVA_W_09_SNP_clipped     99853   0       19459   0.194876\nRI_B_08_SNP_clipped     99853   0       88110   0.882397\nVA_W_08_SNP_clipped     99853   0       83344   0.834667\nVA_B_08_SNP_clipped     99853   0       94221   0.943597\nVA_W_02_merged  99853   0       71050   0.711546\nVA_B_07_merged  99853   0       62687   0.627793\nRI_B_05_merged  99853   0       43945   0.440097\nVA_W_06_merged  99853   0       65164   0.652599\nVA_W_04_merged  99853   0       50911   0.509859\nVA_W_01_merged  99853   0       67150   0.672489\nVA_B_10_SNP_clipped     99853   0       82831   0.829529\nVA_B_06_merged  99853   0       57724   0.57809\nVA_W_05_merged  99853   0       66297   0.663946\nRI_B_09_SNP_clipped     99853   0       85107   0.852323\nVA_W_10_SNP_clipped     99853   0       83234   0.833565\nRI_W_08_SNP_clipped     99853   0       77802   0.779165\nRI_B_06_merged  99853   0       79766   0.798834\nRI_W_10_SNP_clipped     99853   0       91323   0.914574\nRI_B_04_merged  99853   0       46717   0.467858\nVA_W_03_merged  99853   0       64016   0.641102\nRI_B_07_merged  99853   0       70798   0.709022\nRI_W_05_merged  99853   0       57117   0.572011\nRI_W_09_SNP_clipped     99853   0       88492   0.886223\nVA_B_01_merged  99853   0       47217   0.472865\nVA_B_09_SNP_clipped     99853   0       79187   0.793036\nRI_B_10_SNP_clipped     99853   0       82779   0.829009\nRI_W_01_merged  99853   0       73169   0.732767\nRI_B_01_merged  99853   0       69619   0.697215\nVA_B_04_merged  99853   0       58190   0.582757\nRI_B_02_merged  99853   0       87439   0.875677\nRI_W_03_merged  99853   0       38760   0.388171\nVA_B_02_merged  99853   0       78381   0.784964\nVA_W_07_merged  99853   0       64592   0.646871\nVA_B_05_merged  99853   0       54401   0.544811\nRI_B_03_merged  99853   0       80369   0.804873\n\n[kconf001@coreV3-23-002 vcf]$ cut -f 1 out.imiss \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e tail -n +2\nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethen edit on Excel (seperate columns, =LEFT(A1,4), then paste to a popfile.txt\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-002 vcf]$ nano popfile.txt\"\u003e\u003cpre\u003e[kconf001@coreV3-23-002 vcf]$ nano popfile.txt\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e3- Convert your filtered .vcf file to genepop format using the following command:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV3-23-002 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile.txt\nIndivs with genotypes in vcf file: \nRI_W_06_merged  \nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged    \nRI_W_05_merged\nRI_W_09_SNP_clipped  \nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n44 \n1578 \n1578 \n1578 \n1578 \n40 \"\u003e\u003cpre\u003e[kconf001@coreV3-23-002 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile.txt\nIndivs with genotypes \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e vcf file: \nRI_W_06_merged  \nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged    \nRI_W_05_merged\nRI_W_09_SNP_clipped  \nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n44 \n1578 \n1578 \n1578 \n1578 \n40 \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e4- SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/         \nkconf001@turing.hpc.odu.edu's password:\n1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\n100%  343KB   2.5MB/s   00:00    \"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/         \nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e100%  343KB   2.5MB/s   00:00    \u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e5- Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eIn r-studio:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\u0026gt; setwd(\u0026quot;~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog\u0026quot;)\n\u0026gt; datafile\u0026lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)\nConverting data from a Genepop .gen file to a genind object... \nFile description:  AllSNPs \n...done.\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e setwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e datafile\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-read.genepop(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, ncode=2)\nConverting data from a Genepop .gen file to a genind object... \nFile description:  AllSNPs \n...done.\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eadegenet_PCAs.R script\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(\u0026quot;ape\u0026quot;)\nlibrary(\u0026quot;pegas\u0026quot;)\nlibrary(\u0026quot;seqinr\u0026quot;)\nlibrary(\u0026quot;ggplot2\u0026quot;)\nlibrary(\u0026quot;adegenet\u0026quot;)\nsetwd(\u0026quot;~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio\u0026quot;)\ndatafile\u0026lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)\nsum(is.na(datafile$tab))\ndatafile #shows info\nYOURdata\u0026lt;-scaleGen(datafile, NA.method='mean')\nX\u0026lt;-YOURdata\nY\u0026lt;-as.factor(substring(pop(datafile),1,4))\npca1 \u0026lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\n\n#### PCAs ####\n# individual labels\n# s.label(pca1$li)\n\n# population elipses\ns.class(pca1$li, pop(datafile))\n\n#color symbols, pop names\n#pdf(\u0026quot;YOURINITIALS_ColorPCA1v2.pdf\u0026quot;)\ncol \u0026lt;- c(\u0026quot;blue\u0026quot;,\u0026quot;red\u0026quot;, \u0026quot;green\u0026quot;, \u0026quot;black\u0026quot;)\ns.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\u0026quot;topright\u0026quot;)\ntitle(\u0026quot;PCA of DJB_data\\naxes 1-2\u0026quot;)\n#dev.off()\n\n#### Snapclust ####\na.clust\u0026lt;-snapclust(datafile, k = 2)\nclass(a.clust)\nnames(a.clust)\na.tab \u0026lt;- table(pop(datafile), a.clust$group)\ntable.value(a.tab, col.labels = 1:2)\n\ncompoplot(a.clust)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eape\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epegas\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eseqinr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eggplot2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eadegenet\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nsetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndatafile\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-read.genepop(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, ncode=2)\nsum(is.na(datafile\u003cspan class=\"pl-smi\"\u003e$tab\u003c/span\u003e))\ndatafile \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eshows info\u003c/span\u003e\nYOURdata\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-scaleGen(datafile, NA.method=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emean\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\nX\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-YOURdata\nY\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-as.factor(substring(pop(datafile),1,4))\npca1 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### PCAs ####\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e individual labels\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e s.label(pca1$li)\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e population elipses\u003c/span\u003e\ns.class(pca1\u003cspan class=\"pl-smi\"\u003e$li\u003c/span\u003e, pop(datafile))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecolor symbols, pop names\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epdf(\"YOURINITIALS_ColorPCA1v2.pdf\")\u003c/span\u003e\ncol \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ns.class(pca1\u003cspan class=\"pl-smi\"\u003e$li\u003c/span\u003e, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca1\u003cspan class=\"pl-smi\"\u003e$eig\u003c/span\u003e[1:3], 3,1,2, posi=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopright\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ntitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePCA of DJB_data\\naxes 1-2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003edev.off()\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### Snapclust ####\u003c/span\u003e\na.clust\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-snapclust(datafile, k = 2)\nclass(a.clust)\nnames(a.clust)\na.tab \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- table(pop(datafile), a.clust\u003cspan class=\"pl-smi\"\u003e$group\u003c/span\u003e)\ntable.value(a.tab, col.labels = 1:2)\n\ncompoplot(a.clust)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e02/19/2021\u003c/h1\u003e\u003ca id=\"user-content-02192021\" class=\"anchor\" aria-label=\"Permalink: 02/19/2021\" href=\"#02192021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 10\u003c/h2\u003e\u003ca id=\"user-content-homework-day-10\" class=\"anchor\" aria-label=\"Permalink: Homework Day 10\" href=\"#homework-day-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e1- Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e###run step 2 as an sbatch .sh script because it will take a while to finish (Didn't run an sbatch)\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e2- cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~ ssh kconf001@turing.hpc.odu.edu\nkconf001@turing.hpc.odu.edu's password:\nLast login: Wed Feb 17 08:47:42 2021 from ip70-160-48-140.hr.hr.cox.net\n[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams\n[kconf001@turing1 sams]$ salloc\nsalloc: Pending job allocation 9280868\nsalloc: job 9280868 queued and waiting for resources\nsalloc: job 9280868 has been allocated resources\nsalloc: Granted job allocation 9280868\n[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam \"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e ssh kconf001@turing.hpc.odu.edu\nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eLast login: Wed Feb 17 08:47:42 2021 from ip70-160-48-140.hr.hr.cox.net\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@turing1 sams]$ salloc\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: Pending job allocation 9280868\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: job 9280868 queued and waiting for resources\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: job 9280868 has been allocated resources\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: Granted job allocation 9280868\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam \u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e3- Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KristinaCFullCounts_summed.txt NoMatch *_counts.txt\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KristinaCFullCounts_summed.txt NoMatch *_counts.txt\nHits not matched\nRI_B_01_14_clippedtrimmed.fastq_counts.txt=13381\nRI_B_01_18_clippedtrimmed.fastq_counts.txt=13381\nRI_B_01_22_clippedtrimmed.fastq_counts.txt=13381\nRI_B_08_SNP_clippedtrimmed.fastq_counts.txt=13381\nRI_W_01_14_clippedtrimmed.fastq_counts.txt=13381\nRI_W_01_18_clippedtrimmed.fastq_counts.txt=13381\nRI_W_01_22_clippedtrimmed.fastq_counts.txt=13381\nRI_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381\nVA_B_01_14_clippedtrimmed.fastq_counts.txt=13381\nVA_B_01_18_clippedtrimmed.fastq_counts.txt=13381\nVA_B_01_22_clippedtrimmed.fastq_counts.txt=13381\nVA_B_09_SNP_clippedtrimmed.fastq_counts.txt=13381\nVA_W_01_14_clippedtrimmed.fastq_counts.txt=13381\nVA_W_01_18_clippedtrimmed.fastq_counts.txt=13381\nVA_W_01_22_clippedtrimmed.fastq_counts.txt=13381\nVA_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381 \"\u003e\u003cpre\u003e[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KristinaCFullCounts_summed.txt NoMatch \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_counts.txt\nHits not matched\nRI_B_01_14_clippedtrimmed.fastq_counts.txt=13381\nRI_B_01_18_clippedtrimmed.fastq_counts.txt=13381\nRI_B_01_22_clippedtrimmed.fastq_counts.txt=13381\nRI_B_08_SNP_clippedtrimmed.fastq_counts.txt=13381\nRI_W_01_14_clippedtrimmed.fastq_counts.txt=13381\nRI_W_01_18_clippedtrimmed.fastq_counts.txt=13381\nRI_W_01_22_clippedtrimmed.fastq_counts.txt=13381\nRI_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381\nVA_B_01_14_clippedtrimmed.fastq_counts.txt=13381\nVA_B_01_18_clippedtrimmed.fastq_counts.txt=13381\nVA_B_01_22_clippedtrimmed.fastq_counts.txt=13381\nVA_B_09_SNP_clippedtrimmed.fastq_counts.txt=13381\nVA_W_01_14_clippedtrimmed.fastq_counts.txt=13381\nVA_W_01_18_clippedtrimmed.fastq_counts.txt=13381\nVA_W_01_22_clippedtrimmed.fastq_counts.txt=13381\nVA_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381 \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e4- scp YOURNAMEFullCounts_summed.txt to your laptop\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams/KristinaCFullCounts_summed.txt /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/\nkconf001@turing.hpc.odu.edu's password:\nKristinaCFullCounts_summed.txt\n100% 1972KB   3.4MB/s   00:00 \"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams/KristinaCFullCounts_summed.txt /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/\nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKristinaCFullCounts_summed.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e100% 1972KB   3.4MB/s   00:00 \u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e5- edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eFile had no match???\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e02/24/2021\u003c/h1\u003e\u003ca id=\"user-content-02242021\" class=\"anchor\" aria-label=\"Permalink: 02/24/2021\" href=\"#02242021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 11\u003c/h2\u003e\u003ca id=\"user-content-homework-day-11\" class=\"anchor\" aria-label=\"Permalink: Homework Day 11\" href=\"#homework-day-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e1 . Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"### This is an R script that implements the program DESeq2 for gene expression analysis.\n### Much more information on the program and specific function (particularly for checking quality) \n### can be found here: https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html\n### The elements of this script were written by Melissa Pespeni and Daniel Barshis.\n\n#Only need to do this the first time to install the package\n#for R version 4.0\n#if (!requireNamespace(\u0026quot;BiocManager\u0026quot;, quietly = TRUE))\n#  install.packages(\u0026quot;BiocManager\u0026quot;)\n\n#BiocManager::install(\u0026quot;DESeq2\u0026quot;)\n\n#a#for older versions of R\n#source(\u0026quot;https://bioconductor.org/biocLite.R\u0026quot;)\n#biocLite(\u0026quot;DESeq2\u0026quot;)\n\nlibrary(DESeq2)\nlibrary(gplots)\n\nvignette('DESeq2')\n\nsetwd(\u0026quot;Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio/Day11\u0026quot;)  # The drag and drop from finder works in R, too.\n\n#useful functions\n#head() - prints out the top 6 lines\n#dim() - prints the dimensions of a variable\n#nrow() - returns the number of rows in a vector or matrix\n# ?[functionName] - opens documentation describing the function\n\n#read in your data to make counts table\ncountsTable \u0026lt;- read.delim('djbFullCounts_summed.txt', header=TRUE, stringsAsFactors=TRUE, row.names=1)\nhead(countsTable)\ndim(countsTable)\ncolSums(countsTable)\ncolSums(countsTable)[order(colSums(countsTable))]\nLowCounts\u0026lt;-c(\u0026quot;VA_W_02_22\u0026quot;,\u0026quot;VA_W_01_22\u0026quot;,\u0026quot;RI_B_03_18\u0026quot;)\nnames(countsTable)%in%LowCounts\n!(names(countsTable)%in%LowCounts)\ncountsTable\u0026lt;-countsTable[,!(names(countsTable)%in%LowCounts)]\ndim(countsTable)\n\n#make a table with conditions of each individual (e.g. \u0026quot;VA\u0026quot; and \u0026quot;RI\u0026quot;)  There should be the same number of conditions described as there are samples in your data file, and in the same order.\n#NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.\n#Here's an example\n# Sample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp\n# RI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18\n# VA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14\n# VA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22\n# safest way to do this is in R\n\nconds\u0026lt;-data.frame(\u0026quot;Sample\u0026quot;=names(countsTable))\nconds$Origin\u0026lt;-factor(substring(conds$Sample,1,2))\nconds$SymState\u0026lt;-factor(substring(conds$Sample,4,4))\nconds$Temp\u0026lt;-factor(substring(conds$Sample,9,10))\nconds$Origin_SymState\u0026lt;-factor(paste(conds$Origin,conds$SymState, sep=\u0026quot;_\u0026quot;))\nconds$Origin_SymState_Temp\u0026lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=\u0026quot;_\u0026quot;))\n\n##Check null counts per sample\nprop.null \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\n\nbarplot(prop.null, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\npdf(file=\u0026quot;OverallNullCounts.pdf\u0026quot;,14, 14)\nbarplot(prop.null, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\ndev.off()\n\n# how many genes have mean count \u0026gt;=3 \nmeans = apply(countsTable, 1, mean)\ntable(means\u0026gt;=3)\n\ncountsTable\u0026lt;-countsTable[means\u0026gt;=3,]\ndim(countsTable)\n\nprop.nullv2 \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\npdf(file=\u0026quot;Mean3NullCounts.pdf\u0026quot;,14, 14)\nbarplot(prop.nullv2, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\ndev.off()\n\n#make count data set\ndds \u0026lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin_SymState_Temp)\n\ndds \u0026lt;- DESeq(dds)\n\n#transform counts to variance stablized counts\nvstCounts\u0026lt;-varianceStabilizingTransformation(dds)\n\n#plot cluster dendrogram across samples\ndists\u0026lt;-dist(t(assay(vstCounts)))\nplot(hclust(dists))\n\nplotPCA(vstCounts, intgroup=\u0026quot;Origin\u0026quot;)\nplotPCA(vstCounts, intgroup=\u0026quot;SymState\u0026quot;)\nplotPCA(vstCounts, intgroup=\u0026quot;Temp\u0026quot;)\nplotPCA(vstCounts, intgroup=\u0026quot;Origin_SymState\u0026quot;)\nplotPCA(vstCounts, intgroup=\u0026quot;Origin_SymState_Temp\u0026quot;)\n\nresultsNames(dds)\n\ndim(dds)\n\n###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)\nres \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_B_18\u0026quot;, \u0026quot;VA_B_18\u0026quot;))\nhead(res)\nsummary(res)\n\nres \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_W_18\u0026quot;, \u0026quot;VA_W_18\u0026quot;))\nsummary(res)\n\nres \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_B_14\u0026quot;, \u0026quot;VA_B_14\u0026quot;))\nsummary(res)\n\nres \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_W_14\u0026quot;, \u0026quot;VA_W_14\u0026quot;))\nsummary(res)\n\nres \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_B_22\u0026quot;, \u0026quot;VA_B_22\u0026quot;))\nsummary(res)\n\nres \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_W_22\u0026quot;, \u0026quot;VA_W_22\u0026quot;))\nsummary(res)\n\nhead(res[order(res$padj),])\nplotCounts(dds, \u0026quot;TR2367|c0_g1_i1_coral\u0026quot;, intgroup=\u0026quot;Origin_SymState_Temp\u0026quot;)\n\n#count the number of significantly differentially expressed genes\nsum(res$padj \u0026lt; 0.3, na.rm =T)\nsum(res$padj \u0026lt; 0.2, na.rm =T)\nsum(res$padj \u0026lt; 0.1, na.rm =T)\nsum(res$pvalue \u0026lt; 0.05, na.rm =T)\n\t\t\t\t\t\t\t\t\t\t\t \n# Make a counts table that is scaled by the size factors\nscaledcounts = counts(dds, normalized=T)\n\n#building heat map data\nhead(scaledcounts)\ngenes4heatmap\u0026lt;-res[res$pvalue \u0026lt;0.05 \u0026amp; !is.na(res$pvalue),]\nnames(genes4heatmap)\nhead(genes4heatmap)\ndim(genes4heatmap)\ndata4heatmap\u0026lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\ndim(data4heatmap)\nhead(data4heatmap)\n\ntemp = as.matrix(rowMeans(data4heatmap))\nhead(temp)\nscaledmatrix\u0026lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\ndata4heatmapscaled = data4heatmap/scaledmatrix\nhead(data4heatmapscaled)\n\ndim(data4heatmapscaled)\n\npairs.breaks \u0026lt;- seq(0, 3.0, by=0.1)\nlength(pairs.breaks)\nmycol \u0026lt;- colorpanel(n=30, low=\u0026quot;black\u0026quot;, high=\u0026quot;yellow\u0026quot;) \n\npdf(file=\u0026quot;XXXXXX_byrow.pdf\u0026quot;,14,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u0026quot;row\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\ndev.off()\n\npdf(file=\u0026quot;XXXXXX_bycolumn.pdf\u0026quot;,14,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u0026quot;col\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\ndev.off()\n\n\n###PCAs\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin\u0026quot;)\n\nprop.nullv3 \u0026lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))\npdf(file=\u0026quot;ResNullCounts.pdf\u0026quot;,14, 14)\nbarplot(prop.nullv3, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## This is an R script that implements the program DESeq2 for gene expression analysis.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Much more information on the program and specific function (particularly for checking quality) \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## can be found here: https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## The elements of this script were written by Melissa Pespeni and Daniel Barshis.\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eOnly need to do this the first time to install the package\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efor R version 4.0\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eif (!requireNamespace(\"BiocManager\", quietly = TRUE))\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e  install.packages(\"BiocManager\")\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eBiocManager::install(\"DESeq2\")\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ea#for older versions of R\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esource(\"https://bioconductor.org/biocLite.R\")\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ebiocLite(\"DESeq2\")\u003c/span\u003e\n\nlibrary(DESeq2)\nlibrary(gplots)\n\nvignette(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eDESeq2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\nsetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDesktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio/Day11\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e The drag and drop from finder works in R, too.\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003euseful functions\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ehead() - prints out the top 6 lines\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003edim() - prints the dimensions of a variable\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003enrow() - returns the number of rows in a vector or matrix\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ?[functionName] - opens documentation describing the function\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eread in your data to make counts table\u003c/span\u003e\ncountsTable \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- read.delim(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003edjbFullCounts_summed.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, header=TRUE, stringsAsFactors=TRUE, row.names=1)\nhead(countsTable)\ndim(countsTable)\ncolSums(countsTable)\ncolSums(countsTable)[order(colSums(countsTable))]\nLowCounts\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_02_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_01_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_03_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nnames(countsTable)%in%LowCounts\n\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e(names(countsTable)%in%LowCounts)\ncountsTable\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-countsTable[,\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e(names(countsTable)%in%LowCounts)]\ndim(countsTable)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emake a table with conditions of each individual (e.g. \"VA\" and \"RI\")  There should be the same number of conditions described as there are samples in your data file, and in the same order.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eNOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eHere's an example\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Sample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e RI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e VA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e VA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e safest way to do this is in R\u003c/span\u003e\n\nconds\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-data.frame(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e=names(countsTable))\nconds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,1,2))\nconds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,4,4))\nconds\u003cspan class=\"pl-smi\"\u003e$Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,9,10))\nconds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(paste(conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e, sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nconds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(paste(conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$Temp\u003c/span\u003e, sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Check null counts per sample\u003c/span\u003e\nprop.null \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable, 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\n\nbarplot(prop.null, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\npdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOverallNullCounts.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\nbarplot(prop.null, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e how many genes have mean count \u0026gt;=3 \u003c/span\u003e\nmeans = apply(countsTable, 1, mean)\ntable(means\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e=3)\n\ncountsTable\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-countsTable[means\u003cspan class=\"pl-k\"\u003e\u0026gt;=\u003c/span\u003e3,]\ndim(countsTable)\n\nprop.nullv2 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable, 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\npdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMean3NullCounts.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\nbarplot(prop.nullv2, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emake count data set\u003c/span\u003e\ndds \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e Origin_SymState_Temp)\n\ndds \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- DESeq(dds)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etransform counts to variance stablized counts\u003c/span\u003e\nvstCounts\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-varianceStabilizingTransformation(dds)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplot cluster dendrogram across samples\u003c/span\u003e\ndists\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-dist(t(assay(vstCounts)))\nplot(hclust(dists))\n\nplotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSymState\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nplotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nresultsNames(dds)\n\ndim(dds)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##Figure out which contrast you want to examine (i.e. which two groups do you want to compare)\u003c/span\u003e\nres \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_B_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nhead(res)\nsummary(res)\n\nres \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nsummary(res)\n\nres \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nsummary(res)\n\nres \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nsummary(res)\n\nres \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_B_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nsummary(res)\n\nres \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\nsummary(res)\n\nhead(res[order(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e),])\nplotCounts(dds, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTR2367|c0_g1_i1_coral\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecount the number of significantly differentially expressed genes\u003c/span\u003e\nsum(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.3, na.rm =T)\nsum(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.2, na.rm =T)\nsum(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1, na.rm =T)\nsum(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.05, na.rm =T)\n\t\t\t\t\t\t\t\t\t\t\t \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Make a counts table that is scaled by the size factors\u003c/span\u003e\nscaledcounts = counts(dds, normalized=T)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ebuilding heat map data\u003c/span\u003e\nhead(scaledcounts)\ngenes4heatmap\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-res[res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0.05 \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eis.na(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e),]\nnames(genes4heatmap)\nhead(genes4heatmap)\ndim(genes4heatmap)\ndata4heatmap\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\ndim(data4heatmap)\nhead(data4heatmap)\n\ntemp = as.matrix(rowMeans(data4heatmap))\nhead(temp)\nscaledmatrix\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\ndata4heatmapscaled = data4heatmap/scaledmatrix\nhead(data4heatmapscaled)\n\ndim(data4heatmapscaled)\n\npairs.breaks \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- seq(0, 3.0, by=0.1)\nlength(pairs.breaks)\nmycol \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- colorpanel(n=30, low=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, high=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \n\npdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eXXXXXX_byrow.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\n\npdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eXXXXXX_bycolumn.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecol\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##PCAs\u003c/span\u003e\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\nprop.nullv3 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\npdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eResNullCounts.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\nbarplot(prop.nullv3, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        horiz=TRUE, cex.names=0.5, las=1, \n        col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e03/03/2021\u003c/h1\u003e\u003ca id=\"user-content-03032021\" class=\"anchor\" aria-label=\"Permalink: 03/03/2021\" href=\"#03032021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHomework Day 13 (last day!!)\u003c/h2\u003e\u003ca id=\"user-content-homework-day-13-last-day\" class=\"anchor\" aria-label=\"Permalink: Homework Day 13 (last day!!)\" href=\"#homework-day-13-last-day\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003ecreate an sbatch script to rm -r your sandbox/YOURNAME directory\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-007 sandboxes]$ nano KristinaCRemoveSandbox.sh\n[kconf001@coreV2-25-007 sandboxes]$ cat KristinaCRemoveSandbox.sh\n#!/bin/bash -l\n#SBATCH -o KristinaCRemoveSandbox.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kconf001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KristinaCRemoveSandbox\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC\n\"\u003e\u003cpre\u003e[kconf001@coreV2-25-007 sandboxes]$ nano KristinaCRemoveSandbox.sh\n[kconf001@coreV2-25-007 sandboxes]$ cat KristinaCRemoveSandbox.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KristinaCRemoveSandbox.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kconf001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KristinaCRemoveSandbox\u003c/span\u003e\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC\n\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003esubmit your sbatch script and add the entry to your logfile/readme\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"[kconf001@coreV2-25-007 sandboxes]$ sbatch KristinaCRemoveDirectory.sh\n[kconf001@coreV2-25-007 sandboxes]$ squeue -u kconf001\n[kconf001@coreV2-25-007 sandboxes]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes\"\u003e\u003cpre\u003e[kconf001@coreV2-25-007 sandboxes]$ sbatch KristinaCRemoveDirectory.sh\n[kconf001@coreV2-25-007 sandboxes]$ squeue -u kconf001\n[kconf001@coreV2-25-007 sandboxes]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003escp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"$ scp KristinaCREADME.md kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive\"\u003e\u003cpre\u003e$ scp KristinaCREADME.md kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive\u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003escp the following files to your local computer (copied directly to AdvancedGeomicsAnalysis 21SpKristinaCAdvancedGenomicsLog)\nscp \u003ca href=\"mailto:YOURMIDASID@turing.hpc.odu.edu\"\u003eYOURMIDASID@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./\nscp \u003ca href=\"mailto:YOURMIDASID@turing.hpc.odu.edu\"\u003eYOURMIDASID@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./\nscp \u003ca href=\"mailto:YOURMIDASID@turing.hpc.odu.edu\"\u003eYOURMIDASID@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./\nscp \u003ca href=\"mailto:YOURMIDASID@turing.hpc.odu.edu\"\u003eYOURMIDASID@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./\nkconf001@turing.hpc.odu.edu's password:\n15079_Apoc_hostsym.fasta\n100%   13MB   5.7MB/s   00:02\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./\nkconf001@turing.hpc.odu.edu's password:\n15079_Apoc_hostsym.fasta.fai\n100%  641KB   3.8MB/s   00:00 \nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./\nkconf001@turing.hpc.odu.edu's password:\nRI_B_01.bam\nlyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./\nkconf001@turing.hpc.odu.edu's password:\nRI_B_01.bam.bai\n \"\u003e\u003cpre\u003elyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./\nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e15079_Apoc_hostsym.fasta\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e100%   13MB   5.7MB/s   00:02\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003elyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ekconf001@turing.hpc.odu.edu\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003es password:\n15079_Apoc_hostsym.fasta.fai\n100%  641KB   3.8MB/s   00:00 \nlyka@LAPTOP-GFGCMDB6 MINGW64 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\n$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./\nkconf001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_01.bam\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003elyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ekconf001@turing.hpc.odu.edu\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003es password:\nRI_B_01.bam.bai\n \u003c/pre\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003elaunch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the \"genome\" and RI_B_01.bam as the input file\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"01/22/2021","anchor":"01222021","htmlText":"01/22/2021"},{"level":2,"text":"Exercises Day 02","anchor":"exercises-day-02","htmlText":"Exercises Day 02"},{"level":1,"text":"01/27/2021","anchor":"01272021","htmlText":"01/27/2021"},{"level":2,"text":"Homework Day 02 (continuation of Exercise 02)","anchor":"homework-day-02-continuation-of-exercise-02","htmlText":"Homework Day 02 (continuation of Exercise 02)"},{"level":1,"text":"01/27/2021","anchor":"01272021-1","htmlText":"01/27/2021"},{"level":2,"text":"Homework Day 03","anchor":"homework-day-03","htmlText":"Homework Day 03"},{"level":1,"text":"01/29/2021","anchor":"01292021","htmlText":"01/29/2021"},{"level":2,"text":"Homework Day 04","anchor":"homework-day-04","htmlText":"Homework Day 04"},{"level":1,"text":"02/03/2021","anchor":"02032021","htmlText":"02/03/2021"},{"level":2,"text":"Homework Day 05","anchor":"homework-day-05","htmlText":"Homework Day 05"},{"level":1,"text":"02/05/2021","anchor":"02052021","htmlText":"02/05/2021"},{"level":2,"text":"Homework Day 6","anchor":"homework-day-6","htmlText":"Homework Day 6"},{"level":1,"text":"02/12/2021","anchor":"02122021","htmlText":"02/12/2021"},{"level":2,"text":"Homework Day 8","anchor":"homework-day-8","htmlText":"Homework Day 8"},{"level":1,"text":"02/17/2021","anchor":"02172021","htmlText":"02/17/2021"},{"level":2,"text":"Homework Day 09","anchor":"homework-day-09","htmlText":"Homework Day 09"},{"level":1,"text":"02/19/2021","anchor":"02192021","htmlText":"02/19/2021"},{"level":2,"text":"Homework Day 10","anchor":"homework-day-10","htmlText":"Homework Day 10"},{"level":1,"text":"02/24/2021","anchor":"02242021","htmlText":"02/24/2021"},{"level":2,"text":"Homework Day 11","anchor":"homework-day-11","htmlText":"Homework Day 11"},{"level":1,"text":"03/03/2021","anchor":"03032021","htmlText":"03/03/2021"},{"level":2,"text":"Homework Day 13 (last day!!)","anchor":"homework-day-13-last-day","htmlText":"Homework Day 13 (last day!!)"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fkconf001%2F21SpKristinaCAdvancedGenomicsLog"}}],"overviewFilesProcessingTime":194.653135,"createFromTemplatePath":"/new?template_name=21SpKristinaCAdvancedGenomicsLog\u0026template_owner=kconf001"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           39 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      01/22/2021
                     </h1>
                     <a aria-label="Permalink: 01/22/2021" class="anchor" href="#01222021" id="user-content-01222021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Exercises Day 02
                     </h2>
                     <a aria-label="Permalink: Exercises Day 02" class="anchor" href="#exercises-day-02" id="user-content-exercises-day-02">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        Log into the Cluster
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~
$ ssh kconf001@turing.hpc.odu.edu
kconf001@turing.hpc.odu.edu's password:
Last login: Fri Jan 22 10:12:23 2021 from ip70-160-48-140.hr.hr.cox.net" dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>
$ ssh kconf001@turing.hpc.odu.edu
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">Last login: Fri Jan 22 10:12:23 2021 from ip70-160-48-140.hr.hr.cox.net</span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Make directory in course workspace called data
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC
[kconf001@turing1 KristinaC]$ ls
groundrules.txt  KristinaC_Exercise1.txt  scripts
[kconf001@turing1 KristinaC]$ mkdir data
[kconf001@turing1 KristinaC]$ ls
data  groundrules.txt  KristinaC_Exercise1.txt  scripts" dir="auto">
                     <pre>[kconf001@turing1 <span class="pl-k">~</span>]$ <span class="pl-c1">cd</span> /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC
[kconf001@turing1 KristinaC]$ ls
groundrules.txt  KristinaC_Exercise1.txt  scripts
[kconf001@turing1 KristinaC]$ mkdir data
[kconf001@turing1 KristinaC]$ ls
data  groundrules.txt  KristinaC_Exercise1.txt  scripts</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        Make directory called exercises in data directory
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 KristinaC]$ cd data/
[kconf001@turing1 data]$ mkdir exercises
[kconf001@turing1 data]$ ls
exercises 
[kconf001@turing1 exercises]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises  " dir="auto">
                     <pre>[kconf001@turing1 KristinaC]$ <span class="pl-c1">cd</span> data/
[kconf001@turing1 data]$ mkdir exercises
[kconf001@turing1 data]$ ls
exercises 
[kconf001@turing1 exercises]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises  </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        Execute pwd command &amp; start a log of commands with a header for today's date in README.md github logfile in local workspace
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~
$ pwd        
/c/Users/lyka
lyka@LAPTOP-GFGCMDB6 MINGW64 ~
$ cd Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)
$ ls                                                                                                                             
21sp_advgenomics/
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)
$ pwd 
/c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis 
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)
$ mkdir 21SpKristinaCAdvancedGenomicsLog
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis
$ ls
21sp_advgenomics/  21SpKristinaCAdvancedGenomicsLog/
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis
$ cd ./21SpKristinaCAdvancedGenomicsLog
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ nano     " dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>
$ <span class="pl-c1">pwd</span>        
/c/Users/lyka
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>
$ <span class="pl-c1">cd</span> Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)
$ ls                                                                                                                             
21sp_advgenomics/
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)
$ <span class="pl-c1">pwd</span> 
/c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis 
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis (main)
$ mkdir 21SpKristinaCAdvancedGenomicsLog
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis
$ ls
21sp_advgenomics/  21SpKristinaCAdvancedGenomicsLog/
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis
$ <span class="pl-c1">cd</span> ./21SpKristinaCAdvancedGenomicsLog
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ nano     </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <p dir="auto">
                       With nano command, README.md file is created with Exercise 02 Header &amp; logfile.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Logfile is filled at the end of each part of the exercises completed.
                      </p>
                     </li>
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        Copy the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your course exercises directory
                       </li>
                      </ol>
                     </li>
                     <li>
                      <p dir="auto">
                       from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory
                      </p>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./
[kconf001@turing1 exercises]$ ls
Exercise2.fasta.gz  Exercise2.fastq.tar.gz
[kconf001@turing1 exercises]$ ls -alh
total 13M 
drwxrwxrwx 2 kconf001 users   76 Jan 22 14:25 .
drwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz " dir="auto">
                     <pre>[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast<span class="pl-k">*</span> ./
[kconf001@turing1 exercises]$ ls
Exercise2.fasta.gz  Exercise2.fastq.tar.gz
[kconf001@turing1 exercises]$ ls -alh
total 13M 
drwxrwxrwx 2 kconf001 users   76 Jan 22 14:25 <span class="pl-c1">.</span>
drwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="6">
                       <li>
                        Unzip &amp; untar the files
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 exercises]$ gunzip -c Exercise2.fasta.gz &gt; Exercise2.fasta
[kconf001@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq.tar.g
[kconf001@turing1 exercises]$ ls -alh
total 37M
drwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 .
drwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..
-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz
[kconf001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz
Exercise2.fastq
[kconf001@turing1 exercises]$ ls -alh
total 62M
drwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 .                                                                   
drwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 .. 
-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta                                                     
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz
-rw-r--r-- 1 kconf001 users  18M Sep  2  2015 Exercise2.fastq                                                 
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz 
[kconf001@turing1 exercises]$ pwd 
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises                    
[kconf001@turing1 exercises]$ ls                      
Exercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq  Exercise2.fastq.tar.gz" dir="auto">
                     <pre>[kconf001@turing1 exercises]$ gunzip -c Exercise2.fasta.gz <span class="pl-k">&gt;</span> Exercise2.fasta
[kconf001@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq.tar.g
[kconf001@turing1 exercises]$ ls -alh
total 37M
drwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 <span class="pl-c1">.</span>
drwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 ..
-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz
[kconf001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz
Exercise2.fastq
[kconf001@turing1 exercises]$ ls -alh
total 62M
drwxrwxrwx 2 kconf001 users  109 Jan 22 14:26 <span class="pl-c1">.</span>                                                                   
drwxrwxrwx 3 kconf001 users   27 Jan 22 13:25 .. 
-rwxrwxrwx 1 kconf001 users  17M Jan 22 14:26 Exercise2.fasta                                                     
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fasta.gz
-rw-r--r-- 1 kconf001 users  18M Sep  2  2015 Exercise2.fastq                                                 
-rwxr-xr-x 1 kconf001 users 4.3M Jan 22 14:25 Exercise2.fastq.tar.gz 
[kconf001@turing1 exercises]$ <span class="pl-c1">pwd</span> 
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/exercises                    
[kconf001@turing1 exercises]$ ls                      
Exercise2.fasta  Exercise2.fasta.gz  Exercise2.fastq  Exercise2.fastq.tar.gz</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="7">
                       <li>
                        Add commands to notebook file (see above)
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="8">
                       <li>
                        Calculate how many sequences are in each file
                       </li>
                      </ol>
                     </li>
                     <li>
                      <p dir="auto">
                       Fasta file: Sequences start with '&gt;'
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Fastq file: Sequences start with '@'; Each fastq read includes 4 lines.
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       First line- Identifier
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Second line- Sequence
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Third line- Blank line (starts with +, sometimes have the same description as first line)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Fourth line- Quality for each base in Sequence (Second line)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       grep -c = Used to find patterns (find a string in a file or string)
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Can also divide total lines by 4 to find total sequence reads
                      </p>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 exercises]$ head -1 Exercise2.fasta
&gt;scaffold10078|size20675
[kconf001@turing1 exercises]$ grep -c '&gt;' Exercise2.fasta
138
[kconf001@turing1 exercises]$ head -1 Exercise2.fastq
@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC
[kconf001@turing1 exercises]$ grep -c '@HS3' Exercise2.fastq
61304
[kconf001@turing1 exercises]$ wc Exercise2.fastq
245216   306520 18459769 Exercise2.fastq
[kconf001@turing1 exercises]$ echo 245216/4 | bc -l
61304.00000000000000000000  " dir="auto">
                     <pre>[kconf001@turing1 exercises]$ head -1 Exercise2.fasta
<span class="pl-k">&gt;</span>scaffold10078<span class="pl-k">|</span>size20675
[kconf001@turing1 exercises]$ grep -c <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> Exercise2.fasta
138
[kconf001@turing1 exercises]$ head -1 Exercise2.fastq
@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC
[kconf001@turing1 exercises]$ grep -c <span class="pl-s"><span class="pl-pds">'</span>@HS3<span class="pl-pds">'</span></span> Exercise2.fastq
61304
[kconf001@turing1 exercises]$ wc Exercise2.fastq
245216   306520 18459769 Exercise2.fastq
[kconf001@turing1 exercises]$ <span class="pl-c1">echo</span> 245216/4 <span class="pl-k">|</span> bc -l
61304.00000000000000000000  </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="9">
                       <li>
                        Copy the python file from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory
                       </li>
                      </ol>
                     </li>
                     <li>
                      into the class scripts directory
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/
[kconf001@turing1 exercises]$ cd ../
[kconf001@turing1 data]$ cd ../
[kconf001@turing1 KristinaC]$ cd scripts/
[kconf001@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py" dir="auto">
                     <pre>[kconf001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py ../../scripts/
[kconf001@turing1 exercises]$ <span class="pl-c1">cd</span> ../
[kconf001@turing1 data]$ <span class="pl-c1">cd</span> ../
[kconf001@turing1 KristinaC]$ <span class="pl-c1">cd</span> scripts/
[kconf001@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="10">
                       <li>
                        Start interactive compute session and re-navigate to exercises directory
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[kconf001@turing1 scripts]$ cd ../
[kconf001@turing1 KristinaC]$ cd data/
[kconf001@turing1 data]$ cd exercises/
[kconf001@turing1 exercises]$ salloc
salloc: Pending job allocation 9269144
salloc: job 9269144 queued and waiting for resources
salloc: job 9269144 has been allocated resources
salloc: Granted job allocation 9269517
This session will be terminated in 7 days. If your application requires
a longer excution time, please use command "salloc -t N-0" where N is the
number of days that you need.
[kconf001@coreV1-22-016 exercises]$' dir="auto">
                     <pre>[kconf001@turing1 scripts]$ <span class="pl-c1">cd</span> ../
[kconf001@turing1 KristinaC]$ <span class="pl-c1">cd</span> data/
[kconf001@turing1 data]$ <span class="pl-c1">cd</span> exercises/
[kconf001@turing1 exercises]$ salloc
salloc: Pending job allocation 9269144
salloc: job 9269144 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9269144 has been allocated resources
salloc: Granted job allocation 9269517
This session will be terminated <span class="pl-k">in</span> 7 days. If your application requires
a longer excution time, please use <span class="pl-c1">command</span> <span class="pl-s"><span class="pl-pds">"</span>salloc -t N-0<span class="pl-pds">"</span></span> where N is the
number of days that you need.
[kconf001@coreV1-22-016 exercises]$</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="11">
                       <li>
                        Run the python script on the Exercise2.fasta file
                       </li>
                      </ol>
                     </li>
                     <li>
                      by typing the path to the script followed by the Exercise2.fasta file name
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV1-22-016 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 138
contigs &gt;= 1000bp = 138
contigs &gt;= 2000bp = 135
[kconf001@coreV1-22-016 exercises]$ exit
logout
salloc: Relinquishing job allocation 9269517" dir="auto">
                     <pre>[kconf001@coreV1-22-016 exercises]$ ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs <span class="pl-k">&lt;</span> 150bp = 0
contigs <span class="pl-k">&gt;</span>= 500bp = 138
contigs <span class="pl-k">&gt;</span>= 1000bp = 138
contigs <span class="pl-k">&gt;</span>= 2000bp = 135
[kconf001@coreV1-22-016 exercises]$ <span class="pl-c1">exit</span>
<span class="pl-c1">logout</span>
salloc: Relinquishing job allocation 9269517</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="12">
                       <li>
                        Add all parts to notebook README.md file including the results of the script
                       </li>
                      </ol>
                     </li>
                     <li>
                      <p dir="auto">
                       See above
                      </p>
                     </li>
                     <li>
                      <ol dir="auto" start="13">
                       <li>
                        Push notebook file to github page
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git add README.md
warning: LF will be replaced by CRLF in README.md.
The file will have its original line endings in your working directory
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git commit -m 'updating readme'
[main c32c255] updating readme
1 file changed, 161 insertions(+), 4 deletions(-)
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git push -u origin main
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 2.11 KiB | 2.11 MiB/s, done.
Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
To https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git
d677019..c32c255  main -&gt; main
Branch 'main' set up to track remote branch 'main' from 'origin'." dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git add README.md
warning: LF will be replaced by CRLF <span class="pl-k">in</span> README.md.
The file will have its original line endings <span class="pl-k">in</span> your working directory
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
[main c32c255] updating readme
1 file changed, 161 insertions(+), 4 deletions(-)
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git push -u origin main
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 2.11 KiB <span class="pl-k">|</span> 2.11 MiB/s, done.
Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
To https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git
d677019..c32c255  main -<span class="pl-k">&gt;</span> main
Branch <span class="pl-s"><span class="pl-pds">'</span>main<span class="pl-pds">'</span></span> <span class="pl-c1">set</span> up to track remote branch <span class="pl-s"><span class="pl-pds">'</span>main<span class="pl-pds">'</span></span> from <span class="pl-s"><span class="pl-pds">'</span>origin<span class="pl-pds">'</span></span>.</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      01/27/2021
                     </h1>
                     <a aria-label="Permalink: 01/27/2021" class="anchor" href="#01272021" id="user-content-01272021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 02 (continuation of Exercise 02)
                     </h2>
                     <a aria-label="Permalink: Homework Day 02 (continuation of Exercise 02)" class="anchor" href="#homework-day-02-continuation-of-exercise-02" id="user-content-homework-day-02-continuation-of-exercise-02">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        Make sure logged into turing node &amp; in personal data directory.
                       </li>
                      </ol>
                     </li>
                     <li>
                      <p dir="auto">
                       Also input salloc command (see previous day for details).
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Write an sbatch script to cp the files
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       into your own data directory
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       Can use nano or scp
                      </p>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l
#SBATCH -o KristinaCCopylane01.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCCopylane01
cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01*.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCCopylane01.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCCopylane01</span>
cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB01<span class="pl-k">*</span>.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Add sbatch script to data directory (see above)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="nano method (in turing cluster)
[kconf001@coreV2-25-072 data] nano KristinaCCopy1.sh " dir="auto">
                     <pre>nano method (in turing cluster)
[kconf001@coreV2-25-072 data] nano KristinaCCopy1.sh </pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="scp method (in local directory files)
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis 
$ nano KristinaCCopy1.sh
$ scp KristinaCCopylane01.sh kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/
kconf001@turing.hpc.odu.edu's password:
KristinaCCopylane01.sh                                                                           100%  341     8.7KB/s   00:00

Input sbatch script into .sh file" dir="auto">
                     <pre>scp method (in <span class="pl-k">local</span> directory files)
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis 
$ nano KristinaCCopy1.sh
$ scp KristinaCCopylane01.sh kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">KristinaCCopylane01.sh                                                                           100%  341     8.7KB/s   00:00</span>
<span class="pl-s"></span>
<span class="pl-s">Input sbatch script into .sh file</span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        Submit slurm script (sbatch KristinaCCopyl.sh) and verify that it's working (squeue -u kconf001 and/or ls -alh)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-072 data] $ sbatch KristinaCCopy1.sh
Submitted batch job 9270444
[kconf001@coreV2-25-072 data] $ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)  
9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072  
920444      main       sh kconf001  R      01:25      1 coreV2-25-072" dir="auto">
                     <pre>[kconf001@coreV2-25-072 data] $ sbatch KristinaCCopy1.sh
Submitted batch job 9270444
[kconf001@coreV2-25-072 data] $ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)  
9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072  
920444      main       sh kconf001  R      01:25      1 coreV2-25-072</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        Document in README.md / github notebook (see #13 in previous day)
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        Write sbatch script to gunzip all the fastaq.gz files once copying is finished.
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Nano method
#!/bin/bash -l
#SBATCH -o KristinaCCopylane01.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCGunZiplane01
gunzip ./*fastq.gz" dir="auto">
                     <pre>Nano method
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCCopylane01.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCGunZiplane01</span>
gunzip ./<span class="pl-k">*</span>fastq.gz</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      Verify that it is working
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-072 data] $ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072 
9270462      main       sh kconf001  R      25:25      1 coreV2-25-072

[kconf001@coreV2-25-072 data]$ ls -alh
total 58G 
drwxrwxrwx 3 kconf001 users  1017 Jan 27 14:37 . 
drwxrwxrwx 4 kconf001 users   121 Jan 22 11:31 .. 
drwxrwxrwx 2 kconf001 users   142 Jan 25 09:57 exercises
-rw-r--r-- 1 kconf001 users  2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq  
-rw-r--r-- 1 kconf001 users  981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users   13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq
-rw------- 1 kconf001 users  1.2G Jan 27 14:37 HADB01-M_S29_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  657M Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq.gz 
-rw-r--r-- 1 kconf001 users  671M Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq.gz
-rw-r--r-- 1 kconf001 users 1000M Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq.gz
-rw-r--r-- 1 kconf001 users  538M Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq.gz
-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:29 KristinaCCopylane01.txt
-rwxrwxrwx 1 kconf001 users   189 Jan 27 13:49 KristinaCGunZiplane01.sh
-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:52 KristinaCGunZiplane01.txt
-rwxrwxrwx 1 kconf001 users   341 Jan 27 13:20 KristinaCLane01.sh" dir="auto">
                     <pre>[kconf001@coreV2-25-072 data] $ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9270470      main Kristina kconf001  R      17:25      1 coreV2-25-072 
9270462      main       sh kconf001  R      25:25      1 coreV2-25-072

[kconf001@coreV2-25-072 data]$ ls -alh
total 58G 
drwxrwxrwx 3 kconf001 users  1017 Jan 27 14:37 <span class="pl-c1">.</span> 
drwxrwxrwx 4 kconf001 users   121 Jan 22 11:31 .. 
drwxrwxrwx 2 kconf001 users   142 Jan 25 09:57 exercises
-rw-r--r-- 1 kconf001 users  2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq  
-rw-r--r-- 1 kconf001 users  981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users   13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq
-rw------- 1 kconf001 users  1.2G Jan 27 14:37 HADB01-M_S29_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users  657M Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq.gz 
-rw-r--r-- 1 kconf001 users  671M Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq.gz
-rw-r--r-- 1 kconf001 users 1000M Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq.gz
-rw-r--r-- 1 kconf001 users  538M Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq.gz
-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:29 KristinaCCopylane01.txt
-rwxrwxrwx 1 kconf001 users   189 Jan 27 13:49 KristinaCGunZiplane01.sh
-rwxrwxrwx 1 kconf001 users     0 Jan 27 13:52 KristinaCGunZiplane01.txt
-rwxrwxrwx 1 kconf001 users   341 Jan 27 13:20 KristinaCLane01.sh</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="6">
                       <li>
                        Push updated README.md log to github page (see #13 in previous day).
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      01/27/2021
                     </h1>
                     <a aria-label="Permalink: 01/27/2021" class="anchor" href="#01272021-1" id="user-content-01272021-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 03
                     </h2>
                     <a aria-label="Permalink: Homework Day 03" class="anchor" href="#homework-day-03" id="user-content-homework-day-03">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory &amp; files to personal sandbox
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-072 data]$ cd ../
[kconf001@coreV2-25-072 KristinaC]$  cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./ ls
[kconf001@coreV2-25-072 KristinaC]$ ls
data  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts" dir="auto">
                     <pre>[kconf001@coreV2-25-072 data]$ <span class="pl-c1">cd</span> ../
[kconf001@coreV2-25-072 KristinaC]$  cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./ ls
[kconf001@coreV2-25-072 KristinaC]$ ls
data  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Make a new fastq directory (mkdir fastq) in directory and move (mv) all the .fastq files into the fastq directory
Make sure all files have finished gunzipping (ls -alh)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-072 KristinaC]$ cd data
[kconf001@coreV2-25-072 data]$ mkdir fastq
[kconf001@coreV2-25-072 data]$ pwd                                                                                   
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data
[kconf001@coreV2-25-072 data]$ ls -alh                                                                               
total 79G
drwxrwxrwx 4 kconf001 users  980 Jan 27 14:52 .
drwxrwxrwx 5 kconf001 users  144 Jan 27 14:49 ..
drwxrwxrwx 2 kconf001 users  142 Jan 25 09:57 exercises
drwxrwxrwx 2 kconf001 users    0 Jan 27 14:50 fastq
-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq             
-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq                                         
-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:29 KristinaCCopylane01.txt                                                
-rwxrwxrwx 1 kconf001 users  189 Jan 27 13:49 KristinaCGunZiplane01.sh                                               
-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:52 KristinaCGunZiplane01.txt                                              
-rwxrwxrwx 1 kconf001 users  341 Jan 27 13:20 KristinaCLane01.sh   
[kconf001@coreV2-25-072 data]$ mv *.fastq ./fastq/
[kconf001@coreV2-25-072 data]$ ls
exercises  fastq  KristinaCCopylane01.txt  KristinaCGunZiplane01.sh  KristinaCGunZiplane01.txt  KristinaCLane01.sh" dir="auto">
                     <pre>[kconf001@coreV2-25-072 KristinaC]$ <span class="pl-c1">cd</span> data
[kconf001@coreV2-25-072 data]$ mkdir fastq
[kconf001@coreV2-25-072 data]$ <span class="pl-c1">pwd</span>                                                                                   
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data
[kconf001@coreV2-25-072 data]$ ls -alh                                                                               
total 79G
drwxrwxrwx 4 kconf001 users  980 Jan 27 14:52 <span class="pl-c1">.</span>
drwxrwxrwx 5 kconf001 users  144 Jan 27 14:49 ..
drwxrwxrwx 2 kconf001 users  142 Jan 25 09:57 exercises
drwxrwxrwx 2 kconf001 users    0 Jan 27 14:50 fastq
-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 HADB01-A_S17_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 HADB01-B_S18_L002_R1_001.fastq
-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 HADB01-C_S19_L002_R1_001.fastq             
-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 HADB01-D_S20_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 HADB01-E_S21_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 HADB01-F_S22_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 HADB01-G_S23_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 HADB01-H_S24_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 HADB01-I_S25_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 HADB01-J_S26_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-K_S27_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 HADB01-L_S28_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 HADB01-M_S29_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 HADB01-N_S30_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 HADB01-O_S31_L002_R1_001.fastq                                         
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 HADB01-P_S32_L002_R1_001.fastq                                         
-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:29 KristinaCCopylane01.txt                                                
-rwxrwxrwx 1 kconf001 users  189 Jan 27 13:49 KristinaCGunZiplane01.sh                                               
-rwxrwxrwx 1 kconf001 users    0 Jan 27 13:52 KristinaCGunZiplane01.txt                                              
-rwxrwxrwx 1 kconf001 users  341 Jan 27 13:20 KristinaCLane01.sh   
[kconf001@coreV2-25-072 data]$ mv <span class="pl-k">*</span>.fastq ./fastq/
[kconf001@coreV2-25-072 data]$ ls
exercises  fastq  KristinaCCopylane01.txt  KristinaCGunZiplane01.sh  KristinaCGunZiplane01.txt  KristinaCLane01.sh</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        cp the renamingtable_complete.txt from the day03 directory into your fastq directory
                       </li>
                      </ol>
                     </li>
                     <li>
                      and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder
                     </li>
                     <li>
                      and less the new script and check out the usage statement.
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cp the renamingtable_complete.txt from the day03 directory into your fastq directory 
[kconf001@coreV2-25-072 data]$ cp ../day03/renamingtable_complete.txt ./fastq/
[kconf001@coreV2-25-072 data]$ cd fastq/
[kconf001@coreV2-25-072 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq
HADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq
HADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt
HADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq" dir="auto">
                     <pre>cp the renamingtable_complete.txt from the day03 directory into your fastq directory 
[kconf001@coreV2-25-072 data]$ cp ../day03/renamingtable_complete.txt ./fastq/
[kconf001@coreV2-25-072 data]$ <span class="pl-c1">cd</span> fastq/
[kconf001@coreV2-25-072 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq
HADB01-C_S19_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq
HADB01-D_S20_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-E_S21_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  renamingtable_complete.txt
HADB01-F_S22_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder
[kconf001@coreV2-25-072 data]$ cd ../../
[kconf001@coreV2-25-072 KristinaC]$ ls                                                                               
data  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts                                                       
[kconf001@coreV2-25-072 KristinaC]$ cd scripts/
[kconf001@coreV2-25-072 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py  /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/
[kconf001@coreV2-25-072 scripts]$ ls 
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py" dir="auto">
                     <pre>cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder
[kconf001@coreV2-25-072 data]$ <span class="pl-c1">cd</span> ../../
[kconf001@coreV2-25-072 KristinaC]$ ls                                                                               
data  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts                                                       
[kconf001@coreV2-25-072 KristinaC]$ <span class="pl-c1">cd</span> scripts/
[kconf001@coreV2-25-072 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py  /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/
[kconf001@coreV2-25-072 scripts]$ ls 
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py</pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="less (head) new script
[kconf001@coreV2-25-072 scripts]$ head renamer_advbioinf.py 
#!/usr/bin/env python 
####usage renamer.py renamingtable 
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column 
import sys 
import os
fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
linecount+=1" dir="auto">
                     <pre>less (head) new script
[kconf001@coreV2-25-072 scripts]$ head renamer_advbioinf.py 
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python </span>
<span class="pl-c"><span class="pl-c">#</span>###usage renamer.py renamingtable </span>
<span class="pl-c"><span class="pl-c">#</span>### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column </span>
import sys 
import os
fin=open(sys.argv[1],<span class="pl-s"><span class="pl-pds">'</span>r<span class="pl-pds">'</span></span>)
linecount=0
<span class="pl-k">for</span> <span class="pl-smi">line</span> <span class="pl-k">in</span> fin:
linecount+=1</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        Run renamer_advbioinf.py script in fastq folter using renamingtable_complete.txt as practice &amp; verify output to screen by hand.
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~                                                                                                             
$ ssh kconf001@turing.hpc.odu.edu  
kconf001@turing.hpc.odu.edu's password:                                                                         
Last login: Wed Jan 27 15:37:12 2021 from ip70-160-48-140.hr.hr.cox.net
[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/
[kconf001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq
[kconf001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt
mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq
mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq
mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq
mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq
mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq
mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq
mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq
mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq
mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq
mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq
mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq
mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq
mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq
mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq
mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq
mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq
mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq
mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq
mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq
mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq
mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq
mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq
mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq
mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq
mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq
mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq
mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq
mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq
mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq
mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq
mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq
mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq
mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq
mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq
mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq
mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq
mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq
mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq
mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq
mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq
mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq
mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq
mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq
mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq
mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq
mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq
mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq
mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq
mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq
mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq
mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq
mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq
mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq
mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq
mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq
mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq
mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq
mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq
mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq
mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq
mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq
mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq
mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq
mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq
mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq
mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq
mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq
mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq
mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq
mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq
mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq
mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq
mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq
mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq
mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq
mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq
mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq
mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq
mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq
mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq
mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq
mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq
mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq
mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq
mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq
mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq
mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq
mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq
mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq
mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq
mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq
mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq
mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq
mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq
mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq
mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq" dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>                                                                                                             
$ ssh kconf001@turing.hpc.odu.edu  
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:                                                                         </span>
<span class="pl-s">Last login: Wed Jan 27 15:37:12 2021 from ip70-160-48-140.hr.hr.cox.net</span>
<span class="pl-s">[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/</span>
<span class="pl-s">[kconf001@turing1 fastq]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq</span>
<span class="pl-s">[kconf001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt</span>
<span class="pl-s">mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq</span>
<span class="pl-s">mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq</span>
<span class="pl-s">mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq</span>
<span class="pl-s">mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq</span>
<span class="pl-s">mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq</span>
<span class="pl-s">mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq</span>
<span class="pl-s">mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq</span>
<span class="pl-s">mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq</span>
<span class="pl-s">mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq</span>
<span class="pl-s">mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq</span>
<span class="pl-s">mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq</span>
<span class="pl-s">mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq</span>
<span class="pl-s">mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq</span>
<span class="pl-s">mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq</span>
<span class="pl-s">mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq</span>
<span class="pl-s">mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq</span>
<span class="pl-s">mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq</span>
<span class="pl-s">mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq</span>
<span class="pl-s">mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq</span>
<span class="pl-s">mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq</span>
<span class="pl-s">mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq</span>
<span class="pl-s">mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq</span>
<span class="pl-s">mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq</span>
<span class="pl-s">mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq</span>
<span class="pl-s">mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq</span>
<span class="pl-s">mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq</span>
<span class="pl-s">mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq</span>
<span class="pl-s">mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq</span>
<span class="pl-s">mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq</span>
<span class="pl-s">mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq</span>
<span class="pl-s">mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq</span>
<span class="pl-s">mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq</span>
<span class="pl-s">mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq</span>
<span class="pl-s">mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq</span>
<span class="pl-s">mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq</span>
<span class="pl-s">mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq</span>
<span class="pl-s">mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq</span>
<span class="pl-s">mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq</span>
<span class="pl-s">mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq</span>
<span class="pl-s">mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq</span>
<span class="pl-s">mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq</span>
<span class="pl-s">mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq</span>
<span class="pl-s">mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq</span>
<span class="pl-s">mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq</span>
<span class="pl-s">mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq</span>
<span class="pl-s">mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq</span>
<span class="pl-s">mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq</span>
<span class="pl-s">mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq</span>
<span class="pl-s">mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq</span>
<span class="pl-s">mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq</span>
<span class="pl-s">mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq</span>
<span class="pl-s">mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq</span>
<span class="pl-s">mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq</span>
<span class="pl-s">mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq</span>
<span class="pl-s">mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq</span>
<span class="pl-s">mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq</span>
<span class="pl-s">mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq</span>
<span class="pl-s">mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq</span>
<span class="pl-s">mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq</span>
<span class="pl-s">mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq</span>
<span class="pl-s">mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq</span>
<span class="pl-s">mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq</span>
<span class="pl-s">mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq</span>
<span class="pl-s">mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq</span>
<span class="pl-s">mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq</span>
<span class="pl-s">mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq</span>
<span class="pl-s">mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq</span>
<span class="pl-s">mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq</span>
<span class="pl-s">mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq</span>
<span class="pl-s">mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq</span>
<span class="pl-s">mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq</span>
<span class="pl-s">mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq</span>
<span class="pl-s">mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq</span>
<span class="pl-s">mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq</span>
<span class="pl-s">mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq</span>
<span class="pl-s">mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq</span>
<span class="pl-s">mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq</span>
<span class="pl-s">mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq</span>
<span class="pl-s">mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq</span>
<span class="pl-s">mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq</span>
<span class="pl-s">mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq</span>
<span class="pl-s">mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq</span>
<span class="pl-s">mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq</span>
<span class="pl-s">mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq</span>
<span class="pl-s">mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq</span>
<span class="pl-s">mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq</span>
<span class="pl-s">mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq</span>
<span class="pl-s">mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq</span>
<span class="pl-s">mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq</span>
<span class="pl-s">mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq</span>
<span class="pl-s">mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq</span>
<span class="pl-s">mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq</span>
<span class="pl-s">mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq</span>
<span class="pl-s">mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq</span>
<span class="pl-s">mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq</span>
<span class="pl-s">mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq</span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        Uncomment (remove the #) in renaming python script thaht starts with os.open and comment out the print line
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 fastq]$ pwd 
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq
[kconf001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os
linecount+=1
if linecount&gt;=2:
cols=line.rstrip().split('\t')
#print 'mv %s %s' %(cols[0], cols[1])
os.popen('mv %s %s' %(cols[0], cols[1]))" dir="auto">
                     <pre>[kconf001@turing1 fastq]$ <span class="pl-c1">pwd</span> 
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq
[kconf001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python</span>
<span class="pl-c"><span class="pl-c">#</span>###usage renamer.py renamingtable</span>
<span class="pl-c"><span class="pl-c">#</span>### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column</span>
import sys
import os
linecount+=1
<span class="pl-k">if</span> linecount<span class="pl-k">&gt;</span>=2:
cols=line.rstrip().split(<span class="pl-s"><span class="pl-pds">'</span>\t<span class="pl-pds">'</span></span>)
<span class="pl-c"><span class="pl-c">#</span>print 'mv %s %s' %(cols[0], cols[1])</span>
os.popen(<span class="pl-s"><span class="pl-pds">'</span>mv %s %s<span class="pl-pds">'</span></span> %(cols[0], cols[1]))</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="6">
                       <li>
                        Write sbatch script &amp; submit it to rename all the .fastq files according to renaming table using renamer python script
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 fastq]$ nano KristinaCRenamer.sh
[kconf001@turing1 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-D_S20_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-E_S21_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq  KristinaCRenamer.sh
HADB01-C_S19_L002_R1_001.fastq  HADB01-F_S22_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq  renamingtable_complete.txt
[kconf001@turing1 fastq]$ salloc
salloc: Pending job allocation 9270528
salloc: job 9270528 queued and waiting for resources 
salloc: job 9270528 has been allocated resources 
salloc: Granted job allocation 9270528
[kconf001@coreV2-25-072 fastq]$ cat KristinaCRenamer.sh 
#!/bin/bash -l 
#SBATCH -o KristinaCRenamer.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCRenamerFastq
[kconf001@coreV2-25-072 fastq]$ sbatch ./KristinaCRenamer.sh
Submitted batch job 9270529
[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9270529      main Kristina kconf001 PD       0:00      1 (Priority)
9270528      main       sh kconf001  R       0:29      1 coreV2-25-072
Once job is done running: checking to see if renaming is completed. 
[kconf001@coreV2-25-072 fastq]$ ls -alh
total 79G
drwxrwxrwx 2 kconf001 users  667 Jan 27 16:02 .
drwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..
-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh
-rwxrwxrwx 1 kconf001 users  11K Jan 27 16:02 KristinaCRenamer.txt
-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt
-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq
-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq 
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq
-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq
-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq
-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq      " dir="auto">
                     <pre>[kconf001@turing1 fastq]$ nano KristinaCRenamer.sh
[kconf001@turing1 fastq]$ ls
HADB01-A_S17_L002_R1_001.fastq  HADB01-D_S20_L002_R1_001.fastq  HADB01-G_S23_L002_R1_001.fastq  HADB01-J_S26_L002_R1_001.fastq  HADB01-M_S29_L002_R1_001.fastq  HADB01-P_S32_L002_R1_001.fastq
HADB01-B_S18_L002_R1_001.fastq  HADB01-E_S21_L002_R1_001.fastq  HADB01-H_S24_L002_R1_001.fastq  HADB01-K_S27_L002_R1_001.fastq  HADB01-N_S30_L002_R1_001.fastq  KristinaCRenamer.sh
HADB01-C_S19_L002_R1_001.fastq  HADB01-F_S22_L002_R1_001.fastq  HADB01-I_S25_L002_R1_001.fastq  HADB01-L_S28_L002_R1_001.fastq  HADB01-O_S31_L002_R1_001.fastq  renamingtable_complete.txt
[kconf001@turing1 fastq]$ salloc
salloc: Pending job allocation 9270528
salloc: job 9270528 queued and waiting <span class="pl-k">for</span> resources 
salloc: job 9270528 has been allocated resources 
salloc: Granted job allocation 9270528
[kconf001@coreV2-25-072 fastq]$ cat KristinaCRenamer.sh 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCRenamer.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCRenamerFastq</span>
[kconf001@coreV2-25-072 fastq]$ sbatch ./KristinaCRenamer.sh
Submitted batch job 9270529
[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9270529      main Kristina kconf001 PD       0:00      1 (Priority)
9270528      main       sh kconf001  R       0:29      1 coreV2-25-072
Once job is <span class="pl-k">done</span> running: checking to see <span class="pl-k">if</span> renaming is completed. 
[kconf001@coreV2-25-072 fastq]$ ls -alh
total 79G
drwxrwxrwx 2 kconf001 users  667 Jan 27 16:02 <span class="pl-c1">.</span>
drwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..
-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh
-rwxrwxrwx 1 kconf001 users  11K Jan 27 16:02 KristinaCRenamer.txt
-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt
-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq
-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq 
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq
-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq
-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq
-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq      </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="7">
                       <li>
                        Update github (see #13 on Exercises Day 02 for similar workflow)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git add README.md
warning: LF will be replaced by CRLF in README.md.
The file will have its original line endings in your working directory
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git commit -m 'updating readme'
[main 964b72c] updating readme
1 file changed, 235 insertions(+), 2 deletions(-)
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git push -u origin main
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done. 
Writing objects: 100% (3/3), 2.99 KiB | 2.99 MiB/s, done.
Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
To https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git  
b869eb3..964b72c  main -&gt; main 
Branch 'main' set up to track remote branch 'main' from 'origin'." dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git add README.md
warning: LF will be replaced by CRLF <span class="pl-k">in</span> README.md.
The file will have its original line endings <span class="pl-k">in</span> your working directory
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating readme<span class="pl-pds">'</span></span>
[main 964b72c] updating readme
1 file changed, 235 insertions(+), 2 deletions(-)
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ git push -u origin main
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done. 
Writing objects: 100% (3/3), 2.99 KiB <span class="pl-k">|</span> 2.99 MiB/s, done.
Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 <span class="pl-k">local</span> object.
To https://github.com/kconf001/21SpKristinaCAdvancedGenomicsLog.git  
b869eb3..964b72c  main -<span class="pl-k">&gt;</span> main 
Branch <span class="pl-s"><span class="pl-pds">'</span>main<span class="pl-pds">'</span></span> <span class="pl-c1">set</span> up to track remote branch <span class="pl-s"><span class="pl-pds">'</span>main<span class="pl-pds">'</span></span> from <span class="pl-s"><span class="pl-pds">'</span>origin<span class="pl-pds">'</span></span>.</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="8">
                       <li>
                        Naming convention:
                       </li>
                      </ol>
                     </li>
                     <li>
                      <p dir="auto">
                       SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       There are 2 sources: Virginia and Rhode Island
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       There are 2 symbiotic states: Brown and White
                      </p>
                     </li>
                     <li>
                      <ol dir="auto" start="9">
                       <li>
                        Start process of adapter clipping &amp; quality trimming all the renamed .fastq files in batches, by lane.
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="10">
                       <li>
                        cp script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory.
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-072 fastq]$ exit
logout
salloc: Relinquishing job allocation 9270528
[kconf001@turing1 fastq]$ cd ../../
[kconf001@turing1 KristinaC]$ ls
data  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts
[kconf001@turing1 KristinaC]$ cd scripts/
[kconf001@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts
[kconf001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py" dir="auto">
                     <pre>[kconf001@coreV2-25-072 fastq]$ <span class="pl-c1">exit</span>
<span class="pl-c1">logout</span>
salloc: Relinquishing job allocation 9270528
[kconf001@turing1 fastq]$ <span class="pl-c1">cd</span> ../../
[kconf001@turing1 KristinaC]$ ls
data  day03  groundrules.txt  KristinaC_Exercise1.txt  scripts
[kconf001@turing1 KristinaC]$ <span class="pl-c1">cd</span> scripts/
[kconf001@turing1 scripts]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts
[kconf001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="11">
                       <li>
                        less/head new script &amp; check the usage statement
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py
[kconf001@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py
#!/usr/bin/env python
# Written by Dan Barshis
import sys, os
########### Usage #############
# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles
# This should be run from within the folder with all of your original .fastqstrim
# Will quality trim multiple SINGLE-END fastq files
# Things to customize for your particular platform:
[kconf001@turing1 scripts]$  " dir="auto">
                     <pre>[kconf001@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py  Trimclipfilterstatsbatch_advbioinf.py
[kconf001@turing1 scripts]$ head Trimclipfilterstatsbatch_advbioinf.py
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python</span>
<span class="pl-c"><span class="pl-c">#</span> Written by Dan Barshis</span>
import sys, os
<span class="pl-c"><span class="pl-c">#</span>########## Usage #############</span>
<span class="pl-c"><span class="pl-c">#</span> Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles</span>
<span class="pl-c"><span class="pl-c">#</span> This should be run from within the folder with all of your original .fastqstrim</span>
<span class="pl-c"><span class="pl-c">#</span> Will quality trim multiple SINGLE-END fastq files</span>
<span class="pl-c"><span class="pl-c">#</span> Things to customize for your particular platform:</span>
[kconf001@turing1 scripts]$  </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="12">
                       <li>
                        cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 scripts]$ cd ../
[kconf001@turing1 KristinaC]$ cd data/
[kconf001@turing1 data]$ cd fastq/
[kconf001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq
[kconf001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq
[kconf001@turing1 fastq]$ ls 
adapterlist_advbioinf.txt   RI_B_01_18.fastq   RI_W_01_22.fastq   VA_B_09_SNP.fastq
KristinaCRenamer.sh         RI_B_01_22.fastq   RI_W_08_SNP.fastq  VA_W_01_14.fastq
KristinaCRenamer.txt        RI_B_08_SNP.fastq  VA_B_01_14.fastq   VA_W_01_18.fastq
renamingtable_complete.txt  RI_W_01_14.fastq   VA_B_01_18.fastq   VA_W_01_22.fastq
RI_B_01_14.fastq            RI_W_01_18.fastq   VA_B_01_22.fastq   VA_W_08_SNP.fastq" dir="auto">
                     <pre>[kconf001@turing1 scripts]$ <span class="pl-c1">cd</span> ../
[kconf001@turing1 KristinaC]$ <span class="pl-c1">cd</span> data/
[kconf001@turing1 data]$ <span class="pl-c1">cd</span> fastq/
[kconf001@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq
[kconf001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq
[kconf001@turing1 fastq]$ ls 
adapterlist_advbioinf.txt   RI_B_01_18.fastq   RI_W_01_22.fastq   VA_B_09_SNP.fastq
KristinaCRenamer.sh         RI_B_01_22.fastq   RI_W_08_SNP.fastq  VA_W_01_14.fastq
KristinaCRenamer.txt        RI_B_08_SNP.fastq  VA_B_01_14.fastq   VA_W_01_18.fastq
renamingtable_complete.txt  RI_W_01_14.fastq   VA_B_01_18.fastq   VA_W_01_22.fastq
RI_B_01_14.fastq            RI_W_01_18.fastq   VA_B_01_22.fastq   VA_W_08_SNP.fastq</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="13">
                       <li>
                        Make sbatch script for Trimclipfilter, then script &amp; run on fastq files
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
[kconf001@turing1 fastq]$ nano KristinaCTestTrimClip.sh
[kconf001@turing1 fastq]$ cat KristinaCTestTrimClip.sh

#!/bin/bash -l

#SBATCH -o KristinaCTestTrimClip.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCTestTrimClip
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq

[kconf001@turing1 fastq]$ salloc
salloc: Pending job allocation 9270528 
salloc: job 9270528 queued and waiting for resources
salloc: job 9270528 has been allocated resources
salloc: Granted job allocation 9270542
[kconf001@coreV2-25-072 fastq]$ sbatch KristinaCTestTrimClip.sh
Submitted batch job 9270546
[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9270542      main       sh kconf001  R      10:03      1 coreV2-25-072
9270546      main Kristina kconf001  R       0:03      1 coreV2-25-005
[kconf001@coreV2-25-072 fastq]$ ls -alh 
drwxrwxrwx 2 kconf001 users  872 Jan 27 17:35 . 
drwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 .. 
-rwxr-xr-x 1 kconf001 users 7.7K Jan 27 16:46 adapterlist_advbioinf.txt
-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh 
-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt
-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh
-rwxrwxrwx 1 kconf001 users  123 Jan 27 17:22 KristinaCTestTrimClip.txt
-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt
-rwxrwxrwx 1 kconf001 users 959M Jan 27 17:35 RI_B_01_14_clipped.fastq 
-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq 
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq 
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq
-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq
-rwxrwxrwx 1 kconf001 users   45 Jan 27 17:27 trimclipstats.txt 
-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq 
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq
-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq 
-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq
-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq " dir="auto">
                     <pre>[kconf001@turing1 fastq]$ nano KristinaCTestTrimClip.sh
[kconf001@turing1 fastq]$ cat KristinaCTestTrimClip.sh

<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCTestTrimClip.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCTestTrimClip</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt <span class="pl-k">*</span>.fastq

[kconf001@turing1 fastq]$ salloc
salloc: Pending job allocation 9270528 
salloc: job 9270528 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9270528 has been allocated resources
salloc: Granted job allocation 9270542
[kconf001@coreV2-25-072 fastq]$ sbatch KristinaCTestTrimClip.sh
Submitted batch job 9270546
[kconf001@coreV2-25-072 fastq]$ squeue -u kconf001
JOBID   PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9270542      main       sh kconf001  R      10:03      1 coreV2-25-072
9270546      main Kristina kconf001  R       0:03      1 coreV2-25-005
[kconf001@coreV2-25-072 fastq]$ ls -alh 
drwxrwxrwx 2 kconf001 users  872 Jan 27 17:35 <span class="pl-c1">.</span> 
drwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 .. 
-rwxr-xr-x 1 kconf001 users 7.7K Jan 27 16:46 adapterlist_advbioinf.txt
-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh 
-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt
-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh
-rwxrwxrwx 1 kconf001 users  123 Jan 27 17:22 KristinaCTestTrimClip.txt
-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt
-rwxrwxrwx 1 kconf001 users 959M Jan 27 17:35 RI_B_01_14_clipped.fastq 
-rw-r--r-- 1 kconf001 users 1.2G Jan 27 13:30 RI_B_01_14.fastq 
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:32 RI_B_01_22.fastq 
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:40 RI_B_08_SNP.fastq
-rw-r--r-- 1 kconf001 users 981M Jan 27 13:30 RI_W_01_14.fastq
-rw-r--r-- 1 kconf001 users 4.0G Jan 27 13:37 RI_W_01_18.fastq
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:31 RI_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 6.7G Jan 27 13:40 RI_W_08_SNP.fastq
-rwxrwxrwx 1 kconf001 users   45 Jan 27 17:27 trimclipstats.txt 
-rw-r--r-- 1 kconf001 users 3.4G Jan 27 13:30 VA_B_01_14.fastq 
-rw-r--r-- 1 kconf001 users 3.6G Jan 27 13:35 VA_B_01_18.fastq
-rw-r--r-- 1 kconf001 users 4.7G Jan 27 13:30 VA_B_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:39 VA_B_09_SNP.fastq
-rw-r--r-- 1 kconf001 users 2.6G Jan 27 13:29 VA_W_01_14.fastq 
-rw-r--r-- 1 kconf001 users 4.6G Jan 27 13:34 VA_W_01_18.fastq
-rw-r--r-- 1 kconf001 users  13M Jan 27 13:30 VA_W_01_22.fastq
-rw-r--r-- 1 kconf001 users 4.4G Jan 27 13:38 VA_W_08_SNP.fastq </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="14">
                       <li>
                        Job has finished
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 fastq]$ ls -alh                                                                                                                                                                                      
total 584K                                                                                                                                                                                                             
drwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 .
drwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..
drwxrwxrwx 2 kconf001 users 2.2K Jan 28 01:59 filteringstats
-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh
-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt
-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh
-rwxrwxrwx 1 kconf001 users 3.8K Jan 28 01:59 KristinaCTestTrimClip.txt
drwxrwxrwx 2 kconf001 users  591 Jan 28 01:59 originalfastqs
drwxrwxrwx 2 kconf001 users  788 Jan 28 01:59 QCFastqs
-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt" dir="auto">
                     <pre>[kconf001@turing1 fastq]$ ls -alh                                                                                                                                                                                      
total 584K                                                                                                                                                                                                             
drwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 <span class="pl-c1">.</span>
drwxrwxrwx 4 kconf001 users  212 Jan 27 14:59 ..
drwxrwxrwx 2 kconf001 users 2.2K Jan 28 01:59 filteringstats
-rwxrwxrwx 1 kconf001 users  226 Jan 27 16:00 KristinaCRenamer.sh
-rwxrwxrwx 1 kconf001 users  18K Jan 27 16:32 KristinaCRenamer.txt
-rwxrwxrwx 1 kconf001 users  310 Jan 27 17:26 KristinaCTestTrimClip.sh
-rwxrwxrwx 1 kconf001 users 3.8K Jan 28 01:59 KristinaCTestTrimClip.txt
drwxrwxrwx 2 kconf001 users  591 Jan 28 01:59 originalfastqs
drwxrwxrwx 2 kconf001 users  788 Jan 28 01:59 QCFastqs
-rwxr-xr-x 1 kconf001 users 4.6K Jan 27 15:00 renamingtable_complete.txt</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="15">
                       <li>
                        Updated Github &amp; README.md (See #7 on Homework Day 03 for similar workflow)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      01/29/2021
                     </h1>
                     <a aria-label="Permalink: 01/29/2021" class="anchor" href="#01292021" id="user-content-01292021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 04
                     </h2>
                     <a aria-label="Permalink: Homework Day 04" class="anchor" href="#homework-day-04" id="user-content-homework-day-04">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        Add trimclipstats.txt output to full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps
                       </li>
                      </ol>
                     </li>
                     <li>
                      1a. Run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015
This script takes a stats output file from fastx_clipper and converts it into a table.
Usage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt
Options (-c and -v must be listed separately to run together):
-h      Display this help message
-c      Use comma delimiter instead of tabs
-v      Verbose mode (print steps to stdout)
[kconf001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq" dir="auto">
                     <pre>[kconf001@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015
This script takes a stats output file from fastx_clipper and converts it into a table.
Usage: Schafran_trimstatstable.py [-c, -v, <span class="pl-k">-h</span>] inputfile.txt outputfile.txt
Options (-c and -v must be listed separately to run together):
-h      Display this <span class="pl-c1">help</span> message
-c      Use comma delimiter instead of tabs
-v      Verbose mode (print steps to stdout)
[kconf001@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      1b. Run script on data with outputfilename YOURNAME_trimclipstatsout.txt
                     </li>
                     <li>
                      Had to switch to filteringstats directory because this is where the trimclipstats.txt is located.
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 fastq]$ cd filteringstats/ 
[kconf001@turing1 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KristinaC_trimclipstatsout.txt 
[kconf001@turing1 filteringstats]$ ls  
KristinaC_trimclipstatsout.txt        RI_B_08_SNP_clipped_trimmed_stats.txt  RI_W_01_22_nucdist.png                 VA_B_01_18_nucdist.png                 VA_W_01_14_qualbox.png 
RI_B_01_14_clipped_trimmed_stats.txt  RI_B_08_SNP_nucdist.png                RI_W_01_22_qualbox.png                 VA_B_01_18_qualbox.png                 VA_W_01_18_clipped_trimmed_stats.txt 
RI_B_01_14_nucdist.png                RI_B_08_SNP_qualbox.png                RI_W_08_SNP_clipped_trimmed_stats.txt  VA_B_01_22_clipped_trimmed_stats.txt   VA_W_01_18_nucdist.png 
RI_B_01_14_qualbox.png                RI_W_01_14_clipped_trimmed_stats.txt   RI_W_08_SNP_nucdist.png                VA_B_01_22_nucdist.png                 VA_W_01_18_qualbox.png 
RI_B_01_18_clipped_trimmed_stats.txt  RI_W_01_14_nucdist.png                 RI_W_08_SNP_qualbox.png                VA_B_01_22_qualbox.png                 VA_W_01_22_clipped_trimmed_stats.txt 
RI_B_01_18_nucdist.png                RI_W_01_14_qualbox.png                 trimclipstats.txt                      VA_B_09_SNP_clipped_trimmed_stats.txt  VA_W_01_22_nucdist.png 
RI_B_01_18_qualbox.png                RI_W_01_18_clipped_trimmed_stats.txt   VA_B_01_14_clipped_trimmed_stats.txt   VA_B_09_SNP_nucdist.png                VA_W_01_22_qualbox.png         
RI_B_01_22_clipped_trimmed_stats.txt  RI_W_01_18_nucdist.png                 VA_B_01_14_nucdist.png                 VA_B_09_SNP_qualbox.png                VA_W_08_SNP_clipped_trimmed_stats.txt 
RI_B_01_22_nucdist.png                RI_W_01_18_qualbox.png                 VA_B_01_14_qualbox.png                 VA_W_01_14_clipped_trimmed_stats.txt   VA_W_08_SNP_nucdist.png 
RI_B_01_22_qualbox.png                RI_W_01_22_clipped_trimmed_stats.txt   VA_B_01_18_clipped_trimmed_stats.txt   VA_W_01_14_nucdist.png                 VA_W_08_SNP_qualbox.png " dir="auto">
                     <pre>[kconf001@turing1 fastq]$ <span class="pl-c1">cd</span> filteringstats/ 
[kconf001@turing1 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KristinaC_trimclipstatsout.txt 
[kconf001@turing1 filteringstats]$ ls  
KristinaC_trimclipstatsout.txt        RI_B_08_SNP_clipped_trimmed_stats.txt  RI_W_01_22_nucdist.png                 VA_B_01_18_nucdist.png                 VA_W_01_14_qualbox.png 
RI_B_01_14_clipped_trimmed_stats.txt  RI_B_08_SNP_nucdist.png                RI_W_01_22_qualbox.png                 VA_B_01_18_qualbox.png                 VA_W_01_18_clipped_trimmed_stats.txt 
RI_B_01_14_nucdist.png                RI_B_08_SNP_qualbox.png                RI_W_08_SNP_clipped_trimmed_stats.txt  VA_B_01_22_clipped_trimmed_stats.txt   VA_W_01_18_nucdist.png 
RI_B_01_14_qualbox.png                RI_W_01_14_clipped_trimmed_stats.txt   RI_W_08_SNP_nucdist.png                VA_B_01_22_nucdist.png                 VA_W_01_18_qualbox.png 
RI_B_01_18_clipped_trimmed_stats.txt  RI_W_01_14_nucdist.png                 RI_W_08_SNP_qualbox.png                VA_B_01_22_qualbox.png                 VA_W_01_22_clipped_trimmed_stats.txt 
RI_B_01_18_nucdist.png                RI_W_01_14_qualbox.png                 trimclipstats.txt                      VA_B_09_SNP_clipped_trimmed_stats.txt  VA_W_01_22_nucdist.png 
RI_B_01_18_qualbox.png                RI_W_01_18_clipped_trimmed_stats.txt   VA_B_01_14_clipped_trimmed_stats.txt   VA_B_09_SNP_nucdist.png                VA_W_01_22_qualbox.png         
RI_B_01_22_clipped_trimmed_stats.txt  RI_W_01_18_nucdist.png                 VA_B_01_14_nucdist.png                 VA_B_09_SNP_qualbox.png                VA_W_08_SNP_clipped_trimmed_stats.txt 
RI_B_01_22_nucdist.png                RI_W_01_18_qualbox.png                 VA_B_01_14_qualbox.png                 VA_W_01_14_clipped_trimmed_stats.txt   VA_W_08_SNP_nucdist.png 
RI_B_01_22_qualbox.png                RI_W_01_22_clipped_trimmed_stats.txt   VA_B_01_18_clipped_trimmed_stats.txt   VA_W_01_14_nucdist.png                 VA_W_08_SNP_qualbox.png </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      1c. Add KristinaC_trimclipstatsout.txt to the class file (Fulltrimclipstatstable.txt).
                     </li>
                     <li>
                      check to make sure things are in the correct directory first!
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 filteringstats]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/
[kconf001@turing1 Astrangia_poculata]$ ls -alh 
total 320K
drwxrwxrwx 4 dbarshis users  105 Jan 29 01:17 .
drwxrwxrwx 3 dbarshis users   36 Jan 21 23:36 ..
-rwxr-xr-x 1 dbarshis users  29K Jan 29 01:18 Fulltrimclipstatstable.txt 
drwxrwxrwx 2 dbarshis users 4.8K Jan 21 23:37 originalfastqs 
drwxrwxrwx 2 dbarshis users  337 Jan 29 00:52 refassembly 
[kconf001@turing1 Astrangia_poculata]$ cd /cm/shared/courses//dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats/
[kconf001@turing1 filteringstats]$ salloc
salloc: Pending job allocation 9271098
salloc: job 9271098 queued and waiting for resources
salloc: job 9271098 has been allocated resources
salloc: Granted job allocation 9271098 
[kconf001@coreV2-25-072 filteringstats]$ tail -n +2 KristinaC_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt
" dir="auto">
                     <pre>[kconf001@turing1 filteringstats]$ <span class="pl-c1">cd</span> /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/
[kconf001@turing1 Astrangia_poculata]$ ls -alh 
total 320K
drwxrwxrwx 4 dbarshis users  105 Jan 29 01:17 <span class="pl-c1">.</span>
drwxrwxrwx 3 dbarshis users   36 Jan 21 23:36 ..
-rwxr-xr-x 1 dbarshis users  29K Jan 29 01:18 Fulltrimclipstatstable.txt 
drwxrwxrwx 2 dbarshis users 4.8K Jan 21 23:37 originalfastqs 
drwxrwxrwx 2 dbarshis users  337 Jan 29 00:52 refassembly 
[kconf001@turing1 Astrangia_poculata]$ <span class="pl-c1">cd</span> /cm/shared/courses//dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats/
[kconf001@turing1 filteringstats]$ salloc
salloc: Pending job allocation 9271098
salloc: job 9271098 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9271098 has been allocated resources
salloc: Granted job allocation 9271098 
[kconf001@coreV2-25-072 filteringstats]$ tail -n +2 KristinaC_trimclipstatsout.txt <span class="pl-k">&gt;&gt;</span> /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt
</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Map reads back to assembly using Bowtie2 alignment algorithm
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        Write sbatch (in QCFastqs directory) script to do Bowtie2/2.4 on _clippedtrimmedfiltered.fastq data files from data in personal sandbox
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-072 filteringstats]$ cd ../
[kconf001@coreV2-25-072 fastq]$ cd QCFastqs/
[kconf001@coreV2-25-072 QCFastqs]$ nano KristinaCBowtieln.sh
[kconf001@coreV2-25-072 QCFastqs]$ cat KristinaCBowtieln.sh

#!/bin/bash -l

#SBATCH -o KristinaCbowtieln.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCBowtieln

module load bowtie2/2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \
--rg SM:${i%_clippedtrimmed.fastq} \
--very-sensitive -x /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/refassembly/ast_hostsym -U $i \
&gt; ${i%_clippedtrimmedfilterd.fastq}.sam; done

[kconf001@coreV2-25-072 QCFastqs]$ sbatch KristinaCBowtieln.sh 
Submitted batch job 9271110 
[kconf001@coreV2-25-072 QCFastqs]$ squeue -u kconf001 
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271110      main Kristina kconf001  R       0:01      1 coreV2-25-007
           9271098      main       sh kconf001  R      35:31      1 coreV2-25-072 " dir="auto">
                     <pre>[kconf001@coreV2-25-072 filteringstats]$ <span class="pl-c1">cd</span> ../
[kconf001@coreV2-25-072 fastq]$ <span class="pl-c1">cd</span> QCFastqs/
[kconf001@coreV2-25-072 QCFastqs]$ nano KristinaCBowtieln.sh
[kconf001@coreV2-25-072 QCFastqs]$ cat KristinaCBowtieln.sh

<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCbowtieln.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCBowtieln</span>

module load bowtie2/2.4
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>_clippedtrimmed.fastq<span class="pl-k">;</span> <span class="pl-k">do</span> bowtie2 --rg-id <span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmed.fastq}</span> \
--rg SM:<span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmed.fastq}</span> \
--very-sensitive -x /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/refassembly/ast_hostsym -U <span class="pl-smi">$i</span> \
<span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmedfilterd.fastq}</span>.sam<span class="pl-k">;</span> <span class="pl-k">done</span>

[kconf001@coreV2-25-072 QCFastqs]$ sbatch KristinaCBowtieln.sh 
Submitted batch job 9271110 
[kconf001@coreV2-25-072 QCFastqs]$ squeue -u kconf001 
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
           9271110      main Kristina kconf001  R       0:01      1 coreV2-25-007
           9271098      main       sh kconf001  R      35:31      1 coreV2-25-072 </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      Job is complete; should get _clippedtrimmed.fastq.sam files
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 QCFastqs]$ ls -alh
total 184G
drwxrwxrwx 2 kconf001 users 1.7K Jan 29 21:18 .
drwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 ..
-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh 
-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt
-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt
-rwxrwxrwx 1 kconf001 users 1.1G Jan 28 00:45 RI_B_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 1.8G Jan 29 15:25 RI_B_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:47 RI_B_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 15:54 RI_B_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:50 RI_B_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 16:23 RI_B_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.2G Jan 28 00:52 RI_B_08_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.2G Jan 29 16:51 RI_B_08_SNP_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 886M Jan 28 00:53 RI_W_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 1.4G Jan 29 16:59 RI_W_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.6G Jan 28 00:56 RI_W_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.9G Jan 29 17:29 RI_W_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 00:58 RI_W_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 17:53 RI_W_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 5.9G Jan 28 01:02 RI_W_08_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 9.7G Jan 29 18:41 RI_W_08_SNP_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 01:05 VA_B_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 19:04 VA_B_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.3G Jan 28 01:07 VA_B_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.1G Jan 29 19:25 VA_B_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 4.2G Jan 28 01:10 VA_B_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.9G Jan 29 19:58 VA_B_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:13 VA_B_09_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.5G Jan 29 20:29 VA_B_09_SNP_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 2.3G Jan 28 01:15 VA_W_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 3.7G Jan 29 20:46 VA_W_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 4.1G Jan 28 01:18 VA_W_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.7G Jan 29 21:18 VA_W_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users  11M Jan 28 01:18 VA_W_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users  22M Jan 29 21:18 VA_W_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:21 VA_W_08_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.4G Jan 29 21:48 VA_W_08_SNP_clippedtrimmed.fastq.sam" dir="auto">
                     <pre>[kconf001@turing1 QCFastqs]$ ls -alh
total 184G
drwxrwxrwx 2 kconf001 users 1.7K Jan 29 21:18 <span class="pl-c1">.</span>
drwxrwxrwx 5 kconf001 users  294 Jan 28 01:59 ..
-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh 
-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt
-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt
-rwxrwxrwx 1 kconf001 users 1.1G Jan 28 00:45 RI_B_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 1.8G Jan 29 15:25 RI_B_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:47 RI_B_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 15:54 RI_B_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.5G Jan 28 00:50 RI_B_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.7G Jan 29 16:23 RI_B_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.2G Jan 28 00:52 RI_B_08_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.2G Jan 29 16:51 RI_B_08_SNP_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 886M Jan 28 00:53 RI_W_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 1.4G Jan 29 16:59 RI_W_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.6G Jan 28 00:56 RI_W_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.9G Jan 29 17:29 RI_W_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 00:58 RI_W_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 17:53 RI_W_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 5.9G Jan 28 01:02 RI_W_08_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 9.7G Jan 29 18:41 RI_W_08_SNP_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.0G Jan 28 01:05 VA_B_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 4.9G Jan 29 19:04 VA_B_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.3G Jan 28 01:07 VA_B_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 5.1G Jan 29 19:25 VA_B_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 4.2G Jan 28 01:10 VA_B_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.9G Jan 29 19:58 VA_B_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:13 VA_B_09_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.5G Jan 29 20:29 VA_B_09_SNP_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 2.3G Jan 28 01:15 VA_W_01_14_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 3.7G Jan 29 20:46 VA_W_01_14_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 4.1G Jan 28 01:18 VA_W_01_18_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.7G Jan 29 21:18 VA_W_01_18_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users  11M Jan 28 01:18 VA_W_01_22_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users  22M Jan 29 21:18 VA_W_01_22_clippedtrimmed.fastq.sam
-rwxrwxrwx 1 kconf001 users 3.9G Jan 28 01:21 VA_W_08_SNP_clippedtrimmed.fastq
-rwxrwxrwx 1 kconf001 users 6.4G Jan 29 21:48 VA_W_08_SNP_clippedtrimmed.fastq.sam</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        Updated README.md &amp; github (See #7 on Homework Day 03 for similar workflow)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      02/03/2021
                     </h1>
                     <a aria-label="Permalink: 02/03/2021" class="anchor" href="#02032021" id="user-content-02032021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 05
                     </h2>
                     <a aria-label="Permalink: Homework Day 05" class="anchor" href="#homework-day-05" id="user-content-homework-day-05">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        Combine data with partner (KCKC directory- with Katie Crider)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs *.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq" dir="auto">
                     <pre>cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs <span class="pl-k">*</span>.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Run Triniry denovo assembler for two the two combined lanes
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        Modify sbatch script below
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#!/bin/bash -l

#SBATCH -o KCKCTrinitydenovo.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCKCTrinitydenovo

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCKCTrinitydenovo.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 32</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -p himem</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCKCTrinitydenovo</span>

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        Check
                        <a href="https://trinityrnaseq.github.io/" rel="nofollow">
                         https://trinityrnaseq.github.io/
                        </a>
                        for usage info
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        Submit Trinity script
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-072 QCFastqs]$ sbatch KCKCTrinity.sh " dir="auto">
                     <pre>[kconf001@coreV2-25-072 QCFastqs]$ sbatch KCKCTrinity.sh </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="6">
                       <li>
                        We were experiencing some problems with the job input, had to change some parts of the .sbatch script.
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      02/05/2021
                     </h1>
                     <a aria-label="Permalink: 02/05/2021" class="anchor" href="#02052021" id="user-content-02052021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 6
                     </h2>
                     <a aria-label="Permalink: Homework Day 6" class="anchor" href="#homework-day-6" id="user-content-homework-day-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        Start interactive session via salloc &amp; run cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~ $ ssh kconf001@turing.hpc.odu.edu
kconf001@turing.hpc.odu.edu's password:
Last login: Wed Feb  3 15:02:28 2021 from ip70-160-48-140.hr.hr.cox.net
[kconf001@turing1 ~]$ salloc
salloc: Pending job allocation 9272854
salloc: job 9272854 queued and waiting for resources
salloc: job 9272854 has been allocated resources
salloc: Granted job allocation 9272854
[kconf001@coreV3-23-024 ~]$cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir
[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta
################################
## Counts of transcripts, etc.
################################
Total trinity 'genes':  39219
Total trinity transcripts: 41303
Percent GC: 44.54
########################################
Stats based on ALL transcript contigs:
########################################
Contig N10: 1323
Contig N20: 783
Contig N30: 562
Contig N40: 440
Contig N50: 364
Median contig length: 277
Average contig: 371.08
Total assembled bases: 15326666
#####################################################
## Stats based on ONLY LONGEST ISOFORM per 'GENE':
#####################################################
Contig N10: 1094
Contig N20: 682
Contig N30: 511
Contig N40: 410
Contig N50: 347
Median contig length: 274
Average contig: 355.69
Total assembled bases: 13949619" dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span> $ ssh kconf001@turing.hpc.odu.edu
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">Last login: Wed Feb  3 15:02:28 2021 from ip70-160-48-140.hr.hr.cox.net</span>
<span class="pl-s">[kconf001@turing1 ~]$ salloc</span>
<span class="pl-s">salloc: Pending job allocation 9272854</span>
<span class="pl-s">salloc: job 9272854 queued and waiting for resources</span>
<span class="pl-s">salloc: job 9272854 has been allocated resources</span>
<span class="pl-s">salloc: Granted job allocation 9272854</span>
<span class="pl-s">[kconf001@coreV3-23-024 ~]$cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir</span>
<span class="pl-s">[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta</span>
<span class="pl-s">################################</span>
<span class="pl-s">## Counts of transcripts, etc.</span>
<span class="pl-s">################################</span>
<span class="pl-s">Total trinity <span class="pl-pds">'</span></span>genes<span class="pl-s"><span class="pl-pds">'</span>:  39219</span>
<span class="pl-s">Total trinity transcripts: 41303</span>
<span class="pl-s">Percent GC: 44.54</span>
<span class="pl-s">########################################</span>
<span class="pl-s">Stats based on ALL transcript contigs:</span>
<span class="pl-s">########################################</span>
<span class="pl-s">Contig N10: 1323</span>
<span class="pl-s">Contig N20: 783</span>
<span class="pl-s">Contig N30: 562</span>
<span class="pl-s">Contig N40: 440</span>
<span class="pl-s">Contig N50: 364</span>
<span class="pl-s">Median contig length: 277</span>
<span class="pl-s">Average contig: 371.08</span>
<span class="pl-s">Total assembled bases: 15326666</span>
<span class="pl-s">#####################################################</span>
<span class="pl-s">## Stats based on ONLY LONGEST ISOFORM per <span class="pl-pds">'</span></span>GENE<span class="pl-s"><span class="pl-pds">'</span>:</span>
<span class="pl-s">#####################################################</span>
<span class="pl-s">Contig N10: 1094</span>
<span class="pl-s">Contig N20: 682</span>
<span class="pl-s">Contig N30: 511</span>
<span class="pl-s">Contig N40: 410</span>
<span class="pl-s">Contig N50: 347</span>
<span class="pl-s">Median contig length: 274</span>
<span class="pl-s">Average contig: 355.69</span>
<span class="pl-s">Total assembled bases: 13949619</span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Compare this with the output from avg_cov_len_fasta_advbioinf.py
on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
The total number of sequences is 15079
The average sequence length is 876
The total number of bases is 13210470
The minimum sequence length is 500
The maximum sequence length is 10795
The N50 is 881
Median Length = 578
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 15079
contigs &gt;= 1000bp = 3660
contigs &gt;= 2000bp = 536" dir="auto">
                     <pre>[kconf001@coreV3-23-024 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
The total number of sequences is 15079
The average sequence length is 876
The total number of bases is 13210470
The minimum sequence length is 500
The maximum sequence length is 10795
The N50 is 881
Median Length = 578
contigs <span class="pl-k">&lt;</span> 150bp = 0
contigs <span class="pl-k">&gt;</span>= 500bp = 15079
contigs <span class="pl-k">&gt;</span>= 1000bp = 3660
contigs <span class="pl-k">&gt;</span>= 2000bp = 536</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        Less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:
                       </li>
                      </ol>
                     </li>
                     <li>
                      3a. Mean % "Overall alignment rate"
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[kconf001@coreV3-23-024 trinirt_our_dir]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs 
[kconf001@coreV3-23-024 QCFastqs]$ grep "overall alignment rate" KristinaCbowtieln.txt
89.71% overall alignment rate
91.95% overall alignment rate
91.31% overall alignment rate
92.53% overall alignment rate
65.88% overall alignment rate
93.03% overall alignment rate
93.80% overall alignment rate
93.07% overall alignment rate
90.51% overall alignment rate
79.87% overall alignment rate
91.41% overall alignment rate
92.94% overall alignment rate
93.42% overall alignment rate
93.54% overall alignment rate
95.85% overall alignment rate
92.74% overall alignment rate
[kconf001@coreV3-23-024 QCFastqs]$ grep "overall alignment rate" KristinaCbowtieln.txt | cut -d " " -f 1 | cut -d "%" -f 1
89.71
91.95
91.31
92.53
65.88
93.03
93.80
93.07
90.51
79.87
91.41
92.94
93.42
93.54
95.85
92.74 
Calculate mean on Excel, mean % overall alignment rate is 89.89933' dir="auto">
                     <pre>[kconf001@coreV3-23-024 trinirt_our_dir]$ <span class="pl-c1">cd</span> /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs 
[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span> KristinaCbowtieln.txt
89.71% overall alignment rate
91.95% overall alignment rate
91.31% overall alignment rate
92.53% overall alignment rate
65.88% overall alignment rate
93.03% overall alignment rate
93.80% overall alignment rate
93.07% overall alignment rate
90.51% overall alignment rate
79.87% overall alignment rate
91.41% overall alignment rate
92.94% overall alignment rate
93.42% overall alignment rate
93.54% overall alignment rate
95.85% overall alignment rate
92.74% overall alignment rate
[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span> KristinaCbowtieln.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span> -f 1 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1
89.71
91.95
91.31
92.53
65.88
93.03
93.80
93.07
90.51
79.87
91.41
92.94
93.42
93.54
95.85
92.74 
Calculate mean on Excel, mean % overall alignment rate is 89.89933</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      3b. Mean % reads "aligned exactly 1 time"
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[kconf001@coreV3-23-024 QCFastqs]$ grep "aligned exactly 1 time" KristinaCbowtieln.txt
425341 (5.89%) aligned exactly 1 time
1336595 (5.57%) aligned exactly 1 time
1309089 (5.42%) aligned exactly 1 time
1148628 (5.28%) aligned exactly 1 time
313795 (5.25%) aligned exactly 1 time
1058240 (4.28%) aligned exactly 1 time
894094 (4.34%) aligned exactly 1 time
1825387 (4.49%) aligned exactly 1 time
1191456 (5.74%) aligned exactly 1 time
1757128 (7.79%) aligned exactly 1 time
1659193 (5.71%) aligned exactly 1 time
1277294 (4.72%) aligned exactly 1 time
626232 (4.08%) aligned exactly 1 time
1217713 (4.36%) aligned exactly 1 time
1865 (2.60%) aligned exactly 1 time
1229632 (4.58%) aligned exactly 1 time
[kconf001@coreV3-23-024 QCFastqs]$ grep "aligned exactly 1 time" KristinaCbowtieln.txt | cut -d "(" -f 2 | cut -d "%" -f 1
5.89
5.57
5.42
5.28
5.25
4.28
4.34
4.49
5.74
7.79
5.71
4.72
4.08
4.36
2.60
4.58
Calculate mean on Excel, mean % reads aligned exactly 1 time is 5.00625' dir="auto">
                     <pre>[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> KristinaCbowtieln.txt
425341 (5.89%) aligned exactly 1 <span class="pl-k">time</span>
1336595 (5.57%) aligned exactly 1 <span class="pl-k">time</span>
1309089 (5.42%) aligned exactly 1 <span class="pl-k">time</span>
1148628 (5.28%) aligned exactly 1 <span class="pl-k">time</span>
313795 (5.25%) aligned exactly 1 <span class="pl-k">time</span>
1058240 (4.28%) aligned exactly 1 <span class="pl-k">time</span>
894094 (4.34%) aligned exactly 1 <span class="pl-k">time</span>
1825387 (4.49%) aligned exactly 1 <span class="pl-k">time</span>
1191456 (5.74%) aligned exactly 1 <span class="pl-k">time</span>
1757128 (7.79%) aligned exactly 1 <span class="pl-k">time</span>
1659193 (5.71%) aligned exactly 1 <span class="pl-k">time</span>
1277294 (4.72%) aligned exactly 1 <span class="pl-k">time</span>
626232 (4.08%) aligned exactly 1 <span class="pl-k">time</span>
1217713 (4.36%) aligned exactly 1 <span class="pl-k">time</span>
1865 (2.60%) aligned exactly 1 <span class="pl-k">time</span>
1229632 (4.58%) aligned exactly 1 <span class="pl-k">time</span>
[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> KristinaCbowtieln.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span> -f 2 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1
5.89
5.57
5.42
5.28
5.25
4.28
4.34
4.49
5.74
7.79
5.71
4.72
4.08
4.36
2.60
4.58
Calculate mean on Excel, mean % reads aligned exactly 1 <span class="pl-k">time</span> is 5.00625</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      3c. Mean # of reads "aligned exactly 1 time
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[kconf001@coreV3-23-024 QCFastqs]$ grep "aligned exactly 1 time" KristinaCbowtieln.txt
425341 (5.89%) aligned exactly 1 time
1336595 (5.57%) aligned exactly 1 time
1309089 (5.42%) aligned exactly 1 time
1148628 (5.28%) aligned exactly 1 time
313795 (5.25%) aligned exactly 1 time
1058240 (4.28%) aligned exactly 1 time
894094 (4.34%) aligned exactly 1 time
1825387 (4.49%) aligned exactly 1 time
1191456 (5.74%) aligned exactly 1 time
1757128 (7.79%) aligned exactly 1 time
1659193 (5.71%) aligned exactly 1 time
1277294 (4.72%) aligned exactly 1 time
626232 (4.08%) aligned exactly 1 time
1217713 (4.36%) aligned exactly 1 time
1865 (2.60%) aligned exactly 1 time
1229632 (4.58%) aligned exactly 1 time
[kconf001@coreV3-23-024 QCFastqs]$ grep "aligned exactly 1 time" KristinaCbowtieln.txt | cut -d " " -f 5
425341
1336595
1309089
1148628
313795
1058240
894094
1825387
1191456
1757128
1659193
1277294
626232
1217713
1865
1229632
Calculate mean on Excel, mean # of reads alligned exactly 1 time is 1079480' dir="auto">
                     <pre>[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> KristinaCbowtieln.txt
425341 (5.89%) aligned exactly 1 <span class="pl-k">time</span>
1336595 (5.57%) aligned exactly 1 <span class="pl-k">time</span>
1309089 (5.42%) aligned exactly 1 <span class="pl-k">time</span>
1148628 (5.28%) aligned exactly 1 <span class="pl-k">time</span>
313795 (5.25%) aligned exactly 1 <span class="pl-k">time</span>
1058240 (4.28%) aligned exactly 1 <span class="pl-k">time</span>
894094 (4.34%) aligned exactly 1 <span class="pl-k">time</span>
1825387 (4.49%) aligned exactly 1 <span class="pl-k">time</span>
1191456 (5.74%) aligned exactly 1 <span class="pl-k">time</span>
1757128 (7.79%) aligned exactly 1 <span class="pl-k">time</span>
1659193 (5.71%) aligned exactly 1 <span class="pl-k">time</span>
1277294 (4.72%) aligned exactly 1 <span class="pl-k">time</span>
626232 (4.08%) aligned exactly 1 <span class="pl-k">time</span>
1217713 (4.36%) aligned exactly 1 <span class="pl-k">time</span>
1865 (2.60%) aligned exactly 1 <span class="pl-k">time</span>
1229632 (4.58%) aligned exactly 1 <span class="pl-k">time</span>
[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> KristinaCbowtieln.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span> -f 5
425341
1336595
1309089
1148628
313795
1058240
894094
1825387
1191456
1757128
1659193
1277294
626232
1217713
1865
1229632
Calculate mean on Excel, mean <span class="pl-c"><span class="pl-c">#</span> of reads alligned exactly 1 time is 1079480</span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      3d. Mean % reads "aligned &gt;1 times"
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[kconf001@coreV3-23-024 QCFastqs]$ grep "aligned &gt;1 times" KristinaCbowtieln.txt
6056902 (83.83%) aligned &gt;1 times
20712542 (86.38%) aligned &gt;1 times
20741901 (85.89%) aligned &gt;1 times
18972847 (87.25%) aligned &gt;1 times
3627297 (60.63%) aligned &gt;1 times
21931999 (88.74%) aligned &gt;1 times
18425413 (89.46%) aligned &gt;1 times
35983989 (88.58%) aligned &gt;1 times
17579848 (84.76%) aligned &gt;1 times
16251697 (72.08%) aligned &gt;1 times
24925544 (85.71%) aligned &gt;1 times
23877047 (88.22%) aligned &gt;1 times
13722839 (89.34%) aligned &gt;1 times
24899339 (89.18%) aligned &gt;1 times
66899 (93.25%) aligned &gt;1 times
23648840 (88.15%) aligned &gt;1 times
[kconf001@coreV3-23-024 QCFastqs]$ grep "aligned &gt;1 times" KristinaCbowtieln.txt | cut -d "(" -f 2 | cut -d "%" -f 1
83.83
86.38
85.89
87.25
60.63
88.74
89.46
88.58
84.76
72.08
85.71
88.22
89.34
89.18
93.25
88.15
Calculate mean on Excel, mean % reads aligned &gt;1 times is 85.09063.' dir="auto">
                     <pre>[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span> KristinaCbowtieln.txt
6056902 (83.83%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
20712542 (86.38%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
20741901 (85.89%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
18972847 (87.25%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
3627297 (60.63%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
21931999 (88.74%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
18425413 (89.46%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
35983989 (88.58%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
17579848 (84.76%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
16251697 (72.08%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
24925544 (85.71%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
23877047 (88.22%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
13722839 (89.34%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
24899339 (89.18%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
66899 (93.25%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
23648840 (88.15%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
[kconf001@coreV3-23-024 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span> KristinaCbowtieln.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span> -f 2 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1
83.83
86.38
85.89
87.25
60.63
88.74
89.46
88.58
84.76
72.08
85.71
88.22
89.34
89.18
93.25
88.15
Calculate mean on Excel, mean % reads aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span> is 85.09063.</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        Add statistics as single rows to shared table
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-024 QCFastqs]$ nano KristinaCalignstats.txt                                                                                                            
[kconf001@coreV3-23-024 QCFastqs]$ cat KristinaCalignstats.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt" dir="auto">
                     <pre>[kconf001@coreV3-23-024 QCFastqs]$ nano KristinaCalignstats.txt                                                                                                            
[kconf001@coreV3-23-024 QCFastqs]$ cat KristinaCalignstats.txt <span class="pl-k">&gt;&gt;</span> /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      5 &amp; 6: Having issues accessing Trinity.fasta files (file &amp; directory not found)- managed to find Trinity.fasta files, currently sbatch is running
                     </li>
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        Data cleanup &amp; archiving
                       </li>
                      </ol>
                     </li>
                     <li>
                      5a. Move Trinity.fasta file into new testassembly directory
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 testassembly]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/       
[kconf001@coreV3-23-040 testassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/testassembly" dir="auto">
                     <pre>[kconf001@coreV3-23-040 testassembly]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/       
[kconf001@coreV3-23-040 testassembly]$ mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/testassembly</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      5b. sbatch script to rm -r original fastqs * filteringstats (Could not remove -trinity_out_dir since it is already gone- forgot to cp to sandbox
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 testassembly]$ nano Cleanup.sh
[kconf001@coreV3-23-040 testassembly]$ cat Cleanup.sh
#!/bin/bash -l

#SBATCH -o KristinaCCleanup.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCCleanup

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats
[kconf001@coreV3-23-040 testassembly]$ sbatch Cleanup.sh
Submitted batch job 9276432" dir="auto">
                     <pre>[kconf001@coreV3-23-040 testassembly]$ nano Cleanup.sh
[kconf001@coreV3-23-040 testassembly]$ cat Cleanup.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCCleanup.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCCleanup</span>

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/filteringstats
[kconf001@coreV3-23-040 testassembly]$ sbatch Cleanup.sh
Submitted batch job 9276432</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="6">
                       <li>
                        Submit blast job from testassembly directory
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@turing1 testassembly]$ nano KristinaCBlast.sh
[kconf001@turing1 testassembly]$ cat KristinaCBlast.sh
#!/bin/bash -l
#SBATCH -o KristinaCBlast.txt
#SBATCH -n 6
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCBlast
enable_lmod container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
-evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
[kconf001@turing1 testassembly]$ sbatch KristinaCBlast.sh
Submitted batch job 9276434
[kconf001@turing1 testassembly]$ squeue -u kconf001
JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9276434      main Kristina kconf001 PD       0:00      1 (Priority)
9276428      main       sh kconf001  R      45:55      1 coreV3-23-040
9276430      main Kristina kconf001  R      29:25      1 coreV3-23-040                                              " dir="auto">
                     <pre>[kconf001@turing1 testassembly]$ nano KristinaCBlast.sh
[kconf001@turing1 testassembly]$ cat KristinaCBlast.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCBlast.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 6</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCBlast</span>
enable_lmod container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
-evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
[kconf001@turing1 testassembly]$ sbatch KristinaCBlast.sh
Submitted batch job 9276434
[kconf001@turing1 testassembly]$ squeue -u kconf001
JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9276434      main Kristina kconf001 PD       0:00      1 (Priority)
9276428      main       sh kconf001  R      45:55      1 coreV3-23-040
9276430      main Kristina kconf001  R      29:25      1 coreV3-23-040                                              </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="7">
                       <li>
                        Head one of the .sam files to look at the header
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 QCFastqs]$ head VA_W_08_SNP_clippedtrimmed.fastq.sam
@HD     VN:1.0  SO:unsorted
@SQ     SN:AstrangiaT1FW_1      LN:4823
@SQ     SN:AstrangiaT1FW_10     LN:1096
@SQ     SN:AstrangiaT1FW_11     LN:2560
@SQ     SN:AstrangiaT1FW_12     LN:2686
@SQ     SN:AstrangiaT1FW_13     LN:629
@SQ     SN:AstrangiaT1FW_15     LN:582
@SQ     SN:AstrangiaT1FW_16     LN:502
@SQ     SN:AstrangiaT1FW_17     LN:1492
@SQ     SN:AstrangiaT1FW_18     LN:1197" dir="auto">
                     <pre>[kconf001@coreV3-23-040 QCFastqs]$ head VA_W_08_SNP_clippedtrimmed.fastq.sam
@HD     VN:1.0  SO:unsorted
@SQ     SN:AstrangiaT1FW_1      LN:4823
@SQ     SN:AstrangiaT1FW_10     LN:1096
@SQ     SN:AstrangiaT1FW_11     LN:2560
@SQ     SN:AstrangiaT1FW_12     LN:2686
@SQ     SN:AstrangiaT1FW_13     LN:629
@SQ     SN:AstrangiaT1FW_15     LN:582
@SQ     SN:AstrangiaT1FW_16     LN:502
@SQ     SN:AstrangiaT1FW_17     LN:1492
@SQ     SN:AstrangiaT1FW_18     LN:1197</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="8">
                       <li>
                        grep -v '@' sam | head to look at sequence read files, this works because '@' denotes the start of a sequence.
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[kconf001@coreV3-23-040 QCFastqs]$ grep -v "@" VA_W_08_SNP_clippedtrimmed.fastq.sam | head                                                                                                                                                     
K00188:59:HMTFHBBXX:2:1101:4026:1648    0       AstrangiaT1FW_65377     610     1       51M     *       0       0       CNCGTTAATCCATTCATGCGCGTCACTAATTAGATGACGAGGCATTTGGCT     A#AAF-A-AFFJAJFJJA&lt;7-AJJA7JJJ&lt;FJJJJ&lt;F&lt;&lt;JFFJJJJ-AJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:4614:1648    0       AstrangiaT1FW_488087    41      1       51M     *       0       0       GNTCTTGATTAATGAAAACATTCTTGGCAAATGCTTTCGCAGTAGTTCGTC     A#AFFJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJJFJJJJJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5690:1648    16      AstrangiaT1FW_65728     10184   1       51M     *       0       0       AGACGAACTACTGCGAAAGCATTTGCCAAGAATGTTTTCATTAATCAAGNG     FJFAJJJAJJJJJFJJJJJJJJJJF-AJFF&lt;JJJJJJJJJJJJAJJFFA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49A0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       
K00188:59:HMTFHBBXX:2:1101:3772:1666    16      AstrangiaT1FW_65718     4457    1       51M     *       0       0       TTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTGGACATGNG     7JJFFJJAJJF&lt;JJJJFJJJJJJJJ7JFJJ&lt;FFA&lt;JJJFJJJJJJJFFA#&lt;    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:49G1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5051:1666    16      AstrangiaT1FW_131671    3       1       51M     *       0       0       ACAGATGTGCCGCCCCAGCCAAACTCCCAACCTGACGGTGTCTTGGACTNG     JJFJJFJJFJJJJJJFJJJJJFJJJJJJFFJFFJAJJJJJJJJJJJFAA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49C0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5172:1666    0       AstrangiaT1FW_65662     1901    1       51M     *       0       0       TNCGACGGGCGGTGTGTACAAAGGGCAGGGACGTAATCAACGCGAGCTGAT     A#AFFJAJJJJJJJFJFFJJJJFF-AJJJJJJJJJJJJJJJFJJJJJAJJ7    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:0A0G49     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       
K00188:59:HMTFHBBXX:2:1101:5680:1666    0       AstrangiaT1FW_65668     593     1       51M     *       0       0       GNGCATTGCTGAGTCCATTCGTTTCCCTTTCAACGGTTTCACGTACTTTTT     A#-AFFJFJJ7JJJJJJFFJ&lt;FFFF-FJJFJJJF&lt;AFF7FJJAJFJJ7FJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:6674:1666    16      AstrangiaT1FW_65718     1086    1       23M     *       0       0       TCCCAGGAGCATGTCTGTCTGNG FF&lt;JJJJAJJJJJJJJJJFFA#A AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0 NM:i:1  MD:Z:21A1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                                                                                       
K00188:59:HMTFHBBXX:2:1101:7060:1666    0       AstrangiaT1FW_65712     6946    1       51M     *       0       0       CNAAATTTGACGATCGATTTGCACGTCAGAATCGCTACGAGCCTCCACCAG     A#AFFJFFJJJJJFJJJJJJJJJJ&lt;-FJJJJFJJJJJJJJJJJJJJJJJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:7080:1666    0       AstrangiaT1FW_65718     10568   0       51M     *       0       0       TNTTGCAATAGTAATTCTGCTCAGTACGAGAGGAACCGCAGATTCAGACAA     A#AFFJJJFJJJJJJJJJJJ-FJAAAJJJFJJJJJJJJJJJJJJJJJJJJJ    AS:i:-13        XS:i:-19        XN:i:0  XM:i:3  XO:i:0  XG:i:0  NM:i:3  MD:Z:1G16T3G28  YT:Z:UU RG:Z:VA_W_08_SNP     ' dir="auto">
                     <pre>[kconf001@coreV3-23-040 QCFastqs]$ grep -v <span class="pl-s"><span class="pl-pds">"</span>@<span class="pl-pds">"</span></span> VA_W_08_SNP_clippedtrimmed.fastq.sam <span class="pl-k">|</span> head                                                                                                                                                     
K00188:59:HMTFHBBXX:2:1101:4026:1648    0       AstrangiaT1FW_65377     610     1       51M     *       0       0       CNCGTTAATCCATTCATGCGCGTCACTAATTAGATGACGAGGCATTTGGCT     A#AAF-A-AFFJAJFJJA&lt;7-AJJA7JJJ&lt;FJJJJ&lt;F&lt;&lt;JFFJJJJ-AJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:4614:1648    0       AstrangiaT1FW_488087    41      1       51M     *       0       0       GNTCTTGATTAATGAAAACATTCTTGGCAAATGCTTTCGCAGTAGTTCGTC     A#AFFJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJJFJJJJJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1T49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5690:1648    16      AstrangiaT1FW_65728     10184   1       51M     *       0       0       AGACGAACTACTGCGAAAGCATTTGCCAAGAATGTTTTCATTAATCAAGNG     FJFAJJJAJJJJJFJJJJJJJJJJF-AJFF&lt;JJJJJJJJJJJJAJJFFA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49A0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       
K00188:59:HMTFHBBXX:2:1101:3772:1666    16      AstrangiaT1FW_65718     4457    1       51M     *       0       0       TTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTGGACATGNG     7JJFFJJAJJF&lt;JJJJFJJJJJJJJ7JFJJ&lt;FFA&lt;JJJFJJJJJJJFFA#&lt;    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:49G1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5051:1666    16      AstrangiaT1FW_131671    3       1       51M     *       0       0       ACAGATGTGCCGCCCCAGCCAAACTCCCAACCTGACGGTGTCTTGGACTNG     JJFJJFJJFJJJJJJFJJJJJFJJJJJJFFJFFJAJJJJJJJJJJJFAA#A    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:49C0C0     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:5172:1666    0       AstrangiaT1FW_65662     1901    1       51M     *       0       0       TNCGACGGGCGGTGTGTACAAAGGGCAGGGACGTAATCAACGCGAGCTGAT     A#AFFJAJJJJJJJFJFFJJJJFF-AJJJJJJJJJJJJJJJFJJJJJAJJ7    AS:i:-6 XS:i:-6 XN:i:0  XM:i:2  XO:i:0  XG:i:0  NM:i:2  MD:Z:0A0G49     YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       
K00188:59:HMTFHBBXX:2:1101:5680:1666    0       AstrangiaT1FW_65668     593     1       51M     <span class="pl-k">*</span>       0       0       GNGCATTGCTGAGTCCATTCGTTTCCCTTTCAACGGTTTCACGTACTTTTT     A#-AFFJFJJ7JJJJJJFFJ<span class="pl-k">&lt;</span>FFFF-FJJFJJJF<span class="pl-k">&lt;</span>AFF7FJJAJFJJ7FJJ    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:6674:1666    16      AstrangiaT1FW_65718     1086    1       23M     <span class="pl-k">*</span>       0       0       TCCCAGGAGCATGTCTGTCTGNG FF<span class="pl-k">&lt;</span>JJJJAJJJJJJJJJJFFA#A AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0 NM:i:1  MD:Z:21A1       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                                                                                       
K00188:59:HMTFHBBXX:2:1101:7060:1666    0       AstrangiaT1FW_65712     6946    1       51M     <span class="pl-k">*</span>       0       0       CNAAATTTGACGATCGATTTGCACGTCAGAATCGCTACGAGCCTCCACCAG     A#AFFJFFJJJJJFJJJJJJJJJJ<span class="pl-k">&lt;</span>-FJJJJFJJJJJJJJJJJJJJJJJJF    AS:i:-1 XS:i:-1 XN:i:0  XM:i:1  XO:i:0  XG:i:0  NM:i:1  MD:Z:1C49       YT:Z:UU RG:Z:VA_W_08_SNP                                                                                                       K00188:59:HMTFHBBXX:2:1101:7080:1666    0       AstrangiaT1FW_65718     10568   0       51M     <span class="pl-k">*</span>       0       0       TNTTGCAATAGTAATTCTGCTCAGTACGAGAGGAACCGCAGATTCAGACAA     A#AFFJJJFJJJJJJJJJJJ-FJAAAJJJFJJJJJJJJJJJJJJJJJJJJJ    AS:i:-13        XS:i:-19        XN:i:0  XM:i:3  XO:i:0  XG:i:0  NM:i:3  MD:Z:1G16T3G28  YT:Z:UU RG:Z:VA_W_08_SNP     </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="9">
                       <li>
                        Run get_explain_sam_flags phython script on .sam files
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
[kconf001@coreV3-23-040 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py VA_W_01_22_clippedtrimmed.fastq.sam RI_B_01_18_clippedtrimmed.fastq.sam RI_W_01_14_clippedtrimmed.fastq.sam RI_B_01_22_clippedtrimmed.fastq.sam                                                                                                                                                                       
VA_W_01_22_clippedtrimmed.fastq.sam
['0', '4', '16']
0 :
4 :
read unmapped
16 :
read reverse strand
RI_B_01_18_clippedtrimmed.fastq.sam
['0', '4', '16']
0 :
4 :
read unmapped
16 :
read reverse strand
RI_W_01_14_clippedtrimmed.fastq.sam
['0', '4', '16']
0 :
4 :
read unmapped
16 :
read reverse strand
RI_B_01_22_clippedtrimmed.fastq.sam
['0', '4', '16']
0 :
4 :
read unmapped
16 :        " dir="auto">
                     <pre>[kconf001@coreV3-23-040 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py VA_W_01_22_clippedtrimmed.fastq.sam RI_B_01_18_clippedtrimmed.fastq.sam RI_W_01_14_clippedtrimmed.fastq.sam RI_B_01_22_clippedtrimmed.fastq.sam                                                                                                                                                                       
VA_W_01_22_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>4<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
4 <span class="pl-c1">:</span>
<span class="pl-c1">read</span> unmapped
16 <span class="pl-c1">:</span>
<span class="pl-c1">read</span> reverse strand
RI_B_01_18_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>4<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
4 <span class="pl-c1">:</span>
<span class="pl-c1">read</span> unmapped
16 <span class="pl-c1">:</span>
<span class="pl-c1">read</span> reverse strand
RI_W_01_14_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>4<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
4 <span class="pl-c1">:</span>
<span class="pl-c1">read</span> unmapped
16 <span class="pl-c1">:</span>
<span class="pl-c1">read</span> reverse strand
RI_B_01_22_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>4<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
4 <span class="pl-c1">:</span>
<span class="pl-c1">read</span> unmapped
16 <span class="pl-c1">:</span>        </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="10">
                       <li>
                        Run read sorting step fo SNP calling
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCbamsort.sh 
[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCbamsort.sh
#!/bin/bash -l
#SBATCH -o KristinaCBamSort.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCBamSort
enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done
for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done
[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCbamsort.sh
Submitted batch job 9276430" dir="auto">
                     <pre>[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCbamsort.sh 
[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCbamsort.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCBamSort.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCBamSort</span>
enable_lmod
module load samtools/1
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>.sam<span class="pl-k">;</span> <span class="pl-k">do</span> <span class="pl-s"><span class="pl-pds">`</span>samtools view -bS <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>.sam}</span>_UNSORTED.bam<span class="pl-pds">`</span></span><span class="pl-k">;</span> <span class="pl-k">done</span>
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>UNSORTED.bam<span class="pl-k">;</span> <span class="pl-k">do</span> samtools sort <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
samtools index <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
<span class="pl-k">done</span>
[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCbamsort.sh
Submitted batch job 9276430</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      02/12/2021
                     </h1>
                     <a aria-label="Permalink: 02/12/2021" class="anchor" href="#02122021" id="user-content-02122021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 8
                     </h2>
                     <a aria-label="Permalink: Homework Day 8" class="anchor" href="#homework-day-8" id="user-content-homework-day-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        Run the following command on your sprot output file to process into the contig length/match format that trinity examines
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 testassembly]$ nano KristinaCBlastparse.sh
[kconf001@coreV3-23-040 testassembly]$ cat KristinaCBlastparse.sh
#!/bin/bash -l
#SBATCH -o KristinaCBlastparse.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001.odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCBlastparse
/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
[kconf001@coreV3-23-040 testassembly]$ sbatch KristinaCBlastparse.sh 
Submitted batch job 9276467" dir="auto">
                     <pre>[kconf001@coreV3-23-040 testassembly]$ nano KristinaCBlastparse.sh
[kconf001@coreV3-23-040 testassembly]$ cat KristinaCBlastparse.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCBlastparse.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001.odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCBlastparse</span>
/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
[kconf001@coreV3-23-040 testassembly]$ sbatch KristinaCBlastparse.sh 
Submitted batch job 9276467</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      Output of sprot
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist
#hit_pct_cov_bin        count_in_bin    &gt;bin_below
100     212     212
90      98      310
80      104     414
70      137     551
60      182     733
50      256     989
40      413     1402
30      719     2121
20      1440    3561
10      984     4545
[kconf001@coreV3-23-040 testassembly]$ grep -c '&gt;' Trinity.fasta
41303
[kconf001@coreV3-23-040 testassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta
The total number of sequences is 41303
The average sequence length is 371
The total number of bases is 15326666
The minimum sequence length is 184
The maximum sequence length is 10771
The N50 is 364
Median Length = 627
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 5936" dir="auto">
                     <pre>[kconf001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist
<span class="pl-c"><span class="pl-c">#</span>hit_pct_cov_bin        count_in_bin    &gt;bin_below</span>
100     212     212
90      98      310
80      104     414
70      137     551
60      182     733
50      256     989
40      413     1402
30      719     2121
20      1440    3561
10      984     4545
[kconf001@coreV3-23-040 testassembly]$ grep -c <span class="pl-s"><span class="pl-pds">'</span>&gt;<span class="pl-pds">'</span></span> Trinity.fasta
41303
[kconf001@coreV3-23-040 testassembly]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta
The total number of sequences is 41303
The average sequence length is 371
The total number of bases is 15326666
The minimum sequence length is 184
The maximum sequence length is 10771
The N50 is 364
Median Length = 627
contigs <span class="pl-k">&lt;</span> 150bp = 0
contigs <span class="pl-k">&gt;</span>= 500bp = 5936</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Rm UNSORTED.bam's from your QCFastqs directory (or wherever your .bams and .sams are)
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 QCFastqs]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs
[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCrmsortbam.sh
[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCrmsortbam.sh
#!/bin/bash -l
#SBATCH -o KristinaCrmunsortbam.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCrmunsortbam
rm *UNSORTED.bam
[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCrmsortbam.sh
Submitted batch job 9276468" dir="auto">
                     <pre>[kconf001@coreV3-23-040 QCFastqs]$ <span class="pl-c1">cd</span> /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/fastq/QCFastqs
[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCrmsortbam.sh
[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCrmsortbam.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCrmunsortbam.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCrmunsortbam</span>
rm <span class="pl-k">*</span>UNSORTED.bam
[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCrmsortbam.sh
Submitted batch job 9276468</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        Run the following to start genotyping your SNPs for filtering next week
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCfreebayessubref.sh
[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCfreebayessubref.sh
#!/bin/bash -l
#SBATCH -o KristinaCfreebayessubref.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCfreebayessubref
enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam &gt; KristinaCmergedfastqs.vcf
[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCfreebayessubref.sh
Submitted batch job 9276565 " dir="auto">
                     <pre>[kconf001@coreV3-23-040 QCFastqs]$ nano KristinaCfreebayessubref.sh
[kconf001@coreV3-23-040 QCFastqs]$ cat KristinaCfreebayessubref.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCfreebayessubref.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCfreebayessubref</span>
enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta <span class="pl-k">*</span>.bam <span class="pl-k">&gt;</span> KristinaCmergedfastqs.vcf
[kconf001@coreV3-23-040 QCFastqs]$ sbatch KristinaCfreebayessubref.sh
Submitted batch job 9276565 </pre>
                    </div>
                    <p dir="auto">
                     [kconf001@coreV3-23-050 QCFastqs]$ mv *.sam ../../sams/
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="	-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory
```sh
[kconf001@coreV3-23-050 QCFastqs]$ cd ../../
[kconf001@coreV3-23-050 data]$ mkdir bams
[kconf001@coreV3-23-050 data]$ cd fastq/QCFastqs/
[kconf001@coreV3-23-050 QCFastqs]$ mv *.bam *.bam.bai ../../bams/">
                     <pre class="notranslate"><code>	-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory
```sh
[kconf001@coreV3-23-050 QCFastqs]$ cd ../../
[kconf001@coreV3-23-050 data]$ mkdir bams
[kconf001@coreV3-23-050 data]$ cd fastq/QCFastqs/
[kconf001@coreV3-23-050 QCFastqs]$ mv *.bam *.bam.bai ../../bams/
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="-rm your _unfiltered.vcf files and .fastq files if you have any">
                     <pre class="notranslate"><code>-rm your _unfiltered.vcf files and .fastq files if you have any
</code></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 QCFastqs]$ rm *_unfiltered.vcf
[kconf001@coreV3-23-050 QCFastqs]$ rm *.fastq 
[kconf001@coreV3-23-050 QCFastqs]$ ls -alh
total 8.2M
drwxrwxrwx 2 kconf001 users  408 Feb 12 12:56 .
drwxrwxrwx 3 kconf001 users  230 Feb 10 08:52 ..
-rwxrwxrwx 1 kconf001 users   37 Feb  5 14:50 KristinaCalignstats.txt
-rwxrwxrwx 1 kconf001 users  389 Feb 10 08:26 KristinaCbamsort.sh
-rwxrwxrwx 1 kconf001 users  946 Feb 10 09:48 KristinaCBamSort.txt
-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh
-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt
-rwxrwxrwx 1 kconf001 users  398 Feb 10 14:20 KristinaCfreebayessubref.sh
-rwxrwxrwx 1 kconf001 users 1.2K Feb 11 19:14 KristinaCfreebayessubref.txt
-rwxrwxrwx 1 kconf001 users 5.3M Feb 11 19:14 KristinaCmergedfastqs.vcf
-rwxrwxrwx 1 kconf001 users  193 Feb 10 12:48 KristinaCrmsortbam.sh
-rwxrwxrwx 1 kconf001 users    0 Feb 10 12:48 KristinaCrmunsortbam.txt " dir="auto">
                     <pre>[kconf001@coreV3-23-050 QCFastqs]$ rm <span class="pl-k">*</span>_unfiltered.vcf
[kconf001@coreV3-23-050 QCFastqs]$ rm <span class="pl-k">*</span>.fastq 
[kconf001@coreV3-23-050 QCFastqs]$ ls -alh
total 8.2M
drwxrwxrwx 2 kconf001 users  408 Feb 12 12:56 <span class="pl-c1">.</span>
drwxrwxrwx 3 kconf001 users  230 Feb 10 08:52 ..
-rwxrwxrwx 1 kconf001 users   37 Feb  5 14:50 KristinaCalignstats.txt
-rwxrwxrwx 1 kconf001 users  389 Feb 10 08:26 KristinaCbamsort.sh
-rwxrwxrwx 1 kconf001 users  946 Feb 10 09:48 KristinaCBamSort.txt
-rwxrwxrwx 1 kconf001 users  470 Jan 29 15:15 KristinaCBowtieln.sh
-rwxrwxrwx 1 kconf001 users 3.5K Jan 29 21:48 KristinaCbowtieln.txt
-rwxrwxrwx 1 kconf001 users  398 Feb 10 14:20 KristinaCfreebayessubref.sh
-rwxrwxrwx 1 kconf001 users 1.2K Feb 11 19:14 KristinaCfreebayessubref.txt
-rwxrwxrwx 1 kconf001 users 5.3M Feb 11 19:14 KristinaCmergedfastqs.vcf
-rwxrwxrwx 1 kconf001 users  193 Feb 10 12:48 KristinaCrmsortbam.sh
-rwxrwxrwx 1 kconf001 users    0 Feb 10 12:48 KristinaCrmunsortbam.txt </pre>
                    </div>
                    <p dir="auto">
                     -Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 QCFastqs]$ mkdir ../../vcf
[kconf001@coreV3-23-050 QCFastqs]$ cd ../../vcf  " dir="auto">
                     <pre>[kconf001@coreV3-23-050 QCFastqs]$ mkdir ../../vcf
[kconf001@coreV3-23-050 QCFastqs]$ <span class="pl-c1">cd</span> ../../vcf  </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        Start an interactive session via salloc (already done after logging in, see previous assignments for similar workflows)
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf
[kconf001@coreV3-23-050 vcf]$ ls -alh
total 2.1G
drwxrwxrwx 2 kconf001 users   54 Feb 12 12:59 .
drwxrwxrwx 8 kconf001 users  307 Feb 12 12:57 ..
-rwxr-xr-x 1 kconf001 users 1.5G Feb 12 13:00 mergedfastq_HEAAstrangiaAssembly.vcf" dir="auto">
                     <pre>[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf
[kconf001@coreV3-23-050 vcf]$ ls -alh
total 2.1G
drwxrwxrwx 2 kconf001 users   54 Feb 12 12:59 <span class="pl-c1">.</span>
drwxrwxrwx 8 kconf001 users  307 Feb 12 12:57 ..
-rwxr-xr-x 1 kconf001 users 1.5G Feb 12 13:00 mergedfastq_HEAAstrangiaAssembly.vcf</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 5.00 seconds" dir="auto">
                     <pre>[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 5.00 seconds</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf " dir="auto">
                     <pre>[kconf001@coreV3-23-050 vcf]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="6">
                       <li>
                        Run our host vcf extractor on your merged vcf file using the following syntax:
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <p dir="auto">
                     /cm/shared/courses/dbarshis/21AdvBioinf/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf                               
Read in ContigList" dir="auto">
                     <pre>[kconf001@coreV3-23-050 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf                               
Read <span class="pl-k">in</span> ContigList</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      7 Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
After filtering, kept 40 out of 40 Individuals
After filtering, kept 432676 out of a possible 432676 Sites
Run Time = 6.00 seconds
[kconf001@coreV3-23-050 vcf]$
[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
Run Time = 5.00 seconds   ```sh
 " dir="auto">
                     <pre>[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
After filtering, kept 40 out of 40 Individuals
After filtering, kept 432676 out of a possible 432676 Sites
Run Time = 6.00 seconds
[kconf001@coreV3-23-050 vcf]$
[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
Run Time = 5.00 seconds   <span class="pl-s"><span class="pl-pds">`</span><span class="pl-pds">`</span><span class="pl-pds">`</span>sh</span>
<span class="pl-s"> </span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="8">
                       <li>
                        Work through the VCF filtering tutorial until the following step:
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 5.00 seconds
[kconf001@coreV3-23-050 vcf]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3
VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--recode-INFO-all
--mac 3
--minQ 30
--max-missing 0.5
--out raw.g5mac3
--recode
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 432676 Sites
Run Time = 26.00 seconds
[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3
VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf raw.g5mac3.recode.vcf
--recode-INFO-all
--minDP 3
--out raw.g5mac3dp3
--recode
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 24.00 seconds
[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv
VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf raw.g5mac3dp3.recode.vcf
--missing-indv
After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 3.00 seconds
[kconf001@coreV3-23-050 vcf]$ bash
kconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ mawk '!/IN/' out.imiss | cut -f5 &gt; totalmissing
kconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ gnuplot &lt;&lt; \EOF
&gt; set terminal dumb size 120, 30
&gt; set autoscale
&gt; unset label
&gt; set title &quot;Histogram of % missing data per individual&quot;
&gt; set ylabel &quot;Number of Occurrences&quot;
&gt; set xlabel &quot;% of missing data&quot;
&gt; #set yr [0:100000]
&gt; binwidth=0.01
&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0
&gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes
&gt; pause -1
&gt; EOF
â 
Histogram of % missing data per individual
Number of Occurrences
3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++ 
+           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +
|                                                                * *
|                                                                                          
|                                                                * *                                         
|                                                                                          
|                                                                * *                                         
|                                                                                          
|                                                                * *                                         
|                                                                                      
2.5 ++                                                           * *                                          ++
|                                                                * *                                         
|                                                                                          
|                                                                * *                                         
|                                                                                          
|                                                                * *                                         
|                                                                                          
|                                                                * *                                         
|                                                                                          
|                                                                * *                                         
|                                                                                        
2 ++                                                     **      * * ***  ******    ** ****   ****             ++
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                      
1.5 ++                                                   **      * * * *  * *  *    ** ** *   *  *            ++
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
|                                                        **      * * * *  * *  *    ** ** *   *  *           
|                                                                                          
+           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *             +
1 *********************************************************************************************************---++
0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1
% of missing data" dir="auto">
                     <pre>[kconf001@coreV3-23-050 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 5.00 seconds
[kconf001@coreV3-23-050 vcf]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3
VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--recode-INFO-all
--mac 3
--minQ 30
--max-missing 0.5
--out raw.g5mac3
--recode
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 432676 Sites
Run Time = 26.00 seconds
[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3
VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf raw.g5mac3.recode.vcf
--recode-INFO-all
--minDP 3
--out raw.g5mac3dp3
--recode
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 24.00 seconds
[kconf001@coreV3-23-050 vcf]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv
VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf raw.g5mac3dp3.recode.vcf
--missing-indv
After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 3.00 seconds
[kconf001@coreV3-23-050 vcf]$ bash
kconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ mawk <span class="pl-s"><span class="pl-pds">'</span>!/IN/<span class="pl-pds">'</span></span> out.imiss <span class="pl-k">|</span> cut -f5 <span class="pl-k">&gt;</span> totalmissing
kconf001@coreV3-23-050:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf$ gnuplot <span class="pl-s"><span class="pl-k">&lt;&lt;</span> \<span class="pl-k">EOF</span></span>
<span class="pl-s">&gt; set terminal dumb size 120, 30</span>
<span class="pl-s">&gt; set autoscale</span>
<span class="pl-s">&gt; unset label</span>
<span class="pl-s">&gt; set title "Histogram of % missing data per individual"</span>
<span class="pl-s">&gt; set ylabel "Number of Occurrences"</span>
<span class="pl-s">&gt; set xlabel "% of missing data"</span>
<span class="pl-s">&gt; #set yr [0:100000]</span>
<span class="pl-s">&gt; binwidth=0.01</span>
<span class="pl-s">&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0</span>
<span class="pl-s">&gt; plot 'totalmissing' using (bin(<span class="pl-smi">$1</span>,binwidth)):(1.0) smooth freq with boxes</span>
<span class="pl-s">&gt; pause -1</span>
<span class="pl-s">&gt; EOF</span>
<span class="pl-s">â </span>
<span class="pl-s">Histogram of % missing data per individual</span>
<span class="pl-s">Number of Occurrences</span>
<span class="pl-s">3 ++----------+-----------+-----------+-----------+------------+---***-----+-----------+-----------+----------++ </span>
<span class="pl-s">+           +           +           +           +       'totalmissing' using (bin(<span class="pl-smi">$1</span>,binwidth)):(1.0) ****** +</span>
<span class="pl-s">|                                                                * *</span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                      </span>
<span class="pl-s">2.5 ++                                                           * *                                          ++</span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                                * *                                         </span>
<span class="pl-s">|                                                                                        </span>
<span class="pl-s">2 ++                                                     **      * * ***  ******    ** ****   ****             ++</span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                      </span>
<span class="pl-s">1.5 ++                                                   **      * * * *  * *  *    ** ** *   *  *            ++</span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">|                                                        **      * * * *  * *  *    ** ** *   *  *           </span>
<span class="pl-s">|                                                                                          </span>
<span class="pl-s">+           +           +           +           +        **  +   * * * * +* *  *    ** ** *   *  *             +</span>
<span class="pl-s">1 *********************************************************************************************************---++</span>
<span class="pl-s">0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1</span>
<span class="pl-s">% of missing data</span></pre>
                    </div>
                    <p dir="auto">
                     Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.
                    </p>
                    <p dir="auto">
                     vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      02/17/2021
                     </h1>
                     <a aria-label="Permalink: 02/17/2021" class="anchor" href="#02172021" id="user-content-02172021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 09
                     </h2>
                     <a aria-label="Permalink: Homework Day 09" class="anchor" href="#homework-day-09" id="user-content-homework-day-09">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      1-  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters
##Half the class run the following /cm/shared/apps/vcftools/0.1.12b/bin/vcftools
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--maf 0.015
--max-alleles 2
--max-missing 0.5
--minQ 30
--minGQ 20
--minDP 3
--remove-indels
--hwe 0.01
--recode
--recode-INFO-all
--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf 
--max-missing 0.5 
--mac 3 
--minQ 30 
--minDP 10 
--max-alleles 2 
--maf 0.015 
--remove-indels 
--recode 
--recode-INFO-all 
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 18718 out of a possible 432676 Sites
Run Time = 11.00 seconds " dir="auto">
                     <pre>[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf 
--max-missing 0.5 
--mac 3 
--minQ 30 
--minDP 10 
--max-alleles 2 
--maf 0.015 
--remove-indels 
--recode 
--recode-INFO-all 
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 18718 out of a possible 432676 Sites
Run Time = 11.00 seconds </pre>
                    </div>
                    <p dir="auto">
                     ##the other half run the filters we used from the paper
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--max-missing 0.5
--mac 3
--minQ 30
--minDP 10
--max-alleles 2
--maf 0.015
--remove-indels
--recode
--recode-INFO-all
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf 
--max-missing 0.5 
--mac 3 
--minQ 30 
--minDP 10 
--max-alleles 2 
--maf 0.015 
--remove-indels 
--recode 
--recode-INFO-all 
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--recode-INFO-all
--mac 3
--maf 0.015
--max-alleles 2
--minDP 10
--minQ 30
--max-missing 0.5
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
--recode
--remove-indels
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 1578 out of a possible 432676 Sites
Run Time = 5.00 seconds  " dir="auto">
                     <pre>[kconf001@coreV3-23-002 vcf]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf 
--max-missing 0.5 
--mac 3 
--minQ 30 
--minDP 10 
--max-alleles 2 
--maf 0.015 
--remove-indels 
--recode 
--recode-INFO-all 
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--recode-INFO-all
--mac 3
--maf 0.015
--max-alleles 2
--minDP 10
--minQ 30
--max-missing 0.5
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
--recode
--remove-indels
After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 1578 out of a possible 432676 Sites
Run Time = 5.00 seconds  </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      2- Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-002 vcf]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf  &gt; samplenames.txt
[kconf001@coreV3-23-002 vcf]$ cat samplenames.txt
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged
[kconf001@coreV3-23-002 vcf]$ nano samplenames.txt " dir="auto">
                     <pre>[kconf001@coreV3-23-002 vcf]$ grep <span class="pl-s"><span class="pl-pds">'</span>#CHROM<span class="pl-pds">'</span></span> mergedfastq_HEAAstrangiaAssembly_subset.vcf  <span class="pl-k">&gt;</span> samplenames.txt
[kconf001@coreV3-23-002 vcf]$ cat samplenames.txt
<span class="pl-c"><span class="pl-c">#</span>CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  </span>
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged
[kconf001@coreV3-23-002 vcf]$ nano samplenames.txt </pre>
                    </div>
                    <p dir="auto">
                     Need to edit the file on excel or use sed 's/\s+/\n/g' samplenames. txt
                    </p>
                    <p dir="auto">
                     or use .imiss file (the one I used)
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-002 vcf]$ cat out.imiss
INDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS
RI_W_06_merged  99853   0       67075   0.671737
RI_W_07_merged  99853   0       63917   0.640111
RI_W_02_merged  99853   0       73105   0.732126
RI_W_04_merged  99853   0       71159   0.712638
VA_W_09_SNP_clipped     99853   0       19459   0.194876
RI_B_08_SNP_clipped     99853   0       88110   0.882397
VA_W_08_SNP_clipped     99853   0       83344   0.834667
VA_B_08_SNP_clipped     99853   0       94221   0.943597
VA_W_02_merged  99853   0       71050   0.711546
VA_B_07_merged  99853   0       62687   0.627793
RI_B_05_merged  99853   0       43945   0.440097
VA_W_06_merged  99853   0       65164   0.652599
VA_W_04_merged  99853   0       50911   0.509859
VA_W_01_merged  99853   0       67150   0.672489
VA_B_10_SNP_clipped     99853   0       82831   0.829529
VA_B_06_merged  99853   0       57724   0.57809
VA_W_05_merged  99853   0       66297   0.663946
RI_B_09_SNP_clipped     99853   0       85107   0.852323
VA_W_10_SNP_clipped     99853   0       83234   0.833565
RI_W_08_SNP_clipped     99853   0       77802   0.779165
RI_B_06_merged  99853   0       79766   0.798834
RI_W_10_SNP_clipped     99853   0       91323   0.914574
RI_B_04_merged  99853   0       46717   0.467858
VA_W_03_merged  99853   0       64016   0.641102
RI_B_07_merged  99853   0       70798   0.709022
RI_W_05_merged  99853   0       57117   0.572011
RI_W_09_SNP_clipped     99853   0       88492   0.886223
VA_B_01_merged  99853   0       47217   0.472865
VA_B_09_SNP_clipped     99853   0       79187   0.793036
RI_B_10_SNP_clipped     99853   0       82779   0.829009
RI_W_01_merged  99853   0       73169   0.732767
RI_B_01_merged  99853   0       69619   0.697215
VA_B_04_merged  99853   0       58190   0.582757
RI_B_02_merged  99853   0       87439   0.875677
RI_W_03_merged  99853   0       38760   0.388171
VA_B_02_merged  99853   0       78381   0.784964
VA_W_07_merged  99853   0       64592   0.646871
VA_B_05_merged  99853   0       54401   0.544811
RI_B_03_merged  99853   0       80369   0.804873

[kconf001@coreV3-23-002 vcf]$ cut -f 1 out.imiss | tail -n +2
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged" dir="auto">
                     <pre>[kconf001@coreV3-23-002 vcf]$ cat out.imiss
INDV    N_DATA  N_GENOTYPES_FILTERED    N_MISS  F_MISS
RI_W_06_merged  99853   0       67075   0.671737
RI_W_07_merged  99853   0       63917   0.640111
RI_W_02_merged  99853   0       73105   0.732126
RI_W_04_merged  99853   0       71159   0.712638
VA_W_09_SNP_clipped     99853   0       19459   0.194876
RI_B_08_SNP_clipped     99853   0       88110   0.882397
VA_W_08_SNP_clipped     99853   0       83344   0.834667
VA_B_08_SNP_clipped     99853   0       94221   0.943597
VA_W_02_merged  99853   0       71050   0.711546
VA_B_07_merged  99853   0       62687   0.627793
RI_B_05_merged  99853   0       43945   0.440097
VA_W_06_merged  99853   0       65164   0.652599
VA_W_04_merged  99853   0       50911   0.509859
VA_W_01_merged  99853   0       67150   0.672489
VA_B_10_SNP_clipped     99853   0       82831   0.829529
VA_B_06_merged  99853   0       57724   0.57809
VA_W_05_merged  99853   0       66297   0.663946
RI_B_09_SNP_clipped     99853   0       85107   0.852323
VA_W_10_SNP_clipped     99853   0       83234   0.833565
RI_W_08_SNP_clipped     99853   0       77802   0.779165
RI_B_06_merged  99853   0       79766   0.798834
RI_W_10_SNP_clipped     99853   0       91323   0.914574
RI_B_04_merged  99853   0       46717   0.467858
VA_W_03_merged  99853   0       64016   0.641102
RI_B_07_merged  99853   0       70798   0.709022
RI_W_05_merged  99853   0       57117   0.572011
RI_W_09_SNP_clipped     99853   0       88492   0.886223
VA_B_01_merged  99853   0       47217   0.472865
VA_B_09_SNP_clipped     99853   0       79187   0.793036
RI_B_10_SNP_clipped     99853   0       82779   0.829009
RI_W_01_merged  99853   0       73169   0.732767
RI_B_01_merged  99853   0       69619   0.697215
VA_B_04_merged  99853   0       58190   0.582757
RI_B_02_merged  99853   0       87439   0.875677
RI_W_03_merged  99853   0       38760   0.388171
VA_B_02_merged  99853   0       78381   0.784964
VA_W_07_merged  99853   0       64592   0.646871
VA_B_05_merged  99853   0       54401   0.544811
RI_B_03_merged  99853   0       80369   0.804873

[kconf001@coreV3-23-002 vcf]$ cut -f 1 out.imiss <span class="pl-k">|</span> tail -n +2
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged</pre>
                    </div>
                    <p dir="auto">
                     then edit on Excel (seperate columns, =LEFT(A1,4), then paste to a popfile.txt
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-002 vcf]$ nano popfile.txt" dir="auto">
                     <pre>[kconf001@coreV3-23-002 vcf]$ nano popfile.txt</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      3- Convert your filtered .vcf file to genepop format using the following command:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV3-23-002 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile.txt
Indivs with genotypes in vcf file: 
RI_W_06_merged  
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged    
RI_W_05_merged
RI_W_09_SNP_clipped  
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged
44 
1578 
1578 
1578 
1578 
40 " dir="auto">
                     <pre>[kconf001@coreV3-23-002 vcf]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf popfile.txt
Indivs with genotypes <span class="pl-k">in</span> vcf file: 
RI_W_06_merged  
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged    
RI_W_05_merged
RI_W_09_SNP_clipped  
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged
44 
1578 
1578 
1578 
1578 
40 </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      4- SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/         
kconf001@turing.hpc.odu.edu's password:
1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen
100%  343KB   2.5MB/s   00:00    " dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/vcf/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/         
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen</span>
<span class="pl-s">100%  343KB   2.5MB/s   00:00    </span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      5- Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.
                     </li>
                    </ul>
                    <p dir="auto">
                     In r-studio:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="&gt; setwd(&quot;~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog&quot;)
&gt; datafile&lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)
Converting data from a Genepop .gen file to a genind object... 
File description:  AllSNPs 
...done." dir="auto">
                     <pre><span class="pl-k">&gt;</span> setwd(<span class="pl-s"><span class="pl-pds">"</span>~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> datafile<span class="pl-k">&lt;</span>-read.genepop(<span class="pl-s"><span class="pl-pds">'</span>1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen<span class="pl-pds">'</span></span>, ncode=2)
Converting data from a Genepop .gen file to a genind object... 
File description:  AllSNPs 
...done.</pre>
                    </div>
                    <p dir="auto">
                     adegenet_PCAs.R script
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(&quot;ape&quot;)
library(&quot;pegas&quot;)
library(&quot;seqinr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;adegenet&quot;)
setwd(&quot;~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio&quot;)
datafile&lt;-read.genepop('1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen', ncode=2)
sum(is.na(datafile$tab))
datafile #shows info
YOURdata&lt;-scaleGen(datafile, NA.method='mean')
X&lt;-YOURdata
Y&lt;-as.factor(substring(pop(datafile),1,4))
pca1 &lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)

#### PCAs ####
# individual labels
# s.label(pca1$li)

# population elipses
s.class(pca1$li, pop(datafile))

#color symbols, pop names
#pdf(&quot;YOURINITIALS_ColorPCA1v2.pdf&quot;)
col &lt;- c(&quot;blue&quot;,&quot;red&quot;, &quot;green&quot;, &quot;black&quot;)
s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca1$eig[1:3], 3,1,2, posi=&quot;topright&quot;)
title(&quot;PCA of DJB_data\naxes 1-2&quot;)
#dev.off()

#### Snapclust ####
a.clust&lt;-snapclust(datafile, k = 2)
class(a.clust)
names(a.clust)
a.tab &lt;- table(pop(datafile), a.clust$group)
table.value(a.tab, col.labels = 1:2)

compoplot(a.clust)" dir="auto">
                     <pre>library(<span class="pl-s"><span class="pl-pds">"</span>ape<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>pegas<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>seqinr<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>ggplot2<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>adegenet<span class="pl-pds">"</span></span>)
setwd(<span class="pl-s"><span class="pl-pds">"</span>~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio<span class="pl-pds">"</span></span>)
datafile<span class="pl-k">&lt;</span>-read.genepop(<span class="pl-s"><span class="pl-pds">'</span>1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen<span class="pl-pds">'</span></span>, ncode=2)
sum(is.na(datafile<span class="pl-smi">$tab</span>))
datafile <span class="pl-c"><span class="pl-c">#</span>shows info</span>
YOURdata<span class="pl-k">&lt;</span>-scaleGen(datafile, NA.method=<span class="pl-s"><span class="pl-pds">'</span>mean<span class="pl-pds">'</span></span>)
X<span class="pl-k">&lt;</span>-YOURdata
Y<span class="pl-k">&lt;</span>-as.factor(substring(pop(datafile),1,4))
pca1 <span class="pl-k">&lt;</span>- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)

<span class="pl-c"><span class="pl-c">#</span>### PCAs ####</span>
<span class="pl-c"><span class="pl-c">#</span> individual labels</span>
<span class="pl-c"><span class="pl-c">#</span> s.label(pca1$li)</span>

<span class="pl-c"><span class="pl-c">#</span> population elipses</span>
s.class(pca1<span class="pl-smi">$li</span>, pop(datafile))

<span class="pl-c"><span class="pl-c">#</span>color symbols, pop names</span>
<span class="pl-c"><span class="pl-c">#</span>pdf("YOURINITIALS_ColorPCA1v2.pdf")</span>
col <span class="pl-k">&lt;</span>- c(<span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>green<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>)
s.class(pca1<span class="pl-smi">$li</span>, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca1<span class="pl-smi">$eig</span>[1:3], 3,1,2, posi=<span class="pl-s"><span class="pl-pds">"</span>topright<span class="pl-pds">"</span></span>)
title(<span class="pl-s"><span class="pl-pds">"</span>PCA of DJB_data\naxes 1-2<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>dev.off()</span>

<span class="pl-c"><span class="pl-c">#</span>### Snapclust ####</span>
a.clust<span class="pl-k">&lt;</span>-snapclust(datafile, k = 2)
class(a.clust)
names(a.clust)
a.tab <span class="pl-k">&lt;</span>- table(pop(datafile), a.clust<span class="pl-smi">$group</span>)
table.value(a.tab, col.labels = 1:2)

compoplot(a.clust)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      02/19/2021
                     </h1>
                     <a aria-label="Permalink: 02/19/2021" class="anchor" href="#02192021" id="user-content-02192021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 10
                     </h2>
                     <a aria-label="Permalink: Homework Day 10" class="anchor" href="#homework-day-10" id="user-content-homework-day-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      1- Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.
                     </li>
                    </ul>
                    <p dir="auto">
                     ###run step 2 as an sbatch .sh script because it will take a while to finish (Didn't run an sbatch)
                    </p>
                    <ul dir="auto">
                     <li>
                      2- cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~ ssh kconf001@turing.hpc.odu.edu
kconf001@turing.hpc.odu.edu's password:
Last login: Wed Feb 17 08:47:42 2021 from ip70-160-48-140.hr.hr.cox.net
[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams
[kconf001@turing1 sams]$ salloc
salloc: Pending job allocation 9280868
salloc: job 9280868 queued and waiting for resources
salloc: job 9280868 has been allocated resources
salloc: Granted job allocation 9280868
[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam " dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span> ssh kconf001@turing.hpc.odu.edu
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">Last login: Wed Feb 17 08:47:42 2021 from ip70-160-48-140.hr.hr.cox.net</span>
<span class="pl-s">[kconf001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams</span>
<span class="pl-s">[kconf001@turing1 sams]$ salloc</span>
<span class="pl-s">salloc: Pending job allocation 9280868</span>
<span class="pl-s">salloc: job 9280868 queued and waiting for resources</span>
<span class="pl-s">salloc: job 9280868 has been allocated resources</span>
<span class="pl-s">salloc: Granted job allocation 9280868</span>
<span class="pl-s">[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam </span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      3- Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KristinaCFullCounts_summed.txt NoMatch *_counts.txt
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KristinaCFullCounts_summed.txt NoMatch *_counts.txt
Hits not matched
RI_B_01_14_clippedtrimmed.fastq_counts.txt=13381
RI_B_01_18_clippedtrimmed.fastq_counts.txt=13381
RI_B_01_22_clippedtrimmed.fastq_counts.txt=13381
RI_B_08_SNP_clippedtrimmed.fastq_counts.txt=13381
RI_W_01_14_clippedtrimmed.fastq_counts.txt=13381
RI_W_01_18_clippedtrimmed.fastq_counts.txt=13381
RI_W_01_22_clippedtrimmed.fastq_counts.txt=13381
RI_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381
VA_B_01_14_clippedtrimmed.fastq_counts.txt=13381
VA_B_01_18_clippedtrimmed.fastq_counts.txt=13381
VA_B_01_22_clippedtrimmed.fastq_counts.txt=13381
VA_B_09_SNP_clippedtrimmed.fastq_counts.txt=13381
VA_W_01_14_clippedtrimmed.fastq_counts.txt=13381
VA_W_01_18_clippedtrimmed.fastq_counts.txt=13381
VA_W_01_22_clippedtrimmed.fastq_counts.txt=13381
VA_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381 " dir="auto">
                     <pre>[kconf001@coreV2-25-015 sams]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KristinaCFullCounts_summed.txt NoMatch <span class="pl-k">*</span>_counts.txt
Hits not matched
RI_B_01_14_clippedtrimmed.fastq_counts.txt=13381
RI_B_01_18_clippedtrimmed.fastq_counts.txt=13381
RI_B_01_22_clippedtrimmed.fastq_counts.txt=13381
RI_B_08_SNP_clippedtrimmed.fastq_counts.txt=13381
RI_W_01_14_clippedtrimmed.fastq_counts.txt=13381
RI_W_01_18_clippedtrimmed.fastq_counts.txt=13381
RI_W_01_22_clippedtrimmed.fastq_counts.txt=13381
RI_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381
VA_B_01_14_clippedtrimmed.fastq_counts.txt=13381
VA_B_01_18_clippedtrimmed.fastq_counts.txt=13381
VA_B_01_22_clippedtrimmed.fastq_counts.txt=13381
VA_B_09_SNP_clippedtrimmed.fastq_counts.txt=13381
VA_W_01_14_clippedtrimmed.fastq_counts.txt=13381
VA_W_01_18_clippedtrimmed.fastq_counts.txt=13381
VA_W_01_22_clippedtrimmed.fastq_counts.txt=13381
VA_W_08_SNP_clippedtrimmed.fastq_counts.txt=13381 </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      4- scp YOURNAMEFullCounts_summed.txt to your laptop
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams/KristinaCFullCounts_summed.txt /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/
kconf001@turing.hpc.odu.edu's password:
KristinaCFullCounts_summed.txt
100% 1972KB   3.4MB/s   00:00 " dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC/data/sams/KristinaCFullCounts_summed.txt /c/Users/lyka/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">KristinaCFullCounts_summed.txt</span>
<span class="pl-s">100% 1972KB   3.4MB/s   00:00 </span></pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      5- edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)
                     </li>
                    </ul>
                    <p dir="auto">
                     File had no match???
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      02/24/2021
                     </h1>
                     <a aria-label="Permalink: 02/24/2021" class="anchor" href="#02242021" id="user-content-02242021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 11
                     </h2>
                     <a aria-label="Permalink: Homework Day 11" class="anchor" href="#homework-day-11" id="user-content-homework-day-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      1 . Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="### This is an R script that implements the program DESeq2 for gene expression analysis.
### Much more information on the program and specific function (particularly for checking quality) 
### can be found here: https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html
### The elements of this script were written by Melissa Pespeni and Daniel Barshis.

#Only need to do this the first time to install the package
#for R version 4.0
#if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
#  install.packages(&quot;BiocManager&quot;)

#BiocManager::install(&quot;DESeq2&quot;)

#a#for older versions of R
#source(&quot;https://bioconductor.org/biocLite.R&quot;)
#biocLite(&quot;DESeq2&quot;)

library(DESeq2)
library(gplots)

vignette('DESeq2')

setwd(&quot;Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio/Day11&quot;)  # The drag and drop from finder works in R, too.

#useful functions
#head() - prints out the top 6 lines
#dim() - prints the dimensions of a variable
#nrow() - returns the number of rows in a vector or matrix
# ?[functionName] - opens documentation describing the function

#read in your data to make counts table
countsTable &lt;- read.delim('djbFullCounts_summed.txt', header=TRUE, stringsAsFactors=TRUE, row.names=1)
head(countsTable)
dim(countsTable)
colSums(countsTable)
colSums(countsTable)[order(colSums(countsTable))]
LowCounts&lt;-c(&quot;VA_W_02_22&quot;,&quot;VA_W_01_22&quot;,&quot;RI_B_03_18&quot;)
names(countsTable)%in%LowCounts
!(names(countsTable)%in%LowCounts)
countsTable&lt;-countsTable[,!(names(countsTable)%in%LowCounts)]
dim(countsTable)

#make a table with conditions of each individual (e.g. &quot;VA&quot; and &quot;RI&quot;)  There should be the same number of conditions described as there are samples in your data file, and in the same order.
#NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.
#Here's an example
# Sample		Origin	SymState	Temp  Origin_SymState Origin_SymState_Temp
# RI_B_06_18		RI	B	18  RI_B  RI_B_18
# VA_B_07_14		VA	B	14  VA_B  VA_B_14
# VA_W_07_22		VA	W	22  VA_W  VA_W_22
# safest way to do this is in R

conds&lt;-data.frame(&quot;Sample&quot;=names(countsTable))
conds$Origin&lt;-factor(substring(conds$Sample,1,2))
conds$SymState&lt;-factor(substring(conds$Sample,4,4))
conds$Temp&lt;-factor(substring(conds$Sample,9,10))
conds$Origin_SymState&lt;-factor(paste(conds$Origin,conds$SymState, sep=&quot;_&quot;))
conds$Origin_SymState_Temp&lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=&quot;_&quot;))

##Check null counts per sample
prop.null &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))

barplot(prop.null, main=&quot;Percentage of null counts per sample&quot;, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')

pdf(file=&quot;OverallNullCounts.pdf&quot;,14, 14)
barplot(prop.null, main=&quot;Percentage of null counts per sample&quot;, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
dev.off()

# how many genes have mean count &gt;=3 
means = apply(countsTable, 1, mean)
table(means&gt;=3)

countsTable&lt;-countsTable[means&gt;=3,]
dim(countsTable)

prop.nullv2 &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
pdf(file=&quot;Mean3NullCounts.pdf&quot;,14, 14)
barplot(prop.nullv2, main=&quot;Percentage of null counts per sample&quot;, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
dev.off()

#make count data set
dds &lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin_SymState_Temp)

dds &lt;- DESeq(dds)

#transform counts to variance stablized counts
vstCounts&lt;-varianceStabilizingTransformation(dds)

#plot cluster dendrogram across samples
dists&lt;-dist(t(assay(vstCounts)))
plot(hclust(dists))

plotPCA(vstCounts, intgroup=&quot;Origin&quot;)
plotPCA(vstCounts, intgroup=&quot;SymState&quot;)
plotPCA(vstCounts, intgroup=&quot;Temp&quot;)
plotPCA(vstCounts, intgroup=&quot;Origin_SymState&quot;)
plotPCA(vstCounts, intgroup=&quot;Origin_SymState_Temp&quot;)

resultsNames(dds)

dim(dds)

###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)
res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_B_18&quot;, &quot;VA_B_18&quot;))
head(res)
summary(res)

res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_W_18&quot;, &quot;VA_W_18&quot;))
summary(res)

res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_B_14&quot;, &quot;VA_B_14&quot;))
summary(res)

res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_W_14&quot;, &quot;VA_W_14&quot;))
summary(res)

res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_B_22&quot;, &quot;VA_B_22&quot;))
summary(res)

res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_W_22&quot;, &quot;VA_W_22&quot;))
summary(res)

head(res[order(res$padj),])
plotCounts(dds, &quot;TR2367|c0_g1_i1_coral&quot;, intgroup=&quot;Origin_SymState_Temp&quot;)

#count the number of significantly differentially expressed genes
sum(res$padj &lt; 0.3, na.rm =T)
sum(res$padj &lt; 0.2, na.rm =T)
sum(res$padj &lt; 0.1, na.rm =T)
sum(res$pvalue &lt; 0.05, na.rm =T)
											 
# Make a counts table that is scaled by the size factors
scaledcounts = counts(dds, normalized=T)

#building heat map data
head(scaledcounts)
genes4heatmap&lt;-res[res$pvalue &lt;0.05 &amp; !is.na(res$pvalue),]
names(genes4heatmap)
head(genes4heatmap)
dim(genes4heatmap)
data4heatmap&lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
dim(data4heatmap)
head(data4heatmap)

temp = as.matrix(rowMeans(data4heatmap))
head(temp)
scaledmatrix&lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
data4heatmapscaled = data4heatmap/scaledmatrix
head(data4heatmapscaled)

dim(data4heatmapscaled)

pairs.breaks &lt;- seq(0, 3.0, by=0.1)
length(pairs.breaks)
mycol &lt;- colorpanel(n=30, low=&quot;black&quot;, high=&quot;yellow&quot;) 

pdf(file=&quot;XXXXXX_byrow.pdf&quot;,14,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(&quot;row&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
dev.off()

pdf(file=&quot;XXXXXX_bycolumn.pdf&quot;,14,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(&quot;col&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
dev.off()


###PCAs
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin&quot;)

prop.nullv3 &lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))
pdf(file=&quot;ResNullCounts.pdf&quot;,14, 14)
barplot(prop.nullv3, main=&quot;Percentage of null counts per sample&quot;, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
dev.off()
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>## This is an R script that implements the program DESeq2 for gene expression analysis.</span>
<span class="pl-c"><span class="pl-c">#</span>## Much more information on the program and specific function (particularly for checking quality) </span>
<span class="pl-c"><span class="pl-c">#</span>## can be found here: https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</span>
<span class="pl-c"><span class="pl-c">#</span>## The elements of this script were written by Melissa Pespeni and Daniel Barshis.</span>

<span class="pl-c"><span class="pl-c">#</span>Only need to do this the first time to install the package</span>
<span class="pl-c"><span class="pl-c">#</span>for R version 4.0</span>
<span class="pl-c"><span class="pl-c">#</span>if (!requireNamespace("BiocManager", quietly = TRUE))</span>
<span class="pl-c"><span class="pl-c">#</span>  install.packages("BiocManager")</span>

<span class="pl-c"><span class="pl-c">#</span>BiocManager::install("DESeq2")</span>

<span class="pl-c"><span class="pl-c">#</span>a#for older versions of R</span>
<span class="pl-c"><span class="pl-c">#</span>source("https://bioconductor.org/biocLite.R")</span>
<span class="pl-c"><span class="pl-c">#</span>biocLite("DESeq2")</span>

library(DESeq2)
library(gplots)

vignette(<span class="pl-s"><span class="pl-pds">'</span>DESeq2<span class="pl-pds">'</span></span>)

setwd(<span class="pl-s"><span class="pl-pds">"</span>Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog/Rstudio/Day11<span class="pl-pds">"</span></span>)  <span class="pl-c"><span class="pl-c">#</span> The drag and drop from finder works in R, too.</span>

<span class="pl-c"><span class="pl-c">#</span>useful functions</span>
<span class="pl-c"><span class="pl-c">#</span>head() - prints out the top 6 lines</span>
<span class="pl-c"><span class="pl-c">#</span>dim() - prints the dimensions of a variable</span>
<span class="pl-c"><span class="pl-c">#</span>nrow() - returns the number of rows in a vector or matrix</span>
<span class="pl-c"><span class="pl-c">#</span> ?[functionName] - opens documentation describing the function</span>

<span class="pl-c"><span class="pl-c">#</span>read in your data to make counts table</span>
countsTable <span class="pl-k">&lt;</span>- read.delim(<span class="pl-s"><span class="pl-pds">'</span>djbFullCounts_summed.txt<span class="pl-pds">'</span></span>, header=TRUE, stringsAsFactors=TRUE, row.names=1)
head(countsTable)
dim(countsTable)
colSums(countsTable)
colSums(countsTable)[order(colSums(countsTable))]
LowCounts<span class="pl-k">&lt;</span>-c(<span class="pl-s"><span class="pl-pds">"</span>VA_W_02_22<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>VA_W_01_22<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>RI_B_03_18<span class="pl-pds">"</span></span>)
names(countsTable)%in%LowCounts
<span class="pl-k">!</span>(names(countsTable)%in%LowCounts)
countsTable<span class="pl-k">&lt;</span>-countsTable[,<span class="pl-k">!</span>(names(countsTable)%in%LowCounts)]
dim(countsTable)

<span class="pl-c"><span class="pl-c">#</span>make a table with conditions of each individual (e.g. "VA" and "RI")  There should be the same number of conditions described as there are samples in your data file, and in the same order.</span>
<span class="pl-c"><span class="pl-c">#</span>NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.</span>
<span class="pl-c"><span class="pl-c">#</span>Here's an example</span>
<span class="pl-c"><span class="pl-c">#</span> Sample		Origin	SymState	Temp  Origin_SymState Origin_SymState_Temp</span>
<span class="pl-c"><span class="pl-c">#</span> RI_B_06_18		RI	B	18  RI_B  RI_B_18</span>
<span class="pl-c"><span class="pl-c">#</span> VA_B_07_14		VA	B	14  VA_B  VA_B_14</span>
<span class="pl-c"><span class="pl-c">#</span> VA_W_07_22		VA	W	22  VA_W  VA_W_22</span>
<span class="pl-c"><span class="pl-c">#</span> safest way to do this is in R</span>

conds<span class="pl-k">&lt;</span>-data.frame(<span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>=names(countsTable))
conds<span class="pl-smi">$Origin</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,1,2))
conds<span class="pl-smi">$SymState</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,4,4))
conds<span class="pl-smi">$Temp</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,9,10))
conds<span class="pl-smi">$Origin_SymState</span><span class="pl-k">&lt;</span>-factor(paste(conds<span class="pl-smi">$Origin</span>,conds<span class="pl-smi">$SymState</span>, sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))
conds<span class="pl-smi">$Origin_SymState_Temp</span><span class="pl-k">&lt;</span>-factor(paste(conds<span class="pl-smi">$Origin</span>,conds<span class="pl-smi">$SymState</span>,conds<span class="pl-smi">$Temp</span>, sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))

<span class="pl-c"><span class="pl-c">#</span>#Check null counts per sample</span>
prop.null <span class="pl-k">&lt;</span>- apply(countsTable, 2, function(x) 100<span class="pl-k">*</span>mean(x==0))

barplot(prop.null, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)

pdf(file=<span class="pl-s"><span class="pl-pds">"</span>OverallNullCounts.pdf<span class="pl-pds">"</span></span>,14, 14)
barplot(prop.null, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-en">dev.off</span>()

<span class="pl-c"><span class="pl-c">#</span> how many genes have mean count &gt;=3 </span>
means = apply(countsTable, 1, mean)
table(means<span class="pl-k">&gt;</span>=3)

countsTable<span class="pl-k">&lt;</span>-countsTable[means<span class="pl-k">&gt;=</span>3,]
dim(countsTable)

prop.nullv2 <span class="pl-k">&lt;</span>- apply(countsTable, 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
pdf(file=<span class="pl-s"><span class="pl-pds">"</span>Mean3NullCounts.pdf<span class="pl-pds">"</span></span>,14, 14)
barplot(prop.nullv2, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-en">dev.off</span>()

<span class="pl-c"><span class="pl-c">#</span>make count data set</span>
dds <span class="pl-k">&lt;</span>- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=<span class="pl-k">~</span> Origin_SymState_Temp)

dds <span class="pl-k">&lt;</span>- DESeq(dds)

<span class="pl-c"><span class="pl-c">#</span>transform counts to variance stablized counts</span>
vstCounts<span class="pl-k">&lt;</span>-varianceStabilizingTransformation(dds)

<span class="pl-c"><span class="pl-c">#</span>plot cluster dendrogram across samples</span>
dists<span class="pl-k">&lt;</span>-dist(t(assay(vstCounts)))
plot(hclust(dists))

plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)
plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>SymState<span class="pl-pds">"</span></span>)
plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Temp<span class="pl-pds">"</span></span>)
plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState<span class="pl-pds">"</span></span>)
plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>)

resultsNames(dds)

dim(dds)

<span class="pl-c"><span class="pl-c">#</span>##Figure out which contrast you want to examine (i.e. which two groups do you want to compare)</span>
res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_B_18<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_B_18<span class="pl-pds">"</span></span>))
head(res)
summary(res)

res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_18<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_W_18<span class="pl-pds">"</span></span>))
summary(res)

res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_B_14<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_B_14<span class="pl-pds">"</span></span>))
summary(res)

res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_14<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_W_14<span class="pl-pds">"</span></span>))
summary(res)

res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_B_22<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_B_22<span class="pl-pds">"</span></span>))
summary(res)

res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_22<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_W_22<span class="pl-pds">"</span></span>))
summary(res)

head(res[order(res<span class="pl-smi">$padj</span>),])
plotCounts(dds, <span class="pl-s"><span class="pl-pds">"</span>TR2367|c0_g1_i1_coral<span class="pl-pds">"</span></span>, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span>count the number of significantly differentially expressed genes</span>
sum(res<span class="pl-smi">$padj</span> <span class="pl-k">&lt;</span> 0.3, na.rm =T)
sum(res<span class="pl-smi">$padj</span> <span class="pl-k">&lt;</span> 0.2, na.rm =T)
sum(res<span class="pl-smi">$padj</span> <span class="pl-k">&lt;</span> 0.1, na.rm =T)
sum(res<span class="pl-smi">$pvalue</span> <span class="pl-k">&lt;</span> 0.05, na.rm =T)
											 
<span class="pl-c"><span class="pl-c">#</span> Make a counts table that is scaled by the size factors</span>
scaledcounts = counts(dds, normalized=T)

<span class="pl-c"><span class="pl-c">#</span>building heat map data</span>
head(scaledcounts)
genes4heatmap<span class="pl-k">&lt;</span>-res[res<span class="pl-smi">$pvalue</span> <span class="pl-k">&lt;</span>0.05 <span class="pl-k">&amp;</span> <span class="pl-k">!</span>is.na(res<span class="pl-smi">$pvalue</span>),]
names(genes4heatmap)
head(genes4heatmap)
dim(genes4heatmap)
data4heatmap<span class="pl-k">&lt;</span>-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
dim(data4heatmap)
head(data4heatmap)

temp = as.matrix(rowMeans(data4heatmap))
head(temp)
scaledmatrix<span class="pl-k">&lt;</span>-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
data4heatmapscaled = data4heatmap/scaledmatrix
head(data4heatmapscaled)

dim(data4heatmapscaled)

pairs.breaks <span class="pl-k">&lt;</span>- seq(0, 3.0, by=0.1)
length(pairs.breaks)
mycol <span class="pl-k">&lt;</span>- colorpanel(n=30, low=<span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, high=<span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>) 

pdf(file=<span class="pl-s"><span class="pl-pds">"</span>XXXXXX_byrow.pdf<span class="pl-pds">"</span></span>,14,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
<span class="pl-en">dev.off</span>()

pdf(file=<span class="pl-s"><span class="pl-pds">"</span>XXXXXX_bycolumn.pdf<span class="pl-pds">"</span></span>,14,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>col<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
<span class="pl-en">dev.off</span>()


<span class="pl-c"><span class="pl-c">#</span>##PCAs</span>
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)

prop.nullv3 <span class="pl-k">&lt;</span>- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
pdf(file=<span class="pl-s"><span class="pl-pds">"</span>ResNullCounts.pdf<span class="pl-pds">"</span></span>,14, 14)
barplot(prop.nullv3, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
        horiz=TRUE, cex.names=0.5, las=1, 
        col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-en">dev.off</span>()
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      03/03/2021
                     </h1>
                     <a aria-label="Permalink: 03/03/2021" class="anchor" href="#03032021" id="user-content-03032021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Homework Day 13 (last day!!)
                     </h2>
                     <a aria-label="Permalink: Homework Day 13 (last day!!)" class="anchor" href="#homework-day-13-last-day" id="user-content-homework-day-13-last-day">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        create an sbatch script to rm -r your sandbox/YOURNAME directory
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-007 sandboxes]$ nano KristinaCRemoveSandbox.sh
[kconf001@coreV2-25-007 sandboxes]$ cat KristinaCRemoveSandbox.sh
#!/bin/bash -l
#SBATCH -o KristinaCRemoveSandbox.txt
#SBATCH -n 1
#SBATCH --mail-user=kconf001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KristinaCRemoveSandbox
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC
" dir="auto">
                     <pre>[kconf001@coreV2-25-007 sandboxes]$ nano KristinaCRemoveSandbox.sh
[kconf001@coreV2-25-007 sandboxes]$ cat KristinaCRemoveSandbox.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KristinaCRemoveSandbox.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kconf001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KristinaCRemoveSandbox</span>
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/KristinaC
</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        submit your sbatch script and add the entry to your logfile/readme
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="[kconf001@coreV2-25-007 sandboxes]$ sbatch KristinaCRemoveDirectory.sh
[kconf001@coreV2-25-007 sandboxes]$ squeue -u kconf001
[kconf001@coreV2-25-007 sandboxes]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes" dir="auto">
                     <pre>[kconf001@coreV2-25-007 sandboxes]$ sbatch KristinaCRemoveDirectory.sh
[kconf001@coreV2-25-007 sandboxes]$ squeue -u kconf001
[kconf001@coreV2-25-007 sandboxes]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="$ scp KristinaCREADME.md kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive" dir="auto">
                     <pre>$ scp KristinaCREADME.md kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive</pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        scp the following files to your local computer (copied directly to AdvancedGeomicsAnalysis 21SpKristinaCAdvancedGenomicsLog)
scp
                        <a href="mailto:YOURMIDASID@turing.hpc.odu.edu">
                         YOURMIDASID@turing.hpc.odu.edu
                        </a>
                        :/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./
scp
                        <a href="mailto:YOURMIDASID@turing.hpc.odu.edu">
                         YOURMIDASID@turing.hpc.odu.edu
                        </a>
                        :/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./
scp
                        <a href="mailto:YOURMIDASID@turing.hpc.odu.edu">
                         YOURMIDASID@turing.hpc.odu.edu
                        </a>
                        :/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./
scp
                        <a href="mailto:YOURMIDASID@turing.hpc.odu.edu">
                         YOURMIDASID@turing.hpc.odu.edu
                        </a>
                        :/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./
kconf001@turing.hpc.odu.edu's password:
15079_Apoc_hostsym.fasta
100%   13MB   5.7MB/s   00:02
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./
kconf001@turing.hpc.odu.edu's password:
15079_Apoc_hostsym.fasta.fai
100%  641KB   3.8MB/s   00:00 
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./
kconf001@turing.hpc.odu.edu's password:
RI_B_01.bam
lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./
kconf001@turing.hpc.odu.edu's password:
RI_B_01.bam.bai
 " dir="auto">
                     <pre>lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">15079_Apoc_hostsym.fasta</span>
<span class="pl-s">100%   13MB   5.7MB/s   00:02</span>
<span class="pl-s">lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)</span>
<span class="pl-s">$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./</span>
<span class="pl-s">kconf001@turing.hpc.odu.edu<span class="pl-pds">'</span></span>s password:
15079_Apoc_hostsym.fasta.fai
100%  641KB   3.8MB/s   00:00 
lyka@LAPTOP-GFGCMDB6 MINGW64 <span class="pl-k">~</span>/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)
$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./
kconf001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password:</span>
<span class="pl-s">RI_B_01.bam</span>
<span class="pl-s">lyka@LAPTOP-GFGCMDB6 MINGW64 ~/Desktop/Files/ODU/ODUSPRING2021/AdvanceGenomicsAnalysis/21SpKristinaCAdvancedGenomicsLog (main)</span>
<span class="pl-s">$ scp kconf001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./</span>
<span class="pl-s">kconf001@turing.hpc.odu.edu<span class="pl-pds">'</span></span>s password:
RI_B_01.bam.bai
 </pre>
                    </div>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the "genome" and RI_B_01.bam as the input file
                       </li>
                      </ol>
                     </li>
                    </ul>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="MW_jO3tqkI85r3uArfg3uqE-uIN97HGLjDalIoqRdkImt3yU52Ho7zrjgfn7rKp014SVYJ1FChH95YGJjzLoYQ">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Notebook for Advanced Genomics
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/kconf001/21SpKristinaCAdvancedGenomicsLog/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fkconf001%2F21SpKristinaCAdvancedGenomicsLog&amp;report=kconf001+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/kconf001/21SpKristinaCAdvancedGenomicsLog/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/kconf001/packages?repo_name=21SpKristinaCAdvancedGenomicsLog">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/kconf001/21SpKristinaCAdvancedGenomicsLog/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       Â© 2024 GitHub,Â Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You canât perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
